{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# **Automatation of the CLI commands of the Snakemake workflow**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* Environment to use: [Greference_tools](code/environments/Greference_tools.yml)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Libreries\n",
    "import subprocess\n",
    "import os\n",
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "## lists for the loops\n",
    "chr_list        = [\"chr3\", \"chr5\", \"chr7\", \"chr12\", \"chr17\"]\n",
    "\n",
    "sample_list     = [\"ERR696683\", \"ERR753368\", \"ERR753369\" ,\"ERR753370\" ,\"ERR753371\" ,\"ERR753372\", \n",
    "                  \"ERR753373\", \"ERR753374\", \"ERR753375\", \"ERR753376\", \"ERR753377\", \"ERR753378\"]\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. **rule download_data**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Nothing to be done (all requested files are present and up to date).\n",
      "Complete log: .snakemake/log/2023-08-28T131200.585023.snakemake.log\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['snakemake', '--cores', '1', 'download_data'], returncode=0)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"download_data\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* rule pre_processing\n",
    "\n",
    "Filtering the chromosomes, make sure that the chromosomes of the config file are **ch3**. \n",
    "\n",
    "This rule is preatty long too... good luck\n",
    "\n",
    "When it finished, it is recomended to delete the SAM files not oredered."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "pre_processing        1              1              1\n",
      "total                 1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 13:12:25 2023]\n",
      "rule pre_processing:\n",
      "    input: code/03extracting_fastq.sh\n",
      "    output: data/original_bam/filtering/ERR696683_chr3_sorted.bam\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/original_bam/filtering/ERR696683_chr3_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 84123 singletons\n",
      "[M::bam2fq_mainloop] processed 4225255 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 28084 singletons\n",
      "[M::bam2fq_mainloop] processed 4361932 reads\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 29347 singletons\n",
      "[M::bam2fq_mainloop] processed 4672601 reads\n",
      "[bam_sort_core] merging from 3 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 72920 singletons\n",
      "[M::bam2fq_mainloop] processed 5975966 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 18585 singletons\n",
      "[M::bam2fq_mainloop] processed 3894697 reads\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 30337 singletons\n",
      "[M::bam2fq_mainloop] processed 4628597 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 8259 singletons\n",
      "[M::bam2fq_mainloop] processed 3945189 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 7892 singletons\n",
      "[M::bam2fq_mainloop] processed 2332698 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 23741 singletons\n",
      "[M::bam2fq_mainloop] processed 2588099 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 25677 singletons\n",
      "[M::bam2fq_mainloop] processed 3355889 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 49067 singletons\n",
      "[M::bam2fq_mainloop] processed 3506967 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 29704 singletons\n",
      "[M::bam2fq_mainloop] processed 3986056 reads\n",
      "[Mon Aug 28 13:55:09 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T131224.993369.snakemake.log\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "          Chromosomes transformed:\n",
      "           chr3 chr5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "pre_processing        1              1              1\n",
      "total                 1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 13:55:10 2023]\n",
      "rule pre_processing:\n",
      "    input: code/03extracting_fastq.sh\n",
      "    output: data/original_bam/filtering/ERR696683_chr5_sorted.bam\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/original_bam/filtering/ERR696683_chr5_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 74911 singletons\n",
      "[M::bam2fq_mainloop] processed 3565319 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 24550 singletons\n",
      "[M::bam2fq_mainloop] processed 3533184 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 25457 singletons\n",
      "[M::bam2fq_mainloop] processed 3773533 reads\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 64466 singletons\n",
      "[M::bam2fq_mainloop] processed 5100608 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 16803 singletons\n",
      "[M::bam2fq_mainloop] processed 3312473 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 26284 singletons\n",
      "[M::bam2fq_mainloop] processed 3726124 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 7129 singletons\n",
      "[M::bam2fq_mainloop] processed 3545831 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 6660 singletons\n",
      "[M::bam2fq_mainloop] processed 2092598 reads\n",
      "[M::bam2fq_mainloop] discarded 21394 singletons\n",
      "[M::bam2fq_mainloop] processed 2190882 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 22903 singletons\n",
      "[M::bam2fq_mainloop] processed 2828499 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 43852 singletons\n",
      "[M::bam2fq_mainloop] processed 2957702 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 25642 singletons\n",
      "[M::bam2fq_mainloop] processed 3190954 reads\n",
      "[Mon Aug 28 14:33:55 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T135509.674696.snakemake.log\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "          Chromosomes transformed:\n",
      "           chr5 chr7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "pre_processing        1              1              1\n",
      "total                 1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 14:33:57 2023]\n",
      "rule pre_processing:\n",
      "    input: code/03extracting_fastq.sh\n",
      "    output: data/original_bam/filtering/ERR696683_chr7_sorted.bam\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/original_bam/filtering/ERR696683_chr7_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 73902 singletons\n",
      "[M::bam2fq_mainloop] processed 3944452 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 24198 singletons\n",
      "[M::bam2fq_mainloop] processed 3636374 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 25287 singletons\n",
      "[M::bam2fq_mainloop] processed 3883825 reads\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 63834 singletons\n",
      "[M::bam2fq_mainloop] processed 5410754 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 16419 singletons\n",
      "[M::bam2fq_mainloop] processed 3571993 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 26268 singletons\n",
      "[M::bam2fq_mainloop] processed 3882598 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 7172 singletons\n",
      "[M::bam2fq_mainloop] processed 3188190 reads\n",
      "[M::bam2fq_mainloop] discarded 6833 singletons\n",
      "[M::bam2fq_mainloop] processed 1881699 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 21091 singletons\n",
      "[M::bam2fq_mainloop] processed 2428999 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 22516 singletons\n",
      "[M::bam2fq_mainloop] processed 3134720 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 43235 singletons\n",
      "[M::bam2fq_mainloop] processed 3307103 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 25927 singletons\n",
      "[M::bam2fq_mainloop] processed 3349951 reads\n",
      "[Mon Aug 28 15:13:47 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T143356.537170.snakemake.log\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "          Chromosomes transformed:\n",
      "           chr7 chr12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "pre_processing        1              1              1\n",
      "total                 1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 15:13:49 2023]\n",
      "rule pre_processing:\n",
      "    input: code/03extracting_fastq.sh\n",
      "    output: data/original_bam/filtering/ERR696683_chr12_sorted.bam\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/original_bam/filtering/ERR696683_chr12_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 64449 singletons\n",
      "[M::bam2fq_mainloop] processed 3693551 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 22477 singletons\n",
      "[M::bam2fq_mainloop] processed 3818357 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 23327 singletons\n",
      "[M::bam2fq_mainloop] processed 4131307 reads\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 55851 singletons\n",
      "[M::bam2fq_mainloop] processed 5154685 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 14328 singletons\n",
      "[M::bam2fq_mainloop] processed 3395746 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 24300 singletons\n",
      "[M::bam2fq_mainloop] processed 4081066 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 5449 singletons\n",
      "[M::bam2fq_mainloop] processed 2832515 reads\n",
      "[M::bam2fq_mainloop] discarded 5620 singletons\n",
      "[M::bam2fq_mainloop] processed 1705308 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 18852 singletons\n",
      "[M::bam2fq_mainloop] processed 2285894 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 19966 singletons\n",
      "[M::bam2fq_mainloop] processed 2951418 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 37782 singletons\n",
      "[M::bam2fq_mainloop] processed 3107334 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 24204 singletons\n",
      "[M::bam2fq_mainloop] processed 3531638 reads\n",
      "[Mon Aug 28 15:53:02 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T151348.434176.snakemake.log\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "          Chromosomes transformed:\n",
      "           chr12 chr17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "pre_processing        1              1              1\n",
      "total                 1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 15:53:04 2023]\n",
      "rule pre_processing:\n",
      "    input: code/03extracting_fastq.sh\n",
      "    output: data/original_bam/filtering/ERR696683_chr17_sorted.bam\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/original_bam/filtering/ERR696683_chr17_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 49649 singletons\n",
      "[M::bam2fq_mainloop] processed 3373755 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 18673 singletons\n",
      "[M::bam2fq_mainloop] processed 3458829 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 19617 singletons\n",
      "[M::bam2fq_mainloop] processed 3688229 reads\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 44102 singletons\n",
      "[M::bam2fq_mainloop] processed 4435138 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 10297 singletons\n",
      "[M::bam2fq_mainloop] processed 2970261 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 20675 singletons\n",
      "[M::bam2fq_mainloop] processed 3803001 reads\n",
      "[M::bam2fq_mainloop] discarded 3933 singletons\n",
      "[M::bam2fq_mainloop] processed 1690481 reads\n",
      "[M::bam2fq_mainloop] discarded 4157 singletons\n",
      "[M::bam2fq_mainloop] processed 988821 reads\n",
      "[M::bam2fq_mainloop] discarded 13970 singletons\n",
      "[M::bam2fq_mainloop] processed 2066508 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 15068 singletons\n",
      "[M::bam2fq_mainloop] processed 2654178 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 28410 singletons\n",
      "[M::bam2fq_mainloop] processed 2799742 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 20401 singletons\n",
      "[M::bam2fq_mainloop] processed 3329753 reads\n",
      "[Mon Aug 28 16:28:31 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T155303.245665.snakemake.log\n"
     ]
    }
   ],
   "source": [
    "## Running the 3º chromosome\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"data/original_bam/filtering/ERR696683_chr3_sorted.bam\"])\n",
    "\n",
    "## Running the rest of the chromosomes\n",
    "for chr in range(len(chr_list) - 1):\n",
    "    chr1 = chr_list[chr]\n",
    "    chr2 = chr_list[chr + 1]\n",
    "    \n",
    "    print(\"\"\"\n",
    "          Chromosomes transformed:\n",
    "          \"\"\", chr1, chr2)\n",
    "    \n",
    "    subprocess.run([\"sed\", \"-i\", \"6,+12s/\" + chr1 + \"/\" + chr2 + \"/g\", \"config.yaml\"])\n",
    "    subprocess.run([\"sed\", \"-i\", \"39s/\" + chr1 + \"/\" + chr2 + \"/g\", \"config.yaml\"])\n",
    "\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"data/original_bam/filtering/ERR696683_\" + chr2 + \"_sorted.bam\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['sed', '-i', '39s/chr17/chr3/g', 'config.yaml'], returncode=0)"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## reset config \n",
    "subprocess.run([\"sed\", \"-i\", \"6,+12s/chr17/chr3/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"39s/chr17/chr3/g\", \"config.yaml\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* rules reference_genome; fastqc; fastp; fastqc_trimmed; bwa_mapping; sam_to_bam; delete_duplicates; extracting_variants; vep_install; vep_cli; parsing_dataR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job                 count    min threads    max threads\n",
      "----------------  -------  -------------  -------------\n",
      "reference_genome        1              1              1\n",
      "total                   1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:47:08 2023]\n",
      "rule reference_genome:\n",
      "    output: data/reference/genome.fa\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/reference/genome.fa\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "rm: no se puede borrar 'data/reference/genome.fa*': No existe el archivo o el directorio\n",
      "--2023-08-28 16:47:08--  https://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.3.fa.gz\n",
      "Resolviendo ftp.ensembl.org (ftp.ensembl.org)... "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==> If you hadn't a previous reference in the directory and there is an ERROR, is NORMAL <==\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "193.62.193.169\n",
      "Conectando con ftp.ensembl.org (ftp.ensembl.org)[193.62.193.169]:443... conectado.\n",
      "Petición HTTP enviada, esperando respuesta... 200 OK\n",
      "Longitud: 59761837 (57M) [application/x-gzip]\n",
      "Guardando como: ‘data/reference/genome.fa.gz’\n",
      "\n",
      "     0K .......... .......... .......... .......... ..........  0%  452K 2m9s\n",
      "    50K .......... .......... .......... .......... ..........  0% 1,26M 87s\n",
      "   100K .......... .......... .......... .......... ..........  0% 2,53M 65s\n",
      "   150K .......... .......... .......... .......... ..........  0% 1,30M 60s\n",
      "   200K .......... .......... .......... .......... ..........  0% 2,73M 52s\n",
      "   250K .......... .......... .......... .......... ..........  0% 17,0M 44s\n",
      "   300K .......... .......... .......... .......... ..........  0% 65,0M 38s\n",
      "   350K .......... .......... .......... .......... ..........  0% 80,2M 33s\n",
      "   400K .......... .......... .......... .......... ..........  0% 1,37M 34s\n",
      "   450K .......... .......... .......... .......... ..........  0% 2,78M 33s\n",
      "   500K .......... .......... .......... .......... ..........  0% 59,2M 30s\n",
      "   550K .......... .......... .......... .......... ..........  1% 45,2M 27s\n",
      "   600K .......... .......... .......... .......... ..........  1%  104M 25s\n",
      "   650K .......... .......... .......... .......... ..........  1%  122M 23s\n",
      "   700K .......... .......... .......... .......... ..........  1% 99,5M 22s\n",
      "   750K .......... .......... .......... .......... ..........  1%  121M 21s\n",
      "   800K .......... .......... .......... .......... ..........  1% 1,36M 22s\n",
      "   850K .......... .......... .......... .......... ..........  1% 87,4M 21s\n",
      "   900K .......... .......... .......... .......... ..........  1% 74,2M 19s\n",
      "   950K .......... .......... .......... .......... ..........  1% 2,91M 19s\n",
      "  1000K .......... .......... .......... .......... ..........  1% 83,3M 19s\n",
      "  1050K .......... .......... .......... .......... ..........  1% 85,5M 18s\n",
      "  1100K .......... .......... .......... .......... ..........  1% 79,5M 17s\n",
      "  1150K .......... .......... .......... .......... ..........  2% 85,8M 16s\n",
      "  1200K .......... .......... .......... .......... ..........  2% 78,9M 16s\n",
      "  1250K .......... .......... .......... .......... ..........  2% 84,6M 15s\n",
      "  1300K .......... .......... .......... .......... ..........  2% 78,2M 14s\n",
      "  1350K .......... .......... .......... .......... ..........  2% 94,1M 14s\n",
      "  1400K .......... .......... .......... .......... ..........  2%  155M 13s\n",
      "  1450K .......... .......... .......... .......... ..........  2%  136M 13s\n",
      "  1500K .......... .......... .......... .......... ..........  2%  152M 13s\n",
      "  1550K .......... .......... .......... .......... ..........  2%  153M 12s\n",
      "  1600K .......... .......... .......... .......... ..........  2%  135M 12s\n",
      "  1650K .......... .......... .......... .......... ..........  2% 1,52M 13s\n",
      "  1700K .......... .......... .......... .......... ..........  2% 81,0M 12s\n",
      "  1750K .......... .......... .......... .......... ..........  3% 69,6M 12s\n",
      "  1800K .......... .......... .......... .......... ..........  3% 96,4M 12s\n",
      "  1850K .......... .......... .......... .......... ..........  3%  104M 11s\n",
      "  1900K .......... .......... .......... .......... ..........  3% 3,65M 11s\n",
      "  1950K .......... .......... .......... .......... ..........  3% 14,1M 11s\n",
      "  2000K .......... .......... .......... .......... ..........  3%  114M 11s\n",
      "  2050K .......... .......... .......... .......... ..........  3%  114M 11s\n",
      "  2100K .......... .......... .......... .......... ..........  3% 99,5M 10s\n",
      "  2150K .......... .......... .......... .......... ..........  3% 36,7M 10s\n",
      "  2200K .......... .......... .......... .......... ..........  3% 98,5M 10s\n",
      "  2250K .......... .......... .......... .......... ..........  3%  115M 10s\n",
      "  2300K .......... .......... .......... .......... ..........  4%  119M 10s\n",
      "  2350K .......... .......... .......... .......... ..........  4%  101M 9s\n",
      "  2400K .......... .......... .......... .......... ..........  4%  114M 9s\n",
      "  2450K .......... .......... .......... .......... ..........  4%  102M 9s\n",
      "  2500K .......... .......... .......... .......... ..........  4%  117M 9s\n",
      "  2550K .......... .......... .......... .......... ..........  4%  111M 9s\n",
      "  2600K .......... .......... .......... .......... ..........  4% 92,6M 8s\n",
      "  2650K .......... .......... .......... .......... ..........  4%  118M 8s\n",
      "  2700K .......... .......... .......... .......... ..........  4%  100M 8s\n",
      "  2750K .......... .......... .......... .......... ..........  4%  114M 8s\n",
      "  2800K .......... .......... .......... .......... ..........  4%  119M 8s\n",
      "  2850K .......... .......... .......... .......... ..........  4%  106M 8s\n",
      "  2900K .......... .......... .......... .......... ..........  5%  122M 8s\n",
      "  2950K .......... .......... .......... .......... ..........  5%  108M 7s\n",
      "  3000K .......... .......... .......... .......... ..........  5%  126M 7s\n",
      "  3050K .......... .......... .......... .......... ..........  5%  109M 7s\n",
      "  3100K .......... .......... .......... .......... ..........  5%  132M 7s\n",
      "  3150K .......... .......... .......... .......... ..........  5%  155M 7s\n",
      "  3200K .......... .......... .......... .......... ..........  5%  213M 7s\n",
      "  3250K .......... .......... .......... .......... ..........  5%  192M 7s\n",
      "  3300K .......... .......... .......... .......... ..........  5% 13,7M 7s\n",
      "  3350K .......... .......... .......... .......... ..........  5% 1,97M 7s\n",
      "  3400K .......... .......... .......... .......... ..........  5% 26,6M 7s\n",
      "  3450K .......... .......... .......... .......... ..........  5% 24,1M 7s\n",
      "  3500K .......... .......... .......... .......... ..........  6%  103M 7s\n",
      "  3550K .......... .......... .......... .......... ..........  6%  114M 7s\n",
      "  3600K .......... .......... .......... .......... ..........  6% 99,7M 7s\n",
      "  3650K .......... .......... .......... .......... ..........  6%  120M 7s\n",
      "  3700K .......... .......... .......... .......... ..........  6% 80,5M 6s\n",
      "  3750K .......... .......... .......... .......... ..........  6%  112M 6s\n",
      "  3800K .......... .......... .......... .......... ..........  6%  120M 6s\n",
      "  3850K .......... .......... .......... .......... ..........  6% 7,13M 6s\n",
      "  3900K .......... .......... .......... .......... ..........  6% 40,9M 6s\n",
      "  3950K .......... .......... .......... .......... ..........  6% 21,2M 6s\n",
      "  4000K .......... .......... .......... .......... ..........  6% 52,3M 6s\n",
      "  4050K .......... .......... .......... .......... ..........  7% 45,2M 6s\n",
      "  4100K .......... .......... .......... .......... ..........  7% 45,9M 6s\n",
      "  4150K .......... .......... .......... .......... ..........  7% 40,2M 6s\n",
      "  4200K .......... .......... .......... .......... ..........  7% 19,6M 6s\n",
      "  4250K .......... .......... .......... .......... ..........  7% 31,4M 6s\n",
      "  4300K .......... .......... .......... .......... ..........  7% 31,4M 6s\n",
      "  4350K .......... .......... .......... .......... ..........  7%  109M 6s\n",
      "  4400K .......... .......... .......... .......... ..........  7%  128M 6s\n",
      "  4450K .......... .......... .......... .......... ..........  7%  106M 6s\n",
      "  4500K .......... .......... .......... .......... ..........  7%  128M 6s\n",
      "  4550K .......... .......... .......... .......... ..........  7%  132M 5s\n",
      "  4600K .......... .......... .......... .......... ..........  7%  111M 5s\n",
      "  4650K .......... .......... .......... .......... ..........  8%  134M 5s\n",
      "  4700K .......... .......... .......... .......... ..........  8%  108M 5s\n",
      "  4750K .......... .......... .......... .......... ..........  8%  126M 5s\n",
      "  4800K .......... .......... .......... .......... ..........  8%  137M 5s\n",
      "  4850K .......... .......... .......... .......... ..........  8%  121M 5s\n",
      "  4900K .......... .......... .......... .......... ..........  8%  137M 5s\n",
      "  4950K .......... .......... .......... .......... ..........  8%  124M 5s\n",
      "  5000K .......... .......... .......... .......... ..........  8%  141M 5s\n",
      "  5050K .......... .......... .......... .......... ..........  8%  137M 5s\n",
      "  5100K .......... .......... .......... .......... ..........  8% 11,1M 5s\n",
      "  5150K .......... .......... .......... .......... ..........  8% 93,9M 5s\n",
      "  5200K .......... .......... .......... .......... ..........  8% 84,8M 5s\n",
      "  5250K .......... .......... .......... .......... ..........  9% 98,1M 5s\n",
      "  5300K .......... .......... .......... .......... ..........  9% 52,8M 5s\n",
      "  5350K .......... .......... .......... .......... ..........  9% 3,14M 5s\n",
      "  5400K .......... .......... .......... .......... ..........  9%  124M 5s\n",
      "  5450K .......... .......... .......... .......... ..........  9% 2,50M 5s\n",
      "  5500K .......... .......... .......... .......... ..........  9% 90,4M 5s\n",
      "  5550K .......... .......... .......... .......... ..........  9%  110M 5s\n",
      "  5600K .......... .......... .......... .......... ..........  9% 3,37M 5s\n",
      "  5650K .......... .......... .......... .......... ..........  9%  117M 5s\n",
      "  5700K .......... .......... .......... .......... ..........  9%  102M 5s\n",
      "  5750K .......... .......... .......... .......... ..........  9% 52,7M 5s\n",
      "  5800K .......... .......... .......... .......... .......... 10%  120M 5s\n",
      "  5850K .......... .......... .......... .......... .......... 10% 2,77M 5s\n",
      "  5900K .......... .......... .......... .......... .......... 10%  117M 5s\n",
      "  5950K .......... .......... .......... .......... .......... 10%  103M 5s\n",
      "  6000K .......... .......... .......... .......... .......... 10%  109M 5s\n",
      "  6050K .......... .......... .......... .......... .......... 10%  112M 5s\n",
      "  6100K .......... .......... .......... .......... .......... 10%  102M 5s\n",
      "  6150K .......... .......... .......... .......... .......... 10%  121M 5s\n",
      "  6200K .......... .......... .......... .......... .......... 10%  102M 5s\n",
      "  6250K .......... .......... .......... .......... .......... 10%  119M 5s\n",
      "  6300K .......... .......... .......... .......... .......... 10%  119M 5s\n",
      "  6350K .......... .......... .......... .......... .......... 10%  102M 4s\n",
      "  6400K .......... .......... .......... .......... .......... 11%  117M 4s\n",
      "  6450K .......... .......... .......... .......... .......... 11%  102M 4s\n",
      "  6500K .......... .......... .......... .......... .......... 11%  119M 4s\n",
      "  6550K .......... .......... .......... .......... .......... 11%  118M 4s\n",
      "  6600K .......... .......... .......... .......... .......... 11% 99,4M 4s\n",
      "  6650K .......... .......... .......... .......... .......... 11%  126M 4s\n",
      "  6700K .......... .......... .......... .......... .......... 11% 66,8M 4s\n",
      "  6750K .......... .......... .......... .......... .......... 11% 62,5M 4s\n",
      "  6800K .......... .......... .......... .......... .......... 11%  117M 4s\n",
      "  6850K .......... .......... .......... .......... .......... 11%  103M 4s\n",
      "  6900K .......... .......... .......... .......... .......... 11% 62,7M 4s\n",
      "  6950K .......... .......... .......... .......... .......... 11%  107M 4s\n",
      "  7000K .......... .......... .......... .......... .......... 12%  119M 4s\n",
      "  7050K .......... .......... .......... .......... .......... 12%  118M 4s\n",
      "  7100K .......... .......... .......... .......... .......... 12% 28,5M 4s\n",
      "  7150K .......... .......... .......... .......... .......... 12%  116M 4s\n",
      "  7200K .......... .......... .......... .......... .......... 12% 46,1M 4s\n",
      "  7250K .......... .......... .......... .......... .......... 12% 2,81M 4s\n",
      "  7300K .......... .......... .......... .......... .......... 12%  116M 4s\n",
      "  7350K .......... .......... .......... .......... .......... 12%  104M 4s\n",
      "  7400K .......... .......... .......... .......... .......... 12% 71,0M 4s\n",
      "  7450K .......... .......... .......... .......... .......... 12% 51,6M 4s\n",
      "  7500K .......... .......... .......... .......... .......... 12% 3,47M 4s\n",
      "  7550K .......... .......... .......... .......... .......... 13%  120M 4s\n",
      "  7600K .......... .......... .......... .......... .......... 13% 99,5M 4s\n",
      "  7650K .......... .......... .......... .......... .......... 13% 83,7M 4s\n",
      "  7700K .......... .......... .......... .......... .......... 13%  100M 4s\n",
      "  7750K .......... .......... .......... .......... .......... 13%  118M 4s\n",
      "  7800K .......... .......... .......... .......... .......... 13% 4,08M 4s\n",
      "  7850K .......... .......... .......... .......... .......... 13% 86,4M 4s\n",
      "  7900K .......... .......... .......... .......... .......... 13% 2,37M 4s\n",
      "  7950K .......... .......... .......... .......... .......... 13%  102M 4s\n",
      "  8000K .......... .......... .......... .......... .......... 13% 74,1M 4s\n",
      "  8050K .......... .......... .......... .......... .......... 13% 41,3M 4s\n",
      "  8100K .......... .......... .......... .......... .......... 13% 2,40M 4s\n",
      "  8150K .......... .......... .......... .......... .......... 14% 50,3M 4s\n",
      "  8200K .......... .......... .......... .......... .......... 14% 55,3M 4s\n",
      "  8250K .......... .......... .......... .......... .......... 14% 84,8M 4s\n",
      "  8300K .......... .......... .......... .......... .......... 14% 61,4M 4s\n",
      "  8350K .......... .......... .......... .......... .......... 14% 93,5M 4s\n",
      "  8400K .......... .......... .......... .......... .......... 14%  109M 4s\n",
      "  8450K .......... .......... .......... .......... .......... 14%  112M 4s\n",
      "  8500K .......... .......... .......... .......... .......... 14%  194M 4s\n",
      "  8550K .......... .......... .......... .......... .......... 14% 45,0M 4s\n",
      "  8600K .......... .......... .......... .......... .......... 14% 94,9M 4s\n",
      "  8650K .......... .......... .......... .......... .......... 14%  124M 4s\n",
      "  8700K .......... .......... .......... .......... .......... 14% 93,4M 4s\n",
      "  8750K .......... .......... .......... .......... .......... 15%  113M 4s\n",
      "  8800K .......... .......... .......... .......... .......... 15%  111M 4s\n",
      "  8850K .......... .......... .......... .......... .......... 15% 88,4M 4s\n",
      "  8900K .......... .......... .......... .......... .......... 15%  115M 4s\n",
      "  8950K .......... .......... .......... .......... .......... 15% 95,0M 4s\n",
      "  9000K .......... .......... .......... .......... .......... 15%  116M 4s\n",
      "  9050K .......... .......... .......... .......... .......... 15%  131M 4s\n",
      "  9100K .......... .......... .......... .......... .......... 15% 97,1M 4s\n",
      "  9150K .......... .......... .......... .......... .......... 15%  115M 4s\n",
      "  9200K .......... .......... .......... .......... .......... 15% 94,5M 4s\n",
      "  9250K .......... .......... .......... .......... .......... 15%  111M 4s\n",
      "  9300K .......... .......... .......... .......... .......... 16%  104M 3s\n",
      "  9350K .......... .......... .......... .......... .......... 16% 94,4M 3s\n",
      "  9400K .......... .......... .......... .......... .......... 16%  128M 3s\n",
      "  9450K .......... .......... .......... .......... .......... 16%  118M 3s\n",
      "  9500K .......... .......... .......... .......... .......... 16%  126M 3s\n",
      "  9550K .......... .......... .......... .......... .......... 16%  115M 3s\n",
      "  9600K .......... .......... .......... .......... .......... 16%  102M 3s\n",
      "  9650K .......... .......... .......... .......... .......... 16%  119M 3s\n",
      "  9700K .......... .......... .......... .......... .......... 16% 3,14M 3s\n",
      "  9750K .......... .......... .......... .......... .......... 16%  108M 3s\n",
      "  9800K .......... .......... .......... .......... .......... 16%  113M 3s\n",
      "  9850K .......... .......... .......... .......... .......... 16% 91,0M 3s\n",
      "  9900K .......... .......... .......... .......... .......... 17%  120M 3s\n",
      "  9950K .......... .......... .......... .......... .......... 17% 2,68M 3s\n",
      " 10000K .......... .......... .......... .......... .......... 17%  107M 3s\n",
      " 10050K .......... .......... .......... .......... .......... 17%  123M 3s\n",
      " 10100K .......... .......... .......... .......... .......... 17% 90,8M 3s\n",
      " 10150K .......... .......... .......... .......... .......... 17% 45,3M 3s\n",
      " 10200K .......... .......... .......... .......... .......... 17%  103M 3s\n",
      " 10250K .......... .......... .......... .......... .......... 17%  127M 3s\n",
      " 10300K .......... .......... .......... .......... .......... 17%  116M 3s\n",
      " 10350K .......... .......... .......... .......... .......... 17% 67,5M 3s\n",
      " 10400K .......... .......... .......... .......... .......... 17%  126M 3s\n",
      " 10450K .......... .......... .......... .......... .......... 17% 44,6M 3s\n",
      " 10500K .......... .......... .......... .......... .......... 18%  142M 3s\n",
      " 10550K .......... .......... .......... .......... .......... 18%  134M 3s\n",
      " 10600K .......... .......... .......... .......... .......... 18% 53,1M 3s\n",
      " 10650K .......... .......... .......... .......... .......... 18%  132M 3s\n",
      " 10700K .......... .......... .......... .......... .......... 18% 70,6M 3s\n",
      " 10750K .......... .......... .......... .......... .......... 18%  119M 3s\n",
      " 10800K .......... .......... .......... .......... .......... 18%  121M 3s\n",
      " 10850K .......... .......... .......... .......... .......... 18% 45,8M 3s\n",
      " 10900K .......... .......... .......... .......... .......... 18%  116M 3s\n",
      " 10950K .......... .......... .......... .......... .......... 18%  105M 3s\n",
      " 11000K .......... .......... .......... .......... .......... 18% 94,7M 3s\n",
      " 11050K .......... .......... .......... .......... .......... 19%  116M 3s\n",
      " 11100K .......... .......... .......... .......... .......... 19% 89,8M 3s\n",
      " 11150K .......... .......... .......... .......... .......... 19%  120M 3s\n",
      " 11200K .......... .......... .......... .......... .......... 19%  103M 3s\n",
      " 11250K .......... .......... .......... .......... .......... 19% 76,7M 3s\n",
      " 11300K .......... .......... .......... .......... .......... 19% 52,5M 3s\n",
      " 11350K .......... .......... .......... .......... .......... 19% 60,3M 3s\n",
      " 11400K .......... .......... .......... .......... .......... 19%  114M 3s\n",
      " 11450K .......... .......... .......... .......... .......... 19%  103M 3s\n",
      " 11500K .......... .......... .......... .......... .......... 19% 94,3M 3s\n",
      " 11550K .......... .......... .......... .......... .......... 19%  119M 3s\n",
      " 11600K .......... .......... .......... .......... .......... 19%  101M 3s\n",
      " 11650K .......... .......... .......... .......... .......... 20%  120M 3s\n",
      " 11700K .......... .......... .......... .......... .......... 20%  101M 3s\n",
      " 11750K .......... .......... .......... .......... .......... 20% 3,91M 3s\n",
      " 11800K .......... .......... .......... .......... .......... 20%  118M 3s\n",
      " 11850K .......... .......... .......... .......... .......... 20%  103M 3s\n",
      " 11900K .......... .......... .......... .......... .......... 20%  119M 3s\n",
      " 11950K .......... .......... .......... .......... .......... 20%  104M 3s\n",
      " 12000K .......... .......... .......... .......... .......... 20%  119M 3s\n",
      " 12050K .......... .......... .......... .......... .......... 20%  118M 3s\n",
      " 12100K .......... .......... .......... .......... .......... 20%  103M 3s\n",
      " 12150K .......... .......... .......... .......... .......... 20% 99,1M 3s\n",
      " 12200K .......... .......... .......... .......... .......... 20%  104M 3s\n",
      " 12250K .......... .......... .......... .......... .......... 21% 50,6M 3s\n",
      " 12300K .......... .......... .......... .......... .......... 21%  118M 3s\n",
      " 12350K .......... .......... .......... .......... .......... 21% 46,9M 3s\n",
      " 12400K .......... .......... .......... .......... .......... 21% 52,3M 3s\n",
      " 12450K .......... .......... .......... .......... .......... 21%  101M 3s\n",
      " 12500K .......... .......... .......... .......... .......... 21% 30,7M 3s\n",
      " 12550K .......... .......... .......... .......... .......... 21%  120M 3s\n",
      " 12600K .......... .......... .......... .......... .......... 21% 57,1M 3s\n",
      " 12650K .......... .......... .......... .......... .......... 21% 80,0M 3s\n",
      " 12700K .......... .......... .......... .......... .......... 21% 3,63M 3s\n",
      " 12750K .......... .......... .......... .......... .......... 21% 83,9M 3s\n",
      " 12800K .......... .......... .......... .......... .......... 22%  104M 3s\n",
      " 12850K .......... .......... .......... .......... .......... 22%  100M 3s\n",
      " 12900K .......... .......... .......... .......... .......... 22% 71,3M 3s\n",
      " 12950K .......... .......... .......... .......... .......... 22%  101M 3s\n",
      " 13000K .......... .......... .......... .......... .......... 22%  122M 3s\n",
      " 13050K .......... .......... .......... .......... .......... 22% 95,8M 3s\n",
      " 13100K .......... .......... .......... .......... .......... 22%  101M 3s\n",
      " 13150K .......... .......... .......... .......... .......... 22% 27,0M 3s\n",
      " 13200K .......... .......... .......... .......... .......... 22%  103M 3s\n",
      " 13250K .......... .......... .......... .......... .......... 22%  117M 3s\n",
      " 13300K .......... .......... .......... .......... .......... 22% 86,1M 3s\n",
      " 13350K .......... .......... .......... .......... .......... 22%  101M 3s\n",
      " 13400K .......... .......... .......... .......... .......... 23% 51,5M 3s\n",
      " 13450K .......... .......... .......... .......... .......... 23%  106M 3s\n",
      " 13500K .......... .......... .......... .......... .......... 23%  119M 3s\n",
      " 13550K .......... .......... .......... .......... .......... 23%  120M 3s\n",
      " 13600K .......... .......... .......... .......... .......... 23% 98,1M 3s\n",
      " 13650K .......... .......... .......... .......... .......... 23%  122M 3s\n",
      " 13700K .......... .......... .......... .......... .......... 23%  104M 3s\n",
      " 13750K .......... .......... .......... .......... .......... 23%  118M 3s\n",
      " 13800K .......... .......... .......... .......... .......... 23%  123M 2s\n",
      " 13850K .......... .......... .......... .......... .......... 23%  102M 2s\n",
      " 13900K .......... .......... .......... .......... .......... 23%  118M 2s\n",
      " 13950K .......... .......... .......... .......... .......... 23%  102M 2s\n",
      " 14000K .......... .......... .......... .......... .......... 24%  113M 2s\n",
      " 14050K .......... .......... .......... .......... .......... 24%  127M 2s\n",
      " 14100K .......... .......... .......... .......... .......... 24%  101M 2s\n",
      " 14150K .......... .......... .......... .......... .......... 24%  120M 2s\n",
      " 14200K .......... .......... .......... .......... .......... 24%  101M 2s\n",
      " 14250K .......... .......... .......... .......... .......... 24%  118M 2s\n",
      " 14300K .......... .......... .......... .......... .......... 24%  120M 2s\n",
      " 14350K .......... .......... .......... .......... .......... 24%  102M 2s\n",
      " 14400K .......... .......... .......... .......... .......... 24%  120M 2s\n",
      " 14450K .......... .......... .......... .......... .......... 24%  102M 2s\n",
      " 14500K .......... .......... .......... .......... .......... 24%  119M 2s\n",
      " 14550K .......... .......... .......... .......... .......... 25%  119M 2s\n",
      " 14600K .......... .......... .......... .......... .......... 25% 99,4M 2s\n",
      " 14650K .......... .......... .......... .......... .......... 25%  119M 2s\n",
      " 14700K .......... .......... .......... .......... .......... 25%  105M 2s\n",
      " 14750K .......... .......... .......... .......... .......... 25%  117M 2s\n",
      " 14800K .......... .......... .......... .......... .......... 25% 3,77M 2s\n",
      " 14850K .......... .......... .......... .......... .......... 25%  103M 2s\n",
      " 14900K .......... .......... .......... .......... .......... 25%  119M 2s\n",
      " 14950K .......... .......... .......... .......... .......... 25%  102M 2s\n",
      " 15000K .......... .......... .......... .......... .......... 25%  120M 2s\n",
      " 15050K .......... .......... .......... .......... .......... 25%  122M 2s\n",
      " 15100K .......... .......... .......... .......... .......... 25%  102M 2s\n",
      " 15150K .......... .......... .......... .......... .......... 26%  118M 2s\n",
      " 15200K .......... .......... .......... .......... .......... 26%  103M 2s\n",
      " 15250K .......... .......... .......... .......... .......... 26%  120M 2s\n",
      " 15300K .......... .......... .......... .......... .......... 26%  117M 2s\n",
      " 15350K .......... .......... .......... .......... .......... 26%  102M 2s\n",
      " 15400K .......... .......... .......... .......... .......... 26%  122M 2s\n",
      " 15450K .......... .......... .......... .......... .......... 26%  102M 2s\n",
      " 15500K .......... .......... .......... .......... .......... 26% 92,1M 2s\n",
      " 15550K .......... .......... .......... .......... .......... 26% 52,5M 2s\n",
      " 15600K .......... .......... .......... .......... .......... 26% 50,5M 2s\n",
      " 15650K .......... .......... .......... .......... .......... 26% 30,4M 2s\n",
      " 15700K .......... .......... .......... .......... .......... 26%  107M 2s\n",
      " 15750K .......... .......... .......... .......... .......... 27% 3,56M 2s\n",
      " 15800K .......... .......... .......... .......... .......... 27% 46,2M 2s\n",
      " 15850K .......... .......... .......... .......... .......... 27% 55,1M 2s\n",
      " 15900K .......... .......... .......... .......... .......... 27%  115M 2s\n",
      " 15950K .......... .......... .......... .......... .......... 27%  103M 2s\n",
      " 16000K .......... .......... .......... .......... .......... 27%  110M 2s\n",
      " 16050K .......... .......... .......... .......... .......... 27%  118M 2s\n",
      " 16100K .......... .......... .......... .......... .......... 27%  108M 2s\n",
      " 16150K .......... .......... .......... .......... .......... 27%  129M 2s\n",
      " 16200K .......... .......... .......... .......... .......... 27% 16,1M 2s\n",
      " 16250K .......... .......... .......... .......... .......... 27%  121M 2s\n",
      " 16300K .......... .......... .......... .......... .......... 28%  117M 2s\n",
      " 16350K .......... .......... .......... .......... .......... 28%  102M 2s\n",
      " 16400K .......... .......... .......... .......... .......... 28%  120M 2s\n",
      " 16450K .......... .......... .......... .......... .......... 28%  104M 2s\n",
      " 16500K .......... .......... .......... .......... .......... 28%  110M 2s\n",
      " 16550K .......... .......... .......... .......... .......... 28%  129M 2s\n",
      " 16600K .......... .......... .......... .......... .......... 28% 53,2M 2s\n",
      " 16650K .......... .......... .......... .......... .......... 28%  120M 2s\n",
      " 16700K .......... .......... .......... .......... .......... 28%  103M 2s\n",
      " 16750K .......... .......... .......... .......... .......... 28%  118M 2s\n",
      " 16800K .......... .......... .......... .......... .......... 28%  120M 2s\n",
      " 16850K .......... .......... .......... .......... .......... 28%  102M 2s\n",
      " 16900K .......... .......... .......... .......... .......... 29%  118M 2s\n",
      " 16950K .......... .......... .......... .......... .......... 29%  102M 2s\n",
      " 17000K .......... .......... .......... .......... .......... 29%  121M 2s\n",
      " 17050K .......... .......... .......... .......... .......... 29%  117M 2s\n",
      " 17100K .......... .......... .......... .......... .......... 29%  102M 2s\n",
      " 17150K .......... .......... .......... .......... .......... 29%  121M 2s\n",
      " 17200K .......... .......... .......... .......... .......... 29%  102M 2s\n",
      " 17250K .......... .......... .......... .......... .......... 29%  121M 2s\n",
      " 17300K .......... .......... .......... .......... .......... 29%  116M 2s\n",
      " 17350K .......... .......... .......... .......... .......... 29%  103M 2s\n",
      " 17400K .......... .......... .......... .......... .......... 29%  121M 2s\n",
      " 17450K .......... .......... .......... .......... .......... 29%  103M 2s\n",
      " 17500K .......... .......... .......... .......... .......... 30%  117M 2s\n",
      " 17550K .......... .......... .......... .......... .......... 30%  120M 2s\n",
      " 17600K .......... .......... .......... .......... .......... 30%  102M 2s\n",
      " 17650K .......... .......... .......... .......... .......... 30%  121M 2s\n",
      " 17700K .......... .......... .......... .......... .......... 30%  102M 2s\n",
      " 17750K .......... .......... .......... .......... .......... 30%  118M 2s\n",
      " 17800K .......... .......... .......... .......... .......... 30%  121M 2s\n",
      " 17850K .......... .......... .......... .......... .......... 30% 4,32M 2s\n",
      " 17900K .......... .......... .......... .......... .......... 30% 86,5M 2s\n",
      " 17950K .......... .......... .......... .......... .......... 30% 58,7M 2s\n",
      " 18000K .......... .......... .......... .......... .......... 30% 64,5M 2s\n",
      " 18050K .......... .......... .......... .......... .......... 31%  118M 2s\n",
      " 18100K .......... .......... .......... .......... .......... 31%  103M 2s\n",
      " 18150K .......... .......... .......... .......... .......... 31%  120M 2s\n",
      " 18200K .......... .......... .......... .......... .......... 31%  102M 2s\n",
      " 18250K .......... .......... .......... .......... .......... 31%  121M 2s\n",
      " 18300K .......... .......... .......... .......... .......... 31%  118M 2s\n",
      " 18350K .......... .......... .......... .......... .......... 31%  101M 2s\n",
      " 18400K .......... .......... .......... .......... .......... 31%  121M 2s\n",
      " 18450K .......... .......... .......... .......... .......... 31%  102M 2s\n",
      " 18500K .......... .......... .......... .......... .......... 31%  117M 2s\n",
      " 18550K .......... .......... .......... .......... .......... 31%  122M 2s\n",
      " 18600K .......... .......... .......... .......... .......... 31% 91,3M 2s\n",
      " 18650K .......... .......... .......... .......... .......... 32%  117M 2s\n",
      " 18700K .......... .......... .......... .......... .......... 32% 77,0M 2s\n",
      " 18750K .......... .......... .......... .......... .......... 32% 27,5M 2s\n",
      " 18800K .......... .......... .......... .......... .......... 32% 4,06M 2s\n",
      " 18850K .......... .......... .......... .......... .......... 32% 14,7M 2s\n",
      " 18900K .......... .......... .......... .......... .......... 32% 70,0M 2s\n",
      " 18950K .......... .......... .......... .......... .......... 32% 66,6M 2s\n",
      " 19000K .......... .......... .......... .......... .......... 32% 55,1M 2s\n",
      " 19050K .......... .......... .......... .......... .......... 32%  120M 2s\n",
      " 19100K .......... .......... .......... .......... .......... 32%  105M 2s\n",
      " 19150K .......... .......... .......... .......... .......... 32%  119M 2s\n",
      " 19200K .......... .......... .......... .......... .......... 32% 95,8M 2s\n",
      " 19250K .......... .......... .......... .......... .......... 33% 61,2M 2s\n",
      " 19300K .......... .......... .......... .......... .......... 33% 25,2M 2s\n",
      " 19350K .......... .......... .......... .......... .......... 33% 62,0M 2s\n",
      " 19400K .......... .......... .......... .......... .......... 33% 73,2M 2s\n",
      " 19450K .......... .......... .......... .......... .......... 33%  121M 2s\n",
      " 19500K .......... .......... .......... .......... .......... 33%  151M 2s\n",
      " 19550K .......... .......... .......... .......... .......... 33%  147M 2s\n",
      " 19600K .......... .......... .......... .......... .......... 33%  141M 2s\n",
      " 19650K .......... .......... .......... .......... .......... 33%  123M 2s\n",
      " 19700K .......... .......... .......... .......... .......... 33%  102M 2s\n",
      " 19750K .......... .......... .......... .......... .......... 33%  116M 2s\n",
      " 19800K .......... .......... .......... .......... .......... 34% 38,3M 2s\n",
      " 19850K .......... .......... .......... .......... .......... 34%  107M 2s\n",
      " 19900K .......... .......... .......... .......... .......... 34%  118M 2s\n",
      " 19950K .......... .......... .......... .......... .......... 34%  102M 2s\n",
      " 20000K .......... .......... .......... .......... .......... 34%  119M 2s\n",
      " 20050K .......... .......... .......... .......... .......... 34%  119M 2s\n",
      " 20100K .......... .......... .......... .......... .......... 34%  102M 2s\n",
      " 20150K .......... .......... .......... .......... .......... 34%  120M 2s\n",
      " 20200K .......... .......... .......... .......... .......... 34%  104M 2s\n",
      " 20250K .......... .......... .......... .......... .......... 34%  116M 2s\n",
      " 20300K .......... .......... .......... .......... .......... 34%  118M 2s\n",
      " 20350K .......... .......... .......... .......... .......... 34%  104M 2s\n",
      " 20400K .......... .......... .......... .......... .......... 35%  120M 2s\n",
      " 20450K .......... .......... .......... .......... .......... 35%  104M 2s\n",
      " 20500K .......... .......... .......... .......... .......... 35%  116M 2s\n",
      " 20550K .......... .......... .......... .......... .......... 35%  122M 2s\n",
      " 20600K .......... .......... .......... .......... .......... 35% 52,0M 2s\n",
      " 20650K .......... .......... .......... .......... .......... 35%  142M 2s\n",
      " 20700K .......... .......... .......... .......... .......... 35%  157M 2s\n",
      " 20750K .......... .......... .......... .......... .......... 35%  290M 2s\n",
      " 20800K .......... .......... .......... .......... .......... 35%  115M 2s\n",
      " 20850K .......... .......... .......... .......... .......... 35%  102M 2s\n",
      " 20900K .......... .......... .......... .......... .......... 35% 6,67M 2s\n",
      " 20950K .......... .......... .......... .......... .......... 35% 11,5M 2s\n",
      " 21000K .......... .......... .......... .......... .......... 36%  126M 2s\n",
      " 21050K .......... .......... .......... .......... .......... 36% 60,0M 2s\n",
      " 21100K .......... .......... .......... .......... .......... 36%  102M 2s\n",
      " 21150K .......... .......... .......... .......... .......... 36%  122M 2s\n",
      " 21200K .......... .......... .......... .......... .......... 36%  101M 2s\n",
      " 21250K .......... .......... .......... .......... .......... 36%  119M 2s\n",
      " 21300K .......... .......... .......... .......... .......... 36%  121M 2s\n",
      " 21350K .......... .......... .......... .......... .......... 36%  102M 2s\n",
      " 21400K .......... .......... .......... .......... .......... 36%  122M 2s\n",
      " 21450K .......... .......... .......... .......... .......... 36%  102M 2s\n",
      " 21500K .......... .......... .......... .......... .......... 36%  107M 2s\n",
      " 21550K .......... .......... .......... .......... .......... 37%  118M 2s\n",
      " 21600K .......... .......... .......... .......... .......... 37%  116M 2s\n",
      " 21650K .......... .......... .......... .......... .......... 37%  117M 2s\n",
      " 21700K .......... .......... .......... .......... .......... 37%  103M 2s\n",
      " 21750K .......... .......... .......... .......... .......... 37%  121M 2s\n",
      " 21800K .......... .......... .......... .......... .......... 37% 32,6M 2s\n",
      " 21850K .......... .......... .......... .......... .......... 37% 11,5M 2s\n",
      " 21900K .......... .......... .......... .......... .......... 37% 4,37M 2s\n",
      " 21950K .......... .......... .......... .......... .......... 37%  103M 2s\n",
      " 22000K .......... .......... .......... .......... .......... 37% 41,4M 2s\n",
      " 22050K .......... .......... .......... .......... .......... 37% 55,2M 2s\n",
      " 22100K .......... .......... .......... .......... .......... 37%  101M 2s\n",
      " 22150K .......... .......... .......... .......... .......... 38%  119M 2s\n",
      " 22200K .......... .......... .......... .......... .......... 38%  103M 2s\n",
      " 22250K .......... .......... .......... .......... .......... 38% 99,7M 2s\n",
      " 22300K .......... .......... .......... .......... .......... 38% 42,5M 2s\n",
      " 22350K .......... .......... .......... .......... .......... 38% 43,0M 2s\n",
      " 22400K .......... .......... .......... .......... .......... 38% 27,1M 2s\n",
      " 22450K .......... .......... .......... .......... .......... 38%  104M 2s\n",
      " 22500K .......... .......... .......... .......... .......... 38%  122M 2s\n",
      " 22550K .......... .......... .......... .......... .......... 38%  120M 1s\n",
      " 22600K .......... .......... .......... .......... .......... 38% 51,4M 1s\n",
      " 22650K .......... .......... .......... .......... .......... 38%  181M 1s\n",
      " 22700K .......... .......... .......... .......... .......... 38%  171M 1s\n",
      " 22750K .......... .......... .......... .......... .......... 39%  183M 1s\n",
      " 22800K .......... .......... .......... .......... .......... 39%  117M 1s\n",
      " 22850K .......... .......... .......... .......... .......... 39%  103M 1s\n",
      " 22900K .......... .......... .......... .......... .......... 39%  120M 1s\n",
      " 22950K .......... .......... .......... .......... .......... 39%  105M 1s\n",
      " 23000K .......... .......... .......... .......... .......... 39%  118M 1s\n",
      " 23050K .......... .......... .......... .......... .......... 39% 71,5M 1s\n",
      " 23100K .......... .......... .......... .......... .......... 39%  102M 1s\n",
      " 23150K .......... .......... .......... .......... .......... 39%  120M 1s\n",
      " 23200K .......... .......... .......... .......... .......... 39%  103M 1s\n",
      " 23250K .......... .......... .......... .......... .......... 39%  117M 1s\n",
      " 23300K .......... .......... .......... .......... .......... 40%  121M 1s\n",
      " 23350K .......... .......... .......... .......... .......... 40% 85,6M 1s\n",
      " 23400K .......... .......... .......... .......... .......... 40%  117M 1s\n",
      " 23450K .......... .......... .......... .......... .......... 40%  103M 1s\n",
      " 23500K .......... .......... .......... .......... .......... 40%  121M 1s\n",
      " 23550K .......... .......... .......... .......... .......... 40%  121M 1s\n",
      " 23600K .......... .......... .......... .......... .......... 40%  102M 1s\n",
      " 23650K .......... .......... .......... .......... .......... 40%  118M 1s\n",
      " 23700K .......... .......... .......... .......... .......... 40%  102M 1s\n",
      " 23750K .......... .......... .......... .......... .......... 40%  122M 1s\n",
      " 23800K .......... .......... .......... .......... .......... 40%  117M 1s\n",
      " 23850K .......... .......... .......... .......... .......... 40%  103M 1s\n",
      " 23900K .......... .......... .......... .......... .......... 41%  120M 1s\n",
      " 23950K .......... .......... .......... .......... .......... 41% 6,79M 1s\n",
      " 24000K .......... .......... .......... .......... .......... 41% 11,8M 1s\n",
      " 24050K .......... .......... .......... .......... .......... 41% 48,3M 1s\n",
      " 24100K .......... .......... .......... .......... .......... 41%  101M 1s\n",
      " 24150K .......... .......... .......... .......... .......... 41% 76,5M 1s\n",
      " 24200K .......... .......... .......... .......... .......... 41%  156M 1s\n",
      " 24250K .......... .......... .......... .......... .......... 41% 79,3M 1s\n",
      " 24300K .......... .......... .......... .......... .......... 41%  121M 1s\n",
      " 24350K .......... .......... .......... .......... .......... 41%  102M 1s\n",
      " 24400K .......... .......... .......... .......... .......... 41%  119M 1s\n",
      " 24450K .......... .......... .......... .......... .......... 41%  102M 1s\n",
      " 24500K .......... .......... .......... .......... .......... 42%  119M 1s\n",
      " 24550K .......... .......... .......... .......... .......... 42%  118M 1s\n",
      " 24600K .......... .......... .......... .......... .......... 42%  102M 1s\n",
      " 24650K .......... .......... .......... .......... .......... 42%  123M 1s\n",
      " 24700K .......... .......... .......... .......... .......... 42%  102M 1s\n",
      " 24750K .......... .......... .......... .......... .......... 42%  119M 1s\n",
      " 24800K .......... .......... .......... .......... .......... 42%  119M 1s\n",
      " 24850K .......... .......... .......... .......... .......... 42%  103M 1s\n",
      " 24900K .......... .......... .......... .......... .......... 42% 9,17M 1s\n",
      " 24950K .......... .......... .......... .......... .......... 42% 4,84M 1s\n",
      " 25000K .......... .......... .......... .......... .......... 42% 26,8M 1s\n",
      " 25050K .......... .......... .......... .......... .......... 43%  121M 1s\n",
      " 25100K .......... .......... .......... .......... .......... 43% 40,9M 1s\n",
      " 25150K .......... .......... .......... .......... .......... 43%  123M 1s\n",
      " 25200K .......... .......... .......... .......... .......... 43%  101M 1s\n",
      " 25250K .......... .......... .......... .......... .......... 43%  117M 1s\n",
      " 25300K .......... .......... .......... .......... .......... 43%  124M 1s\n",
      " 25350K .......... .......... .......... .......... .......... 43%  101M 1s\n",
      " 25400K .......... .......... .......... .......... .......... 43% 84,8M 1s\n",
      " 25450K .......... .......... .......... .......... .......... 43% 20,7M 1s\n",
      " 25500K .......... .......... .......... .......... .......... 43% 38,4M 1s\n",
      " 25550K .......... .......... .......... .......... .......... 43%  121M 1s\n",
      " 25600K .......... .......... .......... .......... .......... 43% 52,9M 1s\n",
      " 25650K .......... .......... .......... .......... .......... 44%  120M 1s\n",
      " 25700K .......... .......... .......... .......... .......... 44%  103M 1s\n",
      " 25750K .......... .......... .......... .......... .......... 44%  121M 1s\n",
      " 25800K .......... .......... .......... .......... .......... 44%  117M 1s\n",
      " 25850K .......... .......... .......... .......... .......... 44%  102M 1s\n",
      " 25900K .......... .......... .......... .......... .......... 44%  121M 1s\n",
      " 25950K .......... .......... .......... .......... .......... 44%  103M 1s\n",
      " 26000K .......... .......... .......... .......... .......... 44%  117M 1s\n",
      " 26050K .......... .......... .......... .......... .......... 44%  121M 1s\n",
      " 26100K .......... .......... .......... .......... .......... 44%  103M 1s\n",
      " 26150K .......... .......... .......... .......... .......... 44%  117M 1s\n",
      " 26200K .......... .......... .......... .......... .......... 44%  102M 1s\n",
      " 26250K .......... .......... .......... .......... .......... 45%  121M 1s\n",
      " 26300K .......... .......... .......... .......... .......... 45%  121M 1s\n",
      " 26350K .......... .......... .......... .......... .......... 45%  102M 1s\n",
      " 26400K .......... .......... .......... .......... .......... 45%  117M 1s\n",
      " 26450K .......... .......... .......... .......... .......... 45%  102M 1s\n",
      " 26500K .......... .......... .......... .......... .......... 45%  122M 1s\n",
      " 26550K .......... .......... .......... .......... .......... 45%  117M 1s\n",
      " 26600K .......... .......... .......... .......... .......... 45%  103M 1s\n",
      " 26650K .......... .......... .......... .......... .......... 45%  121M 1s\n",
      " 26700K .......... .......... .......... .......... .......... 45% 93,7M 1s\n",
      " 26750K .......... .......... .......... .......... .......... 45%  135M 1s\n",
      " 26800K .......... .......... .......... .......... .......... 46%  118M 1s\n",
      " 26850K .......... .......... .......... .......... .......... 46%  103M 1s\n",
      " 26900K .......... .......... .......... .......... .......... 46%  120M 1s\n",
      " 26950K .......... .......... .......... .......... .......... 46%  103M 1s\n",
      " 27000K .......... .......... .......... .......... .......... 46% 49,3M 1s\n",
      " 27050K .......... .......... .......... .......... .......... 46% 4,69M 1s\n",
      " 27100K .......... .......... .......... .......... .......... 46%  108M 1s\n",
      " 27150K .......... .......... .......... .......... .......... 46% 66,9M 1s\n",
      " 27200K .......... .......... .......... .......... .......... 46%  102M 1s\n",
      " 27250K .......... .......... .......... .......... .......... 46%  121M 1s\n",
      " 27300K .......... .......... .......... .......... .......... 46%  120M 1s\n",
      " 27350K .......... .......... .......... .......... .......... 46%  103M 1s\n",
      " 27400K .......... .......... .......... .......... .......... 47%  117M 1s\n",
      " 27450K .......... .......... .......... .......... .......... 47% 55,8M 1s\n",
      " 27500K .......... .......... .......... .......... .......... 47%  121M 1s\n",
      " 27550K .......... .......... .......... .......... .......... 47%  118M 1s\n",
      " 27600K .......... .......... .......... .......... .......... 47%  102M 1s\n",
      " 27650K .......... .......... .......... .......... .......... 47%  121M 1s\n",
      " 27700K .......... .......... .......... .......... .......... 47%  102M 1s\n",
      " 27750K .......... .......... .......... .......... .......... 47%  117M 1s\n",
      " 27800K .......... .......... .......... .......... .......... 47%  122M 1s\n",
      " 27850K .......... .......... .......... .......... .......... 47%  102M 1s\n",
      " 27900K .......... .......... .......... .......... .......... 47% 79,7M 1s\n",
      " 27950K .......... .......... .......... .......... .......... 47% 24,5M 1s\n",
      " 28000K .......... .......... .......... .......... .......... 48% 4,09M 1s\n",
      " 28050K .......... .......... .......... .......... .......... 48% 21,2M 1s\n",
      " 28100K .......... .......... .......... .......... .......... 48% 29,8M 1s\n",
      " 28150K .......... .......... .......... .......... .......... 48% 64,8M 1s\n",
      " 28200K .......... .......... .......... .......... .......... 48%  113M 1s\n",
      " 28250K .......... .......... .......... .......... .......... 48%  123M 1s\n",
      " 28300K .......... .......... .......... .......... .......... 48%  117M 1s\n",
      " 28350K .......... .......... .......... .......... .......... 48%  102M 1s\n",
      " 28400K .......... .......... .......... .......... .......... 48%  122M 1s\n",
      " 28450K .......... .......... .......... .......... .......... 48%  102M 1s\n",
      " 28500K .......... .......... .......... .......... .......... 48% 30,1M 1s\n",
      " 28550K .......... .......... .......... .......... .......... 49% 25,6M 1s\n",
      " 28600K .......... .......... .......... .......... .......... 49% 44,3M 1s\n",
      " 28650K .......... .......... .......... .......... .......... 49%  121M 1s\n",
      " 28700K .......... .......... .......... .......... .......... 49%  102M 1s\n",
      " 28750K .......... .......... .......... .......... .......... 49% 88,3M 1s\n",
      " 28800K .......... .......... .......... .......... .......... 49%  182M 1s\n",
      " 28850K .......... .......... .......... .......... .......... 49%  102M 1s\n",
      " 28900K .......... .......... .......... .......... .......... 49%  122M 1s\n",
      " 28950K .......... .......... .......... .......... .......... 49%  102M 1s\n",
      " 29000K .......... .......... .......... .......... .......... 49%  117M 1s\n",
      " 29050K .......... .......... .......... .......... .......... 49%  121M 1s\n",
      " 29100K .......... .......... .......... .......... .......... 49%  102M 1s\n",
      " 29150K .......... .......... .......... .......... .......... 50%  118M 1s\n",
      " 29200K .......... .......... .......... .......... .......... 50%  102M 1s\n",
      " 29250K .......... .......... .......... .......... .......... 50%  121M 1s\n",
      " 29300K .......... .......... .......... .......... .......... 50%  117M 1s\n",
      " 29350K .......... .......... .......... .......... .......... 50%  102M 1s\n",
      " 29400K .......... .......... .......... .......... .......... 50%  121M 1s\n",
      " 29450K .......... .......... .......... .......... .......... 50%  102M 1s\n",
      " 29500K .......... .......... .......... .......... .......... 50%  121M 1s\n",
      " 29550K .......... .......... .......... .......... .......... 50%  118M 1s\n",
      " 29600K .......... .......... .......... .......... .......... 50%  102M 1s\n",
      " 29650K .......... .......... .......... .......... .......... 50%  121M 1s\n",
      " 29700K .......... .......... .......... .......... .......... 50%  102M 1s\n",
      " 29750K .......... .......... .......... .......... .......... 51%  117M 1s\n",
      " 29800K .......... .......... .......... .......... .......... 51%  121M 1s\n",
      " 29850K .......... .......... .......... .......... .......... 51%  102M 1s\n",
      " 29900K .......... .......... .......... .......... .......... 51%  118M 1s\n",
      " 29950K .......... .......... .......... .......... .......... 51%  102M 1s\n",
      " 30000K .......... .......... .......... .......... .......... 51%  121M 1s\n",
      " 30050K .......... .......... .......... .......... .......... 51% 13,4M 1s\n",
      " 30100K .......... .......... .......... .......... .......... 51% 7,33M 1s\n",
      " 30150K .......... .......... .......... .......... .......... 51% 45,2M 1s\n",
      " 30200K .......... .......... .......... .......... .......... 51% 39,1M 1s\n",
      " 30250K .......... .......... .......... .......... .......... 51% 55,0M 1s\n",
      " 30300K .......... .......... .......... .......... .......... 52%  117M 1s\n",
      " 30350K .......... .......... .......... .......... .......... 52%  103M 1s\n",
      " 30400K .......... .......... .......... .......... .......... 52%  120M 1s\n",
      " 30450K .......... .......... .......... .......... .......... 52%  103M 1s\n",
      " 30500K .......... .......... .......... .......... .......... 52%  120M 1s\n",
      " 30550K .......... .......... .......... .......... .......... 52%  117M 1s\n",
      " 30600K .......... .......... .......... .......... .......... 52%  103M 1s\n",
      " 30650K .......... .......... .......... .......... .......... 52%  117M 1s\n",
      " 30700K .......... .......... .......... .......... .......... 52% 88,6M 1s\n",
      " 30750K .......... .......... .......... .......... .......... 52%  117M 1s\n",
      " 30800K .......... .......... .......... .......... .......... 52%  109M 1s\n",
      " 30850K .......... .......... .......... .......... .......... 52%  113M 1s\n",
      " 30900K .......... .......... .......... .......... .......... 53%  117M 1s\n",
      " 30950K .......... .......... .......... .......... .......... 53%  103M 1s\n",
      " 31000K .......... .......... .......... .......... .......... 53% 74,0M 1s\n",
      " 31050K .......... .......... .......... .......... .......... 53% 4,68M 1s\n",
      " 31100K .......... .......... .......... .......... .......... 53% 11,0M 1s\n",
      " 31150K .......... .......... .......... .......... .......... 53% 42,8M 1s\n",
      " 31200K .......... .......... .......... .......... .......... 53% 46,2M 1s\n",
      " 31250K .......... .......... .......... .......... .......... 53% 30,6M 1s\n",
      " 31300K .......... .......... .......... .......... .......... 53%  117M 1s\n",
      " 31350K .......... .......... .......... .......... .......... 53%  102M 1s\n",
      " 31400K .......... .......... .......... .......... .......... 53%  121M 1s\n",
      " 31450K .......... .......... .......... .......... .......... 53%  104M 1s\n",
      " 31500K .......... .......... .......... .......... .......... 54%  116M 1s\n",
      " 31550K .......... .......... .......... .......... .......... 54% 27,5M 1s\n",
      " 31600K .......... .......... .......... .......... .......... 54% 24,7M 1s\n",
      " 31650K .......... .......... .......... .......... .......... 54%  120M 1s\n",
      " 31700K .......... .......... .......... .......... .......... 54% 67,2M 1s\n",
      " 31750K .......... .......... .......... .......... .......... 54%  118M 1s\n",
      " 31800K .......... .......... .......... .......... .......... 54%  121M 1s\n",
      " 31850K .......... .......... .......... .......... .......... 54%  102M 1s\n",
      " 31900K .......... .......... .......... .......... .......... 54%  118M 1s\n",
      " 31950K .......... .......... .......... .......... .......... 54%  102M 1s\n",
      " 32000K .......... .......... .......... .......... .......... 54%  122M 1s\n",
      " 32050K .......... .......... .......... .......... .......... 55%  117M 1s\n",
      " 32100K .......... .......... .......... .......... .......... 55%  102M 1s\n",
      " 32150K .......... .......... .......... .......... .......... 55%  122M 1s\n",
      " 32200K .......... .......... .......... .......... .......... 55%  102M 1s\n",
      " 32250K .......... .......... .......... .......... .......... 55%  121M 1s\n",
      " 32300K .......... .......... .......... .......... .......... 55%  117M 1s\n",
      " 32350K .......... .......... .......... .......... .......... 55%  102M 1s\n",
      " 32400K .......... .......... .......... .......... .......... 55%  122M 1s\n",
      " 32450K .......... .......... .......... .......... .......... 55%  102M 1s\n",
      " 32500K .......... .......... .......... .......... .......... 55%  117M 1s\n",
      " 32550K .......... .......... .......... .......... .......... 55%  121M 1s\n",
      " 32600K .......... .......... .......... .......... .......... 55%  102M 1s\n",
      " 32650K .......... .......... .......... .......... .......... 56%  121M 1s\n",
      " 32700K .......... .......... .......... .......... .......... 56%  103M 1s\n",
      " 32750K .......... .......... .......... .......... .......... 56%  117M 1s\n",
      " 32800K .......... .......... .......... .......... .......... 56%  121M 1s\n",
      " 32850K .......... .......... .......... .......... .......... 56%  102M 1s\n",
      " 32900K .......... .......... .......... .......... .......... 56%  117M 1s\n",
      " 32950K .......... .......... .......... .......... .......... 56%  102M 1s\n",
      " 33000K .......... .......... .......... .......... .......... 56%  121M 1s\n",
      " 33050K .......... .......... .......... .......... .......... 56%  117M 1s\n",
      " 33100K .......... .......... .......... .......... .......... 56% 19,5M 1s\n",
      " 33150K .......... .......... .......... .......... .......... 56% 10,0M 1s\n",
      " 33200K .......... .......... .......... .......... .......... 56% 10,8M 1s\n",
      " 33250K .......... .......... .......... .......... .......... 57%  121M 1s\n",
      " 33300K .......... .......... .......... .......... .......... 57% 48,4M 1s\n",
      " 33350K .......... .......... .......... .......... .......... 57%  102M 1s\n",
      " 33400K .......... .......... .......... .......... .......... 57%  121M 1s\n",
      " 33450K .......... .......... .......... .......... .......... 57%  102M 1s\n",
      " 33500K .......... .......... .......... .......... .......... 57%  117M 1s\n",
      " 33550K .......... .......... .......... .......... .......... 57%  121M 1s\n",
      " 33600K .......... .......... .......... .......... .......... 57%  102M 1s\n",
      " 33650K .......... .......... .......... .......... .......... 57%  118M 1s\n",
      " 33700K .......... .......... .......... .......... .......... 57%  103M 1s\n",
      " 33750K .......... .......... .......... .......... .......... 57%  121M 1s\n",
      " 33800K .......... .......... .......... .......... .......... 58%  121M 1s\n",
      " 33850K .......... .......... .......... .......... .......... 58%  102M 1s\n",
      " 33900K .......... .......... .......... .......... .......... 58% 86,5M 1s\n",
      " 33950K .......... .......... .......... .......... .......... 58%  103M 1s\n",
      " 34000K .......... .......... .......... .......... .......... 58%  121M 1s\n",
      " 34050K .......... .......... .......... .......... .......... 58%  117M 1s\n",
      " 34100K .......... .......... .......... .......... .......... 58% 6,21M 1s\n",
      " 34150K .......... .......... .......... .......... .......... 58% 6,26M 1s\n",
      " 34200K .......... .......... .......... .......... .......... 58% 47,7M 1s\n",
      " 34250K .......... .......... .......... .......... .......... 58%  123M 1s\n",
      " 34300K .......... .......... .......... .......... .......... 58% 55,7M 1s\n",
      " 34350K .......... .......... .......... .......... .......... 58% 60,2M 1s\n",
      " 34400K .......... .......... .......... .......... .......... 59% 55,3M 1s\n",
      " 34450K .......... .......... .......... .......... .......... 59%  108M 1s\n",
      " 34500K .......... .......... .......... .......... .......... 59% 83,5M 1s\n",
      " 34550K .......... .......... .......... .......... .......... 59%  122M 1s\n",
      " 34600K .......... .......... .......... .......... .......... 59% 88,1M 1s\n",
      " 34650K .......... .......... .......... .......... .......... 59% 16,7M 1s\n",
      " 34700K .......... .......... .......... .......... .......... 59%  103M 1s\n",
      " 34750K .......... .......... .......... .......... .......... 59% 71,9M 1s\n",
      " 34800K .......... .......... .......... .......... .......... 59%  125M 1s\n",
      " 34850K .......... .......... .......... .......... .......... 59% 99,9M 1s\n",
      " 34900K .......... .......... .......... .......... .......... 59% 61,8M 1s\n",
      " 34950K .......... .......... .......... .......... .......... 59%  103M 1s\n",
      " 35000K .......... .......... .......... .......... .......... 60%  122M 1s\n",
      " 35050K .......... .......... .......... .......... .......... 60%  116M 1s\n",
      " 35100K .......... .......... .......... .......... .......... 60%  104M 1s\n",
      " 35150K .......... .......... .......... .......... .......... 60%  122M 1s\n",
      " 35200K .......... .......... .......... .......... .......... 60%  101M 1s\n",
      " 35250K .......... .......... .......... .......... .......... 60%  118M 1s\n",
      " 35300K .......... .......... .......... .......... .......... 60%  121M 1s\n",
      " 35350K .......... .......... .......... .......... .......... 60%  102M 1s\n",
      " 35400K .......... .......... .......... .......... .......... 60%  121M 1s\n",
      " 35450K .......... .......... .......... .......... .......... 60%  103M 1s\n",
      " 35500K .......... .......... .......... .......... .......... 60%  111M 1s\n",
      " 35550K .......... .......... .......... .......... .......... 60%  122M 1s\n",
      " 35600K .......... .......... .......... .......... .......... 61%  103M 1s\n",
      " 35650K .......... .......... .......... .......... .......... 61%  117M 1s\n",
      " 35700K .......... .......... .......... .......... .......... 61% 86,4M 1s\n",
      " 35750K .......... .......... .......... .......... .......... 61%  148M 1s\n",
      " 35800K .......... .......... .......... .......... .......... 61%  115M 1s\n",
      " 35850K .......... .......... .......... .......... .......... 61%  107M 1s\n",
      " 35900K .......... .......... .......... .......... .......... 61%  122M 1s\n",
      " 35950K .......... .......... .......... .......... .......... 61%  102M 1s\n",
      " 36000K .......... .......... .......... .......... .......... 61%  121M 1s\n",
      " 36050K .......... .......... .......... .......... .......... 61%  118M 1s\n",
      " 36100K .......... .......... .......... .......... .......... 61%  102M 1s\n",
      " 36150K .......... .......... .......... .......... .......... 62%  122M 1s\n",
      " 36200K .......... .......... .......... .......... .......... 62% 12,3M 1s\n",
      " 36250K .......... .......... .......... .......... .......... 62% 7,71M 1s\n",
      " 36300K .......... .......... .......... .......... .......... 62% 26,7M 1s\n",
      " 36350K .......... .......... .......... .......... .......... 62% 73,4M 1s\n",
      " 36400K .......... .......... .......... .......... .......... 62%  120M 1s\n",
      " 36450K .......... .......... .......... .......... .......... 62%  102M 1s\n",
      " 36500K .......... .......... .......... .......... .......... 62%  118M 1s\n",
      " 36550K .......... .......... .......... .......... .......... 62%  120M 1s\n",
      " 36600K .......... .......... .......... .......... .......... 62%  102M 1s\n",
      " 36650K .......... .......... .......... .......... .......... 62%  118M 1s\n",
      " 36700K .......... .......... .......... .......... .......... 62%  103M 1s\n",
      " 36750K .......... .......... .......... .......... .......... 63%  121M 1s\n",
      " 36800K .......... .......... .......... .......... .......... 63%  117M 1s\n",
      " 36850K .......... .......... .......... .......... .......... 63%  102M 1s\n",
      " 36900K .......... .......... .......... .......... .......... 63%  121M 1s\n",
      " 36950K .......... .......... .......... .......... .......... 63%  102M 1s\n",
      " 37000K .......... .......... .......... .......... .......... 63%  122M 1s\n",
      " 37050K .......... .......... .......... .......... .......... 63%  117M 1s\n",
      " 37100K .......... .......... .......... .......... .......... 63%  102M 1s\n",
      " 37150K .......... .......... .......... .......... .......... 63% 10,5M 1s\n",
      " 37200K .......... .......... .......... .......... .......... 63% 4,83M 1s\n",
      " 37250K .......... .......... .......... .......... .......... 63% 28,3M 1s\n",
      " 37300K .......... .......... .......... .......... .......... 63% 67,7M 1s\n",
      " 37350K .......... .......... .......... .......... .......... 64%  102M 1s\n",
      " 37400K .......... .......... .......... .......... .......... 64% 61,7M 1s\n",
      " 37450K .......... .......... .......... .......... .......... 64% 51,5M 1s\n",
      " 37500K .......... .......... .......... .......... .......... 64% 54,5M 1s\n",
      " 37550K .......... .......... .......... .......... .......... 64%  122M 1s\n",
      " 37600K .......... .......... .......... .......... .......... 64%  101M 1s\n",
      " 37650K .......... .......... .......... .......... .......... 64%  116M 1s\n",
      " 37700K .......... .......... .......... .......... .......... 64% 17,7M 1s\n",
      " 37750K .......... .......... .......... .......... .......... 64% 47,9M 1s\n",
      " 37800K .......... .......... .......... .......... .......... 64%  117M 1s\n",
      " 37850K .......... .......... .......... .......... .......... 64% 57,7M 1s\n",
      " 37900K .......... .......... .......... .......... .......... 65%  106M 1s\n",
      " 37950K .......... .......... .......... .......... .......... 65%  120M 1s\n",
      " 38000K .......... .......... .......... .......... .......... 65%  118M 1s\n",
      " 38050K .......... .......... .......... .......... .......... 65%  115M 1s\n",
      " 38100K .......... .......... .......... .......... .......... 65%  107M 1s\n",
      " 38150K .......... .......... .......... .......... .......... 65%  120M 1s\n",
      " 38200K .......... .......... .......... .......... .......... 65%  102M 1s\n",
      " 38250K .......... .......... .......... .......... .......... 65%  117M 1s\n",
      " 38300K .......... .......... .......... .......... .......... 65%  120M 1s\n",
      " 38350K .......... .......... .......... .......... .......... 65%  103M 1s\n",
      " 38400K .......... .......... .......... .......... .......... 65%  119M 1s\n",
      " 38450K .......... .......... .......... .......... .......... 65% 89,8M 1s\n",
      " 38500K .......... .......... .......... .......... .......... 66%  142M 1s\n",
      " 38550K .......... .......... .......... .......... .......... 66%  118M 1s\n",
      " 38600K .......... .......... .......... .......... .......... 66%  101M 1s\n",
      " 38650K .......... .......... .......... .......... .......... 66%  118M 1s\n",
      " 38700K .......... .......... .......... .......... .......... 66%  107M 1s\n",
      " 38750K .......... .......... .......... .......... .......... 66%  118M 1s\n",
      " 38800K .......... .......... .......... .......... .......... 66%  120M 1s\n",
      " 38850K .......... .......... .......... .......... .......... 66%  101M 1s\n",
      " 38900K .......... .......... .......... .......... .......... 66%  121M 1s\n",
      " 38950K .......... .......... .......... .......... .......... 66%  104M 1s\n",
      " 39000K .......... .......... .......... .......... .......... 66%  117M 1s\n",
      " 39050K .......... .......... .......... .......... .......... 66%  121M 1s\n",
      " 39100K .......... .......... .......... .......... .......... 67%  102M 1s\n",
      " 39150K .......... .......... .......... .......... .......... 67%  124M 1s\n",
      " 39200K .......... .......... .......... .......... .......... 67%  102M 1s\n",
      " 39250K .......... .......... .......... .......... .......... 67% 12,6M 1s\n",
      " 39300K .......... .......... .......... .......... .......... 67% 9,79M 1s\n",
      " 39350K .......... .......... .......... .......... .......... 67% 14,4M 1s\n",
      " 39400K .......... .......... .......... .......... .......... 67% 85,2M 1s\n",
      " 39450K .......... .......... .......... .......... .......... 67% 56,7M 1s\n",
      " 39500K .......... .......... .......... .......... .......... 67%  120M 1s\n",
      " 39550K .......... .......... .......... .......... .......... 67%  118M 1s\n",
      " 39600K .......... .......... .......... .......... .......... 67%  103M 1s\n",
      " 39650K .......... .......... .......... .......... .......... 68%  122M 1s\n",
      " 39700K .......... .......... .......... .......... .......... 68%  103M 1s\n",
      " 39750K .......... .......... .......... .......... .......... 68%  120M 1s\n",
      " 39800K .......... .......... .......... .......... .......... 68%  105M 1s\n",
      " 39850K .......... .......... .......... .......... .......... 68%  115M 1s\n",
      " 39900K .......... .......... .......... .......... .......... 68%  121M 1s\n",
      " 39950K .......... .......... .......... .......... .......... 68%  102M 1s\n",
      " 40000K .......... .......... .......... .......... .......... 68%  118M 1s\n",
      " 40050K .......... .......... .......... .......... .......... 68%  121M 1s\n",
      " 40100K .......... .......... .......... .......... .......... 68%  102M 1s\n",
      " 40150K .......... .......... .......... .......... .......... 68%  122M 1s\n",
      " 40200K .......... .......... .......... .......... .......... 68% 57,4M 1s\n",
      " 40250K .......... .......... .......... .......... .......... 69% 4,13M 1s\n",
      " 40300K .......... .......... .......... .......... .......... 69% 16,1M 1s\n",
      " 40350K .......... .......... .......... .......... .......... 69% 40,0M 1s\n",
      " 40400K .......... .......... .......... .......... .......... 69% 59,6M 1s\n",
      " 40450K .......... .......... .......... .......... .......... 69% 71,8M 1s\n",
      " 40500K .......... .......... .......... .......... .......... 69%  120M 1s\n",
      " 40550K .......... .......... .......... .......... .......... 69%  117M 1s\n",
      " 40600K .......... .......... .......... .......... .......... 69% 87,0M 1s\n",
      " 40650K .......... .......... .......... .......... .......... 69%  120M 1s\n",
      " 40700K .......... .......... .......... .......... .......... 69% 20,8M 1s\n",
      " 40750K .......... .......... .......... .......... .......... 69% 51,1M 1s\n",
      " 40800K .......... .......... .......... .......... .......... 69% 35,0M 1s\n",
      " 40850K .......... .......... .......... .......... .......... 70% 51,8M 1s\n",
      " 40900K .......... .......... .......... .......... .......... 70%  121M 1s\n",
      " 40950K .......... .......... .......... .......... .......... 70%  102M 1s\n",
      " 41000K .......... .......... .......... .......... .......... 70%  118M 1s\n",
      " 41050K .......... .......... .......... .......... .......... 70%  121M 1s\n",
      " 41100K .......... .......... .......... .......... .......... 70%  103M 1s\n",
      " 41150K .......... .......... .......... .......... .......... 70%  117M 1s\n",
      " 41200K .......... .......... .......... .......... .......... 70%  102M 1s\n",
      " 41250K .......... .......... .......... .......... .......... 70%  120M 1s\n",
      " 41300K .......... .......... .......... .......... .......... 70%  123M 1s\n",
      " 41350K .......... .......... .......... .......... .......... 70%  102M 1s\n",
      " 41400K .......... .......... .......... .......... .......... 71%  117M 1s\n",
      " 41450K .......... .......... .......... .......... .......... 71%  102M 1s\n",
      " 41500K .......... .......... .......... .......... .......... 71%  122M 1s\n",
      " 41550K .......... .......... .......... .......... .......... 71%  116M 1s\n",
      " 41600K .......... .......... .......... .......... .......... 71%  102M 1s\n",
      " 41650K .......... .......... .......... .......... .......... 71%  122M 1s\n",
      " 41700K .......... .......... .......... .......... .......... 71%  101M 1s\n",
      " 41750K .......... .......... .......... .......... .......... 71%  122M 1s\n",
      " 41800K .......... .......... .......... .......... .......... 71%  117M 1s\n",
      " 41850K .......... .......... .......... .......... .......... 71%  102M 1s\n",
      " 41900K .......... .......... .......... .......... .......... 71%  121M 1s\n",
      " 41950K .......... .......... .......... .......... .......... 71%  102M 1s\n",
      " 42000K .......... .......... .......... .......... .......... 72%  118M 1s\n",
      " 42050K .......... .......... .......... .......... .......... 72%  121M 1s\n",
      " 42100K .......... .......... .......... .......... .......... 72%  102M 1s\n",
      " 42150K .......... .......... .......... .......... .......... 72%  117M 0s\n",
      " 42200K .......... .......... .......... .......... .......... 72% 88,0M 0s\n",
      " 42250K .......... .......... .......... .......... .......... 72%  122M 0s\n",
      " 42300K .......... .......... .......... .......... .......... 72% 17,6M 0s\n",
      " 42350K .......... .......... .......... .......... .......... 72% 10,2M 0s\n",
      " 42400K .......... .......... .......... .......... .......... 72% 13,3M 0s\n",
      " 42450K .......... .......... .......... .......... .......... 72% 25,8M 0s\n",
      " 42500K .......... .......... .......... .......... .......... 72%  122M 0s\n",
      " 42550K .......... .......... .......... .......... .......... 72%  118M 0s\n",
      " 42600K .......... .......... .......... .......... .......... 73%  101M 0s\n",
      " 42650K .......... .......... .......... .......... .......... 73%  124M 0s\n",
      " 42700K .......... .......... .......... .......... .......... 73%  102M 0s\n",
      " 42750K .......... .......... .......... .......... .......... 73%  117M 0s\n",
      " 42800K .......... .......... .......... .......... .......... 73%  121M 0s\n",
      " 42850K .......... .......... .......... .......... .......... 73%  102M 0s\n",
      " 42900K .......... .......... .......... .......... .......... 73%  123M 0s\n",
      " 42950K .......... .......... .......... .......... .......... 73%  103M 0s\n",
      " 43000K .......... .......... .......... .......... .......... 73%  115M 0s\n",
      " 43050K .......... .......... .......... .......... .......... 73%  122M 0s\n",
      " 43100K .......... .......... .......... .......... .......... 73%  102M 0s\n",
      " 43150K .......... .......... .......... .......... .......... 74%  119M 0s\n",
      " 43200K .......... .......... .......... .......... .......... 74%  101M 0s\n",
      " 43250K .......... .......... .......... .......... .......... 74% 49,3M 0s\n",
      " 43300K .......... .......... .......... .......... .......... 74% 7,44M 0s\n",
      " 43350K .......... .......... .......... .......... .......... 74% 5,61M 0s\n",
      " 43400K .......... .......... .......... .......... .......... 74%  125M 0s\n",
      " 43450K .......... .......... .......... .......... .......... 74% 49,6M 0s\n",
      " 43500K .......... .......... .......... .......... .......... 74% 60,1M 0s\n",
      " 43550K .......... .......... .......... .......... .......... 74%  116M 0s\n",
      " 43600K .......... .......... .......... .......... .......... 74%  103M 0s\n",
      " 43650K .......... .......... .......... .......... .......... 74%  120M 0s\n",
      " 43700K .......... .......... .......... .......... .......... 74%  103M 0s\n",
      " 43750K .......... .......... .......... .......... .......... 75% 94,2M 0s\n",
      " 43800K .......... .......... .......... .......... .......... 75% 14,6M 0s\n",
      " 43850K .......... .......... .......... .......... .......... 75%  105M 0s\n",
      " 43900K .......... .......... .......... .......... .......... 75% 53,9M 0s\n",
      " 43950K .......... .......... .......... .......... .......... 75% 58,4M 0s\n",
      " 44000K .......... .......... .......... .......... .......... 75%  112M 0s\n",
      " 44050K .......... .......... .......... .......... .......... 75% 90,5M 0s\n",
      " 44100K .......... .......... .......... .......... .......... 75%  128M 0s\n",
      " 44150K .......... .......... .......... .......... .......... 75%  132M 0s\n",
      " 44200K .......... .......... .......... .......... .......... 75%  104M 0s\n",
      " 44250K .......... .......... .......... .......... .......... 75%  124M 0s\n",
      " 44300K .......... .......... .......... .......... .......... 75%  115M 0s\n",
      " 44350K .......... .......... .......... .......... .......... 76%  102M 0s\n",
      " 44400K .......... .......... .......... .......... .......... 76%  118M 0s\n",
      " 44450K .......... .......... .......... .......... .......... 76%  105M 0s\n",
      " 44500K .......... .......... .......... .......... .......... 76% 87,6M 0s\n",
      " 44550K .......... .......... .......... .......... .......... 76%  117M 0s\n",
      " 44600K .......... .......... .......... .......... .......... 76%  103M 0s\n",
      " 44650K .......... .......... .......... .......... .......... 76%  120M 0s\n",
      " 44700K .......... .......... .......... .......... .......... 76%  104M 0s\n",
      " 44750K .......... .......... .......... .......... .......... 76%  117M 0s\n",
      " 44800K .......... .......... .......... .......... .......... 76%  121M 0s\n",
      " 44850K .......... .......... .......... .......... .......... 76%  101M 0s\n",
      " 44900K .......... .......... .......... .......... .......... 77%  116M 0s\n",
      " 44950K .......... .......... .......... .......... .......... 77%  104M 0s\n",
      " 45000K .......... .......... .......... .......... .......... 77%  121M 0s\n",
      " 45050K .......... .......... .......... .......... .......... 77%  123M 0s\n",
      " 45100K .......... .......... .......... .......... .......... 77%  103M 0s\n",
      " 45150K .......... .......... .......... .......... .......... 77% 85,6M 0s\n",
      " 45200K .......... .......... .......... .......... .......... 77%  102M 0s\n",
      " 45250K .......... .......... .......... .......... .......... 77%  105M 0s\n",
      " 45300K .......... .......... .......... .......... .......... 77%  140M 0s\n",
      " 45350K .......... .......... .......... .......... .......... 77% 11,2M 0s\n",
      " 45400K .......... .......... .......... .......... .......... 77%  120M 0s\n",
      " 45450K .......... .......... .......... .......... .......... 77% 7,17M 0s\n",
      " 45500K .......... .......... .......... .......... .......... 78% 53,3M 0s\n",
      " 45550K .......... .......... .......... .......... .......... 78%  120M 0s\n",
      " 45600K .......... .......... .......... .......... .......... 78%  103M 0s\n",
      " 45650K .......... .......... .......... .......... .......... 78%  120M 0s\n",
      " 45700K .......... .......... .......... .......... .......... 78%  104M 0s\n",
      " 45750K .......... .......... .......... .......... .......... 78%  116M 0s\n",
      " 45800K .......... .......... .......... .......... .......... 78%  121M 0s\n",
      " 45850K .......... .......... .......... .......... .......... 78% 78,1M 0s\n",
      " 45900K .......... .......... .......... .......... .......... 78%  118M 0s\n",
      " 45950K .......... .......... .......... .......... .......... 78%  102M 0s\n",
      " 46000K .......... .......... .......... .......... .......... 78%  121M 0s\n",
      " 46050K .......... .......... .......... .......... .......... 78%  122M 0s\n",
      " 46100K .......... .......... .......... .......... .......... 79%  102M 0s\n",
      " 46150K .......... .......... .......... .......... .......... 79%  117M 0s\n",
      " 46200K .......... .......... .......... .......... .......... 79%  102M 0s\n",
      " 46250K .......... .......... .......... .......... .......... 79%  122M 0s\n",
      " 46300K .......... .......... .......... .......... .......... 79%  118M 0s\n",
      " 46350K .......... .......... .......... .......... .......... 79% 10,1M 0s\n",
      " 46400K .......... .......... .......... .......... .......... 79% 4,66M 0s\n",
      " 46450K .......... .......... .......... .......... .......... 79% 33,5M 0s\n",
      " 46500K .......... .......... .......... .......... .......... 79%  119M 0s\n",
      " 46550K .......... .......... .......... .......... .......... 79% 62,0M 0s\n",
      " 46600K .......... .......... .......... .......... .......... 79% 56,6M 0s\n",
      " 46650K .......... .......... .......... .......... .......... 80%  121M 0s\n",
      " 46700K .......... .......... .......... .......... .......... 80% 99,4M 0s\n",
      " 46750K .......... .......... .......... .......... .......... 80%  124M 0s\n",
      " 46800K .......... .......... .......... .......... .......... 80%  120M 0s\n",
      " 46850K .......... .......... .......... .......... .......... 80% 15,2M 0s\n",
      " 46900K .......... .......... .......... .......... .......... 80% 49,2M 0s\n",
      " 46950K .......... .......... .......... .......... .......... 80% 58,7M 0s\n",
      " 47000K .......... .......... .......... .......... .......... 80%  124M 0s\n",
      " 47050K .......... .......... .......... .......... .......... 80%  116M 0s\n",
      " 47100K .......... .......... .......... .......... .......... 80%  102M 0s\n",
      " 47150K .......... .......... .......... .......... .......... 80%  121M 0s\n",
      " 47200K .......... .......... .......... .......... .......... 80%  101M 0s\n",
      " 47250K .......... .......... .......... .......... .......... 81%  122M 0s\n",
      " 47300K .......... .......... .......... .......... .......... 81%  120M 0s\n",
      " 47350K .......... .......... .......... .......... .......... 81%  101M 0s\n",
      " 47400K .......... .......... .......... .......... .......... 81%  122M 0s\n",
      " 47450K .......... .......... .......... .......... .......... 81%  101M 0s\n",
      " 47500K .......... .......... .......... .......... .......... 81%  115M 0s\n",
      " 47550K .......... .......... .......... .......... .......... 81%  127M 0s\n",
      " 47600K .......... .......... .......... .......... .......... 81%  102M 0s\n",
      " 47650K .......... .......... .......... .......... .......... 81%  121M 0s\n",
      " 47700K .......... .......... .......... .......... .......... 81%  102M 0s\n",
      " 47750K .......... .......... .......... .......... .......... 81%  117M 0s\n",
      " 47800K .......... .......... .......... .......... .......... 81%  121M 0s\n",
      " 47850K .......... .......... .......... .......... .......... 82%  104M 0s\n",
      " 47900K .......... .......... .......... .......... .......... 82%  117M 0s\n",
      " 47950K .......... .......... .......... .......... .......... 82%  102M 0s\n",
      " 48000K .......... .......... .......... .......... .......... 82%  121M 0s\n",
      " 48050K .......... .......... .......... .......... .......... 82%  117M 0s\n",
      " 48100K .......... .......... .......... .......... .......... 82% 91,4M 0s\n",
      " 48150K .......... .......... .......... .......... .......... 82%  140M 0s\n",
      " 48200K .......... .......... .......... .......... .......... 82%  104M 0s\n",
      " 48250K .......... .......... .......... .......... .......... 82%  117M 0s\n",
      " 48300K .......... .......... .......... .......... .......... 82%  118M 0s\n",
      " 48350K .......... .......... .......... .......... .......... 82%  103M 0s\n",
      " 48400K .......... .......... .......... .......... .......... 83%  120M 0s\n",
      " 48450K .......... .......... .......... .......... .......... 83% 8,99M 0s\n",
      " 48500K .......... .......... .......... .......... .......... 83% 8,97M 0s\n",
      " 48550K .......... .......... .......... .......... .......... 83% 31,9M 0s\n",
      " 48600K .......... .......... .......... .......... .......... 83% 98,8M 0s\n",
      " 48650K .......... .......... .......... .......... .......... 83%  121M 0s\n",
      " 48700K .......... .......... .......... .......... .......... 83%  102M 0s\n",
      " 48750K .......... .......... .......... .......... .......... 83%  121M 0s\n",
      " 48800K .......... .......... .......... .......... .......... 83%  121M 0s\n",
      " 48850K .......... .......... .......... .......... .......... 83%  102M 0s\n",
      " 48900K .......... .......... .......... .......... .......... 83%  117M 0s\n",
      " 48950K .......... .......... .......... .......... .......... 83%  104M 0s\n",
      " 49000K .......... .......... .......... .......... .......... 84%  119M 0s\n",
      " 49050K .......... .......... .......... .......... .......... 84%  114M 0s\n",
      " 49100K .......... .......... .......... .......... .......... 84%  104M 0s\n",
      " 49150K .......... .......... .......... .......... .......... 84%  124M 0s\n",
      " 49200K .......... .......... .......... .......... .......... 84%  101M 0s\n",
      " 49250K .......... .......... .......... .......... .......... 84%  121M 0s\n",
      " 49300K .......... .......... .......... .......... .......... 84%  117M 0s\n",
      " 49350K .......... .......... .......... .......... .......... 84% 20,6M 0s\n",
      " 49400K .......... .......... .......... .......... .......... 84%  128M 0s\n",
      " 49450K .......... .......... .......... .......... .......... 84% 4,00M 0s\n",
      " 49500K .......... .......... .......... .......... .......... 84% 36,2M 0s\n",
      " 49550K .......... .......... .......... .......... .......... 84% 26,5M 0s\n",
      " 49600K .......... .......... .......... .......... .......... 85% 53,0M 0s\n",
      " 49650K .......... .......... .......... .......... .......... 85%  121M 0s\n",
      " 49700K .......... .......... .......... .......... .......... 85%  102M 0s\n",
      " 49750K .......... .......... .......... .......... .......... 85%  121M 0s\n",
      " 49800K .......... .......... .......... .......... .......... 85%  121M 0s\n",
      " 49850K .......... .......... .......... .......... .......... 85%  103M 0s\n",
      " 49900K .......... .......... .......... .......... .......... 85% 37,5M 0s\n",
      " 49950K .......... .......... .......... .......... .......... 85% 16,6M 0s\n",
      " 50000K .......... .......... .......... .......... .......... 85% 65,1M 0s\n",
      " 50050K .......... .......... .......... .......... .......... 85%  116M 0s\n",
      " 50100K .......... .......... .......... .......... .......... 85%  103M 0s\n",
      " 50150K .......... .......... .......... .......... .......... 86% 95,2M 0s\n",
      " 50200K .......... .......... .......... .......... .......... 86%  134M 0s\n",
      " 50250K .......... .......... .......... .......... .......... 86%  117M 0s\n",
      " 50300K .......... .......... .......... .......... .......... 86%  121M 0s\n",
      " 50350K .......... .......... .......... .......... .......... 86%  102M 0s\n",
      " 50400K .......... .......... .......... .......... .......... 86%  121M 0s\n",
      " 50450K .......... .......... .......... .......... .......... 86%  103M 0s\n",
      " 50500K .......... .......... .......... .......... .......... 86%  115M 0s\n",
      " 50550K .......... .......... .......... .......... .......... 86%  123M 0s\n",
      " 50600K .......... .......... .......... .......... .......... 86%  102M 0s\n",
      " 50650K .......... .......... .......... .......... .......... 86%  117M 0s\n",
      " 50700K .......... .......... .......... .......... .......... 86%  102M 0s\n",
      " 50750K .......... .......... .......... .......... .......... 87%  121M 0s\n",
      " 50800K .......... .......... .......... .......... .......... 87%  117M 0s\n",
      " 50850K .......... .......... .......... .......... .......... 87%  102M 0s\n",
      " 50900K .......... .......... .......... .......... .......... 87%  121M 0s\n",
      " 50950K .......... .......... .......... .......... .......... 87%  101M 0s\n",
      " 51000K .......... .......... .......... .......... .......... 87%  124M 0s\n",
      " 51050K .......... .......... .......... .......... .......... 87%  118M 0s\n",
      " 51100K .......... .......... .......... .......... .......... 87%  102M 0s\n",
      " 51150K .......... .......... .......... .......... .......... 87%  120M 0s\n",
      " 51200K .......... .......... .......... .......... .......... 87%  103M 0s\n",
      " 51250K .......... .......... .......... .......... .......... 87%  118M 0s\n",
      " 51300K .......... .......... .......... .......... .......... 87%  120M 0s\n",
      " 51350K .......... .......... .......... .......... .......... 88%  103M 0s\n",
      " 51400K .......... .......... .......... .......... .......... 88%  120M 0s\n",
      " 51450K .......... .......... .......... .......... .......... 88% 25,4M 0s\n",
      " 51500K .......... .......... .......... .......... .......... 88% 9,02M 0s\n",
      " 51550K .......... .......... .......... .......... .......... 88%  126M 0s\n",
      " 51600K .......... .......... .......... .......... .......... 88% 10,5M 0s\n",
      " 51650K .......... .......... .......... .......... .......... 88%  119M 0s\n",
      " 51700K .......... .......... .......... .......... .......... 88% 62,4M 0s\n",
      " 51750K .......... .......... .......... .......... .......... 88% 58,9M 0s\n",
      " 51800K .......... .......... .......... .......... .......... 88%  122M 0s\n",
      " 51850K .......... .......... .......... .......... .......... 88%  102M 0s\n",
      " 51900K .......... .......... .......... .......... .......... 89%  121M 0s\n",
      " 51950K .......... .......... .......... .......... .......... 89%  102M 0s\n",
      " 52000K .......... .......... .......... .......... .......... 89%  121M 0s\n",
      " 52050K .......... .......... .......... .......... .......... 89%  117M 0s\n",
      " 52100K .......... .......... .......... .......... .......... 89%  103M 0s\n",
      " 52150K .......... .......... .......... .......... .......... 89%  120M 0s\n",
      " 52200K .......... .......... .......... .......... .......... 89%  102M 0s\n",
      " 52250K .......... .......... .......... .......... .......... 89%  118M 0s\n",
      " 52300K .......... .......... .......... .......... .......... 89%  121M 0s\n",
      " 52350K .......... .......... .......... .......... .......... 89%  102M 0s\n",
      " 52400K .......... .......... .......... .......... .......... 89% 46,2M 0s\n",
      " 52450K .......... .......... .......... .......... .......... 89% 15,4M 0s\n",
      " 52500K .......... .......... .......... .......... .......... 90% 6,50M 0s\n",
      " 52550K .......... .......... .......... .......... .......... 90% 12,7M 0s\n",
      " 52600K .......... .......... .......... .......... .......... 90% 22,3M 0s\n",
      " 52650K .......... .......... .......... .......... .......... 90% 58,7M 0s\n",
      " 52700K .......... .......... .......... .......... .......... 90%  101M 0s\n",
      " 52750K .......... .......... .......... .......... .......... 90%  124M 0s\n",
      " 52800K .......... .......... .......... .......... .......... 90%  116M 0s\n",
      " 52850K .......... .......... .......... .......... .......... 90% 77,7M 0s\n",
      " 52900K .......... .......... .......... .......... .......... 90%  120M 0s\n",
      " 52950K .......... .......... .......... .......... .......... 90% 32,0M 0s\n",
      " 53000K .......... .......... .......... .......... .......... 90% 25,1M 0s\n",
      " 53050K .......... .......... .......... .......... .......... 90% 41,6M 0s\n",
      " 53100K .......... .......... .......... .......... .......... 91% 53,4M 0s\n",
      " 53150K .......... .......... .......... .......... .......... 91%  121M 0s\n",
      " 53200K .......... .......... .......... .......... .......... 91%  102M 0s\n",
      " 53250K .......... .......... .......... .......... .......... 91%  118M 0s\n",
      " 53300K .......... .......... .......... .......... .......... 91%  121M 0s\n",
      " 53350K .......... .......... .......... .......... .......... 91%  102M 0s\n",
      " 53400K .......... .......... .......... .......... .......... 91%  118M 0s\n",
      " 53450K .......... .......... .......... .......... .......... 91%  102M 0s\n",
      " 53500K .......... .......... .......... .......... .......... 91%  121M 0s\n",
      " 53550K .......... .......... .......... .......... .......... 91%  121M 0s\n",
      " 53600K .......... .......... .......... .......... .......... 91%  102M 0s\n",
      " 53650K .......... .......... .......... .......... .......... 92%  117M 0s\n",
      " 53700K .......... .......... .......... .......... .......... 92%  103M 0s\n",
      " 53750K .......... .......... .......... .......... .......... 92%  120M 0s\n",
      " 53800K .......... .......... .......... .......... .......... 92%  115M 0s\n",
      " 53850K .......... .......... .......... .......... .......... 92%  104M 0s\n",
      " 53900K .......... .......... .......... .......... .......... 92%  121M 0s\n",
      " 53950K .......... .......... .......... .......... .......... 92%  102M 0s\n",
      " 54000K .......... .......... .......... .......... .......... 92%  118M 0s\n",
      " 54050K .......... .......... .......... .......... .......... 92%  122M 0s\n",
      " 54100K .......... .......... .......... .......... .......... 92%  102M 0s\n",
      " 54150K .......... .......... .......... .......... .......... 92%  122M 0s\n",
      " 54200K .......... .......... .......... .......... .......... 92%  102M 0s\n",
      " 54250K .......... .......... .......... .......... .......... 93%  117M 0s\n",
      " 54300K .......... .......... .......... .......... .......... 93%  122M 0s\n",
      " 54350K .......... .......... .......... .......... .......... 93%  102M 0s\n",
      " 54400K .......... .......... .......... .......... .......... 93%  117M 0s\n",
      " 54450K .......... .......... .......... .......... .......... 93%  102M 0s\n",
      " 54500K .......... .......... .......... .......... .......... 93% 43,6M 0s\n",
      " 54550K .......... .......... .......... .......... .......... 93% 13,6M 0s\n",
      " 54600K .......... .......... .......... .......... .......... 93% 9,99M 0s\n",
      " 54650K .......... .......... .......... .......... .......... 93% 21,5M 0s\n",
      " 54700K .......... .......... .......... .......... .......... 93%  102M 0s\n",
      " 54750K .......... .......... .......... .......... .......... 93% 78,5M 0s\n",
      " 54800K .......... .......... .......... .......... .......... 93%  116M 0s\n",
      " 54850K .......... .......... .......... .......... .......... 94% 92,4M 0s\n",
      " 54900K .......... .......... .......... .......... .......... 94% 69,5M 0s\n",
      " 54950K .......... .......... .......... .......... .......... 94% 35,1M 0s\n",
      " 55000K .......... .......... .......... .......... .......... 94%  116M 0s\n",
      " 55050K .......... .......... .......... .......... .......... 94%  121M 0s\n",
      " 55100K .......... .......... .......... .......... .......... 94%  103M 0s\n",
      " 55150K .......... .......... .......... .......... .......... 94%  120M 0s\n",
      " 55200K .......... .......... .......... .......... .......... 94%  102M 0s\n",
      " 55250K .......... .......... .......... .......... .......... 94%  117M 0s\n",
      " 55300K .......... .......... .......... .......... .......... 94%  121M 0s\n",
      " 55350K .......... .......... .......... .......... .......... 94%  102M 0s\n",
      " 55400K .......... .......... .......... .......... .......... 95%  119M 0s\n",
      " 55450K .......... .......... .......... .......... .......... 95%  102M 0s\n",
      " 55500K .......... .......... .......... .......... .......... 95% 21,0M 0s\n",
      " 55550K .......... .......... .......... .......... .......... 95% 11,1M 0s\n",
      " 55600K .......... .......... .......... .......... .......... 95% 5,80M 0s\n",
      " 55650K .......... .......... .......... .......... .......... 95% 24,8M 0s\n",
      " 55700K .......... .......... .......... .......... .......... 95%  102M 0s\n",
      " 55750K .......... .......... .......... .......... .......... 95% 75,8M 0s\n",
      " 55800K .......... .......... .......... .......... .......... 95%  118M 0s\n",
      " 55850K .......... .......... .......... .......... .......... 95%  102M 0s\n",
      " 55900K .......... .......... .......... .......... .......... 95%  122M 0s\n",
      " 55950K .......... .......... .......... .......... .......... 95%  102M 0s\n",
      " 56000K .......... .......... .......... .......... .......... 96% 46,5M 0s\n",
      " 56050K .......... .......... .......... .......... .......... 96% 22,2M 0s\n",
      " 56100K .......... .......... .......... .......... .......... 96% 27,2M 0s\n",
      " 56150K .......... .......... .......... .......... .......... 96%  117M 0s\n",
      " 56200K .......... .......... .......... .......... .......... 96%  102M 0s\n",
      " 56250K .......... .......... .......... .......... .......... 96%  121M 0s\n",
      " 56300K .......... .......... .......... .......... .......... 96% 94,3M 0s\n",
      " 56350K .......... .......... .......... .......... .......... 96%  135M 0s\n",
      " 56400K .......... .......... .......... .......... .......... 96%  117M 0s\n",
      " 56450K .......... .......... .......... .......... .......... 96%  102M 0s\n",
      " 56500K .......... .......... .......... .......... .......... 96%  122M 0s\n",
      " 56550K .......... .......... .......... .......... .......... 96%  117M 0s\n",
      " 56600K .......... .......... .......... .......... .......... 97%  102M 0s\n",
      " 56650K .......... .......... .......... .......... .......... 97%  123M 0s\n",
      " 56700K .......... .......... .......... .......... .......... 97%  102M 0s\n",
      " 56750K .......... .......... .......... .......... .......... 97%  122M 0s\n",
      " 56800K .......... .......... .......... .......... .......... 97%  116M 0s\n",
      " 56850K .......... .......... .......... .......... .......... 97%  102M 0s\n",
      " 56900K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 56950K .......... .......... .......... .......... .......... 97%  106M 0s\n",
      " 57000K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 57050K .......... .......... .......... .......... .......... 97%  121M 0s\n",
      " 57100K .......... .......... .......... .......... .......... 97%  102M 0s\n",
      " 57150K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 57200K .......... .......... .......... .......... .......... 98%  103M 0s\n",
      " 57250K .......... .......... .......... .......... .......... 98%  120M 0s\n",
      " 57300K .......... .......... .......... .......... .......... 98%  122M 0s\n",
      " 57350K .......... .......... .......... .......... .......... 98%  102M 0s\n",
      " 57400K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 57450K .......... .......... .......... .......... .......... 98%  102M 0s\n",
      " 57500K .......... .......... .......... .......... .......... 98%  121M 0s\n",
      " 57550K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 57600K .......... .......... .......... .......... .......... 98% 11,7M 0s\n",
      " 57650K .......... .......... .......... .......... .......... 98% 12,3M 0s\n",
      " 57700K .......... .......... .......... .......... .......... 98% 19,1M 0s\n",
      " 57750K .......... .......... .......... .......... .......... 99% 31,8M 0s\n",
      " 57800K .......... .......... .......... .......... .......... 99%  122M 0s\n",
      " 57850K .......... .......... .......... .......... .......... 99% 55,1M 0s\n",
      " 57900K .......... .......... .......... .......... .......... 99%  120M 0s\n",
      " 57950K .......... .......... .......... .......... .......... 99%  103M 0s\n",
      " 58000K .......... .......... .......... .......... .......... 99%  117M 0s\n",
      " 58050K .......... .......... .......... .......... .......... 99%  120M 0s\n",
      " 58100K .......... .......... .......... .......... .......... 99% 44,9M 0s\n",
      " 58150K .......... .......... .......... .......... .......... 99% 44,4M 0s\n",
      " 58200K .......... .......... .......... .......... .......... 99%  103M 0s\n",
      " 58250K .......... .......... .......... .......... .......... 99%  120M 0s\n",
      " 58300K .......... .......... .......... .......... .......... 99%  117M 0s\n",
      " 58350K .......... .                                          100%  140M=1,6s\n",
      "\n",
      "2023-08-28 16:47:11 (35,3 MB/s) - ‘data/reference/genome.fa.gz’ guardado [59761837/59761837]\n",
      "\n",
      "[Mon Aug 28 16:47:12 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T164708.142578.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Creating conda environment code/environments/Greference_tools.yml...\n",
      "Downloading and installing remote packages.\n",
      "Environment for /home/juancarlos/Escritorio/TFM/code/environments/Greference_tools.yml created (location: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_)\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:48:04 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR696683_chr3_1.fastq.gz, data/raw/ERR696683_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR696683_chr3_1_fastqc.html, results/fastqc_result/ERR696683_chr3_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR696683_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 16:48:23 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:48:23 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR696683_chr3_1.fastq.gz, data/raw/ERR696683_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR696683_chr3_1_fastp.fastq.gz, data/processed/ERR696683_chr3_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR696683_chr3_2_fastp.fastq.gz, data/processed/ERR696683_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2070566\n",
      "total bases: 156530982\n",
      "Q20 bases: 155429955(99.2966%)\n",
      "Q30 bases: 151203846(96.5968%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2070566\n",
      "total bases: 156128924\n",
      "Q20 bases: 154231757(98.7849%)\n",
      "Q30 bases: 148068194(94.8371%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1294164\n",
      "total bases: 98291522\n",
      "Q20 bases: 97907734(99.6095%)\n",
      "Q30 bases: 96216656(97.8891%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1294164\n",
      "total bases: 98253340\n",
      "Q20 bases: 97583822(99.3186%)\n",
      "Q30 bases: 95093850(96.7843%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2588328\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1552804\n",
      "reads with adapter trimmed: 2016\n",
      "bases trimmed due to adapters: 53376\n",
      "\n",
      "Duplication rate: 5.84944%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR696683_chr3_1.fastq.gz -I data/raw/ERR696683_chr3_2.fastq.gz -o data/processed/ERR696683_chr3_1_fastp.fastq.gz -O data/processed/ERR696683_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 7 seconds\n",
      "[Mon Aug 28 16:48:30 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:48:30 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR696683_chr3_1_fastp.fastq.gz, data/processed/ERR696683_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR696683_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR696683_chr3_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR696683_chr3_2_fastp.fastq.gz, data/processed/ERR696683_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 16:48:41 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T164712.357193.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:48:42 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753368_chr3_1.fastq.gz, data/raw/ERR753368_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753368_chr3_1_fastp.fastq.gz, data/processed/ERR753368_chr3_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753368_chr3_2_fastp.fastq.gz, data/processed/ERR753368_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2166924\n",
      "total bases: 164537592\n",
      "Q20 bases: 163892626(99.608%)\n",
      "Q30 bases: 159125497(96.7107%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2166924\n",
      "total bases: 164375778\n",
      "Q20 bases: 162943687(99.1288%)\n",
      "Q30 bases: 155573837(94.6452%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1870811\n",
      "total bases: 142174357\n",
      "Q20 bases: 141930007(99.8281%)\n",
      "Q30 bases: 139573591(98.1707%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1870811\n",
      "total bases: 142162640\n",
      "Q20 bases: 141472380(99.5145%)\n",
      "Q30 bases: 137298960(96.5788%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3741622\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 592226\n",
      "reads with adapter trimmed: 1044\n",
      "bases trimmed due to adapters: 26850\n",
      "\n",
      "Duplication rate: 2.05834%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753368_chr3_1.fastq.gz -I data/raw/ERR753368_chr3_2.fastq.gz -o data/processed/ERR753368_chr3_1_fastp.fastq.gz -O data/processed/ERR753368_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 7 seconds\n",
      "[Mon Aug 28 16:48:50 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:48:50 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753368_chr3_1_fastp.fastq.gz, data/processed/ERR753368_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753368_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753368_chr3_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753368_chr3_2_fastp.fastq.gz, data/processed/ERR753368_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 16:49:05 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:49:05 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753368_chr3_1.fastq.gz, data/raw/ERR753368_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753368_chr3_1_fastqc.html, results/fastqc_result/ERR753368_chr3_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753368_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753368_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 16:49:22 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T164841.313387.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:49:23 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753369_chr3_1.fastq.gz, data/raw/ERR753369_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753369_chr3_1_fastqc.html, results/fastqc_result/ERR753369_chr3_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753369_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753369_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 16:49:42 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:49:42 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753369_chr3_1.fastq.gz, data/raw/ERR753369_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753369_chr3_1_fastp.fastq.gz, data/processed/ERR753369_chr3_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753369_chr3_2_fastp.fastq.gz, data/processed/ERR753369_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2321627\n",
      "total bases: 176276293\n",
      "Q20 bases: 175555474(99.5911%)\n",
      "Q30 bases: 170311472(96.6162%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2321627\n",
      "total bases: 176111158\n",
      "Q20 bases: 174567395(99.1234%)\n",
      "Q30 bases: 166614064(94.6073%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 2002164\n",
      "total bases: 152156256\n",
      "Q20 bases: 151876395(99.8161%)\n",
      "Q30 bases: 149254086(98.0926%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 2002164\n",
      "total bases: 152144233\n",
      "Q20 bases: 151399863(99.5107%)\n",
      "Q30 bases: 146897284(96.5513%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 4004328\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 638926\n",
      "reads with adapter trimmed: 1028\n",
      "bases trimmed due to adapters: 24939\n",
      "\n",
      "Duplication rate: 2.38365%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753369_chr3_1.fastq.gz -I data/raw/ERR753369_chr3_2.fastq.gz -o data/processed/ERR753369_chr3_1_fastp.fastq.gz -O data/processed/ERR753369_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 8 seconds\n",
      "[Mon Aug 28 16:49:50 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:49:50 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753369_chr3_1_fastp.fastq.gz, data/processed/ERR753369_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753369_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753369_chr3_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753369_chr3_2_fastp.fastq.gz, data/processed/ERR753369_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 16:50:06 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T164922.299360.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:50:07 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753370_chr3_1.fastq.gz, data/raw/ERR753370_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753370_chr3_1_fastqc.html, results/fastqc_result/ERR753370_chr3_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753370_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753370_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 16:50:34 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:50:34 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753370_chr3_1.fastq.gz, data/raw/ERR753370_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753370_chr3_1_fastp.fastq.gz, data/processed/ERR753370_chr3_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753370_chr3_2_fastp.fastq.gz, data/processed/ERR753370_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2951523\n",
      "total bases: 297130164\n",
      "Q20 bases: 295118642(99.323%)\n",
      "Q30 bases: 286675921(96.4816%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2951523\n",
      "total bases: 297056363\n",
      "Q20 bases: 294260921(99.059%)\n",
      "Q30 bases: 283246706(95.3512%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 2950493\n",
      "total bases: 295883003\n",
      "Q20 bases: 293987676(99.3594%)\n",
      "Q30 bases: 285943525(96.6407%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 2950493\n",
      "total bases: 294939988\n",
      "Q20 bases: 292398511(99.1383%)\n",
      "Q30 bases: 282073643(95.6376%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 5900986\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 2060\n",
      "reads with adapter trimmed: 17930\n",
      "bases trimmed due to adapters: 173930\n",
      "\n",
      "Duplication rate: 4.38811%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 160\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753370_chr3_1.fastq.gz -I data/raw/ERR753370_chr3_2.fastq.gz -o data/processed/ERR753370_chr3_1_fastp.fastq.gz -O data/processed/ERR753370_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 13 seconds\n",
      "[Mon Aug 28 16:50:48 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:50:49 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753370_chr3_1_fastp.fastq.gz, data/processed/ERR753370_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753370_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753370_chr3_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753370_chr3_2_fastp.fastq.gz, data/processed/ERR753370_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 16:51:17 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T165006.673625.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:51:19 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753371_chr3_1.fastq.gz, data/raw/ERR753371_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753371_chr3_1_fastp.fastq.gz, data/processed/ERR753371_chr3_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753371_chr3_2_fastp.fastq.gz, data/processed/ERR753371_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1938056\n",
      "total bases: 146044753\n",
      "Q20 bases: 144732927(99.1018%)\n",
      "Q30 bases: 139492968(95.5139%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1938056\n",
      "total bases: 145676659\n",
      "Q20 bases: 143948860(98.8139%)\n",
      "Q30 bases: 138931875(95.37%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1007584\n",
      "total bases: 76483588\n",
      "Q20 bases: 76112999(99.5155%)\n",
      "Q30 bases: 74225186(97.0472%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1007584\n",
      "total bases: 76440869\n",
      "Q20 bases: 75910140(99.3057%)\n",
      "Q30 bases: 74249432(97.1332%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2015168\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1860944\n",
      "reads with adapter trimmed: 1798\n",
      "bases trimmed due to adapters: 42747\n",
      "\n",
      "Duplication rate: 22.2704%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753371_chr3_1.fastq.gz -I data/raw/ERR753371_chr3_2.fastq.gz -o data/processed/ERR753371_chr3_1_fastp.fastq.gz -O data/processed/ERR753371_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Mon Aug 28 16:51:25 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:51:25 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753371_chr3_1_fastp.fastq.gz, data/processed/ERR753371_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753371_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753371_chr3_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753371_chr3_2_fastp.fastq.gz, data/processed/ERR753371_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 16:51:34 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:51:34 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753371_chr3_1.fastq.gz, data/raw/ERR753371_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753371_chr3_1_fastqc.html, results/fastqc_result/ERR753371_chr3_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753371_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753371_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 16:51:51 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T165117.852212.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:51:52 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753372_chr3_1.fastq.gz, data/raw/ERR753372_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753372_chr3_1_fastqc.html, results/fastqc_result/ERR753372_chr3_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753372_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753372_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 16:52:11 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:52:11 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753372_chr3_1.fastq.gz, data/raw/ERR753372_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753372_chr3_1_fastp.fastq.gz, data/processed/ERR753372_chr3_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753372_chr3_2_fastp.fastq.gz, data/processed/ERR753372_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2299130\n",
      "total bases: 174570404\n",
      "Q20 bases: 173883945(99.6068%)\n",
      "Q30 bases: 168720211(96.6488%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2299130\n",
      "total bases: 174401358\n",
      "Q20 bases: 172878044(99.1265%)\n",
      "Q30 bases: 165079194(94.6548%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1978429\n",
      "total bases: 150352788\n",
      "Q20 bases: 150093042(99.8272%)\n",
      "Q30 bases: 147551682(98.137%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1978429\n",
      "total bases: 150340351\n",
      "Q20 bases: 149614171(99.517%)\n",
      "Q30 bases: 145248530(96.6131%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3956858\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 641402\n",
      "reads with adapter trimmed: 1252\n",
      "bases trimmed due to adapters: 32637\n",
      "\n",
      "Duplication rate: 1.99618%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753372_chr3_1.fastq.gz -I data/raw/ERR753372_chr3_2.fastq.gz -o data/processed/ERR753372_chr3_1_fastp.fastq.gz -O data/processed/ERR753372_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 7 seconds\n",
      "[Mon Aug 28 16:52:19 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:52:19 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753372_chr3_1_fastp.fastq.gz, data/processed/ERR753372_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753372_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753372_chr3_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753372_chr3_2_fastp.fastq.gz, data/processed/ERR753372_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 16:52:36 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T165151.660424.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:52:38 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753373_chr3_1.fastq.gz, data/raw/ERR753373_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753373_chr3_1_fastqc.html, results/fastqc_result/ERR753373_chr3_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753373_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753373_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 16:52:57 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:52:57 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753373_chr3_1.fastq.gz, data/raw/ERR753373_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753373_chr3_1_fastp.fastq.gz, data/processed/ERR753373_chr3_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753373_chr3_2_fastp.fastq.gz, data/processed/ERR753373_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1968465\n",
      "total bases: 188482891\n",
      "Q20 bases: 186996377(99.2113%)\n",
      "Q30 bases: 174397046(92.5267%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1968465\n",
      "total bases: 184512055\n",
      "Q20 bases: 182525877(98.9236%)\n",
      "Q30 bases: 170849210(92.5951%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1922198\n",
      "total bases: 181199250\n",
      "Q20 bases: 180156389(99.4245%)\n",
      "Q30 bases: 169961990(93.7984%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1922198\n",
      "total bases: 175114301\n",
      "Q20 bases: 174290716(99.5297%)\n",
      "Q30 bases: 165407202(94.4567%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3844396\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 92534\n",
      "reads with adapter trimmed: 1350\n",
      "bases trimmed due to adapters: 40609\n",
      "\n",
      "Duplication rate: 1.91392%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 153\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753373_chr3_1.fastq.gz -I data/raw/ERR753373_chr3_2.fastq.gz -o data/processed/ERR753373_chr3_1_fastp.fastq.gz -O data/processed/ERR753373_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 9 seconds\n",
      "[Mon Aug 28 16:53:06 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:53:06 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753373_chr3_1_fastp.fastq.gz, data/processed/ERR753373_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753373_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753373_chr3_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753373_chr3_2_fastp.fastq.gz, data/processed/ERR753373_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 16:53:22 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T165236.879958.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:53:24 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753374_chr3_1.fastq.gz, data/raw/ERR753374_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753374_chr3_1_fastqc.html, results/fastqc_result/ERR753374_chr3_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753374_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753374_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 16:53:37 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:53:37 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753374_chr3_1.fastq.gz, data/raw/ERR753374_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753374_chr3_1_fastp.fastq.gz, data/processed/ERR753374_chr3_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753374_chr3_2_fastp.fastq.gz, data/processed/ERR753374_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1162403\n",
      "total bases: 111198121\n",
      "Q20 bases: 110160620(99.067%)\n",
      "Q30 bases: 100749607(90.6037%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1162403\n",
      "total bases: 108805108\n",
      "Q20 bases: 107679458(98.9654%)\n",
      "Q30 bases: 99151878(91.128%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1126834\n",
      "total bases: 105363400\n",
      "Q20 bases: 104622908(99.2972%)\n",
      "Q30 bases: 97132511(92.1881%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1126834\n",
      "total bases: 102618294\n",
      "Q20 bases: 102036510(99.4331%)\n",
      "Q30 bases: 95498534(93.0619%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2253668\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 71138\n",
      "reads with adapter trimmed: 676\n",
      "bases trimmed due to adapters: 22682\n",
      "\n",
      "Duplication rate: 5.7142%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 152\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753374_chr3_1.fastq.gz -I data/raw/ERR753374_chr3_2.fastq.gz -o data/processed/ERR753374_chr3_1_fastp.fastq.gz -O data/processed/ERR753374_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Mon Aug 28 16:53:42 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:53:42 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753374_chr3_1_fastp.fastq.gz, data/processed/ERR753374_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753374_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753374_chr3_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753374_chr3_2_fastp.fastq.gz, data/processed/ERR753374_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 16:53:53 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T165323.125549.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:53:54 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753375_chr3_1.fastq.gz, data/raw/ERR753375_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753375_chr3_1_fastp.fastq.gz, data/processed/ERR753375_chr3_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753375_chr3_1_fastp.fastq.gz, data/processed/ERR753375_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1282179\n",
      "total bases: 96858250\n",
      "Q20 bases: 96265562(99.3881%)\n",
      "Q30 bases: 94113609(97.1663%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1282179\n",
      "total bases: 96835483\n",
      "Q20 bases: 95589371(98.7132%)\n",
      "Q30 bases: 93289559(96.3382%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 841057\n",
      "total bases: 63862131\n",
      "Q20 bases: 63654927(99.6755%)\n",
      "Q30 bases: 62708243(98.1932%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 841057\n",
      "total bases: 63872391\n",
      "Q20 bases: 63259861(99.041%)\n",
      "Q30 bases: 62279464(97.5061%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1682114\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 882244\n",
      "reads with adapter trimmed: 996\n",
      "bases trimmed due to adapters: 27146\n",
      "\n",
      "Duplication rate: 1.71754%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753375_chr3_1.fastq.gz -I data/raw/ERR753375_chr3_2.fastq.gz -o data/processed/ERR753375_chr3_1_fastp.fastq.gz -O data/processed/ERR753375_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Mon Aug 28 16:53:59 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:53:59 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753375_chr3_1.fastq.gz, data/raw/ERR753375_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753375_chr3_1_fastqc.html, results/fastqc_result/ERR753375_chr3_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753375_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753375_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 16:54:10 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:54:10 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753375_chr3_1_fastp.fastq.gz, data/processed/ERR753375_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753375_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753375_chr3_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753375_chr3_1_fastp.fastq.gz, data/processed/ERR753375_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 16:54:19 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T165353.824829.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:54:20 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753376_chr3_1.fastq.gz, data/raw/ERR753376_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753376_chr3_1_fastqc.html, results/fastqc_result/ERR753376_chr3_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753376_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753376_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 16:54:35 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:54:35 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753376_chr3_1.fastq.gz, data/raw/ERR753376_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753376_chr3_1_fastp.fastq.gz, data/processed/ERR753376_chr3_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753376_chr3_1_fastp.fastq.gz, data/processed/ERR753376_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1665106\n",
      "total bases: 125882357\n",
      "Q20 bases: 125020665(99.3155%)\n",
      "Q30 bases: 121633177(96.6245%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1665106\n",
      "total bases: 125542593\n",
      "Q20 bases: 124005519(98.7757%)\n",
      "Q30 bases: 119018691(94.8034%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1037289\n",
      "total bases: 78781993\n",
      "Q20 bases: 78486134(99.6245%)\n",
      "Q30 bases: 77145479(97.9227%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1037289\n",
      "total bases: 78752401\n",
      "Q20 bases: 78213072(99.3152%)\n",
      "Q30 bases: 76205525(96.766%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2074578\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1255634\n",
      "reads with adapter trimmed: 1176\n",
      "bases trimmed due to adapters: 29546\n",
      "\n",
      "Duplication rate: 9.11624%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753376_chr3_1.fastq.gz -I data/raw/ERR753376_chr3_2.fastq.gz -o data/processed/ERR753376_chr3_1_fastp.fastq.gz -O data/processed/ERR753376_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Mon Aug 28 16:54:41 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:54:41 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753376_chr3_1_fastp.fastq.gz, data/processed/ERR753376_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753376_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753376_chr3_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753376_chr3_1_fastp.fastq.gz, data/processed/ERR753376_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 16:54:51 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T165419.784007.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:54:53 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753377_chr3_1.fastq.gz, data/raw/ERR753377_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753377_chr3_1_fastp.fastq.gz, data/processed/ERR753377_chr3_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753377_chr3_2_fastp.fastq.gz, data/processed/ERR753377_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1728950\n",
      "total bases: 130722103\n",
      "Q20 bases: 129808577(99.3012%)\n",
      "Q30 bases: 126267707(96.5925%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1728950\n",
      "total bases: 130414142\n",
      "Q20 bases: 128895041(98.8352%)\n",
      "Q30 bases: 123933005(95.0303%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1099156\n",
      "total bases: 83481297\n",
      "Q20 bases: 83162064(99.6176%)\n",
      "Q30 bases: 81728416(97.9003%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1099156\n",
      "total bases: 83450542\n",
      "Q20 bases: 82907906(99.3498%)\n",
      "Q30 bases: 80876920(96.916%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2198312\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1259588\n",
      "reads with adapter trimmed: 2030\n",
      "bases trimmed due to adapters: 50654\n",
      "\n",
      "Duplication rate: 9.11006%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 119\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753377_chr3_1.fastq.gz -I data/raw/ERR753377_chr3_2.fastq.gz -o data/processed/ERR753377_chr3_1_fastp.fastq.gz -O data/processed/ERR753377_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 7 seconds\n",
      "[Mon Aug 28 16:55:00 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:55:00 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753377_chr3_1.fastq.gz, data/raw/ERR753377_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753377_chr3_1_fastqc.html, results/fastqc_result/ERR753377_chr3_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753377_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753377_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 16:55:15 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:55:15 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753377_chr3_1_fastp.fastq.gz, data/processed/ERR753377_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753377_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753377_chr3_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753377_chr3_2_fastp.fastq.gz, data/processed/ERR753377_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 16:55:26 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T165451.894331.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:55:27 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753378_chr3_1.fastq.gz, data/raw/ERR753378_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753378_chr3_1_fastqc.html, results/fastqc_result/ERR753378_chr3_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753378_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753378_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 16:55:44 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:55:44 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753378_chr3_1.fastq.gz, data/raw/ERR753378_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753378_chr3_1_fastp.fastq.gz, data/processed/ERR753378_chr3_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753378_chr3_2_fastp.fastq.gz, data/processed/ERR753378_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1978176\n",
      "total bases: 150201904\n",
      "Q20 bases: 149614211(99.6087%)\n",
      "Q30 bases: 145205101(96.6733%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1978176\n",
      "total bases: 150054321\n",
      "Q20 bases: 148742134(99.1255%)\n",
      "Q30 bases: 142005411(94.636%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1708379\n",
      "total bases: 129830167\n",
      "Q20 bases: 129604043(99.8258%)\n",
      "Q30 bases: 127403551(98.1309%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1708379\n",
      "total bases: 129819431\n",
      "Q20 bases: 129183589(99.5102%)\n",
      "Q30 bases: 125347189(96.555%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3416758\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 539594\n",
      "reads with adapter trimmed: 934\n",
      "bases trimmed due to adapters: 22437\n",
      "\n",
      "Duplication rate: 1.97423%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753378_chr3_1.fastq.gz -I data/raw/ERR753378_chr3_2.fastq.gz -o data/processed/ERR753378_chr3_1_fastp.fastq.gz -O data/processed/ERR753378_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 7 seconds\n",
      "[Mon Aug 28 16:55:52 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:55:52 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753378_chr3_1_fastp.fastq.gz, data/processed/ERR753378_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753378_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753378_chr3_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753378_chr3_2_fastp.fastq.gz, data/processed/ERR753378_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 16:56:07 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T165526.496555.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 16:56:08 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR696683_chr3_1_fastp.fastq.gz, data/processed/ERR696683_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR696683_chr3.sam, results/mapped_reads/ERR696683_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR696683_chr3_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR696683_chr3_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.88 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 83.21 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.62 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.43 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 46.62 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 139.348 sec; CPU: 131.753 sec\n",
      "[Mon Aug 28 17:00:21 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:00:21 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR696683_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr3.bam, results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR696683_chr3.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR696683_chr3_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 17:01:12 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:01:12 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR696683_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "17:01:13.384 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 17:01:13 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR696683_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR696683_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 17:01:13 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 17:01:13\tMarkDuplicates\tStart of doWork freeMemory: 528794864; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:01:13\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 17:01:13\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam\n",
      "INFO\t2023-08-28 17:01:15\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:69,122,310\n",
      "INFO\t2023-08-28 17:01:15\tMarkDuplicates\tTracking 804 as yet unmatched pairs. 804 records in RAM.\n",
      "INFO\t2023-08-28 17:01:16\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 3:149,750,768\n",
      "INFO\t2023-08-28 17:01:16\tMarkDuplicates\tTracking 674 as yet unmatched pairs. 674 records in RAM.\n",
      "INFO\t2023-08-28 17:01:17\tMarkDuplicates\tRead 2626393 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 17:01:17\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 847929616; totalMemory: 1199570944; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:01:17\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 17:01:17\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 17:01:18\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 17:01:18\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 17:01:18\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1190092760; totalMemory: 1736441856; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:01:18\tMarkDuplicates\tMarking 245190 records as duplicates.\n",
      "INFO\t2023-08-28 17:01:18\tMarkDuplicates\tFound 114 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam\n",
      "INFO\t2023-08-28 17:01:18\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 17:01:35\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 17:01:35\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 17:01:35\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 17:01:35\tMarkDuplicates\tBefore output close freeMemory: 528092304; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:01:35\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 17:01:35\tMarkDuplicates\tAfter output close freeMemory: 528092304; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 17:01:35 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.37 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 17:01:37 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T165607.206735.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:01:38 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753368_chr3_1_fastp.fastq.gz, data/processed/ERR753368_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753368_chr3.sam, results/mapped_reads/ERR753368_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753368_chr3_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753368_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.70 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 81.82 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.62 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.46 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 46.80 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 136.728 sec; CPU: 130.404 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 17:06:57 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:06:57 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753368_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr3.bam, results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753368_chr3.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753368_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 17:08:03 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:08:03 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753368_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753368_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "17:08:04.586 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 17:08:04 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753368_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753368_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 17:08:04 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 17:08:04\tMarkDuplicates\tStart of doWork freeMemory: 528786624; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:08:04\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 17:08:04\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam\n",
      "INFO\t2023-08-28 17:08:06\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:48,469,724\n",
      "INFO\t2023-08-28 17:08:06\tMarkDuplicates\tTracking 477 as yet unmatched pairs. 477 records in RAM.\n",
      "INFO\t2023-08-28 17:08:07\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 3:108,377,293\n",
      "INFO\t2023-08-28 17:08:07\tMarkDuplicates\tTracking 765 as yet unmatched pairs. 765 records in RAM.\n",
      "INFO\t2023-08-28 17:08:09\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 3:152,591,687\n",
      "INFO\t2023-08-28 17:08:09\tMarkDuplicates\tTracking 385 as yet unmatched pairs. 385 records in RAM.\n",
      "INFO\t2023-08-28 17:08:10\tMarkDuplicates\tRead 3764483 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 17:08:10\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 436740280; totalMemory: 913309696; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:08:10\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 17:08:10\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 17:08:11\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 17:08:11\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 17:08:11\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 903824208; totalMemory: 1450180608; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:08:11\tMarkDuplicates\tMarking 92315 records as duplicates.\n",
      "INFO\t2023-08-28 17:08:11\tMarkDuplicates\tFound 612 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam\n",
      "INFO\t2023-08-28 17:08:11\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 17:08:33\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 17:08:33\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 17:08:33\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 17:08:33\tMarkDuplicates\tBefore output close freeMemory: 528084928; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:08:33\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 17:08:33\tMarkDuplicates\tAfter output close freeMemory: 528084928; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 17:08:33 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.49 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 17:08:36 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T170137.639118.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:08:37 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753369_chr3_1_fastp.fastq.gz, data/processed/ERR753369_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753369_chr3.sam, results/mapped_reads/ERR753369_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753369_chr3_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753369_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.66 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 70.77 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.60 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.40 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 42.01 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 120.901 sec; CPU: 114.429 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 17:13:31 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:13:31 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753369_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr3.bam, results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753369_chr3.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753369_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 17:14:39 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:14:39 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753369_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753369_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "17:14:40.787 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 17:14:40 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753369_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753369_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 17:14:40 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 17:14:40\tMarkDuplicates\tStart of doWork freeMemory: 528788128; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:14:40\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 17:14:40\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam\n",
      "INFO\t2023-08-28 17:14:42\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:47,407,708\n",
      "INFO\t2023-08-28 17:14:42\tMarkDuplicates\tTracking 453 as yet unmatched pairs. 453 records in RAM.\n",
      "INFO\t2023-08-28 17:14:43\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 3:100,386,903\n",
      "INFO\t2023-08-28 17:14:43\tMarkDuplicates\tTracking 651 as yet unmatched pairs. 651 records in RAM.\n",
      "INFO\t2023-08-28 17:14:45\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 3:142,332,829\n",
      "INFO\t2023-08-28 17:14:45\tMarkDuplicates\tTracking 551 as yet unmatched pairs. 551 records in RAM.\n",
      "INFO\t2023-08-28 17:14:46\tMarkDuplicates\tRead     4,000,000 records.  Elapsed time: 00:00:05s.  Time for last 1,000,000:    1s.  Last read position: 3:197,557,443\n",
      "INFO\t2023-08-28 17:14:46\tMarkDuplicates\tTracking 17 as yet unmatched pairs. 17 records in RAM.\n",
      "INFO\t2023-08-28 17:14:46\tMarkDuplicates\tRead 4026673 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 17:14:46\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 349528552; totalMemory: 854589440; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:14:46\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 17:14:46\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 17:14:47\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 17:14:47\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 17:14:47\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 845105904; totalMemory: 1391460352; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:14:47\tMarkDuplicates\tMarking 115047 records as duplicates.\n",
      "INFO\t2023-08-28 17:14:47\tMarkDuplicates\tFound 621 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam\n",
      "INFO\t2023-08-28 17:14:47\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 17:15:11\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 17:15:11\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 17:15:11\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 17:15:11\tMarkDuplicates\tBefore output close freeMemory: 528086152; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:15:11\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 17:15:11\tMarkDuplicates\tAfter output close freeMemory: 528086152; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 17:15:11 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.51 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 17:15:13 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T170836.393862.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:15:15 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753370_chr3_1_fastp.fastq.gz, data/processed/ERR753370_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753370_chr3.sam, results/mapped_reads/ERR753370_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753370_chr3_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753370_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.68 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 78.88 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.59 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.41 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 45.95 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 133.047 sec; CPU: 126.503 sec\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 17:23:16 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:23:16 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753370_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr3.bam, results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753370_chr3.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753370_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 17:25:41 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:25:41 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753370_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753370_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "17:25:41.977 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 17:25:41 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753370_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753370_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 17:25:41 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 17:25:42\tMarkDuplicates\tStart of doWork freeMemory: 528781632; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:25:42\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 17:25:42\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam\n",
      "INFO\t2023-08-28 17:25:43\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:37,998,202\n",
      "INFO\t2023-08-28 17:25:43\tMarkDuplicates\tTracking 989 as yet unmatched pairs. 989 records in RAM.\n",
      "INFO\t2023-08-28 17:25:45\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 3:56,434,427\n",
      "INFO\t2023-08-28 17:25:45\tMarkDuplicates\tTracking 1785 as yet unmatched pairs. 1785 records in RAM.\n",
      "INFO\t2023-08-28 17:25:46\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 3:101,652,466\n",
      "INFO\t2023-08-28 17:25:46\tMarkDuplicates\tTracking 1971 as yet unmatched pairs. 1971 records in RAM.\n",
      "INFO\t2023-08-28 17:25:48\tMarkDuplicates\tRead     4,000,000 records.  Elapsed time: 00:00:06s.  Time for last 1,000,000:    1s.  Last read position: 3:130,736,045\n",
      "INFO\t2023-08-28 17:25:48\tMarkDuplicates\tTracking 1543 as yet unmatched pairs. 1543 records in RAM.\n",
      "INFO\t2023-08-28 17:25:49\tMarkDuplicates\tRead     5,000,000 records.  Elapsed time: 00:00:07s.  Time for last 1,000,000:    1s.  Last read position: 3:167,328,087\n",
      "INFO\t2023-08-28 17:25:49\tMarkDuplicates\tTracking 1029 as yet unmatched pairs. 1029 records in RAM.\n",
      "INFO\t2023-08-28 17:25:51\tMarkDuplicates\tRead     6,000,000 records.  Elapsed time: 00:00:09s.  Time for last 1,000,000:    1s.  Last read position: 3:198,076,827\n",
      "INFO\t2023-08-28 17:25:51\tMarkDuplicates\tTracking 183 as yet unmatched pairs. 183 records in RAM.\n",
      "INFO\t2023-08-28 17:25:51\tMarkDuplicates\tRead 6007564 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 17:25:51\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 514617296; totalMemory: 1225785344; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:25:51\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 17:25:51\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 17:25:52\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 17:25:52\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 17:25:52\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1320103136; totalMemory: 1866465280; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:25:52\tMarkDuplicates\tMarking 429707 records as duplicates.\n",
      "INFO\t2023-08-28 17:25:52\tMarkDuplicates\tFound 269 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam\n",
      "INFO\t2023-08-28 17:25:52\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 17:26:37\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 17:26:37\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 17:26:37\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 17:26:37\tMarkDuplicates\tBefore output close freeMemory: 528079032; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:26:37\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 17:26:37\tMarkDuplicates\tAfter output close freeMemory: 528079032; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 17:26:38 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.93 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 17:26:42 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T171514.177549.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:26:44 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753371_chr3_1_fastp.fastq.gz, data/processed/ERR753371_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753371_chr3.sam, results/mapped_reads/ERR753371_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753371_chr3_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753371_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.68 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 74.19 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.59 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.43 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 43.73 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 126.168 sec; CPU: 119.618 sec\n",
      "[Mon Aug 28 17:30:11 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:30:11 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753371_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr3.bam, results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753371_chr3.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753371_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 17:30:48 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:30:48 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753371_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753371_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "17:30:49.115 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 17:30:49 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753371_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753371_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 17:30:49 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 17:30:49\tMarkDuplicates\tStart of doWork freeMemory: 528796352; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:30:49\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 17:30:49\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam\n",
      "INFO\t2023-08-28 17:30:50\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:108,616,496\n",
      "INFO\t2023-08-28 17:30:50\tMarkDuplicates\tTracking 182 as yet unmatched pairs. 182 records in RAM.\n",
      "INFO\t2023-08-28 17:30:52\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 3:197,282,687\n",
      "INFO\t2023-08-28 17:30:52\tMarkDuplicates\tTracking 56 as yet unmatched pairs. 56 records in RAM.\n",
      "INFO\t2023-08-28 17:30:52\tMarkDuplicates\tRead 2019762 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 17:30:52\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 556785448; totalMemory: 846200832; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:30:52\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 17:30:52\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 17:30:52\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 17:30:53\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 17:30:53\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 874471048; totalMemory: 1420820480; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:30:53\tMarkDuplicates\tMarking 829944 records as duplicates.\n",
      "INFO\t2023-08-28 17:30:53\tMarkDuplicates\tFound 68 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam\n",
      "INFO\t2023-08-28 17:30:53\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 17:31:05\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 17:31:05\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 17:31:05\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 17:31:05\tMarkDuplicates\tBefore output close freeMemory: 528091912; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:31:05\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 17:31:05\tMarkDuplicates\tAfter output close freeMemory: 528091912; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 17:31:05 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.27 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 17:31:06 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T172643.327131.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:31:07 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753372_chr3_1_fastp.fastq.gz, data/processed/ERR753372_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753372_chr3.sam, results/mapped_reads/ERR753372_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753372_chr3_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753372_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.68 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 74.69 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.62 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.42 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 43.49 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 127.079 sec; CPU: 119.896 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 17:36:11 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:36:11 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753372_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr3.bam, results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753372_chr3.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753372_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 17:37:20 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:37:20 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753372_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753372_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "17:37:21.556 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 17:37:21 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753372_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753372_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 17:37:21 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 17:37:21\tMarkDuplicates\tStart of doWork freeMemory: 528826432; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:37:21\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 17:37:21\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam\n",
      "INFO\t2023-08-28 17:37:23\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:47,409,231\n",
      "INFO\t2023-08-28 17:37:23\tMarkDuplicates\tTracking 448 as yet unmatched pairs. 448 records in RAM.\n",
      "INFO\t2023-08-28 17:37:24\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 3:99,848,915\n",
      "INFO\t2023-08-28 17:37:24\tMarkDuplicates\tTracking 678 as yet unmatched pairs. 678 records in RAM.\n",
      "INFO\t2023-08-28 17:37:26\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 3:142,485,802\n",
      "INFO\t2023-08-28 17:37:26\tMarkDuplicates\tTracking 445 as yet unmatched pairs. 445 records in RAM.\n",
      "INFO\t2023-08-28 17:37:27\tMarkDuplicates\tRead 3984056 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 17:37:27\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 720164808; totalMemory: 1220542464; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:37:27\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 17:37:27\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 17:37:28\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 17:37:28\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 17:37:28\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1233119736; totalMemory: 1779433472; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:37:28\tMarkDuplicates\tMarking 95082 records as duplicates.\n",
      "INFO\t2023-08-28 17:37:28\tMarkDuplicates\tFound 627 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam\n",
      "INFO\t2023-08-28 17:37:28\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 17:37:52\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 17:37:52\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 17:37:52\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 17:37:52\tMarkDuplicates\tBefore output close freeMemory: 528125984; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:37:52\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 17:37:52\tMarkDuplicates\tAfter output close freeMemory: 528125984; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 17:37:52 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.51 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 17:37:54 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T173106.716379.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:37:56 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753373_chr3_1_fastp.fastq.gz, data/processed/ERR753373_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753373_chr3.sam, results/mapped_reads/ERR753373_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753373_chr3_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753373_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.67 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 74.59 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.61 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.41 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 43.63 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 126.301 sec; CPU: 119.908 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 17:44:15 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:44:15 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753373_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr3.bam, results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753373_chr3.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753373_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 17:46:15 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:46:15 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753373_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753373_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "17:46:16.164 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 17:46:16 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753373_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753373_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 17:46:16 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 17:46:16\tMarkDuplicates\tStart of doWork freeMemory: 528791096; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:46:16\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 17:46:16\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam\n",
      "INFO\t2023-08-28 17:46:17\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:52,256,390\n",
      "INFO\t2023-08-28 17:46:17\tMarkDuplicates\tTracking 257 as yet unmatched pairs. 257 records in RAM.\n",
      "INFO\t2023-08-28 17:46:19\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 3:108,161,476\n",
      "INFO\t2023-08-28 17:46:19\tMarkDuplicates\tTracking 296 as yet unmatched pairs. 296 records in RAM.\n",
      "INFO\t2023-08-28 17:46:21\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 3:154,972,600\n",
      "INFO\t2023-08-28 17:46:21\tMarkDuplicates\tTracking 208 as yet unmatched pairs. 208 records in RAM.\n",
      "INFO\t2023-08-28 17:46:22\tMarkDuplicates\tRead 3874269 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 17:46:22\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 602615168; totalMemory: 1090519040; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:46:22\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 17:46:22\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 17:46:23\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 17:46:23\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 17:46:23\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1081036952; totalMemory: 1627389952; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:46:23\tMarkDuplicates\tMarking 104535 records as duplicates.\n",
      "INFO\t2023-08-28 17:46:23\tMarkDuplicates\tFound 34216 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam\n",
      "INFO\t2023-08-28 17:46:23\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 17:46:57\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 17:46:57\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 17:46:57\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 17:46:57\tMarkDuplicates\tBefore output close freeMemory: 528088272; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:46:57\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 17:46:57\tMarkDuplicates\tAfter output close freeMemory: 528088272; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 17:46:57 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.68 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 17:47:00 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T173755.090444.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:47:01 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753374_chr3_1_fastp.fastq.gz, data/processed/ERR753374_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753374_chr3.sam, results/mapped_reads/ERR753374_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753374_chr3_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753374_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.64 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 70.44 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.59 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.40 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 41.22 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 119.852 sec; CPU: 113.296 sec\n",
      "[Mon Aug 28 17:51:09 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:51:09 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753374_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr3.bam, results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753374_chr3.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753374_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 17:52:07 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:52:07 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753374_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753374_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "17:52:08.394 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 17:52:08 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753374_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753374_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 17:52:08 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 17:52:08\tMarkDuplicates\tStart of doWork freeMemory: 528832968; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:52:08\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 17:52:08\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam\n",
      "INFO\t2023-08-28 17:52:10\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:96,373,675\n",
      "INFO\t2023-08-28 17:52:10\tMarkDuplicates\tTracking 262 as yet unmatched pairs. 262 records in RAM.\n",
      "INFO\t2023-08-28 17:52:11\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 3:175,259,192\n",
      "INFO\t2023-08-28 17:52:11\tMarkDuplicates\tTracking 108 as yet unmatched pairs. 108 records in RAM.\n",
      "INFO\t2023-08-28 17:52:12\tMarkDuplicates\tRead 2270052 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 17:52:12\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 563343128; totalMemory: 878706688; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:52:12\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 17:52:12\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 17:52:12\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 17:52:12\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 17:52:12\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 869267544; totalMemory: 1415577600; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:52:12\tMarkDuplicates\tMarking 244466 records as duplicates.\n",
      "INFO\t2023-08-28 17:52:12\tMarkDuplicates\tFound 81743 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam\n",
      "INFO\t2023-08-28 17:52:12\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 17:52:31\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 17:52:31\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 17:52:31\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 17:52:31\tMarkDuplicates\tBefore output close freeMemory: 528131288; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:52:31\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 17:52:31\tMarkDuplicates\tAfter output close freeMemory: 528131288; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 17:52:31 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.39 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 17:52:33 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T174700.554081.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:52:34 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753375_chr3_1_fastp.fastq.gz, data/processed/ERR753375_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753375_chr3.sam, results/mapped_reads/ERR753375_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753375_chr3_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753375_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.66 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 69.13 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.58 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.41 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 41.46 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 118.557 sec; CPU: 112.246 sec\n",
      "[Mon Aug 28 17:55:45 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:55:45 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753375_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr3.bam, results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753375_chr3.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753375_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 17:56:18 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:56:18 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753375_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753375_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "17:56:18.933 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 17:56:18 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753375_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753375_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 17:56:18 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 17:56:18\tMarkDuplicates\tStart of doWork freeMemory: 528786488; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:56:18\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 17:56:18\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam\n",
      "INFO\t2023-08-28 17:56:20\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:121,913,097\n",
      "INFO\t2023-08-28 17:56:20\tMarkDuplicates\tTracking 262 as yet unmatched pairs. 262 records in RAM.\n",
      "INFO\t2023-08-28 17:56:21\tMarkDuplicates\tRead 1693364 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 17:56:21\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 348442584; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:56:21\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 17:56:21\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 17:56:22\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 17:56:22\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 17:56:22\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 810500768; totalMemory: 1356857344; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:56:22\tMarkDuplicates\tMarking 82812 records as duplicates.\n",
      "INFO\t2023-08-28 17:56:22\tMarkDuplicates\tFound 199 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam\n",
      "INFO\t2023-08-28 17:56:22\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 17:56:33\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 17:56:33\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 17:56:33\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 17:56:33\tMarkDuplicates\tBefore output close freeMemory: 528084784; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 17:56:33\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 17:56:33\tMarkDuplicates\tAfter output close freeMemory: 528084784; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 17:56:33 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.24 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 17:56:34 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T175233.690982.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 17:56:35 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753376_chr3_1_fastp.fastq.gz, data/processed/ERR753376_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753376_chr3.sam, results/mapped_reads/ERR753376_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753376_chr3_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753376_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.67 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 76.40 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.60 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.40 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 43.59 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 128.612 sec; CPU: 121.665 sec\n",
      "[Mon Aug 28 18:00:10 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:00:10 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753376_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr3.bam, results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753376_chr3.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753376_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 18:00:50 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:00:50 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753376_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753376_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:00:51.684 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 18:00:51 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753376_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753376_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 18:00:51 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 18:00:51\tMarkDuplicates\tStart of doWork freeMemory: 528792280; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 18:00:51\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 18:00:51\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam\n",
      "INFO\t2023-08-28 18:00:53\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:101,484,466\n",
      "INFO\t2023-08-28 18:00:53\tMarkDuplicates\tTracking 283 as yet unmatched pairs. 283 records in RAM.\n",
      "INFO\t2023-08-28 18:00:54\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 3:193,643,851\n",
      "INFO\t2023-08-28 18:00:54\tMarkDuplicates\tTracking 41 as yet unmatched pairs. 41 records in RAM.\n",
      "INFO\t2023-08-28 18:00:54\tMarkDuplicates\tRead 2085915 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 18:00:54\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 563749408; totalMemory: 859832320; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 18:00:54\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 18:00:54\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 18:00:55\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 18:00:55\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 18:00:55\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 881809064; totalMemory: 1428160512; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 18:00:55\tMarkDuplicates\tMarking 302663 records as duplicates.\n",
      "INFO\t2023-08-28 18:00:55\tMarkDuplicates\tFound 89 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam\n",
      "INFO\t2023-08-28 18:00:55\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 18:01:08\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 18:01:08\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 18:01:08\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 18:01:08\tMarkDuplicates\tBefore output close freeMemory: 528090248; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 18:01:08\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 18:01:08\tMarkDuplicates\tAfter output close freeMemory: 528090248; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 18:01:08 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.28 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 18:01:10 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T175634.545827.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:01:11 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753377_chr3_1_fastp.fastq.gz, data/processed/ERR753377_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753377_chr3.sam, results/mapped_reads/ERR753377_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753377_chr3_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753377_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.66 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 76.94 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.59 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.40 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 41.38 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 126.308 sec; CPU: 119.972 sec\n",
      "[Mon Aug 28 18:04:41 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:04:41 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753377_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr3.bam, results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753377_chr3.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753377_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 18:05:22 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:05:22 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753377_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753377_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:05:23.119 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 18:05:23 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753377_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753377_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 18:05:23 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 18:05:23\tMarkDuplicates\tStart of doWork freeMemory: 528833080; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 18:05:23\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 18:05:23\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam\n",
      "INFO\t2023-08-28 18:05:24\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:98,149,597\n",
      "INFO\t2023-08-28 18:05:24\tMarkDuplicates\tTracking 639 as yet unmatched pairs. 639 records in RAM.\n",
      "INFO\t2023-08-28 18:05:25\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 3:183,822,533\n",
      "INFO\t2023-08-28 18:05:25\tMarkDuplicates\tTracking 140 as yet unmatched pairs. 140 records in RAM.\n",
      "INFO\t2023-08-28 18:05:26\tMarkDuplicates\tRead 2212613 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 18:05:26\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 292648296; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 18:05:26\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 18:05:26\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 18:05:26\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 18:05:27\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 18:05:27\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 881850168; totalMemory: 1428160512; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 18:05:27\tMarkDuplicates\tMarking 317042 records as duplicates.\n",
      "INFO\t2023-08-28 18:05:27\tMarkDuplicates\tFound 102 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam\n",
      "INFO\t2023-08-28 18:05:27\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 18:05:40\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 18:05:40\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 18:05:40\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 18:05:40\tMarkDuplicates\tBefore output close freeMemory: 528128800; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 18:05:40\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 18:05:40\tMarkDuplicates\tAfter output close freeMemory: 528128800; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 18:05:40 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.29 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 18:05:41 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T180110.408498.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:05:43 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753378_chr3_1_fastp.fastq.gz, data/processed/ERR753378_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753378_chr3.sam, results/mapped_reads/ERR753378_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753378_chr3_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753378_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.67 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 69.09 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.59 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.40 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 41.54 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 119.005 sec; CPU: 112.289 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 18:10:04 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:10:04 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753378_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr3.bam, results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753378_chr3.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753378_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 18:11:04 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:11:04 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753378_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753378_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:11:05.716 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 18:11:05 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753378_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753378_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 18:11:05 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 18:11:05\tMarkDuplicates\tStart of doWork freeMemory: 528791480; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 18:11:05\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 18:11:05\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam\n",
      "INFO\t2023-08-28 18:11:07\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:49,250,722\n",
      "INFO\t2023-08-28 18:11:07\tMarkDuplicates\tTracking 439 as yet unmatched pairs. 439 records in RAM.\n",
      "INFO\t2023-08-28 18:11:08\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 3:117,903,530\n",
      "INFO\t2023-08-28 18:11:08\tMarkDuplicates\tTracking 566 as yet unmatched pairs. 566 records in RAM.\n",
      "INFO\t2023-08-28 18:11:10\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 3:175,097,199\n",
      "INFO\t2023-08-28 18:11:10\tMarkDuplicates\tTracking 360 as yet unmatched pairs. 360 records in RAM.\n",
      "INFO\t2023-08-28 18:11:10\tMarkDuplicates\tRead 3435097 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 18:11:10\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 631444920; totalMemory: 1072693248; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 18:11:10\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 18:11:10\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 18:11:11\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 18:11:11\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 18:11:11\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1063212152; totalMemory: 1609564160; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 18:11:11\tMarkDuplicates\tMarking 81425 records as duplicates.\n",
      "INFO\t2023-08-28 18:11:11\tMarkDuplicates\tFound 564 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam\n",
      "INFO\t2023-08-28 18:11:11\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 18:11:31\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 18:11:31\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 18:11:31\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 18:11:31\tMarkDuplicates\tBefore output close freeMemory: 528089568; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 18:11:31\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 18:11:31\tMarkDuplicates\tAfter output close freeMemory: 528089568; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 18:11:31 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.42 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 18:11:33 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T180542.174678.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:11:34 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR696683_chr3_dedup.bam\n",
      "    output: results/variants/ERR696683_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR696683_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR696683_chr3.vcf\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "index file data/reference/genome.fa.fai not found, generating...\n",
      "[Mon Aug 28 18:12:30 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T181133.647520.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:12:31 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753368_chr3_dedup.bam\n",
      "    output: results/variants/ERR753368_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753368_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753368_chr3.vcf\n",
      "    wildcards: sample=ERR753368_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 18:13:48 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T181230.752420.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:13:49 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753369_chr3_dedup.bam\n",
      "    output: results/variants/ERR753369_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753369_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753369_chr3.vcf\n",
      "    wildcards: sample=ERR753369_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 18:15:08 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T181348.447126.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:15:09 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753370_chr3_dedup.bam\n",
      "    output: results/variants/ERR753370_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753370_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753370_chr3.vcf\n",
      "    wildcards: sample=ERR753370_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 18:17:16 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T181508.875528.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:17:17 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753371_chr3_dedup.bam\n",
      "    output: results/variants/ERR753371_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753371_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753371_chr3.vcf\n",
      "    wildcards: sample=ERR753371_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 18:17:57 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T181716.434119.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:17:58 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753372_chr3_dedup.bam\n",
      "    output: results/variants/ERR753372_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753372_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753372_chr3.vcf\n",
      "    wildcards: sample=ERR753372_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 18:19:17 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T181757.455551.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:19:18 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753373_chr3_dedup.bam\n",
      "    output: results/variants/ERR753373_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753373_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753373_chr3.vcf\n",
      "    wildcards: sample=ERR753373_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 18:20:54 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T181917.575095.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:20:55 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753374_chr3_dedup.bam\n",
      "    output: results/variants/ERR753374_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753374_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753374_chr3.vcf\n",
      "    wildcards: sample=ERR753374_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 18:21:58 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T182054.447068.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:21:59 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753375_chr3_dedup.bam\n",
      "    output: results/variants/ERR753375_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753375_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753375_chr3.vcf\n",
      "    wildcards: sample=ERR753375_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 18:22:45 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T182158.503353.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:22:46 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753376_chr3_dedup.bam\n",
      "    output: results/variants/ERR753376_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753376_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753376_chr3.vcf\n",
      "    wildcards: sample=ERR753376_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 18:23:33 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T182245.167008.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:23:34 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753377_chr3_dedup.bam\n",
      "    output: results/variants/ERR753377_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753377_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753377_chr3.vcf\n",
      "    wildcards: sample=ERR753377_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 18:24:23 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T182333.831268.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:24:25 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753378_chr3_dedup.bam\n",
      "    output: results/variants/ERR753378_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753378_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753378_chr3.vcf\n",
      "    wildcards: sample=ERR753378_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 18:25:36 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T182423.974840.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "vep_install_db        1              1              1\n",
      "total                 1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:25:37 2023]\n",
      "rule vep_install_db:\n",
      "    output: tasks/11vep_dependencies.done\n",
      "    jobid: 0\n",
      "    reason: Missing output files: tasks/11vep_dependencies.done\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " - getting list of available cache files\n",
      "\n",
      "WARNING: It looks like you already have the cache for homo_sapiens GRCh38 (v109) installed.\n",
      "\n",
      "Delete the folder /home/juancarlos/.vep/homo_sapiens/109_GRCh38 and re-run INSTALL.pl if you want to re-install\n",
      " - skipping homo_sapiens\n",
      "Looks like you already have the FASTA file for homo_sapiens, skipping\n",
      "\n",
      "All done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Touching output file tasks/11vep_dependencies.done.\n",
      "[Mon Aug 28 18:26:09 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T182536.479408.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:26:10 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR696683_chr3.vcf\n",
      "    output: results/variants/vep/ERR696683_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR696683_chr3.txt\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "--2023-08-28 18:26:11--  https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/clinvar.vcf.gz\n",
      "Resolviendo ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.10, 130.14.250.7, 2607:f220:41e:250::12, ...\n",
      "Conectando con ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)[130.14.250.10]:443... conectado.\n",
      "Petición HTTP enviada, esperando respuesta... 200 OK\n",
      "Longitud: 81866312 (78M) [application/x-gzip]\n",
      "Grabando a: “data/ClinVar/clinvar.vcf.gz”\n",
      "\n",
      "     0K .......... .......... .......... .......... ..........  0%  205K 6m30s\n",
      "    50K .......... .......... .......... .......... ..........  0%  414K 4m51s\n",
      "   100K .......... .......... .......... .......... ..........  0% 95,1M 3m14s\n",
      "   150K .......... .......... .......... .......... ..........  0%  413K 3m14s\n",
      "   200K .......... .......... .......... .......... ..........  0% 22,9M 2m36s\n",
      "   250K .......... .......... .......... .......... ..........  0% 71,5M 2m10s\n",
      "   300K .......... .......... .......... .......... ..........  0%  183M 1m51s\n",
      "   350K .......... .......... .......... .......... ..........  0%  420K 2m1s\n",
      "   400K .......... .......... .......... .......... ..........  0% 95,6M 1m48s\n",
      "   450K .......... .......... .......... .......... ..........  0% 55,0M 97s\n",
      "   500K .......... .......... .......... .......... ..........  0% 53,5M 88s\n",
      "   550K .......... .......... .......... .......... ..........  0%  118M 81s\n",
      "   600K .......... .......... .......... .......... ..........  0%  117M 75s\n",
      "   650K .......... .......... .......... .......... ..........  0%  117M 69s\n",
      "   700K .......... .......... .......... .......... ..........  0%  114M 65s\n",
      "   750K .......... .......... .......... .......... ..........  1%  426K 72s\n",
      "   800K .......... .......... .......... .......... ..........  1% 23,9M 68s\n",
      "   850K .......... .......... .......... .......... ..........  1%  119M 64s\n",
      "   900K .......... .......... .......... .......... ..........  1%  121M 61s\n",
      "   950K .......... .......... .......... .......... ..........  1%  129M 58s\n",
      "  1000K .......... .......... .......... .......... ..........  1%  114M 55s\n",
      "  1050K .......... .......... .......... .......... ..........  1%  126M 53s\n",
      "  1100K .......... .......... .......... .......... ..........  1%  128M 50s\n",
      "  1150K .......... .......... .......... .......... ..........  1%  124M 48s\n",
      "  1200K .......... .......... .......... .......... ..........  1%  103M 46s\n",
      "  1250K .......... .......... .......... .......... ..........  1%  114M 45s\n",
      "  1300K .......... .......... .......... .......... ..........  1%  116M 43s\n",
      "  1350K .......... .......... .......... .......... ..........  1%  123M 41s\n",
      "  1400K .......... .......... .......... .......... ..........  1%  107M 40s\n",
      "  1450K .......... .......... .......... .......... ..........  1%  119M 39s\n",
      "  1500K .......... .......... .......... .......... ..........  1%  112M 37s\n",
      "  1550K .......... .......... .......... .......... ..........  2%  438K 42s\n",
      "  1600K .......... .......... .......... .......... ..........  2% 72,5M 40s\n",
      "  1650K .......... .......... .......... .......... ..........  2% 33,6M 39s\n",
      "  1700K .......... .......... .......... .......... ..........  2% 90,7M 38s\n",
      "  1750K .......... .......... .......... .......... ..........  2%  120M 37s\n",
      "  1800K .......... .......... .......... .......... ..........  2%  107M 36s\n",
      "  1850K .......... .......... .......... .......... ..........  2%  116M 35s\n",
      "  1900K .......... .......... .......... .......... ..........  2%  118M 34s\n",
      "  1950K .......... .......... .......... .......... ..........  2%  116M 33s\n",
      "  2000K .......... .......... .......... .......... ..........  2% 60,1M 33s\n",
      "  2050K .......... .......... .......... .......... ..........  2%  125M 32s\n",
      "  2100K .......... .......... .......... .......... ..........  2%  126M 31s\n",
      "  2150K .......... .......... .......... .......... ..........  2%  130M 30s\n",
      "  2200K .......... .......... .......... .......... ..........  2%  117M 30s\n",
      "  2250K .......... .......... .......... .......... ..........  2%  127M 29s\n",
      "  2300K .......... .......... .......... .......... ..........  2%  120M 28s\n",
      "  2350K .......... .......... .......... .......... ..........  3%  126M 28s\n",
      "  2400K .......... .......... .......... .......... ..........  3%  111M 27s\n",
      "  2450K .......... .......... .......... .......... ..........  3%  116M 27s\n",
      "  2500K .......... .......... .......... .......... ..........  3%  113M 26s\n",
      "  2550K .......... .......... .......... .......... ..........  3%  118M 26s\n",
      "  2600K .......... .......... .......... .......... ..........  3%  108M 25s\n",
      "  2650K .......... .......... .......... .......... ..........  3%  123M 25s\n",
      "  2700K .......... .......... .......... .......... ..........  3%  118M 24s\n",
      "  2750K .......... .......... .......... .......... ..........  3%  113M 24s\n",
      "  2800K .......... .......... .......... .......... ..........  3% 92,9M 23s\n",
      "  2850K .......... .......... .......... .......... ..........  3%  118M 23s\n",
      "  2900K .......... .......... .......... .......... ..........  3%  114M 23s\n",
      "  2950K .......... .......... .......... .......... ..........  3%  118M 22s\n",
      "  3000K .......... .......... .......... .......... ..........  3%  111M 22s\n",
      "  3050K .......... .......... .......... .......... ..........  3%  116M 21s\n",
      "  3100K .......... .......... .......... .......... ..........  3%  119M 21s\n",
      "  3150K .......... .......... .......... .......... ..........  4%  466K 23s\n",
      "  3200K .......... .......... .......... .......... ..........  4% 94,7M 23s\n",
      "  3250K .......... .......... .......... .......... ..........  4% 55,7M 23s\n",
      "  3300K .......... .......... .......... .......... ..........  4%  111M 22s\n",
      "  3350K .......... .......... .......... .......... ..........  4%  124M 22s\n",
      "  3400K .......... .......... .......... .......... ..........  4%  110M 22s\n",
      "  3450K .......... .......... .......... .......... ..........  4%  115M 21s\n",
      "  3500K .......... .......... .......... .......... ..........  4%  117M 21s\n",
      "  3550K .......... .......... .......... .......... ..........  4%  116M 21s\n",
      "  3600K .......... .......... .......... .......... ..........  4% 92,6M 20s\n",
      "  3650K .......... .......... .......... .......... ..........  4%  117M 20s\n",
      "  3700K .......... .......... .......... .......... ..........  4%  115M 20s\n",
      "  3750K .......... .......... .......... .......... ..........  4%  111M 20s\n",
      "  3800K .......... .......... .......... .......... ..........  4%  115M 19s\n",
      "  3850K .......... .......... .......... .......... ..........  4%  118M 19s\n",
      "  3900K .......... .......... .......... .......... ..........  4%  114M 19s\n",
      "  3950K .......... .......... .......... .......... ..........  5%  118M 19s\n",
      "  4000K .......... .......... .......... .......... ..........  5% 93,6M 18s\n",
      "  4050K .......... .......... .......... .......... ..........  5% 86,2M 18s\n",
      "  4100K .......... .......... .......... .......... ..........  5%  161M 18s\n",
      "  4150K .......... .......... .......... .......... ..........  5%  126M 18s\n",
      "  4200K .......... .......... .......... .......... ..........  5%  107M 18s\n",
      "  4250K .......... .......... .......... .......... ..........  5%  116M 17s\n",
      "  4300K .......... .......... .......... .......... ..........  5%  120M 17s\n",
      "  4350K .......... .......... .......... .......... ..........  5%  116M 17s\n",
      "  4400K .......... .......... .......... .......... ..........  5% 91,8M 17s\n",
      "  4450K .......... .......... .......... .......... ..........  5%  117M 17s\n",
      "  4500K .......... .......... .......... .......... ..........  5%  118M 16s\n",
      "  4550K .......... .......... .......... .......... ..........  5%  114M 16s\n",
      "  4600K .......... .......... .......... .......... ..........  5%  110M 16s\n",
      "  4650K .......... .......... .......... .......... ..........  5%  117M 16s\n",
      "  4700K .......... .......... .......... .......... ..........  5%  117M 16s\n",
      "  4750K .......... .......... .......... .......... ..........  6% 38,8M 15s\n",
      "  4800K .......... .......... .......... .......... ..........  6% 69,8M 15s\n",
      "  4850K .......... .......... .......... .......... ..........  6%  112M 15s\n",
      "  4900K .......... .......... .......... .......... ..........  6%  106M 15s\n",
      "  4950K .......... .......... .......... .......... ..........  6%  164M 15s\n",
      "  5000K .......... .......... .......... .......... ..........  6%  161M 15s\n",
      "  5050K .......... .......... .......... .......... ..........  6%  172M 15s\n",
      "  5100K .......... .......... .......... .......... ..........  6%  173M 14s\n",
      "  5150K .......... .......... .......... .......... ..........  6%  177M 14s\n",
      "  5200K .......... .......... .......... .......... ..........  6%  149M 14s\n",
      "  5250K .......... .......... .......... .......... ..........  6%  186M 14s\n",
      "  5300K .......... .......... .......... .......... ..........  6%  119M 14s\n",
      "  5350K .......... .......... .......... .......... ..........  6% 93,3M 14s\n",
      "  5400K .......... .......... .......... .......... ..........  6% 84,5M 14s\n",
      "  5450K .......... .......... .......... .......... ..........  6%  128M 13s\n",
      "  5500K .......... .......... .......... .......... ..........  6%  126M 13s\n",
      "  5550K .......... .......... .......... .......... ..........  7%  149M 13s\n",
      "  5600K .......... .......... .......... .......... ..........  7%  118M 13s\n",
      "  5650K .......... .......... .......... .......... ..........  7%  117M 13s\n",
      "  5700K .......... .......... .......... .......... ..........  7%  117M 13s\n",
      "  5750K .......... .......... .......... .......... ..........  7%  117M 13s\n",
      "  5800K .......... .......... .......... .......... ..........  7%  110M 13s\n",
      "  5850K .......... .......... .......... .......... ..........  7%  117M 13s\n",
      "  5900K .......... .......... .......... .......... ..........  7%  120M 12s\n",
      "  5950K .......... .......... .......... .......... ..........  7%  117M 12s\n",
      "  6000K .......... .......... .......... .......... ..........  7% 88,2M 12s\n",
      "  6050K .......... .......... .......... .......... ..........  7%  117M 12s\n",
      "  6100K .......... .......... .......... .......... ..........  7%  118M 12s\n",
      "  6150K .......... .......... .......... .......... ..........  7%  117M 12s\n",
      "  6200K .......... .......... .......... .......... ..........  7%  110M 12s\n",
      "  6250K .......... .......... .......... .......... ..........  7%  524K 13s\n",
      "  6300K .......... .......... .......... .......... ..........  7% 50,8M 13s\n",
      "  6350K .......... .......... .......... .......... ..........  8% 57,2M 13s\n",
      "  6400K .......... .......... .......... .......... ..........  8% 57,5M 13s\n",
      "  6450K .......... .......... .......... .......... ..........  8% 60,7M 12s\n",
      "  6500K .......... .......... .......... .......... ..........  8%  116M 12s\n",
      "  6550K .......... .......... .......... .......... ..........  8%  119M 12s\n",
      "  6600K .......... .......... .......... .......... ..........  8%  104M 12s\n",
      "  6650K .......... .......... .......... .......... ..........  8%  113M 12s\n",
      "  6700K .......... .......... .......... .......... ..........  8%  123M 12s\n",
      "  6750K .......... .......... .......... .......... ..........  8%  118M 12s\n",
      "  6800K .......... .......... .......... .......... ..........  8% 95,6M 12s\n",
      "  6850K .......... .......... .......... .......... ..........  8%  117M 12s\n",
      "  6900K .......... .......... .......... .......... ..........  8%  117M 12s\n",
      "  6950K .......... .......... .......... .......... ..........  8%  114M 12s\n",
      "  7000K .......... .......... .......... .......... ..........  8%  115M 11s\n",
      "  7050K .......... .......... .......... .......... ..........  8%  115M 11s\n",
      "  7100K .......... .......... .......... .......... ..........  8%  118M 11s\n",
      "  7150K .......... .......... .......... .......... ..........  9%  118M 11s\n",
      "  7200K .......... .......... .......... .......... ..........  9% 89,5M 11s\n",
      "  7250K .......... .......... .......... .......... ..........  9%  117M 11s\n",
      "  7300K .......... .......... .......... .......... ..........  9%  118M 11s\n",
      "  7350K .......... .......... .......... .......... ..........  9%  119M 11s\n",
      "  7400K .......... .......... .......... .......... ..........  9%  114M 11s\n",
      "  7450K .......... .......... .......... .......... ..........  9%  113M 11s\n",
      "  7500K .......... .......... .......... .......... ..........  9%  119M 11s\n",
      "  7550K .......... .......... .......... .......... ..........  9%  116M 11s\n",
      "  7600K .......... .......... .......... .......... ..........  9% 89,2M 10s\n",
      "  7650K .......... .......... .......... .......... ..........  9%  118M 10s\n",
      "  7700K .......... .......... .......... .......... ..........  9%  117M 10s\n",
      "  7750K .......... .......... .......... .......... ..........  9%  117M 10s\n",
      "  7800K .......... .......... .......... .......... ..........  9%  114M 10s\n",
      "  7850K .......... .......... .......... .......... ..........  9% 87,4M 10s\n",
      "  7900K .......... .......... .......... .......... ..........  9% 27,4M 10s\n",
      "  7950K .......... .......... .......... .......... .......... 10%  142M 10s\n",
      "  8000K .......... .......... .......... .......... .......... 10% 89,8M 10s\n",
      "  8050K .......... .......... .......... .......... .......... 10%  114M 10s\n",
      "  8100K .......... .......... .......... .......... .......... 10%  117M 10s\n",
      "  8150K .......... .......... .......... .......... .......... 10%  117M 10s\n",
      "  8200K .......... .......... .......... .......... .......... 10%  114M 10s\n",
      "  8250K .......... .......... .......... .......... .......... 10%  117M 10s\n",
      "  8300K .......... .......... .......... .......... .......... 10%  114M 10s\n",
      "  8350K .......... .......... .......... .......... .......... 10%  117M 10s\n",
      "  8400K .......... .......... .......... .......... .......... 10% 90,8M 9s\n",
      "  8450K .......... .......... .......... .......... .......... 10%  114M 9s\n",
      "  8500K .......... .......... .......... .......... .......... 10%  117M 9s\n",
      "  8550K .......... .......... .......... .......... .......... 10%  117M 9s\n",
      "  8600K .......... .......... .......... .......... .......... 10%  115M 9s\n",
      "  8650K .......... .......... .......... .......... .......... 10%  117M 9s\n",
      "  8700K .......... .......... .......... .......... .......... 10%  114M 9s\n",
      "  8750K .......... .......... .......... .......... .......... 11%  117M 9s\n",
      "  8800K .......... .......... .......... .......... .......... 11% 90,1M 9s\n",
      "  8850K .......... .......... .......... .......... .......... 11%  114M 9s\n",
      "  8900K .......... .......... .......... .......... .......... 11%  117M 9s\n",
      "  8950K .......... .......... .......... .......... .......... 11%  117M 9s\n",
      "  9000K .......... .......... .......... .......... .......... 11%  115M 9s\n",
      "  9050K .......... .......... .......... .......... .......... 11%  117M 9s\n",
      "  9100K .......... .......... .......... .......... .......... 11%  113M 9s\n",
      "  9150K .......... .......... .......... .......... .......... 11%  118M 9s\n",
      "  9200K .......... .......... .......... .......... .......... 11% 90,4M 9s\n",
      "  9250K .......... .......... .......... .......... .......... 11% 7,57M 9s\n",
      "  9300K .......... .......... .......... .......... .......... 11%  567K 9s\n",
      "  9350K .......... .......... .......... .......... .......... 11% 52,5M 9s\n",
      "  9400K .......... .......... .......... .......... .......... 11% 59,4M 9s\n",
      "  9450K .......... .......... .......... .......... .......... 11%  100M 9s\n",
      "  9500K .......... .......... .......... .......... .......... 11%  133M 9s\n",
      "  9550K .......... .......... .......... .......... .......... 12%  103M 9s\n",
      "  9600K .......... .......... .......... .......... .......... 12% 68,7M 9s\n",
      "  9650K .......... .......... .......... .......... .......... 12% 93,6M 9s\n",
      "  9700K .......... .......... .......... .......... .......... 12% 94,9M 9s\n",
      "  9750K .......... .......... .......... .......... .......... 12%  101M 9s\n",
      "  9800K .......... .......... .......... .......... .......... 12% 85,1M 9s\n",
      "  9850K .......... .......... .......... .......... .......... 12%  103M 9s\n",
      "  9900K .......... .......... .......... .......... .......... 12% 93,4M 9s\n",
      "  9950K .......... .......... .......... .......... .......... 12% 94,6M 9s\n",
      " 10000K .......... .......... .......... .......... .......... 12% 76,2M 9s\n",
      " 10050K .......... .......... .......... .......... .......... 12% 95,8M 8s\n",
      " 10100K .......... .......... .......... .......... .......... 12% 92,2M 8s\n",
      " 10150K .......... .......... .......... .......... .......... 12% 91,2M 8s\n",
      " 10200K .......... .......... .......... .......... .......... 12% 61,3M 8s\n",
      " 10250K .......... .......... .......... .......... .......... 12%  117M 8s\n",
      " 10300K .......... .......... .......... .......... .......... 12% 93,0M 8s\n",
      " 10350K .......... .......... .......... .......... .......... 13% 86,7M 8s\n",
      " 10400K .......... .......... .......... .......... .......... 13% 62,1M 8s\n",
      " 10450K .......... .......... .......... .......... .......... 13% 88,3M 8s\n",
      " 10500K .......... .......... .......... .......... .......... 13%  131M 8s\n",
      " 10550K .......... .......... .......... .......... .......... 13%  219M 8s\n",
      " 10600K .......... .......... .......... .......... .......... 13%  243M 8s\n",
      " 10650K .......... .......... .......... .......... .......... 13%  285M 8s\n",
      " 10700K .......... .......... .......... .......... .......... 13%  280M 8s\n",
      " 10750K .......... .......... .......... .......... .......... 13%  248M 8s\n",
      " 10800K .......... .......... .......... .......... .......... 13%  227M 8s\n",
      " 10850K .......... .......... .......... .......... .......... 13%  274M 8s\n",
      " 10900K .......... .......... .......... .......... .......... 13%  284M 8s\n",
      " 10950K .......... .......... .......... .......... .......... 13%  297M 8s\n",
      " 11000K .......... .......... .......... .......... .......... 13% 64,6M 8s\n",
      " 11050K .......... .......... .......... .......... .......... 13% 87,0M 8s\n",
      " 11100K .......... .......... .......... .......... .......... 13% 37,7M 8s\n",
      " 11150K .......... .......... .......... .......... .......... 14% 54,6M 8s\n",
      " 11200K .......... .......... .......... .......... .......... 14% 89,7M 8s\n",
      " 11250K .......... .......... .......... .......... .......... 14%  117M 8s\n",
      " 11300K .......... .......... .......... .......... .......... 14%  117M 7s\n",
      " 11350K .......... .......... .......... .......... .......... 14%  114M 7s\n",
      " 11400K .......... .......... .......... .......... .......... 14%  113M 7s\n",
      " 11450K .......... .......... .......... .......... .......... 14%  117M 7s\n",
      " 11500K .......... .......... .......... .......... .......... 14%  117M 7s\n",
      " 11550K .......... .......... .......... .......... .......... 14%  118M 7s\n",
      " 11600K .......... .......... .......... .......... .......... 14% 89,2M 7s\n",
      " 11650K .......... .......... .......... .......... .......... 14%  117M 7s\n",
      " 11700K .......... .......... .......... .......... .......... 14%  117M 7s\n",
      " 11750K .......... .......... .......... .......... .......... 14%  114M 7s\n",
      " 11800K .......... .......... .......... .......... .......... 14%  113M 7s\n",
      " 11850K .......... .......... .......... .......... .......... 14%  118M 7s\n",
      " 11900K .......... .......... .......... .......... .......... 14%  117M 7s\n",
      " 11950K .......... .......... .......... .......... .......... 15%  117M 7s\n",
      " 12000K .......... .......... .......... .......... .......... 15% 88,9M 7s\n",
      " 12050K .......... .......... .......... .......... .......... 15%  118M 7s\n",
      " 12100K .......... .......... .......... .......... .......... 15%  118M 7s\n",
      " 12150K .......... .......... .......... .......... .......... 15%  113M 7s\n",
      " 12200K .......... .......... .......... .......... .......... 15%  114M 7s\n",
      " 12250K .......... .......... .......... .......... .......... 15%  117M 7s\n",
      " 12300K .......... .......... .......... .......... .......... 15% 8,04M 7s\n",
      " 12350K .......... .......... .......... .......... .......... 15%  569K 7s\n",
      " 12400K .......... .......... .......... .......... .......... 15% 32,5M 7s\n",
      " 12450K .......... .......... .......... .......... .......... 15% 62,0M 7s\n",
      " 12500K .......... .......... .......... .......... .......... 15%  126M 7s\n",
      " 12550K .......... .......... .......... .......... .......... 15%  117M 7s\n",
      " 12600K .......... .......... .......... .......... .......... 15%  107M 7s\n",
      " 12650K .......... .......... .......... .......... .......... 15%  116M 7s\n",
      " 12700K .......... .......... .......... .......... .......... 15%  117M 7s\n",
      " 12750K .......... .......... .......... .......... .......... 16%  117M 7s\n",
      " 12800K .......... .......... .......... .......... .......... 16% 92,2M 7s\n",
      " 12850K .......... .......... .......... .......... .......... 16%  117M 7s\n",
      " 12900K .......... .......... .......... .......... .......... 16%  119M 7s\n",
      " 12950K .......... .......... .......... .......... .......... 16%  116M 7s\n",
      " 13000K .......... .......... .......... .......... .......... 16%  104M 7s\n",
      " 13050K .......... .......... .......... .......... .......... 16%  120M 7s\n",
      " 13100K .......... .......... .......... .......... .......... 16%  117M 7s\n",
      " 13150K .......... .......... .......... .......... .......... 16%  116M 7s\n",
      " 13200K .......... .......... .......... .......... .......... 16% 93,6M 7s\n",
      " 13250K .......... .......... .......... .......... .......... 16%  103M 7s\n",
      " 13300K .......... .......... .......... .......... .......... 16%  137M 7s\n",
      " 13350K .......... .......... .......... .......... .......... 16%  115M 7s\n",
      " 13400K .......... .......... .......... .......... .......... 16%  111M 7s\n",
      " 13450K .......... .......... .......... .......... .......... 16%  115M 7s\n",
      " 13500K .......... .......... .......... .......... .......... 16% 68,2M 7s\n",
      " 13550K .......... .......... .......... .......... .......... 17% 56,6M 7s\n",
      " 13600K .......... .......... .......... .......... .......... 17% 62,9M 7s\n",
      " 13650K .......... .......... .......... .......... .......... 17% 84,4M 7s\n",
      " 13700K .......... .......... .......... .......... .......... 17%  117M 7s\n",
      " 13750K .......... .......... .......... .......... .......... 17%  117M 7s\n",
      " 13800K .......... .......... .......... .......... .......... 17%  100M 6s\n",
      " 13850K .......... .......... .......... .......... .......... 17%  119M 6s\n",
      " 13900K .......... .......... .......... .......... .......... 17%  117M 6s\n",
      " 13950K .......... .......... .......... .......... .......... 17%  118M 6s\n",
      " 14000K .......... .......... .......... .......... .......... 17% 89,7M 6s\n",
      " 14050K .......... .......... .......... .......... .......... 17%  118M 6s\n",
      " 14100K .......... .......... .......... .......... .......... 17%  119M 6s\n",
      " 14150K .......... .......... .......... .......... .......... 17%  117M 6s\n",
      " 14200K .......... .......... .......... .......... .......... 17% 62,6M 6s\n",
      " 14250K .......... .......... .......... .......... .......... 17%  114M 6s\n",
      " 14300K .......... .......... .......... .......... .......... 17% 77,8M 6s\n",
      " 14350K .......... .......... .......... .......... .......... 18%  118M 6s\n",
      " 14400K .......... .......... .......... .......... .......... 18% 89,3M 6s\n",
      " 14450K .......... .......... .......... .......... .......... 18%  118M 6s\n",
      " 14500K .......... .......... .......... .......... .......... 18%  117M 6s\n",
      " 14550K .......... .......... .......... .......... .......... 18%  117M 6s\n",
      " 14600K .......... .......... .......... .......... .......... 18%  113M 6s\n",
      " 14650K .......... .......... .......... .......... .......... 18%  115M 6s\n",
      " 14700K .......... .......... .......... .......... .......... 18%  118M 6s\n",
      " 14750K .......... .......... .......... .......... .......... 18%  118M 6s\n",
      " 14800K .......... .......... .......... .......... .......... 18% 88,0M 6s\n",
      " 14850K .......... .......... .......... .......... .......... 18%  116M 6s\n",
      " 14900K .......... .......... .......... .......... .......... 18%  118M 6s\n",
      " 14950K .......... .......... .......... .......... .......... 18%  118M 6s\n",
      " 15000K .......... .......... .......... .......... .......... 18%  113M 6s\n",
      " 15050K .......... .......... .......... .......... .......... 18%  114M 6s\n",
      " 15100K .......... .......... .......... .......... .......... 18%  119M 6s\n",
      " 15150K .......... .......... .......... .......... .......... 19%  117M 6s\n",
      " 15200K .......... .......... .......... .......... .......... 19% 89,0M 6s\n",
      " 15250K .......... .......... .......... .......... .......... 19%  117M 6s\n",
      " 15300K .......... .......... .......... .......... .......... 19%  118M 6s\n",
      " 15350K .......... .......... .......... .......... .......... 19% 10,0M 6s\n",
      " 15400K .......... .......... .......... .......... .......... 19%  567K 6s\n",
      " 15450K .......... .......... .......... .......... .......... 19% 23,4M 6s\n",
      " 15500K .......... .......... .......... .......... .......... 19% 38,7M 6s\n",
      " 15550K .......... .......... .......... .......... .......... 19% 88,6M 6s\n",
      " 15600K .......... .......... .......... .......... .......... 19% 92,7M 6s\n",
      " 15650K .......... .......... .......... .......... .......... 19%  115M 6s\n",
      " 15700K .......... .......... .......... .......... .......... 19%  121M 6s\n",
      " 15750K .......... .......... .......... .......... .......... 19%  119M 6s\n",
      " 15800K .......... .......... .......... .......... .......... 19%  107M 6s\n",
      " 15850K .......... .......... .......... .......... .......... 19%  115M 6s\n",
      " 15900K .......... .......... .......... .......... .......... 19%  112M 6s\n",
      " 15950K .......... .......... .......... .......... .......... 20%  116M 6s\n",
      " 16000K .......... .......... .......... .......... .......... 20%  100M 6s\n",
      " 16050K .......... .......... .......... .......... .......... 20%  111M 6s\n",
      " 16100K .......... .......... .......... .......... .......... 20%  117M 6s\n",
      " 16150K .......... .......... .......... .......... .......... 20%  120M 6s\n",
      " 16200K .......... .......... .......... .......... .......... 20%  108M 6s\n",
      " 16250K .......... .......... .......... .......... .......... 20%  113M 6s\n",
      " 16300K .......... .......... .......... .......... .......... 20%  118M 6s\n",
      " 16350K .......... .......... .......... .......... .......... 20% 73,4M 6s\n",
      " 16400K .......... .......... .......... .......... .......... 20%  102M 6s\n",
      " 16450K .......... .......... .......... .......... .......... 20%  129M 6s\n",
      " 16500K .......... .......... .......... .......... .......... 20%  134M 6s\n",
      " 16550K .......... .......... .......... .......... .......... 20%  133M 6s\n",
      " 16600K .......... .......... .......... .......... .......... 20%  118M 6s\n",
      " 16650K .......... .......... .......... .......... .......... 20%  115M 6s\n",
      " 16700K .......... .......... .......... .......... .......... 20% 70,0M 6s\n",
      " 16750K .......... .......... .......... .......... .......... 21% 75,3M 6s\n",
      " 16800K .......... .......... .......... .......... .......... 21% 95,3M 6s\n",
      " 16850K .......... .......... .......... .......... .......... 21%  115M 6s\n",
      " 16900K .......... .......... .......... .......... .......... 21%  117M 6s\n",
      " 16950K .......... .......... .......... .......... .......... 21%  119M 5s\n",
      " 17000K .......... .......... .......... .......... .......... 21%  108M 5s\n",
      " 17050K .......... .......... .......... .......... .......... 21%  119M 5s\n",
      " 17100K .......... .......... .......... .......... .......... 21%  116M 5s\n",
      " 17150K .......... .......... .......... .......... .......... 21%  117M 5s\n",
      " 17200K .......... .......... .......... .......... .......... 21% 93,8M 5s\n",
      " 17250K .......... .......... .......... .......... .......... 21%  118M 5s\n",
      " 17300K .......... .......... .......... .......... .......... 21%  113M 5s\n",
      " 17350K .......... .......... .......... .......... .......... 21%  120M 5s\n",
      " 17400K .......... .......... .......... .......... .......... 21% 52,2M 5s\n",
      " 17450K .......... .......... .......... .......... .......... 21% 56,5M 5s\n",
      " 17500K .......... .......... .......... .......... .......... 21%  118M 5s\n",
      " 17550K .......... .......... .......... .......... .......... 22%  114M 5s\n",
      " 17600K .......... .......... .......... .......... .......... 22% 92,4M 5s\n",
      " 17650K .......... .......... .......... .......... .......... 22%  118M 5s\n",
      " 17700K .......... .......... .......... .......... .......... 22%  114M 5s\n",
      " 17750K .......... .......... .......... .......... .......... 22%  117M 5s\n",
      " 17800K .......... .......... .......... .......... .......... 22%  111M 5s\n",
      " 17850K .......... .......... .......... .......... .......... 22%  117M 5s\n",
      " 17900K .......... .......... .......... .......... .......... 22%  118M 5s\n",
      " 17950K .......... .......... .......... .......... .......... 22%  116M 5s\n",
      " 18000K .......... .......... .......... .......... .......... 22% 90,9M 5s\n",
      " 18050K .......... .......... .......... .......... .......... 22%  118M 5s\n",
      " 18100K .......... .......... .......... .......... .......... 22%  113M 5s\n",
      " 18150K .......... .......... .......... .......... .......... 22%  117M 5s\n",
      " 18200K .......... .......... .......... .......... .......... 22%  114M 5s\n",
      " 18250K .......... .......... .......... .......... .......... 22%  118M 5s\n",
      " 18300K .......... .......... .......... .......... .......... 22%  117M 5s\n",
      " 18350K .......... .......... .......... .......... .......... 23%  114M 5s\n",
      " 18400K .......... .......... .......... .......... .......... 23% 12,1M 5s\n",
      " 18450K .......... .......... .......... .......... .......... 23%  568K 5s\n",
      " 18500K .......... .......... .......... .......... .......... 23% 17,4M 5s\n",
      " 18550K .......... .......... .......... .......... .......... 23% 28,4M 5s\n",
      " 18600K .......... .......... .......... .......... .......... 23% 53,1M 5s\n",
      " 18650K .......... .......... .......... .......... .......... 23%  116M 5s\n",
      " 18700K .......... .......... .......... .......... .......... 23%  128M 5s\n",
      " 18750K .......... .......... .......... .......... .......... 23%  127M 5s\n",
      " 18800K .......... .......... .......... .......... .......... 23% 98,1M 5s\n",
      " 18850K .......... .......... .......... .......... .......... 23%  114M 5s\n",
      " 18900K .......... .......... .......... .......... .......... 23%  117M 5s\n",
      " 18950K .......... .......... .......... .......... .......... 23%  117M 5s\n",
      " 19000K .......... .......... .......... .......... .......... 23%  103M 5s\n",
      " 19050K .......... .......... .......... .......... .......... 23%  118M 5s\n",
      " 19100K .......... .......... .......... .......... .......... 23%  115M 5s\n",
      " 19150K .......... .......... .......... .......... .......... 24%  119M 5s\n",
      " 19200K .......... .......... .......... .......... .......... 24% 92,7M 5s\n",
      " 19250K .......... .......... .......... .......... .......... 24%  123M 5s\n",
      " 19300K .......... .......... .......... .......... .......... 24%  114M 5s\n",
      " 19350K .......... .......... .......... .......... .......... 24%  113M 5s\n",
      " 19400K .......... .......... .......... .......... .......... 24%  109M 5s\n",
      " 19450K .......... .......... .......... .......... .......... 24%  118M 5s\n",
      " 19500K .......... .......... .......... .......... .......... 24%  116M 5s\n",
      " 19550K .......... .......... .......... .......... .......... 24%  117M 5s\n",
      " 19600K .......... .......... .......... .......... .......... 24% 94,2M 5s\n",
      " 19650K .......... .......... .......... .......... .......... 24%  118M 5s\n",
      " 19700K .......... .......... .......... .......... .......... 24%  114M 5s\n",
      " 19750K .......... .......... .......... .......... .......... 24%  118M 5s\n",
      " 19800K .......... .......... .......... .......... .......... 24%  104M 5s\n",
      " 19850K .......... .......... .......... .......... .......... 24%  119M 5s\n",
      " 19900K .......... .......... .......... .......... .......... 24%  119M 5s\n",
      " 19950K .......... .......... .......... .......... .......... 25%  113M 5s\n",
      " 20000K .......... .......... .......... .......... .......... 25% 95,3M 5s\n",
      " 20050K .......... .......... .......... .......... .......... 25%  118M 5s\n",
      " 20100K .......... .......... .......... .......... .......... 25%  121M 5s\n",
      " 20150K .......... .......... .......... .......... .......... 25%  112M 5s\n",
      " 20200K .......... .......... .......... .......... .......... 25%  109M 5s\n",
      " 20250K .......... .......... .......... .......... .......... 25%  118M 5s\n",
      " 20300K .......... .......... .......... .......... .......... 25%  115M 5s\n",
      " 20350K .......... .......... .......... .......... .......... 25%  125M 5s\n",
      " 20400K .......... .......... .......... .......... .......... 25% 91,2M 5s\n",
      " 20450K .......... .......... .......... .......... .......... 25%  116M 5s\n",
      " 20500K .......... .......... .......... .......... .......... 25%  120M 5s\n",
      " 20550K .......... .......... .......... .......... .......... 25% 92,1M 5s\n",
      " 20600K .......... .......... .......... .......... .......... 25% 62,3M 5s\n",
      " 20650K .......... .......... .......... .......... .......... 25%  118M 5s\n",
      " 20700K .......... .......... .......... .......... .......... 25% 87,3M 5s\n",
      " 20750K .......... .......... .......... .......... .......... 26%  118M 5s\n",
      " 20800K .......... .......... .......... .......... .......... 26% 88,3M 5s\n",
      " 20850K .......... .......... .......... .......... .......... 26%  118M 5s\n",
      " 20900K .......... .......... .......... .......... .......... 26%  117M 5s\n",
      " 20950K .......... .......... .......... .......... .......... 26%  117M 5s\n",
      " 21000K .......... .......... .......... .......... .......... 26%  110M 5s\n",
      " 21050K .......... .......... .......... .......... .......... 26%  118M 4s\n",
      " 21100K .......... .......... .......... .......... .......... 26%  118M 4s\n",
      " 21150K .......... .......... .......... .......... .......... 26%  118M 4s\n",
      " 21200K .......... .......... .......... .......... .......... 26% 88,2M 4s\n",
      " 21250K .......... .......... .......... .......... .......... 26%  118M 4s\n",
      " 21300K .......... .......... .......... .......... .......... 26%  115M 4s\n",
      " 21350K .......... .......... .......... .......... .......... 26%  119M 4s\n",
      " 21400K .......... .......... .......... .......... .......... 26%  110M 4s\n",
      " 21450K .......... .......... .......... .......... .......... 26% 14,2M 4s\n",
      " 21500K .......... .......... .......... .......... .......... 26%  566K 5s\n",
      " 21550K .......... .......... .......... .......... .......... 27% 9,26M 5s\n",
      " 21600K .......... .......... .......... .......... .......... 27% 53,9M 5s\n",
      " 21650K .......... .......... .......... .......... .......... 27% 66,0M 5s\n",
      " 21700K .......... .......... .......... .......... .......... 27% 86,7M 5s\n",
      " 21750K .......... .......... .......... .......... .......... 27% 86,1M 5s\n",
      " 21800K .......... .......... .......... .......... .......... 27%  114M 5s\n",
      " 21850K .......... .......... .......... .......... .......... 27%  123M 5s\n",
      " 21900K .......... .......... .......... .......... .......... 27%  132M 5s\n",
      " 21950K .......... .......... .......... .......... .......... 27%  126M 5s\n",
      " 22000K .......... .......... .......... .......... .......... 27%  104M 5s\n",
      " 22050K .......... .......... .......... .......... .......... 27%  131M 4s\n",
      " 22100K .......... .......... .......... .......... .......... 27%  130M 4s\n",
      " 22150K .......... .......... .......... .......... .......... 27%  134M 4s\n",
      " 22200K .......... .......... .......... .......... .......... 27%  112M 4s\n",
      " 22250K .......... .......... .......... .......... .......... 27%  128M 4s\n",
      " 22300K .......... .......... .......... .......... .......... 27%  176M 4s\n",
      " 22350K .......... .......... .......... .......... .......... 28%  555M 4s\n",
      " 22400K .......... .......... .......... .......... .......... 28% 88,9M 4s\n",
      " 22450K .......... .......... .......... .......... .......... 28%  117M 4s\n",
      " 22500K .......... .......... .......... .......... .......... 28%  109M 4s\n",
      " 22550K .......... .......... .......... .......... .......... 28%  117M 4s\n",
      " 22600K .......... .......... .......... .......... .......... 28%  106M 4s\n",
      " 22650K .......... .......... .......... .......... .......... 28%  113M 4s\n",
      " 22700K .......... .......... .......... .......... .......... 28%  121M 4s\n",
      " 22750K .......... .......... .......... .......... .......... 28%  117M 4s\n",
      " 22800K .......... .......... .......... .......... .......... 28% 93,4M 4s\n",
      " 22850K .......... .......... .......... .......... .......... 28%  117M 4s\n",
      " 22900K .......... .......... .......... .......... .......... 28%  118M 4s\n",
      " 22950K .......... .......... .......... .......... .......... 28%  118M 4s\n",
      " 23000K .......... .......... .......... .......... .......... 28%  109M 4s\n",
      " 23050K .......... .......... .......... .......... .......... 28%  115M 4s\n",
      " 23100K .......... .......... .......... .......... .......... 28%  116M 4s\n",
      " 23150K .......... .......... .......... .......... .......... 29%  117M 4s\n",
      " 23200K .......... .......... .......... .......... .......... 29% 92,8M 4s\n",
      " 23250K .......... .......... .......... .......... .......... 29%  119M 4s\n",
      " 23300K .......... .......... .......... .......... .......... 29%  118M 4s\n",
      " 23350K .......... .......... .......... .......... .......... 29%  117M 4s\n",
      " 23400K .......... .......... .......... .......... .......... 29% 89,6M 4s\n",
      " 23450K .......... .......... .......... .......... .......... 29%  118M 4s\n",
      " 23500K .......... .......... .......... .......... .......... 29%  114M 4s\n",
      " 23550K .......... .......... .......... .......... .......... 29%  120M 4s\n",
      " 23600K .......... .......... .......... .......... .......... 29% 92,6M 4s\n",
      " 23650K .......... .......... .......... .......... .......... 29%  113M 4s\n",
      " 23700K .......... .......... .......... .......... .......... 29%  117M 4s\n",
      " 23750K .......... .......... .......... .......... .......... 29%  120M 4s\n",
      " 23800K .......... .......... .......... .......... .......... 29%  111M 4s\n",
      " 23850K .......... .......... .......... .......... .......... 29%  117M 4s\n",
      " 23900K .......... .......... .......... .......... .......... 29%  115M 4s\n",
      " 23950K .......... .......... .......... .......... .......... 30%  116M 4s\n",
      " 24000K .......... .......... .......... .......... .......... 30% 92,0M 4s\n",
      " 24050K .......... .......... .......... .......... .......... 30%  115M 4s\n",
      " 24100K .......... .......... .......... .......... .......... 30%  117M 4s\n",
      " 24150K .......... .......... .......... .......... .......... 30%  116M 4s\n",
      " 24200K .......... .......... .......... .......... .......... 30%  115M 4s\n",
      " 24250K .......... .......... .......... .......... .......... 30%  117M 4s\n",
      " 24300K .......... .......... .......... .......... .......... 30%  113M 4s\n",
      " 24350K .......... .......... .......... .......... .......... 30%  117M 4s\n",
      " 24400K .......... .......... .......... .......... .......... 30% 91,0M 4s\n",
      " 24450K .......... .......... .......... .......... .......... 30% 18,2M 4s\n",
      " 24500K .......... .......... .......... .......... .......... 30% 72,2M 4s\n",
      " 24550K .......... .......... .......... .......... .......... 30%  545K 4s\n",
      " 24600K .......... .......... .......... .......... .......... 30% 23,4M 4s\n",
      " 24650K .......... .......... .......... .......... .......... 30% 37,0M 4s\n",
      " 24700K .......... .......... .......... .......... .......... 30%  148M 4s\n",
      " 24750K .......... .......... .......... .......... .......... 31% 51,1M 4s\n",
      " 24800K .......... .......... .......... .......... .......... 31% 66,9M 4s\n",
      " 24850K .......... .......... .......... .......... .......... 31% 90,2M 4s\n",
      " 24900K .......... .......... .......... .......... .......... 31% 85,6M 4s\n",
      " 24950K .......... .......... .......... .......... .......... 31% 92,4M 4s\n",
      " 25000K .......... .......... .......... .......... .......... 31% 90,5M 4s\n",
      " 25050K .......... .......... .......... .......... .......... 31% 90,5M 4s\n",
      " 25100K .......... .......... .......... .......... .......... 31% 98,0M 4s\n",
      " 25150K .......... .......... .......... .......... .......... 31% 90,8M 4s\n",
      " 25200K .......... .......... .......... .......... .......... 31% 76,0M 4s\n",
      " 25250K .......... .......... .......... .......... .......... 31% 89,2M 4s\n",
      " 25300K .......... .......... .......... .......... .......... 31% 95,0M 4s\n",
      " 25350K .......... .......... .......... .......... .......... 31%  110M 4s\n",
      " 25400K .......... .......... .......... .......... .......... 31%  102M 4s\n",
      " 25450K .......... .......... .......... .......... .......... 31%  126M 4s\n",
      " 25500K .......... .......... .......... .......... .......... 31%  123M 4s\n",
      " 25550K .......... .......... .......... .......... .......... 32%  129M 4s\n",
      " 25600K .......... .......... .......... .......... .......... 32%  108M 4s\n",
      " 25650K .......... .......... .......... .......... .......... 32%  144M 4s\n",
      " 25700K .......... .......... .......... .......... .......... 32%  133M 4s\n",
      " 25750K .......... .......... .......... .......... .......... 32%  178M 4s\n",
      " 25800K .......... .......... .......... .......... .......... 32%  156M 4s\n",
      " 25850K .......... .......... .......... .......... .......... 32%  178M 4s\n",
      " 25900K .......... .......... .......... .......... .......... 32%  188M 4s\n",
      " 25950K .......... .......... .......... .......... .......... 32%  211M 4s\n",
      " 26000K .......... .......... .......... .......... .......... 32%  184M 4s\n",
      " 26050K .......... .......... .......... .......... .......... 32%  236M 4s\n",
      " 26100K .......... .......... .......... .......... .......... 32%  121M 4s\n",
      " 26150K .......... .......... .......... .......... .......... 32%  117M 4s\n",
      " 26200K .......... .......... .......... .......... .......... 32%  113M 4s\n",
      " 26250K .......... .......... .......... .......... .......... 32%  118M 4s\n",
      " 26300K .......... .......... .......... .......... .......... 32%  117M 4s\n",
      " 26350K .......... .......... .......... .......... .......... 33%  114M 4s\n",
      " 26400K .......... .......... .......... .......... .......... 33% 92,4M 4s\n",
      " 26450K .......... .......... .......... .......... .......... 33%  116M 4s\n",
      " 26500K .......... .......... .......... .......... .......... 33%  117M 4s\n",
      " 26550K .......... .......... .......... .......... .......... 33%  115M 4s\n",
      " 26600K .......... .......... .......... .......... .......... 33%  113M 4s\n",
      " 26650K .......... .......... .......... .......... .......... 33%  117M 4s\n",
      " 26700K .......... .......... .......... .......... .......... 33%  120M 4s\n",
      " 26750K .......... .......... .......... .......... .......... 33%  117M 4s\n",
      " 26800K .......... .......... .......... .......... .......... 33% 88,7M 4s\n",
      " 26850K .......... .......... .......... .......... .......... 33%  118M 4s\n",
      " 26900K .......... .......... .......... .......... .......... 33%  118M 4s\n",
      " 26950K .......... .......... .......... .......... .......... 33%  114M 4s\n",
      " 27000K .......... .......... .......... .......... .......... 33%  114M 4s\n",
      " 27050K .......... .......... .......... .......... .......... 33%  118M 4s\n",
      " 27100K .......... .......... .......... .......... .......... 33%  116M 4s\n",
      " 27150K .......... .......... .......... .......... .......... 34%  118M 4s\n",
      " 27200K .......... .......... .......... .......... .......... 34% 88,7M 4s\n",
      " 27250K .......... .......... .......... .......... .......... 34%  117M 4s\n",
      " 27300K .......... .......... .......... .......... .......... 34%  117M 4s\n",
      " 27350K .......... .......... .......... .......... .......... 34%  114M 4s\n",
      " 27400K .......... .......... .......... .......... .......... 34%  114M 4s\n",
      " 27450K .......... .......... .......... .......... .......... 34%  117M 4s\n",
      " 27500K .......... .......... .......... .......... .......... 34% 36,4M 4s\n",
      " 27550K .......... .......... .......... .......... .......... 34%  562K 4s\n",
      " 27600K .......... .......... .......... .......... .......... 34% 9,21M 4s\n",
      " 27650K .......... .......... .......... .......... .......... 34% 98,4M 4s\n",
      " 27700K .......... .......... .......... .......... .......... 34% 38,0M 4s\n",
      " 27750K .......... .......... .......... .......... .......... 34% 30,4M 4s\n",
      " 27800K .......... .......... .......... .......... .......... 34% 41,9M 4s\n",
      " 27850K .......... .......... .......... .......... .......... 34%  141M 4s\n",
      " 27900K .......... .......... .......... .......... .......... 34% 52,4M 4s\n",
      " 27950K .......... .......... .......... .......... .......... 35% 76,9M 4s\n",
      " 28000K .......... .......... .......... .......... .......... 35% 72,2M 4s\n",
      " 28050K .......... .......... .......... .......... .......... 35% 88,7M 4s\n",
      " 28100K .......... .......... .......... .......... .......... 35% 93,2M 4s\n",
      " 28150K .......... .......... .......... .......... .......... 35% 92,4M 4s\n",
      " 28200K .......... .......... .......... .......... .......... 35% 81,6M 4s\n",
      " 28250K .......... .......... .......... .......... .......... 35% 94,5M 4s\n",
      " 28300K .......... .......... .......... .......... .......... 35% 95,7M 4s\n",
      " 28350K .......... .......... .......... .......... .......... 35% 88,7M 4s\n",
      " 28400K .......... .......... .......... .......... .......... 35% 76,7M 4s\n",
      " 28450K .......... .......... .......... .......... .......... 35% 94,6M 4s\n",
      " 28500K .......... .......... .......... .......... .......... 35% 95,0M 4s\n",
      " 28550K .......... .......... .......... .......... .......... 35% 92,5M 4s\n",
      " 28600K .......... .......... .......... .......... .......... 35% 86,9M 4s\n",
      " 28650K .......... .......... .......... .......... .......... 35% 66,6M 4s\n",
      " 28700K .......... .......... .......... .......... .......... 35% 92,8M 4s\n",
      " 28750K .......... .......... .......... .......... .......... 36% 99,2M 4s\n",
      " 28800K .......... .......... .......... .......... .......... 36% 95,7M 3s\n",
      " 28850K .......... .......... .......... .......... .......... 36%  116M 3s\n",
      " 28900K .......... .......... .......... .......... .......... 36%  119M 3s\n",
      " 28950K .......... .......... .......... .......... .......... 36%  124M 3s\n",
      " 29000K .......... .......... .......... .......... .......... 36%  120M 3s\n",
      " 29050K .......... .......... .......... .......... .......... 36%  144M 3s\n",
      " 29100K .......... .......... .......... .......... .......... 36%  152M 3s\n",
      " 29150K .......... .......... .......... .......... .......... 36%  148M 3s\n",
      " 29200K .......... .......... .......... .......... .......... 36%  143M 3s\n",
      " 29250K .......... .......... .......... .......... .......... 36%  176M 3s\n",
      " 29300K .......... .......... .......... .......... .......... 36%  195M 3s\n",
      " 29350K .......... .......... .......... .......... .......... 36%  210M 3s\n",
      " 29400K .......... .......... .......... .......... .......... 36%  180M 3s\n",
      " 29450K .......... .......... .......... .......... .......... 36%  204M 3s\n",
      " 29500K .......... .......... .......... .......... .......... 36%  216M 3s\n",
      " 29550K .......... .......... .......... .......... .......... 37%  251M 3s\n",
      " 29600K .......... .......... .......... .......... .......... 37%  211M 3s\n",
      " 29650K .......... .......... .......... .......... .......... 37%  236M 3s\n",
      " 29700K .......... .......... .......... .......... .......... 37%  222M 3s\n",
      " 29750K .......... .......... .......... .......... .......... 37%  164M 3s\n",
      " 29800K .......... .......... .......... .......... .......... 37%  112M 3s\n",
      " 29850K .......... .......... .......... .......... .......... 37%  112M 3s\n",
      " 29900K .......... .......... .......... .......... .......... 37%  119M 3s\n",
      " 29950K .......... .......... .......... .......... .......... 37%  118M 3s\n",
      " 30000K .......... .......... .......... .......... .......... 37% 90,0M 3s\n",
      " 30050K .......... .......... .......... .......... .......... 37%  117M 3s\n",
      " 30100K .......... .......... .......... .......... .......... 37%  116M 3s\n",
      " 30150K .......... .......... .......... .......... .......... 37%  118M 3s\n",
      " 30200K .......... .......... .......... .......... .......... 37%  114M 3s\n",
      " 30250K .......... .......... .......... .......... .......... 37%  114M 3s\n",
      " 30300K .......... .......... .......... .......... .......... 37%  118M 3s\n",
      " 30350K .......... .......... .......... .......... .......... 38%  118M 3s\n",
      " 30400K .......... .......... .......... .......... .......... 38% 88,6M 3s\n",
      " 30450K .......... .......... .......... .......... .......... 38%  117M 3s\n",
      " 30500K .......... .......... .......... .......... .......... 38%  118M 3s\n",
      " 30550K .......... .......... .......... .......... .......... 38%  117M 3s\n",
      " 30600K .......... .......... .......... .......... .......... 38%  567K 3s\n",
      " 30650K .......... .......... .......... .......... .......... 38% 9,32M 3s\n",
      " 30700K .......... .......... .......... .......... .......... 38% 33,3M 3s\n",
      " 30750K .......... .......... .......... .......... .......... 38% 36,2M 3s\n",
      " 30800K .......... .......... .......... .......... .......... 38% 37,7M 3s\n",
      " 30850K .......... .......... .......... .......... .......... 38%  114M 3s\n",
      " 30900K .......... .......... .......... .......... .......... 38% 26,1M 3s\n",
      " 30950K .......... .......... .......... .......... .......... 38% 97,2M 3s\n",
      " 31000K .......... .......... .......... .......... .......... 38% 84,6M 3s\n",
      " 31050K .......... .......... .......... .......... .......... 38% 69,1M 3s\n",
      " 31100K .......... .......... .......... .......... .......... 38%  112M 3s\n",
      " 31150K .......... .......... .......... .......... .......... 39%  122M 3s\n",
      " 31200K .......... .......... .......... .......... .......... 39% 94,4M 3s\n",
      " 31250K .......... .......... .......... .......... .......... 39%  114M 3s\n",
      " 31300K .......... .......... .......... .......... .......... 39%  117M 3s\n",
      " 31350K .......... .......... .......... .......... .......... 39%  118M 3s\n",
      " 31400K .......... .......... .......... .......... .......... 39%  111M 3s\n",
      " 31450K .......... .......... .......... .......... .......... 39%  117M 3s\n",
      " 31500K .......... .......... .......... .......... .......... 39%  118M 3s\n",
      " 31550K .......... .......... .......... .......... .......... 39%  114M 3s\n",
      " 31600K .......... .......... .......... .......... .......... 39% 94,7M 3s\n",
      " 31650K .......... .......... .......... .......... .......... 39%  114M 3s\n",
      " 31700K .......... .......... .......... .......... .......... 39%  115M 3s\n",
      " 31750K .......... .......... .......... .......... .......... 39%  118M 3s\n",
      " 31800K .......... .......... .......... .......... .......... 39%  113M 3s\n",
      " 31850K .......... .......... .......... .......... .......... 39%  117M 3s\n",
      " 31900K .......... .......... .......... .......... .......... 39%  116M 3s\n",
      " 31950K .......... .......... .......... .......... .......... 40% 69,6M 3s\n",
      " 32000K .......... .......... .......... .......... .......... 40% 90,4M 3s\n",
      " 32050K .......... .......... .......... .......... .......... 40%  114M 3s\n",
      " 32100K .......... .......... .......... .......... .......... 40%  117M 3s\n",
      " 32150K .......... .......... .......... .......... .......... 40%  118M 3s\n",
      " 32200K .......... .......... .......... .......... .......... 40%  114M 3s\n",
      " 32250K .......... .......... .......... .......... .......... 40%  116M 3s\n",
      " 32300K .......... .......... .......... .......... .......... 40%  113M 3s\n",
      " 32350K .......... .......... .......... .......... .......... 40%  118M 3s\n",
      " 32400K .......... .......... .......... .......... .......... 40% 90,6M 3s\n",
      " 32450K .......... .......... .......... .......... .......... 40%  114M 3s\n",
      " 32500K .......... .......... .......... .......... .......... 40%  117M 3s\n",
      " 32550K .......... .......... .......... .......... .......... 40%  117M 3s\n",
      " 32600K .......... .......... .......... .......... .......... 40%  115M 3s\n",
      " 32650K .......... .......... .......... .......... .......... 40%  117M 3s\n",
      " 32700K .......... .......... .......... .......... .......... 40%  117M 3s\n",
      " 32750K .......... .......... .......... .......... .......... 41%  114M 3s\n",
      " 32800K .......... .......... .......... .......... .......... 41% 90,4M 3s\n",
      " 32850K .......... .......... .......... .......... .......... 41%  117M 3s\n",
      " 32900K .......... .......... .......... .......... .......... 41%  114M 3s\n",
      " 32950K .......... .......... .......... .......... .......... 41%  118M 3s\n",
      " 33000K .......... .......... .......... .......... .......... 41%  115M 3s\n",
      " 33050K .......... .......... .......... .......... .......... 41%  117M 3s\n",
      " 33100K .......... .......... .......... .......... .......... 41%  117M 3s\n",
      " 33150K .......... .......... .......... .......... .......... 41%  113M 3s\n",
      " 33200K .......... .......... .......... .......... .......... 41% 90,4M 3s\n",
      " 33250K .......... .......... .......... .......... .......... 41%  117M 3s\n",
      " 33300K .......... .......... .......... .......... .......... 41%  114M 3s\n",
      " 33350K .......... .......... .......... .......... .......... 41%  117M 3s\n",
      " 33400K .......... .......... .......... .......... .......... 41%  114M 3s\n",
      " 33450K .......... .......... .......... .......... .......... 41%  117M 3s\n",
      " 33500K .......... .......... .......... .......... .......... 41%  117M 3s\n",
      " 33550K .......... .......... .......... .......... .......... 42%  115M 3s\n",
      " 33600K .......... .......... .......... .......... .......... 42%  579K 3s\n",
      " 33650K .......... .......... .......... .......... .......... 42% 30,8M 3s\n",
      " 33700K .......... .......... .......... .......... .......... 42% 9,72M 3s\n",
      " 33750K .......... .......... .......... .......... .......... 42% 20,0M 3s\n",
      " 33800K .......... .......... .......... .......... .......... 42%  114M 3s\n",
      " 33850K .......... .......... .......... .......... .......... 42%  122M 3s\n",
      " 33900K .......... .......... .......... .......... .......... 42% 46,5M 3s\n",
      " 33950K .......... .......... .......... .......... .......... 42% 38,6M 3s\n",
      " 34000K .......... .......... .......... .......... .......... 42% 38,5M 3s\n",
      " 34050K .......... .......... .......... .......... .......... 42%  125M 3s\n",
      " 34100K .......... .......... .......... .......... .......... 42%  134M 3s\n",
      " 34150K .......... .......... .......... .......... .......... 42% 42,8M 3s\n",
      " 34200K .......... .......... .......... .......... .......... 42%  111M 3s\n",
      " 34250K .......... .......... .......... .......... .......... 42%  125M 3s\n",
      " 34300K .......... .......... .......... .......... .......... 42%  129M 3s\n",
      " 34350K .......... .......... .......... .......... .......... 43%  126M 3s\n",
      " 34400K .......... .......... .......... .......... .......... 43% 93,7M 3s\n",
      " 34450K .......... .......... .......... .......... .......... 43%  115M 3s\n",
      " 34500K .......... .......... .......... .......... .......... 43%  117M 3s\n",
      " 34550K .......... .......... .......... .......... .......... 43%  117M 3s\n",
      " 34600K .......... .......... .......... .......... .......... 43%  106M 3s\n",
      " 34650K .......... .......... .......... .......... .......... 43%  115M 3s\n",
      " 34700K .......... .......... .......... .......... .......... 43%  115M 3s\n",
      " 34750K .......... .......... .......... .......... .......... 43%  123M 3s\n",
      " 34800K .......... .......... .......... .......... .......... 43% 72,1M 3s\n",
      " 34850K .......... .......... .......... .......... .......... 43%  156M 3s\n",
      " 34900K .......... .......... .......... .......... .......... 43%  136M 3s\n",
      " 34950K .......... .......... .......... .......... .......... 43%  114M 3s\n",
      " 35000K .......... .......... .......... .......... .......... 43%  107M 3s\n",
      " 35050K .......... .......... .......... .......... .......... 43%  121M 3s\n",
      " 35100K .......... .......... .......... .......... .......... 43%  119M 3s\n",
      " 35150K .......... .......... .......... .......... .......... 44%  117M 3s\n",
      " 35200K .......... .......... .......... .......... .......... 44% 90,3M 3s\n",
      " 35250K .......... .......... .......... .......... .......... 44%  119M 3s\n",
      " 35300K .......... .......... .......... .......... .......... 44%  117M 3s\n",
      " 35350K .......... .......... .......... .......... .......... 44%  117M 3s\n",
      " 35400K .......... .......... .......... .......... .......... 44%  113M 3s\n",
      " 35450K .......... .......... .......... .......... .......... 44%  116M 3s\n",
      " 35500K .......... .......... .......... .......... .......... 44%  117M 3s\n",
      " 35550K .......... .......... .......... .......... .......... 44%  118M 3s\n",
      " 35600K .......... .......... .......... .......... .......... 44% 90,0M 3s\n",
      " 35650K .......... .......... .......... .......... .......... 44%  119M 3s\n",
      " 35700K .......... .......... .......... .......... .......... 44%  118M 3s\n",
      " 35750K .......... .......... .......... .......... .......... 44%  117M 3s\n",
      " 35800K .......... .......... .......... .......... .......... 44%  109M 3s\n",
      " 35850K .......... .......... .......... .......... .......... 44%  118M 3s\n",
      " 35900K .......... .......... .......... .......... .......... 44%  118M 3s\n",
      " 35950K .......... .......... .......... .......... .......... 45%  117M 3s\n",
      " 36000K .......... .......... .......... .......... .......... 45% 88,9M 3s\n",
      " 36050K .......... .......... .......... .......... .......... 45%  117M 3s\n",
      " 36100K .......... .......... .......... .......... .......... 45%  117M 3s\n",
      " 36150K .......... .......... .......... .......... .......... 45%  117M 3s\n",
      " 36200K .......... .......... .......... .......... .......... 45%  111M 3s\n",
      " 36250K .......... .......... .......... .......... .......... 45%  117M 3s\n",
      " 36300K .......... .......... .......... .......... .......... 45%  118M 3s\n",
      " 36350K .......... .......... .......... .......... .......... 45%  119M 3s\n",
      " 36400K .......... .......... .......... .......... .......... 45% 88,0M 3s\n",
      " 36450K .......... .......... .......... .......... .......... 45%  117M 3s\n",
      " 36500K .......... .......... .......... .......... .......... 45%  117M 3s\n",
      " 36550K .......... .......... .......... .......... .......... 45%  117M 3s\n",
      " 36600K .......... .......... .......... .......... .......... 45%  112M 3s\n",
      " 36650K .......... .......... .......... .......... .......... 45%  116M 3s\n",
      " 36700K .......... .......... .......... .......... .......... 45%  580K 3s\n",
      " 36750K .......... .......... .......... .......... .......... 46% 9,63M 3s\n",
      " 36800K .......... .......... .......... .......... .......... 46% 12,8M 3s\n",
      " 36850K .......... .......... .......... .......... .......... 46% 72,7M 3s\n",
      " 36900K .......... .......... .......... .......... .......... 46%  151M 3s\n",
      " 36950K .......... .......... .......... .......... .......... 46%  120M 3s\n",
      " 37000K .......... .......... .......... .......... .......... 46% 24,8M 3s\n",
      " 37050K .......... .......... .......... .......... .......... 46% 61,5M 3s\n",
      " 37100K .......... .......... .......... .......... .......... 46% 60,5M 3s\n",
      " 37150K .......... .......... .......... .......... .......... 46%  142M 3s\n",
      " 37200K .......... .......... .......... .......... .......... 46% 76,4M 3s\n",
      " 37250K .......... .......... .......... .......... .......... 46%  117M 3s\n",
      " 37300K .......... .......... .......... .......... .......... 46%  158M 3s\n",
      " 37350K .......... .......... .......... .......... .......... 46%  117M 3s\n",
      " 37400K .......... .......... .......... .......... .......... 46%  111M 3s\n",
      " 37450K .......... .......... .......... .......... .......... 46%  112M 3s\n",
      " 37500K .......... .......... .......... .......... .......... 46%  121M 3s\n",
      " 37550K .......... .......... .......... .......... .......... 47% 72,2M 3s\n",
      " 37600K .......... .......... .......... .......... .......... 47% 92,1M 3s\n",
      " 37650K .......... .......... .......... .......... .......... 47%  117M 3s\n",
      " 37700K .......... .......... .......... .......... .......... 47%  118M 3s\n",
      " 37750K .......... .......... .......... .......... .......... 47%  117M 3s\n",
      " 37800K .......... .......... .......... .......... .......... 47%  110M 3s\n",
      " 37850K .......... .......... .......... .......... .......... 47%  117M 3s\n",
      " 37900K .......... .......... .......... .......... .......... 47%  118M 3s\n",
      " 37950K .......... .......... .......... .......... .......... 47%  119M 3s\n",
      " 38000K .......... .......... .......... .......... .......... 47% 89,9M 3s\n",
      " 38050K .......... .......... .......... .......... .......... 47%  117M 3s\n",
      " 38100K .......... .......... .......... .......... .......... 47%  118M 3s\n",
      " 38150K .......... .......... .......... .......... .......... 47%  117M 3s\n",
      " 38200K .......... .......... .......... .......... .......... 47%  115M 3s\n",
      " 38250K .......... .......... .......... .......... .......... 47%  114M 3s\n",
      " 38300K .......... .......... .......... .......... .......... 47%  116M 3s\n",
      " 38350K .......... .......... .......... .......... .......... 48%  117M 3s\n",
      " 38400K .......... .......... .......... .......... .......... 48% 89,0M 3s\n",
      " 38450K .......... .......... .......... .......... .......... 48%  118M 3s\n",
      " 38500K .......... .......... .......... .......... .......... 48%  117M 3s\n",
      " 38550K .......... .......... .......... .......... .......... 48%  118M 3s\n",
      " 38600K .......... .......... .......... .......... .......... 48%  114M 3s\n",
      " 38650K .......... .......... .......... .......... .......... 48%  117M 3s\n",
      " 38700K .......... .......... .......... .......... .......... 48%  114M 2s\n",
      " 38750K .......... .......... .......... .......... .......... 48%  117M 2s\n",
      " 38800K .......... .......... .......... .......... .......... 48% 91,1M 2s\n",
      " 38850K .......... .......... .......... .......... .......... 48%  114M 2s\n",
      " 38900K .......... .......... .......... .......... .......... 48%  118M 2s\n",
      " 38950K .......... .......... .......... .......... .......... 48%  117M 2s\n",
      " 39000K .......... .......... .......... .......... .......... 48%  114M 2s\n",
      " 39050K .......... .......... .......... .......... .......... 48%  117M 2s\n",
      " 39100K .......... .......... .......... .......... .......... 48%  115M 2s\n",
      " 39150K .......... .......... .......... .......... .......... 49%  116M 2s\n",
      " 39200K .......... .......... .......... .......... .......... 49% 90,7M 2s\n",
      " 39250K .......... .......... .......... .......... .......... 49%  114M 2s\n",
      " 39300K .......... .......... .......... .......... .......... 49%  117M 2s\n",
      " 39350K .......... .......... .......... .......... .......... 49%  117M 2s\n",
      " 39400K .......... .......... .......... .......... .......... 49%  115M 2s\n",
      " 39450K .......... .......... .......... .......... .......... 49%  116M 2s\n",
      " 39500K .......... .......... .......... .......... .......... 49%  116M 2s\n",
      " 39550K .......... .......... .......... .......... .......... 49%  117M 2s\n",
      " 39600K .......... .......... .......... .......... .......... 49% 90,0M 2s\n",
      " 39650K .......... .......... .......... .......... .......... 49%  114M 2s\n",
      " 39700K .......... .......... .......... .......... .......... 49%  117M 2s\n",
      " 39750K .......... .......... .......... .......... .......... 49%  564K 2s\n",
      " 39800K .......... .......... .......... .......... .......... 49% 14,9M 2s\n",
      " 39850K .......... .......... .......... .......... .......... 49% 43,1M 2s\n",
      " 39900K .......... .......... .......... .......... .......... 49% 23,9M 2s\n",
      " 39950K .......... .......... .......... .......... .......... 50% 45,6M 2s\n",
      " 40000K .......... .......... .......... .......... .......... 50% 23,0M 2s\n",
      " 40050K .......... .......... .......... .......... .......... 50% 72,7M 2s\n",
      " 40100K .......... .......... .......... .......... .......... 50%  113M 2s\n",
      " 40150K .......... .......... .......... .......... .......... 50% 71,5M 2s\n",
      " 40200K .......... .......... .......... .......... .......... 50% 47,7M 2s\n",
      " 40250K .......... .......... .......... .......... .......... 50%  126M 2s\n",
      " 40300K .......... .......... .......... .......... .......... 50%  133M 2s\n",
      " 40350K .......... .......... .......... .......... .......... 50%  131M 2s\n",
      " 40400K .......... .......... .......... .......... .......... 50% 93,2M 2s\n",
      " 40450K .......... .......... .......... .......... .......... 50%  125M 2s\n",
      " 40500K .......... .......... .......... .......... .......... 50%  115M 2s\n",
      " 40550K .......... .......... .......... .......... .......... 50%  118M 2s\n",
      " 40600K .......... .......... .......... .......... .......... 50%  103M 2s\n",
      " 40650K .......... .......... .......... .......... .......... 50%  119M 2s\n",
      " 40700K .......... .......... .......... .......... .......... 50%  119M 2s\n",
      " 40750K .......... .......... .......... .......... .......... 51%  110M 2s\n",
      " 40800K .......... .......... .......... .......... .......... 51% 97,5M 2s\n",
      " 40850K .......... .......... .......... .......... .......... 51%  119M 2s\n",
      " 40900K .......... .......... .......... .......... .......... 51%  111M 2s\n",
      " 40950K .......... .......... .......... .......... .......... 51% 82,0M 2s\n",
      " 41000K .......... .......... .......... .......... .......... 51%  113M 2s\n",
      " 41050K .......... .......... .......... .......... .......... 51%  137M 2s\n",
      " 41100K .......... .......... .......... .......... .......... 51%  132M 2s\n",
      " 41150K .......... .......... .......... .......... .......... 51%  127M 2s\n",
      " 41200K .......... .......... .......... .......... .......... 51% 99,3M 2s\n",
      " 41250K .......... .......... .......... .......... .......... 51%  113M 2s\n",
      " 41300K .......... .......... .......... .......... .......... 51%  118M 2s\n",
      " 41350K .......... .......... .......... .......... .......... 51%  115M 2s\n",
      " 41400K .......... .......... .......... .......... .......... 51%  107M 2s\n",
      " 41450K .......... .......... .......... .......... .......... 51%  119M 2s\n",
      " 41500K .......... .......... .......... .......... .......... 51%  121M 2s\n",
      " 41550K .......... .......... .......... .......... .......... 52%  114M 2s\n",
      " 41600K .......... .......... .......... .......... .......... 52% 94,4M 2s\n",
      " 41650K .......... .......... .......... .......... .......... 52%  117M 2s\n",
      " 41700K .......... .......... .......... .......... .......... 52%  113M 2s\n",
      " 41750K .......... .......... .......... .......... .......... 52%  119M 2s\n",
      " 41800K .......... .......... .......... .......... .......... 52%  111M 2s\n",
      " 41850K .......... .......... .......... .......... .......... 52%  117M 2s\n",
      " 41900K .......... .......... .......... .......... .......... 52%  119M 2s\n",
      " 41950K .......... .......... .......... .......... .......... 52%  117M 2s\n",
      " 42000K .......... .......... .......... .......... .......... 52% 90,9M 2s\n",
      " 42050K .......... .......... .......... .......... .......... 52%  117M 2s\n",
      " 42100K .......... .......... .......... .......... .......... 52%  118M 2s\n",
      " 42150K .......... .......... .......... .......... .......... 52%  115M 2s\n",
      " 42200K .......... .......... .......... .......... .......... 52%  113M 2s\n",
      " 42250K .......... .......... .......... .......... .......... 52%  117M 2s\n",
      " 42300K .......... .......... .......... .......... .......... 52%  118M 2s\n",
      " 42350K .......... .......... .......... .......... .......... 53%  117M 2s\n",
      " 42400K .......... .......... .......... .......... .......... 53% 90,2M 2s\n",
      " 42450K .......... .......... .......... .......... .......... 53%  117M 2s\n",
      " 42500K .......... .......... .......... .......... .......... 53%  117M 2s\n",
      " 42550K .......... .......... .......... .......... .......... 53%  113M 2s\n",
      " 42600K .......... .......... .......... .......... .......... 53%  114M 2s\n",
      " 42650K .......... .......... .......... .......... .......... 53%  119M 2s\n",
      " 42700K .......... .......... .......... .......... .......... 53%  117M 2s\n",
      " 42750K .......... .......... .......... .......... .......... 53%  118M 2s\n",
      " 42800K .......... .......... .......... .......... .......... 53%  573K 2s\n",
      " 42850K .......... .......... .......... .......... .......... 53% 12,7M 2s\n",
      " 42900K .......... .......... .......... .......... .......... 53% 18,6M 2s\n",
      " 42950K .......... .......... .......... .......... .......... 53%  110M 2s\n",
      " 43000K .......... .......... .......... .......... .......... 53% 24,8M 2s\n",
      " 43050K .......... .......... .......... .......... .......... 53% 30,4M 2s\n",
      " 43100K .......... .......... .......... .......... .......... 53% 43,8M 2s\n",
      " 43150K .......... .......... .......... .......... .......... 54% 58,7M 2s\n",
      " 43200K .......... .......... .......... .......... .......... 54% 66,8M 2s\n",
      " 43250K .......... .......... .......... .......... .......... 54% 60,9M 2s\n",
      " 43300K .......... .......... .......... .......... .......... 54%  138M 2s\n",
      " 43350K .......... .......... .......... .......... .......... 54%  144M 2s\n",
      " 43400K .......... .......... .......... .......... .......... 54%  118M 2s\n",
      " 43450K .......... .......... .......... .......... .......... 54%  130M 2s\n",
      " 43500K .......... .......... .......... .......... .......... 54%  125M 2s\n",
      " 43550K .......... .......... .......... .......... .......... 54%  133M 2s\n",
      " 43600K .......... .......... .......... .......... .......... 54%  110M 2s\n",
      " 43650K .......... .......... .......... .......... .......... 54%  134M 2s\n",
      " 43700K .......... .......... .......... .......... .......... 54%  120M 2s\n",
      " 43750K .......... .......... .......... .......... .......... 54%  124M 2s\n",
      " 43800K .......... .......... .......... .......... .......... 54%  106M 2s\n",
      " 43850K .......... .......... .......... .......... .......... 54%  107M 2s\n",
      " 43900K .......... .......... .......... .......... .......... 54%  120M 2s\n",
      " 43950K .......... .......... .......... .......... .......... 55%  119M 2s\n",
      " 44000K .......... .......... .......... .......... .......... 55% 95,4M 2s\n",
      " 44050K .......... .......... .......... .......... .......... 55%  118M 2s\n",
      " 44100K .......... .......... .......... .......... .......... 55%  116M 2s\n",
      " 44150K .......... .......... .......... .......... .......... 55%  116M 2s\n",
      " 44200K .......... .......... .......... .......... .......... 55%  106M 2s\n",
      " 44250K .......... .......... .......... .......... .......... 55%  117M 2s\n",
      " 44300K .......... .......... .......... .......... .......... 55%  123M 2s\n",
      " 44350K .......... .......... .......... .......... .......... 55%  117M 2s\n",
      " 44400K .......... .......... .......... .......... .......... 55% 91,9M 2s\n",
      " 44450K .......... .......... .......... .......... .......... 55%  116M 2s\n",
      " 44500K .......... .......... .......... .......... .......... 55%  119M 2s\n",
      " 44550K .......... .......... .......... .......... .......... 55%  119M 2s\n",
      " 44600K .......... .......... .......... .......... .......... 55%  108M 2s\n",
      " 44650K .......... .......... .......... .......... .......... 55%  118M 2s\n",
      " 44700K .......... .......... .......... .......... .......... 55%  117M 2s\n",
      " 44750K .......... .......... .......... .......... .......... 56%  118M 2s\n",
      " 44800K .......... .......... .......... .......... .......... 56% 90,8M 2s\n",
      " 44850K .......... .......... .......... .......... .......... 56%  120M 2s\n",
      " 44900K .......... .......... .......... .......... .......... 56%  117M 2s\n",
      " 44950K .......... .......... .......... .......... .......... 56%  117M 2s\n",
      " 45000K .......... .......... .......... .......... .......... 56%  114M 2s\n",
      " 45050K .......... .......... .......... .......... .......... 56%  114M 2s\n",
      " 45100K .......... .......... .......... .......... .......... 56%  117M 2s\n",
      " 45150K .......... .......... .......... .......... .......... 56% 12,9M 2s\n",
      " 45200K .......... .......... .......... .......... .......... 56%  635M 2s\n",
      " 45250K .......... .......... .......... .......... .......... 56%  658M 2s\n",
      " 45300K .......... .......... .......... .......... .......... 56%  703M 2s\n",
      " 45350K .......... .......... .......... .......... .......... 56%  735M 2s\n",
      " 45400K .......... .......... .......... .......... .......... 56%  585M 2s\n",
      " 45450K .......... .......... .......... .......... .......... 56%  687M 2s\n",
      " 45500K .......... .......... .......... .......... .......... 56%  645M 2s\n",
      " 45550K .......... .......... .......... .......... .......... 57%  743M 2s\n",
      " 45600K .......... .......... .......... .......... .......... 57%  526M 2s\n",
      " 45650K .......... .......... .......... .......... .......... 57%  117M 2s\n",
      " 45700K .......... .......... .......... .......... .......... 57%  117M 2s\n",
      " 45750K .......... .......... .......... .......... .......... 57%  117M 2s\n",
      " 45800K .......... .......... .......... .......... .......... 57%  115M 2s\n",
      " 45850K .......... .......... .......... .......... .......... 57%  579K 2s\n",
      " 45900K .......... .......... .......... .......... .......... 57% 12,6M 2s\n",
      " 45950K .......... .......... .......... .......... .......... 57% 23,9M 2s\n",
      " 46000K .......... .......... .......... .......... .......... 57% 15,0M 2s\n",
      " 46050K .......... .......... .......... .......... .......... 57% 38,2M 2s\n",
      " 46100K .......... .......... .......... .......... .......... 57%  118M 2s\n",
      " 46150K .......... .......... .......... .......... .......... 57% 29,1M 2s\n",
      " 46200K .......... .......... .......... .......... .......... 57% 79,1M 2s\n",
      " 46250K .......... .......... .......... .......... .......... 57% 67,2M 2s\n",
      " 46300K .......... .......... .......... .......... .......... 57% 90,4M 2s\n",
      " 46350K .......... .......... .......... .......... .......... 58%  101M 2s\n",
      " 46400K .......... .......... .......... .......... .......... 58% 87,2M 2s\n",
      " 46450K .......... .......... .......... .......... .......... 58% 90,4M 2s\n",
      " 46500K .......... .......... .......... .......... .......... 58% 93,2M 2s\n",
      " 46550K .......... .......... .......... .......... .......... 58% 96,8M 2s\n",
      " 46600K .......... .......... .......... .......... .......... 58% 92,2M 2s\n",
      " 46650K .......... .......... .......... .......... .......... 58% 95,7M 2s\n",
      " 46700K .......... .......... .......... .......... .......... 58% 88,8M 2s\n",
      " 46750K .......... .......... .......... .......... .......... 58% 90,6M 2s\n",
      " 46800K .......... .......... .......... .......... .......... 58% 74,8M 2s\n",
      " 46850K .......... .......... .......... .......... .......... 58% 93,1M 2s\n",
      " 46900K .......... .......... .......... .......... .......... 58% 97,0M 2s\n",
      " 46950K .......... .......... .......... .......... .......... 58% 93,3M 2s\n",
      " 47000K .......... .......... .......... .......... .......... 58% 90,3M 2s\n",
      " 47050K .......... .......... .......... .......... .......... 58%  105M 2s\n",
      " 47100K .......... .......... .......... .......... .......... 58% 75,2M 2s\n",
      " 47150K .......... .......... .......... .......... .......... 59%  111M 2s\n",
      " 47200K .......... .......... .......... .......... .......... 59% 87,4M 2s\n",
      " 47250K .......... .......... .......... .......... .......... 59%  136M 2s\n",
      " 47300K .......... .......... .......... .......... .......... 59%  106M 2s\n",
      " 47350K .......... .......... .......... .......... .......... 59%  136M 2s\n",
      " 47400K .......... .......... .......... .......... .......... 59%  143M 2s\n",
      " 47450K .......... .......... .......... .......... .......... 59%  111M 2s\n",
      " 47500K .......... .......... .......... .......... .......... 59%  166M 2s\n",
      " 47550K .......... .......... .......... .......... .......... 59%  180M 2s\n",
      " 47600K .......... .......... .......... .......... .......... 59%  160M 2s\n",
      " 47650K .......... .......... .......... .......... .......... 59%  220M 2s\n",
      " 47700K .......... .......... .......... .......... .......... 59%  192M 2s\n",
      " 47750K .......... .......... .......... .......... .......... 59%  235M 2s\n",
      " 47800K .......... .......... .......... .......... .......... 59%  150M 2s\n",
      " 47850K .......... .......... .......... .......... .......... 59%  194M 2s\n",
      " 47900K .......... .......... .......... .......... .......... 59%  117M 2s\n",
      " 47950K .......... .......... .......... .......... .......... 60%  115M 2s\n",
      " 48000K .......... .......... .......... .......... .......... 60% 91,1M 2s\n",
      " 48050K .......... .......... .......... .......... .......... 60%  117M 2s\n",
      " 48100K .......... .......... .......... .......... .......... 60%  115M 2s\n",
      " 48150K .......... .......... .......... .......... .......... 60% 17,4M 2s\n",
      " 48200K .......... .......... .......... .......... .......... 60%  112M 2s\n",
      " 48250K .......... .......... .......... .......... .......... 60%  118M 2s\n",
      " 48300K .......... .......... .......... .......... .......... 60%  117M 2s\n",
      " 48350K .......... .......... .......... .......... .......... 60%  114M 2s\n",
      " 48400K .......... .......... .......... .......... .......... 60% 88,7M 2s\n",
      " 48450K .......... .......... .......... .......... .......... 60%  123M 2s\n",
      " 48500K .......... .......... .......... .......... .......... 60%  113M 2s\n",
      " 48550K .......... .......... .......... .......... .......... 60%  118M 2s\n",
      " 48600K .......... .......... .......... .......... .......... 60%  112M 2s\n",
      " 48650K .......... .......... .......... .......... .......... 60%  119M 2s\n",
      " 48700K .......... .......... .......... .......... .......... 60%  118M 2s\n",
      " 48750K .......... .......... .......... .......... .......... 61%  108M 2s\n",
      " 48800K .......... .......... .......... .......... .......... 61% 95,0M 2s\n",
      " 48850K .......... .......... .......... .......... .......... 61% 1,24M 2s\n",
      " 48900K .......... .......... .......... .......... .......... 61% 1,03M 2s\n",
      " 48950K .......... .......... .......... .......... .......... 61% 32,1M 2s\n",
      " 49000K .......... .......... .......... .......... .......... 61% 36,6M 2s\n",
      " 49050K .......... .......... .......... .......... .......... 61% 14,7M 2s\n",
      " 49100K .......... .......... .......... .......... .......... 61% 31,1M 2s\n",
      " 49150K .......... .......... .......... .......... .......... 61% 70,1M 2s\n",
      " 49200K .......... .......... .......... .......... .......... 61% 63,4M 2s\n",
      " 49250K .......... .......... .......... .......... .......... 61% 95,8M 2s\n",
      " 49300K .......... .......... .......... .......... .......... 61% 57,0M 2s\n",
      " 49350K .......... .......... .......... .......... .......... 61% 56,6M 2s\n",
      " 49400K .......... .......... .......... .......... .......... 61% 57,3M 2s\n",
      " 49450K .......... .......... .......... .......... .......... 61%  122M 2s\n",
      " 49500K .......... .......... .......... .......... .......... 61% 59,4M 2s\n",
      " 49550K .......... .......... .......... .......... .......... 62%  119M 2s\n",
      " 49600K .......... .......... .......... .......... .......... 62% 91,4M 2s\n",
      " 49650K .......... .......... .......... .......... .......... 62%  124M 2s\n",
      " 49700K .......... .......... .......... .......... .......... 62%  113M 2s\n",
      " 49750K .......... .......... .......... .......... .......... 62%  114M 2s\n",
      " 49800K .......... .......... .......... .......... .......... 62% 65,8M 2s\n",
      " 49850K .......... .......... .......... .......... .......... 62%  116M 2s\n",
      " 49900K .......... .......... .......... .......... .......... 62%  118M 2s\n",
      " 49950K .......... .......... .......... .......... .......... 62%  117M 2s\n",
      " 50000K .......... .......... .......... .......... .......... 62% 90,8M 2s\n",
      " 50050K .......... .......... .......... .......... .......... 62%  121M 2s\n",
      " 50100K .......... .......... .......... .......... .......... 62%  118M 2s\n",
      " 50150K .......... .......... .......... .......... .......... 62%  113M 2s\n",
      " 50200K .......... .......... .......... .......... .......... 62%  106M 2s\n",
      " 50250K .......... .......... .......... .......... .......... 62% 89,6M 2s\n",
      " 50300K .......... .......... .......... .......... .......... 62%  116M 2s\n",
      " 50350K .......... .......... .......... .......... .......... 63%  119M 2s\n",
      " 50400K .......... .......... .......... .......... .......... 63% 93,6M 2s\n",
      " 50450K .......... .......... .......... .......... .......... 63%  119M 2s\n",
      " 50500K .......... .......... .......... .......... .......... 63%  118M 2s\n",
      " 50550K .......... .......... .......... .......... .......... 63%  113M 2s\n",
      " 50600K .......... .......... .......... .......... .......... 63%  108M 2s\n",
      " 50650K .......... .......... .......... .......... .......... 63%  119M 2s\n",
      " 50700K .......... .......... .......... .......... .......... 63%  115M 2s\n",
      " 50750K .......... .......... .......... .......... .......... 63%  125M 2s\n",
      " 50800K .......... .......... .......... .......... .......... 63% 90,4M 2s\n",
      " 50850K .......... .......... .......... .......... .......... 63%  117M 2s\n",
      " 50900K .......... .......... .......... .......... .......... 63%  118M 2s\n",
      " 50950K .......... .......... .......... .......... .......... 63%  103M 2s\n",
      " 51000K .......... .......... .......... .......... .......... 63%  121M 2s\n",
      " 51050K .......... .......... .......... .......... .......... 63%  117M 2s\n",
      " 51100K .......... .......... .......... .......... .......... 63%  119M 2s\n",
      " 51150K .......... .......... .......... .......... .......... 64%  119M 2s\n",
      " 51200K .......... .......... .......... .......... .......... 64% 18,1M 2s\n",
      " 51250K .......... .......... .......... .......... .......... 64%  115M 2s\n",
      " 51300K .......... .......... .......... .......... .......... 64%  119M 2s\n",
      " 51350K .......... .......... .......... .......... .......... 64%  115M 2s\n",
      " 51400K .......... .......... .......... .......... .......... 64%  109M 2s\n",
      " 51450K .......... .......... .......... .......... .......... 64%  118M 2s\n",
      " 51500K .......... .......... .......... .......... .......... 64%  118M 2s\n",
      " 51550K .......... .......... .......... .......... .......... 64%  116M 2s\n",
      " 51600K .......... .......... .......... .......... .......... 64% 90,2M 2s\n",
      " 51650K .......... .......... .......... .......... .......... 64%  118M 2s\n",
      " 51700K .......... .......... .......... .......... .......... 64%  118M 2s\n",
      " 51750K .......... .......... .......... .......... .......... 64%  117M 2s\n",
      " 51800K .......... .......... .......... .......... .......... 64%  111M 2s\n",
      " 51850K .......... .......... .......... .......... .......... 64%  117M 2s\n",
      " 51900K .......... .......... .......... .......... .......... 64%  612K 2s\n",
      " 51950K .......... .......... .......... .......... .......... 65% 12,6M 2s\n",
      " 52000K .......... .......... .......... .......... .......... 65% 16,1M 2s\n",
      " 52050K .......... .......... .......... .......... .......... 65% 21,5M 2s\n",
      " 52100K .......... .......... .......... .......... .......... 65% 86,5M 2s\n",
      " 52150K .......... .......... .......... .......... .......... 65% 17,1M 2s\n",
      " 52200K .......... .......... .......... .......... .......... 65% 94,2M 2s\n",
      " 52250K .......... .......... .......... .......... .......... 65% 51,6M 2s\n",
      " 52300K .......... .......... .......... .......... .......... 65%  107M 2s\n",
      " 52350K .......... .......... .......... .......... .......... 65% 60,3M 2s\n",
      " 52400K .......... .......... .......... .......... .......... 65% 53,1M 2s\n",
      " 52450K .......... .......... .......... .......... .......... 65% 59,0M 2s\n",
      " 52500K .......... .......... .......... .......... .......... 65% 62,1M 2s\n",
      " 52550K .......... .......... .......... .......... .......... 65% 91,1M 2s\n",
      " 52600K .......... .......... .......... .......... .......... 65% 98,3M 2s\n",
      " 52650K .......... .......... .......... .......... .......... 65%  114M 2s\n",
      " 52700K .......... .......... .......... .......... .......... 65%  114M 2s\n",
      " 52750K .......... .......... .......... .......... .......... 66%  107M 2s\n",
      " 52800K .......... .......... .......... .......... .......... 66% 94,5M 2s\n",
      " 52850K .......... .......... .......... .......... .......... 66%  110M 2s\n",
      " 52900K .......... .......... .......... .......... .......... 66%  105M 2s\n",
      " 52950K .......... .......... .......... .......... .......... 66%  226M 2s\n",
      " 53000K .......... .......... .......... .......... .......... 66%  158M 1s\n",
      " 53050K .......... .......... .......... .......... .......... 66%  115M 1s\n",
      " 53100K .......... .......... .......... .......... .......... 66%  119M 1s\n",
      " 53150K .......... .......... .......... .......... .......... 66%  120M 1s\n",
      " 53200K .......... .......... .......... .......... .......... 66% 87,3M 1s\n",
      " 53250K .......... .......... .......... .......... .......... 66%  122M 1s\n",
      " 53300K .......... .......... .......... .......... .......... 66%  118M 1s\n",
      " 53350K .......... .......... .......... .......... .......... 66%  116M 1s\n",
      " 53400K .......... .......... .......... .......... .......... 66%  113M 1s\n",
      " 53450K .......... .......... .......... .......... .......... 66%  115M 1s\n",
      " 53500K .......... .......... .......... .......... .......... 66%  119M 1s\n",
      " 53550K .......... .......... .......... .......... .......... 67%  118M 1s\n",
      " 53600K .......... .......... .......... .......... .......... 67% 89,4M 1s\n",
      " 53650K .......... .......... .......... .......... .......... 67%  117M 1s\n",
      " 53700K .......... .......... .......... .......... .......... 67%  116M 1s\n",
      " 53750K .......... .......... .......... .......... .......... 67%  117M 1s\n",
      " 53800K .......... .......... .......... .......... .......... 67%  114M 1s\n",
      " 53850K .......... .......... .......... .......... .......... 67%  115M 1s\n",
      " 53900K .......... .......... .......... .......... .......... 67%  117M 1s\n",
      " 53950K .......... .......... .......... .......... .......... 67%  119M 1s\n",
      " 54000K .......... .......... .......... .......... .......... 67% 90,7M 1s\n",
      " 54050K .......... .......... .......... .......... .......... 67%  113M 1s\n",
      " 54100K .......... .......... .......... .......... .......... 67%  118M 1s\n",
      " 54150K .......... .......... .......... .......... .......... 67%  118M 1s\n",
      " 54200K .......... .......... .......... .......... .......... 67%  114M 1s\n",
      " 54250K .......... .......... .......... .......... .......... 67%  117M 1s\n",
      " 54300K .......... .......... .......... .......... .......... 67% 17,9M 1s\n",
      " 54350K .......... .......... .......... .......... .......... 68%  116M 1s\n",
      " 54400K .......... .......... .......... .......... .......... 68% 90,9M 1s\n",
      " 54450K .......... .......... .......... .......... .......... 68%  114M 1s\n",
      " 54500K .......... .......... .......... .......... .......... 68%  118M 1s\n",
      " 54550K .......... .......... .......... .......... .......... 68%  117M 1s\n",
      " 54600K .......... .......... .......... .......... .......... 68% 69,5M 1s\n",
      " 54650K .......... .......... .......... .......... .......... 68%  117M 1s\n",
      " 54700K .......... .......... .......... .......... .......... 68%  114M 1s\n",
      " 54750K .......... .......... .......... .......... .......... 68%  118M 1s\n",
      " 54800K .......... .......... .......... .......... .......... 68% 90,6M 1s\n",
      " 54850K .......... .......... .......... .......... .......... 68%  115M 1s\n",
      " 54900K .......... .......... .......... .......... .......... 68% 1,33M 1s\n",
      " 54950K .......... .......... .......... .......... .......... 68% 1,10M 1s\n",
      " 55000K .......... .......... .......... .......... .......... 68% 9,96M 1s\n",
      " 55050K .......... .......... .......... .......... .......... 68% 23,8M 1s\n",
      " 55100K .......... .......... .......... .......... .......... 68% 29,8M 1s\n",
      " 55150K .......... .......... .......... .......... .......... 69% 15,8M 1s\n",
      " 55200K .......... .......... .......... .......... .......... 69% 37,1M 1s\n",
      " 55250K .......... .......... .......... .......... .......... 69% 40,7M 1s\n",
      " 55300K .......... .......... .......... .......... .......... 69% 93,8M 1s\n",
      " 55350K .......... .......... .......... .......... .......... 69%  115M 1s\n",
      " 55400K .......... .......... .......... .......... .......... 69% 36,2M 1s\n",
      " 55450K .......... .......... .......... .......... .......... 69%  119M 1s\n",
      " 55500K .......... .......... .......... .......... .......... 69%  111M 1s\n",
      " 55550K .......... .......... .......... .......... .......... 69%  107M 1s\n",
      " 55600K .......... .......... .......... .......... .......... 69% 58,4M 1s\n",
      " 55650K .......... .......... .......... .......... .......... 69% 58,7M 1s\n",
      " 55700K .......... .......... .......... .......... .......... 69%  113M 1s\n",
      " 55750K .......... .......... .......... .......... .......... 69%  118M 1s\n",
      " 55800K .......... .......... .......... .......... .......... 69%  107M 1s\n",
      " 55850K .......... .......... .......... .......... .......... 69%  117M 1s\n",
      " 55900K .......... .......... .......... .......... .......... 69% 60,6M 1s\n",
      " 55950K .......... .......... .......... .......... .......... 70% 91,5M 1s\n",
      " 56000K .......... .......... .......... .......... .......... 70%  139M 1s\n",
      " 56050K .......... .......... .......... .......... .......... 70%  159M 1s\n",
      " 56100K .......... .......... .......... .......... .......... 70%  167M 1s\n",
      " 56150K .......... .......... .......... .......... .......... 70%  139M 1s\n",
      " 56200K .......... .......... .......... .......... .......... 70%  109M 1s\n",
      " 56250K .......... .......... .......... .......... .......... 70%  127M 1s\n",
      " 56300K .......... .......... .......... .......... .......... 70%  117M 1s\n",
      " 56350K .......... .......... .......... .......... .......... 70%  114M 1s\n",
      " 56400K .......... .......... .......... .......... .......... 70% 93,7M 1s\n",
      " 56450K .......... .......... .......... .......... .......... 70%  119M 1s\n",
      " 56500K .......... .......... .......... .......... .......... 70%  116M 1s\n",
      " 56550K .......... .......... .......... .......... .......... 70%  116M 1s\n",
      " 56600K .......... .......... .......... .......... .......... 70%  111M 1s\n",
      " 56650K .......... .......... .......... .......... .......... 70%  118M 1s\n",
      " 56700K .......... .......... .......... .......... .......... 70%  116M 1s\n",
      " 56750K .......... .......... .......... .......... .......... 71%  115M 1s\n",
      " 56800K .......... .......... .......... .......... .......... 71% 92,5M 1s\n",
      " 56850K .......... .......... .......... .......... .......... 71%  118M 1s\n",
      " 56900K .......... .......... .......... .......... .......... 71%  113M 1s\n",
      " 56950K .......... .......... .......... .......... .......... 71%  117M 1s\n",
      " 57000K .......... .......... .......... .......... .......... 71%  112M 1s\n",
      " 57050K .......... .......... .......... .......... .......... 71%  121M 1s\n",
      " 57100K .......... .......... .......... .......... .......... 71%  118M 1s\n",
      " 57150K .......... .......... .......... .......... .......... 71%  117M 1s\n",
      " 57200K .......... .......... .......... .......... .......... 71% 89,2M 1s\n",
      " 57250K .......... .......... .......... .......... .......... 71%  118M 1s\n",
      " 57300K .......... .......... .......... .......... .......... 71%  101M 1s\n",
      " 57350K .......... .......... .......... .......... .......... 71% 20,9M 1s\n",
      " 57400K .......... .......... .......... .......... .......... 71% 97,8M 1s\n",
      " 57450K .......... .......... .......... .......... .......... 71% 67,6M 1s\n",
      " 57500K .......... .......... .......... .......... .......... 71%  117M 1s\n",
      " 57550K .......... .......... .......... .......... .......... 72%  117M 1s\n",
      " 57600K .......... .......... .......... .......... .......... 72% 90,2M 1s\n",
      " 57650K .......... .......... .......... .......... .......... 72%  116M 1s\n",
      " 57700K .......... .......... .......... .......... .......... 72%  118M 1s\n",
      " 57750K .......... .......... .......... .......... .......... 72% 70,1M 1s\n",
      " 57800K .......... .......... .......... .......... .......... 72%  114M 1s\n",
      " 57850K .......... .......... .......... .......... .......... 72%  116M 1s\n",
      " 57900K .......... .......... .......... .......... .......... 72%  117M 1s\n",
      " 57950K .......... .......... .......... .......... .......... 72% 1,37M 1s\n",
      " 58000K .......... .......... .......... .......... .......... 72% 1,02M 1s\n",
      " 58050K .......... .......... .......... .......... .......... 72% 18,9M 1s\n",
      " 58100K .......... .......... .......... .......... .......... 72% 28,0M 1s\n",
      " 58150K .......... .......... .......... .......... .......... 72% 29,0M 1s\n",
      " 58200K .......... .......... .......... .......... .......... 72% 14,1M 1s\n",
      " 58250K .......... .......... .......... .......... .......... 72% 28,4M 1s\n",
      " 58300K .......... .......... .......... .......... .......... 72%  143M 1s\n",
      " 58350K .......... .......... .......... .......... .......... 73%  151M 1s\n",
      " 58400K .......... .......... .......... .......... .......... 73% 91,3M 1s\n",
      " 58450K .......... .......... .......... .......... .......... 73% 56,9M 1s\n",
      " 58500K .......... .......... .......... .......... .......... 73% 64,7M 1s\n",
      " 58550K .......... .......... .......... .......... .......... 73%  102M 1s\n",
      " 58600K .......... .......... .......... .......... .......... 73% 47,4M 1s\n",
      " 58650K .......... .......... .......... .......... .......... 73% 83,2M 1s\n",
      " 58700K .......... .......... .......... .......... .......... 73% 96,3M 1s\n",
      " 58750K .......... .......... .......... .......... .......... 73%  112M 1s\n",
      " 58800K .......... .......... .......... .......... .......... 73% 88,6M 1s\n",
      " 58850K .......... .......... .......... .......... .......... 73% 63,2M 1s\n",
      " 58900K .......... .......... .......... .......... .......... 73%  108M 1s\n",
      " 58950K .......... .......... .......... .......... .......... 73%  117M 1s\n",
      " 59000K .......... .......... .......... .......... .......... 73%  110M 1s\n",
      " 59050K .......... .......... .......... .......... .......... 73%  128M 1s\n",
      " 59100K .......... .......... .......... .......... .......... 73%  121M 1s\n",
      " 59150K .......... .......... .......... .......... .......... 74%  122M 1s\n",
      " 59200K .......... .......... .......... .......... .......... 74% 92,1M 1s\n",
      " 59250K .......... .......... .......... .......... .......... 74%  117M 1s\n",
      " 59300K .......... .......... .......... .......... .......... 74%  118M 1s\n",
      " 59350K .......... .......... .......... .......... .......... 74% 98,1M 1s\n",
      " 59400K .......... .......... .......... .......... .......... 74%  131M 1s\n",
      " 59450K .......... .......... .......... .......... .......... 74%  119M 1s\n",
      " 59500K .......... .......... .......... .......... .......... 74%  117M 1s\n",
      " 59550K .......... .......... .......... .......... .......... 74%  121M 1s\n",
      " 59600K .......... .......... .......... .......... .......... 74% 90,2M 1s\n",
      " 59650K .......... .......... .......... .......... .......... 74%  117M 1s\n",
      " 59700K .......... .......... .......... .......... .......... 74%  117M 1s\n",
      " 59750K .......... .......... .......... .......... .......... 74%  115M 1s\n",
      " 59800K .......... .......... .......... .......... .......... 74%  111M 1s\n",
      " 59850K .......... .......... .......... .......... .......... 74%  116M 1s\n",
      " 59900K .......... .......... .......... .......... .......... 74%  119M 1s\n",
      " 59950K .......... .......... .......... .......... .......... 75%  118M 1s\n",
      " 60000K .......... .......... .......... .......... .......... 75% 89,5M 1s\n",
      " 60050K .......... .......... .......... .......... .......... 75%  116M 1s\n",
      " 60100K .......... .......... .......... .......... .......... 75%  120M 1s\n",
      " 60150K .......... .......... .......... .......... .......... 75%  117M 1s\n",
      " 60200K .......... .......... .......... .......... .......... 75%  115M 1s\n",
      " 60250K .......... .......... .......... .......... .......... 75%  113M 1s\n",
      " 60300K .......... .......... .......... .......... .......... 75%  118M 1s\n",
      " 60350K .......... .......... .......... .......... .......... 75%  118M 1s\n",
      " 60400K .......... .......... .......... .......... .......... 75% 32,3M 1s\n",
      " 60450K .......... .......... .......... .......... .......... 75% 39,7M 1s\n",
      " 60500K .......... .......... .......... .......... .......... 75%  117M 1s\n",
      " 60550K .......... .......... .......... .......... .......... 75% 80,9M 1s\n",
      " 60600K .......... .......... .......... .......... .......... 75% 64,1M 1s\n",
      " 60650K .......... .......... .......... .......... .......... 75%  114M 1s\n",
      " 60700K .......... .......... .......... .......... .......... 75%  117M 1s\n",
      " 60750K .......... .......... .......... .......... .......... 76%  117M 1s\n",
      " 60800K .......... .......... .......... .......... .......... 76% 88,4M 1s\n",
      " 60850K .......... .......... .......... .......... .......... 76%  116M 1s\n",
      " 60900K .......... .......... .......... .......... .......... 76%  118M 1s\n",
      " 60950K .......... .......... .......... .......... .......... 76%  116M 1s\n",
      " 61000K .......... .......... .......... .......... .......... 76%  631K 1s\n",
      " 61050K .......... .......... .......... .......... .......... 76% 12,6M 1s\n",
      " 61100K .......... .......... .......... .......... .......... 76% 12,1M 1s\n",
      " 61150K .......... .......... .......... .......... .......... 76%  150M 1s\n",
      " 61200K .......... .......... .......... .......... .......... 76% 13,9M 1s\n",
      " 61250K .......... .......... .......... .......... .......... 76% 18,1M 1s\n",
      " 61300K .......... .......... .......... .......... .......... 76% 37,0M 1s\n",
      " 61350K .......... .......... .......... .......... .......... 76% 46,1M 1s\n",
      " 61400K .......... .......... .......... .......... .......... 76%  148M 1s\n",
      " 61450K .......... .......... .......... .......... .......... 76%  118M 1s\n",
      " 61500K .......... .......... .......... .......... .......... 76% 89,9M 1s\n",
      " 61550K .......... .......... .......... .......... .......... 77%  118M 1s\n",
      " 61600K .......... .......... .......... .......... .......... 77%  106M 1s\n",
      " 61650K .......... .......... .......... .......... .......... 77% 44,0M 1s\n",
      " 61700K .......... .......... .......... .......... .......... 77%  118M 1s\n",
      " 61750K .......... .......... .......... .......... .......... 77%  115M 1s\n",
      " 61800K .......... .......... .......... .......... .......... 77% 76,0M 1s\n",
      " 61850K .......... .......... .......... .......... .......... 77%  117M 1s\n",
      " 61900K .......... .......... .......... .......... .......... 77%  116M 1s\n",
      " 61950K .......... .......... .......... .......... .......... 77% 77,0M 1s\n",
      " 62000K .......... .......... .......... .......... .......... 77% 96,8M 1s\n",
      " 62050K .......... .......... .......... .......... .......... 77%  114M 1s\n",
      " 62100K .......... .......... .......... .......... .......... 77%  119M 1s\n",
      " 62150K .......... .......... .......... .......... .......... 77%  116M 1s\n",
      " 62200K .......... .......... .......... .......... .......... 77%  105M 1s\n",
      " 62250K .......... .......... .......... .......... .......... 77% 85,0M 1s\n",
      " 62300K .......... .......... .......... .......... .......... 77%  114M 1s\n",
      " 62350K .......... .......... .......... .......... .......... 78%  113M 1s\n",
      " 62400K .......... .......... .......... .......... .......... 78% 98,7M 1s\n",
      " 62450K .......... .......... .......... .......... .......... 78%  112M 1s\n",
      " 62500K .......... .......... .......... .......... .......... 78%  118M 1s\n",
      " 62550K .......... .......... .......... .......... .......... 78%  119M 1s\n",
      " 62600K .......... .......... .......... .......... .......... 78%  110M 1s\n",
      " 62650K .......... .......... .......... .......... .......... 78%  116M 1s\n",
      " 62700K .......... .......... .......... .......... .......... 78%  113M 1s\n",
      " 62750K .......... .......... .......... .......... .......... 78%  121M 1s\n",
      " 62800K .......... .......... .......... .......... .......... 78% 94,4M 1s\n",
      " 62850K .......... .......... .......... .......... .......... 78%  116M 1s\n",
      " 62900K .......... .......... .......... .......... .......... 78%  117M 1s\n",
      " 62950K .......... .......... .......... .......... .......... 78%  118M 1s\n",
      " 63000K .......... .......... .......... .......... .......... 78%  110M 1s\n",
      " 63050K .......... .......... .......... .......... .......... 78%  116M 1s\n",
      " 63100K .......... .......... .......... .......... .......... 78%  101M 1s\n",
      " 63150K .......... .......... .......... .......... .......... 79%  134M 1s\n",
      " 63200K .......... .......... .......... .......... .......... 79% 95,8M 1s\n",
      " 63250K .......... .......... .......... .......... .......... 79%  117M 1s\n",
      " 63300K .......... .......... .......... .......... .......... 79%  113M 1s\n",
      " 63350K .......... .......... .......... .......... .......... 79%  117M 1s\n",
      " 63400K .......... .......... .......... .......... .......... 79%  112M 1s\n",
      " 63450K .......... .......... .......... .......... .......... 79%  117M 1s\n",
      " 63500K .......... .......... .......... .......... .......... 79% 22,8M 1s\n",
      " 63550K .......... .......... .......... .......... .......... 79%  116M 1s\n",
      " 63600K .......... .......... .......... .......... .......... 79% 50,2M 1s\n",
      " 63650K .......... .......... .......... .......... .......... 79%  117M 1s\n",
      " 63700K .......... .......... .......... .......... .......... 79%  103M 1s\n",
      " 63750K .......... .......... .......... .......... .......... 79% 68,6M 1s\n",
      " 63800K .......... .......... .......... .......... .......... 79%  114M 1s\n",
      " 63850K .......... .......... .......... .......... .......... 79%  117M 1s\n",
      " 63900K .......... .......... .......... .......... .......... 79%  115M 1s\n",
      " 63950K .......... .......... .......... .......... .......... 80%  117M 1s\n",
      " 64000K .......... .......... .......... .......... .......... 80% 91,0M 1s\n",
      " 64050K .......... .......... .......... .......... .......... 80%  639K 1s\n",
      " 64100K .......... .......... .......... .......... .......... 80% 11,5M 1s\n",
      " 64150K .......... .......... .......... .......... .......... 80% 10,5M 1s\n",
      " 64200K .......... .......... .......... .......... .......... 80% 22,5M 1s\n",
      " 64250K .......... .......... .......... .......... .......... 80% 24,9M 1s\n",
      " 64300K .......... .......... .......... .......... .......... 80% 26,5M 1s\n",
      " 64350K .......... .......... .......... .......... .......... 80% 20,4M 1s\n",
      " 64400K .......... .......... .......... .......... .......... 80% 78,5M 1s\n",
      " 64450K .......... .......... .......... .......... .......... 80% 97,3M 1s\n",
      " 64500K .......... .......... .......... .......... .......... 80% 97,3M 1s\n",
      " 64550K .......... .......... .......... .......... .......... 80%  102M 1s\n",
      " 64600K .......... .......... .......... .......... .......... 80% 89,5M 1s\n",
      " 64650K .......... .......... .......... .......... .......... 80%  104M 1s\n",
      " 64700K .......... .......... .......... .......... .......... 80% 74,3M 1s\n",
      " 64750K .......... .......... .......... .......... .......... 81%  123M 1s\n",
      " 64800K .......... .......... .......... .......... .......... 81% 85,8M 1s\n",
      " 64850K .......... .......... .......... .......... .......... 81% 57,3M 1s\n",
      " 64900K .......... .......... .......... .......... .......... 81%  118M 1s\n",
      " 64950K .......... .......... .......... .......... .......... 81%  114M 1s\n",
      " 65000K .......... .......... .......... .......... .......... 81%  109M 1s\n",
      " 65050K .......... .......... .......... .......... .......... 81%  120M 1s\n",
      " 65100K .......... .......... .......... .......... .......... 81%  119M 1s\n",
      " 65150K .......... .......... .......... .......... .......... 81% 93,8M 1s\n",
      " 65200K .......... .......... .......... .......... .......... 81%  111M 1s\n",
      " 65250K .......... .......... .......... .......... .......... 81%  117M 1s\n",
      " 65300K .......... .......... .......... .......... .......... 81%  119M 1s\n",
      " 65350K .......... .......... .......... .......... .......... 81%  114M 1s\n",
      " 65400K .......... .......... .......... .......... .......... 81%  111M 1s\n",
      " 65450K .......... .......... .......... .......... .......... 81%  118M 1s\n",
      " 65500K .......... .......... .......... .......... .......... 81%  116M 1s\n",
      " 65550K .......... .......... .......... .......... .......... 82%  117M 1s\n",
      " 65600K .......... .......... .......... .......... .......... 82% 91,6M 1s\n",
      " 65650K .......... .......... .......... .......... .......... 82%  117M 1s\n",
      " 65700K .......... .......... .......... .......... .......... 82%  117M 1s\n",
      " 65750K .......... .......... .......... .......... .......... 82%  114M 1s\n",
      " 65800K .......... .......... .......... .......... .......... 82%  113M 1s\n",
      " 65850K .......... .......... .......... .......... .......... 82%  117M 1s\n",
      " 65900K .......... .......... .......... .......... .......... 82%  117M 1s\n",
      " 65950K .......... .......... .......... .......... .......... 82%  120M 1s\n",
      " 66000K .......... .......... .......... .......... .......... 82% 89,3M 1s\n",
      " 66050K .......... .......... .......... .......... .......... 82%  117M 1s\n",
      " 66100K .......... .......... .......... .......... .......... 82%  117M 1s\n",
      " 66150K .......... .......... .......... .......... .......... 82%  114M 1s\n",
      " 66200K .......... .......... .......... .......... .......... 82%  112M 1s\n",
      " 66250K .......... .......... .......... .......... .......... 82%  120M 1s\n",
      " 66300K .......... .......... .......... .......... .......... 82%  117M 1s\n",
      " 66350K .......... .......... .......... .......... .......... 83%  117M 1s\n",
      " 66400K .......... .......... .......... .......... .......... 83% 88,2M 1s\n",
      " 66450K .......... .......... .......... .......... .......... 83%  118M 1s\n",
      " 66500K .......... .......... .......... .......... .......... 83%  117M 1s\n",
      " 66550K .......... .......... .......... .......... .......... 83% 38,5M 1s\n",
      " 66600K .......... .......... .......... .......... .......... 83% 39,7M 1s\n",
      " 66650K .......... .......... .......... .......... .......... 83% 53,9M 1s\n",
      " 66700K .......... .......... .......... .......... .......... 83%  121M 1s\n",
      " 66750K .......... .......... .......... .......... .......... 83% 90,9M 1s\n",
      " 66800K .......... .......... .......... .......... .......... 83% 88,4M 1s\n",
      " 66850K .......... .......... .......... .......... .......... 83%  118M 1s\n",
      " 66900K .......... .......... .......... .......... .......... 83%  117M 1s\n",
      " 66950K .......... .......... .......... .......... .......... 83%  117M 1s\n",
      " 67000K .......... .......... .......... .......... .......... 83%  110M 1s\n",
      " 67050K .......... .......... .......... .......... .......... 83%  117M 1s\n",
      " 67100K .......... .......... .......... .......... .......... 83% 1,14M 1s\n",
      " 67150K .......... .......... .......... .......... .......... 84% 1,22M 1s\n",
      " 67200K .......... .......... .......... .......... .......... 84% 12,1M 1s\n",
      " 67250K .......... .......... .......... .......... .......... 84% 25,3M 1s\n",
      " 67300K .......... .......... .......... .......... .......... 84% 12,6M 1s\n",
      " 67350K .......... .......... .......... .......... .......... 84% 28,5M 1s\n",
      " 67400K .......... .......... .......... .......... .......... 84% 25,5M 1s\n",
      " 67450K .......... .......... .......... .......... .......... 84%  120M 1s\n",
      " 67500K .......... .......... .......... .......... .......... 84% 56,4M 1s\n",
      " 67550K .......... .......... .......... .......... .......... 84%  116M 1s\n",
      " 67600K .......... .......... .......... .......... .......... 84% 95,4M 1s\n",
      " 67650K .......... .......... .......... .......... .......... 84%  117M 1s\n",
      " 67700K .......... .......... .......... .......... .......... 84%  113M 1s\n",
      " 67750K .......... .......... .......... .......... .......... 84%  120M 1s\n",
      " 67800K .......... .......... .......... .......... .......... 84%  105M 1s\n",
      " 67850K .......... .......... .......... .......... .......... 84% 46,4M 1s\n",
      " 67900K .......... .......... .......... .......... .......... 84%  116M 1s\n",
      " 67950K .......... .......... .......... .......... .......... 85% 51,8M 1s\n",
      " 68000K .......... .......... .......... .......... .......... 85% 93,5M 1s\n",
      " 68050K .......... .......... .......... .......... .......... 85%  119M 1s\n",
      " 68100K .......... .......... .......... .......... .......... 85%  116M 1s\n",
      " 68150K .......... .......... .......... .......... .......... 85%  115M 1s\n",
      " 68200K .......... .......... .......... .......... .......... 85%  103M 1s\n",
      " 68250K .......... .......... .......... .......... .......... 85%  122M 1s\n",
      " 68300K .......... .......... .......... .......... .......... 85%  115M 1s\n",
      " 68350K .......... .......... .......... .......... .......... 85%  117M 1s\n",
      " 68400K .......... .......... .......... .......... .......... 85% 95,5M 1s\n",
      " 68450K .......... .......... .......... .......... .......... 85%  116M 1s\n",
      " 68500K .......... .......... .......... .......... .......... 85%  119M 1s\n",
      " 68550K .......... .......... .......... .......... .......... 85%  116M 1s\n",
      " 68600K .......... .......... .......... .......... .......... 85%  109M 1s\n",
      " 68650K .......... .......... .......... .......... .......... 85%  116M 1s\n",
      " 68700K .......... .......... .......... .......... .......... 85%  118M 1s\n",
      " 68750K .......... .......... .......... .......... .......... 86%  117M 1s\n",
      " 68800K .......... .......... .......... .......... .......... 86% 92,4M 1s\n",
      " 68850K .......... .......... .......... .......... .......... 86%  117M 1s\n",
      " 68900K .......... .......... .......... .......... .......... 86%  118M 1s\n",
      " 68950K .......... .......... .......... .......... .......... 86%  117M 1s\n",
      " 69000K .......... .......... .......... .......... .......... 86%  110M 1s\n",
      " 69050K .......... .......... .......... .......... .......... 86%  118M 1s\n",
      " 69100K .......... .......... .......... .......... .......... 86%  117M 1s\n",
      " 69150K .......... .......... .......... .......... .......... 86%  118M 1s\n",
      " 69200K .......... .......... .......... .......... .......... 86% 89,6M 1s\n",
      " 69250K .......... .......... .......... .......... .......... 86%  118M 1s\n",
      " 69300K .......... .......... .......... .......... .......... 86%  117M 1s\n",
      " 69350K .......... .......... .......... .......... .......... 86%  117M 1s\n",
      " 69400K .......... .......... .......... .......... .......... 86%  115M 1s\n",
      " 69450K .......... .......... .......... .......... .......... 86%  116M 1s\n",
      " 69500K .......... .......... .......... .......... .......... 86%  114M 1s\n",
      " 69550K .......... .......... .......... .......... .......... 87%  118M 1s\n",
      " 69600K .......... .......... .......... .......... .......... 87% 90,7M 1s\n",
      " 69650K .......... .......... .......... .......... .......... 87% 51,3M 1s\n",
      " 69700K .......... .......... .......... .......... .......... 87% 36,6M 1s\n",
      " 69750K .......... .......... .......... .......... .......... 87% 63,8M 1s\n",
      " 69800K .......... .......... .......... .......... .......... 87%  114M 1s\n",
      " 69850K .......... .......... .......... .......... .......... 87%  117M 1s\n",
      " 69900K .......... .......... .......... .......... .......... 87%  113M 1s\n",
      " 69950K .......... .......... .......... .......... .......... 87%  119M 1s\n",
      " 70000K .......... .......... .......... .......... .......... 87% 91,0M 1s\n",
      " 70050K .......... .......... .......... .......... .......... 87%  113M 1s\n",
      " 70100K .......... .......... .......... .......... .......... 87%  117M 1s\n",
      " 70150K .......... .......... .......... .......... .......... 87% 1,17M 1s\n",
      " 70200K .......... .......... .......... .......... .......... 87% 1,15M 1s\n",
      " 70250K .......... .......... .......... .......... .......... 87% 16,4M 1s\n",
      " 70300K .......... .......... .......... .......... .......... 87% 36,4M 1s\n",
      " 70350K .......... .......... .......... .......... .......... 88% 18,7M 0s\n",
      " 70400K .......... .......... .......... .......... .......... 88% 19,3M 0s\n",
      " 70450K .......... .......... .......... .......... .......... 88% 56,5M 0s\n",
      " 70500K .......... .......... .......... .......... .......... 88% 61,3M 0s\n",
      " 70550K .......... .......... .......... .......... .......... 88%  109M 0s\n",
      " 70600K .......... .......... .......... .......... .......... 88% 27,3M 0s\n",
      " 70650K .......... .......... .......... .......... .......... 88% 77,3M 0s\n",
      " 70700K .......... .......... .......... .......... .......... 88% 93,3M 0s\n",
      " 70750K .......... .......... .......... .......... .......... 88% 93,0M 0s\n",
      " 70800K .......... .......... .......... .......... .......... 88% 78,4M 0s\n",
      " 70850K .......... .......... .......... .......... .......... 88%  101M 0s\n",
      " 70900K .......... .......... .......... .......... .......... 88%  141M 0s\n",
      " 70950K .......... .......... .......... .......... .......... 88% 77,6M 0s\n",
      " 71000K .......... .......... .......... .......... .......... 88% 92,1M 0s\n",
      " 71050K .......... .......... .......... .......... .......... 88% 94,0M 0s\n",
      " 71100K .......... .......... .......... .......... .......... 88%  117M 0s\n",
      " 71150K .......... .......... .......... .......... .......... 89%  112M 0s\n",
      " 71200K .......... .......... .......... .......... .......... 89% 95,5M 0s\n",
      " 71250K .......... .......... .......... .......... .......... 89%  116M 0s\n",
      " 71300K .......... .......... .......... .......... .......... 89% 79,6M 0s\n",
      " 71350K .......... .......... .......... .......... .......... 89%  131M 0s\n",
      " 71400K .......... .......... .......... .......... .......... 89%  118M 0s\n",
      " 71450K .......... .......... .......... .......... .......... 89%  124M 0s\n",
      " 71500K .......... .......... .......... .......... .......... 89%  148M 0s\n",
      " 71550K .......... .......... .......... .......... .......... 89%  115M 0s\n",
      " 71600K .......... .......... .......... .......... .......... 89% 96,0M 0s\n",
      " 71650K .......... .......... .......... .......... .......... 89%  115M 0s\n",
      " 71700K .......... .......... .......... .......... .......... 89%  117M 0s\n",
      " 71750K .......... .......... .......... .......... .......... 89%  118M 0s\n",
      " 71800K .......... .......... .......... .......... .......... 89%  108M 0s\n",
      " 71850K .......... .......... .......... .......... .......... 89%  114M 0s\n",
      " 71900K .......... .......... .......... .......... .......... 89%  121M 0s\n",
      " 71950K .......... .......... .......... .......... .......... 90%  113M 0s\n",
      " 72000K .......... .......... .......... .......... .......... 90% 96,5M 0s\n",
      " 72050K .......... .......... .......... .......... .......... 90%  118M 0s\n",
      " 72100K .......... .......... .......... .......... .......... 90%  114M 0s\n",
      " 72150K .......... .......... .......... .......... .......... 90%  117M 0s\n",
      " 72200K .......... .......... .......... .......... .......... 90%  112M 0s\n",
      " 72250K .......... .......... .......... .......... .......... 90%  118M 0s\n",
      " 72300K .......... .......... .......... .......... .......... 90%  117M 0s\n",
      " 72350K .......... .......... .......... .......... .......... 90%  117M 0s\n",
      " 72400K .......... .......... .......... .......... .......... 90% 91,0M 0s\n",
      " 72450K .......... .......... .......... .......... .......... 90%  116M 0s\n",
      " 72500K .......... .......... .......... .......... .......... 90%  118M 0s\n",
      " 72550K .......... .......... .......... .......... .......... 90%  113M 0s\n",
      " 72600K .......... .......... .......... .......... .......... 90%  113M 0s\n",
      " 72650K .......... .......... .......... .......... .......... 90%  118M 0s\n",
      " 72700K .......... .......... .......... .......... .......... 90%  118M 0s\n",
      " 72750K .......... .......... .......... .......... .......... 91% 89,8M 0s\n",
      " 72800K .......... .......... .......... .......... .......... 91% 19,2M 0s\n",
      " 72850K .......... .......... .......... .......... .......... 91%  117M 0s\n",
      " 72900K .......... .......... .......... .......... .......... 91%  117M 0s\n",
      " 72950K .......... .......... .......... .......... .......... 91%  114M 0s\n",
      " 73000K .......... .......... .......... .......... .......... 91%  113M 0s\n",
      " 73050K .......... .......... .......... .......... .......... 91%  118M 0s\n",
      " 73100K .......... .......... .......... .......... .......... 91%  117M 0s\n",
      " 73150K .......... .......... .......... .......... .......... 91%  119M 0s\n",
      " 73200K .......... .......... .......... .......... .......... 91%  633K 0s\n",
      " 73250K .......... .......... .......... .......... .......... 91% 9,06M 0s\n",
      " 73300K .......... .......... .......... .......... .......... 91% 22,9M 0s\n",
      " 73350K .......... .......... .......... .......... .......... 91% 17,1M 0s\n",
      " 73400K .......... .......... .......... .......... .......... 91% 22,7M 0s\n",
      " 73450K .......... .......... .......... .......... .......... 91% 17,0M 0s\n",
      " 73500K .......... .......... .......... .......... .......... 91% 66,1M 0s\n",
      " 73550K .......... .......... .......... .......... .......... 92% 66,8M 0s\n",
      " 73600K .......... .......... .......... .......... .......... 92% 62,1M 0s\n",
      " 73650K .......... .......... .......... .......... .......... 92%  101M 0s\n",
      " 73700K .......... .......... .......... .......... .......... 92% 48,2M 0s\n",
      " 73750K .......... .......... .......... .......... .......... 92% 66,3M 0s\n",
      " 73800K .......... .......... .......... .......... .......... 92% 87,2M 0s\n",
      " 73850K .......... .......... .......... .......... .......... 92%  170M 0s\n",
      " 73900K .......... .......... .......... .......... .......... 92%  180M 0s\n",
      " 73950K .......... .......... .......... .......... .......... 92%  175M 0s\n",
      " 74000K .......... .......... .......... .......... .......... 92%  146M 0s\n",
      " 74050K .......... .......... .......... .......... .......... 92% 62,1M 0s\n",
      " 74100K .......... .......... .......... .......... .......... 92% 79,5M 0s\n",
      " 74150K .......... .......... .......... .......... .......... 92%  117M 0s\n",
      " 74200K .......... .......... .......... .......... .......... 92% 54,0M 0s\n",
      " 74250K .......... .......... .......... .......... .......... 92%  117M 0s\n",
      " 74300K .......... .......... .......... .......... .......... 92%  115M 0s\n",
      " 74350K .......... .......... .......... .......... .......... 93%  121M 0s\n",
      " 74400K .......... .......... .......... .......... .......... 93% 90,7M 0s\n",
      " 74450K .......... .......... .......... .......... .......... 93%  118M 0s\n",
      " 74500K .......... .......... .......... .......... .......... 93%  117M 0s\n",
      " 74550K .......... .......... .......... .......... .......... 93%  117M 0s\n",
      " 74600K .......... .......... .......... .......... .......... 93%  111M 0s\n",
      " 74650K .......... .......... .......... .......... .......... 93%  112M 0s\n",
      " 74700K .......... .......... .......... .......... .......... 93%  121M 0s\n",
      " 74750K .......... .......... .......... .......... .......... 93%  118M 0s\n",
      " 74800K .......... .......... .......... .......... .......... 93% 91,2M 0s\n",
      " 74850K .......... .......... .......... .......... .......... 93%  117M 0s\n",
      " 74900K .......... .......... .......... .......... .......... 93%  117M 0s\n",
      " 74950K .......... .......... .......... .......... .......... 93%  117M 0s\n",
      " 75000K .......... .......... .......... .......... .......... 93%  109M 0s\n",
      " 75050K .......... .......... .......... .......... .......... 93%  113M 0s\n",
      " 75100K .......... .......... .......... .......... .......... 93%  119M 0s\n",
      " 75150K .......... .......... .......... .......... .......... 94%  122M 0s\n",
      " 75200K .......... .......... .......... .......... .......... 94% 89,5M 0s\n",
      " 75250K .......... .......... .......... .......... .......... 94%  117M 0s\n",
      " 75300K .......... .......... .......... .......... .......... 94%  117M 0s\n",
      " 75350K .......... .......... .......... .......... .......... 94%  117M 0s\n",
      " 75400K .......... .......... .......... .......... .......... 94%  110M 0s\n",
      " 75450K .......... .......... .......... .......... .......... 94%  118M 0s\n",
      " 75500K .......... .......... .......... .......... .......... 94%  118M 0s\n",
      " 75550K .......... .......... .......... .......... .......... 94%  117M 0s\n",
      " 75600K .......... .......... .......... .......... .......... 94% 88,6M 0s\n",
      " 75650K .......... .......... .......... .......... .......... 94%  117M 0s\n",
      " 75700K .......... .......... .......... .......... .......... 94%  118M 0s\n",
      " 75750K .......... .......... .......... .......... .......... 94%  116M 0s\n",
      " 75800K .......... .......... .......... .......... .......... 94%  114M 0s\n",
      " 75850K .......... .......... .......... .......... .......... 94% 22,9M 0s\n",
      " 75900K .......... .......... .......... .......... .......... 94% 85,5M 0s\n",
      " 75950K .......... .......... .......... .......... .......... 95%  117M 0s\n",
      " 76000K .......... .......... .......... .......... .......... 95% 89,3M 0s\n",
      " 76050K .......... .......... .......... .......... .......... 95%  119M 0s\n",
      " 76100K .......... .......... .......... .......... .......... 95%  117M 0s\n",
      " 76150K .......... .......... .......... .......... .......... 95%  118M 0s\n",
      " 76200K .......... .......... .......... .......... .......... 95% 1,22M 0s\n",
      " 76250K .......... .......... .......... .......... .......... 95% 1,14M 0s\n",
      " 76300K .......... .......... .......... .......... .......... 95%  148M 0s\n",
      " 76350K .......... .......... .......... .......... .......... 95% 12,7M 0s\n",
      " 76400K .......... .......... .......... .......... .......... 95% 22,3M 0s\n",
      " 76450K .......... .......... .......... .......... .......... 95% 13,8M 0s\n",
      " 76500K .......... .......... .......... .......... .......... 95% 42,2M 0s\n",
      " 76550K .......... .......... .......... .......... .......... 95%  126M 0s\n",
      " 76600K .......... .......... .......... .......... .......... 95% 21,5M 0s\n",
      " 76650K .......... .......... .......... .......... .......... 95% 63,2M 0s\n",
      " 76700K .......... .......... .......... .......... .......... 96% 94,2M 0s\n",
      " 76750K .......... .......... .......... .......... .......... 96%  100M 0s\n",
      " 76800K .......... .......... .......... .......... .......... 96% 87,3M 0s\n",
      " 76850K .......... .......... .......... .......... .......... 96% 49,0M 0s\n",
      " 76900K .......... .......... .......... .......... .......... 96% 57,3M 0s\n",
      " 76950K .......... .......... .......... .......... .......... 96%  101M 0s\n",
      " 77000K .......... .......... .......... .......... .......... 96%  100M 0s\n",
      " 77050K .......... .......... .......... .......... .......... 96%  114M 0s\n",
      " 77100K .......... .......... .......... .......... .......... 96%  118M 0s\n",
      " 77150K .......... .......... .......... .......... .......... 96%  113M 0s\n",
      " 77200K .......... .......... .......... .......... .......... 96% 91,0M 0s\n",
      " 77250K .......... .......... .......... .......... .......... 96% 55,3M 0s\n",
      " 77300K .......... .......... .......... .......... .......... 96% 74,4M 0s\n",
      " 77350K .......... .......... .......... .......... .......... 96%  161M 0s\n",
      " 77400K .......... .......... .......... .......... .......... 96%  152M 0s\n",
      " 77450K .......... .......... .......... .......... .......... 96% 93,0M 0s\n",
      " 77500K .......... .......... .......... .......... .......... 97%  171M 0s\n",
      " 77550K .......... .......... .......... .......... .......... 97%  137M 0s\n",
      " 77600K .......... .......... .......... .......... .......... 97% 93,6M 0s\n",
      " 77650K .......... .......... .......... .......... .......... 97%  114M 0s\n",
      " 77700K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 77750K .......... .......... .......... .......... .......... 97%  119M 0s\n",
      " 77800K .......... .......... .......... .......... .......... 97%  109M 0s\n",
      " 77850K .......... .......... .......... .......... .......... 97%  115M 0s\n",
      " 77900K .......... .......... .......... .......... .......... 97%  114M 0s\n",
      " 77950K .......... .......... .......... .......... .......... 97%  121M 0s\n",
      " 78000K .......... .......... .......... .......... .......... 97% 93,6M 0s\n",
      " 78050K .......... .......... .......... .......... .......... 97%  114M 0s\n",
      " 78100K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 78150K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 78200K .......... .......... .......... .......... .......... 97%  110M 0s\n",
      " 78250K .......... .......... .......... .......... .......... 97%  119M 0s\n",
      " 78300K .......... .......... .......... .......... .......... 98%  114M 0s\n",
      " 78350K .......... .......... .......... .......... .......... 98%  118M 0s\n",
      " 78400K .......... .......... .......... .......... .......... 98% 92,3M 0s\n",
      " 78450K .......... .......... .......... .......... .......... 98%  114M 0s\n",
      " 78500K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 78550K .......... .......... .......... .......... .......... 98%  119M 0s\n",
      " 78600K .......... .......... .......... .......... .......... 98%  112M 0s\n",
      " 78650K .......... .......... .......... .......... .......... 98%  118M 0s\n",
      " 78700K .......... .......... .......... .......... .......... 98%  118M 0s\n",
      " 78750K .......... .......... .......... .......... .......... 98%  112M 0s\n",
      " 78800K .......... .......... .......... .......... .......... 98% 94,0M 0s\n",
      " 78850K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 78900K .......... .......... .......... .......... .......... 98% 78,8M 0s\n",
      " 78950K .......... .......... .......... .......... .......... 98% 25,8M 0s\n",
      " 79000K .......... .......... .......... .......... .......... 98%  111M 0s\n",
      " 79050K .......... .......... .......... .......... .......... 98%  113M 0s\n",
      " 79100K .......... .......... .......... .......... .......... 99%  117M 0s\n",
      " 79150K .......... .......... .......... .......... .......... 99%  115M 0s\n",
      " 79200K .......... .......... .......... .......... .......... 99% 91,1M 0s\n",
      " 79250K .......... .......... .......... .......... .......... 99% 1,24M 0s\n",
      " 79300K .......... .......... .......... .......... .......... 99% 1,16M 0s\n",
      " 79350K .......... .......... .......... .......... .......... 99% 10,7M 0s\n",
      " 79400K .......... .......... .......... .......... .......... 99% 68,9M 0s\n",
      " 79450K .......... .......... .......... .......... .......... 99% 13,9M 0s\n",
      " 79500K .......... .......... .......... .......... .......... 99% 31,9M 0s\n",
      " 79550K .......... .......... .......... .......... .......... 99% 24,7M 0s\n",
      " 79600K .......... .......... .......... .......... .......... 99% 47,0M 0s\n",
      " 79650K .......... .......... .......... .......... .......... 99% 28,0M 0s\n",
      " 79700K .......... .......... .......... .......... .......... 99%  119M 0s\n",
      " 79750K .......... .......... .......... .......... .......... 99%  113M 0s\n",
      " 79800K .......... .......... .......... .......... .......... 99%  108M 0s\n",
      " 79850K .......... .......... .......... .......... .......... 99% 63,2M 0s\n",
      " 79900K .......... .......... .......... .......... .......   100%  132M=4,1s\n",
      "\n",
      "2023-08-28 18:26:16 (19,3 MB/s) - “data/ClinVar/clinvar.vcf.gz” guardado [81866312/81866312]\n",
      "\n",
      "--2023-08-28 18:26:16--  https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/clinvar.vcf.gz.tbi\n",
      "Resolviendo ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.10, 130.14.250.7, 2607:f220:41e:250::11, ...\n",
      "Conectando con ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)[130.14.250.10]:443... conectado.\n",
      "Petición HTTP enviada, esperando respuesta... 200 OK\n",
      "Longitud: 505871 (494K) [application/x-gzip]\n",
      "Grabando a: “data/ClinVar/clinvar.vcf.gz.tbi”\n",
      "\n",
      "     0K .......... .......... .......... .......... .......... 10%  200K 2s\n",
      "    50K .......... .......... .......... .......... .......... 20%  398K 1s\n",
      "   100K .......... .......... .......... .......... .......... 30% 87,1M 1s\n",
      "   150K .......... .......... .......... .......... .......... 40% 1,24M 1s\n",
      "   200K .......... .......... .......... .......... .......... 50%  569K 0s\n",
      "   250K .......... .......... .......... .......... .......... 60%  110M 0s\n",
      "   300K .......... .......... .......... .......... .......... 70%  131M 0s\n",
      "   350K .......... .......... .......... .......... .......... 80% 1,30M 0s\n",
      "   400K .......... .......... .......... .......... .......... 91%  594K 0s\n",
      "   450K .......... .......... .......... .......... ....      100% 31,5M=0,6s\n",
      "\n",
      "2023-08-28 18:26:17 (787 KB/s) - “data/ClinVar/clinvar.vcf.gz.tbi” guardado [505871/505871]\n",
      "\n",
      "[Mon Aug 28 18:26:48 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T182609.880459.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:26:50 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753368_chr3.vcf\n",
      "    output: results/variants/vep/ERR753368_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753368_chr3.txt\n",
      "    wildcards: sample=ERR753368_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Mon Aug 28 18:27:23 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T182649.120424.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:27:25 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753369_chr3.vcf\n",
      "    output: results/variants/vep/ERR753369_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753369_chr3.txt\n",
      "    wildcards: sample=ERR753369_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Mon Aug 28 18:27:57 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T182724.150974.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:27:59 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753370_chr3.vcf\n",
      "    output: results/variants/vep/ERR753370_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753370_chr3.txt\n",
      "    wildcards: sample=ERR753370_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Mon Aug 28 18:28:36 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T182758.138434.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:28:37 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753371_chr3.vcf\n",
      "    output: results/variants/vep/ERR753371_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753371_chr3.txt\n",
      "    wildcards: sample=ERR753371_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Mon Aug 28 18:29:02 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T182837.009774.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:29:03 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753372_chr3.vcf\n",
      "    output: results/variants/vep/ERR753372_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753372_chr3.txt\n",
      "    wildcards: sample=ERR753372_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Mon Aug 28 18:29:36 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T182902.747357.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:29:38 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753373_chr3.vcf\n",
      "    output: results/variants/vep/ERR753373_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753373_chr3.txt\n",
      "    wildcards: sample=ERR753373_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Mon Aug 28 18:30:13 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T182937.024344.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:30:14 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753374_chr3.vcf\n",
      "    output: results/variants/vep/ERR753374_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753374_chr3.txt\n",
      "    wildcards: sample=ERR753374_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Mon Aug 28 18:30:40 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T183013.428735.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:30:41 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753375_chr3.vcf\n",
      "    output: results/variants/vep/ERR753375_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753375_chr3.txt\n",
      "    wildcards: sample=ERR753375_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Mon Aug 28 18:31:07 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T183040.590744.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:31:08 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753376_chr3.vcf\n",
      "    output: results/variants/vep/ERR753376_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753376_chr3.txt\n",
      "    wildcards: sample=ERR753376_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Mon Aug 28 18:31:35 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T183107.717076.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:31:37 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753377_chr3.vcf\n",
      "    output: results/variants/vep/ERR753377_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753377_chr3.txt\n",
      "    wildcards: sample=ERR753377_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Mon Aug 28 18:32:05 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T183136.148707.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:32:06 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753378_chr3.vcf\n",
      "    output: results/variants/vep/ERR753378_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753378_chr3.txt\n",
      "    wildcards: sample=ERR753378_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Mon Aug 28 18:32:38 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T183205.290676.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job              count    min threads    max threads\n",
      "-------------  -------  -------------  -------------\n",
      "parsing_dataR        1              1              1\n",
      "total                1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 18:32:39 2023]\n",
      "rule parsing_dataR:\n",
      "    input: code/05parsing_vep_data.R\n",
      "    output: tasks/13parsing_dataR.done\n",
      "    jobid: 0\n",
      "    reason: Missing output files: tasks/13parsing_dataR.done\n",
      "    resources: tmpdir=/tmp\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">>> This is your sample list: ERR696683_chr3\n",
      ">>> This is your sample list: ERR753368_chr3\n",
      ">>> This is your sample list: ERR753369_chr3\n",
      ">>> This is your sample list: ERR753370_chr3\n",
      ">>> This is your sample list: ERR753371_chr3\n",
      ">>> This is your sample list: ERR753372_chr3\n",
      ">>> This is your sample list: ERR753373_chr3\n",
      ">>> This is your sample list: ERR753374_chr3\n",
      ">>> This is your sample list: ERR753375_chr3\n",
      ">>> This is your sample list: ERR753376_chr3\n",
      ">>> This is your sample list: ERR753377_chr3\n",
      ">>> This is your sample list: ERR753378_chr3\n",
      ">>> This is your gene: PIK3CA\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 33 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "2 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   downstream…\n",
      "3 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   downstream…\n",
      "4 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "5 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "6 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 27 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 26 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 3_179204486_C/A       3:1792044… A      ENSG… ENST00… Transcript   splice_pol…\n",
      "2 3_179204486_C/A       3:1792044… A      ENSG… ENST00… Transcript   splice_pol…\n",
      "3 3_179204486_C/A       3:1792044… A      ENSG… ENST00… Transcript   splice_pol…\n",
      "4 3_179204486_C/A       3:1792044… A      ENSG… ENST00… Transcript   splice_pol…\n",
      "5 3_179204486_C/A       3:1792044… A      ENSG… ENST00… Transcript   splice_pol…\n",
      "6 3_179204642_A/G       3:1792046… G      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 20 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 8 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 3_179226200_A/ACTTGA  3:1792262… ACTTGA ENSG… ENST00… Transcript   intron_var…\n",
      "2 3_179226200_A/ACTTGA  3:1792262… ACTTGA ENSG… ENST00… Transcript   intron_var…\n",
      "3 3_179226200_A/ACTTGA  3:1792262… ACTTGA ENSG… ENST00… Transcript   intron_var…\n",
      "4 3_179226200_A/ACTTGA  3:1792262… ACTTGA ENSG… ENST00… Transcript   intron_var…\n",
      "5 3_179226200_A/ACTTGA  3:1792262… ACTTGA ENSG… ENST00… Transcript   intron_var…\n",
      "6 3_179226200_A/ACTTGA  3:1792262… ACTTGA ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 2 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 33 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "2 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   downstream…\n",
      "3 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   downstream…\n",
      "4 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "5 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "6 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 27 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 0 × 44\n",
      "# ℹ 44 variables: #uploaded_variation <chr>, location <chr>, allele <chr>,\n",
      "#   gene <chr>, feature <chr>, feature_type <chr>, consequence <chr>,\n",
      "#   cdna_position <chr>, cds_position <chr>, protein_position <chr>,\n",
      "#   amino_acids <chr>, codons <chr>, existing_variation <chr>, impact <chr>,\n",
      "#   distance <chr>, strand <chr>, flags <chr>, variant_class <chr>,\n",
      "#   symbol <chr>, symbol_source <chr>, hgnc_id <chr>, biotype <chr>,\n",
      "#   mane_select <chr>, mane_plus_clinical <chr>, tsl <chr>, appris <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 26 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 3_179204486_C/A       3:1792044… A      ENSG… ENST00… Transcript   splice_pol…\n",
      "2 3_179204486_C/A       3:1792044… A      ENSG… ENST00… Transcript   splice_pol…\n",
      "3 3_179204486_C/A       3:1792044… A      ENSG… ENST00… Transcript   splice_pol…\n",
      "4 3_179204486_C/A       3:1792044… A      ENSG… ENST00… Transcript   splice_pol…\n",
      "5 3_179204486_C/A       3:1792044… A      ENSG… ENST00… Transcript   splice_pol…\n",
      "6 3_179204642_A/G       3:1792046… G      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 20 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 12 × 44\n",
      "  `#uploaded_variation`   location allele gene  feature feature_type consequence\n",
      "  <chr>                   <chr>    <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 3_179170077_CACACACACA… 3:17917… CACAC… ENSG… ENST00… Transcript   intron_var…\n",
      "2 3_179170077_CACACACACA… 3:17917… CACAC… ENSG… ENST00… Transcript   intron_var…\n",
      "3 3_179170077_CACACACACA… 3:17917… CACAC… ENSG… ENST00… Transcript   intron_var…\n",
      "4 3_179170077_CACACACACA… 3:17917… CACAC… ENSG… ENST00… Transcript   intron_var…\n",
      "5 3_179170077_CACACACACA… 3:17917… CACAC… ENSG… ENST00… Transcript   intron_var…\n",
      "6 3_179198731_T/C         3:17919… C      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 6 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 5 × 44\n",
      "  `#uploaded_variation`   location allele gene  feature feature_type consequence\n",
      "  <chr>                   <chr>    <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 3_179170077_CACACACACA… 3:17917… CACAC… ENSG… ENST00… Transcript   intron_var…\n",
      "2 3_179170077_CACACACACA… 3:17917… CACAC… ENSG… ENST00… Transcript   intron_var…\n",
      "3 3_179170077_CACACACACA… 3:17917… CACAC… ENSG… ENST00… Transcript   intron_var…\n",
      "4 3_179170077_CACACACACA… 3:17917… CACAC… ENSG… ENST00… Transcript   intron_var…\n",
      "5 3_179170077_CACACACACA… 3:17917… CACAC… ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>,\n",
      "#   freqs <chr>, clin_sig <chr>, somatic <chr>, pheno <chr>, pubmed <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 31 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "2 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   downstream…\n",
      "3 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   downstream…\n",
      "4 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "5 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "6 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 25 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 8 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 3_179226200_A/ACTTGA  3:1792262… ACTTGA ENSG… ENST00… Transcript   intron_var…\n",
      "2 3_179226200_A/ACTTGA  3:1792262… ACTTGA ENSG… ENST00… Transcript   intron_var…\n",
      "3 3_179226200_A/ACTTGA  3:1792262… ACTTGA ENSG… ENST00… Transcript   intron_var…\n",
      "4 3_179226200_A/ACTTGA  3:1792262… ACTTGA ENSG… ENST00… Transcript   intron_var…\n",
      "5 3_179226200_A/ACTTGA  3:1792262… ACTTGA ENSG… ENST00… Transcript   intron_var…\n",
      "6 3_179226200_A/ACTTGA  3:1792262… ACTTGA ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 2 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 25 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "2 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   downstream…\n",
      "3 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   downstream…\n",
      "4 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "5 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "6 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 19 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "Touching output file tasks/13parsing_dataR.done.\n",
      "[Mon Aug 28 18:32:42 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T183238.963328.snakemake.log\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 86 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 3_179199217_A/G       3:1791992… G      ENSG… ENST00… Transcript   intron_var…\n",
      "2 3_179199217_A/G       3:1791992… G      ENSG… ENST00… Transcript   downstream…\n",
      "3 3_179199217_A/G       3:1791992… G      ENSG… ENST00… Transcript   downstream…\n",
      "4 3_179199217_A/G       3:1791992… G      ENSG… ENST00… Transcript   intron_var…\n",
      "5 3_179199217_A/G       3:1791992… G      ENSG… ENST00… Transcript   upstream_g…\n",
      "6 3_179199217_A/G       3:1791992… G      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 80 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "[1] \"##==========================##\"\n",
      "[1] \"##===> WORK FINISHED!!! <===##\"\n",
      "[1] \"##==========================##\"\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['snakemake', '--cores', '1', 'parsing_dataR'], returncode=0)"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Running the workflow for the 3º chormosome\n",
    "\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"data/reference/genome.fa\"]) ## rule reference_genome \n",
    "\n",
    "for sample in sample_list:\n",
    "    ## Quality inspection\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/fastqc_result/\" + sample + \"_chr3_1_fastqc.html\",   \n",
    "                    \"data/processed/\" + sample + \"_chr3_1_fastp.fastq.gz\",   \n",
    "                    \"results/fastqc_result/trimmed/\" + sample + \"_chr3_1_fastp_fastqc.html\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    ## Mapping reads\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/mapped_reads/\" + sample + \"_chr3_sorted.sam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr3_sorted.bam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr3_dedup.bam\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    # Extracting variants  \n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/\" + sample + \"_chr3.vcf\"])   \n",
    "\n",
    "## Downloading VEP\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"vep_install_db\"])\n",
    "## VEP CLI\n",
    "for sample in sample_list:\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/vep/\" + sample + \"_chr3.txt\"])\n",
    "\n",
    "## Extracting the gene PIKCA3 from the VEP files\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"parsing_dataR\"])\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Changing the config file to do the chromosome 5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['sed', '-i', '65s/PIK3CA/APC/g', 'config.yaml'], returncode=0)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subprocess.run([\"sed\", \"-i\", \"6,+12s/chr3/chr5/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"24,+12s/chr3/chr5/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\",\"39s/chr3/chr5/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"46s/chromosome.3.fa.gz/chromosome.5.fa.gz/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"65s/PIK3CA/APC/g\", \"config.yaml\"])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job                 count    min threads    max threads\n",
      "----------------  -------  -------------  -------------\n",
      "reference_genome        1              1              1\n",
      "total                   1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:40:40 2023]\n",
      "rule reference_genome:\n",
      "    output: data/reference/genome.fa\n",
      "    jobid: 0\n",
      "    reason: Forced execution\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "--2023-08-28 20:40:40--  https://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.5.fa.gz\n",
      "Resolviendo ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.169\n",
      "Conectando con ftp.ensembl.org (ftp.ensembl.org)[193.62.193.169]:443... conectado.\n",
      "Petición HTTP enviada, esperando respuesta... 200 OK\n",
      "Longitud: 54485176 (52M) [application/x-gzip]\n",
      "Guardando como: ‘data/reference/genome.fa.gz’\n",
      "\n",
      "     0K .......... .......... .......... .......... ..........  0%  439K 2m1s\n",
      "    50K .......... .......... .......... .......... ..........  0% 4,94M 66s\n",
      "   100K .......... .......... .......... .......... ..........  0% 1,01M 61s\n",
      "   150K .......... .......... .......... .......... ..........  0% 5,57M 48s\n",
      "   200K .......... .......... .......... .......... ..........  0% 1,02M 48s\n",
      "   250K .......... .......... .......... .......... ..........  0% 67,7M 40s\n",
      "   300K .......... .......... .......... .......... ..........  0% 40,4M 35s\n",
      "   350K .......... .......... .......... .......... ..........  0% 8,17M 31s\n",
      "   400K .......... .......... .......... .......... ..........  0% 22,3M 28s\n",
      "   450K .......... .......... .......... .......... ..........  0% 1,03M 30s\n",
      "   500K .......... .......... .......... .......... ..........  1%  121M 27s\n",
      "   550K .......... .......... .......... .......... ..........  1% 98,6M 25s\n",
      "   600K .......... .......... .......... .......... ..........  1%  103M 23s\n",
      "   650K .......... .......... .......... .......... ..........  1%  120M 22s\n",
      "   700K .......... .......... .......... .......... ..........  1%  103M 20s\n",
      "   750K .......... .......... .......... .......... ..........  1% 62,0M 19s\n",
      "   800K .......... .......... .......... .......... ..........  1% 8,77M 18s\n",
      "   850K .......... .......... .......... .......... ..........  1% 36,6M 17s\n",
      "   900K .......... .......... .......... .......... ..........  1% 49,2M 16s\n",
      "   950K .......... .......... .......... .......... ..........  1% 1,05M 18s\n",
      "  1000K .......... .......... .......... .......... ..........  1%  124M 17s\n",
      "  1050K .......... .......... .......... .......... ..........  2%  141M 16s\n",
      "  1100K .......... .......... .......... .......... ..........  2%  102M 16s\n",
      "  1150K .......... .......... .......... .......... ..........  2%  117M 15s\n",
      "  1200K .......... .......... .......... .......... ..........  2%  102M 14s\n",
      "  1250K .......... .......... .......... .......... ..........  2%  121M 14s\n",
      "  1300K .......... .......... .......... .......... ..........  2%  119M 13s\n",
      "  1350K .......... .......... .......... .......... ..........  2%  102M 13s\n",
      "  1400K .......... .......... .......... .......... ..........  2%  119M 12s\n",
      "  1450K .......... .......... .......... .......... ..........  2%  102M 12s\n",
      "  1500K .......... .......... .......... .......... ..........  2%  122M 12s\n",
      "  1550K .......... .......... .......... .......... ..........  3%  117M 11s\n",
      "  1600K .......... .......... .......... .......... ..........  3% 20,0M 11s\n",
      "  1650K .......... .......... .......... .......... ..........  3% 47,5M 11s\n",
      "  1700K .......... .......... .......... .......... ..........  3% 43,5M 10s\n",
      "  1750K .......... .......... .......... .......... ..........  3%  121M 10s\n",
      "  1800K .......... .......... .......... .......... ..........  3%  118M 10s\n",
      "  1850K .......... .......... .......... .......... ..........  3%  102M 10s\n",
      "  1900K .......... .......... .......... .......... ..........  3% 2,49M 10s\n",
      "  1950K .......... .......... .......... .......... ..........  3% 1,84M 10s\n",
      "  2000K .......... .......... .......... .......... ..........  3%  117M 10s\n",
      "  2050K .......... .......... .......... .......... ..........  3%  121M 10s\n",
      "  2100K .......... .......... .......... .......... ..........  4%  102M 10s\n",
      "  2150K .......... .......... .......... .......... ..........  4%  107M 9s\n",
      "  2200K .......... .......... .......... .......... ..........  4%  112M 9s\n",
      "  2250K .......... .......... .......... .......... ..........  4%  121M 9s\n",
      "  2300K .......... .......... .......... .......... ..........  4%  121M 9s\n",
      "  2350K .......... .......... .......... .......... ..........  4%  102M 9s\n",
      "  2400K .......... .......... .......... .......... ..........  4%  117M 8s\n",
      "  2450K .......... .......... .......... .......... ..........  4%  102M 8s\n",
      "  2500K .......... .......... .......... .......... ..........  4%  121M 8s\n",
      "  2550K .......... .......... .......... .......... ..........  4%  117M 8s\n",
      "  2600K .......... .......... .......... .......... ..........  4%  102M 8s\n",
      "  2650K .......... .......... .......... .......... ..........  5%  121M 8s\n",
      "  2700K .......... .......... .......... .......... ..........  5%  102M 7s\n",
      "  2750K .......... .......... .......... .......... ..........  5%  118M 7s\n",
      "  2800K .......... .......... .......... .......... ..........  5%  121M 7s\n",
      "  2850K .......... .......... .......... .......... ..........  5%  102M 7s\n",
      "  2900K .......... .......... .......... .......... ..........  5%  121M 7s\n",
      "  2950K .......... .......... .......... .......... ..........  5%  102M 7s\n",
      "  3000K .......... .......... .......... .......... ..........  5%  117M 7s\n",
      "  3050K .......... .......... .......... .......... ..........  5%  121M 7s\n",
      "  3100K .......... .......... .......... .......... ..........  5%  102M 7s\n",
      "  3150K .......... .......... .......... .......... ..........  6%  117M 6s\n",
      "  3200K .......... .......... .......... .......... ..........  6%  102M 6s\n",
      "  3250K .......... .......... .......... .......... ..........  6%  121M 6s\n",
      "  3300K .......... .......... .......... .......... ..........  6%  117M 6s\n",
      "  3350K .......... .......... .......... .......... ..........  6%  102M 6s\n",
      "  3400K .......... .......... .......... .......... ..........  6%  121M 6s\n",
      "  3450K .......... .......... .......... .......... ..........  6%  102M 6s\n",
      "  3500K .......... .......... .......... .......... ..........  6%  121M 6s\n",
      "  3550K .......... .......... .......... .......... ..........  6%  117M 6s\n",
      "  3600K .......... .......... .......... .......... ..........  6% 6,42M 6s\n",
      "  3650K .......... .......... .......... .......... ..........  6%  117M 6s\n",
      "  3700K .......... .......... .......... .......... ..........  7%  103M 6s\n",
      "  3750K .......... .......... .......... .......... ..........  7%  117M 6s\n",
      "  3800K .......... .......... .......... .......... ..........  7%  119M 5s\n",
      "  3850K .......... .......... .......... .......... ..........  7% 1,66M 6s\n",
      "  3900K .......... .......... .......... .......... ..........  7% 9,91M 6s\n",
      "  3950K .......... .......... .......... .......... ..........  7%  102M 6s\n",
      "  4000K .......... .......... .......... .......... ..........  7% 25,3M 6s\n",
      "  4050K .......... .......... .......... .......... ..........  7%  112M 6s\n",
      "  4100K .......... .......... .......... .......... ..........  7%  108M 5s\n",
      "  4150K .......... .......... .......... .......... ..........  7%  117M 5s\n",
      "  4200K .......... .......... .......... .......... ..........  7% 77,6M 5s\n",
      "  4250K .......... .......... .......... .......... ..........  8%  119M 5s\n",
      "  4300K .......... .......... .......... .......... ..........  8%  123M 5s\n",
      "  4350K .......... .......... .......... .......... ..........  8%  102M 5s\n",
      "  4400K .......... .......... .......... .......... ..........  8%  117M 5s\n",
      "  4450K .......... .......... .......... .......... ..........  8%  102M 5s\n",
      "  4500K .......... .......... .......... .......... ..........  8%  121M 5s\n",
      "  4550K .......... .......... .......... .......... ..........  8%  111M 5s\n",
      "  4600K .......... .......... .......... .......... ..........  8%  106M 5s\n",
      "  4650K .......... .......... .......... .......... ..........  8%  118M 5s\n",
      "  4700K .......... .......... .......... .......... ..........  8%  101M 5s\n",
      "  4750K .......... .......... .......... .......... ..........  9%  118M 5s\n",
      "  4800K .......... .......... .......... .......... ..........  9%  119M 5s\n",
      "  4850K .......... .......... .......... .......... ..........  9%  103M 5s\n",
      "  4900K .......... .......... .......... .......... ..........  9%  119M 5s\n",
      "  4950K .......... .......... .......... .......... ..........  9%  101M 5s\n",
      "  5000K .......... .......... .......... .......... ..........  9%  120M 4s\n",
      "  5050K .......... .......... .......... .......... ..........  9%  116M 4s\n",
      "  5100K .......... .......... .......... .......... ..........  9% 1,65M 5s\n",
      "  5150K .......... .......... .......... .......... ..........  9% 1,33M 5s\n",
      "  5200K .......... .......... .......... .......... ..........  9%  102M 5s\n",
      "  5250K .......... .......... .......... .......... ..........  9%  118M 5s\n",
      "  5300K .......... .......... .......... .......... .......... 10%  120M 5s\n",
      "  5350K .......... .......... .......... .......... .......... 10%  102M 5s\n",
      "  5400K .......... .......... .......... .......... .......... 10%  113M 5s\n",
      "  5450K .......... .......... .......... .......... .......... 10%  108M 5s\n",
      "  5500K .......... .......... .......... .......... .......... 10%  118M 5s\n",
      "  5550K .......... .......... .......... .......... .......... 10%  119M 5s\n",
      "  5600K .......... .......... .......... .......... .......... 10%  102M 5s\n",
      "  5650K .......... .......... .......... .......... .......... 10%  119M 5s\n",
      "  5700K .......... .......... .......... .......... .......... 10%  102M 4s\n",
      "  5750K .......... .......... .......... .......... .......... 10%  117M 4s\n",
      "  5800K .......... .......... .......... .......... .......... 10%  119M 4s\n",
      "  5850K .......... .......... .......... .......... .......... 11%  103M 4s\n",
      "  5900K .......... .......... .......... .......... .......... 11%  119M 4s\n",
      "  5950K .......... .......... .......... .......... .......... 11%  101M 4s\n",
      "  6000K .......... .......... .......... .......... .......... 11%  121M 4s\n",
      "  6050K .......... .......... .......... .......... .......... 11%  117M 4s\n",
      "  6100K .......... .......... .......... .......... .......... 11%  102M 4s\n",
      "  6150K .......... .......... .......... .......... .......... 11%  119M 4s\n",
      "  6200K .......... .......... .......... .......... .......... 11%  103M 4s\n",
      "  6250K .......... .......... .......... .......... .......... 11%  119M 4s\n",
      "  6300K .......... .......... .......... .......... .......... 11%  120M 4s\n",
      "  6350K .......... .......... .......... .......... .......... 12%  102M 4s\n",
      "  6400K .......... .......... .......... .......... .......... 12%  117M 4s\n",
      "  6450K .......... .......... .......... .......... .......... 12%  102M 4s\n",
      "  6500K .......... .......... .......... .......... .......... 12%  113M 4s\n",
      "  6550K .......... .......... .......... .......... .......... 12%  119M 4s\n",
      "  6600K .......... .......... .......... .......... .......... 12%  102M 4s\n",
      "  6650K .......... .......... .......... .......... .......... 12%  124M 4s\n",
      "  6700K .......... .......... .......... .......... .......... 12%  102M 4s\n",
      "  6750K .......... .......... .......... .......... .......... 12%  121M 4s\n",
      "  6800K .......... .......... .......... .......... .......... 12%  117M 4s\n",
      "  6850K .......... .......... .......... .......... .......... 12%  103M 4s\n",
      "  6900K .......... .......... .......... .......... .......... 13%  119M 4s\n",
      "  6950K .......... .......... .......... .......... .......... 13%  100M 4s\n",
      "  7000K .......... .......... .......... .......... .......... 13%  122M 4s\n",
      "  7050K .......... .......... .......... .......... .......... 13%  116M 4s\n",
      "  7100K .......... .......... .......... .......... .......... 13%  103M 4s\n",
      "  7150K .......... .......... .......... .......... .......... 13%  119M 4s\n",
      "  7200K .......... .......... .......... .......... .......... 13%  102M 4s\n",
      "  7250K .......... .......... .......... .......... .......... 13% 9,34M 4s\n",
      "  7300K .......... .......... .......... .......... .......... 13%  129M 3s\n",
      "  7350K .......... .......... .......... .......... .......... 13% 3,01M 4s\n",
      "  7400K .......... .......... .......... .......... .......... 14%  121M 4s\n",
      "  7450K .......... .......... .......... .......... .......... 14%  102M 4s\n",
      "  7500K .......... .......... .......... .......... .......... 14%  119M 3s\n",
      "  7550K .......... .......... .......... .......... .......... 14%  119M 3s\n",
      "  7600K .......... .......... .......... .......... .......... 14%  102M 3s\n",
      "  7650K .......... .......... .......... .......... .......... 14% 82,3M 3s\n",
      "  7700K .......... .......... .......... .......... .......... 14%  102M 3s\n",
      "  7750K .......... .......... .......... .......... .......... 14%  119M 3s\n",
      "  7800K .......... .......... .......... .......... .......... 14%  119M 3s\n",
      "  7850K .......... .......... .......... .......... .......... 14%  101M 3s\n",
      "  7900K .......... .......... .......... .......... .......... 14% 7,22M 3s\n",
      "  7950K .......... .......... .......... .......... .......... 15%  102M 3s\n",
      "  8000K .......... .......... .......... .......... .......... 15% 8,21M 3s\n",
      "  8050K .......... .......... .......... .......... .......... 15%  780M 3s\n",
      "  8100K .......... .......... .......... .......... .......... 15%  140M 3s\n",
      "  8150K .......... .......... .......... .......... .......... 15%  118M 3s\n",
      "  8200K .......... .......... .......... .......... .......... 15%  103M 3s\n",
      "  8250K .......... .......... .......... .......... .......... 15%  116M 3s\n",
      "  8300K .......... .......... .......... .......... .......... 15%  124M 3s\n",
      "  8350K .......... .......... .......... .......... .......... 15%  101M 3s\n",
      "  8400K .......... .......... .......... .......... .......... 15%  121M 3s\n",
      "  8450K .......... .......... .......... .......... .......... 15% 99,6M 3s\n",
      "  8500K .......... .......... .......... .......... .......... 16%  120M 3s\n",
      "  8550K .......... .......... .......... .......... .......... 16%  117M 3s\n",
      "  8600K .......... .......... .......... .......... .......... 16%  102M 3s\n",
      "  8650K .......... .......... .......... .......... .......... 16%  121M 3s\n",
      "  8700K .......... .......... .......... .......... .......... 16%  101M 3s\n",
      "  8750K .......... .......... .......... .......... .......... 16%  117M 3s\n",
      "  8800K .......... .......... .......... .......... .......... 16%  122M 3s\n",
      "  8850K .......... .......... .......... .......... .......... 16%  102M 3s\n",
      "  8900K .......... .......... .......... .......... .......... 16%  118M 3s\n",
      "  8950K .......... .......... .......... .......... .......... 16%  100M 3s\n",
      "  9000K .......... .......... .......... .......... .......... 17%  121M 3s\n",
      "  9050K .......... .......... .......... .......... .......... 17%  119M 3s\n",
      "  9100K .......... .......... .......... .......... .......... 17%  100M 3s\n",
      "  9150K .......... .......... .......... .......... .......... 17%  121M 3s\n",
      "  9200K .......... .......... .......... .......... .......... 17%  102M 3s\n",
      "  9250K .......... .......... .......... .......... .......... 17%  118M 3s\n",
      "  9300K .......... .......... .......... .......... .......... 17%  120M 3s\n",
      "  9350K .......... .......... .......... .......... .......... 17%  102M 3s\n",
      "  9400K .......... .......... .......... .......... .......... 17%  118M 3s\n",
      "  9450K .......... .......... .......... .......... .......... 17%  102M 3s\n",
      "  9500K .......... .......... .......... .......... .......... 17%  119M 3s\n",
      "  9550K .......... .......... .......... .......... .......... 18%  120M 3s\n",
      "  9600K .......... .......... .......... .......... .......... 18%  101M 3s\n",
      "  9650K .......... .......... .......... .......... .......... 18% 23,9M 3s\n",
      "  9700K .......... .......... .......... .......... .......... 18%  102M 3s\n",
      "  9750K .......... .......... .......... .......... .......... 18%  117M 3s\n",
      "  9800K .......... .......... .......... .......... .......... 18%  121M 3s\n",
      "  9850K .......... .......... .......... .......... .......... 18%  739K 3s\n",
      "  9900K .......... .......... .......... .......... .......... 18%  118M 3s\n",
      "  9950K .......... .......... .......... .......... .......... 18%  102M 3s\n",
      " 10000K .......... .......... .......... .......... .......... 18%  119M 3s\n",
      " 10050K .......... .......... .......... .......... .......... 18%  117M 3s\n",
      " 10100K .......... .......... .......... .......... .......... 19%  102M 3s\n",
      " 10150K .......... .......... .......... .......... .......... 19%  119M 3s\n",
      " 10200K .......... .......... .......... .......... .......... 19%  103M 3s\n",
      " 10250K .......... .......... .......... .......... .......... 19%  120M 3s\n",
      " 10300K .......... .......... .......... .......... .......... 19%  117M 3s\n",
      " 10350K .......... .......... .......... .......... .......... 19%  101M 3s\n",
      " 10400K .......... .......... .......... .......... .......... 19%  120M 3s\n",
      " 10450K .......... .......... .......... .......... .......... 19% 99,8M 3s\n",
      " 10500K .......... .......... .......... .......... .......... 19%  122M 3s\n",
      " 10550K .......... .......... .......... .......... .......... 19%  119M 3s\n",
      " 10600K .......... .......... .......... .......... .......... 20%  103M 3s\n",
      " 10650K .......... .......... .......... .......... .......... 20%  114M 3s\n",
      " 10700K .......... .......... .......... .......... .......... 20%  100M 3s\n",
      " 10750K .......... .......... .......... .......... .......... 20%  128M 3s\n",
      " 10800K .......... .......... .......... .......... .......... 20%  117M 3s\n",
      " 10850K .......... .......... .......... .......... .......... 20% 97,3M 3s\n",
      " 10900K .......... .......... .......... .......... .......... 20%  119M 3s\n",
      " 10950K .......... .......... .......... .......... .......... 20%  107M 3s\n",
      " 11000K .......... .......... .......... .......... .......... 20%  118M 3s\n",
      " 11050K .......... .......... .......... .......... .......... 20%  119M 3s\n",
      " 11100K .......... .......... .......... .......... .......... 20%  101M 3s\n",
      " 11150K .......... .......... .......... .......... .......... 21%  117M 3s\n",
      " 11200K .......... .......... .......... .......... .......... 21%  103M 3s\n",
      " 11250K .......... .......... .......... .......... .......... 21%  117M 3s\n",
      " 11300K .......... .......... .......... .......... .......... 21%  119M 3s\n",
      " 11350K .......... .......... .......... .......... .......... 21%  103M 3s\n",
      " 11400K .......... .......... .......... .......... .......... 21%  119M 3s\n",
      " 11450K .......... .......... .......... .......... .......... 21%  100M 3s\n",
      " 11500K .......... .......... .......... .......... .......... 21%  121M 2s\n",
      " 11550K .......... .......... .......... .......... .......... 21%  119M 2s\n",
      " 11600K .......... .......... .......... .......... .......... 21% 96,7M 2s\n",
      " 11650K .......... .......... .......... .......... .......... 21%  127M 2s\n",
      " 11700K .......... .......... .......... .......... .......... 22% 98,9M 2s\n",
      " 11750K .......... .......... .......... .......... .......... 22% 1,16M 3s\n",
      " 11800K .......... .......... .......... .......... .......... 22%  116M 3s\n",
      " 11850K .......... .......... .......... .......... .......... 22%  104M 3s\n",
      " 11900K .......... .......... .......... .......... .......... 22%  120M 3s\n",
      " 11950K .......... .......... .......... .......... .......... 22%  102M 3s\n",
      " 12000K .......... .......... .......... .......... .......... 22%  119M 3s\n",
      " 12050K .......... .......... .......... .......... .......... 22% 80,3M 3s\n",
      " 12100K .......... .......... .......... .......... .......... 22% 3,43M 3s\n",
      " 12150K .......... .......... .......... .......... .......... 22%  143M 3s\n",
      " 12200K .......... .......... .......... .......... .......... 23%  101M 3s\n",
      " 12250K .......... .......... .......... .......... .......... 23%  113M 3s\n",
      " 12300K .......... .......... .......... .......... .......... 23%  119M 3s\n",
      " 12350K .......... .......... .......... .......... .......... 23%  106M 2s\n",
      " 12400K .......... .......... .......... .......... .......... 23%  114M 2s\n",
      " 12450K .......... .......... .......... .......... .......... 23%  107M 2s\n",
      " 12500K .......... .......... .......... .......... .......... 23%  117M 2s\n",
      " 12550K .......... .......... .......... .......... .......... 23%  121M 2s\n",
      " 12600K .......... .......... .......... .......... .......... 23%  101M 2s\n",
      " 12650K .......... .......... .......... .......... .......... 23%  120M 2s\n",
      " 12700K .......... .......... .......... .......... .......... 23% 97,5M 2s\n",
      " 12750K .......... .......... .......... .......... .......... 24%  126M 2s\n",
      " 12800K .......... .......... .......... .......... .......... 24%  119M 2s\n",
      " 12850K .......... .......... .......... .......... .......... 24% 96,7M 2s\n",
      " 12900K .......... .......... .......... .......... .......... 24%  119M 2s\n",
      " 12950K .......... .......... .......... .......... .......... 24%  102M 2s\n",
      " 13000K .......... .......... .......... .......... .......... 24%  126M 2s\n",
      " 13050K .......... .......... .......... .......... .......... 24%  119M 2s\n",
      " 13100K .......... .......... .......... .......... .......... 24%  100M 2s\n",
      " 13150K .......... .......... .......... .......... .......... 24%  114M 2s\n",
      " 13200K .......... .......... .......... .......... .......... 24%  106M 2s\n",
      " 13250K .......... .......... .......... .......... .......... 24%  119M 2s\n",
      " 13300K .......... .......... .......... .......... .......... 25%  119M 2s\n",
      " 13350K .......... .......... .......... .......... .......... 25%  102M 2s\n",
      " 13400K .......... .......... .......... .......... .......... 25%  119M 2s\n",
      " 13450K .......... .......... .......... .......... .......... 25%  101M 2s\n",
      " 13500K .......... .......... .......... .......... .......... 25%  118M 2s\n",
      " 13550K .......... .......... .......... .......... .......... 25%  119M 2s\n",
      " 13600K .......... .......... .......... .......... .......... 25%  102M 2s\n",
      " 13650K .......... .......... .......... .......... .......... 25%  119M 2s\n",
      " 13700K .......... .......... .......... .......... .......... 25%  102M 2s\n",
      " 13750K .......... .......... .......... .......... .......... 25%  117M 2s\n",
      " 13800K .......... .......... .......... .......... .......... 26%  119M 2s\n",
      " 13850K .......... .......... .......... .......... .......... 26%  102M 2s\n",
      " 13900K .......... .......... .......... .......... .......... 26%  121M 2s\n",
      " 13950K .......... .......... .......... .......... .......... 26%  102M 2s\n",
      " 14000K .......... .......... .......... .......... .......... 26%  117M 2s\n",
      " 14050K .......... .......... .......... .......... .......... 26%  121M 2s\n",
      " 14100K .......... .......... .......... .......... .......... 26%  102M 2s\n",
      " 14150K .......... .......... .......... .......... .......... 26%  119M 2s\n",
      " 14200K .......... .......... .......... .......... .......... 26%  102M 2s\n",
      " 14250K .......... .......... .......... .......... .......... 26%  119M 2s\n",
      " 14300K .......... .......... .......... .......... .......... 26%  119M 2s\n",
      " 14350K .......... .......... .......... .......... .......... 27%  102M 2s\n",
      " 14400K .......... .......... .......... .......... .......... 27%  119M 2s\n",
      " 14450K .......... .......... .......... .......... .......... 27%  102M 2s\n",
      " 14500K .......... .......... .......... .......... .......... 27%  119M 2s\n",
      " 14550K .......... .......... .......... .......... .......... 27%  117M 2s\n",
      " 14600K .......... .......... .......... .......... .......... 27%  103M 2s\n",
      " 14650K .......... .......... .......... .......... .......... 27%  119M 2s\n",
      " 14700K .......... .......... .......... .......... .......... 27%  102M 2s\n",
      " 14750K .......... .......... .......... .......... .......... 27%  116M 2s\n",
      " 14800K .......... .......... .......... .......... .......... 27% 2,73M 2s\n",
      " 14850K .......... .......... .......... .......... .......... 28%  105M 2s\n",
      " 14900K .......... .......... .......... .......... .......... 28%  117M 2s\n",
      " 14950K .......... .......... .......... .......... .......... 28%  102M 2s\n",
      " 15000K .......... .......... .......... .......... .......... 28%  121M 2s\n",
      " 15050K .......... .......... .......... .......... .......... 28%  117M 2s\n",
      " 15100K .......... .......... .......... .......... .......... 28%  102M 2s\n",
      " 15150K .......... .......... .......... .......... .......... 28%  121M 2s\n",
      " 15200K .......... .......... .......... .......... .......... 28% 3,27M 2s\n",
      " 15250K .......... .......... .......... .......... .......... 28%  121M 2s\n",
      " 15300K .......... .......... .......... .......... .......... 28%  117M 2s\n",
      " 15350K .......... .......... .......... .......... .......... 28%  102M 2s\n",
      " 15400K .......... .......... .......... .......... .......... 29%  121M 2s\n",
      " 15450K .......... .......... .......... .......... .......... 29%  102M 2s\n",
      " 15500K .......... .......... .......... .......... .......... 29%  117M 2s\n",
      " 15550K .......... .......... .......... .......... .......... 29%  121M 2s\n",
      " 15600K .......... .......... .......... .......... .......... 29%  102M 2s\n",
      " 15650K .......... .......... .......... .......... .......... 29%  117M 2s\n",
      " 15700K .......... .......... .......... .......... .......... 29%  102M 2s\n",
      " 15750K .......... .......... .......... .......... .......... 29%  120M 2s\n",
      " 15800K .......... .......... .......... .......... .......... 29%  121M 2s\n",
      " 15850K .......... .......... .......... .......... .......... 29%  102M 2s\n",
      " 15900K .......... .......... .......... .......... .......... 29%  117M 2s\n",
      " 15950K .......... .......... .......... .......... .......... 30%  102M 2s\n",
      " 16000K .......... .......... .......... .......... .......... 30%  120M 2s\n",
      " 16050K .......... .......... .......... .......... .......... 30%  117M 2s\n",
      " 16100K .......... .......... .......... .......... .......... 30%  102M 2s\n",
      " 16150K .......... .......... .......... .......... .......... 30%  121M 2s\n",
      " 16200K .......... .......... .......... .......... .......... 30%  102M 2s\n",
      " 16250K .......... .......... .......... .......... .......... 30%  122M 2s\n",
      " 16300K .......... .......... .......... .......... .......... 30%  117M 2s\n",
      " 16350K .......... .......... .......... .......... .......... 30%  102M 2s\n",
      " 16400K .......... .......... .......... .......... .......... 30%  121M 2s\n",
      " 16450K .......... .......... .......... .......... .......... 31%  101M 2s\n",
      " 16500K .......... .......... .......... .......... .......... 31%  119M 2s\n",
      " 16550K .......... .......... .......... .......... .......... 31%  121M 2s\n",
      " 16600K .......... .......... .......... .......... .......... 31%  102M 2s\n",
      " 16650K .......... .......... .......... .......... .......... 31%  117M 2s\n",
      " 16700K .......... .......... .......... .......... .......... 31%  104M 2s\n",
      " 16750K .......... .......... .......... .......... .......... 31%  118M 2s\n",
      " 16800K .......... .......... .......... .......... .......... 31%  121M 2s\n",
      " 16850K .......... .......... .......... .......... .......... 31%  102M 2s\n",
      " 16900K .......... .......... .......... .......... .......... 31%  117M 2s\n",
      " 16950K .......... .......... .......... .......... .......... 31%  102M 2s\n",
      " 17000K .......... .......... .......... .......... .......... 32%  121M 2s\n",
      " 17050K .......... .......... .......... .......... .......... 32%  117M 2s\n",
      " 17100K .......... .......... .......... .......... .......... 32%  102M 2s\n",
      " 17150K .......... .......... .......... .......... .......... 32%  121M 2s\n",
      " 17200K .......... .......... .......... .......... .......... 32%  102M 2s\n",
      " 17250K .......... .......... .......... .......... .......... 32%  117M 2s\n",
      " 17300K .......... .......... .......... .......... .......... 32%  121M 2s\n",
      " 17350K .......... .......... .......... .......... .......... 32%  102M 2s\n",
      " 17400K .......... .......... .......... .......... .......... 32%  121M 2s\n",
      " 17450K .......... .......... .......... .......... .......... 32%  102M 2s\n",
      " 17500K .......... .......... .......... .......... .......... 32%  117M 2s\n",
      " 17550K .......... .......... .......... .......... .......... 33%  121M 2s\n",
      " 17600K .......... .......... .......... .......... .......... 33%  102M 2s\n",
      " 17650K .......... .......... .......... .......... .......... 33%  117M 2s\n",
      " 17700K .......... .......... .......... .......... .......... 33%  102M 2s\n",
      " 17750K .......... .......... .......... .......... .......... 33%  121M 2s\n",
      " 17800K .......... .......... .......... .......... .......... 33%  117M 2s\n",
      " 17850K .......... .......... .......... .......... .......... 33% 3,07M 2s\n",
      " 17900K .......... .......... .......... .......... .......... 33% 27,1M 2s\n",
      " 17950K .......... .......... .......... .......... .......... 33% 54,1M 2s\n",
      " 18000K .......... .......... .......... .......... .......... 33%  121M 2s\n",
      " 18050K .......... .......... .......... .......... .......... 34%  117M 2s\n",
      " 18100K .......... .......... .......... .......... .......... 34%  102M 2s\n",
      " 18150K .......... .......... .......... .......... .......... 34%  121M 2s\n",
      " 18200K .......... .......... .......... .......... .......... 34%  102M 2s\n",
      " 18250K .......... .......... .......... .......... .......... 34% 3,58M 2s\n",
      " 18300K .......... .......... .......... .......... .......... 34% 41,6M 2s\n",
      " 18350K .......... .......... .......... .......... .......... 34% 53,9M 2s\n",
      " 18400K .......... .......... .......... .......... .......... 34%  120M 2s\n",
      " 18450K .......... .......... .......... .......... .......... 34%  102M 2s\n",
      " 18500K .......... .......... .......... .......... .......... 34%  117M 2s\n",
      " 18550K .......... .......... .......... .......... .......... 34%  121M 2s\n",
      " 18600K .......... .......... .......... .......... .......... 35%  102M 2s\n",
      " 18650K .......... .......... .......... .......... .......... 35%  117M 2s\n",
      " 18700K .......... .......... .......... .......... .......... 35%  103M 2s\n",
      " 18750K .......... .......... .......... .......... .......... 35%  121M 2s\n",
      " 18800K .......... .......... .......... .......... .......... 35%  117M 2s\n",
      " 18850K .......... .......... .......... .......... .......... 35%  103M 2s\n",
      " 18900K .......... .......... .......... .......... .......... 35%  120M 2s\n",
      " 18950K .......... .......... .......... .......... .......... 35%  102M 2s\n",
      " 19000K .......... .......... .......... .......... .......... 35%  121M 2s\n",
      " 19050K .......... .......... .......... .......... .......... 35%  117M 2s\n",
      " 19100K .......... .......... .......... .......... .......... 35%  103M 2s\n",
      " 19150K .......... .......... .......... .......... .......... 36%  121M 2s\n",
      " 19200K .......... .......... .......... .......... .......... 36%  102M 2s\n",
      " 19250K .......... .......... .......... .......... .......... 36%  117M 2s\n",
      " 19300K .......... .......... .......... .......... .......... 36%  121M 2s\n",
      " 19350K .......... .......... .......... .......... .......... 36%  102M 2s\n",
      " 19400K .......... .......... .......... .......... .......... 36%  117M 2s\n",
      " 19450K .......... .......... .......... .......... .......... 36%  102M 2s\n",
      " 19500K .......... .......... .......... .......... .......... 36%  121M 2s\n",
      " 19550K .......... .......... .......... .......... .......... 36%  121M 2s\n",
      " 19600K .......... .......... .......... .......... .......... 36%  102M 2s\n",
      " 19650K .......... .......... .......... .......... .......... 37%  117M 2s\n",
      " 19700K .......... .......... .......... .......... .......... 37%  102M 1s\n",
      " 19750K .......... .......... .......... .......... .......... 37%  121M 1s\n",
      " 19800K .......... .......... .......... .......... .......... 37%  117M 1s\n",
      " 19850K .......... .......... .......... .......... .......... 37%  102M 1s\n",
      " 19900K .......... .......... .......... .......... .......... 37%  121M 1s\n",
      " 19950K .......... .......... .......... .......... .......... 37%  102M 1s\n",
      " 20000K .......... .......... .......... .......... .......... 37%  121M 1s\n",
      " 20050K .......... .......... .......... .......... .......... 37%  117M 1s\n",
      " 20100K .......... .......... .......... .......... .......... 37%  102M 1s\n",
      " 20150K .......... .......... .......... .......... .......... 37%  120M 1s\n",
      " 20200K .......... .......... .......... .......... .......... 38%  102M 1s\n",
      " 20250K .......... .......... .......... .......... .......... 38%  117M 1s\n",
      " 20300K .......... .......... .......... .......... .......... 38%  121M 1s\n",
      " 20350K .......... .......... .......... .......... .......... 38%  102M 1s\n",
      " 20400K .......... .......... .......... .......... .......... 38%  118M 1s\n",
      " 20450K .......... .......... .......... .......... .......... 38%  102M 1s\n",
      " 20500K .......... .......... .......... .......... .......... 38%  121M 1s\n",
      " 20550K .......... .......... .......... .......... .......... 38%  121M 1s\n",
      " 20600K .......... .......... .......... .......... .......... 38%  102M 1s\n",
      " 20650K .......... .......... .......... .......... .......... 38%  117M 1s\n",
      " 20700K .......... .......... .......... .......... .......... 38%  102M 1s\n",
      " 20750K .......... .......... .......... .......... .......... 39%  121M 1s\n",
      " 20800K .......... .......... .......... .......... .......... 39%  117M 1s\n",
      " 20850K .......... .......... .......... .......... .......... 39%  102M 1s\n",
      " 20900K .......... .......... .......... .......... .......... 39%  121M 1s\n",
      " 20950K .......... .......... .......... .......... .......... 39% 2,96M 1s\n",
      " 21000K .......... .......... .......... .......... .......... 39% 45,6M 1s\n",
      " 21050K .......... .......... .......... .......... .......... 39%  121M 1s\n",
      " 21100K .......... .......... .......... .......... .......... 39%  102M 1s\n",
      " 21150K .......... .......... .......... .......... .......... 39%  121M 1s\n",
      " 21200K .......... .......... .......... .......... .......... 39% 96,2M 1s\n",
      " 21250K .......... .......... .......... .......... .......... 40%  117M 1s\n",
      " 21300K .......... .......... .......... .......... .......... 40% 3,78M 1s\n",
      " 21350K .......... .......... .......... .......... .......... 40% 19,3M 1s\n",
      " 21400K .......... .......... .......... .......... .......... 40%  117M 1s\n",
      " 21450K .......... .......... .......... .......... .......... 40% 69,9M 1s\n",
      " 21500K .......... .......... .......... .......... .......... 40%  121M 1s\n",
      " 21550K .......... .......... .......... .......... .......... 40%  117M 1s\n",
      " 21600K .......... .......... .......... .......... .......... 40%  102M 1s\n",
      " 21650K .......... .......... .......... .......... .......... 40%  121M 1s\n",
      " 21700K .......... .......... .......... .......... .......... 40%  102M 1s\n",
      " 21750K .......... .......... .......... .......... .......... 40%  121M 1s\n",
      " 21800K .......... .......... .......... .......... .......... 41%  117M 1s\n",
      " 21850K .......... .......... .......... .......... .......... 41%  102M 1s\n",
      " 21900K .......... .......... .......... .......... .......... 41%  121M 1s\n",
      " 21950K .......... .......... .......... .......... .......... 41%  102M 1s\n",
      " 22000K .......... .......... .......... .......... .......... 41%  117M 1s\n",
      " 22050K .......... .......... .......... .......... .......... 41%  121M 1s\n",
      " 22100K .......... .......... .......... .......... .......... 41%  102M 1s\n",
      " 22150K .......... .......... .......... .......... .......... 41%  120M 1s\n",
      " 22200K .......... .......... .......... .......... .......... 41%  103M 1s\n",
      " 22250K .......... .......... .......... .......... .......... 41%  117M 1s\n",
      " 22300K .......... .......... .......... .......... .......... 42%  121M 1s\n",
      " 22350K .......... .......... .......... .......... .......... 42%  102M 1s\n",
      " 22400K .......... .......... .......... .......... .......... 42%  117M 1s\n",
      " 22450K .......... .......... .......... .......... .......... 42%  103M 1s\n",
      " 22500K .......... .......... .......... .......... .......... 42%  121M 1s\n",
      " 22550K .......... .......... .......... .......... .......... 42%  117M 1s\n",
      " 22600K .......... .......... .......... .......... .......... 42%  102M 1s\n",
      " 22650K .......... .......... .......... .......... .......... 42%  121M 1s\n",
      " 22700K .......... .......... .......... .......... .......... 42%  102M 1s\n",
      " 22750K .......... .......... .......... .......... .......... 42%  121M 1s\n",
      " 22800K .......... .......... .......... .......... .......... 42%  117M 1s\n",
      " 22850K .......... .......... .......... .......... .......... 43%  102M 1s\n",
      " 22900K .......... .......... .......... .......... .......... 43%  121M 1s\n",
      " 22950K .......... .......... .......... .......... .......... 43%  102M 1s\n",
      " 23000K .......... .......... .......... .......... .......... 43%  117M 1s\n",
      " 23050K .......... .......... .......... .......... .......... 43%  121M 1s\n",
      " 23100K .......... .......... .......... .......... .......... 43%  102M 1s\n",
      " 23150K .......... .......... .......... .......... .......... 43%  117M 1s\n",
      " 23200K .......... .......... .......... .......... .......... 43%  102M 1s\n",
      " 23250K .......... .......... .......... .......... .......... 43%  121M 1s\n",
      " 23300K .......... .......... .......... .......... .......... 43%  121M 1s\n",
      " 23350K .......... .......... .......... .......... .......... 43%  102M 1s\n",
      " 23400K .......... .......... .......... .......... .......... 44%  117M 1s\n",
      " 23450K .......... .......... .......... .......... .......... 44%  102M 1s\n",
      " 23500K .......... .......... .......... .......... .......... 44%  121M 1s\n",
      " 23550K .......... .......... .......... .......... .......... 44%  117M 1s\n",
      " 23600K .......... .......... .......... .......... .......... 44%  103M 1s\n",
      " 23650K .......... .......... .......... .......... .......... 44%  120M 1s\n",
      " 23700K .......... .......... .......... .......... .......... 44%  102M 1s\n",
      " 23750K .......... .......... .......... .......... .......... 44%  121M 1s\n",
      " 23800K .......... .......... .......... .......... .......... 44%  117M 1s\n",
      " 23850K .......... .......... .......... .......... .......... 44%  102M 1s\n",
      " 23900K .......... .......... .......... .......... .......... 45%  120M 1s\n",
      " 23950K .......... .......... .......... .......... .......... 45% 79,4M 1s\n",
      " 24000K .......... .......... .......... .......... .......... 45% 2,95M 1s\n",
      " 24050K .......... .......... .......... .......... .......... 45%  124M 1s\n",
      " 24100K .......... .......... .......... .......... .......... 45% 58,1M 1s\n",
      " 24150K .......... .......... .......... .......... .......... 45%  117M 1s\n",
      " 24200K .......... .......... .......... .......... .......... 45% 88,8M 1s\n",
      " 24250K .......... .......... .......... .......... .......... 45%  147M 1s\n",
      " 24300K .......... .......... .......... .......... .......... 45%  121M 1s\n",
      " 24350K .......... .......... .......... .......... .......... 45% 4,65M 1s\n",
      " 24400K .......... .......... .......... .......... .......... 45% 10,7M 1s\n",
      " 24450K .......... .......... .......... .......... .......... 46% 61,3M 1s\n",
      " 24500K .......... .......... .......... .......... .......... 46%  121M 1s\n",
      " 24550K .......... .......... .......... .......... .......... 46%  117M 1s\n",
      " 24600K .......... .......... .......... .......... .......... 46%  102M 1s\n",
      " 24650K .......... .......... .......... .......... .......... 46%  114M 1s\n",
      " 24700K .......... .......... .......... .......... .......... 46%  107M 1s\n",
      " 24750K .......... .......... .......... .......... .......... 46%  118M 1s\n",
      " 24800K .......... .......... .......... .......... .......... 46%  120M 1s\n",
      " 24850K .......... .......... .......... .......... .......... 46%  103M 1s\n",
      " 24900K .......... .......... .......... .......... .......... 46%  120M 1s\n",
      " 24950K .......... .......... .......... .......... .......... 46% 99,7M 1s\n",
      " 25000K .......... .......... .......... .......... .......... 47%  121M 1s\n",
      " 25050K .......... .......... .......... .......... .......... 47%  120M 1s\n",
      " 25100K .......... .......... .......... .......... .......... 47%  102M 1s\n",
      " 25150K .......... .......... .......... .......... .......... 47%  117M 1s\n",
      " 25200K .......... .......... .......... .......... .......... 47%  102M 1s\n",
      " 25250K .......... .......... .......... .......... .......... 47%  121M 1s\n",
      " 25300K .......... .......... .......... .......... .......... 47%  117M 1s\n",
      " 25350K .......... .......... .......... .......... .......... 47%  102M 1s\n",
      " 25400K .......... .......... .......... .......... .......... 47%  121M 1s\n",
      " 25450K .......... .......... .......... .......... .......... 47%  103M 1s\n",
      " 25500K .......... .......... .......... .......... .......... 48%  121M 1s\n",
      " 25550K .......... .......... .......... .......... .......... 48%  117M 1s\n",
      " 25600K .......... .......... .......... .......... .......... 48%  102M 1s\n",
      " 25650K .......... .......... .......... .......... .......... 48%  121M 1s\n",
      " 25700K .......... .......... .......... .......... .......... 48%  103M 1s\n",
      " 25750K .......... .......... .......... .......... .......... 48%  117M 1s\n",
      " 25800K .......... .......... .......... .......... .......... 48%  121M 1s\n",
      " 25850K .......... .......... .......... .......... .......... 48%  102M 1s\n",
      " 25900K .......... .......... .......... .......... .......... 48%  121M 1s\n",
      " 25950K .......... .......... .......... .......... .......... 48%  102M 1s\n",
      " 26000K .......... .......... .......... .......... .......... 48%  117M 1s\n",
      " 26050K .......... .......... .......... .......... .......... 49%  121M 1s\n",
      " 26100K .......... .......... .......... .......... .......... 49%  102M 1s\n",
      " 26150K .......... .......... .......... .......... .......... 49%  117M 1s\n",
      " 26200K .......... .......... .......... .......... .......... 49%  102M 1s\n",
      " 26250K .......... .......... .......... .......... .......... 49%  120M 1s\n",
      " 26300K .......... .......... .......... .......... .......... 49%  118M 1s\n",
      " 26350K .......... .......... .......... .......... .......... 49%  103M 1s\n",
      " 26400K .......... .......... .......... .......... .......... 49%  120M 1s\n",
      " 26450K .......... .......... .......... .......... .......... 49%  103M 1s\n",
      " 26500K .......... .......... .......... .......... .......... 49%  118M 1s\n",
      " 26550K .......... .......... .......... .......... .......... 49%  119M 1s\n",
      " 26600K .......... .......... .......... .......... .......... 50%  103M 1s\n",
      " 26650K .......... .......... .......... .......... .......... 50%  121M 1s\n",
      " 26700K .......... .......... .......... .......... .......... 50%  101M 1s\n",
      " 26750K .......... .......... .......... .......... .......... 50%  119M 1s\n",
      " 26800K .......... .......... .......... .......... .......... 50%  120M 1s\n",
      " 26850K .......... .......... .......... .......... .......... 50%  102M 1s\n",
      " 26900K .......... .......... .......... .......... .......... 50%  118M 1s\n",
      " 26950K .......... .......... .......... .......... .......... 50%  102M 1s\n",
      " 27000K .......... .......... .......... .......... .......... 50%  121M 1s\n",
      " 27050K .......... .......... .......... .......... .......... 50% 3,39M 1s\n",
      " 27100K .......... .......... .......... .......... .......... 51% 16,9M 1s\n",
      " 27150K .......... .......... .......... .......... .......... 51%  118M 1s\n",
      " 27200K .......... .......... .......... .......... .......... 51% 53,8M 1s\n",
      " 27250K .......... .......... .......... .......... .......... 51%  121M 1s\n",
      " 27300K .......... .......... .......... .......... .......... 51%  117M 1s\n",
      " 27350K .......... .......... .......... .......... .......... 51% 5,82M 1s\n",
      " 27400K .......... .......... .......... .......... .......... 51% 89,3M 1s\n",
      " 27450K .......... .......... .......... .......... .......... 51% 8,80M 1s\n",
      " 27500K .......... .......... .......... .......... .......... 51% 41,5M 1s\n",
      " 27550K .......... .......... .......... .......... .......... 51% 56,7M 1s\n",
      " 27600K .......... .......... .......... .......... .......... 51%  102M 1s\n",
      " 27650K .......... .......... .......... .......... .......... 52%  121M 1s\n",
      " 27700K .......... .......... .......... .......... .......... 52% 82,7M 1s\n",
      " 27750K .......... .......... .......... .......... .......... 52%  121M 1s\n",
      " 27800K .......... .......... .......... .......... .......... 52%  121M 1s\n",
      " 27850K .......... .......... .......... .......... .......... 52%  102M 1s\n",
      " 27900K .......... .......... .......... .......... .......... 52%  118M 1s\n",
      " 27950K .......... .......... .......... .......... .......... 52%  102M 1s\n",
      " 28000K .......... .......... .......... .......... .......... 52%  121M 1s\n",
      " 28050K .......... .......... .......... .......... .......... 52%  120M 1s\n",
      " 28100K .......... .......... .......... .......... .......... 52%  102M 1s\n",
      " 28150K .......... .......... .......... .......... .......... 52%  117M 1s\n",
      " 28200K .......... .......... .......... .......... .......... 53%  102M 1s\n",
      " 28250K .......... .......... .......... .......... .......... 53%  121M 1s\n",
      " 28300K .......... .......... .......... .......... .......... 53%  117M 1s\n",
      " 28350K .......... .......... .......... .......... .......... 53%  102M 1s\n",
      " 28400K .......... .......... .......... .......... .......... 53%  120M 1s\n",
      " 28450K .......... .......... .......... .......... .......... 53%  103M 1s\n",
      " 28500K .......... .......... .......... .......... .......... 53%  117M 1s\n",
      " 28550K .......... .......... .......... .......... .......... 53%  121M 1s\n",
      " 28600K .......... .......... .......... .......... .......... 53%  102M 1s\n",
      " 28650K .......... .......... .......... .......... .......... 53%  121M 1s\n",
      " 28700K .......... .......... .......... .......... .......... 54%  102M 1s\n",
      " 28750K .......... .......... .......... .......... .......... 54%  117M 1s\n",
      " 28800K .......... .......... .......... .......... .......... 54%  121M 1s\n",
      " 28850K .......... .......... .......... .......... .......... 54%  102M 1s\n",
      " 28900K .......... .......... .......... .......... .......... 54%  117M 1s\n",
      " 28950K .......... .......... .......... .......... .......... 54%  102M 1s\n",
      " 29000K .......... .......... .......... .......... .......... 54%  121M 1s\n",
      " 29050K .......... .......... .......... .......... .......... 54%  118M 1s\n",
      " 29100K .......... .......... .......... .......... .......... 54%  102M 1s\n",
      " 29150K .......... .......... .......... .......... .......... 54%  121M 1s\n",
      " 29200K .......... .......... .......... .......... .......... 54%  103M 1s\n",
      " 29250K .......... .......... .......... .......... .......... 55%  120M 1s\n",
      " 29300K .......... .......... .......... .......... .......... 55%  117M 1s\n",
      " 29350K .......... .......... .......... .......... .......... 55%  102M 1s\n",
      " 29400K .......... .......... .......... .......... .......... 55%  121M 1s\n",
      " 29450K .......... .......... .......... .......... .......... 55%  102M 1s\n",
      " 29500K .......... .......... .......... .......... .......... 55%  118M 1s\n",
      " 29550K .......... .......... .......... .......... .......... 55%  120M 1s\n",
      " 29600K .......... .......... .......... .......... .......... 55%  103M 1s\n",
      " 29650K .......... .......... .......... .......... .......... 55%  120M 1s\n",
      " 29700K .......... .......... .......... .......... .......... 55%  102M 1s\n",
      " 29750K .......... .......... .......... .......... .......... 56%  117M 1s\n",
      " 29800K .......... .......... .......... .......... .......... 56%  121M 1s\n",
      " 29850K .......... .......... .......... .......... .......... 56%  102M 1s\n",
      " 29900K .......... .......... .......... .......... .......... 56%  118M 1s\n",
      " 29950K .......... .......... .......... .......... .......... 56%  102M 1s\n",
      " 30000K .......... .......... .......... .......... .......... 56%  120M 1s\n",
      " 30050K .......... .......... .......... .......... .......... 56%  118M 1s\n",
      " 30100K .......... .......... .......... .......... .......... 56% 3,62M 1s\n",
      " 30150K .......... .......... .......... .......... .......... 56% 14,6M 1s\n",
      " 30200K .......... .......... .......... .......... .......... 56% 43,9M 1s\n",
      " 30250K .......... .......... .......... .......... .......... 56% 66,6M 1s\n",
      " 30300K .......... .......... .......... .......... .......... 57%  117M 1s\n",
      " 30350K .......... .......... .......... .......... .......... 57%  102M 1s\n",
      " 30400K .......... .......... .......... .......... .......... 57% 62,4M 1s\n",
      " 30450K .......... .......... .......... .......... .......... 57% 5,92M 1s\n",
      " 30500K .......... .......... .......... .......... .......... 57% 9,30M 1s\n",
      " 30550K .......... .......... .......... .......... .......... 57% 39,8M 1s\n",
      " 30600K .......... .......... .......... .......... .......... 57% 60,4M 1s\n",
      " 30650K .......... .......... .......... .......... .......... 57%  124M 1s\n",
      " 30700K .......... .......... .......... .......... .......... 57%  102M 1s\n",
      " 30750K .......... .......... .......... .......... .......... 57%  122M 1s\n",
      " 30800K .......... .......... .......... .......... .......... 57%  120M 1s\n",
      " 30850K .......... .......... .......... .......... .......... 58%  102M 1s\n",
      " 30900K .......... .......... .......... .......... .......... 58%  118M 1s\n",
      " 30950K .......... .......... .......... .......... .......... 58%  101M 1s\n",
      " 31000K .......... .......... .......... .......... .......... 58%  122M 1s\n",
      " 31050K .......... .......... .......... .......... .......... 58%  117M 1s\n",
      " 31100K .......... .......... .......... .......... .......... 58%  102M 1s\n",
      " 31150K .......... .......... .......... .......... .......... 58%  121M 1s\n",
      " 31200K .......... .......... .......... .......... .......... 58%  102M 1s\n",
      " 31250K .......... .......... .......... .......... .......... 58%  121M 1s\n",
      " 31300K .......... .......... .......... .......... .......... 58%  117M 1s\n",
      " 31350K .......... .......... .......... .......... .......... 59%  102M 1s\n",
      " 31400K .......... .......... .......... .......... .......... 59%  121M 1s\n",
      " 31450K .......... .......... .......... .......... .......... 59%  102M 1s\n",
      " 31500K .......... .......... .......... .......... .......... 59%  117M 1s\n",
      " 31550K .......... .......... .......... .......... .......... 59%  121M 1s\n",
      " 31600K .......... .......... .......... .......... .......... 59%  102M 1s\n",
      " 31650K .......... .......... .......... .......... .......... 59%  117M 1s\n",
      " 31700K .......... .......... .......... .......... .......... 59%  102M 1s\n",
      " 31750K .......... .......... .......... .......... .......... 59%  121M 1s\n",
      " 31800K .......... .......... .......... .......... .......... 59%  121M 1s\n",
      " 31850K .......... .......... .......... .......... .......... 59%  102M 1s\n",
      " 31900K .......... .......... .......... .......... .......... 60%  117M 1s\n",
      " 31950K .......... .......... .......... .......... .......... 60%  102M 1s\n",
      " 32000K .......... .......... .......... .......... .......... 60%  121M 1s\n",
      " 32050K .......... .......... .......... .......... .......... 60%  117M 1s\n",
      " 32100K .......... .......... .......... .......... .......... 60%  102M 1s\n",
      " 32150K .......... .......... .......... .......... .......... 60%  121M 1s\n",
      " 32200K .......... .......... .......... .......... .......... 60%  102M 1s\n",
      " 32250K .......... .......... .......... .......... .......... 60%  117M 1s\n",
      " 32300K .......... .......... .......... .......... .......... 60%  121M 1s\n",
      " 32350K .......... .......... .......... .......... .......... 60%  102M 1s\n",
      " 32400K .......... .......... .......... .......... .......... 60%  121M 1s\n",
      " 32450K .......... .......... .......... .......... .......... 61%  102M 1s\n",
      " 32500K .......... .......... .......... .......... .......... 61%  117M 1s\n",
      " 32550K .......... .......... .......... .......... .......... 61%  121M 1s\n",
      " 32600K .......... .......... .......... .......... .......... 61%  102M 1s\n",
      " 32650K .......... .......... .......... .......... .......... 61%  117M 1s\n",
      " 32700K .......... .......... .......... .......... .......... 61%  102M 1s\n",
      " 32750K .......... .......... .......... .......... .......... 61%  121M 1s\n",
      " 32800K .......... .......... .......... .......... .......... 61%  117M 1s\n",
      " 32850K .......... .......... .......... .......... .......... 61%  102M 1s\n",
      " 32900K .......... .......... .......... .......... .......... 61%  121M 1s\n",
      " 32950K .......... .......... .......... .......... .......... 62%  102M 1s\n",
      " 33000K .......... .......... .......... .......... .......... 62%  121M 1s\n",
      " 33050K .......... .......... .......... .......... .......... 62%  117M 1s\n",
      " 33100K .......... .......... .......... .......... .......... 62%  102M 1s\n",
      " 33150K .......... .......... .......... .......... .......... 62% 4,64M 1s\n",
      " 33200K .......... .......... .......... .......... .......... 62% 8,19M 1s\n",
      " 33250K .......... .......... .......... .......... .......... 62% 40,1M 1s\n",
      " 33300K .......... .......... .......... .......... .......... 62% 63,5M 1s\n",
      " 33350K .......... .......... .......... .......... .......... 62%  102M 1s\n",
      " 33400K .......... .......... .......... .......... .......... 62%  121M 1s\n",
      " 33450K .......... .......... .......... .......... .......... 62% 83,1M 1s\n",
      " 33500K .......... .......... .......... .......... .......... 63% 5,71M 1s\n",
      " 33550K .......... .......... .......... .......... .......... 63% 23,8M 1s\n",
      " 33600K .......... .......... .......... .......... .......... 63% 10,1M 1s\n",
      " 33650K .......... .......... .......... .......... .......... 63%  118M 1s\n",
      " 33700K .......... .......... .......... .......... .......... 63% 54,7M 1s\n",
      " 33750K .......... .......... .......... .......... .......... 63%  120M 1s\n",
      " 33800K .......... .......... .......... .......... .......... 63%  118M 1s\n",
      " 33850K .......... .......... .......... .......... .......... 63%  102M 1s\n",
      " 33900K .......... .......... .......... .......... .......... 63%  121M 1s\n",
      " 33950K .......... .......... .......... .......... .......... 63%  102M 1s\n",
      " 34000K .......... .......... .......... .......... .......... 63%  121M 1s\n",
      " 34050K .......... .......... .......... .......... .......... 64%  117M 1s\n",
      " 34100K .......... .......... .......... .......... .......... 64%  102M 1s\n",
      " 34150K .......... .......... .......... .......... .......... 64%  121M 1s\n",
      " 34200K .......... .......... .......... .......... .......... 64%  102M 1s\n",
      " 34250K .......... .......... .......... .......... .......... 64%  117M 1s\n",
      " 34300K .......... .......... .......... .......... .......... 64%  121M 1s\n",
      " 34350K .......... .......... .......... .......... .......... 64%  102M 1s\n",
      " 34400K .......... .......... .......... .......... .......... 64%  118M 1s\n",
      " 34450K .......... .......... .......... .......... .......... 64%  102M 1s\n",
      " 34500K .......... .......... .......... .......... .......... 64%  121M 1s\n",
      " 34550K .......... .......... .......... .......... .......... 65%  121M 1s\n",
      " 34600K .......... .......... .......... .......... .......... 65%  102M 1s\n",
      " 34650K .......... .......... .......... .......... .......... 65%  117M 1s\n",
      " 34700K .......... .......... .......... .......... .......... 65%  102M 1s\n",
      " 34750K .......... .......... .......... .......... .......... 65%  122M 1s\n",
      " 34800K .......... .......... .......... .......... .......... 65%  117M 1s\n",
      " 34850K .......... .......... .......... .......... .......... 65%  102M 1s\n",
      " 34900K .......... .......... .......... .......... .......... 65%  121M 1s\n",
      " 34950K .......... .......... .......... .......... .......... 65%  102M 1s\n",
      " 35000K .......... .......... .......... .......... .......... 65%  121M 1s\n",
      " 35050K .......... .......... .......... .......... .......... 65%  117M 1s\n",
      " 35100K .......... .......... .......... .......... .......... 66%  102M 1s\n",
      " 35150K .......... .......... .......... .......... .......... 66%  121M 1s\n",
      " 35200K .......... .......... .......... .......... .......... 66%  102M 1s\n",
      " 35250K .......... .......... .......... .......... .......... 66%  117M 1s\n",
      " 35300K .......... .......... .......... .......... .......... 66%  120M 1s\n",
      " 35350K .......... .......... .......... .......... .......... 66%  102M 1s\n",
      " 35400K .......... .......... .......... .......... .......... 66%  117M 1s\n",
      " 35450K .......... .......... .......... .......... .......... 66%  103M 1s\n",
      " 35500K .......... .......... .......... .......... .......... 66%  121M 1s\n",
      " 35550K .......... .......... .......... .......... .......... 66%  121M 1s\n",
      " 35600K .......... .......... .......... .......... .......... 67%  102M 1s\n",
      " 35650K .......... .......... .......... .......... .......... 67%  118M 1s\n",
      " 35700K .......... .......... .......... .......... .......... 67%  102M 1s\n",
      " 35750K .......... .......... .......... .......... .......... 67%  121M 1s\n",
      " 35800K .......... .......... .......... .......... .......... 67%  117M 1s\n",
      " 35850K .......... .......... .......... .......... .......... 67%  102M 1s\n",
      " 35900K .......... .......... .......... .......... .......... 67%  121M 1s\n",
      " 35950K .......... .......... .......... .......... .......... 67%  102M 1s\n",
      " 36000K .......... .......... .......... .......... .......... 67%  117M 1s\n",
      " 36050K .......... .......... .......... .......... .......... 67%  121M 1s\n",
      " 36100K .......... .......... .......... .......... .......... 67%  102M 1s\n",
      " 36150K .......... .......... .......... .......... .......... 68%  121M 1s\n",
      " 36200K .......... .......... .......... .......... .......... 68%  102M 1s\n",
      " 36250K .......... .......... .......... .......... .......... 68% 3,18M 1s\n",
      " 36300K .......... .......... .......... .......... .......... 68% 39,8M 1s\n",
      " 36350K .......... .......... .......... .......... .......... 68% 42,5M 1s\n",
      " 36400K .......... .......... .......... .......... .......... 68% 51,7M 1s\n",
      " 36450K .......... .......... .......... .......... .......... 68%  102M 1s\n",
      " 36500K .......... .......... .......... .......... .......... 68% 46,2M 1s\n",
      " 36550K .......... .......... .......... .......... .......... 68% 6,06M 1s\n",
      " 36600K .......... .......... .......... .......... .......... 68% 34,9M 1s\n",
      " 36650K .......... .......... .......... .......... .......... 68% 10,6M 1s\n",
      " 36700K .......... .......... .......... .......... .......... 69% 50,5M 1s\n",
      " 36750K .......... .......... .......... .......... .......... 69% 55,7M 1s\n",
      " 36800K .......... .......... .......... .......... .......... 69%  120M 1s\n",
      " 36850K .......... .......... .......... .......... .......... 69%  102M 1s\n",
      " 36900K .......... .......... .......... .......... .......... 69%  121M 1s\n",
      " 36950K .......... .......... .......... .......... .......... 69%  102M 1s\n",
      " 37000K .......... .......... .......... .......... .......... 69%  118M 1s\n",
      " 37050K .......... .......... .......... .......... .......... 69%  120M 1s\n",
      " 37100K .......... .......... .......... .......... .......... 69%  102M 1s\n",
      " 37150K .......... .......... .......... .......... .......... 69%  121M 1s\n",
      " 37200K .......... .......... .......... .......... .......... 70%  102M 1s\n",
      " 37250K .......... .......... .......... .......... .......... 70%  118M 1s\n",
      " 37300K .......... .......... .......... .......... .......... 70%  120M 1s\n",
      " 37350K .......... .......... .......... .......... .......... 70%  102M 1s\n",
      " 37400K .......... .......... .......... .......... .......... 70%  117M 1s\n",
      " 37450K .......... .......... .......... .......... .......... 70%  102M 1s\n",
      " 37500K .......... .......... .......... .......... .......... 70%  121M 1s\n",
      " 37550K .......... .......... .......... .......... .......... 70%  117M 1s\n",
      " 37600K .......... .......... .......... .......... .......... 70%  102M 1s\n",
      " 37650K .......... .......... .......... .......... .......... 70%  121M 1s\n",
      " 37700K .......... .......... .......... .......... .......... 70%  102M 1s\n",
      " 37750K .......... .......... .......... .......... .......... 71%  121M 1s\n",
      " 37800K .......... .......... .......... .......... .......... 71%  117M 0s\n",
      " 37850K .......... .......... .......... .......... .......... 71%  102M 0s\n",
      " 37900K .......... .......... .......... .......... .......... 71%  121M 0s\n",
      " 37950K .......... .......... .......... .......... .......... 71%  102M 0s\n",
      " 38000K .......... .......... .......... .......... .......... 71%  117M 0s\n",
      " 38050K .......... .......... .......... .......... .......... 71%  121M 0s\n",
      " 38100K .......... .......... .......... .......... .......... 71%  102M 0s\n",
      " 38150K .......... .......... .......... .......... .......... 71%  117M 0s\n",
      " 38200K .......... .......... .......... .......... .......... 71%  103M 0s\n",
      " 38250K .......... .......... .......... .......... .......... 71%  121M 0s\n",
      " 38300K .......... .......... .......... .......... .......... 72%  121M 0s\n",
      " 38350K .......... .......... .......... .......... .......... 72%  102M 0s\n",
      " 38400K .......... .......... .......... .......... .......... 72%  117M 0s\n",
      " 38450K .......... .......... .......... .......... .......... 72%  102M 0s\n",
      " 38500K .......... .......... .......... .......... .......... 72%  120M 0s\n",
      " 38550K .......... .......... .......... .......... .......... 72%  118M 0s\n",
      " 38600K .......... .......... .......... .......... .......... 72%  102M 0s\n",
      " 38650K .......... .......... .......... .......... .......... 72%  121M 0s\n",
      " 38700K .......... .......... .......... .......... .......... 72%  102M 0s\n",
      " 38750K .......... .......... .......... .......... .......... 72%  121M 0s\n",
      " 38800K .......... .......... .......... .......... .......... 73%  117M 0s\n",
      " 38850K .......... .......... .......... .......... .......... 73%  103M 0s\n",
      " 38900K .......... .......... .......... .......... .......... 73%  121M 0s\n",
      " 38950K .......... .......... .......... .......... .......... 73%  102M 0s\n",
      " 39000K .......... .......... .......... .......... .......... 73%  117M 0s\n",
      " 39050K .......... .......... .......... .......... .......... 73%  120M 0s\n",
      " 39100K .......... .......... .......... .......... .......... 73%  103M 0s\n",
      " 39150K .......... .......... .......... .......... .......... 73%  117M 0s\n",
      " 39200K .......... .......... .......... .......... .......... 73%  102M 0s\n",
      " 39250K .......... .......... .......... .......... .......... 73%  121M 0s\n",
      " 39300K .......... .......... .......... .......... .......... 73% 3,89M 0s\n",
      " 39350K .......... .......... .......... .......... .......... 74% 10,9M 0s\n",
      " 39400K .......... .......... .......... .......... .......... 74% 50,9M 0s\n",
      " 39450K .......... .......... .......... .......... .......... 74%  104M 0s\n",
      " 39500K .......... .......... .......... .......... .......... 74%  121M 0s\n",
      " 39550K .......... .......... .......... .......... .......... 74%  118M 0s\n",
      " 39600K .......... .......... .......... .......... .......... 74% 6,63M 0s\n",
      " 39650K .......... .......... .......... .......... .......... 74% 12,4M 0s\n",
      " 39700K .......... .......... .......... .......... .......... 74% 15,8M 0s\n",
      " 39750K .......... .......... .......... .......... .......... 74% 34,5M 0s\n",
      " 39800K .......... .......... .......... .......... .......... 74% 62,7M 0s\n",
      " 39850K .......... .......... .......... .......... .......... 74% 47,7M 0s\n",
      " 39900K .......... .......... .......... .......... .......... 75%  121M 0s\n",
      " 39950K .......... .......... .......... .......... .......... 75%  102M 0s\n",
      " 40000K .......... .......... .......... .......... .......... 75%  117M 0s\n",
      " 40050K .......... .......... .......... .......... .......... 75%  122M 0s\n",
      " 40100K .......... .......... .......... .......... .......... 75%  102M 0s\n",
      " 40150K .......... .......... .......... .......... .......... 75%  117M 0s\n",
      " 40200K .......... .......... .......... .......... .......... 75%  102M 0s\n",
      " 40250K .......... .......... .......... .......... .......... 75%  121M 0s\n",
      " 40300K .......... .......... .......... .......... .......... 75%  118M 0s\n",
      " 40350K .......... .......... .......... .......... .......... 75%  102M 0s\n",
      " 40400K .......... .......... .......... .......... .......... 76%  121M 0s\n",
      " 40450K .......... .......... .......... .......... .......... 76%  102M 0s\n",
      " 40500K .......... .......... .......... .......... .......... 76%  121M 0s\n",
      " 40550K .......... .......... .......... .......... .......... 76%  117M 0s\n",
      " 40600K .......... .......... .......... .......... .......... 76%  102M 0s\n",
      " 40650K .......... .......... .......... .......... .......... 76%  121M 0s\n",
      " 40700K .......... .......... .......... .......... .......... 76%  103M 0s\n",
      " 40750K .......... .......... .......... .......... .......... 76%  117M 0s\n",
      " 40800K .......... .......... .......... .......... .......... 76%  121M 0s\n",
      " 40850K .......... .......... .......... .......... .......... 76%  102M 0s\n",
      " 40900K .......... .......... .......... .......... .......... 76%  121M 0s\n",
      " 40950K .......... .......... .......... .......... .......... 77%  102M 0s\n",
      " 41000K .......... .......... .......... .......... .......... 77%  117M 0s\n",
      " 41050K .......... .......... .......... .......... .......... 77%  121M 0s\n",
      " 41100K .......... .......... .......... .......... .......... 77%  102M 0s\n",
      " 41150K .......... .......... .......... .......... .......... 77%  117M 0s\n",
      " 41200K .......... .......... .......... .......... .......... 77%  102M 0s\n",
      " 41250K .......... .......... .......... .......... .......... 77%  121M 0s\n",
      " 41300K .......... .......... .......... .......... .......... 77%  117M 0s\n",
      " 41350K .......... .......... .......... .......... .......... 77%  103M 0s\n",
      " 41400K .......... .......... .......... .......... .......... 77%  120M 0s\n",
      " 41450K .......... .......... .......... .......... .......... 77%  102M 0s\n",
      " 41500K .......... .......... .......... .......... .......... 78%  121M 0s\n",
      " 41550K .......... .......... .......... .......... .......... 78%  116M 0s\n",
      " 41600K .......... .......... .......... .......... .......... 78%  103M 0s\n",
      " 41650K .......... .......... .......... .......... .......... 78%  122M 0s\n",
      " 41700K .......... .......... .......... .......... .......... 78%  103M 0s\n",
      " 41750K .......... .......... .......... .......... .......... 78%  117M 0s\n",
      " 41800K .......... .......... .......... .......... .......... 78%  121M 0s\n",
      " 41850K .......... .......... .......... .......... .......... 78%  102M 0s\n",
      " 41900K .......... .......... .......... .......... .......... 78%  117M 0s\n",
      " 41950K .......... .......... .......... .......... .......... 78%  102M 0s\n",
      " 42000K .......... .......... .......... .......... .......... 79%  121M 0s\n",
      " 42050K .......... .......... .......... .......... .......... 79%  121M 0s\n",
      " 42100K .......... .......... .......... .......... .......... 79%  102M 0s\n",
      " 42150K .......... .......... .......... .......... .......... 79%  118M 0s\n",
      " 42200K .......... .......... .......... .......... .......... 79%  102M 0s\n",
      " 42250K .......... .......... .......... .......... .......... 79%  121M 0s\n",
      " 42300K .......... .......... .......... .......... .......... 79%  117M 0s\n",
      " 42350K .......... .......... .......... .......... .......... 79% 4,07M 0s\n",
      " 42400K .......... .......... .......... .......... .......... 79% 12,0M 0s\n",
      " 42450K .......... .......... .......... .......... .......... 79% 39,3M 0s\n",
      " 42500K .......... .......... .......... .......... .......... 79% 73,6M 0s\n",
      " 42550K .......... .......... .......... .......... .......... 80%  118M 0s\n",
      " 42600K .......... .......... .......... .......... .......... 80%  102M 0s\n",
      " 42650K .......... .......... .......... .......... .......... 80% 17,0M 0s\n",
      " 42700K .......... .......... .......... .......... .......... 80% 6,69M 0s\n",
      " 42750K .......... .......... .......... .......... .......... 80% 18,3M 0s\n",
      " 42800K .......... .......... .......... .......... .......... 80% 15,0M 0s\n",
      " 42850K .......... .......... .......... .......... .......... 80%  103M 0s\n",
      " 42900K .......... .......... .......... .......... .......... 80% 60,1M 0s\n",
      " 42950K .......... .......... .......... .......... .......... 80%  117M 0s\n",
      " 43000K .......... .......... .......... .......... .......... 80%  122M 0s\n",
      " 43050K .......... .......... .......... .......... .......... 81%  120M 0s\n",
      " 43100K .......... .......... .......... .......... .......... 81%  102M 0s\n",
      " 43150K .......... .......... .......... .......... .......... 81%  117M 0s\n",
      " 43200K .......... .......... .......... .......... .......... 81%  102M 0s\n",
      " 43250K .......... .......... .......... .......... .......... 81%  121M 0s\n",
      " 43300K .......... .......... .......... .......... .......... 81%  117M 0s\n",
      " 43350K .......... .......... .......... .......... .......... 81%  102M 0s\n",
      " 43400K .......... .......... .......... .......... .......... 81%  121M 0s\n",
      " 43450K .......... .......... .......... .......... .......... 81%  102M 0s\n",
      " 43500K .......... .......... .......... .......... .......... 81%  117M 0s\n",
      " 43550K .......... .......... .......... .......... .......... 81%  121M 0s\n",
      " 43600K .......... .......... .......... .......... .......... 82%  102M 0s\n",
      " 43650K .......... .......... .......... .......... .......... 82%  121M 0s\n",
      " 43700K .......... .......... .......... .......... .......... 82%  103M 0s\n",
      " 43750K .......... .......... .......... .......... .......... 82%  118M 0s\n",
      " 43800K .......... .......... .......... .......... .......... 82%  121M 0s\n",
      " 43850K .......... .......... .......... .......... .......... 82%  102M 0s\n",
      " 43900K .......... .......... .......... .......... .......... 82%  117M 0s\n",
      " 43950K .......... .......... .......... .......... .......... 82%  102M 0s\n",
      " 44000K .......... .......... .......... .......... .......... 82%  121M 0s\n",
      " 44050K .......... .......... .......... .......... .......... 82%  118M 0s\n",
      " 44100K .......... .......... .......... .......... .......... 82% 66,6M 0s\n",
      " 44150K .......... .......... .......... .......... .......... 83%  121M 0s\n",
      " 44200K .......... .......... .......... .......... .......... 83%  102M 0s\n",
      " 44250K .......... .......... .......... .......... .......... 83%  121M 0s\n",
      " 44300K .......... .......... .......... .......... .......... 83%  117M 0s\n",
      " 44350K .......... .......... .......... .......... .......... 83%  103M 0s\n",
      " 44400K .......... .......... .......... .......... .......... 83%  120M 0s\n",
      " 44450K .......... .......... .......... .......... .......... 83%  102M 0s\n",
      " 44500K .......... .......... .......... .......... .......... 83%  117M 0s\n",
      " 44550K .......... .......... .......... .......... .......... 83%  121M 0s\n",
      " 44600K .......... .......... .......... .......... .......... 83%  102M 0s\n",
      " 44650K .......... .......... .......... .......... .......... 84%  121M 0s\n",
      " 44700K .......... .......... .......... .......... .......... 84%  102M 0s\n",
      " 44750K .......... .......... .......... .......... .......... 84%  118M 0s\n",
      " 44800K .......... .......... .......... .......... .......... 84%  121M 0s\n",
      " 44850K .......... .......... .......... .......... .......... 84%  102M 0s\n",
      " 44900K .......... .......... .......... .......... .......... 84%  117M 0s\n",
      " 44950K .......... .......... .......... .......... .......... 84%  103M 0s\n",
      " 45000K .......... .......... .......... .......... .......... 84%  120M 0s\n",
      " 45050K .......... .......... .......... .......... .......... 84%  117M 0s\n",
      " 45100K .......... .......... .......... .......... .......... 84%  103M 0s\n",
      " 45150K .......... .......... .......... .......... .......... 84%  120M 0s\n",
      " 45200K .......... .......... .......... .......... .......... 85%  102M 0s\n",
      " 45250K .......... .......... .......... .......... .......... 85%  121M 0s\n",
      " 45300K .......... .......... .......... .......... .......... 85%  117M 0s\n",
      " 45350K .......... .......... .......... .......... .......... 85%  102M 0s\n",
      " 45400K .......... .......... .......... .......... .......... 85% 7,18M 0s\n",
      " 45450K .......... .......... .......... .......... .......... 85% 4,97M 0s\n",
      " 45500K .......... .......... .......... .......... .......... 85%  118M 0s\n",
      " 45550K .......... .......... .......... .......... .......... 85% 60,4M 0s\n",
      " 45600K .......... .......... .......... .......... .......... 85%  102M 0s\n",
      " 45650K .......... .......... .......... .......... .......... 85% 40,0M 0s\n",
      " 45700K .......... .......... .......... .......... .......... 85% 41,8M 0s\n",
      " 45750K .......... .......... .......... .......... .......... 86% 6,17M 0s\n",
      " 45800K .......... .......... .......... .......... .......... 86% 27,6M 0s\n",
      " 45850K .......... .......... .......... .......... .......... 86% 14,8M 0s\n",
      " 45900K .......... .......... .......... .......... .......... 86% 30,2M 0s\n",
      " 45950K .......... .......... .......... .......... .......... 86%  105M 0s\n",
      " 46000K .......... .......... .......... .......... .......... 86% 57,2M 0s\n",
      " 46050K .......... .......... .......... .......... .......... 86%  117M 0s\n",
      " 46100K .......... .......... .......... .......... .......... 86%  102M 0s\n",
      " 46150K .......... .......... .......... .......... .......... 86%  121M 0s\n",
      " 46200K .......... .......... .......... .......... .......... 86%  102M 0s\n",
      " 46250K .......... .......... .......... .......... .......... 87%  121M 0s\n",
      " 46300K .......... .......... .......... .......... .......... 87%  117M 0s\n",
      " 46350K .......... .......... .......... .......... .......... 87%  102M 0s\n",
      " 46400K .......... .......... .......... .......... .......... 87%  121M 0s\n",
      " 46450K .......... .......... .......... .......... .......... 87%  102M 0s\n",
      " 46500K .......... .......... .......... .......... .......... 87%  117M 0s\n",
      " 46550K .......... .......... .......... .......... .......... 87%  121M 0s\n",
      " 46600K .......... .......... .......... .......... .......... 87%  102M 0s\n",
      " 46650K .......... .......... .......... .......... .......... 87%  117M 0s\n",
      " 46700K .......... .......... .......... .......... .......... 87%  102M 0s\n",
      " 46750K .......... .......... .......... .......... .......... 87%  121M 0s\n",
      " 46800K .......... .......... .......... .......... .......... 88%  121M 0s\n",
      " 46850K .......... .......... .......... .......... .......... 88%  102M 0s\n",
      " 46900K .......... .......... .......... .......... .......... 88%  117M 0s\n",
      " 46950K .......... .......... .......... .......... .......... 88%  103M 0s\n",
      " 47000K .......... .......... .......... .......... .......... 88%  120M 0s\n",
      " 47050K .......... .......... .......... .......... .......... 88%  117M 0s\n",
      " 47100K .......... .......... .......... .......... .......... 88%  103M 0s\n",
      " 47150K .......... .......... .......... .......... .......... 88%  120M 0s\n",
      " 47200K .......... .......... .......... .......... .......... 88%  103M 0s\n",
      " 47250K .......... .......... .......... .......... .......... 88%  117M 0s\n",
      " 47300K .......... .......... .......... .......... .......... 88%  120M 0s\n",
      " 47350K .......... .......... .......... .......... .......... 89%  102M 0s\n",
      " 47400K .......... .......... .......... .......... .......... 89%  121M 0s\n",
      " 47450K .......... .......... .......... .......... .......... 89%  102M 0s\n",
      " 47500K .......... .......... .......... .......... .......... 89%  118M 0s\n",
      " 47550K .......... .......... .......... .......... .......... 89%  120M 0s\n",
      " 47600K .......... .......... .......... .......... .......... 89%  102M 0s\n",
      " 47650K .......... .......... .......... .......... .......... 89%  118M 0s\n",
      " 47700K .......... .......... .......... .......... .......... 89%  102M 0s\n",
      " 47750K .......... .......... .......... .......... .......... 89%  121M 0s\n",
      " 47800K .......... .......... .......... .......... .......... 89%  118M 0s\n",
      " 47850K .......... .......... .......... .......... .......... 90%  102M 0s\n",
      " 47900K .......... .......... .......... .......... .......... 90%  121M 0s\n",
      " 47950K .......... .......... .......... .......... .......... 90%  102M 0s\n",
      " 48000K .......... .......... .......... .......... .......... 90%  121M 0s\n",
      " 48050K .......... .......... .......... .......... .......... 90%  117M 0s\n",
      " 48100K .......... .......... .......... .......... .......... 90%  102M 0s\n",
      " 48150K .......... .......... .......... .......... .......... 90%  122M 0s\n",
      " 48200K .......... .......... .......... .......... .......... 90%  102M 0s\n",
      " 48250K .......... .......... .......... .......... .......... 90%  117M 0s\n",
      " 48300K .......... .......... .......... .......... .......... 90%  121M 0s\n",
      " 48350K .......... .......... .......... .......... .......... 90%  102M 0s\n",
      " 48400K .......... .......... .......... .......... .......... 91%  121M 0s\n",
      " 48450K .......... .......... .......... .......... .......... 91% 7,62M 0s\n",
      " 48500K .......... .......... .......... .......... .......... 91% 5,12M 0s\n",
      " 48550K .......... .......... .......... .......... .......... 91% 32,4M 0s\n",
      " 48600K .......... .......... .......... .......... .......... 91%  105M 0s\n",
      " 48650K .......... .......... .......... .......... .......... 91% 55,6M 0s\n",
      " 48700K .......... .......... .......... .......... .......... 91% 62,5M 0s\n",
      " 48750K .......... .......... .......... .......... .......... 91% 21,7M 0s\n",
      " 48800K .......... .......... .......... .......... .......... 91% 7,81M 0s\n",
      " 48850K .......... .......... .......... .......... .......... 91% 28,4M 0s\n",
      " 48900K .......... .......... .......... .......... .......... 91% 14,3M 0s\n",
      " 48950K .......... .......... .......... .......... .......... 92% 22,0M 0s\n",
      " 49000K .......... .......... .......... .......... .......... 92% 48,4M 0s\n",
      " 49050K .......... .......... .......... .......... .......... 92%  118M 0s\n",
      " 49100K .......... .......... .......... .......... .......... 92%  102M 0s\n",
      " 49150K .......... .......... .......... .......... .......... 92%  121M 0s\n",
      " 49200K .......... .......... .......... .......... .......... 92% 75,3M 0s\n",
      " 49250K .......... .......... .......... .......... .......... 92%  117M 0s\n",
      " 49300K .......... .......... .......... .......... .......... 92%  121M 0s\n",
      " 49350K .......... .......... .......... .......... .......... 92%  102M 0s\n",
      " 49400K .......... .......... .......... .......... .......... 92%  117M 0s\n",
      " 49450K .......... .......... .......... .......... .......... 93%  102M 0s\n",
      " 49500K .......... .......... .......... .......... .......... 93%  121M 0s\n",
      " 49550K .......... .......... .......... .......... .......... 93%  121M 0s\n",
      " 49600K .......... .......... .......... .......... .......... 93%  102M 0s\n",
      " 49650K .......... .......... .......... .......... .......... 93%  117M 0s\n",
      " 49700K .......... .......... .......... .......... .......... 93%  102M 0s\n",
      " 49750K .......... .......... .......... .......... .......... 93%  121M 0s\n",
      " 49800K .......... .......... .......... .......... .......... 93%  117M 0s\n",
      " 49850K .......... .......... .......... .......... .......... 93%  102M 0s\n",
      " 49900K .......... .......... .......... .......... .......... 93%  121M 0s\n",
      " 49950K .......... .......... .......... .......... .......... 93%  102M 0s\n",
      " 50000K .......... .......... .......... .......... .......... 94%  121M 0s\n",
      " 50050K .......... .......... .......... .......... .......... 94%  117M 0s\n",
      " 50100K .......... .......... .......... .......... .......... 94%  103M 0s\n",
      " 50150K .......... .......... .......... .......... .......... 94%  121M 0s\n",
      " 50200K .......... .......... .......... .......... .......... 94%  102M 0s\n",
      " 50250K .......... .......... .......... .......... .......... 94%  117M 0s\n",
      " 50300K .......... .......... .......... .......... .......... 94%  121M 0s\n",
      " 50350K .......... .......... .......... .......... .......... 94%  102M 0s\n",
      " 50400K .......... .......... .......... .......... .......... 94%  117M 0s\n",
      " 50450K .......... .......... .......... .......... .......... 94%  103M 0s\n",
      " 50500K .......... .......... .......... .......... .......... 95%  121M 0s\n",
      " 50550K .......... .......... .......... .......... .......... 95%  121M 0s\n",
      " 50600K .......... .......... .......... .......... .......... 95%  102M 0s\n",
      " 50650K .......... .......... .......... .......... .......... 95%  118M 0s\n",
      " 50700K .......... .......... .......... .......... .......... 95%  102M 0s\n",
      " 50750K .......... .......... .......... .......... .......... 95%  121M 0s\n",
      " 50800K .......... .......... .......... .......... .......... 95%  117M 0s\n",
      " 50850K .......... .......... .......... .......... .......... 95%  102M 0s\n",
      " 50900K .......... .......... .......... .......... .......... 95%  121M 0s\n",
      " 50950K .......... .......... .......... .......... .......... 95%  102M 0s\n",
      " 51000K .......... .......... .......... .......... .......... 95%  117M 0s\n",
      " 51050K .......... .......... .......... .......... .......... 96%  121M 0s\n",
      " 51100K .......... .......... .......... .......... .......... 96%  102M 0s\n",
      " 51150K .......... .......... .......... .......... .......... 96%  121M 0s\n",
      " 51200K .......... .......... .......... .......... .......... 96%  102M 0s\n",
      " 51250K .......... .......... .......... .......... .......... 96%  117M 0s\n",
      " 51300K .......... .......... .......... .......... .......... 96%  121M 0s\n",
      " 51350K .......... .......... .......... .......... .......... 96%  102M 0s\n",
      " 51400K .......... .......... .......... .......... .......... 96%  117M 0s\n",
      " 51450K .......... .......... .......... .......... .......... 96% 11,6M 0s\n",
      " 51500K .......... .......... .......... .......... .......... 96% 41,9M 0s\n",
      " 51550K .......... .......... .......... .......... .......... 96% 6,16M 0s\n",
      " 51600K .......... .......... .......... .......... .......... 97% 11,6M 0s\n",
      " 51650K .......... .......... .......... .......... .......... 97% 62,7M 0s\n",
      " 51700K .......... .......... .......... .......... .......... 97% 57,2M 0s\n",
      " 51750K .......... .......... .......... .......... .......... 97%  127M 0s\n",
      " 51800K .......... .......... .......... .......... .......... 97% 28,5M 0s\n",
      " 51850K .......... .......... .......... .......... .......... 97% 8,31M 0s\n",
      " 51900K .......... .......... .......... .......... .......... 97% 12,1M 0s\n",
      " 51950K .......... .......... .......... .......... .......... 97% 28,4M 0s\n",
      " 52000K .......... .......... .......... .......... .......... 97% 14,7M 0s\n",
      " 52050K .......... .......... .......... .......... .......... 97%  120M 0s\n",
      " 52100K .......... .......... .......... .......... .......... 98% 68,3M 0s\n",
      " 52150K .......... .......... .......... .......... .......... 98%  121M 0s\n",
      " 52200K .......... .......... .......... .......... .......... 98%  102M 0s\n",
      " 52250K .......... .......... .......... .......... .......... 98%  118M 0s\n",
      " 52300K .......... .......... .......... .......... .......... 98%  120M 0s\n",
      " 52350K .......... .......... .......... .......... .......... 98%  102M 0s\n",
      " 52400K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 52450K .......... .......... .......... .......... .......... 98%  102M 0s\n",
      " 52500K .......... .......... .......... .......... .......... 98%  121M 0s\n",
      " 52550K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 52600K .......... .......... .......... .......... .......... 98%  102M 0s\n",
      " 52650K .......... .......... .......... .......... .......... 99%  121M 0s\n",
      " 52700K .......... .......... .......... .......... .......... 99%  102M 0s\n",
      " 52750K .......... .......... .......... .......... .......... 99%  121M 0s\n",
      " 52800K .......... .......... .......... .......... .......... 99%  117M 0s\n",
      " 52850K .......... .......... .......... .......... .......... 99%  102M 0s\n",
      " 52900K .......... .......... .......... .......... .......... 99%  121M 0s\n",
      " 52950K .......... .......... .......... .......... .......... 99%  102M 0s\n",
      " 53000K .......... .......... .......... .......... .......... 99%  117M 0s\n",
      " 53050K .......... .......... .......... .......... .......... 99%  121M 0s\n",
      " 53100K .......... .......... .......... .......... .......... 99%  102M 0s\n",
      " 53150K .......... .......... .......... .......... .......... 99%  117M 0s\n",
      " 53200K ........                                              100%  236M=1,5s\n",
      "\n",
      "2023-08-28 20:40:42 (34,2 MB/s) - ‘data/reference/genome.fa.gz’ guardado [54485176/54485176]\n",
      "\n",
      "[Mon Aug 28 20:40:43 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T204039.652547.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:40:44 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR696683_chr5_1.fastq.gz, data/raw/ERR696683_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR696683_chr5_1_fastp.fastq.gz, data/processed/ERR696683_chr5_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR696683_chr5_1_fastp.fastq.gz, data/processed/ERR696683_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1745204\n",
      "total bases: 131934728\n",
      "Q20 bases: 131000292(99.2917%)\n",
      "Q30 bases: 127389751(96.5551%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1745204\n",
      "total bases: 131603364\n",
      "Q20 bases: 130011063(98.7901%)\n",
      "Q30 bases: 124792206(94.8245%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1090289\n",
      "total bases: 82806787\n",
      "Q20 bases: 82482544(99.6084%)\n",
      "Q30 bases: 81050211(97.8787%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1090289\n",
      "total bases: 82775596\n",
      "Q20 bases: 82215241(99.323%)\n",
      "Q30 bases: 80124032(96.7967%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2180578\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1309830\n",
      "reads with adapter trimmed: 1530\n",
      "bases trimmed due to adapters: 39157\n",
      "\n",
      "Duplication rate: 5.8833%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 119\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR696683_chr5_1.fastq.gz -I data/raw/ERR696683_chr5_2.fastq.gz -o data/processed/ERR696683_chr5_1_fastp.fastq.gz -O data/processed/ERR696683_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Mon Aug 28 20:40:51 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:40:51 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR696683_chr5_1_fastp.fastq.gz, data/processed/ERR696683_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR696683_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR696683_chr5_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR696683_chr5_1_fastp.fastq.gz, data/processed/ERR696683_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 20:41:01 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:41:01 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR696683_chr5_1.fastq.gz, data/raw/ERR696683_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR696683_chr5_1_fastqc.html, results/fastqc_result/ERR696683_chr5_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR696683_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 20:41:15 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T204043.397579.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:41:16 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753368_chr5_1.fastq.gz, data/raw/ERR753368_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753368_chr5_1_fastp.fastq.gz, data/processed/ERR753368_chr5_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753368_chr5_1_fastp.fastq.gz, data/processed/ERR753368_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1754317\n",
      "total bases: 133207348\n",
      "Q20 bases: 132685869(99.6085%)\n",
      "Q30 bases: 128783661(96.6791%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1754317\n",
      "total bases: 133082490\n",
      "Q20 bases: 131930412(99.1343%)\n",
      "Q30 bases: 125959500(94.6477%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1515191\n",
      "total bases: 115148663\n",
      "Q20 bases: 114951118(99.8284%)\n",
      "Q30 bases: 113024140(98.155%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1515191\n",
      "total bases: 115139136\n",
      "Q20 bases: 114585523(99.5192%)\n",
      "Q30 bases: 111209688(96.5872%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3030382\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 478252\n",
      "reads with adapter trimmed: 2004\n",
      "bases trimmed due to adapters: 46944\n",
      "\n",
      "Duplication rate: 2.06192%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753368_chr5_1.fastq.gz -I data/raw/ERR753368_chr5_2.fastq.gz -o data/processed/ERR753368_chr5_1_fastp.fastq.gz -O data/processed/ERR753368_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Mon Aug 28 20:41:22 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:41:22 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753368_chr5_1_fastp.fastq.gz, data/processed/ERR753368_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753368_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753368_chr5_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753368_chr5_1_fastp.fastq.gz, data/processed/ERR753368_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 20:41:35 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:41:35 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753368_chr5_1.fastq.gz, data/raw/ERR753368_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753368_chr5_1_fastqc.html, results/fastqc_result/ERR753368_chr5_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753368_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753368_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 20:41:50 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T204115.784632.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:41:51 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753369_chr5_1.fastq.gz, data/raw/ERR753369_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753369_chr5_1_fastqc.html, results/fastqc_result/ERR753369_chr5_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753369_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753369_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 20:42:06 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:42:06 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753369_chr5_1.fastq.gz, data/raw/ERR753369_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753369_chr5_1_fastp.fastq.gz, data/processed/ERR753369_chr5_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753369_chr5_1_fastp.fastq.gz, data/processed/ERR753369_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1874038\n",
      "total bases: 142293627\n",
      "Q20 bases: 141712295(99.5915%)\n",
      "Q30 bases: 137456286(96.6005%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1874038\n",
      "total bases: 142166679\n",
      "Q20 bases: 140932525(99.1319%)\n",
      "Q30 bases: 134533978(94.6312%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1618376\n",
      "total bases: 122990005\n",
      "Q20 bases: 122765144(99.8172%)\n",
      "Q30 bases: 120638796(98.0883%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1618376\n",
      "total bases: 122980612\n",
      "Q20 bases: 122387247(99.5175%)\n",
      "Q30 bases: 118766072(96.573%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3236752\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 511324\n",
      "reads with adapter trimmed: 2190\n",
      "bases trimmed due to adapters: 51994\n",
      "\n",
      "Duplication rate: 2.40989%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753369_chr5_1.fastq.gz -I data/raw/ERR753369_chr5_2.fastq.gz -o data/processed/ERR753369_chr5_1_fastp.fastq.gz -O data/processed/ERR753369_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Mon Aug 28 20:42:13 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:42:13 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753369_chr5_1_fastp.fastq.gz, data/processed/ERR753369_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753369_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753369_chr5_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753369_chr5_1_fastp.fastq.gz, data/processed/ERR753369_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 20:42:25 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T204150.636042.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:42:27 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753370_chr5_1.fastq.gz, data/raw/ERR753370_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753370_chr5_1_fastp.fastq.gz, data/processed/ERR753370_chr5_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753370_chr5_2_fastp.fastq.gz, data/processed/ERR753370_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2518071\n",
      "total bases: 253490833\n",
      "Q20 bases: 251767367(99.3201%)\n",
      "Q30 bases: 244487028(96.4481%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2518071\n",
      "total bases: 253433925\n",
      "Q20 bases: 251051683(99.06%)\n",
      "Q30 bases: 241620433(95.3386%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 2517536\n",
      "total bases: 252452388\n",
      "Q20 bases: 250828184(99.3566%)\n",
      "Q30 bases: 243890545(96.6085%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 2517536\n",
      "total bases: 251656989\n",
      "Q20 bases: 249490937(99.1393%)\n",
      "Q30 bases: 240647742(95.6253%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 5035072\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1070\n",
      "reads with adapter trimmed: 14130\n",
      "bases trimmed due to adapters: 108801\n",
      "\n",
      "Duplication rate: 4.44843%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 159\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753370_chr5_1.fastq.gz -I data/raw/ERR753370_chr5_2.fastq.gz -o data/processed/ERR753370_chr5_1_fastp.fastq.gz -O data/processed/ERR753370_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 11 seconds\n",
      "[Mon Aug 28 20:42:39 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:42:39 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753370_chr5_1.fastq.gz, data/raw/ERR753370_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753370_chr5_1_fastqc.html, results/fastqc_result/ERR753370_chr5_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753370_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753370_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 20:43:01 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:43:01 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753370_chr5_1_fastp.fastq.gz, data/processed/ERR753370_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753370_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753370_chr5_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753370_chr5_2_fastp.fastq.gz, data/processed/ERR753370_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 20:43:24 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T204226.036739.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:43:25 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753371_chr5_1.fastq.gz, data/raw/ERR753371_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753371_chr5_1_fastp.fastq.gz, data/processed/ERR753371_chr5_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753371_chr5_2_fastp.fastq.gz, data/processed/ERR753371_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1647835\n",
      "total bases: 124169361\n",
      "Q20 bases: 123052427(99.1005%)\n",
      "Q30 bases: 118563904(95.4856%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1647835\n",
      "total bases: 123864691\n",
      "Q20 bases: 122402253(98.8193%)\n",
      "Q30 bases: 118123304(95.3648%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 856072\n",
      "total bases: 64982245\n",
      "Q20 bases: 64667623(99.5158%)\n",
      "Q30 bases: 63059172(97.0406%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 856072\n",
      "total bases: 64947026\n",
      "Q20 bases: 64500274(99.3121%)\n",
      "Q30 bases: 63094570(97.1477%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1712144\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1583526\n",
      "reads with adapter trimmed: 1440\n",
      "bases trimmed due to adapters: 32385\n",
      "\n",
      "Duplication rate: 22.2932%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 117\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753371_chr5_1.fastq.gz -I data/raw/ERR753371_chr5_2.fastq.gz -o data/processed/ERR753371_chr5_1_fastp.fastq.gz -O data/processed/ERR753371_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Mon Aug 28 20:43:31 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:43:31 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753371_chr5_1.fastq.gz, data/raw/ERR753371_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753371_chr5_1_fastqc.html, results/fastqc_result/ERR753371_chr5_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753371_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753371_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 20:43:44 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:43:44 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753371_chr5_1_fastp.fastq.gz, data/processed/ERR753371_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753371_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753371_chr5_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753371_chr5_2_fastp.fastq.gz, data/processed/ERR753371_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 20:43:52 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T204324.717282.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:43:54 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753372_chr5_1.fastq.gz, data/raw/ERR753372_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753372_chr5_1_fastqc.html, results/fastqc_result/ERR753372_chr5_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753372_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753372_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 20:44:09 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:44:09 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753372_chr5_1.fastq.gz, data/raw/ERR753372_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753372_chr5_1_fastp.fastq.gz, data/processed/ERR753372_chr5_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753372_chr5_1_fastp.fastq.gz, data/processed/ERR753372_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1849920\n",
      "total bases: 140463336\n",
      "Q20 bases: 139906933(99.6039%)\n",
      "Q30 bases: 135706572(96.6135%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1849920\n",
      "total bases: 140332630\n",
      "Q20 bases: 139116664(99.1335%)\n",
      "Q30 bases: 132851025(94.6687%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1592983\n",
      "total bases: 121060371\n",
      "Q20 bases: 120850328(99.8265%)\n",
      "Q30 bases: 118781603(98.1177%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1592983\n",
      "total bases: 121050636\n",
      "Q20 bases: 120473426(99.5232%)\n",
      "Q30 bases: 116971001(96.6298%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3185966\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 513874\n",
      "reads with adapter trimmed: 2008\n",
      "bases trimmed due to adapters: 48830\n",
      "\n",
      "Duplication rate: 2.02262%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753372_chr5_1.fastq.gz -I data/raw/ERR753372_chr5_2.fastq.gz -o data/processed/ERR753372_chr5_1_fastp.fastq.gz -O data/processed/ERR753372_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Mon Aug 28 20:44:15 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:44:15 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753372_chr5_1_fastp.fastq.gz, data/processed/ERR753372_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753372_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753372_chr5_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753372_chr5_1_fastp.fastq.gz, data/processed/ERR753372_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 20:44:28 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T204353.190550.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:44:29 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753373_chr5_1.fastq.gz, data/raw/ERR753373_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753373_chr5_1_fastp.fastq.gz, data/processed/ERR753373_chr5_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753373_chr5_1_fastp.fastq.gz, data/processed/ERR753373_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1769351\n",
      "total bases: 169422654\n",
      "Q20 bases: 168087237(99.2118%)\n",
      "Q30 bases: 156744268(92.5167%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1769351\n",
      "total bases: 165883501\n",
      "Q20 bases: 164108844(98.9302%)\n",
      "Q30 bases: 153648766(92.6245%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1727865\n",
      "total bases: 162896728\n",
      "Q20 bases: 161958252(99.4239%)\n",
      "Q30 bases: 152772191(93.7847%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1727865\n",
      "total bases: 157462847\n",
      "Q20 bases: 156729547(99.5343%)\n",
      "Q30 bases: 148765966(94.4769%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3455730\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 82972\n",
      "reads with adapter trimmed: 1628\n",
      "bases trimmed due to adapters: 56407\n",
      "\n",
      "Duplication rate: 1.9251%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 154\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753373_chr5_1.fastq.gz -I data/raw/ERR753373_chr5_2.fastq.gz -o data/processed/ERR753373_chr5_1_fastp.fastq.gz -O data/processed/ERR753373_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 7 seconds\n",
      "[Mon Aug 28 20:44:37 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:44:37 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753373_chr5_1_fastp.fastq.gz, data/processed/ERR753373_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753373_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753373_chr5_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753373_chr5_1_fastp.fastq.gz, data/processed/ERR753373_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 20:44:52 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:44:52 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753373_chr5_1.fastq.gz, data/raw/ERR753373_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753373_chr5_1_fastqc.html, results/fastqc_result/ERR753373_chr5_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753373_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753373_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 20:45:09 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T204428.541446.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:45:10 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753374_chr5_1.fastq.gz, data/raw/ERR753374_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753374_chr5_1_fastqc.html, results/fastqc_result/ERR753374_chr5_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753374_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753374_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 20:45:21 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:45:21 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753374_chr5_1.fastq.gz, data/raw/ERR753374_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753374_chr5_1_fastp.fastq.gz, data/processed/ERR753374_chr5_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753374_chr5_2_fastp.fastq.gz, data/processed/ERR753374_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1042969\n",
      "total bases: 99775475\n",
      "Q20 bases: 98851672(99.0741%)\n",
      "Q30 bases: 90441720(90.6452%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1042969\n",
      "total bases: 97644504\n",
      "Q20 bases: 96637952(98.9692%)\n",
      "Q30 bases: 89006849(91.154%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1011156\n",
      "total bases: 94571637\n",
      "Q20 bases: 93909953(99.3003%)\n",
      "Q30 bases: 87206490(92.2121%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1011156\n",
      "total bases: 92115109\n",
      "Q20 bases: 91596849(99.4374%)\n",
      "Q30 bases: 85744137(93.0837%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2022312\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 63626\n",
      "reads with adapter trimmed: 920\n",
      "bases trimmed due to adapters: 32756\n",
      "\n",
      "Duplication rate: 5.80197%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 148\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753374_chr5_1.fastq.gz -I data/raw/ERR753374_chr5_2.fastq.gz -o data/processed/ERR753374_chr5_1_fastp.fastq.gz -O data/processed/ERR753374_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Mon Aug 28 20:45:26 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:45:26 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753374_chr5_1_fastp.fastq.gz, data/processed/ERR753374_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753374_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753374_chr5_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753374_chr5_2_fastp.fastq.gz, data/processed/ERR753374_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 20:45:37 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T204509.440211.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:45:38 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753375_chr5_1.fastq.gz, data/raw/ERR753375_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753375_chr5_1_fastp.fastq.gz, data/processed/ERR753375_chr5_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753375_chr5_1_fastp.fastq.gz, data/processed/ERR753375_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1084744\n",
      "total bases: 81943993\n",
      "Q20 bases: 81438535(99.3832%)\n",
      "Q30 bases: 79595776(97.1344%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1084744\n",
      "total bases: 81928209\n",
      "Q20 bases: 80877145(98.7171%)\n",
      "Q30 bases: 78915671(96.323%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 711112\n",
      "total bases: 53994712\n",
      "Q20 bases: 53820596(99.6775%)\n",
      "Q30 bases: 53016677(98.1886%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 711112\n",
      "total bases: 54003767\n",
      "Q20 bases: 53488355(99.0456%)\n",
      "Q30 bases: 52659532(97.5108%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1422224\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 747264\n",
      "reads with adapter trimmed: 856\n",
      "bases trimmed due to adapters: 20807\n",
      "\n",
      "Duplication rate: 1.69478%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 119\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753375_chr5_1.fastq.gz -I data/raw/ERR753375_chr5_2.fastq.gz -o data/processed/ERR753375_chr5_1_fastp.fastq.gz -O data/processed/ERR753375_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Mon Aug 28 20:45:42 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:45:42 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753375_chr5_1_fastp.fastq.gz, data/processed/ERR753375_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753375_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753375_chr5_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753375_chr5_1_fastp.fastq.gz, data/processed/ERR753375_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 20:45:49 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:45:49 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753375_chr5_1.fastq.gz, data/raw/ERR753375_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753375_chr5_1_fastqc.html, results/fastqc_result/ERR753375_chr5_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753375_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753375_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 20:45:59 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T204537.190867.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:46:01 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753376_chr5_1.fastq.gz, data/raw/ERR753376_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753376_chr5_1_fastqc.html, results/fastqc_result/ERR753376_chr5_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753376_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753376_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 20:46:13 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:46:13 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753376_chr5_1.fastq.gz, data/raw/ERR753376_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753376_chr5_1_fastp.fastq.gz, data/processed/ERR753376_chr5_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753376_chr5_2_fastp.fastq.gz, data/processed/ERR753376_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1402798\n",
      "total bases: 106050477\n",
      "Q20 bases: 105320649(99.3118%)\n",
      "Q30 bases: 102421962(96.5785%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1402798\n",
      "total bases: 105773011\n",
      "Q20 bases: 104493043(98.7899%)\n",
      "Q30 bases: 100281618(94.8083%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 875003\n",
      "total bases: 66456395\n",
      "Q20 bases: 66207960(99.6262%)\n",
      "Q30 bases: 65072453(97.9175%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 875003\n",
      "total bases: 66430859\n",
      "Q20 bases: 65983780(99.327%)\n",
      "Q30 bases: 64300470(96.7931%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1750006\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1055590\n",
      "reads with adapter trimmed: 1316\n",
      "bases trimmed due to adapters: 32140\n",
      "\n",
      "Duplication rate: 9.15784%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753376_chr5_1.fastq.gz -I data/raw/ERR753376_chr5_2.fastq.gz -o data/processed/ERR753376_chr5_1_fastp.fastq.gz -O data/processed/ERR753376_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Mon Aug 28 20:46:18 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:46:18 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753376_chr5_1_fastp.fastq.gz, data/processed/ERR753376_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753376_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753376_chr5_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753376_chr5_2_fastp.fastq.gz, data/processed/ERR753376_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 100% complete for ERR753376_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 100% complete for ERR753376_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 20:46:27 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T204600.161464.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:46:28 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753377_chr5_1.fastq.gz, data/raw/ERR753377_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753377_chr5_1_fastp.fastq.gz, data/processed/ERR753377_chr5_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753377_chr5_1_fastp.fastq.gz, data/processed/ERR753377_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1456925\n",
      "total bases: 110159308\n",
      "Q20 bases: 109389376(99.3011%)\n",
      "Q30 bases: 106378169(96.5676%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1456925\n",
      "total bases: 109900000\n",
      "Q20 bases: 108632355(98.8465%)\n",
      "Q30 bases: 104449264(95.0403%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 929607\n",
      "total bases: 70603984\n",
      "Q20 bases: 70333119(99.6164%)\n",
      "Q30 bases: 69114056(97.8897%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 929607\n",
      "total bases: 70578254\n",
      "Q20 bases: 70125662(99.3587%)\n",
      "Q30 bases: 68418786(96.9403%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1859214\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1054636\n",
      "reads with adapter trimmed: 1262\n",
      "bases trimmed due to adapters: 30893\n",
      "\n",
      "Duplication rate: 9.20804%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753377_chr5_1.fastq.gz -I data/raw/ERR753377_chr5_2.fastq.gz -o data/processed/ERR753377_chr5_1_fastp.fastq.gz -O data/processed/ERR753377_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Mon Aug 28 20:46:33 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:46:33 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753377_chr5_1_fastp.fastq.gz, data/processed/ERR753377_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753377_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753377_chr5_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753377_chr5_1_fastp.fastq.gz, data/processed/ERR753377_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 20:46:42 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:46:42 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753377_chr5_1.fastq.gz, data/raw/ERR753377_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753377_chr5_1_fastqc.html, results/fastqc_result/ERR753377_chr5_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753377_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753377_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 20:46:55 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T204627.877489.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:46:56 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753378_chr5_1.fastq.gz, data/raw/ERR753378_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753378_chr5_1_fastqc.html, results/fastqc_result/ERR753378_chr5_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753378_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753378_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 20:47:09 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:47:09 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753378_chr5_1.fastq.gz, data/raw/ERR753378_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753378_chr5_1_fastp.fastq.gz, data/processed/ERR753378_chr5_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753378_chr5_1_fastp.fastq.gz, data/processed/ERR753378_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1582656\n",
      "total bases: 120172394\n",
      "Q20 bases: 119702143(99.6087%)\n",
      "Q30 bases: 116149817(96.6527%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1582656\n",
      "total bases: 120057743\n",
      "Q20 bases: 119018888(99.1347%)\n",
      "Q30 bases: 113653667(94.6658%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1368395\n",
      "total bases: 103992525\n",
      "Q20 bases: 103812087(99.8265%)\n",
      "Q30 bases: 102035807(98.1184%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1368395\n",
      "total bases: 103984175\n",
      "Q20 bases: 103483359(99.5184%)\n",
      "Q30 bases: 100427300(96.5794%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2736790\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 428522\n",
      "reads with adapter trimmed: 1908\n",
      "bases trimmed due to adapters: 45702\n",
      "\n",
      "Duplication rate: 2.03321%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753378_chr5_1.fastq.gz -I data/raw/ERR753378_chr5_2.fastq.gz -o data/processed/ERR753378_chr5_1_fastp.fastq.gz -O data/processed/ERR753378_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Mon Aug 28 20:47:14 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:47:14 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753378_chr5_1_fastp.fastq.gz, data/processed/ERR753378_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753378_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753378_chr5_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753378_chr5_1_fastp.fastq.gz, data/processed/ERR753378_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 20:47:25 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T204655.543070.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:47:26 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR696683_chr5_1_fastp.fastq.gz, data/processed/ERR696683_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR696683_chr5.sam, results/mapped_reads/ERR696683_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR696683_chr5_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR696683_chr5_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.58 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 69.64 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.54 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.38 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 39.31 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 111.177 sec; CPU: 110.447 sec\n",
      "[Mon Aug 28 20:50:43 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:50:43 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR696683_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr5.bam, results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR696683_chr5.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR696683_chr5_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 20:51:22 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:51:22 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR696683_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "20:51:23.597 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 20:51:23 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR696683_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR696683_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 20:51:23 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 20:51:23\tMarkDuplicates\tStart of doWork freeMemory: 528821256; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 20:51:23\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 20:51:23\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam\n",
      "INFO\t2023-08-28 20:51:25\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:81,252,672\n",
      "INFO\t2023-08-28 20:51:25\tMarkDuplicates\tTracking 736 as yet unmatched pairs. 736 records in RAM.\n",
      "INFO\t2023-08-28 20:51:26\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 5:168,199,832\n",
      "INFO\t2023-08-28 20:51:26\tMarkDuplicates\tTracking 289 as yet unmatched pairs. 289 records in RAM.\n",
      "INFO\t2023-08-28 20:51:26\tMarkDuplicates\tRead 2201600 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 20:51:26\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 699233360; totalMemory: 1006632960; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 20:51:26\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 20:51:27\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 20:51:27\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 20:51:27\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 20:51:27\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 997181616; totalMemory: 1543503872; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 20:51:27\tMarkDuplicates\tMarking 201846 records as duplicates.\n",
      "INFO\t2023-08-28 20:51:27\tMarkDuplicates\tFound 99 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam\n",
      "INFO\t2023-08-28 20:51:27\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 20:51:41\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 20:51:41\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 20:51:41\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 20:51:41\tMarkDuplicates\tBefore output close freeMemory: 528119224; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 20:51:41\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 20:51:41\tMarkDuplicates\tAfter output close freeMemory: 528119224; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 20:51:41 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.30 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 20:51:43 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T204725.816565.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:51:44 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753368_chr5_1_fastp.fastq.gz, data/processed/ERR753368_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753368_chr5.sam, results/mapped_reads/ERR753368_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753368_chr5_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753368_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.62 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 68.50 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.55 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.37 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 39.07 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 109.955 sec; CPU: 109.109 sec\n",
      "[Mon Aug 28 20:55:47 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:55:47 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753368_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr5.bam, results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753368_chr5.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753368_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 20:56:34 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:56:34 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753368_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753368_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "20:56:35.210 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 20:56:35 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753368_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753368_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 20:56:35 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 20:56:35\tMarkDuplicates\tStart of doWork freeMemory: 528781720; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 20:56:35\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 20:56:35\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam\n",
      "INFO\t2023-08-28 20:56:36\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:72,252,503\n",
      "INFO\t2023-08-28 20:56:36\tMarkDuplicates\tTracking 479 as yet unmatched pairs. 479 records in RAM.\n",
      "INFO\t2023-08-28 20:56:38\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 5:138,184,159\n",
      "INFO\t2023-08-28 20:56:38\tMarkDuplicates\tTracking 440 as yet unmatched pairs. 440 records in RAM.\n",
      "INFO\t2023-08-28 20:56:39\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 5:179,891,658\n",
      "INFO\t2023-08-28 20:56:39\tMarkDuplicates\tTracking 24 as yet unmatched pairs. 24 records in RAM.\n",
      "INFO\t2023-08-28 20:56:39\tMarkDuplicates\tRead 3048194 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 20:56:39\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 475238872; totalMemory: 874512384; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 20:56:39\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 20:56:39\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 20:56:40\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 20:56:40\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 20:56:40\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 865022128; totalMemory: 1411383296; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 20:56:40\tMarkDuplicates\tMarking 74578 records as duplicates.\n",
      "INFO\t2023-08-28 20:56:40\tMarkDuplicates\tFound 483 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam\n",
      "INFO\t2023-08-28 20:56:40\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 20:56:58\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 20:56:58\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 20:56:58\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 20:56:58\tMarkDuplicates\tBefore output close freeMemory: 528080088; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 20:56:58\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 20:56:58\tMarkDuplicates\tAfter output close freeMemory: 528080088; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 20:56:58 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.39 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 20:57:00 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T205143.292307.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 20:57:01 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753369_chr5_1_fastp.fastq.gz, data/processed/ERR753369_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753369_chr5.sam, results/mapped_reads/ERR753369_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753369_chr5_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR753369_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.62 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 66.12 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.55 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.37 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 39.68 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 113.553 sec; CPU: 107.338 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 21:01:29 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:01:29 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753369_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr5.bam, results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753369_chr5.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753369_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 21:02:27 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:02:27 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753369_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753369_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "21:02:28.903 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 21:02:28 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753369_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753369_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 21:02:28 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 21:02:28\tMarkDuplicates\tStart of doWork freeMemory: 528785776; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:02:28\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 21:02:28\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam\n",
      "INFO\t2023-08-28 21:02:30\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:69,509,472\n",
      "INFO\t2023-08-28 21:02:30\tMarkDuplicates\tTracking 483 as yet unmatched pairs. 483 records in RAM.\n",
      "INFO\t2023-08-28 21:02:32\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 5:132,813,658\n",
      "INFO\t2023-08-28 21:02:32\tMarkDuplicates\tTracking 399 as yet unmatched pairs. 399 records in RAM.\n",
      "INFO\t2023-08-28 21:02:33\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 5:173,999,730\n",
      "INFO\t2023-08-28 21:02:33\tMarkDuplicates\tTracking 121 as yet unmatched pairs. 121 records in RAM.\n",
      "INFO\t2023-08-28 21:02:33\tMarkDuplicates\tRead 3250648 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 21:02:33\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 421328312; totalMemory: 843055104; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:02:33\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 21:02:33\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 21:02:34\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 21:02:34\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 21:02:34\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 833568480; totalMemory: 1379926016; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:02:34\tMarkDuplicates\tMarking 92898 records as duplicates.\n",
      "INFO\t2023-08-28 21:02:34\tMarkDuplicates\tFound 490 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam\n",
      "INFO\t2023-08-28 21:02:34\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 21:02:53\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 21:02:53\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 21:02:53\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 21:02:53\tMarkDuplicates\tBefore output close freeMemory: 528083784; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:02:53\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 21:02:53\tMarkDuplicates\tAfter output close freeMemory: 528083784; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 21:02:53 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.42 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 21:02:56 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T205700.812030.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:02:59 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753370_chr5_1_fastp.fastq.gz, data/processed/ERR753370_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753370_chr5.sam, results/mapped_reads/ERR753370_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753370_chr5_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753370_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.59 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 71.00 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.54 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.37 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 40.12 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 118.526 sec; CPU: 112.612 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 21:09:48 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:09:48 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753370_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr5.bam, results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753370_chr5.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753370_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 21:11:46 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:11:46 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753370_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753370_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "21:11:47.379 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 21:11:47 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753370_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753370_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 21:11:47 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 21:11:47\tMarkDuplicates\tStart of doWork freeMemory: 528830928; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:11:47\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 21:11:47\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam\n",
      "INFO\t2023-08-28 21:11:49\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:42,800,651\n",
      "INFO\t2023-08-28 21:11:49\tMarkDuplicates\tTracking 1050 as yet unmatched pairs. 1050 records in RAM.\n",
      "INFO\t2023-08-28 21:11:50\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 5:75,358,775\n",
      "INFO\t2023-08-28 21:11:50\tMarkDuplicates\tTracking 1655 as yet unmatched pairs. 1655 records in RAM.\n",
      "INFO\t2023-08-28 21:11:52\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 5:115,580,661\n",
      "INFO\t2023-08-28 21:11:52\tMarkDuplicates\tTracking 1493 as yet unmatched pairs. 1493 records in RAM.\n",
      "INFO\t2023-08-28 21:11:53\tMarkDuplicates\tRead     4,000,000 records.  Elapsed time: 00:00:06s.  Time for last 1,000,000:    1s.  Last read position: 5:143,041,762\n",
      "INFO\t2023-08-28 21:11:53\tMarkDuplicates\tTracking 959 as yet unmatched pairs. 959 records in RAM.\n",
      "INFO\t2023-08-28 21:11:55\tMarkDuplicates\tRead     5,000,000 records.  Elapsed time: 00:00:08s.  Time for last 1,000,000:    1s.  Last read position: 5:179,585,760\n",
      "INFO\t2023-08-28 21:11:55\tMarkDuplicates\tTracking 247 as yet unmatched pairs. 247 records in RAM.\n",
      "INFO\t2023-08-28 21:11:55\tMarkDuplicates\tRead 5095159 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 21:11:55\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 424178032; totalMemory: 1040187392; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:11:55\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 21:11:55\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 21:11:56\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 21:11:56\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 21:11:56\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1030746344; totalMemory: 1577058304; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:11:56\tMarkDuplicates\tMarking 357835 records as duplicates.\n",
      "INFO\t2023-08-28 21:11:56\tMarkDuplicates\tFound 194 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam\n",
      "INFO\t2023-08-28 21:11:56\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 21:12:35\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 21:12:35\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 21:12:35\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 21:12:35\tMarkDuplicates\tBefore output close freeMemory: 528129552; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:12:35\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 21:12:35\tMarkDuplicates\tAfter output close freeMemory: 528129552; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 21:12:35 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.80 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 21:12:39 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T210258.730307.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:12:40 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753371_chr5_1_fastp.fastq.gz, data/processed/ERR753371_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753371_chr5.sam, results/mapped_reads/ERR753371_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753371_chr5_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR753371_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.61 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 70.46 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.56 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.39 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 40.06 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 119.072 sec; CPU: 112.084 sec\n",
      "[Mon Aug 28 21:15:54 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:15:54 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753371_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr5.bam, results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753371_chr5.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753371_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 21:16:25 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:16:25 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753371_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753371_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "21:16:26.502 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 21:16:26 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753371_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753371_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 21:16:26 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 21:16:26\tMarkDuplicates\tStart of doWork freeMemory: 528786696; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:16:26\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 21:16:26\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam\n",
      "INFO\t2023-08-28 21:16:28\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:109,819,561\n",
      "INFO\t2023-08-28 21:16:28\tMarkDuplicates\tTracking 137 as yet unmatched pairs. 137 records in RAM.\n",
      "INFO\t2023-08-28 21:16:29\tMarkDuplicates\tRead 1717822 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 21:16:29\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 345154848; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:16:29\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 21:16:29\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 21:16:29\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 21:16:30\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 21:16:30\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 816789472; totalMemory: 1363148800; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:16:30\tMarkDuplicates\tMarking 690690 records as duplicates.\n",
      "INFO\t2023-08-28 21:16:30\tMarkDuplicates\tFound 61 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam\n",
      "INFO\t2023-08-28 21:16:30\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 21:16:40\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 21:16:40\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 21:16:40\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 21:16:40\tMarkDuplicates\tBefore output close freeMemory: 528081920; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:16:40\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 21:16:40\tMarkDuplicates\tAfter output close freeMemory: 528081920; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 21:16:40 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.24 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 21:16:41 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T211239.672231.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:16:43 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753372_chr5_1_fastp.fastq.gz, data/processed/ERR753372_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753372_chr5.sam, results/mapped_reads/ERR753372_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753372_chr5_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753372_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.61 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 71.26 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.56 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.39 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 40.06 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 118.730 sec; CPU: 112.872 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 21:21:13 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:21:13 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753372_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr5.bam, results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753372_chr5.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753372_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 21:22:13 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:22:13 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753372_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753372_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "21:22:13.890 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 21:22:13 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753372_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753372_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 21:22:13 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 21:22:13\tMarkDuplicates\tStart of doWork freeMemory: 528782616; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:22:13\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 21:22:13\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam\n",
      "INFO\t2023-08-28 21:22:15\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:70,941,592\n",
      "INFO\t2023-08-28 21:22:15\tMarkDuplicates\tTracking 508 as yet unmatched pairs. 508 records in RAM.\n",
      "INFO\t2023-08-28 21:22:16\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 5:134,741,301\n",
      "INFO\t2023-08-28 21:22:16\tMarkDuplicates\tTracking 568 as yet unmatched pairs. 568 records in RAM.\n",
      "INFO\t2023-08-28 21:22:18\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 5:176,864,748\n",
      "INFO\t2023-08-28 21:22:18\tMarkDuplicates\tTracking 141 as yet unmatched pairs. 141 records in RAM.\n",
      "INFO\t2023-08-28 21:22:18\tMarkDuplicates\tRead 3202425 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 21:22:18\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 432132584; totalMemory: 848297984; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:22:18\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 21:22:18\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 21:22:19\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 21:22:19\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 21:22:19\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1063203856; totalMemory: 1609564160; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:22:19\tMarkDuplicates\tMarking 77166 records as duplicates.\n",
      "INFO\t2023-08-28 21:22:19\tMarkDuplicates\tFound 518 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam\n",
      "INFO\t2023-08-28 21:22:19\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 21:22:38\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 21:22:38\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 21:22:38\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 21:22:38\tMarkDuplicates\tBefore output close freeMemory: 528080960; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:22:38\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 21:22:38\tMarkDuplicates\tAfter output close freeMemory: 528080960; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 21:22:38 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.42 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 21:22:41 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T211642.086712.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:22:43 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753373_chr5_1_fastp.fastq.gz, data/processed/ERR753373_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753373_chr5.sam, results/mapped_reads/ERR753373_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753373_chr5_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR753373_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.62 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 70.55 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.57 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.38 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 39.97 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 118.082 sec; CPU: 112.086 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 21:28:37 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:28:37 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753373_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr5.bam, results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753373_chr5.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753373_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 21:30:26 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:30:26 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753373_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753373_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "21:30:26.977 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 21:30:26 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753373_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753373_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 21:30:26 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 21:30:26\tMarkDuplicates\tStart of doWork freeMemory: 528792080; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:30:26\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 21:30:26\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam\n",
      "INFO\t2023-08-28 21:30:28\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:56,495,436\n",
      "INFO\t2023-08-28 21:30:28\tMarkDuplicates\tTracking 269 as yet unmatched pairs. 269 records in RAM.\n",
      "INFO\t2023-08-28 21:30:30\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 5:102,420,479\n",
      "INFO\t2023-08-28 21:30:30\tMarkDuplicates\tTracking 274 as yet unmatched pairs. 274 records in RAM.\n",
      "INFO\t2023-08-28 21:30:31\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 5:152,603,484\n",
      "INFO\t2023-08-28 21:30:31\tMarkDuplicates\tTracking 150 as yet unmatched pairs. 150 records in RAM.\n",
      "INFO\t2023-08-28 21:30:32\tMarkDuplicates\tRead 3481677 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 21:30:32\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 537016128; totalMemory: 982515712; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:30:32\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 21:30:32\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 21:30:33\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 21:30:33\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 21:30:33\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 973035840; totalMemory: 1519386624; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:30:33\tMarkDuplicates\tMarking 92722 records as duplicates.\n",
      "INFO\t2023-08-28 21:30:33\tMarkDuplicates\tFound 30641 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam\n",
      "INFO\t2023-08-28 21:30:33\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 21:31:04\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 21:31:04\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 21:31:04\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 21:31:04\tMarkDuplicates\tBefore output close freeMemory: 528090720; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:31:04\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 21:31:04\tMarkDuplicates\tAfter output close freeMemory: 528090720; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 21:31:04 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.63 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 21:31:07 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T212241.835680.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:31:09 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753374_chr5_1_fastp.fastq.gz, data/processed/ERR753374_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753374_chr5.sam, results/mapped_reads/ERR753374_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753374_chr5_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753374_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.61 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 70.54 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.57 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.38 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 40.07 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 118.096 sec; CPU: 112.176 sec\n",
      "[Mon Aug 28 21:35:13 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:35:13 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753374_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr5.bam, results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753374_chr5.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753374_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 21:36:09 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:36:09 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753374_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753374_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "21:36:10.374 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 21:36:10 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753374_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753374_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 21:36:10 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 21:36:10\tMarkDuplicates\tStart of doWork freeMemory: 528825712; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:36:10\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 21:36:10\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam\n",
      "INFO\t2023-08-28 21:36:12\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:88,348,059\n",
      "INFO\t2023-08-28 21:36:12\tMarkDuplicates\tTracking 230 as yet unmatched pairs. 230 records in RAM.\n",
      "INFO\t2023-08-28 21:36:13\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 5:177,739,568\n",
      "INFO\t2023-08-28 21:36:13\tMarkDuplicates\tTracking 64 as yet unmatched pairs. 64 records in RAM.\n",
      "INFO\t2023-08-28 21:36:13\tMarkDuplicates\tRead 2038926 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 21:36:13\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 267682016; totalMemory: 557842432; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:36:13\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 21:36:14\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 21:36:14\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 21:36:14\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 21:36:14\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 871356512; totalMemory: 1417674752; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:36:14\tMarkDuplicates\tMarking 219063 records as duplicates.\n",
      "INFO\t2023-08-28 21:36:14\tMarkDuplicates\tFound 73631 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam\n",
      "INFO\t2023-08-28 21:36:14\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 21:36:32\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 21:36:32\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 21:36:32\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 21:36:32\tMarkDuplicates\tBefore output close freeMemory: 528123136; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:36:32\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 21:36:32\tMarkDuplicates\tAfter output close freeMemory: 528123136; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 21:36:32 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.37 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 21:36:34 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T213107.947985.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:36:35 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753375_chr5_1_fastp.fastq.gz, data/processed/ERR753375_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753375_chr5.sam, results/mapped_reads/ERR753375_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753375_chr5_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753375_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.64 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 70.54 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.57 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.38 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 40.06 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 119.917 sec; CPU: 112.186 sec\n",
      "[Mon Aug 28 21:39:41 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:39:41 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753375_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr5.bam, results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753375_chr5.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753375_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 21:40:08 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:40:08 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753375_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753375_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "21:40:09.373 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 21:40:09 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753375_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753375_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 21:40:09 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 21:40:09\tMarkDuplicates\tStart of doWork freeMemory: 528788296; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:40:09\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 21:40:09\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam\n",
      "INFO\t2023-08-28 21:40:10\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:136,231,424\n",
      "INFO\t2023-08-28 21:40:10\tMarkDuplicates\tTracking 189 as yet unmatched pairs. 189 records in RAM.\n",
      "INFO\t2023-08-28 21:40:11\tMarkDuplicates\tRead 1429511 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 21:40:11\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 311480416; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:40:11\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 21:40:11\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 21:40:12\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 21:40:12\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 21:40:12\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 763316696; totalMemory: 1309671424; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:40:12\tMarkDuplicates\tMarking 67772 records as duplicates.\n",
      "INFO\t2023-08-28 21:40:12\tMarkDuplicates\tFound 152 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam\n",
      "INFO\t2023-08-28 21:40:12\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 21:40:21\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 21:40:21\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 21:40:21\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 21:40:21\tMarkDuplicates\tBefore output close freeMemory: 528086784; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:40:21\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 21:40:21\tMarkDuplicates\tAfter output close freeMemory: 528086784; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 21:40:21 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.20 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 21:40:22 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T213634.511069.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:40:23 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753376_chr5_1_fastp.fastq.gz, data/processed/ERR753376_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753376_chr5.sam, results/mapped_reads/ERR753376_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753376_chr5_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753376_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.63 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 70.63 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.56 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.38 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 39.92 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 118.105 sec; CPU: 112.118 sec\n",
      "[Mon Aug 28 21:43:38 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:43:38 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753376_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr5.bam, results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753376_chr5.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753376_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 21:44:12 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:44:12 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753376_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753376_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "21:44:13.865 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 21:44:13 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753376_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753376_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 21:44:13 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 21:44:13\tMarkDuplicates\tStart of doWork freeMemory: 528784704; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:44:13\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 21:44:13\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam\n",
      "INFO\t2023-08-28 21:44:15\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:109,378,593\n",
      "INFO\t2023-08-28 21:44:15\tMarkDuplicates\tTracking 265 as yet unmatched pairs. 265 records in RAM.\n",
      "INFO\t2023-08-28 21:44:16\tMarkDuplicates\tRead 1758630 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 21:44:16\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 276199760; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:44:16\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 21:44:16\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 21:44:17\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 21:44:17\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 21:44:17\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 801060640; totalMemory: 1347420160; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:44:17\tMarkDuplicates\tMarking 249647 records as duplicates.\n",
      "INFO\t2023-08-28 21:44:17\tMarkDuplicates\tFound 80 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam\n",
      "INFO\t2023-08-28 21:44:17\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 21:44:28\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 21:44:28\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 21:44:28\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 21:44:28\tMarkDuplicates\tBefore output close freeMemory: 528081672; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:44:28\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 21:44:28\tMarkDuplicates\tAfter output close freeMemory: 528081672; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 21:44:28 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.25 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 21:44:30 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T214022.723378.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:44:31 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753377_chr5_1_fastp.fastq.gz, data/processed/ERR753377_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753377_chr5.sam, results/mapped_reads/ERR753377_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753377_chr5_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753377_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.61 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 70.34 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.55 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.37 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 40.01 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 117.870 sec; CPU: 111.892 sec\n",
      "[Mon Aug 28 21:47:51 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:47:51 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753377_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr5.bam, results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753377_chr5.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753377_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 21:48:26 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:48:26 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753377_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753377_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "21:48:27.831 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 21:48:27 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753377_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753377_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 21:48:27 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 21:48:27\tMarkDuplicates\tStart of doWork freeMemory: 528784408; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:48:27\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 21:48:27\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam\n",
      "INFO\t2023-08-28 21:48:29\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:97,737,191\n",
      "INFO\t2023-08-28 21:48:29\tMarkDuplicates\tTracking 399 as yet unmatched pairs. 399 records in RAM.\n",
      "INFO\t2023-08-28 21:48:30\tMarkDuplicates\tRead 1869830 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 21:48:30\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 264286632; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:48:30\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 21:48:30\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 21:48:31\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 21:48:31\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 21:48:31\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 841954832; totalMemory: 1388314624; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:48:31\tMarkDuplicates\tMarking 263276 records as duplicates.\n",
      "INFO\t2023-08-28 21:48:31\tMarkDuplicates\tFound 72 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam\n",
      "INFO\t2023-08-28 21:48:31\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 21:48:43\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 21:48:43\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 21:48:43\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 21:48:43\tMarkDuplicates\tBefore output close freeMemory: 528081480; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:48:43\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 21:48:43\tMarkDuplicates\tAfter output close freeMemory: 528081480; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 21:48:43 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.26 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 21:48:44 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T214430.341803.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:48:45 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753378_chr5_1_fastp.fastq.gz, data/processed/ERR753378_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753378_chr5.sam, results/mapped_reads/ERR753378_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753378_chr5_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753378_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.61 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 69.95 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.56 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.38 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 40.05 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 117.593 sec; CPU: 111.545 sec\n",
      "[Mon Aug 28 21:52:45 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:52:45 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753378_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr5.bam, results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753378_chr5.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753378_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 21:53:27 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:53:27 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753378_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753378_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "21:53:28.672 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 21:53:28 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753378_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753378_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 21:53:28 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 21:53:28\tMarkDuplicates\tStart of doWork freeMemory: 528781344; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:53:28\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 21:53:28\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam\n",
      "INFO\t2023-08-28 21:53:30\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:77,325,554\n",
      "INFO\t2023-08-28 21:53:30\tMarkDuplicates\tTracking 549 as yet unmatched pairs. 549 records in RAM.\n",
      "INFO\t2023-08-28 21:53:31\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 5:141,529,643\n",
      "INFO\t2023-08-28 21:53:31\tMarkDuplicates\tTracking 415 as yet unmatched pairs. 415 records in RAM.\n",
      "INFO\t2023-08-28 21:53:32\tMarkDuplicates\tRead 2755612 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 21:53:32\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 563660256; totalMemory: 931135488; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:53:32\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 21:53:33\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 21:53:33\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 21:53:33\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 21:53:33\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 921644176; totalMemory: 1468006400; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:53:33\tMarkDuplicates\tMarking 66434 records as duplicates.\n",
      "INFO\t2023-08-28 21:53:33\tMarkDuplicates\tFound 462 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam\n",
      "INFO\t2023-08-28 21:53:33\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 21:53:50\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 21:53:50\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 21:53:50\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 21:53:50\tMarkDuplicates\tBefore output close freeMemory: 528079360; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 21:53:50\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 21:53:50\tMarkDuplicates\tAfter output close freeMemory: 528079360; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 21:53:50 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.36 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 21:53:52 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T214844.847914.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:53:53 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR696683_chr5_dedup.bam\n",
      "    output: results/variants/ERR696683_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR696683_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR696683_chr5.vcf\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "index file data/reference/genome.fa.fai not found, generating...\n",
      "[Mon Aug 28 21:54:43 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T215352.335737.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:54:45 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753368_chr5_dedup.bam\n",
      "    output: results/variants/ERR753368_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753368_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753368_chr5.vcf\n",
      "    wildcards: sample=ERR753368_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 21:55:52 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T215444.170459.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:55:53 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753369_chr5_dedup.bam\n",
      "    output: results/variants/ERR753369_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753369_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753369_chr5.vcf\n",
      "    wildcards: sample=ERR753369_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 21:57:02 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T215552.814326.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:57:04 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753370_chr5_dedup.bam\n",
      "    output: results/variants/ERR753370_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753370_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753370_chr5.vcf\n",
      "    wildcards: sample=ERR753370_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 21:58:55 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T215702.975602.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:58:56 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753371_chr5_dedup.bam\n",
      "    output: results/variants/ERR753371_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753371_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753371_chr5.vcf\n",
      "    wildcards: sample=ERR753371_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 21:59:32 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T215855.506924.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 21:59:34 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753372_chr5_dedup.bam\n",
      "    output: results/variants/ERR753372_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753372_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753372_chr5.vcf\n",
      "    wildcards: sample=ERR753372_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 22:00:43 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T215933.137258.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:00:45 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753373_chr5_dedup.bam\n",
      "    output: results/variants/ERR753373_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753373_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753373_chr5.vcf\n",
      "    wildcards: sample=ERR753373_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 22:02:15 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T220044.028768.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:02:16 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753374_chr5_dedup.bam\n",
      "    output: results/variants/ERR753374_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753374_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753374_chr5.vcf\n",
      "    wildcards: sample=ERR753374_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 22:03:12 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T220215.811414.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:03:14 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753375_chr5_dedup.bam\n",
      "    output: results/variants/ERR753375_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753375_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753375_chr5.vcf\n",
      "    wildcards: sample=ERR753375_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 22:03:53 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T220313.163202.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:03:54 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753376_chr5_dedup.bam\n",
      "    output: results/variants/ERR753376_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753376_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753376_chr5.vcf\n",
      "    wildcards: sample=ERR753376_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 22:04:35 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T220353.317727.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:04:36 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753377_chr5_dedup.bam\n",
      "    output: results/variants/ERR753377_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753377_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753377_chr5.vcf\n",
      "    wildcards: sample=ERR753377_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 22:05:18 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T220435.246209.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:05:19 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753378_chr5_dedup.bam\n",
      "    output: results/variants/ERR753378_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753378_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753378_chr5.vcf\n",
      "    wildcards: sample=ERR753378_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 22:06:18 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T220518.213983.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:06:19 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR696683_chr5.vcf\n",
      "    output: results/variants/vep/ERR696683_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR696683_chr5.txt\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Mon Aug 28 22:06:44 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T220618.817574.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:06:46 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753368_chr5.vcf\n",
      "    output: results/variants/vep/ERR753368_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753368_chr5.txt\n",
      "    wildcards: sample=ERR753368_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Mon Aug 28 22:07:13 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T220644.957527.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:07:14 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753369_chr5.vcf\n",
      "    output: results/variants/vep/ERR753369_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753369_chr5.txt\n",
      "    wildcards: sample=ERR753369_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Mon Aug 28 22:07:40 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T220713.734466.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:07:41 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753370_chr5.vcf\n",
      "    output: results/variants/vep/ERR753370_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753370_chr5.txt\n",
      "    wildcards: sample=ERR753370_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Mon Aug 28 22:08:08 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T220740.410033.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:08:09 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753371_chr5.vcf\n",
      "    output: results/variants/vep/ERR753371_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753371_chr5.txt\n",
      "    wildcards: sample=ERR753371_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Mon Aug 28 22:08:29 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T220808.930403.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:08:30 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753372_chr5.vcf\n",
      "    output: results/variants/vep/ERR753372_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753372_chr5.txt\n",
      "    wildcards: sample=ERR753372_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Mon Aug 28 22:08:56 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T220829.700614.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:08:57 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753373_chr5.vcf\n",
      "    output: results/variants/vep/ERR753373_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753373_chr5.txt\n",
      "    wildcards: sample=ERR753373_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Mon Aug 28 22:09:25 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T220856.925047.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:09:26 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753374_chr5.vcf\n",
      "    output: results/variants/vep/ERR753374_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753374_chr5.txt\n",
      "    wildcards: sample=ERR753374_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Mon Aug 28 22:09:47 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T220925.391784.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:09:48 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753375_chr5.vcf\n",
      "    output: results/variants/vep/ERR753375_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753375_chr5.txt\n",
      "    wildcards: sample=ERR753375_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Mon Aug 28 22:10:08 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T220947.626125.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:10:10 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753376_chr5.vcf\n",
      "    output: results/variants/vep/ERR753376_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753376_chr5.txt\n",
      "    wildcards: sample=ERR753376_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Mon Aug 28 22:10:31 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T221009.023617.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:10:32 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753377_chr5.vcf\n",
      "    output: results/variants/vep/ERR753377_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753377_chr5.txt\n",
      "    wildcards: sample=ERR753377_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Mon Aug 28 22:10:53 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T221031.521073.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:10:54 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753378_chr5.vcf\n",
      "    output: results/variants/vep/ERR753378_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753378_chr5.txt\n",
      "    wildcards: sample=ERR753378_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Mon Aug 28 22:11:19 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T221053.906433.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job              count    min threads    max threads\n",
      "-------------  -------  -------------  -------------\n",
      "parsing_dataR        1              1              1\n",
      "total                1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:11:20 2023]\n",
      "rule parsing_dataR:\n",
      "    input: code/05parsing_vep_data.R\n",
      "    output: tasks/13parsing_dataR.done\n",
      "    jobid: 0\n",
      "    reason: Params have changed since last execution\n",
      "    resources: tmpdir=/tmp\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">>> This is your sample list: ERR696683_chr5\n",
      ">>> This is your sample list: ERR753368_chr5\n",
      ">>> This is your sample list: ERR753369_chr5\n",
      ">>> This is your sample list: ERR753370_chr5\n",
      ">>> This is your sample list: ERR753371_chr5\n",
      ">>> This is your sample list: ERR753372_chr5\n",
      ">>> This is your sample list: ERR753373_chr5\n",
      ">>> This is your sample list: ERR753374_chr5\n",
      ">>> This is your sample list: ERR753375_chr5\n",
      ">>> This is your sample list: ERR753376_chr5\n",
      ">>> This is your sample list: ERR753377_chr5\n",
      ">>> This is your sample list: ERR753378_chr5\n",
      ">>> This is your gene: APC\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 0 × 44\n",
      "# ℹ 44 variables: #uploaded_variation <chr>, location <chr>, allele <chr>,\n",
      "#   gene <chr>, feature <chr>, feature_type <chr>, consequence <chr>,\n",
      "#   cdna_position <chr>, cds_position <chr>, protein_position <chr>,\n",
      "#   amino_acids <chr>, codons <chr>, existing_variation <chr>, impact <chr>,\n",
      "#   distance <chr>, strand <chr>, flags <chr>, variant_class <chr>,\n",
      "#   symbol <chr>, symbol_source <chr>, hgnc_id <chr>, biotype <chr>,\n",
      "#   mane_select <chr>, mane_plus_clinical <chr>, tsl <chr>, appris <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 71 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112707850_C/T       5:1127078… T      ENSG… ENST00… Transcript   missense_v…\n",
      "2 5_112707850_C/T       5:1127078… T      ENSG… ENST00… Transcript   missense_v…\n",
      "3 5_112707850_C/T       5:1127078… T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 65 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 69 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112724597_G/C       5:1127245… C      ENSG… ENST00… Transcript   intron_var…\n",
      "2 5_112724597_G/C       5:1127245… C      ENSG… ENST00… Transcript   intron_var…\n",
      "3 5_112724597_G/C       5:1127245… C      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 63 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 86 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "2 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "3 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112778038_C/T       5:1127780… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 80 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 0 × 44\n",
      "# ℹ 44 variables: #uploaded_variation <chr>, location <chr>, allele <chr>,\n",
      "#   gene <chr>, feature <chr>, feature_type <chr>, consequence <chr>,\n",
      "#   cdna_position <chr>, cds_position <chr>, protein_position <chr>,\n",
      "#   amino_acids <chr>, codons <chr>, existing_variation <chr>, impact <chr>,\n",
      "#   distance <chr>, strand <chr>, flags <chr>, variant_class <chr>,\n",
      "#   symbol <chr>, symbol_source <chr>, hgnc_id <chr>, biotype <chr>,\n",
      "#   mane_select <chr>, mane_plus_clinical <chr>, tsl <chr>, appris <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 64 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112707850_C/T       5:1127078… T      ENSG… ENST00… Transcript   missense_v…\n",
      "2 5_112707850_C/T       5:1127078… T      ENSG… ENST00… Transcript   missense_v…\n",
      "3 5_112707850_C/T       5:1127078… T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 58 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 78 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112775373_G/A       5:1127753… A      ENSG… ENST00… Transcript   intron_var…\n",
      "2 5_112775373_G/A       5:1127753… A      ENSG… ENST00… Transcript   intron_var…\n",
      "3 5_112775373_G/A       5:1127753… A      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112775373_G/A       5:1127753… A      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112775373_G/A       5:1127753… A      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 72 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 59 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112729129_G/C       5:1127291… C      ENSG… ENST00… Transcript   intron_var…\n",
      "2 5_112729129_G/C       5:1127291… C      ENSG… ENST00… Transcript   intron_var…\n",
      "3 5_112729129_G/C       5:1127291… C      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112775373_G/A       5:1127753… A      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112775373_G/A       5:1127753… A      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112775373_G/A       5:1127753… A      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 53 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 61 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "2 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "3 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112778038_C/T       5:1127780… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 55 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 71 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "2 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "3 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112778038_C/T       5:1127780… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 65 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 78 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "2 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "3 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112778038_C/T       5:1127780… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 72 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 66 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "2 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "3 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112778038_C/T       5:1127780… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 60 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "[1] \"##==========================##\"\n",
      "[1] \"##===> WORK FINISHED!!! <===##\"\n",
      "[1] \"##==========================##\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "Touching output file tasks/13parsing_dataR.done.\n",
      "[Mon Aug 28 22:11:22 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T221119.790527.snakemake.log\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['snakemake', '--cores', '1', 'parsing_dataR'], returncode=0)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Running the workflow for the 5º chormosome\n",
    "\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--force\", \"data/reference/genome.fa\"]) ## rule reference_genome \n",
    "\n",
    "for sample in sample_list:\n",
    "    ## Quality inspection\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/fastqc_result/\" + sample + \"_chr5_1_fastqc.html\",   \n",
    "                    \"data/processed/\" + sample + \"_chr5_1_fastp.fastq.gz\",   \n",
    "                    \"results/fastqc_result/trimmed/\" + sample + \"_chr5_1_fastp_fastqc.html\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    ## Mapping reads\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/mapped_reads/\" + sample + \"_chr5_sorted.sam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr5_sorted.bam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr5_dedup.bam\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    # Extracting variants  \n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/\" + sample + \"_chr5.vcf\"])   \n",
    "\n",
    "## VEP CLI\n",
    "for sample in sample_list:\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/vep/\" + sample + \"_chr5.txt\"])\n",
    "\n",
    "## Extracting the gene PIKCA3 from the VEP files\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"parsing_dataR\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Changing the config file to do the chromosome 7 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['sed', '-i', '65s/APC/BRAF/g', 'config.yaml'], returncode=0)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subprocess.run([\"sed\", \"-i\", \"6,+12s/chr5/chr7/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"24,+12s/chr5/chr7/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\",\"39s/chr5/chr7/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"46s/chromosome.5.fa.gz/chromosome.7.fa.gz/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"65s/APC/BRAF/g\", \"config.yaml\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job                 count    min threads    max threads\n",
      "----------------  -------  -------------  -------------\n",
      "reference_genome        1              1              1\n",
      "total                   1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:55:34 2023]\n",
      "rule reference_genome:\n",
      "    output: data/reference/genome.fa\n",
      "    jobid: 0\n",
      "    reason: Forced execution\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "--2023-08-28 22:55:34--  https://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.7.fa.gz\n",
      "Resolviendo ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.169\n",
      "Conectando con ftp.ensembl.org (ftp.ensembl.org)[193.62.193.169]:443... conectado.\n",
      "Petición HTTP enviada, esperando respuesta... 200 OK\n",
      "Longitud: 47416846 (45M) [application/x-gzip]\n",
      "Guardando como: ‘data/reference/genome.fa.gz’\n",
      "\n",
      "     0K .......... .......... .......... .......... ..........  0%  461K 1m40s\n",
      "    50K .......... .......... .......... .......... ..........  0% 1,35M 67s\n",
      "   100K .......... .......... .......... .......... ..........  0% 2,34M 51s\n",
      "   150K .......... .......... .......... .......... ..........  0% 15,8M 39s\n",
      "   200K .......... .......... .......... .......... ..........  0%  916K 41s\n",
      "   250K .......... .......... .......... .......... ..........  0% 98,3M 34s\n",
      "   300K .......... .......... .......... .......... ..........  0%  130M 29s\n",
      "   350K .......... .......... .......... .......... ..........  0% 25,7M 26s\n",
      "   400K .......... .......... .......... .......... ..........  0% 1,52M 26s\n",
      "   450K .......... .......... .......... .......... ..........  1% 2,40M 26s\n",
      "   500K .......... .......... .......... .......... ..........  1% 30,4M 23s\n",
      "   550K .......... .......... .......... .......... ..........  1% 49,5M 21s\n",
      "   600K .......... .......... .......... .......... ..........  1% 61,0M 20s\n",
      "   650K .......... .......... .......... .......... ..........  1%  181M 18s\n",
      "   700K .......... .......... .......... .......... ..........  1%  149M 17s\n",
      "   750K .......... .......... .......... .......... ..........  1%  121M 16s\n",
      "   800K .......... .......... .......... .......... ..........  1% 50,2M 15s\n",
      "   850K .......... .......... .......... .......... ..........  1% 1,58M 16s\n",
      "   900K .......... .......... .......... .......... ..........  2% 2,76M 16s\n",
      "   950K .......... .......... .......... .......... ..........  2% 19,0M 15s\n",
      "  1000K .......... .......... .......... .......... ..........  2% 43,8M 14s\n",
      "  1050K .......... .......... .......... .......... ..........  2% 74,1M 14s\n",
      "  1100K .......... .......... .......... .......... ..........  2% 42,8M 13s\n",
      "  1150K .......... .......... .......... .......... ..........  2% 55,5M 13s\n",
      "  1200K .......... .......... .......... .......... ..........  2% 35,2M 12s\n",
      "  1250K .......... .......... .......... .......... ..........  2% 51,4M 12s\n",
      "  1300K .......... .......... .......... .......... ..........  2% 76,3M 11s\n",
      "  1350K .......... .......... .......... .......... ..........  3%  114M 11s\n",
      "  1400K .......... .......... .......... .......... ..........  3%  139M 11s\n",
      "  1450K .......... .......... .......... .......... ..........  3%  139M 10s\n",
      "  1500K .......... .......... .......... .......... ..........  3%  158M 10s\n",
      "  1550K .......... .......... .......... .......... ..........  3%  158M 10s\n",
      "  1600K .......... .......... .......... .......... ..........  3%  137M 9s\n",
      "  1650K .......... .......... .......... .......... ..........  3%  161M 9s\n",
      "  1700K .......... .......... .......... .......... ..........  3%  133M 9s\n",
      "  1750K .......... .......... .......... .......... ..........  3% 1,89M 9s\n",
      "  1800K .......... .......... .......... .......... ..........  3% 74,5M 9s\n",
      "  1850K .......... .......... .......... .......... ..........  4% 2,65M 9s\n",
      "  1900K .......... .......... .......... .......... ..........  4% 47,3M 9s\n",
      "  1950K .......... .......... .......... .......... ..........  4% 64,0M 9s\n",
      "  2000K .......... .......... .......... .......... ..........  4% 44,5M 8s\n",
      "  2050K .......... .......... .......... .......... ..........  4%  120M 8s\n",
      "  2100K .......... .......... .......... .......... ..........  4%  103M 8s\n",
      "  2150K .......... .......... .......... .......... ..........  4% 72,4M 8s\n",
      "  2200K .......... .......... .......... .......... ..........  4%  127M 8s\n",
      "  2250K .......... .......... .......... .......... ..........  4%  144M 8s\n",
      "  2300K .......... .......... .......... .......... ..........  5%  140M 7s\n",
      "  2350K .......... .......... .......... .......... ..........  5%  116M 7s\n",
      "  2400K .......... .......... .......... .......... ..........  5%  115M 7s\n",
      "  2450K .......... .......... .......... .......... ..........  5%  105M 7s\n",
      "  2500K .......... .......... .......... .......... ..........  5%  119M 7s\n",
      "  2550K .......... .......... .......... .......... ..........  5%  116M 7s\n",
      "  2600K .......... .......... .......... .......... ..........  5%  104M 7s\n",
      "  2650K .......... .......... .......... .......... ..........  5%  120M 6s\n",
      "  2700K .......... .......... .......... .......... ..........  5%  102M 6s\n",
      "  2750K .......... .......... .......... .......... ..........  6%  118M 6s\n",
      "  2800K .......... .......... .......... .......... ..........  6%  118M 6s\n",
      "  2850K .......... .......... .......... .......... ..........  6%  103M 6s\n",
      "  2900K .......... .......... .......... .......... ..........  6%  117M 6s\n",
      "  2950K .......... .......... .......... .......... ..........  6%  104M 6s\n",
      "  3000K .......... .......... .......... .......... ..........  6%  118M 6s\n",
      "  3050K .......... .......... .......... .......... ..........  6%  121M 6s\n",
      "  3100K .......... .......... .......... .......... ..........  6%  102M 6s\n",
      "  3150K .......... .......... .......... .......... ..........  6%  118M 5s\n",
      "  3200K .......... .......... .......... .......... ..........  7%  103M 5s\n",
      "  3250K .......... .......... .......... .......... ..........  7%  117M 5s\n",
      "  3300K .......... .......... .......... .......... ..........  7%  120M 5s\n",
      "  3350K .......... .......... .......... .......... ..........  7%  102M 5s\n",
      "  3400K .......... .......... .......... .......... ..........  7%  120M 5s\n",
      "  3450K .......... .......... .......... .......... ..........  7%  104M 5s\n",
      "  3500K .......... .......... .......... .......... ..........  7%  118M 5s\n",
      "  3550K .......... .......... .......... .......... ..........  7% 2,29M 5s\n",
      "  3600K .......... .......... .......... .......... ..........  7% 39,0M 5s\n",
      "  3650K .......... .......... .......... .......... ..........  7% 94,9M 5s\n",
      "  3700K .......... .......... .......... .......... ..........  8% 90,8M 5s\n",
      "  3750K .......... .......... .......... .......... ..........  8% 2,98M 5s\n",
      "  3800K .......... .......... .......... .......... ..........  8% 42,3M 5s\n",
      "  3850K .......... .......... .......... .......... ..........  8% 57,6M 5s\n",
      "  3900K .......... .......... .......... .......... ..........  8%  123M 5s\n",
      "  3950K .......... .......... .......... .......... ..........  8%  102M 5s\n",
      "  4000K .......... .......... .......... .......... ..........  8%  117M 5s\n",
      "  4050K .......... .......... .......... .......... ..........  8%  121M 5s\n",
      "  4100K .......... .......... .......... .......... ..........  8%  102M 5s\n",
      "  4150K .......... .......... .......... .......... ..........  9%  117M 5s\n",
      "  4200K .......... .......... .......... .......... ..........  9%  102M 4s\n",
      "  4250K .......... .......... .......... .......... ..........  9%  120M 4s\n",
      "  4300K .......... .......... .......... .......... ..........  9%  118M 4s\n",
      "  4350K .......... .......... .......... .......... ..........  9%  102M 4s\n",
      "  4400K .......... .......... .......... .......... ..........  9%  120M 4s\n",
      "  4450K .......... .......... .......... .......... ..........  9%  102M 4s\n",
      "  4500K .......... .......... .......... .......... ..........  9%  122M 4s\n",
      "  4550K .......... .......... .......... .......... ..........  9%  117M 4s\n",
      "  4600K .......... .......... .......... .......... .......... 10%  102M 4s\n",
      "  4650K .......... .......... .......... .......... .......... 10%  121M 4s\n",
      "  4700K .......... .......... .......... .......... .......... 10%  102M 4s\n",
      "  4750K .......... .......... .......... .......... .......... 10%  117M 4s\n",
      "  4800K .......... .......... .......... .......... .......... 10%  121M 4s\n",
      "  4850K .......... .......... .......... .......... .......... 10%  102M 4s\n",
      "  4900K .......... .......... .......... .......... .......... 10%  119M 4s\n",
      "  4950K .......... .......... .......... .......... .......... 10%  102M 4s\n",
      "  5000K .......... .......... .......... .......... .......... 10%  121M 4s\n",
      "  5050K .......... .......... .......... .......... .......... 11%  121M 4s\n",
      "  5100K .......... .......... .......... .......... .......... 11%  102M 4s\n",
      "  5150K .......... .......... .......... .......... .......... 11%  119M 4s\n",
      "  5200K .......... .......... .......... .......... .......... 11%  102M 4s\n",
      "  5250K .......... .......... .......... .......... .......... 11%  121M 4s\n",
      "  5300K .......... .......... .......... .......... .......... 11%  117M 4s\n",
      "  5350K .......... .......... .......... .......... .......... 11%  102M 3s\n",
      "  5400K .......... .......... .......... .......... .......... 11%  119M 3s\n",
      "  5450K .......... .......... .......... .......... .......... 11%  105M 3s\n",
      "  5500K .......... .......... .......... .......... .......... 11%  120M 3s\n",
      "  5550K .......... .......... .......... .......... .......... 12%  118M 3s\n",
      "  5600K .......... .......... .......... .......... .......... 12%  102M 3s\n",
      "  5650K .......... .......... .......... .......... .......... 12%  120M 3s\n",
      "  5700K .......... .......... .......... .......... .......... 12%  102M 3s\n",
      "  5750K .......... .......... .......... .......... .......... 12%  118M 3s\n",
      "  5800K .......... .......... .......... .......... .......... 12%  120M 3s\n",
      "  5850K .......... .......... .......... .......... .......... 12%  103M 3s\n",
      "  5900K .......... .......... .......... .......... .......... 12%  118M 3s\n",
      "  5950K .......... .......... .......... .......... .......... 12%  102M 3s\n",
      "  6000K .......... .......... .......... .......... .......... 13%  121M 3s\n",
      "  6050K .......... .......... .......... .......... .......... 13%  121M 3s\n",
      "  6100K .......... .......... .......... .......... .......... 13%  102M 3s\n",
      "  6150K .......... .......... .......... .......... .......... 13%  118M 3s\n",
      "  6200K .......... .......... .......... .......... .......... 13%  103M 3s\n",
      "  6250K .......... .......... .......... .......... .......... 13%  121M 3s\n",
      "  6300K .......... .......... .......... .......... .......... 13%  117M 3s\n",
      "  6350K .......... .......... .......... .......... .......... 13%  102M 3s\n",
      "  6400K .......... .......... .......... .......... .......... 13%  120M 3s\n",
      "  6450K .......... .......... .......... .......... .......... 14%  103M 3s\n",
      "  6500K .......... .......... .......... .......... .......... 14%  118M 3s\n",
      "  6550K .......... .......... .......... .......... .......... 14%  121M 3s\n",
      "  6600K .......... .......... .......... .......... .......... 14% 3,93M 3s\n",
      "  6650K .......... .......... .......... .......... .......... 14% 40,7M 3s\n",
      "  6700K .......... .......... .......... .......... .......... 14% 49,2M 3s\n",
      "  6750K .......... .......... .......... .......... .......... 14%  117M 3s\n",
      "  6800K .......... .......... .......... .......... .......... 14% 14,2M 3s\n",
      "  6850K .......... .......... .......... .......... .......... 14% 3,50M 3s\n",
      "  6900K .......... .......... .......... .......... .......... 15% 57,9M 3s\n",
      "  6950K .......... .......... .......... .......... .......... 15% 54,4M 3s\n",
      "  7000K .......... .......... .......... .......... .......... 15%  121M 3s\n",
      "  7050K .......... .......... .......... .......... .......... 15%  116M 3s\n",
      "  7100K .......... .......... .......... .......... .......... 15%  103M 3s\n",
      "  7150K .......... .......... .......... .......... .......... 15%  119M 3s\n",
      "  7200K .......... .......... .......... .......... .......... 15%  104M 3s\n",
      "  7250K .......... .......... .......... .......... .......... 15%  122M 3s\n",
      "  7300K .......... .......... .......... .......... .......... 15%  114M 3s\n",
      "  7350K .......... .......... .......... .......... .......... 15%  104M 3s\n",
      "  7400K .......... .......... .......... .......... .......... 16%  122M 3s\n",
      "  7450K .......... .......... .......... .......... .......... 16%  102M 3s\n",
      "  7500K .......... .......... .......... .......... .......... 16%  117M 3s\n",
      "  7550K .......... .......... .......... .......... .......... 16%  123M 3s\n",
      "  7600K .......... .......... .......... .......... .......... 16%  101M 3s\n",
      "  7650K .......... .......... .......... .......... .......... 16% 87,6M 3s\n",
      "  7700K .......... .......... .......... .......... .......... 16%  102M 3s\n",
      "  7750K .......... .......... .......... .......... .......... 16%  117M 3s\n",
      "  7800K .......... .......... .......... .......... .......... 16%  121M 3s\n",
      "  7850K .......... .......... .......... .......... .......... 17%  103M 2s\n",
      "  7900K .......... .......... .......... .......... .......... 17%  118M 2s\n",
      "  7950K .......... .......... .......... .......... .......... 17%  102M 2s\n",
      "  8000K .......... .......... .......... .......... .......... 17%  121M 2s\n",
      "  8050K .......... .......... .......... .......... .......... 17%  119M 2s\n",
      "  8100K .......... .......... .......... .......... .......... 17%  101M 2s\n",
      "  8150K .......... .......... .......... .......... .......... 17%  121M 2s\n",
      "  8200K .......... .......... .......... .......... .......... 17%  102M 2s\n",
      "  8250K .......... .......... .......... .......... .......... 17%  123M 2s\n",
      "  8300K .......... .......... .......... .......... .......... 18%  117M 2s\n",
      "  8350K .......... .......... .......... .......... .......... 18%  103M 2s\n",
      "  8400K .......... .......... .......... .......... .......... 18%  120M 2s\n",
      "  8450K .......... .......... .......... .......... .......... 18%  103M 2s\n",
      "  8500K .......... .......... .......... .......... .......... 18%  117M 2s\n",
      "  8550K .......... .......... .......... .......... .......... 18%  119M 2s\n",
      "  8600K .......... .......... .......... .......... .......... 18%  104M 2s\n",
      "  8650K .......... .......... .......... .......... .......... 18%  118M 2s\n",
      "  8700K .......... .......... .......... .......... .......... 18%  103M 2s\n",
      "  8750K .......... .......... .......... .......... .......... 19%  120M 2s\n",
      "  8800K .......... .......... .......... .......... .......... 19%  122M 2s\n",
      "  8850K .......... .......... .......... .......... .......... 19%  102M 2s\n",
      "  8900K .......... .......... .......... .......... .......... 19%  118M 2s\n",
      "  8950K .......... .......... .......... .......... .......... 19%  102M 2s\n",
      "  9000K .......... .......... .......... .......... .......... 19%  121M 2s\n",
      "  9050K .......... .......... .......... .......... .......... 19%  117M 2s\n",
      "  9100K .......... .......... .......... .......... .......... 19%  102M 2s\n",
      "  9150K .......... .......... .......... .......... .......... 19%  121M 2s\n",
      "  9200K .......... .......... .......... .......... .......... 19%  102M 2s\n",
      "  9250K .......... .......... .......... .......... .......... 20%  121M 2s\n",
      "  9300K .......... .......... .......... .......... .......... 20%  117M 2s\n",
      "  9350K .......... .......... .......... .......... .......... 20%  102M 2s\n",
      "  9400K .......... .......... .......... .......... .......... 20%  121M 2s\n",
      "  9450K .......... .......... .......... .......... .......... 20%  103M 2s\n",
      "  9500K .......... .......... .......... .......... .......... 20%  117M 2s\n",
      "  9550K .......... .......... .......... .......... .......... 20%  121M 2s\n",
      "  9600K .......... .......... .......... .......... .......... 20%  102M 2s\n",
      "  9650K .......... .......... .......... .......... .......... 20% 5,25M 2s\n",
      "  9700K .......... .......... .......... .......... .......... 21% 12,7M 2s\n",
      "  9750K .......... .......... .......... .......... .......... 21% 41,0M 2s\n",
      "  9800K .......... .......... .......... .......... .......... 21% 57,1M 2s\n",
      "  9850K .......... .......... .......... .......... .......... 21% 25,4M 2s\n",
      "  9900K .......... .......... .......... .......... .......... 21% 3,30M 2s\n",
      "  9950K .......... .......... .......... .......... .......... 21% 52,3M 2s\n",
      " 10000K .......... .......... .......... .......... .......... 21%  112M 2s\n",
      " 10050K .......... .......... .......... .......... .......... 21% 67,5M 2s\n",
      " 10100K .......... .......... .......... .......... .......... 21% 55,5M 2s\n",
      " 10150K .......... .......... .......... .......... .......... 22%  119M 2s\n",
      " 10200K .......... .......... .......... .......... .......... 22%  103M 2s\n",
      " 10250K .......... .......... .......... .......... .......... 22%  121M 2s\n",
      " 10300K .......... .......... .......... .......... .......... 22%  120M 2s\n",
      " 10350K .......... .......... .......... .......... .......... 22% 69,5M 2s\n",
      " 10400K .......... .......... .......... .......... .......... 22%  127M 2s\n",
      " 10450K .......... .......... .......... .......... .......... 22%  129M 2s\n",
      " 10500K .......... .......... .......... .......... .......... 22%  148M 2s\n",
      " 10550K .......... .......... .......... .......... .......... 22%  128M 2s\n",
      " 10600K .......... .......... .......... .......... .......... 22%  103M 2s\n",
      " 10650K .......... .......... .......... .......... .......... 23%  115M 2s\n",
      " 10700K .......... .......... .......... .......... .......... 23%  105M 2s\n",
      " 10750K .......... .......... .......... .......... .......... 23%  120M 2s\n",
      " 10800K .......... .......... .......... .......... .......... 23%  115M 2s\n",
      " 10850K .......... .......... .......... .......... .......... 23%  103M 2s\n",
      " 10900K .......... .......... .......... .......... .......... 23%  120M 2s\n",
      " 10950K .......... .......... .......... .......... .......... 23%  102M 2s\n",
      " 11000K .......... .......... .......... .......... .......... 23%  118M 2s\n",
      " 11050K .......... .......... .......... .......... .......... 23%  117M 2s\n",
      " 11100K .......... .......... .......... .......... .......... 24%  104M 2s\n",
      " 11150K .......... .......... .......... .......... .......... 24%  122M 2s\n",
      " 11200K .......... .......... .......... .......... .......... 24%  104M 2s\n",
      " 11250K .......... .......... .......... .......... .......... 24%  117M 2s\n",
      " 11300K .......... .......... .......... .......... .......... 24%  121M 2s\n",
      " 11350K .......... .......... .......... .......... .......... 24%  102M 2s\n",
      " 11400K .......... .......... .......... .......... .......... 24%  121M 2s\n",
      " 11450K .......... .......... .......... .......... .......... 24%  102M 2s\n",
      " 11500K .......... .......... .......... .......... .......... 24%  118M 2s\n",
      " 11550K .......... .......... .......... .......... .......... 25%  122M 2s\n",
      " 11600K .......... .......... .......... .......... .......... 25%  101M 2s\n",
      " 11650K .......... .......... .......... .......... .......... 25%  117M 2s\n",
      " 11700K .......... .......... .......... .......... .......... 25%  103M 2s\n",
      " 11750K .......... .......... .......... .......... .......... 25%  120M 2s\n",
      " 11800K .......... .......... .......... .......... .......... 25%  118M 2s\n",
      " 11850K .......... .......... .......... .......... .......... 25%  104M 2s\n",
      " 11900K .......... .......... .......... .......... .......... 25%  120M 2s\n",
      " 11950K .......... .......... .......... .......... .......... 25%  103M 2s\n",
      " 12000K .......... .......... .......... .......... .......... 26%  120M 2s\n",
      " 12050K .......... .......... .......... .......... .......... 26%  117M 2s\n",
      " 12100K .......... .......... .......... .......... .......... 26%  102M 2s\n",
      " 12150K .......... .......... .......... .......... .......... 26%  121M 2s\n",
      " 12200K .......... .......... .......... .......... .......... 26%  102M 2s\n",
      " 12250K .......... .......... .......... .......... .......... 26%  118M 2s\n",
      " 12300K .......... .......... .......... .......... .......... 26%  121M 2s\n",
      " 12350K .......... .......... .......... .......... .......... 26%  102M 2s\n",
      " 12400K .......... .......... .......... .......... .......... 26%  118M 2s\n",
      " 12450K .......... .......... .......... .......... .......... 26%  102M 2s\n",
      " 12500K .......... .......... .......... .......... .......... 27%  121M 2s\n",
      " 12550K .......... .......... .......... .......... .......... 27%  121M 2s\n",
      " 12600K .......... .......... .......... .......... .......... 27%  102M 2s\n",
      " 12650K .......... .......... .......... .......... .......... 27%  118M 2s\n",
      " 12700K .......... .......... .......... .......... .......... 27% 6,92M 2s\n",
      " 12750K .......... .......... .......... .......... .......... 27% 7,46M 2s\n",
      " 12800K .......... .......... .......... .......... .......... 27%  122M 2s\n",
      " 12850K .......... .......... .......... .......... .......... 27% 51,3M 2s\n",
      " 12900K .......... .......... .......... .......... .......... 27% 14,5M 2s\n",
      " 12950K .......... .......... .......... .......... .......... 28% 3,88M 2s\n",
      " 13000K .......... .......... .......... .......... .......... 28% 27,5M 2s\n",
      " 13050K .......... .......... .......... .......... .......... 28% 89,4M 2s\n",
      " 13100K .......... .......... .......... .......... .......... 28% 74,4M 2s\n",
      " 13150K .......... .......... .......... .......... .......... 28% 80,3M 2s\n",
      " 13200K .......... .......... .......... .......... .......... 28% 52,5M 2s\n",
      " 13250K .......... .......... .......... .......... .......... 28%  119M 2s\n",
      " 13300K .......... .......... .......... .......... .......... 28% 56,5M 2s\n",
      " 13350K .......... .......... .......... .......... .......... 28%  102M 2s\n",
      " 13400K .......... .......... .......... .......... .......... 29%  116M 2s\n",
      " 13450K .......... .......... .......... .......... .......... 29%  102M 2s\n",
      " 13500K .......... .......... .......... .......... .......... 29%  120M 2s\n",
      " 13550K .......... .......... .......... .......... .......... 29%  124M 2s\n",
      " 13600K .......... .......... .......... .......... .......... 29%  102M 1s\n",
      " 13650K .......... .......... .......... .......... .......... 29%  118M 1s\n",
      " 13700K .......... .......... .......... .......... .......... 29%  101M 1s\n",
      " 13750K .......... .......... .......... .......... .......... 29%  124M 1s\n",
      " 13800K .......... .......... .......... .......... .......... 29%  115M 1s\n",
      " 13850K .......... .......... .......... .......... .......... 30%  104M 1s\n",
      " 13900K .......... .......... .......... .......... .......... 30%  121M 1s\n",
      " 13950K .......... .......... .......... .......... .......... 30%  103M 1s\n",
      " 14000K .......... .......... .......... .......... .......... 30%  117M 1s\n",
      " 14050K .......... .......... .......... .......... .......... 30%  121M 1s\n",
      " 14100K .......... .......... .......... .......... .......... 30%  102M 1s\n",
      " 14150K .......... .......... .......... .......... .......... 30%  121M 1s\n",
      " 14200K .......... .......... .......... .......... .......... 30%  103M 1s\n",
      " 14250K .......... .......... .......... .......... .......... 30%  117M 1s\n",
      " 14300K .......... .......... .......... .......... .......... 30%  120M 1s\n",
      " 14350K .......... .......... .......... .......... .......... 31%  103M 1s\n",
      " 14400K .......... .......... .......... .......... .......... 31%  118M 1s\n",
      " 14450K .......... .......... .......... .......... .......... 31%  102M 1s\n",
      " 14500K .......... .......... .......... .......... .......... 31%  121M 1s\n",
      " 14550K .......... .......... .......... .......... .......... 31%  118M 1s\n",
      " 14600K .......... .......... .......... .......... .......... 31%  102M 1s\n",
      " 14650K .......... .......... .......... .......... .......... 31%  122M 1s\n",
      " 14700K .......... .......... .......... .......... .......... 31%  102M 1s\n",
      " 14750K .......... .......... .......... .......... .......... 31%  122M 1s\n",
      " 14800K .......... .......... .......... .......... .......... 32%  117M 1s\n",
      " 14850K .......... .......... .......... .......... .......... 32%  102M 1s\n",
      " 14900K .......... .......... .......... .......... .......... 32%  122M 1s\n",
      " 14950K .......... .......... .......... .......... .......... 32%  102M 1s\n",
      " 15000K .......... .......... .......... .......... .......... 32%  117M 1s\n",
      " 15050K .......... .......... .......... .......... .......... 32%  121M 1s\n",
      " 15100K .......... .......... .......... .......... .......... 32%  102M 1s\n",
      " 15150K .......... .......... .......... .......... .......... 32%  121M 1s\n",
      " 15200K .......... .......... .......... .......... .......... 32%  102M 1s\n",
      " 15250K .......... .......... .......... .......... .......... 33%  117M 1s\n",
      " 15300K .......... .......... .......... .......... .......... 33%  121M 1s\n",
      " 15350K .......... .......... .......... .......... .......... 33%  103M 1s\n",
      " 15400K .......... .......... .......... .......... .......... 33%  117M 1s\n",
      " 15450K .......... .......... .......... .......... .......... 33%  102M 1s\n",
      " 15500K .......... .......... .......... .......... .......... 33%  121M 1s\n",
      " 15550K .......... .......... .......... .......... .......... 33%  117M 1s\n",
      " 15600K .......... .......... .......... .......... .......... 33%  102M 1s\n",
      " 15650K .......... .......... .......... .......... .......... 33%  121M 1s\n",
      " 15700K .......... .......... .......... .......... .......... 34%  102M 1s\n",
      " 15750K .......... .......... .......... .......... .......... 34% 7,90M 1s\n",
      " 15800K .......... .......... .......... .......... .......... 34% 7,99M 1s\n",
      " 15850K .......... .......... .......... .......... .......... 34% 40,5M 1s\n",
      " 15900K .......... .......... .......... .......... .......... 34%  123M 1s\n",
      " 15950K .......... .......... .......... .......... .......... 34% 11,3M 1s\n",
      " 16000K .......... .......... .......... .......... .......... 34% 4,20M 1s\n",
      " 16050K .......... .......... .......... .......... .......... 34% 17,8M 1s\n",
      " 16100K .......... .......... .......... .......... .......... 34% 56,8M 1s\n",
      " 16150K .......... .......... .......... .......... .......... 34% 79,9M 1s\n",
      " 16200K .......... .......... .......... .......... .......... 35% 71,4M 1s\n",
      " 16250K .......... .......... .......... .......... .......... 35%  119M 1s\n",
      " 16300K .......... .......... .......... .......... .......... 35%  138M 1s\n",
      " 16350K .......... .......... .......... .......... .......... 35%  178M 1s\n",
      " 16400K .......... .......... .......... .......... .......... 35% 87,3M 1s\n",
      " 16450K .......... .......... .......... .......... .......... 35% 98,8M 1s\n",
      " 16500K .......... .......... .......... .......... .......... 35% 75,3M 1s\n",
      " 16550K .......... .......... .......... .......... .......... 35%  126M 1s\n",
      " 16600K .......... .......... .......... .......... .......... 35%  120M 1s\n",
      " 16650K .......... .......... .......... .......... .......... 36%  148M 1s\n",
      " 16700K .......... .......... .......... .......... .......... 36%  103M 1s\n",
      " 16750K .......... .......... .......... .......... .......... 36%  121M 1s\n",
      " 16800K .......... .......... .......... .......... .......... 36%  117M 1s\n",
      " 16850K .......... .......... .......... .......... .......... 36%  102M 1s\n",
      " 16900K .......... .......... .......... .......... .......... 36%  122M 1s\n",
      " 16950K .......... .......... .......... .......... .......... 36%  102M 1s\n",
      " 17000K .......... .......... .......... .......... .......... 36%  117M 1s\n",
      " 17050K .......... .......... .......... .......... .......... 36%  122M 1s\n",
      " 17100K .......... .......... .......... .......... .......... 37%  102M 1s\n",
      " 17150K .......... .......... .......... .......... .......... 37%  118M 1s\n",
      " 17200K .......... .......... .......... .......... .......... 37%  103M 1s\n",
      " 17250K .......... .......... .......... .......... .......... 37%  119M 1s\n",
      " 17300K .......... .......... .......... .......... .......... 37%  123M 1s\n",
      " 17350K .......... .......... .......... .......... .......... 37%  102M 1s\n",
      " 17400K .......... .......... .......... .......... .......... 37%  117M 1s\n",
      " 17450K .......... .......... .......... .......... .......... 37%  102M 1s\n",
      " 17500K .......... .......... .......... .......... .......... 37%  123M 1s\n",
      " 17550K .......... .......... .......... .......... .......... 38%  117M 1s\n",
      " 17600K .......... .......... .......... .......... .......... 38%  101M 1s\n",
      " 17650K .......... .......... .......... .......... .......... 38%  121M 1s\n",
      " 17700K .......... .......... .......... .......... .......... 38%  103M 1s\n",
      " 17750K .......... .......... .......... .......... .......... 38%  118M 1s\n",
      " 17800K .......... .......... .......... .......... .......... 38%  121M 1s\n",
      " 17850K .......... .......... .......... .......... .......... 38%  101M 1s\n",
      " 17900K .......... .......... .......... .......... .......... 38%  123M 1s\n",
      " 17950K .......... .......... .......... .......... .......... 38%  101M 1s\n",
      " 18000K .......... .......... .......... .......... .......... 38%  119M 1s\n",
      " 18050K .......... .......... .......... .......... .......... 39%  121M 1s\n",
      " 18100K .......... .......... .......... .......... .......... 39%  102M 1s\n",
      " 18150K .......... .......... .......... .......... .......... 39%  118M 1s\n",
      " 18200K .......... .......... .......... .......... .......... 39%  102M 1s\n",
      " 18250K .......... .......... .......... .......... .......... 39%  120M 1s\n",
      " 18300K .......... .......... .......... .......... .......... 39%  118M 1s\n",
      " 18350K .......... .......... .......... .......... .......... 39%  102M 1s\n",
      " 18400K .......... .......... .......... .......... .......... 39%  121M 1s\n",
      " 18450K .......... .......... .......... .......... .......... 39%  102M 1s\n",
      " 18500K .......... .......... .......... .......... .......... 40%  120M 1s\n",
      " 18550K .......... .......... .......... .......... .......... 40%  118M 1s\n",
      " 18600K .......... .......... .......... .......... .......... 40%  102M 1s\n",
      " 18650K .......... .......... .......... .......... .......... 40%  121M 1s\n",
      " 18700K .......... .......... .......... .......... .......... 40%  102M 1s\n",
      " 18750K .......... .......... .......... .......... .......... 40%  118M 1s\n",
      " 18800K .......... .......... .......... .......... .......... 40%  121M 1s\n",
      " 18850K .......... .......... .......... .......... .......... 40% 5,74M 1s\n",
      " 18900K .......... .......... .......... .......... .......... 40% 12,0M 1s\n",
      " 18950K .......... .......... .......... .......... .......... 41% 52,1M 1s\n",
      " 19000K .......... .......... .......... .......... .......... 41% 29,9M 1s\n",
      " 19050K .......... .......... .......... .......... .......... 41% 5,61M 1s\n",
      " 19100K .......... .......... .......... .......... .......... 41% 5,79M 1s\n",
      " 19150K .......... .......... .......... .......... .......... 41% 46,0M 1s\n",
      " 19200K .......... .......... .......... .......... .......... 41%  102M 1s\n",
      " 19250K .......... .......... .......... .......... .......... 41%  119M 1s\n",
      " 19300K .......... .......... .......... .......... .......... 41%  122M 1s\n",
      " 19350K .......... .......... .......... .......... .......... 41%  100M 1s\n",
      " 19400K .......... .......... .......... .......... .......... 42%  122M 1s\n",
      " 19450K .......... .......... .......... .......... .......... 42%  103M 1s\n",
      " 19500K .......... .......... .......... .......... .......... 42%  120M 1s\n",
      " 19550K .......... .......... .......... .......... .......... 42%  119M 1s\n",
      " 19600K .......... .......... .......... .......... .......... 42%  102M 1s\n",
      " 19650K .......... .......... .......... .......... .......... 42% 58,4M 1s\n",
      " 19700K .......... .......... .......... .......... .......... 42%  102M 1s\n",
      " 19750K .......... .......... .......... .......... .......... 42%  119M 1s\n",
      " 19800K .......... .......... .......... .......... .......... 42%  119M 1s\n",
      " 19850K .......... .......... .......... .......... .......... 42% 47,5M 1s\n",
      " 19900K .......... .......... .......... .......... .......... 43%  213M 1s\n",
      " 19950K .......... .......... .......... .......... .......... 43%  230M 1s\n",
      " 20000K .......... .......... .......... .......... .......... 43%  165M 1s\n",
      " 20050K .......... .......... .......... .......... .......... 43%  121M 1s\n",
      " 20100K .......... .......... .......... .......... .......... 43%  102M 1s\n",
      " 20150K .......... .......... .......... .......... .......... 43%  117M 1s\n",
      " 20200K .......... .......... .......... .......... .......... 43%  102M 1s\n",
      " 20250K .......... .......... .......... .......... .......... 43%  122M 1s\n",
      " 20300K .......... .......... .......... .......... .......... 43%  117M 1s\n",
      " 20350K .......... .......... .......... .......... .......... 44%  103M 1s\n",
      " 20400K .......... .......... .......... .......... .......... 44%  120M 1s\n",
      " 20450K .......... .......... .......... .......... .......... 44%  103M 1s\n",
      " 20500K .......... .......... .......... .......... .......... 44%  121M 1s\n",
      " 20550K .......... .......... .......... .......... .......... 44%  117M 1s\n",
      " 20600K .......... .......... .......... .......... .......... 44%  103M 1s\n",
      " 20650K .......... .......... .......... .......... .......... 44%  121M 1s\n",
      " 20700K .......... .......... .......... .......... .......... 44%  103M 1s\n",
      " 20750K .......... .......... .......... .......... .......... 44%  117M 1s\n",
      " 20800K .......... .......... .......... .......... .......... 45%  120M 1s\n",
      " 20850K .......... .......... .......... .......... .......... 45%  103M 1s\n",
      " 20900K .......... .......... .......... .......... .......... 45%  117M 1s\n",
      " 20950K .......... .......... .......... .......... .......... 45%  102M 1s\n",
      " 21000K .......... .......... .......... .......... .......... 45%  121M 1s\n",
      " 21050K .......... .......... .......... .......... .......... 45%  121M 1s\n",
      " 21100K .......... .......... .......... .......... .......... 45%  103M 1s\n",
      " 21150K .......... .......... .......... .......... .......... 45%  117M 1s\n",
      " 21200K .......... .......... .......... .......... .......... 45%  102M 1s\n",
      " 21250K .......... .......... .......... .......... .......... 45%  121M 1s\n",
      " 21300K .......... .......... .......... .......... .......... 46%  118M 1s\n",
      " 21350K .......... .......... .......... .......... .......... 46%  102M 1s\n",
      " 21400K .......... .......... .......... .......... .......... 46%  122M 1s\n",
      " 21450K .......... .......... .......... .......... .......... 46%  102M 1s\n",
      " 21500K .......... .......... .......... .......... .......... 46%  117M 1s\n",
      " 21550K .......... .......... .......... .......... .......... 46%  121M 1s\n",
      " 21600K .......... .......... .......... .......... .......... 46%  102M 1s\n",
      " 21650K .......... .......... .......... .......... .......... 46%  121M 1s\n",
      " 21700K .......... .......... .......... .......... .......... 46%  103M 1s\n",
      " 21750K .......... .......... .......... .......... .......... 47%  117M 1s\n",
      " 21800K .......... .......... .......... .......... .......... 47%  121M 1s\n",
      " 21850K .......... .......... .......... .......... .......... 47% 16,8M 1s\n",
      " 21900K .......... .......... .......... .......... .......... 47% 6,24M 1s\n",
      " 21950K .......... .......... .......... .......... .......... 47% 21,1M 1s\n",
      " 22000K .......... .......... .......... .......... .......... 47%  120M 1s\n",
      " 22050K .......... .......... .......... .......... .......... 47% 31,4M 1s\n",
      " 22100K .......... .......... .......... .......... .......... 47% 13,8M 1s\n",
      " 22150K .......... .......... .......... .......... .......... 47% 3,76M 1s\n",
      " 22200K .......... .......... .......... .......... .......... 48% 26,8M 1s\n",
      " 22250K .......... .......... .......... .......... .......... 48% 57,1M 1s\n",
      " 22300K .......... .......... .......... .......... .......... 48% 52,2M 1s\n",
      " 22350K .......... .......... .......... .......... .......... 48%  106M 1s\n",
      " 22400K .......... .......... .......... .......... .......... 48%  128M 1s\n",
      " 22450K .......... .......... .......... .......... .......... 48%  115M 1s\n",
      " 22500K .......... .......... .......... .......... .......... 48%  144M 1s\n",
      " 22550K .......... .......... .......... .......... .......... 48%  142M 1s\n",
      " 22600K .......... .......... .......... .......... .......... 48% 62,2M 1s\n",
      " 22650K .......... .......... .......... .......... .......... 49% 48,5M 1s\n",
      " 22700K .......... .......... .......... .......... .......... 49%  150M 1s\n",
      " 22750K .......... .......... .......... .......... .......... 49%  226M 1s\n",
      " 22800K .......... .......... .......... .......... .......... 49%  249M 1s\n",
      " 22850K .......... .......... .......... .......... .......... 49%  209M 1s\n",
      " 22900K .......... .......... .......... .......... .......... 49%  162M 1s\n",
      " 22950K .......... .......... .......... .......... .......... 49%  103M 1s\n",
      " 23000K .......... .......... .......... .......... .......... 49%  121M 1s\n",
      " 23050K .......... .......... .......... .......... .......... 49%  117M 1s\n",
      " 23100K .......... .......... .......... .......... .......... 49%  104M 1s\n",
      " 23150K .......... .......... .......... .......... .......... 50%  121M 1s\n",
      " 23200K .......... .......... .......... .......... .......... 50%  102M 1s\n",
      " 23250K .......... .......... .......... .......... .......... 50%  121M 1s\n",
      " 23300K .......... .......... .......... .......... .......... 50%  117M 1s\n",
      " 23350K .......... .......... .......... .......... .......... 50%  102M 1s\n",
      " 23400K .......... .......... .......... .......... .......... 50%  121M 1s\n",
      " 23450K .......... .......... .......... .......... .......... 50%  103M 1s\n",
      " 23500K .......... .......... .......... .......... .......... 50%  117M 1s\n",
      " 23550K .......... .......... .......... .......... .......... 50%  121M 1s\n",
      " 23600K .......... .......... .......... .......... .......... 51%  102M 1s\n",
      " 23650K .......... .......... .......... .......... .......... 51%  118M 1s\n",
      " 23700K .......... .......... .......... .......... .......... 51%  102M 1s\n",
      " 23750K .......... .......... .......... .......... .......... 51%  121M 1s\n",
      " 23800K .......... .......... .......... .......... .......... 51%  120M 1s\n",
      " 23850K .......... .......... .......... .......... .......... 51%  103M 1s\n",
      " 23900K .......... .......... .......... .......... .......... 51%  118M 1s\n",
      " 23950K .......... .......... .......... .......... .......... 51%  102M 1s\n",
      " 24000K .......... .......... .......... .......... .......... 51%  121M 1s\n",
      " 24050K .......... .......... .......... .......... .......... 52%  118M 1s\n",
      " 24100K .......... .......... .......... .......... .......... 52%  102M 1s\n",
      " 24150K .......... .......... .......... .......... .......... 52%  121M 1s\n",
      " 24200K .......... .......... .......... .......... .......... 52%  102M 1s\n",
      " 24250K .......... .......... .......... .......... .......... 52%  121M 1s\n",
      " 24300K .......... .......... .......... .......... .......... 52%  117M 1s\n",
      " 24350K .......... .......... .......... .......... .......... 52%  102M 1s\n",
      " 24400K .......... .......... .......... .......... .......... 52%  121M 1s\n",
      " 24450K .......... .......... .......... .......... .......... 52%  102M 1s\n",
      " 24500K .......... .......... .......... .......... .......... 53%  117M 1s\n",
      " 24550K .......... .......... .......... .......... .......... 53%  120M 1s\n",
      " 24600K .......... .......... .......... .......... .......... 53%  103M 1s\n",
      " 24650K .......... .......... .......... .......... .......... 53%  118M 1s\n",
      " 24700K .......... .......... .......... .......... .......... 53%  102M 1s\n",
      " 24750K .......... .......... .......... .......... .......... 53%  121M 1s\n",
      " 24800K .......... .......... .......... .......... .......... 53%  121M 1s\n",
      " 24850K .......... .......... .......... .......... .......... 53%  103M 1s\n",
      " 24900K .......... .......... .......... .......... .......... 53% 32,1M 1s\n",
      " 24950K .......... .......... .......... .......... .......... 53% 8,24M 1s\n",
      " 25000K .......... .......... .......... .......... .......... 54% 9,48M 1s\n",
      " 25050K .......... .......... .......... .......... .......... 54% 57,2M 1s\n",
      " 25100K .......... .......... .......... .......... .......... 54% 43,5M 1s\n",
      " 25150K .......... .......... .......... .......... .......... 54% 16,8M 1s\n",
      " 25200K .......... .......... .......... .......... .......... 54% 3,68M 1s\n",
      " 25250K .......... .......... .......... .......... .......... 54% 18,4M 1s\n",
      " 25300K .......... .......... .......... .......... .......... 54%  121M 1s\n",
      " 25350K .......... .......... .......... .......... .......... 54% 46,1M 1s\n",
      " 25400K .......... .......... .......... .......... .......... 54%  123M 1s\n",
      " 25450K .......... .......... .......... .......... .......... 55% 87,1M 1s\n",
      " 25500K .......... .......... .......... .......... .......... 55%  117M 1s\n",
      " 25550K .......... .......... .......... .......... .......... 55%  121M 1s\n",
      " 25600K .......... .......... .......... .......... .......... 55% 92,7M 1s\n",
      " 25650K .......... .......... .......... .......... .......... 55%  117M 1s\n",
      " 25700K .......... .......... .......... .......... .......... 55%  101M 1s\n",
      " 25750K .......... .......... .......... .......... .......... 55%  123M 1s\n",
      " 25800K .......... .......... .......... .......... .......... 55%  116M 1s\n",
      " 25850K .......... .......... .......... .......... .......... 55%  103M 1s\n",
      " 25900K .......... .......... .......... .......... .......... 56%  121M 1s\n",
      " 25950K .......... .......... .......... .......... .......... 56%  102M 1s\n",
      " 26000K .......... .......... .......... .......... .......... 56%  121M 1s\n",
      " 26050K .......... .......... .......... .......... .......... 56%  117M 1s\n",
      " 26100K .......... .......... .......... .......... .......... 56%  103M 1s\n",
      " 26150K .......... .......... .......... .......... .......... 56%  121M 1s\n",
      " 26200K .......... .......... .......... .......... .......... 56%  102M 1s\n",
      " 26250K .......... .......... .......... .......... .......... 56%  117M 1s\n",
      " 26300K .......... .......... .......... .......... .......... 56%  121M 1s\n",
      " 26350K .......... .......... .......... .......... .......... 57%  102M 1s\n",
      " 26400K .......... .......... .......... .......... .......... 57%  121M 1s\n",
      " 26450K .......... .......... .......... .......... .......... 57%  102M 1s\n",
      " 26500K .......... .......... .......... .......... .......... 57%  117M 1s\n",
      " 26550K .......... .......... .......... .......... .......... 57%  121M 1s\n",
      " 26600K .......... .......... .......... .......... .......... 57%  102M 1s\n",
      " 26650K .......... .......... .......... .......... .......... 57%  117M 1s\n",
      " 26700K .......... .......... .......... .......... .......... 57%  102M 1s\n",
      " 26750K .......... .......... .......... .......... .......... 57%  122M 1s\n",
      " 26800K .......... .......... .......... .......... .......... 57%  117M 1s\n",
      " 26850K .......... .......... .......... .......... .......... 58%  102M 1s\n",
      " 26900K .......... .......... .......... .......... .......... 58%  121M 1s\n",
      " 26950K .......... .......... .......... .......... .......... 58%  102M 1s\n",
      " 27000K .......... .......... .......... .......... .......... 58%  121M 1s\n",
      " 27050K .......... .......... .......... .......... .......... 58%  117M 1s\n",
      " 27100K .......... .......... .......... .......... .......... 58%  102M 1s\n",
      " 27150K .......... .......... .......... .......... .......... 58%  121M 1s\n",
      " 27200K .......... .......... .......... .......... .......... 58%  102M 1s\n",
      " 27250K .......... .......... .......... .......... .......... 58%  117M 1s\n",
      " 27300K .......... .......... .......... .......... .......... 59%  121M 1s\n",
      " 27350K .......... .......... .......... .......... .......... 59%  102M 1s\n",
      " 27400K .......... .......... .......... .......... .......... 59%  117M 1s\n",
      " 27450K .......... .......... .......... .......... .......... 59%  102M 1s\n",
      " 27500K .......... .......... .......... .......... .......... 59%  121M 1s\n",
      " 27550K .......... .......... .......... .......... .......... 59%  121M 1s\n",
      " 27600K .......... .......... .......... .......... .......... 59%  102M 1s\n",
      " 27650K .......... .......... .......... .......... .......... 59%  118M 1s\n",
      " 27700K .......... .......... .......... .......... .......... 59%  102M 1s\n",
      " 27750K .......... .......... .......... .......... .......... 60%  121M 1s\n",
      " 27800K .......... .......... .......... .......... .......... 60%  117M 1s\n",
      " 27850K .......... .......... .......... .......... .......... 60%  102M 1s\n",
      " 27900K .......... .......... .......... .......... .......... 60%  121M 1s\n",
      " 27950K .......... .......... .......... .......... .......... 60%  102M 1s\n",
      " 28000K .......... .......... .......... .......... .......... 60% 11,5M 1s\n",
      " 28050K .......... .......... .......... .......... .......... 60% 6,31M 1s\n",
      " 28100K .......... .......... .......... .......... .......... 60% 51,0M 1s\n",
      " 28150K .......... .......... .......... .......... .......... 60% 17,5M 1s\n",
      " 28200K .......... .......... .......... .......... .......... 61%  103M 1s\n",
      " 28250K .......... .......... .......... .......... .......... 61% 4,63M 1s\n",
      " 28300K .......... .......... .......... .......... .......... 61% 9,34M 1s\n",
      " 28350K .......... .......... .......... .......... .......... 61% 25,6M 1s\n",
      " 28400K .......... .......... .......... .......... .......... 61% 57,7M 1s\n",
      " 28450K .......... .......... .......... .......... .......... 61%  101M 1s\n",
      " 28500K .......... .......... .......... .......... .......... 61%  122M 1s\n",
      " 28550K .......... .......... .......... .......... .......... 61%  121M 1s\n",
      " 28600K .......... .......... .......... .......... .......... 61%  102M 1s\n",
      " 28650K .......... .......... .......... .......... .......... 61%  118M 1s\n",
      " 28700K .......... .......... .......... .......... .......... 62%  102M 1s\n",
      " 28750K .......... .......... .......... .......... .......... 62% 86,1M 1s\n",
      " 28800K .......... .......... .......... .......... .......... 62%  154M 1s\n",
      " 28850K .......... .......... .......... .......... .......... 62%  118M 1s\n",
      " 28900K .......... .......... .......... .......... .......... 62%  123M 1s\n",
      " 28950K .......... .......... .......... .......... .......... 62%  103M 1s\n",
      " 29000K .......... .......... .......... .......... .......... 62%  117M 1s\n",
      " 29050K .......... .......... .......... .......... .......... 62%  121M 1s\n",
      " 29100K .......... .......... .......... .......... .......... 62%  103M 1s\n",
      " 29150K .......... .......... .......... .......... .......... 63%  121M 1s\n",
      " 29200K .......... .......... .......... .......... .......... 63%  104M 1s\n",
      " 29250K .......... .......... .......... .......... .......... 63%  114M 1s\n",
      " 29300K .......... .......... .......... .......... .......... 63%  121M 1s\n",
      " 29350K .......... .......... .......... .......... .......... 63%  105M 1s\n",
      " 29400K .......... .......... .......... .......... .......... 63%  116M 1s\n",
      " 29450K .......... .......... .......... .......... .......... 63%  102M 1s\n",
      " 29500K .......... .......... .......... .......... .......... 63%  122M 1s\n",
      " 29550K .......... .......... .......... .......... .......... 63%  116M 1s\n",
      " 29600K .......... .......... .......... .......... .......... 64%  105M 1s\n",
      " 29650K .......... .......... .......... .......... .......... 64%  121M 1s\n",
      " 29700K .......... .......... .......... .......... .......... 64%  102M 1s\n",
      " 29750K .......... .......... .......... .......... .......... 64%  122M 1s\n",
      " 29800K .......... .......... .......... .......... .......... 64%  116M 1s\n",
      " 29850K .......... .......... .......... .......... .......... 64%  102M 1s\n",
      " 29900K .......... .......... .......... .......... .......... 64%  121M 1s\n",
      " 29950K .......... .......... .......... .......... .......... 64%  102M 1s\n",
      " 30000K .......... .......... .......... .......... .......... 64%  117M 0s\n",
      " 30050K .......... .......... .......... .......... .......... 65%  121M 0s\n",
      " 30100K .......... .......... .......... .......... .......... 65%  102M 0s\n",
      " 30150K .......... .......... .......... .......... .......... 65%  121M 0s\n",
      " 30200K .......... .......... .......... .......... .......... 65%  102M 0s\n",
      " 30250K .......... .......... .......... .......... .......... 65%  117M 0s\n",
      " 30300K .......... .......... .......... .......... .......... 65%  121M 0s\n",
      " 30350K .......... .......... .......... .......... .......... 65%  102M 0s\n",
      " 30400K .......... .......... .......... .......... .......... 65%  117M 0s\n",
      " 30450K .......... .......... .......... .......... .......... 65%  103M 0s\n",
      " 30500K .......... .......... .......... .......... .......... 65%  120M 0s\n",
      " 30550K .......... .......... .......... .......... .......... 66%  117M 0s\n",
      " 30600K .......... .......... .......... .......... .......... 66%  103M 0s\n",
      " 30650K .......... .......... .......... .......... .......... 66%  121M 0s\n",
      " 30700K .......... .......... .......... .......... .......... 66%  103M 0s\n",
      " 30750K .......... .......... .......... .......... .......... 66%  121M 0s\n",
      " 30800K .......... .......... .......... .......... .......... 66%  118M 0s\n",
      " 30850K .......... .......... .......... .......... .......... 66%  102M 0s\n",
      " 30900K .......... .......... .......... .......... .......... 66%  121M 0s\n",
      " 30950K .......... .......... .......... .......... .......... 66%  102M 0s\n",
      " 31000K .......... .......... .......... .......... .......... 67% 46,9M 0s\n",
      " 31050K .......... .......... .......... .......... .......... 67% 20,3M 0s\n",
      " 31100K .......... .......... .......... .......... .......... 67% 6,95M 0s\n",
      " 31150K .......... .......... .......... .......... .......... 67% 24,3M 0s\n",
      " 31200K .......... .......... .......... .......... .......... 67% 48,3M 0s\n",
      " 31250K .......... .......... .......... .......... .......... 67% 18,0M 0s\n",
      " 31300K .......... .......... .......... .......... .......... 67% 8,49M 0s\n",
      " 31350K .......... .......... .......... .......... .......... 67% 5,33M 0s\n",
      " 31400K .......... .......... .......... .......... .......... 67% 24,3M 0s\n",
      " 31450K .......... .......... .......... .......... .......... 68% 43,8M 0s\n",
      " 31500K .......... .......... .......... .......... .......... 68% 48,2M 0s\n",
      " 31550K .......... .......... .......... .......... .......... 68%  128M 0s\n",
      " 31600K .......... .......... .......... .......... .......... 68% 67,6M 0s\n",
      " 31650K .......... .......... .......... .......... .......... 68%  121M 0s\n",
      " 31700K .......... .......... .......... .......... .......... 68%  102M 0s\n",
      " 31750K .......... .......... .......... .......... .......... 68%  121M 0s\n",
      " 31800K .......... .......... .......... .......... .......... 68% 99,8M 0s\n",
      " 31850K .......... .......... .......... .......... .......... 68% 86,9M 0s\n",
      " 31900K .......... .......... .......... .......... .......... 68%  106M 0s\n",
      " 31950K .......... .......... .......... .......... .......... 69%  104M 0s\n",
      " 32000K .......... .......... .......... .......... .......... 69%  117M 0s\n",
      " 32050K .......... .......... .......... .......... .......... 69%  122M 0s\n",
      " 32100K .......... .......... .......... .......... .......... 69%  103M 0s\n",
      " 32150K .......... .......... .......... .......... .......... 69%  117M 0s\n",
      " 32200K .......... .......... .......... .......... .......... 69%  102M 0s\n",
      " 32250K .......... .......... .......... .......... .......... 69%  121M 0s\n",
      " 32300K .......... .......... .......... .......... .......... 69%  121M 0s\n",
      " 32350K .......... .......... .......... .......... .......... 69%  103M 0s\n",
      " 32400K .......... .......... .......... .......... .......... 70%  118M 0s\n",
      " 32450K .......... .......... .......... .......... .......... 70%  102M 0s\n",
      " 32500K .......... .......... .......... .......... .......... 70%  121M 0s\n",
      " 32550K .......... .......... .......... .......... .......... 70%  117M 0s\n",
      " 32600K .......... .......... .......... .......... .......... 70%  101M 0s\n",
      " 32650K .......... .......... .......... .......... .......... 70%  122M 0s\n",
      " 32700K .......... .......... .......... .......... .......... 70%  102M 0s\n",
      " 32750K .......... .......... .......... .......... .......... 70%  117M 0s\n",
      " 32800K .......... .......... .......... .......... .......... 70%  121M 0s\n",
      " 32850K .......... .......... .......... .......... .......... 71% 96,5M 0s\n",
      " 32900K .......... .......... .......... .......... .......... 71%  130M 0s\n",
      " 32950K .......... .......... .......... .......... .......... 71%  103M 0s\n",
      " 33000K .......... .......... .......... .......... .......... 71%  117M 0s\n",
      " 33050K .......... .......... .......... .......... .......... 71%  121M 0s\n",
      " 33100K .......... .......... .......... .......... .......... 71%  102M 0s\n",
      " 33150K .......... .......... .......... .......... .......... 71%  117M 0s\n",
      " 33200K .......... .......... .......... .......... .......... 71%  102M 0s\n",
      " 33250K .......... .......... .......... .......... .......... 71%  121M 0s\n",
      " 33300K .......... .......... .......... .......... .......... 72%  118M 0s\n",
      " 33350K .......... .......... .......... .......... .......... 72%  102M 0s\n",
      " 33400K .......... .......... .......... .......... .......... 72%  121M 0s\n",
      " 33450K .......... .......... .......... .......... .......... 72%  102M 0s\n",
      " 33500K .......... .......... .......... .......... .......... 72%  121M 0s\n",
      " 33550K .......... .......... .......... .......... .......... 72%  117M 0s\n",
      " 33600K .......... .......... .......... .......... .......... 72%  102M 0s\n",
      " 33650K .......... .......... .......... .......... .......... 72%  121M 0s\n",
      " 33700K .......... .......... .......... .......... .......... 72%  102M 0s\n",
      " 33750K .......... .......... .......... .......... .......... 72%  117M 0s\n",
      " 33800K .......... .......... .......... .......... .......... 73%  121M 0s\n",
      " 33850K .......... .......... .......... .......... .......... 73%  102M 0s\n",
      " 33900K .......... .......... .......... .......... .......... 73%  121M 0s\n",
      " 33950K .......... .......... .......... .......... .......... 73%  102M 0s\n",
      " 34000K .......... .......... .......... .......... .......... 73%  117M 0s\n",
      " 34050K .......... .......... .......... .......... .......... 73%  121M 0s\n",
      " 34100K .......... .......... .......... .......... .......... 73% 41,3M 0s\n",
      " 34150K .......... .......... .......... .......... .......... 73% 8,19M 0s\n",
      " 34200K .......... .......... .......... .......... .......... 73% 12,9M 0s\n",
      " 34250K .......... .......... .......... .......... .......... 74% 40,0M 0s\n",
      " 34300K .......... .......... .......... .......... .......... 74% 56,5M 0s\n",
      " 34350K .......... .......... .......... .......... .......... 74% 5,26M 0s\n",
      " 34400K .......... .......... .......... .......... .......... 74% 6,35M 0s\n",
      " 34450K .......... .......... .......... .......... .......... 74% 26,8M 0s\n",
      " 34500K .......... .......... .......... .......... .......... 74% 27,6M 0s\n",
      " 34550K .......... .......... .......... .......... .......... 74%  106M 0s\n",
      " 34600K .......... .......... .......... .......... .......... 74% 66,8M 0s\n",
      " 34650K .......... .......... .......... .......... .......... 74% 62,1M 0s\n",
      " 34700K .......... .......... .......... .......... .......... 75%  107M 0s\n",
      " 34750K .......... .......... .......... .......... .......... 75% 63,4M 0s\n",
      " 34800K .......... .......... .......... .......... .......... 75% 72,0M 0s\n",
      " 34850K .......... .......... .......... .......... .......... 75% 95,2M 0s\n",
      " 34900K .......... .......... .......... .......... .......... 75% 75,6M 0s\n",
      " 34950K .......... .......... .......... .......... .......... 75% 89,0M 0s\n",
      " 35000K .......... .......... .......... .......... .......... 75%  249M 0s\n",
      " 35050K .......... .......... .......... .......... .......... 75%  267M 0s\n",
      " 35100K .......... .......... .......... .......... .......... 75%  229M 0s\n",
      " 35150K .......... .......... .......... .......... .......... 76%  328M 0s\n",
      " 35200K .......... .......... .......... .......... .......... 76%  105M 0s\n",
      " 35250K .......... .......... .......... .......... .......... 76%  107M 0s\n",
      " 35300K .......... .......... .......... .......... .......... 76%  117M 0s\n",
      " 35350K .......... .......... .......... .......... .......... 76%  102M 0s\n",
      " 35400K .......... .......... .......... .......... .......... 76%  121M 0s\n",
      " 35450K .......... .......... .......... .......... .......... 76%  102M 0s\n",
      " 35500K .......... .......... .......... .......... .......... 76%  121M 0s\n",
      " 35550K .......... .......... .......... .......... .......... 76%  118M 0s\n",
      " 35600K .......... .......... .......... .......... .......... 76%  103M 0s\n",
      " 35650K .......... .......... .......... .......... .......... 77%  120M 0s\n",
      " 35700K .......... .......... .......... .......... .......... 77%  102M 0s\n",
      " 35750K .......... .......... .......... .......... .......... 77%  117M 0s\n",
      " 35800K .......... .......... .......... .......... .......... 77%  122M 0s\n",
      " 35850K .......... .......... .......... .......... .......... 77%  102M 0s\n",
      " 35900K .......... .......... .......... .......... .......... 77%  117M 0s\n",
      " 35950K .......... .......... .......... .......... .......... 77%  101M 0s\n",
      " 36000K .......... .......... .......... .......... .......... 77%  121M 0s\n",
      " 36050K .......... .......... .......... .......... .......... 77%  123M 0s\n",
      " 36100K .......... .......... .......... .......... .......... 78%  103M 0s\n",
      " 36150K .......... .......... .......... .......... .......... 78%  117M 0s\n",
      " 36200K .......... .......... .......... .......... .......... 78%  102M 0s\n",
      " 36250K .......... .......... .......... .......... .......... 78%  121M 0s\n",
      " 36300K .......... .......... .......... .......... .......... 78%  117M 0s\n",
      " 36350K .......... .......... .......... .......... .......... 78%  102M 0s\n",
      " 36400K .......... .......... .......... .......... .......... 78%  121M 0s\n",
      " 36450K .......... .......... .......... .......... .......... 78%  102M 0s\n",
      " 36500K .......... .......... .......... .......... .......... 78%  117M 0s\n",
      " 36550K .......... .......... .......... .......... .......... 79%  121M 0s\n",
      " 36600K .......... .......... .......... .......... .......... 79%  102M 0s\n",
      " 36650K .......... .......... .......... .......... .......... 79%  121M 0s\n",
      " 36700K .......... .......... .......... .......... .......... 79%  102M 0s\n",
      " 36750K .......... .......... .......... .......... .......... 79%  117M 0s\n",
      " 36800K .......... .......... .......... .......... .......... 79%  121M 0s\n",
      " 36850K .......... .......... .......... .......... .......... 79%  102M 0s\n",
      " 36900K .......... .......... .......... .......... .......... 79%  117M 0s\n",
      " 36950K .......... .......... .......... .......... .......... 79%  102M 0s\n",
      " 37000K .......... .......... .......... .......... .......... 80%  121M 0s\n",
      " 37050K .......... .......... .......... .......... .......... 80%  117M 0s\n",
      " 37100K .......... .......... .......... .......... .......... 80%  102M 0s\n",
      " 37150K .......... .......... .......... .......... .......... 80% 49,6M 0s\n",
      " 37200K .......... .......... .......... .......... .......... 80% 14,2M 0s\n",
      " 37250K .......... .......... .......... .......... .......... 80% 7,25M 0s\n",
      " 37300K .......... .......... .......... .......... .......... 80% 43,0M 0s\n",
      " 37350K .......... .......... .......... .......... .......... 80%  103M 0s\n",
      " 37400K .......... .......... .......... .......... .......... 80% 6,27M 0s\n",
      " 37450K .......... .......... .......... .......... .......... 80% 6,55M 0s\n",
      " 37500K .......... .......... .......... .......... .......... 81% 11,6M 0s\n",
      " 37550K .......... .......... .......... .......... .......... 81%  111M 0s\n",
      " 37600K .......... .......... .......... .......... .......... 81% 51,6M 0s\n",
      " 37650K .......... .......... .......... .......... .......... 81% 51,5M 0s\n",
      " 37700K .......... .......... .......... .......... .......... 81% 58,0M 0s\n",
      " 37750K .......... .......... .......... .......... .......... 81% 67,9M 0s\n",
      " 37800K .......... .......... .......... .......... .......... 81%  115M 0s\n",
      " 37850K .......... .......... .......... .......... .......... 81%  105M 0s\n",
      " 37900K .......... .......... .......... .......... .......... 81%  118M 0s\n",
      " 37950K .......... .......... .......... .......... .......... 82%  101M 0s\n",
      " 38000K .......... .......... .......... .......... .......... 82%  125M 0s\n",
      " 38050K .......... .......... .......... .......... .......... 82%  117M 0s\n",
      " 38100K .......... .......... .......... .......... .......... 82%  101M 0s\n",
      " 38150K .......... .......... .......... .......... .......... 82%  123M 0s\n",
      " 38200K .......... .......... .......... .......... .......... 82% 93,2M 0s\n",
      " 38250K .......... .......... .......... .......... .......... 82%  137M 0s\n",
      " 38300K .......... .......... .......... .......... .......... 82%  117M 0s\n",
      " 38350K .......... .......... .......... .......... .......... 82%  103M 0s\n",
      " 38400K .......... .......... .......... .......... .......... 83%  120M 0s\n",
      " 38450K .......... .......... .......... .......... .......... 83%  103M 0s\n",
      " 38500K .......... .......... .......... .......... .......... 83%  117M 0s\n",
      " 38550K .......... .......... .......... .......... .......... 83%  120M 0s\n",
      " 38600K .......... .......... .......... .......... .......... 83%  103M 0s\n",
      " 38650K .......... .......... .......... .......... .......... 83%  117M 0s\n",
      " 38700K .......... .......... .......... .......... .......... 83%  102M 0s\n",
      " 38750K .......... .......... .......... .......... .......... 83%  121M 0s\n",
      " 38800K .......... .......... .......... .......... .......... 83%  121M 0s\n",
      " 38850K .......... .......... .......... .......... .......... 84%  102M 0s\n",
      " 38900K .......... .......... .......... .......... .......... 84%  117M 0s\n",
      " 38950K .......... .......... .......... .......... .......... 84%  102M 0s\n",
      " 39000K .......... .......... .......... .......... .......... 84%  121M 0s\n",
      " 39050K .......... .......... .......... .......... .......... 84%  117M 0s\n",
      " 39100K .......... .......... .......... .......... .......... 84%  102M 0s\n",
      " 39150K .......... .......... .......... .......... .......... 84%  121M 0s\n",
      " 39200K .......... .......... .......... .......... .......... 84%  102M 0s\n",
      " 39250K .......... .......... .......... .......... .......... 84%  121M 0s\n",
      " 39300K .......... .......... .......... .......... .......... 84%  117M 0s\n",
      " 39350K .......... .......... .......... .......... .......... 85%  102M 0s\n",
      " 39400K .......... .......... .......... .......... .......... 85%  121M 0s\n",
      " 39450K .......... .......... .......... .......... .......... 85%  102M 0s\n",
      " 39500K .......... .......... .......... .......... .......... 85%  117M 0s\n",
      " 39550K .......... .......... .......... .......... .......... 85%  121M 0s\n",
      " 39600K .......... .......... .......... .......... .......... 85%  102M 0s\n",
      " 39650K .......... .......... .......... .......... .......... 85%  118M 0s\n",
      " 39700K .......... .......... .......... .......... .......... 85%  102M 0s\n",
      " 39750K .......... .......... .......... .......... .......... 85%  121M 0s\n",
      " 39800K .......... .......... .......... .......... .......... 86%  120M 0s\n",
      " 39850K .......... .......... .......... .......... .......... 86%  102M 0s\n",
      " 39900K .......... .......... .......... .......... .......... 86%  117M 0s\n",
      " 39950K .......... .......... .......... .......... .......... 86%  102M 0s\n",
      " 40000K .......... .......... .......... .......... .......... 86%  121M 0s\n",
      " 40050K .......... .......... .......... .......... .......... 86%  117M 0s\n",
      " 40100K .......... .......... .......... .......... .......... 86%  103M 0s\n",
      " 40150K .......... .......... .......... .......... .......... 86%  120M 0s\n",
      " 40200K .......... .......... .......... .......... .......... 86%  102M 0s\n",
      " 40250K .......... .......... .......... .......... .......... 87% 15,7M 0s\n",
      " 40300K .......... .......... .......... .......... .......... 87% 7,00M 0s\n",
      " 40350K .......... .......... .......... .......... .......... 87% 34,6M 0s\n",
      " 40400K .......... .......... .......... .......... .......... 87% 16,4M 0s\n",
      " 40450K .......... .......... .......... .......... .......... 87% 10,5M 0s\n",
      " 40500K .......... .......... .......... .......... .......... 87% 9,66M 0s\n",
      " 40550K .......... .......... .......... .......... .......... 87% 7,91M 0s\n",
      " 40600K .......... .......... .......... .......... .......... 87% 24,0M 0s\n",
      " 40650K .......... .......... .......... .......... .......... 87% 53,6M 0s\n",
      " 40700K .......... .......... .......... .......... .......... 88% 83,9M 0s\n",
      " 40750K .......... .......... .......... .......... .......... 88% 57,1M 0s\n",
      " 40800K .......... .......... .......... .......... .......... 88% 89,9M 0s\n",
      " 40850K .......... .......... .......... .......... .......... 88% 60,7M 0s\n",
      " 40900K .......... .......... .......... .......... .......... 88%  117M 0s\n",
      " 40950K .......... .......... .......... .......... .......... 88%  101M 0s\n",
      " 41000K .......... .......... .......... .......... .......... 88%  121M 0s\n",
      " 41050K .......... .......... .......... .......... .......... 88% 77,7M 0s\n",
      " 41100K .......... .......... .......... .......... .......... 88%  114M 0s\n",
      " 41150K .......... .......... .......... .......... .......... 88%  133M 0s\n",
      " 41200K .......... .......... .......... .......... .......... 89%  115M 0s\n",
      " 41250K .......... .......... .......... .......... .......... 89%  137M 0s\n",
      " 41300K .......... .......... .......... .......... .......... 89%  134M 0s\n",
      " 41350K .......... .......... .......... .......... .......... 89%  101M 0s\n",
      " 41400K .......... .......... .......... .......... .......... 89%  123M 0s\n",
      " 41450K .......... .......... .......... .......... .......... 89%  101M 0s\n",
      " 41500K .......... .......... .......... .......... .......... 89%  120M 0s\n",
      " 41550K .......... .......... .......... .......... .......... 89%  118M 0s\n",
      " 41600K .......... .......... .......... .......... .......... 89%  105M 0s\n",
      " 41650K .......... .......... .......... .......... .......... 90%  117M 0s\n",
      " 41700K .......... .......... .......... .......... .......... 90%  102M 0s\n",
      " 41750K .......... .......... .......... .......... .......... 90%  122M 0s\n",
      " 41800K .......... .......... .......... .......... .......... 90%  117M 0s\n",
      " 41850K .......... .......... .......... .......... .......... 90%  102M 0s\n",
      " 41900K .......... .......... .......... .......... .......... 90%  121M 0s\n",
      " 41950K .......... .......... .......... .......... .......... 90%  103M 0s\n",
      " 42000K .......... .......... .......... .......... .......... 90%  122M 0s\n",
      " 42050K .......... .......... .......... .......... .......... 90%  117M 0s\n",
      " 42100K .......... .......... .......... .......... .......... 91%  102M 0s\n",
      " 42150K .......... .......... .......... .......... .......... 91%  121M 0s\n",
      " 42200K .......... .......... .......... .......... .......... 91%  103M 0s\n",
      " 42250K .......... .......... .......... .......... .......... 91%  117M 0s\n",
      " 42300K .......... .......... .......... .......... .......... 91%  121M 0s\n",
      " 42350K .......... .......... .......... .......... .......... 91%  102M 0s\n",
      " 42400K .......... .......... .......... .......... .......... 91%  117M 0s\n",
      " 42450K .......... .......... .......... .......... .......... 91%  102M 0s\n",
      " 42500K .......... .......... .......... .......... .......... 91%  121M 0s\n",
      " 42550K .......... .......... .......... .......... .......... 91%  121M 0s\n",
      " 42600K .......... .......... .......... .......... .......... 92%  102M 0s\n",
      " 42650K .......... .......... .......... .......... .......... 92%  118M 0s\n",
      " 42700K .......... .......... .......... .......... .......... 92%  102M 0s\n",
      " 42750K .......... .......... .......... .......... .......... 92%  121M 0s\n",
      " 42800K .......... .......... .......... .......... .......... 92%  117M 0s\n",
      " 42850K .......... .......... .......... .......... .......... 92%  102M 0s\n",
      " 42900K .......... .......... .......... .......... .......... 92%  121M 0s\n",
      " 42950K .......... .......... .......... .......... .......... 92%  102M 0s\n",
      " 43000K .......... .......... .......... .......... .......... 92%  121M 0s\n",
      " 43050K .......... .......... .......... .......... .......... 93%  117M 0s\n",
      " 43100K .......... .......... .......... .......... .......... 93%  102M 0s\n",
      " 43150K .......... .......... .......... .......... .......... 93%  121M 0s\n",
      " 43200K .......... .......... .......... .......... .......... 93%  103M 0s\n",
      " 43250K .......... .......... .......... .......... .......... 93%  117M 0s\n",
      " 43300K .......... .......... .......... .......... .......... 93% 64,0M 0s\n",
      " 43350K .......... .......... .......... .......... .......... 93% 5,61M 0s\n",
      " 43400K .......... .......... .......... .......... .......... 93% 23,1M 0s\n",
      " 43450K .......... .......... .......... .......... .......... 93% 34,1M 0s\n",
      " 43500K .......... .......... .......... .......... .......... 94% 12,9M 0s\n",
      " 43550K .......... .......... .......... .......... .......... 94% 5,16M 0s\n",
      " 43600K .......... .......... .......... .......... .......... 94% 14,6M 0s\n",
      " 43650K .......... .......... .......... .......... .......... 94% 25,6M 0s\n",
      " 43700K .......... .......... .......... .......... .......... 94% 32,4M 0s\n",
      " 43750K .......... .......... .......... .......... .......... 94% 51,8M 0s\n",
      " 43800K .......... .......... .......... .......... .......... 94% 34,9M 0s\n",
      " 43850K .......... .......... .......... .......... .......... 94%  105M 0s\n",
      " 43900K .......... .......... .......... .......... .......... 94%  119M 0s\n",
      " 43950K .......... .......... .......... .......... .......... 95%  104M 0s\n",
      " 44000K .......... .......... .......... .......... .......... 95%  116M 0s\n",
      " 44050K .......... .......... .......... .......... .......... 95%  117M 0s\n",
      " 44100K .......... .......... .......... .......... .......... 95%  106M 0s\n",
      " 44150K .......... .......... .......... .......... .......... 95%  122M 0s\n",
      " 44200K .......... .......... .......... .......... .......... 95%  103M 0s\n",
      " 44250K .......... .......... .......... .......... .......... 95%  118M 0s\n",
      " 44300K .......... .......... .......... .......... .......... 95%  118M 0s\n",
      " 44350K .......... .......... .......... .......... .......... 95%  103M 0s\n",
      " 44400K .......... .......... .......... .......... .......... 95%  119M 0s\n",
      " 44450K .......... .......... .......... .......... .......... 96%  101M 0s\n",
      " 44500K .......... .......... .......... .......... .......... 96%  120M 0s\n",
      " 44550K .......... .......... .......... .......... .......... 96%  119M 0s\n",
      " 44600K .......... .......... .......... .......... .......... 96%  103M 0s\n",
      " 44650K .......... .......... .......... .......... .......... 96%  121M 0s\n",
      " 44700K .......... .......... .......... .......... .......... 96%  102M 0s\n",
      " 44750K .......... .......... .......... .......... .......... 96%  123M 0s\n",
      " 44800K .......... .......... .......... .......... .......... 96%  112M 0s\n",
      " 44850K .......... .......... .......... .......... .......... 96%  105M 0s\n",
      " 44900K .......... .......... .......... .......... .......... 97%  124M 0s\n",
      " 44950K .......... .......... .......... .......... .......... 97%  102M 0s\n",
      " 45000K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 45050K .......... .......... .......... .......... .......... 97%  122M 0s\n",
      " 45100K .......... .......... .......... .......... .......... 97%  102M 0s\n",
      " 45150K .......... .......... .......... .......... .......... 97%  122M 0s\n",
      " 45200K .......... .......... .......... .......... .......... 97%  102M 0s\n",
      " 45250K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 45300K .......... .......... .......... .......... .......... 97%  123M 0s\n",
      " 45350K .......... .......... .......... .......... .......... 98%  101M 0s\n",
      " 45400K .......... .......... .......... .......... .......... 98%  118M 0s\n",
      " 45450K .......... .......... .......... .......... .......... 98%  102M 0s\n",
      " 45500K .......... .......... .......... .......... .......... 98%  121M 0s\n",
      " 45550K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 45600K .......... .......... .......... .......... .......... 98%  103M 0s\n",
      " 45650K .......... .......... .......... .......... .......... 98%  121M 0s\n",
      " 45700K .......... .......... .......... .......... .......... 98%  102M 0s\n",
      " 45750K .......... .......... .......... .......... .......... 98%  121M 0s\n",
      " 45800K .......... .......... .......... .......... .......... 99%  117M 0s\n",
      " 45850K .......... .......... .......... .......... .......... 99%  102M 0s\n",
      " 45900K .......... .......... .......... .......... .......... 99%  121M 0s\n",
      " 45950K .......... .......... .......... .......... .......... 99%  102M 0s\n",
      " 46000K .......... .......... .......... .......... .......... 99%  117M 0s\n",
      " 46050K .......... .......... .......... .......... .......... 99%  121M 0s\n",
      " 46100K .......... .......... .......... .......... .......... 99%  102M 0s\n",
      " 46150K .......... .......... .......... .......... .......... 99%  117M 0s\n",
      " 46200K .......... .......... .......... .......... .......... 99%  102M 0s\n",
      " 46250K .......... .......... .......... .......... .......... 99%  120M 0s\n",
      " 46300K .....                                                 100%  560M=1,2s\n",
      "\n",
      "2023-08-28 22:55:35 (37,2 MB/s) - ‘data/reference/genome.fa.gz’ guardado [47416846/47416846]\n",
      "\n",
      "[Mon Aug 28 22:55:36 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T225533.745482.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:55:37 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR696683_chr7_1.fastq.gz, data/raw/ERR696683_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR696683_chr7_1_fastp.fastq.gz, data/processed/ERR696683_chr7_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR696683_chr7_2_fastp.fastq.gz, data/processed/ERR696683_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1935275\n",
      "total bases: 146263550\n",
      "Q20 bases: 145211647(99.2808%)\n",
      "Q30 bases: 141120433(96.4837%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1935275\n",
      "total bases: 145878500\n",
      "Q20 bases: 144064467(98.7565%)\n",
      "Q30 bases: 138098537(94.6668%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1177764\n",
      "total bases: 89449589\n",
      "Q20 bases: 89095661(99.6043%)\n",
      "Q30 bases: 87513697(97.8358%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1177764\n",
      "total bases: 89414052\n",
      "Q20 bases: 88793329(99.3058%)\n",
      "Q30 bases: 86459019(96.6951%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2355528\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1515022\n",
      "reads with adapter trimmed: 8864\n",
      "bases trimmed due to adapters: 230085\n",
      "\n",
      "Duplication rate: 5.65414%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR696683_chr7_1.fastq.gz -I data/raw/ERR696683_chr7_2.fastq.gz -o data/processed/ERR696683_chr7_1_fastp.fastq.gz -O data/processed/ERR696683_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 7 seconds\n",
      "[Mon Aug 28 22:55:46 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:55:46 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR696683_chr7_1.fastq.gz, data/raw/ERR696683_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR696683_chr7_1_fastqc.html, results/fastqc_result/ERR696683_chr7_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR696683_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 22:56:01 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:56:01 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR696683_chr7_1_fastp.fastq.gz, data/processed/ERR696683_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR696683_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR696683_chr7_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR696683_chr7_2_fastp.fastq.gz, data/processed/ERR696683_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 22:56:11 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T225536.818637.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:56:13 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753368_chr7_1.fastq.gz, data/raw/ERR753368_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753368_chr7_1_fastp.fastq.gz, data/processed/ERR753368_chr7_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753368_chr7_2_fastp.fastq.gz, data/processed/ERR753368_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1806088\n",
      "total bases: 137130775\n",
      "Q20 bases: 136588468(99.6045%)\n",
      "Q30 bases: 132513779(96.6331%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1806088\n",
      "total bases: 136987438\n",
      "Q20 bases: 135772877(99.1134%)\n",
      "Q30 bases: 129457863(94.5035%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1548330\n",
      "total bases: 117666684\n",
      "Q20 bases: 117461408(99.8255%)\n",
      "Q30 bases: 115455876(98.1211%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1548330\n",
      "total bases: 117656644\n",
      "Q20 bases: 117075719(99.5063%)\n",
      "Q30 bases: 113529570(96.4923%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3096660\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 515516\n",
      "reads with adapter trimmed: 5202\n",
      "bases trimmed due to adapters: 140710\n",
      "\n",
      "Duplication rate: 2.03596%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753368_chr7_1.fastq.gz -I data/raw/ERR753368_chr7_2.fastq.gz -o data/processed/ERR753368_chr7_1_fastp.fastq.gz -O data/processed/ERR753368_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Mon Aug 28 22:56:19 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:56:19 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753368_chr7_1_fastp.fastq.gz, data/processed/ERR753368_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753368_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753368_chr7_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753368_chr7_2_fastp.fastq.gz, data/processed/ERR753368_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 22:56:32 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:56:32 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753368_chr7_1.fastq.gz, data/raw/ERR753368_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753368_chr7_1_fastqc.html, results/fastqc_result/ERR753368_chr7_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753368_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753368_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 22:56:47 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T225611.994822.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:56:48 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753369_chr7_1.fastq.gz, data/raw/ERR753369_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753369_chr7_1_fastqc.html, results/fastqc_result/ERR753369_chr7_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753369_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753369_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 22:57:03 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:57:03 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753369_chr7_1.fastq.gz, data/raw/ERR753369_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753369_chr7_1_fastp.fastq.gz, data/processed/ERR753369_chr7_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753369_chr7_1_fastp.fastq.gz, data/processed/ERR753369_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1929269\n",
      "total bases: 146477595\n",
      "Q20 bases: 145872914(99.5872%)\n",
      "Q30 bases: 141418223(96.546%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1929269\n",
      "total bases: 146329545\n",
      "Q20 bases: 145027500(99.1102%)\n",
      "Q30 bases: 138248767(94.4777%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1651945\n",
      "total bases: 125540731\n",
      "Q20 bases: 125307776(99.8144%)\n",
      "Q30 bases: 123093775(98.0509%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1651945\n",
      "total bases: 125530164\n",
      "Q20 bases: 124909386(99.5055%)\n",
      "Q30 bases: 121110875(96.4795%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3303890\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 554648\n",
      "reads with adapter trimmed: 5782\n",
      "bases trimmed due to adapters: 156225\n",
      "\n",
      "Duplication rate: 2.35779%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753369_chr7_1.fastq.gz -I data/raw/ERR753369_chr7_2.fastq.gz -o data/processed/ERR753369_chr7_1_fastp.fastq.gz -O data/processed/ERR753369_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Mon Aug 28 22:57:09 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:57:09 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753369_chr7_1_fastp.fastq.gz, data/processed/ERR753369_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753369_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753369_chr7_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753369_chr7_1_fastp.fastq.gz, data/processed/ERR753369_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 22:57:22 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T225647.337754.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:57:23 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753370_chr7_1.fastq.gz, data/raw/ERR753370_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753370_chr7_1_fastp.fastq.gz, data/processed/ERR753370_chr7_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753370_chr7_2_fastp.fastq.gz, data/processed/ERR753370_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2673460\n",
      "total bases: 269090503\n",
      "Q20 bases: 267232327(99.3095%)\n",
      "Q30 bases: 259317581(96.3682%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2673460\n",
      "total bases: 269019918\n",
      "Q20 bases: 266435611(99.0394%)\n",
      "Q30 bases: 256135213(95.2105%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 2669965\n",
      "total bases: 267638922\n",
      "Q20 bases: 265893698(99.3479%)\n",
      "Q30 bases: 258369403(96.5366%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 2669965\n",
      "total bases: 266755891\n",
      "Q20 bases: 264414765(99.1224%)\n",
      "Q30 bases: 254782415(95.5114%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 5339930\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 6990\n",
      "reads with adapter trimmed: 22426\n",
      "bases trimmed due to adapters: 449745\n",
      "\n",
      "Duplication rate: 4.34572%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 158\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753370_chr7_1.fastq.gz -I data/raw/ERR753370_chr7_2.fastq.gz -o data/processed/ERR753370_chr7_1_fastp.fastq.gz -O data/processed/ERR753370_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 12 seconds\n",
      "[Mon Aug 28 22:57:36 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:57:36 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753370_chr7_1_fastp.fastq.gz, data/processed/ERR753370_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753370_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753370_chr7_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753370_chr7_2_fastp.fastq.gz, data/processed/ERR753370_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 22:57:59 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:57:59 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753370_chr7_1.fastq.gz, data/raw/ERR753370_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753370_chr7_1_fastqc.html, results/fastqc_result/ERR753370_chr7_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753370_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753370_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 22:58:23 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T225722.830199.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:58:24 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753371_chr7_1.fastq.gz, data/raw/ERR753371_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753371_chr7_1_fastqc.html, results/fastqc_result/ERR753371_chr7_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753371_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753371_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 22:58:39 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:58:39 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753371_chr7_1.fastq.gz, data/raw/ERR753371_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753371_chr7_1_fastp.fastq.gz, data/processed/ERR753371_chr7_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753371_chr7_1_fastp.fastq.gz, data/processed/ERR753371_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1777787\n",
      "total bases: 133923240\n",
      "Q20 bases: 132711766(99.0954%)\n",
      "Q30 bases: 127818746(95.4418%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1777787\n",
      "total bases: 133584939\n",
      "Q20 bases: 131975601(98.7953%)\n",
      "Q30 bases: 127258356(95.264%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 901910\n",
      "total bases: 68460468\n",
      "Q20 bases: 68125599(99.5109%)\n",
      "Q30 bases: 66410192(97.0052%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 901910\n",
      "total bases: 68422604\n",
      "Q20 bases: 67941529(99.2969%)\n",
      "Q30 bases: 66426314(97.0824%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1803820\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1751754\n",
      "reads with adapter trimmed: 7008\n",
      "bases trimmed due to adapters: 174651\n",
      "\n",
      "Duplication rate: 21.679%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 118\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753371_chr7_1.fastq.gz -I data/raw/ERR753371_chr7_2.fastq.gz -o data/processed/ERR753371_chr7_1_fastp.fastq.gz -O data/processed/ERR753371_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Mon Aug 28 22:58:44 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:58:44 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753371_chr7_1_fastp.fastq.gz, data/processed/ERR753371_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753371_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753371_chr7_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753371_chr7_1_fastp.fastq.gz, data/processed/ERR753371_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 22:58:53 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T225823.282856.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:58:54 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753372_chr7_1.fastq.gz, data/raw/ERR753372_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753372_chr7_1_fastqc.html, results/fastqc_result/ERR753372_chr7_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753372_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753372_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 22:59:09 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:59:09 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753372_chr7_1.fastq.gz, data/raw/ERR753372_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753372_chr7_1_fastp.fastq.gz, data/processed/ERR753372_chr7_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753372_chr7_1_fastp.fastq.gz, data/processed/ERR753372_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1928165\n",
      "total bases: 146395509\n",
      "Q20 bases: 145810094(99.6001%)\n",
      "Q30 bases: 141369327(96.5667%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1928165\n",
      "total bases: 146239318\n",
      "Q20 bases: 144940401(99.1118%)\n",
      "Q30 bases: 138222059(94.5177%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1646996\n",
      "total bases: 125164638\n",
      "Q20 bases: 124943449(99.8233%)\n",
      "Q30 bases: 122763086(98.0813%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1646996\n",
      "total bases: 125153924\n",
      "Q20 bases: 124539813(99.5093%)\n",
      "Q30 bases: 120808917(96.5283%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3293992\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 562338\n",
      "reads with adapter trimmed: 6016\n",
      "bases trimmed due to adapters: 162856\n",
      "\n",
      "Duplication rate: 1.97996%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753372_chr7_1.fastq.gz -I data/raw/ERR753372_chr7_2.fastq.gz -o data/processed/ERR753372_chr7_1_fastp.fastq.gz -O data/processed/ERR753372_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Mon Aug 28 22:59:15 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:59:15 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753372_chr7_1_fastp.fastq.gz, data/processed/ERR753372_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753372_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753372_chr7_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753372_chr7_1_fastp.fastq.gz, data/processed/ERR753372_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 22:59:28 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T225853.486234.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:59:29 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753373_chr7_1.fastq.gz, data/raw/ERR753373_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753373_chr7_1_fastqc.html, results/fastqc_result/ERR753373_chr7_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753373_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753373_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 22:59:44 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:59:44 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753373_chr7_1.fastq.gz, data/raw/ERR753373_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753373_chr7_1_fastp.fastq.gz, data/processed/ERR753373_chr7_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753373_chr7_1_fastp.fastq.gz, data/processed/ERR753373_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1590509\n",
      "total bases: 152276057\n",
      "Q20 bases: 151053516(99.1972%)\n",
      "Q30 bases: 140667684(92.3768%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1590509\n",
      "total bases: 149042940\n",
      "Q20 bases: 147417333(98.9093%)\n",
      "Q30 bases: 137755782(92.4269%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1550999\n",
      "total bases: 146136096\n",
      "Q20 bases: 145283119(99.4163%)\n",
      "Q30 bases: 136885543(93.6699%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1550999\n",
      "total bases: 141207201\n",
      "Q20 bases: 140529494(99.5201%)\n",
      "Q30 bases: 133173977(94.311%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3101998\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 79020\n",
      "reads with adapter trimmed: 4140\n",
      "bases trimmed due to adapters: 154596\n",
      "\n",
      "Duplication rate: 1.84742%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 153\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753373_chr7_1.fastq.gz -I data/raw/ERR753373_chr7_2.fastq.gz -o data/processed/ERR753373_chr7_1_fastp.fastq.gz -O data/processed/ERR753373_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Mon Aug 28 22:59:51 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 22:59:51 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753373_chr7_1_fastp.fastq.gz, data/processed/ERR753373_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753373_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753373_chr7_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753373_chr7_1_fastp.fastq.gz, data/processed/ERR753373_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 23:00:06 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T225928.810824.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:00:07 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753374_chr7_1.fastq.gz, data/raw/ERR753374_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753374_chr7_1_fastp.fastq.gz, data/processed/ERR753374_chr7_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753374_chr7_1_fastp.fastq.gz, data/processed/ERR753374_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 937433\n",
      "total bases: 89668520\n",
      "Q20 bases: 88820397(99.0542%)\n",
      "Q30 bases: 81109687(90.455%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 937433\n",
      "total bases: 87729358\n",
      "Q20 bases: 86808200(98.95%)\n",
      "Q30 bases: 79810509(90.9735%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 907041\n",
      "total bases: 84770678\n",
      "Q20 bases: 84167468(99.2884%)\n",
      "Q30 bases: 78047285(92.0687%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 907041\n",
      "total bases: 82557850\n",
      "Q20 bases: 82082554(99.4243%)\n",
      "Q30 bases: 76724913(92.9347%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1814082\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 60784\n",
      "reads with adapter trimmed: 2678\n",
      "bases trimmed due to adapters: 102178\n",
      "\n",
      "Duplication rate: 5.6418%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 149\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753374_chr7_1.fastq.gz -I data/raw/ERR753374_chr7_2.fastq.gz -o data/processed/ERR753374_chr7_1_fastp.fastq.gz -O data/processed/ERR753374_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Mon Aug 28 23:00:12 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:00:12 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753374_chr7_1_fastp.fastq.gz, data/processed/ERR753374_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753374_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753374_chr7_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753374_chr7_1_fastp.fastq.gz, data/processed/ERR753374_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 23:00:21 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:00:21 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753374_chr7_1.fastq.gz, data/raw/ERR753374_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753374_chr7_1_fastqc.html, results/fastqc_result/ERR753374_chr7_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753374_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753374_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 23:00:31 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T230006.835535.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:00:33 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753375_chr7_1.fastq.gz, data/raw/ERR753375_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753375_chr7_1_fastp.fastq.gz, data/processed/ERR753375_chr7_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753375_chr7_1_fastp.fastq.gz, data/processed/ERR753375_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1203954\n",
      "total bases: 90919106\n",
      "Q20 bases: 90345397(99.369%)\n",
      "Q30 bases: 88249315(97.0636%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1203954\n",
      "total bases: 90896831\n",
      "Q20 bases: 89692945(98.6755%)\n",
      "Q30 bases: 87415245(96.1697%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 768315\n",
      "total bases: 58337245\n",
      "Q20 bases: 58144802(99.6701%)\n",
      "Q30 bases: 57253055(98.1415%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 768315\n",
      "total bases: 58346791\n",
      "Q20 bases: 57776292(99.0222%)\n",
      "Q30 bases: 56839488(97.4166%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1536630\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 871278\n",
      "reads with adapter trimmed: 4636\n",
      "bases trimmed due to adapters: 122070\n",
      "\n",
      "Duplication rate: 1.68137%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 119\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753375_chr7_1.fastq.gz -I data/raw/ERR753375_chr7_2.fastq.gz -o data/processed/ERR753375_chr7_1_fastp.fastq.gz -O data/processed/ERR753375_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Mon Aug 28 23:00:37 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:00:37 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753375_chr7_1_fastp.fastq.gz, data/processed/ERR753375_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753375_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753375_chr7_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753375_chr7_1_fastp.fastq.gz, data/processed/ERR753375_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 23:00:46 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:00:46 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753375_chr7_1.fastq.gz, data/raw/ERR753375_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753375_chr7_1_fastqc.html, results/fastqc_result/ERR753375_chr7_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753375_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753375_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 23:00:57 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T230032.123263.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:00:58 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753376_chr7_1.fastq.gz, data/raw/ERR753376_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753376_chr7_1_fastqc.html, results/fastqc_result/ERR753376_chr7_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753376_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753376_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 23:01:11 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:01:11 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753376_chr7_1.fastq.gz, data/raw/ERR753376_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753376_chr7_1_fastp.fastq.gz, data/processed/ERR753376_chr7_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753376_chr7_2_fastp.fastq.gz, data/processed/ERR753376_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1556102\n",
      "total bases: 117603181\n",
      "Q20 bases: 116778781(99.299%)\n",
      "Q30 bases: 113491260(96.5036%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1556102\n",
      "total bases: 117282101\n",
      "Q20 bases: 115811340(98.746%)\n",
      "Q30 bases: 110980302(94.6268%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 941456\n",
      "total bases: 71502248\n",
      "Q20 bases: 71230059(99.6193%)\n",
      "Q30 bases: 69975404(97.8646%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 941456\n",
      "total bases: 71475352\n",
      "Q20 bases: 70978000(99.3042%)\n",
      "Q30 bases: 69107114(96.6866%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1882912\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1229292\n",
      "reads with adapter trimmed: 6714\n",
      "bases trimmed due to adapters: 177235\n",
      "\n",
      "Duplication rate: 8.84242%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753376_chr7_1.fastq.gz -I data/raw/ERR753376_chr7_2.fastq.gz -o data/processed/ERR753376_chr7_1_fastp.fastq.gz -O data/processed/ERR753376_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Mon Aug 28 23:01:16 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:01:16 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753376_chr7_1_fastp.fastq.gz, data/processed/ERR753376_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753376_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753376_chr7_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753376_chr7_2_fastp.fastq.gz, data/processed/ERR753376_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 23:01:25 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T230057.365940.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:01:26 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753377_chr7_1.fastq.gz, data/raw/ERR753377_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753377_chr7_1_fastp.fastq.gz, data/processed/ERR753377_chr7_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753377_chr7_2_fastp.fastq.gz, data/processed/ERR753377_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1631934\n",
      "total bases: 123352115\n",
      "Q20 bases: 122472837(99.2872%)\n",
      "Q30 bases: 119023024(96.4905%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1631934\n",
      "total bases: 123051439\n",
      "Q20 bases: 121582205(98.806%)\n",
      "Q30 bases: 116732556(94.8648%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1009848\n",
      "total bases: 76696745\n",
      "Q20 bases: 76397840(99.6103%)\n",
      "Q30 bases: 75043318(97.8442%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1009848\n",
      "total bases: 76668037\n",
      "Q20 bases: 76158958(99.336%)\n",
      "Q30 bases: 74237699(96.8301%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2019696\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1244172\n",
      "reads with adapter trimmed: 6972\n",
      "bases trimmed due to adapters: 178774\n",
      "\n",
      "Duplication rate: 8.81267%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753377_chr7_1.fastq.gz -I data/raw/ERR753377_chr7_2.fastq.gz -o data/processed/ERR753377_chr7_1_fastp.fastq.gz -O data/processed/ERR753377_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Mon Aug 28 23:01:32 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:01:32 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753377_chr7_1.fastq.gz, data/raw/ERR753377_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753377_chr7_1_fastqc.html, results/fastqc_result/ERR753377_chr7_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753377_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753377_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 23:01:44 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:01:44 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753377_chr7_1_fastp.fastq.gz, data/processed/ERR753377_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753377_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753377_chr7_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753377_chr7_2_fastp.fastq.gz, data/processed/ERR753377_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 23:01:53 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T230125.641036.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:01:55 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753378_chr7_1.fastq.gz, data/raw/ERR753378_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753378_chr7_1_fastp.fastq.gz, data/processed/ERR753378_chr7_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753378_chr7_2_fastp.fastq.gz, data/processed/ERR753378_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1662012\n",
      "total bases: 126189423\n",
      "Q20 bases: 125691373(99.6053%)\n",
      "Q30 bases: 121897371(96.5987%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1662012\n",
      "total bases: 126054782\n",
      "Q20 bases: 124937025(99.1133%)\n",
      "Q30 bases: 119131691(94.5079%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1424617\n",
      "total bases: 108264659\n",
      "Q20 bases: 108073674(99.8236%)\n",
      "Q30 bases: 106186267(98.0803%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1424617\n",
      "total bases: 108255476\n",
      "Q20 bases: 107719628(99.505%)\n",
      "Q30 bases: 104436218(96.472%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2849234\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 474790\n",
      "reads with adapter trimmed: 5220\n",
      "bases trimmed due to adapters: 141481\n",
      "\n",
      "Duplication rate: 1.98926%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753378_chr7_1.fastq.gz -I data/raw/ERR753378_chr7_2.fastq.gz -o data/processed/ERR753378_chr7_1_fastp.fastq.gz -O data/processed/ERR753378_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Mon Aug 28 23:02:00 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:02:00 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753378_chr7_1.fastq.gz, data/raw/ERR753378_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753378_chr7_1_fastqc.html, results/fastqc_result/ERR753378_chr7_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753378_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753378_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 100% complete for ERR753378_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 100% complete for ERR753378_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 23:02:13 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:02:13 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753378_chr7_1_fastp.fastq.gz, data/processed/ERR753378_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753378_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753378_chr7_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753378_chr7_2_fastp.fastq.gz, data/processed/ERR753378_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Mon Aug 28 23:02:26 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T230153.965350.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:02:27 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR696683_chr7_1_fastp.fastq.gz, data/processed/ERR696683_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR696683_chr7.sam, results/mapped_reads/ERR696683_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR696683_chr7_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR696683_chr7_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.64 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 58.64 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.48 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.33 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 33.67 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 94.532 sec; CPU: 93.770 sec\n",
      "[Mon Aug 28 23:05:47 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:05:47 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR696683_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr7.bam, results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR696683_chr7.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR696683_chr7_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 23:06:27 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:06:27 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR696683_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "23:06:28.658 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 23:06:28 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR696683_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR696683_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 23:06:28 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 23:06:28\tMarkDuplicates\tStart of doWork freeMemory: 528787456; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:06:28\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 23:06:28\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam\n",
      "INFO\t2023-08-28 23:06:30\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:75,881,850\n",
      "INFO\t2023-08-28 23:06:30\tMarkDuplicates\tTracking 744 as yet unmatched pairs. 744 records in RAM.\n",
      "INFO\t2023-08-28 23:06:31\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 7:138,577,465\n",
      "INFO\t2023-08-28 23:06:31\tMarkDuplicates\tTracking 375 as yet unmatched pairs. 375 records in RAM.\n",
      "INFO\t2023-08-28 23:06:32\tMarkDuplicates\tRead 2393966 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 23:06:32\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 785181840; totalMemory: 1111490560; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:06:32\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 23:06:32\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 23:06:32\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 23:06:32\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 23:06:32\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 565134096; totalMemory: 1111490560; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:06:32\tMarkDuplicates\tMarking 206339 records as duplicates.\n",
      "INFO\t2023-08-28 23:06:32\tMarkDuplicates\tFound 75 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam\n",
      "INFO\t2023-08-28 23:06:32\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 23:06:47\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 23:06:47\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 23:06:47\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 23:06:47\tMarkDuplicates\tBefore output close freeMemory: 528084936; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:06:47\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 23:06:47\tMarkDuplicates\tAfter output close freeMemory: 528084936; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 23:06:47 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.31 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 23:06:48 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T230226.577060.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:06:49 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753368_chr7_1_fastp.fastq.gz, data/processed/ERR753368_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753368_chr7.sam, results/mapped_reads/ERR753368_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753368_chr7_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753368_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.54 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 55.40 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.49 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.33 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 32.59 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 90.107 sec; CPU: 89.346 sec\n",
      "[Mon Aug 28 23:10:44 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:10:44 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753368_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr7.bam, results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753368_chr7.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753368_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 23:11:32 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:11:32 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753368_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753368_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "23:11:33.255 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 23:11:33 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753368_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753368_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 23:11:33 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 23:11:33\tMarkDuplicates\tStart of doWork freeMemory: 528787720; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:11:33\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 23:11:33\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam\n",
      "INFO\t2023-08-28 23:11:34\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:56,377,952\n",
      "INFO\t2023-08-28 23:11:34\tMarkDuplicates\tTracking 421 as yet unmatched pairs. 421 records in RAM.\n",
      "INFO\t2023-08-28 23:11:36\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 7:105,465,318\n",
      "INFO\t2023-08-28 23:11:36\tMarkDuplicates\tTracking 476 as yet unmatched pairs. 476 records in RAM.\n",
      "INFO\t2023-08-28 23:11:37\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 7:152,248,153\n",
      "INFO\t2023-08-28 23:11:37\tMarkDuplicates\tTracking 278 as yet unmatched pairs. 278 records in RAM.\n",
      "INFO\t2023-08-28 23:11:37\tMarkDuplicates\tRead 3118785 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 23:11:37\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 461704424; totalMemory: 868220928; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:11:37\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 23:11:37\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 23:11:38\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 23:11:38\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 23:11:38\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1047479944; totalMemory: 1593835520; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:11:38\tMarkDuplicates\tMarking 74952 records as duplicates.\n",
      "INFO\t2023-08-28 23:11:38\tMarkDuplicates\tFound 515 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam\n",
      "INFO\t2023-08-28 23:11:38\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 23:11:56\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 23:11:56\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 23:11:56\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 23:11:56\tMarkDuplicates\tBefore output close freeMemory: 528086432; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:11:56\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 23:11:56\tMarkDuplicates\tAfter output close freeMemory: 528086432; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 23:11:56 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.39 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 23:11:58 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T230648.783364.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:11:59 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753369_chr7_1_fastp.fastq.gz, data/processed/ERR753369_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753369_chr7.sam, results/mapped_reads/ERR753369_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753369_chr7_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR753369_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.53 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 55.85 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.49 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.33 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 32.55 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 96.446 sec; CPU: 89.750 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 23:16:15 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:16:15 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753369_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr7.bam, results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753369_chr7.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753369_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 23:17:13 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:17:13 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753369_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753369_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "23:17:14.181 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 23:17:14 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753369_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753369_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 23:17:14 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 23:17:14\tMarkDuplicates\tStart of doWork freeMemory: 528775144; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:17:14\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 23:17:14\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam\n",
      "INFO\t2023-08-28 23:17:15\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:53,270,611\n",
      "INFO\t2023-08-28 23:17:15\tMarkDuplicates\tTracking 381 as yet unmatched pairs. 381 records in RAM.\n",
      "INFO\t2023-08-28 23:17:17\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 7:101,057,914\n",
      "INFO\t2023-08-28 23:17:17\tMarkDuplicates\tTracking 524 as yet unmatched pairs. 524 records in RAM.\n",
      "INFO\t2023-08-28 23:17:18\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 7:144,095,196\n",
      "INFO\t2023-08-28 23:17:18\tMarkDuplicates\tTracking 240 as yet unmatched pairs. 240 records in RAM.\n",
      "INFO\t2023-08-28 23:17:19\tMarkDuplicates\tRead 3329144 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 23:17:19\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 797852312; totalMemory: 1226833920; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:17:19\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 23:17:19\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 23:17:19\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 23:17:19\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 23:17:19\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1217336872; totalMemory: 1763704832; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:17:19\tMarkDuplicates\tMarking 92618 records as duplicates.\n",
      "INFO\t2023-08-28 23:17:19\tMarkDuplicates\tFound 528 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam\n",
      "INFO\t2023-08-28 23:17:19\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 23:17:38\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 23:17:38\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 23:17:38\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 23:17:38\tMarkDuplicates\tBefore output close freeMemory: 528073128; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:17:38\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 23:17:38\tMarkDuplicates\tAfter output close freeMemory: 528073128; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 23:17:38 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.41 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 23:17:41 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T231158.882153.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:17:42 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753370_chr7_1_fastp.fastq.gz, data/processed/ERR753370_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753370_chr7.sam, results/mapped_reads/ERR753370_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753370_chr7_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR753370_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.54 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 55.69 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.48 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.33 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 32.51 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 94.777 sec; CPU: 89.553 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 23:24:44 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:24:44 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753370_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr7.bam, results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753370_chr7.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753370_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 23:26:41 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:26:41 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753370_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753370_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "23:26:42.575 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 23:26:42 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753370_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753370_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 23:26:42 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 23:26:42\tMarkDuplicates\tStart of doWork freeMemory: 528786200; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:26:42\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 23:26:42\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam\n",
      "INFO\t2023-08-28 23:26:44\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:33,152,780\n",
      "INFO\t2023-08-28 23:26:44\tMarkDuplicates\tTracking 1013 as yet unmatched pairs. 1013 records in RAM.\n",
      "INFO\t2023-08-28 23:26:45\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 7:72,939,721\n",
      "INFO\t2023-08-28 23:26:45\tMarkDuplicates\tTracking 1922 as yet unmatched pairs. 1922 records in RAM.\n",
      "INFO\t2023-08-28 23:26:47\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 7:95,299,055\n",
      "INFO\t2023-08-28 23:26:47\tMarkDuplicates\tTracking 1672 as yet unmatched pairs. 1672 records in RAM.\n",
      "INFO\t2023-08-28 23:26:48\tMarkDuplicates\tRead     4,000,000 records.  Elapsed time: 00:00:06s.  Time for last 1,000,000:    1s.  Last read position: 7:117,116,342\n",
      "INFO\t2023-08-28 23:26:48\tMarkDuplicates\tTracking 1115 as yet unmatched pairs. 1115 records in RAM.\n",
      "INFO\t2023-08-28 23:26:50\tMarkDuplicates\tRead     5,000,000 records.  Elapsed time: 00:00:07s.  Time for last 1,000,000:    1s.  Last read position: 7:144,268,951\n",
      "INFO\t2023-08-28 23:26:50\tMarkDuplicates\tTracking 719 as yet unmatched pairs. 719 records in RAM.\n",
      "INFO\t2023-08-28 23:26:50\tMarkDuplicates\tRead 5445839 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 23:26:51\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 446061216; totalMemory: 1095761920; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:26:51\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 23:26:51\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 23:26:51\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 23:26:52\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 23:26:52\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1086275432; totalMemory: 1632632832; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:26:52\tMarkDuplicates\tMarking 368338 records as duplicates.\n",
      "INFO\t2023-08-28 23:26:52\tMarkDuplicates\tFound 218 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam\n",
      "INFO\t2023-08-28 23:26:52\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 23:27:30\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 23:27:30\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 23:27:30\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 23:27:30\tMarkDuplicates\tBefore output close freeMemory: 528083600; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:27:30\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 23:27:30\tMarkDuplicates\tAfter output close freeMemory: 528083600; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 23:27:30 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.80 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 23:27:34 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T231741.606044.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:27:35 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753371_chr7_1_fastp.fastq.gz, data/processed/ERR753371_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753371_chr7.sam, results/mapped_reads/ERR753371_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753371_chr7_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753371_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.53 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 55.63 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.48 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.33 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 32.57 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 94.898 sec; CPU: 89.550 sec\n",
      "[Mon Aug 28 23:30:33 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:30:33 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753371_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr7.bam, results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753371_chr7.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753371_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 23:31:05 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:31:05 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753371_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753371_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "23:31:06.194 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 23:31:06 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753371_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753371_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 23:31:06 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 23:31:06\tMarkDuplicates\tStart of doWork freeMemory: 528838600; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:31:06\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 23:31:06\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam\n",
      "INFO\t2023-08-28 23:31:07\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:94,599,583\n",
      "INFO\t2023-08-28 23:31:07\tMarkDuplicates\tTracking 163 as yet unmatched pairs. 163 records in RAM.\n",
      "INFO\t2023-08-28 23:31:08\tMarkDuplicates\tRead 1812344 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 23:31:08\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 335392024; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:31:08\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 23:31:09\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 23:31:09\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 23:31:09\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 23:31:09\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 811600552; totalMemory: 1357905920; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:31:09\tMarkDuplicates\tMarking 702762 records as duplicates.\n",
      "INFO\t2023-08-28 23:31:09\tMarkDuplicates\tFound 57 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam\n",
      "INFO\t2023-08-28 23:31:09\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 23:31:20\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 23:31:20\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 23:31:20\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 23:31:20\tMarkDuplicates\tBefore output close freeMemory: 528135696; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:31:20\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 23:31:20\tMarkDuplicates\tAfter output close freeMemory: 528135696; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 23:31:20 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.24 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 23:31:21 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T232734.685109.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:31:22 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753372_chr7_1_fastp.fastq.gz, data/processed/ERR753372_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753372_chr7.sam, results/mapped_reads/ERR753372_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753372_chr7_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753372_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.54 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 55.92 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.50 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.37 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 34.73 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 97.286 sec; CPU: 92.060 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 23:35:52 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:35:52 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753372_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr7.bam, results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753372_chr7.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753372_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 23:36:53 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:36:53 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753372_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753372_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "23:36:54.832 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 23:36:54 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753372_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753372_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 23:36:54 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 23:36:54\tMarkDuplicates\tStart of doWork freeMemory: 528819528; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:36:54\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 23:36:54\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam\n",
      "INFO\t2023-08-28 23:36:56\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:51,802,758\n",
      "INFO\t2023-08-28 23:36:56\tMarkDuplicates\tTracking 417 as yet unmatched pairs. 417 records in RAM.\n",
      "INFO\t2023-08-28 23:36:57\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 7:101,087,639\n",
      "INFO\t2023-08-28 23:36:57\tMarkDuplicates\tTracking 487 as yet unmatched pairs. 487 records in RAM.\n",
      "INFO\t2023-08-28 23:36:59\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 7:144,371,101\n",
      "INFO\t2023-08-28 23:36:59\tMarkDuplicates\tTracking 231 as yet unmatched pairs. 231 records in RAM.\n",
      "INFO\t2023-08-28 23:36:59\tMarkDuplicates\tRead 3322476 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 23:36:59\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 484127704; totalMemory: 912261120; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:36:59\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 23:36:59\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 23:37:00\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 23:37:00\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 23:37:00\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 902808680; totalMemory: 1449132032; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:37:00\tMarkDuplicates\tMarking 78247 records as duplicates.\n",
      "INFO\t2023-08-28 23:37:00\tMarkDuplicates\tFound 501 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam\n",
      "INFO\t2023-08-28 23:37:00\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 23:37:20\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 23:37:20\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 23:37:20\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 23:37:20\tMarkDuplicates\tBefore output close freeMemory: 528118560; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:37:20\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 23:37:20\tMarkDuplicates\tAfter output close freeMemory: 528118560; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 23:37:21 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.44 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 23:37:23 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T233121.701406.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:37:25 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753373_chr7_1_fastp.fastq.gz, data/processed/ERR753373_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753373_chr7.sam, results/mapped_reads/ERR753373_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753373_chr7_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR753373_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.57 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 60.52 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.50 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.35 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 34.33 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 101.581 sec; CPU: 96.278 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 23:43:43 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:43:43 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753373_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr7.bam, results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753373_chr7.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753373_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Mon Aug 28 23:45:21 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:45:21 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753373_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753373_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "23:45:22.537 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 23:45:22 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753373_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753373_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 23:45:22 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 23:45:22\tMarkDuplicates\tStart of doWork freeMemory: 528785168; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:45:22\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 23:45:22\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam\n",
      "INFO\t2023-08-28 23:45:24\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:54,553,817\n",
      "INFO\t2023-08-28 23:45:24\tMarkDuplicates\tTracking 264 as yet unmatched pairs. 264 records in RAM.\n",
      "INFO\t2023-08-28 23:45:25\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 7:102,664,978\n",
      "INFO\t2023-08-28 23:45:25\tMarkDuplicates\tTracking 232 as yet unmatched pairs. 232 records in RAM.\n",
      "INFO\t2023-08-28 23:45:27\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 7:150,822,207\n",
      "INFO\t2023-08-28 23:45:27\tMarkDuplicates\tTracking 58 as yet unmatched pairs. 58 records in RAM.\n",
      "INFO\t2023-08-28 23:45:27\tMarkDuplicates\tRead 3137876 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 23:45:27\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 604600088; totalMemory: 1011875840; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:45:27\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 23:45:27\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 23:45:28\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 23:45:28\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 23:45:28\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1002389024; totalMemory: 1548746752; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:45:28\tMarkDuplicates\tMarking 79811 records as duplicates.\n",
      "INFO\t2023-08-28 23:45:28\tMarkDuplicates\tFound 26038 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam\n",
      "INFO\t2023-08-28 23:45:28\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 23:45:56\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 23:45:56\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 23:45:56\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 23:45:56\tMarkDuplicates\tBefore output close freeMemory: 528084176; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:45:56\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 23:45:56\tMarkDuplicates\tAfter output close freeMemory: 528084176; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 23:45:56 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.56 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 23:45:59 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T233723.888796.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:46:00 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753374_chr7_1_fastp.fastq.gz, data/processed/ERR753374_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753374_chr7.sam, results/mapped_reads/ERR753374_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753374_chr7_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR753374_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.53 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 61.39 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.50 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.34 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 34.34 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 102.448 sec; CPU: 97.100 sec\n",
      "[Mon Aug 28 23:50:35 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:50:35 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753374_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr7.bam, results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753374_chr7.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753374_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 23:51:24 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:51:24 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753374_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753374_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "23:51:25.830 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 23:51:25 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753374_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753374_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 23:51:25 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 23:51:25\tMarkDuplicates\tStart of doWork freeMemory: 528788904; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:51:25\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 23:51:25\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam\n",
      "INFO\t2023-08-28 23:51:27\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:91,123,229\n",
      "INFO\t2023-08-28 23:51:27\tMarkDuplicates\tTracking 228 as yet unmatched pairs. 228 records in RAM.\n",
      "INFO\t2023-08-28 23:51:28\tMarkDuplicates\tRead 1838287 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 23:51:28\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 269032400; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:51:28\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 23:51:29\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 23:51:29\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 23:51:29\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 23:51:29\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 813647936; totalMemory: 1360003072; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:51:29\tMarkDuplicates\tMarking 189148 records as duplicates.\n",
      "INFO\t2023-08-28 23:51:29\tMarkDuplicates\tFound 63003 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam\n",
      "INFO\t2023-08-28 23:51:29\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 23:51:45\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 23:51:45\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 23:51:45\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 23:51:45\tMarkDuplicates\tBefore output close freeMemory: 528085912; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:51:45\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 23:51:45\tMarkDuplicates\tAfter output close freeMemory: 528085912; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 23:51:45 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.33 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 23:51:47 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T234559.214876.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:51:48 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753375_chr7_1_fastp.fastq.gz, data/processed/ERR753375_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753375_chr7.sam, results/mapped_reads/ERR753375_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753375_chr7_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753375_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.55 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 60.88 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.51 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.36 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 34.19 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 101.919 sec; CPU: 96.490 sec\n",
      "[Mon Aug 28 23:54:50 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:54:50 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753375_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr7.bam, results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753375_chr7.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753375_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 23:55:20 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:55:20 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753375_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753375_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "23:55:20.891 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 23:55:20 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753375_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753375_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 23:55:20 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 23:55:20\tMarkDuplicates\tStart of doWork freeMemory: 528772048; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:55:20\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 23:55:20\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam\n",
      "INFO\t2023-08-28 23:55:22\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:102,659,626\n",
      "INFO\t2023-08-28 23:55:22\tMarkDuplicates\tTracking 230 as yet unmatched pairs. 230 records in RAM.\n",
      "INFO\t2023-08-28 23:55:23\tMarkDuplicates\tRead 1549276 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 23:55:23\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 299179528; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:55:23\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 23:55:23\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 23:55:23\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 23:55:24\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 23:55:24\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 766446456; totalMemory: 1312817152; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:55:24\tMarkDuplicates\tMarking 70160 records as duplicates.\n",
      "INFO\t2023-08-28 23:55:24\tMarkDuplicates\tFound 170 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam\n",
      "INFO\t2023-08-28 23:55:24\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 23:55:34\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 23:55:34\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 23:55:34\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 23:55:34\tMarkDuplicates\tBefore output close freeMemory: 528070920; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:55:34\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 23:55:34\tMarkDuplicates\tAfter output close freeMemory: 528070920; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 23:55:34 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.22 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 23:55:35 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T235147.548282.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:55:36 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753376_chr7_1_fastp.fastq.gz, data/processed/ERR753376_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753376_chr7.sam, results/mapped_reads/ERR753376_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753376_chr7_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753376_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.55 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 60.40 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.50 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.35 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 34.18 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 101.182 sec; CPU: 95.983 sec\n",
      "[Mon Aug 28 23:58:49 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:58:49 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753376_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr7.bam, results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753376_chr7.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753376_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Mon Aug 28 23:59:27 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:59:27 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753376_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753376_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "23:59:28.330 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Aug 28 23:59:28 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753376_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753376_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Mon Aug 28 23:59:28 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-28 23:59:28\tMarkDuplicates\tStart of doWork freeMemory: 528783344; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:59:28\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-28 23:59:28\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam\n",
      "INFO\t2023-08-28 23:59:29\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:93,135,566\n",
      "INFO\t2023-08-28 23:59:29\tMarkDuplicates\tTracking 275 as yet unmatched pairs. 275 records in RAM.\n",
      "INFO\t2023-08-28 23:59:31\tMarkDuplicates\tRead 1892327 records. 0 pairs never matched.\n",
      "INFO\t2023-08-28 23:59:31\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 566804496; totalMemory: 842006528; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:59:31\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-28 23:59:31\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-28 23:59:31\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-28 23:59:32\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-28 23:59:32\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 848245456; totalMemory: 1394606080; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:59:32\tMarkDuplicates\tMarking 255074 records as duplicates.\n",
      "INFO\t2023-08-28 23:59:32\tMarkDuplicates\tFound 82 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam\n",
      "INFO\t2023-08-28 23:59:32\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-28 23:59:44\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-28 23:59:44\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-28 23:59:44\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-28 23:59:44\tMarkDuplicates\tBefore output close freeMemory: 528080728; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-28 23:59:44\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-28 23:59:44\tMarkDuplicates\tAfter output close freeMemory: 528080728; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Mon Aug 28 23:59:44 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.27 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Mon Aug 28 23:59:45 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T235535.537218.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Mon Aug 28 23:59:46 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753377_chr7_1_fastp.fastq.gz, data/processed/ERR753377_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753377_chr7.sam, results/mapped_reads/ERR753377_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753377_chr7_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753377_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.57 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 60.62 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.50 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.35 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 34.21 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 101.353 sec; CPU: 96.255 sec\n",
      "[Tue Aug 29 00:03:07 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:03:07 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753377_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr7.bam, results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753377_chr7.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753377_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 00:03:45 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:03:45 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753377_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753377_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "00:03:46.620 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 00:03:46 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753377_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753377_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 00:03:46 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 00:03:46\tMarkDuplicates\tStart of doWork freeMemory: 528827776; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 00:03:46\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 00:03:46\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam\n",
      "INFO\t2023-08-29 00:03:48\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:88,816,136\n",
      "INFO\t2023-08-29 00:03:48\tMarkDuplicates\tTracking 416 as yet unmatched pairs. 416 records in RAM.\n",
      "INFO\t2023-08-29 00:03:49\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 7:154,976,188\n",
      "INFO\t2023-08-29 00:03:49\tMarkDuplicates\tTracking 144 as yet unmatched pairs. 144 records in RAM.\n",
      "INFO\t2023-08-29 00:03:49\tMarkDuplicates\tRead 2043633 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 00:03:49\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 311917296; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 00:03:49\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 00:03:49\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 00:03:50\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 00:03:50\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 00:03:50\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 852484640; totalMemory: 1398800384; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 00:03:50\tMarkDuplicates\tMarking 269867 records as duplicates.\n",
      "INFO\t2023-08-29 00:03:50\tMarkDuplicates\tFound 71 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam\n",
      "INFO\t2023-08-29 00:03:50\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 00:04:03\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 00:04:03\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 00:04:03\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 00:04:03\tMarkDuplicates\tBefore output close freeMemory: 528125696; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 00:04:03\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 00:04:03\tMarkDuplicates\tAfter output close freeMemory: 528125696; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 00:04:03 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.28 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 00:04:04 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-28T235945.892569.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:04:06 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753378_chr7_1_fastp.fastq.gz, data/processed/ERR753378_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753378_chr7.sam, results/mapped_reads/ERR753378_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753378_chr7_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753378_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.56 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 59.80 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.49 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.33 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 33.90 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 100.542 sec; CPU: 95.086 sec\n",
      "[Tue Aug 29 00:08:01 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:08:01 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753378_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr7.bam, results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753378_chr7.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753378_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 00:08:45 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:08:45 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753378_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753378_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "00:08:46.714 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 00:08:46 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753378_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753378_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 00:08:46 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 00:08:46\tMarkDuplicates\tStart of doWork freeMemory: 528784752; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 00:08:46\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 00:08:46\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam\n",
      "INFO\t2023-08-29 00:08:48\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:65,356,803\n",
      "INFO\t2023-08-29 00:08:48\tMarkDuplicates\tTracking 552 as yet unmatched pairs. 552 records in RAM.\n",
      "INFO\t2023-08-29 00:08:49\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 7:116,559,166\n",
      "INFO\t2023-08-29 00:08:49\tMarkDuplicates\tTracking 427 as yet unmatched pairs. 427 records in RAM.\n",
      "INFO\t2023-08-29 00:08:51\tMarkDuplicates\tRead 2871707 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 00:08:51\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 474730920; totalMemory: 854589440; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 00:08:51\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 00:08:51\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 00:08:51\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 00:08:51\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 00:08:51\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1009727696; totalMemory: 1556086784; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 00:08:51\tMarkDuplicates\tMarking 66980 records as duplicates.\n",
      "INFO\t2023-08-29 00:08:51\tMarkDuplicates\tFound 448 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam\n",
      "INFO\t2023-08-29 00:08:51\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 00:09:08\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 00:09:08\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 00:09:08\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 00:09:08\tMarkDuplicates\tBefore output close freeMemory: 528082648; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 00:09:08\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 00:09:08\tMarkDuplicates\tAfter output close freeMemory: 528082648; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 00:09:08 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.37 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 00:09:10 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T000405.078042.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:09:12 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR696683_chr7_dedup.bam\n",
      "    output: results/variants/ERR696683_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR696683_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR696683_chr7.vcf\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "index file data/reference/genome.fa.fai not found, generating...\n",
      "[Tue Aug 29 00:10:01 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T000911.131003.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:10:02 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753368_chr7_dedup.bam\n",
      "    output: results/variants/ERR753368_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753368_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753368_chr7.vcf\n",
      "    wildcards: sample=ERR753368_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 00:11:08 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T001001.336717.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:11:09 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753369_chr7_dedup.bam\n",
      "    output: results/variants/ERR753369_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753369_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753369_chr7.vcf\n",
      "    wildcards: sample=ERR753369_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 00:12:18 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T001108.780036.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:12:19 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753370_chr7_dedup.bam\n",
      "    output: results/variants/ERR753370_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753370_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753370_chr7.vcf\n",
      "    wildcards: sample=ERR753370_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 00:14:11 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T001218.885705.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:14:12 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753371_chr7_dedup.bam\n",
      "    output: results/variants/ERR753371_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753371_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753371_chr7.vcf\n",
      "    wildcards: sample=ERR753371_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 00:14:47 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T001411.844570.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:14:49 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753372_chr7_dedup.bam\n",
      "    output: results/variants/ERR753372_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753372_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753372_chr7.vcf\n",
      "    wildcards: sample=ERR753372_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 00:15:58 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T001448.022356.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:15:59 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753373_chr7_dedup.bam\n",
      "    output: results/variants/ERR753373_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753373_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753373_chr7.vcf\n",
      "    wildcards: sample=ERR753373_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 00:17:20 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T001558.247815.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:17:21 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753374_chr7_dedup.bam\n",
      "    output: results/variants/ERR753374_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753374_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753374_chr7.vcf\n",
      "    wildcards: sample=ERR753374_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 00:18:13 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T001720.180495.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:18:14 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753375_chr7_dedup.bam\n",
      "    output: results/variants/ERR753375_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753375_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753375_chr7.vcf\n",
      "    wildcards: sample=ERR753375_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 00:18:54 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T001813.889850.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:18:56 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753376_chr7_dedup.bam\n",
      "    output: results/variants/ERR753376_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753376_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753376_chr7.vcf\n",
      "    wildcards: sample=ERR753376_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 00:19:37 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T001855.060300.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:19:39 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753377_chr7_dedup.bam\n",
      "    output: results/variants/ERR753377_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753377_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753377_chr7.vcf\n",
      "    wildcards: sample=ERR753377_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 00:20:21 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T001937.969519.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:20:23 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753378_chr7_dedup.bam\n",
      "    output: results/variants/ERR753378_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753378_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753378_chr7.vcf\n",
      "    wildcards: sample=ERR753378_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 00:21:25 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T002022.121447.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:21:26 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR696683_chr7.vcf\n",
      "    output: results/variants/vep/ERR696683_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR696683_chr7.txt\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 00:21:55 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T002125.275644.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:21:57 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753368_chr7.vcf\n",
      "    output: results/variants/vep/ERR753368_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753368_chr7.txt\n",
      "    wildcards: sample=ERR753368_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 00:22:28 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T002156.018763.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:22:29 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753369_chr7.vcf\n",
      "    output: results/variants/vep/ERR753369_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753369_chr7.txt\n",
      "    wildcards: sample=ERR753369_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 00:23:00 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T002228.402423.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:23:01 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753370_chr7.vcf\n",
      "    output: results/variants/vep/ERR753370_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753370_chr7.txt\n",
      "    wildcards: sample=ERR753370_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 00:23:37 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T002300.387520.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:23:38 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753371_chr7.vcf\n",
      "    output: results/variants/vep/ERR753371_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753371_chr7.txt\n",
      "    wildcards: sample=ERR753371_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 00:24:00 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T002337.213510.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:24:01 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753372_chr7.vcf\n",
      "    output: results/variants/vep/ERR753372_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753372_chr7.txt\n",
      "    wildcards: sample=ERR753372_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 00:24:31 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T002400.355348.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:24:32 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753373_chr7.vcf\n",
      "    output: results/variants/vep/ERR753373_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753373_chr7.txt\n",
      "    wildcards: sample=ERR753373_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 00:25:04 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T002431.595653.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:25:05 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753374_chr7.vcf\n",
      "    output: results/variants/vep/ERR753374_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753374_chr7.txt\n",
      "    wildcards: sample=ERR753374_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 00:25:29 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T002504.764171.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:25:30 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753375_chr7.vcf\n",
      "    output: results/variants/vep/ERR753375_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753375_chr7.txt\n",
      "    wildcards: sample=ERR753375_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 00:25:54 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T002529.718885.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:25:55 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753376_chr7.vcf\n",
      "    output: results/variants/vep/ERR753376_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753376_chr7.txt\n",
      "    wildcards: sample=ERR753376_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 00:26:19 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T002554.580149.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:26:20 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753377_chr7.vcf\n",
      "    output: results/variants/vep/ERR753377_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753377_chr7.txt\n",
      "    wildcards: sample=ERR753377_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 00:26:44 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T002619.832177.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:26:46 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753378_chr7.vcf\n",
      "    output: results/variants/vep/ERR753378_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753378_chr7.txt\n",
      "    wildcards: sample=ERR753378_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 00:27:15 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T002645.026058.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job              count    min threads    max threads\n",
      "-------------  -------  -------------  -------------\n",
      "parsing_dataR        1              1              1\n",
      "total                1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 00:27:15 2023]\n",
      "rule parsing_dataR:\n",
      "    input: code/05parsing_vep_data.R\n",
      "    output: tasks/13parsing_dataR.done\n",
      "    jobid: 0\n",
      "    reason: Params have changed since last execution\n",
      "    resources: tmpdir=/tmp\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">>> This is your sample list: ERR696683_chr7\n",
      ">>> This is your sample list: ERR753368_chr7\n",
      ">>> This is your sample list: ERR753369_chr7\n",
      ">>> This is your sample list: ERR753370_chr7\n",
      ">>> This is your sample list: ERR753371_chr7\n",
      ">>> This is your sample list: ERR753372_chr7\n",
      ">>> This is your sample list: ERR753373_chr7\n",
      ">>> This is your sample list: ERR753374_chr7\n",
      ">>> This is your sample list: ERR753375_chr7\n",
      ">>> This is your sample list: ERR753376_chr7\n",
      ">>> This is your sample list: ERR753377_chr7\n",
      ">>> This is your sample list: ERR753378_chr7\n",
      ">>> This is your gene: BRAF\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 44 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "3 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "4 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   downstream…\n",
      "6 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 38 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 52 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "2 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "3 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "4 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "5 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "6 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 46 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 5 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "3 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "4 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   downstream…\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>,\n",
      "#   freqs <chr>, clin_sig <chr>, somatic <chr>, pheno <chr>, pubmed <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 5 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "3 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "4 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   downstream…\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>,\n",
      "#   freqs <chr>, clin_sig <chr>, somatic <chr>, pheno <chr>, pubmed <chr>, …\n",
      "# A tibble: 31 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "3 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "4 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   downstream…\n",
      "6 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 25 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 31 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "3 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "4 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   downstream…\n",
      "6 7_140749350_T/C       7:1407493… C      ENSG… ENST00… Transcript   synonymous…\n",
      "# ℹ 25 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 52 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "2 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "3 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "4 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "5 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "6 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 46 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 39 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "2 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "3 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "4 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "5 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "6 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 33 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 44 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "3 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "4 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   downstream…\n",
      "6 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 38 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 5 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "3 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "4 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   downstream…\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>,\n",
      "#   freqs <chr>, clin_sig <chr>, somatic <chr>, pheno <chr>, pubmed <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 0 × 44\n",
      "# ℹ 44 variables: #uploaded_variation <chr>, location <chr>, allele <chr>,\n",
      "#   gene <chr>, feature <chr>, feature_type <chr>, consequence <chr>,\n",
      "#   cdna_position <chr>, cds_position <chr>, protein_position <chr>,\n",
      "#   amino_acids <chr>, codons <chr>, existing_variation <chr>, impact <chr>,\n",
      "#   distance <chr>, strand <chr>, flags <chr>, variant_class <chr>,\n",
      "#   symbol <chr>, symbol_source <chr>, hgnc_id <chr>, biotype <chr>,\n",
      "#   mane_select <chr>, mane_plus_clinical <chr>, tsl <chr>, appris <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "Touching output file tasks/13parsing_dataR.done.\n",
      "[Tue Aug 29 00:27:19 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T002715.519399.snakemake.log\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 26 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140749350_T/C       7:1407493… C      ENSG… ENST00… Transcript   synonymous…\n",
      "2 7_140749350_T/C       7:1407493… C      ENSG… ENST00… Transcript   synonymous…\n",
      "3 7_140749350_T/C       7:1407493… C      ENSG… ENST00… Transcript   synonymous…\n",
      "4 7_140749350_T/C       7:1407493… C      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 7_140749350_T/C       7:1407493… C      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 7_140749350_T/C       7:1407493… C      ENSG… ENST00… Transcript   non_coding…\n",
      "# ℹ 20 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "[1] \"##==========================##\"\n",
      "[1] \"##===> WORK FINISHED!!! <===##\"\n",
      "[1] \"##==========================##\"\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['snakemake', '--cores', '1', 'parsing_dataR'], returncode=0)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Running the workflow for the 7º chormosome\n",
    "\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--force\", \"data/reference/genome.fa\"]) ## rule reference_genome \n",
    "\n",
    "for sample in sample_list:\n",
    "    ## Quality inspection\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/fastqc_result/\" + sample + \"_chr7_1_fastqc.html\",   \n",
    "                    \"data/processed/\" + sample + \"_chr7_1_fastp.fastq.gz\",   \n",
    "                    \"results/fastqc_result/trimmed/\" + sample + \"_chr7_1_fastp_fastqc.html\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    ## Mapping reads\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/mapped_reads/\" + sample + \"_chr7_sorted.sam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr7_sorted.bam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr7_dedup.bam\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    # Extracting variants  \n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/\" + sample + \"_chr7.vcf\"])   \n",
    "\n",
    "## VEP CLI\n",
    "for sample in sample_list:\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/vep/\" + sample + \"_chr7.txt\"])\n",
    "\n",
    "## Extracting the gene PIKCA3 from the VEP files\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"parsing_dataR\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Changing the config file to do the chromosome 12 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['sed', '-i', '65s/BRAF/KRAS/g', 'config.yaml'], returncode=0)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subprocess.run([\"sed\", \"-i\", \"6,+12s/chr7/chr12/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"24,+12s/chr7/chr12/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\",\"39s/chr7/chr12/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"46s/chromosome.7.fa.gz/chromosome.12.fa.gz/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"65s/BRAF/KRAS/g\", \"config.yaml\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job                 count    min threads    max threads\n",
      "----------------  -------  -------------  -------------\n",
      "reference_genome        1              1              1\n",
      "total                   1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:24:41 2023]\n",
      "rule reference_genome:\n",
      "    output: data/reference/genome.fa\n",
      "    jobid: 0\n",
      "    reason: Forced execution\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "--2023-08-29 10:24:41--  https://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.12.fa.gz\n",
      "Resolviendo ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.169\n",
      "Conectando con ftp.ensembl.org (ftp.ensembl.org)[193.62.193.169]:443... conectado.\n",
      "Petición HTTP enviada, esperando respuesta... 200 OK\n",
      "Longitud: 39795542 (38M) [application/x-gzip]\n",
      "Guardando como: ‘data/reference/genome.fa.gz’\n",
      "\n",
      "     0K .......... .......... .......... .......... ..........  0%  409K 95s\n",
      "    50K .......... .......... .......... .......... ..........  0% 1,30M 62s\n",
      "   100K .......... .......... .......... .......... ..........  0% 2,04M 47s\n",
      "   150K .......... .......... .......... .......... ..........  0% 42,6M 36s\n",
      "   200K .......... .......... .......... .......... ..........  0%  849K 38s\n",
      "   250K .......... .......... .......... .......... ..........  0% 92,3M 31s\n",
      "   300K .......... .......... .......... .......... ..........  0% 72,0M 27s\n",
      "   350K .......... .......... .......... .......... ..........  1% 55,0M 24s\n",
      "   400K .......... .......... .......... .......... ..........  1% 1,37M 24s\n",
      "   450K .......... .......... .......... .......... ..........  1% 2,15M 23s\n",
      "   500K .......... .......... .......... .......... ..........  1% 48,1M 21s\n",
      "   550K .......... .......... .......... .......... ..........  1%  113M 19s\n",
      "   600K .......... .......... .......... .......... ..........  1%  102M 18s\n",
      "   650K .......... .......... .......... .......... ..........  1%  122M 17s\n",
      "   700K .......... .......... .......... .......... ..........  1%  102M 16s\n",
      "   750K .......... .......... .......... .......... ..........  2%  120M 15s\n",
      "   800K .......... .......... .......... .......... ..........  2%  117M 14s\n",
      "   850K .......... .......... .......... .......... ..........  2% 1,40M 14s\n",
      "   900K .......... .......... .......... .......... ..........  2%  126M 14s\n",
      "   950K .......... .......... .......... .......... ..........  2% 2,26M 14s\n",
      "  1000K .......... .......... .......... .......... ..........  2% 59,8M 13s\n",
      "  1050K .......... .......... .......... .......... ..........  2%  105M 13s\n",
      "  1100K .......... .......... .......... .......... ..........  2%  119M 12s\n",
      "  1150K .......... .......... .......... .......... ..........  3% 72,8M 12s\n",
      "  1200K .......... .......... .......... .......... ..........  3%  104M 11s\n",
      "  1250K .......... .......... .......... .......... ..........  3%  121M 11s\n",
      "  1300K .......... .......... .......... .......... ..........  3%  121M 10s\n",
      "  1350K .......... .......... .......... .......... ..........  3%  102M 10s\n",
      "  1400K .......... .......... .......... .......... ..........  3%  118M 10s\n",
      "  1450K .......... .......... .......... .......... ..........  3%  101M 9s\n",
      "  1500K .......... .......... .......... .......... ..........  3%  121M 9s\n",
      "  1550K .......... .......... .......... .......... ..........  4%  118M 9s\n",
      "  1600K .......... .......... .......... .......... ..........  4%  102M 8s\n",
      "  1650K .......... .......... .......... .......... ..........  4%  121M 8s\n",
      "  1700K .......... .......... .......... .......... ..........  4% 1,61M 9s\n",
      "  1750K .......... .......... .......... .......... ..........  4%  126M 8s\n",
      "  1800K .......... .......... .......... .......... ..........  4%  117M 8s\n",
      "  1850K .......... .......... .......... .......... ..........  4% 53,6M 8s\n",
      "  1900K .......... .......... .......... .......... ..........  5% 26,1M 8s\n",
      "  1950K .......... .......... .......... .......... ..........  5% 2,35M 8s\n",
      "  2000K .......... .......... .......... .......... ..........  5%  121M 8s\n",
      "  2050K .......... .......... .......... .......... ..........  5%  121M 7s\n",
      "  2100K .......... .......... .......... .......... ..........  5%  102M 7s\n",
      "  2150K .......... .......... .......... .......... ..........  5%  117M 7s\n",
      "  2200K .......... .......... .......... .......... ..........  5%  103M 7s\n",
      "  2250K .......... .......... .......... .......... ..........  5%  120M 7s\n",
      "  2300K .......... .......... .......... .......... ..........  6%  122M 7s\n",
      "  2350K .......... .......... .......... .......... ..........  6%  102M 7s\n",
      "  2400K .......... .......... .......... .......... ..........  6%  117M 6s\n",
      "  2450K .......... .......... .......... .......... ..........  6%  103M 6s\n",
      "  2500K .......... .......... .......... .......... ..........  6%  120M 6s\n",
      "  2550K .......... .......... .......... .......... ..........  6%  117M 6s\n",
      "  2600K .......... .......... .......... .......... ..........  6% 96,2M 6s\n",
      "  2650K .......... .......... .......... .......... ..........  6%  129M 6s\n",
      "  2700K .......... .......... .......... .......... ..........  7%  103M 6s\n",
      "  2750K .......... .......... .......... .......... ..........  7%  117M 6s\n",
      "  2800K .......... .......... .......... .......... ..........  7%  122M 5s\n",
      "  2850K .......... .......... .......... .......... ..........  7%  102M 5s\n",
      "  2900K .......... .......... .......... .......... ..........  7%  121M 5s\n",
      "  2950K .......... .......... .......... .......... ..........  7%  102M 5s\n",
      "  3000K .......... .......... .......... .......... ..........  7%  118M 5s\n",
      "  3050K .......... .......... .......... .......... ..........  7%  122M 5s\n",
      "  3100K .......... .......... .......... .......... ..........  8%  102M 5s\n",
      "  3150K .......... .......... .......... .......... ..........  8%  117M 5s\n",
      "  3200K .......... .......... .......... .......... ..........  8%  103M 5s\n",
      "  3250K .......... .......... .......... .......... ..........  8%  120M 5s\n",
      "  3300K .......... .......... .......... .......... ..........  8%  117M 5s\n",
      "  3350K .......... .......... .......... .......... ..........  8%  103M 5s\n",
      "  3400K .......... .......... .......... .......... ..........  8%  120M 5s\n",
      "  3450K .......... .......... .......... .......... ..........  9% 2,00M 5s\n",
      "  3500K .......... .......... .......... .......... ..........  9% 66,1M 5s\n",
      "  3550K .......... .......... .......... .......... ..........  9%  117M 5s\n",
      "  3600K .......... .......... .......... .......... ..........  9%  103M 4s\n",
      "  3650K .......... .......... .......... .......... ..........  9%  120M 4s\n",
      "  3700K .......... .......... .......... .......... ..........  9%  103M 4s\n",
      "  3750K .......... .......... .......... .......... ..........  9%  117M 4s\n",
      "  3800K .......... .......... .......... .......... ..........  9%  120M 4s\n",
      "  3850K .......... .......... .......... .......... .......... 10%  103M 4s\n",
      "  3900K .......... .......... .......... .......... .......... 10% 2,93M 4s\n",
      "  3950K .......... .......... .......... .......... .......... 10% 19,4M 4s\n",
      "  4000K .......... .......... .......... .......... .......... 10% 35,3M 4s\n",
      "  4050K .......... .......... .......... .......... .......... 10%  120M 4s\n",
      "  4100K .......... .......... .......... .......... .......... 10%  107M 4s\n",
      "  4150K .......... .......... .......... .......... .......... 10%  118M 4s\n",
      "  4200K .......... .......... .......... .......... .......... 10%  102M 4s\n",
      "  4250K .......... .......... .......... .......... .......... 11%  121M 4s\n",
      "  4300K .......... .......... .......... .......... .......... 11%  118M 4s\n",
      "  4350K .......... .......... .......... .......... .......... 11%  101M 4s\n",
      "  4400K .......... .......... .......... .......... .......... 11%  124M 4s\n",
      "  4450K .......... .......... .......... .......... .......... 11%  101M 4s\n",
      "  4500K .......... .......... .......... .......... .......... 11%  122M 4s\n",
      "  4550K .......... .......... .......... .......... .......... 11%  117M 4s\n",
      "  4600K .......... .......... .......... .......... .......... 11%  102M 4s\n",
      "  4650K .......... .......... .......... .......... .......... 12%  122M 4s\n",
      "  4700K .......... .......... .......... .......... .......... 12%  102M 4s\n",
      "  4750K .......... .......... .......... .......... .......... 12%  118M 4s\n",
      "  4800K .......... .......... .......... .......... .......... 12%  122M 3s\n",
      "  4850K .......... .......... .......... .......... .......... 12%  102M 3s\n",
      "  4900K .......... .......... .......... .......... .......... 12%  117M 3s\n",
      "  4950K .......... .......... .......... .......... .......... 12%  103M 3s\n",
      "  5000K .......... .......... .......... .......... .......... 12%  120M 3s\n",
      "  5050K .......... .......... .......... .......... .......... 13%  121M 3s\n",
      "  5100K .......... .......... .......... .......... .......... 13%  103M 3s\n",
      "  5150K .......... .......... .......... .......... .......... 13%  117M 3s\n",
      "  5200K .......... .......... .......... .......... .......... 13%  102M 3s\n",
      "  5250K .......... .......... .......... .......... .......... 13%  122M 3s\n",
      "  5300K .......... .......... .......... .......... .......... 13%  117M 3s\n",
      "  5350K .......... .......... .......... .......... .......... 13%  102M 3s\n",
      "  5400K .......... .......... .......... .......... .......... 14%  122M 3s\n",
      "  5450K .......... .......... .......... .......... .......... 14% 99,4M 3s\n",
      "  5500K .......... .......... .......... .......... .......... 14%  124M 3s\n",
      "  5550K .......... .......... .......... .......... .......... 14%  117M 3s\n",
      "  5600K .......... .......... .......... .......... .......... 14%  103M 3s\n",
      "  5650K .......... .......... .......... .......... .......... 14%  120M 3s\n",
      "  5700K .......... .......... .......... .......... .......... 14%  103M 3s\n",
      "  5750K .......... .......... .......... .......... .......... 14%  117M 3s\n",
      "  5800K .......... .......... .......... .......... .......... 15%  120M 3s\n",
      "  5850K .......... .......... .......... .......... .......... 15%  103M 3s\n",
      "  5900K .......... .......... .......... .......... .......... 15%  117M 3s\n",
      "  5950K .......... .......... .......... .......... .......... 15%  102M 3s\n",
      "  6000K .......... .......... .......... .......... .......... 15%  122M 3s\n",
      "  6050K .......... .......... .......... .......... .......... 15%  120M 3s\n",
      "  6100K .......... .......... .......... .......... .......... 15%  103M 3s\n",
      "  6150K .......... .......... .......... .......... .......... 15%  116M 3s\n",
      "  6200K .......... .......... .......... .......... .......... 16%  102M 3s\n",
      "  6250K .......... .......... .......... .......... .......... 16%  121M 3s\n",
      "  6300K .......... .......... .......... .......... .......... 16%  117M 3s\n",
      "  6350K .......... .......... .......... .......... .......... 16%  102M 3s\n",
      "  6400K .......... .......... .......... .......... .......... 16%  122M 3s\n",
      "  6450K .......... .......... .......... .......... .......... 16%  102M 3s\n",
      "  6500K .......... .......... .......... .......... .......... 16% 3,72M 3s\n",
      "  6550K .......... .......... .......... .......... .......... 16% 14,8M 3s\n",
      "  6600K .......... .......... .......... .......... .......... 17%  103M 3s\n",
      "  6650K .......... .......... .......... .......... .......... 17%  122M 3s\n",
      "  6700K .......... .......... .......... .......... .......... 17%  102M 3s\n",
      "  6750K .......... .......... .......... .......... .......... 17%  116M 2s\n",
      "  6800K .......... .......... .......... .......... .......... 17%  121M 2s\n",
      "  6850K .......... .......... .......... .......... .......... 17%  103M 2s\n",
      "  6900K .......... .......... .......... .......... .......... 17% 3,26M 2s\n",
      "  6950K .......... .......... .......... .......... .......... 18% 25,3M 2s\n",
      "  7000K .......... .......... .......... .......... .......... 18% 24,1M 2s\n",
      "  7050K .......... .......... .......... .......... .......... 18% 30,1M 2s\n",
      "  7100K .......... .......... .......... .......... .......... 18% 77,5M 2s\n",
      "  7150K .......... .......... .......... .......... .......... 18% 72,0M 2s\n",
      "  7200K .......... .......... .......... .......... .......... 18%  103M 2s\n",
      "  7250K .......... .......... .......... .......... .......... 18%  113M 2s\n",
      "  7300K .......... .......... .......... .......... .......... 18%  126M 2s\n",
      "  7350K .......... .......... .......... .......... .......... 19%  101M 2s\n",
      "  7400K .......... .......... .......... .......... .......... 19%  122M 2s\n",
      "  7450K .......... .......... .......... .......... .......... 19%  102M 2s\n",
      "  7500K .......... .......... .......... .......... .......... 19%  117M 2s\n",
      "  7550K .......... .......... .......... .......... .......... 19%  122M 2s\n",
      "  7600K .......... .......... .......... .......... .......... 19%  102M 2s\n",
      "  7650K .......... .......... .......... .......... .......... 19%  122M 2s\n",
      "  7700K .......... .......... .......... .......... .......... 19%  101M 2s\n",
      "  7750K .......... .......... .......... .......... .......... 20%  117M 2s\n",
      "  7800K .......... .......... .......... .......... .......... 20%  122M 2s\n",
      "  7850K .......... .......... .......... .......... .......... 20%  102M 2s\n",
      "  7900K .......... .......... .......... .......... .......... 20%  117M 2s\n",
      "  7950K .......... .......... .......... .......... .......... 20%  103M 2s\n",
      "  8000K .......... .......... .......... .......... .......... 20%  120M 2s\n",
      "  8050K .......... .......... .......... .......... .......... 20%  117M 2s\n",
      "  8100K .......... .......... .......... .......... .......... 20%  103M 2s\n",
      "  8150K .......... .......... .......... .......... .......... 21%  120M 2s\n",
      "  8200K .......... .......... .......... .......... .......... 21%  103M 2s\n",
      "  8250K .......... .......... .......... .......... .......... 21%  120M 2s\n",
      "  8300K .......... .......... .......... .......... .......... 21%  117M 2s\n",
      "  8350K .......... .......... .......... .......... .......... 21%  103M 2s\n",
      "  8400K .......... .......... .......... .......... .......... 21%  120M 2s\n",
      "  8450K .......... .......... .......... .......... .......... 21%  103M 2s\n",
      "  8500K .......... .......... .......... .......... .......... 22%  117M 2s\n",
      "  8550K .......... .......... .......... .......... .......... 22%  120M 2s\n",
      "  8600K .......... .......... .......... .......... .......... 22%  103M 2s\n",
      "  8650K .......... .......... .......... .......... .......... 22%  117M 2s\n",
      "  8700K .......... .......... .......... .......... .......... 22%  102M 2s\n",
      "  8750K .......... .......... .......... .......... .......... 22%  122M 2s\n",
      "  8800K .......... .......... .......... .......... .......... 22%  120M 2s\n",
      "  8850K .......... .......... .......... .......... .......... 22%  103M 2s\n",
      "  8900K .......... .......... .......... .......... .......... 23%  117M 2s\n",
      "  8950K .......... .......... .......... .......... .......... 23%  102M 2s\n",
      "  9000K .......... .......... .......... .......... .......... 23%  122M 2s\n",
      "  9050K .......... .......... .......... .......... .......... 23%  117M 2s\n",
      "  9100K .......... .......... .......... .......... .......... 23%  102M 2s\n",
      "  9150K .......... .......... .......... .......... .......... 23%  122M 2s\n",
      "  9200K .......... .......... .......... .......... .......... 23%  101M 2s\n",
      "  9250K .......... .......... .......... .......... .......... 23%  121M 2s\n",
      "  9300K .......... .......... .......... .......... .......... 24%  118M 2s\n",
      "  9350K .......... .......... .......... .......... .......... 24%  102M 2s\n",
      "  9400K .......... .......... .......... .......... .......... 24%  122M 2s\n",
      "  9450K .......... .......... .......... .......... .......... 24%  102M 2s\n",
      "  9500K .......... .......... .......... .......... .......... 24%  118M 2s\n",
      "  9550K .......... .......... .......... .......... .......... 24% 10,4M 2s\n",
      "  9600K .......... .......... .......... .......... .......... 24% 4,29M 2s\n",
      "  9650K .......... .......... .......... .......... .......... 24%  118M 2s\n",
      "  9700K .......... .......... .......... .......... .......... 25% 65,1M 2s\n",
      "  9750K .......... .......... .......... .......... .......... 25%  122M 2s\n",
      "  9800K .......... .......... .......... .......... .......... 25%  120M 2s\n",
      "  9850K .......... .......... .......... .......... .......... 25%  103M 2s\n",
      "  9900K .......... .......... .......... .......... .......... 25%  117M 2s\n",
      "  9950K .......... .......... .......... .......... .......... 25% 3,39M 2s\n",
      " 10000K .......... .......... .......... .......... .......... 25% 28,1M 2s\n",
      " 10050K .......... .......... .......... .......... .......... 25% 31,8M 2s\n",
      " 10100K .......... .......... .......... .......... .......... 26% 14,8M 2s\n",
      " 10150K .......... .......... .......... .......... .......... 26%  122M 2s\n",
      " 10200K .......... .......... .......... .......... .......... 26% 71,3M 2s\n",
      " 10250K .......... .......... .......... .......... .......... 26%  117M 2s\n",
      " 10300K .......... .......... .......... .......... .......... 26%  122M 2s\n",
      " 10350K .......... .......... .......... .......... .......... 26%  102M 2s\n",
      " 10400K .......... .......... .......... .......... .......... 26%  122M 2s\n",
      " 10450K .......... .......... .......... .......... .......... 27%  101M 2s\n",
      " 10500K .......... .......... .......... .......... .......... 27% 69,3M 2s\n",
      " 10550K .......... .......... .......... .......... .......... 27%  121M 2s\n",
      " 10600K .......... .......... .......... .......... .......... 27%  103M 2s\n",
      " 10650K .......... .......... .......... .......... .......... 27%  117M 2s\n",
      " 10700K .......... .......... .......... .......... .......... 27%  102M 2s\n",
      " 10750K .......... .......... .......... .......... .......... 27%  122M 2s\n",
      " 10800K .......... .......... .......... .......... .......... 27%  117M 2s\n",
      " 10850K .......... .......... .......... .......... .......... 28%  102M 2s\n",
      " 10900K .......... .......... .......... .......... .......... 28%  122M 2s\n",
      " 10950K .......... .......... .......... .......... .......... 28%  102M 2s\n",
      " 11000K .......... .......... .......... .......... .......... 28%  120M 2s\n",
      " 11050K .......... .......... .......... .......... .......... 28%  118M 2s\n",
      " 11100K .......... .......... .......... .......... .......... 28%  102M 2s\n",
      " 11150K .......... .......... .......... .......... .......... 28%  122M 2s\n",
      " 11200K .......... .......... .......... .......... .......... 28%  102M 2s\n",
      " 11250K .......... .......... .......... .......... .......... 29%  118M 2s\n",
      " 11300K .......... .......... .......... .......... .......... 29%  120M 2s\n",
      " 11350K .......... .......... .......... .......... .......... 29%  102M 1s\n",
      " 11400K .......... .......... .......... .......... .......... 29%  121M 1s\n",
      " 11450K .......... .......... .......... .......... .......... 29%  103M 1s\n",
      " 11500K .......... .......... .......... .......... .......... 29%  117M 1s\n",
      " 11550K .......... .......... .......... .......... .......... 29%  120M 1s\n",
      " 11600K .......... .......... .......... .......... .......... 29%  103M 1s\n",
      " 11650K .......... .......... .......... .......... .......... 30%  117M 1s\n",
      " 11700K .......... .......... .......... .......... .......... 30%  102M 1s\n",
      " 11750K .......... .......... .......... .......... .......... 30%  122M 1s\n",
      " 11800K .......... .......... .......... .......... .......... 30%  117M 1s\n",
      " 11850K .......... .......... .......... .......... .......... 30%  102M 1s\n",
      " 11900K .......... .......... .......... .......... .......... 30%  122M 1s\n",
      " 11950K .......... .......... .......... .......... .......... 30%  102M 1s\n",
      " 12000K .......... .......... .......... .......... .......... 31%  122M 1s\n",
      " 12050K .......... .......... .......... .......... .......... 31%  117M 1s\n",
      " 12100K .......... .......... .......... .......... .......... 31%  102M 1s\n",
      " 12150K .......... .......... .......... .......... .......... 31%  122M 1s\n",
      " 12200K .......... .......... .......... .......... .......... 31%  102M 1s\n",
      " 12250K .......... .......... .......... .......... .......... 31%  117M 1s\n",
      " 12300K .......... .......... .......... .......... .......... 31%  122M 1s\n",
      " 12350K .......... .......... .......... .......... .......... 31%  102M 1s\n",
      " 12400K .......... .......... .......... .......... .......... 32%  118M 1s\n",
      " 12450K .......... .......... .......... .......... .......... 32%  103M 1s\n",
      " 12500K .......... .......... .......... .......... .......... 32%  120M 1s\n",
      " 12550K .......... .......... .......... .......... .......... 32%  122M 1s\n",
      " 12600K .......... .......... .......... .......... .......... 32% 30,2M 1s\n",
      " 12650K .......... .......... .......... .......... .......... 32% 3,48M 1s\n",
      " 12700K .......... .......... .......... .......... .......... 32%  102M 1s\n",
      " 12750K .......... .......... .......... .......... .......... 32%  121M 1s\n",
      " 12800K .......... .......... .......... .......... .......... 33% 87,1M 1s\n",
      " 12850K .......... .......... .......... .......... .......... 33%  103M 1s\n",
      " 12900K .......... .......... .......... .......... .......... 33%  120M 1s\n",
      " 12950K .......... .......... .......... .......... .......... 33%  103M 1s\n",
      " 13000K .......... .......... .......... .......... .......... 33%  120M 1s\n",
      " 13050K .......... .......... .......... .......... .......... 33% 3,11M 1s\n",
      " 13100K .......... .......... .......... .......... .......... 33% 44,4M 1s\n",
      " 13150K .......... .......... .......... .......... .......... 33% 12,9M 1s\n",
      " 13200K .......... .......... .......... .......... .......... 34%  102M 1s\n",
      " 13250K .......... .......... .......... .......... .......... 34%  117M 1s\n",
      " 13300K .......... .......... .......... .......... .......... 34%  108M 1s\n",
      " 13350K .......... .......... .......... .......... .......... 34%  102M 1s\n",
      " 13400K .......... .......... .......... .......... .......... 34%  117M 1s\n",
      " 13450K .......... .......... .......... .......... .......... 34%  102M 1s\n",
      " 13500K .......... .......... .......... .......... .......... 34%  121M 1s\n",
      " 13550K .......... .......... .......... .......... .......... 34%  120M 1s\n",
      " 13600K .......... .......... .......... .......... .......... 35%  102M 1s\n",
      " 13650K .......... .......... .......... .......... .......... 35%  118M 1s\n",
      " 13700K .......... .......... .......... .......... .......... 35%  103M 1s\n",
      " 13750K .......... .......... .......... .......... .......... 35%  121M 1s\n",
      " 13800K .......... .......... .......... .......... .......... 35% 48,5M 1s\n",
      " 13850K .......... .......... .......... .......... .......... 35%  102M 1s\n",
      " 13900K .......... .......... .......... .......... .......... 35%  121M 1s\n",
      " 13950K .......... .......... .......... .......... .......... 36%  102M 1s\n",
      " 14000K .......... .......... .......... .......... .......... 36%  117M 1s\n",
      " 14050K .......... .......... .......... .......... .......... 36%  122M 1s\n",
      " 14100K .......... .......... .......... .......... .......... 36%  102M 1s\n",
      " 14150K .......... .......... .......... .......... .......... 36%  122M 1s\n",
      " 14200K .......... .......... .......... .......... .......... 36%  101M 1s\n",
      " 14250K .......... .......... .......... .......... .......... 36%  118M 1s\n",
      " 14300K .......... .......... .......... .......... .......... 36%  122M 1s\n",
      " 14350K .......... .......... .......... .......... .......... 37%  101M 1s\n",
      " 14400K .......... .......... .......... .......... .......... 37%  118M 1s\n",
      " 14450K .......... .......... .......... .......... .......... 37%  102M 1s\n",
      " 14500K .......... .......... .......... .......... .......... 37%  107M 1s\n",
      " 14550K .......... .......... .......... .......... .......... 37%  134M 1s\n",
      " 14600K .......... .......... .......... .......... .......... 37%  102M 1s\n",
      " 14650K .......... .......... .......... .......... .......... 37%  121M 1s\n",
      " 14700K .......... .......... .......... .......... .......... 37%  102M 1s\n",
      " 14750K .......... .......... .......... .......... .......... 38%  121M 1s\n",
      " 14800K .......... .......... .......... .......... .......... 38%  117M 1s\n",
      " 14850K .......... .......... .......... .......... .......... 38%  101M 1s\n",
      " 14900K .......... .......... .......... .......... .......... 38%  122M 1s\n",
      " 14950K .......... .......... .......... .......... .......... 38%  102M 1s\n",
      " 15000K .......... .......... .......... .......... .......... 38%  117M 1s\n",
      " 15050K .......... .......... .......... .......... .......... 38%  121M 1s\n",
      " 15100K .......... .......... .......... .......... .......... 38%  102M 1s\n",
      " 15150K .......... .......... .......... .......... .......... 39%  122M 1s\n",
      " 15200K .......... .......... .......... .......... .......... 39%  102M 1s\n",
      " 15250K .......... .......... .......... .......... .......... 39%  117M 1s\n",
      " 15300K .......... .......... .......... .......... .......... 39%  121M 1s\n",
      " 15350K .......... .......... .......... .......... .......... 39%  103M 1s\n",
      " 15400K .......... .......... .......... .......... .......... 39%  117M 1s\n",
      " 15450K .......... .......... .......... .......... .......... 39%  103M 1s\n",
      " 15500K .......... .......... .......... .......... .......... 40%  120M 1s\n",
      " 15550K .......... .......... .......... .......... .......... 40%  117M 1s\n",
      " 15600K .......... .......... .......... .......... .......... 40%  102M 1s\n",
      " 15650K .......... .......... .......... .......... .......... 40%  120M 1s\n",
      " 15700K .......... .......... .......... .......... .......... 40% 3,56M 1s\n",
      " 15750K .......... .......... .......... .......... .......... 40% 30,2M 1s\n",
      " 15800K .......... .......... .......... .......... .......... 40% 69,1M 1s\n",
      " 15850K .......... .......... .......... .......... .......... 40%  103M 1s\n",
      " 15900K .......... .......... .......... .......... .......... 41%  120M 1s\n",
      " 15950K .......... .......... .......... .......... .......... 41%  102M 1s\n",
      " 16000K .......... .......... .......... .......... .......... 41%  118M 1s\n",
      " 16050K .......... .......... .......... .......... .......... 41% 3,73M 1s\n",
      " 16100K .......... .......... .......... .......... .......... 41% 15,9M 1s\n",
      " 16150K .......... .......... .......... .......... .......... 41% 44,2M 1s\n",
      " 16200K .......... .......... .......... .......... .......... 41% 23,0M 1s\n",
      " 16250K .......... .......... .......... .......... .......... 41% 25,9M 1s\n",
      " 16300K .......... .......... .......... .......... .......... 42% 69,6M 1s\n",
      " 16350K .......... .......... .......... .......... .......... 42%  103M 1s\n",
      " 16400K .......... .......... .......... .......... .......... 42%  116M 1s\n",
      " 16450K .......... .......... .......... .......... .......... 42%  102M 1s\n",
      " 16500K .......... .......... .......... .......... .......... 42%  120M 1s\n",
      " 16550K .......... .......... .......... .......... .......... 42%  117M 1s\n",
      " 16600K .......... .......... .......... .......... .......... 42%  103M 1s\n",
      " 16650K .......... .......... .......... .......... .......... 42%  120M 1s\n",
      " 16700K .......... .......... .......... .......... .......... 43%  102M 1s\n",
      " 16750K .......... .......... .......... .......... .......... 43%  120M 1s\n",
      " 16800K .......... .......... .......... .......... .......... 43%  117M 1s\n",
      " 16850K .......... .......... .......... .......... .......... 43%  102M 1s\n",
      " 16900K .......... .......... .......... .......... .......... 43%  121M 1s\n",
      " 16950K .......... .......... .......... .......... .......... 43%  102M 1s\n",
      " 17000K .......... .......... .......... .......... .......... 43% 41,7M 1s\n",
      " 17050K .......... .......... .......... .......... .......... 44%  120M 1s\n",
      " 17100K .......... .......... .......... .......... .......... 44%  103M 1s\n",
      " 17150K .......... .......... .......... .......... .......... 44%  117M 1s\n",
      " 17200K .......... .......... .......... .......... .......... 44%  103M 1s\n",
      " 17250K .......... .......... .......... .......... .......... 44%  122M 1s\n",
      " 17300K .......... .......... .......... .......... .......... 44%  120M 1s\n",
      " 17350K .......... .......... .......... .......... .......... 44%  103M 1s\n",
      " 17400K .......... .......... .......... .......... .......... 44%  117M 1s\n",
      " 17450K .......... .......... .......... .......... .......... 45%  102M 1s\n",
      " 17500K .......... .......... .......... .......... .......... 45%  122M 1s\n",
      " 17550K .......... .......... .......... .......... .......... 45%  118M 1s\n",
      " 17600K .......... .......... .......... .......... .......... 45%  102M 1s\n",
      " 17650K .......... .......... .......... .......... .......... 45%  122M 1s\n",
      " 17700K .......... .......... .......... .......... .......... 45%  102M 1s\n",
      " 17750K .......... .......... .......... .......... .......... 45%  118M 1s\n",
      " 17800K .......... .......... .......... .......... .......... 45%  120M 1s\n",
      " 17850K .......... .......... .......... .......... .......... 46%  103M 1s\n",
      " 17900K .......... .......... .......... .......... .......... 46%  120M 1s\n",
      " 17950K .......... .......... .......... .......... .......... 46%  103M 1s\n",
      " 18000K .......... .......... .......... .......... .......... 46%  117M 1s\n",
      " 18050K .......... .......... .......... .......... .......... 46%  120M 1s\n",
      " 18100K .......... .......... .......... .......... .......... 46%  103M 1s\n",
      " 18150K .......... .......... .......... .......... .......... 46%  117M 1s\n",
      " 18200K .......... .......... .......... .......... .......... 46%  102M 1s\n",
      " 18250K .......... .......... .......... .......... .......... 47%  122M 1s\n",
      " 18300K .......... .......... .......... .......... .......... 47%  117M 1s\n",
      " 18350K .......... .......... .......... .......... .......... 47%  102M 1s\n",
      " 18400K .......... .......... .......... .......... .......... 47%  122M 1s\n",
      " 18450K .......... .......... .......... .......... .......... 47%  101M 1s\n",
      " 18500K .......... .......... .......... .......... .......... 47%  122M 1s\n",
      " 18550K .......... .......... .......... .......... .......... 47%  116M 1s\n",
      " 18600K .......... .......... .......... .......... .......... 47%  102M 1s\n",
      " 18650K .......... .......... .......... .......... .......... 48%  122M 1s\n",
      " 18700K .......... .......... .......... .......... .......... 48%  102M 1s\n",
      " 18750K .......... .......... .......... .......... .......... 48% 4,35M 1s\n",
      " 18800K .......... .......... .......... .......... .......... 48% 12,9M 1s\n",
      " 18850K .......... .......... .......... .......... .......... 48%  104M 1s\n",
      " 18900K .......... .......... .......... .......... .......... 48% 81,0M 1s\n",
      " 18950K .......... .......... .......... .......... .......... 48%  103M 1s\n",
      " 19000K .......... .......... .......... .......... .......... 49%  117M 1s\n",
      " 19050K .......... .......... .......... .......... .......... 49%  120M 1s\n",
      " 19100K .......... .......... .......... .......... .......... 49% 3,76M 1s\n",
      " 19150K .......... .......... .......... .......... .......... 49% 31,4M 1s\n",
      " 19200K .......... .......... .......... .......... .......... 49% 22,8M 1s\n",
      " 19250K .......... .......... .......... .......... .......... 49% 32,2M 1s\n",
      " 19300K .......... .......... .......... .......... .......... 49% 15,9M 1s\n",
      " 19350K .......... .......... .......... .......... .......... 49%  104M 1s\n",
      " 19400K .......... .......... .......... .......... .......... 50% 73,7M 1s\n",
      " 19450K .......... .......... .......... .......... .......... 50%  102M 1s\n",
      " 19500K .......... .......... .......... .......... .......... 50%  121M 1s\n",
      " 19550K .......... .......... .......... .......... .......... 50%  117M 1s\n",
      " 19600K .......... .......... .......... .......... .......... 50%  103M 1s\n",
      " 19650K .......... .......... .......... .......... .......... 50%  120M 1s\n",
      " 19700K .......... .......... .......... .......... .......... 50%  103M 1s\n",
      " 19750K .......... .......... .......... .......... .......... 50%  117M 1s\n",
      " 19800K .......... .......... .......... .......... .......... 51%  120M 1s\n",
      " 19850K .......... .......... .......... .......... .......... 51%  103M 1s\n",
      " 19900K .......... .......... .......... .......... .......... 51%  117M 1s\n",
      " 19950K .......... .......... .......... .......... .......... 51%  102M 1s\n",
      " 20000K .......... .......... .......... .......... .......... 51%  122M 1s\n",
      " 20050K .......... .......... .......... .......... .......... 51%  120M 1s\n",
      " 20100K .......... .......... .......... .......... .......... 51%  103M 1s\n",
      " 20150K .......... .......... .......... .......... .......... 51%  117M 1s\n",
      " 20200K .......... .......... .......... .......... .......... 52%  102M 1s\n",
      " 20250K .......... .......... .......... .......... .......... 52%  122M 1s\n",
      " 20300K .......... .......... .......... .......... .......... 52% 70,1M 1s\n",
      " 20350K .......... .......... .......... .......... .......... 52% 50,5M 1s\n",
      " 20400K .......... .......... .......... .......... .......... 52%  120M 1s\n",
      " 20450K .......... .......... .......... .......... .......... 52%  104M 1s\n",
      " 20500K .......... .......... .......... .......... .......... 52%  120M 1s\n",
      " 20550K .......... .......... .......... .......... .......... 53%  118M 1s\n",
      " 20600K .......... .......... .......... .......... .......... 53%  103M 1s\n",
      " 20650K .......... .......... .......... .......... .......... 53%  115M 1s\n",
      " 20700K .......... .......... .......... .......... .......... 53%  104M 1s\n",
      " 20750K .......... .......... .......... .......... .......... 53%  120M 1s\n",
      " 20800K .......... .......... .......... .......... .......... 53%  121M 1s\n",
      " 20850K .......... .......... .......... .......... .......... 53%  103M 1s\n",
      " 20900K .......... .......... .......... .......... .......... 53%  118M 1s\n",
      " 20950K .......... .......... .......... .......... .......... 54%  102M 1s\n",
      " 21000K .......... .......... .......... .......... .......... 54%  121M 1s\n",
      " 21050K .......... .......... .......... .......... .......... 54%  121M 1s\n",
      " 21100K .......... .......... .......... .......... .......... 54%  102M 1s\n",
      " 21150K .......... .......... .......... .......... .......... 54%  117M 1s\n",
      " 21200K .......... .......... .......... .......... .......... 54%  102M 1s\n",
      " 21250K .......... .......... .......... .......... .......... 54%  121M 1s\n",
      " 21300K .......... .......... .......... .......... .......... 54%  117M 1s\n",
      " 21350K .......... .......... .......... .......... .......... 55%  102M 1s\n",
      " 21400K .......... .......... .......... .......... .......... 55%  122M 1s\n",
      " 21450K .......... .......... .......... .......... .......... 55%  102M 1s\n",
      " 21500K .......... .......... .......... .......... .......... 55%  117M 1s\n",
      " 21550K .......... .......... .......... .......... .......... 55%  122M 1s\n",
      " 21600K .......... .......... .......... .......... .......... 55%  101M 1s\n",
      " 21650K .......... .......... .......... .......... .......... 55%  122M 1s\n",
      " 21700K .......... .......... .......... .......... .......... 55%  102M 1s\n",
      " 21750K .......... .......... .......... .......... .......... 56%  117M 1s\n",
      " 21800K .......... .......... .......... .......... .......... 56% 9,42M 1s\n",
      " 21850K .......... .......... .......... .......... .......... 56% 5,35M 1s\n",
      " 21900K .......... .......... .......... .......... .......... 56% 49,8M 1s\n",
      " 21950K .......... .......... .......... .......... .......... 56%  100M 1s\n",
      " 22000K .......... .......... .......... .......... .......... 56% 79,1M 1s\n",
      " 22050K .......... .......... .......... .......... .......... 56% 66,0M 1s\n",
      " 22100K .......... .......... .......... .......... .......... 56%  103M 1s\n",
      " 22150K .......... .......... .......... .......... .......... 57%  120M 1s\n",
      " 22200K .......... .......... .......... .......... .......... 57% 3,46M 1s\n",
      " 22250K .......... .......... .......... .......... .......... 57% 41,5M 1s\n",
      " 22300K .......... .......... .......... .......... .......... 57% 33,7M 1s\n",
      " 22350K .......... .......... .......... .......... .......... 57% 20,2M 1s\n",
      " 22400K .......... .......... .......... .......... .......... 57% 25,6M 1s\n",
      " 22450K .......... .......... .......... .......... .......... 57% 63,7M 1s\n",
      " 22500K .......... .......... .......... .......... .......... 58%  117M 1s\n",
      " 22550K .......... .......... .......... .......... .......... 58% 82,4M 1s\n",
      " 22600K .......... .......... .......... .......... .......... 58%  103M 1s\n",
      " 22650K .......... .......... .......... .......... .......... 58%  120M 1s\n",
      " 22700K .......... .......... .......... .......... .......... 58%  103M 1s\n",
      " 22750K .......... .......... .......... .......... .......... 58%  117M 1s\n",
      " 22800K .......... .......... .......... .......... .......... 58%  120M 1s\n",
      " 22850K .......... .......... .......... .......... .......... 58%  103M 1s\n",
      " 22900K .......... .......... .......... .......... .......... 59%  117M 1s\n",
      " 22950K .......... .......... .......... .......... .......... 59%  102M 1s\n",
      " 23000K .......... .......... .......... .......... .......... 59%  122M 1s\n",
      " 23050K .......... .......... .......... .......... .......... 59%  117M 1s\n",
      " 23100K .......... .......... .......... .......... .......... 59%  102M 1s\n",
      " 23150K .......... .......... .......... .......... .......... 59%  122M 1s\n",
      " 23200K .......... .......... .......... .......... .......... 59%  102M 1s\n",
      " 23250K .......... .......... .......... .......... .......... 59%  122M 1s\n",
      " 23300K .......... .......... .......... .......... .......... 60%  116M 1s\n",
      " 23350K .......... .......... .......... .......... .......... 60%  102M 1s\n",
      " 23400K .......... .......... .......... .......... .......... 60%  122M 1s\n",
      " 23450K .......... .......... .......... .......... .......... 60% 58,0M 1s\n",
      " 23500K .......... .......... .......... .......... .......... 60%  116M 1s\n",
      " 23550K .......... .......... .......... .......... .......... 60%  121M 1s\n",
      " 23600K .......... .......... .......... .......... .......... 60%  103M 1s\n",
      " 23650K .......... .......... .......... .......... .......... 60%  117M 1s\n",
      " 23700K .......... .......... .......... .......... .......... 61%  102M 1s\n",
      " 23750K .......... .......... .......... .......... .......... 61%  122M 1s\n",
      " 23800K .......... .......... .......... .......... .......... 61%  120M 1s\n",
      " 23850K .......... .......... .......... .......... .......... 61%  103M 1s\n",
      " 23900K .......... .......... .......... .......... .......... 61%  116M 1s\n",
      " 23950K .......... .......... .......... .......... .......... 61%  102M 1s\n",
      " 24000K .......... .......... .......... .......... .......... 61%  122M 1s\n",
      " 24050K .......... .......... .......... .......... .......... 62%  117M 1s\n",
      " 24100K .......... .......... .......... .......... .......... 62%  102M 1s\n",
      " 24150K .......... .......... .......... .......... .......... 62%  121M 1s\n",
      " 24200K .......... .......... .......... .......... .......... 62%  102M 1s\n",
      " 24250K .......... .......... .......... .......... .......... 62%  121M 1s\n",
      " 24300K .......... .......... .......... .......... .......... 62%  118M 1s\n",
      " 24350K .......... .......... .......... .......... .......... 62%  102M 1s\n",
      " 24400K .......... .......... .......... .......... .......... 62%  122M 1s\n",
      " 24450K .......... .......... .......... .......... .......... 63%  102M 1s\n",
      " 24500K .......... .......... .......... .......... .......... 63%  118M 1s\n",
      " 24550K .......... .......... .......... .......... .......... 63%  120M 1s\n",
      " 24600K .......... .......... .......... .......... .......... 63%  103M 1s\n",
      " 24650K .......... .......... .......... .......... .......... 63%  117M 1s\n",
      " 24700K .......... .......... .......... .......... .......... 63%  102M 0s\n",
      " 24750K .......... .......... .......... .......... .......... 63%  122M 0s\n",
      " 24800K .......... .......... .......... .......... .......... 63%  120M 0s\n",
      " 24850K .......... .......... .......... .......... .......... 64% 6,64M 0s\n",
      " 24900K .......... .......... .......... .......... .......... 64% 8,81M 0s\n",
      " 24950K .......... .......... .......... .......... .......... 64% 17,1M 0s\n",
      " 25000K .......... .......... .......... .......... .......... 64%  122M 0s\n",
      " 25050K .......... .......... .......... .......... .......... 64%  118M 0s\n",
      " 25100K .......... .......... .......... .......... .......... 64% 92,6M 0s\n",
      " 25150K .......... .......... .......... .......... .......... 64%  122M 0s\n",
      " 25200K .......... .......... .......... .......... .......... 64% 44,1M 0s\n",
      " 25250K .......... .......... .......... .......... .......... 65% 3,54M 0s\n",
      " 25300K .......... .......... .......... .......... .......... 65% 42,8M 0s\n",
      " 25350K .......... .......... .......... .......... .......... 65% 40,7M 0s\n",
      " 25400K .......... .......... .......... .......... .......... 65% 49,7M 0s\n",
      " 25450K .......... .......... .......... .......... .......... 65% 12,1M 0s\n",
      " 25500K .......... .......... .......... .......... .......... 65% 68,3M 0s\n",
      " 25550K .......... .......... .......... .......... .......... 65%  123M 0s\n",
      " 25600K .......... .......... .......... .......... .......... 66%  103M 0s\n",
      " 25650K .......... .......... .......... .......... .......... 66%  117M 0s\n",
      " 25700K .......... .......... .......... .......... .......... 66%  102M 0s\n",
      " 25750K .......... .......... .......... .......... .......... 66%  121M 0s\n",
      " 25800K .......... .......... .......... .......... .......... 66%  118M 0s\n",
      " 25850K .......... .......... .......... .......... .......... 66%  102M 0s\n",
      " 25900K .......... .......... .......... .......... .......... 66%  122M 0s\n",
      " 25950K .......... .......... .......... .......... .......... 66%  102M 0s\n",
      " 26000K .......... .......... .......... .......... .......... 67%  122M 0s\n",
      " 26050K .......... .......... .......... .......... .......... 67%  117M 0s\n",
      " 26100K .......... .......... .......... .......... .......... 67%  102M 0s\n",
      " 26150K .......... .......... .......... .......... .......... 67%  122M 0s\n",
      " 26200K .......... .......... .......... .......... .......... 67%  102M 0s\n",
      " 26250K .......... .......... .......... .......... .......... 67%  117M 0s\n",
      " 26300K .......... .......... .......... .......... .......... 67%  122M 0s\n",
      " 26350K .......... .......... .......... .......... .......... 67%  102M 0s\n",
      " 26400K .......... .......... .......... .......... .......... 68%  122M 0s\n",
      " 26450K .......... .......... .......... .......... .......... 68%  102M 0s\n",
      " 26500K .......... .......... .......... .......... .......... 68%  118M 0s\n",
      " 26550K .......... .......... .......... .......... .......... 68%  120M 0s\n",
      " 26600K .......... .......... .......... .......... .......... 68%  102M 0s\n",
      " 26650K .......... .......... .......... .......... .......... 68%  118M 0s\n",
      " 26700K .......... .......... .......... .......... .......... 68%  102M 0s\n",
      " 26750K .......... .......... .......... .......... .......... 68%  122M 0s\n",
      " 26800K .......... .......... .......... .......... .......... 69%  117M 0s\n",
      " 26850K .......... .......... .......... .......... .......... 69%  102M 0s\n",
      " 26900K .......... .......... .......... .......... .......... 69%  122M 0s\n",
      " 26950K .......... .......... .......... .......... .......... 69%  102M 0s\n",
      " 27000K .......... .......... .......... .......... .......... 69%  122M 0s\n",
      " 27050K .......... .......... .......... .......... .......... 69%  117M 0s\n",
      " 27100K .......... .......... .......... .......... .......... 69%  102M 0s\n",
      " 27150K .......... .......... .......... .......... .......... 69%  121M 0s\n",
      " 27200K .......... .......... .......... .......... .......... 70%  102M 0s\n",
      " 27250K .......... .......... .......... .......... .......... 70%  117M 0s\n",
      " 27300K .......... .......... .......... .......... .......... 70%  120M 0s\n",
      " 27350K .......... .......... .......... .......... .......... 70%  103M 0s\n",
      " 27400K .......... .......... .......... .......... .......... 70%  117M 0s\n",
      " 27450K .......... .......... .......... .......... .......... 70%  102M 0s\n",
      " 27500K .......... .......... .......... .......... .......... 70%  122M 0s\n",
      " 27550K .......... .......... .......... .......... .......... 71%  120M 0s\n",
      " 27600K .......... .......... .......... .......... .......... 71%  103M 0s\n",
      " 27650K .......... .......... .......... .......... .......... 71%  109M 0s\n",
      " 27700K .......... .......... .......... .......... .......... 71%  102M 0s\n",
      " 27750K .......... .......... .......... .......... .......... 71%  122M 0s\n",
      " 27800K .......... .......... .......... .......... .......... 71%  116M 0s\n",
      " 27850K .......... .......... .......... .......... .......... 71% 13,1M 0s\n",
      " 27900K .......... .......... .......... .......... .......... 71% 16,9M 0s\n",
      " 27950K .......... .......... .......... .......... .......... 72% 7,92M 0s\n",
      " 28000K .......... .......... .......... .......... .......... 72% 20,0M 0s\n",
      " 28050K .......... .......... .......... .......... .......... 72% 31,7M 0s\n",
      " 28100K .......... .......... .......... .......... .......... 72%  103M 0s\n",
      " 28150K .......... .......... .......... .......... .......... 72%  120M 0s\n",
      " 28200K .......... .......... .......... .......... .......... 72%  103M 0s\n",
      " 28250K .......... .......... .......... .......... .......... 72%  117M 0s\n",
      " 28300K .......... .......... .......... .......... .......... 72% 3,97M 0s\n",
      " 28350K .......... .......... .......... .......... .......... 73% 13,5M 0s\n",
      " 28400K .......... .......... .......... .......... .......... 73%  117M 0s\n",
      " 28450K .......... .......... .......... .......... .......... 73% 48,6M 0s\n",
      " 28500K .......... .......... .......... .......... .......... 73% 11,1M 0s\n",
      " 28550K .......... .......... .......... .......... .......... 73% 48,6M 0s\n",
      " 28600K .......... .......... .......... .......... .......... 73%  103M 0s\n",
      " 28650K .......... .......... .......... .......... .......... 73%  117M 0s\n",
      " 28700K .......... .......... .......... .......... .......... 73%  102M 0s\n",
      " 28750K .......... .......... .......... .......... .......... 74%  121M 0s\n",
      " 28800K .......... .......... .......... .......... .......... 74%  118M 0s\n",
      " 28850K .......... .......... .......... .......... .......... 74%  102M 0s\n",
      " 28900K .......... .......... .......... .......... .......... 74%  122M 0s\n",
      " 28950K .......... .......... .......... .......... .......... 74%  102M 0s\n",
      " 29000K .......... .......... .......... .......... .......... 74%  117M 0s\n",
      " 29050K .......... .......... .......... .......... .......... 74%  122M 0s\n",
      " 29100K .......... .......... .......... .......... .......... 75%  102M 0s\n",
      " 29150K .......... .......... .......... .......... .......... 75%  122M 0s\n",
      " 29200K .......... .......... .......... .......... .......... 75%  102M 0s\n",
      " 29250K .......... .......... .......... .......... .......... 75%  117M 0s\n",
      " 29300K .......... .......... .......... .......... .......... 75%  122M 0s\n",
      " 29350K .......... .......... .......... .......... .......... 75%  102M 0s\n",
      " 29400K .......... .......... .......... .......... .......... 75%  117M 0s\n",
      " 29450K .......... .......... .......... .......... .......... 75%  103M 0s\n",
      " 29500K .......... .......... .......... .......... .......... 76%  120M 0s\n",
      " 29550K .......... .......... .......... .......... .......... 76%  117M 0s\n",
      " 29600K .......... .......... .......... .......... .......... 76%  103M 0s\n",
      " 29650K .......... .......... .......... .......... .......... 76%  120M 0s\n",
      " 29700K .......... .......... .......... .......... .......... 76%  103M 0s\n",
      " 29750K .......... .......... .......... .......... .......... 76%  120M 0s\n",
      " 29800K .......... .......... .......... .......... .......... 76%  117M 0s\n",
      " 29850K .......... .......... .......... .......... .......... 76%  103M 0s\n",
      " 29900K .......... .......... .......... .......... .......... 77%  120M 0s\n",
      " 29950K .......... .......... .......... .......... .......... 77%  103M 0s\n",
      " 30000K .......... .......... .......... .......... .......... 77%  117M 0s\n",
      " 30050K .......... .......... .......... .......... .......... 77%  120M 0s\n",
      " 30100K .......... .......... .......... .......... .......... 77%  103M 0s\n",
      " 30150K .......... .......... .......... .......... .......... 77%  120M 0s\n",
      " 30200K .......... .......... .......... .......... .......... 77%  103M 0s\n",
      " 30250K .......... .......... .......... .......... .......... 77%  116M 0s\n",
      " 30300K .......... .......... .......... .......... .......... 78%  121M 0s\n",
      " 30350K .......... .......... .......... .......... .......... 78%  103M 0s\n",
      " 30400K .......... .......... .......... .......... .......... 78%  117M 0s\n",
      " 30450K .......... .......... .......... .......... .......... 78%  102M 0s\n",
      " 30500K .......... .......... .......... .......... .......... 78%  121M 0s\n",
      " 30550K .......... .......... .......... .......... .......... 78%  118M 0s\n",
      " 30600K .......... .......... .......... .......... .......... 78%  102M 0s\n",
      " 30650K .......... .......... .......... .......... .......... 78%  120M 0s\n",
      " 30700K .......... .......... .......... .......... .......... 79%  103M 0s\n",
      " 30750K .......... .......... .......... .......... .......... 79%  120M 0s\n",
      " 30800K .......... .......... .......... .......... .......... 79%  117M 0s\n",
      " 30850K .......... .......... .......... .......... .......... 79%  103M 0s\n",
      " 30900K .......... .......... .......... .......... .......... 79% 18,7M 0s\n",
      " 30950K .......... .......... .......... .......... .......... 79% 21,1M 0s\n",
      " 31000K .......... .......... .......... .......... .......... 79% 7,48M 0s\n",
      " 31050K .......... .......... .......... .......... .......... 80% 16,2M 0s\n",
      " 31100K .......... .......... .......... .......... .......... 80% 22,6M 0s\n",
      " 31150K .......... .......... .......... .......... .......... 80%  117M 0s\n",
      " 31200K .......... .......... .......... .......... .......... 80% 72,5M 0s\n",
      " 31250K .......... .......... .......... .......... .......... 80%  121M 0s\n",
      " 31300K .......... .......... .......... .......... .......... 80%  121M 0s\n",
      " 31350K .......... .......... .......... .......... .......... 80% 4,96M 0s\n",
      " 31400K .......... .......... .......... .......... .......... 80% 8,36M 0s\n",
      " 31450K .......... .......... .......... .......... .......... 81% 39,8M 0s\n",
      " 31500K .......... .......... .......... .......... .......... 81% 27,0M 0s\n",
      " 31550K .......... .......... .......... .......... .......... 81%  120M 0s\n",
      " 31600K .......... .......... .......... .......... .......... 81% 15,9M 0s\n",
      " 31650K .......... .......... .......... .......... .......... 81% 58,3M 0s\n",
      " 31700K .......... .......... .......... .......... .......... 81% 55,0M 0s\n",
      " 31750K .......... .......... .......... .......... .......... 81%  122M 0s\n",
      " 31800K .......... .......... .......... .......... .......... 81%  117M 0s\n",
      " 31850K .......... .......... .......... .......... .......... 82%  102M 0s\n",
      " 31900K .......... .......... .......... .......... .......... 82%  122M 0s\n",
      " 31950K .......... .......... .......... .......... .......... 82%  102M 0s\n",
      " 32000K .......... .......... .......... .......... .......... 82%  117M 0s\n",
      " 32050K .......... .......... .......... .......... .......... 82%  122M 0s\n",
      " 32100K .......... .......... .......... .......... .......... 82%  102M 0s\n",
      " 32150K .......... .......... .......... .......... .......... 82%  116M 0s\n",
      " 32200K .......... .......... .......... .......... .......... 82%  102M 0s\n",
      " 32250K .......... .......... .......... .......... .......... 83%  122M 0s\n",
      " 32300K .......... .......... .......... .......... .......... 83%  122M 0s\n",
      " 32350K .......... .......... .......... .......... .......... 83%  102M 0s\n",
      " 32400K .......... .......... .......... .......... .......... 83%  117M 0s\n",
      " 32450K .......... .......... .......... .......... .......... 83%  103M 0s\n",
      " 32500K .......... .......... .......... .......... .......... 83%  120M 0s\n",
      " 32550K .......... .......... .......... .......... .......... 83%  117M 0s\n",
      " 32600K .......... .......... .......... .......... .......... 84%  103M 0s\n",
      " 32650K .......... .......... .......... .......... .......... 84%  121M 0s\n",
      " 32700K .......... .......... .......... .......... .......... 84%  102M 0s\n",
      " 32750K .......... .......... .......... .......... .......... 84%  117M 0s\n",
      " 32800K .......... .......... .......... .......... .......... 84%  122M 0s\n",
      " 32850K .......... .......... .......... .......... .......... 84%  101M 0s\n",
      " 32900K .......... .......... .......... .......... .......... 84%  122M 0s\n",
      " 32950K .......... .......... .......... .......... .......... 84%  102M 0s\n",
      " 33000K .......... .......... .......... .......... .......... 85%  117M 0s\n",
      " 33050K .......... .......... .......... .......... .......... 85%  121M 0s\n",
      " 33100K .......... .......... .......... .......... .......... 85%  101M 0s\n",
      " 33150K .......... .......... .......... .......... .......... 85%  120M 0s\n",
      " 33200K .......... .......... .......... .......... .......... 85%  102M 0s\n",
      " 33250K .......... .......... .......... .......... .......... 85%  121M 0s\n",
      " 33300K .......... .......... .......... .......... .......... 85%  118M 0s\n",
      " 33350K .......... .......... .......... .......... .......... 85%  103M 0s\n",
      " 33400K .......... .......... .......... .......... .......... 86%  120M 0s\n",
      " 33450K .......... .......... .......... .......... .......... 86%  102M 0s\n",
      " 33500K .......... .......... .......... .......... .......... 86%  121M 0s\n",
      " 33550K .......... .......... .......... .......... .......... 86%  118M 0s\n",
      " 33600K .......... .......... .......... .......... .......... 86%  103M 0s\n",
      " 33650K .......... .......... .......... .......... .......... 86%  120M 0s\n",
      " 33700K .......... .......... .......... .......... .......... 86%  103M 0s\n",
      " 33750K .......... .......... .......... .......... .......... 86%  117M 0s\n",
      " 33800K .......... .......... .......... .......... .......... 87%  120M 0s\n",
      " 33850K .......... .......... .......... .......... .......... 87%  103M 0s\n",
      " 33900K .......... .......... .......... .......... .......... 87% 35,6M 0s\n",
      " 33950K .......... .......... .......... .......... .......... 87% 30,2M 0s\n",
      " 34000K .......... .......... .......... .......... .......... 87% 41,3M 0s\n",
      " 34050K .......... .......... .......... .......... .......... 87% 8,49M 0s\n",
      " 34100K .......... .......... .......... .......... .......... 87% 12,2M 0s\n",
      " 34150K .......... .......... .......... .......... .......... 88% 20,7M 0s\n",
      " 34200K .......... .......... .......... .......... .......... 88% 94,0M 0s\n",
      " 34250K .......... .......... .......... .......... .......... 88%  120M 0s\n",
      " 34300K .......... .......... .......... .......... .......... 88% 83,0M 0s\n",
      " 34350K .......... .......... .......... .......... .......... 88% 6,86M 0s\n",
      " 34400K .......... .......... .......... .......... .......... 88% 17,8M 0s\n",
      " 34450K .......... .......... .......... .......... .......... 88% 8,91M 0s\n",
      " 34500K .......... .......... .......... .......... .......... 88% 29,1M 0s\n",
      " 34550K .......... .......... .......... .......... .......... 89% 45,7M 0s\n",
      " 34600K .......... .......... .......... .......... .......... 89% 27,2M 0s\n",
      " 34650K .......... .......... .......... .......... .......... 89% 14,3M 0s\n",
      " 34700K .......... .......... .......... .......... .......... 89% 50,9M 0s\n",
      " 34750K .......... .......... .......... .......... .......... 89%  117M 0s\n",
      " 34800K .......... .......... .......... .......... .......... 89%  120M 0s\n",
      " 34850K .......... .......... .......... .......... .......... 89%  103M 0s\n",
      " 34900K .......... .......... .......... .......... .......... 89%  117M 0s\n",
      " 34950K .......... .......... .......... .......... .......... 90%  102M 0s\n",
      " 35000K .......... .......... .......... .......... .......... 90%  122M 0s\n",
      " 35050K .......... .......... .......... .......... .......... 90%  120M 0s\n",
      " 35100K .......... .......... .......... .......... .......... 90%  103M 0s\n",
      " 35150K .......... .......... .......... .......... .......... 90%  116M 0s\n",
      " 35200K .......... .......... .......... .......... .......... 90%  102M 0s\n",
      " 35250K .......... .......... .......... .......... .......... 90%  120M 0s\n",
      " 35300K .......... .......... .......... .......... .......... 90%  117M 0s\n",
      " 35350K .......... .......... .......... .......... .......... 91%  103M 0s\n",
      " 35400K .......... .......... .......... .......... .......... 91%  121M 0s\n",
      " 35450K .......... .......... .......... .......... .......... 91%  102M 0s\n",
      " 35500K .......... .......... .......... .......... .......... 91%  121M 0s\n",
      " 35550K .......... .......... .......... .......... .......... 91%  117M 0s\n",
      " 35600K .......... .......... .......... .......... .......... 91%  101M 0s\n",
      " 35650K .......... .......... .......... .......... .......... 91%  123M 0s\n",
      " 35700K .......... .......... .......... .......... .......... 91%  102M 0s\n",
      " 35750K .......... .......... .......... .......... .......... 92%  118M 0s\n",
      " 35800K .......... .......... .......... .......... .......... 92%  121M 0s\n",
      " 35850K .......... .......... .......... .......... .......... 92%  102M 0s\n",
      " 35900K .......... .......... .......... .......... .......... 92%  117M 0s\n",
      " 35950K .......... .......... .......... .......... .......... 92%  103M 0s\n",
      " 36000K .......... .......... .......... .......... .......... 92%  120M 0s\n",
      " 36050K .......... .......... .......... .......... .......... 92%  121M 0s\n",
      " 36100K .......... .......... .......... .......... .......... 93%  103M 0s\n",
      " 36150K .......... .......... .......... .......... .......... 93%  117M 0s\n",
      " 36200K .......... .......... .......... .......... .......... 93%  102M 0s\n",
      " 36250K .......... .......... .......... .......... .......... 93%  121M 0s\n",
      " 36300K .......... .......... .......... .......... .......... 93%  117M 0s\n",
      " 36350K .......... .......... .......... .......... .......... 93%  103M 0s\n",
      " 36400K .......... .......... .......... .......... .......... 93%  120M 0s\n",
      " 36450K .......... .......... .......... .......... .......... 93%  102M 0s\n",
      " 36500K .......... .......... .......... .......... .......... 94%  117M 0s\n",
      " 36550K .......... .......... .......... .......... .......... 94%  121M 0s\n",
      " 36600K .......... .......... .......... .......... .......... 94%  102M 0s\n",
      " 36650K .......... .......... .......... .......... .......... 94%  120M 0s\n",
      " 36700K .......... .......... .......... .......... .......... 94%  102M 0s\n",
      " 36750K .......... .......... .......... .......... .......... 94%  118M 0s\n",
      " 36800K .......... .......... .......... .......... .......... 94%  120M 0s\n",
      " 36850K .......... .......... .......... .......... .......... 94%  103M 0s\n",
      " 36900K .......... .......... .......... .......... .......... 95%  117M 0s\n",
      " 36950K .......... .......... .......... .......... .......... 95%  103M 0s\n",
      " 37000K .......... .......... .......... .......... .......... 95% 45,6M 0s\n",
      " 37050K .......... .......... .......... .......... .......... 95% 10,6M 0s\n",
      " 37100K .......... .......... .......... .......... .......... 95% 17,3M 0s\n",
      " 37150K .......... .......... .......... .......... .......... 95% 12,7M 0s\n",
      " 37200K .......... .......... .......... .......... .......... 95% 15,5M 0s\n",
      " 37250K .......... .......... .......... .......... .......... 95% 44,4M 0s\n",
      " 37300K .......... .......... .......... .......... .......... 96%  117M 0s\n",
      " 37350K .......... .......... .......... .......... .......... 96%  103M 0s\n",
      " 37400K .......... .......... .......... .......... .......... 96% 8,15M 0s\n",
      " 37450K .......... .......... .......... .......... .......... 96% 15,6M 0s\n",
      " 37500K .......... .......... .......... .......... .......... 96% 8,73M 0s\n",
      " 37550K .......... .......... .......... .......... .......... 96% 25,1M 0s\n",
      " 37600K .......... .......... .......... .......... .......... 96% 31,6M 0s\n",
      " 37650K .......... .......... .......... .......... .......... 97% 39,6M 0s\n",
      " 37700K .......... .......... .......... .......... .......... 97% 13,6M 0s\n",
      " 37750K .......... .......... .......... .......... .......... 97% 29,6M 0s\n",
      " 37800K .......... .......... .......... .......... .......... 97%  123M 0s\n",
      " 37850K .......... .......... .......... .......... .......... 97%  101M 0s\n",
      " 37900K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 37950K .......... .......... .......... .......... .......... 97%  104M 0s\n",
      " 38000K .......... .......... .......... .......... .......... 97%  120M 0s\n",
      " 38050K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 38100K .......... .......... .......... .......... .......... 98%  103M 0s\n",
      " 38150K .......... .......... .......... .......... .......... 98%  119M 0s\n",
      " 38200K .......... .......... .......... .......... .......... 98%  104M 0s\n",
      " 38250K .......... .......... .......... .......... .......... 98%  120M 0s\n",
      " 38300K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 38350K .......... .......... .......... .......... .......... 98%  103M 0s\n",
      " 38400K .......... .......... .......... .......... .......... 98%  120M 0s\n",
      " 38450K .......... .......... .......... .......... .......... 99%  103M 0s\n",
      " 38500K .......... .......... .......... .......... .......... 99%  116M 0s\n",
      " 38550K .......... .......... .......... .......... .......... 99%  122M 0s\n",
      " 38600K .......... .......... .......... .......... .......... 99%  102M 0s\n",
      " 38650K .......... .......... .......... .......... .......... 99%  117M 0s\n",
      " 38700K .......... .......... .......... .......... .......... 99%  103M 0s\n",
      " 38750K .......... .......... .......... .......... .......... 99%  120M 0s\n",
      " 38800K .......... .......... .......... .......... .......... 99%  122M 0s\n",
      " 38850K .......... ..                                         100%  126M=1,2s\n",
      "\n",
      "2023-08-29 10:24:43 (32,3 MB/s) - ‘data/reference/genome.fa.gz’ guardado [39795542/39795542]\n",
      "\n",
      "[Tue Aug 29 10:24:43 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T102440.782634.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:24:45 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR696683_chr12_1.fastq.gz, data/raw/ERR696683_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR696683_chr12_1_fastp.fastq.gz, data/processed/ERR696683_chr12_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR696683_chr12_1_fastp.fastq.gz, data/processed/ERR696683_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1814551\n",
      "total bases: 137166702\n",
      "Q20 bases: 136196362(99.2926%)\n",
      "Q30 bases: 132450329(96.5616%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1814551\n",
      "total bases: 136812703\n",
      "Q20 bases: 135133064(98.7723%)\n",
      "Q30 bases: 129660948(94.7726%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1124306\n",
      "total bases: 85390403\n",
      "Q20 bases: 85056739(99.6092%)\n",
      "Q30 bases: 83576597(97.8759%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1124306\n",
      "total bases: 85356933\n",
      "Q20 bases: 84768167(99.3102%)\n",
      "Q30 bases: 82580401(96.7472%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2248612\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1380490\n",
      "reads with adapter trimmed: 812\n",
      "bases trimmed due to adapters: 18871\n",
      "\n",
      "Duplication rate: 5.7663%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR696683_chr12_1.fastq.gz -I data/raw/ERR696683_chr12_2.fastq.gz -o data/processed/ERR696683_chr12_1_fastp.fastq.gz -O data/processed/ERR696683_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Tue Aug 29 10:24:50 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:24:50 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR696683_chr12_1_fastp.fastq.gz, data/processed/ERR696683_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR696683_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR696683_chr12_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR696683_chr12_1_fastp.fastq.gz, data/processed/ERR696683_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 10:25:01 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:25:01 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR696683_chr12_1.fastq.gz, data/raw/ERR696683_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR696683_chr12_1_fastqc.html, results/fastqc_result/ERR696683_chr12_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR696683_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 10:25:16 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T102443.917154.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:25:17 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753368_chr12_1.fastq.gz, data/raw/ERR753368_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753368_chr12_1_fastp.fastq.gz, data/processed/ERR753368_chr12_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753368_chr12_1_fastp.fastq.gz, data/processed/ERR753368_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1897940\n",
      "total bases: 144109638\n",
      "Q20 bases: 143542594(99.6065%)\n",
      "Q30 bases: 139301752(96.6637%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1897940\n",
      "total bases: 143961401\n",
      "Q20 bases: 142692339(99.1185%)\n",
      "Q30 bases: 136129314(94.5596%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1632596\n",
      "total bases: 124070769\n",
      "Q20 bases: 123855197(99.8263%)\n",
      "Q30 bases: 121763665(98.1405%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1632596\n",
      "total bases: 124060324\n",
      "Q20 bases: 123450972(99.5088%)\n",
      "Q30 bases: 119750755(96.5262%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3265192\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 530688\n",
      "reads with adapter trimmed: 1148\n",
      "bases trimmed due to adapters: 26680\n",
      "\n",
      "Duplication rate: 2.0326%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753368_chr12_1.fastq.gz -I data/raw/ERR753368_chr12_2.fastq.gz -o data/processed/ERR753368_chr12_1_fastp.fastq.gz -O data/processed/ERR753368_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Tue Aug 29 10:25:24 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:25:24 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753368_chr12_1_fastp.fastq.gz, data/processed/ERR753368_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753368_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753368_chr12_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753368_chr12_1_fastp.fastq.gz, data/processed/ERR753368_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 10:25:36 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:25:36 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753368_chr12_1.fastq.gz, data/raw/ERR753368_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753368_chr12_1_fastqc.html, results/fastqc_result/ERR753368_chr12_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753368_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753368_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 10:25:51 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T102516.564783.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:25:52 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753369_chr12_1.fastq.gz, data/raw/ERR753369_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753369_chr12_1_fastqc.html, results/fastqc_result/ERR753369_chr12_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753369_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753369_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 10:26:07 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:26:07 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753369_chr12_1.fastq.gz, data/raw/ERR753369_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753369_chr12_1_fastp.fastq.gz, data/processed/ERR753369_chr12_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753369_chr12_2_fastp.fastq.gz, data/processed/ERR753369_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2053990\n",
      "total bases: 155951416\n",
      "Q20 bases: 155309500(99.5884%)\n",
      "Q30 bases: 150604540(96.5714%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2053990\n",
      "total bases: 155798651\n",
      "Q20 bases: 154421725(99.1162%)\n",
      "Q30 bases: 147287629(94.5372%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1766738\n",
      "total bases: 134264590\n",
      "Q20 bases: 134014650(99.8138%)\n",
      "Q30 bases: 131659977(98.0601%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1766738\n",
      "total bases: 134253748\n",
      "Q20 bases: 133591459(99.5067%)\n",
      "Q30 bases: 129560868(96.5045%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3533476\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 574504\n",
      "reads with adapter trimmed: 1156\n",
      "bases trimmed due to adapters: 27060\n",
      "\n",
      "Duplication rate: 2.36761%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753369_chr12_1.fastq.gz -I data/raw/ERR753369_chr12_2.fastq.gz -o data/processed/ERR753369_chr12_1_fastp.fastq.gz -O data/processed/ERR753369_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Tue Aug 29 10:26:14 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:26:14 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753369_chr12_1_fastp.fastq.gz, data/processed/ERR753369_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753369_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753369_chr12_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753369_chr12_2_fastp.fastq.gz, data/processed/ERR753369_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 10:26:28 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T102551.801953.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:26:29 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753370_chr12_1.fastq.gz, data/raw/ERR753370_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753370_chr12_1_fastqc.html, results/fastqc_result/ERR753370_chr12_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753370_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753370_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 10:26:52 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:26:52 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753370_chr12_1.fastq.gz, data/raw/ERR753370_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753370_chr12_1_fastp.fastq.gz, data/processed/ERR753370_chr12_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753370_chr12_1_fastp.fastq.gz, data/processed/ERR753370_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2549417\n",
      "total bases: 256637892\n",
      "Q20 bases: 254886215(99.3175%)\n",
      "Q30 bases: 247502750(96.4405%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2549417\n",
      "total bases: 256574888\n",
      "Q20 bases: 254137882(99.0502%)\n",
      "Q30 bases: 244519685(95.3015%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 2548864\n",
      "total bases: 255580768\n",
      "Q20 bases: 253929976(99.3541%)\n",
      "Q30 bases: 246895663(96.6018%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 2548864\n",
      "total bases: 254756297\n",
      "Q20 bases: 252541907(99.1308%)\n",
      "Q30 bases: 243527730(95.5924%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 5097728\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1106\n",
      "reads with adapter trimmed: 13432\n",
      "bases trimmed due to adapters: 104382\n",
      "\n",
      "Duplication rate: 4.35009%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 160\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753370_chr12_1.fastq.gz -I data/raw/ERR753370_chr12_2.fastq.gz -o data/processed/ERR753370_chr12_1_fastp.fastq.gz -O data/processed/ERR753370_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 12 seconds\n",
      "[Tue Aug 29 10:27:04 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:27:04 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753370_chr12_1_fastp.fastq.gz, data/processed/ERR753370_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753370_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753370_chr12_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753370_chr12_1_fastp.fastq.gz, data/processed/ERR753370_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 10:27:27 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T102628.552565.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:27:28 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753371_chr12_1.fastq.gz, data/raw/ERR753371_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753371_chr12_1_fastqc.html, results/fastqc_result/ERR753371_chr12_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753371_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753371_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 10:27:42 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:27:42 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753371_chr12_1.fastq.gz, data/raw/ERR753371_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753371_chr12_1_fastp.fastq.gz, data/processed/ERR753371_chr12_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753371_chr12_2_fastp.fastq.gz, data/processed/ERR753371_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1690709\n",
      "total bases: 127390591\n",
      "Q20 bases: 126242409(99.0987%)\n",
      "Q30 bases: 121643046(95.4882%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1690709\n",
      "total bases: 127070345\n",
      "Q20 bases: 125554387(98.807%)\n",
      "Q30 bases: 121137819(95.3313%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 872488\n",
      "total bases: 66228440\n",
      "Q20 bases: 65905625(99.5126%)\n",
      "Q30 bases: 64262329(97.0313%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 872488\n",
      "total bases: 66192098\n",
      "Q20 bases: 65729465(99.3011%)\n",
      "Q30 bases: 64277244(97.1071%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1744976\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1636442\n",
      "reads with adapter trimmed: 710\n",
      "bases trimmed due to adapters: 15017\n",
      "\n",
      "Duplication rate: 22.0262%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 116\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753371_chr12_1.fastq.gz -I data/raw/ERR753371_chr12_2.fastq.gz -o data/processed/ERR753371_chr12_1_fastp.fastq.gz -O data/processed/ERR753371_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Tue Aug 29 10:27:47 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:27:47 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753371_chr12_1_fastp.fastq.gz, data/processed/ERR753371_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753371_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753371_chr12_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753371_chr12_2_fastp.fastq.gz, data/processed/ERR753371_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 10:27:56 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T102727.566867.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:27:57 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753372_chr12_1.fastq.gz, data/raw/ERR753372_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753372_chr12_1_fastp.fastq.gz, data/processed/ERR753372_chr12_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753372_chr12_1_fastp.fastq.gz, data/processed/ERR753372_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2028383\n",
      "total bases: 154006268\n",
      "Q20 bases: 153393425(99.6021%)\n",
      "Q30 bases: 148774276(96.6027%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2028383\n",
      "total bases: 153852859\n",
      "Q20 bases: 152494427(99.1171%)\n",
      "Q30 bases: 145498335(94.5698%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1738874\n",
      "total bases: 132147137\n",
      "Q20 bases: 131915460(99.8247%)\n",
      "Q30 bases: 129641227(98.1037%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1738874\n",
      "total bases: 132136009\n",
      "Q20 bases: 131490557(99.5115%)\n",
      "Q30 bases: 127584864(96.5557%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3477748\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 579018\n",
      "reads with adapter trimmed: 1250\n",
      "bases trimmed due to adapters: 26990\n",
      "\n",
      "Duplication rate: 2.0027%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753372_chr12_1.fastq.gz -I data/raw/ERR753372_chr12_2.fastq.gz -o data/processed/ERR753372_chr12_1_fastp.fastq.gz -O data/processed/ERR753372_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Tue Aug 29 10:28:04 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:28:04 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753372_chr12_1_fastp.fastq.gz, data/processed/ERR753372_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753372_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753372_chr12_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753372_chr12_1_fastp.fastq.gz, data/processed/ERR753372_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 10:28:19 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:28:19 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753372_chr12_1.fastq.gz, data/raw/ERR753372_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753372_chr12_1_fastqc.html, results/fastqc_result/ERR753372_chr12_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753372_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753372_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 10:28:34 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T102756.623625.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:28:35 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753373_chr12_1.fastq.gz, data/raw/ERR753373_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753373_chr12_1_fastp.fastq.gz, data/processed/ERR753373_chr12_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753373_chr12_1_fastp.fastq.gz, data/processed/ERR753373_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1413533\n",
      "total bases: 135329456\n",
      "Q20 bases: 134237138(99.1928%)\n",
      "Q30 bases: 125040132(92.3968%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1413533\n",
      "total bases: 132444072\n",
      "Q20 bases: 130991936(98.9036%)\n",
      "Q30 bases: 122419166(92.4308%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1378941\n",
      "total bases: 129926291\n",
      "Q20 bases: 129164081(99.4134%)\n",
      "Q30 bases: 121734716(93.6952%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1378941\n",
      "total bases: 125531307\n",
      "Q20 bases: 124923753(99.516%)\n",
      "Q30 bases: 118402277(94.3209%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2757882\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 69184\n",
      "reads with adapter trimmed: 1192\n",
      "bases trimmed due to adapters: 41078\n",
      "\n",
      "Duplication rate: 1.84862%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 153\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753373_chr12_1.fastq.gz -I data/raw/ERR753373_chr12_2.fastq.gz -o data/processed/ERR753373_chr12_1_fastp.fastq.gz -O data/processed/ERR753373_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Tue Aug 29 10:28:41 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:28:41 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753373_chr12_1_fastp.fastq.gz, data/processed/ERR753373_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753373_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753373_chr12_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753373_chr12_1_fastp.fastq.gz, data/processed/ERR753373_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 10:28:54 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:28:54 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753373_chr12_1.fastq.gz, data/raw/ERR753373_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753373_chr12_1_fastqc.html, results/fastqc_result/ERR753373_chr12_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753373_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753373_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 10:29:07 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T102834.323960.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:29:08 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753374_chr12_1.fastq.gz, data/raw/ERR753374_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753374_chr12_1_fastp.fastq.gz, data/processed/ERR753374_chr12_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753374_chr12_1_fastp.fastq.gz, data/processed/ERR753374_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 849844\n",
      "total bases: 81288111\n",
      "Q20 bases: 80518260(99.0529%)\n",
      "Q30 bases: 73547935(90.4781%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 849844\n",
      "total bases: 79522059\n",
      "Q20 bases: 78682606(98.9444%)\n",
      "Q30 bases: 72336674(90.9643%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 822905\n",
      "total bases: 76905295\n",
      "Q20 bases: 76358183(99.2886%)\n",
      "Q30 bases: 70820041(92.0873%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 822905\n",
      "total bases: 74885957\n",
      "Q20 bases: 74450348(99.4183%)\n",
      "Q30 bases: 69591241(92.9296%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1645810\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 53878\n",
      "reads with adapter trimmed: 802\n",
      "bases trimmed due to adapters: 29662\n",
      "\n",
      "Duplication rate: 5.61101%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 150\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753374_chr12_1.fastq.gz -I data/raw/ERR753374_chr12_2.fastq.gz -o data/processed/ERR753374_chr12_1_fastp.fastq.gz -O data/processed/ERR753374_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Tue Aug 29 10:29:13 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:29:13 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753374_chr12_1_fastp.fastq.gz, data/processed/ERR753374_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753374_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753374_chr12_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753374_chr12_1_fastp.fastq.gz, data/processed/ERR753374_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 10:29:21 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:29:21 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753374_chr12_1.fastq.gz, data/raw/ERR753374_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753374_chr12_1_fastqc.html, results/fastqc_result/ERR753374_chr12_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753374_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753374_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 10:29:30 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T102907.861844.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:29:31 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753375_chr12_1.fastq.gz, data/raw/ERR753375_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753375_chr12_1_fastqc.html, results/fastqc_result/ERR753375_chr12_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753375_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753375_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 10:29:42 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:29:42 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753375_chr12_1.fastq.gz, data/raw/ERR753375_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753375_chr12_1_fastp.fastq.gz, data/processed/ERR753375_chr12_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753375_chr12_1_fastp.fastq.gz, data/processed/ERR753375_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1133521\n",
      "total bases: 85620774\n",
      "Q20 bases: 85092338(99.3828%)\n",
      "Q30 bases: 83171029(97.1388%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1133521\n",
      "total bases: 85602587\n",
      "Q20 bases: 84492851(98.7036%)\n",
      "Q30 bases: 82429218(96.2929%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 738967\n",
      "total bases: 56110081\n",
      "Q20 bases: 55927294(99.6742%)\n",
      "Q30 bases: 55087312(98.1772%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 738967\n",
      "total bases: 56118788\n",
      "Q20 bases: 55577475(99.0354%)\n",
      "Q30 bases: 54704557(97.4799%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1477934\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 789108\n",
      "reads with adapter trimmed: 472\n",
      "bases trimmed due to adapters: 10223\n",
      "\n",
      "Duplication rate: 1.69021%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753375_chr12_1.fastq.gz -I data/raw/ERR753375_chr12_2.fastq.gz -o data/processed/ERR753375_chr12_1_fastp.fastq.gz -O data/processed/ERR753375_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 3 seconds\n",
      "[Tue Aug 29 10:29:46 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:29:46 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753375_chr12_1_fastp.fastq.gz, data/processed/ERR753375_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753375_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753375_chr12_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753375_chr12_1_fastp.fastq.gz, data/processed/ERR753375_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 10:29:53 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T102930.882977.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:29:54 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753376_chr12_1.fastq.gz, data/raw/ERR753376_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753376_chr12_1_fastqc.html, results/fastqc_result/ERR753376_chr12_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753376_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753376_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 10:30:07 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:30:07 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753376_chr12_1.fastq.gz, data/raw/ERR753376_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753376_chr12_1_fastp.fastq.gz, data/processed/ERR753376_chr12_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753376_chr12_2_fastp.fastq.gz, data/processed/ERR753376_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1465726\n",
      "total bases: 110800464\n",
      "Q20 bases: 110033992(99.3082%)\n",
      "Q30 bases: 107011804(96.5806%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1465726\n",
      "total bases: 110496460\n",
      "Q20 bases: 109134987(98.7679%)\n",
      "Q30 bases: 104686265(94.7417%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 904519\n",
      "total bases: 68697720\n",
      "Q20 bases: 68438927(99.6233%)\n",
      "Q30 bases: 67261210(97.9089%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 904519\n",
      "total bases: 68671754\n",
      "Q20 bases: 68200419(99.3136%)\n",
      "Q30 bases: 66433000(96.7399%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1809038\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1122414\n",
      "reads with adapter trimmed: 700\n",
      "bases trimmed due to adapters: 15961\n",
      "\n",
      "Duplication rate: 9.00858%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753376_chr12_1.fastq.gz -I data/raw/ERR753376_chr12_2.fastq.gz -o data/processed/ERR753376_chr12_1_fastp.fastq.gz -O data/processed/ERR753376_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Tue Aug 29 10:30:12 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:30:12 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753376_chr12_1_fastp.fastq.gz, data/processed/ERR753376_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753376_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753376_chr12_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753376_chr12_2_fastp.fastq.gz, data/processed/ERR753376_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 10:30:21 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T102953.745751.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:30:22 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753377_chr12_1.fastq.gz, data/raw/ERR753377_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753377_chr12_1_fastp.fastq.gz, data/processed/ERR753377_chr12_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753377_chr12_2_fastp.fastq.gz, data/processed/ERR753377_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1534776\n",
      "total bases: 116035701\n",
      "Q20 bases: 115218791(99.296%)\n",
      "Q30 bases: 112042570(96.5587%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1534776\n",
      "total bases: 115756078\n",
      "Q20 bases: 114393805(98.8232%)\n",
      "Q30 bases: 109938521(94.9743%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 970154\n",
      "total bases: 73683410\n",
      "Q20 bases: 73398394(99.6132%)\n",
      "Q30 bases: 72118967(97.8768%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 970154\n",
      "total bases: 73656183\n",
      "Q20 bases: 73172261(99.343%)\n",
      "Q30 bases: 71360589(96.8834%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1940308\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1129244\n",
      "reads with adapter trimmed: 748\n",
      "bases trimmed due to adapters: 16700\n",
      "\n",
      "Duplication rate: 9.0325%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 119\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753377_chr12_1.fastq.gz -I data/raw/ERR753377_chr12_2.fastq.gz -o data/processed/ERR753377_chr12_1_fastp.fastq.gz -O data/processed/ERR753377_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Tue Aug 29 10:30:27 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:30:27 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753377_chr12_1.fastq.gz, data/raw/ERR753377_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753377_chr12_1_fastqc.html, results/fastqc_result/ERR753377_chr12_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753377_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753377_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 10:30:40 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:30:40 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753377_chr12_1_fastp.fastq.gz, data/processed/ERR753377_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753377_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753377_chr12_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753377_chr12_2_fastp.fastq.gz, data/processed/ERR753377_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 10:30:49 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T103021.398468.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:30:50 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753378_chr12_1.fastq.gz, data/raw/ERR753378_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753378_chr12_1_fastqc.html, results/fastqc_result/ERR753378_chr12_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753378_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753378_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 10:31:05 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:31:05 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753378_chr12_1.fastq.gz, data/raw/ERR753378_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753378_chr12_1_fastp.fastq.gz, data/processed/ERR753378_chr12_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753378_chr12_1_fastp.fastq.gz, data/processed/ERR753378_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1753717\n",
      "total bases: 133156565\n",
      "Q20 bases: 132634253(99.6077%)\n",
      "Q30 bases: 128678299(96.6368%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1753717\n",
      "total bases: 133019021\n",
      "Q20 bases: 131845170(99.1175%)\n",
      "Q30 bases: 125793123(94.5678%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1509829\n",
      "total bases: 114740679\n",
      "Q20 bases: 114539895(99.825%)\n",
      "Q30 bases: 112565716(98.1045%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1509829\n",
      "total bases: 114731150\n",
      "Q20 bases: 114164648(99.5062%)\n",
      "Q30 bases: 110726794(96.5098%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3019658\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 487776\n",
      "reads with adapter trimmed: 1118\n",
      "bases trimmed due to adapters: 25907\n",
      "\n",
      "Duplication rate: 1.96602%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753378_chr12_1.fastq.gz -I data/raw/ERR753378_chr12_2.fastq.gz -o data/processed/ERR753378_chr12_1_fastp.fastq.gz -O data/processed/ERR753378_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Tue Aug 29 10:31:11 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:31:11 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753378_chr12_1_fastp.fastq.gz, data/processed/ERR753378_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753378_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753378_chr12_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753378_chr12_1_fastp.fastq.gz, data/processed/ERR753378_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 10:31:23 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T103049.295405.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:31:25 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR696683_chr12_1_fastp.fastq.gz, data/processed/ERR696683_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR696683_chr12.sam, results/mapped_reads/ERR696683_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR696683_chr12_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR696683_chr12_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.50 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 48.64 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.39 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.28 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 27.60 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 80.841 sec; CPU: 77.410 sec\n",
      "[Tue Aug 29 10:34:13 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:34:13 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR696683_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr12.bam, results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR696683_chr12.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR696683_chr12_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 10:34:56 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:34:56 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR696683_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "10:34:57.687 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 10:34:57 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR696683_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR696683_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 10:34:57 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 10:34:57\tMarkDuplicates\tStart of doWork freeMemory: 528788792; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 10:34:57\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 10:34:57\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam\n",
      "INFO\t2023-08-29 10:34:59\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:53,179,808\n",
      "INFO\t2023-08-29 10:34:59\tMarkDuplicates\tTracking 637 as yet unmatched pairs. 637 records in RAM.\n",
      "INFO\t2023-08-29 10:35:00\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 12:113,957,746\n",
      "INFO\t2023-08-29 10:35:00\tMarkDuplicates\tTracking 278 as yet unmatched pairs. 278 records in RAM.\n",
      "INFO\t2023-08-29 10:35:01\tMarkDuplicates\tRead 2264185 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 10:35:01\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 287067888; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 10:35:01\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 10:35:01\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 10:35:01\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 10:35:01\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 10:35:01\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 916408592; totalMemory: 1462763520; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 10:35:01\tMarkDuplicates\tMarking 210600 records as duplicates.\n",
      "INFO\t2023-08-29 10:35:01\tMarkDuplicates\tFound 91 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam\n",
      "INFO\t2023-08-29 10:35:01\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 10:35:16\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 10:35:16\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 10:35:16\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 10:35:16\tMarkDuplicates\tBefore output close freeMemory: 528086576; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 10:35:16\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 10:35:16\tMarkDuplicates\tAfter output close freeMemory: 528086576; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 10:35:16 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.31 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 10:35:17 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T103124.104244.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:35:19 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753368_chr12_1_fastp.fastq.gz, data/processed/ERR753368_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753368_chr12.sam, results/mapped_reads/ERR753368_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753368_chr12_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753368_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.46 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 47.16 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.41 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.28 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 27.78 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 80.440 sec; CPU: 76.094 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Tue Aug 29 10:39:04 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:39:04 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753368_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr12.bam, results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753368_chr12.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753368_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Tue Aug 29 10:40:04 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:40:04 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753368_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753368_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "10:40:05.720 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 10:40:05 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753368_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753368_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 10:40:05 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 10:40:05\tMarkDuplicates\tStart of doWork freeMemory: 528780768; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 10:40:05\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 10:40:05\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam\n",
      "INFO\t2023-08-29 10:40:07\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:43,551,116\n",
      "INFO\t2023-08-29 10:40:07\tMarkDuplicates\tTracking 479 as yet unmatched pairs. 479 records in RAM.\n",
      "INFO\t2023-08-29 10:40:08\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 12:76,860,355\n",
      "INFO\t2023-08-29 10:40:08\tMarkDuplicates\tTracking 510 as yet unmatched pairs. 510 records in RAM.\n",
      "INFO\t2023-08-29 10:40:10\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 12:122,377,489\n",
      "INFO\t2023-08-29 10:40:10\tMarkDuplicates\tTracking 227 as yet unmatched pairs. 227 records in RAM.\n",
      "INFO\t2023-08-29 10:40:10\tMarkDuplicates\tRead 3277664 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 10:40:10\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 628131520; totalMemory: 1052770304; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 10:40:10\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 10:40:10\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 10:40:11\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 10:40:11\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 10:40:11\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1043279352; totalMemory: 1589641216; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 10:40:11\tMarkDuplicates\tMarking 79702 records as duplicates.\n",
      "INFO\t2023-08-29 10:40:11\tMarkDuplicates\tFound 488 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam\n",
      "INFO\t2023-08-29 10:40:11\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 10:40:31\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 10:40:31\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 10:40:31\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 10:40:31\tMarkDuplicates\tBefore output close freeMemory: 528079368; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 10:40:31\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 10:40:31\tMarkDuplicates\tAfter output close freeMemory: 528079368; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 10:40:31 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.42 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 10:40:33 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T103517.934870.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:40:35 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753369_chr12_1_fastp.fastq.gz, data/processed/ERR753369_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753369_chr12.sam, results/mapped_reads/ERR753369_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753369_chr12_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753369_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.45 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 47.08 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.42 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.28 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 27.74 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 80.411 sec; CPU: 75.970 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Tue Aug 29 10:44:29 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:44:29 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753369_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr12.bam, results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753369_chr12.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753369_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Tue Aug 29 10:45:32 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:45:32 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753369_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753369_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "10:45:33.516 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 10:45:33 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753369_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753369_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 10:45:33 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 10:45:33\tMarkDuplicates\tStart of doWork freeMemory: 528798488; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 10:45:33\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 10:45:33\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam\n",
      "INFO\t2023-08-29 10:45:35\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:40,105,150\n",
      "INFO\t2023-08-29 10:45:35\tMarkDuplicates\tTracking 394 as yet unmatched pairs. 394 records in RAM.\n",
      "INFO\t2023-08-29 10:45:36\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 12:67,306,228\n",
      "INFO\t2023-08-29 10:45:36\tMarkDuplicates\tTracking 520 as yet unmatched pairs. 520 records in RAM.\n",
      "INFO\t2023-08-29 10:45:37\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 12:112,948,761\n",
      "INFO\t2023-08-29 10:45:37\tMarkDuplicates\tTracking 274 as yet unmatched pairs. 274 records in RAM.\n",
      "INFO\t2023-08-29 10:45:38\tMarkDuplicates\tRead 3547704 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 10:45:38\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 471071848; totalMemory: 924844032; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 10:45:38\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 10:45:38\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 10:45:39\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 10:45:39\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 10:45:39\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 915370416; totalMemory: 1461714944; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 10:45:39\tMarkDuplicates\tMarking 100947 records as duplicates.\n",
      "INFO\t2023-08-29 10:45:39\tMarkDuplicates\tFound 534 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam\n",
      "INFO\t2023-08-29 10:45:39\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 10:46:00\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 10:46:00\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 10:46:00\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 10:46:00\tMarkDuplicates\tBefore output close freeMemory: 528096984; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 10:46:00\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 10:46:00\tMarkDuplicates\tAfter output close freeMemory: 528096984; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 10:46:00 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.45 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 10:46:02 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T104033.969278.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:46:04 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753370_chr12_1_fastp.fastq.gz, data/processed/ERR753370_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753370_chr12.sam, results/mapped_reads/ERR753370_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753370_chr12_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753370_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.46 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 47.31 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.42 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.28 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 27.80 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 80.856 sec; CPU: 76.281 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Tue Aug 29 10:51:42 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:51:42 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753370_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr12.bam, results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753370_chr12.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753370_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Tue Aug 29 10:53:40 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:53:40 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753370_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753370_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "10:53:41.523 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 10:53:41 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753370_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753370_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 10:53:41 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 10:53:41\tMarkDuplicates\tStart of doWork freeMemory: 528779664; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 10:53:41\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 10:53:41\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam\n",
      "INFO\t2023-08-29 10:53:43\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:21,504,457\n",
      "INFO\t2023-08-29 10:53:43\tMarkDuplicates\tTracking 792 as yet unmatched pairs. 792 records in RAM.\n",
      "INFO\t2023-08-29 10:53:44\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 12:49,752,439\n",
      "INFO\t2023-08-29 10:53:44\tMarkDuplicates\tTracking 1407 as yet unmatched pairs. 1407 records in RAM.\n",
      "INFO\t2023-08-29 10:53:46\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 12:69,697,584\n",
      "INFO\t2023-08-29 10:53:46\tMarkDuplicates\tTracking 1412 as yet unmatched pairs. 1412 records in RAM.\n",
      "INFO\t2023-08-29 10:53:47\tMarkDuplicates\tRead     4,000,000 records.  Elapsed time: 00:00:06s.  Time for last 1,000,000:    1s.  Last read position: 12:103,793,214\n",
      "INFO\t2023-08-29 10:53:47\tMarkDuplicates\tTracking 1253 as yet unmatched pairs. 1253 records in RAM.\n",
      "INFO\t2023-08-29 10:53:49\tMarkDuplicates\tRead     5,000,000 records.  Elapsed time: 00:00:07s.  Time for last 1,000,000:    1s.  Last read position: 12:128,503,824\n",
      "INFO\t2023-08-29 10:53:49\tMarkDuplicates\tTracking 275 as yet unmatched pairs. 275 records in RAM.\n",
      "INFO\t2023-08-29 10:53:49\tMarkDuplicates\tRead 5146509 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 10:53:49\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 427678040; totalMemory: 1050673152; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 10:53:49\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 10:53:49\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 10:53:50\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 10:53:51\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 10:53:51\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1041179488; totalMemory: 1587544064; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 10:53:51\tMarkDuplicates\tMarking 368842 records as duplicates.\n",
      "INFO\t2023-08-29 10:53:51\tMarkDuplicates\tFound 219 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam\n",
      "INFO\t2023-08-29 10:53:51\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 10:54:29\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 10:54:29\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 10:54:29\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 10:54:29\tMarkDuplicates\tBefore output close freeMemory: 528076816; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 10:54:29\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 10:54:29\tMarkDuplicates\tAfter output close freeMemory: 528076816; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 10:54:29 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.79 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 10:54:33 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T104603.148402.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:54:34 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753371_chr12_1_fastp.fastq.gz, data/processed/ERR753371_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753371_chr12.sam, results/mapped_reads/ERR753371_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753371_chr12_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753371_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.45 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 47.28 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.42 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.27 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 27.83 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 80.581 sec; CPU: 76.255 sec\n",
      "[Tue Aug 29 10:57:03 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:57:03 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753371_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr12.bam, results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753371_chr12.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753371_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 10:57:35 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:57:35 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753371_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753371_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "10:57:36.529 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 10:57:36 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753371_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753371_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 10:57:36 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 10:57:36\tMarkDuplicates\tStart of doWork freeMemory: 528790312; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 10:57:36\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 10:57:36\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam\n",
      "INFO\t2023-08-29 10:57:38\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:68,764,508\n",
      "INFO\t2023-08-29 10:57:38\tMarkDuplicates\tTracking 112 as yet unmatched pairs. 112 records in RAM.\n",
      "INFO\t2023-08-29 10:57:39\tMarkDuplicates\tRead 1748836 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 10:57:39\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 276607712; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 10:57:39\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 10:57:39\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 10:57:39\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 10:57:40\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 10:57:40\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 795822704; totalMemory: 1342177280; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 10:57:40\tMarkDuplicates\tMarking 714314 records as duplicates.\n",
      "INFO\t2023-08-29 10:57:40\tMarkDuplicates\tFound 65 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam\n",
      "INFO\t2023-08-29 10:57:40\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 10:57:50\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 10:57:50\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 10:57:50\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 10:57:50\tMarkDuplicates\tBefore output close freeMemory: 528086672; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 10:57:50\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 10:57:50\tMarkDuplicates\tAfter output close freeMemory: 528086672; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 10:57:50 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.24 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 10:57:51 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T105433.551413.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 10:57:53 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753372_chr12_1_fastp.fastq.gz, data/processed/ERR753372_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753372_chr12.sam, results/mapped_reads/ERR753372_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753372_chr12_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753372_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.44 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 47.27 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.42 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.28 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 27.81 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 80.477 sec; CPU: 76.216 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Tue Aug 29 11:01:46 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:01:46 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753372_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr12.bam, results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753372_chr12.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753372_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Tue Aug 29 11:02:49 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:02:49 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753372_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753372_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "11:02:50.702 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 11:02:50 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753372_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753372_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 11:02:50 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 11:02:50\tMarkDuplicates\tStart of doWork freeMemory: 528838488; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:02:50\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 11:02:50\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam\n",
      "INFO\t2023-08-29 11:02:52\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:40,527,729\n",
      "INFO\t2023-08-29 11:02:52\tMarkDuplicates\tTracking 546 as yet unmatched pairs. 546 records in RAM.\n",
      "INFO\t2023-08-29 11:02:53\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 12:69,257,716\n",
      "INFO\t2023-08-29 11:02:53\tMarkDuplicates\tTracking 532 as yet unmatched pairs. 532 records in RAM.\n",
      "INFO\t2023-08-29 11:02:55\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 12:116,750,195\n",
      "INFO\t2023-08-29 11:02:55\tMarkDuplicates\tTracking 302 as yet unmatched pairs. 302 records in RAM.\n",
      "INFO\t2023-08-29 11:02:55\tMarkDuplicates\tRead 3494913 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 11:02:55\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 414053536; totalMemory: 861929472; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:02:55\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 11:02:56\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 11:02:56\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 11:02:56\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 11:02:56\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 852496712; totalMemory: 1398800384; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:02:56\tMarkDuplicates\tMarking 84319 records as duplicates.\n",
      "INFO\t2023-08-29 11:02:56\tMarkDuplicates\tFound 567 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam\n",
      "INFO\t2023-08-29 11:02:56\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 11:03:17\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 11:03:17\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 11:03:17\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 11:03:17\tMarkDuplicates\tBefore output close freeMemory: 528138304; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:03:17\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 11:03:17\tMarkDuplicates\tAfter output close freeMemory: 528138304; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 11:03:17 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.45 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 11:03:20 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T105752.160896.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:03:21 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753373_chr12_1_fastp.fastq.gz, data/processed/ERR753373_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753373_chr12.sam, results/mapped_reads/ERR753373_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753373_chr12_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753373_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.49 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 49.82 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.41 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.28 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 28.85 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 84.286 sec; CPU: 79.843 sec\n",
      "[Tue Aug 29 11:07:29 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:07:29 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753373_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr12.bam, results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753373_chr12.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753373_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 11:08:46 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:08:46 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753373_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753373_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "11:08:47.028 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 11:08:47 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753373_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753373_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 11:08:47 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 11:08:47\tMarkDuplicates\tStart of doWork freeMemory: 528789504; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:08:47\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 11:08:47\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam\n",
      "INFO\t2023-08-29 11:08:48\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:46,204,245\n",
      "INFO\t2023-08-29 11:08:48\tMarkDuplicates\tTracking 307 as yet unmatched pairs. 307 records in RAM.\n",
      "INFO\t2023-08-29 11:08:50\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 12:92,183,955\n",
      "INFO\t2023-08-29 11:08:50\tMarkDuplicates\tTracking 218 as yet unmatched pairs. 218 records in RAM.\n",
      "INFO\t2023-08-29 11:08:51\tMarkDuplicates\tRead 2777001 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 11:08:51\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 628296696; totalMemory: 998244352; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:08:51\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 11:08:51\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 11:08:52\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 11:08:52\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 11:08:52\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 988761952; totalMemory: 1535115264; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:08:52\tMarkDuplicates\tMarking 73155 records as duplicates.\n",
      "INFO\t2023-08-29 11:08:52\tMarkDuplicates\tFound 23782 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam\n",
      "INFO\t2023-08-29 11:08:52\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 11:09:16\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 11:09:16\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 11:09:16\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 11:09:16\tMarkDuplicates\tBefore output close freeMemory: 528088136; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:09:16\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 11:09:16\tMarkDuplicates\tAfter output close freeMemory: 528088136; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 11:09:16 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.49 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 11:09:18 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T110320.556919.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:09:20 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753374_chr12_1_fastp.fastq.gz, data/processed/ERR753374_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753374_chr12.sam, results/mapped_reads/ERR753374_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753374_chr12_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753374_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.43 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 49.49 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.38 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.27 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 28.64 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 83.743 sec; CPU: 79.202 sec\n",
      "[Tue Aug 29 11:12:14 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:12:14 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753374_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr12.bam, results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753374_chr12.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753374_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 11:12:59 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:12:59 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753374_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753374_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "11:13:00.267 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 11:13:00 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753374_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753374_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 11:13:00 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 11:13:00\tMarkDuplicates\tStart of doWork freeMemory: 528769368; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:13:00\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 11:13:00\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam\n",
      "INFO\t2023-08-29 11:13:02\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:75,838,604\n",
      "INFO\t2023-08-29 11:13:02\tMarkDuplicates\tTracking 162 as yet unmatched pairs. 162 records in RAM.\n",
      "INFO\t2023-08-29 11:13:03\tMarkDuplicates\tRead 1659483 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 11:13:03\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 287186176; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:13:03\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 11:13:03\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 11:13:03\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 11:13:03\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 11:13:03\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 785316864; totalMemory: 1331691520; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:13:03\tMarkDuplicates\tMarking 175808 records as duplicates.\n",
      "INFO\t2023-08-29 11:13:03\tMarkDuplicates\tFound 58563 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam\n",
      "INFO\t2023-08-29 11:13:03\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 11:13:17\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 11:13:17\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 11:13:17\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 11:13:17\tMarkDuplicates\tBefore output close freeMemory: 528066480; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:13:17\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 11:13:17\tMarkDuplicates\tAfter output close freeMemory: 528066480; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 11:13:17 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.30 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 11:13:19 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T110919.033554.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:13:20 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753375_chr12_1_fastp.fastq.gz, data/processed/ERR753375_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753375_chr12.sam, results/mapped_reads/ERR753375_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753375_chr12_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753375_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.46 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 47.10 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.41 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.28 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 27.78 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 80.222 sec; CPU: 76.030 sec\n",
      "[Tue Aug 29 11:15:41 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:15:41 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753375_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr12.bam, results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753375_chr12.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753375_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 11:16:09 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:16:09 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753375_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753375_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "11:16:10.194 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 11:16:10 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753375_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753375_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 11:16:10 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 11:16:10\tMarkDuplicates\tStart of doWork freeMemory: 528831608; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:16:10\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 11:16:10\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam\n",
      "INFO\t2023-08-29 11:16:11\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:87,996,517\n",
      "INFO\t2023-08-29 11:16:11\tMarkDuplicates\tTracking 188 as yet unmatched pairs. 188 records in RAM.\n",
      "INFO\t2023-08-29 11:16:12\tMarkDuplicates\tRead 1483975 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 11:16:12\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 305606864; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:16:12\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 11:16:12\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 11:16:12\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 11:16:13\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 11:16:13\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 773846872; totalMemory: 1320157184; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:16:13\tMarkDuplicates\tMarking 72144 records as duplicates.\n",
      "INFO\t2023-08-29 11:16:13\tMarkDuplicates\tFound 188 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam\n",
      "INFO\t2023-08-29 11:16:13\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 11:16:22\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 11:16:22\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 11:16:22\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 11:16:22\tMarkDuplicates\tBefore output close freeMemory: 528131144; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:16:22\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 11:16:22\tMarkDuplicates\tAfter output close freeMemory: 528131144; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 11:16:22 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.21 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 11:16:23 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T111319.573031.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:16:24 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753376_chr12_1_fastp.fastq.gz, data/processed/ERR753376_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753376_chr12.sam, results/mapped_reads/ERR753376_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753376_chr12_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753376_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.44 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 47.03 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.39 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.26 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 27.73 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 80.215 sec; CPU: 75.852 sec\n",
      "[Tue Aug 29 11:18:57 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:18:57 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753376_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr12.bam, results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753376_chr12.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753376_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 11:19:32 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:19:32 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753376_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753376_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "11:19:33.163 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 11:19:33 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753376_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753376_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 11:19:33 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 11:19:33\tMarkDuplicates\tStart of doWork freeMemory: 528789440; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:19:33\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 11:19:33\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam\n",
      "INFO\t2023-08-29 11:19:34\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:64,464,346\n",
      "INFO\t2023-08-29 11:19:34\tMarkDuplicates\tTracking 273 as yet unmatched pairs. 273 records in RAM.\n",
      "INFO\t2023-08-29 11:19:35\tMarkDuplicates\tRead 1814904 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 11:19:35\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 592621856; totalMemory: 859832320; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:19:35\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 11:19:36\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 11:19:36\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 11:19:36\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 11:19:36\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 850348968; totalMemory: 1396703232; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:19:36\tMarkDuplicates\tMarking 260473 records as duplicates.\n",
      "INFO\t2023-08-29 11:19:36\tMarkDuplicates\tFound 83 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam\n",
      "INFO\t2023-08-29 11:19:36\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 11:19:48\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 11:19:48\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 11:19:48\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 11:19:48\tMarkDuplicates\tBefore output close freeMemory: 528087240; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:19:48\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 11:19:48\tMarkDuplicates\tAfter output close freeMemory: 528087240; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 11:19:48 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.25 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 11:19:49 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T111623.711230.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:19:50 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753377_chr12_1_fastp.fastq.gz, data/processed/ERR753377_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753377_chr12.sam, results/mapped_reads/ERR753377_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753377_chr12_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753377_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.45 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 46.23 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.41 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.28 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 27.64 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 79.428 sec; CPU: 75.016 sec\n",
      "[Tue Aug 29 11:22:26 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:22:26 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753377_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr12.bam, results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753377_chr12.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753377_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 11:23:03 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:23:03 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753377_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753377_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "11:23:04.229 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 11:23:04 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753377_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753377_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 11:23:04 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 11:23:04\tMarkDuplicates\tStart of doWork freeMemory: 528784184; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:23:04\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 11:23:04\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam\n",
      "INFO\t2023-08-29 11:23:05\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:57,029,289\n",
      "INFO\t2023-08-29 11:23:05\tMarkDuplicates\tTracking 396 as yet unmatched pairs. 396 records in RAM.\n",
      "INFO\t2023-08-29 11:23:07\tMarkDuplicates\tRead 1953622 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 11:23:07\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 320413768; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:23:07\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 11:23:07\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 11:23:07\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 11:23:07\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 11:23:07\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 834614856; totalMemory: 1380974592; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:23:07\tMarkDuplicates\tMarking 278994 records as duplicates.\n",
      "INFO\t2023-08-29 11:23:07\tMarkDuplicates\tFound 103 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam\n",
      "INFO\t2023-08-29 11:23:07\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 11:23:20\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 11:23:20\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 11:23:20\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 11:23:20\tMarkDuplicates\tBefore output close freeMemory: 528080864; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:23:20\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 11:23:20\tMarkDuplicates\tAfter output close freeMemory: 528080864; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 11:23:20 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.27 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 11:23:21 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T111949.723093.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:23:22 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753378_chr12_1_fastp.fastq.gz, data/processed/ERR753378_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753378_chr12.sam, results/mapped_reads/ERR753378_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753378_chr12_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753378_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.47 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 48.03 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.41 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.28 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 29.05 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 83.004 sec; CPU: 78.242 sec\n",
      "[Tue Aug 29 11:26:54 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:26:54 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753378_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr12.bam, results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753378_chr12.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753378_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 11:27:40 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:27:40 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753378_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753378_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "11:27:41.036 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 11:27:41 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753378_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753378_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 11:27:41 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 11:27:41\tMarkDuplicates\tStart of doWork freeMemory: 528835472; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:27:41\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 11:27:41\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam\n",
      "INFO\t2023-08-29 11:27:42\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:47,983,094\n",
      "INFO\t2023-08-29 11:27:42\tMarkDuplicates\tTracking 465 as yet unmatched pairs. 465 records in RAM.\n",
      "INFO\t2023-08-29 11:27:44\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 12:90,841,669\n",
      "INFO\t2023-08-29 11:27:44\tMarkDuplicates\tTracking 410 as yet unmatched pairs. 410 records in RAM.\n",
      "INFO\t2023-08-29 11:27:45\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 12:132,671,826\n",
      "INFO\t2023-08-29 11:27:45\tMarkDuplicates\tTracking 25 as yet unmatched pairs. 25 records in RAM.\n",
      "INFO\t2023-08-29 11:27:45\tMarkDuplicates\tRead 3037191 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 11:27:45\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 498391120; totalMemory: 896532480; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:27:45\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 11:27:45\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 11:27:46\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 11:27:46\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 11:27:46\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 887096664; totalMemory: 1433403392; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:27:46\tMarkDuplicates\tMarking 71532 records as duplicates.\n",
      "INFO\t2023-08-29 11:27:46\tMarkDuplicates\tFound 500 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam\n",
      "INFO\t2023-08-29 11:27:46\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 11:28:03\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 11:28:03\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 11:28:03\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 11:28:03\tMarkDuplicates\tBefore output close freeMemory: 528134584; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 11:28:03\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 11:28:03\tMarkDuplicates\tAfter output close freeMemory: 528134584; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 11:28:03 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.38 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 11:28:06 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T112321.628277.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:28:07 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR696683_chr12_dedup.bam\n",
      "    output: results/variants/ERR696683_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR696683_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR696683_chr12.vcf\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "index file data/reference/genome.fa.fai not found, generating...\n",
      "[Tue Aug 29 11:28:53 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T112806.179161.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:28:54 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753368_chr12_dedup.bam\n",
      "    output: results/variants/ERR753368_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753368_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753368_chr12.vcf\n",
      "    wildcards: sample=ERR753368_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 11:29:59 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T112853.532123.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:30:00 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753369_chr12_dedup.bam\n",
      "    output: results/variants/ERR753369_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753369_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753369_chr12.vcf\n",
      "    wildcards: sample=ERR753369_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 11:31:07 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T112959.318541.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:31:09 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753370_chr12_dedup.bam\n",
      "    output: results/variants/ERR753370_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753370_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753370_chr12.vcf\n",
      "    wildcards: sample=ERR753370_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 11:32:52 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T113108.127577.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:32:53 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753371_chr12_dedup.bam\n",
      "    output: results/variants/ERR753371_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753371_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753371_chr12.vcf\n",
      "    wildcards: sample=ERR753371_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 11:33:24 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T113252.783161.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:33:25 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753372_chr12_dedup.bam\n",
      "    output: results/variants/ERR753372_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753372_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753372_chr12.vcf\n",
      "    wildcards: sample=ERR753372_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 11:34:30 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T113324.814037.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:34:31 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753373_chr12_dedup.bam\n",
      "    output: results/variants/ERR753373_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753373_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753373_chr12.vcf\n",
      "    wildcards: sample=ERR753373_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 11:35:39 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T113430.906694.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:35:40 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753374_chr12_dedup.bam\n",
      "    output: results/variants/ERR753374_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753374_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753374_chr12.vcf\n",
      "    wildcards: sample=ERR753374_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 11:36:25 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T113539.949815.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:36:26 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753375_chr12_dedup.bam\n",
      "    output: results/variants/ERR753375_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753375_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753375_chr12.vcf\n",
      "    wildcards: sample=ERR753375_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 11:37:02 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T113625.758745.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:37:03 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753376_chr12_dedup.bam\n",
      "    output: results/variants/ERR753376_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753376_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753376_chr12.vcf\n",
      "    wildcards: sample=ERR753376_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 11:37:40 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T113702.609776.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:37:42 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753377_chr12_dedup.bam\n",
      "    output: results/variants/ERR753377_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753377_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753377_chr12.vcf\n",
      "    wildcards: sample=ERR753377_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 11:38:20 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T113741.015797.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:38:22 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753378_chr12_dedup.bam\n",
      "    output: results/variants/ERR753378_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753378_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753378_chr12.vcf\n",
      "    wildcards: sample=ERR753378_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 11:39:20 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T113821.046576.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:39:21 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR696683_chr12.vcf\n",
      "    output: results/variants/vep/ERR696683_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR696683_chr12.txt\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 11:39:46 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T113920.563865.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:39:48 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753368_chr12.vcf\n",
      "    output: results/variants/vep/ERR753368_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753368_chr12.txt\n",
      "    wildcards: sample=ERR753368_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 11:40:16 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T113946.738603.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:40:17 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753369_chr12.vcf\n",
      "    output: results/variants/vep/ERR753369_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753369_chr12.txt\n",
      "    wildcards: sample=ERR753369_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 11:40:45 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T114016.744659.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:40:46 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753370_chr12.vcf\n",
      "    output: results/variants/vep/ERR753370_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753370_chr12.txt\n",
      "    wildcards: sample=ERR753370_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 11:41:17 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T114045.746449.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:41:18 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753371_chr12.vcf\n",
      "    output: results/variants/vep/ERR753371_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753371_chr12.txt\n",
      "    wildcards: sample=ERR753371_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 11:41:37 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T114117.195433.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:41:39 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753372_chr12.vcf\n",
      "    output: results/variants/vep/ERR753372_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753372_chr12.txt\n",
      "    wildcards: sample=ERR753372_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 11:42:06 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T114138.064082.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:42:07 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753373_chr12.vcf\n",
      "    output: results/variants/vep/ERR753373_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753373_chr12.txt\n",
      "    wildcards: sample=ERR753373_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 11:42:33 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T114206.677546.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:42:34 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753374_chr12.vcf\n",
      "    output: results/variants/vep/ERR753374_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753374_chr12.txt\n",
      "    wildcards: sample=ERR753374_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 11:42:54 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T114233.622533.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:42:55 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753375_chr12.vcf\n",
      "    output: results/variants/vep/ERR753375_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753375_chr12.txt\n",
      "    wildcards: sample=ERR753375_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 11:43:17 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T114254.952388.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:43:18 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753376_chr12.vcf\n",
      "    output: results/variants/vep/ERR753376_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753376_chr12.txt\n",
      "    wildcards: sample=ERR753376_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 11:43:39 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T114317.368562.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:43:41 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753377_chr12.vcf\n",
      "    output: results/variants/vep/ERR753377_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753377_chr12.txt\n",
      "    wildcards: sample=ERR753377_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 11:44:02 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T114340.103733.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:44:03 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753378_chr12.vcf\n",
      "    output: results/variants/vep/ERR753378_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753378_chr12.txt\n",
      "    wildcards: sample=ERR753378_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 11:44:30 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T114402.749140.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job              count    min threads    max threads\n",
      "-------------  -------  -------------  -------------\n",
      "parsing_dataR        1              1              1\n",
      "total                1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 11:44:30 2023]\n",
      "rule parsing_dataR:\n",
      "    input: code/05parsing_vep_data.R\n",
      "    output: tasks/13parsing_dataR.done\n",
      "    jobid: 0\n",
      "    reason: Params have changed since last execution\n",
      "    resources: tmpdir=/tmp\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">>> This is your sample list: ERR696683_chr12\n",
      ">>> This is your sample list: ERR753368_chr12\n",
      ">>> This is your sample list: ERR753369_chr12\n",
      ">>> This is your sample list: ERR753370_chr12\n",
      ">>> This is your sample list: ERR753371_chr12\n",
      ">>> This is your sample list: ERR753372_chr12\n",
      ">>> This is your sample list: ERR753373_chr12\n",
      ">>> This is your sample list: ERR753374_chr12\n",
      ">>> This is your sample list: ERR753375_chr12\n",
      ">>> This is your sample list: ERR753376_chr12\n",
      ">>> This is your sample list: ERR753377_chr12\n",
      ">>> This is your sample list: ERR753378_chr12\n",
      ">>> This is your gene: KRAS\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 0 × 44\n",
      "# ℹ 44 variables: #uploaded_variation <chr>, location <chr>, allele <chr>,\n",
      "#   gene <chr>, feature <chr>, feature_type <chr>, consequence <chr>,\n",
      "#   cdna_position <chr>, cds_position <chr>, protein_position <chr>,\n",
      "#   amino_acids <chr>, codons <chr>, existing_variation <chr>, impact <chr>,\n",
      "#   distance <chr>, strand <chr>, flags <chr>, variant_class <chr>,\n",
      "#   symbol <chr>, symbol_source <chr>, hgnc_id <chr>, biotype <chr>,\n",
      "#   mane_select <chr>, mane_plus_clinical <chr>, tsl <chr>, appris <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 24 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "3 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "4 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "# ℹ 18 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 36 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "3 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "4 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "5 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "# ℹ 30 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 36 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25209531_G/A       12:252095… A      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 12_25209531_G/A       12:252095… A      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "3 12_25209531_G/A       12:252095… A      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "4 12_25209531_G/A       12:252095… A      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 12_25209531_G/A       12:252095… A      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 12_25209531_G/A       12:252095… A      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "# ℹ 30 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 0 × 44\n",
      "# ℹ 44 variables: #uploaded_variation <chr>, location <chr>, allele <chr>,\n",
      "#   gene <chr>, feature <chr>, feature_type <chr>, consequence <chr>,\n",
      "#   cdna_position <chr>, cds_position <chr>, protein_position <chr>,\n",
      "#   amino_acids <chr>, codons <chr>, existing_variation <chr>, impact <chr>,\n",
      "#   distance <chr>, strand <chr>, flags <chr>, variant_class <chr>,\n",
      "#   symbol <chr>, symbol_source <chr>, hgnc_id <chr>, biotype <chr>,\n",
      "#   mane_select <chr>, mane_plus_clinical <chr>, tsl <chr>, appris <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 36 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "3 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "4 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "# ℹ 30 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 25 × 44\n",
      "  `#uploaded_variation`   location allele gene  feature feature_type consequence\n",
      "  <chr>                   <chr>    <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25230796_AATTAATTA/… 12:2523… AATTA  ENSG… ENST00… Transcript   intron_var…\n",
      "2 12_25230796_AATTAATTA/… 12:2523… AATTA  ENSG… ENST00… Transcript   intron_var…\n",
      "3 12_25230796_AATTAATTA/… 12:2523… AATTA  ENSG… ENST00… Transcript   downstream…\n",
      "4 12_25230796_AATTAATTA/… 12:2523… AATTA  ENSG… ENST00… Transcript   intron_var…\n",
      "5 12_25230796_AATTAATTA/… 12:2523… AATTA  ENSG… ENST00… Transcript   intron_var…\n",
      "6 12_25230796_AATTAATTA/… 12:2523… AATTA  ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 19 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 0 × 44\n",
      "# ℹ 44 variables: #uploaded_variation <chr>, location <chr>, allele <chr>,\n",
      "#   gene <chr>, feature <chr>, feature_type <chr>, consequence <chr>,\n",
      "#   cdna_position <chr>, cds_position <chr>, protein_position <chr>,\n",
      "#   amino_acids <chr>, codons <chr>, existing_variation <chr>, impact <chr>,\n",
      "#   distance <chr>, strand <chr>, flags <chr>, variant_class <chr>,\n",
      "#   symbol <chr>, symbol_source <chr>, hgnc_id <chr>, biotype <chr>,\n",
      "#   mane_select <chr>, mane_plus_clinical <chr>, tsl <chr>, appris <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 24 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "3 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "4 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "5 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "# ℹ 18 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 12 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25209618_A/C       12:252096… C      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 12_25209618_A/C       12:252096… C      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "3 12_25209618_A/C       12:252096… C      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "4 12_25209618_A/C       12:252096… C      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 12_25209618_A/C       12:252096… C      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 12_25209618_A/C       12:252096… C      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "# ℹ 6 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 12 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "3 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "4 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "5 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "# ℹ 6 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 36 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "3 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "4 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "5 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "# ℹ 30 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "[1] \"##==========================##\"\n",
      "[1] \"##===> WORK FINISHED!!! <===##\"\n",
      "[1] \"##==========================##\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "Touching output file tasks/13parsing_dataR.done.\n",
      "[Tue Aug 29 11:44:33 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T114430.423390.snakemake.log\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['snakemake', '--cores', '1', 'parsing_dataR'], returncode=0)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Running the workflow for the 12º chormosome\n",
    "\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--force\", \"data/reference/genome.fa\"]) ## rule reference_genome \n",
    "\n",
    "for sample in sample_list:\n",
    "    ## Quality inspection\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/fastqc_result/\" + sample + \"_chr12_1_fastqc.html\",   \n",
    "                    \"data/processed/\" + sample + \"_chr12_1_fastp.fastq.gz\",   \n",
    "                    \"results/fastqc_result/trimmed/\" + sample + \"_chr12_1_fastp_fastqc.html\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    ## Mapping reads\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/mapped_reads/\" + sample + \"_chr12_sorted.sam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr12_sorted.bam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr12_dedup.bam\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    # Extracting variants  \n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/\" + sample + \"_chr12.vcf\"])   \n",
    "\n",
    "## VEP CLI\n",
    "for sample in sample_list:\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/vep/\" + sample + \"_chr12.txt\"])\n",
    "\n",
    "## Extracting the gene PIKCA3 from the VEP files\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"parsing_dataR\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Changing the config file to do the chromosome 17 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['sed', '-i', '65s/KRAS/TP53/g', 'config.yaml'], returncode=0)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subprocess.run([\"sed\", \"-i\", \"6,+12s/chr12/chr17/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"24,+12s/chr12/chr17/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\",\"39s/chr12/chr17/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"46s/chromosome.12.fa.gz/chromosome.17.fa.gz/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"65s/KRAS/TP53/g\", \"config.yaml\"])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job                 count    min threads    max threads\n",
      "----------------  -------  -------------  -------------\n",
      "reference_genome        1              1              1\n",
      "total                   1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:50:29 2023]\n",
      "rule reference_genome:\n",
      "    output: data/reference/genome.fa\n",
      "    jobid: 0\n",
      "    reason: Forced execution\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "--2023-08-29 17:50:29--  https://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.17.fa.gz\n",
      "Resolviendo ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.169\n",
      "Conectando con ftp.ensembl.org (ftp.ensembl.org)[193.62.193.169]:443... conectado.\n",
      "Petición HTTP enviada, esperando respuesta... 200 OK\n",
      "Longitud: 24104573 (23M) [application/x-gzip]\n",
      "Guardando como: ‘data/reference/genome.fa.gz’\n",
      "\n",
      "     0K .......... .......... .......... .......... ..........  0%  421K 56s\n",
      "    50K .......... .......... .......... .......... ..........  0% 2,53M 32s\n",
      "   100K .......... .......... .......... .......... ..........  0% 1,20M 28s\n",
      "   150K .......... .......... .......... .......... ..........  0% 28,5M 21s\n",
      "   200K .......... .......... .......... .......... ..........  1%  886K 22s\n",
      "   250K .......... .......... .......... .......... ..........  1% 31,4M 18s\n",
      "   300K .......... .......... .......... .......... ..........  1% 67,1M 16s\n",
      "   350K .......... .......... .......... .......... ..........  1% 84,0M 14s\n",
      "   400K .......... .......... .......... .......... ..........  1% 2,72M 13s\n",
      "   450K .......... .......... .......... .......... ..........  2% 1,28M 14s\n",
      "   500K .......... .......... .......... .......... ..........  2% 24,5M 12s\n",
      "   550K .......... .......... .......... .......... ..........  2% 30,4M 11s\n",
      "   600K .......... .......... .......... .......... ..........  2% 36,9M 11s\n",
      "   650K .......... .......... .......... .......... ..........  2%  109M 10s\n",
      "   700K .......... .......... .......... .......... ..........  3% 88,1M 9s\n",
      "   750K .......... .......... .......... .......... ..........  3%  124M 9s\n",
      "   800K .......... .......... .......... .......... ..........  3%  127M 8s\n",
      "   850K .......... .......... .......... .......... ..........  3% 3,54M 8s\n",
      "   900K .......... .......... .......... .......... ..........  4% 32,6M 8s\n",
      "   950K .......... .......... .......... .......... ..........  4% 1,29M 8s\n",
      "  1000K .......... .......... .......... .......... ..........  4% 96,3M 8s\n",
      "  1050K .......... .......... .......... .......... ..........  4% 27,1M 7s\n",
      "  1100K .......... .......... .......... .......... ..........  4% 27,5M 7s\n",
      "  1150K .......... .......... .......... .......... ..........  5%  149M 7s\n",
      "  1200K .......... .......... .......... .......... ..........  5%  132M 6s\n",
      "  1250K .......... .......... .......... .......... ..........  5%  148M 6s\n",
      "  1300K .......... .......... .......... .......... ..........  5%  143M 6s\n",
      "  1350K .......... .......... .......... .......... ..........  5%  124M 6s\n",
      "  1400K .......... .......... .......... .......... ..........  6%  154M 6s\n",
      "  1450K .......... .......... .......... .......... ..........  6%  128M 5s\n",
      "  1500K .......... .......... .......... .......... ..........  6%  148M 5s\n",
      "  1550K .......... .......... .......... .......... ..........  6%  153M 5s\n",
      "  1600K .......... .......... .......... .......... ..........  7%  147M 5s\n",
      "  1650K .......... .......... .......... .......... ..........  7%  165M 5s\n",
      "  1700K .......... .......... .......... .......... ..........  7%  141M 5s\n",
      "  1750K .......... .......... .......... .......... ..........  7% 4,34M 5s\n",
      "  1800K .......... .......... .......... .......... ..........  7% 81,4M 4s\n",
      "  1850K .......... .......... .......... .......... ..........  8%  100M 4s\n",
      "  1900K .......... .......... .......... .......... ..........  8% 40,6M 4s\n",
      "  1950K .......... .......... .......... .......... ..........  8% 1,33M 4s\n",
      "  2000K .......... .......... .......... .......... ..........  8% 26,1M 4s\n",
      "  2050K .......... .......... .......... .......... ..........  8% 98,9M 4s\n",
      "  2100K .......... .......... .......... .......... ..........  9%  102M 4s\n",
      "  2150K .......... .......... .......... .......... ..........  9%  118M 4s\n",
      "  2200K .......... .......... .......... .......... ..........  9%  102M 4s\n",
      "  2250K .......... .......... .......... .......... ..........  9%  120M 4s\n",
      "  2300K .......... .......... .......... .......... ..........  9%  118M 4s\n",
      "  2350K .......... .......... .......... .......... .......... 10%  103M 4s\n",
      "  2400K .......... .......... .......... .......... .......... 10%  120M 4s\n",
      "  2450K .......... .......... .......... .......... .......... 10% 86,6M 4s\n",
      "  2500K .......... .......... .......... .......... .......... 10%  102M 3s\n",
      "  2550K .......... .......... .......... .......... .......... 11%  122M 3s\n",
      "  2600K .......... .......... .......... .......... .......... 11% 92,7M 3s\n",
      "  2650K .......... .......... .......... .......... .......... 11%  188M 3s\n",
      "  2700K .......... .......... .......... .......... .......... 11%  115M 3s\n",
      "  2750K .......... .......... .......... .......... .......... 11%  118M 3s\n",
      "  2800K .......... .......... .......... .......... .......... 12%  121M 3s\n",
      "  2850K .......... .......... .......... .......... .......... 12% 81,5M 3s\n",
      "  2900K .......... .......... .......... .......... .......... 12%  127M 3s\n",
      "  2950K .......... .......... .......... .......... .......... 12%  111M 3s\n",
      "  3000K .......... .......... .......... .......... .......... 12%  133M 3s\n",
      "  3050K .......... .......... .......... .......... .......... 13%  124M 3s\n",
      "  3100K .......... .......... .......... .......... .......... 13% 78,3M 3s\n",
      "  3150K .......... .......... .......... .......... .......... 13% 61,4M 3s\n",
      "  3200K .......... .......... .......... .......... .......... 13% 28,8M 3s\n",
      "  3250K .......... .......... .......... .......... .......... 14% 62,6M 3s\n",
      "  3300K .......... .......... .......... .......... .......... 14%  168M 3s\n",
      "  3350K .......... .......... .......... .......... .......... 14%  120M 3s\n",
      "  3400K .......... .......... .......... .......... .......... 14%  113M 3s\n",
      "  3450K .......... .......... .......... .......... .......... 14% 54,3M 2s\n",
      "  3500K .......... .......... .......... .......... .......... 15% 13,0M 2s\n",
      "  3550K .......... .......... .......... .......... .......... 15%  127M 2s\n",
      "  3600K .......... .......... .......... .......... .......... 15% 78,1M 2s\n",
      "  3650K .......... .......... .......... .......... .......... 15%  115M 2s\n",
      "  3700K .......... .......... .......... .......... .......... 15%  110M 2s\n",
      "  3750K .......... .......... .......... .......... .......... 16%  119M 2s\n",
      "  3800K .......... .......... .......... .......... .......... 16%  122M 2s\n",
      "  3850K .......... .......... .......... .......... .......... 16% 88,8M 2s\n",
      "  3900K .......... .......... .......... .......... .......... 16% 1,38M 2s\n",
      "  3950K .......... .......... .......... .......... .......... 16% 57,8M 2s\n",
      "  4000K .......... .......... .......... .......... .......... 17% 52,1M 2s\n",
      "  4050K .......... .......... .......... .......... .......... 17% 45,1M 2s\n",
      "  4100K .......... .......... .......... .......... .......... 17% 41,1M 2s\n",
      "  4150K .......... .......... .......... .......... .......... 17% 73,1M 2s\n",
      "  4200K .......... .......... .......... .......... .......... 18% 95,5M 2s\n",
      "  4250K .......... .......... .......... .......... .......... 18%  152M 2s\n",
      "  4300K .......... .......... .......... .......... .......... 18%  162M 2s\n",
      "  4350K .......... .......... .......... .......... .......... 18%  138M 2s\n",
      "  4400K .......... .......... .......... .......... .......... 18%  160M 2s\n",
      "  4450K .......... .......... .......... .......... .......... 19%  127M 2s\n",
      "  4500K .......... .......... .......... .......... .......... 19%  141M 2s\n",
      "  4550K .......... .......... .......... .......... .......... 19%  136M 2s\n",
      "  4600K .......... .......... .......... .......... .......... 19%  120M 2s\n",
      "  4650K .......... .......... .......... .......... .......... 19%  146M 2s\n",
      "  4700K .......... .......... .......... .......... .......... 20% 69,1M 2s\n",
      "  4750K .......... .......... .......... .......... .......... 20%  147M 2s\n",
      "  4800K .......... .......... .......... .......... .......... 20%  148M 2s\n",
      "  4850K .......... .......... .......... .......... .......... 20%  122M 2s\n",
      "  4900K .......... .......... .......... .......... .......... 21%  150M 2s\n",
      "  4950K .......... .......... .......... .......... .......... 21%  128M 2s\n",
      "  5000K .......... .......... .......... .......... .......... 21%  145M 2s\n",
      "  5050K .......... .......... .......... .......... .......... 21%  221M 2s\n",
      "  5100K .......... .......... .......... .......... .......... 21%  177M 2s\n",
      "  5150K .......... .......... .......... .......... .......... 22%  180M 2s\n",
      "  5200K .......... .......... .......... .......... .......... 22%  103M 2s\n",
      "  5250K .......... .......... .......... .......... .......... 22%  118M 2s\n",
      "  5300K .......... .......... .......... .......... .......... 22%  121M 2s\n",
      "  5350K .......... .......... .......... .......... .......... 22%  102M 2s\n",
      "  5400K .......... .......... .......... .......... .......... 23%  122M 2s\n",
      "  5450K .......... .......... .......... .......... .......... 23%  103M 2s\n",
      "  5500K .......... .......... .......... .......... .......... 23%  116M 2s\n",
      "  5550K .......... .......... .......... .......... .......... 23%  121M 2s\n",
      "  5600K .......... .......... .......... .......... .......... 24%  103M 2s\n",
      "  5650K .......... .......... .......... .......... .......... 24%  117M 2s\n",
      "  5700K .......... .......... .......... .......... .......... 24%  102M 2s\n",
      "  5750K .......... .......... .......... .......... .......... 24%  123M 2s\n",
      "  5800K .......... .......... .......... .......... .......... 24%  120M 1s\n",
      "  5850K .......... .......... .......... .......... .......... 25%  103M 1s\n",
      "  5900K .......... .......... .......... .......... .......... 25%  109M 1s\n",
      "  5950K .......... .......... .......... .......... .......... 25%  105M 1s\n",
      "  6000K .......... .......... .......... .......... .......... 25%  127M 1s\n",
      "  6050K .......... .......... .......... .......... .......... 25%  115M 1s\n",
      "  6100K .......... .......... .......... .......... .......... 26%  102M 1s\n",
      "  6150K .......... .......... .......... .......... .......... 26%  123M 1s\n",
      "  6200K .......... .......... .......... .......... .......... 26%  103M 1s\n",
      "  6250K .......... .......... .......... .......... .......... 26%  118M 1s\n",
      "  6300K .......... .......... .......... .......... .......... 26%  121M 1s\n",
      "  6350K .......... .......... .......... .......... .......... 27%  103M 1s\n",
      "  6400K .......... .......... .......... .......... .......... 27%  120M 1s\n",
      "  6450K .......... .......... .......... .......... .......... 27%  103M 1s\n",
      "  6500K .......... .......... .......... .......... .......... 27%  117M 1s\n",
      "  6550K .......... .......... .......... .......... .......... 28%  121M 1s\n",
      "  6600K .......... .......... .......... .......... .......... 28%  102M 1s\n",
      "  6650K .......... .......... .......... .......... .......... 28%  118M 1s\n",
      "  6700K .......... .......... .......... .......... .......... 28%  100M 1s\n",
      "  6750K .......... .......... .......... .......... .......... 28%  124M 1s\n",
      "  6800K .......... .......... .......... .......... .......... 29%  117M 1s\n",
      "  6850K .......... .......... .......... .......... .......... 29%  102M 1s\n",
      "  6900K .......... .......... .......... .......... .......... 29%  121M 1s\n",
      "  6950K .......... .......... .......... .......... .......... 29% 1,53M 1s\n",
      "  7000K .......... .......... .......... .......... .......... 29% 46,8M 1s\n",
      "  7050K .......... .......... .......... .......... .......... 30% 47,9M 1s\n",
      "  7100K .......... .......... .......... .......... .......... 30% 61,4M 1s\n",
      "  7150K .......... .......... .......... .......... .......... 30% 59,7M 1s\n",
      "  7200K .......... .......... .......... .......... .......... 30% 55,6M 1s\n",
      "  7250K .......... .......... .......... .......... .......... 31%  111M 1s\n",
      "  7300K .......... .......... .......... .......... .......... 31% 48,8M 1s\n",
      "  7350K .......... .......... .......... .......... .......... 31% 52,4M 1s\n",
      "  7400K .......... .......... .......... .......... .......... 31%  111M 1s\n",
      "  7450K .......... .......... .......... .......... .......... 31%  104M 1s\n",
      "  7500K .......... .......... .......... .......... .......... 32%  124M 1s\n",
      "  7550K .......... .......... .......... .......... .......... 32%  113M 1s\n",
      "  7600K .......... .......... .......... .......... .......... 32%  107M 1s\n",
      "  7650K .......... .......... .......... .......... .......... 32%  122M 1s\n",
      "  7700K .......... .......... .......... .......... .......... 32%  121M 1s\n",
      "  7750K .......... .......... .......... .......... .......... 33%  143M 1s\n",
      "  7800K .......... .......... .......... .......... .......... 33%  133M 1s\n",
      "  7850K .......... .......... .......... .......... .......... 33% 98,2M 1s\n",
      "  7900K .......... .......... .......... .......... .......... 33%  122M 1s\n",
      "  7950K .......... .......... .......... .......... .......... 33%  104M 1s\n",
      "  8000K .......... .......... .......... .......... .......... 34%  122M 1s\n",
      "  8050K .......... .......... .......... .......... .......... 34%  117M 1s\n",
      "  8100K .......... .......... .......... .......... .......... 34%  103M 1s\n",
      "  8150K .......... .......... .......... .......... .......... 34%  121M 1s\n",
      "  8200K .......... .......... .......... .......... .......... 35%  102M 1s\n",
      "  8250K .......... .......... .......... .......... .......... 35%  119M 1s\n",
      "  8300K .......... .......... .......... .......... .......... 35%  121M 1s\n",
      "  8350K .......... .......... .......... .......... .......... 35%  102M 1s\n",
      "  8400K .......... .......... .......... .......... .......... 35%  117M 1s\n",
      "  8450K .......... .......... .......... .......... .......... 36%  103M 1s\n",
      "  8500K .......... .......... .......... .......... .......... 36%  121M 1s\n",
      "  8550K .......... .......... .......... .......... .......... 36%  121M 1s\n",
      "  8600K .......... .......... .......... .......... .......... 36%  103M 1s\n",
      "  8650K .......... .......... .......... .......... .......... 36%  117M 1s\n",
      "  8700K .......... .......... .......... .......... .......... 37%  102M 1s\n",
      "  8750K .......... .......... .......... .......... .......... 37%  100M 1s\n",
      "  8800K .......... .......... .......... .......... .......... 37%  147M 1s\n",
      "  8850K .......... .......... .......... .......... .......... 37%  103M 1s\n",
      "  8900K .......... .......... .......... .......... .......... 38%  121M 1s\n",
      "  8950K .......... .......... .......... .......... .......... 38%  102M 1s\n",
      "  9000K .......... .......... .......... .......... .......... 38%  117M 1s\n",
      "  9050K .......... .......... .......... .......... .......... 38%  121M 1s\n",
      "  9100K .......... .......... .......... .......... .......... 38%  102M 1s\n",
      "  9150K .......... .......... .......... .......... .......... 39%  121M 1s\n",
      "  9200K .......... .......... .......... .......... .......... 39%  102M 1s\n",
      "  9250K .......... .......... .......... .......... .......... 39%  117M 1s\n",
      "  9300K .......... .......... .......... .......... .......... 39%  121M 1s\n",
      "  9350K .......... .......... .......... .......... .......... 39%  102M 1s\n",
      "  9400K .......... .......... .......... .......... .......... 40%  117M 1s\n",
      "  9450K .......... .......... .......... .......... .......... 40%  102M 1s\n",
      "  9500K .......... .......... .......... .......... .......... 40%  121M 1s\n",
      "  9550K .......... .......... .......... .......... .......... 40%  121M 1s\n",
      "  9600K .......... .......... .......... .......... .......... 40%  102M 1s\n",
      "  9650K .......... .......... .......... .......... .......... 41%  117M 1s\n",
      "  9700K .......... .......... .......... .......... .......... 41%  102M 1s\n",
      "  9750K .......... .......... .......... .......... .......... 41%  122M 1s\n",
      "  9800K .......... .......... .......... .......... .......... 41%  117M 1s\n",
      "  9850K .......... .......... .......... .......... .......... 42%  102M 1s\n",
      "  9900K .......... .......... .......... .......... .......... 42%  121M 1s\n",
      "  9950K .......... .......... .......... .......... .......... 42%  102M 1s\n",
      " 10000K .......... .......... .......... .......... .......... 42% 1,70M 1s\n",
      " 10050K .......... .......... .......... .......... .......... 42% 17,2M 1s\n",
      " 10100K .......... .......... .......... .......... .......... 43% 47,8M 1s\n",
      " 10150K .......... .......... .......... .......... .......... 43%  133M 1s\n",
      " 10200K .......... .......... .......... .......... .......... 43% 56,0M 1s\n",
      " 10250K .......... .......... .......... .......... .......... 43% 57,9M 1s\n",
      " 10300K .......... .......... .......... .......... .......... 43%  110M 1s\n",
      " 10350K .......... .......... .......... .......... .......... 44%  111M 1s\n",
      " 10400K .......... .......... .......... .......... .......... 44%  109M 1s\n",
      " 10450K .......... .......... .......... .......... .......... 44% 50,3M 1s\n",
      " 10500K .......... .......... .......... .......... .......... 44% 32,8M 1s\n",
      " 10550K .......... .......... .......... .......... .......... 45%  133M 1s\n",
      " 10600K .......... .......... .......... .......... .......... 45%  103M 1s\n",
      " 10650K .......... .......... .......... .......... .......... 45%  121M 1s\n",
      " 10700K .......... .......... .......... .......... .......... 45%  102M 1s\n",
      " 10750K .......... .......... .......... .......... .......... 45%  121M 1s\n",
      " 10800K .......... .......... .......... .......... .......... 46% 92,4M 1s\n",
      " 10850K .......... .......... .......... .......... .......... 46%  135M 1s\n",
      " 10900K .......... .......... .......... .......... .......... 46%  121M 1s\n",
      " 10950K .......... .......... .......... .......... .......... 46%  102M 1s\n",
      " 11000K .......... .......... .......... .......... .......... 46%  117M 1s\n",
      " 11050K .......... .......... .......... .......... .......... 47%  121M 1s\n",
      " 11100K .......... .......... .......... .......... .......... 47%  104M 1s\n",
      " 11150K .......... .......... .......... .......... .......... 47%  117M 1s\n",
      " 11200K .......... .......... .......... .......... .......... 47%  104M 1s\n",
      " 11250K .......... .......... .......... .......... .......... 48%  118M 1s\n",
      " 11300K .......... .......... .......... .......... .......... 48%  121M 1s\n",
      " 11350K .......... .......... .......... .......... .......... 48%  103M 1s\n",
      " 11400K .......... .......... .......... .......... .......... 48%  117M 1s\n",
      " 11450K .......... .......... .......... .......... .......... 48%  103M 1s\n",
      " 11500K .......... .......... .......... .......... .......... 49%  121M 1s\n",
      " 11550K .......... .......... .......... .......... .......... 49%  117M 1s\n",
      " 11600K .......... .......... .......... .......... .......... 49%  102M 1s\n",
      " 11650K .......... .......... .......... .......... .......... 49%  121M 1s\n",
      " 11700K .......... .......... .......... .......... .......... 49%  102M 1s\n",
      " 11750K .......... .......... .......... .......... .......... 50%  121M 1s\n",
      " 11800K .......... .......... .......... .......... .......... 50%  117M 1s\n",
      " 11850K .......... .......... .......... .......... .......... 50%  103M 1s\n",
      " 11900K .......... .......... .......... .......... .......... 50%  121M 1s\n",
      " 11950K .......... .......... .......... .......... .......... 50%  102M 1s\n",
      " 12000K .......... .......... .......... .......... .......... 51%  117M 1s\n",
      " 12050K .......... .......... .......... .......... .......... 51%  121M 1s\n",
      " 12100K .......... .......... .......... .......... .......... 51%  102M 1s\n",
      " 12150K .......... .......... .......... .......... .......... 51%  117M 1s\n",
      " 12200K .......... .......... .......... .......... .......... 52%  102M 1s\n",
      " 12250K .......... .......... .......... .......... .......... 52%  120M 1s\n",
      " 12300K .......... .......... .......... .......... .......... 52%  122M 1s\n",
      " 12350K .......... .......... .......... .......... .......... 52%  102M 1s\n",
      " 12400K .......... .......... .......... .......... .......... 52%  117M 1s\n",
      " 12450K .......... .......... .......... .......... .......... 53%  102M 1s\n",
      " 12500K .......... .......... .......... .......... .......... 53%  121M 1s\n",
      " 12550K .......... .......... .......... .......... .......... 53%  117M 1s\n",
      " 12600K .......... .......... .......... .......... .......... 53%  103M 1s\n",
      " 12650K .......... .......... .......... .......... .......... 53%  120M 1s\n",
      " 12700K .......... .......... .......... .......... .......... 54%  103M 1s\n",
      " 12750K .......... .......... .......... .......... .......... 54%  117M 1s\n",
      " 12800K .......... .......... .......... .......... .......... 54%  121M 1s\n",
      " 12850K .......... .......... .......... .......... .......... 54%  102M 1s\n",
      " 12900K .......... .......... .......... .......... .......... 55%  121M 1s\n",
      " 12950K .......... .......... .......... .......... .......... 55%  102M 1s\n",
      " 13000K .......... .......... .......... .......... .......... 55%  117M 1s\n",
      " 13050K .......... .......... .......... .......... .......... 55%  121M 0s\n",
      " 13100K .......... .......... .......... .......... .......... 55% 1,62M 1s\n",
      " 13150K .......... .......... .......... .......... .......... 56% 41,1M 1s\n",
      " 13200K .......... .......... .......... .......... .......... 56% 57,9M 1s\n",
      " 13250K .......... .......... .......... .......... .......... 56%  102M 1s\n",
      " 13300K .......... .......... .......... .......... .......... 56% 59,6M 1s\n",
      " 13350K .......... .......... .......... .......... .......... 56% 62,1M 0s\n",
      " 13400K .......... .......... .......... .......... .......... 57% 50,5M 0s\n",
      " 13450K .......... .......... .......... .......... .......... 57%  102M 0s\n",
      " 13500K .......... .......... .......... .......... .......... 57%  121M 0s\n",
      " 13550K .......... .......... .......... .......... .......... 57%  118M 0s\n",
      " 13600K .......... .......... .......... .......... .......... 57% 52,4M 0s\n",
      " 13650K .......... .......... .......... .......... .......... 58% 28,6M 0s\n",
      " 13700K .......... .......... .......... .......... .......... 58%  103M 0s\n",
      " 13750K .......... .......... .......... .......... .......... 58%  117M 0s\n",
      " 13800K .......... .......... .......... .......... .......... 58%  123M 0s\n",
      " 13850K .......... .......... .......... .......... .......... 59%  102M 0s\n",
      " 13900K .......... .......... .......... .......... .......... 59%  121M 0s\n",
      " 13950K .......... .......... .......... .......... .......... 59%  101M 0s\n",
      " 14000K .......... .......... .......... .......... .......... 59%  111M 0s\n",
      " 14050K .......... .......... .......... .......... .......... 59% 57,8M 0s\n",
      " 14100K .......... .......... .......... .......... .......... 60% 63,1M 0s\n",
      " 14150K .......... .......... .......... .......... .......... 60% 68,5M 0s\n",
      " 14200K .......... .......... .......... .......... .......... 60% 61,2M 0s\n",
      " 14250K .......... .......... .......... .......... .......... 60% 69,1M 0s\n",
      " 14300K .......... .......... .......... .......... .......... 60% 69,5M 0s\n",
      " 14350K .......... .......... .......... .......... .......... 61%  119M 0s\n",
      " 14400K .......... .......... .......... .......... .......... 61%  145M 0s\n",
      " 14450K .......... .......... .......... .......... .......... 61%  132M 0s\n",
      " 14500K .......... .......... .......... .......... .......... 61%  149M 0s\n",
      " 14550K .......... .......... .......... .......... .......... 62%  145M 0s\n",
      " 14600K .......... .......... .......... .......... .......... 62%  129M 0s\n",
      " 14650K .......... .......... .......... .......... .......... 62%  142M 0s\n",
      " 14700K .......... .......... .......... .......... .......... 62%  127M 0s\n",
      " 14750K .......... .......... .......... .......... .......... 62%  148M 0s\n",
      " 14800K .......... .......... .......... .......... .......... 63%  146M 0s\n",
      " 14850K .......... .......... .......... .......... .......... 63%  123M 0s\n",
      " 14900K .......... .......... .......... .......... .......... 63% 75,2M 0s\n",
      " 14950K .......... .......... .......... .......... .......... 63%  142M 0s\n",
      " 15000K .......... .......... .......... .......... .......... 63%  276M 0s\n",
      " 15050K .......... .......... .......... .......... .......... 64%  280M 0s\n",
      " 15100K .......... .......... .......... .......... .......... 64%  154M 0s\n",
      " 15150K .......... .......... .......... .......... .......... 64%  145M 0s\n",
      " 15200K .......... .......... .......... .......... .......... 64%  147M 0s\n",
      " 15250K .......... .......... .......... .......... .......... 64%  235M 0s\n",
      " 15300K .......... .......... .......... .......... .......... 65%  182M 0s\n",
      " 15350K .......... .......... .......... .......... .......... 65%  103M 0s\n",
      " 15400K .......... .......... .......... .......... .......... 65%  120M 0s\n",
      " 15450K .......... .......... .......... .......... .......... 65%  101M 0s\n",
      " 15500K .......... .......... .......... .......... .......... 66%  117M 0s\n",
      " 15550K .......... .......... .......... .......... .......... 66%  122M 0s\n",
      " 15600K .......... .......... .......... .......... .......... 66%  102M 0s\n",
      " 15650K .......... .......... .......... .......... .......... 66% 93,7M 0s\n",
      " 15700K .......... .......... .......... .......... .......... 66%  136M 0s\n",
      " 15750K .......... .......... .......... .......... .......... 67%  110M 0s\n",
      " 15800K .......... .......... .......... .......... .......... 67%  127M 0s\n",
      " 15850K .......... .......... .......... .......... .......... 67% 99,8M 0s\n",
      " 15900K .......... .......... .......... .......... .......... 67%  118M 0s\n",
      " 15950K .......... .......... .......... .......... .......... 67%  105M 0s\n",
      " 16000K .......... .......... .......... .......... .......... 68%  118M 0s\n",
      " 16050K .......... .......... .......... .......... .......... 68%  123M 0s\n",
      " 16100K .......... .......... .......... .......... .......... 68% 1,96M 0s\n",
      " 16150K .......... .......... .......... .......... .......... 68% 10,8M 0s\n",
      " 16200K .......... .......... .......... .......... .......... 69% 32,0M 0s\n",
      " 16250K .......... .......... .......... .......... .......... 69% 59,4M 0s\n",
      " 16300K .......... .......... .......... .......... .......... 69% 69,0M 0s\n",
      " 16350K .......... .......... .......... .......... .......... 69% 55,2M 0s\n",
      " 16400K .......... .......... .......... .......... .......... 69% 82,5M 0s\n",
      " 16450K .......... .......... .......... .......... .......... 70%  112M 0s\n",
      " 16500K .......... .......... .......... .......... .......... 70%  139M 0s\n",
      " 16550K .......... .......... .......... .......... .......... 70% 56,9M 0s\n",
      " 16600K .......... .......... .......... .......... .......... 70% 59,2M 0s\n",
      " 16650K .......... .......... .......... .......... .......... 70%  123M 0s\n",
      " 16700K .......... .......... .......... .......... .......... 71% 39,3M 0s\n",
      " 16750K .......... .......... .......... .......... .......... 71%  105M 0s\n",
      " 16800K .......... .......... .......... .......... .......... 71% 56,8M 0s\n",
      " 16850K .......... .......... .......... .......... .......... 71%  101M 0s\n",
      " 16900K .......... .......... .......... .......... .......... 72%  118M 0s\n",
      " 16950K .......... .......... .......... .......... .......... 72% 57,9M 0s\n",
      " 17000K .......... .......... .......... .......... .......... 72%  129M 0s\n",
      " 17050K .......... .......... .......... .......... .......... 72%  129M 0s\n",
      " 17100K .......... .......... .......... .......... .......... 72%  104M 0s\n",
      " 17150K .......... .......... .......... .......... .......... 73%  131M 0s\n",
      " 17200K .......... .......... .......... .......... .......... 73% 64,2M 0s\n",
      " 17250K .......... .......... .......... .......... .......... 73% 55,5M 0s\n",
      " 17300K .......... .......... .......... .......... .......... 73% 64,0M 0s\n",
      " 17350K .......... .......... .......... .......... .......... 73%  101M 0s\n",
      " 17400K .......... .......... .......... .......... .......... 74%  121M 0s\n",
      " 17450K .......... .......... .......... .......... .......... 74%  103M 0s\n",
      " 17500K .......... .......... .......... .......... .......... 74% 65,6M 0s\n",
      " 17550K .......... .......... .......... .......... .......... 74%  122M 0s\n",
      " 17600K .......... .......... .......... .......... .......... 74%  102M 0s\n",
      " 17650K .......... .......... .......... .......... .......... 75%  122M 0s\n",
      " 17700K .......... .......... .......... .......... .......... 75%  102M 0s\n",
      " 17750K .......... .......... .......... .......... .......... 75%  116M 0s\n",
      " 17800K .......... .......... .......... .......... .......... 75%  124M 0s\n",
      " 17850K .......... .......... .......... .......... .......... 76%  103M 0s\n",
      " 17900K .......... .......... .......... .......... .......... 76%  117M 0s\n",
      " 17950K .......... .......... .......... .......... .......... 76%  102M 0s\n",
      " 18000K .......... .......... .......... .......... .......... 76%  122M 0s\n",
      " 18050K .......... .......... .......... .......... .......... 76%  117M 0s\n",
      " 18100K .......... .......... .......... .......... .......... 77%  103M 0s\n",
      " 18150K .......... .......... .......... .......... .......... 77%  121M 0s\n",
      " 18200K .......... .......... .......... .......... .......... 77%  103M 0s\n",
      " 18250K .......... .......... .......... .......... .......... 77%  120M 0s\n",
      " 18300K .......... .......... .......... .......... .......... 77%  117M 0s\n",
      " 18350K .......... .......... .......... .......... .......... 78%  102M 0s\n",
      " 18400K .......... .......... .......... .......... .......... 78%  122M 0s\n",
      " 18450K .......... .......... .......... .......... .......... 78%  102M 0s\n",
      " 18500K .......... .......... .......... .......... .......... 78%  117M 0s\n",
      " 18550K .......... .......... .......... .......... .......... 79%  121M 0s\n",
      " 18600K .......... .......... .......... .......... .......... 79%  102M 0s\n",
      " 18650K .......... .......... .......... .......... .......... 79%  118M 0s\n",
      " 18700K .......... .......... .......... .......... .......... 79%  105M 0s\n",
      " 18750K .......... .......... .......... .......... .......... 79%  118M 0s\n",
      " 18800K .......... .......... .......... .......... .......... 80%  120M 0s\n",
      " 18850K .......... .......... .......... .......... .......... 80%  102M 0s\n",
      " 18900K .......... .......... .......... .......... .......... 80%  118M 0s\n",
      " 18950K .......... .......... .......... .......... .......... 80%  102M 0s\n",
      " 19000K .......... .......... .......... .......... .......... 80%  121M 0s\n",
      " 19050K .......... .......... .......... .......... .......... 81%  118M 0s\n",
      " 19100K .......... .......... .......... .......... .......... 81%  102M 0s\n",
      " 19150K .......... .......... .......... .......... .......... 81% 2,15M 0s\n",
      " 19200K .......... .......... .......... .......... .......... 81% 9,26M 0s\n",
      " 19250K .......... .......... .......... .......... .......... 81% 52,9M 0s\n",
      " 19300K .......... .......... .......... .......... .......... 82% 51,0M 0s\n",
      " 19350K .......... .......... .......... .......... .......... 82% 50,3M 0s\n",
      " 19400K .......... .......... .......... .......... .......... 82% 61,4M 0s\n",
      " 19450K .......... .......... .......... .......... .......... 82% 59,3M 0s\n",
      " 19500K .......... .......... .......... .......... .......... 83%  115M 0s\n",
      " 19550K .......... .......... .......... .......... .......... 83%  111M 0s\n",
      " 19600K .......... .......... .......... .......... .......... 83%  105M 0s\n",
      " 19650K .......... .......... .......... .......... .......... 83%  126M 0s\n",
      " 19700K .......... .......... .......... .......... .......... 83%  106M 0s\n",
      " 19750K .......... .......... .......... .......... .......... 84% 64,4M 0s\n",
      " 19800K .......... .......... .......... .......... .......... 84% 53,5M 0s\n",
      " 19850K .......... .......... .......... .......... .......... 84% 37,3M 0s\n",
      " 19900K .......... .......... .......... .......... .......... 84% 57,1M 0s\n",
      " 19950K .......... .......... .......... .......... .......... 84%  103M 0s\n",
      " 20000K .......... .......... .......... .......... .......... 85%  124M 0s\n",
      " 20050K .......... .......... .......... .......... .......... 85%  118M 0s\n",
      " 20100K .......... .......... .......... .......... .......... 85%  102M 0s\n",
      " 20150K .......... .......... .......... .......... .......... 85%  119M 0s\n",
      " 20200K .......... .......... .......... .......... .......... 86%  103M 0s\n",
      " 20250K .......... .......... .......... .......... .......... 86%  120M 0s\n",
      " 20300K .......... .......... .......... .......... .......... 86% 93,5M 0s\n",
      " 20350K .......... .......... .......... .......... .......... 86% 42,0M 0s\n",
      " 20400K .......... .......... .......... .......... .......... 86% 38,4M 0s\n",
      " 20450K .......... .......... .......... .......... .......... 87%  102M 0s\n",
      " 20500K .......... .......... .......... .......... .......... 87%  117M 0s\n",
      " 20550K .......... .......... .......... .......... .......... 87%  121M 0s\n",
      " 20600K .......... .......... .......... .......... .......... 87%  102M 0s\n",
      " 20650K .......... .......... .......... .......... .......... 87%  117M 0s\n",
      " 20700K .......... .......... .......... .......... .......... 88%  102M 0s\n",
      " 20750K .......... .......... .......... .......... .......... 88%  121M 0s\n",
      " 20800K .......... .......... .......... .......... .......... 88%  121M 0s\n",
      " 20850K .......... .......... .......... .......... .......... 88%  103M 0s\n",
      " 20900K .......... .......... .......... .......... .......... 88%  117M 0s\n",
      " 20950K .......... .......... .......... .......... .......... 89%  102M 0s\n",
      " 21000K .......... .......... .......... .......... .......... 89%  122M 0s\n",
      " 21050K .......... .......... .......... .......... .......... 89%  118M 0s\n",
      " 21100K .......... .......... .......... .......... .......... 89%  103M 0s\n",
      " 21150K .......... .......... .......... .......... .......... 90%  120M 0s\n",
      " 21200K .......... .......... .......... .......... .......... 90%  103M 0s\n",
      " 21250K .......... .......... .......... .......... .......... 90%  118M 0s\n",
      " 21300K .......... .......... .......... .......... .......... 90%  120M 0s\n",
      " 21350K .......... .......... .......... .......... .......... 90%  103M 0s\n",
      " 21400K .......... .......... .......... .......... .......... 91%  120M 0s\n",
      " 21450K .......... .......... .......... .......... .......... 91%  103M 0s\n",
      " 21500K .......... .......... .......... .......... .......... 91%  116M 0s\n",
      " 21550K .......... .......... .......... .......... .......... 91%  121M 0s\n",
      " 21600K .......... .......... .......... .......... .......... 91%  103M 0s\n",
      " 21650K .......... .......... .......... .......... .......... 92%  116M 0s\n",
      " 21700K .......... .......... .......... .......... .......... 92%  103M 0s\n",
      " 21750K .......... .......... .......... .......... .......... 92%  121M 0s\n",
      " 21800K .......... .......... .......... .......... .......... 92%  117M 0s\n",
      " 21850K .......... .......... .......... .......... .......... 93%  102M 0s\n",
      " 21900K .......... .......... .......... .......... .......... 93%  121M 0s\n",
      " 21950K .......... .......... .......... .......... .......... 93%  102M 0s\n",
      " 22000K .......... .......... .......... .......... .......... 93%  121M 0s\n",
      " 22050K .......... .......... .......... .......... .......... 93%  118M 0s\n",
      " 22100K .......... .......... .......... .......... .......... 94%  102M 0s\n",
      " 22150K .......... .......... .......... .......... .......... 94%  121M 0s\n",
      " 22200K .......... .......... .......... .......... .......... 94% 2,19M 0s\n",
      " 22250K .......... .......... .......... .......... .......... 94% 21,3M 0s\n",
      " 22300K .......... .......... .......... .......... .......... 94% 10,6M 0s\n",
      " 22350K .......... .......... .......... .......... .......... 95% 47,6M 0s\n",
      " 22400K .......... .......... .......... .......... .......... 95%  122M 0s\n",
      " 22450K .......... .......... .......... .......... .......... 95% 55,4M 0s\n",
      " 22500K .......... .......... .......... .......... .......... 95%  118M 0s\n",
      " 22550K .......... .......... .......... .......... .......... 96%  104M 0s\n",
      " 22600K .......... .......... .......... .......... .......... 96% 66,9M 0s\n",
      " 22650K .......... .......... .......... .......... .......... 96%  116M 0s\n",
      " 22700K .......... .......... .......... .......... .......... 96% 50,4M 0s\n",
      " 22750K .......... .......... .......... .......... .......... 96%  121M 0s\n",
      " 22800K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 22850K .......... .......... .......... .......... .......... 97% 55,0M 0s\n",
      " 22900K .......... .......... .......... .......... .......... 97% 40,4M 0s\n",
      " 22950K .......... .......... .......... .......... .......... 97% 44,6M 0s\n",
      " 23000K .......... .......... .......... .......... .......... 97% 66,0M 0s\n",
      " 23050K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 23100K .......... .......... .......... .......... .......... 98%  102M 0s\n",
      " 23150K .......... .......... .......... .......... .......... 98%  121M 0s\n",
      " 23200K .......... .......... .......... .......... .......... 98%  102M 0s\n",
      " 23250K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 23300K .......... .......... .......... .......... .......... 99%  121M 0s\n",
      " 23350K .......... .......... .......... .......... .......... 99%  102M 0s\n",
      " 23400K .......... .......... .......... .......... .......... 99%  117M 0s\n",
      " 23450K .......... .......... .......... .......... .......... 99% 68,8M 0s\n",
      " 23500K .......... .......... .......... .........            100% 27,9M=0,8s\n",
      "\n",
      "2023-08-29 17:50:30 (27,1 MB/s) - ‘data/reference/genome.fa.gz’ guardado [24104573/24104573]\n",
      "\n",
      "[Tue Aug 29 17:50:31 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T175028.473823.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:50:32 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR696683_chr17_1.fastq.gz, data/raw/ERR696683_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR696683_chr17_1_fastp.fastq.gz, data/processed/ERR696683_chr17_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR696683_chr17_2_fastp.fastq.gz, data/processed/ERR696683_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1662053\n",
      "total bases: 125542423\n",
      "Q20 bases: 124610599(99.2578%)\n",
      "Q30 bases: 120902230(96.3039%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1662053\n",
      "total bases: 125166559\n",
      "Q20 bases: 123458540(98.6354%)\n",
      "Q30 bases: 117801905(94.1161%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 941109\n",
      "total bases: 71474317\n",
      "Q20 bases: 71185568(99.596%)\n",
      "Q30 bases: 69857451(97.7378%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 941109\n",
      "total bases: 71445379\n",
      "Q20 bases: 70903174(99.2411%)\n",
      "Q30 bases: 68848719(96.3655%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1882218\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1441888\n",
      "reads with adapter trimmed: 1178\n",
      "bases trimmed due to adapters: 33118\n",
      "\n",
      "Duplication rate: 5.19851%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR696683_chr17_1.fastq.gz -I data/raw/ERR696683_chr17_2.fastq.gz -o data/processed/ERR696683_chr17_1_fastp.fastq.gz -O data/processed/ERR696683_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Tue Aug 29 17:50:37 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:50:37 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR696683_chr17_1.fastq.gz, data/raw/ERR696683_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR696683_chr17_1_fastqc.html, results/fastqc_result/ERR696683_chr17_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR696683_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 17:50:50 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:50:50 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR696683_chr17_1_fastp.fastq.gz, data/processed/ERR696683_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR696683_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR696683_chr17_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR696683_chr17_2_fastp.fastq.gz, data/processed/ERR696683_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 17:50:59 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T175031.313206.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:51:00 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753368_chr17_1.fastq.gz, data/raw/ERR753368_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753368_chr17_1_fastqc.html, results/fastqc_result/ERR753368_chr17_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753368_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753368_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 17:51:13 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:51:13 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753368_chr17_1.fastq.gz, data/raw/ERR753368_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753368_chr17_1_fastp.fastq.gz, data/processed/ERR753368_chr17_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753368_chr17_2_fastp.fastq.gz, data/processed/ERR753368_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1720078\n",
      "total bases: 130586939\n",
      "Q20 bases: 130062674(99.5985%)\n",
      "Q30 bases: 126025291(96.5068%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1720078\n",
      "total bases: 130408498\n",
      "Q20 bases: 129167916(99.0487%)\n",
      "Q30 bases: 122638409(94.0417%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1449801\n",
      "total bases: 110178424\n",
      "Q20 bases: 109981943(99.8217%)\n",
      "Q30 bases: 108017025(98.0383%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1449801\n",
      "total bases: 110167371\n",
      "Q20 bases: 109584045(99.4705%)\n",
      "Q30 bases: 105965050(96.1855%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2899602\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 540554\n",
      "reads with adapter trimmed: 1280\n",
      "bases trimmed due to adapters: 33381\n",
      "\n",
      "Duplication rate: 1.94638%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753368_chr17_1.fastq.gz -I data/raw/ERR753368_chr17_2.fastq.gz -o data/processed/ERR753368_chr17_1_fastp.fastq.gz -O data/processed/ERR753368_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Tue Aug 29 17:51:19 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:51:19 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753368_chr17_1_fastp.fastq.gz, data/processed/ERR753368_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753368_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753368_chr17_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753368_chr17_2_fastp.fastq.gz, data/processed/ERR753368_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 17:51:31 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T175059.687410.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:51:32 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753369_chr17_1.fastq.gz, data/raw/ERR753369_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753369_chr17_1_fastqc.html, results/fastqc_result/ERR753369_chr17_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753369_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753369_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 17:51:47 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:51:47 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753369_chr17_1.fastq.gz, data/raw/ERR753369_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753369_chr17_1_fastp.fastq.gz, data/processed/ERR753369_chr17_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753369_chr17_2_fastp.fastq.gz, data/processed/ERR753369_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1834306\n",
      "total bases: 139253654\n",
      "Q20 bases: 138670028(99.5809%)\n",
      "Q30 bases: 134266277(96.4185%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1834306\n",
      "total bases: 139072650\n",
      "Q20 bases: 137744780(99.0452%)\n",
      "Q30 bases: 130771387(94.031%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1545614\n",
      "total bases: 117459470\n",
      "Q20 bases: 117236110(99.8098%)\n",
      "Q30 bases: 115069005(97.9649%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1545614\n",
      "total bases: 117448151\n",
      "Q20 bases: 116824626(99.4691%)\n",
      "Q30 bases: 112958776(96.1776%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3091228\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 577384\n",
      "reads with adapter trimmed: 1210\n",
      "bases trimmed due to adapters: 30992\n",
      "\n",
      "Duplication rate: 2.23195%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753369_chr17_1.fastq.gz -I data/raw/ERR753369_chr17_2.fastq.gz -o data/processed/ERR753369_chr17_1_fastp.fastq.gz -O data/processed/ERR753369_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Tue Aug 29 17:51:53 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:51:53 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753369_chr17_1_fastp.fastq.gz, data/processed/ERR753369_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753369_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753369_chr17_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753369_chr17_2_fastp.fastq.gz, data/processed/ERR753369_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 17:52:06 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T175131.379059.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:52:07 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753370_chr17_1.fastq.gz, data/raw/ERR753370_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753370_chr17_1_fastp.fastq.gz, data/processed/ERR753370_chr17_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753370_chr17_1_fastp.fastq.gz, data/processed/ERR753370_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2195518\n",
      "total bases: 220903960\n",
      "Q20 bases: 219316637(99.2814%)\n",
      "Q30 bases: 212394069(96.1477%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2195518\n",
      "total bases: 220842678\n",
      "Q20 bases: 218611707(98.9898%)\n",
      "Q30 bases: 209497343(94.8627%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 2194882\n",
      "total bases: 219845045\n",
      "Q20 bases: 218360012(99.3245%)\n",
      "Q30 bases: 211791798(96.3369%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 2194882\n",
      "total bases: 219058701\n",
      "Q20 bases: 217044793(99.0807%)\n",
      "Q30 bases: 208534267(95.1956%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 4389764\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1272\n",
      "reads with adapter trimmed: 11652\n",
      "bases trimmed due to adapters: 109912\n",
      "\n",
      "Duplication rate: 4.00825%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 161\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753370_chr17_1.fastq.gz -I data/raw/ERR753370_chr17_2.fastq.gz -o data/processed/ERR753370_chr17_1_fastp.fastq.gz -O data/processed/ERR753370_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 10 seconds\n",
      "[Tue Aug 29 17:52:17 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:52:17 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753370_chr17_1_fastp.fastq.gz, data/processed/ERR753370_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753370_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753370_chr17_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753370_chr17_1_fastp.fastq.gz, data/processed/ERR753370_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 17:52:38 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:52:38 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753370_chr17_1.fastq.gz, data/raw/ERR753370_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753370_chr17_1_fastqc.html, results/fastqc_result/ERR753370_chr17_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753370_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753370_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 17:52:59 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T175206.333058.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:53:00 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753371_chr17_1.fastq.gz, data/raw/ERR753371_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753371_chr17_1_fastqc.html, results/fastqc_result/ERR753371_chr17_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753371_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753371_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 17:53:13 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:53:13 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753371_chr17_1.fastq.gz, data/raw/ERR753371_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753371_chr17_1_fastp.fastq.gz, data/processed/ERR753371_chr17_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753371_chr17_2_fastp.fastq.gz, data/processed/ERR753371_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1479982\n",
      "total bases: 111397900\n",
      "Q20 bases: 110370854(99.078%)\n",
      "Q30 bases: 106159867(95.2979%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1479982\n",
      "total bases: 111084814\n",
      "Q20 bases: 109633816(98.6938%)\n",
      "Q30 bases: 105347657(94.8353%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 690305\n",
      "total bases: 52398376\n",
      "Q20 bases: 52131601(99.4909%)\n",
      "Q30 bases: 50765095(96.883%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 690305\n",
      "total bases: 52368274\n",
      "Q20 bases: 51965373(99.2306%)\n",
      "Q30 bases: 50683984(96.7838%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1380610\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1579354\n",
      "reads with adapter trimmed: 980\n",
      "bases trimmed due to adapters: 24008\n",
      "\n",
      "Duplication rate: 19.6085%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 117\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753371_chr17_1.fastq.gz -I data/raw/ERR753371_chr17_2.fastq.gz -o data/processed/ERR753371_chr17_1_fastp.fastq.gz -O data/processed/ERR753371_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Tue Aug 29 17:53:18 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:53:18 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753371_chr17_1_fastp.fastq.gz, data/processed/ERR753371_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753371_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753371_chr17_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753371_chr17_2_fastp.fastq.gz, data/processed/ERR753371_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 17:53:24 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T175259.507042.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:53:26 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753372_chr17_1.fastq.gz, data/raw/ERR753372_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753372_chr17_1_fastqc.html, results/fastqc_result/ERR753372_chr17_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753372_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753372_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 17:53:40 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:53:40 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753372_chr17_1.fastq.gz, data/raw/ERR753372_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753372_chr17_1_fastp.fastq.gz, data/processed/ERR753372_chr17_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753372_chr17_1_fastp.fastq.gz, data/processed/ERR753372_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1891163\n",
      "total bases: 143570978\n",
      "Q20 bases: 142991472(99.5964%)\n",
      "Q30 bases: 138476280(96.4514%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1891163\n",
      "total bases: 143376057\n",
      "Q20 bases: 142013761(99.0498%)\n",
      "Q30 bases: 134881566(94.0754%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1589698\n",
      "total bases: 120809730\n",
      "Q20 bases: 120591835(99.8196%)\n",
      "Q30 bases: 118389976(97.9971%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1589698\n",
      "total bases: 120797532\n",
      "Q20 bases: 120159672(99.472%)\n",
      "Q30 bases: 116232633(96.221%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3179396\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 602930\n",
      "reads with adapter trimmed: 1446\n",
      "bases trimmed due to adapters: 35677\n",
      "\n",
      "Duplication rate: 1.89059%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753372_chr17_1.fastq.gz -I data/raw/ERR753372_chr17_2.fastq.gz -o data/processed/ERR753372_chr17_1_fastp.fastq.gz -O data/processed/ERR753372_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Tue Aug 29 17:53:47 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:53:47 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753372_chr17_1_fastp.fastq.gz, data/processed/ERR753372_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753372_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753372_chr17_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753372_chr17_1_fastp.fastq.gz, data/processed/ERR753372_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 17:54:00 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T175325.097733.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:54:01 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753373_chr17_1.fastq.gz, data/raw/ERR753373_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753373_chr17_1_fastp.fastq.gz, data/processed/ERR753373_chr17_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753373_chr17_2_fastp.fastq.gz, data/processed/ERR753373_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 843274\n",
      "total bases: 80686867\n",
      "Q20 bases: 79980120(99.1241%)\n",
      "Q30 bases: 74138420(91.8841%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 843274\n",
      "total bases: 78848808\n",
      "Q20 bases: 77912736(98.8128%)\n",
      "Q30 bases: 72303629(91.6991%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 819531\n",
      "total bases: 77028198\n",
      "Q20 bases: 76544780(99.3724%)\n",
      "Q30 bases: 71847320(93.2741%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 819531\n",
      "total bases: 74314773\n",
      "Q20 bases: 73905877(99.4498%)\n",
      "Q30 bases: 69630249(93.6964%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1639062\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 47486\n",
      "reads with adapter trimmed: 1874\n",
      "bases trimmed due to adapters: 72905\n",
      "\n",
      "Duplication rate: 1.52948%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 150\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753373_chr17_1.fastq.gz -I data/raw/ERR753373_chr17_2.fastq.gz -o data/processed/ERR753373_chr17_1_fastp.fastq.gz -O data/processed/ERR753373_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Tue Aug 29 17:54:05 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:54:05 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753373_chr17_1_fastp.fastq.gz, data/processed/ERR753373_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753373_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753373_chr17_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753373_chr17_2_fastp.fastq.gz, data/processed/ERR753373_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 17:54:15 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:54:15 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753373_chr17_1.fastq.gz, data/raw/ERR753373_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753373_chr17_1_fastqc.html, results/fastqc_result/ERR753373_chr17_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753373_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753373_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 17:54:24 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T175400.336983.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:54:25 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753374_chr17_1.fastq.gz, data/raw/ERR753374_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753374_chr17_1_fastp.fastq.gz, data/processed/ERR753374_chr17_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753374_chr17_1_fastp.fastq.gz, data/processed/ERR753374_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 492332\n",
      "total bases: 47062824\n",
      "Q20 bases: 46581392(98.977%)\n",
      "Q30 bases: 42272899(89.8223%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 492332\n",
      "total bases: 45990975\n",
      "Q20 bases: 45452935(98.8301%)\n",
      "Q30 bases: 41465767(90.1607%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 473782\n",
      "total bases: 44142721\n",
      "Q20 bases: 43810033(99.2463%)\n",
      "Q30 bases: 40441158(91.6146%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 473782\n",
      "total bases: 42957145\n",
      "Q20 bases: 42673614(99.34%)\n",
      "Q30 bases: 39641425(92.2813%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 947564\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 37100\n",
      "reads with adapter trimmed: 1076\n",
      "bases trimmed due to adapters: 40498\n",
      "\n",
      "Duplication rate: 4.87295%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 151\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753374_chr17_1.fastq.gz -I data/raw/ERR753374_chr17_2.fastq.gz -o data/processed/ERR753374_chr17_1_fastp.fastq.gz -O data/processed/ERR753374_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 2 seconds\n",
      "[Tue Aug 29 17:54:28 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:54:28 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753374_chr17_1.fastq.gz, data/raw/ERR753374_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753374_chr17_1_fastqc.html, results/fastqc_result/ERR753374_chr17_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753374_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753374_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 17:54:35 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:54:35 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753374_chr17_1_fastp.fastq.gz, data/processed/ERR753374_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753374_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753374_chr17_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753374_chr17_1_fastp.fastq.gz, data/processed/ERR753374_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 17:54:41 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T175424.452416.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:54:43 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753375_chr17_1.fastq.gz, data/raw/ERR753375_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753375_chr17_1_fastqc.html, results/fastqc_result/ERR753375_chr17_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753375_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753375_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 17:54:53 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:54:53 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753375_chr17_1.fastq.gz, data/raw/ERR753375_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753375_chr17_1_fastp.fastq.gz, data/processed/ERR753375_chr17_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753375_chr17_1_fastp.fastq.gz, data/processed/ERR753375_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1026269\n",
      "total bases: 77458309\n",
      "Q20 bases: 76947062(99.34%)\n",
      "Q30 bases: 75053112(96.8948%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1026269\n",
      "total bases: 77416125\n",
      "Q20 bases: 76301088(98.5597%)\n",
      "Q30 bases: 74090313(95.704%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 615090\n",
      "total bases: 46703360\n",
      "Q20 bases: 46543091(99.6568%)\n",
      "Q30 bases: 45786092(98.036%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 615090\n",
      "total bases: 46709566\n",
      "Q20 bases: 46221428(98.955%)\n",
      "Q30 bases: 45368202(97.1283%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1230180\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 822358\n",
      "reads with adapter trimmed: 702\n",
      "bases trimmed due to adapters: 19004\n",
      "\n",
      "Duplication rate: 1.5571%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753375_chr17_1.fastq.gz -I data/raw/ERR753375_chr17_2.fastq.gz -o data/processed/ERR753375_chr17_1_fastp.fastq.gz -O data/processed/ERR753375_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 3 seconds\n",
      "[Tue Aug 29 17:54:57 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:54:57 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753375_chr17_1_fastp.fastq.gz, data/processed/ERR753375_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753375_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753375_chr17_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753375_chr17_1_fastp.fastq.gz, data/processed/ERR753375_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 17:55:04 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T175442.044434.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:55:05 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753376_chr17_1.fastq.gz, data/raw/ERR753376_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753376_chr17_1_fastp.fastq.gz, data/processed/ERR753376_chr17_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753376_chr17_1_fastp.fastq.gz, data/processed/ERR753376_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1319555\n",
      "total bases: 99673497\n",
      "Q20 bases: 98950424(99.2746%)\n",
      "Q30 bases: 96007070(96.3216%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1319555\n",
      "total bases: 99357655\n",
      "Q20 bases: 97988993(98.6225%)\n",
      "Q30 bases: 93463715(94.068%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 743097\n",
      "total bases: 56436043\n",
      "Q20 bases: 56214517(99.6075%)\n",
      "Q30 bases: 55168884(97.7547%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 743097\n",
      "total bases: 56414017\n",
      "Q20 bases: 55982344(99.2348%)\n",
      "Q30 bases: 54346819(96.3357%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1486194\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1152916\n",
      "reads with adapter trimmed: 998\n",
      "bases trimmed due to adapters: 26275\n",
      "\n",
      "Duplication rate: 8.00836%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 117\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753376_chr17_1.fastq.gz -I data/raw/ERR753376_chr17_2.fastq.gz -o data/processed/ERR753376_chr17_1_fastp.fastq.gz -O data/processed/ERR753376_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Tue Aug 29 17:55:10 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:55:10 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753376_chr17_1.fastq.gz, data/raw/ERR753376_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753376_chr17_1_fastqc.html, results/fastqc_result/ERR753376_chr17_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753376_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753376_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 17:55:21 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:55:21 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753376_chr17_1_fastp.fastq.gz, data/processed/ERR753376_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753376_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753376_chr17_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753376_chr17_1_fastp.fastq.gz, data/processed/ERR753376_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 17:55:29 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T175504.716628.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:55:31 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753377_chr17_1.fastq.gz, data/raw/ERR753377_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753377_chr17_1_fastp.fastq.gz, data/processed/ERR753377_chr17_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753377_chr17_2_fastp.fastq.gz, data/processed/ERR753377_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1385666\n",
      "total bases: 104679071\n",
      "Q20 bases: 103910027(99.2653%)\n",
      "Q30 bases: 100812888(96.3066%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1385666\n",
      "total bases: 104385271\n",
      "Q20 bases: 103014264(98.6866%)\n",
      "Q30 bases: 98468698(94.332%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 800065\n",
      "total bases: 60762969\n",
      "Q20 bases: 60520165(99.6004%)\n",
      "Q30 bases: 59388451(97.7379%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 800065\n",
      "total bases: 60740129\n",
      "Q20 bases: 60296945(99.2704%)\n",
      "Q30 bases: 58618077(96.5063%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1600130\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1171202\n",
      "reads with adapter trimmed: 1020\n",
      "bases trimmed due to adapters: 26943\n",
      "\n",
      "Duplication rate: 8.08144%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753377_chr17_1.fastq.gz -I data/raw/ERR753377_chr17_2.fastq.gz -o data/processed/ERR753377_chr17_1_fastp.fastq.gz -O data/processed/ERR753377_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Tue Aug 29 17:55:35 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:55:35 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753377_chr17_1.fastq.gz, data/raw/ERR753377_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753377_chr17_1_fastqc.html, results/fastqc_result/ERR753377_chr17_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753377_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753377_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 17:55:48 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:55:48 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753377_chr17_1_fastp.fastq.gz, data/processed/ERR753377_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753377_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753377_chr17_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753377_chr17_2_fastp.fastq.gz, data/processed/ERR753377_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 17:55:57 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T175530.166864.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:55:58 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753378_chr17_1.fastq.gz, data/raw/ERR753378_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753378_chr17_1_fastp.fastq.gz, data/processed/ERR753378_chr17_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753378_chr17_1_fastp.fastq.gz, data/processed/ERR753378_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1654676\n",
      "total bases: 125620372\n",
      "Q20 bases: 125114317(99.5972%)\n",
      "Q30 bases: 121181168(96.4662%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1654676\n",
      "total bases: 125447036\n",
      "Q20 bases: 124251291(99.0468%)\n",
      "Q30 bases: 118001271(94.0646%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1397009\n",
      "total bases: 106166313\n",
      "Q20 bases: 105973483(99.8184%)\n",
      "Q30 bases: 104029622(97.9874%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1397009\n",
      "total bases: 106155410\n",
      "Q20 bases: 105589103(99.4665%)\n",
      "Q30 bases: 102096673(96.1766%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2794018\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 515334\n",
      "reads with adapter trimmed: 1108\n",
      "bases trimmed due to adapters: 27200\n",
      "\n",
      "Duplication rate: 1.89078%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753378_chr17_1.fastq.gz -I data/raw/ERR753378_chr17_2.fastq.gz -o data/processed/ERR753378_chr17_1_fastp.fastq.gz -O data/processed/ERR753378_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Tue Aug 29 17:56:04 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:56:04 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753378_chr17_1_fastp.fastq.gz, data/processed/ERR753378_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753378_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753378_chr17_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753378_chr17_1_fastp.fastq.gz, data/processed/ERR753378_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 100% complete for ERR753378_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 17:56:16 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:56:16 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753378_chr17_1.fastq.gz, data/raw/ERR753378_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753378_chr17_1_fastqc.html, results/fastqc_result/ERR753378_chr17_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753378_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753378_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 17:56:29 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T175557.649522.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:56:30 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR696683_chr17_1_fastp.fastq.gz, data/processed/ERR696683_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR696683_chr17.sam, results/mapped_reads/ERR696683_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR696683_chr17_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR696683_chr17_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.31 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 26.92 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.26 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.18 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 15.91 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 44.019 sec; CPU: 43.586 sec\n",
      "[Tue Aug 29 17:58:32 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:58:32 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR696683_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr17.bam, results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR696683_chr17.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR696683_chr17_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 17:59:05 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:59:05 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR696683_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "17:59:06.679 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 17:59:06 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR696683_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR696683_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 17:59:06 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 17:59:06\tMarkDuplicates\tStart of doWork freeMemory: 528793448; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 17:59:06\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 17:59:06\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam\n",
      "INFO\t2023-08-29 17:59:08\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:40,133,320\n",
      "INFO\t2023-08-29 17:59:08\tMarkDuplicates\tTracking 319 as yet unmatched pairs. 319 records in RAM.\n",
      "INFO\t2023-08-29 17:59:09\tMarkDuplicates\tRead 1926106 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 17:59:09\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 264037064; totalMemory: 538968064; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 17:59:09\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 17:59:09\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 17:59:10\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 17:59:10\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 17:59:10\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 847207352; totalMemory: 1393557504; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 17:59:10\tMarkDuplicates\tMarking 153612 records as duplicates.\n",
      "INFO\t2023-08-29 17:59:10\tMarkDuplicates\tFound 62 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam\n",
      "INFO\t2023-08-29 17:59:10\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 17:59:22\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 17:59:22\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 17:59:22\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 17:59:22\tMarkDuplicates\tBefore output close freeMemory: 528090936; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 17:59:22\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 17:59:22\tMarkDuplicates\tAfter output close freeMemory: 528090936; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 17:59:22 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.27 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 17:59:24 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T175629.254883.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 17:59:25 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753368_chr17_1_fastp.fastq.gz, data/processed/ERR753368_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753368_chr17.sam, results/mapped_reads/ERR753368_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753368_chr17_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR753368_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.28 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 25.61 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.26 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.18 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 15.22 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 41.991 sec; CPU: 41.554 sec\n",
      "[Tue Aug 29 18:01:55 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:01:55 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753368_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr17.bam, results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753368_chr17.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753368_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 18:02:41 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:02:41 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753368_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753368_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:02:41.994 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 18:02:42 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753368_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753368_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 18:02:42 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 18:02:42\tMarkDuplicates\tStart of doWork freeMemory: 528791544; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:02:42\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 18:02:42\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam\n",
      "INFO\t2023-08-29 18:02:43\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:28,617,147\n",
      "INFO\t2023-08-29 18:02:43\tMarkDuplicates\tTracking 426 as yet unmatched pairs. 426 records in RAM.\n",
      "INFO\t2023-08-29 18:02:44\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 17:50,643,453\n",
      "INFO\t2023-08-29 18:02:44\tMarkDuplicates\tTracking 335 as yet unmatched pairs. 335 records in RAM.\n",
      "INFO\t2023-08-29 18:02:46\tMarkDuplicates\tRead 2989104 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 18:02:46\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 842515416; totalMemory: 1227882496; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:02:46\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 18:02:46\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 18:02:47\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 18:02:47\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 18:02:47\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1218401360; totalMemory: 1764753408; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:02:47\tMarkDuplicates\tMarking 67231 records as duplicates.\n",
      "INFO\t2023-08-29 18:02:47\tMarkDuplicates\tFound 468 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam\n",
      "INFO\t2023-08-29 18:02:47\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 18:03:04\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 18:03:04\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 18:03:04\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 18:03:04\tMarkDuplicates\tBefore output close freeMemory: 528089688; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:03:04\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 18:03:04\tMarkDuplicates\tAfter output close freeMemory: 528089688; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 18:03:04 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.37 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 18:03:06 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T175924.175416.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:03:07 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753369_chr17_1_fastp.fastq.gz, data/processed/ERR753369_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753369_chr17.sam, results/mapped_reads/ERR753369_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753369_chr17_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753369_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.28 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 25.43 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.26 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.18 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 15.11 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 44.145 sec; CPU: 41.260 sec\n",
      "[Tue Aug 29 18:05:51 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:05:51 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753369_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr17.bam, results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753369_chr17.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753369_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 18:06:37 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:06:37 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753369_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753369_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:06:37.978 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 18:06:37 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753369_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753369_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 18:06:37 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 18:06:37\tMarkDuplicates\tStart of doWork freeMemory: 528792656; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:06:37\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 18:06:37\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam\n",
      "INFO\t2023-08-29 18:06:39\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:28,638,437\n",
      "INFO\t2023-08-29 18:06:39\tMarkDuplicates\tTracking 413 as yet unmatched pairs. 413 records in RAM.\n",
      "INFO\t2023-08-29 18:06:40\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 17:49,206,538\n",
      "INFO\t2023-08-29 18:06:40\tMarkDuplicates\tTracking 294 as yet unmatched pairs. 294 records in RAM.\n",
      "INFO\t2023-08-29 18:06:42\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 17:80,386,226\n",
      "INFO\t2023-08-29 18:06:42\tMarkDuplicates\tTracking 114 as yet unmatched pairs. 114 records in RAM.\n",
      "INFO\t2023-08-29 18:06:42\tMarkDuplicates\tRead 3123173 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 18:06:42\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 282528152; totalMemory: 688914432; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:06:42\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 18:06:42\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 18:06:43\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 18:06:43\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 18:06:43\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 679435856; totalMemory: 1225785344; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:06:43\tMarkDuplicates\tMarking 82961 records as duplicates.\n",
      "INFO\t2023-08-29 18:06:43\tMarkDuplicates\tFound 436 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam\n",
      "INFO\t2023-08-29 18:06:43\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 18:07:00\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 18:07:00\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 18:07:00\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 18:07:00\tMarkDuplicates\tBefore output close freeMemory: 528091280; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:07:00\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 18:07:00\tMarkDuplicates\tAfter output close freeMemory: 528091280; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 18:07:00 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.38 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 18:07:02 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T180306.315910.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:07:03 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753370_chr17_1_fastp.fastq.gz, data/processed/ERR753370_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753370_chr17.sam, results/mapped_reads/ERR753370_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753370_chr17_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753370_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.28 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 25.13 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.26 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.18 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 15.16 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 43.712 sec; CPU: 41.011 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Tue Aug 29 18:11:44 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:11:44 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753370_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr17.bam, results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753370_chr17.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753370_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Tue Aug 29 18:13:22 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:13:22 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753370_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753370_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:13:23.861 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 18:13:23 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753370_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753370_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 18:13:23 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 18:13:23\tMarkDuplicates\tStart of doWork freeMemory: 528838480; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:13:23\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 18:13:23\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam\n",
      "INFO\t2023-08-29 18:13:25\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:17,848,792\n",
      "INFO\t2023-08-29 18:13:25\tMarkDuplicates\tTracking 636 as yet unmatched pairs. 636 records in RAM.\n",
      "INFO\t2023-08-29 18:13:27\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 17:36,310,568\n",
      "INFO\t2023-08-29 18:13:27\tMarkDuplicates\tTracking 1066 as yet unmatched pairs. 1066 records in RAM.\n",
      "INFO\t2023-08-29 18:13:28\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 17:47,368,563\n",
      "INFO\t2023-08-29 18:13:28\tMarkDuplicates\tTracking 955 as yet unmatched pairs. 955 records in RAM.\n",
      "INFO\t2023-08-29 18:13:30\tMarkDuplicates\tRead     4,000,000 records.  Elapsed time: 00:00:06s.  Time for last 1,000,000:    1s.  Last read position: 17:68,941,906\n",
      "INFO\t2023-08-29 18:13:30\tMarkDuplicates\tTracking 481 as yet unmatched pairs. 481 records in RAM.\n",
      "INFO\t2023-08-29 18:13:30\tMarkDuplicates\tRead 4523077 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 18:13:30\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 380452448; totalMemory: 926941184; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:13:30\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 18:13:30\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 18:13:31\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 18:13:31\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 18:13:31\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 917507200; totalMemory: 1463812096; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:13:31\tMarkDuplicates\tMarking 290944 records as duplicates.\n",
      "INFO\t2023-08-29 18:13:31\tMarkDuplicates\tFound 164 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam\n",
      "INFO\t2023-08-29 18:13:31\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 18:14:03\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 18:14:03\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 18:14:03\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 18:14:03\tMarkDuplicates\tBefore output close freeMemory: 528136472; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:14:03\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 18:14:03\tMarkDuplicates\tAfter output close freeMemory: 528136472; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 18:14:03 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.66 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 18:14:07 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T180702.836718.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:14:08 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753371_chr17_1_fastp.fastq.gz, data/processed/ERR753371_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753371_chr17.sam, results/mapped_reads/ERR753371_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753371_chr17_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753371_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.28 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 25.59 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.26 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.18 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 15.02 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 43.851 sec; CPU: 41.326 sec\n",
      "[Tue Aug 29 18:15:52 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:15:52 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753371_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr17.bam, results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753371_chr17.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753371_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 18:16:16 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:16:16 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753371_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753371_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:16:17.599 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 18:16:17 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753371_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753371_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 18:16:17 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 18:16:17\tMarkDuplicates\tStart of doWork freeMemory: 528787448; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:16:17\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 18:16:17\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam\n",
      "INFO\t2023-08-29 18:16:19\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:51,502,802\n",
      "INFO\t2023-08-29 18:16:19\tMarkDuplicates\tTracking 114 as yet unmatched pairs. 114 records in RAM.\n",
      "INFO\t2023-08-29 18:16:19\tMarkDuplicates\tRead 1400906 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 18:16:19\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 381076928; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:16:19\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 18:16:19\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 18:16:20\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 18:16:20\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 18:16:20\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 755973264; totalMemory: 1302331392; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:16:20\tMarkDuplicates\tMarking 496417 records as duplicates.\n",
      "INFO\t2023-08-29 18:16:20\tMarkDuplicates\tFound 34 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam\n",
      "INFO\t2023-08-29 18:16:20\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 18:16:28\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 18:16:28\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 18:16:28\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 18:16:28\tMarkDuplicates\tBefore output close freeMemory: 528083432; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:16:28\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 18:16:28\tMarkDuplicates\tAfter output close freeMemory: 528083432; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 18:16:28 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.18 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 18:16:29 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T181407.284443.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:16:30 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753372_chr17_1_fastp.fastq.gz, data/processed/ERR753372_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753372_chr17.sam, results/mapped_reads/ERR753372_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753372_chr17_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753372_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.28 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 24.93 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.26 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.18 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 14.46 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 43.161 sec; CPU: 40.105 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Tue Aug 29 18:19:25 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:19:25 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753372_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr17.bam, results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753372_chr17.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753372_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Tue Aug 29 18:20:22 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:20:22 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753372_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753372_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:20:22.957 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 18:20:22 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753372_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753372_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 18:20:22 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 18:20:22\tMarkDuplicates\tStart of doWork freeMemory: 528774352; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:20:22\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 18:20:22\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam\n",
      "INFO\t2023-08-29 18:20:24\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:26,199,503\n",
      "INFO\t2023-08-29 18:20:24\tMarkDuplicates\tTracking 487 as yet unmatched pairs. 487 records in RAM.\n",
      "INFO\t2023-08-29 18:20:25\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 17:47,247,676\n",
      "INFO\t2023-08-29 18:20:25\tMarkDuplicates\tTracking 328 as yet unmatched pairs. 328 records in RAM.\n",
      "INFO\t2023-08-29 18:20:27\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 17:78,126,201\n",
      "INFO\t2023-08-29 18:20:27\tMarkDuplicates\tTracking 98 as yet unmatched pairs. 98 records in RAM.\n",
      "INFO\t2023-08-29 18:20:27\tMarkDuplicates\tRead 3244765 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 18:20:27\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 517769968; totalMemory: 933232640; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:20:27\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 18:20:27\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 18:20:28\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 18:20:28\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 18:20:28\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 923734616; totalMemory: 1470103552; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:20:28\tMarkDuplicates\tMarking 72263 records as duplicates.\n",
      "INFO\t2023-08-29 18:20:28\tMarkDuplicates\tFound 452 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam\n",
      "INFO\t2023-08-29 18:20:28\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 18:20:46\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 18:20:46\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 18:20:46\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 18:20:46\tMarkDuplicates\tBefore output close freeMemory: 528072464; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:20:46\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 18:20:46\tMarkDuplicates\tAfter output close freeMemory: 528072464; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 18:20:46 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.39 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 18:20:49 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T181629.693838.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:20:50 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753373_chr17_1_fastp.fastq.gz, data/processed/ERR753373_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753373_chr17.sam, results/mapped_reads/ERR753373_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753373_chr17_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753373_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.28 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 24.46 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.24 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.17 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 14.88 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 42.880 sec; CPU: 40.040 sec\n",
      "[Tue Aug 29 18:23:35 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:23:35 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753373_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr17.bam, results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753373_chr17.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753373_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 18:24:22 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:24:22 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753373_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753373_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:24:23.185 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 18:24:23 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753373_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753373_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 18:24:23 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 18:24:23\tMarkDuplicates\tStart of doWork freeMemory: 528781704; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:24:23\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 18:24:23\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam\n",
      "INFO\t2023-08-29 18:24:24\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:34,038,098\n",
      "INFO\t2023-08-29 18:24:24\tMarkDuplicates\tTracking 174 as yet unmatched pairs. 174 records in RAM.\n",
      "INFO\t2023-08-29 18:24:26\tMarkDuplicates\tRead 1994716 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 18:24:26\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 287951472; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:24:26\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 18:24:26\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 18:24:26\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 18:24:26\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 18:24:26\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 782184520; totalMemory: 1328545792; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:24:26\tMarkDuplicates\tMarking 36869 records as duplicates.\n",
      "INFO\t2023-08-29 18:24:26\tMarkDuplicates\tFound 11388 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam\n",
      "INFO\t2023-08-29 18:24:26\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 18:24:41\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 18:24:41\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 18:24:41\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 18:24:41\tMarkDuplicates\tBefore output close freeMemory: 528079776; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:24:41\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 18:24:41\tMarkDuplicates\tAfter output close freeMemory: 528079776; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 18:24:41 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.30 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 18:24:42 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T182049.203155.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:24:43 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753374_chr17_1_fastp.fastq.gz, data/processed/ERR753374_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753374_chr17.sam, results/mapped_reads/ERR753374_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753374_chr17_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753374_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.27 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 24.61 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.25 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.17 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 14.80 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 42.891 sec; CPU: 40.097 sec\n",
      "[Tue Aug 29 18:26:34 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:26:34 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753374_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr17.bam, results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753374_chr17.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753374_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 18:27:00 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:27:00 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753374_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753374_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:27:01.079 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 18:27:01 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753374_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753374_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 18:27:01 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 18:27:01\tMarkDuplicates\tStart of doWork freeMemory: 528792128; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:27:01\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 18:27:01\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam\n",
      "INFO\t2023-08-29 18:27:02\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:62,737,033\n",
      "INFO\t2023-08-29 18:27:02\tMarkDuplicates\tTracking 91 as yet unmatched pairs. 91 records in RAM.\n",
      "INFO\t2023-08-29 18:27:02\tMarkDuplicates\tRead 1192512 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 18:27:02\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 362953528; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:27:02\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 18:27:03\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 18:27:03\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 18:27:03\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 18:27:03\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 672092680; totalMemory: 1218445312; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:27:03\tMarkDuplicates\tMarking 88629 records as duplicates.\n",
      "INFO\t2023-08-29 18:27:03\tMarkDuplicates\tFound 28791 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam\n",
      "INFO\t2023-08-29 18:27:03\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 18:27:11\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 18:27:11\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 18:27:11\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 18:27:11\tMarkDuplicates\tBefore output close freeMemory: 528089072; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:27:11\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 18:27:11\tMarkDuplicates\tAfter output close freeMemory: 528089072; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 18:27:11 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.18 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 18:27:12 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T182442.738068.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:27:13 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753375_chr17_1_fastp.fastq.gz, data/processed/ERR753375_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753375_chr17.sam, results/mapped_reads/ERR753375_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753375_chr17_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753375_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.28 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 24.65 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.26 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.18 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 15.07 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 43.338 sec; CPU: 40.432 sec\n",
      "[Tue Aug 29 18:28:51 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:28:51 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753375_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr17.bam, results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753375_chr17.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753375_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 18:29:13 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:29:13 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753375_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753375_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:29:14.771 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 18:29:14 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753375_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753375_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 18:29:14 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 18:29:14\tMarkDuplicates\tStart of doWork freeMemory: 528835168; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:29:14\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 18:29:14\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam\n",
      "INFO\t2023-08-29 18:29:16\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:61,925,001\n",
      "INFO\t2023-08-29 18:29:16\tMarkDuplicates\tTracking 119 as yet unmatched pairs. 119 records in RAM.\n",
      "INFO\t2023-08-29 18:29:16\tMarkDuplicates\tRead 1249072 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 18:29:16\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 397329648; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:29:16\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 18:29:16\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 18:29:17\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 18:29:17\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 18:29:17\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 731907080; totalMemory: 1278214144; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:29:17\tMarkDuplicates\tMarking 53214 records as duplicates.\n",
      "INFO\t2023-08-29 18:29:17\tMarkDuplicates\tFound 132 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam\n",
      "INFO\t2023-08-29 18:29:17\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 18:29:24\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 18:29:24\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 18:29:24\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 18:29:24\tMarkDuplicates\tBefore output close freeMemory: 528134640; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:29:24\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 18:29:24\tMarkDuplicates\tAfter output close freeMemory: 528134640; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 18:29:24 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.17 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 18:29:25 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T182712.825556.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:29:26 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753376_chr17_1_fastp.fastq.gz, data/processed/ERR753376_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753376_chr17.sam, results/mapped_reads/ERR753376_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753376_chr17_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753376_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.28 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 24.63 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.26 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.18 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 15.06 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 43.156 sec; CPU: 40.412 sec\n",
      "[Tue Aug 29 18:31:13 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:31:13 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753376_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr17.bam, results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753376_chr17.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753376_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 18:31:42 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:31:42 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753376_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753376_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:31:42.955 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 18:31:42 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753376_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753376_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 18:31:42 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 18:31:42\tMarkDuplicates\tStart of doWork freeMemory: 528840824; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:31:42\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 18:31:42\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam\n",
      "INFO\t2023-08-29 18:31:44\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:46,640,106\n",
      "INFO\t2023-08-29 18:31:44\tMarkDuplicates\tTracking 168 as yet unmatched pairs. 168 records in RAM.\n",
      "INFO\t2023-08-29 18:31:45\tMarkDuplicates\tRead 1528048 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 18:31:45\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 304731616; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:31:45\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 18:31:45\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 18:31:45\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 18:31:45\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 18:31:45\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 774903624; totalMemory: 1321205760; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:31:45\tMarkDuplicates\tMarking 187062 records as duplicates.\n",
      "INFO\t2023-08-29 18:31:45\tMarkDuplicates\tFound 58 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam\n",
      "INFO\t2023-08-29 18:31:45\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 18:31:55\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 18:31:55\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 18:31:55\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 18:31:55\tMarkDuplicates\tBefore output close freeMemory: 528139400; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:31:55\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 18:31:55\tMarkDuplicates\tAfter output close freeMemory: 528139400; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 18:31:55 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.20 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 18:31:56 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T182925.908368.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:31:57 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753377_chr17_1_fastp.fastq.gz, data/processed/ERR753377_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753377_chr17.sam, results/mapped_reads/ERR753377_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753377_chr17_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753377_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.28 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 27.58 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.25 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.18 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 15.40 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 46.570 sec; CPU: 43.699 sec\n",
      "[Tue Aug 29 18:33:53 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:33:53 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753377_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr17.bam, results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753377_chr17.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753377_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 18:34:22 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:34:22 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753377_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753377_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:34:23.512 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 18:34:23 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753377_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753377_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 18:34:23 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 18:34:23\tMarkDuplicates\tStart of doWork freeMemory: 528789480; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:34:23\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 18:34:23\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam\n",
      "INFO\t2023-08-29 18:34:25\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:45,172,105\n",
      "INFO\t2023-08-29 18:34:25\tMarkDuplicates\tTracking 190 as yet unmatched pairs. 190 records in RAM.\n",
      "INFO\t2023-08-29 18:34:25\tMarkDuplicates\tRead 1629046 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 18:34:25\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 292388368; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:34:25\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 18:34:26\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 18:34:26\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 18:34:26\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 18:34:26\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 774851576; totalMemory: 1321205760; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:34:26\tMarkDuplicates\tMarking 200629 records as duplicates.\n",
      "INFO\t2023-08-29 18:34:26\tMarkDuplicates\tFound 52 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam\n",
      "INFO\t2023-08-29 18:34:26\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 18:34:36\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 18:34:36\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 18:34:36\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 18:34:36\tMarkDuplicates\tBefore output close freeMemory: 528087000; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:34:36\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 18:34:36\tMarkDuplicates\tAfter output close freeMemory: 528087000; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 18:34:36 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.21 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 18:34:37 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T183156.229557.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:34:38 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753378_chr17_1_fastp.fastq.gz, data/processed/ERR753378_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753378_chr17.sam, results/mapped_reads/ERR753378_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753378_chr17_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753378_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.28 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 25.04 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.26 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.17 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 14.90 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 43.358 sec; CPU: 40.659 sec\n",
      "[Tue Aug 29 18:37:10 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:37:10 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753378_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr17.bam, results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753378_chr17.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753378_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 18:37:54 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:37:54 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753378_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753378_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:37:55.088 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Aug 29 18:37:55 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753378_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753378_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Aug 29 18:37:55 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-29 18:37:55\tMarkDuplicates\tStart of doWork freeMemory: 528792392; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:37:55\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-29 18:37:55\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam\n",
      "INFO\t2023-08-29 18:37:56\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:30,209,102\n",
      "INFO\t2023-08-29 18:37:56\tMarkDuplicates\tTracking 409 as yet unmatched pairs. 409 records in RAM.\n",
      "INFO\t2023-08-29 18:37:58\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 17:58,462,914\n",
      "INFO\t2023-08-29 18:37:58\tMarkDuplicates\tTracking 397 as yet unmatched pairs. 397 records in RAM.\n",
      "INFO\t2023-08-29 18:37:59\tMarkDuplicates\tRead 2826651 records. 0 pairs never matched.\n",
      "INFO\t2023-08-29 18:37:59\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 457549480; totalMemory: 831520768; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:37:59\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-29 18:37:59\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-29 18:37:59\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-29 18:38:00\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-29 18:38:00\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 989812928; totalMemory: 1536163840; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:38:00\tMarkDuplicates\tMarking 63252 records as duplicates.\n",
      "INFO\t2023-08-29 18:38:00\tMarkDuplicates\tFound 446 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam\n",
      "INFO\t2023-08-29 18:38:00\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-29 18:38:16\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-29 18:38:16\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-29 18:38:16\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-29 18:38:16\tMarkDuplicates\tBefore output close freeMemory: 528090832; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-29 18:38:16\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-29 18:38:16\tMarkDuplicates\tAfter output close freeMemory: 528090832; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Tue Aug 29 18:38:16 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.36 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Tue Aug 29 18:38:18 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T183437.551925.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:38:20 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR696683_chr17_dedup.bam\n",
      "    output: results/variants/ERR696683_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR696683_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR696683_chr17.vcf\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "index file data/reference/genome.fa.fai not found, generating...\n",
      "[Tue Aug 29 18:38:54 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T183819.004451.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:38:56 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753368_chr17_dedup.bam\n",
      "    output: results/variants/ERR753368_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753368_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753368_chr17.vcf\n",
      "    wildcards: sample=ERR753368_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 18:39:47 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T183855.097336.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:39:48 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753369_chr17_dedup.bam\n",
      "    output: results/variants/ERR753369_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753369_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753369_chr17.vcf\n",
      "    wildcards: sample=ERR753369_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 18:40:41 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T183947.619341.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:40:42 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753370_chr17_dedup.bam\n",
      "    output: results/variants/ERR753370_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753370_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753370_chr17.vcf\n",
      "    wildcards: sample=ERR753370_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 18:42:06 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T184041.740210.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:42:08 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753371_chr17_dedup.bam\n",
      "    output: results/variants/ERR753371_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753371_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753371_chr17.vcf\n",
      "    wildcards: sample=ERR753371_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 18:42:29 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T184207.062079.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:42:31 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753372_chr17_dedup.bam\n",
      "    output: results/variants/ERR753372_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753372_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753372_chr17.vcf\n",
      "    wildcards: sample=ERR753372_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 18:43:26 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T184230.031123.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:43:27 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753373_chr17_dedup.bam\n",
      "    output: results/variants/ERR753373_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753373_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753373_chr17.vcf\n",
      "    wildcards: sample=ERR753373_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 18:44:08 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T184326.791624.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:44:09 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753374_chr17_dedup.bam\n",
      "    output: results/variants/ERR753374_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753374_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753374_chr17.vcf\n",
      "    wildcards: sample=ERR753374_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 18:44:35 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T184408.260883.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:44:36 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753375_chr17_dedup.bam\n",
      "    output: results/variants/ERR753375_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753375_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753375_chr17.vcf\n",
      "    wildcards: sample=ERR753375_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 18:45:03 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T184435.497079.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:45:04 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753376_chr17_dedup.bam\n",
      "    output: results/variants/ERR753376_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753376_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753376_chr17.vcf\n",
      "    wildcards: sample=ERR753376_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 18:45:32 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T184503.579474.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:45:33 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753377_chr17_dedup.bam\n",
      "    output: results/variants/ERR753377_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753377_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753377_chr17.vcf\n",
      "    wildcards: sample=ERR753377_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 18:46:02 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T184532.750509.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:46:03 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753378_chr17_dedup.bam\n",
      "    output: results/variants/ERR753378_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753378_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753378_chr17.vcf\n",
      "    wildcards: sample=ERR753378_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Tue Aug 29 18:46:53 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T184602.758518.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:46:54 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR696683_chr17.vcf\n",
      "    output: results/variants/vep/ERR696683_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR696683_chr17.txt\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 18:47:20 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T184653.241527.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:47:22 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753368_chr17.vcf\n",
      "    output: results/variants/vep/ERR753368_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753368_chr17.txt\n",
      "    wildcards: sample=ERR753368_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 18:47:55 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T184720.802718.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:47:56 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753369_chr17.vcf\n",
      "    output: results/variants/vep/ERR753369_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753369_chr17.txt\n",
      "    wildcards: sample=ERR753369_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 18:48:28 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T184755.524749.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:48:29 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753370_chr17.vcf\n",
      "    output: results/variants/vep/ERR753370_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753370_chr17.txt\n",
      "    wildcards: sample=ERR753370_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 18:49:04 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T184828.893626.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:49:05 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753371_chr17.vcf\n",
      "    output: results/variants/vep/ERR753371_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753371_chr17.txt\n",
      "    wildcards: sample=ERR753371_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 18:49:25 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T184904.613199.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:49:27 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753372_chr17.vcf\n",
      "    output: results/variants/vep/ERR753372_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753372_chr17.txt\n",
      "    wildcards: sample=ERR753372_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 18:50:01 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T184926.032182.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:50:02 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753373_chr17.vcf\n",
      "    output: results/variants/vep/ERR753373_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753373_chr17.txt\n",
      "    wildcards: sample=ERR753373_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 18:50:27 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T185001.380608.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:50:29 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753374_chr17.vcf\n",
      "    output: results/variants/vep/ERR753374_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753374_chr17.txt\n",
      "    wildcards: sample=ERR753374_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 18:50:49 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T185028.151498.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:50:50 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753375_chr17.vcf\n",
      "    output: results/variants/vep/ERR753375_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753375_chr17.txt\n",
      "    wildcards: sample=ERR753375_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 18:51:13 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T185049.429715.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:51:14 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753376_chr17.vcf\n",
      "    output: results/variants/vep/ERR753376_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753376_chr17.txt\n",
      "    wildcards: sample=ERR753376_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 18:51:37 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T185113.270442.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:51:38 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753377_chr17.vcf\n",
      "    output: results/variants/vep/ERR753377_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753377_chr17.txt\n",
      "    wildcards: sample=ERR753377_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 18:52:02 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T185137.480957.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:52:03 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753378_chr17.vcf\n",
      "    output: results/variants/vep/ERR753378_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753378_chr17.txt\n",
      "    wildcards: sample=ERR753378_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Tue Aug 29 18:52:35 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T185202.244145.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job              count    min threads    max threads\n",
      "-------------  -------  -------------  -------------\n",
      "parsing_dataR        1              1              1\n",
      "total                1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:52:36 2023]\n",
      "rule parsing_dataR:\n",
      "    input: code/05parsing_vep_data.R\n",
      "    output: tasks/13parsing_dataR.done\n",
      "    jobid: 0\n",
      "    reason: Params have changed since last execution\n",
      "    resources: tmpdir=/tmp\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">>> This is your sample list: ERR696683_chr17\n",
      ">>> This is your sample list: ERR753368_chr17\n",
      ">>> This is your sample list: ERR753369_chr17\n",
      ">>> This is your sample list: ERR753370_chr17\n",
      ">>> This is your sample list: ERR753371_chr17\n",
      ">>> This is your sample list: ERR753372_chr17\n",
      ">>> This is your sample list: ERR753373_chr17\n",
      ">>> This is your sample list: ERR753374_chr17\n",
      ">>> This is your sample list: ERR753375_chr17\n",
      ">>> This is your sample list: ERR753376_chr17\n",
      ">>> This is your sample list: ERR753377_chr17\n",
      ">>> This is your sample list: ERR753378_chr17\n",
      ">>> This is your gene: TP53\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 26 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "2 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "3 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "4 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "5 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "6 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "# ℹ 20 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 152 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7674109_G/A        17:7674109 A      ENSG… ENST00… Transcript   intron_var…\n",
      "2 17_7674109_G/A        17:7674109 A      ENSG… ENST00… Transcript   intron_var…\n",
      "3 17_7674109_G/A        17:7674109 A      ENSG… ENST00… Transcript   intron_var…\n",
      "4 17_7674109_G/A        17:7674109 A      ENSG… ENST00… Transcript   intron_var…\n",
      "5 17_7674109_G/A        17:7674109 A      ENSG… ENST00… Transcript   intron_var…\n",
      "6 17_7674109_G/A        17:7674109 A      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 146 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 249 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   downstream…\n",
      "2 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   intron_var…\n",
      "3 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   intron_var…\n",
      "4 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   downstream…\n",
      "5 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   downstream…\n",
      "6 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   downstream…\n",
      "# ℹ 243 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 130 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "2 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "3 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "4 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "5 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "6 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 124 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 58 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "2 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "3 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "4 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "5 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "6 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "# ℹ 52 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 126 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "2 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "3 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "4 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "5 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "6 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 120 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 68 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "2 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "3 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 62 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 0 × 44\n",
      "# ℹ 44 variables: #uploaded_variation <chr>, location <chr>, allele <chr>,\n",
      "#   gene <chr>, feature <chr>, feature_type <chr>, consequence <chr>,\n",
      "#   cdna_position <chr>, cds_position <chr>, protein_position <chr>,\n",
      "#   amino_acids <chr>, codons <chr>, existing_variation <chr>, impact <chr>,\n",
      "#   distance <chr>, strand <chr>, flags <chr>, variant_class <chr>,\n",
      "#   symbol <chr>, symbol_source <chr>, hgnc_id <chr>, biotype <chr>,\n",
      "#   mane_select <chr>, mane_plus_clinical <chr>, tsl <chr>, appris <chr>, …\n",
      "# A tibble: 26 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "2 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "3 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 20 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 52 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "2 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "3 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "4 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "5 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "6 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "# ℹ 46 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 52 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "2 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "3 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "4 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "5 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "6 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "# ℹ 46 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "Touching output file tasks/13parsing_dataR.done.\n",
      "[Tue Aug 29 18:52:39 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T185235.823072.snakemake.log\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 149 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   downstream…\n",
      "2 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   intron_var…\n",
      "3 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   intron_var…\n",
      "4 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   downstream…\n",
      "5 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   downstream…\n",
      "6 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   downstream…\n",
      "# ℹ 143 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "[1] \"##==========================##\"\n",
      "[1] \"##===> WORK FINISHED!!! <===##\"\n",
      "[1] \"##==========================##\"\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['snakemake', '--cores', '1', 'parsing_dataR'], returncode=0)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Running the workflow for the 17º chormosome\n",
    "\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--force\", \"data/reference/genome.fa\"]) ## rule reference_genome \n",
    "\n",
    "for sample in sample_list:\n",
    "    ## Quality inspection\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/fastqc_result/\" + sample + \"_chr17_1_fastqc.html\",   \n",
    "                    \"data/processed/\" + sample + \"_chr17_1_fastp.fastq.gz\",   \n",
    "                    \"results/fastqc_result/trimmed/\" + sample + \"_chr17_1_fastp_fastqc.html\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    ## Mapping reads\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/mapped_reads/\" + sample + \"_chr17_sorted.sam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr17_sorted.bam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr17_dedup.bam\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    # Extracting variants  \n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/\" + sample + \"_chr17.vcf\"])   \n",
    "\n",
    "## VEP CLI\n",
    "for sample in sample_list:\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/vep/\" + sample + \"_chr17.txt\"])\n",
    "\n",
    "## Extracting the gene PIKCA3 from the VEP files\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"parsing_dataR\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['sed', '-i', '65s/TP53/PIK3CA/g', 'config.yaml'], returncode=0)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## reset config \n",
    "subprocess.run([\"sed\", \"-i\", \"6,+12s/chr17/chr3/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"24,+12s/chr17/chr3/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\",\"39s/chr17/chr3/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"46s/chromosome.17.fa.gz/chromosome.3.fa.gz/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"65s/TP53/PIK3CA/g\", \"config.yaml\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## **Last rule**\n",
    "\n",
    "### **rule R_plotting**\n",
    "\n",
    "Plotting the data of the 5 gene tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job           count    min threads    max threads\n",
      "----------  -------  -------------  -------------\n",
      "R_plotting        1              1              1\n",
      "total             1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Tue Aug 29 18:54:28 2023]\n",
      "rule R_plotting:\n",
      "    input: code/06final_plot.R\n",
      "    output: results/biostatistics/plots/final_plot.png, results/biostatistics/plots/other_plot.png\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/biostatistics/plots/final_plot.png, results/biostatistics/plots/other_plot.png\n",
      "    resources: tmpdir=/tmp\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"##==========================##\"\n",
      "[1] \"##===> WORK FINISHED!!! <===##\"\n",
      "[1] \"##==========================##\"\n",
      "THE SCRIPT RAN WELL congrats :)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Tue Aug 29 18:54:31 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-29T185428.517384.snakemake.log\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['snakemake', '--cores', '1', 'R_plotting'], returncode=0)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"R_plotting\"])"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Greference_tools",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
