{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# **Automatation of the CLI commands of the Snakemake workflow**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* Environment to use: [Greference_tools](code/environments/Greference_tools.yml)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Libreries\n",
    "import subprocess\n",
    "import os\n",
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "## lists for the loops\n",
    "chr_list        = [\"chr3\", \"chr5\", \"chr7\", \"chr12\", \"chr17\"]\n",
    "\n",
    "url_list        = [\"chromosome.3.fa.gz\", \"chromosome.5.fa.gz\", \"chromosome.7.fa.gz\", \n",
    "                   \"chromosome.12.fa.gz\", \"chromosome.17.fa.gz\"]\n",
    "\n",
    "gene_list       = [\"PIK3CA\", \"APC\", \"BRAF\", \"KRAS\", \"TP53\"]\n",
    "\n",
    "sample_list     = [\"ERR696683\", \"ERR753368\", \"ERR753369\" ,\"ERR753370\" ,\"ERR753371\" ,\"ERR753372\", \n",
    "                  \"ERR753373\", \"ERR753374\", \"ERR753375\", \"ERR753376\", \"ERR753377\", \"ERR753378\"]\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. **rule download_data**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"download_data\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* rule pre_processing\n",
    "\n",
    "Filtering the chromosomes, make sure that the chromosomes of the config file are **ch3**. \n",
    "\n",
    "This rule is preatty long too... good luck\n",
    "\n",
    "When it finished, it is recomended to delete the SAM files not oredered."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Running the 3º chromosome\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"data/original_bam/filtering/ERR696683_chr3_sorted.bam\"])\n",
    "\n",
    "## Running the rest of the chromosomes\n",
    "for chr in range(len(chr_list) - 1):\n",
    "    chr1 = chr_list[chr]\n",
    "    chr2 = chr_list[chr + 1]\n",
    "    \n",
    "    print(\"\"\"\n",
    "          Chromosomes transformed:\n",
    "          \"\"\", chr1, chr2)\n",
    "    \n",
    "    subprocess.run([\"sed\", \"-i\", \"6,+12s/\" + chr1 + \"/\" + chr2 + \"/g\", \"config.yaml\"])\n",
    "    subprocess.run([\"sed\", \"-i\", \"24s/\" + chr1 + \"/\" + chr2 + \"/g\", \"config.yaml\"])\n",
    "\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"data/original_bam/filtering/ERR696683_\" + chr2 + \"_sorted.bam\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['sed', '-i', '24s/chr17/chr3/g', 'config.yaml'], returncode=0)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## reset config \n",
    "subprocess.run([\"sed\", \"-i\", \"6,+12s/chr17/chr3/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"24s/chr17/chr3/g\", \"config.yaml\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* rules reference_genome; fastqc; fastp; fastqc_trimmed; bwa_mapping; sam_to_bam; delete_duplicates; extracting_variants; vep_install; vep_cli; parsing_dataR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job                 count    min threads    max threads\n",
      "----------------  -------  -------------  -------------\n",
      "reference_genome        1              1              1\n",
      "total                   1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 14:55:46 2023]\n",
      "rule reference_genome:\n",
      "    output: data/reference/genome.fa\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/reference/genome.fa\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "rm: no se puede borrar 'data/reference/genome.fa*': No existe el archivo o el directorio\n",
      "--2023-08-18 14:55:46--  https://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.3.fa.gz\n",
      "Resolviendo ftp.ensembl.org (ftp.ensembl.org)... "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==> If you hadn't a previous reference in the directory and there is an ERROR, is NORMAL <==\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "193.62.193.169\n",
      "Conectando con ftp.ensembl.org (ftp.ensembl.org)[193.62.193.169]:443... conectado.\n",
      "Petición HTTP enviada, esperando respuesta... 200 OK\n",
      "Longitud: 59761837 (57M) [application/x-gzip]\n",
      "Guardando como: ‘data/reference/genome.fa.gz’\n",
      "\n",
      "     0K .......... .......... .......... .......... ..........  0%  458K 2m7s\n",
      "    50K .......... .......... .......... .......... ..........  0% 4,25M 70s\n",
      "   100K .......... .......... .......... .......... ..........  0% 1018K 66s\n",
      "   150K .......... .......... .......... .......... ..........  0% 91,1M 50s\n",
      "   200K .......... .......... .......... .......... ..........  0%  896K 53s\n",
      "   250K .......... .......... .......... .......... ..........  0% 80,4M 44s\n",
      "   300K .......... .......... .......... .......... ..........  0% 90,8M 38s\n",
      "   350K .......... .......... .......... .......... ..........  0% 79,6M 33s\n",
      "   400K .......... .......... .......... .......... ..........  0% 6,10M 30s\n",
      "   450K .......... .......... .......... .......... ..........  0% 1,04M 33s\n",
      "   500K .......... .......... .......... .......... ..........  0% 73,6M 30s\n",
      "   550K .......... .......... .......... .......... ..........  1%  113M 27s\n",
      "   600K .......... .......... .......... .......... ..........  1%  104M 25s\n",
      "   650K .......... .......... .......... .......... ..........  1%  123M 23s\n",
      "   700K .......... .......... .......... .......... ..........  1%  103M 22s\n",
      "   750K .......... .......... .......... .......... ..........  1%  121M 21s\n",
      "   800K .......... .......... .......... .......... ..........  1%  115M 19s\n",
      "   850K .......... .......... .......... .......... ..........  1% 9,36M 19s\n",
      "   900K .......... .......... .......... .......... ..........  1% 26,6M 18s\n",
      "   950K .......... .......... .......... .......... ..........  1% 1,05M 19s\n",
      "  1000K .......... .......... .......... .......... ..........  1% 33,0M 19s\n",
      "  1050K .......... .......... .......... .......... ..........  1%  115M 18s\n",
      "  1100K .......... .......... .......... .......... ..........  1%  107M 17s\n",
      "  1150K .......... .......... .......... .......... ..........  2%  115M 16s\n",
      "  1200K .......... .......... .......... .......... ..........  2% 61,7M 16s\n",
      "  1250K .......... .......... .......... .......... ..........  2% 86,4M 15s\n",
      "  1300K .......... .......... .......... .......... ..........  2%  113M 15s\n",
      "  1350K .......... .......... .......... .......... ..........  2%  119M 14s\n",
      "  1400K .......... .......... .......... .......... ..........  2%  154M 14s\n",
      "  1450K .......... .......... .......... .......... ..........  2%  111M 13s\n",
      "  1500K .......... .......... .......... .......... ..........  2%  102M 13s\n",
      "  1550K .......... .......... .......... .......... ..........  2% 83,8M 12s\n",
      "  1600K .......... .......... .......... .......... ..........  2% 87,7M 12s\n",
      "  1650K .......... .......... .......... .......... ..........  2%  144M 12s\n",
      "  1700K .......... .......... .......... .......... ..........  2%  103M 11s\n",
      "  1750K .......... .......... .......... .......... ..........  3%  119M 11s\n",
      "  1800K .......... .......... .......... .......... ..........  3% 45,5M 11s\n",
      "  1850K .......... .......... .......... .......... ..........  3%  103M 10s\n",
      "  1900K .......... .......... .......... .......... ..........  3%  123M 10s\n",
      "  1950K .......... .......... .......... .......... ..........  3% 1,04M 11s\n",
      "  2000K .......... .......... .......... .......... ..........  3%  139M 11s\n",
      "  2050K .......... .......... .......... .......... ..........  3% 42,2M 11s\n",
      "  2100K .......... .......... .......... .......... ..........  3% 41,9M 10s\n",
      "  2150K .......... .......... .......... .......... ..........  3% 51,8M 10s\n",
      "  2200K .......... .......... .......... .......... ..........  3% 96,8M 10s\n",
      "  2250K .......... .......... .......... .......... ..........  3%  115M 10s\n",
      "  2300K .......... .......... .......... .......... ..........  4%  114M 10s\n",
      "  2350K .......... .......... .......... .......... ..........  4%  101M 9s\n",
      "  2400K .......... .......... .......... .......... ..........  4%  117M 9s\n",
      "  2450K .......... .......... .......... .......... ..........  4% 98,9M 9s\n",
      "  2500K .......... .......... .......... .......... ..........  4%  111M 9s\n",
      "  2550K .......... .......... .......... .......... ..........  4%  115M 9s\n",
      "  2600K .......... .......... .......... .......... ..........  4%  100M 9s\n",
      "  2650K .......... .......... .......... .......... ..........  4%  119M 8s\n",
      "  2700K .......... .......... .......... .......... ..........  4% 99,8M 8s\n",
      "  2750K .......... .......... .......... .......... ..........  4%  117M 8s\n",
      "  2800K .......... .......... .......... .......... ..........  4%  121M 8s\n",
      "  2850K .......... .......... .......... .......... ..........  4% 96,4M 8s\n",
      "  2900K .......... .......... .......... .......... ..........  5%  116M 8s\n",
      "  2950K .......... .......... .......... .......... ..........  5% 91,3M 8s\n",
      "  3000K .......... .......... .......... .......... ..........  5%  128M 7s\n",
      "  3050K .......... .......... .......... .......... ..........  5%  129M 7s\n",
      "  3100K .......... .......... .......... .......... ..........  5%  103M 7s\n",
      "  3150K .......... .......... .......... .......... ..........  5%  120M 7s\n",
      "  3200K .......... .......... .......... .......... ..........  5%  101M 7s\n",
      "  3250K .......... .......... .......... .......... ..........  5%  117M 7s\n",
      "  3300K .......... .......... .......... .......... ..........  5%  117M 7s\n",
      "  3350K .......... .......... .......... .......... ..........  5%  120M 7s\n",
      "  3400K .......... .......... .......... .......... ..........  5%  157M 7s\n",
      "  3450K .......... .......... .......... .......... ..........  5%  128M 6s\n",
      "  3500K .......... .......... .......... .......... ..........  6%  151M 6s\n",
      "  3550K .......... .......... .......... .......... ..........  6%  165M 6s\n",
      "  3600K .......... .......... .......... .......... ..........  6%  137M 6s\n",
      "  3650K .......... .......... .......... .......... ..........  6%  171M 6s\n",
      "  3700K .......... .......... .......... .......... ..........  6%  148M 6s\n",
      "  3750K .......... .......... .......... .......... ..........  6%  212M 6s\n",
      "  3800K .......... .......... .......... .......... ..........  6%  222M 6s\n",
      "  3850K .......... .......... .......... .......... ..........  6%  178M 6s\n",
      "  3900K .......... .......... .......... .......... ..........  6% 1,27M 6s\n",
      "  3950K .......... .......... .......... .......... ..........  6% 33,5M 6s\n",
      "  4000K .......... .......... .......... .......... ..........  6% 73,7M 6s\n",
      "  4050K .......... .......... .......... .......... ..........  7% 77,2M 6s\n",
      "  4100K .......... .......... .......... .......... ..........  7%  113M 6s\n",
      "  4150K .......... .......... .......... .......... ..........  7%  128M 6s\n",
      "  4200K .......... .......... .......... .......... ..........  7% 72,1M 6s\n",
      "  4250K .......... .......... .......... .......... ..........  7%  150M 6s\n",
      "  4300K .......... .......... .......... .......... ..........  7%  168M 6s\n",
      "  4350K .......... .......... .......... .......... ..........  7%  145M 6s\n",
      "  4400K .......... .......... .......... .......... ..........  7%  132M 6s\n",
      "  4450K .......... .......... .......... .......... ..........  7% 91,8M 5s\n",
      "  4500K .......... .......... .......... .......... ..........  7%  137M 5s\n",
      "  4550K .......... .......... .......... .......... ..........  7%  185M 5s\n",
      "  4600K .......... .......... .......... .......... ..........  7%  101M 5s\n",
      "  4650K .......... .......... .......... .......... ..........  8%  123M 5s\n",
      "  4700K .......... .......... .......... .......... ..........  8%  103M 5s\n",
      "  4750K .......... .......... .......... .......... ..........  8%  116M 5s\n",
      "  4800K .......... .......... .......... .......... ..........  8%  122M 5s\n",
      "  4850K .......... .......... .......... .......... ..........  8%  102M 5s\n",
      "  4900K .......... .......... .......... .......... ..........  8%  117M 5s\n",
      "  4950K .......... .......... .......... .......... ..........  8%  102M 5s\n",
      "  5000K .......... .......... .......... .......... ..........  8%  122M 5s\n",
      "  5050K .......... .......... .......... .......... ..........  8%  121M 5s\n",
      "  5100K .......... .......... .......... .......... ..........  8%  101M 5s\n",
      "  5150K .......... .......... .......... .......... ..........  8%  117M 5s\n",
      "  5200K .......... .......... .......... .......... ..........  8%  103M 5s\n",
      "  5250K .......... .......... .......... .......... ..........  9%  120M 5s\n",
      "  5300K .......... .......... .......... .......... ..........  9%  119M 5s\n",
      "  5350K .......... .......... .......... .......... ..........  9%  102M 5s\n",
      "  5400K .......... .......... .......... .......... ..........  9%  120M 5s\n",
      "  5450K .......... .......... .......... .......... ..........  9% 96,6M 4s\n",
      "  5500K .......... .......... .......... .......... ..........  9%  131M 4s\n",
      "  5550K .......... .......... .......... .......... ..........  9%  117M 4s\n",
      "  5600K .......... .......... .......... .......... ..........  9%  102M 4s\n",
      "  5650K .......... .......... .......... .......... ..........  9%  121M 4s\n",
      "  5700K .......... .......... .......... .......... ..........  9%  102M 4s\n",
      "  5750K .......... .......... .......... .......... ..........  9%  117M 4s\n",
      "  5800K .......... .......... .......... .......... .......... 10%  121M 4s\n",
      "  5850K .......... .......... .......... .......... .......... 10% 11,5M 4s\n",
      "  5900K .......... .......... .......... .......... .......... 10%  121M 4s\n",
      "  5950K .......... .......... .......... .......... .......... 10% 97,7M 4s\n",
      "  6000K .......... .......... .......... .......... .......... 10%  104M 4s\n",
      "  6050K .......... .......... .......... .......... .......... 10% 78,9M 4s\n",
      "  6100K .......... .......... .......... .......... .......... 10% 81,4M 4s\n",
      "  6150K .......... .......... .......... .......... .......... 10% 76,8M 4s\n",
      "  6200K .......... .......... .......... .......... .......... 10%  156M 4s\n",
      "  6250K .......... .......... .......... .......... .......... 10%  150M 4s\n",
      "  6300K .......... .......... .......... .......... .......... 10%  268M 4s\n",
      "  6350K .......... .......... .......... .......... .......... 10%  151M 4s\n",
      "  6400K .......... .......... .......... .......... .......... 11%  118M 4s\n",
      "  6450K .......... .......... .......... .......... .......... 11%  103M 4s\n",
      "  6500K .......... .......... .......... .......... .......... 11%  117M 4s\n",
      "  6550K .......... .......... .......... .......... .......... 11%  122M 4s\n",
      "  6600K .......... .......... .......... .......... .......... 11%  102M 4s\n",
      "  6650K .......... .......... .......... .......... .......... 11%  119M 4s\n",
      "  6700K .......... .......... .......... .......... .......... 11%  102M 4s\n",
      "  6750K .......... .......... .......... .......... .......... 11%  119M 4s\n",
      "  6800K .......... .......... .......... .......... .......... 11%  117M 4s\n",
      "  6850K .......... .......... .......... .......... .......... 11%  103M 4s\n",
      "  6900K .......... .......... .......... .......... .......... 11%  118M 4s\n",
      "  6950K .......... .......... .......... .......... .......... 11% 2,02M 4s\n",
      "  7000K .......... .......... .......... .......... .......... 12% 14,7M 4s\n",
      "  7050K .......... .......... .......... .......... .......... 12% 84,1M 4s\n",
      "  7100K .......... .......... .......... .......... .......... 12% 53,0M 4s\n",
      "  7150K .......... .......... .......... .......... .......... 12% 74,8M 4s\n",
      "  7200K .......... .......... .......... .......... .......... 12% 67,3M 4s\n",
      "  7250K .......... .......... .......... .......... .......... 12% 82,9M 4s\n",
      "  7300K .......... .......... .......... .......... .......... 12% 86,7M 4s\n",
      "  7350K .......... .......... .......... .......... .......... 12% 62,5M 4s\n",
      "  7400K .......... .......... .......... .......... .......... 12% 88,8M 4s\n",
      "  7450K .......... .......... .......... .......... .......... 12% 70,8M 4s\n",
      "  7500K .......... .......... .......... .......... .......... 12% 74,1M 3s\n",
      "  7550K .......... .......... .......... .......... .......... 13%  117M 3s\n",
      "  7600K .......... .......... .......... .......... .......... 13% 97,1M 3s\n",
      "  7650K .......... .......... .......... .......... .......... 13% 97,0M 3s\n",
      "  7700K .......... .......... .......... .......... .......... 13% 64,4M 3s\n",
      "  7750K .......... .......... .......... .......... .......... 13%  161M 3s\n",
      "  7800K .......... .......... .......... .......... .......... 13%  158M 3s\n",
      "  7850K .......... .......... .......... .......... .......... 13%  142M 3s\n",
      "  7900K .......... .......... .......... .......... .......... 13%  168M 3s\n",
      "  7950K .......... .......... .......... .......... .......... 13%  146M 3s\n",
      "  8000K .......... .......... .......... .......... .......... 13%  157M 3s\n",
      "  8050K .......... .......... .......... .......... .......... 13%  189M 3s\n",
      "  8100K .......... .......... .......... .......... .......... 13%  159M 3s\n",
      "  8150K .......... .......... .......... .......... .......... 14%  197M 3s\n",
      "  8200K .......... .......... .......... .......... .......... 14%  154M 3s\n",
      "  8250K .......... .......... .......... .......... .......... 14%  180M 3s\n",
      "  8300K .......... .......... .......... .......... .......... 14%  118M 3s\n",
      "  8350K .......... .......... .......... .......... .......... 14%  229M 3s\n",
      "  8400K .......... .......... .......... .......... .......... 14%  275M 3s\n",
      "  8450K .......... .......... .......... .......... .......... 14%  119M 3s\n",
      "  8500K .......... .......... .......... .......... .......... 14%  122M 3s\n",
      "  8550K .......... .......... .......... .......... .......... 14%  117M 3s\n",
      "  8600K .......... .......... .......... .......... .......... 14%  103M 3s\n",
      "  8650K .......... .......... .......... .......... .......... 14%  120M 3s\n",
      "  8700K .......... .......... .......... .......... .......... 14%  103M 3s\n",
      "  8750K .......... .......... .......... .......... .......... 15%  120M 3s\n",
      "  8800K .......... .......... .......... .......... .......... 15%  117M 3s\n",
      "  8850K .......... .......... .......... .......... .......... 15%  103M 3s\n",
      "  8900K .......... .......... .......... .......... .......... 15% 14,0M 3s\n",
      "  8950K .......... .......... .......... .......... .......... 15% 36,7M 3s\n",
      "  9000K .......... .......... .......... .......... .......... 15%  125M 3s\n",
      "  9050K .......... .......... .......... .......... .......... 15%  121M 3s\n",
      "  9100K .......... .......... .......... .......... .......... 15%  102M 3s\n",
      "  9150K .......... .......... .......... .......... .......... 15%  120M 3s\n",
      "  9200K .......... .......... .......... .......... .......... 15%  103M 3s\n",
      "  9250K .......... .......... .......... .......... .......... 15%  117M 3s\n",
      "  9300K .......... .......... .......... .......... .......... 16%  121M 3s\n",
      "  9350K .......... .......... .......... .......... .......... 16%  102M 3s\n",
      "  9400K .......... .......... .......... .......... .......... 16%  114M 3s\n",
      "  9450K .......... .......... .......... .......... .......... 16%  104M 3s\n",
      "  9500K .......... .......... .......... .......... .......... 16%  122M 3s\n",
      "  9550K .......... .......... .......... .......... .......... 16%  117M 3s\n",
      "  9600K .......... .......... .......... .......... .......... 16%  102M 3s\n",
      "  9650K .......... .......... .......... .......... .......... 16%  121M 3s\n",
      "  9700K .......... .......... .......... .......... .......... 16%  102M 3s\n",
      "  9750K .......... .......... .......... .......... .......... 16%  120M 3s\n",
      "  9800K .......... .......... .......... .......... .......... 16%  118M 3s\n",
      "  9850K .......... .......... .......... .......... .......... 16%  102M 3s\n",
      "  9900K .......... .......... .......... .......... .......... 17%  122M 3s\n",
      "  9950K .......... .......... .......... .......... .......... 17%  101M 3s\n",
      " 10000K .......... .......... .......... .......... .......... 17% 2,32M 3s\n",
      " 10050K .......... .......... .......... .......... .......... 17% 9,48M 3s\n",
      " 10100K .......... .......... .......... .......... .......... 17% 23,2M 3s\n",
      " 10150K .......... .......... .......... .......... .......... 17% 61,4M 3s\n",
      " 10200K .......... .......... .......... .......... .......... 17% 62,3M 3s\n",
      " 10250K .......... .......... .......... .......... .......... 17% 70,9M 3s\n",
      " 10300K .......... .......... .......... .......... .......... 17% 81,0M 3s\n",
      " 10350K .......... .......... .......... .......... .......... 17% 43,3M 3s\n",
      " 10400K .......... .......... .......... .......... .......... 17% 81,1M 3s\n",
      " 10450K .......... .......... .......... .......... .......... 17% 69,9M 3s\n",
      " 10500K .......... .......... .......... .......... .......... 18% 77,8M 3s\n",
      " 10550K .......... .......... .......... .......... .......... 18%  185M 3s\n",
      " 10600K .......... .......... .......... .......... .......... 18%  173M 3s\n",
      " 10650K .......... .......... .......... .......... .......... 18%  189M 3s\n",
      " 10700K .......... .......... .......... .......... .......... 18%  123M 3s\n",
      " 10750K .......... .......... .......... .......... .......... 18%  148M 3s\n",
      " 10800K .......... .......... .......... .......... .......... 18%  191M 3s\n",
      " 10850K .......... .......... .......... .......... .......... 18%  154M 3s\n",
      " 10900K .......... .......... .......... .......... .......... 18%  174M 3s\n",
      " 10950K .......... .......... .......... .......... .......... 18%  116M 2s\n",
      " 11000K .......... .......... .......... .......... .......... 18%  134M 2s\n",
      " 11050K .......... .......... .......... .......... .......... 19%  123M 2s\n",
      " 11100K .......... .......... .......... .......... .......... 19%  136M 2s\n",
      " 11150K .......... .......... .......... .......... .......... 19%  130M 2s\n",
      " 11200K .......... .......... .......... .......... .......... 19%  105M 2s\n",
      " 11250K .......... .......... .......... .......... .......... 19%  121M 2s\n",
      " 11300K .......... .......... .......... .......... .......... 19%  121M 2s\n",
      " 11350K .......... .......... .......... .......... .......... 19%  102M 2s\n",
      " 11400K .......... .......... .......... .......... .......... 19%  116M 2s\n",
      " 11450K .......... .......... .......... .......... .......... 19%  104M 2s\n",
      " 11500K .......... .......... .......... .......... .......... 19%  116M 2s\n",
      " 11550K .......... .......... .......... .......... .......... 19% 93,3M 2s\n",
      " 11600K .......... .......... .......... .......... .......... 19%  115M 2s\n",
      " 11650K .......... .......... .......... .......... .......... 20%  120M 2s\n",
      " 11700K .......... .......... .......... .......... .......... 20%  101M 2s\n",
      " 11750K .......... .......... .......... .......... .......... 20%  112M 2s\n",
      " 11800K .......... .......... .......... .......... .......... 20%  114M 2s\n",
      " 11850K .......... .......... .......... .......... .......... 20% 95,1M 2s\n",
      " 11900K .......... .......... .......... .......... .......... 20%  113M 2s\n",
      " 11950K .......... .......... .......... .......... .......... 20% 74,7M 2s\n",
      " 12000K .......... .......... .......... .......... .......... 20% 24,6M 2s\n",
      " 12050K .......... .......... .......... .......... .......... 20% 62,2M 2s\n",
      " 12100K .......... .......... .......... .......... .......... 20%  105M 2s\n",
      " 12150K .......... .......... .......... .......... .......... 20% 52,3M 2s\n",
      " 12200K .......... .......... .......... .......... .......... 20%  101M 2s\n",
      " 12250K .......... .......... .......... .......... .......... 21%  122M 2s\n",
      " 12300K .......... .......... .......... .......... .......... 21%  116M 2s\n",
      " 12350K .......... .......... .......... .......... .......... 21%  103M 2s\n",
      " 12400K .......... .......... .......... .......... .......... 21%  122M 2s\n",
      " 12450K .......... .......... .......... .......... .......... 21%  101M 2s\n",
      " 12500K .......... .......... .......... .......... .......... 21%  121M 2s\n",
      " 12550K .......... .......... .......... .......... .......... 21%  117M 2s\n",
      " 12600K .......... .......... .......... .......... .......... 21%  102M 2s\n",
      " 12650K .......... .......... .......... .......... .......... 21%  122M 2s\n",
      " 12700K .......... .......... .......... .......... .......... 21% 53,6M 2s\n",
      " 12750K .......... .......... .......... .......... .......... 21%  507M 2s\n",
      " 12800K .......... .......... .......... .......... .......... 22%  170M 2s\n",
      " 12850K .......... .......... .......... .......... .......... 22%  102M 2s\n",
      " 12900K .......... .......... .......... .......... .......... 22%  121M 2s\n",
      " 12950K .......... .......... .......... .......... .......... 22%  102M 2s\n",
      " 13000K .......... .......... .......... .......... .......... 22% 78,9M 2s\n",
      " 13050K .......... .......... .......... .......... .......... 22% 9,56M 2s\n",
      " 13100K .......... .......... .......... .......... .......... 22% 2,37M 2s\n",
      " 13150K .......... .......... .......... .......... .......... 22% 24,6M 2s\n",
      " 13200K .......... .......... .......... .......... .......... 22% 40,3M 2s\n",
      " 13250K .......... .......... .......... .......... .......... 22% 55,5M 2s\n",
      " 13300K .......... .......... .......... .......... .......... 22%  125M 2s\n",
      " 13350K .......... .......... .......... .......... .......... 22% 64,9M 2s\n",
      " 13400K .......... .......... .......... .......... .......... 23%  125M 2s\n",
      " 13450K .......... .......... .......... .......... .......... 23%  102M 2s\n",
      " 13500K .......... .......... .......... .......... .......... 23%  119M 2s\n",
      " 13550K .......... .......... .......... .......... .......... 23%  120M 2s\n",
      " 13600K .......... .......... .......... .......... .......... 23%  102M 2s\n",
      " 13650K .......... .......... .......... .......... .......... 23% 57,9M 2s\n",
      " 13700K .......... .......... .......... .......... .......... 23% 62,0M 2s\n",
      " 13750K .......... .......... .......... .......... .......... 23%  118M 2s\n",
      " 13800K .......... .......... .......... .......... .......... 23% 95,0M 2s\n",
      " 13850K .......... .......... .......... .......... .......... 23% 95,3M 2s\n",
      " 13900K .......... .......... .......... .......... .......... 23%  107M 2s\n",
      " 13950K .......... .......... .......... .......... .......... 23%  147M 2s\n",
      " 14000K .......... .......... .......... .......... .......... 24%  136M 2s\n",
      " 14050K .......... .......... .......... .......... .......... 24% 83,0M 2s\n",
      " 14100K .......... .......... .......... .......... .......... 24%  103M 2s\n",
      " 14150K .......... .......... .......... .......... .......... 24%  209M 2s\n",
      " 14200K .......... .......... .......... .......... .......... 24%  102M 2s\n",
      " 14250K .......... .......... .......... .......... .......... 24%  121M 2s\n",
      " 14300K .......... .......... .......... .......... .......... 24%  116M 2s\n",
      " 14350K .......... .......... .......... .......... .......... 24%  103M 2s\n",
      " 14400K .......... .......... .......... .......... .......... 24%  121M 2s\n",
      " 14450K .......... .......... .......... .......... .......... 24%  102M 2s\n",
      " 14500K .......... .......... .......... .......... .......... 24%  118M 2s\n",
      " 14550K .......... .......... .......... .......... .......... 25%  121M 2s\n",
      " 14600K .......... .......... .......... .......... .......... 25%  100M 2s\n",
      " 14650K .......... .......... .......... .......... .......... 25%  122M 2s\n",
      " 14700K .......... .......... .......... .......... .......... 25%  103M 2s\n",
      " 14750K .......... .......... .......... .......... .......... 25%  117M 2s\n",
      " 14800K .......... .......... .......... .......... .......... 25%  122M 2s\n",
      " 14850K .......... .......... .......... .......... .......... 25%  102M 2s\n",
      " 14900K .......... .......... .......... .......... .......... 25%  117M 2s\n",
      " 14950K .......... .......... .......... .......... .......... 25%  102M 2s\n",
      " 15000K .......... .......... .......... .......... .......... 25%  121M 2s\n",
      " 15050K .......... .......... .......... .......... .......... 25%  121M 2s\n",
      " 15100K .......... .......... .......... .......... .......... 25% 37,0M 2s\n",
      " 15150K .......... .......... .......... .......... .......... 26% 46,8M 2s\n",
      " 15200K .......... .......... .......... .......... .......... 26% 55,6M 2s\n",
      " 15250K .......... .......... .......... .......... .......... 26%  123M 2s\n",
      " 15300K .......... .......... .......... .......... .......... 26% 73,7M 2s\n",
      " 15350K .......... .......... .......... .......... .......... 26%  102M 2s\n",
      " 15400K .......... .......... .......... .......... .......... 26%  121M 2s\n",
      " 15450K .......... .......... .......... .......... .......... 26%  102M 2s\n",
      " 15500K .......... .......... .......... .......... .......... 26%  117M 2s\n",
      " 15550K .......... .......... .......... .......... .......... 26%  121M 2s\n",
      " 15600K .......... .......... .......... .......... .......... 26%  102M 2s\n",
      " 15650K .......... .......... .......... .......... .......... 26%  121M 2s\n",
      " 15700K .......... .......... .......... .......... .......... 26%  102M 2s\n",
      " 15750K .......... .......... .......... .......... .......... 27%  117M 2s\n",
      " 15800K .......... .......... .......... .......... .......... 27%  121M 2s\n",
      " 15850K .......... .......... .......... .......... .......... 27%  102M 2s\n",
      " 15900K .......... .......... .......... .......... .......... 27%  117M 2s\n",
      " 15950K .......... .......... .......... .......... .......... 27%  103M 2s\n",
      " 16000K .......... .......... .......... .......... .......... 27%  120M 2s\n",
      " 16050K .......... .......... .......... .......... .......... 27%  117M 2s\n",
      " 16100K .......... .......... .......... .......... .......... 27% 10,8M 2s\n",
      " 16150K .......... .......... .......... .......... .......... 27% 2,38M 2s\n",
      " 16200K .......... .......... .......... .......... .......... 27% 26,5M 2s\n",
      " 16250K .......... .......... .......... .......... .......... 27% 30,4M 2s\n",
      " 16300K .......... .......... .......... .......... .......... 28% 37,2M 2s\n",
      " 16350K .......... .......... .......... .......... .......... 28% 36,5M 2s\n",
      " 16400K .......... .......... .......... .......... .......... 28%  109M 2s\n",
      " 16450K .......... .......... .......... .......... .......... 28%  102M 2s\n",
      " 16500K .......... .......... .......... .......... .......... 28% 59,3M 2s\n",
      " 16550K .......... .......... .......... .......... .......... 28%  111M 2s\n",
      " 16600K .......... .......... .......... .......... .......... 28% 96,2M 2s\n",
      " 16650K .......... .......... .......... .......... .......... 28%  116M 2s\n",
      " 16700K .......... .......... .......... .......... .......... 28%  106M 2s\n",
      " 16750K .......... .......... .......... .......... .......... 28%  110M 2s\n",
      " 16800K .......... .......... .......... .......... .......... 28%  135M 2s\n",
      " 16850K .......... .......... .......... .......... .......... 28%  100M 2s\n",
      " 16900K .......... .......... .......... .......... .......... 29%  137M 2s\n",
      " 16950K .......... .......... .......... .......... .......... 29%  122M 2s\n",
      " 17000K .......... .......... .......... .......... .......... 29%  136M 2s\n",
      " 17050K .......... .......... .......... .......... .......... 29%  133M 2s\n",
      " 17100K .......... .......... .......... .......... .......... 29%  116M 2s\n",
      " 17150K .......... .......... .......... .......... .......... 29%  137M 2s\n",
      " 17200K .......... .......... .......... .......... .......... 29%  102M 2s\n",
      " 17250K .......... .......... .......... .......... .......... 29%  121M 2s\n",
      " 17300K .......... .......... .......... .......... .......... 29%  120M 2s\n",
      " 17350K .......... .......... .......... .......... .......... 29%  100M 2s\n",
      " 17400K .......... .......... .......... .......... .......... 29%  125M 2s\n",
      " 17450K .......... .......... .......... .......... .......... 29%  101M 2s\n",
      " 17500K .......... .......... .......... .......... .......... 30%  118M 2s\n",
      " 17550K .......... .......... .......... .......... .......... 30%  121M 2s\n",
      " 17600K .......... .......... .......... .......... .......... 30%  103M 2s\n",
      " 17650K .......... .......... .......... .......... .......... 30%  118M 2s\n",
      " 17700K .......... .......... .......... .......... .......... 30%  103M 2s\n",
      " 17750K .......... .......... .......... .......... .......... 30%  122M 2s\n",
      " 17800K .......... .......... .......... .......... .......... 30%  122M 2s\n",
      " 17850K .......... .......... .......... .......... .......... 30%  102M 2s\n",
      " 17900K .......... .......... .......... .......... .......... 30%  117M 2s\n",
      " 17950K .......... .......... .......... .......... .......... 30% 95,7M 2s\n",
      " 18000K .......... .......... .......... .......... .......... 30%  129M 2s\n",
      " 18050K .......... .......... .......... .......... .......... 31%  113M 2s\n",
      " 18100K .......... .......... .......... .......... .......... 31%  108M 2s\n",
      " 18150K .......... .......... .......... .......... .......... 31% 89,1M 2s\n",
      " 18200K .......... .......... .......... .......... .......... 31% 30,5M 2s\n",
      " 18250K .......... .......... .......... .......... .......... 31% 51,6M 2s\n",
      " 18300K .......... .......... .......... .......... .......... 31% 62,3M 2s\n",
      " 18350K .......... .......... .......... .......... .......... 31%  106M 2s\n",
      " 18400K .......... .......... .......... .......... .......... 31%  120M 2s\n",
      " 18450K .......... .......... .......... .......... .......... 31%  102M 2s\n",
      " 18500K .......... .......... .......... .......... .......... 31%  118M 2s\n",
      " 18550K .......... .......... .......... .......... .......... 31%  121M 2s\n",
      " 18600K .......... .......... .......... .......... .......... 31%  102M 2s\n",
      " 18650K .......... .......... .......... .......... .......... 32%  117M 2s\n",
      " 18700K .......... .......... .......... .......... .......... 32%  103M 2s\n",
      " 18750K .......... .......... .......... .......... .......... 32%  121M 1s\n",
      " 18800K .......... .......... .......... .......... .......... 32%  121M 1s\n",
      " 18850K .......... .......... .......... .......... .......... 32%  102M 1s\n",
      " 18900K .......... .......... .......... .......... .......... 32%  118M 1s\n",
      " 18950K .......... .......... .......... .......... .......... 32%  102M 1s\n",
      " 19000K .......... .......... .......... .......... .......... 32%  121M 1s\n",
      " 19050K .......... .......... .......... .......... .......... 32%  117M 1s\n",
      " 19100K .......... .......... .......... .......... .......... 32%  102M 1s\n",
      " 19150K .......... .......... .......... .......... .......... 32% 16,8M 1s\n",
      " 19200K .......... .......... .......... .......... .......... 32% 2,58M 1s\n",
      " 19250K .......... .......... .......... .......... .......... 33% 11,0M 1s\n",
      " 19300K .......... .......... .......... .......... .......... 33% 43,5M 1s\n",
      " 19350K .......... .......... .......... .......... .......... 33% 33,3M 1s\n",
      " 19400K .......... .......... .......... .......... .......... 33% 35,2M 1s\n",
      " 19450K .......... .......... .......... .......... .......... 33% 57,4M 1s\n",
      " 19500K .......... .......... .......... .......... .......... 33% 60,4M 1s\n",
      " 19550K .......... .......... .......... .......... .......... 33%  125M 1s\n",
      " 19600K .......... .......... .......... .......... .......... 33% 57,6M 1s\n",
      " 19650K .......... .......... .......... .......... .......... 33%  112M 1s\n",
      " 19700K .......... .......... .......... .......... .......... 33%  107M 1s\n",
      " 19750K .......... .......... .......... .......... .......... 33%  110M 1s\n",
      " 19800K .......... .......... .......... .......... .......... 34%  129M 1s\n",
      " 19850K .......... .......... .......... .......... .......... 34%  104M 1s\n",
      " 19900K .......... .......... .......... .......... .......... 34%  121M 1s\n",
      " 19950K .......... .......... .......... .......... .......... 34%  102M 1s\n",
      " 20000K .......... .......... .......... .......... .......... 34%  121M 1s\n",
      " 20050K .......... .......... .......... .......... .......... 34%  117M 1s\n",
      " 20100K .......... .......... .......... .......... .......... 34% 85,5M 1s\n",
      " 20150K .......... .......... .......... .......... .......... 34%  158M 1s\n",
      " 20200K .......... .......... .......... .......... .......... 34%  102M 1s\n",
      " 20250K .......... .......... .......... .......... .......... 34%  114M 1s\n",
      " 20300K .......... .......... .......... .......... .......... 34%  125M 1s\n",
      " 20350K .......... .......... .......... .......... .......... 34%  102M 1s\n",
      " 20400K .......... .......... .......... .......... .......... 35%  122M 1s\n",
      " 20450K .......... .......... .......... .......... .......... 35%  102M 1s\n",
      " 20500K .......... .......... .......... .......... .......... 35%  115M 1s\n",
      " 20550K .......... .......... .......... .......... .......... 35%  124M 1s\n",
      " 20600K .......... .......... .......... .......... .......... 35%  102M 1s\n",
      " 20650K .......... .......... .......... .......... .......... 35%  118M 1s\n",
      " 20700K .......... .......... .......... .......... .......... 35%  103M 1s\n",
      " 20750K .......... .......... .......... .......... .......... 35%  121M 1s\n",
      " 20800K .......... .......... .......... .......... .......... 35%  117M 1s\n",
      " 20850K .......... .......... .......... .......... .......... 35%  102M 1s\n",
      " 20900K .......... .......... .......... .......... .......... 35%  121M 1s\n",
      " 20950K .......... .......... .......... .......... .......... 35%  102M 1s\n",
      " 21000K .......... .......... .......... .......... .......... 36%  121M 1s\n",
      " 21050K .......... .......... .......... .......... .......... 36%  117M 1s\n",
      " 21100K .......... .......... .......... .......... .......... 36%  103M 1s\n",
      " 21150K .......... .......... .......... .......... .......... 36%  120M 1s\n",
      " 21200K .......... .......... .......... .......... .......... 36%  102M 1s\n",
      " 21250K .......... .......... .......... .......... .......... 36%  119M 1s\n",
      " 21300K .......... .......... .......... .......... .......... 36% 43,6M 1s\n",
      " 21350K .......... .......... .......... .......... .......... 36% 38,7M 1s\n",
      " 21400K .......... .......... .......... .......... .......... 36% 76,5M 1s\n",
      " 21450K .......... .......... .......... .......... .......... 36%  102M 1s\n",
      " 21500K .......... .......... .......... .......... .......... 36%  121M 1s\n",
      " 21550K .......... .......... .......... .......... .......... 37%  121M 1s\n",
      " 21600K .......... .......... .......... .......... .......... 37%  102M 1s\n",
      " 21650K .......... .......... .......... .......... .......... 37%  117M 1s\n",
      " 21700K .......... .......... .......... .......... .......... 37%  102M 1s\n",
      " 21750K .......... .......... .......... .......... .......... 37%  121M 1s\n",
      " 21800K .......... .......... .......... .......... .......... 37%  117M 1s\n",
      " 21850K .......... .......... .......... .......... .......... 37%  102M 1s\n",
      " 21900K .......... .......... .......... .......... .......... 37%  118M 1s\n",
      " 21950K .......... .......... .......... .......... .......... 37%  104M 1s\n",
      " 22000K .......... .......... .......... .......... .......... 37%  118M 1s\n",
      " 22050K .......... .......... .......... .......... .......... 37%  120M 1s\n",
      " 22100K .......... .......... .......... .......... .......... 37%  103M 1s\n",
      " 22150K .......... .......... .......... .......... .......... 38%  120M 1s\n",
      " 22200K .......... .......... .......... .......... .......... 38%  101M 1s\n",
      " 22250K .......... .......... .......... .......... .......... 38% 2,43M 1s\n",
      " 22300K .......... .......... .......... .......... .......... 38% 7,62M 1s\n",
      " 22350K .......... .......... .......... .......... .......... 38% 56,6M 1s\n",
      " 22400K .......... .......... .......... .......... .......... 38% 41,9M 1s\n",
      " 22450K .......... .......... .......... .......... .......... 38% 67,5M 1s\n",
      " 22500K .......... .......... .......... .......... .......... 38% 29,0M 1s\n",
      " 22550K .......... .......... .......... .......... .......... 38% 42,6M 1s\n",
      " 22600K .......... .......... .......... .......... .......... 38% 77,9M 1s\n",
      " 22650K .......... .......... .......... .......... .......... 38% 83,4M 1s\n",
      " 22700K .......... .......... .......... .......... .......... 38%  157M 1s\n",
      " 22750K .......... .......... .......... .......... .......... 39%  122M 1s\n",
      " 22800K .......... .......... .......... .......... .......... 39%  118M 1s\n",
      " 22850K .......... .......... .......... .......... .......... 39%  102M 1s\n",
      " 22900K .......... .......... .......... .......... .......... 39%  122M 1s\n",
      " 22950K .......... .......... .......... .......... .......... 39%  103M 1s\n",
      " 23000K .......... .......... .......... .......... .......... 39%  117M 1s\n",
      " 23050K .......... .......... .......... .......... .......... 39%  121M 1s\n",
      " 23100K .......... .......... .......... .......... .......... 39% 37,2M 1s\n",
      " 23150K .......... .......... .......... .......... .......... 39%  249M 1s\n",
      " 23200K .......... .......... .......... .......... .......... 39%  192M 1s\n",
      " 23250K .......... .......... .......... .......... .......... 39%  179M 1s\n",
      " 23300K .......... .......... .......... .......... .......... 40%  187M 1s\n",
      " 23350K .......... .......... .......... .......... .......... 40%  136M 1s\n",
      " 23400K .......... .......... .......... .......... .......... 40%  115M 1s\n",
      " 23450K .......... .......... .......... .......... .......... 40% 91,9M 1s\n",
      " 23500K .......... .......... .......... .......... .......... 40%  140M 1s\n",
      " 23550K .......... .......... .......... .......... .......... 40%  122M 1s\n",
      " 23600K .......... .......... .......... .......... .......... 40%  102M 1s\n",
      " 23650K .......... .......... .......... .......... .......... 40%  122M 1s\n",
      " 23700K .......... .......... .......... .......... .......... 40% 95,8M 1s\n",
      " 23750K .......... .......... .......... .......... .......... 40%  130M 1s\n",
      " 23800K .......... .......... .......... .......... .......... 40%  118M 1s\n",
      " 23850K .......... .......... .......... .......... .......... 40%  101M 1s\n",
      " 23900K .......... .......... .......... .......... .......... 41%  111M 1s\n",
      " 23950K .......... .......... .......... .......... .......... 41%  110M 1s\n",
      " 24000K .......... .......... .......... .......... .......... 41%  117M 1s\n",
      " 24050K .......... .......... .......... .......... .......... 41%  122M 1s\n",
      " 24100K .......... .......... .......... .......... .......... 41%  102M 1s\n",
      " 24150K .......... .......... .......... .......... .......... 41%  121M 1s\n",
      " 24200K .......... .......... .......... .......... .......... 41%  103M 1s\n",
      " 24250K .......... .......... .......... .......... .......... 41%  116M 1s\n",
      " 24300K .......... .......... .......... .......... .......... 41%  121M 1s\n",
      " 24350K .......... .......... .......... .......... .......... 41%  102M 1s\n",
      " 24400K .......... .......... .......... .......... .......... 41% 66,2M 1s\n",
      " 24450K .......... .......... .......... .......... .......... 41% 25,7M 1s\n",
      " 24500K .......... .......... .......... .......... .......... 42%  120M 1s\n",
      " 24550K .......... .......... .......... .......... .......... 42%  117M 1s\n",
      " 24600K .......... .......... .......... .......... .......... 42%  102M 1s\n",
      " 24650K .......... .......... .......... .......... .......... 42%  121M 1s\n",
      " 24700K .......... .......... .......... .......... .......... 42%  102M 1s\n",
      " 24750K .......... .......... .......... .......... .......... 42%  121M 1s\n",
      " 24800K .......... .......... .......... .......... .......... 42%  117M 1s\n",
      " 24850K .......... .......... .......... .......... .......... 42%  102M 1s\n",
      " 24900K .......... .......... .......... .......... .......... 42%  121M 1s\n",
      " 24950K .......... .......... .......... .......... .......... 42%  102M 1s\n",
      " 25000K .......... .......... .......... .......... .......... 42%  117M 1s\n",
      " 25050K .......... .......... .......... .......... .......... 43%  121M 1s\n",
      " 25100K .......... .......... .......... .......... .......... 43%  102M 1s\n",
      " 25150K .......... .......... .......... .......... .......... 43%  117M 1s\n",
      " 25200K .......... .......... .......... .......... .......... 43%  102M 1s\n",
      " 25250K .......... .......... .......... .......... .......... 43% 20,3M 1s\n",
      " 25300K .......... .......... .......... .......... .......... 43% 17,6M 1s\n",
      " 25350K .......... .......... .......... .......... .......... 43% 2,27M 1s\n",
      " 25400K .......... .......... .......... .......... .......... 43% 20,6M 1s\n",
      " 25450K .......... .......... .......... .......... .......... 43%  102M 1s\n",
      " 25500K .......... .......... .......... .......... .......... 43%  118M 1s\n",
      " 25550K .......... .......... .......... .......... .......... 43%  118M 1s\n",
      " 25600K .......... .......... .......... .......... .......... 43% 28,7M 1s\n",
      " 25650K .......... .......... .......... .......... .......... 44% 52,4M 1s\n",
      " 25700K .......... .......... .......... .......... .......... 44%  102M 1s\n",
      " 25750K .......... .......... .......... .......... .......... 44%  112M 1s\n",
      " 25800K .......... .......... .......... .......... .......... 44%  121M 1s\n",
      " 25850K .......... .......... .......... .......... .......... 44% 94,8M 1s\n",
      " 25900K .......... .......... .......... .......... .......... 44%  137M 1s\n",
      " 25950K .......... .......... .......... .......... .......... 44%  105M 1s\n",
      " 26000K .......... .......... .......... .......... .......... 44%  116M 1s\n",
      " 26050K .......... .......... .......... .......... .......... 44%  122M 1s\n",
      " 26100K .......... .......... .......... .......... .......... 44% 77,6M 1s\n",
      " 26150K .......... .......... .......... .......... .......... 44%  131M 1s\n",
      " 26200K .......... .......... .......... .......... .......... 44%  103M 1s\n",
      " 26250K .......... .......... .......... .......... .......... 45%  121M 1s\n",
      " 26300K .......... .......... .......... .......... .......... 45%  120M 1s\n",
      " 26350K .......... .......... .......... .......... .......... 45%  103M 1s\n",
      " 26400K .......... .......... .......... .......... .......... 45%  117M 1s\n",
      " 26450K .......... .......... .......... .......... .......... 45%  102M 1s\n",
      " 26500K .......... .......... .......... .......... .......... 45%  121M 1s\n",
      " 26550K .......... .......... .......... .......... .......... 45%  117M 1s\n",
      " 26600K .......... .......... .......... .......... .......... 45%  102M 1s\n",
      " 26650K .......... .......... .......... .......... .......... 45%  122M 1s\n",
      " 26700K .......... .......... .......... .......... .......... 45% 88,1M 1s\n",
      " 26750K .......... .......... .......... .......... .......... 45%  143M 1s\n",
      " 26800K .......... .......... .......... .......... .......... 46%  121M 1s\n",
      " 26850K .......... .......... .......... .......... .......... 46%  102M 1s\n",
      " 26900K .......... .......... .......... .......... .......... 46%  121M 1s\n",
      " 26950K .......... .......... .......... .......... .......... 46%  102M 1s\n",
      " 27000K .......... .......... .......... .......... .......... 46%  118M 1s\n",
      " 27050K .......... .......... .......... .......... .......... 46%  120M 1s\n",
      " 27100K .......... .......... .......... .......... .......... 46%  102M 1s\n",
      " 27150K .......... .......... .......... .......... .......... 46%  118M 1s\n",
      " 27200K .......... .......... .......... .......... .......... 46%  100M 1s\n",
      " 27250K .......... .......... .......... .......... .......... 46%  119M 1s\n",
      " 27300K .......... .......... .......... .......... .......... 46%  120M 1s\n",
      " 27350K .......... .......... .......... .......... .......... 46%  100M 1s\n",
      " 27400K .......... .......... .......... .......... .......... 47%  118M 1s\n",
      " 27450K .......... .......... .......... .......... .......... 47%  108M 1s\n",
      " 27500K .......... .......... .......... .......... .......... 47% 30,3M 1s\n",
      " 27550K .......... .......... .......... .......... .......... 47%  101M 1s\n",
      " 27600K .......... .......... .......... .......... .......... 47% 63,7M 1s\n",
      " 27650K .......... .......... .......... .......... .......... 47%  120M 1s\n",
      " 27700K .......... .......... .......... .......... .......... 47% 94,6M 1s\n",
      " 27750K .......... .......... .......... .......... .......... 47%  129M 1s\n",
      " 27800K .......... .......... .......... .......... .......... 47%  122M 1s\n",
      " 27850K .......... .......... .......... .......... .......... 47%  102M 1s\n",
      " 27900K .......... .......... .......... .......... .......... 47%  121M 1s\n",
      " 27950K .......... .......... .......... .......... .......... 47%  102M 1s\n",
      " 28000K .......... .......... .......... .......... .......... 48%  117M 1s\n",
      " 28050K .......... .......... .......... .......... .......... 48%  121M 1s\n",
      " 28100K .......... .......... .......... .......... .......... 48%  100M 1s\n",
      " 28150K .......... .......... .......... .......... .......... 48%  120M 1s\n",
      " 28200K .......... .......... .......... .......... .......... 48%  102M 1s\n",
      " 28250K .......... .......... .......... .......... .......... 48%  121M 1s\n",
      " 28300K .......... .......... .......... .......... .......... 48% 28,6M 1s\n",
      " 28350K .......... .......... .......... .......... .......... 48% 14,6M 1s\n",
      " 28400K .......... .......... .......... .......... .......... 48% 2,32M 1s\n",
      " 28450K .......... .......... .......... .......... .......... 48% 16,5M 1s\n",
      " 28500K .......... .......... .......... .......... .......... 48%  117M 1s\n",
      " 28550K .......... .......... .......... .......... .......... 49% 69,7M 1s\n",
      " 28600K .......... .......... .......... .......... .......... 49% 76,4M 1s\n",
      " 28650K .......... .......... .......... .......... .......... 49% 49,3M 1s\n",
      " 28700K .......... .......... .......... .......... .......... 49% 41,0M 1s\n",
      " 28750K .......... .......... .......... .......... .......... 49% 83,1M 1s\n",
      " 28800K .......... .......... .......... .......... .......... 49%  117M 1s\n",
      " 28850K .......... .......... .......... .......... .......... 49%  101M 1s\n",
      " 28900K .......... .......... .......... .......... .......... 49%  118M 1s\n",
      " 28950K .......... .......... .......... .......... .......... 49%  103M 1s\n",
      " 29000K .......... .......... .......... .......... .......... 49%  120M 1s\n",
      " 29050K .......... .......... .......... .......... .......... 49%  122M 1s\n",
      " 29100K .......... .......... .......... .......... .......... 49%  103M 1s\n",
      " 29150K .......... .......... .......... .......... .......... 50%  117M 1s\n",
      " 29200K .......... .......... .......... .......... .......... 50%  102M 1s\n",
      " 29250K .......... .......... .......... .......... .......... 50%  121M 1s\n",
      " 29300K .......... .......... .......... .......... .......... 50%  117M 1s\n",
      " 29350K .......... .......... .......... .......... .......... 50%  101M 1s\n",
      " 29400K .......... .......... .......... .......... .......... 50%  122M 1s\n",
      " 29450K .......... .......... .......... .......... .......... 50%  103M 1s\n",
      " 29500K .......... .......... .......... .......... .......... 50%  117M 1s\n",
      " 29550K .......... .......... .......... .......... .......... 50%  119M 1s\n",
      " 29600K .......... .......... .......... .......... .......... 50%  103M 1s\n",
      " 29650K .......... .......... .......... .......... .......... 50%  120M 1s\n",
      " 29700K .......... .......... .......... .......... .......... 50%  103M 1s\n",
      " 29750K .......... .......... .......... .......... .......... 51%  117M 1s\n",
      " 29800K .......... .......... .......... .......... .......... 51%  118M 1s\n",
      " 29850K .......... .......... .......... .......... .......... 51%  101M 1s\n",
      " 29900K .......... .......... .......... .......... .......... 51%  119M 1s\n",
      " 29950K .......... .......... .......... .......... .......... 51%  102M 1s\n",
      " 30000K .......... .......... .......... .......... .......... 51%  123M 1s\n",
      " 30050K .......... .......... .......... .......... .......... 51%  121M 1s\n",
      " 30100K .......... .......... .......... .......... .......... 51%  102M 1s\n",
      " 30150K .......... .......... .......... .......... .......... 51%  119M 1s\n",
      " 30200K .......... .......... .......... .......... .......... 51%  102M 1s\n",
      " 30250K .......... .......... .......... .......... .......... 51%  122M 1s\n",
      " 30300K .......... .......... .......... .......... .......... 52%  118M 1s\n",
      " 30350K .......... .......... .......... .......... .......... 52%  102M 1s\n",
      " 30400K .......... .......... .......... .......... .......... 52%  121M 1s\n",
      " 30450K .......... .......... .......... .......... .......... 52%  102M 1s\n",
      " 30500K .......... .......... .......... .......... .......... 52%  118M 1s\n",
      " 30550K .......... .......... .......... .......... .......... 52% 47,3M 1s\n",
      " 30600K .......... .......... .......... .......... .......... 52% 49,0M 1s\n",
      " 30650K .......... .......... .......... .......... .......... 52%  121M 1s\n",
      " 30700K .......... .......... .......... .......... .......... 52% 58,5M 1s\n",
      " 30750K .......... .......... .......... .......... .......... 52% 78,0M 1s\n",
      " 30800K .......... .......... .......... .......... .......... 52%  121M 1s\n",
      " 30850K .......... .......... .......... .......... .......... 52% 72,1M 1s\n",
      " 30900K .......... .......... .......... .......... .......... 53%  118M 1s\n",
      " 30950K .......... .......... .......... .......... .......... 53%  102M 1s\n",
      " 31000K .......... .......... .......... .......... .......... 53%  121M 1s\n",
      " 31050K .......... .......... .......... .......... .......... 53%  117M 1s\n",
      " 31100K .......... .......... .......... .......... .......... 53% 98,2M 1s\n",
      " 31150K .......... .......... .......... .......... .......... 53%  114M 1s\n",
      " 31200K .......... .......... .......... .......... .......... 53%  112M 1s\n",
      " 31250K .......... .......... .......... .......... .......... 53%  121M 1s\n",
      " 31300K .......... .......... .......... .......... .......... 53%  118M 1s\n",
      " 31350K .......... .......... .......... .......... .......... 53%  102M 1s\n",
      " 31400K .......... .......... .......... .......... .......... 53% 13,4M 1s\n",
      " 31450K .......... .......... .......... .......... .......... 53% 2,48M 1s\n",
      " 31500K .......... .......... .......... .......... .......... 54% 9,87M 1s\n",
      " 31550K .......... .......... .......... .......... .......... 54% 62,8M 1s\n",
      " 31600K .......... .......... .......... .......... .......... 54%  103M 1s\n",
      " 31650K .......... .......... .......... .......... .......... 54% 52,0M 1s\n",
      " 31700K .......... .......... .......... .......... .......... 54%  101M 1s\n",
      " 31750K .......... .......... .......... .......... .......... 54%  118M 1s\n",
      " 31800K .......... .......... .......... .......... .......... 54%  120M 1s\n",
      " 31850K .......... .......... .......... .......... .......... 54%  104M 1s\n",
      " 31900K .......... .......... .......... .......... .......... 54% 37,0M 1s\n",
      " 31950K .......... .......... .......... .......... .......... 54% 66,9M 1s\n",
      " 32000K .......... .......... .......... .......... .......... 54%  121M 1s\n",
      " 32050K .......... .......... .......... .......... .......... 55%  117M 1s\n",
      " 32100K .......... .......... .......... .......... .......... 55%  102M 1s\n",
      " 32150K .......... .......... .......... .......... .......... 55%  121M 1s\n",
      " 32200K .......... .......... .......... .......... .......... 55%  103M 1s\n",
      " 32250K .......... .......... .......... .......... .......... 55%  121M 1s\n",
      " 32300K .......... .......... .......... .......... .......... 55%  117M 1s\n",
      " 32350K .......... .......... .......... .......... .......... 55%  103M 1s\n",
      " 32400K .......... .......... .......... .......... .......... 55%  121M 1s\n",
      " 32450K .......... .......... .......... .......... .......... 55%  102M 1s\n",
      " 32500K .......... .......... .......... .......... .......... 55%  118M 1s\n",
      " 32550K .......... .......... .......... .......... .......... 55%  121M 1s\n",
      " 32600K .......... .......... .......... .......... .......... 55%  102M 1s\n",
      " 32650K .......... .......... .......... .......... .......... 56%  118M 1s\n",
      " 32700K .......... .......... .......... .......... .......... 56% 96,8M 1s\n",
      " 32750K .......... .......... .......... .......... .......... 56%  129M 1s\n",
      " 32800K .......... .......... .......... .......... .......... 56%  122M 1s\n",
      " 32850K .......... .......... .......... .......... .......... 56% 87,6M 1s\n",
      " 32900K .......... .......... .......... .......... .......... 56%  144M 1s\n",
      " 32950K .......... .......... .......... .......... .......... 56%  102M 1s\n",
      " 33000K .......... .......... .......... .......... .......... 56%  122M 1s\n",
      " 33050K .......... .......... .......... .......... .......... 56%  117M 1s\n",
      " 33100K .......... .......... .......... .......... .......... 56%  102M 1s\n",
      " 33150K .......... .......... .......... .......... .......... 56%  121M 1s\n",
      " 33200K .......... .......... .......... .......... .......... 56%  102M 1s\n",
      " 33250K .......... .......... .......... .......... .......... 57%  118M 1s\n",
      " 33300K .......... .......... .......... .......... .......... 57%  120M 1s\n",
      " 33350K .......... .......... .......... .......... .......... 57%  102M 1s\n",
      " 33400K .......... .......... .......... .......... .......... 57%  121M 1s\n",
      " 33450K .......... .......... .......... .......... .......... 57%  102M 1s\n",
      " 33500K .......... .......... .......... .......... .......... 57%  118M 1s\n",
      " 33550K .......... .......... .......... .......... .......... 57%  121M 1s\n",
      " 33600K .......... .......... .......... .......... .......... 57%  102M 1s\n",
      " 33650K .......... .......... .......... .......... .......... 57%  119M 1s\n",
      " 33700K .......... .......... .......... .......... .......... 57% 27,4M 1s\n",
      " 33750K .......... .......... .......... .......... .......... 57%  122M 1s\n",
      " 33800K .......... .......... .......... .......... .......... 58%  120M 1s\n",
      " 33850K .......... .......... .......... .......... .......... 58% 75,5M 1s\n",
      " 33900K .......... .......... .......... .......... .......... 58%  120M 1s\n",
      " 33950K .......... .......... .......... .......... .......... 58%  103M 1s\n",
      " 34000K .......... .......... .......... .......... .......... 58%  120M 1s\n",
      " 34050K .......... .......... .......... .......... .......... 58%  117M 1s\n",
      " 34100K .......... .......... .......... .......... .......... 58%  102M 1s\n",
      " 34150K .......... .......... .......... .......... .......... 58%  121M 1s\n",
      " 34200K .......... .......... .......... .......... .......... 58%  102M 1s\n",
      " 34250K .......... .......... .......... .......... .......... 58%  117M 1s\n",
      " 34300K .......... .......... .......... .......... .......... 58%  121M 1s\n",
      " 34350K .......... .......... .......... .......... .......... 58%  103M 1s\n",
      " 34400K .......... .......... .......... .......... .......... 59%  120M 1s\n",
      " 34450K .......... .......... .......... .......... .......... 59% 12,5M 1s\n",
      " 34500K .......... .......... .......... .......... .......... 59% 2,91M 1s\n",
      " 34550K .......... .......... .......... .......... .......... 59% 7,31M 1s\n",
      " 34600K .......... .......... .......... .......... .......... 59% 21,6M 1s\n",
      " 34650K .......... .......... .......... .......... .......... 59% 59,3M 1s\n",
      " 34700K .......... .......... .......... .......... .......... 59%  104M 1s\n",
      " 34750K .......... .......... .......... .......... .......... 59%  121M 1s\n",
      " 34800K .......... .......... .......... .......... .......... 59%  117M 1s\n",
      " 34850K .......... .......... .......... .......... .......... 59% 99,0M 1s\n",
      " 34900K .......... .......... .......... .......... .......... 59%  126M 1s\n",
      " 34950K .......... .......... .......... .......... .......... 59% 46,6M 1s\n",
      " 35000K .......... .......... .......... .......... .......... 60%  116M 1s\n",
      " 35050K .......... .......... .......... .......... .......... 60% 80,7M 1s\n",
      " 35100K .......... .......... .......... .......... .......... 60% 56,4M 1s\n",
      " 35150K .......... .......... .......... .......... .......... 60% 83,3M 1s\n",
      " 35200K .......... .......... .......... .......... .......... 60%  107M 1s\n",
      " 35250K .......... .......... .......... .......... .......... 60%  147M 1s\n",
      " 35300K .......... .......... .......... .......... .......... 60%  148M 1s\n",
      " 35350K .......... .......... .......... .......... .......... 60%  122M 1s\n",
      " 35400K .......... .......... .......... .......... .......... 60%  121M 1s\n",
      " 35450K .......... .......... .......... .......... .......... 60%  101M 1s\n",
      " 35500K .......... .......... .......... .......... .......... 60%  118M 1s\n",
      " 35550K .......... .......... .......... .......... .......... 60%  122M 1s\n",
      " 35600K .......... .......... .......... .......... .......... 61%  102M 1s\n",
      " 35650K .......... .......... .......... .......... .......... 61%  116M 1s\n",
      " 35700K .......... .......... .......... .......... .......... 61%  104M 1s\n",
      " 35750K .......... .......... .......... .......... .......... 61%  119M 1s\n",
      " 35800K .......... .......... .......... .......... .......... 61%  117M 1s\n",
      " 35850K .......... .......... .......... .......... .......... 61% 73,3M 1s\n",
      " 35900K .......... .......... .......... .......... .......... 61%  238M 1s\n",
      " 35950K .......... .......... .......... .......... .......... 61%  103M 1s\n",
      " 36000K .......... .......... .......... .......... .......... 61%  121M 1s\n",
      " 36050K .......... .......... .......... .......... .......... 61%  117M 1s\n",
      " 36100K .......... .......... .......... .......... .......... 61%  103M 1s\n",
      " 36150K .......... .......... .......... .......... .......... 62%  121M 1s\n",
      " 36200K .......... .......... .......... .......... .......... 62%  103M 1s\n",
      " 36250K .......... .......... .......... .......... .......... 62%  117M 1s\n",
      " 36300K .......... .......... .......... .......... .......... 62%  120M 1s\n",
      " 36350K .......... .......... .......... .......... .......... 62%  102M 1s\n",
      " 36400K .......... .......... .......... .......... .......... 62%  118M 1s\n",
      " 36450K .......... .......... .......... .......... .......... 62%  102M 1s\n",
      " 36500K .......... .......... .......... .......... .......... 62%  119M 1s\n",
      " 36550K .......... .......... .......... .......... .......... 62%  123M 1s\n",
      " 36600K .......... .......... .......... .......... .......... 62%  102M 1s\n",
      " 36650K .......... .......... .......... .......... .......... 62%  117M 1s\n",
      " 36700K .......... .......... .......... .......... .......... 62%  102M 1s\n",
      " 36750K .......... .......... .......... .......... .......... 63%  122M 1s\n",
      " 36800K .......... .......... .......... .......... .......... 63% 57,6M 1s\n",
      " 36850K .......... .......... .......... .......... .......... 63% 29,0M 1s\n",
      " 36900K .......... .......... .......... .......... .......... 63%  121M 1s\n",
      " 36950K .......... .......... .......... .......... .......... 63%  102M 1s\n",
      " 37000K .......... .......... .......... .......... .......... 63%  117M 1s\n",
      " 37050K .......... .......... .......... .......... .......... 63%  121M 1s\n",
      " 37100K .......... .......... .......... .......... .......... 63%  103M 1s\n",
      " 37150K .......... .......... .......... .......... .......... 63%  121M 1s\n",
      " 37200K .......... .......... .......... .......... .......... 63%  102M 1s\n",
      " 37250K .......... .......... .......... .......... .......... 63%  118M 1s\n",
      " 37300K .......... .......... .......... .......... .......... 63%  121M 1s\n",
      " 37350K .......... .......... .......... .......... .......... 64%  102M 1s\n",
      " 37400K .......... .......... .......... .......... .......... 64%  117M 1s\n",
      " 37450K .......... .......... .......... .......... .......... 64%  102M 1s\n",
      " 37500K .......... .......... .......... .......... .......... 64% 19,7M 1s\n",
      " 37550K .......... .......... .......... .......... .......... 64% 4,47M 1s\n",
      " 37600K .......... .......... .......... .......... .......... 64% 4,16M 1s\n",
      " 37650K .......... .......... .......... .......... .......... 64% 15,4M 1s\n",
      " 37700K .......... .......... .......... .......... .......... 64% 28,1M 1s\n",
      " 37750K .......... .......... .......... .......... .......... 64% 69,3M 1s\n",
      " 37800K .......... .......... .......... .......... .......... 64% 62,3M 1s\n",
      " 37850K .......... .......... .......... .......... .......... 64%  101M 1s\n",
      " 37900K .......... .......... .......... .......... .......... 65% 70,7M 1s\n",
      " 37950K .......... .......... .......... .......... .......... 65%  109M 1s\n",
      " 38000K .......... .......... .......... .......... .......... 65%  127M 1s\n",
      " 38050K .......... .......... .......... .......... .......... 65%  128M 1s\n",
      " 38100K .......... .......... .......... .......... .......... 65%  110M 1s\n",
      " 38150K .......... .......... .......... .......... .......... 65%  127M 1s\n",
      " 38200K .......... .......... .......... .......... .......... 65% 31,7M 1s\n",
      " 38250K .......... .......... .......... .......... .......... 65%  119M 1s\n",
      " 38300K .......... .......... .......... .......... .......... 65%  121M 1s\n",
      " 38350K .......... .......... .......... .......... .......... 65%  101M 1s\n",
      " 38400K .......... .......... .......... .......... .......... 65%  122M 1s\n",
      " 38450K .......... .......... .......... .......... .......... 65%  102M 1s\n",
      " 38500K .......... .......... .......... .......... .......... 66%  121M 1s\n",
      " 38550K .......... .......... .......... .......... .......... 66%  117M 1s\n",
      " 38600K .......... .......... .......... .......... .......... 66%  103M 1s\n",
      " 38650K .......... .......... .......... .......... .......... 66%  120M 1s\n",
      " 38700K .......... .......... .......... .......... .......... 66%  103M 1s\n",
      " 38750K .......... .......... .......... .......... .......... 66%  121M 1s\n",
      " 38800K .......... .......... .......... .......... .......... 66%  117M 1s\n",
      " 38850K .......... .......... .......... .......... .......... 66%  103M 1s\n",
      " 38900K .......... .......... .......... .......... .......... 66%  113M 1s\n",
      " 38950K .......... .......... .......... .......... .......... 66%  108M 1s\n",
      " 39000K .......... .......... .......... .......... .......... 66%  108M 1s\n",
      " 39050K .......... .......... .......... .......... .......... 66%  135M 1s\n",
      " 39100K .......... .......... .......... .......... .......... 67%  102M 1s\n",
      " 39150K .......... .......... .......... .......... .......... 67%  120M 1s\n",
      " 39200K .......... .......... .......... .......... .......... 67%  103M 1s\n",
      " 39250K .......... .......... .......... .......... .......... 67%  117M 1s\n",
      " 39300K .......... .......... .......... .......... .......... 67%  114M 1s\n",
      " 39350K .......... .......... .......... .......... .......... 67%  104M 1s\n",
      " 39400K .......... .......... .......... .......... .......... 67%  119M 1s\n",
      " 39450K .......... .......... .......... .......... .......... 67% 98,7M 1s\n",
      " 39500K .......... .......... .......... .......... .......... 67%  124M 1s\n",
      " 39550K .......... .......... .......... .......... .......... 67%  118M 1s\n",
      " 39600K .......... .......... .......... .......... .......... 67%  103M 1s\n",
      " 39650K .......... .......... .......... .......... .......... 68%  127M 1s\n",
      " 39700K .......... .......... .......... .......... .......... 68%  102M 1s\n",
      " 39750K .......... .......... .......... .......... .......... 68%  122M 1s\n",
      " 39800K .......... .......... .......... .......... .......... 68%  117M 1s\n",
      " 39850K .......... .......... .......... .......... .......... 68%  102M 1s\n",
      " 39900K .......... .......... .......... .......... .......... 68% 29,2M 1s\n",
      " 39950K .......... .......... .......... .......... .......... 68% 87,3M 1s\n",
      " 40000K .......... .......... .......... .......... .......... 68%  143M 1s\n",
      " 40050K .......... .......... .......... .......... .......... 68%  120M 1s\n",
      " 40100K .......... .......... .......... .......... .......... 68%  103M 1s\n",
      " 40150K .......... .......... .......... .......... .......... 68%  117M 1s\n",
      " 40200K .......... .......... .......... .......... .......... 68%  102M 1s\n",
      " 40250K .......... .......... .......... .......... .......... 69%  120M 0s\n",
      " 40300K .......... .......... .......... .......... .......... 69%  121M 0s\n",
      " 40350K .......... .......... .......... .......... .......... 69% 99,6M 0s\n",
      " 40400K .......... .......... .......... .......... .......... 69%  115M 0s\n",
      " 40450K .......... .......... .......... .......... .......... 69%  108M 0s\n",
      " 40500K .......... .......... .......... .......... .......... 69%  104M 0s\n",
      " 40550K .......... .......... .......... .......... .......... 69% 53,6M 0s\n",
      " 40600K .......... .......... .......... .......... .......... 69% 4,80M 0s\n",
      " 40650K .......... .......... .......... .......... .......... 69% 3,78M 0s\n",
      " 40700K .......... .......... .......... .......... .......... 69% 11,4M 0s\n",
      " 40750K .......... .......... .......... .......... .......... 69% 49,0M 0s\n",
      " 40800K .......... .......... .......... .......... .......... 69% 42,8M 0s\n",
      " 40850K .......... .......... .......... .......... .......... 70% 61,2M 0s\n",
      " 40900K .......... .......... .......... .......... .......... 70% 56,4M 0s\n",
      " 40950K .......... .......... .......... .......... .......... 70%  103M 0s\n",
      " 41000K .......... .......... .......... .......... .......... 70%  119M 0s\n",
      " 41050K .......... .......... .......... .......... .......... 70%  123M 0s\n",
      " 41100K .......... .......... .......... .......... .......... 70%  101M 0s\n",
      " 41150K .......... .......... .......... .......... .......... 70%  119M 0s\n",
      " 41200K .......... .......... .......... .......... .......... 70%  102M 0s\n",
      " 41250K .......... .......... .......... .......... .......... 70%  119M 0s\n",
      " 41300K .......... .......... .......... .......... .......... 70%  123M 0s\n",
      " 41350K .......... .......... .......... .......... .......... 70% 68,0M 0s\n",
      " 41400K .......... .......... .......... .......... .......... 71% 60,3M 0s\n",
      " 41450K .......... .......... .......... .......... .......... 71% 75,2M 0s\n",
      " 41500K .......... .......... .......... .......... .......... 71%  121M 0s\n",
      " 41550K .......... .......... .......... .......... .......... 71%  117M 0s\n",
      " 41600K .......... .......... .......... .......... .......... 71%  103M 0s\n",
      " 41650K .......... .......... .......... .......... .......... 71%  123M 0s\n",
      " 41700K .......... .......... .......... .......... .......... 71% 97,5M 0s\n",
      " 41750K .......... .......... .......... .......... .......... 71%  109M 0s\n",
      " 41800K .......... .......... .......... .......... .......... 71%  116M 0s\n",
      " 41850K .......... .......... .......... .......... .......... 71% 99,0M 0s\n",
      " 41900K .......... .......... .......... .......... .......... 71%  153M 0s\n",
      " 41950K .......... .......... .......... .......... .......... 71%  102M 0s\n",
      " 42000K .......... .......... .......... .......... .......... 72% 82,0M 0s\n",
      " 42050K .......... .......... .......... .......... .......... 72%  218M 0s\n",
      " 42100K .......... .......... .......... .......... .......... 72%  103M 0s\n",
      " 42150K .......... .......... .......... .......... .......... 72%  117M 0s\n",
      " 42200K .......... .......... .......... .......... .......... 72%  102M 0s\n",
      " 42250K .......... .......... .......... .......... .......... 72%  122M 0s\n",
      " 42300K .......... .......... .......... .......... .......... 72%  117M 0s\n",
      " 42350K .......... .......... .......... .......... .......... 72%  102M 0s\n",
      " 42400K .......... .......... .......... .......... .......... 72%  123M 0s\n",
      " 42450K .......... .......... .......... .......... .......... 72%  102M 0s\n",
      " 42500K .......... .......... .......... .......... .......... 72%  117M 0s\n",
      " 42550K .......... .......... .......... .......... .......... 72%  121M 0s\n",
      " 42600K .......... .......... .......... .......... .......... 73%  103M 0s\n",
      " 42650K .......... .......... .......... .......... .......... 73%  120M 0s\n",
      " 42700K .......... .......... .......... .......... .......... 73%  102M 0s\n",
      " 42750K .......... .......... .......... .......... .......... 73%  118M 0s\n",
      " 42800K .......... .......... .......... .......... .......... 73%  121M 0s\n",
      " 42850K .......... .......... .......... .......... .......... 73%  102M 0s\n",
      " 42900K .......... .......... .......... .......... .......... 73%  121M 0s\n",
      " 42950K .......... .......... .......... .......... .......... 73%  102M 0s\n",
      " 43000K .......... .......... .......... .......... .......... 73% 37,4M 0s\n",
      " 43050K .......... .......... .......... .......... .......... 73% 32,9M 0s\n",
      " 43100K .......... .......... .......... .......... .......... 73%  107M 0s\n",
      " 43150K .......... .......... .......... .......... .......... 74%  117M 0s\n",
      " 43200K .......... .......... .......... .......... .......... 74%  102M 0s\n",
      " 43250K .......... .......... .......... .......... .......... 74%  121M 0s\n",
      " 43300K .......... .......... .......... .......... .......... 74%  117M 0s\n",
      " 43350K .......... .......... .......... .......... .......... 74%  103M 0s\n",
      " 43400K .......... .......... .......... .......... .......... 74%  120M 0s\n",
      " 43450K .......... .......... .......... .......... .......... 74%  102M 0s\n",
      " 43500K .......... .......... .......... .......... .......... 74%  121M 0s\n",
      " 43550K .......... .......... .......... .......... .......... 74%  117M 0s\n",
      " 43600K .......... .......... .......... .......... .......... 74%  102M 0s\n",
      " 43650K .......... .......... .......... .......... .......... 74% 25,3M 0s\n",
      " 43700K .......... .......... .......... .......... .......... 74% 2,45M 0s\n",
      " 43750K .......... .......... .......... .......... .......... 75% 14,8M 0s\n",
      " 43800K .......... .......... .......... .......... .......... 75% 17,3M 0s\n",
      " 43850K .......... .......... .......... .......... .......... 75% 26,5M 0s\n",
      " 43900K .......... .......... .......... .......... .......... 75%  120M 0s\n",
      " 43950K .......... .......... .......... .......... .......... 75% 53,3M 0s\n",
      " 44000K .......... .......... .......... .......... .......... 75%  119M 0s\n",
      " 44050K .......... .......... .......... .......... .......... 75% 85,2M 0s\n",
      " 44100K .......... .......... .......... .......... .......... 75%  156M 0s\n",
      " 44150K .......... .......... .......... .......... .......... 75%  119M 0s\n",
      " 44200K .......... .......... .......... .......... .......... 75%  103M 0s\n",
      " 44250K .......... .......... .......... .......... .......... 75% 67,9M 0s\n",
      " 44300K .......... .......... .......... .......... .......... 75%  220M 0s\n",
      " 44350K .......... .......... .......... .......... .......... 76%  138M 0s\n",
      " 44400K .......... .......... .......... .......... .......... 76% 90,2M 0s\n",
      " 44450K .......... .......... .......... .......... .......... 76% 52,7M 0s\n",
      " 44500K .......... .......... .......... .......... .......... 76% 57,9M 0s\n",
      " 44550K .......... .......... .......... .......... .......... 76%  120M 0s\n",
      " 44600K .......... .......... .......... .......... .......... 76%  102M 0s\n",
      " 44650K .......... .......... .......... .......... .......... 76%  121M 0s\n",
      " 44700K .......... .......... .......... .......... .......... 76%  102M 0s\n",
      " 44750K .......... .......... .......... .......... .......... 76%  118M 0s\n",
      " 44800K .......... .......... .......... .......... .......... 76%  121M 0s\n",
      " 44850K .......... .......... .......... .......... .......... 76%  103M 0s\n",
      " 44900K .......... .......... .......... .......... .......... 77%  116M 0s\n",
      " 44950K .......... .......... .......... .......... .......... 77%  103M 0s\n",
      " 45000K .......... .......... .......... .......... .......... 77%  120M 0s\n",
      " 45050K .......... .......... .......... .......... .......... 77%  120M 0s\n",
      " 45100K .......... .......... .......... .......... .......... 77%  101M 0s\n",
      " 45150K .......... .......... .......... .......... .......... 77%  119M 0s\n",
      " 45200K .......... .......... .......... .......... .......... 77%  103M 0s\n",
      " 45250K .......... .......... .......... .......... .......... 77%  121M 0s\n",
      " 45300K .......... .......... .......... .......... .......... 77%  117M 0s\n",
      " 45350K .......... .......... .......... .......... .......... 77%  102M 0s\n",
      " 45400K .......... .......... .......... .......... .......... 77%  121M 0s\n",
      " 45450K .......... .......... .......... .......... .......... 77%  103M 0s\n",
      " 45500K .......... .......... .......... .......... .......... 78%  117M 0s\n",
      " 45550K .......... .......... .......... .......... .......... 78%  121M 0s\n",
      " 45600K .......... .......... .......... .......... .......... 78%  102M 0s\n",
      " 45650K .......... .......... .......... .......... .......... 78%  118M 0s\n",
      " 45700K .......... .......... .......... .......... .......... 78%  105M 0s\n",
      " 45750K .......... .......... .......... .......... .......... 78%  116M 0s\n",
      " 45800K .......... .......... .......... .......... .......... 78%  123M 0s\n",
      " 45850K .......... .......... .......... .......... .......... 78%  103M 0s\n",
      " 45900K .......... .......... .......... .......... .......... 78%  117M 0s\n",
      " 45950K .......... .......... .......... .......... .......... 78%  102M 0s\n",
      " 46000K .......... .......... .......... .......... .......... 78%  121M 0s\n",
      " 46050K .......... .......... .......... .......... .......... 78%  118M 0s\n",
      " 46100K .......... .......... .......... .......... .......... 79% 26,6M 0s\n",
      " 46150K .......... .......... .......... .......... .......... 79% 74,1M 0s\n",
      " 46200K .......... .......... .......... .......... .......... 79%  102M 0s\n",
      " 46250K .......... .......... .......... .......... .......... 79%  120M 0s\n",
      " 46300K .......... .......... .......... .......... .......... 79%  118M 0s\n",
      " 46350K .......... .......... .......... .......... .......... 79%  102M 0s\n",
      " 46400K .......... .......... .......... .......... .......... 79%  121M 0s\n",
      " 46450K .......... .......... .......... .......... .......... 79% 91,8M 0s\n",
      " 46500K .......... .......... .......... .......... .......... 79%  117M 0s\n",
      " 46550K .......... .......... .......... .......... .......... 79%  121M 0s\n",
      " 46600K .......... .......... .......... .......... .......... 79%  102M 0s\n",
      " 46650K .......... .......... .......... .......... .......... 80%  121M 0s\n",
      " 46700K .......... .......... .......... .......... .......... 80% 50,6M 0s\n",
      " 46750K .......... .......... .......... .......... .......... 80% 2,83M 0s\n",
      " 46800K .......... .......... .......... .......... .......... 80% 7,26M 0s\n",
      " 46850K .......... .......... .......... .......... .......... 80% 18,6M 0s\n",
      " 46900K .......... .......... .......... .......... .......... 80% 22,4M 0s\n",
      " 46950K .......... .......... .......... .......... .......... 80% 44,4M 0s\n",
      " 47000K .......... .......... .......... .......... .......... 80%  118M 0s\n",
      " 47050K .......... .......... .......... .......... .......... 80%  121M 0s\n",
      " 47100K .......... .......... .......... .......... .......... 80% 91,0M 0s\n",
      " 47150K .......... .......... .......... .......... .......... 80%  120M 0s\n",
      " 47200K .......... .......... .......... .......... .......... 80%  103M 0s\n",
      " 47250K .......... .......... .......... .......... .......... 81%  121M 0s\n",
      " 47300K .......... .......... .......... .......... .......... 81%  117M 0s\n",
      " 47350K .......... .......... .......... .......... .......... 81%  102M 0s\n",
      " 47400K .......... .......... .......... .......... .......... 81%  122M 0s\n",
      " 47450K .......... .......... .......... .......... .......... 81%  102M 0s\n",
      " 47500K .......... .......... .......... .......... .......... 81%  117M 0s\n",
      " 47550K .......... .......... .......... .......... .......... 81% 36,2M 0s\n",
      " 47600K .......... .......... .......... .......... .......... 81%  104M 0s\n",
      " 47650K .......... .......... .......... .......... .......... 81% 94,1M 0s\n",
      " 47700K .......... .......... .......... .......... .......... 81%  102M 0s\n",
      " 47750K .......... .......... .......... .......... .......... 81%  121M 0s\n",
      " 47800K .......... .......... .......... .......... .......... 81%  121M 0s\n",
      " 47850K .......... .......... .......... .......... .......... 82%  102M 0s\n",
      " 47900K .......... .......... .......... .......... .......... 82%  119M 0s\n",
      " 47950K .......... .......... .......... .......... .......... 82%  102M 0s\n",
      " 48000K .......... .......... .......... .......... .......... 82%  121M 0s\n",
      " 48050K .......... .......... .......... .......... .......... 82%  118M 0s\n",
      " 48100K .......... .......... .......... .......... .......... 82%  102M 0s\n",
      " 48150K .......... .......... .......... .......... .......... 82%  120M 0s\n",
      " 48200K .......... .......... .......... .......... .......... 82% 68,9M 0s\n",
      " 48250K .......... .......... .......... .......... .......... 82%  125M 0s\n",
      " 48300K .......... .......... .......... .......... .......... 82%  121M 0s\n",
      " 48350K .......... .......... .......... .......... .......... 82%  102M 0s\n",
      " 48400K .......... .......... .......... .......... .......... 83%  121M 0s\n",
      " 48450K .......... .......... .......... .......... .......... 83% 99,3M 0s\n",
      " 48500K .......... .......... .......... .......... .......... 83%  119M 0s\n",
      " 48550K .......... .......... .......... .......... .......... 83%  123M 0s\n",
      " 48600K .......... .......... .......... .......... .......... 83%  102M 0s\n",
      " 48650K .......... .......... .......... .......... .......... 83%  118M 0s\n",
      " 48700K .......... .......... .......... .......... .......... 83%  102M 0s\n",
      " 48750K .......... .......... .......... .......... .......... 83%  121M 0s\n",
      " 48800K .......... .......... .......... .......... .......... 83%  121M 0s\n",
      " 48850K .......... .......... .......... .......... .......... 83%  102M 0s\n",
      " 48900K .......... .......... .......... .......... .......... 83%  118M 0s\n",
      " 48950K .......... .......... .......... .......... .......... 83%  102M 0s\n",
      " 49000K .......... .......... .......... .......... .......... 84%  121M 0s\n",
      " 49050K .......... .......... .......... .......... .......... 84%  117M 0s\n",
      " 49100K .......... .......... .......... .......... .......... 84%  102M 0s\n",
      " 49150K .......... .......... .......... .......... .......... 84%  121M 0s\n",
      " 49200K .......... .......... .......... .......... .......... 84% 30,0M 0s\n",
      " 49250K .......... .......... .......... .......... .......... 84%  117M 0s\n",
      " 49300K .......... .......... .......... .......... .......... 84% 96,2M 0s\n",
      " 49350K .......... .......... .......... .......... .......... 84%  103M 0s\n",
      " 49400K .......... .......... .......... .......... .......... 84%  121M 0s\n",
      " 49450K .......... .......... .......... .......... .......... 84%  102M 0s\n",
      " 49500K .......... .......... .......... .......... .......... 84%  117M 0s\n",
      " 49550K .......... .......... .......... .......... .......... 84%  120M 0s\n",
      " 49600K .......... .......... .......... .......... .......... 85%  103M 0s\n",
      " 49650K .......... .......... .......... .......... .......... 85%  117M 0s\n",
      " 49700K .......... .......... .......... .......... .......... 85%  102M 0s\n",
      " 49750K .......... .......... .......... .......... .......... 85% 34,7M 0s\n",
      " 49800K .......... .......... .......... .......... .......... 85% 3,78M 0s\n",
      " 49850K .......... .......... .......... .......... .......... 85% 4,51M 0s\n",
      " 49900K .......... .......... .......... .......... .......... 85% 35,4M 0s\n",
      " 49950K .......... .......... .......... .......... .......... 85% 20,2M 0s\n",
      " 50000K .......... .......... .......... .......... .......... 85% 37,0M 0s\n",
      " 50050K .......... .......... .......... .......... .......... 85% 48,5M 0s\n",
      " 50100K .......... .......... .......... .......... .......... 85%  102M 0s\n",
      " 50150K .......... .......... .......... .......... .......... 86%  121M 0s\n",
      " 50200K .......... .......... .......... .......... .......... 86% 67,3M 0s\n",
      " 50250K .......... .......... .......... .......... .......... 86%  130M 0s\n",
      " 50300K .......... .......... .......... .......... .......... 86%  121M 0s\n",
      " 50350K .......... .......... .......... .......... .......... 86% 93,9M 0s\n",
      " 50400K .......... .......... .......... .......... .......... 86%  134M 0s\n",
      " 50450K .......... .......... .......... .......... .......... 86%  102M 0s\n",
      " 50500K .......... .......... .......... .......... .......... 86%  118M 0s\n",
      " 50550K .......... .......... .......... .......... .......... 86%  121M 0s\n",
      " 50600K .......... .......... .......... .......... .......... 86% 92,5M 0s\n",
      " 50650K .......... .......... .......... .......... .......... 86%  136M 0s\n",
      " 50700K .......... .......... .......... .......... .......... 86% 63,9M 0s\n",
      " 50750K .......... .......... .......... .......... .......... 87% 57,4M 0s\n",
      " 50800K .......... .......... .......... .......... .......... 87% 85,7M 0s\n",
      " 50850K .......... .......... .......... .......... .......... 87%  107M 0s\n",
      " 50900K .......... .......... .......... .......... .......... 87%  123M 0s\n",
      " 50950K .......... .......... .......... .......... .......... 87% 95,5M 0s\n",
      " 51000K .......... .......... .......... .......... .......... 87%  127M 0s\n",
      " 51050K .......... .......... .......... .......... .......... 87%  117M 0s\n",
      " 51100K .......... .......... .......... .......... .......... 87%  105M 0s\n",
      " 51150K .......... .......... .......... .......... .......... 87%  120M 0s\n",
      " 51200K .......... .......... .......... .......... .......... 87%  103M 0s\n",
      " 51250K .......... .......... .......... .......... .......... 87%  117M 0s\n",
      " 51300K .......... .......... .......... .......... .......... 87%  122M 0s\n",
      " 51350K .......... .......... .......... .......... .......... 88%  102M 0s\n",
      " 51400K .......... .......... .......... .......... .......... 88%  117M 0s\n",
      " 51450K .......... .......... .......... .......... .......... 88%  102M 0s\n",
      " 51500K .......... .......... .......... .......... .......... 88%  122M 0s\n",
      " 51550K .......... .......... .......... .......... .......... 88%  121M 0s\n",
      " 51600K .......... .......... .......... .......... .......... 88%  103M 0s\n",
      " 51650K .......... .......... .......... .......... .......... 88%  117M 0s\n",
      " 51700K .......... .......... .......... .......... .......... 88%  102M 0s\n",
      " 51750K .......... .......... .......... .......... .......... 88%  121M 0s\n",
      " 51800K .......... .......... .......... .......... .......... 88%  119M 0s\n",
      " 51850K .......... .......... .......... .......... .......... 88%  102M 0s\n",
      " 51900K .......... .......... .......... .......... .......... 89%  121M 0s\n",
      " 51950K .......... .......... .......... .......... .......... 89%  102M 0s\n",
      " 52000K .......... .......... .......... .......... .......... 89%  118M 0s\n",
      " 52050K .......... .......... .......... .......... .......... 89%  121M 0s\n",
      " 52100K .......... .......... .......... .......... .......... 89%  102M 0s\n",
      " 52150K .......... .......... .......... .......... .......... 89%  121M 0s\n",
      " 52200K .......... .......... .......... .......... .......... 89% 88,8M 0s\n",
      " 52250K .......... .......... .......... .......... .......... 89% 50,2M 0s\n",
      " 52300K .......... .......... .......... .......... .......... 89% 59,6M 0s\n",
      " 52350K .......... .......... .......... .......... .......... 89% 99,7M 0s\n",
      " 52400K .......... .......... .......... .......... .......... 89% 72,1M 0s\n",
      " 52450K .......... .......... .......... .......... .......... 89% 52,0M 0s\n",
      " 52500K .......... .......... .......... .......... .......... 90%  121M 0s\n",
      " 52550K .......... .......... .......... .......... .......... 90%  121M 0s\n",
      " 52600K .......... .......... .......... .......... .......... 90%  102M 0s\n",
      " 52650K .......... .......... .......... .......... .......... 90%  118M 0s\n",
      " 52700K .......... .......... .......... .......... .......... 90%  102M 0s\n",
      " 52750K .......... .......... .......... .......... .......... 90%  121M 0s\n",
      " 52800K .......... .......... .......... .......... .......... 90% 22,6M 0s\n",
      " 52850K .......... .......... .......... .......... .......... 90% 7,27M 0s\n",
      " 52900K .......... .......... .......... .......... .......... 90% 3,53M 0s\n",
      " 52950K .......... .......... .......... .......... .......... 90% 12,9M 0s\n",
      " 53000K .......... .......... .......... .......... .......... 90% 46,3M 0s\n",
      " 53050K .......... .......... .......... .......... .......... 90% 35,9M 0s\n",
      " 53100K .......... .......... .......... .......... .......... 91% 39,6M 0s\n",
      " 53150K .......... .......... .......... .......... .......... 91% 62,8M 0s\n",
      " 53200K .......... .......... .......... .......... .......... 91% 96,9M 0s\n",
      " 53250K .......... .......... .......... .......... .......... 91%  114M 0s\n",
      " 53300K .......... .......... .......... .......... .......... 91%  112M 0s\n",
      " 53350K .......... .......... .......... .......... .......... 91%  132M 0s\n",
      " 53400K .......... .......... .......... .......... .......... 91%  104M 0s\n",
      " 53450K .......... .......... .......... .......... .......... 91% 56,3M 0s\n",
      " 53500K .......... .......... .......... .......... .......... 91%  132M 0s\n",
      " 53550K .......... .......... .......... .......... .......... 91% 77,0M 0s\n",
      " 53600K .......... .......... .......... .......... .......... 91%  102M 0s\n",
      " 53650K .......... .......... .......... .......... .......... 92%  121M 0s\n",
      " 53700K .......... .......... .......... .......... .......... 92%  103M 0s\n",
      " 53750K .......... .......... .......... .......... .......... 92%  122M 0s\n",
      " 53800K .......... .......... .......... .......... .......... 92%  117M 0s\n",
      " 53850K .......... .......... .......... .......... .......... 92% 61,8M 0s\n",
      " 53900K .......... .......... .......... .......... .......... 92% 48,4M 0s\n",
      " 53950K .......... .......... .......... .......... .......... 92%  102M 0s\n",
      " 54000K .......... .......... .......... .......... .......... 92%  117M 0s\n",
      " 54050K .......... .......... .......... .......... .......... 92%  121M 0s\n",
      " 54100K .......... .......... .......... .......... .......... 92%  104M 0s\n",
      " 54150K .......... .......... .......... .......... .......... 92%  119M 0s\n",
      " 54200K .......... .......... .......... .......... .......... 92%  103M 0s\n",
      " 54250K .......... .......... .......... .......... .......... 93%  117M 0s\n",
      " 54300K .......... .......... .......... .......... .......... 93%  116M 0s\n",
      " 54350K .......... .......... .......... .......... .......... 93%  106M 0s\n",
      " 54400K .......... .......... .......... .......... .......... 93%  118M 0s\n",
      " 54450K .......... .......... .......... .......... .......... 93%  103M 0s\n",
      " 54500K .......... .......... .......... .......... .......... 93%  121M 0s\n",
      " 54550K .......... .......... .......... .......... .......... 93%  118M 0s\n",
      " 54600K .......... .......... .......... .......... .......... 93%  102M 0s\n",
      " 54650K .......... .......... .......... .......... .......... 93%  121M 0s\n",
      " 54700K .......... .......... .......... .......... .......... 93%  102M 0s\n",
      " 54750K .......... .......... .......... .......... .......... 93%  122M 0s\n",
      " 54800K .......... .......... .......... .......... .......... 93%  117M 0s\n",
      " 54850K .......... .......... .......... .......... .......... 94%  102M 0s\n",
      " 54900K .......... .......... .......... .......... .......... 94%  120M 0s\n",
      " 54950K .......... .......... .......... .......... .......... 94% 92,5M 0s\n",
      " 55000K .......... .......... .......... .......... .......... 94%  134M 0s\n",
      " 55050K .......... .......... .......... .......... .......... 94%  121M 0s\n",
      " 55100K .......... .......... .......... .......... .......... 94%  102M 0s\n",
      " 55150K .......... .......... .......... .......... .......... 94%  117M 0s\n",
      " 55200K .......... .......... .......... .......... .......... 94%  102M 0s\n",
      " 55250K .......... .......... .......... .......... .......... 94%  121M 0s\n",
      " 55300K .......... .......... .......... .......... .......... 94%  121M 0s\n",
      " 55350K .......... .......... .......... .......... .......... 94% 66,2M 0s\n",
      " 55400K .......... .......... .......... .......... .......... 95% 42,7M 0s\n",
      " 55450K .......... .......... .......... .......... .......... 95% 50,9M 0s\n",
      " 55500K .......... .......... .......... .......... .......... 95%  121M 0s\n",
      " 55550K .......... .......... .......... .......... .......... 95%  117M 0s\n",
      " 55600K .......... .......... .......... .......... .......... 95%  102M 0s\n",
      " 55650K .......... .......... .......... .......... .......... 95%  121M 0s\n",
      " 55700K .......... .......... .......... .......... .......... 95%  102M 0s\n",
      " 55750K .......... .......... .......... .......... .......... 95%  117M 0s\n",
      " 55800K .......... .......... .......... .......... .......... 95%  121M 0s\n",
      " 55850K .......... .......... .......... .......... .......... 95% 55,6M 0s\n",
      " 55900K .......... .......... .......... .......... .......... 95% 8,42M 0s\n",
      " 55950K .......... .......... .......... .......... .......... 95% 4,11M 0s\n",
      " 56000K .......... .......... .......... .......... .......... 96% 6,69M 0s\n",
      " 56050K .......... .......... .......... .......... .......... 96% 20,0M 0s\n",
      " 56100K .......... .......... .......... .......... .......... 96% 40,1M 0s\n",
      " 56150K .......... .......... .......... .......... .......... 96% 90,0M 0s\n",
      " 56200K .......... .......... .......... .......... .......... 96% 35,4M 0s\n",
      " 56250K .......... .......... .......... .......... .......... 96%  118M 0s\n",
      " 56300K .......... .......... .......... .......... .......... 96%  121M 0s\n",
      " 56350K .......... .......... .......... .......... .......... 96% 88,6M 0s\n",
      " 56400K .......... .......... .......... .......... .......... 96%  142M 0s\n",
      " 56450K .......... .......... .......... .......... .......... 96%  102M 0s\n",
      " 56500K .......... .......... .......... .......... .......... 96%  121M 0s\n",
      " 56550K .......... .......... .......... .......... .......... 96%  116M 0s\n",
      " 56600K .......... .......... .......... .......... .......... 97%  103M 0s\n",
      " 56650K .......... .......... .......... .......... .......... 97%  122M 0s\n",
      " 56700K .......... .......... .......... .......... .......... 97%  103M 0s\n",
      " 56750K .......... .......... .......... .......... .......... 97%  116M 0s\n",
      " 56800K .......... .......... .......... .......... .......... 97%  121M 0s\n",
      " 56850K .......... .......... .......... .......... .......... 97%  104M 0s\n",
      " 56900K .......... .......... .......... .......... .......... 97%  121M 0s\n",
      " 56950K .......... .......... .......... .......... .......... 97% 53,6M 0s\n",
      " 57000K .......... .......... .......... .......... .......... 97% 57,6M 0s\n",
      " 57050K .......... .......... .......... .......... .......... 97%  121M 0s\n",
      " 57100K .......... .......... .......... .......... .......... 97%  102M 0s\n",
      " 57150K .......... .......... .......... .......... .......... 98%  105M 0s\n",
      " 57200K .......... .......... .......... .......... .......... 98% 60,2M 0s\n",
      " 57250K .......... .......... .......... .......... .......... 98%  121M 0s\n",
      " 57300K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 57350K .......... .......... .......... .......... .......... 98%  102M 0s\n",
      " 57400K .......... .......... .......... .......... .......... 98%  121M 0s\n",
      " 57450K .......... .......... .......... .......... .......... 98%  102M 0s\n",
      " 57500K .......... .......... .......... .......... .......... 98%  121M 0s\n",
      " 57550K .......... .......... .......... .......... .......... 98%  101M 0s\n",
      " 57600K .......... .......... .......... .......... .......... 98%  109M 0s\n",
      " 57650K .......... .......... .......... .......... .......... 98%  133M 0s\n",
      " 57700K .......... .......... .......... .......... .......... 98%  102M 0s\n",
      " 57750K .......... .......... .......... .......... .......... 99%  117M 0s\n",
      " 57800K .......... .......... .......... .......... .......... 99%  122M 0s\n",
      " 57850K .......... .......... .......... .......... .......... 99%  102M 0s\n",
      " 57900K .......... .......... .......... .......... .......... 99%  122M 0s\n",
      " 57950K .......... .......... .......... .......... .......... 99%  102M 0s\n",
      " 58000K .......... .......... .......... .......... .......... 99%  117M 0s\n",
      " 58050K .......... .......... .......... .......... .......... 99%  122M 0s\n",
      " 58100K .......... .......... .......... .......... .......... 99%  102M 0s\n",
      " 58150K .......... .......... .......... .......... .......... 99%  118M 0s\n",
      " 58200K .......... .......... .......... .......... .......... 99%  103M 0s\n",
      " 58250K .......... .......... .......... .......... .......... 99%  120M 0s\n",
      " 58300K .......... .......... .......... .......... .......... 99%  117M 0s\n",
      " 58350K .......... .                                          100%  147M=1,5s\n",
      "\n",
      "2023-08-18 14:55:48 (39,2 MB/s) - ‘data/reference/genome.fa.gz’ guardado [59761837/59761837]\n",
      "\n",
      "[Fri Aug 18 14:55:49 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T145546.099442.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Creating conda environment code/environments/Greference_tools.yml...\n",
      "Downloading and installing remote packages.\n",
      "Environment for /home/juancarlos/Escritorio/TFM/code/environments/Greference_tools.yml created (location: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_)\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 14:56:49 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR696683_chr3_1.fastq.gz, data/raw/ERR696683_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR696683_chr3_1_fastqc.html, results/fastqc_result/ERR696683_chr3_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR696683_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 14:57:07 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 14:57:07 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR696683_chr3_1.fastq.gz, data/raw/ERR696683_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR696683_chr3_1_fastp.fastq.gz, data/processed/ERR696683_chr3_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR696683_chr3_1_fastp.fastq.gz, data/processed/ERR696683_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2070566\n",
      "total bases: 156530982\n",
      "Q20 bases: 155429955(99.2966%)\n",
      "Q30 bases: 151203846(96.5968%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2070566\n",
      "total bases: 156128924\n",
      "Q20 bases: 154231757(98.7849%)\n",
      "Q30 bases: 148068194(94.8371%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1294164\n",
      "total bases: 98291522\n",
      "Q20 bases: 97907734(99.6095%)\n",
      "Q30 bases: 96216656(97.8891%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1294164\n",
      "total bases: 98253340\n",
      "Q20 bases: 97583822(99.3186%)\n",
      "Q30 bases: 95093850(96.7843%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2588328\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1552804\n",
      "reads with adapter trimmed: 2016\n",
      "bases trimmed due to adapters: 53376\n",
      "\n",
      "Duplication rate: 5.84855%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR696683_chr3_1.fastq.gz -I data/raw/ERR696683_chr3_2.fastq.gz -o data/processed/ERR696683_chr3_1_fastp.fastq.gz -O data/processed/ERR696683_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Fri Aug 18 14:57:14 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 14:57:14 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR696683_chr3_1_fastp.fastq.gz, data/processed/ERR696683_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR696683_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR696683_chr3_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR696683_chr3_1_fastp.fastq.gz, data/processed/ERR696683_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 14:57:25 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T145550.029356.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 14:57:26 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753368_chr3_1.fastq.gz, data/raw/ERR753368_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753368_chr3_1_fastqc.html, results/fastqc_result/ERR753368_chr3_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753368_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753368_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 14:57:43 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 14:57:43 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753368_chr3_1.fastq.gz, data/raw/ERR753368_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753368_chr3_1_fastp.fastq.gz, data/processed/ERR753368_chr3_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753368_chr3_1_fastp.fastq.gz, data/processed/ERR753368_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2166924\n",
      "total bases: 164537592\n",
      "Q20 bases: 163892626(99.608%)\n",
      "Q30 bases: 159125497(96.7107%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2166924\n",
      "total bases: 164375778\n",
      "Q20 bases: 162943687(99.1288%)\n",
      "Q30 bases: 155573837(94.6452%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1870811\n",
      "total bases: 142174357\n",
      "Q20 bases: 141930007(99.8281%)\n",
      "Q30 bases: 139573591(98.1707%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1870811\n",
      "total bases: 142162640\n",
      "Q20 bases: 141472380(99.5145%)\n",
      "Q30 bases: 137298960(96.5788%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3741622\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 592226\n",
      "reads with adapter trimmed: 1044\n",
      "bases trimmed due to adapters: 26850\n",
      "\n",
      "Duplication rate: 2.05843%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753368_chr3_1.fastq.gz -I data/raw/ERR753368_chr3_2.fastq.gz -o data/processed/ERR753368_chr3_1_fastp.fastq.gz -O data/processed/ERR753368_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 7 seconds\n",
      "[Fri Aug 18 14:57:51 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 14:57:51 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753368_chr3_1_fastp.fastq.gz, data/processed/ERR753368_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753368_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753368_chr3_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753368_chr3_1_fastp.fastq.gz, data/processed/ERR753368_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 14:58:06 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T145725.418841.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 14:58:07 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753369_chr3_1.fastq.gz, data/raw/ERR753369_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753369_chr3_1_fastqc.html, results/fastqc_result/ERR753369_chr3_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753369_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753369_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 14:58:26 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 14:58:26 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753369_chr3_1.fastq.gz, data/raw/ERR753369_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753369_chr3_1_fastp.fastq.gz, data/processed/ERR753369_chr3_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753369_chr3_1_fastp.fastq.gz, data/processed/ERR753369_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2321627\n",
      "total bases: 176276293\n",
      "Q20 bases: 175555474(99.5911%)\n",
      "Q30 bases: 170311472(96.6162%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2321627\n",
      "total bases: 176111158\n",
      "Q20 bases: 174567395(99.1234%)\n",
      "Q30 bases: 166614064(94.6073%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 2002164\n",
      "total bases: 152156256\n",
      "Q20 bases: 151876395(99.8161%)\n",
      "Q30 bases: 149254086(98.0926%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 2002164\n",
      "total bases: 152144233\n",
      "Q20 bases: 151399863(99.5107%)\n",
      "Q30 bases: 146897284(96.5513%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 4004328\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 638926\n",
      "reads with adapter trimmed: 1028\n",
      "bases trimmed due to adapters: 24939\n",
      "\n",
      "Duplication rate: 2.38335%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753369_chr3_1.fastq.gz -I data/raw/ERR753369_chr3_2.fastq.gz -o data/processed/ERR753369_chr3_1_fastp.fastq.gz -O data/processed/ERR753369_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 7 seconds\n",
      "[Fri Aug 18 14:58:34 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 14:58:34 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753369_chr3_1_fastp.fastq.gz, data/processed/ERR753369_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753369_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753369_chr3_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753369_chr3_1_fastp.fastq.gz, data/processed/ERR753369_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 14:58:49 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T145806.225041.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 14:58:50 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753370_chr3_1.fastq.gz, data/raw/ERR753370_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753370_chr3_1_fastp.fastq.gz, data/processed/ERR753370_chr3_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753370_chr3_1_fastp.fastq.gz, data/processed/ERR753370_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2951523\n",
      "total bases: 297130164\n",
      "Q20 bases: 295118642(99.323%)\n",
      "Q30 bases: 286675921(96.4816%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2951523\n",
      "total bases: 297056363\n",
      "Q20 bases: 294260921(99.059%)\n",
      "Q30 bases: 283246706(95.3512%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 2950493\n",
      "total bases: 295883003\n",
      "Q20 bases: 293987676(99.3594%)\n",
      "Q30 bases: 285943525(96.6407%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 2950493\n",
      "total bases: 294939988\n",
      "Q20 bases: 292398511(99.1383%)\n",
      "Q30 bases: 282073643(95.6376%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 5900986\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 2060\n",
      "reads with adapter trimmed: 17930\n",
      "bases trimmed due to adapters: 173930\n",
      "\n",
      "Duplication rate: 4.38761%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 158\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753370_chr3_1.fastq.gz -I data/raw/ERR753370_chr3_2.fastq.gz -o data/processed/ERR753370_chr3_1_fastp.fastq.gz -O data/processed/ERR753370_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 15 seconds\n",
      "[Fri Aug 18 14:59:06 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 14:59:06 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753370_chr3_1_fastp.fastq.gz, data/processed/ERR753370_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753370_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753370_chr3_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753370_chr3_1_fastp.fastq.gz, data/processed/ERR753370_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 14:59:33 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 14:59:33 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753370_chr3_1.fastq.gz, data/raw/ERR753370_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753370_chr3_1_fastqc.html, results/fastqc_result/ERR753370_chr3_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753370_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753370_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 15:00:00 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T145849.580412.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:00:01 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753371_chr3_1.fastq.gz, data/raw/ERR753371_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753371_chr3_1_fastqc.html, results/fastqc_result/ERR753371_chr3_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753371_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753371_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 15:00:17 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:00:17 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753371_chr3_1.fastq.gz, data/raw/ERR753371_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753371_chr3_1_fastp.fastq.gz, data/processed/ERR753371_chr3_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753371_chr3_2_fastp.fastq.gz, data/processed/ERR753371_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1938056\n",
      "total bases: 146044753\n",
      "Q20 bases: 144732927(99.1018%)\n",
      "Q30 bases: 139492968(95.5139%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1938056\n",
      "total bases: 145676659\n",
      "Q20 bases: 143948860(98.8139%)\n",
      "Q30 bases: 138931875(95.37%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1007584\n",
      "total bases: 76483588\n",
      "Q20 bases: 76112999(99.5155%)\n",
      "Q30 bases: 74225186(97.0472%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1007584\n",
      "total bases: 76440869\n",
      "Q20 bases: 75910140(99.3057%)\n",
      "Q30 bases: 74249432(97.1332%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2015168\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1860944\n",
      "reads with adapter trimmed: 1798\n",
      "bases trimmed due to adapters: 42747\n",
      "\n",
      "Duplication rate: 22.2728%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753371_chr3_1.fastq.gz -I data/raw/ERR753371_chr3_2.fastq.gz -o data/processed/ERR753371_chr3_1_fastp.fastq.gz -O data/processed/ERR753371_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Fri Aug 18 15:00:24 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:00:24 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753371_chr3_1_fastp.fastq.gz, data/processed/ERR753371_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753371_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753371_chr3_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753371_chr3_2_fastp.fastq.gz, data/processed/ERR753371_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 15:00:32 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T150000.451268.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:00:34 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753372_chr3_1.fastq.gz, data/raw/ERR753372_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753372_chr3_1_fastqc.html, results/fastqc_result/ERR753372_chr3_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753372_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753372_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 15:00:53 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:00:53 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753372_chr3_1.fastq.gz, data/raw/ERR753372_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753372_chr3_1_fastp.fastq.gz, data/processed/ERR753372_chr3_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753372_chr3_2_fastp.fastq.gz, data/processed/ERR753372_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2299130\n",
      "total bases: 174570404\n",
      "Q20 bases: 173883945(99.6068%)\n",
      "Q30 bases: 168720211(96.6488%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2299130\n",
      "total bases: 174401358\n",
      "Q20 bases: 172878044(99.1265%)\n",
      "Q30 bases: 165079194(94.6548%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1978429\n",
      "total bases: 150352788\n",
      "Q20 bases: 150093042(99.8272%)\n",
      "Q30 bases: 147551682(98.137%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1978429\n",
      "total bases: 150340351\n",
      "Q20 bases: 149614171(99.517%)\n",
      "Q30 bases: 145248530(96.6131%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3956858\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 641402\n",
      "reads with adapter trimmed: 1252\n",
      "bases trimmed due to adapters: 32637\n",
      "\n",
      "Duplication rate: 1.99609%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753372_chr3_1.fastq.gz -I data/raw/ERR753372_chr3_2.fastq.gz -o data/processed/ERR753372_chr3_1_fastp.fastq.gz -O data/processed/ERR753372_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 8 seconds\n",
      "[Fri Aug 18 15:01:01 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:01:01 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753372_chr3_1_fastp.fastq.gz, data/processed/ERR753372_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753372_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753372_chr3_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753372_chr3_2_fastp.fastq.gz, data/processed/ERR753372_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 15:01:16 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T150033.058684.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:01:17 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753373_chr3_1.fastq.gz, data/raw/ERR753373_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753373_chr3_1_fastqc.html, results/fastqc_result/ERR753373_chr3_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753373_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753373_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 15:01:36 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:01:36 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753373_chr3_1.fastq.gz, data/raw/ERR753373_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753373_chr3_1_fastp.fastq.gz, data/processed/ERR753373_chr3_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753373_chr3_2_fastp.fastq.gz, data/processed/ERR753373_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1968465\n",
      "total bases: 188482891\n",
      "Q20 bases: 186996377(99.2113%)\n",
      "Q30 bases: 174397046(92.5267%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1968465\n",
      "total bases: 184512055\n",
      "Q20 bases: 182525877(98.9236%)\n",
      "Q30 bases: 170849210(92.5951%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1922198\n",
      "total bases: 181199250\n",
      "Q20 bases: 180156389(99.4245%)\n",
      "Q30 bases: 169961990(93.7984%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1922198\n",
      "total bases: 175114301\n",
      "Q20 bases: 174290716(99.5297%)\n",
      "Q30 bases: 165407202(94.4567%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3844396\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 92534\n",
      "reads with adapter trimmed: 1350\n",
      "bases trimmed due to adapters: 40609\n",
      "\n",
      "Duplication rate: 1.91361%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 153\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753373_chr3_1.fastq.gz -I data/raw/ERR753373_chr3_2.fastq.gz -o data/processed/ERR753373_chr3_1_fastp.fastq.gz -O data/processed/ERR753373_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 8 seconds\n",
      "[Fri Aug 18 15:01:45 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:01:45 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753373_chr3_1_fastp.fastq.gz, data/processed/ERR753373_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753373_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753373_chr3_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753373_chr3_2_fastp.fastq.gz, data/processed/ERR753373_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 15:02:02 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T150116.570739.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:02:03 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753374_chr3_1.fastq.gz, data/raw/ERR753374_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753374_chr3_1_fastqc.html, results/fastqc_result/ERR753374_chr3_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753374_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753374_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 15:02:16 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:02:16 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753374_chr3_1.fastq.gz, data/raw/ERR753374_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753374_chr3_1_fastp.fastq.gz, data/processed/ERR753374_chr3_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753374_chr3_2_fastp.fastq.gz, data/processed/ERR753374_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1162403\n",
      "total bases: 111198121\n",
      "Q20 bases: 110160620(99.067%)\n",
      "Q30 bases: 100749607(90.6037%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1162403\n",
      "total bases: 108805108\n",
      "Q20 bases: 107679458(98.9654%)\n",
      "Q30 bases: 99151878(91.128%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1126834\n",
      "total bases: 105363400\n",
      "Q20 bases: 104622908(99.2972%)\n",
      "Q30 bases: 97132511(92.1881%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1126834\n",
      "total bases: 102618294\n",
      "Q20 bases: 102036510(99.4331%)\n",
      "Q30 bases: 95498534(93.0619%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2253668\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 71138\n",
      "reads with adapter trimmed: 676\n",
      "bases trimmed due to adapters: 22682\n",
      "\n",
      "Duplication rate: 5.71367%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 153\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753374_chr3_1.fastq.gz -I data/raw/ERR753374_chr3_2.fastq.gz -o data/processed/ERR753374_chr3_1_fastp.fastq.gz -O data/processed/ERR753374_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Fri Aug 18 15:02:22 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:02:22 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753374_chr3_1_fastp.fastq.gz, data/processed/ERR753374_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753374_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753374_chr3_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753374_chr3_2_fastp.fastq.gz, data/processed/ERR753374_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 15:02:33 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T150202.783357.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:02:34 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753375_chr3_1.fastq.gz, data/raw/ERR753375_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753375_chr3_1_fastp.fastq.gz, data/processed/ERR753375_chr3_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753375_chr3_2_fastp.fastq.gz, data/processed/ERR753375_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1282179\n",
      "total bases: 96858250\n",
      "Q20 bases: 96265562(99.3881%)\n",
      "Q30 bases: 94113609(97.1663%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1282179\n",
      "total bases: 96835483\n",
      "Q20 bases: 95589371(98.7132%)\n",
      "Q30 bases: 93289559(96.3382%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 841057\n",
      "total bases: 63862131\n",
      "Q20 bases: 63654927(99.6755%)\n",
      "Q30 bases: 62708243(98.1932%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 841057\n",
      "total bases: 63872391\n",
      "Q20 bases: 63259861(99.041%)\n",
      "Q30 bases: 62279464(97.5061%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1682114\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 882244\n",
      "reads with adapter trimmed: 996\n",
      "bases trimmed due to adapters: 27146\n",
      "\n",
      "Duplication rate: 1.7172%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753375_chr3_1.fastq.gz -I data/raw/ERR753375_chr3_2.fastq.gz -o data/processed/ERR753375_chr3_1_fastp.fastq.gz -O data/processed/ERR753375_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Fri Aug 18 15:02:39 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:02:39 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753375_chr3_1.fastq.gz, data/raw/ERR753375_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753375_chr3_1_fastqc.html, results/fastqc_result/ERR753375_chr3_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753375_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753375_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 15:02:50 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:02:50 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753375_chr3_1_fastp.fastq.gz, data/processed/ERR753375_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753375_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753375_chr3_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753375_chr3_2_fastp.fastq.gz, data/processed/ERR753375_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 15:02:58 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T150233.365030.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:03:00 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753376_chr3_1.fastq.gz, data/raw/ERR753376_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753376_chr3_1_fastp.fastq.gz, data/processed/ERR753376_chr3_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753376_chr3_1_fastp.fastq.gz, data/processed/ERR753376_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1665106\n",
      "total bases: 125882357\n",
      "Q20 bases: 125020665(99.3155%)\n",
      "Q30 bases: 121633177(96.6245%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1665106\n",
      "total bases: 125542593\n",
      "Q20 bases: 124005519(98.7757%)\n",
      "Q30 bases: 119018691(94.8034%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1037289\n",
      "total bases: 78781993\n",
      "Q20 bases: 78486134(99.6245%)\n",
      "Q30 bases: 77145479(97.9227%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1037289\n",
      "total bases: 78752401\n",
      "Q20 bases: 78213072(99.3152%)\n",
      "Q30 bases: 76205525(96.766%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2074578\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1255634\n",
      "reads with adapter trimmed: 1176\n",
      "bases trimmed due to adapters: 29546\n",
      "\n",
      "Duplication rate: 9.11476%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753376_chr3_1.fastq.gz -I data/raw/ERR753376_chr3_2.fastq.gz -o data/processed/ERR753376_chr3_1_fastp.fastq.gz -O data/processed/ERR753376_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Fri Aug 18 15:03:06 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:03:06 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753376_chr3_1.fastq.gz, data/raw/ERR753376_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753376_chr3_1_fastqc.html, results/fastqc_result/ERR753376_chr3_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753376_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753376_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 15:03:21 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:03:21 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753376_chr3_1_fastp.fastq.gz, data/processed/ERR753376_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753376_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753376_chr3_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753376_chr3_1_fastp.fastq.gz, data/processed/ERR753376_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 15:03:30 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T150259.152327.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:03:31 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753377_chr3_1.fastq.gz, data/raw/ERR753377_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753377_chr3_1_fastqc.html, results/fastqc_result/ERR753377_chr3_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753377_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753377_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 15:03:47 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:03:47 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753377_chr3_1.fastq.gz, data/raw/ERR753377_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753377_chr3_1_fastp.fastq.gz, data/processed/ERR753377_chr3_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753377_chr3_2_fastp.fastq.gz, data/processed/ERR753377_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1728950\n",
      "total bases: 130722103\n",
      "Q20 bases: 129808577(99.3012%)\n",
      "Q30 bases: 126267707(96.5925%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1728950\n",
      "total bases: 130414142\n",
      "Q20 bases: 128895041(98.8352%)\n",
      "Q30 bases: 123933005(95.0303%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1099156\n",
      "total bases: 83481297\n",
      "Q20 bases: 83162064(99.6176%)\n",
      "Q30 bases: 81728416(97.9003%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1099156\n",
      "total bases: 83450542\n",
      "Q20 bases: 82907906(99.3498%)\n",
      "Q30 bases: 80876920(96.916%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2198312\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1259588\n",
      "reads with adapter trimmed: 2030\n",
      "bases trimmed due to adapters: 50654\n",
      "\n",
      "Duplication rate: 9.10903%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 119\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753377_chr3_1.fastq.gz -I data/raw/ERR753377_chr3_2.fastq.gz -o data/processed/ERR753377_chr3_1_fastp.fastq.gz -O data/processed/ERR753377_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Fri Aug 18 15:03:53 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:03:53 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753377_chr3_1_fastp.fastq.gz, data/processed/ERR753377_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753377_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753377_chr3_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753377_chr3_2_fastp.fastq.gz, data/processed/ERR753377_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 15:04:04 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T150330.715666.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:04:05 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753378_chr3_1.fastq.gz, data/raw/ERR753378_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753378_chr3_1_fastp.fastq.gz, data/processed/ERR753378_chr3_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753378_chr3_1_fastp.fastq.gz, data/processed/ERR753378_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1978176\n",
      "total bases: 150201904\n",
      "Q20 bases: 149614211(99.6087%)\n",
      "Q30 bases: 145205101(96.6733%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1978176\n",
      "total bases: 150054321\n",
      "Q20 bases: 148742134(99.1255%)\n",
      "Q30 bases: 142005411(94.636%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1708379\n",
      "total bases: 129830167\n",
      "Q20 bases: 129604043(99.8258%)\n",
      "Q30 bases: 127403551(98.1309%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1708379\n",
      "total bases: 129819431\n",
      "Q20 bases: 129183589(99.5102%)\n",
      "Q30 bases: 125347189(96.555%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3416758\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 539594\n",
      "reads with adapter trimmed: 934\n",
      "bases trimmed due to adapters: 22437\n",
      "\n",
      "Duplication rate: 1.97397%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753378_chr3_1.fastq.gz -I data/raw/ERR753378_chr3_2.fastq.gz -o data/processed/ERR753378_chr3_1_fastp.fastq.gz -O data/processed/ERR753378_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Fri Aug 18 15:04:12 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:04:12 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753378_chr3_1.fastq.gz, data/raw/ERR753378_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753378_chr3_1_fastqc.html, results/fastqc_result/ERR753378_chr3_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753378_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753378_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 15:04:28 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:04:28 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753378_chr3_1_fastp.fastq.gz, data/processed/ERR753378_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753378_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753378_chr3_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753378_chr3_1_fastp.fastq.gz, data/processed/ERR753378_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 15:04:43 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T150404.668018.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:04:44 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR696683_chr3_1_fastp.fastq.gz, data/processed/ERR696683_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR696683_chr3.sam, results/mapped_reads/ERR696683_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR696683_chr3_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR696683_chr3_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.85 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 78.68 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.63 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.43 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 45.40 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 131.650 sec; CPU: 125.986 sec\n",
      "[Fri Aug 18 15:08:50 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:08:50 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR696683_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr3.bam, results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR696683_chr3.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR696683_chr3_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 15:09:41 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:09:41 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR696683_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "15:09:42.639 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 15:09:42 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR696683_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR696683_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 15:09:42 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 15:09:42\tMarkDuplicates\tStart of doWork freeMemory: 528784192; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:09:42\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 15:09:42\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam\n",
      "INFO\t2023-08-18 15:09:44\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:69,122,304\n",
      "INFO\t2023-08-18 15:09:44\tMarkDuplicates\tTracking 808 as yet unmatched pairs. 808 records in RAM.\n",
      "INFO\t2023-08-18 15:09:45\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 3:149,750,765\n",
      "INFO\t2023-08-18 15:09:45\tMarkDuplicates\tTracking 675 as yet unmatched pairs. 675 records in RAM.\n",
      "INFO\t2023-08-18 15:09:46\tMarkDuplicates\tRead 2626393 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 15:09:46\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 577429864; totalMemory: 929038336; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:09:46\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 15:09:46\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 15:09:47\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 15:09:47\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 15:09:47\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 919549912; totalMemory: 1465909248; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:09:47\tMarkDuplicates\tMarking 245188 records as duplicates.\n",
      "INFO\t2023-08-18 15:09:47\tMarkDuplicates\tFound 114 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam\n",
      "INFO\t2023-08-18 15:09:47\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 15:10:04\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 15:10:04\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 15:10:04\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 15:10:04\tMarkDuplicates\tBefore output close freeMemory: 528081992; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:10:04\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 15:10:04\tMarkDuplicates\tAfter output close freeMemory: 528081992; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 15:10:04 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.37 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 15:10:06 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T150443.815368.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:10:08 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753368_chr3_1_fastp.fastq.gz, data/processed/ERR753368_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753368_chr3.sam, results/mapped_reads/ERR753368_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753368_chr3_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753368_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.70 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 80.99 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.65 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.42 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 45.65 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 135.614 sec; CPU: 128.424 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 15:15:25 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:15:25 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753368_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr3.bam, results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753368_chr3.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753368_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 15:16:35 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:16:35 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753368_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753368_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "15:16:36.328 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 15:16:36 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753368_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753368_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 15:16:36 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 15:16:36\tMarkDuplicates\tStart of doWork freeMemory: 528785632; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:16:36\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 15:16:36\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam\n",
      "INFO\t2023-08-18 15:16:38\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:48,469,722\n",
      "INFO\t2023-08-18 15:16:38\tMarkDuplicates\tTracking 476 as yet unmatched pairs. 476 records in RAM.\n",
      "INFO\t2023-08-18 15:16:39\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 3:108,377,297\n",
      "INFO\t2023-08-18 15:16:39\tMarkDuplicates\tTracking 765 as yet unmatched pairs. 765 records in RAM.\n",
      "INFO\t2023-08-18 15:16:40\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 3:152,590,728\n",
      "INFO\t2023-08-18 15:16:40\tMarkDuplicates\tTracking 382 as yet unmatched pairs. 382 records in RAM.\n",
      "INFO\t2023-08-18 15:16:42\tMarkDuplicates\tRead 3764483 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 15:16:42\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 383444120; totalMemory: 859832320; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:16:42\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 15:16:42\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 15:16:42\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 15:16:43\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 15:16:43\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 850346320; totalMemory: 1396703232; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:16:43\tMarkDuplicates\tMarking 92309 records as duplicates.\n",
      "INFO\t2023-08-18 15:16:43\tMarkDuplicates\tFound 612 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam\n",
      "INFO\t2023-08-18 15:16:43\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 15:17:06\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 15:17:06\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 15:17:06\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 15:17:06\tMarkDuplicates\tBefore output close freeMemory: 528083936; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:17:06\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 15:17:06\tMarkDuplicates\tAfter output close freeMemory: 528083936; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 15:17:06 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.51 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 15:17:09 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T151007.181413.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:17:10 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753369_chr3_1_fastp.fastq.gz, data/processed/ERR753369_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753369_chr3.sam, results/mapped_reads/ERR753369_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753369_chr3_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR753369_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.69 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 75.44 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.59 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.42 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 42.45 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 126.270 sec; CPU: 119.587 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 15:22:23 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:22:23 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753369_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr3.bam, results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753369_chr3.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753369_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 15:23:35 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:23:35 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753369_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753369_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "15:23:36.442 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 15:23:36 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753369_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753369_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 15:23:36 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 15:23:36\tMarkDuplicates\tStart of doWork freeMemory: 528779664; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:23:36\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 15:23:36\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam\n",
      "INFO\t2023-08-18 15:23:38\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:47,407,712\n",
      "INFO\t2023-08-18 15:23:38\tMarkDuplicates\tTracking 449 as yet unmatched pairs. 449 records in RAM.\n",
      "INFO\t2023-08-18 15:23:39\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 3:100,386,910\n",
      "INFO\t2023-08-18 15:23:39\tMarkDuplicates\tTracking 644 as yet unmatched pairs. 644 records in RAM.\n",
      "INFO\t2023-08-18 15:23:41\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 3:142,332,832\n",
      "INFO\t2023-08-18 15:23:41\tMarkDuplicates\tTracking 551 as yet unmatched pairs. 551 records in RAM.\n",
      "INFO\t2023-08-18 15:23:42\tMarkDuplicates\tRead     4,000,000 records.  Elapsed time: 00:00:05s.  Time for last 1,000,000:    1s.  Last read position: 3:197,557,367\n",
      "INFO\t2023-08-18 15:23:42\tMarkDuplicates\tTracking 15 as yet unmatched pairs. 15 records in RAM.\n",
      "INFO\t2023-08-18 15:23:42\tMarkDuplicates\tRead 4026673 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 15:23:42\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 722733744; totalMemory: 1227882496; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:23:42\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 15:23:42\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 15:23:43\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 15:23:43\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 15:23:43\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1243557512; totalMemory: 1789919232; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:23:43\tMarkDuplicates\tMarking 115047 records as duplicates.\n",
      "INFO\t2023-08-18 15:23:43\tMarkDuplicates\tFound 621 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam\n",
      "INFO\t2023-08-18 15:23:43\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 15:24:07\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 15:24:07\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 15:24:07\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 15:24:07\tMarkDuplicates\tBefore output close freeMemory: 528077304; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:24:07\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 15:24:07\tMarkDuplicates\tAfter output close freeMemory: 528077304; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 15:24:07 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.53 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 15:24:10 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T151709.720429.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:24:13 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753370_chr3_1_fastp.fastq.gz, data/processed/ERR753370_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753370_chr3.sam, results/mapped_reads/ERR753370_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753370_chr3_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753370_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.67 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 80.42 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.61 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.43 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 45.31 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 133.856 sec; CPU: 127.434 sec\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 15:32:02 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:32:02 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753370_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr3.bam, results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753370_chr3.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753370_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 15:34:31 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:34:31 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753370_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753370_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "15:34:31.906 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 15:34:31 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753370_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753370_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 15:34:31 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 15:34:31\tMarkDuplicates\tStart of doWork freeMemory: 528790392; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:34:31\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 15:34:31\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam\n",
      "INFO\t2023-08-18 15:34:33\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:37,998,222\n",
      "INFO\t2023-08-18 15:34:33\tMarkDuplicates\tTracking 992 as yet unmatched pairs. 992 records in RAM.\n",
      "INFO\t2023-08-18 15:34:35\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 3:56,434,430\n",
      "INFO\t2023-08-18 15:34:35\tMarkDuplicates\tTracking 1784 as yet unmatched pairs. 1784 records in RAM.\n",
      "INFO\t2023-08-18 15:34:36\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 3:101,652,466\n",
      "INFO\t2023-08-18 15:34:36\tMarkDuplicates\tTracking 1978 as yet unmatched pairs. 1978 records in RAM.\n",
      "INFO\t2023-08-18 15:34:38\tMarkDuplicates\tRead     4,000,000 records.  Elapsed time: 00:00:06s.  Time for last 1,000,000:    1s.  Last read position: 3:130,736,077\n",
      "INFO\t2023-08-18 15:34:38\tMarkDuplicates\tTracking 1536 as yet unmatched pairs. 1536 records in RAM.\n",
      "INFO\t2023-08-18 15:34:40\tMarkDuplicates\tRead     5,000,000 records.  Elapsed time: 00:00:08s.  Time for last 1,000,000:    1s.  Last read position: 3:167,328,091\n",
      "INFO\t2023-08-18 15:34:40\tMarkDuplicates\tTracking 1026 as yet unmatched pairs. 1026 records in RAM.\n",
      "INFO\t2023-08-18 15:34:41\tMarkDuplicates\tRead     6,000,000 records.  Elapsed time: 00:00:09s.  Time for last 1,000,000:    1s.  Last read position: 3:198,076,827\n",
      "INFO\t2023-08-18 15:34:41\tMarkDuplicates\tTracking 183 as yet unmatched pairs. 183 records in RAM.\n",
      "INFO\t2023-08-18 15:34:41\tMarkDuplicates\tRead 6007564 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 15:34:41\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 489551056; totalMemory: 1199570944; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:34:41\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 15:34:41\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 15:34:42\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 15:34:43\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 15:34:43\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1190088536; totalMemory: 1736441856; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:34:43\tMarkDuplicates\tMarking 429693 records as duplicates.\n",
      "INFO\t2023-08-18 15:34:43\tMarkDuplicates\tFound 269 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam\n",
      "INFO\t2023-08-18 15:34:43\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 15:35:28\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 15:35:28\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 15:35:28\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 15:35:28\tMarkDuplicates\tBefore output close freeMemory: 528087848; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:35:28\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 15:35:28\tMarkDuplicates\tAfter output close freeMemory: 528087848; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 15:35:28 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.95 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 15:35:33 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T152411.247422.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:35:34 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753371_chr3_1_fastp.fastq.gz, data/processed/ERR753371_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753371_chr3.sam, results/mapped_reads/ERR753371_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753371_chr3_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR753371_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.67 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 80.17 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.60 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.42 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 45.16 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 133.620 sec; CPU: 127.018 sec\n",
      "[Fri Aug 18 15:39:12 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:39:12 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753371_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr3.bam, results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753371_chr3.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753371_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 15:39:49 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:39:49 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753371_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753371_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "15:39:50.531 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 15:39:50 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753371_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753371_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 15:39:50 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 15:39:50\tMarkDuplicates\tStart of doWork freeMemory: 528785064; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:39:50\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 15:39:50\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam\n",
      "INFO\t2023-08-18 15:39:52\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:108,616,469\n",
      "INFO\t2023-08-18 15:39:52\tMarkDuplicates\tTracking 178 as yet unmatched pairs. 178 records in RAM.\n",
      "INFO\t2023-08-18 15:39:53\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 3:197,282,696\n",
      "INFO\t2023-08-18 15:39:53\tMarkDuplicates\tTracking 58 as yet unmatched pairs. 58 records in RAM.\n",
      "INFO\t2023-08-18 15:39:53\tMarkDuplicates\tRead 2019762 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 15:39:53\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 312231256; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:39:53\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 15:39:53\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 15:39:54\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 15:39:54\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 15:39:54\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 848245800; totalMemory: 1394606080; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:39:54\tMarkDuplicates\tMarking 829946 records as duplicates.\n",
      "INFO\t2023-08-18 15:39:54\tMarkDuplicates\tFound 68 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam\n",
      "INFO\t2023-08-18 15:39:54\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 15:40:07\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 15:40:07\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 15:40:07\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 15:40:07\tMarkDuplicates\tBefore output close freeMemory: 528081000; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:40:07\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 15:40:07\tMarkDuplicates\tAfter output close freeMemory: 528081000; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 15:40:07 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.28 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 15:40:08 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T153533.830635.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:40:10 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753372_chr3_1_fastp.fastq.gz, data/processed/ERR753372_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753372_chr3.sam, results/mapped_reads/ERR753372_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753372_chr3_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753372_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.69 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 80.03 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.61 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.41 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 45.11 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 133.478 sec; CPU: 126.859 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 15:45:30 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:45:30 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753372_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr3.bam, results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753372_chr3.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753372_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 15:46:41 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:46:41 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753372_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753372_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "15:46:42.627 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 15:46:42 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753372_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753372_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 15:46:42 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 15:46:42\tMarkDuplicates\tStart of doWork freeMemory: 528784192; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:46:42\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 15:46:42\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam\n",
      "INFO\t2023-08-18 15:46:44\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:47,409,236\n",
      "INFO\t2023-08-18 15:46:44\tMarkDuplicates\tTracking 449 as yet unmatched pairs. 449 records in RAM.\n",
      "INFO\t2023-08-18 15:46:45\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 3:99,848,915\n",
      "INFO\t2023-08-18 15:46:45\tMarkDuplicates\tTracking 680 as yet unmatched pairs. 680 records in RAM.\n",
      "INFO\t2023-08-18 15:46:47\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 3:142,485,451\n",
      "INFO\t2023-08-18 15:46:47\tMarkDuplicates\tTracking 445 as yet unmatched pairs. 445 records in RAM.\n",
      "INFO\t2023-08-18 15:46:48\tMarkDuplicates\tRead 3984056 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 15:46:48\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 345942808; totalMemory: 846200832; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:46:48\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 15:46:48\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 15:46:49\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 15:46:49\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 15:46:49\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 836713520; totalMemory: 1383071744; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:46:49\tMarkDuplicates\tMarking 95090 records as duplicates.\n",
      "INFO\t2023-08-18 15:46:49\tMarkDuplicates\tFound 627 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam\n",
      "INFO\t2023-08-18 15:46:49\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 15:47:14\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 15:47:14\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 15:47:14\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 15:47:14\tMarkDuplicates\tBefore output close freeMemory: 528083240; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:47:14\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 15:47:14\tMarkDuplicates\tAfter output close freeMemory: 528083240; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 15:47:14 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.52 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 15:47:16 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T154008.964463.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:47:18 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753373_chr3_1_fastp.fastq.gz, data/processed/ERR753373_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753373_chr3.sam, results/mapped_reads/ERR753373_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753373_chr3_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR753373_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.70 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 80.00 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.61 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.42 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 45.26 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 133.605 sec; CPU: 126.982 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 15:53:50 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:53:50 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753373_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr3.bam, results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753373_chr3.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753373_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 15:55:51 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:55:51 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753373_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753373_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "15:55:52.010 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 15:55:52 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753373_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753373_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 15:55:52 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 15:55:52\tMarkDuplicates\tStart of doWork freeMemory: 528783904; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:55:52\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 15:55:52\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam\n",
      "INFO\t2023-08-18 15:55:53\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:52,256,364\n",
      "INFO\t2023-08-18 15:55:53\tMarkDuplicates\tTracking 260 as yet unmatched pairs. 260 records in RAM.\n",
      "INFO\t2023-08-18 15:55:55\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 3:108,162,251\n",
      "INFO\t2023-08-18 15:55:55\tMarkDuplicates\tTracking 295 as yet unmatched pairs. 295 records in RAM.\n",
      "INFO\t2023-08-18 15:55:57\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:05s.  Time for last 1,000,000:    1s.  Last read position: 3:154,974,926\n",
      "INFO\t2023-08-18 15:55:57\tMarkDuplicates\tTracking 206 as yet unmatched pairs. 206 records in RAM.\n",
      "INFO\t2023-08-18 15:55:58\tMarkDuplicates\tRead 3874269 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 15:55:58\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 447498304; totalMemory: 935329792; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:55:58\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 15:55:58\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 15:55:59\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 15:55:59\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 15:55:59\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 925841064; totalMemory: 1472200704; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:55:59\tMarkDuplicates\tMarking 104535 records as duplicates.\n",
      "INFO\t2023-08-18 15:55:59\tMarkDuplicates\tFound 34216 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam\n",
      "INFO\t2023-08-18 15:55:59\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 15:56:34\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 15:56:34\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 15:56:34\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 15:56:34\tMarkDuplicates\tBefore output close freeMemory: 528081632; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 15:56:34\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 15:56:34\tMarkDuplicates\tAfter output close freeMemory: 528081632; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 15:56:34 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.72 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 15:56:38 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T154717.093094.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 15:56:39 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753374_chr3_1_fastp.fastq.gz, data/processed/ERR753374_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753374_chr3.sam, results/mapped_reads/ERR753374_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753374_chr3_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753374_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.70 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 92.20 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.59 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.45 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 44.52 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 144.986 sec; CPU: 138.452 sec\n",
      "[Fri Aug 18 16:01:25 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:01:25 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753374_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr3.bam, results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753374_chr3.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753374_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 16:02:28 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:02:28 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753374_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753374_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "16:02:29.024 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 16:02:29 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753374_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753374_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 16:02:29 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 16:02:29\tMarkDuplicates\tStart of doWork freeMemory: 528821344; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 16:02:29\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 16:02:29\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam\n",
      "INFO\t2023-08-18 16:02:30\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:96,373,400\n",
      "INFO\t2023-08-18 16:02:30\tMarkDuplicates\tTracking 262 as yet unmatched pairs. 262 records in RAM.\n",
      "INFO\t2023-08-18 16:02:32\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 3:175,256,665\n",
      "INFO\t2023-08-18 16:02:32\tMarkDuplicates\tTracking 112 as yet unmatched pairs. 112 records in RAM.\n",
      "INFO\t2023-08-18 16:02:32\tMarkDuplicates\tRead 2270052 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 16:02:32\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 659966232; totalMemory: 975175680; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 16:02:32\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 16:02:32\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 16:02:33\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 16:02:33\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 16:02:33\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 965724744; totalMemory: 1512046592; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 16:02:33\tMarkDuplicates\tMarking 244462 records as duplicates.\n",
      "INFO\t2023-08-18 16:02:33\tMarkDuplicates\tFound 81740 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam\n",
      "INFO\t2023-08-18 16:02:33\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 16:02:52\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 16:02:52\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 16:02:52\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 16:02:52\tMarkDuplicates\tBefore output close freeMemory: 528118736; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 16:02:52\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 16:02:52\tMarkDuplicates\tAfter output close freeMemory: 528118736; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 16:02:52 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.39 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 16:02:54 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T155638.549683.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:02:55 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753375_chr3_1_fastp.fastq.gz, data/processed/ERR753375_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753375_chr3.sam, results/mapped_reads/ERR753375_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753375_chr3_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR753375_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.67 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 81.98 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.61 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.43 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 48.63 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 139.131 sec; CPU: 132.321 sec\n",
      "[Fri Aug 18 16:06:27 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:06:27 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753375_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr3.bam, results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753375_chr3.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753375_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 16:06:58 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:06:58 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753375_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753375_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "16:06:59.292 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 16:06:59 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753375_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753375_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 16:06:59 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 16:06:59\tMarkDuplicates\tStart of doWork freeMemory: 528776696; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 16:06:59\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 16:06:59\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam\n",
      "INFO\t2023-08-18 16:07:00\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:121,913,100\n",
      "INFO\t2023-08-18 16:07:00\tMarkDuplicates\tTracking 260 as yet unmatched pairs. 260 records in RAM.\n",
      "INFO\t2023-08-18 16:07:01\tMarkDuplicates\tRead 1693364 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 16:07:01\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 283447160; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 16:07:01\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 16:07:01\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 16:07:02\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 16:07:02\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 16:07:02\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 789518736; totalMemory: 1335885824; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 16:07:02\tMarkDuplicates\tMarking 82810 records as duplicates.\n",
      "INFO\t2023-08-18 16:07:02\tMarkDuplicates\tFound 199 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam\n",
      "INFO\t2023-08-18 16:07:02\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 16:07:13\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 16:07:13\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 16:07:13\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 16:07:13\tMarkDuplicates\tBefore output close freeMemory: 528074920; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 16:07:13\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 16:07:13\tMarkDuplicates\tAfter output close freeMemory: 528074920; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 16:07:13 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.23 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 16:07:14 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T160254.420428.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:07:15 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753376_chr3_1_fastp.fastq.gz, data/processed/ERR753376_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753376_chr3.sam, results/mapped_reads/ERR753376_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753376_chr3_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753376_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.70 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 70.52 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.59 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.40 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 41.84 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 120.695 sec; CPU: 114.044 sec\n",
      "[Fri Aug 18 16:10:36 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:10:36 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753376_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr3.bam, results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753376_chr3.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753376_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 16:11:16 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:11:16 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753376_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753376_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "16:11:17.087 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 16:11:17 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753376_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753376_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 16:11:17 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 16:11:17\tMarkDuplicates\tStart of doWork freeMemory: 528785232; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 16:11:17\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 16:11:17\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam\n",
      "INFO\t2023-08-18 16:11:18\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:101,476,507\n",
      "INFO\t2023-08-18 16:11:18\tMarkDuplicates\tTracking 284 as yet unmatched pairs. 284 records in RAM.\n",
      "INFO\t2023-08-18 16:11:19\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 3:193,643,851\n",
      "INFO\t2023-08-18 16:11:19\tMarkDuplicates\tTracking 41 as yet unmatched pairs. 41 records in RAM.\n",
      "INFO\t2023-08-18 16:11:20\tMarkDuplicates\tRead 2085915 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 16:11:20\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 306239408; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 16:11:20\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 16:11:20\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 16:11:20\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 16:11:20\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 16:11:20\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 857684352; totalMemory: 1404043264; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 16:11:20\tMarkDuplicates\tMarking 302643 records as duplicates.\n",
      "INFO\t2023-08-18 16:11:20\tMarkDuplicates\tFound 89 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam\n",
      "INFO\t2023-08-18 16:11:20\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 16:11:33\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 16:11:33\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 16:11:33\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 16:11:33\tMarkDuplicates\tBefore output close freeMemory: 528083112; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 16:11:33\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 16:11:33\tMarkDuplicates\tAfter output close freeMemory: 528083112; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 16:11:33 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.27 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 16:11:34 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T160714.770042.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:11:36 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753377_chr3_1_fastp.fastq.gz, data/processed/ERR753377_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753377_chr3.sam, results/mapped_reads/ERR753377_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753377_chr3_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR753377_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.67 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 70.04 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.59 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.40 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 41.72 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 120.191 sec; CPU: 113.427 sec\n",
      "[Fri Aug 18 16:15:02 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:15:02 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753377_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr3.bam, results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753377_chr3.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753377_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 16:15:42 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:15:42 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753377_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753377_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "16:15:43.629 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 16:15:43 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753377_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753377_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 16:15:43 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 16:15:43\tMarkDuplicates\tStart of doWork freeMemory: 528787576; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 16:15:43\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 16:15:43\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam\n",
      "INFO\t2023-08-18 16:15:45\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:98,149,596\n",
      "INFO\t2023-08-18 16:15:45\tMarkDuplicates\tTracking 643 as yet unmatched pairs. 643 records in RAM.\n",
      "INFO\t2023-08-18 16:15:46\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 3:183,827,213\n",
      "INFO\t2023-08-18 16:15:46\tMarkDuplicates\tTracking 140 as yet unmatched pairs. 140 records in RAM.\n",
      "INFO\t2023-08-18 16:15:46\tMarkDuplicates\tRead 2212613 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 16:15:46\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 292614680; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 16:15:46\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 16:15:46\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 16:15:47\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 16:15:47\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 16:15:47\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 879706112; totalMemory: 1426063360; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 16:15:47\tMarkDuplicates\tMarking 317028 records as duplicates.\n",
      "INFO\t2023-08-18 16:15:47\tMarkDuplicates\tFound 102 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam\n",
      "INFO\t2023-08-18 16:15:47\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 16:16:01\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 16:16:01\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 16:16:01\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 16:16:01\tMarkDuplicates\tBefore output close freeMemory: 528084512; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 16:16:01\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 16:16:01\tMarkDuplicates\tAfter output close freeMemory: 528084512; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 16:16:01 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.29 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 16:16:02 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T161135.083840.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:16:03 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753378_chr3_1_fastp.fastq.gz, data/processed/ERR753378_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753378_chr3.sam, results/mapped_reads/ERR753378_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753378_chr3_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR753378_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.66 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 70.09 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.60 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.40 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 41.86 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 120.211 sec; CPU: 113.617 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 16:20:34 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:20:34 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753378_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr3.bam, results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753378_chr3.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753378_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 16:21:34 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:21:34 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753378_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753378_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "16:21:35.838 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 16:21:35 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753378_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753378_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 16:21:35 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 16:21:35\tMarkDuplicates\tStart of doWork freeMemory: 528785456; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 16:21:35\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 16:21:35\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam\n",
      "INFO\t2023-08-18 16:21:37\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:49,251,356\n",
      "INFO\t2023-08-18 16:21:37\tMarkDuplicates\tTracking 438 as yet unmatched pairs. 438 records in RAM.\n",
      "INFO\t2023-08-18 16:21:38\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 3:117,911,449\n",
      "INFO\t2023-08-18 16:21:38\tMarkDuplicates\tTracking 566 as yet unmatched pairs. 566 records in RAM.\n",
      "INFO\t2023-08-18 16:21:40\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 3:175,097,202\n",
      "INFO\t2023-08-18 16:21:40\tMarkDuplicates\tTracking 362 as yet unmatched pairs. 362 records in RAM.\n",
      "INFO\t2023-08-18 16:21:40\tMarkDuplicates\tRead 3435097 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 16:21:40\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 733007640; totalMemory: 1174405120; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 16:21:40\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 16:21:40\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 16:21:41\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 16:21:41\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 16:21:41\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1164918912; totalMemory: 1711276032; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 16:21:41\tMarkDuplicates\tMarking 81437 records as duplicates.\n",
      "INFO\t2023-08-18 16:21:41\tMarkDuplicates\tFound 564 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam\n",
      "INFO\t2023-08-18 16:21:41\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 16:22:01\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 16:22:01\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 16:22:01\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 16:22:01\tMarkDuplicates\tBefore output close freeMemory: 528084120; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 16:22:01\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 16:22:01\tMarkDuplicates\tAfter output close freeMemory: 528084120; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 16:22:01 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.42 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 16:22:03 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T161602.656122.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:22:05 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR696683_chr3_dedup.bam\n",
      "    output: results/variants/ERR696683_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR696683_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR696683_chr3.vcf\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "index file data/reference/genome.fa.fai not found, generating...\n",
      "[Fri Aug 18 16:23:01 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T162204.252616.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:23:03 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753368_chr3_dedup.bam\n",
      "    output: results/variants/ERR753368_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753368_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753368_chr3.vcf\n",
      "    wildcards: sample=ERR753368_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 16:24:19 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T162302.007835.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:24:20 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753369_chr3_dedup.bam\n",
      "    output: results/variants/ERR753369_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753369_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753369_chr3.vcf\n",
      "    wildcards: sample=ERR753369_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 16:25:41 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T162419.988945.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:25:43 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753370_chr3_dedup.bam\n",
      "    output: results/variants/ERR753370_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753370_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753370_chr3.vcf\n",
      "    wildcards: sample=ERR753370_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 16:27:48 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T162542.004533.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:27:50 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753371_chr3_dedup.bam\n",
      "    output: results/variants/ERR753371_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753371_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753371_chr3.vcf\n",
      "    wildcards: sample=ERR753371_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 16:28:29 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T162749.011142.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:28:30 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753372_chr3_dedup.bam\n",
      "    output: results/variants/ERR753372_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753372_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753372_chr3.vcf\n",
      "    wildcards: sample=ERR753372_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 16:29:49 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T162829.830489.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:29:50 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753373_chr3_dedup.bam\n",
      "    output: results/variants/ERR753373_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753373_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753373_chr3.vcf\n",
      "    wildcards: sample=ERR753373_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 16:31:29 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T162949.900831.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:31:30 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753374_chr3_dedup.bam\n",
      "    output: results/variants/ERR753374_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753374_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753374_chr3.vcf\n",
      "    wildcards: sample=ERR753374_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 16:32:38 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T163129.267200.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:32:40 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753375_chr3_dedup.bam\n",
      "    output: results/variants/ERR753375_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753375_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753375_chr3.vcf\n",
      "    wildcards: sample=ERR753375_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 16:33:29 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T163238.982902.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:33:30 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753376_chr3_dedup.bam\n",
      "    output: results/variants/ERR753376_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753376_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753376_chr3.vcf\n",
      "    wildcards: sample=ERR753376_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 16:34:21 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T163329.696363.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:34:23 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753377_chr3_dedup.bam\n",
      "    output: results/variants/ERR753377_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753377_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753377_chr3.vcf\n",
      "    wildcards: sample=ERR753377_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 16:35:15 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T163422.136102.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:35:17 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753378_chr3_dedup.bam\n",
      "    output: results/variants/ERR753378_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753378_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753378_chr3.vcf\n",
      "    wildcards: sample=ERR753378_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 16:36:34 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T163516.156273.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Creating conda environment code/environments/vep.yml...\n",
      "Downloading and installing remote packages.\n",
      "Environment for /home/juancarlos/Escritorio/TFM/code/environments/vep.yml created (location: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_)\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "vep_install_db        1              1              1\n",
      "total                 1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:37:17 2023]\n",
      "rule vep_install_db:\n",
      "    output: tasks/11vep_dependencies.done\n",
      "    jobid: 0\n",
      "    reason: Missing output files: tasks/11vep_dependencies.done\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " - getting list of available cache files\n",
      " - downloading https://ftp.ensembl.org/pub/release-109/variation/indexed_vep_cache/homo_sapiens_vep_109_GRCh38.tar.gz\n",
      " - unpacking homo_sapiens_vep_109_GRCh38.tar.gz\n",
      " - converting cache, this may take some time but will allow VEP to look up variants and frequency data much faster\n",
      " - use CTRL-C to cancel if you do not wish to convert this cache now (you may run convert_cache.pl later)\n",
      "2023-08-18 16:55:02 - Processing homo_sapiens\n",
      "2023-08-18 16:55:02 - Processing version 109_GRCh38\n",
      "2023-08-18 16:55:02 - No unprocessed types remaining, skipping\n",
      "2023-08-18 16:55:02 - All done!\n",
      " - downloading Homo_sapiens.GRCh38.dna.toplevel.fa.gz\n",
      " - downloading Homo_sapiens.GRCh38.dna.toplevel.fa.gz.fai\n",
      " - downloading Homo_sapiens.GRCh38.dna.toplevel.fa.gz.gzi\n",
      "\n",
      "The FASTA file should be automatically detected by the VEP when using --cache or --offline.\n",
      "If it is not, use \"--fasta /home/juancarlos/.vep/homo_sapiens/109_GRCh38/Homo_sapiens.GRCh38.dna.toplevel.fa.gz\"\n",
      "\n",
      "\n",
      "All done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Touching output file tasks/11vep_dependencies.done.\n",
      "[Fri Aug 18 16:55:28 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T163634.231224.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:55:44 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR696683_chr3.vcf\n",
      "    output: results/variants/vep/ERR696683_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR696683_chr3.txt\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "--2023-08-18 16:55:44--  https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/clinvar.vcf.gz\n",
      "Resolviendo ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.7, 130.14.250.10, 2607:f220:41e:250::7, ...\n",
      "Conectando con ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)[130.14.250.7]:443... conectado.\n",
      "Petición HTTP enviada, esperando respuesta... 200 OK\n",
      "Longitud: 81792225 (78M) [application/x-gzip]\n",
      "Grabando a: “data/ClinVar/clinvar.vcf.gz”\n",
      "\n",
      "     0K .......... .......... .......... .......... ..........  0%  205K 6m29s\n",
      "    50K .......... .......... .......... .......... ..........  0%  415K 4m51s\n",
      "   100K .......... .......... .......... .......... ..........  0% 45,5M 3m14s\n",
      "   150K .......... .......... .......... .......... ..........  0% 47,7M 2m26s\n",
      "   200K .......... .......... .......... .......... ..........  0%  411K 2m35s\n",
      "   250K .......... .......... .......... .......... ..........  0%  108M 2m10s\n",
      "   300K .......... .......... .......... .......... ..........  0%  144M 1m51s\n",
      "   350K .......... .......... .......... .......... ..........  0%  138M 97s\n",
      "   400K .......... .......... .......... .......... ..........  0%  418K 1m47s\n",
      "   450K .......... .......... .......... .......... ..........  0% 39,1M 97s\n",
      "   500K .......... .......... .......... .......... ..........  0%  114M 88s\n",
      "   550K .......... .......... .......... .......... ..........  0%  115M 81s\n",
      "   600K .......... .......... .......... .......... ..........  0%  115M 75s\n",
      "   650K .......... .......... .......... .......... ..........  0%  191M 69s\n",
      "   700K .......... .......... .......... .......... ..........  0%  139M 65s\n",
      "   750K .......... .......... .......... .......... ..........  1%  113M 61s\n",
      "   800K .......... .......... .......... .......... ..........  1% 95,5M 57s\n",
      "   850K .......... .......... .......... .......... ..........  1%  431K 64s\n",
      "   900K .......... .......... .......... .......... ..........  1% 28,0M 61s\n",
      "   950K .......... .......... .......... .......... ..........  1% 50,4M 58s\n",
      "  1000K .......... .......... .......... .......... ..........  1% 58,7M 55s\n",
      "  1050K .......... .......... .......... .......... ..........  1%  111M 52s\n",
      "  1100K .......... .......... .......... .......... ..........  1%  123M 50s\n",
      "  1150K .......... .......... .......... .......... ..........  1%  115M 48s\n",
      "  1200K .......... .......... .......... .......... ..........  1% 97,1M 46s\n",
      "  1250K .......... .......... .......... .......... ..........  1%  110M 44s\n",
      "  1300K .......... .......... .......... .......... ..........  1%  121M 43s\n",
      "  1350K .......... .......... .......... .......... ..........  1%  116M 41s\n",
      "  1400K .......... .......... .......... .......... ..........  1%  109M 40s\n",
      "  1450K .......... .......... .......... .......... ..........  1%  114M 38s\n",
      "  1500K .......... .......... .......... .......... ..........  1%  117M 37s\n",
      "  1550K .......... .......... .......... .......... ..........  2%  115M 36s\n",
      "  1600K .......... .......... .......... .......... ..........  2% 96,0M 35s\n",
      "  1650K .......... .......... .......... .......... ..........  2%  111M 34s\n",
      "  1700K .......... .......... .......... .......... ..........  2% 1,47M 34s\n",
      "  1750K .......... .......... .......... .......... ..........  2%  624K 37s\n",
      "  1800K .......... .......... .......... .......... ..........  2%  103M 36s\n",
      "  1850K .......... .......... .......... .......... ..........  2% 62,2M 35s\n",
      "  1900K .......... .......... .......... .......... ..........  2% 54,7M 34s\n",
      "  1950K .......... .......... .......... .......... ..........  2%  126M 33s\n",
      "  2000K .......... .......... .......... .......... ..........  2% 93,6M 32s\n",
      "  2050K .......... .......... .......... .......... ..........  2%  122M 32s\n",
      "  2100K .......... .......... .......... .......... ..........  2%  115M 31s\n",
      "  2150K .......... .......... .......... .......... ..........  2% 68,6M 30s\n",
      "  2200K .......... .......... .......... .......... ..........  2%  153M 30s\n",
      "  2250K .......... .......... .......... .......... ..........  2%  170M 29s\n",
      "  2300K .......... .......... .......... .......... ..........  2%  127M 28s\n",
      "  2350K .......... .......... .......... .......... ..........  3%  118M 28s\n",
      "  2400K .......... .......... .......... .......... ..........  3% 92,6M 27s\n",
      "  2450K .......... .......... .......... .......... ..........  3%  117M 27s\n",
      "  2500K .......... .......... .......... .......... ..........  3% 92,4M 26s\n",
      "  2550K .......... .......... .......... .......... ..........  3%  133M 26s\n",
      "  2600K .......... .......... .......... .......... ..........  3%  129M 25s\n",
      "  2650K .......... .......... .......... .......... ..........  3%  117M 25s\n",
      "  2700K .......... .......... .......... .......... ..........  3%  118M 24s\n",
      "  2750K .......... .......... .......... .......... ..........  3%  114M 24s\n",
      "  2800K .......... .......... .......... .......... ..........  3% 90,5M 23s\n",
      "  2850K .......... .......... .......... .......... ..........  3%  118M 23s\n",
      "  2900K .......... .......... .......... .......... ..........  3%  116M 22s\n",
      "  2950K .......... .......... .......... .......... ..........  3%  117M 22s\n",
      "  3000K .......... .......... .......... .......... ..........  3%  112M 22s\n",
      "  3050K .......... .......... .......... .......... ..........  3%  118M 21s\n",
      "  3100K .......... .......... .......... .......... ..........  3%  118M 21s\n",
      "  3150K .......... .......... .......... .......... ..........  4%  114M 21s\n",
      "  3200K .......... .......... .......... .......... ..........  4% 91,2M 20s\n",
      "  3250K .......... .......... .......... .......... ..........  4%  117M 20s\n",
      "  3300K .......... .......... .......... .......... ..........  4%  113M 20s\n",
      "  3350K .......... .......... .......... .......... ..........  4%  117M 19s\n",
      "  3400K .......... .......... .......... .......... ..........  4%  115M 19s\n",
      "  3450K .......... .......... .......... .......... ..........  4% 11,9M 19s\n",
      "  3500K .......... .......... .......... .......... ..........  4% 2,08M 19s\n",
      "  3550K .......... .......... .......... .......... ..........  4%  640K 21s\n",
      "  3600K .......... .......... .......... .......... ..........  4% 31,9M 20s\n",
      "  3650K .......... .......... .......... .......... ..........  4% 43,7M 20s\n",
      "  3700K .......... .......... .......... .......... ..........  4%  123M 20s\n",
      "  3750K .......... .......... .......... .......... ..........  4%  130M 20s\n",
      "  3800K .......... .......... .......... .......... ..........  4%  116M 19s\n",
      "  3850K .......... .......... .......... .......... ..........  4%  127M 19s\n",
      "  3900K .......... .......... .......... .......... ..........  4%  129M 19s\n",
      "  3950K .......... .......... .......... .......... ..........  5%  131M 19s\n",
      "  4000K .......... .......... .......... .......... ..........  5%  111M 18s\n",
      "  4050K .......... .......... .......... .......... ..........  5%  139M 18s\n",
      "  4100K .......... .......... .......... .......... ..........  5%  138M 18s\n",
      "  4150K .......... .......... .......... .......... ..........  5%  126M 18s\n",
      "  4200K .......... .......... .......... .......... ..........  5% 59,2M 17s\n",
      "  4250K .......... .......... .......... .......... ..........  5%  126M 17s\n",
      "  4300K .......... .......... .......... .......... ..........  5%  137M 17s\n",
      "  4350K .......... .......... .......... .......... ..........  5%  137M 17s\n",
      "  4400K .......... .......... .......... .......... ..........  5%  114M 17s\n",
      "  4450K .......... .......... .......... .......... ..........  5%  129M 16s\n",
      "  4500K .......... .......... .......... .......... ..........  5%  137M 16s\n",
      "  4550K .......... .......... .......... .......... ..........  5%  130M 16s\n",
      "  4600K .......... .......... .......... .......... ..........  5%  115M 16s\n",
      "  4650K .......... .......... .......... .......... ..........  5%  130M 16s\n",
      "  4700K .......... .......... .......... .......... ..........  5%  126M 16s\n",
      "  4750K .......... .......... .......... .......... ..........  6%  137M 15s\n",
      "  4800K .......... .......... .......... .......... ..........  6%  106M 15s\n",
      "  4850K .......... .......... .......... .......... ..........  6%  129M 15s\n",
      "  4900K .......... .......... .......... .......... ..........  6%  132M 15s\n",
      "  4950K .......... .......... .......... .......... ..........  6%  115M 15s\n",
      "  5000K .......... .......... .......... .......... ..........  6%  103M 15s\n",
      "  5050K .......... .......... .......... .......... ..........  6%  127M 14s\n",
      "  5100K .......... .......... .......... .......... ..........  6%  111M 14s\n",
      "  5150K .......... .......... .......... .......... ..........  6%  124M 14s\n",
      "  5200K .......... .......... .......... .......... ..........  6% 93,2M 14s\n",
      "  5250K .......... .......... .......... .......... ..........  6%  114M 14s\n",
      "  5300K .......... .......... .......... .......... ..........  6%  122M 14s\n",
      "  5350K .......... .......... .......... .......... ..........  6%  116M 14s\n",
      "  5400K .......... .......... .......... .......... ..........  6%  107M 14s\n",
      "  5450K .......... .......... .......... .......... ..........  6%  116M 13s\n",
      "  5500K .......... .......... .......... .......... ..........  6%  118M 13s\n",
      "  5550K .......... .......... .......... .......... ..........  7%  120M 13s\n",
      "  5600K .......... .......... .......... .......... ..........  7% 93,5M 13s\n",
      "  5650K .......... .......... .......... .......... ..........  7%  115M 13s\n",
      "  5700K .......... .......... .......... .......... ..........  7%  111M 13s\n",
      "  5750K .......... .......... .......... .......... ..........  7% 94,2M 13s\n",
      "  5800K .......... .......... .......... .......... ..........  7% 84,6M 13s\n",
      "  5850K .......... .......... .......... .......... ..........  7% 94,9M 12s\n",
      "  5900K .......... .......... .......... .......... ..........  7% 90,7M 12s\n",
      "  5950K .......... .......... .......... .......... ..........  7%  193M 12s\n",
      "  6000K .......... .......... .......... .......... ..........  7%  220M 12s\n",
      "  6050K .......... .......... .......... .......... ..........  7%  116M 12s\n",
      "  6100K .......... .......... .......... .......... ..........  7%  117M 12s\n",
      "  6150K .......... .......... .......... .......... ..........  7%  116M 12s\n",
      "  6200K .......... .......... .......... .......... ..........  7%  105M 12s\n",
      "  6250K .......... .......... .......... .......... ..........  7%  124M 12s\n",
      "  6300K .......... .......... .......... .......... ..........  7%  115M 12s\n",
      "  6350K .......... .......... .......... .......... ..........  8%  123M 11s\n",
      "  6400K .......... .......... .......... .......... ..........  8% 88,7M 11s\n",
      "  6450K .......... .......... .......... .......... ..........  8%  117M 11s\n",
      "  6500K .......... .......... .......... .......... ..........  8%  115M 11s\n",
      "  6550K .......... .......... .......... .......... ..........  8% 2,57M 11s\n",
      "  6600K .......... .......... .......... .......... ..........  8%  646K 12s\n",
      "  6650K .......... .......... .......... .......... ..........  8% 85,7M 12s\n",
      "  6700K .......... .......... .......... .......... ..........  8%  113M 12s\n",
      "  6750K .......... .......... .......... .......... ..........  8% 25,6M 12s\n",
      "  6800K .......... .......... .......... .......... ..........  8% 49,7M 12s\n",
      "  6850K .......... .......... .......... .......... ..........  8% 91,9M 12s\n",
      "  6900K .......... .......... .......... .......... ..........  8% 96,0M 12s\n",
      "  6950K .......... .......... .......... .......... ..........  8% 92,3M 11s\n",
      "  7000K .......... .......... .......... .......... ..........  8% 88,4M 11s\n",
      "  7050K .......... .......... .......... .......... ..........  8% 99,2M 11s\n",
      "  7100K .......... .......... .......... .......... ..........  8% 99,8M 11s\n",
      "  7150K .......... .......... .......... .......... ..........  9% 77,8M 11s\n",
      "  7200K .......... .......... .......... .......... ..........  9%  129M 11s\n",
      "  7250K .......... .......... .......... .......... ..........  9%  169M 11s\n",
      "  7300K .......... .......... .......... .......... ..........  9%  166M 11s\n",
      "  7350K .......... .......... .......... .......... ..........  9%  176M 11s\n",
      "  7400K .......... .......... .......... .......... ..........  9%  153M 11s\n",
      "  7450K .......... .......... .......... .......... ..........  9%  174M 11s\n",
      "  7500K .......... .......... .......... .......... ..........  9%  197M 11s\n",
      "  7550K .......... .......... .......... .......... ..........  9%  236M 11s\n",
      "  7600K .......... .......... .......... .......... ..........  9%  193M 10s\n",
      "  7650K .......... .......... .......... .......... ..........  9%  221M 10s\n",
      "  7700K .......... .......... .......... .......... ..........  9%  236M 10s\n",
      "  7750K .......... .......... .......... .......... ..........  9%  167M 10s\n",
      "  7800K .......... .......... .......... .......... ..........  9%  112M 10s\n",
      "  7850K .......... .......... .......... .......... ..........  9%  111M 10s\n",
      "  7900K .......... .......... .......... .......... ..........  9%  108M 10s\n",
      "  7950K .......... .......... .......... .......... .......... 10%  129M 10s\n",
      "  8000K .......... .......... .......... .......... .......... 10% 94,6M 10s\n",
      "  8050K .......... .......... .......... .......... .......... 10%  115M 10s\n",
      "  8100K .......... .......... .......... .......... .......... 10%  117M 10s\n",
      "  8150K .......... .......... .......... .......... .......... 10%  117M 10s\n",
      "  8200K .......... .......... .......... .......... .......... 10%  114M 10s\n",
      "  8250K .......... .......... .......... .......... .......... 10%  116M 10s\n",
      "  8300K .......... .......... .......... .......... .......... 10% 25,5M 10s\n",
      "  8350K .......... .......... .......... .......... .......... 10%  296M 9s\n",
      "  8400K .......... .......... .......... .......... .......... 10%  266M 9s\n",
      "  8450K .......... .......... .......... .......... .......... 10%  357M 9s\n",
      "  8500K .......... .......... .......... .......... .......... 10%  389M 9s\n",
      "  8550K .......... .......... .......... .......... .......... 10%  370M 9s\n",
      "  8600K .......... .......... .......... .......... .......... 10%  121M 9s\n",
      "  8650K .......... .......... .......... .......... .......... 10%  118M 9s\n",
      "  8700K .......... .......... .......... .......... .......... 10%  114M 9s\n",
      "  8750K .......... .......... .......... .......... .......... 11%  118M 9s\n",
      "  8800K .......... .......... .......... .......... .......... 11% 92,0M 9s\n",
      "  8850K .......... .......... .......... .......... .......... 11%  113M 9s\n",
      "  8900K .......... .......... .......... .......... .......... 11%  117M 9s\n",
      "  8950K .......... .......... .......... .......... .......... 11%  118M 9s\n",
      "  9000K .......... .......... .......... .......... .......... 11%  114M 9s\n",
      "  9050K .......... .......... .......... .......... .......... 11%  117M 9s\n",
      "  9100K .......... .......... .......... .......... .......... 11%  113M 9s\n",
      "  9150K .......... .......... .......... .......... .......... 11%  119M 9s\n",
      "  9200K .......... .......... .......... .......... .......... 11% 90,5M 9s\n",
      "  9250K .......... .......... .......... .......... .......... 11%  114M 8s\n",
      "  9300K .......... .......... .......... .......... .......... 11%  117M 8s\n",
      "  9350K .......... .......... .......... .......... .......... 11%  117M 8s\n",
      "  9400K .......... .......... .......... .......... .......... 11%  114M 8s\n",
      "  9450K .......... .......... .......... .......... .......... 11%  118M 8s\n",
      "  9500K .......... .......... .......... .......... .......... 11%  117M 8s\n",
      "  9550K .......... .......... .......... .......... .......... 12% 99,5M 8s\n",
      "  9600K .......... .......... .......... .......... .......... 12%  538K 9s\n",
      "  9650K .......... .......... .......... .......... .......... 12% 17,9M 9s\n",
      "  9700K .......... .......... .......... .......... .......... 12%  100M 9s\n",
      "  9750K .......... .......... .......... .......... .......... 12% 84,6M 9s\n",
      "  9800K .......... .......... .......... .......... .......... 12% 54,7M 9s\n",
      "  9850K .......... .......... .......... .......... .......... 12%  121M 9s\n",
      "  9900K .......... .......... .......... .......... .......... 12% 35,1M 9s\n",
      "  9950K .......... .......... .......... .......... .......... 12% 94,0M 9s\n",
      " 10000K .......... .......... .......... .......... .......... 12% 60,1M 8s\n",
      " 10050K .......... .......... .......... .......... .......... 12%  115M 8s\n",
      " 10100K .......... .......... .......... .......... .......... 12%  112M 8s\n",
      " 10150K .......... .......... .......... .......... .......... 12%  117M 8s\n",
      " 10200K .......... .......... .......... .......... .......... 12% 63,1M 8s\n",
      " 10250K .......... .......... .......... .......... .......... 12%  117M 8s\n",
      " 10300K .......... .......... .......... .......... .......... 12%  118M 8s\n",
      " 10350K .......... .......... .......... .......... .......... 13% 83,4M 8s\n",
      " 10400K .......... .......... .......... .......... .......... 13%  133M 8s\n",
      " 10450K .......... .......... .......... .......... .......... 13%  119M 8s\n",
      " 10500K .......... .......... .......... .......... .......... 13%  113M 8s\n",
      " 10550K .......... .......... .......... .......... .......... 13%  117M 8s\n",
      " 10600K .......... .......... .......... .......... .......... 13%  109M 8s\n",
      " 10650K .......... .......... .......... .......... .......... 13%  117M 8s\n",
      " 10700K .......... .......... .......... .......... .......... 13%  116M 8s\n",
      " 10750K .......... .......... .......... .......... .......... 13%  113M 8s\n",
      " 10800K .......... .......... .......... .......... .......... 13% 93,0M 8s\n",
      " 10850K .......... .......... .......... .......... .......... 13% 57,7M 8s\n",
      " 10900K .......... .......... .......... .......... .......... 13%  115M 8s\n",
      " 10950K .......... .......... .......... .......... .......... 13%  138M 8s\n",
      " 11000K .......... .......... .......... .......... .......... 13%  124M 8s\n",
      " 11050K .......... .......... .......... .......... .......... 13%  143M 8s\n",
      " 11100K .......... .......... .......... .......... .......... 13%  138M 8s\n",
      " 11150K .......... .......... .......... .......... .......... 14%  135M 8s\n",
      " 11200K .......... .......... .......... .......... .......... 14%  116M 8s\n",
      " 11250K .......... .......... .......... .......... .......... 14%  121M 7s\n",
      " 11300K .......... .......... .......... .......... .......... 14%  117M 7s\n",
      " 11350K .......... .......... .......... .......... .......... 14% 95,9M 7s\n",
      " 11400K .......... .......... .......... .......... .......... 14% 91,5M 7s\n",
      " 11450K .......... .......... .......... .......... .......... 14%  116M 7s\n",
      " 11500K .......... .......... .......... .......... .......... 14%  119M 7s\n",
      " 11550K .......... .......... .......... .......... .......... 14%  121M 7s\n",
      " 11600K .......... .......... .......... .......... .......... 14% 92,3M 7s\n",
      " 11650K .......... .......... .......... .......... .......... 14%  117M 7s\n",
      " 11700K .......... .......... .......... .......... .......... 14%  118M 7s\n",
      " 11750K .......... .......... .......... .......... .......... 14%  115M 7s\n",
      " 11800K .......... .......... .......... .......... .......... 14%  108M 7s\n",
      " 11850K .......... .......... .......... .......... .......... 14%  123M 7s\n",
      " 11900K .......... .......... .......... .......... .......... 14%  117M 7s\n",
      " 11950K .......... .......... .......... .......... .......... 15%  118M 7s\n",
      " 12000K .......... .......... .......... .......... .......... 15% 89,9M 7s\n",
      " 12050K .......... .......... .......... .......... .......... 15%  117M 7s\n",
      " 12100K .......... .......... .......... .......... .......... 15%  117M 7s\n",
      " 12150K .......... .......... .......... .......... .......... 15%  115M 7s\n",
      " 12200K .......... .......... .......... .......... .......... 15%  113M 7s\n",
      " 12250K .......... .......... .......... .......... .......... 15%  115M 7s\n",
      " 12300K .......... .......... .......... .......... .......... 15%  122M 7s\n",
      " 12350K .......... .......... .......... .......... .......... 15%  115M 7s\n",
      " 12400K .......... .......... .......... .......... .......... 15% 89,3M 7s\n",
      " 12450K .......... .......... .......... .......... .......... 15%  118M 7s\n",
      " 12500K .......... .......... .......... .......... .......... 15%  117M 7s\n",
      " 12550K .......... .......... .......... .......... .......... 15%  119M 7s\n",
      " 12600K .......... .......... .......... .......... .......... 15%  111M 7s\n",
      " 12650K .......... .......... .......... .......... .......... 15%  117M 7s\n",
      " 12700K .......... .......... .......... .......... .......... 15%  532K 7s\n",
      " 12750K .......... .......... .......... .......... .......... 16% 61,6M 7s\n",
      " 12800K .......... .......... .......... .......... .......... 16% 66,7M 7s\n",
      " 12850K .......... .......... .......... .......... .......... 16% 57,6M 7s\n",
      " 12900K .......... .......... .......... .......... .......... 16% 59,2M 7s\n",
      " 12950K .......... .......... .......... .......... .......... 16%  120M 7s\n",
      " 13000K .......... .......... .......... .......... .......... 16% 51,9M 7s\n",
      " 13050K .......... .......... .......... .......... .......... 16%  113M 7s\n",
      " 13100K .......... .......... .......... .......... .......... 16%  118M 7s\n",
      " 13150K .......... .......... .......... .......... .......... 16%  121M 7s\n",
      " 13200K .......... .......... .......... .......... .......... 16% 92,2M 7s\n",
      " 13250K .......... .......... .......... .......... .......... 16%  117M 7s\n",
      " 13300K .......... .......... .......... .......... .......... 16%  121M 7s\n",
      " 13350K .......... .......... .......... .......... .......... 16%  113M 7s\n",
      " 13400K .......... .......... .......... .......... .......... 16%  105M 7s\n",
      " 13450K .......... .......... .......... .......... .......... 16%  116M 7s\n",
      " 13500K .......... .......... .......... .......... .......... 16%  121M 7s\n",
      " 13550K .......... .......... .......... .......... .......... 17%  119M 7s\n",
      " 13600K .......... .......... .......... .......... .......... 17% 92,6M 7s\n",
      " 13650K .......... .......... .......... .......... .......... 17%  117M 7s\n",
      " 13700K .......... .......... .......... .......... .......... 17%  119M 6s\n",
      " 13750K .......... .......... .......... .......... .......... 17%  114M 6s\n",
      " 13800K .......... .......... .......... .......... .......... 17%  105M 6s\n",
      " 13850K .......... .......... .......... .......... .......... 17%  116M 6s\n",
      " 13900K .......... .......... .......... .......... .......... 17%  119M 6s\n",
      " 13950K .......... .......... .......... .......... .......... 17%  115M 6s\n",
      " 14000K .......... .......... .......... .......... .......... 17% 73,0M 6s\n",
      " 14050K .......... .......... .......... .......... .......... 17%  122M 6s\n",
      " 14100K .......... .......... .......... .......... .......... 17%  117M 6s\n",
      " 14150K .......... .......... .......... .......... .......... 17%  117M 6s\n",
      " 14200K .......... .......... .......... .......... .......... 17%  110M 6s\n",
      " 14250K .......... .......... .......... .......... .......... 17%  113M 6s\n",
      " 14300K .......... .......... .......... .......... .......... 17%  119M 6s\n",
      " 14350K .......... .......... .......... .......... .......... 18%  121M 6s\n",
      " 14400K .......... .......... .......... .......... .......... 18% 90,3M 6s\n",
      " 14450K .......... .......... .......... .......... .......... 18%  114M 6s\n",
      " 14500K .......... .......... .......... .......... .......... 18%  121M 6s\n",
      " 14550K .......... .......... .......... .......... .......... 18%  118M 6s\n",
      " 14600K .......... .......... .......... .......... .......... 18%  111M 6s\n",
      " 14650K .......... .......... .......... .......... .......... 18%  119M 6s\n",
      " 14700K .......... .......... .......... .......... .......... 18%  117M 6s\n",
      " 14750K .......... .......... .......... .......... .......... 18% 87,5M 6s\n",
      " 14800K .......... .......... .......... .......... .......... 18% 89,7M 6s\n",
      " 14850K .......... .......... .......... .......... .......... 18%  117M 6s\n",
      " 14900K .......... .......... .......... .......... .......... 18%  117M 6s\n",
      " 14950K .......... .......... .......... .......... .......... 18%  119M 6s\n",
      " 15000K .......... .......... .......... .......... .......... 18%  114M 6s\n",
      " 15050K .......... .......... .......... .......... .......... 18%  112M 6s\n",
      " 15100K .......... .......... .......... .......... .......... 18%  115M 6s\n",
      " 15150K .......... .......... .......... .......... .......... 19%  121M 6s\n",
      " 15200K .......... .......... .......... .......... .......... 19% 88,7M 6s\n",
      " 15250K .......... .......... .......... .......... .......... 19%  117M 6s\n",
      " 15300K .......... .......... .......... .......... .......... 19%  117M 6s\n",
      " 15350K .......... .......... .......... .......... .......... 19%  118M 6s\n",
      " 15400K .......... .......... .......... .......... .......... 19%  113M 6s\n",
      " 15450K .......... .......... .......... .......... .......... 19%  114M 6s\n",
      " 15500K .......... .......... .......... .......... .......... 19%  119M 6s\n",
      " 15550K .......... .......... .......... .......... .......... 19%  117M 6s\n",
      " 15600K .......... .......... .......... .......... .......... 19% 90,5M 6s\n",
      " 15650K .......... .......... .......... .......... .......... 19%  114M 6s\n",
      " 15700K .......... .......... .......... .......... .......... 19%  553K 6s\n",
      " 15750K .......... .......... .......... .......... .......... 19% 13,2M 6s\n",
      " 15800K .......... .......... .......... .......... .......... 19% 39,0M 6s\n",
      " 15850K .......... .......... .......... .......... .......... 19%  102M 6s\n",
      " 15900K .......... .......... .......... .......... .......... 19% 83,3M 6s\n",
      " 15950K .......... .......... .......... .......... .......... 20% 55,9M 6s\n",
      " 16000K .......... .......... .......... .......... .......... 20% 51,7M 6s\n",
      " 16050K .......... .......... .......... .......... .......... 20% 63,9M 6s\n",
      " 16100K .......... .......... .......... .......... .......... 20%  121M 6s\n",
      " 16150K .......... .......... .......... .......... .......... 20%  134M 6s\n",
      " 16200K .......... .......... .......... .......... .......... 20%  104M 6s\n",
      " 16250K .......... .......... .......... .......... .......... 20%  131M 6s\n",
      " 16300K .......... .......... .......... .......... .......... 20%  113M 6s\n",
      " 16350K .......... .......... .......... .......... .......... 20%  115M 6s\n",
      " 16400K .......... .......... .......... .......... .......... 20% 97,4M 6s\n",
      " 16450K .......... .......... .......... .......... .......... 20%  111M 6s\n",
      " 16500K .......... .......... .......... .......... .......... 20% 96,1M 6s\n",
      " 16550K .......... .......... .......... .......... .......... 20%  151M 6s\n",
      " 16600K .......... .......... .......... .......... .......... 20%  108M 6s\n",
      " 16650K .......... .......... .......... .......... .......... 20%  118M 6s\n",
      " 16700K .......... .......... .......... .......... .......... 20%  118M 6s\n",
      " 16750K .......... .......... .......... .......... .......... 21%  118M 6s\n",
      " 16800K .......... .......... .......... .......... .......... 21% 92,3M 6s\n",
      " 16850K .......... .......... .......... .......... .......... 21%  113M 5s\n",
      " 16900K .......... .......... .......... .......... .......... 21%  119M 5s\n",
      " 16950K .......... .......... .......... .......... .......... 21%  118M 5s\n",
      " 17000K .......... .......... .......... .......... .......... 21%  107M 5s\n",
      " 17050K .......... .......... .......... .......... .......... 21%  118M 5s\n",
      " 17100K .......... .......... .......... .......... .......... 21%  116M 5s\n",
      " 17150K .......... .......... .......... .......... .......... 21%  118M 5s\n",
      " 17200K .......... .......... .......... .......... .......... 21% 92,8M 5s\n",
      " 17250K .......... .......... .......... .......... .......... 21%  117M 5s\n",
      " 17300K .......... .......... .......... .......... .......... 21%  115M 5s\n",
      " 17350K .......... .......... .......... .......... .......... 21%  117M 5s\n",
      " 17400K .......... .......... .......... .......... .......... 21%  110M 5s\n",
      " 17450K .......... .......... .......... .......... .......... 21%  120M 5s\n",
      " 17500K .......... .......... .......... .......... .......... 21%  117M 5s\n",
      " 17550K .......... .......... .......... .......... .......... 22%  114M 5s\n",
      " 17600K .......... .......... .......... .......... .......... 22% 92,7M 5s\n",
      " 17650K .......... .......... .......... .......... .......... 22%  116M 5s\n",
      " 17700K .......... .......... .......... .......... .......... 22%  117M 5s\n",
      " 17750K .......... .......... .......... .......... .......... 22%  117M 5s\n",
      " 17800K .......... .......... .......... .......... .......... 22%  111M 5s\n",
      " 17850K .......... .......... .......... .......... .......... 22%  116M 5s\n",
      " 17900K .......... .......... .......... .......... .......... 22%  118M 5s\n",
      " 17950K .......... .......... .......... .......... .......... 22% 69,1M 5s\n",
      " 18000K .......... .......... .......... .......... .......... 22%  198M 5s\n",
      " 18050K .......... .......... .......... .......... .......... 22%  110M 5s\n",
      " 18100K .......... .......... .......... .......... .......... 22%  122M 5s\n",
      " 18150K .......... .......... .......... .......... .......... 22%  117M 5s\n",
      " 18200K .......... .......... .......... .......... .......... 22%  113M 5s\n",
      " 18250K .......... .......... .......... .......... .......... 22%  116M 5s\n",
      " 18300K .......... .......... .......... .......... .......... 22%  118M 5s\n",
      " 18350K .......... .......... .......... .......... .......... 23%  106M 5s\n",
      " 18400K .......... .......... .......... .......... .......... 23% 95,1M 5s\n",
      " 18450K .......... .......... .......... .......... .......... 23%  117M 5s\n",
      " 18500K .......... .......... .......... .......... .......... 23%  115M 5s\n",
      " 18550K .......... .......... .......... .......... .......... 23% 93,9M 5s\n",
      " 18600K .......... .......... .......... .......... .......... 23%  142M 5s\n",
      " 18650K .......... .......... .......... .......... .......... 23%  117M 5s\n",
      " 18700K .......... .......... .......... .......... .......... 23%  118M 5s\n",
      " 18750K .......... .......... .......... .......... .......... 23%  549K 5s\n",
      " 18800K .......... .......... .......... .......... .......... 23% 17,7M 5s\n",
      " 18850K .......... .......... .......... .......... .......... 23% 30,5M 5s\n",
      " 18900K .......... .......... .......... .......... .......... 23% 88,5M 5s\n",
      " 18950K .......... .......... .......... .......... .......... 23%  114M 5s\n",
      " 19000K .......... .......... .......... .......... .......... 23% 61,5M 5s\n",
      " 19050K .......... .......... .......... .......... .......... 23% 65,0M 5s\n",
      " 19100K .......... .......... .......... .......... .......... 23% 56,6M 5s\n",
      " 19150K .......... .......... .......... .......... .......... 24%  117M 5s\n",
      " 19200K .......... .......... .......... .......... .......... 24% 65,6M 5s\n",
      " 19250K .......... .......... .......... .......... .......... 24%  116M 5s\n",
      " 19300K .......... .......... .......... .......... .......... 24% 65,3M 5s\n",
      " 19350K .......... .......... .......... .......... .......... 24% 68,4M 5s\n",
      " 19400K .......... .......... .......... .......... .......... 24%  111M 5s\n",
      " 19450K .......... .......... .......... .......... .......... 24%  118M 5s\n",
      " 19500K .......... .......... .......... .......... .......... 24%  115M 5s\n",
      " 19550K .......... .......... .......... .......... .......... 24%  120M 5s\n",
      " 19600K .......... .......... .......... .......... .......... 24% 92,2M 5s\n",
      " 19650K .......... .......... .......... .......... .......... 24%  118M 5s\n",
      " 19700K .......... .......... .......... .......... .......... 24%  117M 5s\n",
      " 19750K .......... .......... .......... .......... .......... 24%  114M 5s\n",
      " 19800K .......... .......... .......... .......... .......... 24%  100M 5s\n",
      " 19850K .......... .......... .......... .......... .......... 24%  133M 5s\n",
      " 19900K .......... .......... .......... .......... .......... 24%  118M 5s\n",
      " 19950K .......... .......... .......... .......... .......... 25%  117M 5s\n",
      " 20000K .......... .......... .......... .......... .......... 25% 90,3M 5s\n",
      " 20050K .......... .......... .......... .......... .......... 25%  118M 5s\n",
      " 20100K .......... .......... .......... .......... .......... 25%  117M 5s\n",
      " 20150K .......... .......... .......... .......... .......... 25%  114M 5s\n",
      " 20200K .......... .......... .......... .......... .......... 25%  115M 5s\n",
      " 20250K .......... .......... .......... .......... .......... 25%  118M 5s\n",
      " 20300K .......... .......... .......... .......... .......... 25%  117M 5s\n",
      " 20350K .......... .......... .......... .......... .......... 25%  117M 5s\n",
      " 20400K .......... .......... .......... .......... .......... 25% 89,7M 5s\n",
      " 20450K .......... .......... .......... .......... .......... 25%  117M 5s\n",
      " 20500K .......... .......... .......... .......... .......... 25%  118M 5s\n",
      " 20550K .......... .......... .......... .......... .......... 25%  118M 5s\n",
      " 20600K .......... .......... .......... .......... .......... 25%  110M 5s\n",
      " 20650K .......... .......... .......... .......... .......... 25%  118M 5s\n",
      " 20700K .......... .......... .......... .......... .......... 25%  116M 5s\n",
      " 20750K .......... .......... .......... .......... .......... 26%  116M 5s\n",
      " 20800K .......... .......... .......... .......... .......... 26% 89,7M 5s\n",
      " 20850K .......... .......... .......... .......... .......... 26%  120M 5s\n",
      " 20900K .......... .......... .......... .......... .......... 26%  117M 5s\n",
      " 20950K .......... .......... .......... .......... .......... 26%  116M 5s\n",
      " 21000K .......... .......... .......... .......... .......... 26%  111M 4s\n",
      " 21050K .......... .......... .......... .......... .......... 26%  118M 4s\n",
      " 21100K .......... .......... .......... .......... .......... 26%  118M 4s\n",
      " 21150K .......... .......... .......... .......... .......... 26%  117M 4s\n",
      " 21200K .......... .......... .......... .......... .......... 26% 87,8M 4s\n",
      " 21250K .......... .......... .......... .......... .......... 26%  117M 4s\n",
      " 21300K .......... .......... .......... .......... .......... 26%  117M 4s\n",
      " 21350K .......... .......... .......... .......... .......... 26%  118M 4s\n",
      " 21400K .......... .......... .......... .......... .......... 26%  112M 4s\n",
      " 21450K .......... .......... .......... .......... .......... 26%  117M 4s\n",
      " 21500K .......... .......... .......... .......... .......... 26%  116M 4s\n",
      " 21550K .......... .......... .......... .......... .......... 27%  118M 4s\n",
      " 21600K .......... .......... .......... .......... .......... 27% 88,6M 4s\n",
      " 21650K .......... .......... .......... .......... .......... 27%  116M 4s\n",
      " 21700K .......... .......... .......... .......... .......... 27%  117M 4s\n",
      " 21750K .......... .......... .......... .......... .......... 27%  117M 4s\n",
      " 21800K .......... .......... .......... .......... .......... 27%  542K 5s\n",
      " 21850K .......... .......... .......... .......... .......... 27% 27,5M 5s\n",
      " 21900K .......... .......... .......... .......... .......... 27% 40,3M 5s\n",
      " 21950K .......... .......... .......... .......... .......... 27% 64,4M 4s\n",
      " 22000K .......... .......... .......... .......... .......... 27% 54,7M 4s\n",
      " 22050K .......... .......... .......... .......... .......... 27%  117M 4s\n",
      " 22100K .......... .......... .......... .......... .......... 27%  114M 4s\n",
      " 22150K .......... .......... .......... .......... .......... 27% 88,7M 4s\n",
      " 22200K .......... .......... .......... .......... .......... 27% 65,5M 4s\n",
      " 22250K .......... .......... .......... .......... .......... 27%  115M 4s\n",
      " 22300K .......... .......... .......... .......... .......... 27%  119M 4s\n",
      " 22350K .......... .......... .......... .......... .......... 28%  115M 4s\n",
      " 22400K .......... .......... .......... .......... .......... 28% 95,3M 4s\n",
      " 22450K .......... .......... .......... .......... .......... 28%  117M 4s\n",
      " 22500K .......... .......... .......... .......... .......... 28% 65,6M 4s\n",
      " 22550K .......... .......... .......... .......... .......... 28%  122M 4s\n",
      " 22600K .......... .......... .......... .......... .......... 28% 59,5M 4s\n",
      " 22650K .......... .......... .......... .......... .......... 28% 55,2M 4s\n",
      " 22700K .......... .......... .......... .......... .......... 28%  120M 4s\n",
      " 22750K .......... .......... .......... .......... .......... 28%  117M 4s\n",
      " 22800K .......... .......... .......... .......... .......... 28% 93,6M 4s\n",
      " 22850K .......... .......... .......... .......... .......... 28%  118M 4s\n",
      " 22900K .......... .......... .......... .......... .......... 28%  116M 4s\n",
      " 22950K .......... .......... .......... .......... .......... 28%  119M 4s\n",
      " 23000K .......... .......... .......... .......... .......... 28%  107M 4s\n",
      " 23050K .......... .......... .......... .......... .......... 28%  114M 4s\n",
      " 23100K .......... .......... .......... .......... .......... 28%  119M 4s\n",
      " 23150K .......... .......... .......... .......... .......... 29%  118M 4s\n",
      " 23200K .......... .......... .......... .......... .......... 29% 92,7M 4s\n",
      " 23250K .......... .......... .......... .......... .......... 29%  117M 4s\n",
      " 23300K .......... .......... .......... .......... .......... 29%  120M 4s\n",
      " 23350K .......... .......... .......... .......... .......... 29%  119M 4s\n",
      " 23400K .......... .......... .......... .......... .......... 29%  108M 4s\n",
      " 23450K .......... .......... .......... .......... .......... 29%  119M 4s\n",
      " 23500K .......... .......... .......... .......... .......... 29% 87,9M 4s\n",
      " 23550K .......... .......... .......... .......... .......... 29%  138M 4s\n",
      " 23600K .......... .......... .......... .......... .......... 29%  106M 4s\n",
      " 23650K .......... .......... .......... .......... .......... 29%  115M 4s\n",
      " 23700K .......... .......... .......... .......... .......... 29%  119M 4s\n",
      " 23750K .......... .......... .......... .......... .......... 29%  117M 4s\n",
      " 23800K .......... .......... .......... .......... .......... 29%  112M 4s\n",
      " 23850K .......... .......... .......... .......... .......... 29%  114M 4s\n",
      " 23900K .......... .......... .......... .......... .......... 29%  112M 4s\n",
      " 23950K .......... .......... .......... .......... .......... 30%  120M 4s\n",
      " 24000K .......... .......... .......... .......... .......... 30% 92,4M 4s\n",
      " 24050K .......... .......... .......... .......... .......... 30%  114M 4s\n",
      " 24100K .......... .......... .......... .......... .......... 30%  118M 4s\n",
      " 24150K .......... .......... .......... .......... .......... 30%  117M 4s\n",
      " 24200K .......... .......... .......... .......... .......... 30%  118M 4s\n",
      " 24250K .......... .......... .......... .......... .......... 30%  117M 4s\n",
      " 24300K .......... .......... .......... .......... .......... 30%  113M 4s\n",
      " 24350K .......... .......... .......... .......... .......... 30%  118M 4s\n",
      " 24400K .......... .......... .......... .......... .......... 30% 90,7M 4s\n",
      " 24450K .......... .......... .......... .......... .......... 30%  114M 4s\n",
      " 24500K .......... .......... .......... .......... .......... 30%  118M 4s\n",
      " 24550K .......... .......... .......... .......... .......... 30%  117M 4s\n",
      " 24600K .......... .......... .......... .......... .......... 30%  114M 4s\n",
      " 24650K .......... .......... .......... .......... .......... 30%  117M 4s\n",
      " 24700K .......... .......... .......... .......... .......... 30%  113M 4s\n",
      " 24750K .......... .......... .......... .......... .......... 31%  118M 4s\n",
      " 24800K .......... .......... .......... .......... .......... 31% 1,17M 4s\n",
      " 24850K .......... .......... .......... .......... .......... 31% 1012K 4s\n",
      " 24900K .......... .......... .......... .......... .......... 31% 15,8M 4s\n",
      " 24950K .......... .......... .......... .......... .......... 31% 23,2M 4s\n",
      " 25000K .......... .......... .......... .......... .......... 31%  117M 4s\n",
      " 25050K .......... .......... .......... .......... .......... 31%  180M 4s\n",
      " 25100K .......... .......... .......... .......... .......... 31%  153M 4s\n",
      " 25150K .......... .......... .......... .......... .......... 31%  108M 4s\n",
      " 25200K .......... .......... .......... .......... .......... 31% 60,1M 4s\n",
      " 25250K .......... .......... .......... .......... .......... 31%  118M 4s\n",
      " 25300K .......... .......... .......... .......... .......... 31%  110M 4s\n",
      " 25350K .......... .......... .......... .......... .......... 31% 66,1M 4s\n",
      " 25400K .......... .......... .......... .......... .......... 31%  109M 4s\n",
      " 25450K .......... .......... .......... .......... .......... 31%  119M 4s\n",
      " 25500K .......... .......... .......... .......... .......... 31%  118M 4s\n",
      " 25550K .......... .......... .......... .......... .......... 32%  114M 4s\n",
      " 25600K .......... .......... .......... .......... .......... 32% 93,9M 4s\n",
      " 25650K .......... .......... .......... .......... .......... 32%  118M 4s\n",
      " 25700K .......... .......... .......... .......... .......... 32%  114M 4s\n",
      " 25750K .......... .......... .......... .......... .......... 32%  114M 4s\n",
      " 25800K .......... .......... .......... .......... .......... 32% 71,5M 4s\n",
      " 25850K .......... .......... .......... .......... .......... 32% 55,1M 4s\n",
      " 25900K .......... .......... .......... .......... .......... 32%  118M 4s\n",
      " 25950K .......... .......... .......... .......... .......... 32% 63,8M 4s\n",
      " 26000K .......... .......... .......... .......... .......... 32% 92,6M 4s\n",
      " 26050K .......... .......... .......... .......... .......... 32%  117M 4s\n",
      " 26100K .......... .......... .......... .......... .......... 32%  114M 4s\n",
      " 26150K .......... .......... .......... .......... .......... 32%  117M 4s\n",
      " 26200K .......... .......... .......... .......... .......... 32%  112M 4s\n",
      " 26250K .......... .......... .......... .......... .......... 32%  115M 4s\n",
      " 26300K .......... .......... .......... .......... .......... 32%  119M 4s\n",
      " 26350K .......... .......... .......... .......... .......... 33%  115M 4s\n",
      " 26400K .......... .......... .......... .......... .......... 33% 91,8M 4s\n",
      " 26450K .......... .......... .......... .......... .......... 33%  117M 4s\n",
      " 26500K .......... .......... .......... .......... .......... 33%  118M 4s\n",
      " 26550K .......... .......... .......... .......... .......... 33%  113M 4s\n",
      " 26600K .......... .......... .......... .......... .......... 33%  110M 4s\n",
      " 26650K .......... .......... .......... .......... .......... 33%  117M 4s\n",
      " 26700K .......... .......... .......... .......... .......... 33%  126M 4s\n",
      " 26750K .......... .......... .......... .......... .......... 33%  118M 4s\n",
      " 26800K .......... .......... .......... .......... .......... 33% 89,1M 4s\n",
      " 26850K .......... .......... .......... .......... .......... 33%  117M 4s\n",
      " 26900K .......... .......... .......... .......... .......... 33%  117M 4s\n",
      " 26950K .......... .......... .......... .......... .......... 33%  115M 4s\n",
      " 27000K .......... .......... .......... .......... .......... 33%  113M 4s\n",
      " 27050K .......... .......... .......... .......... .......... 33%  117M 4s\n",
      " 27100K .......... .......... .......... .......... .......... 33%  119M 4s\n",
      " 27150K .......... .......... .......... .......... .......... 34%  117M 4s\n",
      " 27200K .......... .......... .......... .......... .......... 34% 88,9M 4s\n",
      " 27250K .......... .......... .......... .......... .......... 34%  116M 4s\n",
      " 27300K .......... .......... .......... .......... .......... 34%  117M 4s\n",
      " 27350K .......... .......... .......... .......... .......... 34%  115M 4s\n",
      " 27400K .......... .......... .......... .......... .......... 34%  114M 4s\n",
      " 27450K .......... .......... .......... .......... .......... 34%  118M 4s\n",
      " 27500K .......... .......... .......... .......... .......... 34%  117M 4s\n",
      " 27550K .......... .......... .......... .......... .......... 34%  118M 4s\n",
      " 27600K .......... .......... .......... .......... .......... 34% 87,8M 4s\n",
      " 27650K .......... .......... .......... .......... .......... 34%  117M 3s\n",
      " 27700K .......... .......... .......... .......... .......... 34%  118M 3s\n",
      " 27750K .......... .......... .......... .......... .......... 34%  113M 3s\n",
      " 27800K .......... .......... .......... .......... .......... 34% 1,20M 4s\n",
      " 27850K .......... .......... .......... .......... .......... 34% 64,8M 4s\n",
      " 27900K .......... .......... .......... .......... .......... 34% 1,04M 4s\n",
      " 27950K .......... .......... .......... .......... .......... 35% 11,1M 4s\n",
      " 28000K .......... .......... .......... .......... .......... 35% 16,4M 4s\n",
      " 28050K .......... .......... .......... .......... .......... 35% 71,3M 4s\n",
      " 28100K .......... .......... .......... .......... .......... 35%  128M 4s\n",
      " 28150K .......... .......... .......... .......... .......... 35%  120M 4s\n",
      " 28200K .......... .......... .......... .......... .......... 35% 57,3M 4s\n",
      " 28250K .......... .......... .......... .......... .......... 35%  102M 4s\n",
      " 28300K .......... .......... .......... .......... .......... 35%  116M 4s\n",
      " 28350K .......... .......... .......... .......... .......... 35%  101M 4s\n",
      " 28400K .......... .......... .......... .......... .......... 35% 92,0M 4s\n",
      " 28450K .......... .......... .......... .......... .......... 35% 86,5M 4s\n",
      " 28500K .......... .......... .......... .......... .......... 35%  115M 4s\n",
      " 28550K .......... .......... .......... .......... .......... 35%  122M 4s\n",
      " 28600K .......... .......... .......... .......... .......... 35% 80,6M 4s\n",
      " 28650K .......... .......... .......... .......... .......... 35% 64,7M 3s\n",
      " 28700K .......... .......... .......... .......... .......... 35%  118M 3s\n",
      " 28750K .......... .......... .......... .......... .......... 36%  116M 3s\n",
      " 28800K .......... .......... .......... .......... .......... 36% 75,4M 3s\n",
      " 28850K .......... .......... .......... .......... .......... 36%  168M 3s\n",
      " 28900K .......... .......... .......... .......... .......... 36%  115M 3s\n",
      " 28950K .......... .......... .......... .......... .......... 36% 83,6M 3s\n",
      " 29000K .......... .......... .......... .......... .......... 36% 34,0M 3s\n",
      " 29050K .......... .......... .......... .......... .......... 36%  130M 3s\n",
      " 29100K .......... .......... .......... .......... .......... 36%  103M 3s\n",
      " 29150K .......... .......... .......... .......... .......... 36%  117M 3s\n",
      " 29200K .......... .......... .......... .......... .......... 36% 93,4M 3s\n",
      " 29250K .......... .......... .......... .......... .......... 36% 67,4M 3s\n",
      " 29300K .......... .......... .......... .......... .......... 36%  130M 3s\n",
      " 29350K .......... .......... .......... .......... .......... 36%  135M 3s\n",
      " 29400K .......... .......... .......... .......... .......... 36%  132M 3s\n",
      " 29450K .......... .......... .......... .......... .......... 36%  165M 3s\n",
      " 29500K .......... .......... .......... .......... .......... 36%  121M 3s\n",
      " 29550K .......... .......... .......... .......... .......... 37%  117M 3s\n",
      " 29600K .......... .......... .......... .......... .......... 37% 92,5M 3s\n",
      " 29650K .......... .......... .......... .......... .......... 37%  119M 3s\n",
      " 29700K .......... .......... .......... .......... .......... 37%  118M 3s\n",
      " 29750K .......... .......... .......... .......... .......... 37%  118M 3s\n",
      " 29800K .......... .......... .......... .......... .......... 37%  109M 3s\n",
      " 29850K .......... .......... .......... .......... .......... 37%  115M 3s\n",
      " 29900K .......... .......... .......... .......... .......... 37%  117M 3s\n",
      " 29950K .......... .......... .......... .......... .......... 37%  117M 3s\n",
      " 30000K .......... .......... .......... .......... .......... 37% 93,0M 3s\n",
      " 30050K .......... .......... .......... .......... .......... 37%  117M 3s\n",
      " 30100K .......... .......... .......... .......... .......... 37%  116M 3s\n",
      " 30150K .......... .......... .......... .......... .......... 37%  117M 3s\n",
      " 30200K .......... .......... .......... .......... .......... 37%  113M 3s\n",
      " 30250K .......... .......... .......... .......... .......... 37%  115M 3s\n",
      " 30300K .......... .......... .......... .......... .......... 37%  117M 3s\n",
      " 30350K .......... .......... .......... .......... .......... 38%  120M 3s\n",
      " 30400K .......... .......... .......... .......... .......... 38% 88,4M 3s\n",
      " 30450K .......... .......... .......... .......... .......... 38%  117M 3s\n",
      " 30500K .......... .......... .......... .......... .......... 38%  118M 3s\n",
      " 30550K .......... .......... .......... .......... .......... 38%  118M 3s\n",
      " 30600K .......... .......... .......... .......... .......... 38%  114M 3s\n",
      " 30650K .......... .......... .......... .......... .......... 38%  112M 3s\n",
      " 30700K .......... .......... .......... .......... .......... 38%  119M 3s\n",
      " 30750K .......... .......... .......... .......... .......... 38%  119M 3s\n",
      " 30800K .......... .......... .......... .......... .......... 38% 87,9M 3s\n",
      " 30850K .......... .......... .......... .......... .......... 38%  118M 3s\n",
      " 30900K .......... .......... .......... .......... .......... 38%  579K 3s\n",
      " 30950K .......... .......... .......... .......... .......... 38% 18,0M 3s\n",
      " 31000K .......... .......... .......... .......... .......... 38% 12,9M 3s\n",
      " 31050K .......... .......... .......... .......... .......... 38% 29,7M 3s\n",
      " 31100K .......... .......... .......... .......... .......... 38% 56,8M 3s\n",
      " 31150K .......... .......... .......... .......... .......... 39%  114M 3s\n",
      " 31200K .......... .......... .......... .......... .......... 39% 57,0M 3s\n",
      " 31250K .......... .......... .......... .......... .......... 39% 50,1M 3s\n",
      " 31300K .......... .......... .......... .......... .......... 39% 65,0M 3s\n",
      " 31350K .......... .......... .......... .......... .......... 39%  117M 3s\n",
      " 31400K .......... .......... .......... .......... .......... 39%  106M 3s\n",
      " 31450K .......... .......... .......... .......... .......... 39%  119M 3s\n",
      " 31500K .......... .......... .......... .......... .......... 39%  115M 3s\n",
      " 31550K .......... .......... .......... .......... .......... 39%  107M 3s\n",
      " 31600K .......... .......... .......... .......... .......... 39%  103M 3s\n",
      " 31650K .......... .......... .......... .......... .......... 39%  113M 3s\n",
      " 31700K .......... .......... .......... .......... .......... 39%  117M 3s\n",
      " 31750K .......... .......... .......... .......... .......... 39% 76,2M 3s\n",
      " 31800K .......... .......... .......... .......... .......... 39%  130M 3s\n",
      " 31850K .......... .......... .......... .......... .......... 39%  118M 3s\n",
      " 31900K .......... .......... .......... .......... .......... 39%  114M 3s\n",
      " 31950K .......... .......... .......... .......... .......... 40%  119M 3s\n",
      " 32000K .......... .......... .......... .......... .......... 40% 93,6M 3s\n",
      " 32050K .......... .......... .......... .......... .......... 40% 62,4M 3s\n",
      " 32100K .......... .......... .......... .......... .......... 40% 28,0M 3s\n",
      " 32150K .......... .......... .......... .......... .......... 40%  117M 3s\n",
      " 32200K .......... .......... .......... .......... .......... 40% 54,2M 3s\n",
      " 32250K .......... .......... .......... .......... .......... 40%  118M 3s\n",
      " 32300K .......... .......... .......... .......... .......... 40%  113M 3s\n",
      " 32350K .......... .......... .......... .......... .......... 40% 76,9M 3s\n",
      " 32400K .......... .......... .......... .......... .......... 40%  158M 3s\n",
      " 32450K .......... .......... .......... .......... .......... 40%  114M 3s\n",
      " 32500K .......... .......... .......... .......... .......... 40%  118M 3s\n",
      " 32550K .......... .......... .......... .......... .......... 40%  117M 3s\n",
      " 32600K .......... .......... .......... .......... .......... 40%  114M 3s\n",
      " 32650K .......... .......... .......... .......... .......... 40%  117M 3s\n",
      " 32700K .......... .......... .......... .......... .......... 41%  117M 3s\n",
      " 32750K .......... .......... .......... .......... .......... 41%  114M 3s\n",
      " 32800K .......... .......... .......... .......... .......... 41% 92,0M 3s\n",
      " 32850K .......... .......... .......... .......... .......... 41% 90,8M 3s\n",
      " 32900K .......... .......... .......... .......... .......... 41%  159M 3s\n",
      " 32950K .......... .......... .......... .......... .......... 41%  118M 3s\n",
      " 33000K .......... .......... .......... .......... .......... 41%  114M 3s\n",
      " 33050K .......... .......... .......... .......... .......... 41%  116M 3s\n",
      " 33100K .......... .......... .......... .......... .......... 41%  119M 3s\n",
      " 33150K .......... .......... .......... .......... .......... 41%  113M 3s\n",
      " 33200K .......... .......... .......... .......... .......... 41% 91,6M 3s\n",
      " 33250K .......... .......... .......... .......... .......... 41%  118M 3s\n",
      " 33300K .......... .......... .......... .......... .......... 41%  112M 3s\n",
      " 33350K .......... .......... .......... .......... .......... 41%  119M 3s\n",
      " 33400K .......... .......... .......... .......... .......... 41%  114M 3s\n",
      " 33450K .......... .......... .......... .......... .......... 41%  118M 3s\n",
      " 33500K .......... .......... .......... .......... .......... 42%  109M 3s\n",
      " 33550K .......... .......... .......... .......... .......... 42%  122M 3s\n",
      " 33600K .......... .......... .......... .......... .......... 42% 91,4M 3s\n",
      " 33650K .......... .......... .......... .......... .......... 42%  117M 3s\n",
      " 33700K .......... .......... .......... .......... .......... 42%  114M 3s\n",
      " 33750K .......... .......... .......... .......... .......... 42%  117M 3s\n",
      " 33800K .......... .......... .......... .......... .......... 42%  115M 3s\n",
      " 33850K .......... .......... .......... .......... .......... 42%  118M 3s\n",
      " 33900K .......... .......... .......... .......... .......... 42%  117M 3s\n",
      " 33950K .......... .......... .......... .......... .......... 42%  590K 3s\n",
      " 34000K .......... .......... .......... .......... .......... 42% 12,0M 3s\n",
      " 34050K .......... .......... .......... .......... .......... 42% 10,3M 3s\n",
      " 34100K .......... .......... .......... .......... .......... 42%  118M 3s\n",
      " 34150K .......... .......... .......... .......... .......... 42% 48,5M 3s\n",
      " 34200K .......... .......... .......... .......... .......... 42%  107M 3s\n",
      " 34250K .......... .......... .......... .......... .......... 42% 41,4M 3s\n",
      " 34300K .......... .......... .......... .......... .......... 43%  109M 3s\n",
      " 34350K .......... .......... .......... .......... .......... 43% 87,7M 3s\n",
      " 34400K .......... .......... .......... .......... .......... 43% 92,6M 3s\n",
      " 34450K .......... .......... .......... .......... .......... 43% 84,8M 3s\n",
      " 34500K .......... .......... .......... .......... .......... 43%  117M 3s\n",
      " 34550K .......... .......... .......... .......... .......... 43%  112M 3s\n",
      " 34600K .......... .......... .......... .......... .......... 43%  108M 3s\n",
      " 34650K .......... .......... .......... .......... .......... 43%  120M 3s\n",
      " 34700K .......... .......... .......... .......... .......... 43%  119M 3s\n",
      " 34750K .......... .......... .......... .......... .......... 43%  117M 3s\n",
      " 34800K .......... .......... .......... .......... .......... 43% 90,0M 3s\n",
      " 34850K .......... .......... .......... .......... .......... 43%  121M 3s\n",
      " 34900K .......... .......... .......... .......... .......... 43% 69,7M 3s\n",
      " 34950K .......... .......... .......... .......... .......... 43%  166M 3s\n",
      " 35000K .......... .......... .......... .......... .......... 43%  103M 3s\n",
      " 35050K .......... .......... .......... .......... .......... 43%  118M 3s\n",
      " 35100K .......... .......... .......... .......... .......... 44%  118M 3s\n",
      " 35150K .......... .......... .......... .......... .......... 44%  116M 3s\n",
      " 35200K .......... .......... .......... .......... .......... 44% 32,5M 3s\n",
      " 35250K .......... .......... .......... .......... .......... 44% 28,5M 3s\n",
      " 35300K .......... .......... .......... .......... .......... 44%  116M 3s\n",
      " 35350K .......... .......... .......... .......... .......... 44%  116M 3s\n",
      " 35400K .......... .......... .......... .......... .......... 44%  107M 3s\n",
      " 35450K .......... .......... .......... .......... .......... 44%  120M 3s\n",
      " 35500K .......... .......... .......... .......... .......... 44%  120M 3s\n",
      " 35550K .......... .......... .......... .......... .......... 44%  115M 3s\n",
      " 35600K .......... .......... .......... .......... .......... 44% 92,0M 3s\n",
      " 35650K .......... .......... .......... .......... .......... 44%  119M 3s\n",
      " 35700K .......... .......... .......... .......... .......... 44%  116M 3s\n",
      " 35750K .......... .......... .......... .......... .......... 44%  118M 3s\n",
      " 35800K .......... .......... .......... .......... .......... 44%  111M 3s\n",
      " 35850K .......... .......... .......... .......... .......... 44%  117M 3s\n",
      " 35900K .......... .......... .......... .......... .......... 45%  119M 3s\n",
      " 35950K .......... .......... .......... .......... .......... 45%  115M 3s\n",
      " 36000K .......... .......... .......... .......... .......... 45% 89,3M 3s\n",
      " 36050K .......... .......... .......... .......... .......... 45%  118M 3s\n",
      " 36100K .......... .......... .......... .......... .......... 45%  118M 3s\n",
      " 36150K .......... .......... .......... .......... .......... 45%  120M 3s\n",
      " 36200K .......... .......... .......... .......... .......... 45%  110M 3s\n",
      " 36250K .......... .......... .......... .......... .......... 45%  117M 3s\n",
      " 36300K .......... .......... .......... .......... .......... 45%  117M 3s\n",
      " 36350K .......... .......... .......... .......... .......... 45%  117M 3s\n",
      " 36400K .......... .......... .......... .......... .......... 45% 89,6M 3s\n",
      " 36450K .......... .......... .......... .......... .......... 45%  117M 3s\n",
      " 36500K .......... .......... .......... .......... .......... 45%  119M 3s\n",
      " 36550K .......... .......... .......... .......... .......... 45%  118M 3s\n",
      " 36600K .......... .......... .......... .......... .......... 45%  110M 3s\n",
      " 36650K .......... .......... .......... .......... .......... 45%  117M 3s\n",
      " 36700K .......... .......... .......... .......... .......... 46%  118M 3s\n",
      " 36750K .......... .......... .......... .......... .......... 46%  118M 3s\n",
      " 36800K .......... .......... .......... .......... .......... 46% 88,9M 3s\n",
      " 36850K .......... .......... .......... .......... .......... 46%  118M 3s\n",
      " 36900K .......... .......... .......... .......... .......... 46%  117M 3s\n",
      " 36950K .......... .......... .......... .......... .......... 46%  118M 3s\n",
      " 37000K .......... .......... .......... .......... .......... 46%  600K 3s\n",
      " 37050K .......... .......... .......... .......... .......... 46% 13,1M 3s\n",
      " 37100K .......... .......... .......... .......... .......... 46% 9,64M 3s\n",
      " 37150K .......... .......... .......... .......... .......... 46% 30,9M 3s\n",
      " 37200K .......... .......... .......... .......... .......... 46% 37,8M 3s\n",
      " 37250K .......... .......... .......... .......... .......... 46%  114M 3s\n",
      " 37300K .......... .......... .......... .......... .......... 46%  111M 3s\n",
      " 37350K .......... .......... .......... .......... .......... 46% 60,0M 3s\n",
      " 37400K .......... .......... .......... .......... .......... 46%  132M 3s\n",
      " 37450K .......... .......... .......... .......... .......... 46% 52,8M 3s\n",
      " 37500K .......... .......... .......... .......... .......... 47%  133M 3s\n",
      " 37550K .......... .......... .......... .......... .......... 47%  117M 3s\n",
      " 37600K .......... .......... .......... .......... .......... 47% 93,2M 3s\n",
      " 37650K .......... .......... .......... .......... .......... 47%  112M 3s\n",
      " 37700K .......... .......... .......... .......... .......... 47%  122M 3s\n",
      " 37750K .......... .......... .......... .......... .......... 47%  117M 3s\n",
      " 37800K .......... .......... .......... .......... .......... 47%  110M 3s\n",
      " 37850K .......... .......... .......... .......... .......... 47%  113M 3s\n",
      " 37900K .......... .......... .......... .......... .......... 47%  121M 3s\n",
      " 37950K .......... .......... .......... .......... .......... 47%  116M 3s\n",
      " 38000K .......... .......... .......... .......... .......... 47% 91,5M 3s\n",
      " 38050K .......... .......... .......... .......... .......... 47%  117M 3s\n",
      " 38100K .......... .......... .......... .......... .......... 47%  118M 3s\n",
      " 38150K .......... .......... .......... .......... .......... 47%  115M 3s\n",
      " 38200K .......... .......... .......... .......... .......... 47%  113M 3s\n",
      " 38250K .......... .......... .......... .......... .......... 47% 50,3M 3s\n",
      " 38300K .......... .......... .......... .......... .......... 48% 20,9M 3s\n",
      " 38350K .......... .......... .......... .......... .......... 48%  117M 3s\n",
      " 38400K .......... .......... .......... .......... .......... 48% 58,9M 3s\n",
      " 38450K .......... .......... .......... .......... .......... 48%  118M 3s\n",
      " 38500K .......... .......... .......... .......... .......... 48%  117M 3s\n",
      " 38550K .......... .......... .......... .......... .......... 48% 99,0M 2s\n",
      " 38600K .......... .......... .......... .......... .......... 48%  122M 2s\n",
      " 38650K .......... .......... .......... .......... .......... 48%  122M 2s\n",
      " 38700K .......... .......... .......... .......... .......... 48%  111M 2s\n",
      " 38750K .......... .......... .......... .......... .......... 48%  114M 2s\n",
      " 38800K .......... .......... .......... .......... .......... 48% 99,9M 2s\n",
      " 38850K .......... .......... .......... .......... .......... 48%  110M 2s\n",
      " 38900K .......... .......... .......... .......... .......... 48%  121M 2s\n",
      " 38950K .......... .......... .......... .......... .......... 48%  115M 2s\n",
      " 39000K .......... .......... .......... .......... .......... 48% 61,7M 2s\n",
      " 39050K .......... .......... .......... .......... .......... 48%  174M 2s\n",
      " 39100K .......... .......... .......... .......... .......... 49%  173M 2s\n",
      " 39150K .......... .......... .......... .......... .......... 49%  141M 2s\n",
      " 39200K .......... .......... .......... .......... .......... 49% 93,9M 2s\n",
      " 39250K .......... .......... .......... .......... .......... 49%  116M 2s\n",
      " 39300K .......... .......... .......... .......... .......... 49%  117M 2s\n",
      " 39350K .......... .......... .......... .......... .......... 49%  117M 2s\n",
      " 39400K .......... .......... .......... .......... .......... 49%  109M 2s\n",
      " 39450K .......... .......... .......... .......... .......... 49%  122M 2s\n",
      " 39500K .......... .......... .......... .......... .......... 49%  113M 2s\n",
      " 39550K .......... .......... .......... .......... .......... 49%  117M 2s\n",
      " 39600K .......... .......... .......... .......... .......... 49% 93,0M 2s\n",
      " 39650K .......... .......... .......... .......... .......... 49%  113M 2s\n",
      " 39700K .......... .......... .......... .......... .......... 49%  117M 2s\n",
      " 39750K .......... .......... .......... .......... .......... 49%  118M 2s\n",
      " 39800K .......... .......... .......... .......... .......... 49%  114M 2s\n",
      " 39850K .......... .......... .......... .......... .......... 49%  118M 2s\n",
      " 39900K .......... .......... .......... .......... .......... 50%  114M 2s\n",
      " 39950K .......... .......... .......... .......... .......... 50%  117M 2s\n",
      " 40000K .......... .......... .......... .......... .......... 50% 1,35M 2s\n",
      " 40050K .......... .......... .......... .......... .......... 50% 1,02M 2s\n",
      " 40100K .......... .......... .......... .......... .......... 50% 18,7M 2s\n",
      " 40150K .......... .......... .......... .......... .......... 50% 8,48M 2s\n",
      " 40200K .......... .......... .......... .......... .......... 50% 25,4M 2s\n",
      " 40250K .......... .......... .......... .......... .......... 50% 54,6M 2s\n",
      " 40300K .......... .......... .......... .......... .......... 50%  123M 2s\n",
      " 40350K .......... .......... .......... .......... .......... 50% 71,9M 2s\n",
      " 40400K .......... .......... .......... .......... .......... 50%  109M 2s\n",
      " 40450K .......... .......... .......... .......... .......... 50%  118M 2s\n",
      " 40500K .......... .......... .......... .......... .......... 50%  116M 2s\n",
      " 40550K .......... .......... .......... .......... .......... 50%  117M 2s\n",
      " 40600K .......... .......... .......... .......... .......... 50%  109M 2s\n",
      " 40650K .......... .......... .......... .......... .......... 50% 37,4M 2s\n",
      " 40700K .......... .......... .......... .......... .......... 51% 57,3M 2s\n",
      " 40750K .......... .......... .......... .......... .......... 51% 90,9M 2s\n",
      " 40800K .......... .......... .......... .......... .......... 51% 49,8M 2s\n",
      " 40850K .......... .......... .......... .......... .......... 51%  178M 2s\n",
      " 40900K .......... .......... .......... .......... .......... 51% 91,1M 2s\n",
      " 40950K .......... .......... .......... .......... .......... 51% 57,1M 2s\n",
      " 41000K .......... .......... .......... .......... .......... 51%  119M 2s\n",
      " 41050K .......... .......... .......... .......... .......... 51%  113M 2s\n",
      " 41100K .......... .......... .......... .......... .......... 51% 51,5M 2s\n",
      " 41150K .......... .......... .......... .......... .......... 51% 90,7M 2s\n",
      " 41200K .......... .......... .......... .......... .......... 51% 49,9M 2s\n",
      " 41250K .......... .......... .......... .......... .......... 51% 93,4M 2s\n",
      " 41300K .......... .......... .......... .......... .......... 51%  113M 2s\n",
      " 41350K .......... .......... .......... .......... .......... 51% 23,8M 2s\n",
      " 41400K .......... .......... .......... .......... .......... 51% 55,9M 2s\n",
      " 41450K .......... .......... .......... .......... .......... 51% 91,7M 2s\n",
      " 41500K .......... .......... .......... .......... .......... 52%  120M 2s\n",
      " 41550K .......... .......... .......... .......... .......... 52% 56,1M 2s\n",
      " 41600K .......... .......... .......... .......... .......... 52% 91,1M 2s\n",
      " 41650K .......... .......... .......... .......... .......... 52%  117M 2s\n",
      " 41700K .......... .......... .......... .......... .......... 52%  114M 2s\n",
      " 41750K .......... .......... .......... .......... .......... 52%  118M 2s\n",
      " 41800K .......... .......... .......... .......... .......... 52% 50,2M 2s\n",
      " 41850K .......... .......... .......... .......... .......... 52%  118M 2s\n",
      " 41900K .......... .......... .......... .......... .......... 52% 32,3M 2s\n",
      " 41950K .......... .......... .......... .......... .......... 52% 93,3M 2s\n",
      " 42000K .......... .......... .......... .......... .......... 52% 49,1M 2s\n",
      " 42050K .......... .......... .......... .......... .......... 52%  117M 2s\n",
      " 42100K .......... .......... .......... .......... .......... 52%  117M 2s\n",
      " 42150K .......... .......... .......... .......... .......... 52% 91,1M 2s\n",
      " 42200K .......... .......... .......... .......... .......... 52%  115M 2s\n",
      " 42250K .......... .......... .......... .......... .......... 52% 55,3M 2s\n",
      " 42300K .......... .......... .......... .......... .......... 53%  124M 2s\n",
      " 42350K .......... .......... .......... .......... .......... 53% 93,2M 2s\n",
      " 42400K .......... .......... .......... .......... .......... 53% 49,2M 2s\n",
      " 42450K .......... .......... .......... .......... .......... 53%  117M 2s\n",
      " 42500K .......... .......... .......... .......... .......... 53%  117M 2s\n",
      " 42550K .......... .......... .......... .......... .......... 53% 50,0M 2s\n",
      " 42600K .......... .......... .......... .......... .......... 53%  116M 2s\n",
      " 42650K .......... .......... .......... .......... .......... 53% 50,4M 2s\n",
      " 42700K .......... .......... .......... .......... .......... 53%  119M 2s\n",
      " 42750K .......... .......... .......... .......... .......... 53% 93,4M 2s\n",
      " 42800K .......... .......... .......... .......... .......... 53% 49,0M 2s\n",
      " 42850K .......... .......... .......... .......... .......... 53%  117M 2s\n",
      " 42900K .......... .......... .......... .......... .......... 53% 93,3M 2s\n",
      " 42950K .......... .......... .......... .......... .......... 53%  114M 2s\n",
      " 43000K .......... .......... .......... .......... .......... 53% 56,6M 2s\n",
      " 43050K .......... .......... .......... .......... .......... 53% 1,83M 2s\n",
      " 43100K .......... .......... .......... .......... .......... 54% 1,03M 2s\n",
      " 43150K .......... .......... .......... .......... .......... 54% 12,5M 2s\n",
      " 43200K .......... .......... .......... .......... .......... 54% 8,84M 2s\n",
      " 43250K .......... .......... .......... .......... .......... 54% 17,2M 2s\n",
      " 43300K .......... .......... .......... .......... .......... 54% 50,8M 2s\n",
      " 43350K .......... .......... .......... .......... .......... 54% 98,2M 2s\n",
      " 43400K .......... .......... .......... .......... .......... 54% 99,6M 2s\n",
      " 43450K .......... .......... .......... .......... .......... 54%  114M 2s\n",
      " 43500K .......... .......... .......... .......... .......... 54%  112M 2s\n",
      " 43550K .......... .......... .......... .......... .......... 54%  115M 2s\n",
      " 43600K .......... .......... .......... .......... .......... 54% 91,5M 2s\n",
      " 43650K .......... .......... .......... .......... .......... 54%  114M 2s\n",
      " 43700K .......... .......... .......... .......... .......... 54%  109M 2s\n",
      " 43750K .......... .......... .......... .......... .......... 54%  110M 2s\n",
      " 43800K .......... .......... .......... .......... .......... 54% 47,2M 2s\n",
      " 43850K .......... .......... .......... .......... .......... 54% 75,5M 2s\n",
      " 43900K .......... .......... .......... .......... .......... 55%  104M 2s\n",
      " 43950K .......... .......... .......... .......... .......... 55% 99,4M 2s\n",
      " 44000K .......... .......... .......... .......... .......... 55% 46,6M 2s\n",
      " 44050K .......... .......... .......... .......... .......... 55%  124M 2s\n",
      " 44100K .......... .......... .......... .......... .......... 55% 75,8M 2s\n",
      " 44150K .......... .......... .......... .......... .......... 55%  114M 2s\n",
      " 44200K .......... .......... .......... .......... .......... 55% 70,8M 2s\n",
      " 44250K .......... .......... .......... .......... .......... 55% 58,1M 2s\n",
      " 44300K .......... .......... .......... .......... .......... 55% 69,4M 2s\n",
      " 44350K .......... .......... .......... .......... .......... 55%  125M 2s\n",
      " 44400K .......... .......... .......... .......... .......... 55% 36,0M 2s\n",
      " 44450K .......... .......... .......... .......... .......... 55% 64,7M 2s\n",
      " 44500K .......... .......... .......... .......... .......... 55% 65,2M 2s\n",
      " 44550K .......... .......... .......... .......... .......... 55% 56,5M 2s\n",
      " 44600K .......... .......... .......... .......... .......... 55%  109M 2s\n",
      " 44650K .......... .......... .......... .......... .......... 55% 65,3M 2s\n",
      " 44700K .......... .......... .......... .......... .......... 56%  124M 2s\n",
      " 44750K .......... .......... .......... .......... .......... 56%  120M 2s\n",
      " 44800K .......... .......... .......... .......... .......... 56% 55,3M 2s\n",
      " 44850K .......... .......... .......... .......... .......... 56%  117M 2s\n",
      " 44900K .......... .......... .......... .......... .......... 56%  117M 2s\n",
      " 44950K .......... .......... .......... .......... .......... 56% 34,1M 2s\n",
      " 45000K .......... .......... .......... .......... .......... 56% 54,5M 2s\n",
      " 45050K .......... .......... .......... .......... .......... 56% 98,0M 2s\n",
      " 45100K .......... .......... .......... .......... .......... 56%  118M 2s\n",
      " 45150K .......... .......... .......... .......... .......... 56% 88,9M 2s\n",
      " 45200K .......... .......... .......... .......... .......... 56% 80,4M 2s\n",
      " 45250K .......... .......... .......... .......... .......... 56% 56,4M 2s\n",
      " 45300K .......... .......... .......... .......... .......... 56%  117M 2s\n",
      " 45350K .......... .......... .......... .......... .......... 56%  117M 2s\n",
      " 45400K .......... .......... .......... .......... .......... 56%  114M 2s\n",
      " 45450K .......... .......... .......... .......... .......... 56% 67,8M 2s\n",
      " 45500K .......... .......... .......... .......... .......... 57%  117M 2s\n",
      " 45550K .......... .......... .......... .......... .......... 57%  117M 2s\n",
      " 45600K .......... .......... .......... .......... .......... 57% 59,2M 2s\n",
      " 45650K .......... .......... .......... .......... .......... 57% 70,0M 2s\n",
      " 45700K .......... .......... .......... .......... .......... 57% 57,2M 2s\n",
      " 45750K .......... .......... .......... .......... .......... 57% 74,1M 2s\n",
      " 45800K .......... .......... .......... .......... .......... 57%  113M 2s\n",
      " 45850K .......... .......... .......... .......... .......... 57% 94,3M 2s\n",
      " 45900K .......... .......... .......... .......... .......... 57% 95,5M 2s\n",
      " 45950K .......... .......... .......... .......... .......... 57% 69,3M 2s\n",
      " 46000K .......... .......... .......... .......... .......... 57% 46,2M 2s\n",
      " 46050K .......... .......... .......... .......... .......... 57% 1,81M 2s\n",
      " 46100K .......... .......... .......... .......... .......... 57% 1,05M 2s\n",
      " 46150K .......... .......... .......... .......... .......... 57% 12,9M 2s\n",
      " 46200K .......... .......... .......... .......... .......... 57% 7,89M 2s\n",
      " 46250K .......... .......... .......... .......... .......... 57% 17,7M 2s\n",
      " 46300K .......... .......... .......... .......... .......... 58% 29,9M 2s\n",
      " 46350K .......... .......... .......... .......... .......... 58% 53,6M 2s\n",
      " 46400K .......... .......... .......... .......... .......... 58% 31,1M 2s\n",
      " 46450K .......... .......... .......... .......... .......... 58%  107M 2s\n",
      " 46500K .......... .......... .......... .......... .......... 58% 97,9M 2s\n",
      " 46550K .......... .......... .......... .......... .......... 58%  160M 2s\n",
      " 46600K .......... .......... .......... .......... .......... 58%  150M 2s\n",
      " 46650K .......... .......... .......... .......... .......... 58%  163M 2s\n",
      " 46700K .......... .......... .......... .......... .......... 58%  133M 2s\n",
      " 46750K .......... .......... .......... .......... .......... 58%  116M 2s\n",
      " 46800K .......... .......... .......... .......... .......... 58% 94,7M 2s\n",
      " 46850K .......... .......... .......... .......... .......... 58%  113M 2s\n",
      " 46900K .......... .......... .......... .......... .......... 58%  119M 2s\n",
      " 46950K .......... .......... .......... .......... .......... 58% 66,6M 2s\n",
      " 47000K .......... .......... .......... .......... .......... 58%  109M 2s\n",
      " 47050K .......... .......... .......... .......... .......... 58% 96,9M 2s\n",
      " 47100K .......... .......... .......... .......... .......... 59% 67,5M 2s\n",
      " 47150K .......... .......... .......... .......... .......... 59%  118M 2s\n",
      " 47200K .......... .......... .......... .......... .......... 59% 61,6M 2s\n",
      " 47250K .......... .......... .......... .......... .......... 59%  100M 2s\n",
      " 47300K .......... .......... .......... .......... .......... 59%  118M 2s\n",
      " 47350K .......... .......... .......... .......... .......... 59% 75,9M 2s\n",
      " 47400K .......... .......... .......... .......... .......... 59% 43,6M 2s\n",
      " 47450K .......... .......... .......... .......... .......... 59% 55,6M 2s\n",
      " 47500K .......... .......... .......... .......... .......... 59%  115M 2s\n",
      " 47550K .......... .......... .......... .......... .......... 59% 48,5M 2s\n",
      " 47600K .......... .......... .......... .......... .......... 59% 50,9M 2s\n",
      " 47650K .......... .......... .......... .......... .......... 59% 58,5M 2s\n",
      " 47700K .......... .......... .......... .......... .......... 59%  118M 2s\n",
      " 47750K .......... .......... .......... .......... .......... 59%  116M 2s\n",
      " 47800K .......... .......... .......... .......... .......... 59%  112M 2s\n",
      " 47850K .......... .......... .......... .......... .......... 59% 85,3M 2s\n",
      " 47900K .......... .......... .......... .......... .......... 60%  118M 2s\n",
      " 47950K .......... .......... .......... .......... .......... 60%  115M 2s\n",
      " 48000K .......... .......... .......... .......... .......... 60% 72,0M 2s\n",
      " 48050K .......... .......... .......... .......... .......... 60% 50,2M 2s\n",
      " 48100K .......... .......... .......... .......... .......... 60% 43,5M 2s\n",
      " 48150K .......... .......... .......... .......... .......... 60% 29,3M 2s\n",
      " 48200K .......... .......... .......... .......... .......... 60%  112M 2s\n",
      " 48250K .......... .......... .......... .......... .......... 60%  117M 2s\n",
      " 48300K .......... .......... .......... .......... .......... 60%  118M 2s\n",
      " 48350K .......... .......... .......... .......... .......... 60%  114M 2s\n",
      " 48400K .......... .......... .......... .......... .......... 60% 93,3M 2s\n",
      " 48450K .......... .......... .......... .......... .......... 60%  110M 2s\n",
      " 48500K .......... .......... .......... .......... .......... 60%  120M 2s\n",
      " 48550K .......... .......... .......... .......... .......... 60%  118M 2s\n",
      " 48600K .......... .......... .......... .......... .......... 60%  113M 2s\n",
      " 48650K .......... .......... .......... .......... .......... 60% 66,9M 2s\n",
      " 48700K .......... .......... .......... .......... .......... 61%  119M 2s\n",
      " 48750K .......... .......... .......... .......... .......... 61%  115M 2s\n",
      " 48800K .......... .......... .......... .......... .......... 61% 79,8M 2s\n",
      " 48850K .......... .......... .......... .......... .......... 61% 70,2M 2s\n",
      " 48900K .......... .......... .......... .......... .......... 61% 59,7M 2s\n",
      " 48950K .......... .......... .......... .......... .......... 61%  152M 2s\n",
      " 49000K .......... .......... .......... .......... .......... 61%  114M 2s\n",
      " 49050K .......... .......... .......... .......... .......... 61% 57,1M 2s\n",
      " 49100K .......... .......... .......... .......... .......... 61% 1,77M 2s\n",
      " 49150K .......... .......... .......... .......... .......... 61% 1,05M 2s\n",
      " 49200K .......... .......... .......... .......... .......... 61% 10,3M 2s\n",
      " 49250K .......... .......... .......... .......... .......... 61% 7,22M 2s\n",
      " 49300K .......... .......... .......... .......... .......... 61% 41,2M 2s\n",
      " 49350K .......... .......... .......... .......... .......... 61% 24,9M 2s\n",
      " 49400K .......... .......... .......... .......... .......... 61% 39,5M 2s\n",
      " 49450K .......... .......... .......... .......... .......... 61% 50,1M 2s\n",
      " 49500K .......... .......... .......... .......... .......... 62%  117M 2s\n",
      " 49550K .......... .......... .......... .......... .......... 62% 37,8M 2s\n",
      " 49600K .......... .......... .......... .......... .......... 62% 56,7M 2s\n",
      " 49650K .......... .......... .......... .......... .......... 62%  118M 2s\n",
      " 49700K .......... .......... .......... .......... .......... 62%  117M 2s\n",
      " 49750K .......... .......... .......... .......... .......... 62%  114M 2s\n",
      " 49800K .......... .......... .......... .......... .......... 62%  110M 2s\n",
      " 49850K .......... .......... .......... .......... .......... 62%  117M 2s\n",
      " 49900K .......... .......... .......... .......... .......... 62% 88,7M 2s\n",
      " 49950K .......... .......... .......... .......... .......... 62%  146M 2s\n",
      " 50000K .......... .......... .......... .......... .......... 62% 99,9M 2s\n",
      " 50050K .......... .......... .......... .......... .......... 62%  120M 2s\n",
      " 50100K .......... .......... .......... .......... .......... 62%  117M 2s\n",
      " 50150K .......... .......... .......... .......... .......... 62%  116M 2s\n",
      " 50200K .......... .......... .......... .......... .......... 62%  114M 2s\n",
      " 50250K .......... .......... .......... .......... .......... 62%  116M 2s\n",
      " 50300K .......... .......... .......... .......... .......... 63%  117M 2s\n",
      " 50350K .......... .......... .......... .......... .......... 63%  117M 2s\n",
      " 50400K .......... .......... .......... .......... .......... 63% 89,0M 2s\n",
      " 50450K .......... .......... .......... .......... .......... 63% 93,8M 2s\n",
      " 50500K .......... .......... .......... .......... .......... 63% 56,8M 2s\n",
      " 50550K .......... .......... .......... .......... .......... 63% 57,6M 2s\n",
      " 50600K .......... .......... .......... .......... .......... 63%  113M 2s\n",
      " 50650K .......... .......... .......... .......... .......... 63% 56,8M 2s\n",
      " 50700K .......... .......... .......... .......... .......... 63% 51,7M 2s\n",
      " 50750K .......... .......... .......... .......... .......... 63% 60,6M 2s\n",
      " 50800K .......... .......... .......... .......... .......... 63% 48,3M 2s\n",
      " 50850K .......... .......... .......... .......... .......... 63% 81,4M 2s\n",
      " 50900K .......... .......... .......... .......... .......... 63%  118M 2s\n",
      " 50950K .......... .......... .......... .......... .......... 63%  117M 2s\n",
      " 51000K .......... .......... .......... .......... .......... 63%  111M 2s\n",
      " 51050K .......... .......... .......... .......... .......... 63%  111M 2s\n",
      " 51100K .......... .......... .......... .......... .......... 64%  124M 2s\n",
      " 51150K .......... .......... .......... .......... .......... 64% 45,0M 2s\n",
      " 51200K .......... .......... .......... .......... .......... 64% 15,8M 2s\n",
      " 51250K .......... .......... .......... .......... .......... 64%  128M 2s\n",
      " 51300K .......... .......... .......... .......... .......... 64%  137M 2s\n",
      " 51350K .......... .......... .......... .......... .......... 64%  118M 2s\n",
      " 51400K .......... .......... .......... .......... .......... 64%  110M 2s\n",
      " 51450K .......... .......... .......... .......... .......... 64%  117M 2s\n",
      " 51500K .......... .......... .......... .......... .......... 64%  118M 2s\n",
      " 51550K .......... .......... .......... .......... .......... 64%  116M 2s\n",
      " 51600K .......... .......... .......... .......... .......... 64% 90,1M 2s\n",
      " 51650K .......... .......... .......... .......... .......... 64%  117M 2s\n",
      " 51700K .......... .......... .......... .......... .......... 64%  117M 2s\n",
      " 51750K .......... .......... .......... .......... .......... 64%  117M 2s\n",
      " 51800K .......... .......... .......... .......... .......... 64%  111M 2s\n",
      " 51850K .......... .......... .......... .......... .......... 64%  118M 2s\n",
      " 51900K .......... .......... .......... .......... .......... 65%  118M 2s\n",
      " 51950K .......... .......... .......... .......... .......... 65%  117M 2s\n",
      " 52000K .......... .......... .......... .......... .......... 65% 88,3M 2s\n",
      " 52050K .......... .......... .......... .......... .......... 65% 51,1M 2s\n",
      " 52100K .......... .......... .......... .......... .......... 65%  118M 2s\n",
      " 52150K .......... .......... .......... .......... .......... 65% 1,76M 2s\n",
      " 52200K .......... .......... .......... .......... .......... 65% 1,02M 2s\n",
      " 52250K .......... .......... .......... .......... .......... 65% 6,13M 2s\n",
      " 52300K .......... .......... .......... .......... .......... 65% 39,3M 2s\n",
      " 52350K .......... .......... .......... .......... .......... 65% 25,9M 2s\n",
      " 52400K .......... .......... .......... .......... .......... 65% 17,2M 2s\n",
      " 52450K .......... .......... .......... .......... .......... 65% 21,8M 2s\n",
      " 52500K .......... .......... .......... .......... .......... 65% 61,4M 2s\n",
      " 52550K .......... .......... .......... .......... .......... 65% 86,9M 2s\n",
      " 52600K .......... .......... .......... .......... .......... 65% 95,8M 2s\n",
      " 52650K .......... .......... .......... .......... .......... 65%  111M 2s\n",
      " 52700K .......... .......... .......... .......... .......... 66%  107M 2s\n",
      " 52750K .......... .......... .......... .......... .......... 66% 76,9M 2s\n",
      " 52800K .......... .......... .......... .......... .......... 66%  143M 2s\n",
      " 52850K .......... .......... .......... .......... .......... 66%  185M 1s\n",
      " 52900K .......... .......... .......... .......... .......... 66%  131M 1s\n",
      " 52950K .......... .......... .......... .......... .......... 66%  119M 1s\n",
      " 53000K .......... .......... .......... .......... .......... 66%  106M 1s\n",
      " 53050K .......... .......... .......... .......... .......... 66%  118M 1s\n",
      " 53100K .......... .......... .......... .......... .......... 66%  118M 1s\n",
      " 53150K .......... .......... .......... .......... .......... 66%  118M 1s\n",
      " 53200K .......... .......... .......... .......... .......... 66% 90,8M 1s\n",
      " 53250K .......... .......... .......... .......... .......... 66%  117M 1s\n",
      " 53300K .......... .......... .......... .......... .......... 66%  117M 1s\n",
      " 53350K .......... .......... .......... .......... .......... 66% 82,0M 1s\n",
      " 53400K .......... .......... .......... .......... .......... 66%  167M 1s\n",
      " 53450K .......... .......... .......... .......... .......... 66%  113M 1s\n",
      " 53500K .......... .......... .......... .......... .......... 67%  119M 1s\n",
      " 53550K .......... .......... .......... .......... .......... 67%  117M 1s\n",
      " 53600K .......... .......... .......... .......... .......... 67% 91,9M 1s\n",
      " 53650K .......... .......... .......... .......... .......... 67% 49,3M 1s\n",
      " 53700K .......... .......... .......... .......... .......... 67% 38,8M 1s\n",
      " 53750K .......... .......... .......... .......... .......... 67%  118M 1s\n",
      " 53800K .......... .......... .......... .......... .......... 67% 55,7M 1s\n",
      " 53850K .......... .......... .......... .......... .......... 67% 40,4M 1s\n",
      " 53900K .......... .......... .......... .......... .......... 67% 77,7M 1s\n",
      " 53950K .......... .......... .......... .......... .......... 67%  147M 1s\n",
      " 54000K .......... .......... .......... .......... .......... 67% 67,7M 1s\n",
      " 54050K .......... .......... .......... .......... .......... 67%  113M 1s\n",
      " 54100K .......... .......... .......... .......... .......... 67%  115M 1s\n",
      " 54150K .......... .......... .......... .......... .......... 67%  121M 1s\n",
      " 54200K .......... .......... .......... .......... .......... 67%  108M 1s\n",
      " 54250K .......... .......... .......... .......... .......... 67% 13,0M 1s\n",
      " 54300K .......... .......... .......... .......... .......... 68%  114M 1s\n",
      " 54350K .......... .......... .......... .......... .......... 68% 76,4M 1s\n",
      " 54400K .......... .......... .......... .......... .......... 68% 93,2M 1s\n",
      " 54450K .......... .......... .......... .......... .......... 68%  114M 1s\n",
      " 54500K .......... .......... .......... .......... .......... 68%  118M 1s\n",
      " 54550K .......... .......... .......... .......... .......... 68%  118M 1s\n",
      " 54600K .......... .......... .......... .......... .......... 68%  110M 1s\n",
      " 54650K .......... .......... .......... .......... .......... 68%  117M 1s\n",
      " 54700K .......... .......... .......... .......... .......... 68%  115M 1s\n",
      " 54750K .......... .......... .......... .......... .......... 68%  118M 1s\n",
      " 54800K .......... .......... .......... .......... .......... 68% 94,1M 1s\n",
      " 54850K .......... .......... .......... .......... .......... 68%  113M 1s\n",
      " 54900K .......... .......... .......... .......... .......... 68%  103M 1s\n",
      " 54950K .......... .......... .......... .......... .......... 68% 92,1M 1s\n",
      " 55000K .......... .......... .......... .......... .......... 68%  174M 1s\n",
      " 55050K .......... .......... .......... .......... .......... 68%  123M 1s\n",
      " 55100K .......... .......... .......... .......... .......... 69%  113M 1s\n",
      " 55150K .......... .......... .......... .......... .......... 69%  120M 1s\n",
      " 55200K .......... .......... .......... .......... .......... 69%  685K 1s\n",
      " 55250K .......... .......... .......... .......... .......... 69% 12,2M 1s\n",
      " 55300K .......... .......... .......... .......... .......... 69% 21,8M 1s\n",
      " 55350K .......... .......... .......... .......... .......... 69% 8,26M 1s\n",
      " 55400K .......... .......... .......... .......... .......... 69% 24,1M 1s\n",
      " 55450K .......... .......... .......... .......... .......... 69% 21,3M 1s\n",
      " 55500K .......... .......... .......... .......... .......... 69% 19,5M 1s\n",
      " 55550K .......... .......... .......... .......... .......... 69% 42,1M 1s\n",
      " 55600K .......... .......... .......... .......... .......... 69% 51,6M 1s\n",
      " 55650K .......... .......... .......... .......... .......... 69% 62,4M 1s\n",
      " 55700K .......... .......... .......... .......... .......... 69% 98,3M 1s\n",
      " 55750K .......... .......... .......... .......... .......... 69% 95,7M 1s\n",
      " 55800K .......... .......... .......... .......... .......... 69% 88,9M 1s\n",
      " 55850K .......... .......... .......... .......... .......... 69% 93,9M 1s\n",
      " 55900K .......... .......... .......... .......... .......... 70%  159M 1s\n",
      " 55950K .......... .......... .......... .......... .......... 70%  166M 1s\n",
      " 56000K .......... .......... .......... .......... .......... 70%  141M 1s\n",
      " 56050K .......... .......... .......... .......... .......... 70%  169M 1s\n",
      " 56100K .......... .......... .......... .......... .......... 70%  192M 1s\n",
      " 56150K .......... .......... .......... .......... .......... 70%  132M 1s\n",
      " 56200K .......... .......... .......... .......... .......... 70%  110M 1s\n",
      " 56250K .......... .......... .......... .......... .......... 70%  118M 1s\n",
      " 56300K .......... .......... .......... .......... .......... 70%  118M 1s\n",
      " 56350K .......... .......... .......... .......... .......... 70%  113M 1s\n",
      " 56400K .......... .......... .......... .......... .......... 70% 92,6M 1s\n",
      " 56450K .......... .......... .......... .......... .......... 70%  118M 1s\n",
      " 56500K .......... .......... .......... .......... .......... 70%  115M 1s\n",
      " 56550K .......... .......... .......... .......... .......... 70%  118M 1s\n",
      " 56600K .......... .......... .......... .......... .......... 70%  112M 1s\n",
      " 56650K .......... .......... .......... .......... .......... 70%  117M 1s\n",
      " 56700K .......... .......... .......... .......... .......... 71%  117M 1s\n",
      " 56750K .......... .......... .......... .......... .......... 71%  114M 1s\n",
      " 56800K .......... .......... .......... .......... .......... 71% 44,5M 1s\n",
      " 56850K .......... .......... .......... .......... .......... 71%  122M 1s\n",
      " 56900K .......... .......... .......... .......... .......... 71% 26,3M 1s\n",
      " 56950K .......... .......... .......... .......... .......... 71%  117M 1s\n",
      " 57000K .......... .......... .......... .......... .......... 71% 46,3M 1s\n",
      " 57050K .......... .......... .......... .......... .......... 71%  120M 1s\n",
      " 57100K .......... .......... .......... .......... .......... 71%  118M 1s\n",
      " 57150K .......... .......... .......... .......... .......... 71%  108M 1s\n",
      " 57200K .......... .......... .......... .......... .......... 71% 92,9M 1s\n",
      " 57250K .......... .......... .......... .......... .......... 71% 30,2M 1s\n",
      " 57300K .......... .......... .......... .......... .......... 71% 18,7M 1s\n",
      " 57350K .......... .......... .......... .......... .......... 71%  113M 1s\n",
      " 57400K .......... .......... .......... .......... .......... 71% 60,1M 1s\n",
      " 57450K .......... .......... .......... .......... .......... 71%  121M 1s\n",
      " 57500K .......... .......... .......... .......... .......... 72%  126M 1s\n",
      " 57550K .......... .......... .......... .......... .......... 72%  117M 1s\n",
      " 57600K .......... .......... .......... .......... .......... 72% 88,9M 1s\n",
      " 57650K .......... .......... .......... .......... .......... 72%  118M 1s\n",
      " 57700K .......... .......... .......... .......... .......... 72%  103M 1s\n",
      " 57750K .......... .......... .......... .......... .......... 72%  132M 1s\n",
      " 57800K .......... .......... .......... .......... .......... 72%  112M 1s\n",
      " 57850K .......... .......... .......... .......... .......... 72%  117M 1s\n",
      " 57900K .......... .......... .......... .......... .......... 72%  117M 1s\n",
      " 57950K .......... .......... .......... .......... .......... 72%  118M 1s\n",
      " 58000K .......... .......... .......... .......... .......... 72% 90,3M 1s\n",
      " 58050K .......... .......... .......... .......... .......... 72%  117M 1s\n",
      " 58100K .......... .......... .......... .......... .......... 72%  117M 1s\n",
      " 58150K .......... .......... .......... .......... .......... 72%  114M 1s\n",
      " 58200K .......... .......... .......... .......... .......... 72%  692K 1s\n",
      " 58250K .......... .......... .......... .......... .......... 72% 39,0M 1s\n",
      " 58300K .......... .......... .......... .......... .......... 73% 19,6M 1s\n",
      " 58350K .......... .......... .......... .......... .......... 73% 19,6M 1s\n",
      " 58400K .......... .......... .......... .......... .......... 73% 8,99M 1s\n",
      " 58450K .......... .......... .......... .......... .......... 73% 13,8M 1s\n",
      " 58500K .......... .......... .......... .......... .......... 73% 13,6M 1s\n",
      " 58550K .......... .......... .......... .......... .......... 73%  117M 1s\n",
      " 58600K .......... .......... .......... .......... .......... 73% 28,4M 1s\n",
      " 58650K .......... .......... .......... .......... .......... 73% 38,8M 1s\n",
      " 58700K .......... .......... .......... .......... .......... 73%  118M 1s\n",
      " 58750K .......... .......... .......... .......... .......... 73%  120M 1s\n",
      " 58800K .......... .......... .......... .......... .......... 73% 93,8M 1s\n",
      " 58850K .......... .......... .......... .......... .......... 73%  117M 1s\n",
      " 58900K .......... .......... .......... .......... .......... 73%  118M 1s\n",
      " 58950K .......... .......... .......... .......... .......... 73%  120M 1s\n",
      " 59000K .......... .......... .......... .......... .......... 73%  101M 1s\n",
      " 59050K .......... .......... .......... .......... .......... 73%  119M 1s\n",
      " 59100K .......... .......... .......... .......... .......... 74%  118M 1s\n",
      " 59150K .......... .......... .......... .......... .......... 74%  116M 1s\n",
      " 59200K .......... .......... .......... .......... .......... 74% 94,0M 1s\n",
      " 59250K .......... .......... .......... .......... .......... 74%  113M 1s\n",
      " 59300K .......... .......... .......... .......... .......... 74%  121M 1s\n",
      " 59350K .......... .......... .......... .......... .......... 74%  118M 1s\n",
      " 59400K .......... .......... .......... .......... .......... 74%  102M 1s\n",
      " 59450K .......... .......... .......... .......... .......... 74%  118M 1s\n",
      " 59500K .......... .......... .......... .......... .......... 74%  119M 1s\n",
      " 59550K .......... .......... .......... .......... .......... 74%  114M 1s\n",
      " 59600K .......... .......... .......... .......... .......... 74% 94,5M 1s\n",
      " 59650K .......... .......... .......... .......... .......... 74%  120M 1s\n",
      " 59700K .......... .......... .......... .......... .......... 74%  118M 1s\n",
      " 59750K .......... .......... .......... .......... .......... 74% 68,6M 1s\n",
      " 59800K .......... .......... .......... .......... .......... 74%  120M 1s\n",
      " 59850K .......... .......... .......... .......... .......... 74%  134M 1s\n",
      " 59900K .......... .......... .......... .......... .......... 75%  130M 1s\n",
      " 59950K .......... .......... .......... .......... .......... 75% 33,1M 1s\n",
      " 60000K .......... .......... .......... .......... .......... 75% 88,0M 1s\n",
      " 60050K .......... .......... .......... .......... .......... 75% 68,0M 1s\n",
      " 60100K .......... .......... .......... .......... .......... 75%  117M 1s\n",
      " 60150K .......... .......... .......... .......... .......... 75%  106M 1s\n",
      " 60200K .......... .......... .......... .......... .......... 75%  110M 1s\n",
      " 60250K .......... .......... .......... .......... .......... 75% 66,8M 1s\n",
      " 60300K .......... .......... .......... .......... .......... 75% 77,6M 1s\n",
      " 60350K .......... .......... .......... .......... .......... 75% 13,8M 1s\n",
      " 60400K .......... .......... .......... .......... .......... 75% 72,8M 1s\n",
      " 60450K .......... .......... .......... .......... .......... 75%  117M 1s\n",
      " 60500K .......... .......... .......... .......... .......... 75% 48,4M 1s\n",
      " 60550K .......... .......... .......... .......... .......... 75% 83,0M 1s\n",
      " 60600K .......... .......... .......... .......... .......... 75%  112M 1s\n",
      " 60650K .......... .......... .......... .......... .......... 75%  107M 1s\n",
      " 60700K .......... .......... .......... .......... .......... 76%  118M 1s\n",
      " 60750K .......... .......... .......... .......... .......... 76%  117M 1s\n",
      " 60800K .......... .......... .......... .......... .......... 76% 93,0M 1s\n",
      " 60850K .......... .......... .......... .......... .......... 76%  115M 1s\n",
      " 60900K .......... .......... .......... .......... .......... 76%  118M 1s\n",
      " 60950K .......... .......... .......... .......... .......... 76%  121M 1s\n",
      " 61000K .......... .......... .......... .......... .......... 76%  113M 1s\n",
      " 61050K .......... .......... .......... .......... .......... 76%  115M 1s\n",
      " 61100K .......... .......... .......... .......... .......... 76%  117M 1s\n",
      " 61150K .......... .......... .......... .......... .......... 76%  117M 1s\n",
      " 61200K .......... .......... .......... .......... .......... 76% 90,0M 1s\n",
      " 61250K .......... .......... .......... .......... .......... 76%  699K 1s\n",
      " 61300K .......... .......... .......... .......... .......... 76% 15,6M 1s\n",
      " 61350K .......... .......... .......... .......... .......... 76% 30,4M 1s\n",
      " 61400K .......... .......... .......... .......... .......... 76% 12,4M 1s\n",
      " 61450K .......... .......... .......... .......... .......... 76% 16,6M 1s\n",
      " 61500K .......... .......... .......... .......... .......... 77% 16,0M 1s\n",
      " 61550K .......... .......... .......... .......... .......... 77% 13,3M 1s\n",
      " 61600K .......... .......... .......... .......... .......... 77% 21,2M 1s\n",
      " 61650K .......... .......... .......... .......... .......... 77%  118M 1s\n",
      " 61700K .......... .......... .......... .......... .......... 77% 51,1M 1s\n",
      " 61750K .......... .......... .......... .......... .......... 77% 63,1M 1s\n",
      " 61800K .......... .......... .......... .......... .......... 77% 55,6M 1s\n",
      " 61850K .......... .......... .......... .......... .......... 77%  116M 1s\n",
      " 61900K .......... .......... .......... .......... .......... 77%  113M 1s\n",
      " 61950K .......... .......... .......... .......... .......... 77%  121M 1s\n",
      " 62000K .......... .......... .......... .......... .......... 77% 86,3M 1s\n",
      " 62050K .......... .......... .......... .......... .......... 77%  124M 1s\n",
      " 62100K .......... .......... .......... .......... .......... 77%  120M 1s\n",
      " 62150K .......... .......... .......... .......... .......... 77%  116M 1s\n",
      " 62200K .......... .......... .......... .......... .......... 77%  113M 1s\n",
      " 62250K .......... .......... .......... .......... .......... 77%  117M 1s\n",
      " 62300K .......... .......... .......... .......... .......... 78% 59,2M 1s\n",
      " 62350K .......... .......... .......... .......... .......... 78%  167M 1s\n",
      " 62400K .......... .......... .......... .......... .......... 78%  195M 1s\n",
      " 62450K .......... .......... .......... .......... .......... 78%  115M 1s\n",
      " 62500K .......... .......... .......... .......... .......... 78%  117M 1s\n",
      " 62550K .......... .......... .......... .......... .......... 78%  114M 1s\n",
      " 62600K .......... .......... .......... .......... .......... 78%  110M 1s\n",
      " 62650K .......... .......... .......... .......... .......... 78%  118M 1s\n",
      " 62700K .......... .......... .......... .......... .......... 78%  114M 1s\n",
      " 62750K .......... .......... .......... .......... .......... 78%  117M 1s\n",
      " 62800K .......... .......... .......... .......... .......... 78% 96,4M 1s\n",
      " 62850K .......... .......... .......... .......... .......... 78%  114M 1s\n",
      " 62900K .......... .......... .......... .......... .......... 78%  117M 1s\n",
      " 62950K .......... .......... .......... .......... .......... 78%  118M 1s\n",
      " 63000K .......... .......... .......... .......... .......... 78%  114M 1s\n",
      " 63050K .......... .......... .......... .......... .......... 78% 32,7M 1s\n",
      " 63100K .......... .......... .......... .......... .......... 79%  114M 1s\n",
      " 63150K .......... .......... .......... .......... .......... 79% 92,0M 1s\n",
      " 63200K .......... .......... .......... .......... .......... 79% 49,2M 1s\n",
      " 63250K .......... .......... .......... .......... .......... 79%  117M 1s\n",
      " 63300K .......... .......... .......... .......... .......... 79%  112M 1s\n",
      " 63350K .......... .......... .......... .......... .......... 79%  118M 1s\n",
      " 63400K .......... .......... .......... .......... .......... 79% 33,9M 1s\n",
      " 63450K .......... .......... .......... .......... .......... 79% 14,7M 1s\n",
      " 63500K .......... .......... .......... .......... .......... 79% 62,4M 1s\n",
      " 63550K .......... .......... .......... .......... .......... 79%  117M 1s\n",
      " 63600K .......... .......... .......... .......... .......... 79% 91,9M 1s\n",
      " 63650K .......... .......... .......... .......... .......... 79%  119M 1s\n",
      " 63700K .......... .......... .......... .......... .......... 79%  113M 1s\n",
      " 63750K .......... .......... .......... .......... .......... 79%  117M 1s\n",
      " 63800K .......... .......... .......... .......... .......... 79%  112M 1s\n",
      " 63850K .......... .......... .......... .......... .......... 79%  117M 1s\n",
      " 63900K .......... .......... .......... .......... .......... 80%  118M 1s\n",
      " 63950K .......... .......... .......... .......... .......... 80%  116M 1s\n",
      " 64000K .......... .......... .......... .......... .......... 80% 91,1M 1s\n",
      " 64050K .......... .......... .......... .......... .......... 80%  117M 1s\n",
      " 64100K .......... .......... .......... .......... .......... 80%  113M 1s\n",
      " 64150K .......... .......... .......... .......... .......... 80%  117M 1s\n",
      " 64200K .......... .......... .......... .......... .......... 80%  114M 1s\n",
      " 64250K .......... .......... .......... .......... .......... 80%  712K 1s\n",
      " 64300K .......... .......... .......... .......... .......... 80% 55,2M 1s\n",
      " 64350K .......... .......... .......... .......... .......... 80% 15,3M 1s\n",
      " 64400K .......... .......... .......... .......... .......... 80% 8,38M 1s\n",
      " 64450K .......... .......... .......... .......... .......... 80% 44,9M 1s\n",
      " 64500K .......... .......... .......... .......... .......... 80% 11,8M 1s\n",
      " 64550K .......... .......... .......... .......... .......... 80% 70,6M 1s\n",
      " 64600K .......... .......... .......... .......... .......... 80% 12,3M 1s\n",
      " 64650K .......... .......... .......... .......... .......... 81% 24,4M 1s\n",
      " 64700K .......... .......... .......... .......... .......... 81% 45,7M 1s\n",
      " 64750K .......... .......... .......... .......... .......... 81%  114M 1s\n",
      " 64800K .......... .......... .......... .......... .......... 81% 54,6M 1s\n",
      " 64850K .......... .......... .......... .......... .......... 81% 64,7M 1s\n",
      " 64900K .......... .......... .......... .......... .......... 81%  117M 1s\n",
      " 64950K .......... .......... .......... .......... .......... 81%  112M 1s\n",
      " 65000K .......... .......... .......... .......... .......... 81%  108M 1s\n",
      " 65050K .......... .......... .......... .......... .......... 81% 64,6M 1s\n",
      " 65100K .......... .......... .......... .......... .......... 81%  117M 1s\n",
      " 65150K .......... .......... .......... .......... .......... 81%  119M 1s\n",
      " 65200K .......... .......... .......... .......... .......... 81% 70,4M 1s\n",
      " 65250K .......... .......... .......... .......... .......... 81%  111M 1s\n",
      " 65300K .......... .......... .......... .......... .......... 81%  171M 1s\n",
      " 65350K .......... .......... .......... .......... .......... 81%  131M 1s\n",
      " 65400K .......... .......... .......... .......... .......... 81%  113M 1s\n",
      " 65450K .......... .......... .......... .......... .......... 82%  116M 1s\n",
      " 65500K .......... .......... .......... .......... .......... 82%  116M 1s\n",
      " 65550K .......... .......... .......... .......... .......... 82%  117M 1s\n",
      " 65600K .......... .......... .......... .......... .......... 82% 86,7M 1s\n",
      " 65650K .......... .......... .......... .......... .......... 82%  118M 1s\n",
      " 65700K .......... .......... .......... .......... .......... 82%  118M 1s\n",
      " 65750K .......... .......... .......... .......... .......... 82%  114M 1s\n",
      " 65800K .......... .......... .......... .......... .......... 82%  110M 1s\n",
      " 65850K .......... .......... .......... .......... .......... 82%  118M 1s\n",
      " 65900K .......... .......... .......... .......... .......... 82% 86,3M 1s\n",
      " 65950K .......... .......... .......... .......... .......... 82%  183M 1s\n",
      " 66000K .......... .......... .......... .......... .......... 82% 72,6M 1s\n",
      " 66050K .......... .......... .......... .......... .......... 82% 75,3M 1s\n",
      " 66100K .......... .......... .......... .......... .......... 82% 35,5M 1s\n",
      " 66150K .......... .......... .......... .......... .......... 82%  115M 1s\n",
      " 66200K .......... .......... .......... .......... .......... 82%  108M 1s\n",
      " 66250K .......... .......... .......... .......... .......... 83%  117M 1s\n",
      " 66300K .......... .......... .......... .......... .......... 83%  117M 1s\n",
      " 66350K .......... .......... .......... .......... .......... 83%  117M 1s\n",
      " 66400K .......... .......... .......... .......... .......... 83% 61,9M 1s\n",
      " 66450K .......... .......... .......... .......... .......... 83% 64,2M 1s\n",
      " 66500K .......... .......... .......... .......... .......... 83% 11,0M 1s\n",
      " 66550K .......... .......... .......... .......... .......... 83% 71,2M 1s\n",
      " 66600K .......... .......... .......... .......... .......... 83%  105M 1s\n",
      " 66650K .......... .......... .......... .......... .......... 83% 82,9M 1s\n",
      " 66700K .......... .......... .......... .......... .......... 83%  122M 1s\n",
      " 66750K .......... .......... .......... .......... .......... 83%  117M 1s\n",
      " 66800K .......... .......... .......... .......... .......... 83% 91,1M 1s\n",
      " 66850K .......... .......... .......... .......... .......... 83%  118M 1s\n",
      " 66900K .......... .......... .......... .......... .......... 83%  117M 1s\n",
      " 66950K .......... .......... .......... .......... .......... 83%  117M 1s\n",
      " 67000K .......... .......... .......... .......... .......... 83%  108M 1s\n",
      " 67050K .......... .......... .......... .......... .......... 84%  116M 1s\n",
      " 67100K .......... .......... .......... .......... .......... 84%  121M 1s\n",
      " 67150K .......... .......... .......... .......... .......... 84%  117M 1s\n",
      " 67200K .......... .......... .......... .......... .......... 84% 89,7M 1s\n",
      " 67250K .......... .......... .......... .......... .......... 84%  117M 1s\n",
      " 67300K .......... .......... .......... .......... .......... 84%  724K 1s\n",
      " 67350K .......... .......... .......... .......... .......... 84% 14,2M 1s\n",
      " 67400K .......... .......... .......... .......... .......... 84% 26,3M 1s\n",
      " 67450K .......... .......... .......... .......... .......... 84% 8,33M 1s\n",
      " 67500K .......... .......... .......... .......... .......... 84% 35,7M 1s\n",
      " 67550K .......... .......... .......... .......... .......... 84% 18,8M 1s\n",
      " 67600K .......... .......... .......... .......... .......... 84% 12,6M 1s\n",
      " 67650K .......... .......... .......... .......... .......... 84% 32,1M 1s\n",
      " 67700K .......... .......... .......... .......... .......... 84% 27,6M 1s\n",
      " 67750K .......... .......... .......... .......... .......... 84% 25,6M 1s\n",
      " 67800K .......... .......... .......... .......... .......... 84% 53,2M 1s\n",
      " 67850K .......... .......... .......... .......... .......... 85% 67,0M 1s\n",
      " 67900K .......... .......... .......... .......... .......... 85%  169M 1s\n",
      " 67950K .......... .......... .......... .......... .......... 85% 91,5M 1s\n",
      " 68000K .......... .......... .......... .......... .......... 85%  147M 1s\n",
      " 68050K .......... .......... .......... .......... .......... 85%  176M 1s\n",
      " 68100K .......... .......... .......... .......... .......... 85%  173M 1s\n",
      " 68150K .......... .......... .......... .......... .......... 85%  150M 1s\n",
      " 68200K .......... .......... .......... .......... .......... 85%  111M 1s\n",
      " 68250K .......... .......... .......... .......... .......... 85%  114M 1s\n",
      " 68300K .......... .......... .......... .......... .......... 85%  120M 1s\n",
      " 68350K .......... .......... .......... .......... .......... 85%  117M 1s\n",
      " 68400K .......... .......... .......... .......... .......... 85% 86,4M 1s\n",
      " 68450K .......... .......... .......... .......... .......... 85%  118M 1s\n",
      " 68500K .......... .......... .......... .......... .......... 85%  117M 1s\n",
      " 68550K .......... .......... .......... .......... .......... 85% 64,0M 1s\n",
      " 68600K .......... .......... .......... .......... .......... 85%  112M 1s\n",
      " 68650K .......... .......... .......... .......... .......... 86%  113M 1s\n",
      " 68700K .......... .......... .......... .......... .......... 86%  118M 1s\n",
      " 68750K .......... .......... .......... .......... .......... 86%  114M 1s\n",
      " 68800K .......... .......... .......... .......... .......... 86% 92,4M 1s\n",
      " 68850K .......... .......... .......... .......... .......... 86%  118M 1s\n",
      " 68900K .......... .......... .......... .......... .......... 86%  118M 1s\n",
      " 68950K .......... .......... .......... .......... .......... 86%  117M 1s\n",
      " 69000K .......... .......... .......... .......... .......... 86%  113M 1s\n",
      " 69050K .......... .......... .......... .......... .......... 86%  113M 1s\n",
      " 69100K .......... .......... .......... .......... .......... 86%  118M 1s\n",
      " 69150K .......... .......... .......... .......... .......... 86%  106M 1s\n",
      " 69200K .......... .......... .......... .......... .......... 86% 30,1M 1s\n",
      " 69250K .......... .......... .......... .......... .......... 86%  115M 1s\n",
      " 69300K .......... .......... .......... .......... .......... 86% 89,9M 1s\n",
      " 69350K .......... .......... .......... .......... .......... 86%  112M 1s\n",
      " 69400K .......... .......... .......... .......... .......... 86%  162M 1s\n",
      " 69450K .......... .......... .......... .......... .......... 87% 97,8M 1s\n",
      " 69500K .......... .......... .......... .......... .......... 87% 17,4M 1s\n",
      " 69550K .......... .......... .......... .......... .......... 87% 23,1M 1s\n",
      " 69600K .......... .......... .......... .......... .......... 87% 37,9M 1s\n",
      " 69650K .......... .......... .......... .......... .......... 87% 63,9M 1s\n",
      " 69700K .......... .......... .......... .......... .......... 87%  117M 1s\n",
      " 69750K .......... .......... .......... .......... .......... 87%  118M 1s\n",
      " 69800K .......... .......... .......... .......... .......... 87%  112M 1s\n",
      " 69850K .......... .......... .......... .......... .......... 87%  112M 1s\n",
      " 69900K .......... .......... .......... .......... .......... 87%  120M 1s\n",
      " 69950K .......... .......... .......... .......... .......... 87%  118M 1s\n",
      " 70000K .......... .......... .......... .......... .......... 87% 91,3M 1s\n",
      " 70050K .......... .......... .......... .......... .......... 87%  114M 0s\n",
      " 70100K .......... .......... .......... .......... .......... 87%  118M 0s\n",
      " 70150K .......... .......... .......... .......... .......... 87%  117M 0s\n",
      " 70200K .......... .......... .......... .......... .......... 87%  114M 0s\n",
      " 70250K .......... .......... .......... .......... .......... 88%  118M 0s\n",
      " 70300K .......... .......... .......... .......... .......... 88%  113M 0s\n",
      " 70350K .......... .......... .......... .......... .......... 88%  732K 0s\n",
      " 70400K .......... .......... .......... .......... .......... 88% 14,7M 0s\n",
      " 70450K .......... .......... .......... .......... .......... 88% 8,46M 0s\n",
      " 70500K .......... .......... .......... .......... .......... 88% 16,4M 0s\n",
      " 70550K .......... .......... .......... .......... .......... 88% 18,5M 0s\n",
      " 70600K .......... .......... .......... .......... .......... 88% 31,9M 0s\n",
      " 70650K .......... .......... .......... .......... .......... 88% 18,6M 0s\n",
      " 70700K .......... .......... .......... .......... .......... 88% 17,9M 0s\n",
      " 70750K .......... .......... .......... .......... .......... 88% 67,0M 0s\n",
      " 70800K .......... .......... .......... .......... .......... 88% 19,6M 0s\n",
      " 70850K .......... .......... .......... .......... .......... 88% 50,3M 0s\n",
      " 70900K .......... .......... .......... .......... .......... 88%  119M 0s\n",
      " 70950K .......... .......... .......... .......... .......... 88% 73,8M 0s\n",
      " 71000K .......... .......... .......... .......... .......... 88%  112M 0s\n",
      " 71050K .......... .......... .......... .......... .......... 89%  116M 0s\n",
      " 71100K .......... .......... .......... .......... .......... 89%  114M 0s\n",
      " 71150K .......... .......... .......... .......... .......... 89%  117M 0s\n",
      " 71200K .......... .......... .......... .......... .......... 89% 94,2M 0s\n",
      " 71250K .......... .......... .......... .......... .......... 89%  117M 0s\n",
      " 71300K .......... .......... .......... .......... .......... 89%  114M 0s\n",
      " 71350K .......... .......... .......... .......... .......... 89%  117M 0s\n",
      " 71400K .......... .......... .......... .......... .......... 89%  109M 0s\n",
      " 71450K .......... .......... .......... .......... .......... 89%  117M 0s\n",
      " 71500K .......... .......... .......... .......... .......... 89%  117M 0s\n",
      " 71550K .......... .......... .......... .......... .......... 89%  116M 0s\n",
      " 71600K .......... .......... .......... .......... .......... 89% 93,7M 0s\n",
      " 71650K .......... .......... .......... .......... .......... 89%  118M 0s\n",
      " 71700K .......... .......... .......... .......... .......... 89%  114M 0s\n",
      " 71750K .......... .......... .......... .......... .......... 89%  119M 0s\n",
      " 71800K .......... .......... .......... .......... .......... 89%  107M 0s\n",
      " 71850K .......... .......... .......... .......... .......... 90%  117M 0s\n",
      " 71900K .......... .......... .......... .......... .......... 90%  118M 0s\n",
      " 71950K .......... .......... .......... .......... .......... 90%  116M 0s\n",
      " 72000K .......... .......... .......... .......... .......... 90% 93,2M 0s\n",
      " 72050K .......... .......... .......... .......... .......... 90% 91,3M 0s\n",
      " 72100K .......... .......... .......... .......... .......... 90%  158M 0s\n",
      " 72150K .......... .......... .......... .......... .......... 90%  118M 0s\n",
      " 72200K .......... .......... .......... .......... .......... 90%  112M 0s\n",
      " 72250K .......... .......... .......... .......... .......... 90% 55,8M 0s\n",
      " 72300K .......... .......... .......... .......... .......... 90% 34,4M 0s\n",
      " 72350K .......... .......... .......... .......... .......... 90%  127M 0s\n",
      " 72400K .......... .......... .......... .......... .......... 90% 94,4M 0s\n",
      " 72450K .......... .......... .......... .......... .......... 90%  118M 0s\n",
      " 72500K .......... .......... .......... .......... .......... 90%  118M 0s\n",
      " 72550K .......... .......... .......... .......... .......... 90% 49,0M 0s\n",
      " 72600K .......... .......... .......... .......... .......... 90% 14,4M 0s\n",
      " 72650K .......... .......... .......... .......... .......... 91% 29,3M 0s\n",
      " 72700K .......... .......... .......... .......... .......... 91%  120M 0s\n",
      " 72750K .......... .......... .......... .......... .......... 91% 69,9M 0s\n",
      " 72800K .......... .......... .......... .......... .......... 91% 56,2M 0s\n",
      " 72850K .......... .......... .......... .......... .......... 91%  116M 0s\n",
      " 72900K .......... .......... .......... .......... .......... 91%  111M 0s\n",
      " 72950K .......... .......... .......... .......... .......... 91%  115M 0s\n",
      " 73000K .......... .......... .......... .......... .......... 91%  108M 0s\n",
      " 73050K .......... .......... .......... .......... .......... 91%  128M 0s\n",
      " 73100K .......... .......... .......... .......... .......... 91%  117M 0s\n",
      " 73150K .......... .......... .......... .......... .......... 91%  103M 0s\n",
      " 73200K .......... .......... .......... .......... .......... 91% 89,0M 0s\n",
      " 73250K .......... .......... .......... .......... .......... 91%  133M 0s\n",
      " 73300K .......... .......... .......... .......... .......... 91%  126M 0s\n",
      " 73350K .......... .......... .......... .......... .......... 91% 1,13M 0s\n",
      " 73400K .......... .......... .......... .......... .......... 91% 1,89M 0s\n",
      " 73450K .......... .......... .......... .......... .......... 92% 23,8M 0s\n",
      " 73500K .......... .......... .......... .......... .......... 92% 8,80M 0s\n",
      " 73550K .......... .......... .......... .......... .......... 92% 20,5M 0s\n",
      " 73600K .......... .......... .......... .......... .......... 92% 14,7M 0s\n",
      " 73650K .......... .......... .......... .......... .......... 92% 21,0M 0s\n",
      " 73700K .......... .......... .......... .......... .......... 92% 36,8M 0s\n",
      " 73750K .......... .......... .......... .......... .......... 92% 14,9M 0s\n",
      " 73800K .......... .......... .......... .......... .......... 92% 21,2M 0s\n",
      " 73850K .......... .......... .......... .......... .......... 92% 43,5M 0s\n",
      " 73900K .......... .......... .......... .......... .......... 92% 26,3M 0s\n",
      " 73950K .......... .......... .......... .......... .......... 92%  116M 0s\n",
      " 74000K .......... .......... .......... .......... .......... 92% 60,6M 0s\n",
      " 74050K .......... .......... .......... .......... .......... 92%  117M 0s\n",
      " 74100K .......... .......... .......... .......... .......... 92%  108M 0s\n",
      " 74150K .......... .......... .......... .......... .......... 92%  128M 0s\n",
      " 74200K .......... .......... .......... .......... .......... 92%  108M 0s\n",
      " 74250K .......... .......... .......... .......... .......... 93%  118M 0s\n",
      " 74300K .......... .......... .......... .......... .......... 93%  117M 0s\n",
      " 74350K .......... .......... .......... .......... .......... 93%  118M 0s\n",
      " 74400K .......... .......... .......... .......... .......... 93% 90,1M 0s\n",
      " 74450K .......... .......... .......... .......... .......... 93%  119M 0s\n",
      " 74500K .......... .......... .......... .......... .......... 93%  111M 0s\n",
      " 74550K .......... .......... .......... .......... .......... 93%  125M 0s\n",
      " 74600K .......... .......... .......... .......... .......... 93%  110M 0s\n",
      " 74650K .......... .......... .......... .......... .......... 93%  118M 0s\n",
      " 74700K .......... .......... .......... .......... .......... 93%  117M 0s\n",
      " 74750K .......... .......... .......... .......... .......... 93%  118M 0s\n",
      " 74800K .......... .......... .......... .......... .......... 93% 90,2M 0s\n",
      " 74850K .......... .......... .......... .......... .......... 93% 99,4M 0s\n",
      " 74900K .......... .......... .......... .......... .......... 93%  144M 0s\n",
      " 74950K .......... .......... .......... .......... .......... 93%  117M 0s\n",
      " 75000K .......... .......... .......... .......... .......... 93%  110M 0s\n",
      " 75050K .......... .......... .......... .......... .......... 94%  117M 0s\n",
      " 75100K .......... .......... .......... .......... .......... 94%  118M 0s\n",
      " 75150K .......... .......... .......... .......... .......... 94%  118M 0s\n",
      " 75200K .......... .......... .......... .......... .......... 94% 88,2M 0s\n",
      " 75250K .......... .......... .......... .......... .......... 94% 95,7M 0s\n",
      " 75300K .......... .......... .......... .......... .......... 94% 28,1M 0s\n",
      " 75350K .......... .......... .......... .......... .......... 94%  126M 0s\n",
      " 75400K .......... .......... .......... .......... .......... 94%  110M 0s\n",
      " 75450K .......... .......... .......... .......... .......... 94%  115M 0s\n",
      " 75500K .......... .......... .......... .......... .......... 94%  120M 0s\n",
      " 75550K .......... .......... .......... .......... .......... 94%  117M 0s\n",
      " 75600K .......... .......... .......... .......... .......... 94% 87,4M 0s\n",
      " 75650K .......... .......... .......... .......... .......... 94% 15,0M 0s\n",
      " 75700K .......... .......... .......... .......... .......... 94% 27,5M 0s\n",
      " 75750K .......... .......... .......... .......... .......... 94% 43,2M 0s\n",
      " 75800K .......... .......... .......... .......... .......... 94%  109M 0s\n",
      " 75850K .......... .......... .......... .......... .......... 95% 56,8M 0s\n",
      " 75900K .......... .......... .......... .......... .......... 95%  118M 0s\n",
      " 75950K .......... .......... .......... .......... .......... 95%  115M 0s\n",
      " 76000K .......... .......... .......... .......... .......... 95% 91,2M 0s\n",
      " 76050K .......... .......... .......... .......... .......... 95%  118M 0s\n",
      " 76100K .......... .......... .......... .......... .......... 95%  118M 0s\n",
      " 76150K .......... .......... .......... .......... .......... 95%  116M 0s\n",
      " 76200K .......... .......... .......... .......... .......... 95%  101M 0s\n",
      " 76250K .......... .......... .......... .......... .......... 95%  132M 0s\n",
      " 76300K .......... .......... .......... .......... .......... 95%  118M 0s\n",
      " 76350K .......... .......... .......... .......... .......... 95%  118M 0s\n",
      " 76400K .......... .......... .......... .......... .......... 95% 1,20M 0s\n",
      " 76450K .......... .......... .......... .......... .......... 95% 1,71M 0s\n",
      " 76500K .......... .......... .......... .......... .......... 95% 8,83M 0s\n",
      " 76550K .......... .......... .......... .......... .......... 95%  132M 0s\n",
      " 76600K .......... .......... .......... .......... .......... 95% 12,0M 0s\n",
      " 76650K .......... .......... .......... .......... .......... 96% 15,9M 0s\n",
      " 76700K .......... .......... .......... .......... .......... 96% 20,6M 0s\n",
      " 76750K .......... .......... .......... .......... .......... 96% 98,3M 0s\n",
      " 76800K .......... .......... .......... .......... .......... 96% 14,1M 0s\n",
      " 76850K .......... .......... .......... .......... .......... 96% 25,0M 0s\n",
      " 76900K .......... .......... .......... .......... .......... 96% 18,8M 0s\n",
      " 76950K .......... .......... .......... .......... .......... 96% 36,5M 0s\n",
      " 77000K .......... .......... .......... .......... .......... 96%  108M 0s\n",
      " 77050K .......... .......... .......... .......... .......... 96%  117M 0s\n",
      " 77100K .......... .......... .......... .......... .......... 96%  115M 0s\n",
      " 77150K .......... .......... .......... .......... .......... 96%  116M 0s\n",
      " 77200K .......... .......... .......... .......... .......... 96% 96,5M 0s\n",
      " 77250K .......... .......... .......... .......... .......... 96%  111M 0s\n",
      " 77300K .......... .......... .......... .......... .......... 96% 92,8M 0s\n",
      " 77350K .......... .......... .......... .......... .......... 96% 59,7M 0s\n",
      " 77400K .......... .......... .......... .......... .......... 96%  111M 0s\n",
      " 77450K .......... .......... .......... .......... .......... 97%  116M 0s\n",
      " 77500K .......... .......... .......... .......... .......... 97%  115M 0s\n",
      " 77550K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 77600K .......... .......... .......... .......... .......... 97% 66,0M 0s\n",
      " 77650K .......... .......... .......... .......... .......... 97%  120M 0s\n",
      " 77700K .......... .......... .......... .......... .......... 97%  168M 0s\n",
      " 77750K .......... .......... .......... .......... .......... 97%  145M 0s\n",
      " 77800K .......... .......... .......... .......... .......... 97%  110M 0s\n",
      " 77850K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 77900K .......... .......... .......... .......... .......... 97%  114M 0s\n",
      " 77950K .......... .......... .......... .......... .......... 97%  115M 0s\n",
      " 78000K .......... .......... .......... .......... .......... 97% 94,3M 0s\n",
      " 78050K .......... .......... .......... .......... .......... 97%  114M 0s\n",
      " 78100K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 78150K .......... .......... .......... .......... .......... 97%  118M 0s\n",
      " 78200K .......... .......... .......... .......... .......... 97% 98,2M 0s\n",
      " 78250K .......... .......... .......... .......... .......... 98%  137M 0s\n",
      " 78300K .......... .......... .......... .......... .......... 98%  113M 0s\n",
      " 78350K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 78400K .......... .......... .......... .......... .......... 98% 35,7M 0s\n",
      " 78450K .......... .......... .......... .......... .......... 98%  115M 0s\n",
      " 78500K .......... .......... .......... .......... .......... 98% 77,7M 0s\n",
      " 78550K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 78600K .......... .......... .......... .......... .......... 98%  112M 0s\n",
      " 78650K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 78700K .......... .......... .......... .......... .......... 98% 62,6M 0s\n",
      " 78750K .......... .......... .......... .......... .......... 98% 13,2M 0s\n",
      " 78800K .......... .......... .......... .......... .......... 98% 45,8M 0s\n",
      " 78850K .......... .......... .......... .......... .......... 98% 70,2M 0s\n",
      " 78900K .......... .......... .......... .......... .......... 98% 55,0M 0s\n",
      " 78950K .......... .......... .......... .......... .......... 98%  122M 0s\n",
      " 79000K .......... .......... .......... .......... .......... 98%  110M 0s\n",
      " 79050K .......... .......... .......... .......... .......... 99%  106M 0s\n",
      " 79100K .......... .......... .......... .......... .......... 99% 67,0M 0s\n",
      " 79150K .......... .......... .......... .......... .......... 99%  116M 0s\n",
      " 79200K .......... .......... .......... .......... .......... 99% 91,8M 0s\n",
      " 79250K .......... .......... .......... .......... .......... 99%  117M 0s\n",
      " 79300K .......... .......... .......... .......... .......... 99%  114M 0s\n",
      " 79350K .......... .......... .......... .......... .......... 99%  118M 0s\n",
      " 79400K .......... .......... .......... .......... .......... 99%  112M 0s\n",
      " 79450K .......... .......... .......... .......... .......... 99%  757K 0s\n",
      " 79500K .......... .......... .......... .......... .......... 99% 29,9M 0s\n",
      " 79550K .......... .......... .......... .......... .......... 99% 8,25M 0s\n",
      " 79600K .......... .......... .......... .......... .......... 99% 11,7M 0s\n",
      " 79650K .......... .......... .......... .......... .......... 99% 45,5M 0s\n",
      " 79700K .......... .......... .......... .......... .......... 99% 24,9M 0s\n",
      " 79750K .......... .......... .......... .......... .......... 99% 17,7M 0s\n",
      " 79800K .......... .......... .......... .......... .......... 99% 11,5M 0s\n",
      " 79850K .......... .......... .....                           100%  138M=4,0s\n",
      "\n",
      "2023-08-18 16:55:49 (19,3 MB/s) - “data/ClinVar/clinvar.vcf.gz” guardado [81792225/81792225]\n",
      "\n",
      "--2023-08-18 16:55:49--  https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/clinvar.vcf.gz.tbi\n",
      "Resolviendo ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.7, 130.14.250.10, 2607:f220:41f:250::228, ...\n",
      "Conectando con ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)[130.14.250.7]:443... conectado.\n",
      "Petición HTTP enviada, esperando respuesta... 200 OK\n",
      "Longitud: 506114 (494K) [application/x-gzip]\n",
      "Grabando a: “data/ClinVar/clinvar.vcf.gz.tbi”\n",
      "\n",
      "     0K .......... .......... .......... .......... .......... 10%  192K 2s\n",
      "    50K .......... .......... .......... .......... .......... 20%  385K 2s\n",
      "   100K .......... .......... .......... .......... .......... 30% 27,7M 1s\n",
      "   150K .......... .......... .......... .......... .......... 40%  390K 1s\n",
      "   200K .......... .......... .......... .......... .......... 50% 52,3M 1s\n",
      "   250K .......... .......... .......... .......... .......... 60% 34,7M 0s\n",
      "   300K .......... .......... .......... .......... .......... 70% 70,3M 0s\n",
      "   350K .......... .......... .......... .......... .......... 80%  395K 0s\n",
      "   400K .......... .......... .......... .......... .......... 91% 31,4M 0s\n",
      "   450K .......... .......... .......... .......... ....      100% 95,8M=0,7s\n",
      "\n",
      "2023-08-18 16:55:50 (758 KB/s) - “data/ClinVar/clinvar.vcf.gz.tbi” guardado [506114/506114]\n",
      "\n",
      "[Fri Aug 18 16:56:21 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T165539.270947.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:56:23 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753368_chr3.vcf\n",
      "    output: results/variants/vep/ERR753368_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753368_chr3.txt\n",
      "    wildcards: sample=ERR753368_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 16:56:56 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T165622.082924.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:56:58 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753369_chr3.vcf\n",
      "    output: results/variants/vep/ERR753369_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753369_chr3.txt\n",
      "    wildcards: sample=ERR753369_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 16:57:32 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T165657.155105.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:57:33 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753370_chr3.vcf\n",
      "    output: results/variants/vep/ERR753370_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753370_chr3.txt\n",
      "    wildcards: sample=ERR753370_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 16:58:13 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T165732.690860.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:58:14 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753371_chr3.vcf\n",
      "    output: results/variants/vep/ERR753371_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753371_chr3.txt\n",
      "    wildcards: sample=ERR753371_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 16:58:40 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T165813.378999.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:58:41 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753372_chr3.vcf\n",
      "    output: results/variants/vep/ERR753372_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753372_chr3.txt\n",
      "    wildcards: sample=ERR753372_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 16:59:16 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T165840.496372.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:59:17 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753373_chr3.vcf\n",
      "    output: results/variants/vep/ERR753373_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753373_chr3.txt\n",
      "    wildcards: sample=ERR753373_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 16:59:54 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T165916.506462.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 16:59:56 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753374_chr3.vcf\n",
      "    output: results/variants/vep/ERR753374_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753374_chr3.txt\n",
      "    wildcards: sample=ERR753374_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 17:00:23 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T165955.146032.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:00:24 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753375_chr3.vcf\n",
      "    output: results/variants/vep/ERR753375_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753375_chr3.txt\n",
      "    wildcards: sample=ERR753375_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 17:00:52 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T170023.868146.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:00:53 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753376_chr3.vcf\n",
      "    output: results/variants/vep/ERR753376_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753376_chr3.txt\n",
      "    wildcards: sample=ERR753376_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 17:01:22 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T170052.781028.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:01:24 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753377_chr3.vcf\n",
      "    output: results/variants/vep/ERR753377_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753377_chr3.txt\n",
      "    wildcards: sample=ERR753377_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 17:01:54 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T170123.050831.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:01:55 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753378_chr3.vcf\n",
      "    output: results/variants/vep/ERR753378_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753378_chr3.txt\n",
      "    wildcards: sample=ERR753378_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 17:02:29 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T170154.510173.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job              count    min threads    max threads\n",
      "-------------  -------  -------------  -------------\n",
      "parsing_dataR        1              1              1\n",
      "total                1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:02:30 2023]\n",
      "rule parsing_dataR:\n",
      "    input: code/05parsing_vep_data.R\n",
      "    output: tasks/13parsing_dataR.done\n",
      "    jobid: 0\n",
      "    reason: Missing output files: tasks/13parsing_dataR.done\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Warning message:\n",
      "package ‘glue’ was built under R version 4.2.3 \n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">>> This is your sample list: ERR696683_chr3\n",
      ">>> This is your sample list: ERR753368_chr3\n",
      ">>> This is your sample list: ERR753369_chr3\n",
      ">>> This is your sample list: ERR753370_chr3\n",
      ">>> This is your sample list: ERR753371_chr3\n",
      ">>> This is your sample list: ERR753372_chr3\n",
      ">>> This is your sample list: ERR753373_chr3\n",
      ">>> This is your sample list: ERR753374_chr3\n",
      ">>> This is your sample list: ERR753375_chr3\n",
      ">>> This is your sample list: ERR753376_chr3\n",
      ">>> This is your sample list: ERR753377_chr3\n",
      ">>> This is your sample list: ERR753378_chr3\n",
      ">>> This is your gene: PIK3CA\n",
      "# A tibble: 33 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "2 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   downstream…\n",
      "3 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   downstream…\n",
      "4 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "5 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "6 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 27 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 26 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 3_179204486_C/A       3:1792044… A      ENSG… ENST00… Transcript   splice_pol…\n",
      "2 3_179204486_C/A       3:1792044… A      ENSG… ENST00… Transcript   splice_pol…\n",
      "3 3_179204486_C/A       3:1792044… A      ENSG… ENST00… Transcript   splice_pol…\n",
      "4 3_179204486_C/A       3:1792044… A      ENSG… ENST00… Transcript   splice_pol…\n",
      "5 3_179204486_C/A       3:1792044… A      ENSG… ENST00… Transcript   splice_pol…\n",
      "6 3_179204642_A/G       3:1792046… G      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 20 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 8 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 3_179226200_A/ACTTGA  3:1792262… ACTTGA ENSG… ENST00… Transcript   intron_var…\n",
      "2 3_179226200_A/ACTTGA  3:1792262… ACTTGA ENSG… ENST00… Transcript   intron_var…\n",
      "3 3_179226200_A/ACTTGA  3:1792262… ACTTGA ENSG… ENST00… Transcript   intron_var…\n",
      "4 3_179226200_A/ACTTGA  3:1792262… ACTTGA ENSG… ENST00… Transcript   intron_var…\n",
      "5 3_179226200_A/ACTTGA  3:1792262… ACTTGA ENSG… ENST00… Transcript   intron_var…\n",
      "6 3_179226200_A/ACTTGA  3:1792262… ACTTGA ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 2 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 33 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "2 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   downstream…\n",
      "3 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   downstream…\n",
      "4 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "5 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "6 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 27 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 0 × 44\n",
      "# ℹ 44 variables: #uploaded_variation <chr>, location <chr>, allele <chr>,\n",
      "#   gene <chr>, feature <chr>, feature_type <chr>, consequence <chr>,\n",
      "#   cdna_position <chr>, cds_position <chr>, protein_position <chr>,\n",
      "#   amino_acids <chr>, codons <chr>, existing_variation <chr>, impact <chr>,\n",
      "#   distance <chr>, strand <chr>, flags <chr>, variant_class <chr>,\n",
      "#   symbol <chr>, symbol_source <chr>, hgnc_id <chr>, biotype <chr>,\n",
      "#   mane_select <chr>, mane_plus_clinical <chr>, tsl <chr>, appris <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 26 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 3_179204486_C/A       3:1792044… A      ENSG… ENST00… Transcript   splice_pol…\n",
      "2 3_179204486_C/A       3:1792044… A      ENSG… ENST00… Transcript   splice_pol…\n",
      "3 3_179204486_C/A       3:1792044… A      ENSG… ENST00… Transcript   splice_pol…\n",
      "4 3_179204486_C/A       3:1792044… A      ENSG… ENST00… Transcript   splice_pol…\n",
      "5 3_179204486_C/A       3:1792044… A      ENSG… ENST00… Transcript   splice_pol…\n",
      "6 3_179204642_A/G       3:1792046… G      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 20 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 12 × 44\n",
      "  `#uploaded_variation`   location allele gene  feature feature_type consequence\n",
      "  <chr>                   <chr>    <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 3_179170077_CACACACACA… 3:17917… CACAC… ENSG… ENST00… Transcript   intron_var…\n",
      "2 3_179170077_CACACACACA… 3:17917… CACAC… ENSG… ENST00… Transcript   intron_var…\n",
      "3 3_179170077_CACACACACA… 3:17917… CACAC… ENSG… ENST00… Transcript   intron_var…\n",
      "4 3_179170077_CACACACACA… 3:17917… CACAC… ENSG… ENST00… Transcript   intron_var…\n",
      "5 3_179170077_CACACACACA… 3:17917… CACAC… ENSG… ENST00… Transcript   intron_var…\n",
      "6 3_179198731_T/C         3:17919… C      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 6 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 5 × 44\n",
      "  `#uploaded_variation`   location allele gene  feature feature_type consequence\n",
      "  <chr>                   <chr>    <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 3_179170077_CACACACACA… 3:17917… CACAC… ENSG… ENST00… Transcript   intron_var…\n",
      "2 3_179170077_CACACACACA… 3:17917… CACAC… ENSG… ENST00… Transcript   intron_var…\n",
      "3 3_179170077_CACACACACA… 3:17917… CACAC… ENSG… ENST00… Transcript   intron_var…\n",
      "4 3_179170077_CACACACACA… 3:17917… CACAC… ENSG… ENST00… Transcript   intron_var…\n",
      "5 3_179170077_CACACACACA… 3:17917… CACAC… ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>,\n",
      "#   freqs <chr>, clin_sig <chr>, somatic <chr>, pheno <chr>, pubmed <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 31 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "2 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   downstream…\n",
      "3 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   downstream…\n",
      "4 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "5 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "6 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 25 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 8 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 3_179226200_A/ACTTGA  3:1792262… ACTTGA ENSG… ENST00… Transcript   intron_var…\n",
      "2 3_179226200_A/ACTTGA  3:1792262… ACTTGA ENSG… ENST00… Transcript   intron_var…\n",
      "3 3_179226200_A/ACTTGA  3:1792262… ACTTGA ENSG… ENST00… Transcript   intron_var…\n",
      "4 3_179226200_A/ACTTGA  3:1792262… ACTTGA ENSG… ENST00… Transcript   intron_var…\n",
      "5 3_179226200_A/ACTTGA  3:1792262… ACTTGA ENSG… ENST00… Transcript   intron_var…\n",
      "6 3_179226200_A/ACTTGA  3:1792262… ACTTGA ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 2 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 25 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "2 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   downstream…\n",
      "3 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   downstream…\n",
      "4 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "5 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "6 3_179203851_C/A       3:1792038… A      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 19 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "Touching output file tasks/13parsing_dataR.done.\n",
      "[Fri Aug 18 17:02:34 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T170230.070985.snakemake.log\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 86 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 3_179199217_A/G       3:1791992… G      ENSG… ENST00… Transcript   intron_var…\n",
      "2 3_179199217_A/G       3:1791992… G      ENSG… ENST00… Transcript   downstream…\n",
      "3 3_179199217_A/G       3:1791992… G      ENSG… ENST00… Transcript   downstream…\n",
      "4 3_179199217_A/G       3:1791992… G      ENSG… ENST00… Transcript   intron_var…\n",
      "5 3_179199217_A/G       3:1791992… G      ENSG… ENST00… Transcript   upstream_g…\n",
      "6 3_179199217_A/G       3:1791992… G      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 80 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "[1] \"##==========================##\"\n",
      "[1] \"##===> WORK FINISHED!!! <===##\"\n",
      "[1] \"##==========================##\"\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['snakemake', '--cores', '1', 'parsing_dataR'], returncode=0)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Running the workflow for the 3º chormosome\n",
    "\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"data/reference/genome.fa\"]) ## rule reference_genome \n",
    "\n",
    "for sample in sample_list:\n",
    "    ## Quality inspection\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/fastqc_result/\" + sample + \"_chr3_1_fastqc.html\",   \n",
    "                    \"data/processed/\" + sample + \"_chr3_1_fastp.fastq.gz\",   \n",
    "                    \"results/fastqc_result/trimmed/\" + sample + \"_chr3_1_fastp_fastqc.html\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    ## Mapping reads\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/mapped_reads/\" + sample + \"_chr3_sorted.sam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr3_sorted.bam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr3_dedup.bam\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    # Extracting variants  \n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/\" + sample + \"_chr3.vcf\"])   \n",
    "\n",
    "## Downloading VEP\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"vep_install_db\"])\n",
    "## VEP CLI\n",
    "for sample in sample_list:\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/vep/\" + sample + \"_chr3.txt\"])\n",
    "\n",
    "## Extracting the gene PIKCA3 from the VEP files\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"parsing_dataR\"])\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Changing the config file to do the chromosome 5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['sed', '-i', '50s/PIK3CA/APC/g', 'config.yaml'], returncode=0)"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subprocess.run([\"sed\", \"-i\", \"6,+12s/chr3/chr5/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\",\"24s/chr3/chr5/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"31s/chromosome.3.fa.gz/chromosome.5.fa.gz/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"50s/PIK3CA/APC/g\", \"config.yaml\"])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job                 count    min threads    max threads\n",
      "----------------  -------  -------------  -------------\n",
      "reference_genome        1              1              1\n",
      "total                   1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:02:54 2023]\n",
      "rule reference_genome:\n",
      "    output: data/reference/genome.fa\n",
      "    jobid: 0\n",
      "    reason: Forced execution\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "--2023-08-18 17:02:54--  https://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.5.fa.gz\n",
      "Resolviendo ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.169\n",
      "Conectando con ftp.ensembl.org (ftp.ensembl.org)[193.62.193.169]:443... conectado.\n",
      "Petición HTTP enviada, esperando respuesta... 200 OK\n",
      "Longitud: 54485176 (52M) [application/x-gzip]\n",
      "Guardando como: ‘data/reference/genome.fa.gz’\n",
      "\n",
      "     0K .......... .......... .......... .......... ..........  0%  440K 2m1s\n",
      "    50K .......... .......... .......... .......... ..........  0% 1,51M 78s\n",
      "   100K .......... .......... .......... .......... ..........  0% 1,89M 61s\n",
      "   150K .......... .......... .......... .......... ..........  0% 1,60M 54s\n",
      "   200K .......... .......... .......... .......... ..........  0% 1,96M 48s\n",
      "   250K .......... .......... .......... .......... ..........  0% 29,2M 40s\n",
      "   300K .......... .......... .......... .......... ..........  0% 94,0M 35s\n",
      "   350K .......... .......... .......... .......... ..........  0% 72,0M 30s\n",
      "   400K .......... .......... .......... .......... ..........  0% 1,58M 31s\n",
      "   450K .......... .......... .......... .......... ..........  0% 2,04M 30s\n",
      "   500K .......... .......... .......... .......... ..........  1% 61,2M 27s\n",
      "   550K .......... .......... .......... .......... ..........  1% 59,1M 25s\n",
      "   600K .......... .......... .......... .......... ..........  1% 47,6M 23s\n",
      "   650K .......... .......... .......... .......... ..........  1%  116M 22s\n",
      "   700K .......... .......... .......... .......... ..........  1% 99,3M 20s\n",
      "   750K .......... .......... .......... .......... ..........  1%  154M 19s\n",
      "   800K .......... .......... .......... .......... ..........  1% 1,74M 20s\n",
      "   850K .......... .......... .......... .......... ..........  1% 27,8M 19s\n",
      "   900K .......... .......... .......... .......... ..........  1% 62,1M 18s\n",
      "   950K .......... .......... .......... .......... ..........  1% 2,16M 18s\n",
      "  1000K .......... .......... .......... .......... ..........  1%  118M 17s\n",
      "  1050K .......... .......... .......... .......... ..........  2% 68,1M 16s\n",
      "  1100K .......... .......... .......... .......... ..........  2% 63,7M 16s\n",
      "  1150K .......... .......... .......... .......... ..........  2%  118M 15s\n",
      "  1200K .......... .......... .......... .......... ..........  2%  103M 14s\n",
      "  1250K .......... .......... .......... .......... ..........  2%  121M 14s\n",
      "  1300K .......... .......... .......... .......... ..........  2%  120M 13s\n",
      "  1350K .......... .......... .......... .......... ..........  2%  102M 13s\n",
      "  1400K .......... .......... .......... .......... ..........  2%  118M 12s\n",
      "  1450K .......... .......... .......... .......... ..........  2%  103M 12s\n",
      "  1500K .......... .......... .......... .......... ..........  2%  120M 12s\n",
      "  1550K .......... .......... .......... .......... ..........  3%  118M 11s\n",
      "  1600K .......... .......... .......... .......... ..........  3%  103M 11s\n",
      "  1650K .......... .......... .......... .......... ..........  3% 1,88M 11s\n",
      "  1700K .......... .......... .......... .......... ..........  3% 69,4M 11s\n",
      "  1750K .......... .......... .......... .......... ..........  3% 52,0M 11s\n",
      "  1800K .......... .......... .......... .......... ..........  3%  114M 10s\n",
      "  1850K .......... .......... .......... .......... ..........  3%  100M 10s\n",
      "  1900K .......... .......... .......... .......... ..........  3% 2,38M 10s\n",
      "  1950K .......... .......... .......... .......... ..........  3% 24,2M 10s\n",
      "  2000K .......... .......... .......... .......... ..........  3%  112M 10s\n",
      "  2050K .......... .......... .......... .......... ..........  3%  115M 10s\n",
      "  2100K .......... .......... .......... .......... ..........  4%  107M 10s\n",
      "  2150K .......... .......... .......... .......... ..........  4% 54,2M 9s\n",
      "  2200K .......... .......... .......... .......... ..........  4%  104M 9s\n",
      "  2250K .......... .......... .......... .......... ..........  4%  119M 9s\n",
      "  2300K .......... .......... .......... .......... ..........  4%  118M 9s\n",
      "  2350K .......... .......... .......... .......... ..........  4%  105M 9s\n",
      "  2400K .......... .......... .......... .......... ..........  4%  117M 8s\n",
      "  2450K .......... .......... .......... .......... ..........  4%  102M 8s\n",
      "  2500K .......... .......... .......... .......... ..........  4%  110M 8s\n",
      "  2550K .......... .......... .......... .......... ..........  4%  117M 8s\n",
      "  2600K .......... .......... .......... .......... ..........  4%  102M 8s\n",
      "  2650K .......... .......... .......... .......... ..........  5%  124M 8s\n",
      "  2700K .......... .......... .......... .......... ..........  5%  105M 7s\n",
      "  2750K .......... .......... .......... .......... ..........  5%  120M 7s\n",
      "  2800K .......... .......... .......... .......... ..........  5%  117M 7s\n",
      "  2850K .......... .......... .......... .......... ..........  5%  100M 7s\n",
      "  2900K .......... .......... .......... .......... ..........  5%  135M 7s\n",
      "  2950K .......... .......... .......... .......... ..........  5%  119M 7s\n",
      "  3000K .......... .......... .......... .......... ..........  5%  131M 7s\n",
      "  3050K .......... .......... .......... .......... ..........  5%  141M 7s\n",
      "  3100K .......... .......... .......... .......... ..........  5%  127M 7s\n",
      "  3150K .......... .......... .......... .......... ..........  6%  152M 6s\n",
      "  3200K .......... .......... .......... .......... ..........  6%  140M 6s\n",
      "  3250K .......... .......... .......... .......... ..........  6%  125M 6s\n",
      "  3300K .......... .......... .......... .......... ..........  6%  118M 6s\n",
      "  3350K .......... .......... .......... .......... ..........  6% 2,56M 6s\n",
      "  3400K .......... .......... .......... .......... ..........  6% 24,7M 6s\n",
      "  3450K .......... .......... .......... .......... ..........  6% 55,4M 6s\n",
      "  3500K .......... .......... .......... .......... ..........  6%  125M 6s\n",
      "  3550K .......... .......... .......... .......... ..........  6%  118M 6s\n",
      "  3600K .......... .......... .......... .......... ..........  6%  100M 6s\n",
      "  3650K .......... .......... .......... .......... ..........  6%  120M 6s\n",
      "  3700K .......... .......... .......... .......... ..........  7%  101M 6s\n",
      "  3750K .......... .......... .......... .......... ..........  7%  117M 6s\n",
      "  3800K .......... .......... .......... .......... ..........  7%  127M 6s\n",
      "  3850K .......... .......... .......... .......... ..........  7% 2,74M 6s\n",
      "  3900K .......... .......... .......... .......... ..........  7% 25,6M 6s\n",
      "  3950K .......... .......... .......... .......... ..........  7% 42,1M 6s\n",
      "  4000K .......... .......... .......... .......... ..........  7%  116M 6s\n",
      "  4050K .......... .......... .......... .......... ..........  7%  121M 5s\n",
      "  4100K .......... .......... .......... .......... ..........  7%  100M 5s\n",
      "  4150K .......... .......... .......... .......... ..........  7%  122M 5s\n",
      "  4200K .......... .......... .......... .......... ..........  7% 67,5M 5s\n",
      "  4250K .......... .......... .......... .......... ..........  8%  140M 5s\n",
      "  4300K .......... .......... .......... .......... ..........  8%  149M 5s\n",
      "  4350K .......... .......... .......... .......... ..........  8%  118M 5s\n",
      "  4400K .......... .......... .......... .......... ..........  8%  127M 5s\n",
      "  4450K .......... .......... .......... .......... ..........  8%  104M 5s\n",
      "  4500K .......... .......... .......... .......... ..........  8%  122M 5s\n",
      "  4550K .......... .......... .......... .......... ..........  8%  118M 5s\n",
      "  4600K .......... .......... .......... .......... ..........  8%  102M 5s\n",
      "  4650K .......... .......... .......... .......... ..........  8%  123M 5s\n",
      "  4700K .......... .......... .......... .......... ..........  8%  100M 5s\n",
      "  4750K .......... .......... .......... .......... ..........  9%  121M 5s\n",
      "  4800K .......... .......... .......... .......... ..........  9%  121M 5s\n",
      "  4850K .......... .......... .......... .......... ..........  9%  102M 5s\n",
      "  4900K .......... .......... .......... .......... ..........  9%  116M 5s\n",
      "  4950K .......... .......... .......... .......... ..........  9%  102M 4s\n",
      "  5000K .......... .......... .......... .......... ..........  9%  121M 4s\n",
      "  5050K .......... .......... .......... .......... ..........  9%  125M 4s\n",
      "  5100K .......... .......... .......... .......... ..........  9%  101M 4s\n",
      "  5150K .......... .......... .......... .......... ..........  9%  120M 4s\n",
      "  5200K .......... .......... .......... .......... ..........  9%  100M 4s\n",
      "  5250K .......... .......... .......... .......... ..........  9%  122M 4s\n",
      "  5300K .......... .......... .......... .......... .......... 10%  118M 4s\n",
      "  5350K .......... .......... .......... .......... .......... 10%  103M 4s\n",
      "  5400K .......... .......... .......... .......... .......... 10%  122M 4s\n",
      "  5450K .......... .......... .......... .......... .......... 10% 49,5M 4s\n",
      "  5500K .......... .......... .......... .......... .......... 10%  123M 4s\n",
      "  5550K .......... .......... .......... .......... .......... 10%  323M 4s\n",
      "  5600K .......... .......... .......... .......... .......... 10%  207M 4s\n",
      "  5650K .......... .......... .......... .......... .......... 10%  120M 4s\n",
      "  5700K .......... .......... .......... .......... .......... 10%  100M 4s\n",
      "  5750K .......... .......... .......... .......... .......... 10%  116M 4s\n",
      "  5800K .......... .......... .......... .......... .......... 10%  123M 4s\n",
      "  5850K .......... .......... .......... .......... .......... 11%  104M 4s\n",
      "  5900K .......... .......... .......... .......... .......... 11%  117M 4s\n",
      "  5950K .......... .......... .......... .......... .......... 11%  103M 4s\n",
      "  6000K .......... .......... .......... .......... .......... 11%  120M 4s\n",
      "  6050K .......... .......... .......... .......... .......... 11%  122M 4s\n",
      "  6100K .......... .......... .......... .......... .......... 11%  102M 4s\n",
      "  6150K .......... .......... .......... .......... .......... 11%  117M 4s\n",
      "  6200K .......... .......... .......... .......... .......... 11%  103M 4s\n",
      "  6250K .......... .......... .......... .......... .......... 11%  120M 4s\n",
      "  6300K .......... .......... .......... .......... .......... 11%  117M 4s\n",
      "  6350K .......... .......... .......... .......... .......... 12%  102M 3s\n",
      "  6400K .......... .......... .......... .......... .......... 12% 6,23M 4s\n",
      "  6450K .......... .......... .......... .......... .......... 12% 11,9M 4s\n",
      "  6500K .......... .......... .......... .......... .......... 12% 51,5M 3s\n",
      "  6550K .......... .......... .......... .......... .......... 12% 72,6M 3s\n",
      "  6600K .......... .......... .......... .......... .......... 12% 98,1M 3s\n",
      "  6650K .......... .......... .......... .......... .......... 12%  125M 3s\n",
      "  6700K .......... .......... .......... .......... .......... 12%  105M 3s\n",
      "  6750K .......... .......... .......... .......... .......... 12%  116M 3s\n",
      "  6800K .......... .......... .......... .......... .......... 12%  120M 3s\n",
      "  6850K .......... .......... .......... .......... .......... 12%  102M 3s\n",
      "  6900K .......... .......... .......... .......... .......... 13% 3,20M 3s\n",
      "  6950K .......... .......... .......... .......... .......... 13% 11,6M 3s\n",
      "  7000K .......... .......... .......... .......... .......... 13% 49,9M 3s\n",
      "  7050K .......... .......... .......... .......... .......... 13% 72,2M 3s\n",
      "  7100K .......... .......... .......... .......... .......... 13% 91,9M 3s\n",
      "  7150K .......... .......... .......... .......... .......... 13%  137M 3s\n",
      "  7200K .......... .......... .......... .......... .......... 13% 99,8M 3s\n",
      "  7250K .......... .......... .......... .......... .......... 13%  128M 3s\n",
      "  7300K .......... .......... .......... .......... .......... 13%  115M 3s\n",
      "  7350K .......... .......... .......... .......... .......... 13%  100M 3s\n",
      "  7400K .......... .......... .......... .......... .......... 14%  121M 3s\n",
      "  7450K .......... .......... .......... .......... .......... 14%  104M 3s\n",
      "  7500K .......... .......... .......... .......... .......... 14%  116M 3s\n",
      "  7550K .......... .......... .......... .......... .......... 14%  118M 3s\n",
      "  7600K .......... .......... .......... .......... .......... 14%  104M 3s\n",
      "  7650K .......... .......... .......... .......... .......... 14%  123M 3s\n",
      "  7700K .......... .......... .......... .......... .......... 14%  103M 3s\n",
      "  7750K .......... .......... .......... .......... .......... 14%  114M 3s\n",
      "  7800K .......... .......... .......... .......... .......... 14%  125M 3s\n",
      "  7850K .......... .......... .......... .......... .......... 14%  102M 3s\n",
      "  7900K .......... .......... .......... .......... .......... 14%  117M 3s\n",
      "  7950K .......... .......... .......... .......... .......... 15%  104M 3s\n",
      "  8000K .......... .......... .......... .......... .......... 15% 79,6M 3s\n",
      "  8050K .......... .......... .......... .......... .......... 15%  177M 3s\n",
      "  8100K .......... .......... .......... .......... .......... 15%  123M 3s\n",
      "  8150K .......... .......... .......... .......... .......... 15%  119M 3s\n",
      "  8200K .......... .......... .......... .......... .......... 15%  101M 3s\n",
      "  8250K .......... .......... .......... .......... .......... 15%  125M 3s\n",
      "  8300K .......... .......... .......... .......... .......... 15% 56,1M 3s\n",
      "  8350K .......... .......... .......... .......... .......... 15% 92,7M 3s\n",
      "  8400K .......... .......... .......... .......... .......... 15%  111M 3s\n",
      "  8450K .......... .......... .......... .......... .......... 15%  198M 3s\n",
      "  8500K .......... .......... .......... .......... .......... 16%  215M 3s\n",
      "  8550K .......... .......... .......... .......... .......... 16%  170M 3s\n",
      "  8600K .......... .......... .......... .......... .......... 16%  102M 3s\n",
      "  8650K .......... .......... .......... .......... .......... 16%  119M 3s\n",
      "  8700K .......... .......... .......... .......... .......... 16%  102M 3s\n",
      "  8750K .......... .......... .......... .......... .......... 16%  121M 3s\n",
      "  8800K .......... .......... .......... .......... .......... 16%  122M 3s\n",
      "  8850K .......... .......... .......... .......... .......... 16%  103M 3s\n",
      "  8900K .......... .......... .......... .......... .......... 16%  117M 3s\n",
      "  8950K .......... .......... .......... .......... .......... 16%  102M 3s\n",
      "  9000K .......... .......... .......... .......... .......... 17%  121M 3s\n",
      "  9050K .......... .......... .......... .......... .......... 17%  117M 3s\n",
      "  9100K .......... .......... .......... .......... .......... 17%  103M 3s\n",
      "  9150K .......... .......... .......... .......... .......... 17%  121M 3s\n",
      "  9200K .......... .......... .......... .......... .......... 17%  102M 3s\n",
      "  9250K .......... .......... .......... .......... .......... 17%  121M 3s\n",
      "  9300K .......... .......... .......... .......... .......... 17%  117M 3s\n",
      "  9350K .......... .......... .......... .......... .......... 17%  102M 2s\n",
      "  9400K .......... .......... .......... .......... .......... 17%  121M 2s\n",
      "  9450K .......... .......... .......... .......... .......... 17%  102M 2s\n",
      "  9500K .......... .......... .......... .......... .......... 17% 4,17M 3s\n",
      "  9550K .......... .......... .......... .......... .......... 18% 99,8M 2s\n",
      "  9600K .......... .......... .......... .......... .......... 18% 66,8M 2s\n",
      "  9650K .......... .......... .......... .......... .......... 18%  122M 2s\n",
      "  9700K .......... .......... .......... .......... .......... 18%  107M 2s\n",
      "  9750K .......... .......... .......... .......... .......... 18%  121M 2s\n",
      "  9800K .......... .......... .......... .......... .......... 18% 77,9M 2s\n",
      "  9850K .......... .......... .......... .......... .......... 18%  102M 2s\n",
      "  9900K .......... .......... .......... .......... .......... 18% 86,4M 2s\n",
      "  9950K .......... .......... .......... .......... .......... 18% 5,54M 2s\n",
      " 10000K .......... .......... .......... .......... .......... 18% 4,58M 2s\n",
      " 10050K .......... .......... .......... .......... .......... 18%  116M 2s\n",
      " 10100K .......... .......... .......... .......... .......... 19% 56,4M 2s\n",
      " 10150K .......... .......... .......... .......... .......... 19% 67,9M 2s\n",
      " 10200K .......... .......... .......... .......... .......... 19%  103M 2s\n",
      " 10250K .......... .......... .......... .......... .......... 19%  114M 2s\n",
      " 10300K .......... .......... .......... .......... .......... 19%  126M 2s\n",
      " 10350K .......... .......... .......... .......... .......... 19% 83,8M 2s\n",
      " 10400K .......... .......... .......... .......... .......... 19%  145M 2s\n",
      " 10450K .......... .......... .......... .......... .......... 19%  103M 2s\n",
      " 10500K .......... .......... .......... .......... .......... 19%  118M 2s\n",
      " 10550K .......... .......... .......... .......... .......... 19%  121M 2s\n",
      " 10600K .......... .......... .......... .......... .......... 20%  103M 2s\n",
      " 10650K .......... .......... .......... .......... .......... 20%  115M 2s\n",
      " 10700K .......... .......... .......... .......... .......... 20% 50,0M 2s\n",
      " 10750K .......... .......... .......... .......... .......... 20%  250M 2s\n",
      " 10800K .......... .......... .......... .......... .......... 20%  420M 2s\n",
      " 10850K .......... .......... .......... .......... .......... 20%  102M 2s\n",
      " 10900K .......... .......... .......... .......... .......... 20%  118M 2s\n",
      " 10950K .......... .......... .......... .......... .......... 20%  103M 2s\n",
      " 11000K .......... .......... .......... .......... .......... 20%  125M 2s\n",
      " 11050K .......... .......... .......... .......... .......... 20%  117M 2s\n",
      " 11100K .......... .......... .......... .......... .......... 20%  102M 2s\n",
      " 11150K .......... .......... .......... .......... .......... 21%  122M 2s\n",
      " 11200K .......... .......... .......... .......... .......... 21%  102M 2s\n",
      " 11250K .......... .......... .......... .......... .......... 21%  121M 2s\n",
      " 11300K .......... .......... .......... .......... .......... 21% 77,7M 2s\n",
      " 11350K .......... .......... .......... .......... .......... 21% 61,7M 2s\n",
      " 11400K .......... .......... .......... .......... .......... 21%  122M 2s\n",
      " 11450K .......... .......... .......... .......... .......... 21%  102M 2s\n",
      " 11500K .......... .......... .......... .......... .......... 21%  117M 2s\n",
      " 11550K .......... .......... .......... .......... .......... 21%  122M 2s\n",
      " 11600K .......... .......... .......... .......... .......... 21%  102M 2s\n",
      " 11650K .......... .......... .......... .......... .......... 21%  117M 2s\n",
      " 11700K .......... .......... .......... .......... .......... 22%  103M 2s\n",
      " 11750K .......... .......... .......... .......... .......... 22%  120M 2s\n",
      " 11800K .......... .......... .......... .......... .......... 22%  117M 2s\n",
      " 11850K .......... .......... .......... .......... .......... 22%  103M 2s\n",
      " 11900K .......... .......... .......... .......... .......... 22%  120M 2s\n",
      " 11950K .......... .......... .......... .......... .......... 22%  103M 2s\n",
      " 12000K .......... .......... .......... .......... .......... 22%  120M 2s\n",
      " 12050K .......... .......... .......... .......... .......... 22%  117M 2s\n",
      " 12100K .......... .......... .......... .......... .......... 22%  101M 2s\n",
      " 12150K .......... .......... .......... .......... .......... 22%  124M 2s\n",
      " 12200K .......... .......... .......... .......... .......... 23% 99,8M 2s\n",
      " 12250K .......... .......... .......... .......... .......... 23% 71,4M 2s\n",
      " 12300K .......... .......... .......... .......... .......... 23%  110M 2s\n",
      " 12350K .......... .......... .......... .......... .......... 23%  101M 2s\n",
      " 12400K .......... .......... .......... .......... .......... 23% 58,0M 2s\n",
      " 12450K .......... .......... .......... .......... .......... 23%  109M 2s\n",
      " 12500K .......... .......... .......... .......... .......... 23% 10,8M 2s\n",
      " 12550K .......... .......... .......... .......... .......... 23% 10,4M 2s\n",
      " 12600K .......... .......... .......... .......... .......... 23% 22,0M 2s\n",
      " 12650K .......... .......... .......... .......... .......... 23%  120M 2s\n",
      " 12700K .......... .......... .......... .......... .......... 23% 55,1M 2s\n",
      " 12750K .......... .......... .......... .......... .......... 24%  122M 2s\n",
      " 12800K .......... .......... .......... .......... .......... 24%  118M 2s\n",
      " 12850K .......... .......... .......... .......... .......... 24%  102M 2s\n",
      " 12900K .......... .......... .......... .......... .......... 24%  122M 2s\n",
      " 12950K .......... .......... .......... .......... .......... 24% 8,40M 2s\n",
      " 13000K .......... .......... .......... .......... .......... 24% 23,2M 2s\n",
      " 13050K .......... .......... .......... .......... .......... 24% 4,50M 2s\n",
      " 13100K .......... .......... .......... .......... .......... 24% 38,4M 2s\n",
      " 13150K .......... .......... .......... .......... .......... 24% 43,7M 2s\n",
      " 13200K .......... .......... .......... .......... .......... 24% 57,0M 2s\n",
      " 13250K .......... .......... .......... .......... .......... 24%  135M 2s\n",
      " 13300K .......... .......... .......... .......... .......... 25%  120M 2s\n",
      " 13350K .......... .......... .......... .......... .......... 25%  103M 2s\n",
      " 13400K .......... .......... .......... .......... .......... 25%  117M 2s\n",
      " 13450K .......... .......... .......... .......... .......... 25%  102M 2s\n",
      " 13500K .......... .......... .......... .......... .......... 25%  121M 2s\n",
      " 13550K .......... .......... .......... .......... .......... 25%  121M 2s\n",
      " 13600K .......... .......... .......... .......... .......... 25%  102M 2s\n",
      " 13650K .......... .......... .......... .......... .......... 25%  118M 2s\n",
      " 13700K .......... .......... .......... .......... .......... 25%  103M 2s\n",
      " 13750K .......... .......... .......... .......... .......... 25%  121M 2s\n",
      " 13800K .......... .......... .......... .......... .......... 26%  118M 2s\n",
      " 13850K .......... .......... .......... .......... .......... 26%  102M 2s\n",
      " 13900K .......... .......... .......... .......... .......... 26%  121M 2s\n",
      " 13950K .......... .......... .......... .......... .......... 26%  102M 2s\n",
      " 14000K .......... .......... .......... .......... .......... 26%  117M 2s\n",
      " 14050K .......... .......... .......... .......... .......... 26%  121M 2s\n",
      " 14100K .......... .......... .......... .......... .......... 26%  102M 2s\n",
      " 14150K .......... .......... .......... .......... .......... 26%  113M 2s\n",
      " 14200K .......... .......... .......... .......... .......... 26%  104M 2s\n",
      " 14250K .......... .......... .......... .......... .......... 26%  122M 2s\n",
      " 14300K .......... .......... .......... .......... .......... 26%  119M 2s\n",
      " 14350K .......... .......... .......... .......... .......... 27%  103M 2s\n",
      " 14400K .......... .......... .......... .......... .......... 27%  116M 2s\n",
      " 14450K .......... .......... .......... .......... .......... 27%  103M 2s\n",
      " 14500K .......... .......... .......... .......... .......... 27% 90,2M 2s\n",
      " 14550K .......... .......... .......... .......... .......... 27%  116M 2s\n",
      " 14600K .......... .......... .......... .......... .......... 27%  111M 2s\n",
      " 14650K .......... .......... .......... .......... .......... 27%  115M 2s\n",
      " 14700K .......... .......... .......... .......... .......... 27% 90,1M 2s\n",
      " 14750K .......... .......... .......... .......... .......... 27%  157M 2s\n",
      " 14800K .......... .......... .......... .......... .......... 27%  117M 2s\n",
      " 14850K .......... .......... .......... .......... .......... 28%  102M 2s\n",
      " 14900K .......... .......... .......... .......... .......... 28%  122M 2s\n",
      " 14950K .......... .......... .......... .......... .......... 28%  102M 2s\n",
      " 15000K .......... .......... .......... .......... .......... 28%  117M 2s\n",
      " 15050K .......... .......... .......... .......... .......... 28%  122M 2s\n",
      " 15100K .......... .......... .......... .......... .......... 28%  102M 2s\n",
      " 15150K .......... .......... .......... .......... .......... 28%  121M 2s\n",
      " 15200K .......... .......... .......... .......... .......... 28%  102M 2s\n",
      " 15250K .......... .......... .......... .......... .......... 28%  117M 2s\n",
      " 15300K .......... .......... .......... .......... .......... 28%  121M 2s\n",
      " 15350K .......... .......... .......... .......... .......... 28%  102M 2s\n",
      " 15400K .......... .......... .......... .......... .......... 29%  117M 2s\n",
      " 15450K .......... .......... .......... .......... .......... 29% 50,4M 2s\n",
      " 15500K .......... .......... .......... .......... .......... 29% 11,6M 2s\n",
      " 15550K .......... .......... .......... .......... .......... 29% 39,9M 2s\n",
      " 15600K .......... .......... .......... .......... .......... 29% 10,4M 2s\n",
      " 15650K .......... .......... .......... .......... .......... 29% 20,6M 2s\n",
      " 15700K .......... .......... .......... .......... .......... 29% 60,6M 2s\n",
      " 15750K .......... .......... .......... .......... .......... 29%  120M 2s\n",
      " 15800K .......... .......... .......... .......... .......... 29%  117M 2s\n",
      " 15850K .......... .......... .......... .......... .......... 29%  103M 2s\n",
      " 15900K .......... .......... .......... .......... .......... 29%  120M 2s\n",
      " 15950K .......... .......... .......... .......... .......... 30% 70,8M 2s\n",
      " 16000K .......... .......... .......... .......... .......... 30% 9,09M 2s\n",
      " 16050K .......... .......... .......... .......... .......... 30% 39,2M 2s\n",
      " 16100K .......... .......... .......... .......... .......... 30% 4,40M 2s\n",
      " 16150K .......... .......... .......... .......... .......... 30% 18,5M 2s\n",
      " 16200K .......... .......... .......... .......... .......... 30% 51,0M 2s\n",
      " 16250K .......... .......... .......... .......... .......... 30%  120M 2s\n",
      " 16300K .......... .......... .......... .......... .......... 30%  121M 2s\n",
      " 16350K .......... .......... .......... .......... .......... 30%  102M 2s\n",
      " 16400K .......... .......... .......... .......... .......... 30%  118M 2s\n",
      " 16450K .......... .......... .......... .......... .......... 31%  102M 2s\n",
      " 16500K .......... .......... .......... .......... .......... 31%  114M 2s\n",
      " 16550K .......... .......... .......... .......... .......... 31%  124M 2s\n",
      " 16600K .......... .......... .......... .......... .......... 31%  103M 2s\n",
      " 16650K .......... .......... .......... .......... .......... 31%  119M 2s\n",
      " 16700K .......... .......... .......... .......... .......... 31%  103M 2s\n",
      " 16750K .......... .......... .......... .......... .......... 31%  122M 2s\n",
      " 16800K .......... .......... .......... .......... .......... 31%  117M 1s\n",
      " 16850K .......... .......... .......... .......... .......... 31%  102M 1s\n",
      " 16900K .......... .......... .......... .......... .......... 31%  122M 1s\n",
      " 16950K .......... .......... .......... .......... .......... 31%  102M 1s\n",
      " 17000K .......... .......... .......... .......... .......... 32%  112M 1s\n",
      " 17050K .......... .......... .......... .......... .......... 32%  122M 1s\n",
      " 17100K .......... .......... .......... .......... .......... 32%  106M 1s\n",
      " 17150K .......... .......... .......... .......... .......... 32%  117M 1s\n",
      " 17200K .......... .......... .......... .......... .......... 32% 91,6M 1s\n",
      " 17250K .......... .......... .......... .......... .......... 32% 96,0M 1s\n",
      " 17300K .......... .......... .......... .......... .......... 32%  154M 1s\n",
      " 17350K .......... .......... .......... .......... .......... 32%  117M 1s\n",
      " 17400K .......... .......... .......... .......... .......... 32%  121M 1s\n",
      " 17450K .......... .......... .......... .......... .......... 32% 97,7M 1s\n",
      " 17500K .......... .......... .......... .......... .......... 32%  126M 1s\n",
      " 17550K .......... .......... .......... .......... .......... 33%  115M 1s\n",
      " 17600K .......... .......... .......... .......... .......... 33%  106M 1s\n",
      " 17650K .......... .......... .......... .......... .......... 33%  122M 1s\n",
      " 17700K .......... .......... .......... .......... .......... 33%  102M 1s\n",
      " 17750K .......... .......... .......... .......... .......... 33%  117M 1s\n",
      " 17800K .......... .......... .......... .......... .......... 33%  122M 1s\n",
      " 17850K .......... .......... .......... .......... .......... 33%  102M 1s\n",
      " 17900K .......... .......... .......... .......... .......... 33%  120M 1s\n",
      " 17950K .......... .......... .......... .......... .......... 33%  103M 1s\n",
      " 18000K .......... .......... .......... .......... .......... 33%  117M 1s\n",
      " 18050K .......... .......... .......... .......... .......... 34%  120M 1s\n",
      " 18100K .......... .......... .......... .......... .......... 34%  103M 1s\n",
      " 18150K .......... .......... .......... .......... .......... 34%  117M 1s\n",
      " 18200K .......... .......... .......... .......... .......... 34%  102M 1s\n",
      " 18250K .......... .......... .......... .......... .......... 34%  121M 1s\n",
      " 18300K .......... .......... .......... .......... .......... 34%  119M 1s\n",
      " 18350K .......... .......... .......... .......... .......... 34%  102M 1s\n",
      " 18400K .......... .......... .......... .......... .......... 34%  121M 1s\n",
      " 18450K .......... .......... .......... .......... .......... 34%  102M 1s\n",
      " 18500K .......... .......... .......... .......... .......... 34%  102M 1s\n",
      " 18550K .......... .......... .......... .......... .......... 34% 10,5M 1s\n",
      " 18600K .......... .......... .......... .......... .......... 35% 24,6M 1s\n",
      " 18650K .......... .......... .......... .......... .......... 35%  120M 1s\n",
      " 18700K .......... .......... .......... .......... .......... 35% 8,77M 1s\n",
      " 18750K .......... .......... .......... .......... .......... 35% 46,0M 1s\n",
      " 18800K .......... .......... .......... .......... .......... 35% 64,9M 1s\n",
      " 18850K .......... .......... .......... .......... .......... 35%  102M 1s\n",
      " 18900K .......... .......... .......... .......... .......... 35%  120M 1s\n",
      " 18950K .......... .......... .......... .......... .......... 35%  103M 1s\n",
      " 19000K .......... .......... .......... .......... .......... 35%  117M 1s\n",
      " 19050K .......... .......... .......... .......... .......... 35% 10,0M 1s\n",
      " 19100K .......... .......... .......... .......... .......... 35% 7,96M 1s\n",
      " 19150K .......... .......... .......... .......... .......... 36% 12,1M 1s\n",
      " 19200K .......... .......... .......... .......... .......... 36% 9,82M 1s\n",
      " 19250K .......... .......... .......... .......... .......... 36% 44,5M 1s\n",
      " 19300K .......... .......... .......... .......... .......... 36% 54,8M 1s\n",
      " 19350K .......... .......... .......... .......... .......... 36%  102M 1s\n",
      " 19400K .......... .......... .......... .......... .......... 36%  122M 1s\n",
      " 19450K .......... .......... .......... .......... .......... 36%  102M 1s\n",
      " 19500K .......... .......... .......... .......... .......... 36%  121M 1s\n",
      " 19550K .......... .......... .......... .......... .......... 36%  117M 1s\n",
      " 19600K .......... .......... .......... .......... .......... 36%  102M 1s\n",
      " 19650K .......... .......... .......... .......... .......... 37%  104M 1s\n",
      " 19700K .......... .......... .......... .......... .......... 37%  115M 1s\n",
      " 19750K .......... .......... .......... .......... .......... 37%  122M 1s\n",
      " 19800K .......... .......... .......... .......... .......... 37%  118M 1s\n",
      " 19850K .......... .......... .......... .......... .......... 37%  103M 1s\n",
      " 19900K .......... .......... .......... .......... .......... 37%  119M 1s\n",
      " 19950K .......... .......... .......... .......... .......... 37%  102M 1s\n",
      " 20000K .......... .......... .......... .......... .......... 37%  120M 1s\n",
      " 20050K .......... .......... .......... .......... .......... 37%  120M 1s\n",
      " 20100K .......... .......... .......... .......... .......... 37%  103M 1s\n",
      " 20150K .......... .......... .......... .......... .......... 37%  118M 1s\n",
      " 20200K .......... .......... .......... .......... .......... 38%  102M 1s\n",
      " 20250K .......... .......... .......... .......... .......... 38%  122M 1s\n",
      " 20300K .......... .......... .......... .......... .......... 38%  117M 1s\n",
      " 20350K .......... .......... .......... .......... .......... 38%  102M 1s\n",
      " 20400K .......... .......... .......... .......... .......... 38%  122M 1s\n",
      " 20450K .......... .......... .......... .......... .......... 38%  102M 1s\n",
      " 20500K .......... .......... .......... .......... .......... 38%  122M 1s\n",
      " 20550K .......... .......... .......... .......... .......... 38%  117M 1s\n",
      " 20600K .......... .......... .......... .......... .......... 38%  102M 1s\n",
      " 20650K .......... .......... .......... .......... .......... 38%  121M 1s\n",
      " 20700K .......... .......... .......... .......... .......... 38%  102M 1s\n",
      " 20750K .......... .......... .......... .......... .......... 39%  117M 1s\n",
      " 20800K .......... .......... .......... .......... .......... 39%  122M 1s\n",
      " 20850K .......... .......... .......... .......... .......... 39%  102M 1s\n",
      " 20900K .......... .......... .......... .......... .......... 39%  117M 1s\n",
      " 20950K .......... .......... .......... .......... .......... 39%  102M 1s\n",
      " 21000K .......... .......... .......... .......... .......... 39%  121M 1s\n",
      " 21050K .......... .......... .......... .......... .......... 39%  121M 1s\n",
      " 21100K .......... .......... .......... .......... .......... 39%  102M 1s\n",
      " 21150K .......... .......... .......... .......... .......... 39%  117M 1s\n",
      " 21200K .......... .......... .......... .......... .......... 39%  101M 1s\n",
      " 21250K .......... .......... .......... .......... .......... 40%  122M 1s\n",
      " 21300K .......... .......... .......... .......... .......... 40%  117M 1s\n",
      " 21350K .......... .......... .......... .......... .......... 40%  102M 1s\n",
      " 21400K .......... .......... .......... .......... .......... 40%  121M 1s\n",
      " 21450K .......... .......... .......... .......... .......... 40%  103M 1s\n",
      " 21500K .......... .......... .......... .......... .......... 40%  116M 1s\n",
      " 21550K .......... .......... .......... .......... .......... 40%  121M 1s\n",
      " 21600K .......... .......... .......... .......... .......... 40% 9,78M 1s\n",
      " 21650K .......... .......... .......... .......... .......... 40% 40,3M 1s\n",
      " 21700K .......... .......... .......... .......... .......... 40% 49,1M 1s\n",
      " 21750K .......... .......... .......... .......... .......... 40% 8,50M 1s\n",
      " 21800K .......... .......... .......... .......... .......... 41%  127M 1s\n",
      " 21850K .......... .......... .......... .......... .......... 41% 48,0M 1s\n",
      " 21900K .......... .......... .......... .......... .......... 41% 59,9M 1s\n",
      " 21950K .......... .......... .......... .......... .......... 41%  102M 1s\n",
      " 22000K .......... .......... .......... .......... .......... 41%  122M 1s\n",
      " 22050K .......... .......... .......... .......... .......... 41%  116M 1s\n",
      " 22100K .......... .......... .......... .......... .......... 41% 15,0M 1s\n",
      " 22150K .......... .......... .......... .......... .......... 41% 11,2M 1s\n",
      " 22200K .......... .......... .......... .......... .......... 41% 8,21M 1s\n",
      " 22250K .......... .......... .......... .......... .......... 41% 10,0M 1s\n",
      " 22300K .......... .......... .......... .......... .......... 42% 22,6M 1s\n",
      " 22350K .......... .......... .......... .......... .......... 42% 43,8M 1s\n",
      " 22400K .......... .......... .......... .......... .......... 42% 41,0M 1s\n",
      " 22450K .......... .......... .......... .......... .......... 42% 97,9M 1s\n",
      " 22500K .......... .......... .......... .......... .......... 42%  120M 1s\n",
      " 22550K .......... .......... .......... .......... .......... 42%  121M 1s\n",
      " 22600K .......... .......... .......... .......... .......... 42% 93,7M 1s\n",
      " 22650K .......... .......... .......... .......... .......... 42%  119M 1s\n",
      " 22700K .......... .......... .......... .......... .......... 42% 97,3M 1s\n",
      " 22750K .......... .......... .......... .......... .......... 42%  116M 1s\n",
      " 22800K .......... .......... .......... .......... .......... 42%  120M 1s\n",
      " 22850K .......... .......... .......... .......... .......... 43%  110M 1s\n",
      " 22900K .......... .......... .......... .......... .......... 43%  202M 1s\n",
      " 22950K .......... .......... .......... .......... .......... 43%  155M 1s\n",
      " 23000K .......... .......... .......... .......... .......... 43%  120M 1s\n",
      " 23050K .......... .......... .......... .......... .......... 43%  118M 1s\n",
      " 23100K .......... .......... .......... .......... .......... 43%  103M 1s\n",
      " 23150K .......... .......... .......... .......... .......... 43%  119M 1s\n",
      " 23200K .......... .......... .......... .......... .......... 43%  102M 1s\n",
      " 23250K .......... .......... .......... .......... .......... 43%  122M 1s\n",
      " 23300K .......... .......... .......... .......... .......... 43%  116M 1s\n",
      " 23350K .......... .......... .......... .......... .......... 43% 78,4M 1s\n",
      " 23400K .......... .......... .......... .......... .......... 44%  191M 1s\n",
      " 23450K .......... .......... .......... .......... .......... 44%  103M 1s\n",
      " 23500K .......... .......... .......... .......... .......... 44%  117M 1s\n",
      " 23550K .......... .......... .......... .......... .......... 44%  120M 1s\n",
      " 23600K .......... .......... .......... .......... .......... 44%  103M 1s\n",
      " 23650K .......... .......... .......... .......... .......... 44%  118M 1s\n",
      " 23700K .......... .......... .......... .......... .......... 44%  102M 1s\n",
      " 23750K .......... .......... .......... .......... .......... 44%  119M 1s\n",
      " 23800K .......... .......... .......... .......... .......... 44%  121M 1s\n",
      " 23850K .......... .......... .......... .......... .......... 44%  102M 1s\n",
      " 23900K .......... .......... .......... .......... .......... 45%  118M 1s\n",
      " 23950K .......... .......... .......... .......... .......... 45%  102M 1s\n",
      " 24000K .......... .......... .......... .......... .......... 45%  122M 1s\n",
      " 24050K .......... .......... .......... .......... .......... 45%  119M 1s\n",
      " 24100K .......... .......... .......... .......... .......... 45%  102M 1s\n",
      " 24150K .......... .......... .......... .......... .......... 45%  122M 1s\n",
      " 24200K .......... .......... .......... .......... .......... 45%  102M 1s\n",
      " 24250K .......... .......... .......... .......... .......... 45%  122M 1s\n",
      " 24300K .......... .......... .......... .......... .......... 45%  116M 1s\n",
      " 24350K .......... .......... .......... .......... .......... 45%  102M 1s\n",
      " 24400K .......... .......... .......... .......... .......... 45%  121M 1s\n",
      " 24450K .......... .......... .......... .......... .......... 46%  103M 1s\n",
      " 24500K .......... .......... .......... .......... .......... 46%  117M 1s\n",
      " 24550K .......... .......... .......... .......... .......... 46%  121M 1s\n",
      " 24600K .......... .......... .......... .......... .......... 46%  102M 1s\n",
      " 24650K .......... .......... .......... .......... .......... 46% 20,1M 1s\n",
      " 24700K .......... .......... .......... .......... .......... 46% 12,2M 1s\n",
      " 24750K .......... .......... .......... .......... .......... 46% 43,6M 1s\n",
      " 24800K .......... .......... .......... .......... .......... 46% 22,3M 1s\n",
      " 24850K .......... .......... .......... .......... .......... 46% 13,6M 1s\n",
      " 24900K .......... .......... .......... .......... .......... 46%  118M 1s\n",
      " 24950K .......... .......... .......... .......... .......... 46% 56,4M 1s\n",
      " 25000K .......... .......... .......... .......... .......... 47%  121M 1s\n",
      " 25050K .......... .......... .......... .......... .......... 47%  110M 1s\n",
      " 25100K .......... .......... .......... .......... .......... 47% 54,3M 1s\n",
      " 25150K .......... .......... .......... .......... .......... 47%  125M 1s\n",
      " 25200K .......... .......... .......... .......... .......... 47% 6,82M 1s\n",
      " 25250K .......... .......... .......... .......... .......... 47% 34,9M 1s\n",
      " 25300K .......... .......... .......... .......... .......... 47% 5,62M 1s\n",
      " 25350K .......... .......... .......... .......... .......... 47% 16,5M 1s\n",
      " 25400K .......... .......... .......... .......... .......... 47% 81,6M 1s\n",
      " 25450K .......... .......... .......... .......... .......... 47% 61,5M 1s\n",
      " 25500K .......... .......... .......... .......... .......... 48% 62,2M 1s\n",
      " 25550K .......... .......... .......... .......... .......... 48% 59,8M 1s\n",
      " 25600K .......... .......... .......... .......... .......... 48%  103M 1s\n",
      " 25650K .......... .......... .......... .......... .......... 48%  115M 1s\n",
      " 25700K .......... .......... .......... .......... .......... 48%  103M 1s\n",
      " 25750K .......... .......... .......... .......... .......... 48%  121M 1s\n",
      " 25800K .......... .......... .......... .......... .......... 48%  117M 1s\n",
      " 25850K .......... .......... .......... .......... .......... 48%  101M 1s\n",
      " 25900K .......... .......... .......... .......... .......... 48%  124M 1s\n",
      " 25950K .......... .......... .......... .......... .......... 48%  103M 1s\n",
      " 26000K .......... .......... .......... .......... .......... 48%  120M 1s\n",
      " 26050K .......... .......... .......... .......... .......... 49%  117M 1s\n",
      " 26100K .......... .......... .......... .......... .......... 49%  102M 1s\n",
      " 26150K .......... .......... .......... .......... .......... 49%  121M 1s\n",
      " 26200K .......... .......... .......... .......... .......... 49%  101M 1s\n",
      " 26250K .......... .......... .......... .......... .......... 49%  120M 1s\n",
      " 26300K .......... .......... .......... .......... .......... 49%  121M 1s\n",
      " 26350K .......... .......... .......... .......... .......... 49% 97,4M 1s\n",
      " 26400K .......... .......... .......... .......... .......... 49%  126M 1s\n",
      " 26450K .......... .......... .......... .......... .......... 49%  105M 1s\n",
      " 26500K .......... .......... .......... .......... .......... 49%  117M 1s\n",
      " 26550K .......... .......... .......... .......... .......... 49%  120M 1s\n",
      " 26600K .......... .......... .......... .......... .......... 50%  103M 1s\n",
      " 26650K .......... .......... .......... .......... .......... 50%  117M 1s\n",
      " 26700K .......... .......... .......... .......... .......... 50%  103M 1s\n",
      " 26750K .......... .......... .......... .......... .......... 50%  121M 1s\n",
      " 26800K .......... .......... .......... .......... .......... 50%  118M 1s\n",
      " 26850K .......... .......... .......... .......... .......... 50%  102M 1s\n",
      " 26900K .......... .......... .......... .......... .......... 50%  122M 1s\n",
      " 26950K .......... .......... .......... .......... .......... 50%  101M 1s\n",
      " 27000K .......... .......... .......... .......... .......... 50%  123M 1s\n",
      " 27050K .......... .......... .......... .......... .......... 50%  117M 1s\n",
      " 27100K .......... .......... .......... .......... .......... 51%  102M 1s\n",
      " 27150K .......... .......... .......... .......... .......... 51%  121M 1s\n",
      " 27200K .......... .......... .......... .......... .......... 51%  102M 1s\n",
      " 27250K .......... .......... .......... .......... .......... 51%  117M 1s\n",
      " 27300K .......... .......... .......... .......... .......... 51%  121M 1s\n",
      " 27350K .......... .......... .......... .......... .......... 51%  102M 1s\n",
      " 27400K .......... .......... .......... .......... .......... 51%  118M 1s\n",
      " 27450K .......... .......... .......... .......... .......... 51% 95,8M 1s\n",
      " 27500K .......... .......... .......... .......... .......... 51%  131M 1s\n",
      " 27550K .......... .......... .......... .......... .......... 51%  121M 1s\n",
      " 27600K .......... .......... .......... .......... .......... 51%  103M 1s\n",
      " 27650K .......... .......... .......... .......... .......... 52%  117M 1s\n",
      " 27700K .......... .......... .......... .......... .......... 52% 19,6M 1s\n",
      " 27750K .......... .......... .......... .......... .......... 52% 25,8M 1s\n",
      " 27800K .......... .......... .......... .......... .......... 52% 17,1M 1s\n",
      " 27850K .......... .......... .......... .......... .......... 52% 36,6M 1s\n",
      " 27900K .......... .......... .......... .......... .......... 52% 12,8M 1s\n",
      " 27950K .......... .......... .......... .......... .......... 52% 47,2M 1s\n",
      " 28000K .......... .......... .......... .......... .......... 52% 55,6M 1s\n",
      " 28050K .......... .......... .......... .......... .......... 52%  113M 1s\n",
      " 28100K .......... .......... .......... .......... .......... 52% 80,9M 1s\n",
      " 28150K .......... .......... .......... .......... .......... 52% 95,2M 1s\n",
      " 28200K .......... .......... .......... .......... .......... 53% 80,2M 1s\n",
      " 28250K .......... .......... .......... .......... .......... 53% 8,48M 1s\n",
      " 28300K .......... .......... .......... .......... .......... 53% 22,1M 1s\n",
      " 28350K .......... .......... .......... .......... .......... 53% 6,34M 1s\n",
      " 28400K .......... .......... .......... .......... .......... 53% 12,4M 1s\n",
      " 28450K .......... .......... .......... .......... .......... 53% 36,5M 1s\n",
      " 28500K .......... .......... .......... .......... .......... 53%  125M 1s\n",
      " 28550K .......... .......... .......... .......... .......... 53% 84,9M 1s\n",
      " 28600K .......... .......... .......... .......... .......... 53% 62,8M 1s\n",
      " 28650K .......... .......... .......... .......... .......... 53% 56,3M 1s\n",
      " 28700K .......... .......... .......... .......... .......... 54%  101M 1s\n",
      " 28750K .......... .......... .......... .......... .......... 54%  121M 1s\n",
      " 28800K .......... .......... .......... .......... .......... 54%  115M 1s\n",
      " 28850K .......... .......... .......... .......... .......... 54%  100M 1s\n",
      " 28900K .......... .......... .......... .......... .......... 54%  118M 1s\n",
      " 28950K .......... .......... .......... .......... .......... 54%  104M 1s\n",
      " 29000K .......... .......... .......... .......... .......... 54%  119M 1s\n",
      " 29050K .......... .......... .......... .......... .......... 54%  121M 1s\n",
      " 29100K .......... .......... .......... .......... .......... 54%  104M 1s\n",
      " 29150K .......... .......... .......... .......... .......... 54%  115M 1s\n",
      " 29200K .......... .......... .......... .......... .......... 54%  104M 1s\n",
      " 29250K .......... .......... .......... .......... .......... 55%  117M 1s\n",
      " 29300K .......... .......... .......... .......... .......... 55%  121M 1s\n",
      " 29350K .......... .......... .......... .......... .......... 55%  104M 1s\n",
      " 29400K .......... .......... .......... .......... .......... 55%  117M 1s\n",
      " 29450K .......... .......... .......... .......... .......... 55%  102M 1s\n",
      " 29500K .......... .......... .......... .......... .......... 55% 90,0M 1s\n",
      " 29550K .......... .......... .......... .......... .......... 55%  176M 1s\n",
      " 29600K .......... .......... .......... .......... .......... 55%  101M 1s\n",
      " 29650K .......... .......... .......... .......... .......... 55%  123M 1s\n",
      " 29700K .......... .......... .......... .......... .......... 55%  102M 1s\n",
      " 29750K .......... .......... .......... .......... .......... 56%  122M 1s\n",
      " 29800K .......... .......... .......... .......... .......... 56%  117M 1s\n",
      " 29850K .......... .......... .......... .......... .......... 56%  102M 1s\n",
      " 29900K .......... .......... .......... .......... .......... 56%  121M 1s\n",
      " 29950K .......... .......... .......... .......... .......... 56%  103M 1s\n",
      " 30000K .......... .......... .......... .......... .......... 56%  117M 1s\n",
      " 30050K .......... .......... .......... .......... .......... 56%  121M 1s\n",
      " 30100K .......... .......... .......... .......... .......... 56%  103M 1s\n",
      " 30150K .......... .......... .......... .......... .......... 56%  116M 1s\n",
      " 30200K .......... .......... .......... .......... .......... 56%  103M 1s\n",
      " 30250K .......... .......... .......... .......... .......... 56%  116M 1s\n",
      " 30300K .......... .......... .......... .......... .......... 57%  126M 1s\n",
      " 30350K .......... .......... .......... .......... .......... 57%  101M 1s\n",
      " 30400K .......... .......... .......... .......... .......... 57%  120M 1s\n",
      " 30450K .......... .......... .......... .......... .......... 57%  101M 1s\n",
      " 30500K .......... .......... .......... .......... .......... 57%  122M 1s\n",
      " 30550K .......... .......... .......... .......... .......... 57% 86,0M 1s\n",
      " 30600K .......... .......... .......... .......... .......... 57% 63,0M 1s\n",
      " 30650K .......... .......... .......... .......... .......... 57% 65,9M 1s\n",
      " 30700K .......... .......... .......... .......... .......... 57%  106M 1s\n",
      " 30750K .......... .......... .......... .......... .......... 57%  159M 1s\n",
      " 30800K .......... .......... .......... .......... .......... 57% 21,3M 1s\n",
      " 30850K .......... .......... .......... .......... .......... 58% 13,9M 1s\n",
      " 30900K .......... .......... .......... .......... .......... 58% 45,4M 1s\n",
      " 30950K .......... .......... .......... .......... .......... 58% 15,9M 1s\n",
      " 31000K .......... .......... .......... .......... .......... 58% 23,2M 1s\n",
      " 31050K .......... .......... .......... .......... .......... 58% 63,1M 1s\n",
      " 31100K .......... .......... .......... .......... .......... 58%  100M 1s\n",
      " 31150K .......... .......... .......... .......... .......... 58%  120M 1s\n",
      " 31200K .......... .......... .......... .......... .......... 58%  102M 1s\n",
      " 31250K .......... .......... .......... .......... .......... 58% 76,7M 1s\n",
      " 31300K .......... .......... .......... .......... .......... 58% 16,2M 1s\n",
      " 31350K .......... .......... .......... .......... .......... 59% 9,86M 1s\n",
      " 31400K .......... .......... .......... .......... .......... 59% 7,37M 1s\n",
      " 31450K .......... .......... .......... .......... .......... 59% 8,13M 1s\n",
      " 31500K .......... .......... .......... .......... .......... 59% 50,4M 1s\n",
      " 31550K .......... .......... .......... .......... .......... 59% 48,1M 1s\n",
      " 31600K .......... .......... .......... .......... .......... 59%  122M 1s\n",
      " 31650K .......... .......... .......... .......... .......... 59%  145M 1s\n",
      " 31700K .......... .......... .......... .......... .......... 59%  126M 1s\n",
      " 31750K .......... .......... .......... .......... .......... 59%  152M 1s\n",
      " 31800K .......... .......... .......... .......... .......... 59% 79,4M 1s\n",
      " 31850K .......... .......... .......... .......... .......... 59%  103M 1s\n",
      " 31900K .......... .......... .......... .......... .......... 60%  110M 1s\n",
      " 31950K .......... .......... .......... .......... .......... 60%  113M 1s\n",
      " 32000K .......... .......... .......... .......... .......... 60%  115M 1s\n",
      " 32050K .......... .......... .......... .......... .......... 60%  124M 1s\n",
      " 32100K .......... .......... .......... .......... .......... 60%  103M 1s\n",
      " 32150K .......... .......... .......... .......... .......... 60%  119M 1s\n",
      " 32200K .......... .......... .......... .......... .......... 60%  103M 1s\n",
      " 32250K .......... .......... .......... .......... .......... 60% 91,2M 1s\n",
      " 32300K .......... .......... .......... .......... .......... 60%  124M 1s\n",
      " 32350K .......... .......... .......... .......... .......... 60%  101M 1s\n",
      " 32400K .......... .......... .......... .......... .......... 60%  120M 1s\n",
      " 32450K .......... .......... .......... .......... .......... 61%  102M 1s\n",
      " 32500K .......... .......... .......... .......... .......... 61%  119M 1s\n",
      " 32550K .......... .......... .......... .......... .......... 61%  119M 1s\n",
      " 32600K .......... .......... .......... .......... .......... 61%  102M 1s\n",
      " 32650K .......... .......... .......... .......... .......... 61%  121M 1s\n",
      " 32700K .......... .......... .......... .......... .......... 61%  102M 1s\n",
      " 32750K .......... .......... .......... .......... .......... 61%  118M 1s\n",
      " 32800K .......... .......... .......... .......... .......... 61%  122M 1s\n",
      " 32850K .......... .......... .......... .......... .......... 61%  101M 1s\n",
      " 32900K .......... .......... .......... .......... .......... 61%  124M 1s\n",
      " 32950K .......... .......... .......... .......... .......... 62%  102M 1s\n",
      " 33000K .......... .......... .......... .......... .......... 62%  117M 1s\n",
      " 33050K .......... .......... .......... .......... .......... 62%  117M 1s\n",
      " 33100K .......... .......... .......... .......... .......... 62%  107M 1s\n",
      " 33150K .......... .......... .......... .......... .......... 62%  110M 1s\n",
      " 33200K .......... .......... .......... .......... .......... 62%  103M 1s\n",
      " 33250K .......... .......... .......... .......... .......... 62%  124M 1s\n",
      " 33300K .......... .......... .......... .......... .......... 62%  115M 1s\n",
      " 33350K .......... .......... .......... .......... .......... 62%  107M 1s\n",
      " 33400K .......... .......... .......... .......... .......... 62%  121M 1s\n",
      " 33450K .......... .......... .......... .......... .......... 62%  103M 1s\n",
      " 33500K .......... .......... .......... .......... .......... 63%  120M 1s\n",
      " 33550K .......... .......... .......... .......... .......... 63%  117M 1s\n",
      " 33600K .......... .......... .......... .......... .......... 63% 97,4M 1s\n",
      " 33650K .......... .......... .......... .......... .......... 63%  129M 1s\n",
      " 33700K .......... .......... .......... .......... .......... 63%  102M 1s\n",
      " 33750K .......... .......... .......... .......... .......... 63% 67,9M 1s\n",
      " 33800K .......... .......... .......... .......... .......... 63%  121M 1s\n",
      " 33850K .......... .......... .......... .......... .......... 63% 40,6M 1s\n",
      " 33900K .......... .......... .......... .......... .......... 63% 21,1M 1s\n",
      " 33950K .......... .......... .......... .......... .......... 63% 12,1M 1s\n",
      " 34000K .......... .......... .......... .......... .......... 63% 20,5M 1s\n",
      " 34050K .......... .......... .......... .......... .......... 64% 23,7M 1s\n",
      " 34100K .......... .......... .......... .......... .......... 64% 38,5M 1s\n",
      " 34150K .......... .......... .......... .......... .......... 64% 64,5M 1s\n",
      " 34200K .......... .......... .......... .......... .......... 64%  102M 1s\n",
      " 34250K .......... .......... .......... .......... .......... 64%  121M 1s\n",
      " 34300K .......... .......... .......... .......... .......... 64%  117M 1s\n",
      " 34350K .......... .......... .......... .......... .......... 64% 30,6M 1s\n",
      " 34400K .......... .......... .......... .......... .......... 64% 8,62M 1s\n",
      " 34450K .......... .......... .......... .......... .......... 64% 9,00M 1s\n",
      " 34500K .......... .......... .......... .......... .......... 64% 7,66M 1s\n",
      " 34550K .......... .......... .......... .......... .......... 65% 20,5M 1s\n",
      " 34600K .......... .......... .......... .......... .......... 65% 42,2M 1s\n",
      " 34650K .......... .......... .......... .......... .......... 65%  123M 1s\n",
      " 34700K .......... .......... .......... .......... .......... 65%  103M 1s\n",
      " 34750K .......... .......... .......... .......... .......... 65%  118M 1s\n",
      " 34800K .......... .......... .......... .......... .......... 65%  121M 1s\n",
      " 34850K .......... .......... .......... .......... .......... 65%  102M 1s\n",
      " 34900K .......... .......... .......... .......... .......... 65%  114M 1s\n",
      " 34950K .......... .......... .......... .......... .......... 65%  104M 1s\n",
      " 35000K .......... .......... .......... .......... .......... 65%  123M 1s\n",
      " 35050K .......... .......... .......... .......... .......... 65%  117M 1s\n",
      " 35100K .......... .......... .......... .......... .......... 66%  103M 1s\n",
      " 35150K .......... .......... .......... .......... .......... 66%  118M 1s\n",
      " 35200K .......... .......... .......... .......... .......... 66%  104M 1s\n",
      " 35250K .......... .......... .......... .......... .......... 66%  119M 1s\n",
      " 35300K .......... .......... .......... .......... .......... 66%  114M 1s\n",
      " 35350K .......... .......... .......... .......... .......... 66%  102M 1s\n",
      " 35400K .......... .......... .......... .......... .......... 66% 88,6M 1s\n",
      " 35450K .......... .......... .......... .......... .......... 66% 78,5M 1s\n",
      " 35500K .......... .......... .......... .......... .......... 66%  113M 1s\n",
      " 35550K .......... .......... .......... .......... .......... 66%  128M 1s\n",
      " 35600K .......... .......... .......... .......... .......... 67%  125M 1s\n",
      " 35650K .......... .......... .......... .......... .......... 67% 76,3M 1s\n",
      " 35700K .......... .......... .......... .......... .......... 67%  128M 1s\n",
      " 35750K .......... .......... .......... .......... .......... 67% 82,3M 1s\n",
      " 35800K .......... .......... .......... .......... .......... 67%  113M 1s\n",
      " 35850K .......... .......... .......... .......... .......... 67%  122M 1s\n",
      " 35900K .......... .......... .......... .......... .......... 67%  149M 1s\n",
      " 35950K .......... .......... .......... .......... .......... 67%  129M 1s\n",
      " 36000K .......... .......... .......... .......... .......... 67%  153M 0s\n",
      " 36050K .......... .......... .......... .......... .......... 67%  150M 0s\n",
      " 36100K .......... .......... .......... .......... .......... 67%  131M 0s\n",
      " 36150K .......... .......... .......... .......... .......... 68%  127M 0s\n",
      " 36200K .......... .......... .......... .......... .......... 68%  102M 0s\n",
      " 36250K .......... .......... .......... .......... .......... 68%  124M 0s\n",
      " 36300K .......... .......... .......... .......... .......... 68%  115M 0s\n",
      " 36350K .......... .......... .......... .......... .......... 68%  102M 0s\n",
      " 36400K .......... .......... .......... .......... .......... 68%  122M 0s\n",
      " 36450K .......... .......... .......... .......... .......... 68%  102M 0s\n",
      " 36500K .......... .......... .......... .......... .......... 68%  117M 0s\n",
      " 36550K .......... .......... .......... .......... .......... 68%  121M 0s\n",
      " 36600K .......... .......... .......... .......... .......... 68%  102M 0s\n",
      " 36650K .......... .......... .......... .......... .......... 68%  122M 0s\n",
      " 36700K .......... .......... .......... .......... .......... 69%  103M 0s\n",
      " 36750K .......... .......... .......... .......... .......... 69%  117M 0s\n",
      " 36800K .......... .......... .......... .......... .......... 69%  121M 0s\n",
      " 36850K .......... .......... .......... .......... .......... 69%  102M 0s\n",
      " 36900K .......... .......... .......... .......... .......... 69%  118M 0s\n",
      " 36950K .......... .......... .......... .......... .......... 69% 37,3M 0s\n",
      " 37000K .......... .......... .......... .......... .......... 69% 8,35M 0s\n",
      " 37050K .......... .......... .......... .......... .......... 69% 36,7M 0s\n",
      " 37100K .......... .......... .......... .......... .......... 69% 26,3M 0s\n",
      " 37150K .......... .......... .......... .......... .......... 69% 15,7M 0s\n",
      " 37200K .......... .......... .......... .......... .......... 70% 78,2M 0s\n",
      " 37250K .......... .......... .......... .......... .......... 70% 57,4M 0s\n",
      " 37300K .......... .......... .......... .......... .......... 70%  117M 0s\n",
      " 37350K .......... .......... .......... .......... .......... 70%  125M 0s\n",
      " 37400K .......... .......... .......... .......... .......... 70%  159M 0s\n",
      " 37450K .......... .......... .......... .......... .......... 70% 6,97M 0s\n",
      " 37500K .......... .......... .......... .......... .......... 70% 35,0M 0s\n",
      " 37550K .......... .......... .......... .......... .......... 70% 7,81M 0s\n",
      " 37600K .......... .......... .......... .......... .......... 70% 9,14M 0s\n",
      " 37650K .......... .......... .......... .......... .......... 70% 43,9M 0s\n",
      " 37700K .......... .......... .......... .......... .......... 70% 36,6M 0s\n",
      " 37750K .......... .......... .......... .......... .......... 71% 80,6M 0s\n",
      " 37800K .......... .......... .......... .......... .......... 71%  125M 0s\n",
      " 37850K .......... .......... .......... .......... .......... 71% 38,3M 0s\n",
      " 37900K .......... .......... .......... .......... .......... 71%  132M 0s\n",
      " 37950K .......... .......... .......... .......... .......... 71%  123M 0s\n",
      " 38000K .......... .......... .......... .......... .......... 71%  142M 0s\n",
      " 38050K .......... .......... .......... .......... .......... 71%  134M 0s\n",
      " 38100K .......... .......... .......... .......... .......... 71% 83,6M 0s\n",
      " 38150K .......... .......... .......... .......... .......... 71%  127M 0s\n",
      " 38200K .......... .......... .......... .......... .......... 71%  128M 0s\n",
      " 38250K .......... .......... .......... .......... .......... 71%  147M 0s\n",
      " 38300K .......... .......... .......... .......... .......... 72%  149M 0s\n",
      " 38350K .......... .......... .......... .......... .......... 72%  129M 0s\n",
      " 38400K .......... .......... .......... .......... .......... 72%  143M 0s\n",
      " 38450K .......... .......... .......... .......... .......... 72%  130M 0s\n",
      " 38500K .......... .......... .......... .......... .......... 72%  167M 0s\n",
      " 38550K .......... .......... .......... .......... .......... 72%  124M 0s\n",
      " 38600K .......... .......... .......... .......... .......... 72%  100M 0s\n",
      " 38650K .......... .......... .......... .......... .......... 72%  118M 0s\n",
      " 38700K .......... .......... .......... .......... .......... 72%  102M 0s\n",
      " 38750K .......... .......... .......... .......... .......... 72%  121M 0s\n",
      " 38800K .......... .......... .......... .......... .......... 73%  121M 0s\n",
      " 38850K .......... .......... .......... .......... .......... 73%  103M 0s\n",
      " 38900K .......... .......... .......... .......... .......... 73% 94,0M 0s\n",
      " 38950K .......... .......... .......... .......... .......... 73%  104M 0s\n",
      " 39000K .......... .......... .......... .......... .......... 73%  122M 0s\n",
      " 39050K .......... .......... .......... .......... .......... 73%  116M 0s\n",
      " 39100K .......... .......... .......... .......... .......... 73%  105M 0s\n",
      " 39150K .......... .......... .......... .......... .......... 73%  117M 0s\n",
      " 39200K .......... .......... .......... .......... .......... 73%  103M 0s\n",
      " 39250K .......... .......... .......... .......... .......... 73%  121M 0s\n",
      " 39300K .......... .......... .......... .......... .......... 73%  118M 0s\n",
      " 39350K .......... .......... .......... .......... .......... 74%  102M 0s\n",
      " 39400K .......... .......... .......... .......... .......... 74%  112M 0s\n",
      " 39450K .......... .......... .......... .......... .......... 74%  110M 0s\n",
      " 39500K .......... .......... .......... .......... .......... 74%  117M 0s\n",
      " 39550K .......... .......... .......... .......... .......... 74%  121M 0s\n",
      " 39600K .......... .......... .......... .......... .......... 74%  103M 0s\n",
      " 39650K .......... .......... .......... .......... .......... 74%  117M 0s\n",
      " 39700K .......... .......... .......... .......... .......... 74%  103M 0s\n",
      " 39750K .......... .......... .......... .......... .......... 74%  121M 0s\n",
      " 39800K .......... .......... .......... .......... .......... 74%  121M 0s\n",
      " 39850K .......... .......... .......... .......... .......... 74%  103M 0s\n",
      " 39900K .......... .......... .......... .......... .......... 75%  117M 0s\n",
      " 39950K .......... .......... .......... .......... .......... 75%  102M 0s\n",
      " 40000K .......... .......... .......... .......... .......... 75%  122M 0s\n",
      " 40050K .......... .......... .......... .......... .......... 75% 8,92M 0s\n",
      " 40100K .......... .......... .......... .......... .......... 75% 31,1M 0s\n",
      " 40150K .......... .......... .......... .......... .......... 75% 35,3M 0s\n",
      " 40200K .......... .......... .......... .......... .......... 75% 13,9M 0s\n",
      " 40250K .......... .......... .......... .......... .......... 75% 24,8M 0s\n",
      " 40300K .......... .......... .......... .......... .......... 75% 65,6M 0s\n",
      " 40350K .......... .......... .......... .......... .......... 75% 60,2M 0s\n",
      " 40400K .......... .......... .......... .......... .......... 76%  238M 0s\n",
      " 40450K .......... .......... .......... .......... .......... 76%  215M 0s\n",
      " 40500K .......... .......... .......... .......... .......... 76% 26,9M 0s\n",
      " 40550K .......... .......... .......... .......... .......... 76% 8,46M 0s\n",
      " 40600K .......... .......... .......... .......... .......... 76% 9,40M 0s\n",
      " 40650K .......... .......... .......... .......... .......... 76% 8,80M 0s\n",
      " 40700K .......... .......... .......... .......... .......... 76% 18,2M 0s\n",
      " 40750K .......... .......... .......... .......... .......... 76% 68,7M 0s\n",
      " 40800K .......... .......... .......... .......... .......... 76% 91,0M 0s\n",
      " 40850K .......... .......... .......... .......... .......... 76%  107M 0s\n",
      " 40900K .......... .......... .......... .......... .......... 76%  112M 0s\n",
      " 40950K .......... .......... .......... .......... .......... 77% 88,3M 0s\n",
      " 41000K .......... .......... .......... .......... .......... 77%  105M 0s\n",
      " 41050K .......... .......... .......... .......... .......... 77%  149M 0s\n",
      " 41100K .......... .......... .......... .......... .......... 77%  114M 0s\n",
      " 41150K .......... .......... .......... .......... .......... 77%  110M 0s\n",
      " 41200K .......... .......... .......... .......... .......... 77%  105M 0s\n",
      " 41250K .......... .......... .......... .......... .......... 77%  127M 0s\n",
      " 41300K .......... .......... .......... .......... .......... 77%  120M 0s\n",
      " 41350K .......... .......... .......... .......... .......... 77%  102M 0s\n",
      " 41400K .......... .......... .......... .......... .......... 77%  115M 0s\n",
      " 41450K .......... .......... .......... .......... .......... 77%  108M 0s\n",
      " 41500K .......... .......... .......... .......... .......... 78%  118M 0s\n",
      " 41550K .......... .......... .......... .......... .......... 78%  122M 0s\n",
      " 41600K .......... .......... .......... .......... .......... 78%  102M 0s\n",
      " 41650K .......... .......... .......... .......... .......... 78%  117M 0s\n",
      " 41700K .......... .......... .......... .......... .......... 78%  102M 0s\n",
      " 41750K .......... .......... .......... .......... .......... 78%  121M 0s\n",
      " 41800K .......... .......... .......... .......... .......... 78%  106M 0s\n",
      " 41850K .......... .......... .......... .......... .......... 78%  113M 0s\n",
      " 41900K .......... .......... .......... .......... .......... 78%  119M 0s\n",
      " 41950K .......... .......... .......... .......... .......... 78%  103M 0s\n",
      " 42000K .......... .......... .......... .......... .......... 79%  113M 0s\n",
      " 42050K .......... .......... .......... .......... .......... 79% 57,3M 0s\n",
      " 42100K .......... .......... .......... .......... .......... 79%  106M 0s\n",
      " 42150K .......... .......... .......... .......... .......... 79%  147M 0s\n",
      " 42200K .......... .......... .......... .......... .......... 79%  124M 0s\n",
      " 42250K .......... .......... .......... .......... .......... 79%  411M 0s\n",
      " 42300K .......... .......... .......... .......... .......... 79%  120M 0s\n",
      " 42350K .......... .......... .......... .......... .......... 79%  103M 0s\n",
      " 42400K .......... .......... .......... .......... .......... 79%  117M 0s\n",
      " 42450K .......... .......... .......... .......... .......... 79%  103M 0s\n",
      " 42500K .......... .......... .......... .......... .......... 79%  120M 0s\n",
      " 42550K .......... .......... .......... .......... .......... 80%  122M 0s\n",
      " 42600K .......... .......... .......... .......... .......... 80%  102M 0s\n",
      " 42650K .......... .......... .......... .......... .......... 80%  117M 0s\n",
      " 42700K .......... .......... .......... .......... .......... 80%  102M 0s\n",
      " 42750K .......... .......... .......... .......... .......... 80%  121M 0s\n",
      " 42800K .......... .......... .......... .......... .......... 80%  117M 0s\n",
      " 42850K .......... .......... .......... .......... .......... 80%  102M 0s\n",
      " 42900K .......... .......... .......... .......... .......... 80%  121M 0s\n",
      " 42950K .......... .......... .......... .......... .......... 80%  102M 0s\n",
      " 43000K .......... .......... .......... .......... .......... 80%  121M 0s\n",
      " 43050K .......... .......... .......... .......... .......... 81%  117M 0s\n",
      " 43100K .......... .......... .......... .......... .......... 81% 17,7M 0s\n",
      " 43150K .......... .......... .......... .......... .......... 81% 8,52M 0s\n",
      " 43200K .......... .......... .......... .......... .......... 81%  105M 0s\n",
      " 43250K .......... .......... .......... .......... .......... 81% 29,3M 0s\n",
      " 43300K .......... .......... .......... .......... .......... 81% 20,6M 0s\n",
      " 43350K .......... .......... .......... .......... .......... 81% 59,9M 0s\n",
      " 43400K .......... .......... .......... .......... .......... 81% 61,5M 0s\n",
      " 43450K .......... .......... .......... .......... .......... 81%  107M 0s\n",
      " 43500K .......... .......... .......... .......... .......... 81% 66,7M 0s\n",
      " 43550K .......... .......... .......... .......... .......... 81% 19,6M 0s\n",
      " 43600K .......... .......... .......... .......... .......... 82% 8,41M 0s\n",
      " 43650K .......... .......... .......... .......... .......... 82% 12,6M 0s\n",
      " 43700K .......... .......... .......... .......... .......... 82% 9,20M 0s\n",
      " 43750K .......... .......... .......... .......... .......... 82% 17,9M 0s\n",
      " 43800K .......... .......... .......... .......... .......... 82% 28,4M 0s\n",
      " 43850K .......... .......... .......... .......... .......... 82% 52,2M 0s\n",
      " 43900K .......... .......... .......... .......... .......... 82% 88,9M 0s\n",
      " 43950K .......... .......... .......... .......... .......... 82% 78,2M 0s\n",
      " 44000K .......... .......... .......... .......... .......... 82%  101M 0s\n",
      " 44050K .......... .......... .......... .......... .......... 82% 93,7M 0s\n",
      " 44100K .......... .......... .......... .......... .......... 82% 84,7M 0s\n",
      " 44150K .......... .......... .......... .......... .......... 83%  120M 0s\n",
      " 44200K .......... .......... .......... .......... .......... 83%  105M 0s\n",
      " 44250K .......... .......... .......... .......... .......... 83%  121M 0s\n",
      " 44300K .......... .......... .......... .......... .......... 83%  124M 0s\n",
      " 44350K .......... .......... .......... .......... .......... 83%  187M 0s\n",
      " 44400K .......... .......... .......... .......... .......... 83%  181M 0s\n",
      " 44450K .......... .......... .......... .......... .......... 83%  185M 0s\n",
      " 44500K .......... .......... .......... .......... .......... 83%  125M 0s\n",
      " 44550K .......... .......... .......... .......... .......... 83%  136M 0s\n",
      " 44600K .......... .......... .......... .......... .......... 83%  160M 0s\n",
      " 44650K .......... .......... .......... .......... .......... 84%  125M 0s\n",
      " 44700K .......... .......... .......... .......... .......... 84%  103M 0s\n",
      " 44750K .......... .......... .......... .......... .......... 84%  123M 0s\n",
      " 44800K .......... .......... .......... .......... .......... 84%  119M 0s\n",
      " 44850K .......... .......... .......... .......... .......... 84%  101M 0s\n",
      " 44900K .......... .......... .......... .......... .......... 84%  119M 0s\n",
      " 44950K .......... .......... .......... .......... .......... 84%  102M 0s\n",
      " 45000K .......... .......... .......... .......... .......... 84%  122M 0s\n",
      " 45050K .......... .......... .......... .......... .......... 84%  118M 0s\n",
      " 45100K .......... .......... .......... .......... .......... 84%  104M 0s\n",
      " 45150K .......... .......... .......... .......... .......... 84%  124M 0s\n",
      " 45200K .......... .......... .......... .......... .......... 85%  102M 0s\n",
      " 45250K .......... .......... .......... .......... .......... 85%  117M 0s\n",
      " 45300K .......... .......... .......... .......... .......... 85%  121M 0s\n",
      " 45350K .......... .......... .......... .......... .......... 85%  102M 0s\n",
      " 45400K .......... .......... .......... .......... .......... 85%  117M 0s\n",
      " 45450K .......... .......... .......... .......... .......... 85%  102M 0s\n",
      " 45500K .......... .......... .......... .......... .......... 85%  121M 0s\n",
      " 45550K .......... .......... .......... .......... .......... 85%  117M 0s\n",
      " 45600K .......... .......... .......... .......... .......... 85% 97,0M 0s\n",
      " 45650K .......... .......... .......... .......... .......... 85%  131M 0s\n",
      " 45700K .......... .......... .......... .......... .......... 85%  102M 0s\n",
      " 45750K .......... .......... .......... .......... .......... 86%  122M 0s\n",
      " 45800K .......... .......... .......... .......... .......... 86%  117M 0s\n",
      " 45850K .......... .......... .......... .......... .......... 86%  103M 0s\n",
      " 45900K .......... .......... .......... .......... .......... 86%  120M 0s\n",
      " 45950K .......... .......... .......... .......... .......... 86% 93,6M 0s\n",
      " 46000K .......... .......... .......... .......... .......... 86%  126M 0s\n",
      " 46050K .......... .......... .......... .......... .......... 86%  125M 0s\n",
      " 46100K .......... .......... .......... .......... .......... 86%  102M 0s\n",
      " 46150K .......... .......... .......... .......... .......... 86% 27,1M 0s\n",
      " 46200K .......... .......... .......... .......... .......... 86% 9,10M 0s\n",
      " 46250K .......... .......... .......... .......... .......... 87% 25,3M 0s\n",
      " 46300K .......... .......... .......... .......... .......... 87%  118M 0s\n",
      " 46350K .......... .......... .......... .......... .......... 87% 12,9M 0s\n",
      " 46400K .......... .......... .......... .......... .......... 87% 45,1M 0s\n",
      " 46450K .......... .......... .......... .......... .......... 87% 88,4M 0s\n",
      " 46500K .......... .......... .......... .......... .......... 87%  130M 0s\n",
      " 46550K .......... .......... .......... .......... .......... 87%  238M 0s\n",
      " 46600K .......... .......... .......... .......... .......... 87% 15,3M 0s\n",
      " 46650K .......... .......... .......... .......... .......... 87% 20,0M 0s\n",
      " 46700K .......... .......... .......... .......... .......... 87% 9,52M 0s\n",
      " 46750K .......... .......... .......... .......... .......... 87% 8,94M 0s\n",
      " 46800K .......... .......... .......... .......... .......... 88% 15,7M 0s\n",
      " 46850K .......... .......... .......... .......... .......... 88% 21,9M 0s\n",
      " 46900K .......... .......... .......... .......... .......... 88% 35,3M 0s\n",
      " 46950K .......... .......... .......... .......... .......... 88% 62,3M 0s\n",
      " 47000K .......... .......... .......... .......... .......... 88% 74,9M 0s\n",
      " 47050K .......... .......... .......... .......... .......... 88% 90,1M 0s\n",
      " 47100K .......... .......... .......... .......... .......... 88% 81,7M 0s\n",
      " 47150K .......... .......... .......... .......... .......... 88%  103M 0s\n",
      " 47200K .......... .......... .......... .......... .......... 88% 94,2M 0s\n",
      " 47250K .......... .......... .......... .......... .......... 88%  136M 0s\n",
      " 47300K .......... .......... .......... .......... .......... 88%  127M 0s\n",
      " 47350K .......... .......... .......... .......... .......... 89% 74,7M 0s\n",
      " 47400K .......... .......... .......... .......... .......... 89% 90,2M 0s\n",
      " 47450K .......... .......... .......... .......... .......... 89%  106M 0s\n",
      " 47500K .......... .......... .......... .......... .......... 89% 93,9M 0s\n",
      " 47550K .......... .......... .......... .......... .......... 89%  117M 0s\n",
      " 47600K .......... .......... .......... .......... .......... 89%  125M 0s\n",
      " 47650K .......... .......... .......... .......... .......... 89% 76,9M 0s\n",
      " 47700K .......... .......... .......... .......... .......... 89%  187M 0s\n",
      " 47750K .......... .......... .......... .......... .......... 89%  242M 0s\n",
      " 47800K .......... .......... .......... .......... .......... 89%  250M 0s\n",
      " 47850K .......... .......... .......... .......... .......... 90%  132M 0s\n",
      " 47900K .......... .......... .......... .......... .......... 90%  121M 0s\n",
      " 47950K .......... .......... .......... .......... .......... 90% 70,9M 0s\n",
      " 48000K .......... .......... .......... .......... .......... 90% 88,4M 0s\n",
      " 48050K .......... .......... .......... .......... .......... 90%  106M 0s\n",
      " 48100K .......... .......... .......... .......... .......... 90% 96,9M 0s\n",
      " 48150K .......... .......... .......... .......... .......... 90%  122M 0s\n",
      " 48200K .......... .......... .......... .......... .......... 90%  174M 0s\n",
      " 48250K .......... .......... .......... .......... .......... 90%  273M 0s\n",
      " 48300K .......... .......... .......... .......... .......... 90%  118M 0s\n",
      " 48350K .......... .......... .......... .......... .......... 90%  102M 0s\n",
      " 48400K .......... .......... .......... .......... .......... 91%  119M 0s\n",
      " 48450K .......... .......... .......... .......... .......... 91%  103M 0s\n",
      " 48500K .......... .......... .......... .......... .......... 91%  121M 0s\n",
      " 48550K .......... .......... .......... .......... .......... 91%  118M 0s\n",
      " 48600K .......... .......... .......... .......... .......... 91%  103M 0s\n",
      " 48650K .......... .......... .......... .......... .......... 91%  123M 0s\n",
      " 48700K .......... .......... .......... .......... .......... 91%  101M 0s\n",
      " 48750K .......... .......... .......... .......... .......... 91%  118M 0s\n",
      " 48800K .......... .......... .......... .......... .......... 91%  123M 0s\n",
      " 48850K .......... .......... .......... .......... .......... 91%  102M 0s\n",
      " 48900K .......... .......... .......... .......... .......... 91%  121M 0s\n",
      " 48950K .......... .......... .......... .......... .......... 92%  102M 0s\n",
      " 49000K .......... .......... .......... .......... .......... 92%  117M 0s\n",
      " 49050K .......... .......... .......... .......... .......... 92%  121M 0s\n",
      " 49100K .......... .......... .......... .......... .......... 92%  102M 0s\n",
      " 49150K .......... .......... .......... .......... .......... 92%  117M 0s\n",
      " 49200K .......... .......... .......... .......... .......... 92% 11,9M 0s\n",
      " 49250K .......... .......... .......... .......... .......... 92% 18,4M 0s\n",
      " 49300K .......... .......... .......... .......... .......... 92% 41,7M 0s\n",
      " 49350K .......... .......... .......... .......... .......... 92% 33,0M 0s\n",
      " 49400K .......... .......... .......... .......... .......... 92% 33,1M 0s\n",
      " 49450K .......... .......... .......... .......... .......... 93% 17,7M 0s\n",
      " 49500K .......... .......... .......... .......... .......... 93%  121M 0s\n",
      " 49550K .......... .......... .......... .......... .......... 93% 61,5M 0s\n",
      " 49600K .......... .......... .......... .......... .......... 93% 54,3M 0s\n",
      " 49650K .......... .......... .......... .......... .......... 93% 32,9M 0s\n",
      " 49700K .......... .......... .......... .......... .......... 93% 19,2M 0s\n",
      " 49750K .......... .......... .......... .......... .......... 93% 8,37M 0s\n",
      " 49800K .......... .......... .......... .......... .......... 93% 17,6M 0s\n",
      " 49850K .......... .......... .......... .......... .......... 93% 8,90M 0s\n",
      " 49900K .......... .......... .......... .......... .......... 93% 30,0M 0s\n",
      " 49950K .......... .......... .......... .......... .......... 93% 31,1M 0s\n",
      " 50000K .......... .......... .......... .......... .......... 94% 29,4M 0s\n",
      " 50050K .......... .......... .......... .......... .......... 94% 51,3M 0s\n",
      " 50100K .......... .......... .......... .......... .......... 94%  102M 0s\n",
      " 50150K .......... .......... .......... .......... .......... 94%  118M 0s\n",
      " 50200K .......... .......... .......... .......... .......... 94%  102M 0s\n",
      " 50250K .......... .......... .......... .......... .......... 94%  122M 0s\n",
      " 50300K .......... .......... .......... .......... .......... 94%  118M 0s\n",
      " 50350K .......... .......... .......... .......... .......... 94%  102M 0s\n",
      " 50400K .......... .......... .......... .......... .......... 94%  122M 0s\n",
      " 50450K .......... .......... .......... .......... .......... 94% 79,4M 0s\n",
      " 50500K .......... .......... .......... .......... .......... 95%  122M 0s\n",
      " 50550K .......... .......... .......... .......... .......... 95%  117M 0s\n",
      " 50600K .......... .......... .......... .......... .......... 95%  103M 0s\n",
      " 50650K .......... .......... .......... .......... .......... 95%  121M 0s\n",
      " 50700K .......... .......... .......... .......... .......... 95%  102M 0s\n",
      " 50750K .......... .......... .......... .......... .......... 95%  118M 0s\n",
      " 50800K .......... .......... .......... .......... .......... 95%  121M 0s\n",
      " 50850K .......... .......... .......... .......... .......... 95%  103M 0s\n",
      " 50900K .......... .......... .......... .......... .......... 95%  113M 0s\n",
      " 50950K .......... .......... .......... .......... .......... 95%  104M 0s\n",
      " 51000K .......... .......... .......... .......... .......... 95%  123M 0s\n",
      " 51050K .......... .......... .......... .......... .......... 96%  121M 0s\n",
      " 51100K .......... .......... .......... .......... .......... 96% 93,1M 0s\n",
      " 51150K .......... .......... .......... .......... .......... 96%  127M 0s\n",
      " 51200K .......... .......... .......... .......... .......... 96%  105M 0s\n",
      " 51250K .......... .......... .......... .......... .......... 96%  121M 0s\n",
      " 51300K .......... .......... .......... .......... .......... 96%  117M 0s\n",
      " 51350K .......... .......... .......... .......... .......... 96%  102M 0s\n",
      " 51400K .......... .......... .......... .......... .......... 96%  121M 0s\n",
      " 51450K .......... .......... .......... .......... .......... 96%  102M 0s\n",
      " 51500K .......... .......... .......... .......... .......... 96%  117M 0s\n",
      " 51550K .......... .......... .......... .......... .......... 96%  122M 0s\n",
      " 51600K .......... .......... .......... .......... .......... 97%  103M 0s\n",
      " 51650K .......... .......... .......... .......... .......... 97%  121M 0s\n",
      " 51700K .......... .......... .......... .......... .......... 97%  102M 0s\n",
      " 51750K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 51800K .......... .......... .......... .......... .......... 97%  121M 0s\n",
      " 51850K .......... .......... .......... .......... .......... 97%  102M 0s\n",
      " 51900K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 51950K .......... .......... .......... .......... .......... 97%  103M 0s\n",
      " 52000K .......... .......... .......... .......... .......... 97%  120M 0s\n",
      " 52050K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 52100K .......... .......... .......... .......... .......... 98%  103M 0s\n",
      " 52150K .......... .......... .......... .......... .......... 98%  121M 0s\n",
      " 52200K .......... .......... .......... .......... .......... 98%  102M 0s\n",
      " 52250K .......... .......... .......... .......... .......... 98% 28,6M 0s\n",
      " 52300K .......... .......... .......... .......... .......... 98% 10,1M 0s\n",
      " 52350K .......... .......... .......... .......... .......... 98%  103M 0s\n",
      " 52400K .......... .......... .......... .......... .......... 98% 56,4M 0s\n",
      " 52450K .......... .......... .......... .......... .......... 98% 51,8M 0s\n",
      " 52500K .......... .......... .......... .......... .......... 98% 13,9M 0s\n",
      " 52550K .......... .......... .......... .......... .......... 98% 48,6M 0s\n",
      " 52600K .......... .......... .......... .......... .......... 98%  108M 0s\n",
      " 52650K .......... .......... .......... .......... .......... 99% 57,1M 0s\n",
      " 52700K .......... .......... .......... .......... .......... 99% 52,9M 0s\n",
      " 52750K .......... .......... .......... .......... .......... 99% 18,2M 0s\n",
      " 52800K .......... .......... .......... .......... .......... 99% 11,9M 0s\n",
      " 52850K .......... .......... .......... .......... .......... 99% 11,4M 0s\n",
      " 52900K .......... .......... .......... .......... .......... 99% 10,7M 0s\n",
      " 52950K .......... .......... .......... .......... .......... 99% 22,3M 0s\n",
      " 53000K .......... .......... .......... .......... .......... 99% 19,7M 0s\n",
      " 53050K .......... .......... .......... .......... .......... 99% 21,6M 0s\n",
      " 53100K .......... .......... .......... .......... .......... 99%  103M 0s\n",
      " 53150K .......... .......... .......... .......... .......... 99% 78,1M 0s\n",
      " 53200K ........                                              100%  237M=1,4s\n",
      "\n",
      "2023-08-18 17:02:56 (37,4 MB/s) - ‘data/reference/genome.fa.gz’ guardado [54485176/54485176]\n",
      "\n",
      "[Fri Aug 18 17:02:56 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T170254.064254.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:02:58 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR696683_chr5_1.fastq.gz, data/raw/ERR696683_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR696683_chr5_1_fastp.fastq.gz, data/processed/ERR696683_chr5_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR696683_chr5_2_fastp.fastq.gz, data/processed/ERR696683_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1745204\n",
      "total bases: 131934728\n",
      "Q20 bases: 131000292(99.2917%)\n",
      "Q30 bases: 127389751(96.5551%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1745204\n",
      "total bases: 131603364\n",
      "Q20 bases: 130011063(98.7901%)\n",
      "Q30 bases: 124792206(94.8245%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1090289\n",
      "total bases: 82806787\n",
      "Q20 bases: 82482544(99.6084%)\n",
      "Q30 bases: 81050211(97.8787%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1090289\n",
      "total bases: 82775596\n",
      "Q20 bases: 82215241(99.323%)\n",
      "Q30 bases: 80124032(96.7967%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2180578\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1309830\n",
      "reads with adapter trimmed: 1530\n",
      "bases trimmed due to adapters: 39157\n",
      "\n",
      "Duplication rate: 5.88368%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR696683_chr5_1.fastq.gz -I data/raw/ERR696683_chr5_2.fastq.gz -o data/processed/ERR696683_chr5_1_fastp.fastq.gz -O data/processed/ERR696683_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 7 seconds\n",
      "[Fri Aug 18 17:03:06 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:03:06 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR696683_chr5_1_fastp.fastq.gz, data/processed/ERR696683_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR696683_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR696683_chr5_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR696683_chr5_2_fastp.fastq.gz, data/processed/ERR696683_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 17:03:17 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:03:17 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR696683_chr5_1.fastq.gz, data/raw/ERR696683_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR696683_chr5_1_fastqc.html, results/fastqc_result/ERR696683_chr5_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR696683_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 17:03:31 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T170257.114925.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:03:33 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753368_chr5_1.fastq.gz, data/raw/ERR753368_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753368_chr5_1_fastp.fastq.gz, data/processed/ERR753368_chr5_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753368_chr5_1_fastp.fastq.gz, data/processed/ERR753368_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1754317\n",
      "total bases: 133207348\n",
      "Q20 bases: 132685869(99.6085%)\n",
      "Q30 bases: 128783661(96.6791%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1754317\n",
      "total bases: 133082490\n",
      "Q20 bases: 131930412(99.1343%)\n",
      "Q30 bases: 125959500(94.6477%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1515191\n",
      "total bases: 115148663\n",
      "Q20 bases: 114951118(99.8284%)\n",
      "Q30 bases: 113024140(98.155%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1515191\n",
      "total bases: 115139136\n",
      "Q20 bases: 114585523(99.5192%)\n",
      "Q30 bases: 111209688(96.5872%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3030382\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 478252\n",
      "reads with adapter trimmed: 2004\n",
      "bases trimmed due to adapters: 46944\n",
      "\n",
      "Duplication rate: 2.06205%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753368_chr5_1.fastq.gz -I data/raw/ERR753368_chr5_2.fastq.gz -o data/processed/ERR753368_chr5_1_fastp.fastq.gz -O data/processed/ERR753368_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 8 seconds\n",
      "[Fri Aug 18 17:03:41 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:03:41 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753368_chr5_1.fastq.gz, data/raw/ERR753368_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753368_chr5_1_fastqc.html, results/fastqc_result/ERR753368_chr5_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753368_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753368_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 17:03:56 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:03:56 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753368_chr5_1_fastp.fastq.gz, data/processed/ERR753368_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753368_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753368_chr5_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753368_chr5_1_fastp.fastq.gz, data/processed/ERR753368_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 17:04:09 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T170332.177002.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:04:10 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753369_chr5_1.fastq.gz, data/raw/ERR753369_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753369_chr5_1_fastp.fastq.gz, data/processed/ERR753369_chr5_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753369_chr5_1_fastp.fastq.gz, data/processed/ERR753369_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1874038\n",
      "total bases: 142293627\n",
      "Q20 bases: 141712295(99.5915%)\n",
      "Q30 bases: 137456286(96.6005%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1874038\n",
      "total bases: 142166679\n",
      "Q20 bases: 140932525(99.1319%)\n",
      "Q30 bases: 134533978(94.6312%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1618376\n",
      "total bases: 122990005\n",
      "Q20 bases: 122765144(99.8172%)\n",
      "Q30 bases: 120638796(98.0883%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1618376\n",
      "total bases: 122980612\n",
      "Q20 bases: 122387247(99.5175%)\n",
      "Q30 bases: 118766072(96.573%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3236752\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 511324\n",
      "reads with adapter trimmed: 2190\n",
      "bases trimmed due to adapters: 51994\n",
      "\n",
      "Duplication rate: 2.41004%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753369_chr5_1.fastq.gz -I data/raw/ERR753369_chr5_2.fastq.gz -o data/processed/ERR753369_chr5_1_fastp.fastq.gz -O data/processed/ERR753369_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Fri Aug 18 17:04:17 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:04:17 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753369_chr5_1_fastp.fastq.gz, data/processed/ERR753369_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753369_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753369_chr5_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753369_chr5_1_fastp.fastq.gz, data/processed/ERR753369_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 17:04:30 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:04:30 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753369_chr5_1.fastq.gz, data/raw/ERR753369_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753369_chr5_1_fastqc.html, results/fastqc_result/ERR753369_chr5_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753369_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753369_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 17:04:45 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T170409.303148.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:04:46 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753370_chr5_1.fastq.gz, data/raw/ERR753370_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753370_chr5_1_fastp.fastq.gz, data/processed/ERR753370_chr5_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753370_chr5_2_fastp.fastq.gz, data/processed/ERR753370_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2518071\n",
      "total bases: 253490833\n",
      "Q20 bases: 251767367(99.3201%)\n",
      "Q30 bases: 244487028(96.4481%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2518071\n",
      "total bases: 253433925\n",
      "Q20 bases: 251051683(99.06%)\n",
      "Q30 bases: 241620433(95.3386%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 2517536\n",
      "total bases: 252452388\n",
      "Q20 bases: 250828184(99.3566%)\n",
      "Q30 bases: 243890545(96.6085%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 2517536\n",
      "total bases: 251656989\n",
      "Q20 bases: 249490937(99.1393%)\n",
      "Q30 bases: 240647742(95.6253%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 5035072\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1070\n",
      "reads with adapter trimmed: 14130\n",
      "bases trimmed due to adapters: 108801\n",
      "\n",
      "Duplication rate: 4.4485%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 158\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753370_chr5_1.fastq.gz -I data/raw/ERR753370_chr5_2.fastq.gz -o data/processed/ERR753370_chr5_1_fastp.fastq.gz -O data/processed/ERR753370_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 13 seconds\n",
      "[Fri Aug 18 17:05:00 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:05:00 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753370_chr5_1_fastp.fastq.gz, data/processed/ERR753370_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753370_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753370_chr5_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753370_chr5_2_fastp.fastq.gz, data/processed/ERR753370_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 17:05:23 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:05:23 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753370_chr5_1.fastq.gz, data/raw/ERR753370_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753370_chr5_1_fastqc.html, results/fastqc_result/ERR753370_chr5_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753370_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753370_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 17:05:46 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T170445.615692.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:05:48 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753371_chr5_1.fastq.gz, data/raw/ERR753371_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753371_chr5_1_fastqc.html, results/fastqc_result/ERR753371_chr5_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753371_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753371_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 17:06:00 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:06:00 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753371_chr5_1.fastq.gz, data/raw/ERR753371_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753371_chr5_1_fastp.fastq.gz, data/processed/ERR753371_chr5_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753371_chr5_1_fastp.fastq.gz, data/processed/ERR753371_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1647835\n",
      "total bases: 124169361\n",
      "Q20 bases: 123052427(99.1005%)\n",
      "Q30 bases: 118563904(95.4856%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1647835\n",
      "total bases: 123864691\n",
      "Q20 bases: 122402253(98.8193%)\n",
      "Q30 bases: 118123304(95.3648%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 856072\n",
      "total bases: 64982245\n",
      "Q20 bases: 64667623(99.5158%)\n",
      "Q30 bases: 63059172(97.0406%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 856072\n",
      "total bases: 64947026\n",
      "Q20 bases: 64500274(99.3121%)\n",
      "Q30 bases: 63094570(97.1477%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1712144\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1583526\n",
      "reads with adapter trimmed: 1440\n",
      "bases trimmed due to adapters: 32385\n",
      "\n",
      "Duplication rate: 22.2902%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 117\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753371_chr5_1.fastq.gz -I data/raw/ERR753371_chr5_2.fastq.gz -o data/processed/ERR753371_chr5_1_fastp.fastq.gz -O data/processed/ERR753371_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Fri Aug 18 17:06:06 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:06:06 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753371_chr5_1_fastp.fastq.gz, data/processed/ERR753371_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753371_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753371_chr5_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753371_chr5_1_fastp.fastq.gz, data/processed/ERR753371_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 17:06:15 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T170547.076840.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:06:16 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753372_chr5_1.fastq.gz, data/raw/ERR753372_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753372_chr5_1_fastqc.html, results/fastqc_result/ERR753372_chr5_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753372_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753372_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 17:06:30 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:06:30 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753372_chr5_1.fastq.gz, data/raw/ERR753372_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753372_chr5_1_fastp.fastq.gz, data/processed/ERR753372_chr5_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753372_chr5_2_fastp.fastq.gz, data/processed/ERR753372_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1849920\n",
      "total bases: 140463336\n",
      "Q20 bases: 139906933(99.6039%)\n",
      "Q30 bases: 135706572(96.6135%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1849920\n",
      "total bases: 140332630\n",
      "Q20 bases: 139116664(99.1335%)\n",
      "Q30 bases: 132851025(94.6687%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1592983\n",
      "total bases: 121060371\n",
      "Q20 bases: 120850328(99.8265%)\n",
      "Q30 bases: 118781603(98.1177%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1592983\n",
      "total bases: 121050636\n",
      "Q20 bases: 120473426(99.5232%)\n",
      "Q30 bases: 116971001(96.6298%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3185966\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 513874\n",
      "reads with adapter trimmed: 2008\n",
      "bases trimmed due to adapters: 48830\n",
      "\n",
      "Duplication rate: 2.0226%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753372_chr5_1.fastq.gz -I data/raw/ERR753372_chr5_2.fastq.gz -o data/processed/ERR753372_chr5_1_fastp.fastq.gz -O data/processed/ERR753372_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Fri Aug 18 17:06:37 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:06:37 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753372_chr5_1_fastp.fastq.gz, data/processed/ERR753372_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753372_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753372_chr5_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753372_chr5_2_fastp.fastq.gz, data/processed/ERR753372_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 17:06:50 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T170615.188140.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:06:51 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753373_chr5_1.fastq.gz, data/raw/ERR753373_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753373_chr5_1_fastqc.html, results/fastqc_result/ERR753373_chr5_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753373_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753373_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 17:07:10 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:07:10 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753373_chr5_1.fastq.gz, data/raw/ERR753373_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753373_chr5_1_fastp.fastq.gz, data/processed/ERR753373_chr5_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753373_chr5_1_fastp.fastq.gz, data/processed/ERR753373_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1769351\n",
      "total bases: 169422654\n",
      "Q20 bases: 168087237(99.2118%)\n",
      "Q30 bases: 156744268(92.5167%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1769351\n",
      "total bases: 165883501\n",
      "Q20 bases: 164108844(98.9302%)\n",
      "Q30 bases: 153648766(92.6245%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1727865\n",
      "total bases: 162896728\n",
      "Q20 bases: 161958252(99.4239%)\n",
      "Q30 bases: 152772191(93.7847%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1727865\n",
      "total bases: 157462847\n",
      "Q20 bases: 156729547(99.5343%)\n",
      "Q30 bases: 148765966(94.4769%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3455730\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 82972\n",
      "reads with adapter trimmed: 1628\n",
      "bases trimmed due to adapters: 56407\n",
      "\n",
      "Duplication rate: 1.9249%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 153\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753373_chr5_1.fastq.gz -I data/raw/ERR753373_chr5_2.fastq.gz -o data/processed/ERR753373_chr5_1_fastp.fastq.gz -O data/processed/ERR753373_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 8 seconds\n",
      "[Fri Aug 18 17:07:18 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:07:18 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753373_chr5_1_fastp.fastq.gz, data/processed/ERR753373_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753373_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753373_chr5_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753373_chr5_1_fastp.fastq.gz, data/processed/ERR753373_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 17:07:32 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T170650.282774.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:07:34 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753374_chr5_1.fastq.gz, data/raw/ERR753374_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753374_chr5_1_fastp.fastq.gz, data/processed/ERR753374_chr5_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753374_chr5_1_fastp.fastq.gz, data/processed/ERR753374_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1042969\n",
      "total bases: 99775475\n",
      "Q20 bases: 98851672(99.0741%)\n",
      "Q30 bases: 90441720(90.6452%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1042969\n",
      "total bases: 97644504\n",
      "Q20 bases: 96637952(98.9692%)\n",
      "Q30 bases: 89006849(91.154%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1011156\n",
      "total bases: 94571637\n",
      "Q20 bases: 93909953(99.3003%)\n",
      "Q30 bases: 87206490(92.2121%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1011156\n",
      "total bases: 92115109\n",
      "Q20 bases: 91596849(99.4374%)\n",
      "Q30 bases: 85744137(93.0837%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2022312\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 63626\n",
      "reads with adapter trimmed: 920\n",
      "bases trimmed due to adapters: 32756\n",
      "\n",
      "Duplication rate: 5.80193%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 152\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753374_chr5_1.fastq.gz -I data/raw/ERR753374_chr5_2.fastq.gz -o data/processed/ERR753374_chr5_1_fastp.fastq.gz -O data/processed/ERR753374_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Fri Aug 18 17:07:39 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:07:39 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753374_chr5_1_fastp.fastq.gz, data/processed/ERR753374_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753374_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753374_chr5_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753374_chr5_1_fastp.fastq.gz, data/processed/ERR753374_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 17:07:49 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:07:49 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753374_chr5_1.fastq.gz, data/raw/ERR753374_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753374_chr5_1_fastqc.html, results/fastqc_result/ERR753374_chr5_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753374_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753374_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 17:08:00 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T170733.068462.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:08:01 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753375_chr5_1.fastq.gz, data/raw/ERR753375_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753375_chr5_1_fastqc.html, results/fastqc_result/ERR753375_chr5_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753375_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753375_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 17:08:13 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:08:13 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753375_chr5_1.fastq.gz, data/raw/ERR753375_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753375_chr5_1_fastp.fastq.gz, data/processed/ERR753375_chr5_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753375_chr5_1_fastp.fastq.gz, data/processed/ERR753375_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1084744\n",
      "total bases: 81943993\n",
      "Q20 bases: 81438535(99.3832%)\n",
      "Q30 bases: 79595776(97.1344%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1084744\n",
      "total bases: 81928209\n",
      "Q20 bases: 80877145(98.7171%)\n",
      "Q30 bases: 78915671(96.323%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 711112\n",
      "total bases: 53994712\n",
      "Q20 bases: 53820596(99.6775%)\n",
      "Q30 bases: 53016677(98.1886%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 711112\n",
      "total bases: 54003767\n",
      "Q20 bases: 53488355(99.0456%)\n",
      "Q30 bases: 52659532(97.5108%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1422224\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 747264\n",
      "reads with adapter trimmed: 856\n",
      "bases trimmed due to adapters: 20807\n",
      "\n",
      "Duplication rate: 1.69493%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753375_chr5_1.fastq.gz -I data/raw/ERR753375_chr5_2.fastq.gz -o data/processed/ERR753375_chr5_1_fastp.fastq.gz -O data/processed/ERR753375_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 3 seconds\n",
      "[Fri Aug 18 17:08:17 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:08:17 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753375_chr5_1_fastp.fastq.gz, data/processed/ERR753375_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753375_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753375_chr5_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753375_chr5_1_fastp.fastq.gz, data/processed/ERR753375_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 17:08:24 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T170800.898905.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:08:25 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753376_chr5_1.fastq.gz, data/raw/ERR753376_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753376_chr5_1_fastp.fastq.gz, data/processed/ERR753376_chr5_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753376_chr5_1_fastp.fastq.gz, data/processed/ERR753376_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1402798\n",
      "total bases: 106050477\n",
      "Q20 bases: 105320649(99.3118%)\n",
      "Q30 bases: 102421962(96.5785%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1402798\n",
      "total bases: 105773011\n",
      "Q20 bases: 104493043(98.7899%)\n",
      "Q30 bases: 100281618(94.8083%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 875003\n",
      "total bases: 66456395\n",
      "Q20 bases: 66207960(99.6262%)\n",
      "Q30 bases: 65072453(97.9175%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 875003\n",
      "total bases: 66430859\n",
      "Q20 bases: 65983780(99.327%)\n",
      "Q30 bases: 64300470(96.7931%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1750006\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1055590\n",
      "reads with adapter trimmed: 1316\n",
      "bases trimmed due to adapters: 32140\n",
      "\n",
      "Duplication rate: 9.15698%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753376_chr5_1.fastq.gz -I data/raw/ERR753376_chr5_2.fastq.gz -o data/processed/ERR753376_chr5_1_fastp.fastq.gz -O data/processed/ERR753376_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Fri Aug 18 17:08:30 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:08:30 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753376_chr5_1_fastp.fastq.gz, data/processed/ERR753376_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753376_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753376_chr5_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753376_chr5_1_fastp.fastq.gz, data/processed/ERR753376_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 100% complete for ERR753376_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 100% complete for ERR753376_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 17:08:39 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:08:39 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753376_chr5_1.fastq.gz, data/raw/ERR753376_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753376_chr5_1_fastqc.html, results/fastqc_result/ERR753376_chr5_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753376_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753376_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 17:08:52 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T170824.815650.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:08:53 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753377_chr5_1.fastq.gz, data/raw/ERR753377_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753377_chr5_1_fastqc.html, results/fastqc_result/ERR753377_chr5_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753377_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753377_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 17:09:06 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:09:06 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753377_chr5_1.fastq.gz, data/raw/ERR753377_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753377_chr5_1_fastp.fastq.gz, data/processed/ERR753377_chr5_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753377_chr5_2_fastp.fastq.gz, data/processed/ERR753377_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1456925\n",
      "total bases: 110159308\n",
      "Q20 bases: 109389376(99.3011%)\n",
      "Q30 bases: 106378169(96.5676%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1456925\n",
      "total bases: 109900000\n",
      "Q20 bases: 108632355(98.8465%)\n",
      "Q30 bases: 104449264(95.0403%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 929607\n",
      "total bases: 70603984\n",
      "Q20 bases: 70333119(99.6164%)\n",
      "Q30 bases: 69114056(97.8897%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 929607\n",
      "total bases: 70578254\n",
      "Q20 bases: 70125662(99.3587%)\n",
      "Q30 bases: 68418786(96.9403%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1859214\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1054636\n",
      "reads with adapter trimmed: 1262\n",
      "bases trimmed due to adapters: 30893\n",
      "\n",
      "Duplication rate: 9.20844%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 119\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753377_chr5_1.fastq.gz -I data/raw/ERR753377_chr5_2.fastq.gz -o data/processed/ERR753377_chr5_1_fastp.fastq.gz -O data/processed/ERR753377_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Fri Aug 18 17:09:11 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:09:11 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753377_chr5_1_fastp.fastq.gz, data/processed/ERR753377_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753377_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753377_chr5_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753377_chr5_2_fastp.fastq.gz, data/processed/ERR753377_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 17:09:20 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T170852.467771.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:09:21 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753378_chr5_1.fastq.gz, data/raw/ERR753378_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753378_chr5_1_fastqc.html, results/fastqc_result/ERR753378_chr5_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753378_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753378_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 17:09:35 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:09:35 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753378_chr5_1.fastq.gz, data/raw/ERR753378_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753378_chr5_1_fastp.fastq.gz, data/processed/ERR753378_chr5_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753378_chr5_2_fastp.fastq.gz, data/processed/ERR753378_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1582656\n",
      "total bases: 120172394\n",
      "Q20 bases: 119702143(99.6087%)\n",
      "Q30 bases: 116149817(96.6527%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1582656\n",
      "total bases: 120057743\n",
      "Q20 bases: 119018888(99.1347%)\n",
      "Q30 bases: 113653667(94.6658%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1368395\n",
      "total bases: 103992525\n",
      "Q20 bases: 103812087(99.8265%)\n",
      "Q30 bases: 102035807(98.1184%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1368395\n",
      "total bases: 103984175\n",
      "Q20 bases: 103483359(99.5184%)\n",
      "Q30 bases: 100427300(96.5794%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2736790\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 428522\n",
      "reads with adapter trimmed: 1908\n",
      "bases trimmed due to adapters: 45702\n",
      "\n",
      "Duplication rate: 2.03313%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753378_chr5_1.fastq.gz -I data/raw/ERR753378_chr5_2.fastq.gz -o data/processed/ERR753378_chr5_1_fastp.fastq.gz -O data/processed/ERR753378_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Fri Aug 18 17:09:41 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:09:41 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753378_chr5_1_fastp.fastq.gz, data/processed/ERR753378_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753378_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753378_chr5_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753378_chr5_2_fastp.fastq.gz, data/processed/ERR753378_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 17:09:52 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T170920.702352.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:09:53 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR696683_chr5_1_fastp.fastq.gz, data/processed/ERR696683_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR696683_chr5.sam, results/mapped_reads/ERR696683_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR696683_chr5_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR696683_chr5_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.74 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 70.80 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.55 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.39 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 40.75 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 118.805 sec; CPU: 113.240 sec\n",
      "[Fri Aug 18 17:13:30 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:13:30 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR696683_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr5.bam, results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR696683_chr5.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR696683_chr5_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 17:14:13 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:14:13 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR696683_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "17:14:14.813 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 17:14:14 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR696683_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR696683_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 17:14:14 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 17:14:14\tMarkDuplicates\tStart of doWork freeMemory: 528793352; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:14:14\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 17:14:14\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam\n",
      "INFO\t2023-08-18 17:14:16\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:81,252,589\n",
      "INFO\t2023-08-18 17:14:16\tMarkDuplicates\tTracking 728 as yet unmatched pairs. 728 records in RAM.\n",
      "INFO\t2023-08-18 17:14:17\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 5:168,199,832\n",
      "INFO\t2023-08-18 17:14:17\tMarkDuplicates\tTracking 289 as yet unmatched pairs. 289 records in RAM.\n",
      "INFO\t2023-08-18 17:14:18\tMarkDuplicates\tRead 2201600 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 17:14:18\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 539934248; totalMemory: 847249408; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:14:18\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 17:14:18\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 17:14:18\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 17:14:18\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 17:14:19\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 904878296; totalMemory: 1451229184; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:14:19\tMarkDuplicates\tMarking 201846 records as duplicates.\n",
      "INFO\t2023-08-18 17:14:19\tMarkDuplicates\tFound 99 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam\n",
      "INFO\t2023-08-18 17:14:19\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 17:14:33\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 17:14:33\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 17:14:33\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 17:14:33\tMarkDuplicates\tBefore output close freeMemory: 528090704; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:14:33\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 17:14:33\tMarkDuplicates\tAfter output close freeMemory: 528090704; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 17:14:33 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.30 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 17:14:34 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T170952.470295.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:14:35 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753368_chr5_1_fastp.fastq.gz, data/processed/ERR753368_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753368_chr5.sam, results/mapped_reads/ERR753368_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753368_chr5_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753368_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.65 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 70.27 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.56 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.39 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 38.84 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 116.539 sec; CPU: 110.706 sec\n",
      "[Fri Aug 18 17:18:44 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:18:44 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753368_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr5.bam, results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753368_chr5.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753368_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 17:19:32 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:19:32 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753368_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753368_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "17:19:33.317 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 17:19:33 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753368_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753368_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 17:19:33 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 17:19:33\tMarkDuplicates\tStart of doWork freeMemory: 528788440; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:19:33\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 17:19:33\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam\n",
      "INFO\t2023-08-18 17:19:34\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:72,250,220\n",
      "INFO\t2023-08-18 17:19:34\tMarkDuplicates\tTracking 478 as yet unmatched pairs. 478 records in RAM.\n",
      "INFO\t2023-08-18 17:19:36\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 5:138,184,157\n",
      "INFO\t2023-08-18 17:19:36\tMarkDuplicates\tTracking 440 as yet unmatched pairs. 440 records in RAM.\n",
      "INFO\t2023-08-18 17:19:37\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 5:179,891,658\n",
      "INFO\t2023-08-18 17:19:37\tMarkDuplicates\tTracking 24 as yet unmatched pairs. 24 records in RAM.\n",
      "INFO\t2023-08-18 17:19:37\tMarkDuplicates\tRead 3048194 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 17:19:37\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 460307552; totalMemory: 859832320; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:19:37\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 17:19:38\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 17:19:38\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 17:19:38\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 17:19:38\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1039092464; totalMemory: 1585446912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:19:38\tMarkDuplicates\tMarking 74561 records as duplicates.\n",
      "INFO\t2023-08-18 17:19:38\tMarkDuplicates\tFound 483 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam\n",
      "INFO\t2023-08-18 17:19:38\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 17:19:57\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 17:19:57\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 17:19:57\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 17:19:57\tMarkDuplicates\tBefore output close freeMemory: 528087072; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:19:57\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 17:19:57\tMarkDuplicates\tAfter output close freeMemory: 528087072; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 17:19:57 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.40 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 17:19:59 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T171434.774587.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:20:01 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753369_chr5_1_fastp.fastq.gz, data/processed/ERR753369_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753369_chr5.sam, results/mapped_reads/ERR753369_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753369_chr5_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753369_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.64 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 69.54 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.58 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.39 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 40.14 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 117.316 sec; CPU: 111.291 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 17:24:30 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:24:30 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753369_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr5.bam, results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753369_chr5.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753369_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 17:25:28 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:25:28 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753369_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753369_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "17:25:29.555 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 17:25:29 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753369_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753369_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 17:25:29 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 17:25:29\tMarkDuplicates\tStart of doWork freeMemory: 528784504; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:25:29\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 17:25:29\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam\n",
      "INFO\t2023-08-18 17:25:31\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:69,509,473\n",
      "INFO\t2023-08-18 17:25:31\tMarkDuplicates\tTracking 481 as yet unmatched pairs. 481 records in RAM.\n",
      "INFO\t2023-08-18 17:25:32\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 5:132,812,908\n",
      "INFO\t2023-08-18 17:25:32\tMarkDuplicates\tTracking 397 as yet unmatched pairs. 397 records in RAM.\n",
      "INFO\t2023-08-18 17:25:34\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 5:173,999,730\n",
      "INFO\t2023-08-18 17:25:34\tMarkDuplicates\tTracking 121 as yet unmatched pairs. 121 records in RAM.\n",
      "INFO\t2023-08-18 17:25:34\tMarkDuplicates\tRead 3250648 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 17:25:34\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 479056912; totalMemory: 900726784; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:25:34\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 17:25:34\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 17:25:35\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 17:25:35\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 17:25:35\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 891239336; totalMemory: 1437597696; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:25:35\tMarkDuplicates\tMarking 92900 records as duplicates.\n",
      "INFO\t2023-08-18 17:25:35\tMarkDuplicates\tFound 490 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam\n",
      "INFO\t2023-08-18 17:25:35\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 17:25:54\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 17:25:54\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 17:25:54\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 17:25:54\tMarkDuplicates\tBefore output close freeMemory: 528082896; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:25:54\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 17:25:54\tMarkDuplicates\tAfter output close freeMemory: 528082896; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 17:25:54 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.42 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 17:25:57 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T171959.846575.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:25:58 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753370_chr5_1_fastp.fastq.gz, data/processed/ERR753370_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753370_chr5.sam, results/mapped_reads/ERR753370_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753370_chr5_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753370_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.61 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 71.10 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.55 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.39 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 39.95 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 119.292 sec; CPU: 112.610 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 17:32:56 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:32:56 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753370_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr5.bam, results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753370_chr5.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753370_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 17:34:54 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:34:54 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753370_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753370_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "17:34:55.915 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 17:34:55 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753370_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753370_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 17:34:55 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 17:34:55\tMarkDuplicates\tStart of doWork freeMemory: 528785264; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:34:55\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 17:34:55\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam\n",
      "INFO\t2023-08-18 17:34:57\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:42,800,639\n",
      "INFO\t2023-08-18 17:34:57\tMarkDuplicates\tTracking 1038 as yet unmatched pairs. 1038 records in RAM.\n",
      "INFO\t2023-08-18 17:34:59\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 5:75,358,770\n",
      "INFO\t2023-08-18 17:34:59\tMarkDuplicates\tTracking 1655 as yet unmatched pairs. 1655 records in RAM.\n",
      "INFO\t2023-08-18 17:35:00\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 5:115,580,658\n",
      "INFO\t2023-08-18 17:35:00\tMarkDuplicates\tTracking 1498 as yet unmatched pairs. 1498 records in RAM.\n",
      "INFO\t2023-08-18 17:35:02\tMarkDuplicates\tRead     4,000,000 records.  Elapsed time: 00:00:06s.  Time for last 1,000,000:    1s.  Last read position: 5:143,040,715\n",
      "INFO\t2023-08-18 17:35:02\tMarkDuplicates\tTracking 956 as yet unmatched pairs. 956 records in RAM.\n",
      "INFO\t2023-08-18 17:35:04\tMarkDuplicates\tRead     5,000,000 records.  Elapsed time: 00:00:08s.  Time for last 1,000,000:    1s.  Last read position: 5:179,585,760\n",
      "INFO\t2023-08-18 17:35:04\tMarkDuplicates\tTracking 247 as yet unmatched pairs. 247 records in RAM.\n",
      "INFO\t2023-08-18 17:35:04\tMarkDuplicates\tRead 5095159 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 17:35:04\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 429032424; totalMemory: 1047527424; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:35:04\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 17:35:04\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 17:35:05\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 17:35:05\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 17:35:05\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1038039376; totalMemory: 1584398336; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:35:05\tMarkDuplicates\tMarking 357808 records as duplicates.\n",
      "INFO\t2023-08-18 17:35:05\tMarkDuplicates\tFound 194 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam\n",
      "INFO\t2023-08-18 17:35:05\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 17:35:43\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 17:35:43\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 17:35:43\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 17:35:43\tMarkDuplicates\tBefore output close freeMemory: 528082544; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:35:43\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 17:35:43\tMarkDuplicates\tAfter output close freeMemory: 528082544; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 17:35:43 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.80 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 17:35:48 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T172557.199396.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:35:49 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753371_chr5_1_fastp.fastq.gz, data/processed/ERR753371_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753371_chr5.sam, results/mapped_reads/ERR753371_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753371_chr5_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753371_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.61 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 71.87 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.56 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.38 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 39.54 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 119.029 sec; CPU: 112.970 sec\n",
      "[Fri Aug 18 17:39:03 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:39:03 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753371_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr5.bam, results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753371_chr5.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753371_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 17:39:34 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:39:34 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753371_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753371_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "17:39:35.566 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 17:39:35 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753371_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753371_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 17:39:35 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 17:39:35\tMarkDuplicates\tStart of doWork freeMemory: 528778128; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:39:35\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 17:39:35\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam\n",
      "INFO\t2023-08-18 17:39:37\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:109,819,561\n",
      "INFO\t2023-08-18 17:39:37\tMarkDuplicates\tTracking 139 as yet unmatched pairs. 139 records in RAM.\n",
      "INFO\t2023-08-18 17:39:38\tMarkDuplicates\tRead 1717822 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 17:39:38\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 345010904; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:39:38\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 17:39:38\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 17:39:38\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 17:39:39\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 17:39:39\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 818878928; totalMemory: 1365245952; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:39:39\tMarkDuplicates\tMarking 690637 records as duplicates.\n",
      "INFO\t2023-08-18 17:39:39\tMarkDuplicates\tFound 61 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam\n",
      "INFO\t2023-08-18 17:39:39\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 17:39:49\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 17:39:49\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 17:39:49\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 17:39:49\tMarkDuplicates\tBefore output close freeMemory: 528074168; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:39:49\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 17:39:49\tMarkDuplicates\tAfter output close freeMemory: 528074168; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 17:39:49 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.24 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 17:39:50 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T173548.359160.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:39:52 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753372_chr5_1_fastp.fastq.gz, data/processed/ERR753372_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753372_chr5.sam, results/mapped_reads/ERR753372_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753372_chr5_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR753372_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.63 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 71.11 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.53 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.38 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 38.11 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 116.829 sec; CPU: 110.762 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 17:44:10 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:44:10 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753372_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr5.bam, results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753372_chr5.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753372_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 17:45:07 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:45:07 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753372_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753372_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "17:45:08.466 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 17:45:08 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753372_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753372_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 17:45:08 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 17:45:08\tMarkDuplicates\tStart of doWork freeMemory: 528817024; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:45:08\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 17:45:08\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam\n",
      "INFO\t2023-08-18 17:45:10\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:70,941,594\n",
      "INFO\t2023-08-18 17:45:10\tMarkDuplicates\tTracking 503 as yet unmatched pairs. 503 records in RAM.\n",
      "INFO\t2023-08-18 17:45:11\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 5:134,741,303\n",
      "INFO\t2023-08-18 17:45:11\tMarkDuplicates\tTracking 569 as yet unmatched pairs. 569 records in RAM.\n",
      "INFO\t2023-08-18 17:45:12\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 5:176,864,726\n",
      "INFO\t2023-08-18 17:45:12\tMarkDuplicates\tTracking 137 as yet unmatched pairs. 137 records in RAM.\n",
      "INFO\t2023-08-18 17:45:13\tMarkDuplicates\tRead 3202425 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 17:45:13\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 752995184; totalMemory: 1169162240; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:45:13\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 17:45:13\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 17:45:13\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 17:45:14\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 17:45:14\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1159707280; totalMemory: 1706033152; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:45:14\tMarkDuplicates\tMarking 77158 records as duplicates.\n",
      "INFO\t2023-08-18 17:45:14\tMarkDuplicates\tFound 518 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam\n",
      "INFO\t2023-08-18 17:45:14\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 17:45:32\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 17:45:32\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 17:45:32\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 17:45:32\tMarkDuplicates\tBefore output close freeMemory: 528115592; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:45:32\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 17:45:32\tMarkDuplicates\tAfter output close freeMemory: 528115592; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 17:45:32 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.39 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 17:45:34 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T173951.083832.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:45:35 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753373_chr5_1_fastp.fastq.gz, data/processed/ERR753373_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753373_chr5.sam, results/mapped_reads/ERR753373_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753373_chr5_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR753373_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.61 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 62.83 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.54 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.37 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 37.29 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 107.297 sec; CPU: 101.643 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 17:50:50 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:50:50 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753373_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr5.bam, results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753373_chr5.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753373_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 17:52:25 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:52:25 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753373_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753373_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "17:52:26.438 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 17:52:26 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753373_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753373_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 17:52:26 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 17:52:26\tMarkDuplicates\tStart of doWork freeMemory: 528820440; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:52:26\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 17:52:26\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam\n",
      "INFO\t2023-08-18 17:52:28\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:56,495,436\n",
      "INFO\t2023-08-18 17:52:28\tMarkDuplicates\tTracking 269 as yet unmatched pairs. 269 records in RAM.\n",
      "INFO\t2023-08-18 17:52:29\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 5:102,420,273\n",
      "INFO\t2023-08-18 17:52:29\tMarkDuplicates\tTracking 269 as yet unmatched pairs. 269 records in RAM.\n",
      "INFO\t2023-08-18 17:52:31\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 5:152,603,320\n",
      "INFO\t2023-08-18 17:52:31\tMarkDuplicates\tTracking 150 as yet unmatched pairs. 150 records in RAM.\n",
      "INFO\t2023-08-18 17:52:31\tMarkDuplicates\tRead 3481677 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 17:52:31\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 556905496; totalMemory: 1002438656; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:52:31\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 17:52:32\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 17:52:32\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 17:52:32\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 17:52:32\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 992987464; totalMemory: 1539309568; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:52:32\tMarkDuplicates\tMarking 92730 records as duplicates.\n",
      "INFO\t2023-08-18 17:52:32\tMarkDuplicates\tFound 30640 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam\n",
      "INFO\t2023-08-18 17:52:32\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 17:53:01\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 17:53:01\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 17:53:01\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 17:53:01\tMarkDuplicates\tBefore output close freeMemory: 528119328; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:53:01\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 17:53:01\tMarkDuplicates\tAfter output close freeMemory: 528119328; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 17:53:01 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.59 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 17:53:04 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T174534.409159.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:53:05 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753374_chr5_1_fastp.fastq.gz, data/processed/ERR753374_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753374_chr5.sam, results/mapped_reads/ERR753374_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753374_chr5_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR753374_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.60 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 62.80 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.55 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.37 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 37.27 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 102.438 sec; CPU: 101.589 sec\n",
      "[Fri Aug 18 17:56:39 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:56:39 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753374_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr5.bam, results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753374_chr5.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753374_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 17:57:28 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:57:28 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753374_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753374_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "17:57:28.922 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 17:57:28 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753374_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753374_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 17:57:28 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 17:57:28\tMarkDuplicates\tStart of doWork freeMemory: 528788184; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:57:28\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 17:57:28\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam\n",
      "INFO\t2023-08-18 17:57:30\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:88,349,068\n",
      "INFO\t2023-08-18 17:57:30\tMarkDuplicates\tTracking 233 as yet unmatched pairs. 233 records in RAM.\n",
      "INFO\t2023-08-18 17:57:32\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 5:177,739,569\n",
      "INFO\t2023-08-18 17:57:32\tMarkDuplicates\tTracking 64 as yet unmatched pairs. 64 records in RAM.\n",
      "INFO\t2023-08-18 17:57:32\tMarkDuplicates\tRead 2038926 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 17:57:32\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 598827512; totalMemory: 889192448; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:57:32\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 17:57:32\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 17:57:32\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 17:57:32\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 17:57:32\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 879707104; totalMemory: 1426063360; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:57:32\tMarkDuplicates\tMarking 219071 records as duplicates.\n",
      "INFO\t2023-08-18 17:57:32\tMarkDuplicates\tFound 73634 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam\n",
      "INFO\t2023-08-18 17:57:32\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 17:57:49\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 17:57:49\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 17:57:49\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 17:57:49\tMarkDuplicates\tBefore output close freeMemory: 528085192; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 17:57:49\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 17:57:49\tMarkDuplicates\tAfter output close freeMemory: 528085192; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 17:57:49 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.35 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 17:57:51 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T175304.640801.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 17:57:52 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753375_chr5_1_fastp.fastq.gz, data/processed/ERR753375_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753375_chr5.sam, results/mapped_reads/ERR753375_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753375_chr5_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753375_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.60 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 62.73 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.54 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.37 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 37.34 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 102.445 sec; CPU: 101.589 sec\n",
      "[Fri Aug 18 18:00:30 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:00:30 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753375_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr5.bam, results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753375_chr5.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753375_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 18:00:54 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:00:54 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753375_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753375_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:00:55.293 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 18:00:55 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753375_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753375_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 18:00:55 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 18:00:55\tMarkDuplicates\tStart of doWork freeMemory: 528782776; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 18:00:55\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 18:00:55\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam\n",
      "INFO\t2023-08-18 18:00:56\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:136,231,428\n",
      "INFO\t2023-08-18 18:00:56\tMarkDuplicates\tTracking 187 as yet unmatched pairs. 187 records in RAM.\n",
      "INFO\t2023-08-18 18:00:57\tMarkDuplicates\tRead 1429511 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 18:00:57\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 311660720; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 18:00:57\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 18:00:57\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 18:00:57\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 18:00:58\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 18:00:58\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 761213960; totalMemory: 1307574272; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 18:00:58\tMarkDuplicates\tMarking 67768 records as duplicates.\n",
      "INFO\t2023-08-18 18:00:58\tMarkDuplicates\tFound 153 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam\n",
      "INFO\t2023-08-18 18:00:58\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 18:01:06\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 18:01:06\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 18:01:06\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 18:01:06\tMarkDuplicates\tBefore output close freeMemory: 528081208; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 18:01:06\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 18:01:06\tMarkDuplicates\tAfter output close freeMemory: 528081208; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 18:01:06 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.19 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 18:01:07 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T175751.768475.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:01:08 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753376_chr5_1_fastp.fastq.gz, data/processed/ERR753376_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753376_chr5.sam, results/mapped_reads/ERR753376_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753376_chr5_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR753376_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.60 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 62.83 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.54 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.36 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 37.66 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 102.834 sec; CPU: 101.988 sec\n",
      "[Fri Aug 18 18:03:56 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:03:56 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753376_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr5.bam, results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753376_chr5.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753376_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 18:04:27 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:04:27 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753376_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753376_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:04:28.616 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 18:04:28 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753376_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753376_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 18:04:28 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 18:04:28\tMarkDuplicates\tStart of doWork freeMemory: 528824792; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 18:04:28\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 18:04:28\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam\n",
      "INFO\t2023-08-18 18:04:30\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:109,378,596\n",
      "INFO\t2023-08-18 18:04:30\tMarkDuplicates\tTracking 265 as yet unmatched pairs. 265 records in RAM.\n",
      "INFO\t2023-08-18 18:04:31\tMarkDuplicates\tRead 1758630 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 18:04:31\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 341063680; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 18:04:31\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 18:04:31\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 18:04:31\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 18:04:31\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 18:04:31\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 822073544; totalMemory: 1368391680; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 18:04:31\tMarkDuplicates\tMarking 249671 records as duplicates.\n",
      "INFO\t2023-08-18 18:04:31\tMarkDuplicates\tFound 80 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam\n",
      "INFO\t2023-08-18 18:04:31\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 18:04:42\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 18:04:42\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 18:04:42\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 18:04:42\tMarkDuplicates\tBefore output close freeMemory: 528123472; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 18:04:42\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 18:04:42\tMarkDuplicates\tAfter output close freeMemory: 528123472; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 18:04:42 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.23 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 18:04:43 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T180107.824959.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:04:44 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753377_chr5_1_fastp.fastq.gz, data/processed/ERR753377_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753377_chr5.sam, results/mapped_reads/ERR753377_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753377_chr5_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753377_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.61 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 62.25 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.54 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.37 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 37.09 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 101.713 sec; CPU: 100.862 sec\n",
      "[Fri Aug 18 18:07:34 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:07:34 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753377_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr5.bam, results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753377_chr5.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753377_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 18:08:05 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:08:05 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753377_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753377_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:08:06.583 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 18:08:06 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753377_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753377_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 18:08:06 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 18:08:06\tMarkDuplicates\tStart of doWork freeMemory: 528779968; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 18:08:06\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 18:08:06\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam\n",
      "INFO\t2023-08-18 18:08:08\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:97,717,058\n",
      "INFO\t2023-08-18 18:08:08\tMarkDuplicates\tTracking 399 as yet unmatched pairs. 399 records in RAM.\n",
      "INFO\t2023-08-18 18:08:09\tMarkDuplicates\tRead 1869830 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 18:08:09\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 329201256; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 18:08:09\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 18:08:09\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 18:08:09\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 18:08:09\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 18:08:09\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 822027984; totalMemory: 1368391680; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 18:08:09\tMarkDuplicates\tMarking 263270 records as duplicates.\n",
      "INFO\t2023-08-18 18:08:09\tMarkDuplicates\tFound 72 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam\n",
      "INFO\t2023-08-18 18:08:09\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 18:08:21\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 18:08:21\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 18:08:21\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 18:08:21\tMarkDuplicates\tBefore output close freeMemory: 528077768; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 18:08:21\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 18:08:21\tMarkDuplicates\tAfter output close freeMemory: 528077768; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 18:08:21 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.24 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 18:08:22 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T180443.970338.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:08:23 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753378_chr5_1_fastp.fastq.gz, data/processed/ERR753378_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753378_chr5.sam, results/mapped_reads/ERR753378_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753378_chr5_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR753378_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.61 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 62.56 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.54 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.37 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 37.30 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 102.242 sec; CPU: 101.390 sec\n",
      "[Fri Aug 18 18:11:51 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:11:51 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753378_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr5.bam, results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753378_chr5.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753378_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 18:12:32 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:12:32 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753378_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753378_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:12:33.700 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 18:12:33 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753378_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753378_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 18:12:33 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 18:12:33\tMarkDuplicates\tStart of doWork freeMemory: 528783936; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 18:12:33\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 18:12:33\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam\n",
      "INFO\t2023-08-18 18:12:35\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:77,325,563\n",
      "INFO\t2023-08-18 18:12:35\tMarkDuplicates\tTracking 548 as yet unmatched pairs. 548 records in RAM.\n",
      "INFO\t2023-08-18 18:12:36\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 5:141,529,639\n",
      "INFO\t2023-08-18 18:12:36\tMarkDuplicates\tTracking 415 as yet unmatched pairs. 415 records in RAM.\n",
      "INFO\t2023-08-18 18:12:37\tMarkDuplicates\tRead 2755612 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 18:12:37\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 598246792; totalMemory: 965738496; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 18:12:37\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 18:12:37\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 18:12:38\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 18:12:38\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 18:12:38\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 956249576; totalMemory: 1502609408; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 18:12:38\tMarkDuplicates\tMarking 66436 records as duplicates.\n",
      "INFO\t2023-08-18 18:12:38\tMarkDuplicates\tFound 462 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam\n",
      "INFO\t2023-08-18 18:12:38\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 18:12:53\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 18:12:53\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 18:12:53\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 18:12:53\tMarkDuplicates\tBefore output close freeMemory: 528081752; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 18:12:53\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 18:12:53\tMarkDuplicates\tAfter output close freeMemory: 528081752; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 18:12:53 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.34 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 18:12:55 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T180822.563003.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:12:56 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR696683_chr5_dedup.bam\n",
      "    output: results/variants/ERR696683_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR696683_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR696683_chr5.vcf\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "index file data/reference/genome.fa.fai not found, generating...\n",
      "[Fri Aug 18 18:13:44 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T181255.950813.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:13:45 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753368_chr5_dedup.bam\n",
      "    output: results/variants/ERR753368_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753368_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753368_chr5.vcf\n",
      "    wildcards: sample=ERR753368_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 18:14:49 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T181344.613070.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:14:50 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753369_chr5_dedup.bam\n",
      "    output: results/variants/ERR753369_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753369_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753369_chr5.vcf\n",
      "    wildcards: sample=ERR753369_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 18:15:56 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T181449.575347.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:15:57 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753370_chr5_dedup.bam\n",
      "    output: results/variants/ERR753370_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753370_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753370_chr5.vcf\n",
      "    wildcards: sample=ERR753370_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 18:17:42 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T181556.731876.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:17:43 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753371_chr5_dedup.bam\n",
      "    output: results/variants/ERR753371_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753371_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753371_chr5.vcf\n",
      "    wildcards: sample=ERR753371_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 18:18:17 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T181742.439552.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:18:19 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753372_chr5_dedup.bam\n",
      "    output: results/variants/ERR753372_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753372_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753372_chr5.vcf\n",
      "    wildcards: sample=ERR753372_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 18:19:24 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T181818.027782.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:19:25 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753373_chr5_dedup.bam\n",
      "    output: results/variants/ERR753373_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753373_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753373_chr5.vcf\n",
      "    wildcards: sample=ERR753373_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 18:20:50 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T181924.728338.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:20:52 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753374_chr5_dedup.bam\n",
      "    output: results/variants/ERR753374_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753374_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753374_chr5.vcf\n",
      "    wildcards: sample=ERR753374_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 18:21:48 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T182051.090778.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:21:49 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753375_chr5_dedup.bam\n",
      "    output: results/variants/ERR753375_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753375_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753375_chr5.vcf\n",
      "    wildcards: sample=ERR753375_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 18:22:28 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T182148.448306.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:22:29 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753376_chr5_dedup.bam\n",
      "    output: results/variants/ERR753376_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753376_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753376_chr5.vcf\n",
      "    wildcards: sample=ERR753376_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 18:23:10 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T182228.801488.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:23:11 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753377_chr5_dedup.bam\n",
      "    output: results/variants/ERR753377_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753377_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753377_chr5.vcf\n",
      "    wildcards: sample=ERR753377_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 18:23:53 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T182310.449144.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:23:54 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753378_chr5_dedup.bam\n",
      "    output: results/variants/ERR753378_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753378_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753378_chr5.vcf\n",
      "    wildcards: sample=ERR753378_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 18:24:53 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T182353.265468.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:24:54 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR696683_chr5.vcf\n",
      "    output: results/variants/vep/ERR696683_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR696683_chr5.txt\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 18:25:18 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T182453.170943.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:25:19 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753368_chr5.vcf\n",
      "    output: results/variants/vep/ERR753368_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753368_chr5.txt\n",
      "    wildcards: sample=ERR753368_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 18:25:47 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T182518.605467.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:25:48 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753369_chr5.vcf\n",
      "    output: results/variants/vep/ERR753369_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753369_chr5.txt\n",
      "    wildcards: sample=ERR753369_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 18:26:13 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T182547.316461.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:26:14 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753370_chr5.vcf\n",
      "    output: results/variants/vep/ERR753370_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753370_chr5.txt\n",
      "    wildcards: sample=ERR753370_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 18:26:41 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T182613.638720.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:26:42 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753371_chr5.vcf\n",
      "    output: results/variants/vep/ERR753371_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753371_chr5.txt\n",
      "    wildcards: sample=ERR753371_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 18:27:02 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T182641.901288.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:27:03 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753372_chr5.vcf\n",
      "    output: results/variants/vep/ERR753372_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753372_chr5.txt\n",
      "    wildcards: sample=ERR753372_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 18:27:29 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T182702.460912.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:27:30 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753373_chr5.vcf\n",
      "    output: results/variants/vep/ERR753373_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753373_chr5.txt\n",
      "    wildcards: sample=ERR753373_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 18:27:57 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T182729.543017.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:27:58 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753374_chr5.vcf\n",
      "    output: results/variants/vep/ERR753374_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753374_chr5.txt\n",
      "    wildcards: sample=ERR753374_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 18:28:19 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T182757.592513.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:28:20 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753375_chr5.vcf\n",
      "    output: results/variants/vep/ERR753375_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753375_chr5.txt\n",
      "    wildcards: sample=ERR753375_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 18:28:40 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T182819.558389.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:28:41 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753376_chr5.vcf\n",
      "    output: results/variants/vep/ERR753376_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753376_chr5.txt\n",
      "    wildcards: sample=ERR753376_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 18:29:03 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T182840.892214.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:29:04 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753377_chr5.vcf\n",
      "    output: results/variants/vep/ERR753377_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753377_chr5.txt\n",
      "    wildcards: sample=ERR753377_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 18:29:25 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T182903.166819.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:29:26 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753378_chr5.vcf\n",
      "    output: results/variants/vep/ERR753378_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753378_chr5.txt\n",
      "    wildcards: sample=ERR753378_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 18:29:51 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T182925.735173.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job              count    min threads    max threads\n",
      "-------------  -------  -------------  -------------\n",
      "parsing_dataR        1              1              1\n",
      "total                1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 18:29:51 2023]\n",
      "rule parsing_dataR:\n",
      "    input: code/05parsing_vep_data.R\n",
      "    output: tasks/13parsing_dataR.done\n",
      "    jobid: 0\n",
      "    reason: Params have changed since last execution\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Warning message:\n",
      "package ‘glue’ was built under R version 4.2.3 \n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">>> This is your sample list: ERR696683_chr5\n",
      ">>> This is your sample list: ERR753368_chr5\n",
      ">>> This is your sample list: ERR753369_chr5\n",
      ">>> This is your sample list: ERR753370_chr5\n",
      ">>> This is your sample list: ERR753371_chr5\n",
      ">>> This is your sample list: ERR753372_chr5\n",
      ">>> This is your sample list: ERR753373_chr5\n",
      ">>> This is your sample list: ERR753374_chr5\n",
      ">>> This is your sample list: ERR753375_chr5\n",
      ">>> This is your sample list: ERR753376_chr5\n",
      ">>> This is your sample list: ERR753377_chr5\n",
      ">>> This is your sample list: ERR753378_chr5\n",
      ">>> This is your gene: APC\n",
      "# A tibble: 0 × 44\n",
      "# ℹ 44 variables: #uploaded_variation <chr>, location <chr>, allele <chr>,\n",
      "#   gene <chr>, feature <chr>, feature_type <chr>, consequence <chr>,\n",
      "#   cdna_position <chr>, cds_position <chr>, protein_position <chr>,\n",
      "#   amino_acids <chr>, codons <chr>, existing_variation <chr>, impact <chr>,\n",
      "#   distance <chr>, strand <chr>, flags <chr>, variant_class <chr>,\n",
      "#   symbol <chr>, symbol_source <chr>, hgnc_id <chr>, biotype <chr>,\n",
      "#   mane_select <chr>, mane_plus_clinical <chr>, tsl <chr>, appris <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 71 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112707850_C/T       5:1127078… T      ENSG… ENST00… Transcript   missense_v…\n",
      "2 5_112707850_C/T       5:1127078… T      ENSG… ENST00… Transcript   missense_v…\n",
      "3 5_112707850_C/T       5:1127078… T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 65 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 69 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112724597_G/C       5:1127245… C      ENSG… ENST00… Transcript   intron_var…\n",
      "2 5_112724597_G/C       5:1127245… C      ENSG… ENST00… Transcript   intron_var…\n",
      "3 5_112724597_G/C       5:1127245… C      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 63 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 86 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "2 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "3 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112778038_C/T       5:1127780… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 80 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 0 × 44\n",
      "# ℹ 44 variables: #uploaded_variation <chr>, location <chr>, allele <chr>,\n",
      "#   gene <chr>, feature <chr>, feature_type <chr>, consequence <chr>,\n",
      "#   cdna_position <chr>, cds_position <chr>, protein_position <chr>,\n",
      "#   amino_acids <chr>, codons <chr>, existing_variation <chr>, impact <chr>,\n",
      "#   distance <chr>, strand <chr>, flags <chr>, variant_class <chr>,\n",
      "#   symbol <chr>, symbol_source <chr>, hgnc_id <chr>, biotype <chr>,\n",
      "#   mane_select <chr>, mane_plus_clinical <chr>, tsl <chr>, appris <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 64 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112707850_C/T       5:1127078… T      ENSG… ENST00… Transcript   missense_v…\n",
      "2 5_112707850_C/T       5:1127078… T      ENSG… ENST00… Transcript   missense_v…\n",
      "3 5_112707850_C/T       5:1127078… T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 58 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 78 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112775373_G/A       5:1127753… A      ENSG… ENST00… Transcript   intron_var…\n",
      "2 5_112775373_G/A       5:1127753… A      ENSG… ENST00… Transcript   intron_var…\n",
      "3 5_112775373_G/A       5:1127753… A      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112775373_G/A       5:1127753… A      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112775373_G/A       5:1127753… A      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 72 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 59 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112729129_G/C       5:1127291… C      ENSG… ENST00… Transcript   intron_var…\n",
      "2 5_112729129_G/C       5:1127291… C      ENSG… ENST00… Transcript   intron_var…\n",
      "3 5_112729129_G/C       5:1127291… C      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112775373_G/A       5:1127753… A      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112775373_G/A       5:1127753… A      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112775373_G/A       5:1127753… A      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 53 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 61 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "2 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "3 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112778038_C/T       5:1127780… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 55 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 71 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "2 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "3 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112778038_C/T       5:1127780… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 65 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 78 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "2 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "3 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112778038_C/T       5:1127780… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 72 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 66 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "2 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "3 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112778038_C/T       5:1127780… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 60 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "[1] \"##==========================##\"\n",
      "[1] \"##===> WORK FINISHED!!! <===##\"\n",
      "[1] \"##==========================##\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "Touching output file tasks/13parsing_dataR.done.\n",
      "[Fri Aug 18 18:29:55 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T182951.504585.snakemake.log\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['snakemake', '--cores', '1', 'parsing_dataR'], returncode=0)"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Running the workflow for the 5º chormosome\n",
    "\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--force\", \"data/reference/genome.fa\"]) ## rule reference_genome \n",
    "\n",
    "for sample in sample_list:\n",
    "    ## Quality inspection\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/fastqc_result/\" + sample + \"_chr5_1_fastqc.html\",   \n",
    "                    \"data/processed/\" + sample + \"_chr5_1_fastp.fastq.gz\",   \n",
    "                    \"results/fastqc_result/trimmed/\" + sample + \"_chr5_1_fastp_fastqc.html\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    ## Mapping reads\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/mapped_reads/\" + sample + \"_chr5_sorted.sam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr5_sorted.bam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr5_dedup.bam\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    # Extracting variants  \n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/\" + sample + \"_chr5.vcf\"])   \n",
    "\n",
    "## VEP CLI\n",
    "for sample in sample_list:\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/vep/\" + sample + \"_chr5.txt\"])\n",
    "\n",
    "## Extracting the gene PIKCA3 from the VEP files\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"parsing_dataR\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Changing the config file to do the chromosome 7 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['sed', '-i', '50s/APC/BRAF/g', 'config.yaml'], returncode=0)"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subprocess.run([\"sed\", \"-i\", \"6,+12s/chr5/chr7/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\",\"24s/chr5/chr7/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"31s/chromosome.5.fa.gz/chromosome.7.fa.gz/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"50s/APC/BRAF/g\", \"config.yaml\"])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job                 count    min threads    max threads\n",
      "----------------  -------  -------------  -------------\n",
      "reference_genome        1              1              1\n",
      "total                   1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:45:27 2023]\n",
      "rule reference_genome:\n",
      "    output: data/reference/genome.fa\n",
      "    jobid: 0\n",
      "    reason: Forced execution\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "--2023-08-18 20:45:27--  https://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.7.fa.gz\n",
      "Resolviendo ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.169\n",
      "Conectando con ftp.ensembl.org (ftp.ensembl.org)[193.62.193.169]:443... conectado.\n",
      "Petición HTTP enviada, esperando respuesta... 200 OK\n",
      "Longitud: 47416846 (45M) [application/x-gzip]\n",
      "Guardando como: ‘data/reference/genome.fa.gz’\n",
      "\n",
      "     0K .......... .......... .......... .......... ..........  0%  432K 1m47s\n",
      "    50K .......... .......... .......... .......... ..........  0%  867K 80s\n",
      "   100K .......... .......... .......... .......... ..........  0%  867K 71s\n",
      "   150K .......... .......... .......... .......... ..........  0% 18,3M 54s\n",
      "   200K .......... .......... .......... .......... ..........  0% 33,4M 43s\n",
      "   250K .......... .......... .......... .......... ..........  0%  127M 36s\n",
      "   300K .......... .......... .......... .......... ..........  0%  918K 38s\n",
      "   350K .......... .......... .......... .......... ..........  0% 25,4M 34s\n",
      "   400K .......... .......... .......... .......... ..........  0% 64,9M 30s\n",
      "   450K .......... .......... .......... .......... ..........  1% 31,8M 27s\n",
      "   500K .......... .......... .......... .......... ..........  1%  124M 25s\n",
      "   550K .......... .......... .......... .......... ..........  1%  119M 22s\n",
      "   600K .......... .......... .......... .......... ..........  1%  961K 24s\n",
      "   650K .......... .......... .......... .......... ..........  1% 20,0M 23s\n",
      "   700K .......... .......... .......... .......... ..........  1%  118M 21s\n",
      "   750K .......... .......... .......... .......... ..........  1%  141M 20s\n",
      "   800K .......... .......... .......... .......... ..........  1%  140M 19s\n",
      "   850K .......... .......... .......... .......... ..........  1%  117M 18s\n",
      "   900K .......... .......... .......... .......... ..........  2%  119M 17s\n",
      "   950K .......... .......... .......... .......... ..........  2% 47,8M 16s\n",
      "  1000K .......... .......... .......... .......... ..........  2% 43,0M 15s\n",
      "  1050K .......... .......... .......... .......... ..........  2%  137M 15s\n",
      "  1100K .......... .......... .......... .......... ..........  2% 3,35M 14s\n",
      "  1150K .......... .......... .......... .......... ..........  2%  117M 14s\n",
      "  1200K .......... .......... .......... .......... ..........  2% 2,19M 14s\n",
      "  1250K .......... .......... .......... .......... ..........  2% 3,15M 14s\n",
      "  1300K .......... .......... .......... .......... ..........  2%  139M 14s\n",
      "  1350K .......... .......... .......... .......... ..........  3% 30,7M 13s\n",
      "  1400K .......... .......... .......... .......... ..........  3%  134M 13s\n",
      "  1450K .......... .......... .......... .......... ..........  3%  128M 12s\n",
      "  1500K .......... .......... .......... .......... ..........  3% 2,15M 12s\n",
      "  1550K .......... .......... .......... .......... ..........  3% 6,74M 12s\n",
      "  1600K .......... .......... .......... .......... ..........  3% 93,7M 12s\n",
      "  1650K .......... .......... .......... .......... ..........  3%  144M 12s\n",
      "  1700K .......... .......... .......... .......... ..........  3%  123M 11s\n",
      "  1750K .......... .......... .......... .......... ..........  3% 1000K 12s\n",
      "  1800K .......... .......... .......... .......... ..........  3%  132M 12s\n",
      "  1850K .......... .......... .......... .......... ..........  4%  121M 11s\n",
      "  1900K .......... .......... .......... .......... ..........  4%  130M 11s\n",
      "  1950K .......... .......... .......... .......... ..........  4%  103M 11s\n",
      "  2000K .......... .......... .......... .......... ..........  4% 10,4M 11s\n",
      "  2050K .......... .......... .......... .......... ..........  4%  117M 10s\n",
      "  2100K .......... .......... .......... .......... ..........  4%  101M 10s\n",
      "  2150K .......... .......... .......... .......... ..........  4% 77,2M 10s\n",
      "  2200K .......... .......... .......... .......... ..........  4%  698K 11s\n",
      "  2250K .......... .......... .......... .......... ..........  4% 43,3K 33s\n",
      "  2300K .......... .......... .......... .......... ..........  5%  119M 32s\n",
      "  2350K .......... .......... .......... .......... ..........  5%  857K 33s\n",
      "  2400K .......... .......... .......... .......... ..........  5%  121M 32s\n",
      "  2450K .......... .......... .......... .......... ..........  5%  102M 31s\n",
      "  2500K .......... .......... .......... .......... ..........  5%  123M 31s\n",
      "  2550K .......... .......... .......... .......... ..........  5%  892K 31s\n",
      "  2600K .......... .......... .......... .......... ..........  5% 20,9M 30s\n",
      "  2650K .......... .......... .......... .......... ..........  5%  140M 30s\n",
      "  2700K .......... .......... .......... .......... ..........  5%  102M 29s\n",
      "  2750K .......... .......... .......... .......... ..........  6%  118M 29s\n",
      "  2800K .......... .......... .......... .......... ..........  6%  122M 28s\n",
      "  2850K .......... .......... .......... .......... ..........  6%  223K 31s\n",
      "  2900K .......... .......... .......... .......... ..........  6%  137M 30s\n",
      "  2950K .......... .......... .......... .......... ..........  6%  124M 30s\n",
      "  3000K .......... .......... .......... .......... ..........  6%  147M 29s\n",
      "  3050K .......... .......... .......... .......... ..........  6%  372K 31s\n",
      "  3100K .......... .......... .......... .......... ..........  6%  110M 30s\n",
      "  3150K .......... .......... .......... .......... ..........  6%  146M 30s\n",
      "  3200K .......... .......... .......... .......... ..........  7%  128M 29s\n",
      "  3250K .......... .......... .......... .......... ..........  7%  121M 29s\n",
      "  3300K .......... .......... .......... .......... ..........  7% 21,1M 28s\n",
      "  3350K .......... .......... .......... .......... ..........  7%  118M 28s\n",
      "  3400K .......... .......... .......... .......... ..........  7%  118M 27s\n",
      "  3450K .......... .......... .......... .......... ..........  7%  318K 29s\n",
      "  3500K .......... .......... .......... .......... ..........  7%  118M 28s\n",
      "  3550K .......... .......... .......... .......... ..........  7%  120M 28s\n",
      "  3600K .......... .......... .......... .......... ..........  7%  102M 28s\n",
      "  3650K .......... .......... .......... .......... ..........  7%  118M 27s\n",
      "  3700K .......... .......... .......... .......... ..........  8% 98,8M 27s\n",
      "  3750K .......... .......... .......... .......... ..........  8%  123M 26s\n",
      "  3800K .......... .......... .......... .......... ..........  8%  114M 26s\n",
      "  3850K .......... .......... .......... .......... ..........  8%  106M 26s\n",
      "  3900K .......... .......... .......... .......... ..........  8%  115M 25s\n",
      "  3950K .......... .......... .......... .......... ..........  8%  104M 25s\n",
      "  4000K .......... .......... .......... .......... ..........  8% 75,3M 25s\n",
      "  4050K .......... .......... .......... .......... ..........  8%  930K 25s\n",
      "  4100K .......... .......... .......... .......... ..........  8% 22,7M 25s\n",
      "  4150K .......... .......... .......... .......... ..........  9%  121M 24s\n",
      "  4200K .......... .......... .......... .......... ..........  9% 61,0M 24s\n",
      "  4250K .......... .......... .......... .......... ..........  9%  116M 24s\n",
      "  4300K .......... .......... .......... .......... ..........  9%  172M 23s\n",
      "  4350K .......... .......... .......... .......... ..........  9% 13,5M 23s\n",
      "  4400K .......... .......... .......... .......... ..........  9%  134M 23s\n",
      "  4450K .......... .......... .......... .......... ..........  9%  133M 23s\n",
      "  4500K .......... .......... .......... .......... ..........  9%  154M 22s\n",
      "  4550K .......... .......... .......... .......... ..........  9%  144M 22s\n",
      "  4600K .......... .......... .......... .......... .......... 10%  125M 22s\n",
      "  4650K .......... .......... .......... .......... .......... 10%  144M 22s\n",
      "  4700K .......... .......... .......... .......... .......... 10%  126M 21s\n",
      "  4750K .......... .......... .......... .......... .......... 10%  118M 21s\n",
      "  4800K .......... .......... .......... .......... .......... 10%  117M 21s\n",
      "  4850K .......... .......... .......... .......... .......... 10% 2,18M 21s\n",
      "  4900K .......... .......... .......... .......... .......... 10% 1,96M 21s\n",
      "  4950K .......... .......... .......... .......... .......... 10% 48,6M 21s\n",
      "  5000K .......... .......... .......... .......... .......... 10%  121M 20s\n",
      "  5050K .......... .......... .......... .......... .......... 11%  169M 20s\n",
      "  5100K .......... .......... .......... .......... .......... 11% 6,69M 20s\n",
      "  5150K .......... .......... .......... .......... .......... 11% 99,9M 20s\n",
      "  5200K .......... .......... .......... .......... .......... 11% 92,5M 19s\n",
      "  5250K .......... .......... .......... .......... .......... 11%  106M 19s\n",
      "  5300K .......... .......... .......... .......... .......... 11%  114M 19s\n",
      "  5350K .......... .......... .......... .......... .......... 11% 92,3M 19s\n",
      "  5400K .......... .......... .......... .......... .......... 11%  107M 19s\n",
      "  5450K .......... .......... .......... .......... .......... 11% 97,4M 19s\n",
      "  5500K .......... .......... .......... .......... .......... 11%  110M 18s\n",
      "  5550K .......... .......... .......... .......... .......... 12%  109M 18s\n",
      "  5600K .......... .......... .......... .......... .......... 12% 91,4M 18s\n",
      "  5650K .......... .......... .......... .......... .......... 12%  109M 18s\n",
      "  5700K .......... .......... .......... .......... .......... 12% 94,2M 18s\n",
      "  5750K .......... .......... .......... .......... .......... 12%  107M 17s\n",
      "  5800K .......... .......... .......... .......... .......... 12%  105M 17s\n",
      "  5850K .......... .......... .......... .......... .......... 12% 94,4M 17s\n",
      "  5900K .......... .......... .......... .......... .......... 12%  116M 17s\n",
      "  5950K .......... .......... .......... .......... .......... 12%  142M 17s\n",
      "  6000K .......... .......... .......... .......... .......... 13%  211M 17s\n",
      "  6050K .......... .......... .......... .......... .......... 13%  563K 17s\n",
      "  6100K .......... .......... .......... .......... .......... 13% 97,4M 17s\n",
      "  6150K .......... .......... .......... .......... .......... 13%  113M 17s\n",
      "  6200K .......... .......... .......... .......... .......... 13% 99,4M 17s\n",
      "  6250K .......... .......... .......... .......... .......... 13% 73,7M 16s\n",
      "  6300K .......... .......... .......... .......... .......... 13%  119M 16s\n",
      "  6350K .......... .......... .......... .......... .......... 13% 82,4M 16s\n",
      "  6400K .......... .......... .......... .......... .......... 13% 74,2M 16s\n",
      "  6450K .......... .......... .......... .......... .......... 14%  140M 16s\n",
      "  6500K .......... .......... .......... .......... .......... 14%  156M 16s\n",
      "  6550K .......... .......... .......... .......... .......... 14%  127M 16s\n",
      "  6600K .......... .......... .......... .......... .......... 14%  102M 15s\n",
      "  6650K .......... .......... .......... .......... .......... 14%  122M 15s\n",
      "  6700K .......... .......... .......... .......... .......... 14% 99,0M 15s\n",
      "  6750K .......... .......... .......... .......... .......... 14% 2,88M 15s\n",
      "  6800K .......... .......... .......... .......... .......... 14% 52,6M 15s\n",
      "  6850K .......... .......... .......... .......... .......... 14%  135M 15s\n",
      "  6900K .......... .......... .......... .......... .......... 15%  450K 15s\n",
      "  6950K .......... .......... .......... .......... .......... 15%  127M 15s\n",
      "  7000K .......... .......... .......... .......... .......... 15%  147M 15s\n",
      "  7050K .......... .......... .......... .......... .......... 15%  139M 15s\n",
      "  7100K .......... .......... .......... .......... .......... 15%  125M 15s\n",
      "  7150K .......... .......... .......... .......... .......... 15%  147M 15s\n",
      "  7200K .......... .......... .......... .......... .......... 15%  127M 15s\n",
      "  7250K .......... .......... .......... .......... .......... 15%  150M 15s\n",
      "  7300K .......... .......... .......... .......... .......... 15%  139M 14s\n",
      "  7350K .......... .......... .......... .......... .......... 15%  136M 14s\n",
      "  7400K .......... .......... .......... .......... .......... 16%  159M 14s\n",
      "  7450K .......... .......... .......... .......... .......... 16%  124M 14s\n",
      "  7500K .......... .......... .......... .......... .......... 16%  139M 14s\n",
      "  7550K .......... .......... .......... .......... .......... 16%  148M 14s\n",
      "  7600K .......... .......... .......... .......... .......... 16%  127M 14s\n",
      "  7650K .......... .......... .......... .......... .......... 16%  144M 14s\n",
      "  7700K .......... .......... .......... .......... .......... 16%  123M 14s\n",
      "  7750K .......... .......... .......... .......... .......... 16%  144M 13s\n",
      "  7800K .......... .......... .......... .......... .......... 16%  143M 13s\n",
      "  7850K .......... .......... .......... .......... .......... 17% 94,9M 13s\n",
      "  7900K .......... .......... .......... .......... .......... 17%  131M 13s\n",
      "  7950K .......... .......... .......... .......... .......... 17%  103M 13s\n",
      "  8000K .......... .......... .......... .......... .......... 17%  122M 13s\n",
      "  8050K .......... .......... .......... .......... .......... 17% 42,5M 13s\n",
      "  8100K .......... .......... .......... .......... .......... 17% 22,0M 13s\n",
      "  8150K .......... .......... .......... .......... .......... 17%  123M 13s\n",
      "  8200K .......... .......... .......... .......... .......... 17%  103M 13s\n",
      "  8250K .......... .......... .......... .......... .......... 17%  119M 13s\n",
      "  8300K .......... .......... .......... .......... .......... 18% 33,4M 12s\n",
      "  8350K .......... .......... .......... .......... .......... 18%  102M 12s\n",
      "  8400K .......... .......... .......... .......... .......... 18% 3,91M 12s\n",
      "  8450K .......... .......... .......... .......... .......... 18% 95,7M 12s\n",
      "  8500K .......... .......... .......... .......... .......... 18% 4,74M 12s\n",
      "  8550K .......... .......... .......... .......... .......... 18%  118M 12s\n",
      "  8600K .......... .......... .......... .......... .......... 18% 99,0M 12s\n",
      "  8650K .......... .......... .......... .......... .......... 18%  124M 12s\n",
      "  8700K .......... .......... .......... .......... .......... 18% 84,1M 12s\n",
      "  8750K .......... .......... .......... .......... .......... 19% 3,25M 12s\n",
      "  8800K .......... .......... .......... .......... .......... 19%  124M 12s\n",
      "  8850K .......... .......... .......... .......... .......... 19%  102M 12s\n",
      "  8900K .......... .......... .......... .......... .......... 19%  121M 12s\n",
      "  8950K .......... .......... .......... .......... .......... 19% 71,0M 12s\n",
      "  9000K .......... .......... .......... .......... .......... 19%  114M 11s\n",
      "  9050K .......... .......... .......... .......... .......... 19%  121M 11s\n",
      "  9100K .......... .......... .......... .......... .......... 19%  101M 11s\n",
      "  9150K .......... .......... .......... .......... .......... 19%  117M 11s\n",
      "  9200K .......... .......... .......... .......... .......... 19%  105M 11s\n",
      "  9250K .......... .......... .......... .......... .......... 20%  119M 11s\n",
      "  9300K .......... .......... .......... .......... .......... 20% 13,2M 11s\n",
      "  9350K .......... .......... .......... .......... .......... 20%  102M 11s\n",
      "  9400K .......... .......... .......... .......... .......... 20%  114M 11s\n",
      "  9450K .......... .......... .......... .......... .......... 20%  107M 11s\n",
      "  9500K .......... .......... .......... .......... .......... 20% 3,20M 11s\n",
      "  9550K .......... .......... .......... .......... .......... 20%  112M 11s\n",
      "  9600K .......... .......... .......... .......... .......... 20%  108M 11s\n",
      "  9650K .......... .......... .......... .......... .......... 20%  119M 11s\n",
      "  9700K .......... .......... .......... .......... .......... 21% 89,6M 11s\n",
      "  9750K .......... .......... .......... .......... .......... 21%  123M 10s\n",
      "  9800K .......... .......... .......... .......... .......... 21%  118M 10s\n",
      "  9850K .......... .......... .......... .......... .......... 21% 2,85M 10s\n",
      "  9900K .......... .......... .......... .......... .......... 21%  120M 10s\n",
      "  9950K .......... .......... .......... .......... .......... 21%  104M 10s\n",
      " 10000K .......... .......... .......... .......... .......... 21% 1,32M 10s\n",
      " 10050K .......... .......... .......... .......... .......... 21%  114M 10s\n",
      " 10100K .......... .......... .......... .......... .......... 21%  125M 10s\n",
      " 10150K .......... .......... .......... .......... .......... 22%  121M 10s\n",
      " 10200K .......... .......... .......... .......... .......... 22%  112M 10s\n",
      " 10250K .......... .......... .......... .......... .......... 22%  113M 10s\n",
      " 10300K .......... .......... .......... .......... .......... 22%  123M 10s\n",
      " 10350K .......... .......... .......... .......... .......... 22% 73,8M 10s\n",
      " 10400K .......... .......... .......... .......... .......... 22%  142M 10s\n",
      " 10450K .......... .......... .......... .......... .......... 22% 73,7M 10s\n",
      " 10500K .......... .......... .......... .......... .......... 22%  110M 10s\n",
      " 10550K .......... .......... .......... .......... .......... 22%  136M 10s\n",
      " 10600K .......... .......... .......... .......... .......... 22%  121M 10s\n",
      " 10650K .......... .......... .......... .......... .......... 23%  114M 10s\n",
      " 10700K .......... .......... .......... .......... .......... 23%  101M 9s\n",
      " 10750K .......... .......... .......... .......... .......... 23%  127M 9s\n",
      " 10800K .......... .......... .......... .......... .......... 23% 31,8M 9s\n",
      " 10850K .......... .......... .......... .......... .......... 23%  116M 9s\n",
      " 10900K .......... .......... .......... .......... .......... 23%  113M 9s\n",
      " 10950K .......... .......... .......... .......... .......... 23%  105M 9s\n",
      " 11000K .......... .......... .......... .......... .......... 23%  112M 9s\n",
      " 11050K .......... .......... .......... .......... .......... 23%  121M 9s\n",
      " 11100K .......... .......... .......... .......... .......... 24%  109M 9s\n",
      " 11150K .......... .......... .......... .......... .......... 24%  118M 9s\n",
      " 11200K .......... .......... .......... .......... .......... 24%  102M 9s\n",
      " 11250K .......... .......... .......... .......... .......... 24% 16,7M 9s\n",
      " 11300K .......... .......... .......... .......... .......... 24% 20,8M 9s\n",
      " 11350K .......... .......... .......... .......... .......... 24%  100M 9s\n",
      " 11400K .......... .......... .......... .......... .......... 24%  122M 9s\n",
      " 11450K .......... .......... .......... .......... .......... 24%  100M 9s\n",
      " 11500K .......... .......... .......... .......... .......... 24%  120M 9s\n",
      " 11550K .......... .......... .......... .......... .......... 25% 16,4M 9s\n",
      " 11600K .......... .......... .......... .......... .......... 25%  103M 9s\n",
      " 11650K .......... .......... .......... .......... .......... 25%  118M 9s\n",
      " 11700K .......... .......... .......... .......... .......... 25%  103M 8s\n",
      " 11750K .......... .......... .......... .......... .......... 25% 30,3M 8s\n",
      " 11800K .......... .......... .......... .......... .......... 25% 64,3M 8s\n",
      " 11850K .......... .......... .......... .......... .......... 25%  101M 8s\n",
      " 11900K .......... .......... .......... .......... .......... 25%  121M 8s\n",
      " 11950K .......... .......... .......... .......... .......... 25%  101M 8s\n",
      " 12000K .......... .......... .......... .......... .......... 26% 73,9M 8s\n",
      " 12050K .......... .......... .......... .......... .......... 26% 7,73M 8s\n",
      " 12100K .......... .......... .......... .......... .......... 26% 99,3M 8s\n",
      " 12150K .......... .......... .......... .......... .......... 26%  122M 8s\n",
      " 12200K .......... .......... .......... .......... .......... 26%  103M 8s\n",
      " 12250K .......... .......... .......... .......... .......... 26%  104M 8s\n",
      " 12300K .......... .......... .......... .......... .......... 26%  121M 8s\n",
      " 12350K .......... .......... .......... .......... .......... 26%  104M 8s\n",
      " 12400K .......... .......... .......... .......... .......... 26% 87,7M 8s\n",
      " 12450K .......... .......... .......... .......... .......... 26%  127M 8s\n",
      " 12500K .......... .......... .......... .......... .......... 27%  128M 8s\n",
      " 12550K .......... .......... .......... .......... .......... 27%  117M 8s\n",
      " 12600K .......... .......... .......... .......... .......... 27%  108M 8s\n",
      " 12650K .......... .......... .......... .......... .......... 27%  115M 8s\n",
      " 12700K .......... .......... .......... .......... .......... 27%  102M 8s\n",
      " 12750K .......... .......... .......... .......... .......... 27%  122M 8s\n",
      " 12800K .......... .......... .......... .......... .......... 27%  117M 8s\n",
      " 12850K .......... .......... .......... .......... .......... 27%  102M 8s\n",
      " 12900K .......... .......... .......... .......... .......... 27%  121M 7s\n",
      " 12950K .......... .......... .......... .......... .......... 28%  102M 7s\n",
      " 13000K .......... .......... .......... .......... .......... 28%  119M 7s\n",
      " 13050K .......... .......... .......... .......... .......... 28%  118M 7s\n",
      " 13100K .......... .......... .......... .......... .......... 28% 2,86M 7s\n",
      " 13150K .......... .......... .......... .......... .......... 28%  124M 7s\n",
      " 13200K .......... .......... .......... .......... .......... 28%  101M 7s\n",
      " 13250K .......... .......... .......... .......... .......... 28%  116M 7s\n",
      " 13300K .......... .......... .......... .......... .......... 28%  121M 7s\n",
      " 13350K .......... .......... .......... .......... .......... 28%  105M 7s\n",
      " 13400K .......... .......... .......... .......... .......... 29%  110M 7s\n",
      " 13450K .......... .......... .......... .......... .......... 29%  110M 7s\n",
      " 13500K .......... .......... .......... .......... .......... 29%  116M 7s\n",
      " 13550K .......... .......... .......... .......... .......... 29%  120M 7s\n",
      " 13600K .......... .......... .......... .......... .......... 29% 74,9M 7s\n",
      " 13650K .......... .......... .......... .......... .......... 29%  119M 7s\n",
      " 13700K .......... .......... .......... .......... .......... 29%  104M 7s\n",
      " 13750K .......... .......... .......... .......... .......... 29%  120M 7s\n",
      " 13800K .......... .......... .......... .......... .......... 29%  119K 8s\n",
      " 13850K .......... .......... .......... .......... .......... 30%  122M 8s\n",
      " 13900K .......... .......... .......... .......... .......... 30%  147M 8s\n",
      " 13950K .......... .......... .......... .......... .......... 30% 95,0M 8s\n",
      " 14000K .......... .......... .......... .......... .......... 30%  130M 8s\n",
      " 14050K .......... .......... .......... .......... .......... 30%  128M 8s\n",
      " 14100K .......... .......... .......... .......... .......... 30% 98,0M 8s\n",
      " 14150K .......... .......... .......... .......... .......... 30%  128M 8s\n",
      " 14200K .......... .......... .......... .......... .......... 30% 95,8M 8s\n",
      " 14250K .......... .......... .......... .......... .......... 30%  122M 7s\n",
      " 14300K .......... .......... .......... .......... .......... 30%  121M 7s\n",
      " 14350K .......... .......... .......... .......... .......... 31%  104M 7s\n",
      " 14400K .......... .......... .......... .......... .......... 31%  117M 7s\n",
      " 14450K .......... .......... .......... .......... .......... 31% 80,9M 7s\n",
      " 14500K .......... .......... .......... .......... .......... 31%  138M 7s\n",
      " 14550K .......... .......... .......... .......... .......... 31%  126M 7s\n",
      " 14600K .......... .......... .......... .......... .......... 31%  102M 7s\n",
      " 14650K .......... .......... .......... .......... .......... 31%  120M 7s\n",
      " 14700K .......... .......... .......... .......... .......... 31% 19,5M 7s\n",
      " 14750K .......... .......... .......... .......... .......... 31%  121M 7s\n",
      " 14800K .......... .......... .......... .......... .......... 32%  116M 7s\n",
      " 14850K .......... .......... .......... .......... .......... 32%  102M 7s\n",
      " 14900K .......... .......... .......... .......... .......... 32%  121M 7s\n",
      " 14950K .......... .......... .......... .......... .......... 32% 50,5M 7s\n",
      " 15000K .......... .......... .......... .......... .......... 32%  119M 7s\n",
      " 15050K .......... .......... .......... .......... .......... 32%  120M 7s\n",
      " 15100K .......... .......... .......... .......... .......... 32%  102M 7s\n",
      " 15150K .......... .......... .......... .......... .......... 32%  119M 7s\n",
      " 15200K .......... .......... .......... .......... .......... 32% 97,4M 7s\n",
      " 15250K .......... .......... .......... .......... .......... 33%  118M 7s\n",
      " 15300K .......... .......... .......... .......... .......... 33%  127M 7s\n",
      " 15350K .......... .......... .......... .......... .......... 33%  100M 7s\n",
      " 15400K .......... .......... .......... .......... .......... 33%  122M 7s\n",
      " 15450K .......... .......... .......... .......... .......... 33%  100M 7s\n",
      " 15500K .......... .......... .......... .......... .......... 33%  121M 7s\n",
      " 15550K .......... .......... .......... .......... .......... 33%  118M 7s\n",
      " 15600K .......... .......... .......... .......... .......... 33%  103M 7s\n",
      " 15650K .......... .......... .......... .......... .......... 33%  120M 7s\n",
      " 15700K .......... .......... .......... .......... .......... 34%  103M 7s\n",
      " 15750K .......... .......... .......... .......... .......... 34%  115M 6s\n",
      " 15800K .......... .......... .......... .......... .......... 34%  120M 6s\n",
      " 15850K .......... .......... .......... .......... .......... 34%  101M 6s\n",
      " 15900K .......... .......... .......... .......... .......... 34%  123M 6s\n",
      " 15950K .......... .......... .......... .......... .......... 34%  102M 6s\n",
      " 16000K .......... .......... .......... .......... .......... 34%  118M 6s\n",
      " 16050K .......... .......... .......... .......... .......... 34%  117M 6s\n",
      " 16100K .......... .......... .......... .......... .......... 34%  103M 6s\n",
      " 16150K .......... .......... .......... .......... .......... 34%  119M 6s\n",
      " 16200K .......... .......... .......... .......... .......... 35%  102M 6s\n",
      " 16250K .......... .......... .......... .......... .......... 35%  121M 6s\n",
      " 16300K .......... .......... .......... .......... .......... 35%  118M 6s\n",
      " 16350K .......... .......... .......... .......... .......... 35%  102M 6s\n",
      " 16400K .......... .......... .......... .......... .......... 35%  117M 6s\n",
      " 16450K .......... .......... .......... .......... .......... 35%  103M 6s\n",
      " 16500K .......... .......... .......... .......... .......... 35%  120M 6s\n",
      " 16550K .......... .......... .......... .......... .......... 35%  118M 6s\n",
      " 16600K .......... .......... .......... .......... .......... 35%  101M 6s\n",
      " 16650K .......... .......... .......... .......... .......... 36%  121M 6s\n",
      " 16700K .......... .......... .......... .......... .......... 36%  100M 6s\n",
      " 16750K .......... .......... .......... .......... .......... 36%  118M 6s\n",
      " 16800K .......... .......... .......... .......... .......... 36%  122M 6s\n",
      " 16850K .......... .......... .......... .......... .......... 36% 1,64M 6s\n",
      " 16900K .......... .......... .......... .......... .......... 36% 46,0M 6s\n",
      " 16950K .......... .......... .......... .......... .......... 36% 68,8M 6s\n",
      " 17000K .......... .......... .......... .......... .......... 36%  122M 6s\n",
      " 17050K .......... .......... .......... .......... .......... 36%  119M 6s\n",
      " 17100K .......... .......... .......... .......... .......... 37% 98,2M 6s\n",
      " 17150K .......... .......... .......... .......... .......... 37%  122M 6s\n",
      " 17200K .......... .......... .......... .......... .......... 37%  102M 6s\n",
      " 17250K .......... .......... .......... .......... .......... 37%  444K 6s\n",
      " 17300K .......... .......... .......... .......... .......... 37% 35,3M 6s\n",
      " 17350K .......... .......... .......... .......... .......... 37% 99,2M 6s\n",
      " 17400K .......... .......... .......... .......... .......... 37%  140M 6s\n",
      " 17450K .......... .......... .......... .......... .......... 37% 1,59M 6s\n",
      " 17500K .......... .......... .......... .......... .......... 37%  129M 6s\n",
      " 17550K .......... .......... .......... .......... .......... 38%  115M 6s\n",
      " 17600K .......... .......... .......... .......... .......... 38%  105M 6s\n",
      " 17650K .......... .......... .......... .......... .......... 38%  118M 6s\n",
      " 17700K .......... .......... .......... .......... .......... 38%  103M 6s\n",
      " 17750K .......... .......... .......... .......... .......... 38%  121M 6s\n",
      " 17800K .......... .......... .......... .......... .......... 38%  115M 6s\n",
      " 17850K .......... .......... .......... .......... .......... 38%  104M 6s\n",
      " 17900K .......... .......... .......... .......... .......... 38%  119M 6s\n",
      " 17950K .......... .......... .......... .......... .......... 38%  101M 6s\n",
      " 18000K .......... .......... .......... .......... .......... 38%  119M 6s\n",
      " 18050K .......... .......... .......... .......... .......... 39%  107M 6s\n",
      " 18100K .......... .......... .......... .......... .......... 39%  111M 6s\n",
      " 18150K .......... .......... .......... .......... .......... 39% 39,5M 5s\n",
      " 18200K .......... .......... .......... .......... .......... 39% 81,4M 5s\n",
      " 18250K .......... .......... .......... .......... .......... 39%  116M 5s\n",
      " 18300K .......... .......... .......... .......... .......... 39%  117M 5s\n",
      " 18350K .......... .......... .......... .......... .......... 39%  105M 5s\n",
      " 18400K .......... .......... .......... .......... .......... 39%  119M 5s\n",
      " 18450K .......... .......... .......... .......... .......... 39%  102M 5s\n",
      " 18500K .......... .......... .......... .......... .......... 40%  121M 5s\n",
      " 18550K .......... .......... .......... .......... .......... 40% 37,1M 5s\n",
      " 18600K .......... .......... .......... .......... .......... 40%  106M 5s\n",
      " 18650K .......... .......... .......... .......... .......... 40%  119M 5s\n",
      " 18700K .......... .......... .......... .......... .......... 40%  103M 5s\n",
      " 18750K .......... .......... .......... .......... .......... 40%  115M 5s\n",
      " 18800K .......... .......... .......... .......... .......... 40%  119M 5s\n",
      " 18850K .......... .......... .......... .......... .......... 40%  102M 5s\n",
      " 18900K .......... .......... .......... .......... .......... 40% 69,9M 5s\n",
      " 18950K .......... .......... .......... .......... .......... 41%  102M 5s\n",
      " 19000K .......... .......... .......... .......... .......... 41%  114M 5s\n",
      " 19050K .......... .......... .......... .......... .......... 41%  123M 5s\n",
      " 19100K .......... .......... .......... .......... .......... 41%  101M 5s\n",
      " 19150K .......... .......... .......... .......... .......... 41%  119M 5s\n",
      " 19200K .......... .......... .......... .......... .......... 41%  104M 5s\n",
      " 19250K .......... .......... .......... .......... .......... 41%  117M 5s\n",
      " 19300K .......... .......... .......... .......... .......... 41%  121M 5s\n",
      " 19350K .......... .......... .......... .......... .......... 41% 98,8M 5s\n",
      " 19400K .......... .......... .......... .......... .......... 42%  121M 5s\n",
      " 19450K .......... .......... .......... .......... .......... 42% 95,9M 5s\n",
      " 19500K .......... .......... .......... .......... .......... 42%  128M 5s\n",
      " 19550K .......... .......... .......... .......... .......... 42%  118M 5s\n",
      " 19600K .......... .......... .......... .......... .......... 42%  100M 5s\n",
      " 19650K .......... .......... .......... .......... .......... 42%  124M 5s\n",
      " 19700K .......... .......... .......... .......... .......... 42%  101M 5s\n",
      " 19750K .......... .......... .......... .......... .......... 42%  122M 5s\n",
      " 19800K .......... .......... .......... .......... .......... 42%  604K 5s\n",
      " 19850K .......... .......... .......... .......... .......... 42%  114M 5s\n",
      " 19900K .......... .......... .......... .......... .......... 43%  126M 5s\n",
      " 19950K .......... .......... .......... .......... .......... 43%  125M 5s\n",
      " 20000K .......... .......... .......... .......... .......... 43%  143M 5s\n",
      " 20050K .......... .......... .......... .......... .......... 43%  145M 5s\n",
      " 20100K .......... .......... .......... .......... .......... 43%  125M 5s\n",
      " 20150K .......... .......... .......... .......... .......... 43%  135M 5s\n",
      " 20200K .......... .......... .......... .......... .......... 43%  100M 5s\n",
      " 20250K .......... .......... .......... .......... .......... 43%  116M 5s\n",
      " 20300K .......... .......... .......... .......... .......... 43%  123M 5s\n",
      " 20350K .......... .......... .......... .......... .......... 44% 47,2M 5s\n",
      " 20400K .......... .......... .......... .......... .......... 44%  131M 5s\n",
      " 20450K .......... .......... .......... .......... .......... 44%  117M 5s\n",
      " 20500K .......... .......... .......... .......... .......... 44%  138M 5s\n",
      " 20550K .......... .......... .......... .......... .......... 44%  112M 5s\n",
      " 20600K .......... .......... .......... .......... .......... 44% 74,0M 5s\n",
      " 20650K .......... .......... .......... .......... .......... 44%  149M 5s\n",
      " 20700K .......... .......... .......... .......... .......... 44% 99,9M 5s\n",
      " 20750K .......... .......... .......... .......... .......... 44%  121M 4s\n",
      " 20800K .......... .......... .......... .......... .......... 45%  117M 4s\n",
      " 20850K .......... .......... .......... .......... .......... 45% 70,7M 4s\n",
      " 20900K .......... .......... .......... .......... .......... 45%  119M 4s\n",
      " 20950K .......... .......... .......... .......... .......... 45% 96,5M 4s\n",
      " 21000K .......... .......... .......... .......... .......... 45%  124M 4s\n",
      " 21050K .......... .......... .......... .......... .......... 45%  121M 4s\n",
      " 21100K .......... .......... .......... .......... .......... 45%  831K 4s\n",
      " 21150K .......... .......... .......... .......... .......... 45%  154M 4s\n",
      " 21200K .......... .......... .......... .......... .......... 45%  135M 4s\n",
      " 21250K .......... .......... .......... .......... .......... 45% 66,9M 4s\n",
      " 21300K .......... .......... .......... .......... .......... 46% 98,5M 4s\n",
      " 21350K .......... .......... .......... .......... .......... 46%  106M 4s\n",
      " 21400K .......... .......... .......... .......... .......... 46%  120M 4s\n",
      " 21450K .......... .......... .......... .......... .......... 46% 92,7M 4s\n",
      " 21500K .......... .......... .......... .......... .......... 46%  136M 4s\n",
      " 21550K .......... .......... .......... .......... .......... 46%  117M 4s\n",
      " 21600K .......... .......... .......... .......... .......... 46%  103M 4s\n",
      " 21650K .......... .......... .......... .......... .......... 46%  122M 4s\n",
      " 21700K .......... .......... .......... .......... .......... 46%  101M 4s\n",
      " 21750K .......... .......... .......... .......... .......... 47%  117M 4s\n",
      " 21800K .......... .......... .......... .......... .......... 47%  118M 4s\n",
      " 21850K .......... .......... .......... .......... .......... 47% 97,4M 4s\n",
      " 21900K .......... .......... .......... .......... .......... 47%  130M 4s\n",
      " 21950K .......... .......... .......... .......... .......... 47%  102M 4s\n",
      " 22000K .......... .......... .......... .......... .......... 47%  117M 4s\n",
      " 22050K .......... .......... .......... .......... .......... 47%  120M 4s\n",
      " 22100K .......... .......... .......... .......... .......... 47%  102M 4s\n",
      " 22150K .......... .......... .......... .......... .......... 47%  116M 4s\n",
      " 22200K .......... .......... .......... .......... .......... 48%  102M 4s\n",
      " 22250K .......... .......... .......... .......... .......... 48%  125M 4s\n",
      " 22300K .......... .......... .......... .......... .......... 48%  120M 4s\n",
      " 22350K .......... .......... .......... .......... .......... 48% 2,05M 4s\n",
      " 22400K .......... .......... .......... .......... .......... 48%  127M 4s\n",
      " 22450K .......... .......... .......... .......... .......... 48%  102M 4s\n",
      " 22500K .......... .......... .......... .......... .......... 48%  121M 4s\n",
      " 22550K .......... .......... .......... .......... .......... 48%  115M 4s\n",
      " 22600K .......... .......... .......... .......... .......... 48% 77,4M 4s\n",
      " 22650K .......... .......... .......... .......... .......... 49%  134M 4s\n",
      " 22700K .......... .......... .......... .......... .......... 49%  128M 4s\n",
      " 22750K .......... .......... .......... .......... .......... 49%  122M 4s\n",
      " 22800K .......... .......... .......... .......... .......... 49%  120M 4s\n",
      " 22850K .......... .......... .......... .......... .......... 49%  104M 4s\n",
      " 22900K .......... .......... .......... .......... .......... 49%  118M 4s\n",
      " 22950K .......... .......... .......... .......... .......... 49% 99,6M 4s\n",
      " 23000K .......... .......... .......... .......... .......... 49%  117M 4s\n",
      " 23050K .......... .......... .......... .......... .......... 49%  120M 4s\n",
      " 23100K .......... .......... .......... .......... .......... 49%  102M 4s\n",
      " 23150K .......... .......... .......... .......... .......... 50%  120M 4s\n",
      " 23200K .......... .......... .......... .......... .......... 50%  102M 4s\n",
      " 23250K .......... .......... .......... .......... .......... 50%  117M 4s\n",
      " 23300K .......... .......... .......... .......... .......... 50%  123M 4s\n",
      " 23350K .......... .......... .......... .......... .......... 50% 99,4M 4s\n",
      " 23400K .......... .......... .......... .......... .......... 50%  119M 4s\n",
      " 23450K .......... .......... .......... .......... .......... 50%  102M 4s\n",
      " 23500K .......... .......... .......... .......... .......... 50%  122M 4s\n",
      " 23550K .......... .......... .......... .......... .......... 50%  120M 4s\n",
      " 23600K .......... .......... .......... .......... .......... 51% 98,9M 4s\n",
      " 23650K .......... .......... .......... .......... .......... 51%  120M 4s\n",
      " 23700K .......... .......... .......... .......... .......... 51%  103M 4s\n",
      " 23750K .......... .......... .......... .......... .......... 51%  118M 4s\n",
      " 23800K .......... .......... .......... .......... .......... 51%  119M 4s\n",
      " 23850K .......... .......... .......... .......... .......... 51%  102M 4s\n",
      " 23900K .......... .......... .......... .......... .......... 51%  119M 4s\n",
      " 23950K .......... .......... .......... .......... .......... 51%  103M 4s\n",
      " 24000K .......... .......... .......... .......... .......... 51%  116M 3s\n",
      " 24050K .......... .......... .......... .......... .......... 52%  122M 3s\n",
      " 24100K .......... .......... .......... .......... .......... 52%  100M 3s\n",
      " 24150K .......... .......... .......... .......... .......... 52% 5,43M 3s\n",
      " 24200K .......... .......... .......... .......... .......... 52% 44,4M 3s\n",
      " 24250K .......... .......... .......... .......... .......... 52% 41,9M 3s\n",
      " 24300K .......... .......... .......... .......... .......... 52% 88,3M 3s\n",
      " 24350K .......... .......... .......... .......... .......... 52%  129M 3s\n",
      " 24400K .......... .......... .......... .......... .......... 52%  149M 3s\n",
      " 24450K .......... .......... .......... .......... .......... 52%  148M 3s\n",
      " 24500K .......... .......... .......... .......... .......... 53%  226M 3s\n",
      " 24550K .......... .......... .......... .......... .......... 53% 90,8M 3s\n",
      " 24600K .......... .......... .......... .......... .......... 53%  101M 3s\n",
      " 24650K .......... .......... .......... .......... .......... 53%  118M 3s\n",
      " 24700K .......... .......... .......... .......... .......... 53% 94,0M 3s\n",
      " 24750K .......... .......... .......... .......... .......... 53%  134M 3s\n",
      " 24800K .......... .......... .......... .......... .......... 53% 89,5M 3s\n",
      " 24850K .......... .......... .......... .......... .......... 53% 86,9M 3s\n",
      " 24900K .......... .......... .......... .......... .......... 53%  121M 3s\n",
      " 24950K .......... .......... .......... .......... .......... 53% 32,1M 3s\n",
      " 25000K .......... .......... .......... .......... .......... 54%  119M 3s\n",
      " 25050K .......... .......... .......... .......... .......... 54%  119M 3s\n",
      " 25100K .......... .......... .......... .......... .......... 54%  101M 3s\n",
      " 25150K .......... .......... .......... .......... .......... 54%  122M 3s\n",
      " 25200K .......... .......... .......... .......... .......... 54% 55,2M 3s\n",
      " 25250K .......... .......... .......... .......... .......... 54%  119M 3s\n",
      " 25300K .......... .......... .......... .......... .......... 54%  121M 3s\n",
      " 25350K .......... .......... .......... .......... .......... 54%  101M 3s\n",
      " 25400K .......... .......... .......... .......... .......... 54%  118M 3s\n",
      " 25450K .......... .......... .......... .......... .......... 55% 2,26M 3s\n",
      " 25500K .......... .......... .......... .......... .......... 55%  100M 3s\n",
      " 25550K .......... .......... .......... .......... .......... 55%  321K 3s\n",
      " 25600K .......... .......... .......... .......... .......... 55% 76,0M 3s\n",
      " 25650K .......... .......... .......... .......... .......... 55%  114M 3s\n",
      " 25700K .......... .......... .......... .......... .......... 55% 95,4M 3s\n",
      " 25750K .......... .......... .......... .......... .......... 55%  108M 3s\n",
      " 25800K .......... .......... .......... .......... .......... 55%  109M 3s\n",
      " 25850K .......... .......... .......... .......... .......... 55% 94,1M 3s\n",
      " 25900K .......... .......... .......... .......... .......... 56%  101M 3s\n",
      " 25950K .......... .......... .......... .......... .......... 56% 96,4M 3s\n",
      " 26000K .......... .......... .......... .......... .......... 56%  114M 3s\n",
      " 26050K .......... .......... .......... .......... .......... 56%  119M 3s\n",
      " 26100K .......... .......... .......... .......... .......... 56%  104M 3s\n",
      " 26150K .......... .......... .......... .......... .......... 56%  111M 3s\n",
      " 26200K .......... .......... .......... .......... .......... 56% 92,4M 3s\n",
      " 26250K .......... .......... .......... .......... .......... 56%  108M 3s\n",
      " 26300K .......... .......... .......... .......... .......... 56%  105M 3s\n",
      " 26350K .......... .......... .......... .......... .......... 57% 98,0M 3s\n",
      " 26400K .......... .......... .......... .......... .......... 57%  121M 3s\n",
      " 26450K .......... .......... .......... .......... .......... 57%  159M 3s\n",
      " 26500K .......... .......... .......... .......... .......... 57%  647M 3s\n",
      " 26550K .......... .......... .......... .......... .......... 57%  162M 3s\n",
      " 26600K .......... .......... .......... .......... .......... 57%  102M 3s\n",
      " 26650K .......... .......... .......... .......... .......... 57%  118M 3s\n",
      " 26700K .......... .......... .......... .......... .......... 57% 30,5M 3s\n",
      " 26750K .......... .......... .......... .......... .......... 57% 68,1M 3s\n",
      " 26800K .......... .......... .......... .......... .......... 57% 74,3M 3s\n",
      " 26850K .......... .......... .......... .......... .......... 58% 58,7M 3s\n",
      " 26900K .......... .......... .......... .......... .......... 58% 65,4M 3s\n",
      " 26950K .......... .......... .......... .......... .......... 58% 56,1M 3s\n",
      " 27000K .......... .......... .......... .......... .......... 58%  110M 3s\n",
      " 27050K .......... .......... .......... .......... .......... 58%  109M 3s\n",
      " 27100K .......... .......... .......... .......... .......... 58% 96,2M 3s\n",
      " 27150K .......... .......... .......... .......... .......... 58%  116M 3s\n",
      " 27200K .......... .......... .......... .......... .......... 58%  116M 3s\n",
      " 27250K .......... .......... .......... .......... .......... 58%  137M 3s\n",
      " 27300K .......... .......... .......... .......... .......... 59%  141M 3s\n",
      " 27350K .......... .......... .......... .......... .......... 59%  117M 3s\n",
      " 27400K .......... .......... .......... .......... .......... 59%  134M 3s\n",
      " 27450K .......... .......... .......... .......... .......... 59%  126M 3s\n",
      " 27500K .......... .......... .......... .......... .......... 59%  164M 3s\n",
      " 27550K .......... .......... .......... .......... .......... 59%  161M 3s\n",
      " 27600K .......... .......... .......... .......... .......... 59%  140M 3s\n",
      " 27650K .......... .......... .......... .......... .......... 59%  164M 3s\n",
      " 27700K .......... .......... .......... .......... .......... 59%  140M 3s\n",
      " 27750K .......... .......... .......... .......... .......... 60%  161M 3s\n",
      " 27800K .......... .......... .......... .......... .......... 60%  162M 3s\n",
      " 27850K .......... .......... .......... .......... .......... 60%  173M 3s\n",
      " 27900K .......... .......... .......... .......... .......... 60%  205M 3s\n",
      " 27950K .......... .......... .......... .......... .......... 60%  181M 3s\n",
      " 28000K .......... .......... .......... .......... .......... 60%  129M 3s\n",
      " 28050K .......... .......... .......... .......... .......... 60%  119M 3s\n",
      " 28100K .......... .......... .......... .......... .......... 60%  103M 3s\n",
      " 28150K .......... .......... .......... .......... .......... 60%  119M 3s\n",
      " 28200K .......... .......... .......... .......... .......... 61%  104M 3s\n",
      " 28250K .......... .......... .......... .......... .......... 61%  120M 3s\n",
      " 28300K .......... .......... .......... .......... .......... 61%  121M 3s\n",
      " 28350K .......... .......... .......... .......... .......... 61%  100M 3s\n",
      " 28400K .......... .......... .......... .......... .......... 61%  119M 3s\n",
      " 28450K .......... .......... .......... .......... .......... 61%  104M 3s\n",
      " 28500K .......... .......... .......... .......... .......... 61%  119M 2s\n",
      " 28550K .......... .......... .......... .......... .......... 61%  119M 2s\n",
      " 28600K .......... .......... .......... .......... .......... 61% 1,58M 2s\n",
      " 28650K .......... .......... .......... .......... .......... 61% 34,7M 2s\n",
      " 28700K .......... .......... .......... .......... .......... 62% 93,9M 2s\n",
      " 28750K .......... .......... .......... .......... .......... 62% 73,3M 2s\n",
      " 28800K .......... .......... .......... .......... .......... 62%  110M 2s\n",
      " 28850K .......... .......... .......... .......... .......... 62% 91,6M 2s\n",
      " 28900K .......... .......... .......... .......... .......... 62%  120M 2s\n",
      " 28950K .......... .......... .......... .......... .......... 62% 99,4M 2s\n",
      " 29000K .......... .......... .......... .......... .......... 62%  180M 2s\n",
      " 29050K .......... .......... .......... .......... .......... 62% 88,7M 2s\n",
      " 29100K .......... .......... .......... .......... .......... 62% 93,6M 2s\n",
      " 29150K .......... .......... .......... .......... .......... 63%  108M 2s\n",
      " 29200K .......... .......... .......... .......... .......... 63% 93,7M 2s\n",
      " 29250K .......... .......... .......... .......... .......... 63%  106M 2s\n",
      " 29300K .......... .......... .......... .......... .......... 63%  111M 2s\n",
      " 29350K .......... .......... .......... .......... .......... 63% 97,0M 2s\n",
      " 29400K .......... .......... .......... .......... .......... 63% 1,03M 2s\n",
      " 29450K .......... .......... .......... .......... .......... 63% 36,5M 2s\n",
      " 29500K .......... .......... .......... .......... .......... 63% 61,8M 2s\n",
      " 29550K .......... .......... .......... .......... .......... 63% 61,1M 2s\n",
      " 29600K .......... .......... .......... .......... .......... 64% 60,9M 2s\n",
      " 29650K .......... .......... .......... .......... .......... 64% 68,6M 2s\n",
      " 29700K .......... .......... .......... .......... .......... 64% 55,8M 2s\n",
      " 29750K .......... .......... .......... .......... .......... 64% 85,0M 2s\n",
      " 29800K .......... .......... .......... .......... .......... 64% 66,2M 2s\n",
      " 29850K .......... .......... .......... .......... .......... 64% 59,0M 2s\n",
      " 29900K .......... .......... .......... .......... .......... 64%  106M 2s\n",
      " 29950K .......... .......... .......... .......... .......... 64%  150M 2s\n",
      " 30000K .......... .......... .......... .......... .......... 64%  173M 2s\n",
      " 30050K .......... .......... .......... .......... .......... 65%  177M 2s\n",
      " 30100K .......... .......... .......... .......... .......... 65%  145M 2s\n",
      " 30150K .......... .......... .......... .......... .......... 65%  181M 2s\n",
      " 30200K .......... .......... .......... .......... .......... 65%  149M 2s\n",
      " 30250K .......... .......... .......... .......... .......... 65%  187M 2s\n",
      " 30300K .......... .......... .......... .......... .......... 65%  148M 2s\n",
      " 30350K .......... .......... .......... .......... .......... 65% 76,6M 2s\n",
      " 30400K .......... .......... .......... .......... .......... 65% 91,3M 2s\n",
      " 30450K .......... .......... .......... .......... .......... 65%  105M 2s\n",
      " 30500K .......... .......... .......... .......... .......... 65%  114M 2s\n",
      " 30550K .......... .......... .......... .......... .......... 66%  107M 2s\n",
      " 30600K .......... .......... .......... .......... .......... 66%  108M 2s\n",
      " 30650K .......... .......... .......... .......... .......... 66%  143M 2s\n",
      " 30700K .......... .......... .......... .......... .......... 66%  132M 2s\n",
      " 30750K .......... .......... .......... .......... .......... 66%  144M 2s\n",
      " 30800K .......... .......... .......... .......... .......... 66%  110M 2s\n",
      " 30850K .......... .......... .......... .......... .......... 66%  160M 2s\n",
      " 30900K .......... .......... .......... .......... .......... 66%  189M 2s\n",
      " 30950K .......... .......... .......... .......... .......... 66%  156M 2s\n",
      " 31000K .......... .......... .......... .......... .......... 67%  177M 2s\n",
      " 31050K .......... .......... .......... .......... .......... 67%  168M 2s\n",
      " 31100K .......... .......... .......... .......... .......... 67%  102M 2s\n",
      " 31150K .......... .......... .......... .......... .......... 67%  119M 2s\n",
      " 31200K .......... .......... .......... .......... .......... 67%  105M 2s\n",
      " 31250K .......... .......... .......... .......... .......... 67%  119M 2s\n",
      " 31300K .......... .......... .......... .......... .......... 67%  113M 2s\n",
      " 31350K .......... .......... .......... .......... .......... 67% 92,7M 2s\n",
      " 31400K .......... .......... .......... .......... .......... 67%  119M 2s\n",
      " 31450K .......... .......... .......... .......... .......... 68% 80,8M 2s\n",
      " 31500K .......... .......... .......... .......... .......... 68%  121M 2s\n",
      " 31550K .......... .......... .......... .......... .......... 68%  119M 2s\n",
      " 31600K .......... .......... .......... .......... .......... 68% 93,3M 2s\n",
      " 31650K .......... .......... .......... .......... .......... 68%  119M 2s\n",
      " 31700K .......... .......... .......... .......... .......... 68%  103M 2s\n",
      " 31750K .......... .......... .......... .......... .......... 68%  120M 2s\n",
      " 31800K .......... .......... .......... .......... .......... 68%  117M 2s\n",
      " 31850K .......... .......... .......... .......... .......... 68%  103M 2s\n",
      " 31900K .......... .......... .......... .......... .......... 68%  119M 2s\n",
      " 31950K .......... .......... .......... .......... .......... 69% 25,0M 2s\n",
      " 32000K .......... .......... .......... .......... .......... 69% 72,4M 2s\n",
      " 32050K .......... .......... .......... .......... .......... 69%  119M 2s\n",
      " 32100K .......... .......... .......... .......... .......... 69%  102M 2s\n",
      " 32150K .......... .......... .......... .......... .......... 69%  119M 2s\n",
      " 32200K .......... .......... .......... .......... .......... 69%  101M 2s\n",
      " 32250K .......... .......... .......... .......... .......... 69%  119M 2s\n",
      " 32300K .......... .......... .......... .......... .......... 69%  278K 2s\n",
      " 32350K .......... .......... .......... .......... .......... 69% 62,4M 2s\n",
      " 32400K .......... .......... .......... .......... .......... 70%  184M 2s\n",
      " 32450K .......... .......... .......... .......... .......... 70% 59,9M 2s\n",
      " 32500K .......... .......... .......... .......... .......... 70% 64,8M 2s\n",
      " 32550K .......... .......... .......... .......... .......... 70% 66,6M 2s\n",
      " 32600K .......... .......... .......... .......... .......... 70% 58,4M 2s\n",
      " 32650K .......... .......... .......... .......... .......... 70% 67,5M 2s\n",
      " 32700K .......... .......... .......... .......... .......... 70% 56,0M 2s\n",
      " 32750K .......... .......... .......... .......... .......... 70%  130M 2s\n",
      " 32800K .......... .......... .......... .......... .......... 70%  592M 2s\n",
      " 32850K .......... .......... .......... .......... .......... 71%  417M 2s\n",
      " 32900K .......... .......... .......... .......... .......... 71%  579M 2s\n",
      " 32950K .......... .......... .......... .......... .......... 71%  533M 2s\n",
      " 33000K .......... .......... .......... .......... .......... 71%  626M 2s\n",
      " 33050K .......... .......... .......... .......... .......... 71%  583M 2s\n",
      " 33100K .......... .......... .......... .......... .......... 71%  515M 2s\n",
      " 33150K .......... .......... .......... .......... .......... 71%  128M 2s\n",
      " 33200K .......... .......... .......... .......... .......... 71%  106M 2s\n",
      " 33250K .......... .......... .......... .......... .......... 71%  102M 2s\n",
      " 33300K .......... .......... .......... .......... .......... 72% 72,3M 2s\n",
      " 33350K .......... .......... .......... .......... .......... 72%  105M 2s\n",
      " 33400K .......... .......... .......... .......... .......... 72%  122M 2s\n",
      " 33450K .......... .......... .......... .......... .......... 72%  101M 2s\n",
      " 33500K .......... .......... .......... .......... .......... 72%  121M 2s\n",
      " 33550K .......... .......... .......... .......... .......... 72%  117M 2s\n",
      " 33600K .......... .......... .......... .......... .......... 72%  101M 2s\n",
      " 33650K .......... .......... .......... .......... .......... 72%  129M 2s\n",
      " 33700K .......... .......... .......... .......... .......... 72%  113M 2s\n",
      " 33750K .......... .......... .......... .......... .......... 72%  125M 2s\n",
      " 33800K .......... .......... .......... .......... .......... 73%  144M 2s\n",
      " 33850K .......... .......... .......... .......... .......... 73%  125M 2s\n",
      " 33900K .......... .......... .......... .......... .......... 73%  136M 2s\n",
      " 33950K .......... .......... .......... .......... .......... 73%  103M 2s\n",
      " 34000K .......... .......... .......... .......... .......... 73%  116M 2s\n",
      " 34050K .......... .......... .......... .......... .......... 73%  118M 2s\n",
      " 34100K .......... .......... .......... .......... .......... 73%  102M 2s\n",
      " 34150K .......... .......... .......... .......... .......... 73%  123M 2s\n",
      " 34200K .......... .......... .......... .......... .......... 73%  100M 2s\n",
      " 34250K .......... .......... .......... .......... .......... 74%  119M 2s\n",
      " 34300K .......... .......... .......... .......... .......... 74%  121M 2s\n",
      " 34350K .......... .......... .......... .......... .......... 74%  101M 1s\n",
      " 34400K .......... .......... .......... .......... .......... 74%  121M 1s\n",
      " 34450K .......... .......... .......... .......... .......... 74%  101M 1s\n",
      " 34500K .......... .......... .......... .......... .......... 74%  122M 1s\n",
      " 34550K .......... .......... .......... .......... .......... 74%  121M 1s\n",
      " 34600K .......... .......... .......... .......... .......... 74% 98,7M 1s\n",
      " 34650K .......... .......... .......... .......... .......... 74%  122M 1s\n",
      " 34700K .......... .......... .......... .......... .......... 75%  102M 1s\n",
      " 34750K .......... .......... .......... .......... .......... 75%  119M 1s\n",
      " 34800K .......... .......... .......... .......... .......... 75% 82,4M 1s\n",
      " 34850K .......... .......... .......... .......... .......... 75%  104M 1s\n",
      " 34900K .......... .......... .......... .......... .......... 75%  120M 1s\n",
      " 34950K .......... .......... .......... .......... .......... 75%  101M 1s\n",
      " 35000K .......... .......... .......... .......... .......... 75%  118M 1s\n",
      " 35050K .......... .......... .......... .......... .......... 75%  121M 1s\n",
      " 35100K .......... .......... .......... .......... .......... 75%  102M 1s\n",
      " 35150K .......... .......... .......... .......... .......... 76% 38,3M 1s\n",
      " 35200K .......... .......... .......... .......... .......... 76% 68,5M 1s\n",
      " 35250K .......... .......... .......... .......... .......... 76%  123M 1s\n",
      " 35300K .......... .......... .......... .......... .......... 76%  120M 1s\n",
      " 35350K .......... .......... .......... .......... .......... 76% 1,56M 1s\n",
      " 35400K .......... .......... .......... .......... .......... 76% 47,5M 1s\n",
      " 35450K .......... .......... .......... .......... .......... 76% 56,3M 1s\n",
      " 35500K .......... .......... .......... .......... .......... 76%  106M 1s\n",
      " 35550K .......... .......... .......... .......... .......... 76% 56,9M 1s\n",
      " 35600K .......... .......... .......... .......... .......... 76% 82,7M 1s\n",
      " 35650K .......... .......... .......... .......... .......... 77%  110M 1s\n",
      " 35700K .......... .......... .......... .......... .......... 77% 91,0M 1s\n",
      " 35750K .......... .......... .......... .......... .......... 77%  165M 1s\n",
      " 35800K .......... .......... .......... .......... .......... 77% 59,4M 1s\n",
      " 35850K .......... .......... .......... .......... .......... 77% 91,9M 1s\n",
      " 35900K .......... .......... .......... .......... .......... 77%  113M 1s\n",
      " 35950K .......... .......... .......... .......... .......... 77% 92,7M 1s\n",
      " 36000K .......... .......... .......... .......... .......... 77%  108M 1s\n",
      " 36050K .......... .......... .......... .......... .......... 77%  107M 1s\n",
      " 36100K .......... .......... .......... .......... .......... 78% 91,0M 1s\n",
      " 36150K .......... .......... .......... .......... .......... 78%  113M 1s\n",
      " 36200K .......... .......... .......... .......... .......... 78%  104M 1s\n",
      " 36250K .......... .......... .......... .......... .......... 78%  244K 1s\n",
      " 36300K .......... .......... .......... .......... .......... 78%  175M 1s\n",
      " 36350K .......... .......... .......... .......... .......... 78% 30,0M 1s\n",
      " 36400K .......... .......... .......... .......... .......... 78%  116M 1s\n",
      " 36450K .......... .......... .......... .......... .......... 78% 47,1M 1s\n",
      " 36500K .......... .......... .......... .......... .......... 78% 95,5M 1s\n",
      " 36550K .......... .......... .......... .......... .......... 79% 93,9M 1s\n",
      " 36600K .......... .......... .......... .......... .......... 79% 59,5M 1s\n",
      " 36650K .......... .......... .......... .......... .......... 79%  110M 1s\n",
      " 36700K .......... .......... .......... .......... .......... 79% 90,3M 1s\n",
      " 36750K .......... .......... .......... .......... .......... 79%  109M 1s\n",
      " 36800K .......... .......... .......... .......... .......... 79%  116M 1s\n",
      " 36850K .......... .......... .......... .......... .......... 79% 85,2M 1s\n",
      " 36900K .......... .......... .......... .......... .......... 79%  110M 1s\n",
      " 36950K .......... .......... .......... .......... .......... 79% 55,2M 1s\n",
      " 37000K .......... .......... .......... .......... .......... 80%  112M 1s\n",
      " 37050K .......... .......... .......... .......... .......... 80%  117M 1s\n",
      " 37100K .......... .......... .......... .......... .......... 80%  102M 1s\n",
      " 37150K .......... .......... .......... .......... .......... 80%  123M 1s\n",
      " 37200K .......... .......... .......... .......... .......... 80%  101M 1s\n",
      " 37250K .......... .......... .......... .......... .......... 80%  116M 1s\n",
      " 37300K .......... .......... .......... .......... .......... 80%  109M 1s\n",
      " 37350K .......... .......... .......... .......... .......... 80% 92,1M 1s\n",
      " 37400K .......... .......... .......... .......... .......... 80%  111M 1s\n",
      " 37450K .......... .......... .......... .......... .......... 80% 88,5M 1s\n",
      " 37500K .......... .......... .......... .......... .......... 81%  116M 1s\n",
      " 37550K .......... .......... .......... .......... .......... 81%  122M 1s\n",
      " 37600K .......... .......... .......... .......... .......... 81%  109M 1s\n",
      " 37650K .......... .......... .......... .......... .......... 81%  137M 1s\n",
      " 37700K .......... .......... .......... .......... .......... 81%  111M 1s\n",
      " 37750K .......... .......... .......... .......... .......... 81%  136M 1s\n",
      " 37800K .......... .......... .......... .......... .......... 81%  134M 1s\n",
      " 37850K .......... .......... .......... .......... .......... 81%  119M 1s\n",
      " 37900K .......... .......... .......... .......... .......... 81%  162M 1s\n",
      " 37950K .......... .......... .......... .......... .......... 82%  139M 1s\n",
      " 38000K .......... .......... .......... .......... .......... 82%  162M 1s\n",
      " 38050K .......... .......... .......... .......... .......... 82%  161M 1s\n",
      " 38100K .......... .......... .......... .......... .......... 82%  102M 1s\n",
      " 38150K .......... .......... .......... .......... .......... 82%  118M 1s\n",
      " 38200K .......... .......... .......... .......... .......... 82%  104M 1s\n",
      " 38250K .......... .......... .......... .......... .......... 82%  117M 1s\n",
      " 38300K .......... .......... .......... .......... .......... 82%  117M 1s\n",
      " 38350K .......... .......... .......... .......... .......... 82%  105M 1s\n",
      " 38400K .......... .......... .......... .......... .......... 83%  116M 1s\n",
      " 38450K .......... .......... .......... .......... .......... 83%  101M 1s\n",
      " 38500K .......... .......... .......... .......... .......... 83%  124M 1s\n",
      " 38550K .......... .......... .......... .......... .......... 83%  115M 1s\n",
      " 38600K .......... .......... .......... .......... .......... 83%  106M 1s\n",
      " 38650K .......... .......... .......... .......... .......... 83%  116M 1s\n",
      " 38700K .......... .......... .......... .......... .......... 83%  102M 1s\n",
      " 38750K .......... .......... .......... .......... .......... 83%  121M 1s\n",
      " 38800K .......... .......... .......... .......... .......... 83%  120M 1s\n",
      " 38850K .......... .......... .......... .......... .......... 84%  103M 1s\n",
      " 38900K .......... .......... .......... .......... .......... 84%  119M 1s\n",
      " 38950K .......... .......... .......... .......... .......... 84%  101M 1s\n",
      " 39000K .......... .......... .......... .......... .......... 84%  118M 1s\n",
      " 39050K .......... .......... .......... .......... .......... 84%  121M 1s\n",
      " 39100K .......... .......... .......... .......... .......... 84%  104M 1s\n",
      " 39150K .......... .......... .......... .......... .......... 84%  116M 1s\n",
      " 39200K .......... .......... .......... .......... .......... 84%  105M 1s\n",
      " 39250K .......... .......... .......... .......... .......... 84%  118M 1s\n",
      " 39300K .......... .......... .......... .......... .......... 84% 1,62M 1s\n",
      " 39350K .......... .......... .......... .......... .......... 85% 35,2M 1s\n",
      " 39400K .......... .......... .......... .......... .......... 85% 38,3M 1s\n",
      " 39450K .......... .......... .......... .......... .......... 85%  110M 1s\n",
      " 39500K .......... .......... .......... .......... .......... 85% 92,8M 1s\n",
      " 39550K .......... .......... .......... .......... .......... 85% 31,7M 1s\n",
      " 39600K .......... .......... .......... .......... .......... 85% 86,3M 1s\n",
      " 39650K .......... .......... .......... .......... .......... 85%  187M 1s\n",
      " 39700K .......... .......... .......... .......... .......... 85% 93,5M 1s\n",
      " 39750K .......... .......... .......... .......... .......... 85%  108M 1s\n",
      " 39800K .......... .......... .......... .......... .......... 86%  112M 1s\n",
      " 39850K .......... .......... .......... .......... .......... 86% 93,6M 1s\n",
      " 39900K .......... .......... .......... .......... .......... 86%  107M 1s\n",
      " 39950K .......... .......... .......... .......... .......... 86%  104M 1s\n",
      " 40000K .......... .......... .......... .......... .......... 86%  173M 1s\n",
      " 40050K .......... .......... .......... .......... .......... 86% 69,5M 1s\n",
      " 40100K .......... .......... .......... .......... .......... 86% 99,8M 1s\n",
      " 40150K .......... .......... .......... .......... .......... 86%  119M 1s\n",
      " 40200K .......... .......... .......... .......... .......... 86%  105M 1s\n",
      " 40250K .......... .......... .......... .......... .......... 87%  123M 1s\n",
      " 40300K .......... .......... .......... .......... .......... 87%  136M 1s\n",
      " 40350K .......... .......... .......... .......... .......... 87%  119M 1s\n",
      " 40400K .......... .......... .......... .......... .......... 87%  135M 1s\n",
      " 40450K .......... .......... .......... .......... .......... 87%  110M 1s\n",
      " 40500K .......... .......... .......... .......... .......... 87%  119M 1s\n",
      " 40550K .......... .......... .......... .......... .......... 87%  123M 1s\n",
      " 40600K .......... .......... .......... .......... .......... 87%  104M 1s\n",
      " 40650K .......... .......... .......... .......... .......... 87%  115M 1s\n",
      " 40700K .......... .......... .......... .......... .......... 88%  102M 1s\n",
      " 40750K .......... .......... .......... .......... .......... 88%  123M 1s\n",
      " 40800K .......... .......... .......... .......... .......... 88%  117M 1s\n",
      " 40850K .......... .......... .......... .......... .......... 88%  103M 1s\n",
      " 40900K .......... .......... .......... .......... .......... 88%  121M 1s\n",
      " 40950K .......... .......... .......... .......... .......... 88%  102M 1s\n",
      " 41000K .......... .......... .......... .......... .......... 88%  123M 1s\n",
      " 41050K .......... .......... .......... .......... .......... 88%  111M 1s\n",
      " 41100K .......... .......... .......... .......... .......... 88%  107M 1s\n",
      " 41150K .......... .......... .......... .......... .......... 88%  121M 1s\n",
      " 41200K .......... .......... .......... .......... .......... 89%  102M 1s\n",
      " 41250K .......... .......... .......... .......... .......... 89%  118M 1s\n",
      " 41300K .......... .......... .......... .......... .......... 89% 54,8M 1s\n",
      " 41350K .......... .......... .......... .......... .......... 89%  456M 1s\n",
      " 41400K .......... .......... .......... .......... .......... 89%  171M 1s\n",
      " 41450K .......... .......... .......... .......... .......... 89%  103M 1s\n",
      " 41500K .......... .......... .......... .......... .......... 89%  118M 1s\n",
      " 41550K .......... .......... .......... .......... .......... 89%  120M 1s\n",
      " 41600K .......... .......... .......... .......... .......... 89%  102M 1s\n",
      " 41650K .......... .......... .......... .......... .......... 90%  118M 1s\n",
      " 41700K .......... .......... .......... .......... .......... 90%  102M 1s\n",
      " 41750K .......... .......... .......... .......... .......... 90%  121M 1s\n",
      " 41800K .......... .......... .......... .......... .......... 90%  118M 0s\n",
      " 41850K .......... .......... .......... .......... .......... 90%  102M 0s\n",
      " 41900K .......... .......... .......... .......... .......... 90%  121M 0s\n",
      " 41950K .......... .......... .......... .......... .......... 90%  102M 0s\n",
      " 42000K .......... .......... .......... .......... .......... 90%  121M 0s\n",
      " 42050K .......... .......... .......... .......... .......... 90%  117M 0s\n",
      " 42100K .......... .......... .......... .......... .......... 91%  102M 0s\n",
      " 42150K .......... .......... .......... .......... .......... 91%  121M 0s\n",
      " 42200K .......... .......... .......... .......... .......... 91%  102M 0s\n",
      " 42250K .......... .......... .......... .......... .......... 91%  117M 0s\n",
      " 42300K .......... .......... .......... .......... .......... 91%  121M 0s\n",
      " 42350K .......... .......... .......... .......... .......... 91% 1,67M 0s\n",
      " 42400K .......... .......... .......... .......... .......... 91% 37,2M 0s\n",
      " 42450K .......... .......... .......... .......... .......... 91% 52,6M 0s\n",
      " 42500K .......... .......... .......... .......... .......... 91% 50,5M 0s\n",
      " 42550K .......... .......... .......... .......... .......... 91% 99,5M 0s\n",
      " 42600K .......... .......... .......... .......... .......... 92%  102M 0s\n",
      " 42650K .......... .......... .......... .......... .......... 92% 32,2M 0s\n",
      " 42700K .......... .......... .......... .......... .......... 92% 90,7M 0s\n",
      " 42750K .......... .......... .......... .......... .......... 92%  120M 0s\n",
      " 42800K .......... .......... .......... .......... .......... 92%  121M 0s\n",
      " 42850K .......... .......... .......... .......... .......... 92%  101M 0s\n",
      " 42900K .......... .......... .......... .......... .......... 92%  121M 0s\n",
      " 42950K .......... .......... .......... .......... .......... 92%  101M 0s\n",
      " 43000K .......... .......... .......... .......... .......... 92%  122M 0s\n",
      " 43050K .......... .......... .......... .......... .......... 93%  119M 0s\n",
      " 43100K .......... .......... .......... .......... .......... 93% 72,8M 0s\n",
      " 43150K .......... .......... .......... .......... .......... 93%  146M 0s\n",
      " 43200K .......... .......... .......... .......... .......... 93%  134M 0s\n",
      " 43250K .......... .......... .......... .......... .......... 93% 49,8M 0s\n",
      " 43300K .......... .......... .......... .......... .......... 93% 65,4M 0s\n",
      " 43350K .......... .......... .......... .......... .......... 93%  106M 0s\n",
      " 43400K .......... .......... .......... .......... .......... 93%  116M 0s\n",
      " 43450K .......... .......... .......... .......... .......... 93%  104M 0s\n",
      " 43500K .......... .......... .......... .......... .......... 94%  120M 0s\n",
      " 43550K .......... .......... .......... .......... .......... 94%  121M 0s\n",
      " 43600K .......... .......... .......... .......... .......... 94%  102M 0s\n",
      " 43650K .......... .......... .......... .......... .......... 94%  119M 0s\n",
      " 43700K .......... .......... .......... .......... .......... 94%  101M 0s\n",
      " 43750K .......... .......... .......... .......... .......... 94%  121M 0s\n",
      " 43800K .......... .......... .......... .......... .......... 94%  117M 0s\n",
      " 43850K .......... .......... .......... .......... .......... 94%  103M 0s\n",
      " 43900K .......... .......... .......... .......... .......... 94%  120M 0s\n",
      " 43950K .......... .......... .......... .......... .......... 95%  100M 0s\n",
      " 44000K .......... .......... .......... .......... .......... 95%  121M 0s\n",
      " 44050K .......... .......... .......... .......... .......... 95%  121M 0s\n",
      " 44100K .......... .......... .......... .......... .......... 95%  102M 0s\n",
      " 44150K .......... .......... .......... .......... .......... 95%  122M 0s\n",
      " 44200K .......... .......... .......... .......... .......... 95%  102M 0s\n",
      " 44250K .......... .......... .......... .......... .......... 95%  117M 0s\n",
      " 44300K .......... .......... .......... .......... .......... 95%  122M 0s\n",
      " 44350K .......... .......... .......... .......... .......... 95%  102M 0s\n",
      " 44400K .......... .......... .......... .......... .......... 95%  117M 0s\n",
      " 44450K .......... .......... .......... .......... .......... 96%  102M 0s\n",
      " 44500K .......... .......... .......... .......... .......... 96%  122M 0s\n",
      " 44550K .......... .......... .......... .......... .......... 96%  117M 0s\n",
      " 44600K .......... .......... .......... .......... .......... 96%  103M 0s\n",
      " 44650K .......... .......... .......... .......... .......... 96%  120M 0s\n",
      " 44700K .......... .......... .......... .......... .......... 96%  103M 0s\n",
      " 44750K .......... .......... .......... .......... .......... 96%  120M 0s\n",
      " 44800K .......... .......... .......... .......... .......... 96%  117M 0s\n",
      " 44850K .......... .......... .......... .......... .......... 96%  103M 0s\n",
      " 44900K .......... .......... .......... .......... .......... 97%  121M 0s\n",
      " 44950K .......... .......... .......... .......... .......... 97%  102M 0s\n",
      " 45000K .......... .......... .......... .......... .......... 97%  118M 0s\n",
      " 45050K .......... .......... .......... .......... .......... 97%  120M 0s\n",
      " 45100K .......... .......... .......... .......... .......... 97%  103M 0s\n",
      " 45150K .......... .......... .......... .......... .......... 97%  120M 0s\n",
      " 45200K .......... .......... .......... .......... .......... 97%  102M 0s\n",
      " 45250K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 45300K .......... .......... .......... .......... .......... 97%  121M 0s\n",
      " 45350K .......... .......... .......... .......... .......... 98%  102M 0s\n",
      " 45400K .......... .......... .......... .......... .......... 98% 1,73M 0s\n",
      " 45450K .......... .......... .......... .......... .......... 98% 22,6M 0s\n",
      " 45500K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 45550K .......... .......... .......... .......... .......... 98% 61,7M 0s\n",
      " 45600K .......... .......... .......... .......... .......... 98% 55,9M 0s\n",
      " 45650K .......... .......... .......... .......... .......... 98%  120M 0s\n",
      " 45700K .......... .......... .......... .......... .......... 98% 89,4M 0s\n",
      " 45750K .......... .......... .......... .......... .......... 98%  118M 0s\n",
      " 45800K .......... .......... .......... .......... .......... 99% 64,7M 0s\n",
      " 45850K .......... .......... .......... .......... .......... 99%  104M 0s\n",
      " 45900K .......... .......... .......... .......... .......... 99%  127M 0s\n",
      " 45950K .......... .......... .......... .......... .......... 99%  101M 0s\n",
      " 46000K .......... .......... .......... .......... .......... 99%  119M 0s\n",
      " 46050K .......... .......... .......... .......... .......... 99%  121M 0s\n",
      " 46100K .......... .......... .......... .......... .......... 99%  101M 0s\n",
      " 46150K .......... .......... .......... .......... .......... 99%  119M 0s\n",
      " 46200K .......... .......... .......... .......... .......... 99% 81,9M 0s\n",
      " 46250K .......... .......... .......... .......... .......... 99%  120M 0s\n",
      " 46300K .....                                                 100%  117M=4,8s\n",
      "\n",
      "2023-08-18 20:45:33 (9,50 MB/s) - ‘data/reference/genome.fa.gz’ guardado [47416846/47416846]\n",
      "\n",
      "[Fri Aug 18 20:45:33 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T204526.677880.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:45:34 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR696683_chr7_1.fastq.gz, data/raw/ERR696683_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR696683_chr7_1_fastqc.html, results/fastqc_result/ERR696683_chr7_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR696683_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 20:45:49 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:45:49 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR696683_chr7_1.fastq.gz, data/raw/ERR696683_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR696683_chr7_1_fastp.fastq.gz, data/processed/ERR696683_chr7_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR696683_chr7_2_fastp.fastq.gz, data/processed/ERR696683_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1935275\n",
      "total bases: 146263550\n",
      "Q20 bases: 145211647(99.2808%)\n",
      "Q30 bases: 141120433(96.4837%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1935275\n",
      "total bases: 145878500\n",
      "Q20 bases: 144064467(98.7565%)\n",
      "Q30 bases: 138098537(94.6668%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1177764\n",
      "total bases: 89449589\n",
      "Q20 bases: 89095661(99.6043%)\n",
      "Q30 bases: 87513697(97.8358%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1177764\n",
      "total bases: 89414052\n",
      "Q20 bases: 88793329(99.3058%)\n",
      "Q30 bases: 86459019(96.6951%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2355528\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1515022\n",
      "reads with adapter trimmed: 8864\n",
      "bases trimmed due to adapters: 230085\n",
      "\n",
      "Duplication rate: 5.65647%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR696683_chr7_1.fastq.gz -I data/raw/ERR696683_chr7_2.fastq.gz -o data/processed/ERR696683_chr7_1_fastp.fastq.gz -O data/processed/ERR696683_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Fri Aug 18 20:45:55 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:45:55 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR696683_chr7_1_fastp.fastq.gz, data/processed/ERR696683_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR696683_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR696683_chr7_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR696683_chr7_2_fastp.fastq.gz, data/processed/ERR696683_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 20:46:04 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T204533.974225.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:46:05 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753368_chr7_1.fastq.gz, data/raw/ERR753368_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753368_chr7_1_fastp.fastq.gz, data/processed/ERR753368_chr7_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753368_chr7_1_fastp.fastq.gz, data/processed/ERR753368_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1806088\n",
      "total bases: 137130775\n",
      "Q20 bases: 136588468(99.6045%)\n",
      "Q30 bases: 132513779(96.6331%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1806088\n",
      "total bases: 136987438\n",
      "Q20 bases: 135772877(99.1134%)\n",
      "Q30 bases: 129457863(94.5035%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1548330\n",
      "total bases: 117666684\n",
      "Q20 bases: 117461408(99.8255%)\n",
      "Q30 bases: 115455876(98.1211%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1548330\n",
      "total bases: 117656644\n",
      "Q20 bases: 117075719(99.5063%)\n",
      "Q30 bases: 113529570(96.4923%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3096660\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 515516\n",
      "reads with adapter trimmed: 5202\n",
      "bases trimmed due to adapters: 140710\n",
      "\n",
      "Duplication rate: 2.03583%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753368_chr7_1.fastq.gz -I data/raw/ERR753368_chr7_2.fastq.gz -o data/processed/ERR753368_chr7_1_fastp.fastq.gz -O data/processed/ERR753368_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Fri Aug 18 20:46:12 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:46:12 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753368_chr7_1.fastq.gz, data/raw/ERR753368_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753368_chr7_1_fastqc.html, results/fastqc_result/ERR753368_chr7_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753368_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753368_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 20:46:24 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:46:24 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753368_chr7_1_fastp.fastq.gz, data/processed/ERR753368_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753368_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753368_chr7_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753368_chr7_1_fastp.fastq.gz, data/processed/ERR753368_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 20:46:37 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T204604.876502.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:46:38 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753369_chr7_1.fastq.gz, data/raw/ERR753369_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753369_chr7_1_fastp.fastq.gz, data/processed/ERR753369_chr7_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753369_chr7_1_fastp.fastq.gz, data/processed/ERR753369_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1929269\n",
      "total bases: 146477595\n",
      "Q20 bases: 145872914(99.5872%)\n",
      "Q30 bases: 141418223(96.546%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1929269\n",
      "total bases: 146329545\n",
      "Q20 bases: 145027500(99.1102%)\n",
      "Q30 bases: 138248767(94.4777%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1651945\n",
      "total bases: 125540731\n",
      "Q20 bases: 125307776(99.8144%)\n",
      "Q30 bases: 123093775(98.0509%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1651945\n",
      "total bases: 125530164\n",
      "Q20 bases: 124909386(99.5055%)\n",
      "Q30 bases: 121110875(96.4795%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3303890\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 554648\n",
      "reads with adapter trimmed: 5782\n",
      "bases trimmed due to adapters: 156225\n",
      "\n",
      "Duplication rate: 2.35789%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753369_chr7_1.fastq.gz -I data/raw/ERR753369_chr7_2.fastq.gz -o data/processed/ERR753369_chr7_1_fastp.fastq.gz -O data/processed/ERR753369_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Fri Aug 18 20:46:45 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:46:45 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753369_chr7_1_fastp.fastq.gz, data/processed/ERR753369_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753369_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753369_chr7_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753369_chr7_1_fastp.fastq.gz, data/processed/ERR753369_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 20:46:57 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:46:57 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753369_chr7_1.fastq.gz, data/raw/ERR753369_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753369_chr7_1_fastqc.html, results/fastqc_result/ERR753369_chr7_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753369_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753369_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 20:47:12 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T204637.716104.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:47:13 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753370_chr7_1.fastq.gz, data/raw/ERR753370_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753370_chr7_1_fastqc.html, results/fastqc_result/ERR753370_chr7_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753370_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753370_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 20:47:36 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:47:36 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753370_chr7_1.fastq.gz, data/raw/ERR753370_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753370_chr7_1_fastp.fastq.gz, data/processed/ERR753370_chr7_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753370_chr7_1_fastp.fastq.gz, data/processed/ERR753370_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2673460\n",
      "total bases: 269090503\n",
      "Q20 bases: 267232327(99.3095%)\n",
      "Q30 bases: 259317581(96.3682%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2673460\n",
      "total bases: 269019918\n",
      "Q20 bases: 266435611(99.0394%)\n",
      "Q30 bases: 256135213(95.2105%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 2669965\n",
      "total bases: 267638922\n",
      "Q20 bases: 265893698(99.3479%)\n",
      "Q30 bases: 258369403(96.5366%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 2669965\n",
      "total bases: 266755891\n",
      "Q20 bases: 264414765(99.1224%)\n",
      "Q30 bases: 254782415(95.5114%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 5339930\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 6990\n",
      "reads with adapter trimmed: 22426\n",
      "bases trimmed due to adapters: 449745\n",
      "\n",
      "Duplication rate: 4.34648%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 159\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753370_chr7_1.fastq.gz -I data/raw/ERR753370_chr7_2.fastq.gz -o data/processed/ERR753370_chr7_1_fastp.fastq.gz -O data/processed/ERR753370_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 11 seconds\n",
      "[Fri Aug 18 20:47:48 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:47:48 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753370_chr7_1_fastp.fastq.gz, data/processed/ERR753370_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753370_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753370_chr7_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753370_chr7_1_fastp.fastq.gz, data/processed/ERR753370_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 20:48:11 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T204712.913622.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:48:12 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753371_chr7_1.fastq.gz, data/raw/ERR753371_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753371_chr7_1_fastqc.html, results/fastqc_result/ERR753371_chr7_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753371_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753371_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 20:48:27 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:48:27 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753371_chr7_1.fastq.gz, data/raw/ERR753371_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753371_chr7_1_fastp.fastq.gz, data/processed/ERR753371_chr7_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753371_chr7_2_fastp.fastq.gz, data/processed/ERR753371_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1777787\n",
      "total bases: 133923240\n",
      "Q20 bases: 132711766(99.0954%)\n",
      "Q30 bases: 127818746(95.4418%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1777787\n",
      "total bases: 133584939\n",
      "Q20 bases: 131975601(98.7953%)\n",
      "Q30 bases: 127258356(95.264%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 901910\n",
      "total bases: 68460468\n",
      "Q20 bases: 68125599(99.5109%)\n",
      "Q30 bases: 66410192(97.0052%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 901910\n",
      "total bases: 68422604\n",
      "Q20 bases: 67941529(99.2969%)\n",
      "Q30 bases: 66426314(97.0824%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1803820\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1751754\n",
      "reads with adapter trimmed: 7008\n",
      "bases trimmed due to adapters: 174651\n",
      "\n",
      "Duplication rate: 21.6814%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 118\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753371_chr7_1.fastq.gz -I data/raw/ERR753371_chr7_2.fastq.gz -o data/processed/ERR753371_chr7_1_fastp.fastq.gz -O data/processed/ERR753371_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Fri Aug 18 20:48:33 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:48:33 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753371_chr7_1_fastp.fastq.gz, data/processed/ERR753371_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753371_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753371_chr7_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753371_chr7_2_fastp.fastq.gz, data/processed/ERR753371_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 20:48:42 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T204811.841548.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:48:43 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753372_chr7_1.fastq.gz, data/raw/ERR753372_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753372_chr7_1_fastqc.html, results/fastqc_result/ERR753372_chr7_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753372_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753372_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 20:48:59 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:48:59 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753372_chr7_1.fastq.gz, data/raw/ERR753372_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753372_chr7_1_fastp.fastq.gz, data/processed/ERR753372_chr7_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753372_chr7_1_fastp.fastq.gz, data/processed/ERR753372_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1928165\n",
      "total bases: 146395509\n",
      "Q20 bases: 145810094(99.6001%)\n",
      "Q30 bases: 141369327(96.5667%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1928165\n",
      "total bases: 146239318\n",
      "Q20 bases: 144940401(99.1118%)\n",
      "Q30 bases: 138222059(94.5177%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1646996\n",
      "total bases: 125164638\n",
      "Q20 bases: 124943449(99.8233%)\n",
      "Q30 bases: 122763086(98.0813%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1646996\n",
      "total bases: 125153924\n",
      "Q20 bases: 124539813(99.5093%)\n",
      "Q30 bases: 120808917(96.5283%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3293992\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 562338\n",
      "reads with adapter trimmed: 6016\n",
      "bases trimmed due to adapters: 162856\n",
      "\n",
      "Duplication rate: 1.98008%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753372_chr7_1.fastq.gz -I data/raw/ERR753372_chr7_2.fastq.gz -o data/processed/ERR753372_chr7_1_fastp.fastq.gz -O data/processed/ERR753372_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Fri Aug 18 20:49:06 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:49:06 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753372_chr7_1_fastp.fastq.gz, data/processed/ERR753372_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753372_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753372_chr7_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753372_chr7_1_fastp.fastq.gz, data/processed/ERR753372_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 20:49:18 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T204842.353133.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:49:20 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753373_chr7_1.fastq.gz, data/raw/ERR753373_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753373_chr7_1_fastp.fastq.gz, data/processed/ERR753373_chr7_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753373_chr7_1_fastp.fastq.gz, data/processed/ERR753373_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1590509\n",
      "total bases: 152276057\n",
      "Q20 bases: 151053516(99.1972%)\n",
      "Q30 bases: 140667684(92.3768%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1590509\n",
      "total bases: 149042940\n",
      "Q20 bases: 147417333(98.9093%)\n",
      "Q30 bases: 137755782(92.4269%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1550999\n",
      "total bases: 146136096\n",
      "Q20 bases: 145283119(99.4163%)\n",
      "Q30 bases: 136885543(93.6699%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1550999\n",
      "total bases: 141207201\n",
      "Q20 bases: 140529494(99.5201%)\n",
      "Q30 bases: 133173977(94.311%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3101998\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 79020\n",
      "reads with adapter trimmed: 4140\n",
      "bases trimmed due to adapters: 154596\n",
      "\n",
      "Duplication rate: 1.84731%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 153\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753373_chr7_1.fastq.gz -I data/raw/ERR753373_chr7_2.fastq.gz -o data/processed/ERR753373_chr7_1_fastp.fastq.gz -O data/processed/ERR753373_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 7 seconds\n",
      "[Fri Aug 18 20:49:27 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:49:27 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753373_chr7_1.fastq.gz, data/raw/ERR753373_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753373_chr7_1_fastqc.html, results/fastqc_result/ERR753373_chr7_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753373_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753373_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 20:49:42 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:49:42 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753373_chr7_1_fastp.fastq.gz, data/processed/ERR753373_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753373_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753373_chr7_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753373_chr7_1_fastp.fastq.gz, data/processed/ERR753373_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 20:49:56 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T204919.012747.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:49:58 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753374_chr7_1.fastq.gz, data/raw/ERR753374_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753374_chr7_1_fastp.fastq.gz, data/processed/ERR753374_chr7_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753374_chr7_2_fastp.fastq.gz, data/processed/ERR753374_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 937433\n",
      "total bases: 89668520\n",
      "Q20 bases: 88820397(99.0542%)\n",
      "Q30 bases: 81109687(90.455%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 937433\n",
      "total bases: 87729358\n",
      "Q20 bases: 86808200(98.95%)\n",
      "Q30 bases: 79810509(90.9735%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 907041\n",
      "total bases: 84770678\n",
      "Q20 bases: 84167468(99.2884%)\n",
      "Q30 bases: 78047285(92.0687%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 907041\n",
      "total bases: 82557850\n",
      "Q20 bases: 82082554(99.4243%)\n",
      "Q30 bases: 76724913(92.9347%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1814082\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 60784\n",
      "reads with adapter trimmed: 2678\n",
      "bases trimmed due to adapters: 102178\n",
      "\n",
      "Duplication rate: 5.64177%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 149\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753374_chr7_1.fastq.gz -I data/raw/ERR753374_chr7_2.fastq.gz -o data/processed/ERR753374_chr7_1_fastp.fastq.gz -O data/processed/ERR753374_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Fri Aug 18 20:50:02 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:50:02 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753374_chr7_1_fastp.fastq.gz, data/processed/ERR753374_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753374_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753374_chr7_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753374_chr7_2_fastp.fastq.gz, data/processed/ERR753374_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 20:50:13 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:50:13 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753374_chr7_1.fastq.gz, data/raw/ERR753374_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753374_chr7_1_fastqc.html, results/fastqc_result/ERR753374_chr7_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753374_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753374_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 20:50:24 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T204956.993412.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:50:25 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753375_chr7_1.fastq.gz, data/raw/ERR753375_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753375_chr7_1_fastqc.html, results/fastqc_result/ERR753375_chr7_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753375_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753375_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 20:50:36 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:50:36 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753375_chr7_1.fastq.gz, data/raw/ERR753375_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753375_chr7_1_fastp.fastq.gz, data/processed/ERR753375_chr7_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753375_chr7_1_fastp.fastq.gz, data/processed/ERR753375_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1203954\n",
      "total bases: 90919106\n",
      "Q20 bases: 90345397(99.369%)\n",
      "Q30 bases: 88249315(97.0636%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1203954\n",
      "total bases: 90896831\n",
      "Q20 bases: 89692945(98.6755%)\n",
      "Q30 bases: 87415245(96.1697%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 768315\n",
      "total bases: 58337245\n",
      "Q20 bases: 58144802(99.6701%)\n",
      "Q30 bases: 57253055(98.1415%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 768315\n",
      "total bases: 58346791\n",
      "Q20 bases: 57776292(99.0222%)\n",
      "Q30 bases: 56839488(97.4166%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1536630\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 871278\n",
      "reads with adapter trimmed: 4636\n",
      "bases trimmed due to adapters: 122070\n",
      "\n",
      "Duplication rate: 1.68137%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753375_chr7_1.fastq.gz -I data/raw/ERR753375_chr7_2.fastq.gz -o data/processed/ERR753375_chr7_1_fastp.fastq.gz -O data/processed/ERR753375_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Fri Aug 18 20:50:40 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:50:40 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753375_chr7_1_fastp.fastq.gz, data/processed/ERR753375_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753375_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753375_chr7_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753375_chr7_1_fastp.fastq.gz, data/processed/ERR753375_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 20:50:48 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T205024.411290.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:50:49 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753376_chr7_1.fastq.gz, data/raw/ERR753376_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753376_chr7_1_fastp.fastq.gz, data/processed/ERR753376_chr7_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753376_chr7_1_fastp.fastq.gz, data/processed/ERR753376_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1556102\n",
      "total bases: 117603181\n",
      "Q20 bases: 116778781(99.299%)\n",
      "Q30 bases: 113491260(96.5036%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1556102\n",
      "total bases: 117282101\n",
      "Q20 bases: 115811340(98.746%)\n",
      "Q30 bases: 110980302(94.6268%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 941456\n",
      "total bases: 71502248\n",
      "Q20 bases: 71230059(99.6193%)\n",
      "Q30 bases: 69975404(97.8646%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 941456\n",
      "total bases: 71475352\n",
      "Q20 bases: 70978000(99.3042%)\n",
      "Q30 bases: 69107114(96.6866%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1882912\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1229292\n",
      "reads with adapter trimmed: 6714\n",
      "bases trimmed due to adapters: 177235\n",
      "\n",
      "Duplication rate: 8.84146%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753376_chr7_1.fastq.gz -I data/raw/ERR753376_chr7_2.fastq.gz -o data/processed/ERR753376_chr7_1_fastp.fastq.gz -O data/processed/ERR753376_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Fri Aug 18 20:50:54 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:50:54 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753376_chr7_1.fastq.gz, data/raw/ERR753376_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753376_chr7_1_fastqc.html, results/fastqc_result/ERR753376_chr7_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753376_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753376_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 20:51:07 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:51:07 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753376_chr7_1_fastp.fastq.gz, data/processed/ERR753376_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753376_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753376_chr7_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753376_chr7_1_fastp.fastq.gz, data/processed/ERR753376_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 20:51:16 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T205048.582472.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:51:17 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753377_chr7_1.fastq.gz, data/raw/ERR753377_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753377_chr7_1_fastqc.html, results/fastqc_result/ERR753377_chr7_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753377_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753377_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 20:51:30 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:51:30 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753377_chr7_1.fastq.gz, data/raw/ERR753377_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753377_chr7_1_fastp.fastq.gz, data/processed/ERR753377_chr7_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753377_chr7_1_fastp.fastq.gz, data/processed/ERR753377_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1631934\n",
      "total bases: 123352115\n",
      "Q20 bases: 122472837(99.2872%)\n",
      "Q30 bases: 119023024(96.4905%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1631934\n",
      "total bases: 123051439\n",
      "Q20 bases: 121582205(98.806%)\n",
      "Q30 bases: 116732556(94.8648%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1009848\n",
      "total bases: 76696745\n",
      "Q20 bases: 76397840(99.6103%)\n",
      "Q30 bases: 75043318(97.8442%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1009848\n",
      "total bases: 76668037\n",
      "Q20 bases: 76158958(99.336%)\n",
      "Q30 bases: 74237699(96.8301%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2019696\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1244172\n",
      "reads with adapter trimmed: 6972\n",
      "bases trimmed due to adapters: 178774\n",
      "\n",
      "Duplication rate: 8.81519%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 119\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753377_chr7_1.fastq.gz -I data/raw/ERR753377_chr7_2.fastq.gz -o data/processed/ERR753377_chr7_1_fastp.fastq.gz -O data/processed/ERR753377_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Fri Aug 18 20:51:35 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:51:35 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753377_chr7_1_fastp.fastq.gz, data/processed/ERR753377_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753377_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753377_chr7_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753377_chr7_1_fastp.fastq.gz, data/processed/ERR753377_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 20:51:44 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T205116.548263.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:51:45 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753378_chr7_1.fastq.gz, data/raw/ERR753378_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753378_chr7_1_fastqc.html, results/fastqc_result/ERR753378_chr7_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753378_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753378_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 100% complete for ERR753378_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 100% complete for ERR753378_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 20:51:58 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:51:58 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753378_chr7_1.fastq.gz, data/raw/ERR753378_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753378_chr7_1_fastp.fastq.gz, data/processed/ERR753378_chr7_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753378_chr7_2_fastp.fastq.gz, data/processed/ERR753378_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1662012\n",
      "total bases: 126189423\n",
      "Q20 bases: 125691373(99.6053%)\n",
      "Q30 bases: 121897371(96.5987%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1662012\n",
      "total bases: 126054782\n",
      "Q20 bases: 124937025(99.1133%)\n",
      "Q30 bases: 119131691(94.5079%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1424617\n",
      "total bases: 108264659\n",
      "Q20 bases: 108073674(99.8236%)\n",
      "Q30 bases: 106186267(98.0803%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1424617\n",
      "total bases: 108255476\n",
      "Q20 bases: 107719628(99.505%)\n",
      "Q30 bases: 104436218(96.472%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2849234\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 474790\n",
      "reads with adapter trimmed: 5220\n",
      "bases trimmed due to adapters: 141481\n",
      "\n",
      "Duplication rate: 1.98912%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753378_chr7_1.fastq.gz -I data/raw/ERR753378_chr7_2.fastq.gz -o data/processed/ERR753378_chr7_1_fastp.fastq.gz -O data/processed/ERR753378_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Fri Aug 18 20:52:04 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:52:04 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753378_chr7_1_fastp.fastq.gz, data/processed/ERR753378_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753378_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753378_chr7_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753378_chr7_2_fastp.fastq.gz, data/processed/ERR753378_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 20:52:15 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T205144.907572.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:52:16 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR696683_chr7_1_fastp.fastq.gz, data/processed/ERR696683_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR696683_chr7.sam, results/mapped_reads/ERR696683_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR696683_chr7_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR696683_chr7_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.51 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 62.75 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.48 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.33 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 34.01 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 98.714 sec; CPU: 98.073 sec\n",
      "[Fri Aug 18 20:55:40 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:55:40 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR696683_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr7.bam, results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR696683_chr7.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR696683_chr7_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 20:56:22 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:56:22 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR696683_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "20:56:23.418 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 20:56:23 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR696683_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR696683_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 20:56:23 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 20:56:23\tMarkDuplicates\tStart of doWork freeMemory: 528829056; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 20:56:23\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 20:56:23\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam\n",
      "INFO\t2023-08-18 20:56:25\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:75,871,766\n",
      "INFO\t2023-08-18 20:56:25\tMarkDuplicates\tTracking 731 as yet unmatched pairs. 731 records in RAM.\n",
      "INFO\t2023-08-18 20:56:26\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 7:138,577,465\n",
      "INFO\t2023-08-18 20:56:26\tMarkDuplicates\tTracking 375 as yet unmatched pairs. 375 records in RAM.\n",
      "INFO\t2023-08-18 20:56:26\tMarkDuplicates\tRead 2393966 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 20:56:27\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 313328592; totalMemory: 639631360; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 20:56:27\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 20:56:27\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 20:56:27\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 20:56:28\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 20:56:28\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 934275736; totalMemory: 1480589312; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 20:56:28\tMarkDuplicates\tMarking 206351 records as duplicates.\n",
      "INFO\t2023-08-18 20:56:28\tMarkDuplicates\tFound 75 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam\n",
      "INFO\t2023-08-18 20:56:28\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 20:56:43\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 20:56:43\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 20:56:43\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 20:56:43\tMarkDuplicates\tBefore output close freeMemory: 528128464; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 20:56:43\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 20:56:43\tMarkDuplicates\tAfter output close freeMemory: 528128464; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 20:56:43 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.33 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 20:56:44 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T205215.509217.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 20:56:46 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753368_chr7_1_fastp.fastq.gz, data/processed/ERR753368_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753368_chr7.sam, results/mapped_reads/ERR753368_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753368_chr7_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753368_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.52 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 59.22 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.49 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.33 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 34.19 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 95.493 sec; CPU: 94.746 sec\n",
      "[Fri Aug 18 21:00:47 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:00:47 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753368_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr7.bam, results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753368_chr7.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753368_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 21:01:36 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:01:36 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753368_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753368_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "21:01:37.033 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 21:01:37 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753368_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753368_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 21:01:37 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 21:01:37\tMarkDuplicates\tStart of doWork freeMemory: 528785808; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:01:37\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 21:01:37\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam\n",
      "INFO\t2023-08-18 21:01:38\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:56,377,969\n",
      "INFO\t2023-08-18 21:01:38\tMarkDuplicates\tTracking 421 as yet unmatched pairs. 421 records in RAM.\n",
      "INFO\t2023-08-18 21:01:40\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 7:105,465,325\n",
      "INFO\t2023-08-18 21:01:40\tMarkDuplicates\tTracking 472 as yet unmatched pairs. 472 records in RAM.\n",
      "INFO\t2023-08-18 21:01:41\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 7:152,248,151\n",
      "INFO\t2023-08-18 21:01:41\tMarkDuplicates\tTracking 275 as yet unmatched pairs. 275 records in RAM.\n",
      "INFO\t2023-08-18 21:01:41\tMarkDuplicates\tRead 3118785 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 21:01:41\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 406162792; totalMemory: 812646400; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:01:41\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 21:01:41\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 21:01:42\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 21:01:42\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 21:01:42\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 803159648; totalMemory: 1349517312; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:01:42\tMarkDuplicates\tMarking 74958 records as duplicates.\n",
      "INFO\t2023-08-18 21:01:42\tMarkDuplicates\tFound 515 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam\n",
      "INFO\t2023-08-18 21:01:42\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 21:02:01\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 21:02:01\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 21:02:01\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 21:02:01\tMarkDuplicates\tBefore output close freeMemory: 528084072; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:02:01\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 21:02:01\tMarkDuplicates\tAfter output close freeMemory: 528084072; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 21:02:01 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.40 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 21:02:03 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T205644.957945.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:02:04 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753369_chr7_1_fastp.fastq.gz, data/processed/ERR753369_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753369_chr7.sam, results/mapped_reads/ERR753369_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753369_chr7_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753369_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.53 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 59.48 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.50 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.33 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 33.11 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 99.385 sec; CPU: 93.943 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 21:06:29 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:06:29 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753369_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr7.bam, results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753369_chr7.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753369_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 21:07:29 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:07:29 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753369_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753369_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "21:07:30.522 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 21:07:30 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753369_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753369_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 21:07:30 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 21:07:30\tMarkDuplicates\tStart of doWork freeMemory: 528819976; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:07:30\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 21:07:30\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam\n",
      "INFO\t2023-08-18 21:07:32\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:53,276,537\n",
      "INFO\t2023-08-18 21:07:32\tMarkDuplicates\tTracking 381 as yet unmatched pairs. 381 records in RAM.\n",
      "INFO\t2023-08-18 21:07:33\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 7:101,057,915\n",
      "INFO\t2023-08-18 21:07:33\tMarkDuplicates\tTracking 525 as yet unmatched pairs. 525 records in RAM.\n",
      "INFO\t2023-08-18 21:07:35\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 7:144,095,190\n",
      "INFO\t2023-08-18 21:07:35\tMarkDuplicates\tTracking 240 as yet unmatched pairs. 240 records in RAM.\n",
      "INFO\t2023-08-18 21:07:35\tMarkDuplicates\tRead 3329144 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 21:07:35\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 561966248; totalMemory: 990904320; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:07:35\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 21:07:35\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 21:07:36\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 21:07:36\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 21:07:36\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 981452360; totalMemory: 1527775232; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:07:36\tMarkDuplicates\tMarking 92630 records as duplicates.\n",
      "INFO\t2023-08-18 21:07:36\tMarkDuplicates\tFound 528 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam\n",
      "INFO\t2023-08-18 21:07:36\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 21:07:55\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 21:07:55\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 21:07:55\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 21:07:55\tMarkDuplicates\tBefore output close freeMemory: 528118640; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:07:55\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 21:07:55\tMarkDuplicates\tAfter output close freeMemory: 528118640; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 21:07:55 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.42 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 21:07:58 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T210203.411383.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:07:59 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753370_chr7_1_fastp.fastq.gz, data/processed/ERR753370_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753370_chr7.sam, results/mapped_reads/ERR753370_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753370_chr7_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753370_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.54 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 58.10 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.49 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.33 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 32.65 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 97.152 sec; CPU: 92.101 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 21:15:05 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:15:05 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753370_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr7.bam, results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753370_chr7.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753370_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 21:17:05 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:17:05 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753370_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753370_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "21:17:05.977 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 21:17:05 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753370_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753370_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 21:17:05 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 21:17:05\tMarkDuplicates\tStart of doWork freeMemory: 528786440; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:17:05\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 21:17:05\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam\n",
      "INFO\t2023-08-18 21:17:07\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:33,152,778\n",
      "INFO\t2023-08-18 21:17:07\tMarkDuplicates\tTracking 1010 as yet unmatched pairs. 1010 records in RAM.\n",
      "INFO\t2023-08-18 21:17:09\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 7:72,939,724\n",
      "INFO\t2023-08-18 21:17:09\tMarkDuplicates\tTracking 1942 as yet unmatched pairs. 1942 records in RAM.\n",
      "INFO\t2023-08-18 21:17:10\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 7:95,299,077\n",
      "INFO\t2023-08-18 21:17:10\tMarkDuplicates\tTracking 1649 as yet unmatched pairs. 1649 records in RAM.\n",
      "INFO\t2023-08-18 21:17:12\tMarkDuplicates\tRead     4,000,000 records.  Elapsed time: 00:00:06s.  Time for last 1,000,000:    1s.  Last read position: 7:117,114,803\n",
      "INFO\t2023-08-18 21:17:12\tMarkDuplicates\tTracking 1114 as yet unmatched pairs. 1114 records in RAM.\n",
      "INFO\t2023-08-18 21:17:13\tMarkDuplicates\tRead     5,000,000 records.  Elapsed time: 00:00:07s.  Time for last 1,000,000:    1s.  Last read position: 7:144,268,951\n",
      "INFO\t2023-08-18 21:17:13\tMarkDuplicates\tTracking 717 as yet unmatched pairs. 717 records in RAM.\n",
      "INFO\t2023-08-18 21:17:14\tMarkDuplicates\tRead 5445839 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 21:17:14\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 492304360; totalMemory: 1141899264; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:17:14\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 21:17:14\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 21:17:15\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 21:17:15\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 21:17:15\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1132413000; totalMemory: 1678770176; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:17:15\tMarkDuplicates\tMarking 368385 records as duplicates.\n",
      "INFO\t2023-08-18 21:17:15\tMarkDuplicates\tFound 219 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam\n",
      "INFO\t2023-08-18 21:17:15\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 21:17:53\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 21:17:53\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 21:17:53\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 21:17:53\tMarkDuplicates\tBefore output close freeMemory: 528084088; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:17:53\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 21:17:53\tMarkDuplicates\tAfter output close freeMemory: 528084088; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 21:17:53 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.79 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 21:17:57 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T210758.309020.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:17:58 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753371_chr7_1_fastp.fastq.gz, data/processed/ERR753371_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753371_chr7.sam, results/mapped_reads/ERR753371_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753371_chr7_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR753371_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.51 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 55.52 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.49 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.33 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 32.60 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 94.789 sec; CPU: 89.453 sec\n",
      "[Fri Aug 18 21:20:56 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:20:56 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753371_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr7.bam, results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753371_chr7.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753371_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 21:21:28 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:21:28 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753371_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753371_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "21:21:29.091 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 21:21:29 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753371_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753371_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 21:21:29 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 21:21:29\tMarkDuplicates\tStart of doWork freeMemory: 528789056; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:21:29\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 21:21:29\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam\n",
      "INFO\t2023-08-18 21:21:30\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:94,599,576\n",
      "INFO\t2023-08-18 21:21:30\tMarkDuplicates\tTracking 159 as yet unmatched pairs. 159 records in RAM.\n",
      "INFO\t2023-08-18 21:21:31\tMarkDuplicates\tRead 1812344 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 21:21:31\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 335213560; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:21:31\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 21:21:31\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 21:21:32\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 21:21:32\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 21:21:32\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 811549816; totalMemory: 1357905920; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:21:32\tMarkDuplicates\tMarking 702725 records as duplicates.\n",
      "INFO\t2023-08-18 21:21:32\tMarkDuplicates\tFound 58 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam\n",
      "INFO\t2023-08-18 21:21:32\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 21:21:43\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 21:21:43\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 21:21:43\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 21:21:43\tMarkDuplicates\tBefore output close freeMemory: 528085096; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:21:43\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 21:21:43\tMarkDuplicates\tAfter output close freeMemory: 528085096; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 21:21:43 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.24 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 21:21:44 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T211757.813433.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:21:45 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753372_chr7_1_fastp.fastq.gz, data/processed/ERR753372_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753372_chr7.sam, results/mapped_reads/ERR753372_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753372_chr7_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753372_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.53 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 54.22 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.47 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.31 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 31.84 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 92.618 sec; CPU: 87.367 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 21:25:56 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:25:56 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753372_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr7.bam, results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753372_chr7.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753372_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 21:26:55 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:26:55 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753372_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753372_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "21:26:56.101 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 21:26:56 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753372_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753372_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 21:26:56 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 21:26:56\tMarkDuplicates\tStart of doWork freeMemory: 528789560; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:26:56\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 21:26:56\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam\n",
      "INFO\t2023-08-18 21:26:57\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:51,800,134\n",
      "INFO\t2023-08-18 21:26:57\tMarkDuplicates\tTracking 415 as yet unmatched pairs. 415 records in RAM.\n",
      "INFO\t2023-08-18 21:26:59\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 7:101,087,619\n",
      "INFO\t2023-08-18 21:26:59\tMarkDuplicates\tTracking 490 as yet unmatched pairs. 490 records in RAM.\n",
      "INFO\t2023-08-18 21:27:00\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 7:144,371,095\n",
      "INFO\t2023-08-18 21:27:00\tMarkDuplicates\tTracking 223 as yet unmatched pairs. 223 records in RAM.\n",
      "INFO\t2023-08-18 21:27:00\tMarkDuplicates\tRead 3322476 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 21:27:00\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 432052768; totalMemory: 859832320; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:27:00\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 21:27:00\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 21:27:01\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 21:27:01\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 21:27:01\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1081036288; totalMemory: 1627389952; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:27:01\tMarkDuplicates\tMarking 78229 records as duplicates.\n",
      "INFO\t2023-08-18 21:27:01\tMarkDuplicates\tFound 500 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam\n",
      "INFO\t2023-08-18 21:27:01\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 21:27:20\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 21:27:20\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 21:27:20\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 21:27:20\tMarkDuplicates\tBefore output close freeMemory: 528088336; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:27:20\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 21:27:20\tMarkDuplicates\tAfter output close freeMemory: 528088336; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 21:27:20 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.41 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 21:27:22 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T212144.528484.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:27:24 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753373_chr7_1_fastp.fastq.gz, data/processed/ERR753373_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753373_chr7.sam, results/mapped_reads/ERR753373_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753373_chr7_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR753373_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.52 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 54.02 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.47 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.32 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 31.94 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 92.412 sec; CPU: 87.258 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 21:33:31 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:33:31 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753373_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr7.bam, results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753373_chr7.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753373_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 21:35:09 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:35:09 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753373_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753373_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "21:35:10.142 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 21:35:10 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753373_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753373_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 21:35:10 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 21:35:10\tMarkDuplicates\tStart of doWork freeMemory: 528784448; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:35:10\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 21:35:10\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam\n",
      "INFO\t2023-08-18 21:35:11\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:54,553,814\n",
      "INFO\t2023-08-18 21:35:11\tMarkDuplicates\tTracking 260 as yet unmatched pairs. 260 records in RAM.\n",
      "INFO\t2023-08-18 21:35:13\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 7:102,664,313\n",
      "INFO\t2023-08-18 21:35:13\tMarkDuplicates\tTracking 229 as yet unmatched pairs. 229 records in RAM.\n",
      "INFO\t2023-08-18 21:35:14\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 7:150,822,160\n",
      "INFO\t2023-08-18 21:35:14\tMarkDuplicates\tTracking 56 as yet unmatched pairs. 56 records in RAM.\n",
      "INFO\t2023-08-18 21:35:15\tMarkDuplicates\tRead 3137876 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 21:35:15\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 611922800; totalMemory: 1019215872; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:35:15\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 21:35:15\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 21:35:15\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 21:35:15\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 21:35:15\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1009728152; totalMemory: 1556086784; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:35:15\tMarkDuplicates\tMarking 79803 records as duplicates.\n",
      "INFO\t2023-08-18 21:35:15\tMarkDuplicates\tFound 26038 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam\n",
      "INFO\t2023-08-18 21:35:15\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 21:35:43\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 21:35:43\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 21:35:43\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 21:35:43\tMarkDuplicates\tBefore output close freeMemory: 528082416; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:35:43\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 21:35:43\tMarkDuplicates\tAfter output close freeMemory: 528082416; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 21:35:43 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.56 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 21:35:46 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T212722.980552.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:35:47 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753374_chr7_1_fastp.fastq.gz, data/processed/ERR753374_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753374_chr7.sam, results/mapped_reads/ERR753374_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753374_chr7_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753374_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.55 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 62.07 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.50 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.33 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 34.11 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 103.236 sec; CPU: 97.564 sec\n",
      "[Fri Aug 18 21:40:23 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:40:23 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753374_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr7.bam, results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753374_chr7.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753374_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 21:41:12 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:41:12 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753374_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753374_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "21:41:13.297 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 21:41:13 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753374_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753374_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 21:41:13 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 21:41:13\tMarkDuplicates\tStart of doWork freeMemory: 528785816; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:41:13\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 21:41:13\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam\n",
      "INFO\t2023-08-18 21:41:15\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:91,121,710\n",
      "INFO\t2023-08-18 21:41:15\tMarkDuplicates\tTracking 227 as yet unmatched pairs. 227 records in RAM.\n",
      "INFO\t2023-08-18 21:41:16\tMarkDuplicates\tRead 1838287 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 21:41:16\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 269111264; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:41:16\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 21:41:16\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 21:41:16\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 21:41:16\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 21:41:17\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 815742048; totalMemory: 1362100224; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:41:17\tMarkDuplicates\tMarking 189114 records as duplicates.\n",
      "INFO\t2023-08-18 21:41:17\tMarkDuplicates\tFound 63001 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam\n",
      "INFO\t2023-08-18 21:41:17\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 21:41:32\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 21:41:32\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 21:41:32\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 21:41:32\tMarkDuplicates\tBefore output close freeMemory: 528083464; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:41:32\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 21:41:32\tMarkDuplicates\tAfter output close freeMemory: 528083464; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 21:41:32 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.32 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 21:41:33 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T213546.431794.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:41:35 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753375_chr7_1_fastp.fastq.gz, data/processed/ERR753375_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753375_chr7.sam, results/mapped_reads/ERR753375_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753375_chr7_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753375_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.56 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 60.83 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.47 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.33 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 34.15 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 101.524 sec; CPU: 96.345 sec\n",
      "[Fri Aug 18 21:44:36 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:44:36 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753375_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr7.bam, results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753375_chr7.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753375_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 21:45:05 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:45:05 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753375_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753375_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "21:45:06.469 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 21:45:06 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753375_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753375_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 21:45:06 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 21:45:06\tMarkDuplicates\tStart of doWork freeMemory: 528779272; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:45:06\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 21:45:06\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam\n",
      "INFO\t2023-08-18 21:45:08\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:102,659,617\n",
      "INFO\t2023-08-18 21:45:08\tMarkDuplicates\tTracking 226 as yet unmatched pairs. 226 records in RAM.\n",
      "INFO\t2023-08-18 21:45:08\tMarkDuplicates\tRead 1549276 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 21:45:08\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 299193104; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:45:08\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 21:45:09\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 21:45:09\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 21:45:09\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 21:45:09\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 761210024; totalMemory: 1307574272; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:45:09\tMarkDuplicates\tMarking 70136 records as duplicates.\n",
      "INFO\t2023-08-18 21:45:09\tMarkDuplicates\tFound 170 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam\n",
      "INFO\t2023-08-18 21:45:09\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 21:45:19\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 21:45:19\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 21:45:19\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 21:45:19\tMarkDuplicates\tBefore output close freeMemory: 528077440; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:45:19\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 21:45:19\tMarkDuplicates\tAfter output close freeMemory: 528077440; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 21:45:19 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.22 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 21:45:20 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T214134.011959.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:45:21 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753376_chr7_1_fastp.fastq.gz, data/processed/ERR753376_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753376_chr7.sam, results/mapped_reads/ERR753376_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753376_chr7_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR753376_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.55 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 59.98 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.50 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.34 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 34.02 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 100.784 sec; CPU: 95.386 sec\n",
      "[Fri Aug 18 21:48:35 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:48:35 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753376_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr7.bam, results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753376_chr7.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753376_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 21:49:13 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:49:13 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753376_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753376_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "21:49:14.625 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 21:49:14 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753376_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753376_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 21:49:14 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 21:49:14\tMarkDuplicates\tStart of doWork freeMemory: 528827128; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:49:14\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 21:49:14\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam\n",
      "INFO\t2023-08-18 21:49:16\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:93,135,591\n",
      "INFO\t2023-08-18 21:49:16\tMarkDuplicates\tTracking 287 as yet unmatched pairs. 287 records in RAM.\n",
      "INFO\t2023-08-18 21:49:17\tMarkDuplicates\tRead 1892327 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 21:49:17\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 263856408; totalMemory: 538968064; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:49:17\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 21:49:17\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 21:49:18\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 21:49:18\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 21:49:18\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 843047504; totalMemory: 1389363200; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:49:18\tMarkDuplicates\tMarking 255064 records as duplicates.\n",
      "INFO\t2023-08-18 21:49:18\tMarkDuplicates\tFound 82 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam\n",
      "INFO\t2023-08-18 21:49:18\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 21:49:31\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 21:49:31\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 21:49:31\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 21:49:31\tMarkDuplicates\tBefore output close freeMemory: 528125192; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:49:31\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 21:49:31\tMarkDuplicates\tAfter output close freeMemory: 528125192; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 21:49:31 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.27 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 21:49:32 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T214520.825902.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:49:33 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753377_chr7_1_fastp.fastq.gz, data/processed/ERR753377_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753377_chr7.sam, results/mapped_reads/ERR753377_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753377_chr7_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753377_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.58 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 64.33 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.51 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.35 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 35.52 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 106.686 sec; CPU: 101.287 sec\n",
      "[Fri Aug 18 21:53:02 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:53:02 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753377_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr7.bam, results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753377_chr7.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753377_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 21:53:42 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:53:42 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753377_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753377_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "21:53:42.991 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 21:53:43 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753377_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753377_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 21:53:43 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 21:53:43\tMarkDuplicates\tStart of doWork freeMemory: 528789920; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:53:43\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 21:53:43\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam\n",
      "INFO\t2023-08-18 21:53:44\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:88,825,890\n",
      "INFO\t2023-08-18 21:53:44\tMarkDuplicates\tTracking 415 as yet unmatched pairs. 415 records in RAM.\n",
      "INFO\t2023-08-18 21:53:46\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 7:154,976,188\n",
      "INFO\t2023-08-18 21:53:46\tMarkDuplicates\tTracking 144 as yet unmatched pairs. 144 records in RAM.\n",
      "INFO\t2023-08-18 21:53:46\tMarkDuplicates\tRead 2043633 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 21:53:46\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 551895920; totalMemory: 842006528; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:53:46\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 21:53:46\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 21:53:46\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 21:53:47\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 21:53:47\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 873418384; totalMemory: 1419771904; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:53:47\tMarkDuplicates\tMarking 269941 records as duplicates.\n",
      "INFO\t2023-08-18 21:53:47\tMarkDuplicates\tFound 71 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam\n",
      "INFO\t2023-08-18 21:53:47\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 21:54:00\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 21:54:00\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 21:54:00\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 21:54:00\tMarkDuplicates\tBefore output close freeMemory: 528088176; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:54:00\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 21:54:00\tMarkDuplicates\tAfter output close freeMemory: 528088176; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 21:54:00 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.29 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 21:54:01 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T214932.583472.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:54:03 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753378_chr7_1_fastp.fastq.gz, data/processed/ERR753378_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753378_chr7.sam, results/mapped_reads/ERR753378_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753378_chr7_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753378_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.54 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 61.28 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.52 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.35 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 34.53 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 102.684 sec; CPU: 97.225 sec\n",
      "[Fri Aug 18 21:58:02 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:58:02 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753378_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr7.bam, results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753378_chr7.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753378_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 21:58:46 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:58:46 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753378_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753378_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "21:58:47.613 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 21:58:47 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753378_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753378_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 21:58:47 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 21:58:47\tMarkDuplicates\tStart of doWork freeMemory: 528774384; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:58:47\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 21:58:47\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam\n",
      "INFO\t2023-08-18 21:58:49\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:65,356,793\n",
      "INFO\t2023-08-18 21:58:49\tMarkDuplicates\tTracking 550 as yet unmatched pairs. 550 records in RAM.\n",
      "INFO\t2023-08-18 21:58:50\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 7:116,559,171\n",
      "INFO\t2023-08-18 21:58:50\tMarkDuplicates\tTracking 427 as yet unmatched pairs. 427 records in RAM.\n",
      "INFO\t2023-08-18 21:58:51\tMarkDuplicates\tRead 2871707 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 21:58:51\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 840847728; totalMemory: 1220542464; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:58:51\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 21:58:52\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 21:58:52\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 21:58:52\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 21:58:52\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1211044560; totalMemory: 1757413376; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:58:52\tMarkDuplicates\tMarking 67000 records as duplicates.\n",
      "INFO\t2023-08-18 21:58:52\tMarkDuplicates\tFound 448 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam\n",
      "INFO\t2023-08-18 21:58:52\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 21:59:09\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 21:59:09\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 21:59:09\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 21:59:09\tMarkDuplicates\tBefore output close freeMemory: 528072856; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 21:59:09\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 21:59:09\tMarkDuplicates\tAfter output close freeMemory: 528072856; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 21:59:09 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.36 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 21:59:10 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T215401.944575.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 21:59:12 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR696683_chr7_dedup.bam\n",
      "    output: results/variants/ERR696683_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR696683_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR696683_chr7.vcf\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "index file data/reference/genome.fa.fai not found, generating...\n",
      "[Fri Aug 18 22:00:00 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T215911.138806.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:00:01 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753368_chr7_dedup.bam\n",
      "    output: results/variants/ERR753368_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753368_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753368_chr7.vcf\n",
      "    wildcards: sample=ERR753368_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 22:01:03 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T220000.353611.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:01:04 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753369_chr7_dedup.bam\n",
      "    output: results/variants/ERR753369_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753369_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753369_chr7.vcf\n",
      "    wildcards: sample=ERR753369_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 22:02:09 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T220103.989539.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:02:10 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753370_chr7_dedup.bam\n",
      "    output: results/variants/ERR753370_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753370_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753370_chr7.vcf\n",
      "    wildcards: sample=ERR753370_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 22:03:57 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T220209.876964.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:03:58 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753371_chr7_dedup.bam\n",
      "    output: results/variants/ERR753371_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753371_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753371_chr7.vcf\n",
      "    wildcards: sample=ERR753371_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 22:04:32 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T220357.417987.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:04:33 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753372_chr7_dedup.bam\n",
      "    output: results/variants/ERR753372_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753372_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753372_chr7.vcf\n",
      "    wildcards: sample=ERR753372_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 22:05:40 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T220432.389585.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:05:41 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753373_chr7_dedup.bam\n",
      "    output: results/variants/ERR753373_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753373_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753373_chr7.vcf\n",
      "    wildcards: sample=ERR753373_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 22:06:57 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T220540.460464.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:06:58 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753374_chr7_dedup.bam\n",
      "    output: results/variants/ERR753374_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753374_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753374_chr7.vcf\n",
      "    wildcards: sample=ERR753374_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 22:07:48 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T220657.751846.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:07:49 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753375_chr7_dedup.bam\n",
      "    output: results/variants/ERR753375_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753375_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753375_chr7.vcf\n",
      "    wildcards: sample=ERR753375_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 22:08:27 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T220748.827849.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:08:29 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753376_chr7_dedup.bam\n",
      "    output: results/variants/ERR753376_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753376_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753376_chr7.vcf\n",
      "    wildcards: sample=ERR753376_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 22:09:09 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T220828.046176.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:09:10 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753377_chr7_dedup.bam\n",
      "    output: results/variants/ERR753377_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753377_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753377_chr7.vcf\n",
      "    wildcards: sample=ERR753377_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 22:09:51 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T220909.590433.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:09:53 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753378_chr7_dedup.bam\n",
      "    output: results/variants/ERR753378_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753378_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753378_chr7.vcf\n",
      "    wildcards: sample=ERR753378_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 22:10:52 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T220952.041785.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:10:54 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR696683_chr7.vcf\n",
      "    output: results/variants/vep/ERR696683_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR696683_chr7.txt\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 22:11:21 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T221053.093688.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:11:23 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753368_chr7.vcf\n",
      "    output: results/variants/vep/ERR753368_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753368_chr7.txt\n",
      "    wildcards: sample=ERR753368_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 22:11:53 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T221121.674003.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:11:54 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753369_chr7.vcf\n",
      "    output: results/variants/vep/ERR753369_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753369_chr7.txt\n",
      "    wildcards: sample=ERR753369_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 22:12:23 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T221153.445461.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:12:25 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753370_chr7.vcf\n",
      "    output: results/variants/vep/ERR753370_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753370_chr7.txt\n",
      "    wildcards: sample=ERR753370_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 22:12:59 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T221224.059922.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:13:00 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753371_chr7.vcf\n",
      "    output: results/variants/vep/ERR753371_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753371_chr7.txt\n",
      "    wildcards: sample=ERR753371_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 22:13:22 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T221259.421841.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:13:23 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753372_chr7.vcf\n",
      "    output: results/variants/vep/ERR753372_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753372_chr7.txt\n",
      "    wildcards: sample=ERR753372_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 22:13:52 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T221322.221111.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:13:53 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753373_chr7.vcf\n",
      "    output: results/variants/vep/ERR753373_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753373_chr7.txt\n",
      "    wildcards: sample=ERR753373_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 22:14:24 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T221352.790259.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:14:25 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753374_chr7.vcf\n",
      "    output: results/variants/vep/ERR753374_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753374_chr7.txt\n",
      "    wildcards: sample=ERR753374_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 22:14:48 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T221424.782877.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:14:49 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753375_chr7.vcf\n",
      "    output: results/variants/vep/ERR753375_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753375_chr7.txt\n",
      "    wildcards: sample=ERR753375_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 22:15:12 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T221448.992728.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:15:13 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753376_chr7.vcf\n",
      "    output: results/variants/vep/ERR753376_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753376_chr7.txt\n",
      "    wildcards: sample=ERR753376_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 22:15:36 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T221512.907066.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:15:38 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753377_chr7.vcf\n",
      "    output: results/variants/vep/ERR753377_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753377_chr7.txt\n",
      "    wildcards: sample=ERR753377_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 22:16:01 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T221537.146718.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:16:02 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753378_chr7.vcf\n",
      "    output: results/variants/vep/ERR753378_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753378_chr7.txt\n",
      "    wildcards: sample=ERR753378_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Fri Aug 18 22:16:30 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T221601.762940.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job              count    min threads    max threads\n",
      "-------------  -------  -------------  -------------\n",
      "parsing_dataR        1              1              1\n",
      "total                1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:16:31 2023]\n",
      "rule parsing_dataR:\n",
      "    input: code/05parsing_vep_data.R\n",
      "    output: tasks/13parsing_dataR.done\n",
      "    jobid: 0\n",
      "    reason: Params have changed since last execution\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Warning message:\n",
      "package ‘glue’ was built under R version 4.2.3 \n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">>> This is your sample list: ERR696683_chr7\n",
      ">>> This is your sample list: ERR753368_chr7\n",
      ">>> This is your sample list: ERR753369_chr7\n",
      ">>> This is your sample list: ERR753370_chr7\n",
      ">>> This is your sample list: ERR753371_chr7\n",
      ">>> This is your sample list: ERR753372_chr7\n",
      ">>> This is your sample list: ERR753373_chr7\n",
      ">>> This is your sample list: ERR753374_chr7\n",
      ">>> This is your sample list: ERR753375_chr7\n",
      ">>> This is your sample list: ERR753376_chr7\n",
      ">>> This is your sample list: ERR753377_chr7\n",
      ">>> This is your sample list: ERR753378_chr7\n",
      ">>> This is your gene: BRAF\n",
      "# A tibble: 44 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "3 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "4 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   downstream…\n",
      "6 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 38 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 52 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "2 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "3 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "4 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "5 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "6 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 46 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 5 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "3 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "4 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   downstream…\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>,\n",
      "#   freqs <chr>, clin_sig <chr>, somatic <chr>, pheno <chr>, pubmed <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 5 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "3 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "4 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   downstream…\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>,\n",
      "#   freqs <chr>, clin_sig <chr>, somatic <chr>, pheno <chr>, pubmed <chr>, …\n",
      "# A tibble: 31 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "3 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "4 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   downstream…\n",
      "6 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 25 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 31 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "3 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "4 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   downstream…\n",
      "6 7_140749350_T/C       7:1407493… C      ENSG… ENST00… Transcript   synonymous…\n",
      "# ℹ 25 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 52 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "2 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "3 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "4 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "5 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "6 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 46 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 39 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "2 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "3 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "4 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "5 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "6 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 33 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 44 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "3 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "4 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   downstream…\n",
      "6 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 38 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 5 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "3 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "4 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   downstream…\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>,\n",
      "#   freqs <chr>, clin_sig <chr>, somatic <chr>, pheno <chr>, pubmed <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 0 × 44\n",
      "# ℹ 44 variables: #uploaded_variation <chr>, location <chr>, allele <chr>,\n",
      "#   gene <chr>, feature <chr>, feature_type <chr>, consequence <chr>,\n",
      "#   cdna_position <chr>, cds_position <chr>, protein_position <chr>,\n",
      "#   amino_acids <chr>, codons <chr>, existing_variation <chr>, impact <chr>,\n",
      "#   distance <chr>, strand <chr>, flags <chr>, variant_class <chr>,\n",
      "#   symbol <chr>, symbol_source <chr>, hgnc_id <chr>, biotype <chr>,\n",
      "#   mane_select <chr>, mane_plus_clinical <chr>, tsl <chr>, appris <chr>, …\n",
      "# A tibble: 26 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140749350_T/C       7:1407493… C      ENSG… ENST00… Transcript   synonymous…\n",
      "2 7_140749350_T/C       7:1407493… C      ENSG… ENST00… Transcript   synonymous…\n",
      "3 7_140749350_T/C       7:1407493… C      ENSG… ENST00… Transcript   synonymous…\n",
      "4 7_140749350_T/C       7:1407493… C      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 7_140749350_T/C       7:1407493… C      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 7_140749350_T/C       7:1407493… C      ENSG… ENST00… Transcript   non_coding…\n",
      "# ℹ 20 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "[1] \"##==========================##\"\n",
      "[1] \"##===> WORK FINISHED!!! <===##\"\n",
      "[1] \"##==========================##\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "Touching output file tasks/13parsing_dataR.done.\n",
      "[Fri Aug 18 22:16:35 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T221631.141305.snakemake.log\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['snakemake', '--cores', '1', 'parsing_dataR'], returncode=0)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Running the workflow for the 7º chormosome\n",
    "\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--force\", \"data/reference/genome.fa\"]) ## rule reference_genome \n",
    "\n",
    "for sample in sample_list:\n",
    "    ## Quality inspection\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/fastqc_result/\" + sample + \"_chr7_1_fastqc.html\",   \n",
    "                    \"data/processed/\" + sample + \"_chr7_1_fastp.fastq.gz\",   \n",
    "                    \"results/fastqc_result/trimmed/\" + sample + \"_chr7_1_fastp_fastqc.html\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    ## Mapping reads\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/mapped_reads/\" + sample + \"_chr7_sorted.sam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr7_sorted.bam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr7_dedup.bam\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    # Extracting variants  \n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/\" + sample + \"_chr7.vcf\"])   \n",
    "\n",
    "## VEP CLI\n",
    "for sample in sample_list:\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/vep/\" + sample + \"_chr7.txt\"])\n",
    "\n",
    "## Extracting the gene PIKCA3 from the VEP files\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"parsing_dataR\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Changing the config file to do the chromosome 12 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['sed', '-i', '50s/BRAF/KRAS/g', 'config.yaml'], returncode=0)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subprocess.run([\"sed\", \"-i\", \"6,+12s/chr7/chr12/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\",\"24s/chr7/chr12/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"31s/chromosome.7.fa.gz/chromosome.12.fa.gz/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"50s/BRAF/KRAS/g\", \"config.yaml\"])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job                 count    min threads    max threads\n",
      "----------------  -------  -------------  -------------\n",
      "reference_genome        1              1              1\n",
      "total                   1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:53:29 2023]\n",
      "rule reference_genome:\n",
      "    output: data/reference/genome.fa\n",
      "    jobid: 0\n",
      "    reason: Forced execution\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "--2023-08-18 22:53:29--  https://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.12.fa.gz\n",
      "Resolviendo ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.169\n",
      "Conectando con ftp.ensembl.org (ftp.ensembl.org)[193.62.193.169]:443... conectado.\n",
      "Petición HTTP enviada, esperando respuesta... 200 OK\n",
      "Longitud: 39795542 (38M) [application/x-gzip]\n",
      "Guardando como: ‘data/reference/genome.fa.gz’\n",
      "\n",
      "     0K .......... .......... .......... .......... ..........  0%  411K 94s\n",
      "    50K .......... .......... .......... .......... ..........  0% 4,73M 51s\n",
      "   100K .......... .......... .......... .......... ..........  0%  976K 47s\n",
      "   150K .......... .......... .......... .......... ..........  0% 5,07M 37s\n",
      "   200K .......... .......... .......... .......... ..........  0% 1005K 37s\n",
      "   250K .......... .......... .......... .......... ..........  0% 48,4M 31s\n",
      "   300K .......... .......... .......... .......... ..........  0% 43,5M 27s\n",
      "   350K .......... .......... .......... .......... ..........  1% 93,6M 24s\n",
      "   400K .......... .......... .......... .......... ..........  1% 5,40M 22s\n",
      "   450K .......... .......... .......... .......... ..........  1% 1,01M 23s\n",
      "   500K .......... .......... .......... .......... ..........  1% 35,9M 21s\n",
      "   550K .......... .......... .......... .......... ..........  1%  110M 19s\n",
      "   600K .......... .......... .......... .......... ..........  1% 94,1M 18s\n",
      "   650K .......... .......... .......... .......... ..........  1%  117M 17s\n",
      "   700K .......... .......... .......... .......... ..........  1% 97,2M 16s\n",
      "   750K .......... .......... .......... .......... ..........  2%  112M 15s\n",
      "   800K .......... .......... .......... .......... ..........  2% 7,22M 14s\n",
      "   850K .......... .......... .......... .......... ..........  2% 25,8M 13s\n",
      "   900K .......... .......... .......... .......... ..........  2% 45,9M 13s\n",
      "   950K .......... .......... .......... .......... ..........  2% 1,05M 14s\n",
      "  1000K .......... .......... .......... .......... ..........  2% 45,0M 13s\n",
      "  1050K .......... .......... .......... .......... ..........  2% 34,0M 13s\n",
      "  1100K .......... .......... .......... .......... ..........  2%  105M 12s\n",
      "  1150K .......... .......... .......... .......... ..........  3%  118M 11s\n",
      "  1200K .......... .......... .......... .......... ..........  3%  103M 11s\n",
      "  1250K .......... .......... .......... .......... ..........  3%  121M 11s\n",
      "  1300K .......... .......... .......... .......... ..........  3%  112M 10s\n",
      "  1350K .......... .......... .......... .......... ..........  3%  104M 10s\n",
      "  1400K .......... .......... .......... .......... ..........  3%  124M 9s\n",
      "  1450K .......... .......... .......... .......... ..........  3%  102M 9s\n",
      "  1500K .......... .......... .......... .......... ..........  3%  114M 9s\n",
      "  1550K .......... .......... .......... .......... ..........  4%  123M 9s\n",
      "  1600K .......... .......... .......... .......... ..........  4%  101M 8s\n",
      "  1650K .......... .......... .......... .......... ..........  4% 20,4M 8s\n",
      "  1700K .......... .......... .......... .......... ..........  4% 21,1M 8s\n",
      "  1750K .......... .......... .......... .......... ..........  4% 68,1M 8s\n",
      "  1800K .......... .......... .......... .......... ..........  4% 63,3M 8s\n",
      "  1850K .......... .......... .......... .......... ..........  4% 54,3M 7s\n",
      "  1900K .......... .......... .......... .......... ..........  5% 1,26M 8s\n",
      "  1950K .......... .......... .......... .......... ..........  5% 6,76M 8s\n",
      "  2000K .......... .......... .......... .......... ..........  5% 53,7M 8s\n",
      "  2050K .......... .......... .......... .......... ..........  5% 40,3M 7s\n",
      "  2100K .......... .......... .......... .......... ..........  5% 97,6M 7s\n",
      "  2150K .......... .......... .......... .......... ..........  5% 73,3M 7s\n",
      "  2200K .......... .......... .......... .......... ..........  5% 95,6M 7s\n",
      "  2250K .......... .......... .......... .......... ..........  5%  115M 7s\n",
      "  2300K .......... .......... .......... .......... ..........  6%  126M 7s\n",
      "  2350K .......... .......... .......... .......... ..........  6% 99,5M 7s\n",
      "  2400K .......... .......... .......... .......... ..........  6%  126M 6s\n",
      "  2450K .......... .......... .......... .......... ..........  6%  101M 6s\n",
      "  2500K .......... .......... .......... .......... ..........  6%  119M 6s\n",
      "  2550K .......... .......... .......... .......... ..........  6%  121M 6s\n",
      "  2600K .......... .......... .......... .......... ..........  6% 99,6M 6s\n",
      "  2650K .......... .......... .......... .......... ..........  6%  120M 6s\n",
      "  2700K .......... .......... .......... .......... ..........  7%  102M 6s\n",
      "  2750K .......... .......... .......... .......... ..........  7%  118M 6s\n",
      "  2800K .......... .......... .......... .......... ..........  7%  120M 5s\n",
      "  2850K .......... .......... .......... .......... ..........  7%  102M 5s\n",
      "  2900K .......... .......... .......... .......... ..........  7%  122M 5s\n",
      "  2950K .......... .......... .......... .......... ..........  7%  102M 5s\n",
      "  3000K .......... .......... .......... .......... ..........  7%  117M 5s\n",
      "  3050K .......... .......... .......... .......... ..........  7%  121M 5s\n",
      "  3100K .......... .......... .......... .......... ..........  8%  101M 5s\n",
      "  3150K .......... .......... .......... .......... ..........  8%  121M 5s\n",
      "  3200K .......... .......... .......... .......... ..........  8%  102M 5s\n",
      "  3250K .......... .......... .......... .......... ..........  8%  124M 5s\n",
      "  3300K .......... .......... .......... .......... ..........  8%  115M 5s\n",
      "  3350K .......... .......... .......... .......... ..........  8%  103M 5s\n",
      "  3400K .......... .......... .......... .......... ..........  8%  121M 4s\n",
      "  3450K .......... .......... .......... .......... ..........  9%  102M 4s\n",
      "  3500K .......... .......... .......... .......... ..........  9%  120M 4s\n",
      "  3550K .......... .......... .......... .......... ..........  9%  121M 4s\n",
      "  3600K .......... .......... .......... .......... ..........  9%  102M 4s\n",
      "  3650K .......... .......... .......... .......... ..........  9%  118M 4s\n",
      "  3700K .......... .......... .......... .......... ..........  9%  101M 4s\n",
      "  3750K .......... .......... .......... .......... ..........  9%  119M 4s\n",
      "  3800K .......... .......... .......... .......... ..........  9%  120M 4s\n",
      "  3850K .......... .......... .......... .......... .......... 10%  103M 4s\n",
      "  3900K .......... .......... .......... .......... .......... 10% 1,17M 4s\n",
      "  3950K .......... .......... .......... .......... .......... 10% 33,7M 4s\n",
      "  4000K .......... .......... .......... .......... .......... 10% 61,4M 4s\n",
      "  4050K .......... .......... .......... .......... .......... 10%  117M 4s\n",
      "  4100K .......... .......... .......... .......... .......... 10%  101M 4s\n",
      "  4150K .......... .......... .......... .......... .......... 10%  121M 4s\n",
      "  4200K .......... .......... .......... .......... .......... 10% 72,0M 4s\n",
      "  4250K .......... .......... .......... .......... .......... 11%  133M 4s\n",
      "  4300K .......... .......... .......... .......... .......... 11%  147M 4s\n",
      "  4350K .......... .......... .......... .......... .......... 11%  124M 4s\n",
      "  4400K .......... .......... .......... .......... .......... 11%  120M 4s\n",
      "  4450K .......... .......... .......... .......... .......... 11%  103M 4s\n",
      "  4500K .......... .......... .......... .......... .......... 11%  120M 4s\n",
      "  4550K .......... .......... .......... .......... .......... 11%  115M 4s\n",
      "  4600K .......... .......... .......... .......... .......... 11%  106M 4s\n",
      "  4650K .......... .......... .......... .......... .......... 12%  289K 5s\n",
      "  4700K .......... .......... .......... .......... .......... 12% 96,2M 5s\n",
      "  4750K .......... .......... .......... .......... .......... 12%  105M 5s\n",
      "  4800K .......... .......... .......... .......... .......... 12%  103M 5s\n",
      "  4850K .......... .......... .......... .......... .......... 12%  109M 5s\n",
      "  4900K .......... .......... .......... .......... .......... 12%  123M 5s\n",
      "  4950K .......... .......... .......... .......... .......... 12% 81,9M 5s\n",
      "  5000K .......... .......... .......... .......... .......... 12%  138M 4s\n",
      "  5050K .......... .......... .......... .......... .......... 13%  132M 4s\n",
      "  5100K .......... .......... .......... .......... .......... 13%  108M 4s\n",
      "  5150K .......... .......... .......... .......... .......... 13%  117M 4s\n",
      "  5200K .......... .......... .......... .......... .......... 13% 99,9M 4s\n",
      "  5250K .......... .......... .......... .......... .......... 13%  122M 4s\n",
      "  5300K .......... .......... .......... .......... .......... 13%  118M 4s\n",
      "  5350K .......... .......... .......... .......... .......... 13%  104M 4s\n",
      "  5400K .......... .......... .......... .......... .......... 14%  109M 4s\n",
      "  5450K .......... .......... .......... .......... .......... 14%  113M 4s\n",
      "  5500K .......... .......... .......... .......... .......... 14%  111M 4s\n",
      "  5550K .......... .......... .......... .......... .......... 14%  128M 4s\n",
      "  5600K .......... .......... .......... .......... .......... 14% 4,68M 4s\n",
      "  5650K .......... .......... .......... .......... .......... 14%  123M 4s\n",
      "  5700K .......... .......... .......... .......... .......... 14%  103M 4s\n",
      "  5750K .......... .......... .......... .......... .......... 14%  118M 4s\n",
      "  5800K .......... .......... .......... .......... .......... 15%  117M 4s\n",
      "  5850K .......... .......... .......... .......... .......... 15% 5,38M 4s\n",
      "  5900K .......... .......... .......... .......... .......... 15%  121M 4s\n",
      "  5950K .......... .......... .......... .......... .......... 15%  104M 4s\n",
      "  6000K .......... .......... .......... .......... .......... 15%  117M 4s\n",
      "  6050K .......... .......... .......... .......... .......... 15%  118M 4s\n",
      "  6100K .......... .......... .......... .......... .......... 15%  100M 4s\n",
      "  6150K .......... .......... .......... .......... .......... 15%  124M 4s\n",
      "  6200K .......... .......... .......... .......... .......... 16%  104M 4s\n",
      "  6250K .......... .......... .......... .......... .......... 16% 82,1M 4s\n",
      "  6300K .......... .......... .......... .......... .......... 16%  147M 4s\n",
      "  6350K .......... .......... .......... .......... .......... 16% 2,10M 4s\n",
      "  6400K .......... .......... .......... .......... .......... 16%  119M 4s\n",
      "  6450K .......... .......... .......... .......... .......... 16%  111M 4s\n",
      "  6500K .......... .......... .......... .......... .......... 16%  117M 4s\n",
      "  6550K .......... .......... .......... .......... .......... 16%  122M 4s\n",
      "  6600K .......... .......... .......... .......... .......... 17%  105M 3s\n",
      "  6650K .......... .......... .......... .......... .......... 17%  116M 3s\n",
      "  6700K .......... .......... .......... .......... .......... 17%  102M 3s\n",
      "  6750K .......... .......... .......... .......... .......... 17%  115M 3s\n",
      "  6800K .......... .......... .......... .......... .......... 17%  124M 3s\n",
      "  6850K .......... .......... .......... .......... .......... 17%  102M 3s\n",
      "  6900K .......... .......... .......... .......... .......... 17%  118M 3s\n",
      "  6950K .......... .......... .......... .......... .......... 18%  102M 3s\n",
      "  7000K .......... .......... .......... .......... .......... 18%  119M 3s\n",
      "  7050K .......... .......... .......... .......... .......... 18%  116M 3s\n",
      "  7100K .......... .......... .......... .......... .......... 18%  101M 3s\n",
      "  7150K .......... .......... .......... .......... .......... 18%  123M 3s\n",
      "  7200K .......... .......... .......... .......... .......... 18%  102M 3s\n",
      "  7250K .......... .......... .......... .......... .......... 18%  117M 3s\n",
      "  7300K .......... .......... .......... .......... .......... 18%  118M 3s\n",
      "  7350K .......... .......... .......... .......... .......... 19%  104M 3s\n",
      "  7400K .......... .......... .......... .......... .......... 19%  115M 3s\n",
      "  7450K .......... .......... .......... .......... .......... 19%  107M 3s\n",
      "  7500K .......... .......... .......... .......... .......... 19%  116M 3s\n",
      "  7550K .......... .......... .......... .......... .......... 19%  120M 3s\n",
      "  7600K .......... .......... .......... .......... .......... 19% 3,37M 3s\n",
      "  7650K .......... .......... .......... .......... .......... 19%  121M 3s\n",
      "  7700K .......... .......... .......... .......... .......... 19%  102M 3s\n",
      "  7750K .......... .......... .......... .......... .......... 20%  119M 3s\n",
      "  7800K .......... .......... .......... .......... .......... 20%  121M 3s\n",
      "  7850K .......... .......... .......... .......... .......... 20%  100M 3s\n",
      "  7900K .......... .......... .......... .......... .......... 20%  122M 3s\n",
      "  7950K .......... .......... .......... .......... .......... 20%  103M 3s\n",
      "  8000K .......... .......... .......... .......... .......... 20%  119M 3s\n",
      "  8050K .......... .......... .......... .......... .......... 20%  119M 3s\n",
      "  8100K .......... .......... .......... .......... .......... 20%  102M 3s\n",
      "  8150K .......... .......... .......... .......... .......... 21%  117M 3s\n",
      "  8200K .......... .......... .......... .......... .......... 21% 11,0M 3s\n",
      "  8250K .......... .......... .......... .......... .......... 21%  118M 3s\n",
      "  8300K .......... .......... .......... .......... .......... 21%  116M 3s\n",
      "  8350K .......... .......... .......... .......... .......... 21%  103M 3s\n",
      "  8400K .......... .......... .......... .......... .......... 21%  120M 3s\n",
      "  8450K .......... .......... .......... .......... .......... 21%  101M 3s\n",
      "  8500K .......... .......... .......... .......... .......... 22%  120M 3s\n",
      "  8550K .......... .......... .......... .......... .......... 22%  120M 3s\n",
      "  8600K .......... .......... .......... .......... .......... 22%  103M 3s\n",
      "  8650K .......... .......... .......... .......... .......... 22%  119M 3s\n",
      "  8700K .......... .......... .......... .......... .......... 22% 22,4M 3s\n",
      "  8750K .......... .......... .......... .......... .......... 22%  117M 3s\n",
      "  8800K .......... .......... .......... .......... .......... 22%  121M 3s\n",
      "  8850K .......... .......... .......... .......... .......... 22%  101M 3s\n",
      "  8900K .......... .......... .......... .......... .......... 23%  120M 3s\n",
      "  8950K .......... .......... .......... .......... .......... 23%  103M 3s\n",
      "  9000K .......... .......... .......... .......... .......... 23% 3,72M 3s\n",
      "  9050K .......... .......... .......... .......... .......... 23%  124M 3s\n",
      "  9100K .......... .......... .......... .......... .......... 23%  102M 3s\n",
      "  9150K .......... .......... .......... .......... .......... 23%  119M 3s\n",
      "  9200K .......... .......... .......... .......... .......... 23% 33,5M 2s\n",
      "  9250K .......... .......... .......... .......... .......... 23%  115M 2s\n",
      "  9300K .......... .......... .......... .......... .......... 24%  127M 2s\n",
      "  9350K .......... .......... .......... .......... .......... 24% 99,6M 2s\n",
      "  9400K .......... .......... .......... .......... .......... 24% 82,9M 2s\n",
      "  9450K .......... .......... .......... .......... .......... 24% 39,1M 2s\n",
      "  9500K .......... .......... .......... .......... .......... 24% 59,7M 2s\n",
      "  9550K .......... .......... .......... .......... .......... 24% 59,6M 2s\n",
      "  9600K .......... .......... .......... .......... .......... 24%  101M 2s\n",
      "  9650K .......... .......... .......... .......... .......... 24%  124M 2s\n",
      "  9700K .......... .......... .......... .......... .......... 25%  100M 2s\n",
      "  9750K .......... .......... .......... .......... .......... 25%  122M 2s\n",
      "  9800K .......... .......... .......... .......... .......... 25%  120M 2s\n",
      "  9850K .......... .......... .......... .......... .......... 25% 1,22M 2s\n",
      "  9900K .......... .......... .......... .......... .......... 25%  136M 2s\n",
      "  9950K .......... .......... .......... .......... .......... 25%  116M 2s\n",
      " 10000K .......... .......... .......... .......... .......... 25%  108M 2s\n",
      " 10050K .......... .......... .......... .......... .......... 25%  127M 2s\n",
      " 10100K .......... .......... .......... .......... .......... 26% 91,6M 2s\n",
      " 10150K .......... .......... .......... .......... .......... 26%  135M 2s\n",
      " 10200K .......... .......... .......... .......... .......... 26% 93,2M 2s\n",
      " 10250K .......... .......... .......... .......... .......... 26%  142M 2s\n",
      " 10300K .......... .......... .......... .......... .......... 26%  116M 2s\n",
      " 10350K .......... .......... .......... .......... .......... 26% 80,4M 2s\n",
      " 10400K .......... .......... .......... .......... .......... 26%  143M 2s\n",
      " 10450K .......... .......... .......... .......... .......... 27%  114M 2s\n",
      " 10500K .......... .......... .......... .......... .......... 27%  124M 2s\n",
      " 10550K .......... .......... .......... .......... .......... 27%  119M 2s\n",
      " 10600K .......... .......... .......... .......... .......... 27%  100M 2s\n",
      " 10650K .......... .......... .......... .......... .......... 27%  119M 2s\n",
      " 10700K .......... .......... .......... .......... .......... 27%  105M 2s\n",
      " 10750K .......... .......... .......... .......... .......... 27%  114M 2s\n",
      " 10800K .......... .......... .......... .......... .......... 27%  121M 2s\n",
      " 10850K .......... .......... .......... .......... .......... 28%  101M 2s\n",
      " 10900K .......... .......... .......... .......... .......... 28%  122M 2s\n",
      " 10950K .......... .......... .......... .......... .......... 28%  103M 2s\n",
      " 11000K .......... .......... .......... .......... .......... 28%  114M 2s\n",
      " 11050K .......... .......... .......... .......... .......... 28%  120M 2s\n",
      " 11100K .......... .......... .......... .......... .......... 28%  104M 2s\n",
      " 11150K .......... .......... .......... .......... .......... 28%  116M 2s\n",
      " 11200K .......... .......... .......... .......... .......... 28%  102M 2s\n",
      " 11250K .......... .......... .......... .......... .......... 29%  118M 2s\n",
      " 11300K .......... .......... .......... .......... .......... 29%  124M 2s\n",
      " 11350K .......... .......... .......... .......... .......... 29%  101M 2s\n",
      " 11400K .......... .......... .......... .......... .......... 29%  117M 2s\n",
      " 11450K .......... .......... .......... .......... .......... 29% 99,8M 2s\n",
      " 11500K .......... .......... .......... .......... .......... 29%  124M 2s\n",
      " 11550K .......... .......... .......... .......... .......... 29%  119M 2s\n",
      " 11600K .......... .......... .......... .......... .......... 29%  100M 2s\n",
      " 11650K .......... .......... .......... .......... .......... 30%  120M 2s\n",
      " 11700K .......... .......... .......... .......... .......... 30%  102M 2s\n",
      " 11750K .......... .......... .......... .......... .......... 30%  118M 2s\n",
      " 11800K .......... .......... .......... .......... .......... 30%  122M 2s\n",
      " 11850K .......... .......... .......... .......... .......... 30%  102M 2s\n",
      " 11900K .......... .......... .......... .......... .......... 30%  120M 2s\n",
      " 11950K .......... .......... .......... .......... .......... 30%  102M 2s\n",
      " 12000K .......... .......... .......... .......... .......... 31%  116M 2s\n",
      " 12050K .......... .......... .......... .......... .......... 31%  122M 2s\n",
      " 12100K .......... .......... .......... .......... .......... 31%  102M 2s\n",
      " 12150K .......... .......... .......... .......... .......... 31%  117M 2s\n",
      " 12200K .......... .......... .......... .......... .......... 31% 4,97M 2s\n",
      " 12250K .......... .......... .......... .......... .......... 31%  116M 2s\n",
      " 12300K .......... .......... .......... .......... .......... 31%  130M 2s\n",
      " 12350K .......... .......... .......... .......... .......... 31% 2,19M 2s\n",
      " 12400K .......... .......... .......... .......... .......... 32% 5,08M 2s\n",
      " 12450K .......... .......... .......... .......... .......... 32% 98,7M 2s\n",
      " 12500K .......... .......... .......... .......... .......... 32% 4,73M 2s\n",
      " 12550K .......... .......... .......... .......... .......... 32%  138M 2s\n",
      " 12600K .......... .......... .......... .......... .......... 32% 99,2M 2s\n",
      " 12650K .......... .......... .......... .......... .......... 32%  116M 2s\n",
      " 12700K .......... .......... .......... .......... .......... 32%  104M 2s\n",
      " 12750K .......... .......... .......... .......... .......... 32%  119M 2s\n",
      " 12800K .......... .......... .......... .......... .......... 33%  118M 2s\n",
      " 12850K .......... .......... .......... .......... .......... 33%  104M 2s\n",
      " 12900K .......... .......... .......... .......... .......... 33%  120M 2s\n",
      " 12950K .......... .......... .......... .......... .......... 33%  101M 2s\n",
      " 13000K .......... .......... .......... .......... .......... 33%  121M 2s\n",
      " 13050K .......... .......... .......... .......... .......... 33%  117M 2s\n",
      " 13100K .......... .......... .......... .......... .......... 33%  101M 2s\n",
      " 13150K .......... .......... .......... .......... .......... 33%  120M 2s\n",
      " 13200K .......... .......... .......... .......... .......... 34%  101M 2s\n",
      " 13250K .......... .......... .......... .......... .......... 34%  120M 2s\n",
      " 13300K .......... .......... .......... .......... .......... 34%  119M 2s\n",
      " 13350K .......... .......... .......... .......... .......... 34% 99,8M 2s\n",
      " 13400K .......... .......... .......... .......... .......... 34%  118M 2s\n",
      " 13450K .......... .......... .......... .......... .......... 34%  102M 2s\n",
      " 13500K .......... .......... .......... .......... .......... 34%  119M 2s\n",
      " 13550K .......... .......... .......... .......... .......... 34%  119M 2s\n",
      " 13600K .......... .......... .......... .......... .......... 35%  102M 2s\n",
      " 13650K .......... .......... .......... .......... .......... 35%  120M 2s\n",
      " 13700K .......... .......... .......... .......... .......... 35%  102M 2s\n",
      " 13750K .......... .......... .......... .......... .......... 35%  120M 2s\n",
      " 13800K .......... .......... .......... .......... .......... 35%  119M 2s\n",
      " 13850K .......... .......... .......... .......... .......... 35%  100M 2s\n",
      " 13900K .......... .......... .......... .......... .......... 35%  123M 2s\n",
      " 13950K .......... .......... .......... .......... .......... 36%  101M 2s\n",
      " 14000K .......... .......... .......... .......... .......... 36%  114M 2s\n",
      " 14050K .......... .......... .......... .......... .......... 36%  121M 2s\n",
      " 14100K .......... .......... .......... .......... .......... 36%  103M 2s\n",
      " 14150K .......... .......... .......... .......... .......... 36%  118M 2s\n",
      " 14200K .......... .......... .......... .......... .......... 36% 98,0M 2s\n",
      " 14250K .......... .......... .......... .......... .......... 36%  123M 2s\n",
      " 14300K .......... .......... .......... .......... .......... 36%  121M 2s\n",
      " 14350K .......... .......... .......... .......... .......... 37%  104M 2s\n",
      " 14400K .......... .......... .......... .......... .......... 37%  118M 2s\n",
      " 14450K .......... .......... .......... .......... .......... 37%  103M 2s\n",
      " 14500K .......... .......... .......... .......... .......... 37%  103M 2s\n",
      " 14550K .......... .......... .......... .......... .......... 37% 11,9M 2s\n",
      " 14600K .......... .......... .......... .......... .......... 37%  105M 2s\n",
      " 14650K .......... .......... .......... .......... .......... 37%  134M 2s\n",
      " 14700K .......... .......... .......... .......... .......... 37%  107M 2s\n",
      " 14750K .......... .......... .......... .......... .......... 38%  121M 1s\n",
      " 14800K .......... .......... .......... .......... .......... 38%  120M 1s\n",
      " 14850K .......... .......... .......... .......... .......... 38%  102M 1s\n",
      " 14900K .......... .......... .......... .......... .......... 38%  117M 1s\n",
      " 14950K .......... .......... .......... .......... .......... 38%  100M 1s\n",
      " 15000K .......... .......... .......... .......... .......... 38%  122M 1s\n",
      " 15050K .......... .......... .......... .......... .......... 38%  119M 1s\n",
      " 15100K .......... .......... .......... .......... .......... 38% 99,9M 1s\n",
      " 15150K .......... .......... .......... .......... .......... 39%  118M 1s\n",
      " 15200K .......... .......... .......... .......... .......... 39%  103M 1s\n",
      " 15250K .......... .......... .......... .......... .......... 39%  122M 1s\n",
      " 15300K .......... .......... .......... .......... .......... 39%  119M 1s\n",
      " 15350K .......... .......... .......... .......... .......... 39% 1,54M 1s\n",
      " 15400K .......... .......... .......... .......... .......... 39% 83,8M 1s\n",
      " 15450K .......... .......... .......... .......... .......... 39% 68,5M 1s\n",
      " 15500K .......... .......... .......... .......... .......... 40% 77,2M 1s\n",
      " 15550K .......... .......... .......... .......... .......... 40% 84,2M 1s\n",
      " 15600K .......... .......... .......... .......... .......... 40% 74,3M 1s\n",
      " 15650K .......... .......... .......... .......... .......... 40% 82,6M 1s\n",
      " 15700K .......... .......... .......... .......... .......... 40% 68,6M 1s\n",
      " 15750K .......... .......... .......... .......... .......... 40% 76,1M 1s\n",
      " 15800K .......... .......... .......... .......... .......... 40% 83,2M 1s\n",
      " 15850K .......... .......... .......... .......... .......... 40% 68,1M 1s\n",
      " 15900K .......... .......... .......... .......... .......... 41%  160M 1s\n",
      " 15950K .......... .......... .......... .......... .......... 41%  148M 1s\n",
      " 16000K .......... .......... .......... .......... .......... 41%  168M 1s\n",
      " 16050K .......... .......... .......... .......... .......... 41%  169M 1s\n",
      " 16100K .......... .......... .......... .......... .......... 41%  142M 1s\n",
      " 16150K .......... .......... .......... .......... .......... 41%  167M 1s\n",
      " 16200K .......... .......... .......... .......... .......... 41%  140M 1s\n",
      " 16250K .......... .......... .......... .......... .......... 41%  120M 1s\n",
      " 16300K .......... .......... .......... .......... .......... 42%  167M 1s\n",
      " 16350K .......... .......... .......... .......... .......... 42%  145M 1s\n",
      " 16400K .......... .......... .......... .......... .......... 42%  168M 1s\n",
      " 16450K .......... .......... .......... .......... .......... 42%  144M 1s\n",
      " 16500K .......... .......... .......... .......... .......... 42% 99,5M 1s\n",
      " 16550K .......... .......... .......... .......... .......... 42%  164M 1s\n",
      " 16600K .......... .......... .......... .......... .......... 42%  141M 1s\n",
      " 16650K .......... .......... .......... .......... .......... 42%  163M 1s\n",
      " 16700K .......... .......... .......... .......... .......... 43%  120M 1s\n",
      " 16750K .......... .......... .......... .......... .......... 43%  123M 1s\n",
      " 16800K .......... .......... .......... .......... .......... 43%  117M 1s\n",
      " 16850K .......... .......... .......... .......... .......... 43%  102M 1s\n",
      " 16900K .......... .......... .......... .......... .......... 43%  121M 1s\n",
      " 16950K .......... .......... .......... .......... .......... 43% 99,4M 1s\n",
      " 17000K .......... .......... .......... .......... .......... 43%  121M 1s\n",
      " 17050K .......... .......... .......... .......... .......... 44%  122M 1s\n",
      " 17100K .......... .......... .......... .......... .......... 44%  104M 1s\n",
      " 17150K .......... .......... .......... .......... .......... 44%  118M 1s\n",
      " 17200K .......... .......... .......... .......... .......... 44%  102M 1s\n",
      " 17250K .......... .......... .......... .......... .......... 44%  121M 1s\n",
      " 17300K .......... .......... .......... .......... .......... 44%  120M 1s\n",
      " 17350K .......... .......... .......... .......... .......... 44% 67,0M 1s\n",
      " 17400K .......... .......... .......... .......... .......... 44%  139M 1s\n",
      " 17450K .......... .......... .......... .......... .......... 45%  144M 1s\n",
      " 17500K .......... .......... .......... .......... .......... 45%  138M 1s\n",
      " 17550K .......... .......... .......... .......... .......... 45%  113M 1s\n",
      " 17600K .......... .......... .......... .......... .......... 45% 14,9M 1s\n",
      " 17650K .......... .......... .......... .......... .......... 45%  125M 1s\n",
      " 17700K .......... .......... .......... .......... .......... 45% 52,9M 1s\n",
      " 17750K .......... .......... .......... .......... .......... 45% 53,4M 1s\n",
      " 17800K .......... .......... .......... .......... .......... 45%  146M 1s\n",
      " 17850K .......... .......... .......... .......... .......... 46%  102M 1s\n",
      " 17900K .......... .......... .......... .......... .......... 46%  120M 1s\n",
      " 17950K .......... .......... .......... .......... .......... 46%  104M 1s\n",
      " 18000K .......... .......... .......... .......... .......... 46%  117M 1s\n",
      " 18050K .......... .......... .......... .......... .......... 46%  121M 1s\n",
      " 18100K .......... .......... .......... .......... .......... 46%  103M 1s\n",
      " 18150K .......... .......... .......... .......... .......... 46%  120M 1s\n",
      " 18200K .......... .......... .......... .......... .......... 46%  103M 1s\n",
      " 18250K .......... .......... .......... .......... .......... 47%  117M 1s\n",
      " 18300K .......... .......... .......... .......... .......... 47%  121M 1s\n",
      " 18350K .......... .......... .......... .......... .......... 47% 4,87M 1s\n",
      " 18400K .......... .......... .......... .......... .......... 47% 2,36M 1s\n",
      " 18450K .......... .......... .......... .......... .......... 47% 27,0M 1s\n",
      " 18500K .......... .......... .......... .......... .......... 47%  111M 1s\n",
      " 18550K .......... .......... .......... .......... .......... 47% 89,6M 1s\n",
      " 18600K .......... .......... .......... .......... .......... 47%  105M 1s\n",
      " 18650K .......... .......... .......... .......... .......... 48%  123M 1s\n",
      " 18700K .......... .......... .......... .......... .......... 48%  105M 1s\n",
      " 18750K .......... .......... .......... .......... .......... 48%  119M 1s\n",
      " 18800K .......... .......... .......... .......... .......... 48%  118M 1s\n",
      " 18850K .......... .......... .......... .......... .......... 48% 57,8M 1s\n",
      " 18900K .......... .......... .......... .......... .......... 48%  121M 1s\n",
      " 18950K .......... .......... .......... .......... .......... 48% 59,3M 1s\n",
      " 19000K .......... .......... .......... .......... .......... 49%  115M 1s\n",
      " 19050K .......... .......... .......... .......... .......... 49%  125M 1s\n",
      " 19100K .......... .......... .......... .......... .......... 49% 99,1M 1s\n",
      " 19150K .......... .......... .......... .......... .......... 49%  122M 1s\n",
      " 19200K .......... .......... .......... .......... .......... 49%  103M 1s\n",
      " 19250K .......... .......... .......... .......... .......... 49%  118M 1s\n",
      " 19300K .......... .......... .......... .......... .......... 49%  123M 1s\n",
      " 19350K .......... .......... .......... .......... .......... 49%  102M 1s\n",
      " 19400K .......... .......... .......... .......... .......... 50%  118M 1s\n",
      " 19450K .......... .......... .......... .......... .......... 50%  100M 1s\n",
      " 19500K .......... .......... .......... .......... .......... 50%  121M 1s\n",
      " 19550K .......... .......... .......... .......... .......... 50%  119M 1s\n",
      " 19600K .......... .......... .......... .......... .......... 50%  101M 1s\n",
      " 19650K .......... .......... .......... .......... .......... 50%  123M 1s\n",
      " 19700K .......... .......... .......... .......... .......... 50%  104M 1s\n",
      " 19750K .......... .......... .......... .......... .......... 50%  118M 1s\n",
      " 19800K .......... .......... .......... .......... .......... 51%  114M 1s\n",
      " 19850K .......... .......... .......... .......... .......... 51%  105M 1s\n",
      " 19900K .......... .......... .......... .......... .......... 51%  121M 1s\n",
      " 19950K .......... .......... .......... .......... .......... 51% 2,06M 1s\n",
      " 20000K .......... .......... .......... .......... .......... 51% 62,3M 1s\n",
      " 20050K .......... .......... .......... .......... .......... 51% 62,2M 1s\n",
      " 20100K .......... .......... .......... .......... .......... 51% 55,9M 1s\n",
      " 20150K .......... .......... .......... .......... .......... 51% 69,0M 1s\n",
      " 20200K .......... .......... .......... .......... .......... 52% 53,8M 1s\n",
      " 20250K .......... .......... .......... .......... .......... 52% 57,4M 1s\n",
      " 20300K .......... .......... .......... .......... .......... 52% 65,5M 1s\n",
      " 20350K .......... .......... .......... .......... .......... 52% 96,4M 1s\n",
      " 20400K .......... .......... .......... .......... .......... 52%  113M 1s\n",
      " 20450K .......... .......... .......... .......... .......... 52% 92,8M 1s\n",
      " 20500K .......... .......... .......... .......... .......... 52%  108M 1s\n",
      " 20550K .......... .......... .......... .......... .......... 53%  107M 1s\n",
      " 20600K .......... .......... .......... .......... .......... 53% 66,5M 1s\n",
      " 20650K .......... .......... .......... .......... .......... 53%  123M 1s\n",
      " 20700K .......... .......... .......... .......... .......... 53%  103M 1s\n",
      " 20750K .......... .......... .......... .......... .......... 53%  153M 1s\n",
      " 20800K .......... .......... .......... .......... .......... 53%  160M 1s\n",
      " 20850K .......... .......... .......... .......... .......... 53%  137M 1s\n",
      " 20900K .......... .......... .......... .......... .......... 53%  158M 1s\n",
      " 20950K .......... .......... .......... .......... .......... 54%  135M 1s\n",
      " 21000K .......... .......... .......... .......... .......... 54%  148M 1s\n",
      " 21050K .......... .......... .......... .......... .......... 54%  155M 1s\n",
      " 21100K .......... .......... .......... .......... .......... 54%  156M 1s\n",
      " 21150K .......... .......... .......... .......... .......... 54%  316M 1s\n",
      " 21200K .......... .......... .......... .......... .......... 54%  268M 1s\n",
      " 21250K .......... .......... .......... .......... .......... 54%  310M 1s\n",
      " 21300K .......... .......... .......... .......... .......... 54%  317M 1s\n",
      " 21350K .......... .......... .......... .......... .......... 55%  259M 1s\n",
      " 21400K .......... .......... .......... .......... .......... 55%  306M 1s\n",
      " 21450K .......... .......... .......... .......... .......... 55% 4,97M 1s\n",
      " 21500K .......... .......... .......... .......... .......... 55% 43,8M 1s\n",
      " 21550K .......... .......... .......... .......... .......... 55% 91,6M 1s\n",
      " 21600K .......... .......... .......... .......... .......... 55% 91,4M 1s\n",
      " 21650K .......... .......... .......... .......... .......... 55%  105M 1s\n",
      " 21700K .......... .......... .......... .......... .......... 55% 91,1M 1s\n",
      " 21750K .......... .......... .......... .......... .......... 56%  111M 1s\n",
      " 21800K .......... .......... .......... .......... .......... 56%  125M 1s\n",
      " 21850K .......... .......... .......... .......... .......... 56% 97,9M 1s\n",
      " 21900K .......... .......... .......... .......... .......... 56%  130M 1s\n",
      " 21950K .......... .......... .......... .......... .......... 56%  116M 1s\n",
      " 22000K .......... .......... .......... .......... .......... 56%  129M 1s\n",
      " 22050K .......... .......... .......... .......... .......... 56%  138M 1s\n",
      " 22100K .......... .......... .......... .......... .......... 56%  125M 1s\n",
      " 22150K .......... .......... .......... .......... .......... 57%  161M 1s\n",
      " 22200K .......... .......... .......... .......... .......... 57%  116M 1s\n",
      " 22250K .......... .......... .......... .......... .......... 57%  116M 1s\n",
      " 22300K .......... .......... .......... .......... .......... 57%  120M 1s\n",
      " 22350K .......... .......... .......... .......... .......... 57% 95,5M 1s\n",
      " 22400K .......... .......... .......... .......... .......... 57%  137M 1s\n",
      " 22450K .......... .......... .......... .......... .......... 57%  101M 1s\n",
      " 22500K .......... .......... .......... .......... .......... 58% 88,6M 1s\n",
      " 22550K .......... .......... .......... .......... .......... 58%  149M 1s\n",
      " 22600K .......... .......... .......... .......... .......... 58%  106M 1s\n",
      " 22650K .......... .......... .......... .......... .......... 58%  132M 1s\n",
      " 22700K .......... .......... .......... .......... .......... 58%  103M 1s\n",
      " 22750K .......... .......... .......... .......... .......... 58%  120M 1s\n",
      " 22800K .......... .......... .......... .......... .......... 58%  117M 1s\n",
      " 22850K .......... .......... .......... .......... .......... 58%  103M 1s\n",
      " 22900K .......... .......... .......... .......... .......... 59%  120M 1s\n",
      " 22950K .......... .......... .......... .......... .......... 59% 3,50M 1s\n",
      " 23000K .......... .......... .......... .......... .......... 59% 4,92M 1s\n",
      " 23050K .......... .......... .......... .......... .......... 59% 30,7M 1s\n",
      " 23100K .......... .......... .......... .......... .......... 59%  103M 1s\n",
      " 23150K .......... .......... .......... .......... .......... 59% 57,5M 1s\n",
      " 23200K .......... .......... .......... .......... .......... 59% 64,9M 1s\n",
      " 23250K .......... .......... .......... .......... .......... 59%  119M 1s\n",
      " 23300K .......... .......... .......... .......... .......... 60% 66,9M 1s\n",
      " 23350K .......... .......... .......... .......... .......... 60% 47,7M 1s\n",
      " 23400K .......... .......... .......... .......... .......... 60%  114M 1s\n",
      " 23450K .......... .......... .......... .......... .......... 60%  104M 1s\n",
      " 23500K .......... .......... .......... .......... .......... 60%  121M 1s\n",
      " 23550K .......... .......... .......... .......... .......... 60%  124M 1s\n",
      " 23600K .......... .......... .......... .......... .......... 60% 54,0M 1s\n",
      " 23650K .......... .......... .......... .......... .......... 60%  118M 1s\n",
      " 23700K .......... .......... .......... .......... .......... 61%  102M 1s\n",
      " 23750K .......... .......... .......... .......... .......... 61%  121M 1s\n",
      " 23800K .......... .......... .......... .......... .......... 61%  117M 1s\n",
      " 23850K .......... .......... .......... .......... .......... 61%  102M 1s\n",
      " 23900K .......... .......... .......... .......... .......... 61%  121M 1s\n",
      " 23950K .......... .......... .......... .......... .......... 61%  100M 1s\n",
      " 24000K .......... .......... .......... .......... .......... 61%  124M 1s\n",
      " 24050K .......... .......... .......... .......... .......... 62%  117M 1s\n",
      " 24100K .......... .......... .......... .......... .......... 62%  102M 1s\n",
      " 24150K .......... .......... .......... .......... .......... 62%  121M 1s\n",
      " 24200K .......... .......... .......... .......... .......... 62%  102M 1s\n",
      " 24250K .......... .......... .......... .......... .......... 62% 7,04M 1s\n",
      " 24300K .......... .......... .......... .......... .......... 62%  121M 1s\n",
      " 24350K .......... .......... .......... .......... .......... 62%  100M 1s\n",
      " 24400K .......... .......... .......... .......... .......... 62%  118M 1s\n",
      " 24450K .......... .......... .......... .......... .......... 63%  102M 1s\n",
      " 24500K .......... .......... .......... .......... .......... 63% 49,2M 1s\n",
      " 24550K .......... .......... .......... .......... .......... 63% 41,9M 1s\n",
      " 24600K .......... .......... .......... .......... .......... 63% 51,1M 1s\n",
      " 24650K .......... .......... .......... .......... .......... 63% 66,4M 1s\n",
      " 24700K .......... .......... .......... .......... .......... 63%  103M 1s\n",
      " 24750K .......... .......... .......... .......... .......... 63% 52,6M 1s\n",
      " 24800K .......... .......... .......... .......... .......... 63%  117M 1s\n",
      " 24850K .......... .......... .......... .......... .......... 64%  102M 1s\n",
      " 24900K .......... .......... .......... .......... .......... 64%  121M 1s\n",
      " 24950K .......... .......... .......... .......... .......... 64%  102M 1s\n",
      " 25000K .......... .......... .......... .......... .......... 64%  117M 1s\n",
      " 25050K .......... .......... .......... .......... .......... 64%  121M 1s\n",
      " 25100K .......... .......... .......... .......... .......... 64%  102M 1s\n",
      " 25150K .......... .......... .......... .......... .......... 64%  121M 1s\n",
      " 25200K .......... .......... .......... .......... .......... 64%  102M 1s\n",
      " 25250K .......... .......... .......... .......... .......... 65%  118M 1s\n",
      " 25300K .......... .......... .......... .......... .......... 65%  121M 1s\n",
      " 25350K .......... .......... .......... .......... .......... 65%  102M 1s\n",
      " 25400K .......... .......... .......... .......... .......... 65%  117M 1s\n",
      " 25450K .......... .......... .......... .......... .......... 65%  102M 1s\n",
      " 25500K .......... .......... .......... .......... .......... 65%  121M 1s\n",
      " 25550K .......... .......... .......... .......... .......... 65%  121M 1s\n",
      " 25600K .......... .......... .......... .......... .......... 66%  102M 1s\n",
      " 25650K .......... .......... .......... .......... .......... 66%  114M 1s\n",
      " 25700K .......... .......... .......... .......... .......... 66% 97,0M 1s\n",
      " 25750K .......... .......... .......... .......... .......... 66%  121M 1s\n",
      " 25800K .......... .......... .......... .......... .......... 66%  117M 1s\n",
      " 25850K .......... .......... .......... .......... .......... 66%  102M 1s\n",
      " 25900K .......... .......... .......... .......... .......... 66%  121M 1s\n",
      " 25950K .......... .......... .......... .......... .......... 66%  102M 1s\n",
      " 26000K .......... .......... .......... .......... .......... 67%  117M 1s\n",
      " 26050K .......... .......... .......... .......... .......... 67% 2,16M 1s\n",
      " 26100K .......... .......... .......... .......... .......... 67% 27,2M 1s\n",
      " 26150K .......... .......... .......... .......... .......... 67%  121M 1s\n",
      " 26200K .......... .......... .......... .......... .......... 67% 54,0M 1s\n",
      " 26250K .......... .......... .......... .......... .......... 67%  115M 1s\n",
      " 26300K .......... .......... .......... .......... .......... 67%  118M 1s\n",
      " 26350K .......... .......... .......... .......... .......... 67%  106M 1s\n",
      " 26400K .......... .......... .......... .......... .......... 68% 61,4M 1s\n",
      " 26450K .......... .......... .......... .......... .......... 68%  103M 1s\n",
      " 26500K .......... .......... .......... .......... .......... 68%  121M 1s\n",
      " 26550K .......... .......... .......... .......... .......... 68%  118M 1s\n",
      " 26600K .......... .......... .......... .......... .......... 68%  104M 1s\n",
      " 26650K .......... .......... .......... .......... .......... 68%  121M 1s\n",
      " 26700K .......... .......... .......... .......... .......... 68% 75,8M 1s\n",
      " 26750K .......... .......... .......... .......... .......... 68%  174M 1s\n",
      " 26800K .......... .......... .......... .......... .......... 69% 92,4M 1s\n",
      " 26850K .......... .......... .......... .......... .......... 69%  102M 1s\n",
      " 26900K .......... .......... .......... .......... .......... 69% 74,1M 1s\n",
      " 26950K .......... .......... .......... .......... .......... 69%  102M 1s\n",
      " 27000K .......... .......... .......... .......... .......... 69%  117M 1s\n",
      " 27050K .......... .......... .......... .......... .......... 69%  122M 1s\n",
      " 27100K .......... .......... .......... .......... .......... 69%  102M 1s\n",
      " 27150K .......... .......... .......... .......... .......... 69%  117M 1s\n",
      " 27200K .......... .......... .......... .......... .......... 70%  102M 1s\n",
      " 27250K .......... .......... .......... .......... .......... 70%  121M 1s\n",
      " 27300K .......... .......... .......... .......... .......... 70% 7,58M 1s\n",
      " 27350K .......... .......... .......... .......... .......... 70% 22,3M 1s\n",
      " 27400K .......... .......... .......... .......... .......... 70%  119M 0s\n",
      " 27450K .......... .......... .......... .......... .......... 70%  101M 0s\n",
      " 27500K .......... .......... .......... .......... .......... 70%  123M 0s\n",
      " 27550K .......... .......... .......... .......... .......... 71%  118M 0s\n",
      " 27600K .......... .......... .......... .......... .......... 71%  102M 0s\n",
      " 27650K .......... .......... .......... .......... .......... 71% 27,5M 0s\n",
      " 27700K .......... .......... .......... .......... .......... 71%  102M 0s\n",
      " 27750K .......... .......... .......... .......... .......... 71% 75,0M 0s\n",
      " 27800K .......... .......... .......... .......... .......... 71%  118M 0s\n",
      " 27850K .......... .......... .......... .......... .......... 71% 60,3M 0s\n",
      " 27900K .......... .......... .......... .......... .......... 71%  122M 0s\n",
      " 27950K .......... .......... .......... .......... .......... 72%  101M 0s\n",
      " 28000K .......... .......... .......... .......... .......... 72%  119M 0s\n",
      " 28050K .......... .......... .......... .......... .......... 72%  120M 0s\n",
      " 28100K .......... .......... .......... .......... .......... 72%  103M 0s\n",
      " 28150K .......... .......... .......... .......... .......... 72%  117M 0s\n",
      " 28200K .......... .......... .......... .......... .......... 72%  102M 0s\n",
      " 28250K .......... .......... .......... .......... .......... 72%  121M 0s\n",
      " 28300K .......... .......... .......... .......... .......... 72%  121M 0s\n",
      " 28350K .......... .......... .......... .......... .......... 73%  102M 0s\n",
      " 28400K .......... .......... .......... .......... .......... 73%  117M 0s\n",
      " 28450K .......... .......... .......... .......... .......... 73%  102M 0s\n",
      " 28500K .......... .......... .......... .......... .......... 73%  121M 0s\n",
      " 28550K .......... .......... .......... .......... .......... 73%  117M 0s\n",
      " 28600K .......... .......... .......... .......... .......... 73%  103M 0s\n",
      " 28650K .......... .......... .......... .......... .......... 73%  121M 0s\n",
      " 28700K .......... .......... .......... .......... .......... 73%  102M 0s\n",
      " 28750K .......... .......... .......... .......... .......... 74%  117M 0s\n",
      " 28800K .......... .......... .......... .......... .......... 74%  121M 0s\n",
      " 28850K .......... .......... .......... .......... .......... 74%  102M 0s\n",
      " 28900K .......... .......... .......... .......... .......... 74%  121M 0s\n",
      " 28950K .......... .......... .......... .......... .......... 74%  102M 0s\n",
      " 29000K .......... .......... .......... .......... .......... 74%  118M 0s\n",
      " 29050K .......... .......... .......... .......... .......... 74%  120M 0s\n",
      " 29100K .......... .......... .......... .......... .......... 75% 2,27M 0s\n",
      " 29150K .......... .......... .......... .......... .......... 75% 21,5M 0s\n",
      " 29200K .......... .......... .......... .......... .......... 75% 25,3M 0s\n",
      " 29250K .......... .......... .......... .......... .......... 75%  121M 0s\n",
      " 29300K .......... .......... .......... .......... .......... 75%  119M 0s\n",
      " 29350K .......... .......... .......... .......... .......... 75%  104M 0s\n",
      " 29400K .......... .......... .......... .......... .......... 75% 64,0M 0s\n",
      " 29450K .......... .......... .......... .......... .......... 75%  102M 0s\n",
      " 29500K .......... .......... .......... .......... .......... 76%  120M 0s\n",
      " 29550K .......... .......... .......... .......... .......... 76%  118M 0s\n",
      " 29600K .......... .......... .......... .......... .......... 76%  102M 0s\n",
      " 29650K .......... .......... .......... .......... .......... 76%  123M 0s\n",
      " 29700K .......... .......... .......... .......... .......... 76%  102M 0s\n",
      " 29750K .......... .......... .......... .......... .......... 76%  117M 0s\n",
      " 29800K .......... .......... .......... .......... .......... 76%  121M 0s\n",
      " 29850K .......... .......... .......... .......... .......... 76%  102M 0s\n",
      " 29900K .......... .......... .......... .......... .......... 77%  121M 0s\n",
      " 29950K .......... .......... .......... .......... .......... 77%  103M 0s\n",
      " 30000K .......... .......... .......... .......... .......... 77%  117M 0s\n",
      " 30050K .......... .......... .......... .......... .......... 77%  120M 0s\n",
      " 30100K .......... .......... .......... .......... .......... 77%  104M 0s\n",
      " 30150K .......... .......... .......... .......... .......... 77%  117M 0s\n",
      " 30200K .......... .......... .......... .......... .......... 77%  102M 0s\n",
      " 30250K .......... .......... .......... .......... .......... 77%  122M 0s\n",
      " 30300K .......... .......... .......... .......... .......... 78%  117M 0s\n",
      " 30350K .......... .......... .......... .......... .......... 78% 8,70M 0s\n",
      " 30400K .......... .......... .......... .......... .......... 78% 15,3M 0s\n",
      " 30450K .......... .......... .......... .......... .......... 78%  104M 0s\n",
      " 30500K .......... .......... .......... .......... .......... 78%  119M 0s\n",
      " 30550K .......... .......... .......... .......... .......... 78%  116M 0s\n",
      " 30600K .......... .......... .......... .......... .......... 78%  103M 0s\n",
      " 30650K .......... .......... .......... .......... .......... 78% 70,0M 0s\n",
      " 30700K .......... .......... .......... .......... .......... 79%  102M 0s\n",
      " 30750K .......... .......... .......... .......... .......... 79% 34,7M 0s\n",
      " 30800K .......... .......... .......... .......... .......... 79% 53,4M 0s\n",
      " 30850K .......... .......... .......... .......... .......... 79%  138M 0s\n",
      " 30900K .......... .......... .......... .......... .......... 79%  116M 0s\n",
      " 30950K .......... .......... .......... .......... .......... 79%  103M 0s\n",
      " 31000K .......... .......... .......... .......... .......... 79%  122M 0s\n",
      " 31050K .......... .......... .......... .......... .......... 80%  120M 0s\n",
      " 31100K .......... .......... .......... .......... .......... 80% 85,1M 0s\n",
      " 31150K .......... .......... .......... .......... .......... 80%  117M 0s\n",
      " 31200K .......... .......... .......... .......... .......... 80%  102M 0s\n",
      " 31250K .......... .......... .......... .......... .......... 80%  103M 0s\n",
      " 31300K .......... .......... .......... .......... .......... 80%  118M 0s\n",
      " 31350K .......... .......... .......... .......... .......... 80%  102M 0s\n",
      " 31400K .......... .......... .......... .......... .......... 80%  121M 0s\n",
      " 31450K .......... .......... .......... .......... .......... 81%  102M 0s\n",
      " 31500K .......... .......... .......... .......... .......... 81%  121M 0s\n",
      " 31550K .......... .......... .......... .......... .......... 81%  117M 0s\n",
      " 31600K .......... .......... .......... .......... .......... 81%  102M 0s\n",
      " 31650K .......... .......... .......... .......... .......... 81%  122M 0s\n",
      " 31700K .......... .......... .......... .......... .......... 81%  102M 0s\n",
      " 31750K .......... .......... .......... .......... .......... 81%  117M 0s\n",
      " 31800K .......... .......... .......... .......... .......... 81%  122M 0s\n",
      " 31850K .......... .......... .......... .......... .......... 82%  102M 0s\n",
      " 31900K .......... .......... .......... .......... .......... 82%  118M 0s\n",
      " 31950K .......... .......... .......... .......... .......... 82%  103M 0s\n",
      " 32000K .......... .......... .......... .......... .......... 82%  121M 0s\n",
      " 32050K .......... .......... .......... .......... .......... 82%  121M 0s\n",
      " 32100K .......... .......... .......... .......... .......... 82% 4,53M 0s\n",
      " 32150K .......... .......... .......... .......... .......... 82% 5,64M 0s\n",
      " 32200K .......... .......... .......... .......... .......... 82% 9,35M 0s\n",
      " 32250K .......... .......... .......... .......... .......... 83% 39,4M 0s\n",
      " 32300K .......... .......... .......... .......... .......... 83% 60,3M 0s\n",
      " 32350K .......... .......... .......... .......... .......... 83% 55,7M 0s\n",
      " 32400K .......... .......... .......... .......... .......... 83%  122M 0s\n",
      " 32450K .......... .......... .......... .......... .......... 83%  103M 0s\n",
      " 32500K .......... .......... .......... .......... .......... 83%  107M 0s\n",
      " 32550K .......... .......... .......... .......... .......... 83%  134M 0s\n",
      " 32600K .......... .......... .......... .......... .......... 84%  100M 0s\n",
      " 32650K .......... .......... .......... .......... .......... 84%  122M 0s\n",
      " 32700K .......... .......... .......... .......... .......... 84%  101M 0s\n",
      " 32750K .......... .......... .......... .......... .......... 84%  111M 0s\n",
      " 32800K .......... .......... .......... .......... .......... 84%  127M 0s\n",
      " 32850K .......... .......... .......... .......... .......... 84% 68,4M 0s\n",
      " 32900K .......... .......... .......... .......... .......... 84%  141M 0s\n",
      " 32950K .......... .......... .......... .......... .......... 84% 98,7M 0s\n",
      " 33000K .......... .......... .......... .......... .......... 85%  159M 0s\n",
      " 33050K .......... .......... .......... .......... .......... 85%  149M 0s\n",
      " 33100K .......... .......... .......... .......... .......... 85%  104M 0s\n",
      " 33150K .......... .......... .......... .......... .......... 85%  123M 0s\n",
      " 33200K .......... .......... .......... .......... .......... 85%  103M 0s\n",
      " 33250K .......... .......... .......... .......... .......... 85%  120M 0s\n",
      " 33300K .......... .......... .......... .......... .......... 85%  111M 0s\n",
      " 33350K .......... .......... .......... .......... .......... 85%  108M 0s\n",
      " 33400K .......... .......... .......... .......... .......... 86% 9,60M 0s\n",
      " 33450K .......... .......... .......... .......... .......... 86% 23,6M 0s\n",
      " 33500K .......... .......... .......... .......... .......... 86% 23,9M 0s\n",
      " 33550K .......... .......... .......... .......... .......... 86%  120M 0s\n",
      " 33600K .......... .......... .......... .......... .......... 86%  102M 0s\n",
      " 33650K .......... .......... .......... .......... .......... 86%  121M 0s\n",
      " 33700K .......... .......... .......... .......... .......... 86%  102M 0s\n",
      " 33750K .......... .......... .......... .......... .......... 86%  117M 0s\n",
      " 33800K .......... .......... .......... .......... .......... 87%  122M 0s\n",
      " 33850K .......... .......... .......... .......... .......... 87% 77,4M 0s\n",
      " 33900K .......... .......... .......... .......... .......... 87% 47,7M 0s\n",
      " 33950K .......... .......... .......... .......... .......... 87% 25,7M 0s\n",
      " 34000K .......... .......... .......... .......... .......... 87%  120M 0s\n",
      " 34050K .......... .......... .......... .......... .......... 87%  118M 0s\n",
      " 34100K .......... .......... .......... .......... .......... 87%  102M 0s\n",
      " 34150K .......... .......... .......... .......... .......... 88%  122M 0s\n",
      " 34200K .......... .......... .......... .......... .......... 88%  102M 0s\n",
      " 34250K .......... .......... .......... .......... .......... 88%  122M 0s\n",
      " 34300K .......... .......... .......... .......... .......... 88%  117M 0s\n",
      " 34350K .......... .......... .......... .......... .......... 88%  103M 0s\n",
      " 34400K .......... .......... .......... .......... .......... 88%  120M 0s\n",
      " 34450K .......... .......... .......... .......... .......... 88%  103M 0s\n",
      " 34500K .......... .......... .......... .......... .......... 88%  117M 0s\n",
      " 34550K .......... .......... .......... .......... .......... 89%  120M 0s\n",
      " 34600K .......... .......... .......... .......... .......... 89%  103M 0s\n",
      " 34650K .......... .......... .......... .......... .......... 89%  117M 0s\n",
      " 34700K .......... .......... .......... .......... .......... 89%  101M 0s\n",
      " 34750K .......... .......... .......... .......... .......... 89%  122M 0s\n",
      " 34800K .......... .......... .......... .......... .......... 89%  120M 0s\n",
      " 34850K .......... .......... .......... .......... .......... 89%  102M 0s\n",
      " 34900K .......... .......... .......... .......... .......... 89%  114M 0s\n",
      " 34950K .......... .......... .......... .......... .......... 90%  105M 0s\n",
      " 35000K .......... .......... .......... .......... .......... 90%  122M 0s\n",
      " 35050K .......... .......... .......... .......... .......... 90%  117M 0s\n",
      " 35100K .......... .......... .......... .......... .......... 90%  102M 0s\n",
      " 35150K .......... .......... .......... .......... .......... 90% 5,10M 0s\n",
      " 35200K .......... .......... .......... .......... .......... 90% 5,09M 0s\n",
      " 35250K .......... .......... .......... .......... .......... 90% 12,4M 0s\n",
      " 35300K .......... .......... .......... .......... .......... 90% 23,9M 0s\n",
      " 35350K .......... .......... .......... .......... .......... 91% 40,1M 0s\n",
      " 35400K .......... .......... .......... .......... .......... 91% 56,7M 0s\n",
      " 35450K .......... .......... .......... .......... .......... 91%  105M 0s\n",
      " 35500K .......... .......... .......... .......... .......... 91%  115M 0s\n",
      " 35550K .......... .......... .......... .......... .......... 91%  103M 0s\n",
      " 35600K .......... .......... .......... .......... .......... 91%  101M 0s\n",
      " 35650K .......... .......... .......... .......... .......... 91%  117M 0s\n",
      " 35700K .......... .......... .......... .......... .......... 91%  100M 0s\n",
      " 35750K .......... .......... .......... .......... .......... 92%  120M 0s\n",
      " 35800K .......... .......... .......... .......... .......... 92%  127M 0s\n",
      " 35850K .......... .......... .......... .......... .......... 92%  102M 0s\n",
      " 35900K .......... .......... .......... .......... .......... 92%  117M 0s\n",
      " 35950K .......... .......... .......... .......... .......... 92%  101M 0s\n",
      " 36000K .......... .......... .......... .......... .......... 92%  124M 0s\n",
      " 36050K .......... .......... .......... .......... .......... 92%  116M 0s\n",
      " 36100K .......... .......... .......... .......... .......... 93%  102M 0s\n",
      " 36150K .......... .......... .......... .......... .......... 93%  121M 0s\n",
      " 36200K .......... .......... .......... .......... .......... 93%  103M 0s\n",
      " 36250K .......... .......... .......... .......... .......... 93%  118M 0s\n",
      " 36300K .......... .......... .......... .......... .......... 93%  118M 0s\n",
      " 36350K .......... .......... .......... .......... .......... 93%  106M 0s\n",
      " 36400K .......... .......... .......... .......... .......... 93% 14,4M 0s\n",
      " 36450K .......... .......... .......... .......... .......... 93% 15,8M 0s\n",
      " 36500K .......... .......... .......... .......... .......... 94%  121M 0s\n",
      " 36550K .......... .......... .......... .......... .......... 94% 33,5M 0s\n",
      " 36600K .......... .......... .......... .......... .......... 94% 37,1M 0s\n",
      " 36650K .......... .......... .......... .......... .......... 94% 78,0M 0s\n",
      " 36700K .......... .......... .......... .......... .......... 94% 53,7M 0s\n",
      " 36750K .......... .......... .......... .......... .......... 94%  118M 0s\n",
      " 36800K .......... .......... .......... .......... .......... 94%  124M 0s\n",
      " 36850K .......... .......... .......... .......... .......... 94%  103M 0s\n",
      " 36900K .......... .......... .......... .......... .......... 95%  116M 0s\n",
      " 36950K .......... .......... .......... .......... .......... 95% 94,8M 0s\n",
      " 37000K .......... .......... .......... .......... .......... 95% 24,3M 0s\n",
      " 37050K .......... .......... .......... .......... .......... 95%  117M 0s\n",
      " 37100K .......... .......... .......... .......... .......... 95% 74,3M 0s\n",
      " 37150K .......... .......... .......... .......... .......... 95%  121M 0s\n",
      " 37200K .......... .......... .......... .......... .......... 95%  104M 0s\n",
      " 37250K .......... .......... .......... .......... .......... 95%  111M 0s\n",
      " 37300K .......... .......... .......... .......... .......... 96%  127M 0s\n",
      " 37350K .......... .......... .......... .......... .......... 96%  103M 0s\n",
      " 37400K .......... .......... .......... .......... .......... 96% 84,4M 0s\n",
      " 37450K .......... .......... .......... .......... .......... 96%  103M 0s\n",
      " 37500K .......... .......... .......... .......... .......... 96%  117M 0s\n",
      " 37550K .......... .......... .......... .......... .......... 96%  121M 0s\n",
      " 37600K .......... .......... .......... .......... .......... 96%  102M 0s\n",
      " 37650K .......... .......... .......... .......... .......... 97%  119M 0s\n",
      " 37700K .......... .......... .......... .......... .......... 97%  102M 0s\n",
      " 37750K .......... .......... .......... .......... .......... 97%  123M 0s\n",
      " 37800K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 37850K .......... .......... .......... .......... .......... 97%  102M 0s\n",
      " 37900K .......... .......... .......... .......... .......... 97%  120M 0s\n",
      " 37950K .......... .......... .......... .......... .......... 97%  102M 0s\n",
      " 38000K .......... .......... .......... .......... .......... 97%  121M 0s\n",
      " 38050K .......... .......... .......... .......... .......... 98%  118M 0s\n",
      " 38100K .......... .......... .......... .......... .......... 98%  102M 0s\n",
      " 38150K .......... .......... .......... .......... .......... 98%  120M 0s\n",
      " 38200K .......... .......... .......... .......... .......... 98% 5,26M 0s\n",
      " 38250K .......... .......... .......... .......... .......... 98% 6,21M 0s\n",
      " 38300K .......... .......... .......... .......... .......... 98% 10,8M 0s\n",
      " 38350K .......... .......... .......... .......... .......... 98% 13,5M 0s\n",
      " 38400K .......... .......... .......... .......... .......... 98%  110M 0s\n",
      " 38450K .......... .......... .......... .......... .......... 99% 50,7M 0s\n",
      " 38500K .......... .......... .......... .......... .......... 99% 58,3M 0s\n",
      " 38550K .......... .......... .......... .......... .......... 99%  115M 0s\n",
      " 38600K .......... .......... .......... .......... .......... 99% 99,1M 0s\n",
      " 38650K .......... .......... .......... .......... .......... 99%  109M 0s\n",
      " 38700K .......... .......... .......... .......... .......... 99%  101M 0s\n",
      " 38750K .......... .......... .......... .......... .......... 99%  109M 0s\n",
      " 38800K .......... .......... .......... .......... .......... 99%  120M 0s\n",
      " 38850K .......... ..                                         100%  143M=1,4s\n",
      "\n",
      "2023-08-18 22:53:31 (26,5 MB/s) - ‘data/reference/genome.fa.gz’ guardado [39795542/39795542]\n",
      "\n",
      "[Fri Aug 18 22:53:32 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T225329.336237.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:53:33 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR696683_chr12_1.fastq.gz, data/raw/ERR696683_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR696683_chr12_1_fastqc.html, results/fastqc_result/ERR696683_chr12_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR696683_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 22:53:49 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:53:49 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR696683_chr12_1.fastq.gz, data/raw/ERR696683_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR696683_chr12_1_fastp.fastq.gz, data/processed/ERR696683_chr12_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR696683_chr12_2_fastp.fastq.gz, data/processed/ERR696683_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1814551\n",
      "total bases: 137166702\n",
      "Q20 bases: 136196362(99.2926%)\n",
      "Q30 bases: 132450329(96.5616%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1814551\n",
      "total bases: 136812703\n",
      "Q20 bases: 135133064(98.7723%)\n",
      "Q30 bases: 129660948(94.7726%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1124306\n",
      "total bases: 85390403\n",
      "Q20 bases: 85056739(99.6092%)\n",
      "Q30 bases: 83576597(97.8759%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1124306\n",
      "total bases: 85356933\n",
      "Q20 bases: 84768167(99.3102%)\n",
      "Q30 bases: 82580401(96.7472%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2248612\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1380490\n",
      "reads with adapter trimmed: 812\n",
      "bases trimmed due to adapters: 18871\n",
      "\n",
      "Duplication rate: 5.7679%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR696683_chr12_1.fastq.gz -I data/raw/ERR696683_chr12_2.fastq.gz -o data/processed/ERR696683_chr12_1_fastp.fastq.gz -O data/processed/ERR696683_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Fri Aug 18 22:53:54 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:53:54 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR696683_chr12_1_fastp.fastq.gz, data/processed/ERR696683_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR696683_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR696683_chr12_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR696683_chr12_2_fastp.fastq.gz, data/processed/ERR696683_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 22:54:03 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T225332.614110.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:54:04 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753368_chr12_1.fastq.gz, data/raw/ERR753368_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753368_chr12_1_fastqc.html, results/fastqc_result/ERR753368_chr12_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753368_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753368_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 22:54:19 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:54:19 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753368_chr12_1.fastq.gz, data/raw/ERR753368_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753368_chr12_1_fastp.fastq.gz, data/processed/ERR753368_chr12_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753368_chr12_2_fastp.fastq.gz, data/processed/ERR753368_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1897940\n",
      "total bases: 144109638\n",
      "Q20 bases: 143542594(99.6065%)\n",
      "Q30 bases: 139301752(96.6637%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1897940\n",
      "total bases: 143961401\n",
      "Q20 bases: 142692339(99.1185%)\n",
      "Q30 bases: 136129314(94.5596%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1632596\n",
      "total bases: 124070769\n",
      "Q20 bases: 123855197(99.8263%)\n",
      "Q30 bases: 121763665(98.1405%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1632596\n",
      "total bases: 124060324\n",
      "Q20 bases: 123450972(99.5088%)\n",
      "Q30 bases: 119750755(96.5262%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3265192\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 530688\n",
      "reads with adapter trimmed: 1148\n",
      "bases trimmed due to adapters: 26680\n",
      "\n",
      "Duplication rate: 2.03283%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753368_chr12_1.fastq.gz -I data/raw/ERR753368_chr12_2.fastq.gz -o data/processed/ERR753368_chr12_1_fastp.fastq.gz -O data/processed/ERR753368_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Fri Aug 18 22:54:25 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:54:25 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753368_chr12_1_fastp.fastq.gz, data/processed/ERR753368_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753368_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753368_chr12_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753368_chr12_2_fastp.fastq.gz, data/processed/ERR753368_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 22:54:38 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T225403.590682.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:54:39 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753369_chr12_1.fastq.gz, data/raw/ERR753369_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753369_chr12_1_fastqc.html, results/fastqc_result/ERR753369_chr12_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753369_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753369_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 22:54:56 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:54:56 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753369_chr12_1.fastq.gz, data/raw/ERR753369_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753369_chr12_1_fastp.fastq.gz, data/processed/ERR753369_chr12_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753369_chr12_2_fastp.fastq.gz, data/processed/ERR753369_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2053990\n",
      "total bases: 155951416\n",
      "Q20 bases: 155309500(99.5884%)\n",
      "Q30 bases: 150604540(96.5714%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2053990\n",
      "total bases: 155798651\n",
      "Q20 bases: 154421725(99.1162%)\n",
      "Q30 bases: 147287629(94.5372%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1766738\n",
      "total bases: 134264590\n",
      "Q20 bases: 134014650(99.8138%)\n",
      "Q30 bases: 131659977(98.0601%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1766738\n",
      "total bases: 134253748\n",
      "Q20 bases: 133591459(99.5067%)\n",
      "Q30 bases: 129560868(96.5045%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3533476\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 574504\n",
      "reads with adapter trimmed: 1156\n",
      "bases trimmed due to adapters: 27060\n",
      "\n",
      "Duplication rate: 2.36772%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753369_chr12_1.fastq.gz -I data/raw/ERR753369_chr12_2.fastq.gz -o data/processed/ERR753369_chr12_1_fastp.fastq.gz -O data/processed/ERR753369_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Fri Aug 18 22:55:03 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:55:03 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753369_chr12_1_fastp.fastq.gz, data/processed/ERR753369_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753369_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753369_chr12_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753369_chr12_2_fastp.fastq.gz, data/processed/ERR753369_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 22:55:16 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T225438.903825.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:55:17 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753370_chr12_1.fastq.gz, data/raw/ERR753370_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753370_chr12_1_fastqc.html, results/fastqc_result/ERR753370_chr12_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753370_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753370_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 22:55:40 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:55:40 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753370_chr12_1.fastq.gz, data/raw/ERR753370_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753370_chr12_1_fastp.fastq.gz, data/processed/ERR753370_chr12_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753370_chr12_1_fastp.fastq.gz, data/processed/ERR753370_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2549417\n",
      "total bases: 256637892\n",
      "Q20 bases: 254886215(99.3175%)\n",
      "Q30 bases: 247502750(96.4405%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2549417\n",
      "total bases: 256574888\n",
      "Q20 bases: 254137882(99.0502%)\n",
      "Q30 bases: 244519685(95.3015%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 2548864\n",
      "total bases: 255580768\n",
      "Q20 bases: 253929976(99.3541%)\n",
      "Q30 bases: 246895663(96.6018%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 2548864\n",
      "total bases: 254756297\n",
      "Q20 bases: 252541907(99.1308%)\n",
      "Q30 bases: 243527730(95.5924%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 5097728\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1106\n",
      "reads with adapter trimmed: 13432\n",
      "bases trimmed due to adapters: 104382\n",
      "\n",
      "Duplication rate: 4.35037%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 160\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753370_chr12_1.fastq.gz -I data/raw/ERR753370_chr12_2.fastq.gz -o data/processed/ERR753370_chr12_1_fastp.fastq.gz -O data/processed/ERR753370_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 11 seconds\n",
      "[Fri Aug 18 22:55:52 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:55:52 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753370_chr12_1_fastp.fastq.gz, data/processed/ERR753370_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753370_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753370_chr12_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753370_chr12_1_fastp.fastq.gz, data/processed/ERR753370_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 22:56:15 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T225516.720136.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:56:16 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753371_chr12_1.fastq.gz, data/raw/ERR753371_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753371_chr12_1_fastp.fastq.gz, data/processed/ERR753371_chr12_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753371_chr12_2_fastp.fastq.gz, data/processed/ERR753371_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1690709\n",
      "total bases: 127390591\n",
      "Q20 bases: 126242409(99.0987%)\n",
      "Q30 bases: 121643046(95.4882%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1690709\n",
      "total bases: 127070345\n",
      "Q20 bases: 125554387(98.807%)\n",
      "Q30 bases: 121137819(95.3313%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 872488\n",
      "total bases: 66228440\n",
      "Q20 bases: 65905625(99.5126%)\n",
      "Q30 bases: 64262329(97.0313%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 872488\n",
      "total bases: 66192098\n",
      "Q20 bases: 65729465(99.3011%)\n",
      "Q30 bases: 64277244(97.1071%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1744976\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1636442\n",
      "reads with adapter trimmed: 710\n",
      "bases trimmed due to adapters: 15017\n",
      "\n",
      "Duplication rate: 22.0345%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 116\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753371_chr12_1.fastq.gz -I data/raw/ERR753371_chr12_2.fastq.gz -o data/processed/ERR753371_chr12_1_fastp.fastq.gz -O data/processed/ERR753371_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Fri Aug 18 22:56:22 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:56:22 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753371_chr12_1_fastp.fastq.gz, data/processed/ERR753371_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753371_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753371_chr12_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753371_chr12_2_fastp.fastq.gz, data/processed/ERR753371_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 22:56:30 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:56:30 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753371_chr12_1.fastq.gz, data/raw/ERR753371_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753371_chr12_1_fastqc.html, results/fastqc_result/ERR753371_chr12_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753371_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753371_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 22:56:43 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T225615.449203.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:56:44 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753372_chr12_1.fastq.gz, data/raw/ERR753372_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753372_chr12_1_fastqc.html, results/fastqc_result/ERR753372_chr12_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753372_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753372_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 22:56:59 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:56:59 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753372_chr12_1.fastq.gz, data/raw/ERR753372_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753372_chr12_1_fastp.fastq.gz, data/processed/ERR753372_chr12_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753372_chr12_2_fastp.fastq.gz, data/processed/ERR753372_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2028383\n",
      "total bases: 154006268\n",
      "Q20 bases: 153393425(99.6021%)\n",
      "Q30 bases: 148774276(96.6027%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2028383\n",
      "total bases: 153852859\n",
      "Q20 bases: 152494427(99.1171%)\n",
      "Q30 bases: 145498335(94.5698%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1738874\n",
      "total bases: 132147137\n",
      "Q20 bases: 131915460(99.8247%)\n",
      "Q30 bases: 129641227(98.1037%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1738874\n",
      "total bases: 132136009\n",
      "Q20 bases: 131490557(99.5115%)\n",
      "Q30 bases: 127584864(96.5557%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3477748\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 579018\n",
      "reads with adapter trimmed: 1250\n",
      "bases trimmed due to adapters: 26990\n",
      "\n",
      "Duplication rate: 2.00291%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753372_chr12_1.fastq.gz -I data/raw/ERR753372_chr12_2.fastq.gz -o data/processed/ERR753372_chr12_1_fastp.fastq.gz -O data/processed/ERR753372_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Fri Aug 18 22:57:06 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:57:06 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753372_chr12_1_fastp.fastq.gz, data/processed/ERR753372_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753372_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753372_chr12_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753372_chr12_2_fastp.fastq.gz, data/processed/ERR753372_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 22:57:19 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T225643.881023.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:57:20 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753373_chr12_1.fastq.gz, data/raw/ERR753373_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753373_chr12_1_fastqc.html, results/fastqc_result/ERR753373_chr12_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753373_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753373_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 22:57:34 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:57:34 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753373_chr12_1.fastq.gz, data/raw/ERR753373_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753373_chr12_1_fastp.fastq.gz, data/processed/ERR753373_chr12_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753373_chr12_2_fastp.fastq.gz, data/processed/ERR753373_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1413533\n",
      "total bases: 135329456\n",
      "Q20 bases: 134237138(99.1928%)\n",
      "Q30 bases: 125040132(92.3968%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1413533\n",
      "total bases: 132444072\n",
      "Q20 bases: 130991936(98.9036%)\n",
      "Q30 bases: 122419166(92.4308%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1378941\n",
      "total bases: 129926291\n",
      "Q20 bases: 129164081(99.4134%)\n",
      "Q30 bases: 121734716(93.6952%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1378941\n",
      "total bases: 125531307\n",
      "Q20 bases: 124923753(99.516%)\n",
      "Q30 bases: 118402277(94.3209%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2757882\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 69184\n",
      "reads with adapter trimmed: 1192\n",
      "bases trimmed due to adapters: 41078\n",
      "\n",
      "Duplication rate: 1.84912%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 153\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753373_chr12_1.fastq.gz -I data/raw/ERR753373_chr12_2.fastq.gz -o data/processed/ERR753373_chr12_1_fastp.fastq.gz -O data/processed/ERR753373_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Fri Aug 18 22:57:40 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:57:40 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753373_chr12_1_fastp.fastq.gz, data/processed/ERR753373_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753373_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753373_chr12_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753373_chr12_2_fastp.fastq.gz, data/processed/ERR753373_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 22:57:53 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T225719.382074.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:57:54 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753374_chr12_1.fastq.gz, data/raw/ERR753374_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753374_chr12_1_fastp.fastq.gz, data/processed/ERR753374_chr12_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753374_chr12_1_fastp.fastq.gz, data/processed/ERR753374_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 849844\n",
      "total bases: 81288111\n",
      "Q20 bases: 80518260(99.0529%)\n",
      "Q30 bases: 73547935(90.4781%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 849844\n",
      "total bases: 79522059\n",
      "Q20 bases: 78682606(98.9444%)\n",
      "Q30 bases: 72336674(90.9643%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 822905\n",
      "total bases: 76905295\n",
      "Q20 bases: 76358183(99.2886%)\n",
      "Q30 bases: 70820041(92.0873%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 822905\n",
      "total bases: 74885957\n",
      "Q20 bases: 74450348(99.4183%)\n",
      "Q30 bases: 69591241(92.9296%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1645810\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 53878\n",
      "reads with adapter trimmed: 802\n",
      "bases trimmed due to adapters: 29662\n",
      "\n",
      "Duplication rate: 5.60966%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 154\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753374_chr12_1.fastq.gz -I data/raw/ERR753374_chr12_2.fastq.gz -o data/processed/ERR753374_chr12_1_fastp.fastq.gz -O data/processed/ERR753374_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 3 seconds\n",
      "[Fri Aug 18 22:57:58 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:57:58 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753374_chr12_1.fastq.gz, data/raw/ERR753374_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753374_chr12_1_fastqc.html, results/fastqc_result/ERR753374_chr12_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753374_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753374_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 22:58:07 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:58:07 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753374_chr12_1_fastp.fastq.gz, data/processed/ERR753374_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753374_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753374_chr12_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753374_chr12_1_fastp.fastq.gz, data/processed/ERR753374_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 22:58:16 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T225753.671496.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:58:17 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753375_chr12_1.fastq.gz, data/raw/ERR753375_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753375_chr12_1_fastqc.html, results/fastqc_result/ERR753375_chr12_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753375_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753375_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 22:58:28 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:58:28 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753375_chr12_1.fastq.gz, data/raw/ERR753375_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753375_chr12_1_fastp.fastq.gz, data/processed/ERR753375_chr12_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753375_chr12_1_fastp.fastq.gz, data/processed/ERR753375_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1133521\n",
      "total bases: 85620774\n",
      "Q20 bases: 85092338(99.3828%)\n",
      "Q30 bases: 83171029(97.1388%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1133521\n",
      "total bases: 85602587\n",
      "Q20 bases: 84492851(98.7036%)\n",
      "Q30 bases: 82429218(96.2929%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 738967\n",
      "total bases: 56110081\n",
      "Q20 bases: 55927294(99.6742%)\n",
      "Q30 bases: 55087312(98.1772%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 738967\n",
      "total bases: 56118788\n",
      "Q20 bases: 55577475(99.0354%)\n",
      "Q30 bases: 54704557(97.4799%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1477934\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 789108\n",
      "reads with adapter trimmed: 472\n",
      "bases trimmed due to adapters: 10223\n",
      "\n",
      "Duplication rate: 1.69069%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753375_chr12_1.fastq.gz -I data/raw/ERR753375_chr12_2.fastq.gz -o data/processed/ERR753375_chr12_1_fastp.fastq.gz -O data/processed/ERR753375_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Fri Aug 18 22:58:32 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:58:32 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753375_chr12_1_fastp.fastq.gz, data/processed/ERR753375_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753375_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753375_chr12_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753375_chr12_1_fastp.fastq.gz, data/processed/ERR753375_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 22:58:40 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T225816.562732.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:58:41 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753376_chr12_1.fastq.gz, data/raw/ERR753376_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753376_chr12_1_fastqc.html, results/fastqc_result/ERR753376_chr12_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753376_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753376_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 22:58:54 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:58:54 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753376_chr12_1.fastq.gz, data/raw/ERR753376_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753376_chr12_1_fastp.fastq.gz, data/processed/ERR753376_chr12_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753376_chr12_2_fastp.fastq.gz, data/processed/ERR753376_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1465726\n",
      "total bases: 110800464\n",
      "Q20 bases: 110033992(99.3082%)\n",
      "Q30 bases: 107011804(96.5806%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1465726\n",
      "total bases: 110496460\n",
      "Q20 bases: 109134987(98.7679%)\n",
      "Q30 bases: 104686265(94.7417%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 904519\n",
      "total bases: 68697720\n",
      "Q20 bases: 68438927(99.6233%)\n",
      "Q30 bases: 67261210(97.9089%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 904519\n",
      "total bases: 68671754\n",
      "Q20 bases: 68200419(99.3136%)\n",
      "Q30 bases: 66433000(96.7399%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1809038\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1122414\n",
      "reads with adapter trimmed: 700\n",
      "bases trimmed due to adapters: 15961\n",
      "\n",
      "Duplication rate: 9.00905%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753376_chr12_1.fastq.gz -I data/raw/ERR753376_chr12_2.fastq.gz -o data/processed/ERR753376_chr12_1_fastp.fastq.gz -O data/processed/ERR753376_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Fri Aug 18 22:58:59 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:58:59 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753376_chr12_1_fastp.fastq.gz, data/processed/ERR753376_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753376_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753376_chr12_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753376_chr12_2_fastp.fastq.gz, data/processed/ERR753376_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 22:59:08 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T225840.377645.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:59:09 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753377_chr12_1.fastq.gz, data/raw/ERR753377_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753377_chr12_1_fastp.fastq.gz, data/processed/ERR753377_chr12_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753377_chr12_1_fastp.fastq.gz, data/processed/ERR753377_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1534776\n",
      "total bases: 116035701\n",
      "Q20 bases: 115218791(99.296%)\n",
      "Q30 bases: 112042570(96.5587%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1534776\n",
      "total bases: 115756078\n",
      "Q20 bases: 114393805(98.8232%)\n",
      "Q30 bases: 109938521(94.9743%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 970154\n",
      "total bases: 73683410\n",
      "Q20 bases: 73398394(99.6132%)\n",
      "Q30 bases: 72118967(97.8768%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 970154\n",
      "total bases: 73656183\n",
      "Q20 bases: 73172261(99.343%)\n",
      "Q30 bases: 71360589(96.8834%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1940308\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1129244\n",
      "reads with adapter trimmed: 748\n",
      "bases trimmed due to adapters: 16700\n",
      "\n",
      "Duplication rate: 9.03266%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 119\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753377_chr12_1.fastq.gz -I data/raw/ERR753377_chr12_2.fastq.gz -o data/processed/ERR753377_chr12_1_fastp.fastq.gz -O data/processed/ERR753377_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Fri Aug 18 22:59:14 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:59:14 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753377_chr12_1_fastp.fastq.gz, data/processed/ERR753377_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753377_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753377_chr12_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753377_chr12_1_fastp.fastq.gz, data/processed/ERR753377_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 22:59:23 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:59:23 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753377_chr12_1.fastq.gz, data/raw/ERR753377_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753377_chr12_1_fastqc.html, results/fastqc_result/ERR753377_chr12_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753377_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753377_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 22:59:36 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T225908.291482.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:59:37 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753378_chr12_1.fastq.gz, data/raw/ERR753378_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753378_chr12_1_fastqc.html, results/fastqc_result/ERR753378_chr12_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753378_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753378_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 22:59:52 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:59:52 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753378_chr12_1.fastq.gz, data/raw/ERR753378_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753378_chr12_1_fastp.fastq.gz, data/processed/ERR753378_chr12_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753378_chr12_1_fastp.fastq.gz, data/processed/ERR753378_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1753717\n",
      "total bases: 133156565\n",
      "Q20 bases: 132634253(99.6077%)\n",
      "Q30 bases: 128678299(96.6368%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1753717\n",
      "total bases: 133019021\n",
      "Q20 bases: 131845170(99.1175%)\n",
      "Q30 bases: 125793123(94.5678%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1509829\n",
      "total bases: 114740679\n",
      "Q20 bases: 114539895(99.825%)\n",
      "Q30 bases: 112565716(98.1045%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1509829\n",
      "total bases: 114731150\n",
      "Q20 bases: 114164648(99.5062%)\n",
      "Q30 bases: 110726794(96.5098%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3019658\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 487776\n",
      "reads with adapter trimmed: 1118\n",
      "bases trimmed due to adapters: 25907\n",
      "\n",
      "Duplication rate: 1.96634%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753378_chr12_1.fastq.gz -I data/raw/ERR753378_chr12_2.fastq.gz -o data/processed/ERR753378_chr12_1_fastp.fastq.gz -O data/processed/ERR753378_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Fri Aug 18 22:59:58 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 22:59:58 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753378_chr12_1_fastp.fastq.gz, data/processed/ERR753378_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753378_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753378_chr12_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753378_chr12_1_fastp.fastq.gz, data/processed/ERR753378_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Fri Aug 18 23:00:10 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T225936.250164.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:00:12 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR696683_chr12_1_fastp.fastq.gz, data/processed/ERR696683_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR696683_chr12.sam, results/mapped_reads/ERR696683_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR696683_chr12_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR696683_chr12_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.52 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 46.85 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.40 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.27 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 27.97 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 76.642 sec; CPU: 76.006 sec\n",
      "[Fri Aug 18 23:02:48 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:02:48 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR696683_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr12.bam, results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR696683_chr12.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR696683_chr12_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 23:03:28 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:03:28 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR696683_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "23:03:29.040 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 23:03:29 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR696683_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR696683_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 23:03:29 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 23:03:29\tMarkDuplicates\tStart of doWork freeMemory: 528833064; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:03:29\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 23:03:29\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam\n",
      "INFO\t2023-08-18 23:03:30\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:53,179,811\n",
      "INFO\t2023-08-18 23:03:30\tMarkDuplicates\tTracking 631 as yet unmatched pairs. 631 records in RAM.\n",
      "INFO\t2023-08-18 23:03:32\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 12:113,957,746\n",
      "INFO\t2023-08-18 23:03:32\tMarkDuplicates\tTracking 278 as yet unmatched pairs. 278 records in RAM.\n",
      "INFO\t2023-08-18 23:03:32\tMarkDuplicates\tRead 2264185 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 23:03:32\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 537783160; totalMemory: 852492288; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:03:32\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 23:03:32\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 23:03:32\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 23:03:33\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 23:03:33\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 920647848; totalMemory: 1466957824; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:03:33\tMarkDuplicates\tMarking 210604 records as duplicates.\n",
      "INFO\t2023-08-18 23:03:33\tMarkDuplicates\tFound 91 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam\n",
      "INFO\t2023-08-18 23:03:33\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 23:03:46\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 23:03:46\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 23:03:46\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 23:03:46\tMarkDuplicates\tBefore output close freeMemory: 528131680; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:03:46\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 23:03:46\tMarkDuplicates\tAfter output close freeMemory: 528131680; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 23:03:46 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.30 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 23:03:48 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T230011.164813.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:03:49 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753368_chr12_1_fastp.fastq.gz, data/processed/ERR753368_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753368_chr12.sam, results/mapped_reads/ERR753368_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753368_chr12_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR753368_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.44 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 43.07 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.42 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.28 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 26.13 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 70.988 sec; CPU: 70.337 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 23:07:17 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:07:17 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753368_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr12.bam, results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753368_chr12.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753368_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 23:08:16 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:08:16 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753368_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753368_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "23:08:16.900 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 23:08:16 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753368_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753368_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 23:08:16 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 23:08:16\tMarkDuplicates\tStart of doWork freeMemory: 528788552; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:08:16\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 23:08:16\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam\n",
      "INFO\t2023-08-18 23:08:18\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:43,551,107\n",
      "INFO\t2023-08-18 23:08:18\tMarkDuplicates\tTracking 477 as yet unmatched pairs. 477 records in RAM.\n",
      "INFO\t2023-08-18 23:08:19\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 12:76,860,351\n",
      "INFO\t2023-08-18 23:08:19\tMarkDuplicates\tTracking 506 as yet unmatched pairs. 506 records in RAM.\n",
      "INFO\t2023-08-18 23:08:21\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 12:122,377,487\n",
      "INFO\t2023-08-18 23:08:21\tMarkDuplicates\tTracking 227 as yet unmatched pairs. 227 records in RAM.\n",
      "INFO\t2023-08-18 23:08:21\tMarkDuplicates\tRead 3277664 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 23:08:21\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 672001408; totalMemory: 1096810496; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:08:21\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 23:08:21\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 23:08:22\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 23:08:22\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 23:08:22\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1087326960; totalMemory: 1633681408; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:08:22\tMarkDuplicates\tMarking 79706 records as duplicates.\n",
      "INFO\t2023-08-18 23:08:22\tMarkDuplicates\tFound 488 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam\n",
      "INFO\t2023-08-18 23:08:22\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 23:08:41\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 23:08:41\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 23:08:41\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 23:08:41\tMarkDuplicates\tBefore output close freeMemory: 528087048; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:08:41\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 23:08:41\tMarkDuplicates\tAfter output close freeMemory: 528087048; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 23:08:41 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.40 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 23:08:43 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T230348.433514.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:08:45 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753369_chr12_1_fastp.fastq.gz, data/processed/ERR753369_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753369_chr12.sam, results/mapped_reads/ERR753369_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753369_chr12_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR753369_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.43 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 43.00 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.41 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.27 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 26.19 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 74.688 sec; CPU: 70.289 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 23:12:26 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:12:26 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753369_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr12.bam, results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753369_chr12.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753369_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 23:13:27 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:13:27 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753369_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753369_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "23:13:28.063 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 23:13:28 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753369_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753369_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 23:13:28 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 23:13:28\tMarkDuplicates\tStart of doWork freeMemory: 528790752; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:13:28\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 23:13:28\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam\n",
      "INFO\t2023-08-18 23:13:29\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:40,105,150\n",
      "INFO\t2023-08-18 23:13:29\tMarkDuplicates\tTracking 394 as yet unmatched pairs. 394 records in RAM.\n",
      "INFO\t2023-08-18 23:13:30\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 12:67,306,227\n",
      "INFO\t2023-08-18 23:13:30\tMarkDuplicates\tTracking 518 as yet unmatched pairs. 518 records in RAM.\n",
      "INFO\t2023-08-18 23:13:32\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 12:112,948,761\n",
      "INFO\t2023-08-18 23:13:32\tMarkDuplicates\tTracking 274 as yet unmatched pairs. 274 records in RAM.\n",
      "INFO\t2023-08-18 23:13:33\tMarkDuplicates\tRead 3547704 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 23:13:33\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 776555280; totalMemory: 1231028224; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:13:33\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 23:13:33\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 23:13:33\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 23:13:33\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 23:13:33\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1157585792; totalMemory: 1703936000; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:13:33\tMarkDuplicates\tMarking 100943 records as duplicates.\n",
      "INFO\t2023-08-18 23:13:33\tMarkDuplicates\tFound 534 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam\n",
      "INFO\t2023-08-18 23:13:33\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 23:13:53\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 23:13:53\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 23:13:53\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 23:13:53\tMarkDuplicates\tBefore output close freeMemory: 528089240; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:13:53\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 23:13:53\tMarkDuplicates\tAfter output close freeMemory: 528089240; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 23:13:53 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.42 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 23:13:56 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T230844.349189.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:13:57 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753370_chr12_1_fastp.fastq.gz, data/processed/ERR753370_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753370_chr12.sam, results/mapped_reads/ERR753370_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753370_chr12_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753370_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.44 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 43.04 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.40 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.26 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 26.29 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 74.948 sec; CPU: 70.437 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 23:19:16 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:19:16 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753370_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr12.bam, results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753370_chr12.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753370_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 23:21:09 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:21:09 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753370_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753370_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "23:21:10.465 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 23:21:10 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753370_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753370_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 23:21:10 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 23:21:10\tMarkDuplicates\tStart of doWork freeMemory: 528787576; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:21:10\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 23:21:10\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam\n",
      "INFO\t2023-08-18 23:21:12\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:21,504,452\n",
      "INFO\t2023-08-18 23:21:12\tMarkDuplicates\tTracking 786 as yet unmatched pairs. 786 records in RAM.\n",
      "INFO\t2023-08-18 23:21:13\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 12:49,752,444\n",
      "INFO\t2023-08-18 23:21:13\tMarkDuplicates\tTracking 1418 as yet unmatched pairs. 1418 records in RAM.\n",
      "INFO\t2023-08-18 23:21:15\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 12:69,697,586\n",
      "INFO\t2023-08-18 23:21:15\tMarkDuplicates\tTracking 1415 as yet unmatched pairs. 1415 records in RAM.\n",
      "INFO\t2023-08-18 23:21:16\tMarkDuplicates\tRead     4,000,000 records.  Elapsed time: 00:00:06s.  Time for last 1,000,000:    1s.  Last read position: 12:103,793,216\n",
      "INFO\t2023-08-18 23:21:16\tMarkDuplicates\tTracking 1251 as yet unmatched pairs. 1251 records in RAM.\n",
      "INFO\t2023-08-18 23:21:18\tMarkDuplicates\tRead     5,000,000 records.  Elapsed time: 00:00:07s.  Time for last 1,000,000:    1s.  Last read position: 12:128,502,441\n",
      "INFO\t2023-08-18 23:21:18\tMarkDuplicates\tTracking 275 as yet unmatched pairs. 275 records in RAM.\n",
      "INFO\t2023-08-18 23:21:18\tMarkDuplicates\tRead 5146509 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 23:21:18\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 429523840; totalMemory: 1052770304; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:21:18\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 23:21:18\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 23:21:19\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 23:21:19\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 23:21:19\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1043285256; totalMemory: 1589641216; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:21:19\tMarkDuplicates\tMarking 368838 records as duplicates.\n",
      "INFO\t2023-08-18 23:21:19\tMarkDuplicates\tFound 219 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam\n",
      "INFO\t2023-08-18 23:21:19\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 23:21:55\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 23:21:55\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 23:21:55\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 23:21:55\tMarkDuplicates\tBefore output close freeMemory: 528085104; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:21:55\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 23:21:55\tMarkDuplicates\tAfter output close freeMemory: 528085104; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 23:21:55 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.76 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 23:21:59 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T231356.752765.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:22:00 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753371_chr12_1_fastp.fastq.gz, data/processed/ERR753371_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753371_chr12.sam, results/mapped_reads/ERR753371_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753371_chr12_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753371_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.44 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 43.05 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.40 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.27 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 26.28 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 74.804 sec; CPU: 70.433 sec\n",
      "[Fri Aug 18 23:24:21 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:24:21 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753371_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr12.bam, results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753371_chr12.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753371_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 23:24:52 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:24:52 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753371_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753371_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "23:24:53.343 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 23:24:53 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753371_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753371_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 23:24:53 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 23:24:53\tMarkDuplicates\tStart of doWork freeMemory: 528786240; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:24:53\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 23:24:53\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam\n",
      "INFO\t2023-08-18 23:24:54\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:68,764,508\n",
      "INFO\t2023-08-18 23:24:54\tMarkDuplicates\tTracking 112 as yet unmatched pairs. 112 records in RAM.\n",
      "INFO\t2023-08-18 23:24:55\tMarkDuplicates\tRead 1748836 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 23:24:55\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 341756744; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:24:55\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 23:24:55\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 23:24:56\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 23:24:56\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 23:24:56\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 822032952; totalMemory: 1368391680; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:24:56\tMarkDuplicates\tMarking 714306 records as duplicates.\n",
      "INFO\t2023-08-18 23:24:56\tMarkDuplicates\tFound 65 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam\n",
      "INFO\t2023-08-18 23:24:56\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 23:25:06\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 23:25:06\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 23:25:06\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 23:25:06\tMarkDuplicates\tBefore output close freeMemory: 528082560; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:25:06\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 23:25:06\tMarkDuplicates\tAfter output close freeMemory: 528082560; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 23:25:06 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.23 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 23:25:08 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T232159.937742.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:25:09 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753372_chr12_1_fastp.fastq.gz, data/processed/ERR753372_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753372_chr12.sam, results/mapped_reads/ERR753372_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753372_chr12_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753372_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.45 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 43.20 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.40 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.28 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 26.24 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 75.103 sec; CPU: 70.556 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 23:28:49 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:28:49 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753372_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr12.bam, results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753372_chr12.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753372_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Fri Aug 18 23:29:49 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:29:49 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753372_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753372_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "23:29:50.719 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 23:29:50 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753372_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753372_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 23:29:50 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 23:29:50\tMarkDuplicates\tStart of doWork freeMemory: 528829352; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:29:50\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 23:29:50\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam\n",
      "INFO\t2023-08-18 23:29:52\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:40,527,733\n",
      "INFO\t2023-08-18 23:29:52\tMarkDuplicates\tTracking 548 as yet unmatched pairs. 548 records in RAM.\n",
      "INFO\t2023-08-18 23:29:53\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 12:69,257,716\n",
      "INFO\t2023-08-18 23:29:53\tMarkDuplicates\tTracking 532 as yet unmatched pairs. 532 records in RAM.\n",
      "INFO\t2023-08-18 23:29:55\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 12:116,750,195\n",
      "INFO\t2023-08-18 23:29:55\tMarkDuplicates\tTracking 302 as yet unmatched pairs. 302 records in RAM.\n",
      "INFO\t2023-08-18 23:29:55\tMarkDuplicates\tRead 3494913 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 23:29:55\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 514599552; totalMemory: 962592768; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:29:55\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 23:29:55\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 23:29:56\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 23:29:56\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 23:29:56\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 953150536; totalMemory: 1499463680; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:29:56\tMarkDuplicates\tMarking 84321 records as duplicates.\n",
      "INFO\t2023-08-18 23:29:56\tMarkDuplicates\tFound 567 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam\n",
      "INFO\t2023-08-18 23:29:56\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 23:30:15\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 23:30:15\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 23:30:15\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 23:30:15\tMarkDuplicates\tBefore output close freeMemory: 528128968; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:30:15\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 23:30:15\tMarkDuplicates\tAfter output close freeMemory: 528128968; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 23:30:15 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.42 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 23:30:18 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T232508.245430.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:30:19 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753373_chr12_1_fastp.fastq.gz, data/processed/ERR753373_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753373_chr12.sam, results/mapped_reads/ERR753373_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753373_chr12_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753373_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.44 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 43.30 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.40 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.27 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 26.24 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 74.900 sec; CPU: 70.658 sec\n",
      "[Fri Aug 18 23:33:59 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:33:59 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753373_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr12.bam, results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753373_chr12.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753373_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 23:35:13 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:35:13 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753373_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753373_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "23:35:13.976 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 23:35:13 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753373_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753373_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 23:35:13 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 23:35:13\tMarkDuplicates\tStart of doWork freeMemory: 528828656; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:35:13\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 23:35:13\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam\n",
      "INFO\t2023-08-18 23:35:15\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:46,204,250\n",
      "INFO\t2023-08-18 23:35:15\tMarkDuplicates\tTracking 307 as yet unmatched pairs. 307 records in RAM.\n",
      "INFO\t2023-08-18 23:35:17\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 12:92,184,282\n",
      "INFO\t2023-08-18 23:35:17\tMarkDuplicates\tTracking 218 as yet unmatched pairs. 218 records in RAM.\n",
      "INFO\t2023-08-18 23:35:18\tMarkDuplicates\tRead 2777001 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 23:35:18\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 675499312; totalMemory: 1045430272; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:35:18\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 23:35:18\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 23:35:18\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 23:35:19\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 23:35:19\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1035987384; totalMemory: 1582301184; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:35:19\tMarkDuplicates\tMarking 73153 records as duplicates.\n",
      "INFO\t2023-08-18 23:35:19\tMarkDuplicates\tFound 23782 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam\n",
      "INFO\t2023-08-18 23:35:19\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 23:35:41\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 23:35:41\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 23:35:41\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 23:35:41\tMarkDuplicates\tBefore output close freeMemory: 528127520; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:35:41\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 23:35:41\tMarkDuplicates\tAfter output close freeMemory: 528127520; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 23:35:41 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.46 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 23:35:44 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T233018.660027.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:35:46 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753374_chr12_1_fastp.fastq.gz, data/processed/ERR753374_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753374_chr12.sam, results/mapped_reads/ERR753374_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753374_chr12_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753374_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.43 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 43.17 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.40 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.27 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 26.13 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 74.908 sec; CPU: 70.409 sec\n",
      "[Fri Aug 18 23:38:34 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:38:34 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753374_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr12.bam, results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753374_chr12.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753374_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 23:39:20 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:39:20 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753374_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753374_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "23:39:21.124 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 23:39:21 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753374_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753374_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 23:39:21 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 23:39:21\tMarkDuplicates\tStart of doWork freeMemory: 528775400; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:39:21\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 23:39:21\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam\n",
      "INFO\t2023-08-18 23:39:22\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:75,838,860\n",
      "INFO\t2023-08-18 23:39:22\tMarkDuplicates\tTracking 162 as yet unmatched pairs. 162 records in RAM.\n",
      "INFO\t2023-08-18 23:39:24\tMarkDuplicates\tRead 1659483 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 23:39:24\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 287336952; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:39:24\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 23:39:24\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 23:39:24\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 23:39:24\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 23:39:24\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 782177184; totalMemory: 1328545792; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:39:24\tMarkDuplicates\tMarking 175806 records as duplicates.\n",
      "INFO\t2023-08-18 23:39:24\tMarkDuplicates\tFound 58563 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam\n",
      "INFO\t2023-08-18 23:39:24\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 23:39:39\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 23:39:39\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 23:39:39\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 23:39:39\tMarkDuplicates\tBefore output close freeMemory: 528072584; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:39:39\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 23:39:39\tMarkDuplicates\tAfter output close freeMemory: 528072584; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 23:39:39 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.30 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 23:39:40 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T233545.094885.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:39:42 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753375_chr12_1_fastp.fastq.gz, data/processed/ERR753375_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753375_chr12.sam, results/mapped_reads/ERR753375_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753375_chr12_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753375_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.46 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 50.78 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.38 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.27 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 28.86 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 85.328 sec; CPU: 80.754 sec\n",
      "[Fri Aug 18 23:42:11 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:42:11 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753375_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr12.bam, results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753375_chr12.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753375_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 23:42:39 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:42:39 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753375_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753375_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "23:42:40.425 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 23:42:40 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753375_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753375_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 23:42:40 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 23:42:40\tMarkDuplicates\tStart of doWork freeMemory: 528788232; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:42:40\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 23:42:40\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam\n",
      "INFO\t2023-08-18 23:42:42\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:87,996,517\n",
      "INFO\t2023-08-18 23:42:42\tMarkDuplicates\tTracking 188 as yet unmatched pairs. 188 records in RAM.\n",
      "INFO\t2023-08-18 23:42:42\tMarkDuplicates\tRead 1483975 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 23:42:42\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 370500208; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:42:42\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 23:42:43\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 23:42:43\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 23:42:43\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 23:42:43\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 773802512; totalMemory: 1320157184; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:42:43\tMarkDuplicates\tMarking 72154 records as duplicates.\n",
      "INFO\t2023-08-18 23:42:43\tMarkDuplicates\tFound 188 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam\n",
      "INFO\t2023-08-18 23:42:43\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 23:42:52\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 23:42:52\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 23:42:52\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 23:42:52\tMarkDuplicates\tBefore output close freeMemory: 528086784; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:42:52\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 23:42:52\tMarkDuplicates\tAfter output close freeMemory: 528086784; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 23:42:52 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.21 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 23:42:53 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T233940.972618.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:42:55 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753376_chr12_1_fastp.fastq.gz, data/processed/ERR753376_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753376_chr12.sam, results/mapped_reads/ERR753376_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753376_chr12_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753376_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.46 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 49.51 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.41 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.30 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 31.76 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 86.813 sec; CPU: 82.435 sec\n",
      "[Fri Aug 18 23:45:38 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:45:38 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753376_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr12.bam, results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753376_chr12.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753376_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 23:46:15 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:46:15 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753376_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753376_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "23:46:16.528 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 23:46:16 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753376_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753376_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 23:46:16 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 23:46:16\tMarkDuplicates\tStart of doWork freeMemory: 528779136; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:46:16\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 23:46:16\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam\n",
      "INFO\t2023-08-18 23:46:18\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:64,464,346\n",
      "INFO\t2023-08-18 23:46:18\tMarkDuplicates\tTracking 273 as yet unmatched pairs. 273 records in RAM.\n",
      "INFO\t2023-08-18 23:46:19\tMarkDuplicates\tRead 1814904 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 23:46:19\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 334644456; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:46:19\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 23:46:19\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 23:46:19\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 23:46:20\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 23:46:20\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 808395552; totalMemory: 1354760192; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:46:20\tMarkDuplicates\tMarking 260477 records as duplicates.\n",
      "INFO\t2023-08-18 23:46:20\tMarkDuplicates\tFound 83 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam\n",
      "INFO\t2023-08-18 23:46:20\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 23:46:32\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 23:46:32\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 23:46:32\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 23:46:32\tMarkDuplicates\tBefore output close freeMemory: 528076608; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:46:32\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 23:46:32\tMarkDuplicates\tAfter output close freeMemory: 528076608; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 23:46:32 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.26 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 23:46:33 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T234254.080528.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:46:35 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753377_chr12_1_fastp.fastq.gz, data/processed/ERR753377_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753377_chr12.sam, results/mapped_reads/ERR753377_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753377_chr12_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR753377_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.47 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 54.10 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.41 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.30 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 30.78 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 90.926 sec; CPU: 86.057 sec\n",
      "[Fri Aug 18 23:49:27 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:49:27 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753377_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr12.bam, results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753377_chr12.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753377_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 23:50:05 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:50:05 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753377_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753377_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "23:50:06.030 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 23:50:06 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753377_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753377_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 23:50:06 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 23:50:06\tMarkDuplicates\tStart of doWork freeMemory: 528786096; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:50:06\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 23:50:06\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam\n",
      "INFO\t2023-08-18 23:50:07\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:57,029,289\n",
      "INFO\t2023-08-18 23:50:07\tMarkDuplicates\tTracking 396 as yet unmatched pairs. 396 records in RAM.\n",
      "INFO\t2023-08-18 23:50:09\tMarkDuplicates\tRead 1953622 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 23:50:09\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 255391168; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:50:09\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 23:50:09\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 23:50:09\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 23:50:09\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 23:50:09\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 831470904; totalMemory: 1377828864; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:50:09\tMarkDuplicates\tMarking 278994 records as duplicates.\n",
      "INFO\t2023-08-18 23:50:09\tMarkDuplicates\tFound 103 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam\n",
      "INFO\t2023-08-18 23:50:09\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 23:50:22\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 23:50:22\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 23:50:22\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 23:50:22\tMarkDuplicates\tBefore output close freeMemory: 528083080; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:50:22\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 23:50:22\tMarkDuplicates\tAfter output close freeMemory: 528083080; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 23:50:22 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.28 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 23:50:24 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T234633.885413.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:50:25 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753378_chr12_1_fastp.fastq.gz, data/processed/ERR753378_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753378_chr12.sam, results/mapped_reads/ERR753378_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753378_chr12_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753378_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.46 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 51.94 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.41 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.28 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 29.87 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 87.892 sec; CPU: 82.969 sec\n",
      "[Fri Aug 18 23:54:06 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:54:06 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753378_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr12.bam, results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753378_chr12.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753378_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 23:54:53 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:54:53 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753378_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753378_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "23:54:54.840 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Fri Aug 18 23:54:54 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753378_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753378_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Fri Aug 18 23:54:54 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-18 23:54:54\tMarkDuplicates\tStart of doWork freeMemory: 528816736; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:54:54\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-18 23:54:54\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam\n",
      "INFO\t2023-08-18 23:54:56\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:47,983,100\n",
      "INFO\t2023-08-18 23:54:56\tMarkDuplicates\tTracking 461 as yet unmatched pairs. 461 records in RAM.\n",
      "INFO\t2023-08-18 23:54:57\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 12:90,841,669\n",
      "INFO\t2023-08-18 23:54:57\tMarkDuplicates\tTracking 410 as yet unmatched pairs. 410 records in RAM.\n",
      "INFO\t2023-08-18 23:54:59\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 12:132,671,826\n",
      "INFO\t2023-08-18 23:54:59\tMarkDuplicates\tTracking 25 as yet unmatched pairs. 25 records in RAM.\n",
      "INFO\t2023-08-18 23:54:59\tMarkDuplicates\tRead 3037191 records. 0 pairs never matched.\n",
      "INFO\t2023-08-18 23:54:59\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 424830472; totalMemory: 823132160; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:54:59\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-18 23:54:59\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-18 23:55:00\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-18 23:55:00\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-18 23:55:00\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1035975704; totalMemory: 1582301184; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:55:00\tMarkDuplicates\tMarking 71536 records as duplicates.\n",
      "INFO\t2023-08-18 23:55:00\tMarkDuplicates\tFound 500 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam\n",
      "INFO\t2023-08-18 23:55:00\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-18 23:55:18\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-18 23:55:18\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-18 23:55:18\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-18 23:55:18\tMarkDuplicates\tBefore output close freeMemory: 528114920; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-18 23:55:18\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-18 23:55:18\tMarkDuplicates\tAfter output close freeMemory: 528114920; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Fri Aug 18 23:55:18 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.40 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Fri Aug 18 23:55:21 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T235024.398102.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:55:22 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR696683_chr12_dedup.bam\n",
      "    output: results/variants/ERR696683_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR696683_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR696683_chr12.vcf\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "index file data/reference/genome.fa.fai not found, generating...\n",
      "[Fri Aug 18 23:56:10 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T235521.228398.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:56:12 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753368_chr12_dedup.bam\n",
      "    output: results/variants/ERR753368_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753368_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753368_chr12.vcf\n",
      "    wildcards: sample=ERR753368_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 23:57:18 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T235611.051099.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:57:20 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753369_chr12_dedup.bam\n",
      "    output: results/variants/ERR753369_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753369_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753369_chr12.vcf\n",
      "    wildcards: sample=ERR753369_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Fri Aug 18 23:58:30 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T235719.007548.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Fri Aug 18 23:58:31 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753370_chr12_dedup.bam\n",
      "    output: results/variants/ERR753370_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753370_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753370_chr12.vcf\n",
      "    wildcards: sample=ERR753370_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 00:00:23 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-18T235830.645787.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:00:24 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753371_chr12_dedup.bam\n",
      "    output: results/variants/ERR753371_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753371_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753371_chr12.vcf\n",
      "    wildcards: sample=ERR753371_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 00:00:58 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T000023.839700.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:01:00 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753372_chr12_dedup.bam\n",
      "    output: results/variants/ERR753372_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753372_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753372_chr12.vcf\n",
      "    wildcards: sample=ERR753372_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 00:02:09 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T000058.949699.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:02:11 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753373_chr12_dedup.bam\n",
      "    output: results/variants/ERR753373_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753373_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753373_chr12.vcf\n",
      "    wildcards: sample=ERR753373_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 00:03:22 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T000209.739190.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:03:24 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753374_chr12_dedup.bam\n",
      "    output: results/variants/ERR753374_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753374_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753374_chr12.vcf\n",
      "    wildcards: sample=ERR753374_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 00:04:12 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T000323.366745.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:04:13 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753375_chr12_dedup.bam\n",
      "    output: results/variants/ERR753375_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753375_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753375_chr12.vcf\n",
      "    wildcards: sample=ERR753375_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 00:04:51 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T000412.391493.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:04:52 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753376_chr12_dedup.bam\n",
      "    output: results/variants/ERR753376_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753376_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753376_chr12.vcf\n",
      "    wildcards: sample=ERR753376_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 00:05:31 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T000451.286991.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:05:32 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753377_chr12_dedup.bam\n",
      "    output: results/variants/ERR753377_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753377_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753377_chr12.vcf\n",
      "    wildcards: sample=ERR753377_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 00:06:13 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T000531.826655.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:06:14 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753378_chr12_dedup.bam\n",
      "    output: results/variants/ERR753378_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753378_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753378_chr12.vcf\n",
      "    wildcards: sample=ERR753378_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 00:07:16 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T000613.487879.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:07:17 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR696683_chr12.vcf\n",
      "    output: results/variants/vep/ERR696683_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR696683_chr12.txt\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Sat Aug 19 00:07:47 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T000716.661030.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:07:49 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753368_chr12.vcf\n",
      "    output: results/variants/vep/ERR753368_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753368_chr12.txt\n",
      "    wildcards: sample=ERR753368_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Sat Aug 19 00:08:19 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T000748.147912.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:08:20 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753369_chr12.vcf\n",
      "    output: results/variants/vep/ERR753369_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753369_chr12.txt\n",
      "    wildcards: sample=ERR753369_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Sat Aug 19 00:08:49 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T000819.744554.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:08:50 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753370_chr12.vcf\n",
      "    output: results/variants/vep/ERR753370_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753370_chr12.txt\n",
      "    wildcards: sample=ERR753370_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Sat Aug 19 00:09:22 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T000849.863176.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:09:23 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753371_chr12.vcf\n",
      "    output: results/variants/vep/ERR753371_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753371_chr12.txt\n",
      "    wildcards: sample=ERR753371_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Sat Aug 19 00:09:44 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T000922.951437.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:09:45 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753372_chr12.vcf\n",
      "    output: results/variants/vep/ERR753372_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753372_chr12.txt\n",
      "    wildcards: sample=ERR753372_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Sat Aug 19 00:10:13 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T000944.568151.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:10:15 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753373_chr12.vcf\n",
      "    output: results/variants/vep/ERR753373_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753373_chr12.txt\n",
      "    wildcards: sample=ERR753373_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Sat Aug 19 00:10:41 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T001014.010937.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:10:43 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753374_chr12.vcf\n",
      "    output: results/variants/vep/ERR753374_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753374_chr12.txt\n",
      "    wildcards: sample=ERR753374_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Sat Aug 19 00:11:04 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T001042.079978.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:11:05 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753375_chr12.vcf\n",
      "    output: results/variants/vep/ERR753375_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753375_chr12.txt\n",
      "    wildcards: sample=ERR753375_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Sat Aug 19 00:11:27 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T001104.314931.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:11:28 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753376_chr12.vcf\n",
      "    output: results/variants/vep/ERR753376_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753376_chr12.txt\n",
      "    wildcards: sample=ERR753376_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Sat Aug 19 00:11:51 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T001127.682800.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:11:52 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753377_chr12.vcf\n",
      "    output: results/variants/vep/ERR753377_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753377_chr12.txt\n",
      "    wildcards: sample=ERR753377_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Sat Aug 19 00:12:14 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T001151.238906.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:12:15 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753378_chr12.vcf\n",
      "    output: results/variants/vep/ERR753378_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753378_chr12.txt\n",
      "    wildcards: sample=ERR753378_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Sat Aug 19 00:12:43 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T001214.761073.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job              count    min threads    max threads\n",
      "-------------  -------  -------------  -------------\n",
      "parsing_dataR        1              1              1\n",
      "total                1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:12:43 2023]\n",
      "rule parsing_dataR:\n",
      "    input: code/05parsing_vep_data.R\n",
      "    output: tasks/13parsing_dataR.done\n",
      "    jobid: 0\n",
      "    reason: Params have changed since last execution\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Warning message:\n",
      "package ‘glue’ was built under R version 4.2.3 \n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">>> This is your sample list: ERR696683_chr12\n",
      ">>> This is your sample list: ERR753368_chr12\n",
      ">>> This is your sample list: ERR753369_chr12\n",
      ">>> This is your sample list: ERR753370_chr12\n",
      ">>> This is your sample list: ERR753371_chr12\n",
      ">>> This is your sample list: ERR753372_chr12\n",
      ">>> This is your sample list: ERR753373_chr12\n",
      ">>> This is your sample list: ERR753374_chr12\n",
      ">>> This is your sample list: ERR753375_chr12\n",
      ">>> This is your sample list: ERR753376_chr12\n",
      ">>> This is your sample list: ERR753377_chr12\n",
      ">>> This is your sample list: ERR753378_chr12\n",
      ">>> This is your gene: KRAS\n",
      "# A tibble: 0 × 44\n",
      "# ℹ 44 variables: #uploaded_variation <chr>, location <chr>, allele <chr>,\n",
      "#   gene <chr>, feature <chr>, feature_type <chr>, consequence <chr>,\n",
      "#   cdna_position <chr>, cds_position <chr>, protein_position <chr>,\n",
      "#   amino_acids <chr>, codons <chr>, existing_variation <chr>, impact <chr>,\n",
      "#   distance <chr>, strand <chr>, flags <chr>, variant_class <chr>,\n",
      "#   symbol <chr>, symbol_source <chr>, hgnc_id <chr>, biotype <chr>,\n",
      "#   mane_select <chr>, mane_plus_clinical <chr>, tsl <chr>, appris <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 24 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "3 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "4 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "# ℹ 18 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 36 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "3 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "4 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "5 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "# ℹ 30 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 36 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25209531_G/A       12:252095… A      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 12_25209531_G/A       12:252095… A      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "3 12_25209531_G/A       12:252095… A      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "4 12_25209531_G/A       12:252095… A      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 12_25209531_G/A       12:252095… A      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 12_25209531_G/A       12:252095… A      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "# ℹ 30 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 0 × 44\n",
      "# ℹ 44 variables: #uploaded_variation <chr>, location <chr>, allele <chr>,\n",
      "#   gene <chr>, feature <chr>, feature_type <chr>, consequence <chr>,\n",
      "#   cdna_position <chr>, cds_position <chr>, protein_position <chr>,\n",
      "#   amino_acids <chr>, codons <chr>, existing_variation <chr>, impact <chr>,\n",
      "#   distance <chr>, strand <chr>, flags <chr>, variant_class <chr>,\n",
      "#   symbol <chr>, symbol_source <chr>, hgnc_id <chr>, biotype <chr>,\n",
      "#   mane_select <chr>, mane_plus_clinical <chr>, tsl <chr>, appris <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 36 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "3 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "4 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "# ℹ 30 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 25 × 44\n",
      "  `#uploaded_variation`   location allele gene  feature feature_type consequence\n",
      "  <chr>                   <chr>    <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25230796_AATTAATTA/… 12:2523… AATTA  ENSG… ENST00… Transcript   intron_var…\n",
      "2 12_25230796_AATTAATTA/… 12:2523… AATTA  ENSG… ENST00… Transcript   intron_var…\n",
      "3 12_25230796_AATTAATTA/… 12:2523… AATTA  ENSG… ENST00… Transcript   downstream…\n",
      "4 12_25230796_AATTAATTA/… 12:2523… AATTA  ENSG… ENST00… Transcript   intron_var…\n",
      "5 12_25230796_AATTAATTA/… 12:2523… AATTA  ENSG… ENST00… Transcript   intron_var…\n",
      "6 12_25230796_AATTAATTA/… 12:2523… AATTA  ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 19 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 0 × 44\n",
      "# ℹ 44 variables: #uploaded_variation <chr>, location <chr>, allele <chr>,\n",
      "#   gene <chr>, feature <chr>, feature_type <chr>, consequence <chr>,\n",
      "#   cdna_position <chr>, cds_position <chr>, protein_position <chr>,\n",
      "#   amino_acids <chr>, codons <chr>, existing_variation <chr>, impact <chr>,\n",
      "#   distance <chr>, strand <chr>, flags <chr>, variant_class <chr>,\n",
      "#   symbol <chr>, symbol_source <chr>, hgnc_id <chr>, biotype <chr>,\n",
      "#   mane_select <chr>, mane_plus_clinical <chr>, tsl <chr>, appris <chr>, …\n",
      "# A tibble: 24 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "3 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "4 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "5 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "# ℹ 18 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 12 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25209618_A/C       12:252096… C      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 12_25209618_A/C       12:252096… C      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "3 12_25209618_A/C       12:252096… C      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "4 12_25209618_A/C       12:252096… C      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 12_25209618_A/C       12:252096… C      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 12_25209618_A/C       12:252096… C      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "# ℹ 6 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 12 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "3 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "4 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "5 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "# ℹ 6 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 36 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "3 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "4 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "5 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "# ℹ 30 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "[1] \"##==========================##\"\n",
      "[1] \"##===> WORK FINISHED!!! <===##\"\n",
      "[1] \"##==========================##\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "Touching output file tasks/13parsing_dataR.done.\n",
      "[Sat Aug 19 00:12:47 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T001243.679343.snakemake.log\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['snakemake', '--cores', '1', 'parsing_dataR'], returncode=0)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Running the workflow for the 12º chormosome\n",
    "\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--force\", \"data/reference/genome.fa\"]) ## rule reference_genome \n",
    "\n",
    "for sample in sample_list:\n",
    "    ## Quality inspection\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/fastqc_result/\" + sample + \"_chr12_1_fastqc.html\",   \n",
    "                    \"data/processed/\" + sample + \"_chr12_1_fastp.fastq.gz\",   \n",
    "                    \"results/fastqc_result/trimmed/\" + sample + \"_chr12_1_fastp_fastqc.html\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    ## Mapping reads\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/mapped_reads/\" + sample + \"_chr12_sorted.sam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr12_sorted.bam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr12_dedup.bam\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    # Extracting variants  \n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/\" + sample + \"_chr12.vcf\"])   \n",
    "\n",
    "## VEP CLI\n",
    "for sample in sample_list:\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/vep/\" + sample + \"_chr12.txt\"])\n",
    "\n",
    "## Extracting the gene PIKCA3 from the VEP files\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"parsing_dataR\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Changing the config file to do the chromosome 17 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['sed', '-i', '50s/KRAS/TP53/g', 'config.yaml'], returncode=0)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subprocess.run([\"sed\", \"-i\", \"6,+12s/chr12/chr17/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\",\"24s/chr12/chr17/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"31s/chromosome.12.fa.gz/chromosome.17.fa.gz/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"50s/KRAS/TP53/g\", \"config.yaml\"])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job                 count    min threads    max threads\n",
      "----------------  -------  -------------  -------------\n",
      "reference_genome        1              1              1\n",
      "total                   1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:13:23 2023]\n",
      "rule reference_genome:\n",
      "    output: data/reference/genome.fa\n",
      "    jobid: 0\n",
      "    reason: Forced execution\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "--2023-08-19 00:13:23--  https://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.17.fa.gz\n",
      "Resolviendo ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.169\n",
      "Conectando con ftp.ensembl.org (ftp.ensembl.org)[193.62.193.169]:443... conectado.\n",
      "Petición HTTP enviada, esperando respuesta... 200 OK\n",
      "Longitud: 24104573 (23M) [application/x-gzip]\n",
      "Guardando como: ‘data/reference/genome.fa.gz’\n",
      "\n",
      "     0K .......... .......... .......... .......... ..........  0%  443K 53s\n",
      "    50K .......... .......... .......... .......... ..........  0%  901K 39s\n",
      "   100K .......... .......... .......... .......... ..........  0% 21,4M 27s\n",
      "   150K .......... .......... .......... .......... ..........  0% 51,1M 20s\n",
      "   200K .......... .......... .......... .......... ..........  1%  894K 21s\n",
      "   250K .......... .......... .......... .......... ..........  1% 50,5M 18s\n",
      "   300K .......... .......... .......... .......... ..........  1% 85,8M 15s\n",
      "   350K .......... .......... .......... .......... ..........  1%  154M 13s\n",
      "   400K .......... .......... .......... .......... ..........  1%  949K 14s\n",
      "   450K .......... .......... .......... .......... ..........  2% 21,9M 13s\n",
      "   500K .......... .......... .......... .......... ..........  2% 16,1M 12s\n",
      "   550K .......... .......... .......... .......... ..........  2% 78,3M 11s\n",
      "   600K .......... .......... .......... .......... ..........  2% 72,9M 10s\n",
      "   650K .......... .......... .......... .......... ..........  2% 95,6M 9s\n",
      "   700K .......... .......... .......... .......... ..........  3% 81,5M 9s\n",
      "   750K .......... .......... .......... .......... ..........  3% 92,6M 8s\n",
      "   800K .......... .......... .......... .......... ..........  3% 50,5M 8s\n",
      "   850K .......... .......... .......... .......... ..........  3%  302K 11s\n",
      "   900K .......... .......... .......... .......... ..........  4% 75,0M 11s\n",
      "   950K .......... .......... .......... .......... ..........  4%  231M 10s\n",
      "  1000K .......... .......... .......... .......... ..........  4%  145M 10s\n",
      "  1050K .......... .......... .......... .......... ..........  4%  145M 9s\n",
      "  1100K .......... .......... .......... .......... ..........  4%  130M 9s\n",
      "  1150K .......... .......... .......... .......... ..........  5%  160M 9s\n",
      "  1200K .......... .......... .......... .......... ..........  5%  141M 8s\n",
      "  1250K .......... .......... .......... .......... ..........  5% 3,69M 8s\n",
      "  1300K .......... .......... .......... .......... ..........  5%  165M 8s\n",
      "  1350K .......... .......... .......... .......... ..........  5%  139M 7s\n",
      "  1400K .......... .......... .......... .......... ..........  6%  161M 7s\n",
      "  1450K .......... .......... .......... .......... ..........  6% 3,11M 7s\n",
      "  1500K .......... .......... .......... .......... ..........  6% 50,6M 7s\n",
      "  1550K .......... .......... .......... .......... ..........  6%  102M 7s\n",
      "  1600K .......... .......... .......... .......... ..........  7%  167M 7s\n",
      "  1650K .......... .......... .......... .......... ..........  7% 96,9M 6s\n",
      "  1700K .......... .......... .......... .......... ..........  7% 82,7M 6s\n",
      "  1750K .......... .......... .......... .......... ..........  7% 2,32M 6s\n",
      "  1800K .......... .......... .......... .......... ..........  7% 95,5M 6s\n",
      "  1850K .......... .......... .......... .......... ..........  8% 84,3M 6s\n",
      "  1900K .......... .......... .......... .......... ..........  8%  144M 6s\n",
      "  1950K .......... .......... .......... .......... ..........  8%  130M 6s\n",
      "  2000K .......... .......... .......... .......... ..........  8% 1,42M 6s\n",
      "  2050K .......... .......... .......... .......... ..........  8%  122M 6s\n",
      "  2100K .......... .......... .......... .......... ..........  9%  125M 5s\n",
      "  2150K .......... .......... .......... .......... ..........  9%  119M 5s\n",
      "  2200K .......... .......... .......... .......... ..........  9%  102M 5s\n",
      "  2250K .......... .......... .......... .......... ..........  9% 77,7M 5s\n",
      "  2300K .......... .......... .......... .......... ..........  9% 61,0M 5s\n",
      "  2350K .......... .......... .......... .......... .......... 10% 96,0M 5s\n",
      "  2400K .......... .......... .......... .......... .......... 10%  116M 5s\n",
      "  2450K .......... .......... .......... .......... .......... 10%  100M 5s\n",
      "  2500K .......... .......... .......... .......... .......... 10%  181M 5s\n",
      "  2550K .......... .......... .......... .......... .......... 11% 4,67M 5s\n",
      "  2600K .......... .......... .......... .......... .......... 11% 91,3M 4s\n",
      "  2650K .......... .......... .......... .......... .......... 11%  737K 5s\n",
      "  2700K .......... .......... .......... .......... .......... 11% 88,8M 5s\n",
      "  2750K .......... .......... .......... .......... .......... 11%  110M 5s\n",
      "  2800K .......... .......... .......... .......... .......... 12%  114M 5s\n",
      "  2850K .......... .......... .......... .......... .......... 12% 1,12M 5s\n",
      "  2900K .......... .......... .......... .......... .......... 12% 63,8M 5s\n",
      "  2950K .......... .......... .......... .......... .......... 12% 25,9M 5s\n",
      "  3000K .......... .......... .......... .......... .......... 12% 23,5M 5s\n",
      "  3050K .......... .......... .......... .......... .......... 13% 78,4M 5s\n",
      "  3100K .......... .......... .......... .......... .......... 13% 78,6M 4s\n",
      "  3150K .......... .......... .......... .......... .......... 13% 88,7M 4s\n",
      "  3200K .......... .......... .......... .......... .......... 13% 69,8M 4s\n",
      "  3250K .......... .......... .......... .......... .......... 14% 81,0M 4s\n",
      "  3300K .......... .......... .......... .......... .......... 14% 89,1M 4s\n",
      "  3350K .......... .......... .......... .......... .......... 14% 71,7M 4s\n",
      "  3400K .......... .......... .......... .......... .......... 14% 91,6M 4s\n",
      "  3450K .......... .......... .......... .......... .......... 14% 82,1M 4s\n",
      "  3500K .......... .......... .......... .......... .......... 15% 92,3M 4s\n",
      "  3550K .......... .......... .......... .......... .......... 15% 4,17M 4s\n",
      "  3600K .......... .......... .......... .......... .......... 15% 62,3M 4s\n",
      "  3650K .......... .......... .......... .......... .......... 15% 77,7M 4s\n",
      "  3700K .......... .......... .......... .......... .......... 15% 58,3M 4s\n",
      "  3750K .......... .......... .......... .......... .......... 16%  141M 4s\n",
      "  3800K .......... .......... .......... .......... .......... 16%  159M 4s\n",
      "  3850K .......... .......... .......... .......... .......... 16%  141M 4s\n",
      "  3900K .......... .......... .......... .......... .......... 16%  165M 3s\n",
      "  3950K .......... .......... .......... .......... .......... 16%  145M 3s\n",
      "  4000K .......... .......... .......... .......... .......... 17% 2,37M 3s\n",
      "  4050K .......... .......... .......... .......... .......... 17%  118M 3s\n",
      "  4100K .......... .......... .......... .......... .......... 17%  103M 3s\n",
      "  4150K .......... .......... .......... .......... .......... 17% 4,30M 3s\n",
      "  4200K .......... .......... .......... .......... .......... 18%  101M 3s\n",
      "  4250K .......... .......... .......... .......... .......... 18% 1,45M 3s\n",
      "  4300K .......... .......... .......... .......... .......... 18%  109M 3s\n",
      "  4350K .......... .......... .......... .......... .......... 18% 88,0M 3s\n",
      "  4400K .......... .......... .......... .......... .......... 18%  111M 3s\n",
      "  4450K .......... .......... .......... .......... .......... 19% 48,5M 3s\n",
      "  4500K .......... .......... .......... .......... .......... 19%  107M 3s\n",
      "  4550K .......... .......... .......... .......... .......... 19%  130M 3s\n",
      "  4600K .......... .......... .......... .......... .......... 19% 27,5M 3s\n",
      "  4650K .......... .......... .......... .......... .......... 19%  106M 3s\n",
      "  4700K .......... .......... .......... .......... .......... 20% 99,9M 3s\n",
      "  4750K .......... .......... .......... .......... .......... 20% 48,1M 3s\n",
      "  4800K .......... .......... .......... .......... .......... 20% 88,7M 3s\n",
      "  4850K .......... .......... .......... .......... .......... 20% 45,9M 3s\n",
      "  4900K .......... .......... .......... .......... .......... 21% 61,2M 3s\n",
      "  4950K .......... .......... .......... .......... .......... 21%  113M 3s\n",
      "  5000K .......... .......... .......... .......... .......... 21%  116M 3s\n",
      "  5050K .......... .......... .......... .......... .......... 21%  118M 3s\n",
      "  5100K .......... .......... .......... .......... .......... 21% 93,9M 3s\n",
      "  5150K .......... .......... .......... .......... .......... 22%  104M 3s\n",
      "  5200K .......... .......... .......... .......... .......... 22% 65,4M 3s\n",
      "  5250K .......... .......... .......... .......... .......... 22%  117M 3s\n",
      "  5300K .......... .......... .......... .......... .......... 22%  123M 3s\n",
      "  5350K .......... .......... .......... .......... .......... 22%  103M 3s\n",
      "  5400K .......... .......... .......... .......... .......... 23%  118M 3s\n",
      "  5450K .......... .......... .......... .......... .......... 23% 4,56M 3s\n",
      "  5500K .......... .......... .......... .......... .......... 23%  101M 3s\n",
      "  5550K .......... .......... .......... .......... .......... 23% 27,9M 3s\n",
      "  5600K .......... .......... .......... .......... .......... 24% 90,8M 3s\n",
      "  5650K .......... .......... .......... .......... .......... 24%  120M 3s\n",
      "  5700K .......... .......... .......... .......... .......... 24% 99,6M 2s\n",
      "  5750K .......... .......... .......... .......... .......... 24%  115M 2s\n",
      "  5800K .......... .......... .......... .......... .......... 24% 36,6M 2s\n",
      "  5850K .......... .......... .......... .......... .......... 25%  104M 2s\n",
      "  5900K .......... .......... .......... .......... .......... 25%  115M 2s\n",
      "  5950K .......... .......... .......... .......... .......... 25%  106M 2s\n",
      "  6000K .......... .......... .......... .......... .......... 25%  116M 2s\n",
      "  6050K .......... .......... .......... .......... .......... 25% 11,8M 2s\n",
      "  6100K .......... .......... .......... .......... .......... 26%  119M 2s\n",
      "  6150K .......... .......... .......... .......... .......... 26%  149M 2s\n",
      "  6200K .......... .......... .......... .......... .......... 26%  105M 2s\n",
      "  6250K .......... .......... .......... .......... .......... 26%  126M 2s\n",
      "  6300K .......... .......... .......... .......... .......... 26% 2,22M 2s\n",
      "  6350K .......... .......... .......... .......... .......... 27%  125M 2s\n",
      "  6400K .......... .......... .......... .......... .......... 27%  165M 2s\n",
      "  6450K .......... .......... .......... .......... .......... 27% 1,87M 2s\n",
      "  6500K .......... .......... .......... .......... .......... 27%  132M 2s\n",
      "  6550K .......... .......... .......... .......... .......... 28% 74,5M 2s\n",
      "  6600K .......... .......... .......... .......... .......... 28%  101M 2s\n",
      "  6650K .......... .......... .......... .......... .......... 28%  130M 2s\n",
      "  6700K .......... .......... .......... .......... .......... 28%  116M 2s\n",
      "  6750K .......... .......... .......... .......... .......... 28%  140M 2s\n",
      "  6800K .......... .......... .......... .......... .......... 29% 14,4M 2s\n",
      "  6850K .......... .......... .......... .......... .......... 29% 87,4M 2s\n",
      "  6900K .......... .......... .......... .......... .......... 29%  122M 2s\n",
      "  6950K .......... .......... .......... .......... .......... 29% 61,9M 2s\n",
      "  7000K .......... .......... .......... .......... .......... 29%  128M 2s\n",
      "  7050K .......... .......... .......... .......... .......... 30%  124M 2s\n",
      "  7100K .......... .......... .......... .......... .......... 30%  108M 2s\n",
      "  7150K .......... .......... .......... .......... .......... 30%  134M 2s\n",
      "  7200K .......... .......... .......... .......... .......... 30% 18,7M 2s\n",
      "  7250K .......... .......... .......... .......... .......... 31% 76,4M 2s\n",
      "  7300K .......... .......... .......... .......... .......... 31%  118M 2s\n",
      "  7350K .......... .......... .......... .......... .......... 31%  101M 2s\n",
      "  7400K .......... .......... .......... .......... .......... 31%  115M 2s\n",
      "  7450K .......... .......... .......... .......... .......... 31%  105M 2s\n",
      "  7500K .......... .......... .......... .......... .......... 32% 1,50M 2s\n",
      "  7550K .......... .......... .......... .......... .......... 32%  129M 2s\n",
      "  7600K .......... .......... .......... .......... .......... 32%  104M 2s\n",
      "  7650K .......... .......... .......... .......... .......... 32%  118M 2s\n",
      "  7700K .......... .......... .......... .......... .......... 32% 99,2M 2s\n",
      "  7750K .......... .......... .......... .......... .......... 33%  120M 2s\n",
      "  7800K .......... .......... .......... .......... .......... 33% 70,4M 2s\n",
      "  7850K .......... .......... .......... .......... .......... 33%  107M 2s\n",
      "  7900K .......... .......... .......... .......... .......... 33%  119M 2s\n",
      "  7950K .......... .......... .......... .......... .......... 33% 99,8M 2s\n",
      "  8000K .......... .......... .......... .......... .......... 34%  127M 2s\n",
      "  8050K .......... .......... .......... .......... .......... 34%  117M 2s\n",
      "  8100K .......... .......... .......... .......... .......... 34%  101M 2s\n",
      "  8150K .......... .......... .......... .......... .......... 34%  118M 2s\n",
      "  8200K .......... .......... .......... .......... .......... 35% 7,45M 2s\n",
      "  8250K .......... .......... .......... .......... .......... 35%  106M 2s\n",
      "  8300K .......... .......... .......... .......... .......... 35%  110M 2s\n",
      "  8350K .......... .......... .......... .......... .......... 35% 93,4M 2s\n",
      "  8400K .......... .......... .......... .......... .......... 35% 23,4M 2s\n",
      "  8450K .......... .......... .......... .......... .......... 36%  102M 2s\n",
      "  8500K .......... .......... .......... .......... .......... 36%  114M 2s\n",
      "  8550K .......... .......... .......... .......... .......... 36%  125M 2s\n",
      "  8600K .......... .......... .......... .......... .......... 36% 99,7M 2s\n",
      "  8650K .......... .......... .......... .......... .......... 36%  123M 2s\n",
      "  8700K .......... .......... .......... .......... .......... 37% 1,24M 2s\n",
      "  8750K .......... .......... .......... .......... .......... 37%  119M 2s\n",
      "  8800K .......... .......... .......... .......... .......... 37% 38,6M 2s\n",
      "  8850K .......... .......... .......... .......... .......... 37% 67,2M 2s\n",
      "  8900K .......... .......... .......... .......... .......... 38%  169M 2s\n",
      "  8950K .......... .......... .......... .......... .......... 38%  145M 2s\n",
      "  9000K .......... .......... .......... .......... .......... 38% 72,6M 2s\n",
      "  9050K .......... .......... .......... .......... .......... 38%  141M 2s\n",
      "  9100K .......... .......... .......... .......... .......... 38%  114M 2s\n",
      "  9150K .......... .......... .......... .......... .......... 39%  144M 2s\n",
      "  9200K .......... .......... .......... .......... .......... 39%  118M 1s\n",
      "  9250K .......... .......... .......... .......... .......... 39% 86,0M 1s\n",
      "  9300K .......... .......... .......... .......... .......... 39%  100M 1s\n",
      "  9350K .......... .......... .......... .......... .......... 39% 98,8M 1s\n",
      "  9400K .......... .......... .......... .......... .......... 40%  121M 1s\n",
      "  9450K .......... .......... .......... .......... .......... 40%  104M 1s\n",
      "  9500K .......... .......... .......... .......... .......... 40%  118M 1s\n",
      "  9550K .......... .......... .......... .......... .......... 40%  119M 1s\n",
      "  9600K .......... .......... .......... .......... .......... 40%  103M 1s\n",
      "  9650K .......... .......... .......... .......... .......... 41%  118M 1s\n",
      "  9700K .......... .......... .......... .......... .......... 41%  102M 1s\n",
      "  9750K .......... .......... .......... .......... .......... 41%  119M 1s\n",
      "  9800K .......... .......... .......... .......... .......... 41%  122M 1s\n",
      "  9850K .......... .......... .......... .......... .......... 42%  100M 1s\n",
      "  9900K .......... .......... .......... .......... .......... 42% 10,5M 1s\n",
      "  9950K .......... .......... .......... .......... .......... 42%  144M 1s\n",
      " 10000K .......... .......... .......... .......... .......... 42%  115M 1s\n",
      " 10050K .......... .......... .......... .......... .......... 42%  121M 1s\n",
      " 10100K .......... .......... .......... .......... .......... 43%  102M 1s\n",
      " 10150K .......... .......... .......... .......... .......... 43%  118M 1s\n",
      " 10200K .......... .......... .......... .......... .......... 43% 1,31M 1s\n",
      " 10250K .......... .......... .......... .......... .......... 43%  135M 1s\n",
      " 10300K .......... .......... .......... .......... .......... 43%  143M 1s\n",
      " 10350K .......... .......... .......... .......... .......... 44%  103M 1s\n",
      " 10400K .......... .......... .......... .......... .......... 44%  119M 1s\n",
      " 10450K .......... .......... .......... .......... .......... 44% 23,8M 1s\n",
      " 10500K .......... .......... .......... .......... .......... 44%  114M 1s\n",
      " 10550K .......... .......... .......... .......... .......... 45%  110M 1s\n",
      " 10600K .......... .......... .......... .......... .......... 45%  108M 1s\n",
      " 10650K .......... .......... .......... .......... .......... 45%  126M 1s\n",
      " 10700K .......... .......... .......... .......... .......... 45% 58,6M 1s\n",
      " 10750K .......... .......... .......... .......... .......... 45%  101M 1s\n",
      " 10800K .......... .......... .......... .......... .......... 46%  122M 1s\n",
      " 10850K .......... .......... .......... .......... .......... 46%  101M 1s\n",
      " 10900K .......... .......... .......... .......... .......... 46%  120M 1s\n",
      " 10950K .......... .......... .......... .......... .......... 46%  103M 1s\n",
      " 11000K .......... .......... .......... .......... .......... 46% 1,84M 1s\n",
      " 11050K .......... .......... .......... .......... .......... 47% 80,1M 1s\n",
      " 11100K .......... .......... .......... .......... .......... 47%  124M 1s\n",
      " 11150K .......... .......... .......... .......... .......... 47%  141M 1s\n",
      " 11200K .......... .......... .......... .......... .......... 47%  127M 1s\n",
      " 11250K .......... .......... .......... .......... .......... 48%  100M 1s\n",
      " 11300K .......... .......... .......... .......... .......... 48%  125M 1s\n",
      " 11350K .......... .......... .......... .......... .......... 48% 99,8M 1s\n",
      " 11400K .......... .......... .......... .......... .......... 48%  119M 1s\n",
      " 11450K .......... .......... .......... .......... .......... 48%  101M 1s\n",
      " 11500K .......... .......... .......... .......... .......... 49% 90,4M 1s\n",
      " 11550K .......... .......... .......... .......... .......... 49%  118M 1s\n",
      " 11600K .......... .......... .......... .......... .......... 49% 98,6M 1s\n",
      " 11650K .......... .......... .......... .......... .......... 49%  121M 1s\n",
      " 11700K .......... .......... .......... .......... .......... 49%  102M 1s\n",
      " 11750K .......... .......... .......... .......... .......... 50%  102M 1s\n",
      " 11800K .......... .......... .......... .......... .......... 50%  118M 1s\n",
      " 11850K .......... .......... .......... .......... .......... 50%  101M 1s\n",
      " 11900K .......... .......... .......... .......... .......... 50%  121M 1s\n",
      " 11950K .......... .......... .......... .......... .......... 50%  102M 1s\n",
      " 12000K .......... .......... .......... .......... .......... 51%  119M 1s\n",
      " 12050K .......... .......... .......... .......... .......... 51%  116M 1s\n",
      " 12100K .......... .......... .......... .......... .......... 51%  103M 1s\n",
      " 12150K .......... .......... .......... .......... .......... 51% 51,8M 1s\n",
      " 12200K .......... .......... .......... .......... .......... 52%  101M 1s\n",
      " 12250K .......... .......... .......... .......... .......... 52%  122M 1s\n",
      " 12300K .......... .......... .......... .......... .......... 52%  117M 1s\n",
      " 12350K .......... .......... .......... .......... .......... 52%  103M 1s\n",
      " 12400K .......... .......... .......... .......... .......... 52%  117M 1s\n",
      " 12450K .......... .......... .......... .......... .......... 53% 1,66M 1s\n",
      " 12500K .......... .......... .......... .......... .......... 53%  132M 1s\n",
      " 12550K .......... .......... .......... .......... .......... 53%  117M 1s\n",
      " 12600K .......... .......... .......... .......... .......... 53% 3,62M 1s\n",
      " 12650K .......... .......... .......... .......... .......... 53% 1,75M 1s\n",
      " 12700K .......... .......... .......... .......... .......... 54% 63,6M 1s\n",
      " 12750K .......... .......... .......... .......... .......... 54% 77,8M 1s\n",
      " 12800K .......... .......... .......... .......... .......... 54%  106M 1s\n",
      " 12850K .......... .......... .......... .......... .......... 54%  102M 1s\n",
      " 12900K .......... .......... .......... .......... .......... 55%  116M 1s\n",
      " 12950K .......... .......... .......... .......... .......... 55% 96,6M 1s\n",
      " 13000K .......... .......... .......... .......... .......... 55%  110M 1s\n",
      " 13050K .......... .......... .......... .......... .......... 55%  121M 1s\n",
      " 13100K .......... .......... .......... .......... .......... 55% 4,57M 1s\n",
      " 13150K .......... .......... .......... .......... .......... 56% 5,58M 1s\n",
      " 13200K .......... .......... .......... .......... .......... 56%  104M 1s\n",
      " 13250K .......... .......... .......... .......... .......... 56%  118M 1s\n",
      " 13300K .......... .......... .......... .......... .......... 56%  118M 1s\n",
      " 13350K .......... .......... .......... .......... .......... 56%  102M 1s\n",
      " 13400K .......... .......... .......... .......... .......... 57%  119M 1s\n",
      " 13450K .......... .......... .......... .......... .......... 57%  100M 1s\n",
      " 13500K .......... .......... .......... .......... .......... 57%  121M 1s\n",
      " 13550K .......... .......... .......... .......... .......... 57% 85,5M 1s\n",
      " 13600K .......... .......... .......... .......... .......... 57% 99,8M 1s\n",
      " 13650K .......... .......... .......... .......... .......... 58% 2,55M 1s\n",
      " 13700K .......... .......... .......... .......... .......... 58% 95,7M 1s\n",
      " 13750K .......... .......... .......... .......... .......... 58%  138M 1s\n",
      " 13800K .......... .......... .......... .......... .......... 58%  135M 1s\n",
      " 13850K .......... .......... .......... .......... .......... 59%  105M 1s\n",
      " 13900K .......... .......... .......... .......... .......... 59%  120M 1s\n",
      " 13950K .......... .......... .......... .......... .......... 59% 87,8M 1s\n",
      " 14000K .......... .......... .......... .......... .......... 59%  142M 1s\n",
      " 14050K .......... .......... .......... .......... .......... 59%  123M 1s\n",
      " 14100K .......... .......... .......... .......... .......... 60% 39,2M 1s\n",
      " 14150K .......... .......... .......... .......... .......... 60%  118M 1s\n",
      " 14200K .......... .......... .......... .......... .......... 60% 2,79M 1s\n",
      " 14250K .......... .......... .......... .......... .......... 60%  105M 1s\n",
      " 14300K .......... .......... .......... .......... .......... 60%  109M 1s\n",
      " 14350K .......... .......... .......... .......... .......... 61% 94,8M 1s\n",
      " 14400K .......... .......... .......... .......... .......... 61%  110M 1s\n",
      " 14450K .......... .......... .......... .......... .......... 61% 97,7M 1s\n",
      " 14500K .......... .......... .......... .......... .......... 61%  110M 1s\n",
      " 14550K .......... .......... .......... .......... .......... 62%  109M 1s\n",
      " 14600K .......... .......... .......... .......... .......... 62% 94,7M 1s\n",
      " 14650K .......... .......... .......... .......... .......... 62%  110M 1s\n",
      " 14700K .......... .......... .......... .......... .......... 62% 90,9M 1s\n",
      " 14750K .......... .......... .......... .......... .......... 62%  110M 1s\n",
      " 14800K .......... .......... .......... .......... .......... 63%  114M 1s\n",
      " 14850K .......... .......... .......... .......... .......... 63% 93,7M 1s\n",
      " 14900K .......... .......... .......... .......... .......... 63%  163M 1s\n",
      " 14950K .......... .......... .......... .......... .......... 63% 1,88M 1s\n",
      " 15000K .......... .......... .......... .......... .......... 63%  110M 1s\n",
      " 15050K .......... .......... .......... .......... .......... 64%  125M 1s\n",
      " 15100K .......... .......... .......... .......... .......... 64%  113M 1s\n",
      " 15150K .......... .......... .......... .......... .......... 64% 75,8M 1s\n",
      " 15200K .......... .......... .......... .......... .......... 64%  284M 1s\n",
      " 15250K .......... .......... .......... .......... .......... 64%  202M 1s\n",
      " 15300K .......... .......... .......... .......... .......... 65%  122M 1s\n",
      " 15350K .......... .......... .......... .......... .......... 65%  102M 1s\n",
      " 15400K .......... .......... .......... .......... .......... 65%  119M 1s\n",
      " 15450K .......... .......... .......... .......... .......... 65% 85,1M 1s\n",
      " 15500K .......... .......... .......... .......... .......... 66% 33,1M 1s\n",
      " 15550K .......... .......... .......... .......... .......... 66%  124M 1s\n",
      " 15600K .......... .......... .......... .......... .......... 66% 28,8M 1s\n",
      " 15650K .......... .......... .......... .......... .......... 66%  118M 1s\n",
      " 15700K .......... .......... .......... .......... .......... 66%  104M 1s\n",
      " 15750K .......... .......... .......... .......... .......... 67%  119M 1s\n",
      " 15800K .......... .......... .......... .......... .......... 67% 2,43M 1s\n",
      " 15850K .......... .......... .......... .......... .......... 67%  101M 1s\n",
      " 15900K .......... .......... .......... .......... .......... 67%  121M 1s\n",
      " 15950K .......... .......... .......... .......... .......... 67% 99,7M 1s\n",
      " 16000K .......... .......... .......... .......... .......... 68%  110M 1s\n",
      " 16050K .......... .......... .......... .......... .......... 68%  132M 1s\n",
      " 16100K .......... .......... .......... .......... .......... 68% 99,3M 1s\n",
      " 16150K .......... .......... .......... .......... .......... 68%  126M 1s\n",
      " 16200K .......... .......... .......... .......... .......... 69% 71,7M 1s\n",
      " 16250K .......... .......... .......... .......... .......... 69% 20,4M 1s\n",
      " 16300K .......... .......... .......... .......... .......... 69% 57,4M 1s\n",
      " 16350K .......... .......... .......... .......... .......... 69% 94,4M 1s\n",
      " 16400K .......... .......... .......... .......... .......... 69%  133M 1s\n",
      " 16450K .......... .......... .......... .......... .......... 70%  122M 1s\n",
      " 16500K .......... .......... .......... .......... .......... 70%  150M 1s\n",
      " 16550K .......... .......... .......... .......... .......... 70%  146M 1s\n",
      " 16600K .......... .......... .......... .......... .......... 70%  120M 1s\n",
      " 16650K .......... .......... .......... .......... .......... 70%  133M 1s\n",
      " 16700K .......... .......... .......... .......... .......... 71% 2,57M 1s\n",
      " 16750K .......... .......... .......... .......... .......... 71% 12,7M 1s\n",
      " 16800K .......... .......... .......... .......... .......... 71% 82,3M 1s\n",
      " 16850K .......... .......... .......... .......... .......... 71% 1,91M 1s\n",
      " 16900K .......... .......... .......... .......... .......... 72%  132M 1s\n",
      " 16950K .......... .......... .......... .......... .......... 72%  109M 1s\n",
      " 17000K .......... .......... .......... .......... .......... 72%  622K 1s\n",
      " 17050K .......... .......... .......... .......... .......... 72%  134M 1s\n",
      " 17100K .......... .......... .......... .......... .......... 72%  106M 1s\n",
      " 17150K .......... .......... .......... .......... .......... 73%  124M 1s\n",
      " 17200K .......... .......... .......... .......... .......... 73% 56,0M 1s\n",
      " 17250K .......... .......... .......... .......... .......... 73%  123M 1s\n",
      " 17300K .......... .......... .......... .......... .......... 73%  943K 1s\n",
      " 17350K .......... .......... .......... .......... .......... 73% 89,6M 1s\n",
      " 17400K .......... .......... .......... .......... .......... 74%  120M 1s\n",
      " 17450K .......... .......... .......... .......... .......... 74%  103M 1s\n",
      " 17500K .......... .......... .......... .......... .......... 74% 50,4M 1s\n",
      " 17550K .......... .......... .......... .......... .......... 74%  142M 0s\n",
      " 17600K .......... .......... .......... .......... .......... 74%  113M 0s\n",
      " 17650K .......... .......... .......... .......... .......... 75%  147M 0s\n",
      " 17700K .......... .......... .......... .......... .......... 75%  124M 0s\n",
      " 17750K .......... .......... .......... .......... .......... 75%  148M 0s\n",
      " 17800K .......... .......... .......... .......... .......... 75%  118M 0s\n",
      " 17850K .......... .......... .......... .......... .......... 76%  105M 0s\n",
      " 17900K .......... .......... .......... .......... .......... 76%  120M 0s\n",
      " 17950K .......... .......... .......... .......... .......... 76% 99,4M 0s\n",
      " 18000K .......... .......... .......... .......... .......... 76%  128M 0s\n",
      " 18050K .......... .......... .......... .......... .......... 76%  116M 0s\n",
      " 18100K .......... .......... .......... .......... .......... 77%  104M 0s\n",
      " 18150K .......... .......... .......... .......... .......... 77%  116M 0s\n",
      " 18200K .......... .......... .......... .......... .......... 77%  103M 0s\n",
      " 18250K .......... .......... .......... .......... .......... 77% 21,1M 0s\n",
      " 18300K .......... .......... .......... .......... .......... 77%  120M 0s\n",
      " 18350K .......... .......... .......... .......... .......... 78%  101M 0s\n",
      " 18400K .......... .......... .......... .......... .......... 78%  121M 0s\n",
      " 18450K .......... .......... .......... .......... .......... 78%  101M 0s\n",
      " 18500K .......... .......... .......... .......... .......... 78%  121M 0s\n",
      " 18550K .......... .......... .......... .......... .......... 79%  117M 0s\n",
      " 18600K .......... .......... .......... .......... .......... 79%  100M 0s\n",
      " 18650K .......... .......... .......... .......... .......... 79%  116M 0s\n",
      " 18700K .......... .......... .......... .......... .......... 79%  103M 0s\n",
      " 18750K .......... .......... .......... .......... .......... 79%  117M 0s\n",
      " 18800K .......... .......... .......... .......... .......... 80%  121M 0s\n",
      " 18850K .......... .......... .......... .......... .......... 80% 54,1M 0s\n",
      " 18900K .......... .......... .......... .......... .......... 80%  124M 0s\n",
      " 18950K .......... .......... .......... .......... .......... 80%  128M 0s\n",
      " 19000K .......... .......... .......... .......... .......... 80%  147M 0s\n",
      " 19050K .......... .......... .......... .......... .......... 81%  144M 0s\n",
      " 19100K .......... .......... .......... .......... .......... 81%  137M 0s\n",
      " 19150K .......... .......... .......... .......... .......... 81% 84,6M 0s\n",
      " 19200K .......... .......... .......... .......... .......... 81%  104M 0s\n",
      " 19250K .......... .......... .......... .......... .......... 81%  123M 0s\n",
      " 19300K .......... .......... .......... .......... .......... 82% 85,2M 0s\n",
      " 19350K .......... .......... .......... .......... .......... 82%  100M 0s\n",
      " 19400K .......... .......... .......... .......... .......... 82%  120M 0s\n",
      " 19450K .......... .......... .......... .......... .......... 82%  102M 0s\n",
      " 19500K .......... .......... .......... .......... .......... 83% 2,37M 0s\n",
      " 19550K .......... .......... .......... .......... .......... 83%  103M 0s\n",
      " 19600K .......... .......... .......... .......... .......... 83%  118M 0s\n",
      " 19650K .......... .......... .......... .......... .......... 83%  121M 0s\n",
      " 19700K .......... .......... .......... .......... .......... 83%  101M 0s\n",
      " 19750K .......... .......... .......... .......... .......... 84%  119M 0s\n",
      " 19800K .......... .......... .......... .......... .......... 84% 38,4M 0s\n",
      " 19850K .......... .......... .......... .......... .......... 84%  101M 0s\n",
      " 19900K .......... .......... .......... .......... .......... 84%  120M 0s\n",
      " 19950K .......... .......... .......... .......... .......... 84%  102M 0s\n",
      " 20000K .......... .......... .......... .......... .......... 85%  118M 0s\n",
      " 20050K .......... .......... .......... .......... .......... 85% 5,31M 0s\n",
      " 20100K .......... .......... .......... .......... .......... 85% 98,8M 0s\n",
      " 20150K .......... .......... .......... .......... .......... 85%  119M 0s\n",
      " 20200K .......... .......... .......... .......... .......... 86%  104M 0s\n",
      " 20250K .......... .......... .......... .......... .......... 86%  748K 0s\n",
      " 20300K .......... .......... .......... .......... .......... 86%  141M 0s\n",
      " 20350K .......... .......... .......... .......... .......... 86% 80,2M 0s\n",
      " 20400K .......... .......... .......... .......... .......... 86% 84,9M 0s\n",
      " 20450K .......... .......... .......... .......... .......... 87% 69,4M 0s\n",
      " 20500K .......... .......... .......... .......... .......... 87%  113M 0s\n",
      " 20550K .......... .......... .......... .......... .......... 87%  121M 0s\n",
      " 20600K .......... .......... .......... .......... .......... 87% 82,4M 0s\n",
      " 20650K .......... .......... .......... .......... .......... 87% 80,3M 0s\n",
      " 20700K .......... .......... .......... .......... .......... 88% 86,7M 0s\n",
      " 20750K .......... .......... .......... .......... .......... 88%  109M 0s\n",
      " 20800K .......... .......... .......... .......... .......... 88% 98,0M 0s\n",
      " 20850K .......... .......... .......... .......... .......... 88%  102M 0s\n",
      " 20900K .......... .......... .......... .......... .......... 88%  119M 0s\n",
      " 20950K .......... .......... .......... .......... .......... 89% 64,4M 0s\n",
      " 21000K .......... .......... .......... .......... .......... 89%  119M 0s\n",
      " 21050K .......... .......... .......... .......... .......... 89%  121M 0s\n",
      " 21100K .......... .......... .......... .......... .......... 89% 83,5M 0s\n",
      " 21150K .......... .......... .......... .......... .......... 90%  113M 0s\n",
      " 21200K .......... .......... .......... .......... .......... 90% 75,6M 0s\n",
      " 21250K .......... .......... .......... .......... .......... 90%  124M 0s\n",
      " 21300K .......... .......... .......... .......... .......... 90%  120M 0s\n",
      " 21350K .......... .......... .......... .......... .......... 90%  101M 0s\n",
      " 21400K .......... .......... .......... .......... .......... 91%  121M 0s\n",
      " 21450K .......... .......... .......... .......... .......... 91%  103M 0s\n",
      " 21500K .......... .......... .......... .......... .......... 91%  119M 0s\n",
      " 21550K .......... .......... .......... .......... .......... 91%  117M 0s\n",
      " 21600K .......... .......... .......... .......... .......... 91% 70,2M 0s\n",
      " 21650K .......... .......... .......... .......... .......... 92%  257M 0s\n",
      " 21700K .......... .......... .......... .......... .......... 92%  101M 0s\n",
      " 21750K .......... .......... .......... .......... .......... 92%  119M 0s\n",
      " 21800K .......... .......... .......... .......... .......... 92%  120M 0s\n",
      " 21850K .......... .......... .......... .......... .......... 93%  102M 0s\n",
      " 21900K .......... .......... .......... .......... .......... 93%  118M 0s\n",
      " 21950K .......... .......... .......... .......... .......... 93%  103M 0s\n",
      " 22000K .......... .......... .......... .......... .......... 93%  103M 0s\n",
      " 22050K .......... .......... .......... .......... .......... 93%  128M 0s\n",
      " 22100K .......... .......... .......... .......... .......... 94%  108M 0s\n",
      " 22150K .......... .......... .......... .......... .......... 94%  118M 0s\n",
      " 22200K .......... .......... .......... .......... .......... 94%  103M 0s\n",
      " 22250K .......... .......... .......... .......... .......... 94%  119M 0s\n",
      " 22300K .......... .......... .......... .......... .......... 94%  119M 0s\n",
      " 22350K .......... .......... .......... .......... .......... 95%  101M 0s\n",
      " 22400K .......... .......... .......... .......... .......... 95%  120M 0s\n",
      " 22450K .......... .......... .......... .......... .......... 95%  740K 0s\n",
      " 22500K .......... .......... .......... .......... .......... 95%  101M 0s\n",
      " 22550K .......... .......... .......... .......... .......... 96%  105M 0s\n",
      " 22600K .......... .......... .......... .......... .......... 96% 90,0M 0s\n",
      " 22650K .......... .......... .......... .......... .......... 96%  109M 0s\n",
      " 22700K .......... .......... .......... .......... .......... 96% 93,0M 0s\n",
      " 22750K .......... .......... .......... .......... .......... 96%  100M 0s\n",
      " 22800K .......... .......... .......... .......... .......... 97%  112M 0s\n",
      " 22850K .......... .......... .......... .......... .......... 97% 92,7M 0s\n",
      " 22900K .......... .......... .......... .......... .......... 97%  107M 0s\n",
      " 22950K .......... .......... .......... .......... .......... 97% 86,5M 0s\n",
      " 23000K .......... .......... .......... .......... .......... 97%  118M 0s\n",
      " 23050K .......... .......... .......... .......... .......... 98%  116M 0s\n",
      " 23100K .......... .......... .......... .......... .......... 98%  104M 0s\n",
      " 23150K .......... .......... .......... .......... .......... 98%  127M 0s\n",
      " 23200K .......... .......... .......... .......... .......... 98%  177M 0s\n",
      " 23250K .......... .......... .......... .......... .......... 98% 2,01M 0s\n",
      " 23300K .......... .......... .......... .......... .......... 99%  121M 0s\n",
      " 23350K .......... .......... .......... .......... .......... 99%  483K 0s\n",
      " 23400K .......... .......... .......... .......... .......... 99% 3,66M 0s\n",
      " 23450K .......... .......... .......... .......... .......... 99%  114M 0s\n",
      " 23500K .......... .......... .......... .........            100%  155M=1,8s\n",
      "\n",
      "2023-08-19 00:13:25 (12,6 MB/s) - ‘data/reference/genome.fa.gz’ guardado [24104573/24104573]\n",
      "\n",
      "[Sat Aug 19 00:13:26 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T001322.922136.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:13:27 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR696683_chr17_1.fastq.gz, data/raw/ERR696683_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR696683_chr17_1_fastp.fastq.gz, data/processed/ERR696683_chr17_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR696683_chr17_1_fastp.fastq.gz, data/processed/ERR696683_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1662053\n",
      "total bases: 125542423\n",
      "Q20 bases: 124610599(99.2578%)\n",
      "Q30 bases: 120902230(96.3039%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1662053\n",
      "total bases: 125166559\n",
      "Q20 bases: 123458540(98.6354%)\n",
      "Q30 bases: 117801905(94.1161%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 941109\n",
      "total bases: 71474317\n",
      "Q20 bases: 71185568(99.596%)\n",
      "Q30 bases: 69857451(97.7378%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 941109\n",
      "total bases: 71445379\n",
      "Q20 bases: 70903174(99.2411%)\n",
      "Q30 bases: 68848719(96.3655%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1882218\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1441888\n",
      "reads with adapter trimmed: 1178\n",
      "bases trimmed due to adapters: 33118\n",
      "\n",
      "Duplication rate: 5.19829%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR696683_chr17_1.fastq.gz -I data/raw/ERR696683_chr17_2.fastq.gz -o data/processed/ERR696683_chr17_1_fastp.fastq.gz -O data/processed/ERR696683_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Sat Aug 19 00:13:33 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:13:33 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR696683_chr17_1.fastq.gz, data/raw/ERR696683_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR696683_chr17_1_fastqc.html, results/fastqc_result/ERR696683_chr17_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR696683_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 00:13:47 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:13:47 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR696683_chr17_1_fastp.fastq.gz, data/processed/ERR696683_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR696683_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR696683_chr17_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR696683_chr17_1_fastp.fastq.gz, data/processed/ERR696683_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 00:13:55 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T001326.499894.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:13:57 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753368_chr17_1.fastq.gz, data/raw/ERR753368_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753368_chr17_1_fastqc.html, results/fastqc_result/ERR753368_chr17_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753368_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753368_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 00:14:10 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:14:10 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753368_chr17_1.fastq.gz, data/raw/ERR753368_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753368_chr17_1_fastp.fastq.gz, data/processed/ERR753368_chr17_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753368_chr17_1_fastp.fastq.gz, data/processed/ERR753368_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1720078\n",
      "total bases: 130586939\n",
      "Q20 bases: 130062674(99.5985%)\n",
      "Q30 bases: 126025291(96.5068%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1720078\n",
      "total bases: 130408498\n",
      "Q20 bases: 129167916(99.0487%)\n",
      "Q30 bases: 122638409(94.0417%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1449801\n",
      "total bases: 110178424\n",
      "Q20 bases: 109981943(99.8217%)\n",
      "Q30 bases: 108017025(98.0383%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1449801\n",
      "total bases: 110167371\n",
      "Q20 bases: 109584045(99.4705%)\n",
      "Q30 bases: 105965050(96.1855%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2899602\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 540554\n",
      "reads with adapter trimmed: 1280\n",
      "bases trimmed due to adapters: 33381\n",
      "\n",
      "Duplication rate: 1.94669%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753368_chr17_1.fastq.gz -I data/raw/ERR753368_chr17_2.fastq.gz -o data/processed/ERR753368_chr17_1_fastp.fastq.gz -O data/processed/ERR753368_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Sat Aug 19 00:14:17 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:14:17 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753368_chr17_1_fastp.fastq.gz, data/processed/ERR753368_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753368_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753368_chr17_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753368_chr17_1_fastp.fastq.gz, data/processed/ERR753368_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 00:14:28 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T001356.141635.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:14:30 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753369_chr17_1.fastq.gz, data/raw/ERR753369_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753369_chr17_1_fastqc.html, results/fastqc_result/ERR753369_chr17_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753369_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753369_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 00:14:46 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:14:46 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753369_chr17_1.fastq.gz, data/raw/ERR753369_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753369_chr17_1_fastp.fastq.gz, data/processed/ERR753369_chr17_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753369_chr17_2_fastp.fastq.gz, data/processed/ERR753369_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1834306\n",
      "total bases: 139253654\n",
      "Q20 bases: 138670028(99.5809%)\n",
      "Q30 bases: 134266277(96.4185%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1834306\n",
      "total bases: 139072650\n",
      "Q20 bases: 137744780(99.0452%)\n",
      "Q30 bases: 130771387(94.031%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1545614\n",
      "total bases: 117459470\n",
      "Q20 bases: 117236110(99.8098%)\n",
      "Q30 bases: 115069005(97.9649%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1545614\n",
      "total bases: 117448151\n",
      "Q20 bases: 116824626(99.4691%)\n",
      "Q30 bases: 112958776(96.1776%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3091228\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 577384\n",
      "reads with adapter trimmed: 1210\n",
      "bases trimmed due to adapters: 30992\n",
      "\n",
      "Duplication rate: 2.23224%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753369_chr17_1.fastq.gz -I data/raw/ERR753369_chr17_2.fastq.gz -o data/processed/ERR753369_chr17_1_fastp.fastq.gz -O data/processed/ERR753369_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Sat Aug 19 00:14:53 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:14:53 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753369_chr17_1_fastp.fastq.gz, data/processed/ERR753369_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753369_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753369_chr17_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753369_chr17_2_fastp.fastq.gz, data/processed/ERR753369_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 00:15:06 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T001429.058808.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:15:07 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753370_chr17_1.fastq.gz, data/raw/ERR753370_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753370_chr17_1_fastp.fastq.gz, data/processed/ERR753370_chr17_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753370_chr17_1_fastp.fastq.gz, data/processed/ERR753370_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2195518\n",
      "total bases: 220903960\n",
      "Q20 bases: 219316637(99.2814%)\n",
      "Q30 bases: 212394069(96.1477%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2195518\n",
      "total bases: 220842678\n",
      "Q20 bases: 218611707(98.9898%)\n",
      "Q30 bases: 209497343(94.8627%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 2194882\n",
      "total bases: 219845045\n",
      "Q20 bases: 218360012(99.3245%)\n",
      "Q30 bases: 211791798(96.3369%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 2194882\n",
      "total bases: 219058701\n",
      "Q20 bases: 217044793(99.0807%)\n",
      "Q30 bases: 208534267(95.1956%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 4389764\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1272\n",
      "reads with adapter trimmed: 11652\n",
      "bases trimmed due to adapters: 109912\n",
      "\n",
      "Duplication rate: 4.00861%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 157\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753370_chr17_1.fastq.gz -I data/raw/ERR753370_chr17_2.fastq.gz -o data/processed/ERR753370_chr17_1_fastp.fastq.gz -O data/processed/ERR753370_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 10 seconds\n",
      "[Sat Aug 19 00:15:17 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:15:17 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753370_chr17_1.fastq.gz, data/raw/ERR753370_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753370_chr17_1_fastqc.html, results/fastqc_result/ERR753370_chr17_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753370_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753370_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 00:15:38 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:15:38 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753370_chr17_1_fastp.fastq.gz, data/processed/ERR753370_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753370_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753370_chr17_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753370_chr17_1_fastp.fastq.gz, data/processed/ERR753370_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 00:15:58 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T001506.323801.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:15:59 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753371_chr17_1.fastq.gz, data/raw/ERR753371_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753371_chr17_1_fastqc.html, results/fastqc_result/ERR753371_chr17_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753371_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753371_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 00:16:12 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:16:12 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753371_chr17_1.fastq.gz, data/raw/ERR753371_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753371_chr17_1_fastp.fastq.gz, data/processed/ERR753371_chr17_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753371_chr17_1_fastp.fastq.gz, data/processed/ERR753371_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1479982\n",
      "total bases: 111397900\n",
      "Q20 bases: 110370854(99.078%)\n",
      "Q30 bases: 106159867(95.2979%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1479982\n",
      "total bases: 111084814\n",
      "Q20 bases: 109633816(98.6938%)\n",
      "Q30 bases: 105347657(94.8353%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 690305\n",
      "total bases: 52398376\n",
      "Q20 bases: 52131601(99.4909%)\n",
      "Q30 bases: 50765095(96.883%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 690305\n",
      "total bases: 52368274\n",
      "Q20 bases: 51965373(99.2306%)\n",
      "Q30 bases: 50683984(96.7838%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1380610\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1579354\n",
      "reads with adapter trimmed: 980\n",
      "bases trimmed due to adapters: 24008\n",
      "\n",
      "Duplication rate: 19.6058%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 117\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753371_chr17_1.fastq.gz -I data/raw/ERR753371_chr17_2.fastq.gz -o data/processed/ERR753371_chr17_1_fastp.fastq.gz -O data/processed/ERR753371_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Sat Aug 19 00:16:17 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:16:17 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753371_chr17_1_fastp.fastq.gz, data/processed/ERR753371_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753371_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753371_chr17_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753371_chr17_1_fastp.fastq.gz, data/processed/ERR753371_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 00:16:23 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T001558.536707.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:16:25 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753372_chr17_1.fastq.gz, data/raw/ERR753372_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753372_chr17_1_fastp.fastq.gz, data/processed/ERR753372_chr17_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753372_chr17_1_fastp.fastq.gz, data/processed/ERR753372_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1891163\n",
      "total bases: 143570978\n",
      "Q20 bases: 142991472(99.5964%)\n",
      "Q30 bases: 138476280(96.4514%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1891163\n",
      "total bases: 143376057\n",
      "Q20 bases: 142013761(99.0498%)\n",
      "Q30 bases: 134881566(94.0754%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1589698\n",
      "total bases: 120809730\n",
      "Q20 bases: 120591835(99.8196%)\n",
      "Q30 bases: 118389976(97.9971%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1589698\n",
      "total bases: 120797532\n",
      "Q20 bases: 120159672(99.472%)\n",
      "Q30 bases: 116232633(96.221%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3179396\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 602930\n",
      "reads with adapter trimmed: 1446\n",
      "bases trimmed due to adapters: 35677\n",
      "\n",
      "Duplication rate: 1.89058%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753372_chr17_1.fastq.gz -I data/raw/ERR753372_chr17_2.fastq.gz -o data/processed/ERR753372_chr17_1_fastp.fastq.gz -O data/processed/ERR753372_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Sat Aug 19 00:16:31 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:16:31 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753372_chr17_1.fastq.gz, data/raw/ERR753372_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753372_chr17_1_fastqc.html, results/fastqc_result/ERR753372_chr17_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753372_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753372_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 00:16:46 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:16:46 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753372_chr17_1_fastp.fastq.gz, data/processed/ERR753372_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753372_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753372_chr17_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753372_chr17_1_fastp.fastq.gz, data/processed/ERR753372_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 00:16:59 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T001624.075420.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:17:00 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753373_chr17_1.fastq.gz, data/raw/ERR753373_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753373_chr17_1_fastqc.html, results/fastqc_result/ERR753373_chr17_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753373_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753373_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 00:17:09 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:17:09 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753373_chr17_1.fastq.gz, data/raw/ERR753373_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753373_chr17_1_fastp.fastq.gz, data/processed/ERR753373_chr17_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753373_chr17_2_fastp.fastq.gz, data/processed/ERR753373_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 843274\n",
      "total bases: 80686867\n",
      "Q20 bases: 79980120(99.1241%)\n",
      "Q30 bases: 74138420(91.8841%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 843274\n",
      "total bases: 78848808\n",
      "Q20 bases: 77912736(98.8128%)\n",
      "Q30 bases: 72303629(91.6991%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 819531\n",
      "total bases: 77028198\n",
      "Q20 bases: 76544780(99.3724%)\n",
      "Q30 bases: 71847320(93.2741%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 819531\n",
      "total bases: 74314773\n",
      "Q20 bases: 73905877(99.4498%)\n",
      "Q30 bases: 69630249(93.6964%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1639062\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 47486\n",
      "reads with adapter trimmed: 1874\n",
      "bases trimmed due to adapters: 72905\n",
      "\n",
      "Duplication rate: 1.52989%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 153\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753373_chr17_1.fastq.gz -I data/raw/ERR753373_chr17_2.fastq.gz -o data/processed/ERR753373_chr17_1_fastp.fastq.gz -O data/processed/ERR753373_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 3 seconds\n",
      "[Sat Aug 19 00:17:13 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:17:13 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753373_chr17_1_fastp.fastq.gz, data/processed/ERR753373_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753373_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753373_chr17_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753373_chr17_2_fastp.fastq.gz, data/processed/ERR753373_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 00:17:23 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T001659.878892.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:17:25 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753374_chr17_1.fastq.gz, data/raw/ERR753374_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753374_chr17_1_fastqc.html, results/fastqc_result/ERR753374_chr17_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753374_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753374_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 00:17:31 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:17:31 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753374_chr17_1.fastq.gz, data/raw/ERR753374_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753374_chr17_1_fastp.fastq.gz, data/processed/ERR753374_chr17_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753374_chr17_1_fastp.fastq.gz, data/processed/ERR753374_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 492332\n",
      "total bases: 47062824\n",
      "Q20 bases: 46581392(98.977%)\n",
      "Q30 bases: 42272899(89.8223%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 492332\n",
      "total bases: 45990975\n",
      "Q20 bases: 45452935(98.8301%)\n",
      "Q30 bases: 41465767(90.1607%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 473782\n",
      "total bases: 44142721\n",
      "Q20 bases: 43810033(99.2463%)\n",
      "Q30 bases: 40441158(91.6146%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 473782\n",
      "total bases: 42957145\n",
      "Q20 bases: 42673614(99.34%)\n",
      "Q30 bases: 39641425(92.2813%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 947564\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 37100\n",
      "reads with adapter trimmed: 1076\n",
      "bases trimmed due to adapters: 40498\n",
      "\n",
      "Duplication rate: 4.87442%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 150\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753374_chr17_1.fastq.gz -I data/raw/ERR753374_chr17_2.fastq.gz -o data/processed/ERR753374_chr17_1_fastp.fastq.gz -O data/processed/ERR753374_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 2 seconds\n",
      "[Sat Aug 19 00:17:34 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:17:34 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753374_chr17_1_fastp.fastq.gz, data/processed/ERR753374_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753374_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753374_chr17_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753374_chr17_1_fastp.fastq.gz, data/processed/ERR753374_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 00:17:41 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T001724.001950.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:17:42 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753375_chr17_1.fastq.gz, data/raw/ERR753375_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753375_chr17_1_fastp.fastq.gz, data/processed/ERR753375_chr17_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753375_chr17_1_fastp.fastq.gz, data/processed/ERR753375_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1026269\n",
      "total bases: 77458309\n",
      "Q20 bases: 76947062(99.34%)\n",
      "Q30 bases: 75053112(96.8948%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1026269\n",
      "total bases: 77416125\n",
      "Q20 bases: 76301088(98.5597%)\n",
      "Q30 bases: 74090313(95.704%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 615090\n",
      "total bases: 46703360\n",
      "Q20 bases: 46543091(99.6568%)\n",
      "Q30 bases: 45786092(98.036%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 615090\n",
      "total bases: 46709566\n",
      "Q20 bases: 46221428(98.955%)\n",
      "Q30 bases: 45368202(97.1283%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1230180\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 822358\n",
      "reads with adapter trimmed: 702\n",
      "bases trimmed due to adapters: 19004\n",
      "\n",
      "Duplication rate: 1.55678%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753375_chr17_1.fastq.gz -I data/raw/ERR753375_chr17_2.fastq.gz -o data/processed/ERR753375_chr17_1_fastp.fastq.gz -O data/processed/ERR753375_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Sat Aug 19 00:17:47 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:17:47 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753375_chr17_1_fastp.fastq.gz, data/processed/ERR753375_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753375_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753375_chr17_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753375_chr17_1_fastp.fastq.gz, data/processed/ERR753375_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 00:17:54 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:17:54 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753375_chr17_1.fastq.gz, data/raw/ERR753375_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753375_chr17_1_fastqc.html, results/fastqc_result/ERR753375_chr17_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753375_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753375_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 00:18:03 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T001741.639263.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:18:04 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753376_chr17_1.fastq.gz, data/raw/ERR753376_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753376_chr17_1_fastqc.html, results/fastqc_result/ERR753376_chr17_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753376_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753376_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 00:18:16 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:18:16 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753376_chr17_1.fastq.gz, data/raw/ERR753376_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753376_chr17_1_fastp.fastq.gz, data/processed/ERR753376_chr17_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753376_chr17_2_fastp.fastq.gz, data/processed/ERR753376_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1319555\n",
      "total bases: 99673497\n",
      "Q20 bases: 98950424(99.2746%)\n",
      "Q30 bases: 96007070(96.3216%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1319555\n",
      "total bases: 99357655\n",
      "Q20 bases: 97988993(98.6225%)\n",
      "Q30 bases: 93463715(94.068%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 743097\n",
      "total bases: 56436043\n",
      "Q20 bases: 56214517(99.6075%)\n",
      "Q30 bases: 55168884(97.7547%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 743097\n",
      "total bases: 56414017\n",
      "Q20 bases: 55982344(99.2348%)\n",
      "Q30 bases: 54346819(96.3357%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1486194\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1152916\n",
      "reads with adapter trimmed: 998\n",
      "bases trimmed due to adapters: 26275\n",
      "\n",
      "Duplication rate: 8.00848%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753376_chr17_1.fastq.gz -I data/raw/ERR753376_chr17_2.fastq.gz -o data/processed/ERR753376_chr17_1_fastp.fastq.gz -O data/processed/ERR753376_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Sat Aug 19 00:18:21 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:18:21 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753376_chr17_1_fastp.fastq.gz, data/processed/ERR753376_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753376_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753376_chr17_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753376_chr17_2_fastp.fastq.gz, data/processed/ERR753376_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 00:18:28 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T001803.602515.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:18:29 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753377_chr17_1.fastq.gz, data/raw/ERR753377_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753377_chr17_1_fastqc.html, results/fastqc_result/ERR753377_chr17_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753377_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753377_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 00:18:41 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:18:41 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753377_chr17_1.fastq.gz, data/raw/ERR753377_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753377_chr17_1_fastp.fastq.gz, data/processed/ERR753377_chr17_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753377_chr17_1_fastp.fastq.gz, data/processed/ERR753377_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1385666\n",
      "total bases: 104679071\n",
      "Q20 bases: 103910027(99.2653%)\n",
      "Q30 bases: 100812888(96.3066%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1385666\n",
      "total bases: 104385271\n",
      "Q20 bases: 103014264(98.6866%)\n",
      "Q30 bases: 98468698(94.332%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 800065\n",
      "total bases: 60762969\n",
      "Q20 bases: 60520165(99.6004%)\n",
      "Q30 bases: 59388451(97.7379%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 800065\n",
      "total bases: 60740129\n",
      "Q20 bases: 60296945(99.2704%)\n",
      "Q30 bases: 58618077(96.5063%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1600130\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1171202\n",
      "reads with adapter trimmed: 1020\n",
      "bases trimmed due to adapters: 26943\n",
      "\n",
      "Duplication rate: 8.08277%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753377_chr17_1.fastq.gz -I data/raw/ERR753377_chr17_2.fastq.gz -o data/processed/ERR753377_chr17_1_fastp.fastq.gz -O data/processed/ERR753377_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Sat Aug 19 00:18:45 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:18:45 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753377_chr17_1_fastp.fastq.gz, data/processed/ERR753377_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753377_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753377_chr17_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753377_chr17_1_fastp.fastq.gz, data/processed/ERR753377_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 00:18:54 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T001828.362534.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:18:56 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753378_chr17_1.fastq.gz, data/raw/ERR753378_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753378_chr17_1_fastqc.html, results/fastqc_result/ERR753378_chr17_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753378_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753378_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 00:19:08 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:19:08 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753378_chr17_1.fastq.gz, data/raw/ERR753378_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753378_chr17_1_fastp.fastq.gz, data/processed/ERR753378_chr17_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753378_chr17_2_fastp.fastq.gz, data/processed/ERR753378_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1654676\n",
      "total bases: 125620372\n",
      "Q20 bases: 125114317(99.5972%)\n",
      "Q30 bases: 121181168(96.4662%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1654676\n",
      "total bases: 125447036\n",
      "Q20 bases: 124251291(99.0468%)\n",
      "Q30 bases: 118001271(94.0646%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1397009\n",
      "total bases: 106166313\n",
      "Q20 bases: 105973483(99.8184%)\n",
      "Q30 bases: 104029622(97.9874%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1397009\n",
      "total bases: 106155410\n",
      "Q20 bases: 105589103(99.4665%)\n",
      "Q30 bases: 102096673(96.1766%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2794018\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 515334\n",
      "reads with adapter trimmed: 1108\n",
      "bases trimmed due to adapters: 27200\n",
      "\n",
      "Duplication rate: 1.89105%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753378_chr17_1.fastq.gz -I data/raw/ERR753378_chr17_2.fastq.gz -o data/processed/ERR753378_chr17_1_fastp.fastq.gz -O data/processed/ERR753378_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Sat Aug 19 00:19:14 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:19:14 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753378_chr17_1_fastp.fastq.gz, data/processed/ERR753378_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753378_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753378_chr17_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753378_chr17_2_fastp.fastq.gz, data/processed/ERR753378_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 100% complete for ERR753378_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 100% complete for ERR753378_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 00:19:27 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T001854.973776.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:19:28 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR696683_chr17_1_fastp.fastq.gz, data/processed/ERR696683_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR696683_chr17.sam, results/mapped_reads/ERR696683_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR696683_chr17_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR696683_chr17_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.28 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 28.31 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.25 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.18 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 16.35 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 47.577 sec; CPU: 45.368 sec\n",
      "[Sat Aug 19 00:21:42 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:21:42 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR696683_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr17.bam, results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR696683_chr17.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR696683_chr17_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 00:22:19 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:22:19 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR696683_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "00:22:20.465 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Sat Aug 19 00:22:20 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR696683_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR696683_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Sat Aug 19 00:22:20 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-19 00:22:20\tMarkDuplicates\tStart of doWork freeMemory: 528787200; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:22:20\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-19 00:22:20\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam\n",
      "INFO\t2023-08-19 00:22:22\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:40,133,370\n",
      "INFO\t2023-08-19 00:22:22\tMarkDuplicates\tTracking 324 as yet unmatched pairs. 324 records in RAM.\n",
      "INFO\t2023-08-19 00:22:23\tMarkDuplicates\tRead 1926106 records. 0 pairs never matched.\n",
      "INFO\t2023-08-19 00:22:23\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 261920136; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:22:23\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-19 00:22:23\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-19 00:22:23\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-19 00:22:24\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-19 00:22:24\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 822035312; totalMemory: 1368391680; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:22:24\tMarkDuplicates\tMarking 153605 records as duplicates.\n",
      "INFO\t2023-08-19 00:22:24\tMarkDuplicates\tFound 62 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam\n",
      "INFO\t2023-08-19 00:22:24\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-19 00:22:36\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-19 00:22:36\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-19 00:22:36\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-19 00:22:36\tMarkDuplicates\tBefore output close freeMemory: 528085320; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:22:36\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-19 00:22:36\tMarkDuplicates\tAfter output close freeMemory: 528085320; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Sat Aug 19 00:22:36 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.27 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Sat Aug 19 00:22:37 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T001927.635004.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:22:39 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753368_chr17_1_fastp.fastq.gz, data/processed/ERR753368_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753368_chr17.sam, results/mapped_reads/ERR753368_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753368_chr17_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR753368_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.28 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 28.02 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.26 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.21 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 16.01 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 47.813 sec; CPU: 44.783 sec\n",
      "[Sat Aug 19 00:25:29 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:25:29 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753368_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr17.bam, results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753368_chr17.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753368_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 00:26:16 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:26:16 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753368_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753368_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "00:26:16.857 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Sat Aug 19 00:26:16 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753368_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753368_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Sat Aug 19 00:26:16 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-19 00:26:16\tMarkDuplicates\tStart of doWork freeMemory: 528821016; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:26:16\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-19 00:26:16\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam\n",
      "INFO\t2023-08-19 00:26:18\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:28,617,154\n",
      "INFO\t2023-08-19 00:26:18\tMarkDuplicates\tTracking 428 as yet unmatched pairs. 428 records in RAM.\n",
      "INFO\t2023-08-19 00:26:19\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 17:50,643,524\n",
      "INFO\t2023-08-19 00:26:19\tMarkDuplicates\tTracking 335 as yet unmatched pairs. 335 records in RAM.\n",
      "INFO\t2023-08-19 00:26:21\tMarkDuplicates\tRead 2989104 records. 0 pairs never matched.\n",
      "INFO\t2023-08-19 00:26:21\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 762904384; totalMemory: 1148190720; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:26:21\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-19 00:26:21\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-19 00:26:21\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-19 00:26:22\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-19 00:26:22\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1138739784; totalMemory: 1685061632; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:26:22\tMarkDuplicates\tMarking 67240 records as duplicates.\n",
      "INFO\t2023-08-19 00:26:22\tMarkDuplicates\tFound 468 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam\n",
      "INFO\t2023-08-19 00:26:22\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-19 00:26:39\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-19 00:26:39\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-19 00:26:39\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-19 00:26:39\tMarkDuplicates\tBefore output close freeMemory: 528119952; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:26:39\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-19 00:26:39\tMarkDuplicates\tAfter output close freeMemory: 528119952; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Sat Aug 19 00:26:39 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.38 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Sat Aug 19 00:26:41 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T002238.113248.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:26:43 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753369_chr17_1_fastp.fastq.gz, data/processed/ERR753369_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753369_chr17.sam, results/mapped_reads/ERR753369_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753369_chr17_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753369_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.30 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 27.64 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.27 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.23 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 16.89 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 47.974 sec; CPU: 45.325 sec\n",
      "[Sat Aug 19 00:29:44 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:29:44 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753369_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr17.bam, results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753369_chr17.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753369_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 00:30:32 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:30:32 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753369_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753369_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "00:30:33.524 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Sat Aug 19 00:30:33 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753369_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753369_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Sat Aug 19 00:30:33 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-19 00:30:33\tMarkDuplicates\tStart of doWork freeMemory: 528775992; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:30:33\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-19 00:30:33\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam\n",
      "INFO\t2023-08-19 00:30:35\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:28,638,437\n",
      "INFO\t2023-08-19 00:30:35\tMarkDuplicates\tTracking 414 as yet unmatched pairs. 414 records in RAM.\n",
      "INFO\t2023-08-19 00:30:36\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 17:49,206,145\n",
      "INFO\t2023-08-19 00:30:36\tMarkDuplicates\tTracking 295 as yet unmatched pairs. 295 records in RAM.\n",
      "INFO\t2023-08-19 00:30:38\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 17:80,386,227\n",
      "INFO\t2023-08-19 00:30:38\tMarkDuplicates\tTracking 116 as yet unmatched pairs. 116 records in RAM.\n",
      "INFO\t2023-08-19 00:30:38\tMarkDuplicates\tRead 3123173 records. 0 pairs never matched.\n",
      "INFO\t2023-08-19 00:30:38\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 602515376; totalMemory: 1008730112; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:30:38\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-19 00:30:38\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-19 00:30:39\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-19 00:30:39\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-19 00:30:39\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 999234296; totalMemory: 1545601024; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:30:39\tMarkDuplicates\tMarking 82978 records as duplicates.\n",
      "INFO\t2023-08-19 00:30:39\tMarkDuplicates\tFound 436 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam\n",
      "INFO\t2023-08-19 00:30:39\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-19 00:30:57\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-19 00:30:57\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-19 00:30:57\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-19 00:30:57\tMarkDuplicates\tBefore output close freeMemory: 528074792; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:30:57\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-19 00:30:57\tMarkDuplicates\tAfter output close freeMemory: 528074792; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Sat Aug 19 00:30:57 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.40 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Sat Aug 19 00:30:59 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T002642.050147.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:31:00 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753370_chr17_1_fastp.fastq.gz, data/processed/ERR753370_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753370_chr17.sam, results/mapped_reads/ERR753370_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753370_chr17_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753370_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.28 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 26.94 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.26 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.18 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 16.32 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 48.002 sec; CPU: 43.985 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Sat Aug 19 00:36:10 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:36:10 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753370_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr17.bam, results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753370_chr17.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753370_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Sat Aug 19 00:37:52 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:37:52 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753370_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753370_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "00:37:53.203 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Sat Aug 19 00:37:53 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753370_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753370_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Sat Aug 19 00:37:53 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-19 00:37:53\tMarkDuplicates\tStart of doWork freeMemory: 528825752; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:37:53\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-19 00:37:53\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam\n",
      "INFO\t2023-08-19 00:37:55\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:17,848,756\n",
      "INFO\t2023-08-19 00:37:55\tMarkDuplicates\tTracking 640 as yet unmatched pairs. 640 records in RAM.\n",
      "INFO\t2023-08-19 00:37:56\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 17:36,310,552\n",
      "INFO\t2023-08-19 00:37:56\tMarkDuplicates\tTracking 1045 as yet unmatched pairs. 1045 records in RAM.\n",
      "INFO\t2023-08-19 00:37:58\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 17:47,368,569\n",
      "INFO\t2023-08-19 00:37:58\tMarkDuplicates\tTracking 951 as yet unmatched pairs. 951 records in RAM.\n",
      "INFO\t2023-08-19 00:37:59\tMarkDuplicates\tRead     4,000,000 records.  Elapsed time: 00:00:06s.  Time for last 1,000,000:    1s.  Last read position: 17:68,941,907\n",
      "INFO\t2023-08-19 00:37:59\tMarkDuplicates\tTracking 485 as yet unmatched pairs. 485 records in RAM.\n",
      "INFO\t2023-08-19 00:38:00\tMarkDuplicates\tRead 4523067 records. 0 pairs never matched.\n",
      "INFO\t2023-08-19 00:38:00\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 378241872; totalMemory: 924844032; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:38:00\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-19 00:38:00\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-19 00:38:01\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-19 00:38:01\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-19 00:38:01\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1309662112; totalMemory: 1855979520; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:38:01\tMarkDuplicates\tMarking 290968 records as duplicates.\n",
      "INFO\t2023-08-19 00:38:01\tMarkDuplicates\tFound 164 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam\n",
      "INFO\t2023-08-19 00:38:01\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-19 00:38:35\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-19 00:38:35\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-19 00:38:35\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-19 00:38:35\tMarkDuplicates\tBefore output close freeMemory: 528123784; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:38:35\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-19 00:38:35\tMarkDuplicates\tAfter output close freeMemory: 528123784; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Sat Aug 19 00:38:35 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.70 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Sat Aug 19 00:38:38 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T003059.747962.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:38:40 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753371_chr17_1_fastp.fastq.gz, data/processed/ERR753371_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753371_chr17.sam, results/mapped_reads/ERR753371_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753371_chr17_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR753371_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.28 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 30.94 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.26 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.18 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 17.46 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 51.919 sec; CPU: 49.118 sec\n",
      "[Sat Aug 19 00:40:37 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:40:37 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753371_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr17.bam, results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753371_chr17.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753371_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 00:41:03 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:41:03 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753371_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753371_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "00:41:04.002 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Sat Aug 19 00:41:04 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753371_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753371_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Sat Aug 19 00:41:04 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-19 00:41:04\tMarkDuplicates\tStart of doWork freeMemory: 528803032; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:41:04\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-19 00:41:04\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam\n",
      "INFO\t2023-08-19 00:41:05\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:51,502,814\n",
      "INFO\t2023-08-19 00:41:05\tMarkDuplicates\tTracking 120 as yet unmatched pairs. 120 records in RAM.\n",
      "INFO\t2023-08-19 00:41:06\tMarkDuplicates\tRead 1400906 records. 0 pairs never matched.\n",
      "INFO\t2023-08-19 00:41:06\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 315957264; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:41:06\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-19 00:41:06\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-19 00:41:06\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-19 00:41:06\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-19 00:41:06\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 753892752; totalMemory: 1300234240; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:41:06\tMarkDuplicates\tMarking 496472 records as duplicates.\n",
      "INFO\t2023-08-19 00:41:06\tMarkDuplicates\tFound 34 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam\n",
      "INFO\t2023-08-19 00:41:06\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-19 00:41:15\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-19 00:41:15\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-19 00:41:15\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-19 00:41:15\tMarkDuplicates\tBefore output close freeMemory: 528100264; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:41:15\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-19 00:41:15\tMarkDuplicates\tAfter output close freeMemory: 528100264; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Sat Aug 19 00:41:15 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.19 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Sat Aug 19 00:41:16 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T003839.203682.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:41:17 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753372_chr17_1_fastp.fastq.gz, data/processed/ERR753372_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753372_chr17.sam, results/mapped_reads/ERR753372_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753372_chr17_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753372_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.30 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 27.13 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.28 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.22 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 15.85 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 46.733 sec; CPU: 43.786 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Sat Aug 19 00:44:27 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:44:27 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753372_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr17.bam, results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753372_chr17.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753372_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Sat Aug 19 00:45:26 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:45:26 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753372_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753372_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "00:45:27.041 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Sat Aug 19 00:45:27 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753372_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753372_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Sat Aug 19 00:45:27 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-19 00:45:27\tMarkDuplicates\tStart of doWork freeMemory: 528832848; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:45:27\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-19 00:45:27\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam\n",
      "INFO\t2023-08-19 00:45:28\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:26,199,503\n",
      "INFO\t2023-08-19 00:45:28\tMarkDuplicates\tTracking 485 as yet unmatched pairs. 485 records in RAM.\n",
      "INFO\t2023-08-19 00:45:30\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 17:47,244,348\n",
      "INFO\t2023-08-19 00:45:30\tMarkDuplicates\tTracking 332 as yet unmatched pairs. 332 records in RAM.\n",
      "INFO\t2023-08-19 00:45:31\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 17:78,126,195\n",
      "INFO\t2023-08-19 00:45:31\tMarkDuplicates\tTracking 96 as yet unmatched pairs. 96 records in RAM.\n",
      "INFO\t2023-08-19 00:45:31\tMarkDuplicates\tRead 3244765 records. 0 pairs never matched.\n",
      "INFO\t2023-08-19 00:45:31\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 444221192; totalMemory: 859832320; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:45:31\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-19 00:45:31\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-19 00:45:32\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-19 00:45:32\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-19 00:45:32\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1067447984; totalMemory: 1613758464; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:45:32\tMarkDuplicates\tMarking 72231 records as duplicates.\n",
      "INFO\t2023-08-19 00:45:32\tMarkDuplicates\tFound 452 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam\n",
      "INFO\t2023-08-19 00:45:32\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-19 00:45:51\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-19 00:45:51\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-19 00:45:51\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-19 00:45:51\tMarkDuplicates\tBefore output close freeMemory: 528131616; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:45:51\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-19 00:45:51\tMarkDuplicates\tAfter output close freeMemory: 528131616; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Sat Aug 19 00:45:51 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.41 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Sat Aug 19 00:45:54 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T004116.825725.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:45:55 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753373_chr17_1_fastp.fastq.gz, data/processed/ERR753373_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753373_chr17.sam, results/mapped_reads/ERR753373_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753373_chr17_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753373_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.28 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 28.27 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.27 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.19 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 16.42 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 48.136 sec; CPU: 45.436 sec\n",
      "[Sat Aug 19 00:48:58 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:48:58 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753373_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr17.bam, results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753373_chr17.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753373_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 00:49:47 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:49:47 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753373_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753373_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "00:49:48.741 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Sat Aug 19 00:49:48 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753373_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753373_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Sat Aug 19 00:49:48 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-19 00:49:48\tMarkDuplicates\tStart of doWork freeMemory: 528792808; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:49:48\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-19 00:49:48\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam\n",
      "INFO\t2023-08-19 00:49:50\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:34,038,094\n",
      "INFO\t2023-08-19 00:49:50\tMarkDuplicates\tTracking 177 as yet unmatched pairs. 177 records in RAM.\n",
      "INFO\t2023-08-19 00:49:51\tMarkDuplicates\tRead 1994716 records. 0 pairs never matched.\n",
      "INFO\t2023-08-19 00:49:51\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 288033120; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:49:51\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-19 00:49:51\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-19 00:49:52\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-19 00:49:52\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-19 00:49:52\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 782194896; totalMemory: 1328545792; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:49:52\tMarkDuplicates\tMarking 36873 records as duplicates.\n",
      "INFO\t2023-08-19 00:49:52\tMarkDuplicates\tFound 11387 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam\n",
      "INFO\t2023-08-19 00:49:52\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-19 00:50:07\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-19 00:50:07\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-19 00:50:07\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-19 00:50:07\tMarkDuplicates\tBefore output close freeMemory: 528090800; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:50:07\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-19 00:50:07\tMarkDuplicates\tAfter output close freeMemory: 528090800; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Sat Aug 19 00:50:07 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.32 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Sat Aug 19 00:50:09 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T004554.354769.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:50:10 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753374_chr17_1_fastp.fastq.gz, data/processed/ERR753374_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753374_chr17.sam, results/mapped_reads/ERR753374_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753374_chr17_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753374_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.30 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 28.42 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.26 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.19 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 16.47 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 48.182 sec; CPU: 45.642 sec\n",
      "[Sat Aug 19 00:52:13 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:52:13 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753374_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr17.bam, results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753374_chr17.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753374_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 00:52:41 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:52:41 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753374_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753374_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "00:52:41.942 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Sat Aug 19 00:52:41 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753374_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753374_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Sat Aug 19 00:52:41 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-19 00:52:41\tMarkDuplicates\tStart of doWork freeMemory: 528785192; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:52:41\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-19 00:52:41\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam\n",
      "INFO\t2023-08-19 00:52:43\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:62,737,033\n",
      "INFO\t2023-08-19 00:52:43\tMarkDuplicates\tTracking 91 as yet unmatched pairs. 91 records in RAM.\n",
      "INFO\t2023-08-19 00:52:43\tMarkDuplicates\tRead 1192512 records. 0 pairs never matched.\n",
      "INFO\t2023-08-19 00:52:43\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 362868712; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:52:43\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-19 00:52:44\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-19 00:52:44\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-19 00:52:44\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-19 00:52:44\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 669989736; totalMemory: 1216348160; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:52:44\tMarkDuplicates\tMarking 88631 records as duplicates.\n",
      "INFO\t2023-08-19 00:52:44\tMarkDuplicates\tFound 28790 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam\n",
      "INFO\t2023-08-19 00:52:44\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-19 00:52:53\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-19 00:52:53\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-19 00:52:53\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-19 00:52:53\tMarkDuplicates\tBefore output close freeMemory: 528082672; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:52:53\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-19 00:52:53\tMarkDuplicates\tAfter output close freeMemory: 528082672; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Sat Aug 19 00:52:53 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.19 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Sat Aug 19 00:52:54 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T005009.588958.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:52:55 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753375_chr17_1_fastp.fastq.gz, data/processed/ERR753375_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753375_chr17.sam, results/mapped_reads/ERR753375_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753375_chr17_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR753375_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.30 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 28.33 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.27 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.20 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 16.44 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 48.253 sec; CPU: 45.539 sec\n",
      "[Sat Aug 19 00:54:43 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:54:43 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753375_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr17.bam, results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753375_chr17.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753375_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 00:55:07 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:55:07 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753375_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753375_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "00:55:08.105 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Sat Aug 19 00:55:08 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753375_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753375_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Sat Aug 19 00:55:08 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-19 00:55:08\tMarkDuplicates\tStart of doWork freeMemory: 528780704; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:55:08\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-19 00:55:08\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam\n",
      "INFO\t2023-08-19 00:55:09\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:61,924,974\n",
      "INFO\t2023-08-19 00:55:09\tMarkDuplicates\tTracking 111 as yet unmatched pairs. 111 records in RAM.\n",
      "INFO\t2023-08-19 00:55:10\tMarkDuplicates\tRead 1249072 records. 0 pairs never matched.\n",
      "INFO\t2023-08-19 00:55:10\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 397344728; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:55:10\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-19 00:55:10\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-19 00:55:10\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-19 00:55:10\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-19 00:55:10\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 729754288; totalMemory: 1276116992; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:55:10\tMarkDuplicates\tMarking 53195 records as duplicates.\n",
      "INFO\t2023-08-19 00:55:10\tMarkDuplicates\tFound 132 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam\n",
      "INFO\t2023-08-19 00:55:10\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-19 00:55:18\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-19 00:55:18\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-19 00:55:18\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-19 00:55:18\tMarkDuplicates\tBefore output close freeMemory: 528078912; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:55:18\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-19 00:55:18\tMarkDuplicates\tAfter output close freeMemory: 528078912; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Sat Aug 19 00:55:18 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.18 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Sat Aug 19 00:55:19 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T005254.591330.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:55:21 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753376_chr17_1_fastp.fastq.gz, data/processed/ERR753376_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753376_chr17.sam, results/mapped_reads/ERR753376_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753376_chr17_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753376_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.30 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 28.15 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.27 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.19 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 16.49 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 48.361 sec; CPU: 45.411 sec\n",
      "[Sat Aug 19 00:57:18 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:57:18 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753376_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr17.bam, results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753376_chr17.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753376_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 00:57:48 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:57:48 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753376_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753376_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "00:57:49.535 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Sat Aug 19 00:57:49 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753376_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753376_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Sat Aug 19 00:57:49 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-19 00:57:49\tMarkDuplicates\tStart of doWork freeMemory: 528770944; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:57:49\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-19 00:57:49\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam\n",
      "INFO\t2023-08-19 00:57:51\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:46,640,092\n",
      "INFO\t2023-08-19 00:57:51\tMarkDuplicates\tTracking 173 as yet unmatched pairs. 173 records in RAM.\n",
      "INFO\t2023-08-19 00:57:51\tMarkDuplicates\tRead 1528048 records. 0 pairs never matched.\n",
      "INFO\t2023-08-19 00:57:51\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 304531312; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:57:51\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-19 00:57:52\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-19 00:57:52\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-19 00:57:52\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-19 00:57:52\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 773783872; totalMemory: 1320157184; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:57:52\tMarkDuplicates\tMarking 187073 records as duplicates.\n",
      "INFO\t2023-08-19 00:57:52\tMarkDuplicates\tFound 58 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam\n",
      "INFO\t2023-08-19 00:57:52\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-19 00:58:02\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-19 00:58:02\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-19 00:58:02\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-19 00:58:02\tMarkDuplicates\tBefore output close freeMemory: 528068120; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 00:58:02\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-19 00:58:02\tMarkDuplicates\tAfter output close freeMemory: 528068120; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Sat Aug 19 00:58:02 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.22 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Sat Aug 19 00:58:03 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T005520.017643.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 00:58:04 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753377_chr17_1_fastp.fastq.gz, data/processed/ERR753377_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753377_chr17.sam, results/mapped_reads/ERR753377_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753377_chr17_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753377_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.30 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 28.28 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.27 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.20 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 16.44 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 48.160 sec; CPU: 45.496 sec\n",
      "[Sat Aug 19 01:00:04 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:00:04 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753377_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr17.bam, results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753377_chr17.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753377_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 01:00:36 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:00:36 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753377_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753377_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "01:00:37.063 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Sat Aug 19 01:00:37 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753377_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753377_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Sat Aug 19 01:00:37 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-19 01:00:37\tMarkDuplicates\tStart of doWork freeMemory: 528783344; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 01:00:37\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-19 01:00:37\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam\n",
      "INFO\t2023-08-19 01:00:38\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:45,187,382\n",
      "INFO\t2023-08-19 01:00:38\tMarkDuplicates\tTracking 190 as yet unmatched pairs. 190 records in RAM.\n",
      "INFO\t2023-08-19 01:00:39\tMarkDuplicates\tRead 1629046 records. 0 pairs never matched.\n",
      "INFO\t2023-08-19 01:00:39\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 292343384; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 01:00:39\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-19 01:00:39\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-19 01:00:40\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-19 01:00:40\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-19 01:00:40\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 774845392; totalMemory: 1321205760; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 01:00:40\tMarkDuplicates\tMarking 200588 records as duplicates.\n",
      "INFO\t2023-08-19 01:00:40\tMarkDuplicates\tFound 53 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam\n",
      "INFO\t2023-08-19 01:00:40\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-19 01:00:50\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-19 01:00:50\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-19 01:00:50\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-19 01:00:50\tMarkDuplicates\tBefore output close freeMemory: 528080824; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 01:00:50\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-19 01:00:50\tMarkDuplicates\tAfter output close freeMemory: 528080824; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Sat Aug 19 01:00:50 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.23 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Sat Aug 19 01:00:51 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T005803.809680.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:00:53 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753378_chr17_1_fastp.fastq.gz, data/processed/ERR753378_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753378_chr17.sam, results/mapped_reads/ERR753378_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753378_chr17_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753378_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.30 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 28.32 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.28 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.19 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 16.41 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 48.299 sec; CPU: 45.492 sec\n",
      "[Sat Aug 19 01:03:38 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:03:38 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753378_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr17.bam, results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753378_chr17.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753378_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 01:04:22 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:04:22 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753378_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753378_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "01:04:23.735 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Sat Aug 19 01:04:23 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753378_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753378_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Sat Aug 19 01:04:23 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-19 01:04:23\tMarkDuplicates\tStart of doWork freeMemory: 528792528; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 01:04:23\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-19 01:04:23\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam\n",
      "INFO\t2023-08-19 01:04:25\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:30,209,102\n",
      "INFO\t2023-08-19 01:04:25\tMarkDuplicates\tTracking 410 as yet unmatched pairs. 410 records in RAM.\n",
      "INFO\t2023-08-19 01:04:26\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 17:58,462,903\n",
      "INFO\t2023-08-19 01:04:26\tMarkDuplicates\tTracking 397 as yet unmatched pairs. 397 records in RAM.\n",
      "INFO\t2023-08-19 01:04:28\tMarkDuplicates\tRead 2826651 records. 0 pairs never matched.\n",
      "INFO\t2023-08-19 01:04:28\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 543619504; totalMemory: 917504000; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 01:04:28\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-19 01:04:28\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-19 01:04:28\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-19 01:04:28\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-19 01:04:28\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 908023880; totalMemory: 1454374912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 01:04:28\tMarkDuplicates\tMarking 63270 records as duplicates.\n",
      "INFO\t2023-08-19 01:04:28\tMarkDuplicates\tFound 446 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam\n",
      "INFO\t2023-08-19 01:04:28\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-19 01:04:45\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-19 01:04:45\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-19 01:04:45\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-19 01:04:45\tMarkDuplicates\tBefore output close freeMemory: 528090704; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-19 01:04:45\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-19 01:04:45\tMarkDuplicates\tAfter output close freeMemory: 528090704; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Sat Aug 19 01:04:45 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.36 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Sat Aug 19 01:04:47 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T010052.189997.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:04:48 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR696683_chr17_dedup.bam\n",
      "    output: results/variants/ERR696683_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR696683_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR696683_chr17.vcf\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "index file data/reference/genome.fa.fai not found, generating...\n",
      "[Sat Aug 19 01:05:24 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T010447.810824.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:05:25 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753368_chr17_dedup.bam\n",
      "    output: results/variants/ERR753368_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753368_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753368_chr17.vcf\n",
      "    wildcards: sample=ERR753368_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 01:06:20 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T010524.494929.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:06:21 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753369_chr17_dedup.bam\n",
      "    output: results/variants/ERR753369_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753369_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753369_chr17.vcf\n",
      "    wildcards: sample=ERR753369_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 01:07:18 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T010620.725562.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:07:19 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753370_chr17_dedup.bam\n",
      "    output: results/variants/ERR753370_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753370_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753370_chr17.vcf\n",
      "    wildcards: sample=ERR753370_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 01:08:48 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T010718.886926.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:08:49 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753371_chr17_dedup.bam\n",
      "    output: results/variants/ERR753371_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753371_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753371_chr17.vcf\n",
      "    wildcards: sample=ERR753371_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 01:09:12 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T010848.459588.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:09:14 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753372_chr17_dedup.bam\n",
      "    output: results/variants/ERR753372_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753372_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753372_chr17.vcf\n",
      "    wildcards: sample=ERR753372_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 01:10:12 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T010912.991517.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:10:13 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753373_chr17_dedup.bam\n",
      "    output: results/variants/ERR753373_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753373_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753373_chr17.vcf\n",
      "    wildcards: sample=ERR753373_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 01:10:56 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T011012.912853.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:10:57 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753374_chr17_dedup.bam\n",
      "    output: results/variants/ERR753374_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753374_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753374_chr17.vcf\n",
      "    wildcards: sample=ERR753374_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 01:11:25 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T011056.579602.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:11:26 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753375_chr17_dedup.bam\n",
      "    output: results/variants/ERR753375_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753375_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753375_chr17.vcf\n",
      "    wildcards: sample=ERR753375_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 01:11:54 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T011125.611706.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:11:55 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753376_chr17_dedup.bam\n",
      "    output: results/variants/ERR753376_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753376_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753376_chr17.vcf\n",
      "    wildcards: sample=ERR753376_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 01:12:24 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T011154.582441.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:12:25 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753377_chr17_dedup.bam\n",
      "    output: results/variants/ERR753377_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753377_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753377_chr17.vcf\n",
      "    wildcards: sample=ERR753377_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 01:12:56 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T011224.808769.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:12:57 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753378_chr17_dedup.bam\n",
      "    output: results/variants/ERR753378_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753378_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753378_chr17.vcf\n",
      "    wildcards: sample=ERR753378_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Sat Aug 19 01:13:50 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T011256.623457.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:13:51 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR696683_chr17.vcf\n",
      "    output: results/variants/vep/ERR696683_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR696683_chr17.txt\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Sat Aug 19 01:14:19 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T011350.879515.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:14:20 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753368_chr17.vcf\n",
      "    output: results/variants/vep/ERR753368_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753368_chr17.txt\n",
      "    wildcards: sample=ERR753368_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Sat Aug 19 01:14:56 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T011419.385239.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:14:57 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753369_chr17.vcf\n",
      "    output: results/variants/vep/ERR753369_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753369_chr17.txt\n",
      "    wildcards: sample=ERR753369_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Sat Aug 19 01:15:32 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T011456.352776.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:15:33 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753370_chr17.vcf\n",
      "    output: results/variants/vep/ERR753370_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753370_chr17.txt\n",
      "    wildcards: sample=ERR753370_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Sat Aug 19 01:16:11 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T011532.567224.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:16:12 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753371_chr17.vcf\n",
      "    output: results/variants/vep/ERR753371_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753371_chr17.txt\n",
      "    wildcards: sample=ERR753371_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Sat Aug 19 01:16:34 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T011611.300397.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:16:35 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753372_chr17.vcf\n",
      "    output: results/variants/vep/ERR753372_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753372_chr17.txt\n",
      "    wildcards: sample=ERR753372_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Sat Aug 19 01:17:12 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T011634.265627.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:17:13 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753373_chr17.vcf\n",
      "    output: results/variants/vep/ERR753373_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753373_chr17.txt\n",
      "    wildcards: sample=ERR753373_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Sat Aug 19 01:17:41 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T011712.250945.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:17:42 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753374_chr17.vcf\n",
      "    output: results/variants/vep/ERR753374_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753374_chr17.txt\n",
      "    wildcards: sample=ERR753374_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Sat Aug 19 01:18:04 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T011741.276058.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:18:05 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753375_chr17.vcf\n",
      "    output: results/variants/vep/ERR753375_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753375_chr17.txt\n",
      "    wildcards: sample=ERR753375_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Sat Aug 19 01:18:30 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T011804.272937.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:18:31 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753376_chr17.vcf\n",
      "    output: results/variants/vep/ERR753376_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753376_chr17.txt\n",
      "    wildcards: sample=ERR753376_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Sat Aug 19 01:18:56 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T011830.270154.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:18:57 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753377_chr17.vcf\n",
      "    output: results/variants/vep/ERR753377_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753377_chr17.txt\n",
      "    wildcards: sample=ERR753377_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Sat Aug 19 01:19:22 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T011856.477244.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:19:24 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753378_chr17.vcf\n",
      "    output: results/variants/vep/ERR753378_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753378_chr17.txt\n",
      "    wildcards: sample=ERR753378_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Sat Aug 19 01:19:59 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T011923.119361.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job              count    min threads    max threads\n",
      "-------------  -------  -------------  -------------\n",
      "parsing_dataR        1              1              1\n",
      "total                1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:19:59 2023]\n",
      "rule parsing_dataR:\n",
      "    input: code/05parsing_vep_data.R\n",
      "    output: tasks/13parsing_dataR.done\n",
      "    jobid: 0\n",
      "    reason: Params have changed since last execution\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Warning message:\n",
      "package ‘glue’ was built under R version 4.2.3 \n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">>> This is your sample list: ERR696683_chr17\n",
      ">>> This is your sample list: ERR753368_chr17\n",
      ">>> This is your sample list: ERR753369_chr17\n",
      ">>> This is your sample list: ERR753370_chr17\n",
      ">>> This is your sample list: ERR753371_chr17\n",
      ">>> This is your sample list: ERR753372_chr17\n",
      ">>> This is your sample list: ERR753373_chr17\n",
      ">>> This is your sample list: ERR753374_chr17\n",
      ">>> This is your sample list: ERR753375_chr17\n",
      ">>> This is your sample list: ERR753376_chr17\n",
      ">>> This is your sample list: ERR753377_chr17\n",
      ">>> This is your sample list: ERR753378_chr17\n",
      ">>> This is your gene: TP53\n",
      "# A tibble: 26 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "2 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "3 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "4 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "5 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "6 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "# ℹ 20 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 152 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7674109_G/A        17:7674109 A      ENSG… ENST00… Transcript   intron_var…\n",
      "2 17_7674109_G/A        17:7674109 A      ENSG… ENST00… Transcript   intron_var…\n",
      "3 17_7674109_G/A        17:7674109 A      ENSG… ENST00… Transcript   intron_var…\n",
      "4 17_7674109_G/A        17:7674109 A      ENSG… ENST00… Transcript   intron_var…\n",
      "5 17_7674109_G/A        17:7674109 A      ENSG… ENST00… Transcript   intron_var…\n",
      "6 17_7674109_G/A        17:7674109 A      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 146 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 249 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   downstream…\n",
      "2 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   intron_var…\n",
      "3 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   intron_var…\n",
      "4 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   downstream…\n",
      "5 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   downstream…\n",
      "6 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   downstream…\n",
      "# ℹ 243 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 130 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "2 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "3 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "4 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "5 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "6 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 124 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 58 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "2 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "3 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "4 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "5 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "6 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "# ℹ 52 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 126 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "2 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "3 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "4 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "5 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "6 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 120 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 68 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "2 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "3 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 62 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 0 × 44\n",
      "# ℹ 44 variables: #uploaded_variation <chr>, location <chr>, allele <chr>,\n",
      "#   gene <chr>, feature <chr>, feature_type <chr>, consequence <chr>,\n",
      "#   cdna_position <chr>, cds_position <chr>, protein_position <chr>,\n",
      "#   amino_acids <chr>, codons <chr>, existing_variation <chr>, impact <chr>,\n",
      "#   distance <chr>, strand <chr>, flags <chr>, variant_class <chr>,\n",
      "#   symbol <chr>, symbol_source <chr>, hgnc_id <chr>, biotype <chr>,\n",
      "#   mane_select <chr>, mane_plus_clinical <chr>, tsl <chr>, appris <chr>, …\n",
      "# A tibble: 26 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "2 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "3 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 20 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 52 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "2 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "3 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "4 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "5 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "6 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "# ℹ 46 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 52 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "2 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "3 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "4 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "5 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "6 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "# ℹ 46 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 149 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   downstream…\n",
      "2 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   intron_var…\n",
      "3 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   intron_var…\n",
      "4 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   downstream…\n",
      "5 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   downstream…\n",
      "6 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   downstream…\n",
      "# ℹ 143 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "[1] \"##==========================##\"\n",
      "[1] \"##===> WORK FINISHED!!! <===##\"\n",
      "[1] \"##==========================##\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Touching output file tasks/13parsing_dataR.done.\n",
      "[Sat Aug 19 01:20:03 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T011959.421244.snakemake.log\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['snakemake', '--cores', '1', 'parsing_dataR'], returncode=0)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Running the workflow for the 17º chormosome\n",
    "\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--force\", \"data/reference/genome.fa\"]) ## rule reference_genome \n",
    "\n",
    "for sample in sample_list:\n",
    "    ## Quality inspection\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/fastqc_result/\" + sample + \"_chr17_1_fastqc.html\",   \n",
    "                    \"data/processed/\" + sample + \"_chr17_1_fastp.fastq.gz\",   \n",
    "                    \"results/fastqc_result/trimmed/\" + sample + \"_chr17_1_fastp_fastqc.html\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    ## Mapping reads\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/mapped_reads/\" + sample + \"_chr17_sorted.sam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr17_sorted.bam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr17_dedup.bam\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    # Extracting variants  \n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/\" + sample + \"_chr17.vcf\"])   \n",
    "\n",
    "## VEP CLI\n",
    "for sample in sample_list:\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/vep/\" + sample + \"_chr17.txt\"])\n",
    "\n",
    "## Extracting the gene PIKCA3 from the VEP files\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"parsing_dataR\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['sed', '-i', '50s/TP53/PIK3CA/g', 'config.yaml'], returncode=0)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## reset config \n",
    "subprocess.run([\"sed\", \"-i\", \"6,+12s/chr17/chr3/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"24s/chr17/chr3/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"31s/chromosome.17.fa.gz/chromosome.3.fa.gz/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"50s/TP53/PIK3CA/g\", \"config.yaml\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## **Last rule**\n",
    "\n",
    "### **rule R_plotting**\n",
    "\n",
    "Plotting the data of the 5 gene tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job           count    min threads    max threads\n",
      "----------  -------  -------------  -------------\n",
      "R_plotting        1              1              1\n",
      "total             1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Sat Aug 19 01:25:37 2023]\n",
      "rule R_plotting:\n",
      "    input: code/06final_plot.R\n",
      "    output: results/biostatistics/plots/final_plot.png, results/biostatistics/plots/other_plot.png\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/biostatistics/plots/other_plot.png, results/biostatistics/plots/final_plot.png\n",
      "    resources: tmpdir=/tmp\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"##==========================##\"\n",
      "[1] \"##===> WORK FINISHED!!! <===##\"\n",
      "[1] \"##==========================##\"\n",
      "THE SCRIPT RAN WELL congrats :)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Sat Aug 19 01:25:41 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-19T012536.811650.snakemake.log\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['snakemake', '--cores', '1', 'R_plotting'], returncode=0)"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"R_plotting\"])"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Greference_tools",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
