{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# **Automatation of the CLI commands of the Snakemake workflow**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* Environment to use: [Greference_tools](code/environments/Greference_tools.yml)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Libreries\n",
    "import subprocess\n",
    "import os\n",
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "## lists for the loops\n",
    "chr_list        = [\"chr3\", \"chr5\", \"chr7\", \"chr12\", \"chr17\"]\n",
    "\n",
    "url_list        = [\"chromosome.3.fa.gz\", \"chromosome.5.fa.gz\", \"chromosome.7.fa.gz\", \n",
    "                   \"chromosome.12.fa.gz\", \"chromosome.17.fa.gz\"]\n",
    "\n",
    "gene_list       = [\"PIK3CA\", \"APC\", \"BRAF\", \"KRAS\", \"TP53\"]\n",
    "\n",
    "sample_list     = [\"ERR696683\", \"ERR753368\", \"ERR753369\" ,\"ERR753370\" ,\"ERR753371\" ,\"ERR753372\", \n",
    "                  \"ERR753373\", \"ERR753374\", \"ERR753375\", \"ERR753376\", \"ERR753377\", \"ERR753378\"]\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. **rule download_data**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job              count    min threads    max threads\n",
      "-------------  -------  -------------  -------------\n",
      "download_data        1              1              1\n",
      "total                1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Wed Aug 16 13:19:58 2023]\n",
      "rule download_data:\n",
      "    input: code/01dl_rawdata.bash, code/02rename.py\n",
      "    output: tasks/01download_data.done\n",
      "    jobid: 0\n",
      "    reason: Missing output files: tasks/01download_data.done\n",
      "    resources: tmpdir=/tmp\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==> THE FILE data/original_bam/C2466_GCCAAT.bwa.sorted.rmdup.recal.realigned.fixed.bam ALREADY EXIST!!! <==\n",
      "\n",
      "==> THE FILE data/original_bam/C2467_CGATGT.bwa.sorted.rmdup.recal.realigned.fixed.bam ALREADY EXIST!!! <==\n",
      "\n",
      "==> THE FILE data/original_bam/C2468_TTAGGC.bwa.sorted.rmdup.recal.realigned.fixed.bam ALREADY EXIST!!! <==\n",
      "\n",
      "==> THE FILE data/original_bam/C3481_GCCAAT.bwa.sorted.rmdup.recal.realigned.fixed.bam ALREADY EXIST!!! <==\n",
      "\n",
      "==> THE FILE data/original_bam/C3852_CTTGTA.bwa.sorted.rmdup.recal.realigned.fixed.bam ALREADY EXIST!!! <==\n",
      "\n",
      "==> THE FILE data/original_bam/C3890_TGACCA.bwa.sorted.rmdup.recal.realigned.fixed.bam ALREADY EXIST!!! <==\n",
      "\n",
      "==> THE FILE data/original_bam/NG-5502_C2463.bwa.sorted.rmdup.recal.realigned.sorted.fixed.bam ALREADY EXIST!!! <==\n",
      "\n",
      "==> THE FILE data/original_bam/NG-5502_C2461.bwa.sorted.rmdup.recal.realigned.sorted.fixed.bam ALREADY EXIST!!! <==\n",
      "\n",
      "==> THE FILE data/original_bam/C-3861_GCCAAT_L007.bwa.sorted.rmdup.recal.realigned.fixed.bam ALREADY EXIST!!! <==\n",
      "\n",
      "==> THE FILE data/original_bam/C2213_CGATGT.bwa.sorted.rmdup.recal.realigned.fixed.bam ALREADY EXIST!!! <==\n",
      "\n",
      "==> THE FILE data/original_bam/C896_TGACCA.bwa.sorted.rmdup.recal.realigned.fixed.bam ALREADY EXIST!!! <==\n",
      "\n",
      "==> THE FILE data/original_bam/C3988_ACAGTG.bwa.sorted.rmdup.recal.realigned.fixed.bam ALREADY EXIST!!! <==\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Touching output file tasks/01download_data.done.\n",
      "[Wed Aug 16 13:19:59 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-16T131958.278608.snakemake.log\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['snakemake', '--cores', '1', 'download_data'], returncode=0)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"download_data\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* rule pre_processing\n",
    "\n",
    "Filtering the chromosomes, make sure that the chromosomes of the config file are **ch3**. \n",
    "\n",
    "This rule is preatty long too... good luck\n",
    "\n",
    "When it finished, it is recomended to delete the SAM files not oredered."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "pre_processing        1              1              1\n",
      "total                 1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Wed Aug 16 15:13:13 2023]\n",
      "rule pre_processing:\n",
      "    input: code/03extracting_fastq.sh\n",
      "    output: data/original_bam/filtering/ERR696683_chr3_sorted.bam\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/original_bam/filtering/ERR696683_chr3_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 84123 singletons\n",
      "[M::bam2fq_mainloop] processed 4225255 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 28084 singletons\n",
      "[M::bam2fq_mainloop] processed 4361932 reads\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 29347 singletons\n",
      "[M::bam2fq_mainloop] processed 4672601 reads\n",
      "[bam_sort_core] merging from 3 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 72920 singletons\n",
      "[M::bam2fq_mainloop] processed 5975966 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 18585 singletons\n",
      "[M::bam2fq_mainloop] processed 3894697 reads\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 30337 singletons\n",
      "[M::bam2fq_mainloop] processed 4628597 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 8259 singletons\n",
      "[M::bam2fq_mainloop] processed 3945189 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 7892 singletons\n",
      "[M::bam2fq_mainloop] processed 2332698 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 23741 singletons\n",
      "[M::bam2fq_mainloop] processed 2588099 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 25677 singletons\n",
      "[M::bam2fq_mainloop] processed 3355889 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 49067 singletons\n",
      "[M::bam2fq_mainloop] processed 3506967 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 29704 singletons\n",
      "[M::bam2fq_mainloop] processed 3986056 reads\n",
      "[Wed Aug 16 15:56:18 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-16T151313.645596.snakemake.log\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "          Chromosomes transformed:\n",
      "           chr3 chr5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "pre_processing        1              1              1\n",
      "total                 1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Wed Aug 16 15:56:19 2023]\n",
      "rule pre_processing:\n",
      "    input: code/03extracting_fastq.sh\n",
      "    output: data/original_bam/filtering/ERR696683_chr5_sorted.bam\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/original_bam/filtering/ERR696683_chr5_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 74911 singletons\n",
      "[M::bam2fq_mainloop] processed 3565319 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 24550 singletons\n",
      "[M::bam2fq_mainloop] processed 3533184 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 25457 singletons\n",
      "[M::bam2fq_mainloop] processed 3773533 reads\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 64466 singletons\n",
      "[M::bam2fq_mainloop] processed 5100608 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 16803 singletons\n",
      "[M::bam2fq_mainloop] processed 3312473 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 26284 singletons\n",
      "[M::bam2fq_mainloop] processed 3726124 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 7129 singletons\n",
      "[M::bam2fq_mainloop] processed 3545831 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 6660 singletons\n",
      "[M::bam2fq_mainloop] processed 2092598 reads\n",
      "[M::bam2fq_mainloop] discarded 21394 singletons\n",
      "[M::bam2fq_mainloop] processed 2190882 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 22903 singletons\n",
      "[M::bam2fq_mainloop] processed 2828499 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 43852 singletons\n",
      "[M::bam2fq_mainloop] processed 2957702 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 25642 singletons\n",
      "[M::bam2fq_mainloop] processed 3190954 reads\n",
      "[Wed Aug 16 16:35:52 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-16T155618.676069.snakemake.log\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "          Chromosomes transformed:\n",
      "           chr5 chr7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "pre_processing        1              1              1\n",
      "total                 1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Wed Aug 16 16:35:54 2023]\n",
      "rule pre_processing:\n",
      "    input: code/03extracting_fastq.sh\n",
      "    output: data/original_bam/filtering/ERR696683_chr7_sorted.bam\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/original_bam/filtering/ERR696683_chr7_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 73902 singletons\n",
      "[M::bam2fq_mainloop] processed 3944452 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 24198 singletons\n",
      "[M::bam2fq_mainloop] processed 3636374 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 25287 singletons\n",
      "[M::bam2fq_mainloop] processed 3883825 reads\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 63834 singletons\n",
      "[M::bam2fq_mainloop] processed 5410754 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 16419 singletons\n",
      "[M::bam2fq_mainloop] processed 3571993 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 26268 singletons\n",
      "[M::bam2fq_mainloop] processed 3882598 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 7172 singletons\n",
      "[M::bam2fq_mainloop] processed 3188190 reads\n",
      "[M::bam2fq_mainloop] discarded 6833 singletons\n",
      "[M::bam2fq_mainloop] processed 1881699 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 21091 singletons\n",
      "[M::bam2fq_mainloop] processed 2428999 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 22516 singletons\n",
      "[M::bam2fq_mainloop] processed 3134720 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 43235 singletons\n",
      "[M::bam2fq_mainloop] processed 3307103 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 25927 singletons\n",
      "[M::bam2fq_mainloop] processed 3349951 reads\n",
      "[Wed Aug 16 17:15:52 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-16T163553.338929.snakemake.log\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "          Chromosomes transformed:\n",
      "           chr7 chr12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "pre_processing        1              1              1\n",
      "total                 1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Wed Aug 16 17:15:54 2023]\n",
      "rule pre_processing:\n",
      "    input: code/03extracting_fastq.sh\n",
      "    output: data/original_bam/filtering/ERR696683_chr12_sorted.bam\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/original_bam/filtering/ERR696683_chr12_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 64449 singletons\n",
      "[M::bam2fq_mainloop] processed 3693551 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 22477 singletons\n",
      "[M::bam2fq_mainloop] processed 3818357 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 23327 singletons\n",
      "[M::bam2fq_mainloop] processed 4131307 reads\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 55851 singletons\n",
      "[M::bam2fq_mainloop] processed 5154685 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 14328 singletons\n",
      "[M::bam2fq_mainloop] processed 3395746 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 24300 singletons\n",
      "[M::bam2fq_mainloop] processed 4081066 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 5449 singletons\n",
      "[M::bam2fq_mainloop] processed 2832515 reads\n",
      "[M::bam2fq_mainloop] discarded 5620 singletons\n",
      "[M::bam2fq_mainloop] processed 1705308 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 18852 singletons\n",
      "[M::bam2fq_mainloop] processed 2285894 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 19966 singletons\n",
      "[M::bam2fq_mainloop] processed 2951418 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 37782 singletons\n",
      "[M::bam2fq_mainloop] processed 3107334 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 24204 singletons\n",
      "[M::bam2fq_mainloop] processed 3531638 reads\n",
      "[Wed Aug 16 17:54:13 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-16T171553.450974.snakemake.log\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "          Chromosomes transformed:\n",
      "           chr12 chr17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "pre_processing        1              1              1\n",
      "total                 1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Wed Aug 16 17:54:14 2023]\n",
      "rule pre_processing:\n",
      "    input: code/03extracting_fastq.sh\n",
      "    output: data/original_bam/filtering/ERR696683_chr17_sorted.bam\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/original_bam/filtering/ERR696683_chr17_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 49649 singletons\n",
      "[M::bam2fq_mainloop] processed 3373755 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 18673 singletons\n",
      "[M::bam2fq_mainloop] processed 3458829 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 19617 singletons\n",
      "[M::bam2fq_mainloop] processed 3688229 reads\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 44102 singletons\n",
      "[M::bam2fq_mainloop] processed 4435138 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 10297 singletons\n",
      "[M::bam2fq_mainloop] processed 2970261 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 20675 singletons\n",
      "[M::bam2fq_mainloop] processed 3803001 reads\n",
      "[M::bam2fq_mainloop] discarded 3933 singletons\n",
      "[M::bam2fq_mainloop] processed 1690481 reads\n",
      "[M::bam2fq_mainloop] discarded 4157 singletons\n",
      "[M::bam2fq_mainloop] processed 988821 reads\n",
      "[M::bam2fq_mainloop] discarded 13970 singletons\n",
      "[M::bam2fq_mainloop] processed 2066508 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 15068 singletons\n",
      "[M::bam2fq_mainloop] processed 2654178 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 28410 singletons\n",
      "[M::bam2fq_mainloop] processed 2799742 reads\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[M::bam2fq_mainloop] discarded 20401 singletons\n",
      "[M::bam2fq_mainloop] processed 3329753 reads\n",
      "[Wed Aug 16 18:28:34 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-16T175413.784143.snakemake.log\n"
     ]
    }
   ],
   "source": [
    "## Running the 3º chromosome\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"data/original_bam/filtering/ERR696683_chr3_sorted.bam\"])\n",
    "\n",
    "## Running the rest of the chromosomes\n",
    "for chr in range(len(chr_list) - 1):\n",
    "    chr1 = chr_list[chr]\n",
    "    chr2 = chr_list[chr + 1]\n",
    "    \n",
    "    print(\"\"\"\n",
    "          Chromosomes transformed:\n",
    "          \"\"\", chr1, chr2)\n",
    "    \n",
    "    subprocess.run([\"sed\", \"-i\", \"6,+12s/\" + chr1 + \"/\" + chr2 + \"/g\", \"config.yaml\"])\n",
    "    subprocess.run([\"sed\", \"-i\", \"24s/\" + chr1 + \"/\" + chr2 + \"/g\", \"config.yaml\"])\n",
    "\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"data/original_bam/filtering/ERR696683_\" + chr2 + \"_sorted.bam\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['sed', '-i', '24s/chr17/chr3/g', 'config.yaml'], returncode=0)"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## reset config \n",
    "subprocess.run([\"sed\", \"-i\", \"6,+12s/chr17/chr3/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"24s/chr17/chr3/g\", \"config.yaml\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* rules reference_genome; fastqc; fastp; fastqc_trimmed; bwa_mapping; sam_to_bam; delete_duplicates; extracting_variants; vep_install; vep_cli; parsing_dataR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Nothing to be done (all requested files are present and up to date).\n",
      "Complete log: .snakemake/log/2023-08-17T091933.602241.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:19:35 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR696683_chr3_1.fastq.gz, data/raw/ERR696683_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR696683_chr3_1_fastqc.html, results/fastqc_result/ERR696683_chr3_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR696683_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 09:19:52 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:19:52 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR696683_chr3_1.fastq.gz, data/raw/ERR696683_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR696683_chr3_1_fastp.fastq.gz, data/processed/ERR696683_chr3_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR696683_chr3_1_fastp.fastq.gz, data/processed/ERR696683_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2070566\n",
      "total bases: 156530982\n",
      "Q20 bases: 155429955(99.2966%)\n",
      "Q30 bases: 151203846(96.5968%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2070566\n",
      "total bases: 156128924\n",
      "Q20 bases: 154231757(98.7849%)\n",
      "Q30 bases: 148068194(94.8371%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1294164\n",
      "total bases: 98291522\n",
      "Q20 bases: 97907734(99.6095%)\n",
      "Q30 bases: 96216656(97.8891%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1294164\n",
      "total bases: 98253340\n",
      "Q20 bases: 97583822(99.3186%)\n",
      "Q30 bases: 95093850(96.7843%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2588328\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1552804\n",
      "reads with adapter trimmed: 2016\n",
      "bases trimmed due to adapters: 53376\n",
      "\n",
      "Duplication rate: 5.84877%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 119\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR696683_chr3_1.fastq.gz -I data/raw/ERR696683_chr3_2.fastq.gz -o data/processed/ERR696683_chr3_1_fastp.fastq.gz -O data/processed/ERR696683_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Thu Aug 17 09:19:58 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:19:58 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR696683_chr3_1_fastp.fastq.gz, data/processed/ERR696683_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR696683_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR696683_chr3_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR696683_chr3_1_fastp.fastq.gz, data/processed/ERR696683_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 09:20:09 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T091934.064106.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:20:11 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753368_chr3_1.fastq.gz, data/raw/ERR753368_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753368_chr3_1_fastp.fastq.gz, data/processed/ERR753368_chr3_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753368_chr3_1_fastp.fastq.gz, data/processed/ERR753368_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2166924\n",
      "total bases: 164537592\n",
      "Q20 bases: 163892626(99.608%)\n",
      "Q30 bases: 159125497(96.7107%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2166924\n",
      "total bases: 164375778\n",
      "Q20 bases: 162943687(99.1288%)\n",
      "Q30 bases: 155573837(94.6452%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1870811\n",
      "total bases: 142174357\n",
      "Q20 bases: 141930007(99.8281%)\n",
      "Q30 bases: 139573591(98.1707%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1870811\n",
      "total bases: 142162640\n",
      "Q20 bases: 141472380(99.5145%)\n",
      "Q30 bases: 137298960(96.5788%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3741622\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 592226\n",
      "reads with adapter trimmed: 1044\n",
      "bases trimmed due to adapters: 26850\n",
      "\n",
      "Duplication rate: 2.0582%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753368_chr3_1.fastq.gz -I data/raw/ERR753368_chr3_2.fastq.gz -o data/processed/ERR753368_chr3_1_fastp.fastq.gz -O data/processed/ERR753368_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 7 seconds\n",
      "[Thu Aug 17 09:20:18 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:20:18 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753368_chr3_1_fastp.fastq.gz, data/processed/ERR753368_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753368_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753368_chr3_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753368_chr3_1_fastp.fastq.gz, data/processed/ERR753368_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 09:20:32 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:20:32 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753368_chr3_1.fastq.gz, data/raw/ERR753368_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753368_chr3_1_fastqc.html, results/fastqc_result/ERR753368_chr3_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753368_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753368_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 09:20:48 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T092009.885970.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:20:49 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753369_chr3_1.fastq.gz, data/raw/ERR753369_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753369_chr3_1_fastp.fastq.gz, data/processed/ERR753369_chr3_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753369_chr3_1_fastp.fastq.gz, data/processed/ERR753369_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2321627\n",
      "total bases: 176276293\n",
      "Q20 bases: 175555474(99.5911%)\n",
      "Q30 bases: 170311472(96.6162%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2321627\n",
      "total bases: 176111158\n",
      "Q20 bases: 174567395(99.1234%)\n",
      "Q30 bases: 166614064(94.6073%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 2002164\n",
      "total bases: 152156256\n",
      "Q20 bases: 151876395(99.8161%)\n",
      "Q30 bases: 149254086(98.0926%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 2002164\n",
      "total bases: 152144233\n",
      "Q20 bases: 151399863(99.5107%)\n",
      "Q30 bases: 146897284(96.5513%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 4004328\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 638926\n",
      "reads with adapter trimmed: 1028\n",
      "bases trimmed due to adapters: 24939\n",
      "\n",
      "Duplication rate: 2.38342%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753369_chr3_1.fastq.gz -I data/raw/ERR753369_chr3_2.fastq.gz -o data/processed/ERR753369_chr3_1_fastp.fastq.gz -O data/processed/ERR753369_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 7 seconds\n",
      "[Thu Aug 17 09:20:57 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:20:57 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753369_chr3_1.fastq.gz, data/raw/ERR753369_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753369_chr3_1_fastqc.html, results/fastqc_result/ERR753369_chr3_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753369_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753369_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 09:21:14 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:21:14 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753369_chr3_1_fastp.fastq.gz, data/processed/ERR753369_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753369_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753369_chr3_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753369_chr3_1_fastp.fastq.gz, data/processed/ERR753369_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 09:21:29 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T092048.422085.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:21:30 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753370_chr3_1.fastq.gz, data/raw/ERR753370_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753370_chr3_1_fastqc.html, results/fastqc_result/ERR753370_chr3_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753370_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753370_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 09:21:55 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:21:55 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753370_chr3_1.fastq.gz, data/raw/ERR753370_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753370_chr3_1_fastp.fastq.gz, data/processed/ERR753370_chr3_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753370_chr3_2_fastp.fastq.gz, data/processed/ERR753370_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2951523\n",
      "total bases: 297130164\n",
      "Q20 bases: 295118642(99.323%)\n",
      "Q30 bases: 286675921(96.4816%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2951523\n",
      "total bases: 297056363\n",
      "Q20 bases: 294260921(99.059%)\n",
      "Q30 bases: 283246706(95.3512%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 2950493\n",
      "total bases: 295883003\n",
      "Q20 bases: 293987676(99.3594%)\n",
      "Q30 bases: 285943525(96.6407%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 2950493\n",
      "total bases: 294939988\n",
      "Q20 bases: 292398511(99.1383%)\n",
      "Q30 bases: 282073643(95.6376%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 5900986\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 2060\n",
      "reads with adapter trimmed: 17930\n",
      "bases trimmed due to adapters: 173930\n",
      "\n",
      "Duplication rate: 4.38804%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 161\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753370_chr3_1.fastq.gz -I data/raw/ERR753370_chr3_2.fastq.gz -o data/processed/ERR753370_chr3_1_fastp.fastq.gz -O data/processed/ERR753370_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 13 seconds\n",
      "[Thu Aug 17 09:22:08 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:22:08 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753370_chr3_1_fastp.fastq.gz, data/processed/ERR753370_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753370_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753370_chr3_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753370_chr3_2_fastp.fastq.gz, data/processed/ERR753370_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 09:22:33 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T092129.313407.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:22:35 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753371_chr3_1.fastq.gz, data/raw/ERR753371_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753371_chr3_1_fastp.fastq.gz, data/processed/ERR753371_chr3_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753371_chr3_1_fastp.fastq.gz, data/processed/ERR753371_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1938056\n",
      "total bases: 146044753\n",
      "Q20 bases: 144732927(99.1018%)\n",
      "Q30 bases: 139492968(95.5139%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1938056\n",
      "total bases: 145676659\n",
      "Q20 bases: 143948860(98.8139%)\n",
      "Q30 bases: 138931875(95.37%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1007584\n",
      "total bases: 76483588\n",
      "Q20 bases: 76112999(99.5155%)\n",
      "Q30 bases: 74225186(97.0472%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1007584\n",
      "total bases: 76440869\n",
      "Q20 bases: 75910140(99.3057%)\n",
      "Q30 bases: 74249432(97.1332%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2015168\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1860944\n",
      "reads with adapter trimmed: 1798\n",
      "bases trimmed due to adapters: 42747\n",
      "\n",
      "Duplication rate: 22.2679%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753371_chr3_1.fastq.gz -I data/raw/ERR753371_chr3_2.fastq.gz -o data/processed/ERR753371_chr3_1_fastp.fastq.gz -O data/processed/ERR753371_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Thu Aug 17 09:22:41 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:22:41 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753371_chr3_1.fastq.gz, data/raw/ERR753371_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753371_chr3_1_fastqc.html, results/fastqc_result/ERR753371_chr3_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753371_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753371_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 09:22:56 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:22:56 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753371_chr3_1_fastp.fastq.gz, data/processed/ERR753371_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753371_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753371_chr3_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753371_chr3_1_fastp.fastq.gz, data/processed/ERR753371_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 09:23:04 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T092233.941667.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:23:06 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753372_chr3_1.fastq.gz, data/raw/ERR753372_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753372_chr3_1_fastqc.html, results/fastqc_result/ERR753372_chr3_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753372_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753372_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 09:23:23 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:23:23 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753372_chr3_1.fastq.gz, data/raw/ERR753372_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753372_chr3_1_fastp.fastq.gz, data/processed/ERR753372_chr3_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753372_chr3_2_fastp.fastq.gz, data/processed/ERR753372_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2299130\n",
      "total bases: 174570404\n",
      "Q20 bases: 173883945(99.6068%)\n",
      "Q30 bases: 168720211(96.6488%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2299130\n",
      "total bases: 174401358\n",
      "Q20 bases: 172878044(99.1265%)\n",
      "Q30 bases: 165079194(94.6548%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1978429\n",
      "total bases: 150352788\n",
      "Q20 bases: 150093042(99.8272%)\n",
      "Q30 bases: 147551682(98.137%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1978429\n",
      "total bases: 150340351\n",
      "Q20 bases: 149614171(99.517%)\n",
      "Q30 bases: 145248530(96.6131%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3956858\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 641402\n",
      "reads with adapter trimmed: 1252\n",
      "bases trimmed due to adapters: 32637\n",
      "\n",
      "Duplication rate: 1.99606%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753372_chr3_1.fastq.gz -I data/raw/ERR753372_chr3_2.fastq.gz -o data/processed/ERR753372_chr3_1_fastp.fastq.gz -O data/processed/ERR753372_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 7 seconds\n",
      "[Thu Aug 17 09:23:30 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:23:30 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753372_chr3_1_fastp.fastq.gz, data/processed/ERR753372_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753372_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753372_chr3_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753372_chr3_2_fastp.fastq.gz, data/processed/ERR753372_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 09:23:45 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T092305.035442.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:23:46 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753373_chr3_1.fastq.gz, data/raw/ERR753373_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753373_chr3_1_fastp.fastq.gz, data/processed/ERR753373_chr3_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753373_chr3_2_fastp.fastq.gz, data/processed/ERR753373_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1968465\n",
      "total bases: 188482891\n",
      "Q20 bases: 186996377(99.2113%)\n",
      "Q30 bases: 174397046(92.5267%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1968465\n",
      "total bases: 184512055\n",
      "Q20 bases: 182525877(98.9236%)\n",
      "Q30 bases: 170849210(92.5951%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1922198\n",
      "total bases: 181199250\n",
      "Q20 bases: 180156389(99.4245%)\n",
      "Q30 bases: 169961990(93.7984%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1922198\n",
      "total bases: 175114301\n",
      "Q20 bases: 174290716(99.5297%)\n",
      "Q30 bases: 165407202(94.4567%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3844396\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 92534\n",
      "reads with adapter trimmed: 1350\n",
      "bases trimmed due to adapters: 40609\n",
      "\n",
      "Duplication rate: 1.91409%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 153\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753373_chr3_1.fastq.gz -I data/raw/ERR753373_chr3_2.fastq.gz -o data/processed/ERR753373_chr3_1_fastp.fastq.gz -O data/processed/ERR753373_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 8 seconds\n",
      "[Thu Aug 17 09:23:55 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:23:55 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753373_chr3_1_fastp.fastq.gz, data/processed/ERR753373_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753373_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753373_chr3_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753373_chr3_2_fastp.fastq.gz, data/processed/ERR753373_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 09:24:11 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:24:11 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753373_chr3_1.fastq.gz, data/raw/ERR753373_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753373_chr3_1_fastqc.html, results/fastqc_result/ERR753373_chr3_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753373_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753373_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 09:24:28 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T092345.746770.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:24:29 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753374_chr3_1.fastq.gz, data/raw/ERR753374_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753374_chr3_1_fastqc.html, results/fastqc_result/ERR753374_chr3_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753374_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753374_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 09:24:40 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:24:40 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753374_chr3_1.fastq.gz, data/raw/ERR753374_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753374_chr3_1_fastp.fastq.gz, data/processed/ERR753374_chr3_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753374_chr3_1_fastp.fastq.gz, data/processed/ERR753374_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1162403\n",
      "total bases: 111198121\n",
      "Q20 bases: 110160620(99.067%)\n",
      "Q30 bases: 100749607(90.6037%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1162403\n",
      "total bases: 108805108\n",
      "Q20 bases: 107679458(98.9654%)\n",
      "Q30 bases: 99151878(91.128%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1126834\n",
      "total bases: 105363400\n",
      "Q20 bases: 104622908(99.2972%)\n",
      "Q30 bases: 97132511(92.1881%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1126834\n",
      "total bases: 102618294\n",
      "Q20 bases: 102036510(99.4331%)\n",
      "Q30 bases: 95498534(93.0619%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2253668\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 71138\n",
      "reads with adapter trimmed: 676\n",
      "bases trimmed due to adapters: 22682\n",
      "\n",
      "Duplication rate: 5.71343%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 152\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753374_chr3_1.fastq.gz -I data/raw/ERR753374_chr3_2.fastq.gz -o data/processed/ERR753374_chr3_1_fastp.fastq.gz -O data/processed/ERR753374_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Thu Aug 17 09:24:45 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:24:45 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753374_chr3_1_fastp.fastq.gz, data/processed/ERR753374_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753374_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753374_chr3_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753374_chr3_1_fastp.fastq.gz, data/processed/ERR753374_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 09:24:56 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T092428.498263.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:24:58 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753375_chr3_1.fastq.gz, data/raw/ERR753375_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753375_chr3_1_fastp.fastq.gz, data/processed/ERR753375_chr3_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753375_chr3_1_fastp.fastq.gz, data/processed/ERR753375_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1282179\n",
      "total bases: 96858250\n",
      "Q20 bases: 96265562(99.3881%)\n",
      "Q30 bases: 94113609(97.1663%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1282179\n",
      "total bases: 96835483\n",
      "Q20 bases: 95589371(98.7132%)\n",
      "Q30 bases: 93289559(96.3382%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 841057\n",
      "total bases: 63862131\n",
      "Q20 bases: 63654927(99.6755%)\n",
      "Q30 bases: 62708243(98.1932%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 841057\n",
      "total bases: 63872391\n",
      "Q20 bases: 63259861(99.041%)\n",
      "Q30 bases: 62279464(97.5061%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1682114\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 882244\n",
      "reads with adapter trimmed: 996\n",
      "bases trimmed due to adapters: 27146\n",
      "\n",
      "Duplication rate: 1.71778%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753375_chr3_1.fastq.gz -I data/raw/ERR753375_chr3_2.fastq.gz -o data/processed/ERR753375_chr3_1_fastp.fastq.gz -O data/processed/ERR753375_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Thu Aug 17 09:25:02 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:25:02 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753375_chr3_1.fastq.gz, data/raw/ERR753375_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753375_chr3_1_fastqc.html, results/fastqc_result/ERR753375_chr3_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753375_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753375_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 09:25:13 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:25:13 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753375_chr3_1_fastp.fastq.gz, data/processed/ERR753375_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753375_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753375_chr3_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753375_chr3_1_fastp.fastq.gz, data/processed/ERR753375_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 09:25:22 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T092457.008479.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:25:23 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753376_chr3_1.fastq.gz, data/raw/ERR753376_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753376_chr3_1_fastqc.html, results/fastqc_result/ERR753376_chr3_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753376_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753376_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 09:25:36 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:25:36 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753376_chr3_1.fastq.gz, data/raw/ERR753376_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753376_chr3_1_fastp.fastq.gz, data/processed/ERR753376_chr3_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753376_chr3_1_fastp.fastq.gz, data/processed/ERR753376_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1665106\n",
      "total bases: 125882357\n",
      "Q20 bases: 125020665(99.3155%)\n",
      "Q30 bases: 121633177(96.6245%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1665106\n",
      "total bases: 125542593\n",
      "Q20 bases: 124005519(98.7757%)\n",
      "Q30 bases: 119018691(94.8034%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1037289\n",
      "total bases: 78781993\n",
      "Q20 bases: 78486134(99.6245%)\n",
      "Q30 bases: 77145479(97.9227%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1037289\n",
      "total bases: 78752401\n",
      "Q20 bases: 78213072(99.3152%)\n",
      "Q30 bases: 76205525(96.766%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2074578\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1255634\n",
      "reads with adapter trimmed: 1176\n",
      "bases trimmed due to adapters: 29546\n",
      "\n",
      "Duplication rate: 9.11294%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 119\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753376_chr3_1.fastq.gz -I data/raw/ERR753376_chr3_2.fastq.gz -o data/processed/ERR753376_chr3_1_fastp.fastq.gz -O data/processed/ERR753376_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Thu Aug 17 09:25:41 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:25:41 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753376_chr3_1_fastp.fastq.gz, data/processed/ERR753376_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753376_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753376_chr3_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753376_chr3_1_fastp.fastq.gz, data/processed/ERR753376_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 09:25:50 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T092522.618559.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:25:52 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753377_chr3_1.fastq.gz, data/raw/ERR753377_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753377_chr3_1_fastqc.html, results/fastqc_result/ERR753377_chr3_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753377_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753377_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 09:26:05 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:26:05 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753377_chr3_1.fastq.gz, data/raw/ERR753377_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753377_chr3_1_fastp.fastq.gz, data/processed/ERR753377_chr3_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753377_chr3_2_fastp.fastq.gz, data/processed/ERR753377_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1728950\n",
      "total bases: 130722103\n",
      "Q20 bases: 129808577(99.3012%)\n",
      "Q30 bases: 126267707(96.5925%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1728950\n",
      "total bases: 130414142\n",
      "Q20 bases: 128895041(98.8352%)\n",
      "Q30 bases: 123933005(95.0303%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1099156\n",
      "total bases: 83481297\n",
      "Q20 bases: 83162064(99.6176%)\n",
      "Q30 bases: 81728416(97.9003%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1099156\n",
      "total bases: 83450542\n",
      "Q20 bases: 82907906(99.3498%)\n",
      "Q30 bases: 80876920(96.916%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2198312\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1259588\n",
      "reads with adapter trimmed: 2030\n",
      "bases trimmed due to adapters: 50654\n",
      "\n",
      "Duplication rate: 9.10942%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 119\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753377_chr3_1.fastq.gz -I data/raw/ERR753377_chr3_2.fastq.gz -o data/processed/ERR753377_chr3_1_fastp.fastq.gz -O data/processed/ERR753377_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Thu Aug 17 09:26:10 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:26:10 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753377_chr3_1_fastp.fastq.gz, data/processed/ERR753377_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753377_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753377_chr3_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753377_chr3_2_fastp.fastq.gz, data/processed/ERR753377_chr3_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 09:26:19 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T092550.991094.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:26:20 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753378_chr3_1.fastq.gz, data/raw/ERR753378_chr3_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753378_chr3_1_fastqc.html, results/fastqc_result/ERR753378_chr3_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753378_chr3_1_fastqc.html\n",
      "    wildcards: sample=ERR753378_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr3_1.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr3_1.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr3_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr3_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr3_2.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr3_2.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr3_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 09:26:35 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:26:35 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753378_chr3_1.fastq.gz, data/raw/ERR753378_chr3_2.fastq.gz\n",
      "    output: data/processed/ERR753378_chr3_1_fastp.fastq.gz, data/processed/ERR753378_chr3_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753378_chr3_1_fastp.fastq.gz, data/processed/ERR753378_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1978176\n",
      "total bases: 150201904\n",
      "Q20 bases: 149614211(99.6087%)\n",
      "Q30 bases: 145205101(96.6733%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1978176\n",
      "total bases: 150054321\n",
      "Q20 bases: 148742134(99.1255%)\n",
      "Q30 bases: 142005411(94.636%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1708379\n",
      "total bases: 129830167\n",
      "Q20 bases: 129604043(99.8258%)\n",
      "Q30 bases: 127403551(98.1309%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1708379\n",
      "total bases: 129819431\n",
      "Q20 bases: 129183589(99.5102%)\n",
      "Q30 bases: 125347189(96.555%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3416758\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 539594\n",
      "reads with adapter trimmed: 934\n",
      "bases trimmed due to adapters: 22437\n",
      "\n",
      "Duplication rate: 1.97416%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753378_chr3_1.fastq.gz -I data/raw/ERR753378_chr3_2.fastq.gz -o data/processed/ERR753378_chr3_1_fastp.fastq.gz -O data/processed/ERR753378_chr3_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Thu Aug 17 09:26:42 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:26:42 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753378_chr3_1_fastp.fastq.gz, data/processed/ERR753378_chr3_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753378_chr3_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753378_chr3_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753378_chr3_1_fastp.fastq.gz, data/processed/ERR753378_chr3_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr3_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr3_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr3_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr3_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 09:26:55 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T092619.705761.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:26:56 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR696683_chr3_1_fastp.fastq.gz, data/processed/ERR696683_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR696683_chr3.sam, results/mapped_reads/ERR696683_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR696683_chr3_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR696683_chr3_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.63 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 70.75 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.57 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.41 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 41.56 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 114.863 sec; CPU: 113.925 sec\n",
      "[Thu Aug 17 09:30:23 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:30:23 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR696683_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr3.bam, results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR696683_chr3.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR696683_chr3_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 09:31:08 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:31:08 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR696683_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "09:31:09.338 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 09:31:09 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR696683_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR696683_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 09:31:09 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 09:31:09\tMarkDuplicates\tStart of doWork freeMemory: 528780256; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 09:31:09\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 09:31:09\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam\n",
      "INFO\t2023-08-17 09:31:10\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:69,122,311\n",
      "INFO\t2023-08-17 09:31:10\tMarkDuplicates\tTracking 802 as yet unmatched pairs. 802 records in RAM.\n",
      "INFO\t2023-08-17 09:31:12\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 3:149,750,771\n",
      "INFO\t2023-08-17 09:31:12\tMarkDuplicates\tTracking 676 as yet unmatched pairs. 676 records in RAM.\n",
      "INFO\t2023-08-17 09:31:13\tMarkDuplicates\tRead 2626393 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 09:31:13\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 474431496; totalMemory: 826277888; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 09:31:13\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 09:31:13\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 09:31:13\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 09:31:13\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 09:31:13\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 978266184; totalMemory: 1524629504; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 09:31:13\tMarkDuplicates\tMarking 245192 records as duplicates.\n",
      "INFO\t2023-08-17 09:31:13\tMarkDuplicates\tFound 114 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr3_sorted.bam\n",
      "INFO\t2023-08-17 09:31:13\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 09:31:29\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 09:31:29\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 09:31:29\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 09:31:29\tMarkDuplicates\tBefore output close freeMemory: 528078208; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 09:31:29\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 09:31:29\tMarkDuplicates\tAfter output close freeMemory: 528078208; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 09:31:29 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.34 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 09:31:31 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T092655.538298.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:31:33 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753368_chr3_1_fastp.fastq.gz, data/processed/ERR753368_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753368_chr3.sam, results/mapped_reads/ERR753368_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753368_chr3_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753368_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.66 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 70.92 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.60 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.39 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 42.04 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 115.726 sec; CPU: 114.615 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 09:35:57 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:35:57 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753368_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr3.bam, results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753368_chr3.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753368_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 09:37:03 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:37:03 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753368_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753368_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "09:37:04.598 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 09:37:04 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753368_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753368_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 09:37:04 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 09:37:04\tMarkDuplicates\tStart of doWork freeMemory: 528784968; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 09:37:04\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 09:37:04\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam\n",
      "INFO\t2023-08-17 09:37:06\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:48,469,724\n",
      "INFO\t2023-08-17 09:37:06\tMarkDuplicates\tTracking 478 as yet unmatched pairs. 478 records in RAM.\n",
      "INFO\t2023-08-17 09:37:07\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 3:108,377,297\n",
      "INFO\t2023-08-17 09:37:07\tMarkDuplicates\tTracking 764 as yet unmatched pairs. 764 records in RAM.\n",
      "INFO\t2023-08-17 09:37:08\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 3:152,590,573\n",
      "INFO\t2023-08-17 09:37:08\tMarkDuplicates\tTracking 383 as yet unmatched pairs. 383 records in RAM.\n",
      "INFO\t2023-08-17 09:37:09\tMarkDuplicates\tRead 3764483 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 09:37:09\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 681911496; totalMemory: 1158676480; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 09:37:09\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 09:37:10\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 09:37:10\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 09:37:10\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 09:37:10\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1149189320; totalMemory: 1695547392; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 09:37:10\tMarkDuplicates\tMarking 92311 records as duplicates.\n",
      "INFO\t2023-08-17 09:37:10\tMarkDuplicates\tFound 612 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr3_sorted.bam\n",
      "INFO\t2023-08-17 09:37:10\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 09:37:33\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 09:37:33\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 09:37:33\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 09:37:33\tMarkDuplicates\tBefore output close freeMemory: 528083528; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 09:37:33\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 09:37:33\tMarkDuplicates\tAfter output close freeMemory: 528083528; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 09:37:33 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.48 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 09:37:35 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T093131.782619.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:37:37 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753369_chr3_1_fastp.fastq.gz, data/processed/ERR753369_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753369_chr3.sam, results/mapped_reads/ERR753369_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753369_chr3_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753369_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.66 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 70.86 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.59 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.39 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 41.78 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 120.658 sec; CPU: 114.286 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 09:42:31 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:42:31 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753369_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr3.bam, results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753369_chr3.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753369_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 09:43:40 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:43:40 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753369_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753369_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "09:43:41.673 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 09:43:41 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753369_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753369_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 09:43:41 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 09:43:41\tMarkDuplicates\tStart of doWork freeMemory: 528789888; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 09:43:41\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 09:43:41\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam\n",
      "INFO\t2023-08-17 09:43:43\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:47,407,707\n",
      "INFO\t2023-08-17 09:43:43\tMarkDuplicates\tTracking 452 as yet unmatched pairs. 452 records in RAM.\n",
      "INFO\t2023-08-17 09:43:44\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 3:100,386,906\n",
      "INFO\t2023-08-17 09:43:44\tMarkDuplicates\tTracking 645 as yet unmatched pairs. 645 records in RAM.\n",
      "INFO\t2023-08-17 09:43:46\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 3:142,332,829\n",
      "INFO\t2023-08-17 09:43:46\tMarkDuplicates\tTracking 549 as yet unmatched pairs. 549 records in RAM.\n",
      "INFO\t2023-08-17 09:43:47\tMarkDuplicates\tRead     4,000,000 records.  Elapsed time: 00:00:05s.  Time for last 1,000,000:    1s.  Last read position: 3:197,558,108\n",
      "INFO\t2023-08-17 09:43:47\tMarkDuplicates\tTracking 17 as yet unmatched pairs. 17 records in RAM.\n",
      "INFO\t2023-08-17 09:43:47\tMarkDuplicates\tRead 4026673 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 09:43:47\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 447449248; totalMemory: 952107008; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 09:43:47\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 09:43:47\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 09:43:48\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 09:43:48\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 09:43:48\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 942624472; totalMemory: 1488977920; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 09:43:48\tMarkDuplicates\tMarking 115043 records as duplicates.\n",
      "INFO\t2023-08-17 09:43:48\tMarkDuplicates\tFound 621 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr3_sorted.bam\n",
      "INFO\t2023-08-17 09:43:48\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 09:44:11\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 09:44:11\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 09:44:11\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 09:44:11\tMarkDuplicates\tBefore output close freeMemory: 528088224; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 09:44:11\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 09:44:11\tMarkDuplicates\tAfter output close freeMemory: 528088224; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 09:44:11 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.50 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 09:44:14 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T093736.137898.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:44:15 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753370_chr3_1_fastp.fastq.gz, data/processed/ERR753370_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753370_chr3.sam, results/mapped_reads/ERR753370_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753370_chr3_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753370_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.65 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 70.82 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.60 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.39 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 41.91 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 116.880 sec; CPU: 114.361 sec\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 09:51:21 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:51:21 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753370_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr3.bam, results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753370_chr3.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753370_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 09:53:41 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:53:41 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753370_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753370_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "09:53:42.852 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 09:53:42 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753370_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753370_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 09:53:42 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 09:53:42\tMarkDuplicates\tStart of doWork freeMemory: 528824496; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 09:53:42\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 09:53:42\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam\n",
      "INFO\t2023-08-17 09:53:44\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:37,998,197\n",
      "INFO\t2023-08-17 09:53:44\tMarkDuplicates\tTracking 988 as yet unmatched pairs. 988 records in RAM.\n",
      "INFO\t2023-08-17 09:53:46\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 3:56,434,427\n",
      "INFO\t2023-08-17 09:53:46\tMarkDuplicates\tTracking 1783 as yet unmatched pairs. 1783 records in RAM.\n",
      "INFO\t2023-08-17 09:53:47\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 3:101,652,465\n",
      "INFO\t2023-08-17 09:53:47\tMarkDuplicates\tTracking 1971 as yet unmatched pairs. 1971 records in RAM.\n",
      "INFO\t2023-08-17 09:53:48\tMarkDuplicates\tRead     4,000,000 records.  Elapsed time: 00:00:06s.  Time for last 1,000,000:    1s.  Last read position: 3:130,736,040\n",
      "INFO\t2023-08-17 09:53:48\tMarkDuplicates\tTracking 1546 as yet unmatched pairs. 1546 records in RAM.\n",
      "INFO\t2023-08-17 09:53:50\tMarkDuplicates\tRead     5,000,000 records.  Elapsed time: 00:00:07s.  Time for last 1,000,000:    1s.  Last read position: 3:167,328,091\n",
      "INFO\t2023-08-17 09:53:50\tMarkDuplicates\tTracking 1026 as yet unmatched pairs. 1026 records in RAM.\n",
      "INFO\t2023-08-17 09:53:51\tMarkDuplicates\tRead     6,000,000 records.  Elapsed time: 00:00:08s.  Time for last 1,000,000:    1s.  Last read position: 3:198,076,827\n",
      "INFO\t2023-08-17 09:53:51\tMarkDuplicates\tTracking 183 as yet unmatched pairs. 183 records in RAM.\n",
      "INFO\t2023-08-17 09:53:51\tMarkDuplicates\tRead 6007564 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 09:53:51\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 488792200; totalMemory: 1199570944; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 09:53:51\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 09:53:52\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 09:53:52\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 09:53:53\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 09:53:53\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1190123320; totalMemory: 1736441856; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 09:53:53\tMarkDuplicates\tMarking 429709 records as duplicates.\n",
      "INFO\t2023-08-17 09:53:53\tMarkDuplicates\tFound 269 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr3_sorted.bam\n",
      "INFO\t2023-08-17 09:53:53\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 09:54:35\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 09:54:35\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 09:54:35\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 09:54:35\tMarkDuplicates\tBefore output close freeMemory: 528122632; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 09:54:35\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 09:54:35\tMarkDuplicates\tAfter output close freeMemory: 528122632; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 09:54:35 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.87 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 09:54:40 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T094414.495868.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:54:42 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753371_chr3_1_fastp.fastq.gz, data/processed/ERR753371_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753371_chr3.sam, results/mapped_reads/ERR753371_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753371_chr3_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753371_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.63 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 69.34 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.59 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.39 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 41.46 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 118.845 sec; CPU: 112.419 sec\n",
      "[Thu Aug 17 09:57:51 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:57:51 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753371_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr3.bam, results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753371_chr3.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753371_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 09:58:23 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:58:23 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753371_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753371_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "09:58:24.643 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 09:58:24 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753371_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753371_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 09:58:24 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 09:58:24\tMarkDuplicates\tStart of doWork freeMemory: 528786208; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 09:58:24\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 09:58:24\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam\n",
      "INFO\t2023-08-17 09:58:26\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:108,616,467\n",
      "INFO\t2023-08-17 09:58:26\tMarkDuplicates\tTracking 176 as yet unmatched pairs. 176 records in RAM.\n",
      "INFO\t2023-08-17 09:58:27\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 3:197,282,714\n",
      "INFO\t2023-08-17 09:58:27\tMarkDuplicates\tTracking 70 as yet unmatched pairs. 70 records in RAM.\n",
      "INFO\t2023-08-17 09:58:27\tMarkDuplicates\tRead 2019762 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 09:58:27\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 312323368; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 09:58:27\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 09:58:27\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 09:58:28\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 09:58:28\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 09:58:28\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 847198344; totalMemory: 1393557504; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 09:58:28\tMarkDuplicates\tMarking 829938 records as duplicates.\n",
      "INFO\t2023-08-17 09:58:28\tMarkDuplicates\tFound 68 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr3_sorted.bam\n",
      "INFO\t2023-08-17 09:58:28\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 09:58:40\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 09:58:40\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 09:58:40\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 09:58:40\tMarkDuplicates\tBefore output close freeMemory: 528082256; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 09:58:40\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 09:58:40\tMarkDuplicates\tAfter output close freeMemory: 528082256; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 09:58:40 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.26 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 09:58:41 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T095440.768045.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 09:58:43 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753372_chr3_1_fastp.fastq.gz, data/processed/ERR753372_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753372_chr3.sam, results/mapped_reads/ERR753372_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753372_chr3_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753372_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.65 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 69.30 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.58 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.39 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 41.32 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 113.422 sec; CPU: 112.240 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 10:03:25 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:03:25 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753372_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr3.bam, results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753372_chr3.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753372_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 10:04:33 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:04:33 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753372_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753372_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "10:04:34.693 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 10:04:34 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753372_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753372_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 10:04:34 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 10:04:34\tMarkDuplicates\tStart of doWork freeMemory: 528786376; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:04:34\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 10:04:34\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam\n",
      "INFO\t2023-08-17 10:04:36\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:47,409,234\n",
      "INFO\t2023-08-17 10:04:36\tMarkDuplicates\tTracking 449 as yet unmatched pairs. 449 records in RAM.\n",
      "INFO\t2023-08-17 10:04:37\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 3:99,848,912\n",
      "INFO\t2023-08-17 10:04:37\tMarkDuplicates\tTracking 681 as yet unmatched pairs. 681 records in RAM.\n",
      "INFO\t2023-08-17 10:04:39\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 3:142,486,165\n",
      "INFO\t2023-08-17 10:04:39\tMarkDuplicates\tTracking 448 as yet unmatched pairs. 448 records in RAM.\n",
      "INFO\t2023-08-17 10:04:40\tMarkDuplicates\tRead 3984054 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 10:04:40\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 500588192; totalMemory: 1000341504; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:04:40\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 10:04:40\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 10:04:41\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 10:04:41\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 10:04:41\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 990855192; totalMemory: 1537212416; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:04:41\tMarkDuplicates\tMarking 95088 records as duplicates.\n",
      "INFO\t2023-08-17 10:04:41\tMarkDuplicates\tFound 627 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr3_sorted.bam\n",
      "INFO\t2023-08-17 10:04:41\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 10:05:03\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 10:05:03\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 10:05:03\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 10:05:03\tMarkDuplicates\tBefore output close freeMemory: 528084776; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:05:03\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 10:05:03\tMarkDuplicates\tAfter output close freeMemory: 528084776; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 10:05:03 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.48 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 10:05:06 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T095841.820747.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:05:07 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753373_chr3_1_fastp.fastq.gz, data/processed/ERR753373_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753373_chr3.sam, results/mapped_reads/ERR753373_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753373_chr3_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753373_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.64 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 72.24 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.57 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.39 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 41.27 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 121.728 sec; CPU: 115.112 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 10:10:58 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:10:58 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753373_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr3.bam, results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753373_chr3.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753373_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 10:12:52 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:12:52 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753373_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753373_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "10:12:53.103 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 10:12:53 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753373_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753373_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 10:12:53 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 10:12:53\tMarkDuplicates\tStart of doWork freeMemory: 528779528; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:12:53\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 10:12:53\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam\n",
      "INFO\t2023-08-17 10:12:54\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:52,256,336\n",
      "INFO\t2023-08-17 10:12:54\tMarkDuplicates\tTracking 258 as yet unmatched pairs. 258 records in RAM.\n",
      "INFO\t2023-08-17 10:12:56\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 3:108,161,461\n",
      "INFO\t2023-08-17 10:12:56\tMarkDuplicates\tTracking 296 as yet unmatched pairs. 296 records in RAM.\n",
      "INFO\t2023-08-17 10:12:57\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 3:154,972,789\n",
      "INFO\t2023-08-17 10:12:57\tMarkDuplicates\tTracking 205 as yet unmatched pairs. 205 records in RAM.\n",
      "INFO\t2023-08-17 10:12:59\tMarkDuplicates\tRead 3874269 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 10:12:59\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 509477552; totalMemory: 997195776; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:12:59\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 10:12:59\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 10:12:59\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 10:13:00\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 10:13:00\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 987702336; totalMemory: 1534066688; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:13:00\tMarkDuplicates\tMarking 104535 records as duplicates.\n",
      "INFO\t2023-08-17 10:13:00\tMarkDuplicates\tFound 34216 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr3_sorted.bam\n",
      "INFO\t2023-08-17 10:13:00\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 10:13:32\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 10:13:32\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 10:13:32\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 10:13:32\tMarkDuplicates\tBefore output close freeMemory: 528076912; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:13:32\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 10:13:32\tMarkDuplicates\tAfter output close freeMemory: 528076912; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 10:13:32 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.66 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 10:13:35 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T100506.493377.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:13:37 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753374_chr3_1_fastp.fastq.gz, data/processed/ERR753374_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753374_chr3.sam, results/mapped_reads/ERR753374_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753374_chr3_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753374_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.64 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 69.54 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.58 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.40 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 41.35 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 119.664 sec; CPU: 112.516 sec\n",
      "[Thu Aug 17 10:17:45 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:17:45 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753374_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr3.bam, results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753374_chr3.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753374_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 10:18:44 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:18:44 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753374_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753374_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "10:18:45.014 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 10:18:45 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753374_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753374_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 10:18:45 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 10:18:45\tMarkDuplicates\tStart of doWork freeMemory: 528784824; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:18:45\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 10:18:45\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam\n",
      "INFO\t2023-08-17 10:18:46\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:96,373,675\n",
      "INFO\t2023-08-17 10:18:46\tMarkDuplicates\tTracking 262 as yet unmatched pairs. 262 records in RAM.\n",
      "INFO\t2023-08-17 10:18:48\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 3:175,256,684\n",
      "INFO\t2023-08-17 10:18:48\tMarkDuplicates\tTracking 110 as yet unmatched pairs. 110 records in RAM.\n",
      "INFO\t2023-08-17 10:18:48\tMarkDuplicates\tRead 2270052 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 10:18:48\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 574840944; totalMemory: 890241024; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:18:48\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 10:18:48\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 10:18:49\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 10:18:49\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 10:18:49\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 880752784; totalMemory: 1427111936; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:18:49\tMarkDuplicates\tMarking 244464 records as duplicates.\n",
      "INFO\t2023-08-17 10:18:49\tMarkDuplicates\tFound 81740 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr3_sorted.bam\n",
      "INFO\t2023-08-17 10:18:49\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 10:19:08\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 10:19:08\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 10:19:08\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 10:19:08\tMarkDuplicates\tBefore output close freeMemory: 528082064; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:19:08\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 10:19:08\tMarkDuplicates\tAfter output close freeMemory: 528082064; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 10:19:08 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.39 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 10:19:10 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T101335.712937.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:19:11 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753375_chr3_1_fastp.fastq.gz, data/processed/ERR753375_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753375_chr3.sam, results/mapped_reads/ERR753375_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753375_chr3_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753375_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.65 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 69.47 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.58 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.39 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 41.41 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 115.863 sec; CPU: 112.504 sec\n",
      "[Thu Aug 17 10:22:11 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:22:11 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753375_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr3.bam, results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753375_chr3.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753375_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 10:22:43 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:22:43 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753375_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753375_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "10:22:44.070 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 10:22:44 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753375_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753375_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 10:22:44 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 10:22:44\tMarkDuplicates\tStart of doWork freeMemory: 528787272; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:22:44\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 10:22:44\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam\n",
      "INFO\t2023-08-17 10:22:45\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:121,913,100\n",
      "INFO\t2023-08-17 10:22:45\tMarkDuplicates\tTracking 260 as yet unmatched pairs. 260 records in RAM.\n",
      "INFO\t2023-08-17 10:22:46\tMarkDuplicates\tRead 1693364 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 10:22:46\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 283567912; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:22:46\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 10:22:46\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 10:22:47\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 10:22:47\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 10:22:47\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 787433296; totalMemory: 1333788672; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:22:47\tMarkDuplicates\tMarking 82804 records as duplicates.\n",
      "INFO\t2023-08-17 10:22:47\tMarkDuplicates\tFound 199 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr3_sorted.bam\n",
      "INFO\t2023-08-17 10:22:47\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 10:22:57\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 10:22:57\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 10:22:57\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 10:22:57\tMarkDuplicates\tBefore output close freeMemory: 528085432; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:22:57\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 10:22:57\tMarkDuplicates\tAfter output close freeMemory: 528085432; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 10:22:57 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.23 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 10:22:58 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T101910.284943.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:23:00 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753376_chr3_1_fastp.fastq.gz, data/processed/ERR753376_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753376_chr3.sam, results/mapped_reads/ERR753376_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753376_chr3_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753376_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.65 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 73.57 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.63 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.43 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 45.35 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 127.034 sec; CPU: 120.618 sec\n",
      "[Thu Aug 17 10:26:35 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:26:35 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753376_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr3.bam, results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753376_chr3.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753376_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 10:27:18 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:27:18 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753376_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753376_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "10:27:19.144 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 10:27:19 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753376_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753376_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 10:27:19 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 10:27:19\tMarkDuplicates\tStart of doWork freeMemory: 528786640; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:27:19\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 10:27:19\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam\n",
      "INFO\t2023-08-17 10:27:20\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:101,493,693\n",
      "INFO\t2023-08-17 10:27:20\tMarkDuplicates\tTracking 285 as yet unmatched pairs. 285 records in RAM.\n",
      "INFO\t2023-08-17 10:27:22\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 3:193,643,851\n",
      "INFO\t2023-08-17 10:27:22\tMarkDuplicates\tTracking 41 as yet unmatched pairs. 41 records in RAM.\n",
      "INFO\t2023-08-17 10:27:22\tMarkDuplicates\tRead 2085915 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 10:27:22\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 592187856; totalMemory: 888143872; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:27:22\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 10:27:22\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 10:27:22\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 10:27:23\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 10:27:23\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 881802688; totalMemory: 1428160512; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:27:23\tMarkDuplicates\tMarking 302661 records as duplicates.\n",
      "INFO\t2023-08-17 10:27:23\tMarkDuplicates\tFound 89 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr3_sorted.bam\n",
      "INFO\t2023-08-17 10:27:23\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 10:27:37\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 10:27:37\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 10:27:37\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 10:27:37\tMarkDuplicates\tBefore output close freeMemory: 528083872; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:27:37\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 10:27:37\tMarkDuplicates\tAfter output close freeMemory: 528083872; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 10:27:37 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.30 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 10:27:38 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T102259.040619.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:27:40 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753377_chr3_1_fastp.fastq.gz, data/processed/ERR753377_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753377_chr3.sam, results/mapped_reads/ERR753377_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753377_chr3_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR753377_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.69 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 80.37 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.65 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.43 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 45.33 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 134.270 sec; CPU: 127.459 sec\n",
      "[Thu Aug 17 10:31:28 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:31:28 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753377_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr3.bam, results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753377_chr3.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753377_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 10:32:11 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:32:11 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753377_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753377_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "10:32:12.707 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 10:32:12 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753377_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753377_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 10:32:12 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 10:32:12\tMarkDuplicates\tStart of doWork freeMemory: 528785368; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:32:12\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 10:32:12\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam\n",
      "INFO\t2023-08-17 10:32:14\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:98,149,597\n",
      "INFO\t2023-08-17 10:32:14\tMarkDuplicates\tTracking 639 as yet unmatched pairs. 639 records in RAM.\n",
      "INFO\t2023-08-17 10:32:15\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 3:183,827,468\n",
      "INFO\t2023-08-17 10:32:15\tMarkDuplicates\tTracking 142 as yet unmatched pairs. 142 records in RAM.\n",
      "INFO\t2023-08-17 10:32:16\tMarkDuplicates\tRead 2212613 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 10:32:16\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 305126560; totalMemory: 614465536; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:32:16\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 10:32:16\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 10:32:16\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 10:32:16\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 10:32:16\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 604977552; totalMemory: 1151336448; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:32:16\tMarkDuplicates\tMarking 317026 records as duplicates.\n",
      "INFO\t2023-08-17 10:32:16\tMarkDuplicates\tFound 102 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr3_sorted.bam\n",
      "INFO\t2023-08-17 10:32:16\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 10:32:31\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 10:32:31\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 10:32:31\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 10:32:31\tMarkDuplicates\tBefore output close freeMemory: 528082640; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:32:31\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 10:32:31\tMarkDuplicates\tAfter output close freeMemory: 528082640; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 10:32:31 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.31 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 10:32:32 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T102739.034801.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:32:34 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753378_chr3_1_fastp.fastq.gz, data/processed/ERR753378_chr3_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753378_chr3.sam, results/mapped_reads/ERR753378_chr3_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753378_chr3_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR753378_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.67 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=396591118, availableWord=39905300\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 65826158 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 121609790 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 171186446 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 215246238 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 254402686 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 289201022 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 320125822 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 347607806 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 372029854 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 393732190 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 102 iterations.\n",
      "[bwa_index] 79.56 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.62 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.42 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 45.33 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 133.161 sec; CPU: 126.610 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 10:37:30 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:37:30 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753378_chr3_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr3.bam, results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753378_chr3.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753378_chr3_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 10:38:32 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:38:32 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr3_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753378_chr3_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam\n",
      "    wildcards: sample=ERR753378_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "10:38:33.767 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 10:38:33 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753378_chr3_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753378_chr3_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 10:38:33 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 10:38:33\tMarkDuplicates\tStart of doWork freeMemory: 528772560; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:38:33\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 10:38:33\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam\n",
      "INFO\t2023-08-17 10:38:35\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 3:49,250,853\n",
      "INFO\t2023-08-17 10:38:35\tMarkDuplicates\tTracking 439 as yet unmatched pairs. 439 records in RAM.\n",
      "INFO\t2023-08-17 10:38:36\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 3:117,907,861\n",
      "INFO\t2023-08-17 10:38:36\tMarkDuplicates\tTracking 565 as yet unmatched pairs. 565 records in RAM.\n",
      "INFO\t2023-08-17 10:38:38\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 3:175,097,206\n",
      "INFO\t2023-08-17 10:38:38\tMarkDuplicates\tTracking 362 as yet unmatched pairs. 362 records in RAM.\n",
      "INFO\t2023-08-17 10:38:38\tMarkDuplicates\tRead 3435097 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 10:38:38\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 702830656; totalMemory: 1143996416; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:38:38\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 10:38:39\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 10:38:39\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 10:38:39\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 10:38:39\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1134496992; totalMemory: 1680867328; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:38:39\tMarkDuplicates\tMarking 81431 records as duplicates.\n",
      "INFO\t2023-08-17 10:38:39\tMarkDuplicates\tFound 564 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr3_sorted.bam\n",
      "INFO\t2023-08-17 10:38:39\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 10:39:00\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 10:39:00\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 10:39:00\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 10:39:00\tMarkDuplicates\tBefore output close freeMemory: 528071168; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 10:39:00\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 10:39:00\tMarkDuplicates\tAfter output close freeMemory: 528071168; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 10:39:00 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.45 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 10:39:03 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T103233.029392.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:39:04 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR696683_chr3_dedup.bam\n",
      "    output: results/variants/ERR696683_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR696683_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR696683_chr3.vcf\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "index file data/reference/genome.fa.fai not found, generating...\n",
      "[Thu Aug 17 10:40:02 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T103903.346738.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:40:03 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753368_chr3_dedup.bam\n",
      "    output: results/variants/ERR753368_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753368_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753368_chr3.vcf\n",
      "    wildcards: sample=ERR753368_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 10:41:26 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T104002.687586.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:41:27 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753369_chr3_dedup.bam\n",
      "    output: results/variants/ERR753369_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753369_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753369_chr3.vcf\n",
      "    wildcards: sample=ERR753369_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 10:42:49 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T104126.595908.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:42:51 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753370_chr3_dedup.bam\n",
      "    output: results/variants/ERR753370_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753370_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753370_chr3.vcf\n",
      "    wildcards: sample=ERR753370_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 10:45:01 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T104249.828692.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:45:03 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753371_chr3_dedup.bam\n",
      "    output: results/variants/ERR753371_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753371_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753371_chr3.vcf\n",
      "    wildcards: sample=ERR753371_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 10:45:45 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T104501.853628.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:45:46 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753372_chr3_dedup.bam\n",
      "    output: results/variants/ERR753372_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753372_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753372_chr3.vcf\n",
      "    wildcards: sample=ERR753372_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 10:47:09 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T104545.230250.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:47:10 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753373_chr3_dedup.bam\n",
      "    output: results/variants/ERR753373_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753373_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753373_chr3.vcf\n",
      "    wildcards: sample=ERR753373_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 10:48:54 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T104709.647713.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:48:55 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753374_chr3_dedup.bam\n",
      "    output: results/variants/ERR753374_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753374_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753374_chr3.vcf\n",
      "    wildcards: sample=ERR753374_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 10:50:03 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T104854.681624.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:50:04 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753375_chr3_dedup.bam\n",
      "    output: results/variants/ERR753375_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753375_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753375_chr3.vcf\n",
      "    wildcards: sample=ERR753375_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 10:50:53 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T105003.432291.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:50:54 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753376_chr3_dedup.bam\n",
      "    output: results/variants/ERR753376_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753376_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753376_chr3.vcf\n",
      "    wildcards: sample=ERR753376_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 10:51:45 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T105053.509706.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:51:46 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753377_chr3_dedup.bam\n",
      "    output: results/variants/ERR753377_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753377_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753377_chr3.vcf\n",
      "    wildcards: sample=ERR753377_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 10:52:39 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T105145.701631.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:52:41 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753378_chr3_dedup.bam\n",
      "    output: results/variants/ERR753378_chr3.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753378_chr3.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753378_chr3.vcf\n",
      "    wildcards: sample=ERR753378_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 10:53:58 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T105240.104332.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "vep_install_db        1              1              1\n",
      "total                 1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:53:59 2023]\n",
      "rule vep_install_db:\n",
      "    output: tasks/11vep_dependencies.done\n",
      "    jobid: 0\n",
      "    reason: Missing output files: tasks/11vep_dependencies.done\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " - getting list of available cache files\n",
      "\n",
      "WARNING: It looks like you already have the cache for homo_sapiens GRCh38 (v109) installed.\n",
      "\n",
      "Delete the folder /home/juancarlos/.vep/homo_sapiens/109_GRCh38 and re-run INSTALL.pl if you want to re-install\n",
      " - skipping homo_sapiens\n",
      "Looks like you already have the FASTA file for homo_sapiens, skipping\n",
      "\n",
      "All done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Touching output file tasks/11vep_dependencies.done.\n",
      "[Thu Aug 17 10:54:31 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T105358.413301.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:54:33 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR696683_chr3.vcf\n",
      "    output: results/variants/vep/ERR696683_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR696683_chr3.txt\n",
      "    wildcards: sample=ERR696683_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "--2023-08-17 10:54:33--  https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/clinvar.vcf.gz\n",
      "Resolviendo ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.10, 130.14.250.12, 2607:f220:41e:250::10, ...\n",
      "Conectando con ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)[130.14.250.10]:443... conectado.\n",
      "Petición HTTP enviada, esperando respuesta... 200 OK\n",
      "Longitud: 81792225 (78M) [application/x-gzip]\n",
      "Grabando a: “data/ClinVar/clinvar.vcf.gz”\n",
      "\n",
      "     0K .......... .......... .......... .......... ..........  0%  198K 6m42s\n",
      "    50K .......... .......... .......... .......... ..........  0%  398K 5m1s\n",
      "   100K .......... .......... .......... .......... ..........  0% 26,5M 3m22s\n",
      "   150K .......... .......... .......... .......... ..........  0%  402K 3m21s\n",
      "   200K .......... .......... .......... .......... ..........  0% 57,4M 2m41s\n",
      "   250K .......... .......... .......... .......... ..........  0% 90,3M 2m14s\n",
      "   300K .......... .......... .......... .......... ..........  0%  134M 1m55s\n",
      "   350K .......... .......... .......... .......... ..........  0%  404K 2m5s\n",
      "   400K .......... .......... .......... .......... ..........  0% 39,0M 1m51s\n",
      "   450K .......... .......... .......... .......... ..........  0%  114M 1m40s\n",
      "   500K .......... .......... .......... .......... ..........  0%  111M 91s\n",
      "   550K .......... .......... .......... .......... ..........  0%  118M 84s\n",
      "   600K .......... .......... .......... .......... ..........  0%  115M 77s\n",
      "   650K .......... .......... .......... .......... ..........  0%  113M 72s\n",
      "   700K .......... .......... .......... .......... ..........  0%  119M 67s\n",
      "   750K .......... .......... .......... .......... ..........  1%  409K 75s\n",
      "   800K .......... .......... .......... .......... ..........  1% 77,0M 70s\n",
      "   850K .......... .......... .......... .......... ..........  1% 51,0M 66s\n",
      "   900K .......... .......... .......... .......... ..........  1% 75,0M 63s\n",
      "   950K .......... .......... .......... .......... ..........  1% 72,9M 60s\n",
      "  1000K .......... .......... .......... .......... ..........  1% 67,7M 57s\n",
      "  1050K .......... .......... .......... .......... ..........  1% 80,8M 54s\n",
      "  1100K .......... .......... .......... .......... ..........  1% 66,7M 52s\n",
      "  1150K .......... .......... .......... .......... ..........  1%  131M 50s\n",
      "  1200K .......... .......... .......... .......... ..........  1%  113M 48s\n",
      "  1250K .......... .......... .......... .......... ..........  1%  131M 46s\n",
      "  1300K .......... .......... .......... .......... ..........  1%  134M 44s\n",
      "  1350K .......... .......... .......... .......... ..........  1%  135M 43s\n",
      "  1400K .......... .......... .......... .......... ..........  1%  138M 41s\n",
      "  1450K .......... .......... .......... .......... ..........  1%  192M 40s\n",
      "  1500K .......... .......... .......... .......... ..........  1%  428K 44s\n",
      "  1550K .......... .......... .......... .......... ..........  2% 41,2M 43s\n",
      "  1600K .......... .......... .......... .......... ..........  2% 41,7M 42s\n",
      "  1650K .......... .......... .......... .......... ..........  2%  115M 41s\n",
      "  1700K .......... .......... .......... .......... ..........  2%  117M 39s\n",
      "  1750K .......... .......... .......... .......... ..........  2%  116M 38s\n",
      "  1800K .......... .......... .......... .......... ..........  2%  113M 37s\n",
      "  1850K .......... .......... .......... .......... ..........  2%  116M 36s\n",
      "  1900K .......... .......... .......... .......... ..........  2%  118M 35s\n",
      "  1950K .......... .......... .......... .......... ..........  2%  114M 34s\n",
      "  2000K .......... .......... .......... .......... ..........  2% 92,5M 34s\n",
      "  2050K .......... .......... .......... .......... ..........  2%  117M 33s\n",
      "  2100K .......... .......... .......... .......... ..........  2%  115M 32s\n",
      "  2150K .......... .......... .......... .......... ..........  2%  116M 31s\n",
      "  2200K .......... .......... .......... .......... ..........  2% 96,3M 31s\n",
      "  2250K .......... .......... .......... .......... ..........  2% 53,1M 30s\n",
      "  2300K .......... .......... .......... .......... ..........  2%  192M 29s\n",
      "  2350K .......... .......... .......... .......... ..........  3% 42,7M 29s\n",
      "  2400K .......... .......... .......... .......... ..........  3% 23,4M 28s\n",
      "  2450K .......... .......... .......... .......... ..........  3% 86,0M 28s\n",
      "  2500K .......... .......... .......... .......... ..........  3%  464M 27s\n",
      "  2550K .......... .......... .......... .......... ..........  3%  467M 27s\n",
      "  2600K .......... .......... .......... .......... ..........  3%  401M 26s\n",
      "  2650K .......... .......... .......... .......... ..........  3%  470M 26s\n",
      "  2700K .......... .......... .......... .......... ..........  3%  474M 25s\n",
      "  2750K .......... .......... .......... .......... ..........  3%  483M 25s\n",
      "  2800K .......... .......... .......... .......... ..........  3%  423M 24s\n",
      "  2850K .......... .......... .......... .......... ..........  3%  551M 24s\n",
      "  2900K .......... .......... .......... .......... ..........  3%  211M 23s\n",
      "  2950K .......... .......... .......... .......... ..........  3%  117M 23s\n",
      "  3000K .......... .......... .......... .......... ..........  3%  113M 23s\n",
      "  3050K .......... .......... .......... .......... ..........  3%  449K 25s\n",
      "  3100K .......... .......... .......... .......... ..........  3% 36,7M 25s\n",
      "  3150K .......... .......... .......... .......... ..........  4% 70,4M 24s\n",
      "  3200K .......... .......... .......... .......... ..........  4% 50,2M 24s\n",
      "  3250K .......... .......... .......... .......... ..........  4% 48,4M 23s\n",
      "  3300K .......... .......... .......... .......... ..........  4% 42,6M 23s\n",
      "  3350K .......... .......... .......... .......... ..........  4% 74,9M 23s\n",
      "  3400K .......... .......... .......... .......... ..........  4% 65,9M 22s\n",
      "  3450K .......... .......... .......... .......... ..........  4%  115M 22s\n",
      "  3500K .......... .......... .......... .......... ..........  4%  132M 22s\n",
      "  3550K .......... .......... .......... .......... ..........  4%  133M 21s\n",
      "  3600K .......... .......... .......... .......... ..........  4%  107M 21s\n",
      "  3650K .......... .......... .......... .......... ..........  4%  131M 21s\n",
      "  3700K .......... .......... .......... .......... ..........  4%  131M 21s\n",
      "  3750K .......... .......... .......... .......... ..........  4%  133M 20s\n",
      "  3800K .......... .......... .......... .......... ..........  4%  113M 20s\n",
      "  3850K .......... .......... .......... .......... ..........  4%  130M 20s\n",
      "  3900K .......... .......... .......... .......... ..........  4%  125M 20s\n",
      "  3950K .......... .......... .......... .......... ..........  5%  134M 19s\n",
      "  4000K .......... .......... .......... .......... ..........  5%  113M 19s\n",
      "  4050K .......... .......... .......... .......... ..........  5%  137M 19s\n",
      "  4100K .......... .......... .......... .......... ..........  5%  145M 19s\n",
      "  4150K .......... .......... .......... .......... ..........  5%  144M 18s\n",
      "  4200K .......... .......... .......... .......... ..........  5% 62,5M 18s\n",
      "  4250K .......... .......... .......... .......... ..........  5%  145M 18s\n",
      "  4300K .......... .......... .......... .......... ..........  5%  171M 18s\n",
      "  4350K .......... .......... .......... .......... ..........  5%  166M 17s\n",
      "  4400K .......... .......... .......... .......... ..........  5%  140M 17s\n",
      "  4450K .......... .......... .......... .......... ..........  5%  155M 17s\n",
      "  4500K .......... .......... .......... .......... ..........  5%  184M 17s\n",
      "  4550K .......... .......... .......... .......... ..........  5%  183M 17s\n",
      "  4600K .......... .......... .......... .......... ..........  5%  165M 17s\n",
      "  4650K .......... .......... .......... .......... ..........  5%  184M 16s\n",
      "  4700K .......... .......... .......... .......... ..........  5%  167M 16s\n",
      "  4750K .......... .......... .......... .......... ..........  6%  182M 16s\n",
      "  4800K .......... .......... .......... .......... ..........  6% 84,6M 16s\n",
      "  4850K .......... .......... .......... .......... ..........  6%  109M 16s\n",
      "  4900K .......... .......... .......... .......... ..........  6%  149M 15s\n",
      "  4950K .......... .......... .......... .......... ..........  6%  117M 15s\n",
      "  5000K .......... .......... .......... .......... ..........  6%  109M 15s\n",
      "  5050K .......... .......... .......... .......... ..........  6%  116M 15s\n",
      "  5100K .......... .......... .......... .......... ..........  6%  118M 15s\n",
      "  5150K .......... .......... .......... .......... ..........  6%  117M 15s\n",
      "  5200K .......... .......... .......... .......... ..........  6% 90,5M 15s\n",
      "  5250K .......... .......... .......... .......... ..........  6%  118M 14s\n",
      "  5300K .......... .......... .......... .......... ..........  6%  117M 14s\n",
      "  5350K .......... .......... .......... .......... ..........  6%  118M 14s\n",
      "  5400K .......... .......... .......... .......... ..........  6%  112M 14s\n",
      "  5450K .......... .......... .......... .......... ..........  6%  117M 14s\n",
      "  5500K .......... .......... .......... .......... ..........  6%  118M 14s\n",
      "  5550K .......... .......... .......... .......... ..........  7%  117M 14s\n",
      "  5600K .......... .......... .......... .......... ..........  7% 88,3M 14s\n",
      "  5650K .......... .......... .......... .......... ..........  7%  118M 13s\n",
      "  5700K .......... .......... .......... .......... ..........  7%  117M 13s\n",
      "  5750K .......... .......... .......... .......... ..........  7%  117M 13s\n",
      "  5800K .......... .......... .......... .......... ..........  7%  111M 13s\n",
      "  5850K .......... .......... .......... .......... ..........  7%  117M 13s\n",
      "  5900K .......... .......... .......... .......... ..........  7%  118M 13s\n",
      "  5950K .......... .......... .......... .......... ..........  7%  117M 13s\n",
      "  6000K .......... .......... .......... .......... ..........  7% 87,9M 13s\n",
      "  6050K .......... .......... .......... .......... ..........  7%  119M 13s\n",
      "  6100K .......... .......... .......... .......... ..........  7%  118M 12s\n",
      "  6150K .......... .......... .......... .......... ..........  7%  503K 13s\n",
      "  6200K .......... .......... .......... .......... ..........  7% 41,7M 13s\n",
      "  6250K .......... .......... .......... .......... ..........  7%  102M 13s\n",
      "  6300K .......... .......... .......... .......... ..........  7% 74,6M 13s\n",
      "  6350K .......... .......... .......... .......... ..........  8% 63,4M 13s\n",
      "  6400K .......... .......... .......... .......... ..........  8% 90,1M 13s\n",
      "  6450K .......... .......... .......... .......... ..........  8% 45,4M 13s\n",
      "  6500K .......... .......... .......... .......... ..........  8%  120M 13s\n",
      "  6550K .......... .......... .......... .......... ..........  8%  115M 13s\n",
      "  6600K .......... .......... .......... .......... ..........  8% 92,0M 13s\n",
      "  6650K .......... .......... .......... .......... ..........  8%  113M 12s\n",
      "  6700K .......... .......... .......... .......... ..........  8%  115M 12s\n",
      "  6750K .......... .......... .......... .......... ..........  8%  117M 12s\n",
      "  6800K .......... .......... .......... .......... ..........  8% 91,9M 12s\n",
      "  6850K .......... .......... .......... .......... ..........  8%  118M 12s\n",
      "  6900K .......... .......... .......... .......... ..........  8%  117M 12s\n",
      "  6950K .......... .......... .......... .......... ..........  8%  117M 12s\n",
      "  7000K .......... .......... .......... .......... ..........  8%  114M 12s\n",
      "  7050K .......... .......... .......... .......... ..........  8%  109M 12s\n",
      "  7100K .......... .......... .......... .......... ..........  8%  110M 12s\n",
      "  7150K .......... .......... .......... .......... ..........  9%  104M 12s\n",
      "  7200K .......... .......... .......... .......... ..........  9%  103M 11s\n",
      "  7250K .......... .......... .......... .......... ..........  9% 99,4M 11s\n",
      "  7300K .......... .......... .......... .......... ..........  9%  142M 11s\n",
      "  7350K .......... .......... .......... .......... ..........  9% 86,6M 11s\n",
      "  7400K .......... .......... .......... .......... ..........  9%  167M 11s\n",
      "  7450K .......... .......... .......... .......... ..........  9%  118M 11s\n",
      "  7500K .......... .......... .......... .......... ..........  9%  118M 11s\n",
      "  7550K .......... .......... .......... .......... ..........  9%  118M 11s\n",
      "  7600K .......... .......... .......... .......... ..........  9% 89,0M 11s\n",
      "  7650K .......... .......... .......... .......... ..........  9%  118M 11s\n",
      "  7700K .......... .......... .......... .......... ..........  9%  118M 11s\n",
      "  7750K .......... .......... .......... .......... ..........  9%  117M 11s\n",
      "  7800K .......... .......... .......... .......... ..........  9%  118M 11s\n",
      "  7850K .......... .......... .......... .......... ..........  9%  115M 10s\n",
      "  7900K .......... .......... .......... .......... ..........  9%  117M 10s\n",
      "  7950K .......... .......... .......... .......... .......... 10%  118M 10s\n",
      "  8000K .......... .......... .......... .......... .......... 10% 89,1M 10s\n",
      "  8050K .......... .......... .......... .......... .......... 10%  115M 10s\n",
      "  8100K .......... .......... .......... .......... .......... 10%  117M 10s\n",
      "  8150K .......... .......... .......... .......... .......... 10%  117M 10s\n",
      "  8200K .......... .......... .......... .......... .......... 10%  116M 10s\n",
      "  8250K .......... .......... .......... .......... .......... 10%  117M 10s\n",
      "  8300K .......... .......... .......... .......... .......... 10%  113M 10s\n",
      "  8350K .......... .......... .......... .......... .......... 10%  118M 10s\n",
      "  8400K .......... .......... .......... .......... .......... 10% 89,2M 10s\n",
      "  8450K .......... .......... .......... .......... .......... 10%  113M 10s\n",
      "  8500K .......... .......... .......... .......... .......... 10%  117M 10s\n",
      "  8550K .......... .......... .......... .......... .......... 10%  118M 10s\n",
      "  8600K .......... .......... .......... .......... .......... 10%  114M 10s\n",
      "  8650K .......... .......... .......... .......... .......... 10%  117M 9s\n",
      "  8700K .......... .......... .......... .......... .......... 10%  114M 9s\n",
      "  8750K .......... .......... .......... .......... .......... 11%  117M 9s\n",
      "  8800K .......... .......... .......... .......... .......... 11% 90,9M 9s\n",
      "  8850K .......... .......... .......... .......... .......... 11%  114M 9s\n",
      "  8900K .......... .......... .......... .......... .......... 11%  117M 9s\n",
      "  8950K .......... .......... .......... .......... .......... 11%  114M 9s\n",
      "  9000K .......... .......... .......... .......... .......... 11%  120M 9s\n",
      "  9050K .......... .......... .......... .......... .......... 11%  117M 9s\n",
      "  9100K .......... .......... .......... .......... .......... 11%  114M 9s\n",
      "  9150K .......... .......... .......... .......... .......... 11%  117M 9s\n",
      "  9200K .......... .......... .......... .......... .......... 11%  506K 10s\n",
      "  9250K .......... .......... .......... .......... .......... 11% 50,7M 10s\n",
      "  9300K .......... .......... .......... .......... .......... 11% 65,0M 10s\n",
      "  9350K .......... .......... .......... .......... .......... 11% 68,5M 9s\n",
      "  9400K .......... .......... .......... .......... .......... 11%  144M 9s\n",
      "  9450K .......... .......... .......... .......... .......... 11%  158M 9s\n",
      "  9500K .......... .......... .......... .......... .......... 11% 67,9M 9s\n",
      "  9550K .......... .......... .......... .......... .......... 12%  112M 9s\n",
      "  9600K .......... .......... .......... .......... .......... 12% 52,7M 9s\n",
      "  9650K .......... .......... .......... .......... .......... 12%  116M 9s\n",
      "  9700K .......... .......... .......... .......... .......... 12%  117M 9s\n",
      "  9750K .......... .......... .......... .......... .......... 12%  116M 9s\n",
      "  9800K .......... .......... .......... .......... .......... 12%  108M 9s\n",
      "  9850K .......... .......... .......... .......... .......... 12%  119M 9s\n",
      "  9900K .......... .......... .......... .......... .......... 12%  119M 9s\n",
      "  9950K .......... .......... .......... .......... .......... 12%  113M 9s\n",
      " 10000K .......... .......... .......... .......... .......... 12% 94,5M 9s\n",
      " 10050K .......... .......... .......... .......... .......... 12%  117M 9s\n",
      " 10100K .......... .......... .......... .......... .......... 12%  114M 9s\n",
      " 10150K .......... .......... .......... .......... .......... 12%  118M 9s\n",
      " 10200K .......... .......... .......... .......... .......... 12%  112M 9s\n",
      " 10250K .......... .......... .......... .......... .......... 12%  118M 9s\n",
      " 10300K .......... .......... .......... .......... .......... 12%  116M 9s\n",
      " 10350K .......... .......... .......... .......... .......... 13%  112M 8s\n",
      " 10400K .......... .......... .......... .......... .......... 13% 94,2M 8s\n",
      " 10450K .......... .......... .......... .......... .......... 13%  118M 8s\n",
      " 10500K .......... .......... .......... .......... .......... 13%  115M 8s\n",
      " 10550K .......... .......... .......... .......... .......... 13%  117M 8s\n",
      " 10600K .......... .......... .......... .......... .......... 13%  111M 8s\n",
      " 10650K .......... .......... .......... .......... .......... 13%  119M 8s\n",
      " 10700K .......... .......... .......... .......... .......... 13%  117M 8s\n",
      " 10750K .......... .......... .......... .......... .......... 13% 98,3M 8s\n",
      " 10800K .......... .......... .......... .......... .......... 13%  102M 8s\n",
      " 10850K .......... .......... .......... .......... .......... 13%  121M 8s\n",
      " 10900K .......... .......... .......... .......... .......... 13%  114M 8s\n",
      " 10950K .......... .......... .......... .......... .......... 13%  118M 8s\n",
      " 11000K .......... .......... .......... .......... .......... 13%  116M 8s\n",
      " 11050K .......... .......... .......... .......... .......... 13%  117M 8s\n",
      " 11100K .......... .......... .......... .......... .......... 13%  112M 8s\n",
      " 11150K .......... .......... .......... .......... .......... 14%  115M 8s\n",
      " 11200K .......... .......... .......... .......... .......... 14% 91,3M 8s\n",
      " 11250K .......... .......... .......... .......... .......... 14%  119M 8s\n",
      " 11300K .......... .......... .......... .......... .......... 14%  118M 8s\n",
      " 11350K .......... .......... .......... .......... .......... 14%  114M 8s\n",
      " 11400K .......... .......... .......... .......... .......... 14%  115M 8s\n",
      " 11450K .......... .......... .......... .......... .......... 14%  117M 8s\n",
      " 11500K .......... .......... .......... .......... .......... 14%  118M 8s\n",
      " 11550K .......... .......... .......... .......... .......... 14%  119M 8s\n",
      " 11600K .......... .......... .......... .......... .......... 14% 87,3M 8s\n",
      " 11650K .......... .......... .......... .......... .......... 14%  117M 7s\n",
      " 11700K .......... .......... .......... .......... .......... 14%  117M 7s\n",
      " 11750K .......... .......... .......... .......... .......... 14%  114M 7s\n",
      " 11800K .......... .......... .......... .......... .......... 14%  115M 7s\n",
      " 11850K .......... .......... .......... .......... .......... 14%  118M 7s\n",
      " 11900K .......... .......... .......... .......... .......... 14%  117M 7s\n",
      " 11950K .......... .......... .......... .......... .......... 15%  116M 7s\n",
      " 12000K .......... .......... .......... .......... .......... 15% 88,3M 7s\n",
      " 12050K .......... .......... .......... .......... .......... 15%  117M 7s\n",
      " 12100K .......... .......... .......... .......... .......... 15%  117M 7s\n",
      " 12150K .......... .......... .......... .......... .......... 15%  114M 7s\n",
      " 12200K .......... .......... .......... .......... .......... 15% 6,49M 7s\n",
      " 12250K .......... .......... .......... .......... .......... 15%  547K 8s\n",
      " 12300K .......... .......... .......... .......... .......... 15% 32,6M 8s\n",
      " 12350K .......... .......... .......... .......... .......... 15% 62,8M 8s\n",
      " 12400K .......... .......... .......... .......... .......... 15% 87,7M 8s\n",
      " 12450K .......... .......... .......... .......... .......... 15%  117M 7s\n",
      " 12500K .......... .......... .......... .......... .......... 15%  117M 7s\n",
      " 12550K .......... .......... .......... .......... .......... 15%  117M 7s\n",
      " 12600K .......... .......... .......... .......... .......... 15%  112M 7s\n",
      " 12650K .......... .......... .......... .......... .......... 15% 74,0M 7s\n",
      " 12700K .......... .......... .......... .......... .......... 15%  122M 7s\n",
      " 12750K .......... .......... .......... .......... .......... 16%  118M 7s\n",
      " 12800K .......... .......... .......... .......... .......... 16% 87,5M 7s\n",
      " 12850K .......... .......... .......... .......... .......... 16%  118M 7s\n",
      " 12900K .......... .......... .......... .......... .......... 16%  117M 7s\n",
      " 12950K .......... .......... .......... .......... .......... 16%  117M 7s\n",
      " 13000K .......... .......... .......... .......... .......... 16% 80,3M 7s\n",
      " 13050K .......... .......... .......... .......... .......... 16%  118M 7s\n",
      " 13100K .......... .......... .......... .......... .......... 16%  117M 7s\n",
      " 13150K .......... .......... .......... .......... .......... 16%  117M 7s\n",
      " 13200K .......... .......... .......... .......... .......... 16% 87,9M 7s\n",
      " 13250K .......... .......... .......... .......... .......... 16%  117M 7s\n",
      " 13300K .......... .......... .......... .......... .......... 16%  117M 7s\n",
      " 13350K .......... .......... .......... .......... .......... 16%  117M 7s\n",
      " 13400K .......... .......... .......... .......... .......... 16%  113M 7s\n",
      " 13450K .......... .......... .......... .......... .......... 16%  117M 7s\n",
      " 13500K .......... .......... .......... .......... .......... 16%  103M 7s\n",
      " 13550K .......... .......... .......... .......... .......... 17%  114M 7s\n",
      " 13600K .......... .......... .......... .......... .......... 17% 99,1M 7s\n",
      " 13650K .......... .......... .......... .......... .......... 17%  114M 7s\n",
      " 13700K .......... .......... .......... .......... .......... 17%  119M 7s\n",
      " 13750K .......... .......... .......... .......... .......... 17%  120M 7s\n",
      " 13800K .......... .......... .......... .......... .......... 17%  112M 7s\n",
      " 13850K .......... .......... .......... .......... .......... 17%  118M 7s\n",
      " 13900K .......... .......... .......... .......... .......... 17%  117M 7s\n",
      " 13950K .......... .......... .......... .......... .......... 17%  117M 7s\n",
      " 14000K .......... .......... .......... .......... .......... 17% 87,8M 7s\n",
      " 14050K .......... .......... .......... .......... .......... 17%  117M 7s\n",
      " 14100K .......... .......... .......... .......... .......... 17%  117M 7s\n",
      " 14150K .......... .......... .......... .......... .......... 17%  117M 6s\n",
      " 14200K .......... .......... .......... .......... .......... 17%  116M 6s\n",
      " 14250K .......... .......... .......... .......... .......... 17%  114M 6s\n",
      " 14300K .......... .......... .......... .......... .......... 17%  115M 6s\n",
      " 14350K .......... .......... .......... .......... .......... 18%  118M 6s\n",
      " 14400K .......... .......... .......... .......... .......... 18% 88,5M 6s\n",
      " 14450K .......... .......... .......... .......... .......... 18%  117M 6s\n",
      " 14500K .......... .......... .......... .......... .......... 18%  117M 6s\n",
      " 14550K .......... .......... .......... .......... .......... 18%  117M 6s\n",
      " 14600K .......... .......... .......... .......... .......... 18%  103M 6s\n",
      " 14650K .......... .......... .......... .......... .......... 18%  105M 6s\n",
      " 14700K .......... .......... .......... .......... .......... 18%  143M 6s\n",
      " 14750K .......... .......... .......... .......... .......... 18%  114M 6s\n",
      " 14800K .......... .......... .......... .......... .......... 18% 91,5M 6s\n",
      " 14850K .......... .......... .......... .......... .......... 18%  117M 6s\n",
      " 14900K .......... .......... .......... .......... .......... 18%  117M 6s\n",
      " 14950K .......... .......... .......... .......... .......... 18%  117M 6s\n",
      " 15000K .......... .......... .......... .......... .......... 18%  116M 6s\n",
      " 15050K .......... .......... .......... .......... .......... 18%  115M 6s\n",
      " 15100K .......... .......... .......... .......... .......... 18%  117M 6s\n",
      " 15150K .......... .......... .......... .......... .......... 19%  116M 6s\n",
      " 15200K .......... .......... .......... .......... .......... 19% 87,7M 6s\n",
      " 15250K .......... .......... .......... .......... .......... 19% 7,31M 6s\n",
      " 15300K .......... .......... .......... .......... .......... 19%  548K 6s\n",
      " 15350K .......... .......... .......... .......... .......... 19% 17,4M 6s\n",
      " 15400K .......... .......... .......... .......... .......... 19% 52,7M 6s\n",
      " 15450K .......... .......... .......... .......... .......... 19%  121M 6s\n",
      " 15500K .......... .......... .......... .......... .......... 19% 60,0M 6s\n",
      " 15550K .......... .......... .......... .......... .......... 19%  147M 6s\n",
      " 15600K .......... .......... .......... .......... .......... 19%  123M 6s\n",
      " 15650K .......... .......... .......... .......... .......... 19%  150M 6s\n",
      " 15700K .......... .......... .......... .......... .......... 19%  155M 6s\n",
      " 15750K .......... .......... .......... .......... .......... 19%  113M 6s\n",
      " 15800K .......... .......... .......... .......... .......... 19%  107M 6s\n",
      " 15850K .......... .......... .......... .......... .......... 19%  126M 6s\n",
      " 15900K .......... .......... .......... .......... .......... 19%  115M 6s\n",
      " 15950K .......... .......... .......... .......... .......... 20%  119M 6s\n",
      " 16000K .......... .......... .......... .......... .......... 20% 93,7M 6s\n",
      " 16050K .......... .......... .......... .......... .......... 20%  115M 6s\n",
      " 16100K .......... .......... .......... .......... .......... 20%  117M 6s\n",
      " 16150K .......... .......... .......... .......... .......... 20%  118M 6s\n",
      " 16200K .......... .......... .......... .......... .......... 20%  111M 6s\n",
      " 16250K .......... .......... .......... .......... .......... 20%  115M 6s\n",
      " 16300K .......... .......... .......... .......... .......... 20%  116M 6s\n",
      " 16350K .......... .......... .......... .......... .......... 20%  115M 6s\n",
      " 16400K .......... .......... .......... .......... .......... 20% 95,1M 6s\n",
      " 16450K .......... .......... .......... .......... .......... 20%  113M 6s\n",
      " 16500K .......... .......... .......... .......... .......... 20%  113M 6s\n",
      " 16550K .......... .......... .......... .......... .......... 20%  123M 6s\n",
      " 16600K .......... .......... .......... .......... .......... 20%  111M 6s\n",
      " 16650K .......... .......... .......... .......... .......... 20%  105M 6s\n",
      " 16700K .......... .......... .......... .......... .......... 20% 96,9M 6s\n",
      " 16750K .......... .......... .......... .......... .......... 21%  117M 6s\n",
      " 16800K .......... .......... .......... .......... .......... 21% 92,3M 6s\n",
      " 16850K .......... .......... .......... .......... .......... 21%  114M 6s\n",
      " 16900K .......... .......... .......... .......... .......... 21%  118M 6s\n",
      " 16950K .......... .......... .......... .......... .......... 21%  117M 6s\n",
      " 17000K .......... .......... .......... .......... .......... 21%  110M 6s\n",
      " 17050K .......... .......... .......... .......... .......... 21%  117M 6s\n",
      " 17100K .......... .......... .......... .......... .......... 21%  114M 6s\n",
      " 17150K .......... .......... .......... .......... .......... 21% 86,8M 6s\n",
      " 17200K .......... .......... .......... .......... .......... 21% 93,7M 6s\n",
      " 17250K .......... .......... .......... .......... .......... 21%  117M 6s\n",
      " 17300K .......... .......... .......... .......... .......... 21%  114M 6s\n",
      " 17350K .......... .......... .......... .......... .......... 21%  117M 6s\n",
      " 17400K .......... .......... .......... .......... .......... 21%  111M 5s\n",
      " 17450K .......... .......... .......... .......... .......... 21%  120M 5s\n",
      " 17500K .......... .......... .......... .......... .......... 21%  117M 5s\n",
      " 17550K .......... .......... .......... .......... .......... 22%  114M 5s\n",
      " 17600K .......... .......... .......... .......... .......... 22% 91,4M 5s\n",
      " 17650K .......... .......... .......... .......... .......... 22%  117M 5s\n",
      " 17700K .......... .......... .......... .......... .......... 22%  114M 5s\n",
      " 17750K .......... .......... .......... .......... .......... 22%  118M 5s\n",
      " 17800K .......... .......... .......... .......... .......... 22%  113M 5s\n",
      " 17850K .......... .......... .......... .......... .......... 22%  117M 5s\n",
      " 17900K .......... .......... .......... .......... .......... 22%  118M 5s\n",
      " 17950K .......... .......... .......... .......... .......... 22%  114M 5s\n",
      " 18000K .......... .......... .......... .......... .......... 22% 91,4M 5s\n",
      " 18050K .......... .......... .......... .......... .......... 22%  117M 5s\n",
      " 18100K .......... .......... .......... .......... .......... 22%  114M 5s\n",
      " 18150K .......... .......... .......... .......... .......... 22%  115M 5s\n",
      " 18200K .......... .......... .......... .......... .......... 22%  115M 5s\n",
      " 18250K .......... .......... .......... .......... .......... 22%  117M 5s\n",
      " 18300K .......... .......... .......... .......... .......... 22% 8,64M 5s\n",
      " 18350K .......... .......... .......... .......... .......... 23%  545K 5s\n",
      " 18400K .......... .......... .......... .......... .......... 23% 11,7M 5s\n",
      " 18450K .......... .......... .......... .......... .......... 23% 64,8M 5s\n",
      " 18500K .......... .......... .......... .......... .......... 23%  113M 5s\n",
      " 18550K .......... .......... .......... .......... .......... 23% 59,4M 5s\n",
      " 18600K .......... .......... .......... .......... .......... 23%  181M 5s\n",
      " 18650K .......... .......... .......... .......... .......... 23%  243M 5s\n",
      " 18700K .......... .......... .......... .......... .......... 23%  160M 5s\n",
      " 18750K .......... .......... .......... .......... .......... 23%  117M 5s\n",
      " 18800K .......... .......... .......... .......... .......... 23% 89,3M 5s\n",
      " 18850K .......... .......... .......... .......... .......... 23%  117M 5s\n",
      " 18900K .......... .......... .......... .......... .......... 23%  117M 5s\n",
      " 18950K .......... .......... .......... .......... .......... 23%  115M 5s\n",
      " 19000K .......... .......... .......... .......... .......... 23%  113M 5s\n",
      " 19050K .......... .......... .......... .......... .......... 23%  119M 5s\n",
      " 19100K .......... .......... .......... .......... .......... 23%  119M 5s\n",
      " 19150K .......... .......... .......... .......... .......... 24%  117M 5s\n",
      " 19200K .......... .......... .......... .......... .......... 24% 89,0M 5s\n",
      " 19250K .......... .......... .......... .......... .......... 24%  117M 5s\n",
      " 19300K .......... .......... .......... .......... .......... 24%  117M 5s\n",
      " 19350K .......... .......... .......... .......... .......... 24%  115M 5s\n",
      " 19400K .......... .......... .......... .......... .......... 24%  115M 5s\n",
      " 19450K .......... .......... .......... .......... .......... 24%  117M 5s\n",
      " 19500K .......... .......... .......... .......... .......... 24%  117M 5s\n",
      " 19550K .......... .......... .......... .......... .......... 24%  117M 5s\n",
      " 19600K .......... .......... .......... .......... .......... 24% 88,1M 5s\n",
      " 19650K .......... .......... .......... .......... .......... 24%  117M 5s\n",
      " 19700K .......... .......... .......... .......... .......... 24%  118M 5s\n",
      " 19750K .......... .......... .......... .......... .......... 24%  115M 5s\n",
      " 19800K .......... .......... .......... .......... .......... 24%  116M 5s\n",
      " 19850K .......... .......... .......... .......... .......... 24%  112M 5s\n",
      " 19900K .......... .......... .......... .......... .......... 24% 95,9M 5s\n",
      " 19950K .......... .......... .......... .......... .......... 25%  134M 5s\n",
      " 20000K .......... .......... .......... .......... .......... 25% 86,8M 5s\n",
      " 20050K .......... .......... .......... .......... .......... 25%  133M 5s\n",
      " 20100K .......... .......... .......... .......... .......... 25%  117M 5s\n",
      " 20150K .......... .......... .......... .......... .......... 25%  118M 5s\n",
      " 20200K .......... .......... .......... .......... .......... 25%  114M 5s\n",
      " 20250K .......... .......... .......... .......... .......... 25%  117M 5s\n",
      " 20300K .......... .......... .......... .......... .......... 25%  118M 5s\n",
      " 20350K .......... .......... .......... .......... .......... 25%  115M 5s\n",
      " 20400K .......... .......... .......... .......... .......... 25% 89,6M 5s\n",
      " 20450K .......... .......... .......... .......... .......... 25%  117M 5s\n",
      " 20500K .......... .......... .......... .......... .......... 25%  119M 5s\n",
      " 20550K .......... .......... .......... .......... .......... 25%  117M 5s\n",
      " 20600K .......... .......... .......... .......... .......... 25%  110M 5s\n",
      " 20650K .......... .......... .......... .......... .......... 25% 95,9M 5s\n",
      " 20700K .......... .......... .......... .......... .......... 25%  133M 5s\n",
      " 20750K .......... .......... .......... .......... .......... 26%  124M 5s\n",
      " 20800K .......... .......... .......... .......... .......... 26% 91,8M 5s\n",
      " 20850K .......... .......... .......... .......... .......... 26%  117M 5s\n",
      " 20900K .......... .......... .......... .......... .......... 26%  118M 5s\n",
      " 20950K .......... .......... .......... .......... .......... 26%  115M 5s\n",
      " 21000K .......... .......... .......... .......... .......... 26%  112M 5s\n",
      " 21050K .......... .......... .......... .......... .......... 26%  115M 5s\n",
      " 21100K .......... .......... .......... .......... .......... 26%  117M 5s\n",
      " 21150K .......... .......... .......... .......... .......... 26%  117M 5s\n",
      " 21200K .......... .......... .......... .......... .......... 26% 90,0M 5s\n",
      " 21250K .......... .......... .......... .......... .......... 26%  118M 5s\n",
      " 21300K .......... .......... .......... .......... .......... 26% 10,6M 5s\n",
      " 21350K .......... .......... .......... .......... .......... 26% 29,4M 5s\n",
      " 21400K .......... .......... .......... .......... .......... 26%  531K 5s\n",
      " 21450K .......... .......... .......... .......... .......... 26% 31,4M 5s\n",
      " 21500K .......... .......... .......... .......... .......... 26% 76,4M 5s\n",
      " 21550K .......... .......... .......... .......... .......... 27%  123M 5s\n",
      " 21600K .......... .......... .......... .......... .......... 27% 52,1M 5s\n",
      " 21650K .......... .......... .......... .......... .......... 27%  131M 5s\n",
      " 21700K .......... .......... .......... .......... .......... 27%  100M 5s\n",
      " 21750K .......... .......... .......... .......... .......... 27%  136M 5s\n",
      " 21800K .......... .......... .......... .......... .......... 27%  112M 5s\n",
      " 21850K .......... .......... .......... .......... .......... 27%  114M 5s\n",
      " 21900K .......... .......... .......... .......... .......... 27%  119M 5s\n",
      " 21950K .......... .......... .......... .......... .......... 27%  118M 5s\n",
      " 22000K .......... .......... .......... .......... .......... 27% 91,1M 5s\n",
      " 22050K .......... .......... .......... .......... .......... 27%  120M 5s\n",
      " 22100K .......... .......... .......... .......... .......... 27%  117M 5s\n",
      " 22150K .......... .......... .......... .......... .......... 27%  111M 5s\n",
      " 22200K .......... .......... .......... .......... .......... 27%  117M 5s\n",
      " 22250K .......... .......... .......... .......... .......... 27%  114M 5s\n",
      " 22300K .......... .......... .......... .......... .......... 27%  118M 5s\n",
      " 22350K .......... .......... .......... .......... .......... 28%  120M 5s\n",
      " 22400K .......... .......... .......... .......... .......... 28% 89,1M 5s\n",
      " 22450K .......... .......... .......... .......... .......... 28% 98,5M 5s\n",
      " 22500K .......... .......... .......... .......... .......... 28%  143M 5s\n",
      " 22550K .......... .......... .......... .......... .......... 28%  118M 5s\n",
      " 22600K .......... .......... .......... .......... .......... 28%  113M 4s\n",
      " 22650K .......... .......... .......... .......... .......... 28%  116M 4s\n",
      " 22700K .......... .......... .......... .......... .......... 28%  116M 4s\n",
      " 22750K .......... .......... .......... .......... .......... 28%  118M 4s\n",
      " 22800K .......... .......... .......... .......... .......... 28% 88,8M 4s\n",
      " 22850K .......... .......... .......... .......... .......... 28%  118M 4s\n",
      " 22900K .......... .......... .......... .......... .......... 28%  114M 4s\n",
      " 22950K .......... .......... .......... .......... .......... 28% 61,0M 4s\n",
      " 23000K .......... .......... .......... .......... .......... 28%  336M 4s\n",
      " 23050K .......... .......... .......... .......... .......... 28%  141M 4s\n",
      " 23100K .......... .......... .......... .......... .......... 28%  124M 4s\n",
      " 23150K .......... .......... .......... .......... .......... 29%  118M 4s\n",
      " 23200K .......... .......... .......... .......... .......... 29% 89,5M 4s\n",
      " 23250K .......... .......... .......... .......... .......... 29%  117M 4s\n",
      " 23300K .......... .......... .......... .......... .......... 29%  118M 4s\n",
      " 23350K .......... .......... .......... .......... .......... 29%  118M 4s\n",
      " 23400K .......... .......... .......... .......... .......... 29%  114M 4s\n",
      " 23450K .......... .......... .......... .......... .......... 29%  118M 4s\n",
      " 23500K .......... .......... .......... .......... .......... 29%  113M 4s\n",
      " 23550K .......... .......... .......... .......... .......... 29%  117M 4s\n",
      " 23600K .......... .......... .......... .......... .......... 29% 91,6M 4s\n",
      " 23650K .......... .......... .......... .......... .......... 29%  114M 4s\n",
      " 23700K .......... .......... .......... .......... .......... 29%  117M 4s\n",
      " 23750K .......... .......... .......... .......... .......... 29%  118M 4s\n",
      " 23800K .......... .......... .......... .......... .......... 29%  114M 4s\n",
      " 23850K .......... .......... .......... .......... .......... 29%  116M 4s\n",
      " 23900K .......... .......... .......... .......... .......... 29%  114M 4s\n",
      " 23950K .......... .......... .......... .......... .......... 30%  119M 4s\n",
      " 24000K .......... .......... .......... .......... .......... 30% 90,0M 4s\n",
      " 24050K .......... .......... .......... .......... .......... 30%  114M 4s\n",
      " 24100K .......... .......... .......... .......... .......... 30%  118M 4s\n",
      " 24150K .......... .......... .......... .......... .......... 30%  117M 4s\n",
      " 24200K .......... .......... .......... .......... .......... 30%  115M 4s\n",
      " 24250K .......... .......... .......... .......... .......... 30%  118M 4s\n",
      " 24300K .......... .......... .......... .......... .......... 30%  114M 4s\n",
      " 24350K .......... .......... .......... .......... .......... 30% 15,6M 4s\n",
      " 24400K .......... .......... .......... .......... .......... 30% 4,29M 4s\n",
      " 24450K .......... .......... .......... .......... .......... 30% 1,10M 4s\n",
      " 24500K .......... .......... .......... .......... .......... 30% 1,17M 4s\n",
      " 24550K .......... .......... .......... .......... .......... 30% 35,2M 4s\n",
      " 24600K .......... .......... .......... .......... .......... 30%  105M 4s\n",
      " 24650K .......... .......... .......... .......... .......... 30%  119M 4s\n",
      " 24700K .......... .......... .......... .......... .......... 30%  114M 4s\n",
      " 24750K .......... .......... .......... .......... .......... 31%  118M 4s\n",
      " 24800K .......... .......... .......... .......... .......... 31% 91,2M 4s\n",
      " 24850K .......... .......... .......... .......... .......... 31%  115M 4s\n",
      " 24900K .......... .......... .......... .......... .......... 31%  117M 4s\n",
      " 24950K .......... .......... .......... .......... .......... 31%  119M 4s\n",
      " 25000K .......... .......... .......... .......... .......... 31%  112M 4s\n",
      " 25050K .......... .......... .......... .......... .......... 31%  117M 4s\n",
      " 25100K .......... .......... .......... .......... .......... 31%  118M 4s\n",
      " 25150K .......... .......... .......... .......... .......... 31%  115M 4s\n",
      " 25200K .......... .......... .......... .......... .......... 31% 90,6M 4s\n",
      " 25250K .......... .......... .......... .......... .......... 31%  117M 4s\n",
      " 25300K .......... .......... .......... .......... .......... 31%  113M 4s\n",
      " 25350K .......... .......... .......... .......... .......... 31%  116M 4s\n",
      " 25400K .......... .......... .......... .......... .......... 31%  113M 4s\n",
      " 25450K .......... .......... .......... .......... .......... 31%  118M 4s\n",
      " 25500K .......... .......... .......... .......... .......... 31%  117M 4s\n",
      " 25550K .......... .......... .......... .......... .......... 32%  113M 4s\n",
      " 25600K .......... .......... .......... .......... .......... 32% 92,2M 4s\n",
      " 25650K .......... .......... .......... .......... .......... 32%  118M 4s\n",
      " 25700K .......... .......... .......... .......... .......... 32%  113M 4s\n",
      " 25750K .......... .......... .......... .......... .......... 32%  117M 4s\n",
      " 25800K .......... .......... .......... .......... .......... 32%  113M 4s\n",
      " 25850K .......... .......... .......... .......... .......... 32%  118M 4s\n",
      " 25900K .......... .......... .......... .......... .......... 32%  118M 4s\n",
      " 25950K .......... .......... .......... .......... .......... 32%  115M 4s\n",
      " 26000K .......... .......... .......... .......... .......... 32% 90,6M 4s\n",
      " 26050K .......... .......... .......... .......... .......... 32%  117M 4s\n",
      " 26100K .......... .......... .......... .......... .......... 32%  115M 4s\n",
      " 26150K .......... .......... .......... .......... .......... 32%  117M 4s\n",
      " 26200K .......... .......... .......... .......... .......... 32% 64,9M 4s\n",
      " 26250K .......... .......... .......... .......... .......... 32%  108M 4s\n",
      " 26300K .......... .......... .......... .......... .......... 32%  160M 4s\n",
      " 26350K .......... .......... .......... .......... .......... 33% 74,3M 4s\n",
      " 26400K .......... .......... .......... .......... .......... 33% 91,6M 4s\n",
      " 26450K .......... .......... .......... .......... .......... 33% 74,0M 4s\n",
      " 26500K .......... .......... .......... .......... .......... 33%  110M 4s\n",
      " 26550K .......... .......... .......... .......... .......... 33% 36,3M 4s\n",
      " 26600K .......... .......... .......... .......... .......... 33%  130M 4s\n",
      " 26650K .......... .......... .......... .......... .......... 33%  155M 4s\n",
      " 26700K .......... .......... .......... .......... .......... 33%  154M 4s\n",
      " 26750K .......... .......... .......... .......... .......... 33%  124M 4s\n",
      " 26800K .......... .......... .......... .......... .......... 33%  102M 4s\n",
      " 26850K .......... .......... .......... .......... .......... 33% 97,2M 4s\n",
      " 26900K .......... .......... .......... .......... .......... 33%  152M 4s\n",
      " 26950K .......... .......... .......... .......... .......... 33%  144M 4s\n",
      " 27000K .......... .......... .......... .......... .......... 33%  155M 4s\n",
      " 27050K .......... .......... .......... .......... .......... 33%  159M 4s\n",
      " 27100K .......... .......... .......... .......... .......... 33%  161M 4s\n",
      " 27150K .......... .......... .......... .......... .......... 34%  183M 4s\n",
      " 27200K .......... .......... .......... .......... .......... 34%  139M 4s\n",
      " 27250K .......... .......... .......... .......... .......... 34%  163M 4s\n",
      " 27300K .......... .......... .......... .......... .......... 34%  137M 4s\n",
      " 27350K .......... .......... .......... .......... .......... 34%  144M 4s\n",
      " 27400K .......... .......... .......... .......... .......... 34% 14,2M 4s\n",
      " 27450K .......... .......... .......... .......... .......... 34% 5,00M 4s\n",
      " 27500K .......... .......... .......... .......... .......... 34% 1,09M 4s\n",
      " 27550K .......... .......... .......... .......... .......... 34% 1,15M 4s\n",
      " 27600K .......... .......... .......... .......... .......... 34% 34,3M 4s\n",
      " 27650K .......... .......... .......... .......... .......... 34% 56,7M 4s\n",
      " 27700K .......... .......... .......... .......... .......... 34%  101M 4s\n",
      " 27750K .......... .......... .......... .......... .......... 34%  141M 4s\n",
      " 27800K .......... .......... .......... .......... .......... 34%  108M 4s\n",
      " 27850K .......... .......... .......... .......... .......... 34%  122M 4s\n",
      " 27900K .......... .......... .......... .......... .......... 34%  116M 4s\n",
      " 27950K .......... .......... .......... .......... .......... 35%  117M 4s\n",
      " 28000K .......... .......... .......... .......... .......... 35% 92,2M 4s\n",
      " 28050K .......... .......... .......... .......... .......... 35% 86,8M 4s\n",
      " 28100K .......... .......... .......... .......... .......... 35%  120M 4s\n",
      " 28150K .......... .......... .......... .......... .......... 35%  114M 4s\n",
      " 28200K .......... .......... .......... .......... .......... 35%  110M 4s\n",
      " 28250K .......... .......... .......... .......... .......... 35%  117M 4s\n",
      " 28300K .......... .......... .......... .......... .......... 35%  118M 4s\n",
      " 28350K .......... .......... .......... .......... .......... 35%  121M 4s\n",
      " 28400K .......... .......... .......... .......... .......... 35% 90,8M 4s\n",
      " 28450K .......... .......... .......... .......... .......... 35%  116M 4s\n",
      " 28500K .......... .......... .......... .......... .......... 35%  118M 4s\n",
      " 28550K .......... .......... .......... .......... .......... 35%  117M 4s\n",
      " 28600K .......... .......... .......... .......... .......... 35% 94,6M 4s\n",
      " 28650K .......... .......... .......... .......... .......... 35%  143M 4s\n",
      " 28700K .......... .......... .......... .......... .......... 35%  117M 4s\n",
      " 28750K .......... .......... .......... .......... .......... 36%  118M 4s\n",
      " 28800K .......... .......... .......... .......... .......... 36% 90,6M 4s\n",
      " 28850K .......... .......... .......... .......... .......... 36%  117M 4s\n",
      " 28900K .......... .......... .......... .......... .......... 36%  117M 4s\n",
      " 28950K .......... .......... .......... .......... .......... 36% 97,6M 4s\n",
      " 29000K .......... .......... .......... .......... .......... 36%  109M 4s\n",
      " 29050K .......... .......... .......... .......... .......... 36% 64,0M 4s\n",
      " 29100K .......... .......... .......... .......... .......... 36%  197M 4s\n",
      " 29150K .......... .......... .......... .......... .......... 36%  209M 4s\n",
      " 29200K .......... .......... .......... .......... .......... 36%  107M 4s\n",
      " 29250K .......... .......... .......... .......... .......... 36%  109M 4s\n",
      " 29300K .......... .......... .......... .......... .......... 36%  118M 4s\n",
      " 29350K .......... .......... .......... .......... .......... 36%  120M 4s\n",
      " 29400K .......... .......... .......... .......... .......... 36%  117M 3s\n",
      " 29450K .......... .......... .......... .......... .......... 36%  102M 3s\n",
      " 29500K .......... .......... .......... .......... .......... 36%  117M 3s\n",
      " 29550K .......... .......... .......... .......... .......... 37% 65,9M 3s\n",
      " 29600K .......... .......... .......... .......... .......... 37% 88,1M 3s\n",
      " 29650K .......... .......... .......... .......... .......... 37%  117M 3s\n",
      " 29700K .......... .......... .......... .......... .......... 37%  117M 3s\n",
      " 29750K .......... .......... .......... .......... .......... 37%  118M 3s\n",
      " 29800K .......... .......... .......... .......... .......... 37%  114M 3s\n",
      " 29850K .......... .......... .......... .......... .......... 37% 79,9M 3s\n",
      " 29900K .......... .......... .......... .......... .......... 37%  117M 3s\n",
      " 29950K .......... .......... .......... .......... .......... 37%  117M 3s\n",
      " 30000K .......... .......... .......... .......... .......... 37% 88,0M 3s\n",
      " 30050K .......... .......... .......... .......... .......... 37%  117M 3s\n",
      " 30100K .......... .......... .......... .......... .......... 37%  118M 3s\n",
      " 30150K .......... .......... .......... .......... .......... 37%  117M 3s\n",
      " 30200K .......... .......... .......... .......... .......... 37%  115M 3s\n",
      " 30250K .......... .......... .......... .......... .......... 37%  113M 3s\n",
      " 30300K .......... .......... .......... .......... .......... 37%  118M 3s\n",
      " 30350K .......... .......... .......... .......... .......... 38%  117M 3s\n",
      " 30400K .......... .......... .......... .......... .......... 38% 78,1M 3s\n",
      " 30450K .......... .......... .......... .......... .......... 38% 4,34M 3s\n",
      " 30500K .......... .......... .......... .......... .......... 38% 34,0M 3s\n",
      " 30550K .......... .......... .......... .......... .......... 38%  585K 3s\n",
      " 30600K .......... .......... .......... .......... .......... 38% 18,8M 3s\n",
      " 30650K .......... .......... .......... .......... .......... 38% 51,1M 3s\n",
      " 30700K .......... .......... .......... .......... .......... 38%  118M 3s\n",
      " 30750K .......... .......... .......... .......... .......... 38%  118M 3s\n",
      " 30800K .......... .......... .......... .......... .......... 38% 83,1M 3s\n",
      " 30850K .......... .......... .......... .......... .......... 38%  134M 3s\n",
      " 30900K .......... .......... .......... .......... .......... 38% 70,5M 3s\n",
      " 30950K .......... .......... .......... .......... .......... 38%  161M 3s\n",
      " 31000K .......... .......... .......... .......... .......... 38%  156M 3s\n",
      " 31050K .......... .......... .......... .......... .......... 38%  128M 3s\n",
      " 31100K .......... .......... .......... .......... .......... 38%  116M 3s\n",
      " 31150K .......... .......... .......... .......... .......... 39%  115M 3s\n",
      " 31200K .......... .......... .......... .......... .......... 39% 91,9M 3s\n",
      " 31250K .......... .......... .......... .......... .......... 39%  101M 3s\n",
      " 31300K .......... .......... .......... .......... .......... 39%  127M 3s\n",
      " 31350K .......... .......... .......... .......... .......... 39%  122M 3s\n",
      " 31400K .......... .......... .......... .......... .......... 39%  104M 3s\n",
      " 31450K .......... .......... .......... .......... .......... 39%  111M 3s\n",
      " 31500K .......... .......... .......... .......... .......... 39%  128M 3s\n",
      " 31550K .......... .......... .......... .......... .......... 39%  116M 3s\n",
      " 31600K .......... .......... .......... .......... .......... 39% 94,8M 3s\n",
      " 31650K .......... .......... .......... .......... .......... 39%  113M 3s\n",
      " 31700K .......... .......... .......... .......... .......... 39%  106M 3s\n",
      " 31750K .......... .......... .......... .......... .......... 39%  133M 3s\n",
      " 31800K .......... .......... .......... .......... .......... 39% 91,8M 3s\n",
      " 31850K .......... .......... .......... .......... .......... 39%  117M 3s\n",
      " 31900K .......... .......... .......... .......... .......... 39%  116M 3s\n",
      " 31950K .......... .......... .......... .......... .......... 40%  117M 3s\n",
      " 32000K .......... .......... .......... .......... .......... 40% 94,3M 3s\n",
      " 32050K .......... .......... .......... .......... .......... 40%  114M 3s\n",
      " 32100K .......... .......... .......... .......... .......... 40%  118M 3s\n",
      " 32150K .......... .......... .......... .......... .......... 40%  116M 3s\n",
      " 32200K .......... .......... .......... .......... .......... 40% 43,6M 3s\n",
      " 32250K .......... .......... .......... .......... .......... 40%  118M 3s\n",
      " 32300K .......... .......... .......... .......... .......... 40%  114M 3s\n",
      " 32350K .......... .......... .......... .......... .......... 40%  116M 3s\n",
      " 32400K .......... .......... .......... .......... .......... 40% 93,6M 3s\n",
      " 32450K .......... .......... .......... .......... .......... 40%  106M 3s\n",
      " 32500K .......... .......... .......... .......... .......... 40%  119M 3s\n",
      " 32550K .......... .......... .......... .......... .......... 40% 85,3M 3s\n",
      " 32600K .......... .......... .......... .......... .......... 40% 89,3M 3s\n",
      " 32650K .......... .......... .......... .......... .......... 40%  103M 3s\n",
      " 32700K .......... .......... .......... .......... .......... 41% 38,5M 3s\n",
      " 32750K .......... .......... .......... .......... .......... 41%  116M 3s\n",
      " 32800K .......... .......... .......... .......... .......... 41% 94,1M 3s\n",
      " 32850K .......... .......... .......... .......... .......... 41%  101M 3s\n",
      " 32900K .......... .......... .......... .......... .......... 41% 98,3M 3s\n",
      " 32950K .......... .......... .......... .......... .......... 41%  112M 3s\n",
      " 33000K .......... .......... .......... .......... .......... 41%  105M 3s\n",
      " 33050K .......... .......... .......... .......... .......... 41%  103M 3s\n",
      " 33100K .......... .......... .......... .......... .......... 41%  119M 3s\n",
      " 33150K .......... .......... .......... .......... .......... 41%  117M 3s\n",
      " 33200K .......... .......... .......... .......... .......... 41%  110M 3s\n",
      " 33250K .......... .......... .......... .......... .......... 41%  155M 3s\n",
      " 33300K .......... .......... .......... .......... .......... 41%  125M 3s\n",
      " 33350K .......... .......... .......... .......... .......... 41%  115M 3s\n",
      " 33400K .......... .......... .......... .......... .......... 41% 96,4M 3s\n",
      " 33450K .......... .......... .......... .......... .......... 41%  115M 3s\n",
      " 33500K .......... .......... .......... .......... .......... 42% 5,31M 3s\n",
      " 33550K .......... .......... .......... .......... .......... 42%  102M 3s\n",
      " 33600K .......... .......... .......... .......... .......... 42%  579K 3s\n",
      " 33650K .......... .......... .......... .......... .......... 42% 21,2M 3s\n",
      " 33700K .......... .......... .......... .......... .......... 42% 28,5M 3s\n",
      " 33750K .......... .......... .......... .......... .......... 42% 94,3M 3s\n",
      " 33800K .......... .......... .......... .......... .......... 42%  113M 3s\n",
      " 33850K .......... .......... .......... .......... .......... 42% 69,1M 3s\n",
      " 33900K .......... .......... .......... .......... .......... 42%  117M 3s\n",
      " 33950K .......... .......... .......... .......... .......... 42%  115M 3s\n",
      " 34000K .......... .......... .......... .......... .......... 42% 92,7M 3s\n",
      " 34050K .......... .......... .......... .......... .......... 42%  117M 3s\n",
      " 34100K .......... .......... .......... .......... .......... 42% 95,2M 3s\n",
      " 34150K .......... .......... .......... .......... .......... 42% 85,3M 3s\n",
      " 34200K .......... .......... .......... .......... .......... 42%  111M 3s\n",
      " 34250K .......... .......... .......... .......... .......... 42%  109M 3s\n",
      " 34300K .......... .......... .......... .......... .......... 43%  117M 3s\n",
      " 34350K .......... .......... .......... .......... .......... 43%  116M 3s\n",
      " 34400K .......... .......... .......... .......... .......... 43% 91,4M 3s\n",
      " 34450K .......... .......... .......... .......... .......... 43%  122M 3s\n",
      " 34500K .......... .......... .......... .......... .......... 43%  117M 3s\n",
      " 34550K .......... .......... .......... .......... .......... 43%  114M 3s\n",
      " 34600K .......... .......... .......... .......... .......... 43%  103M 3s\n",
      " 34650K .......... .......... .......... .......... .......... 43%  122M 3s\n",
      " 34700K .......... .......... .......... .......... .......... 43%  117M 3s\n",
      " 34750K .......... .......... .......... .......... .......... 43% 53,7M 3s\n",
      " 34800K .......... .......... .......... .......... .......... 43%  110M 3s\n",
      " 34850K .......... .......... .......... .......... .......... 43%  124M 3s\n",
      " 34900K .......... .......... .......... .......... .......... 43%  135M 3s\n",
      " 34950K .......... .......... .......... .......... .......... 43%  144M 3s\n",
      " 35000K .......... .......... .......... .......... .......... 43%  125M 3s\n",
      " 35050K .......... .......... .......... .......... .......... 43%  130M 3s\n",
      " 35100K .......... .......... .......... .......... .......... 44%  134M 3s\n",
      " 35150K .......... .......... .......... .......... .......... 44%  132M 3s\n",
      " 35200K .......... .......... .......... .......... .......... 44%  101M 3s\n",
      " 35250K .......... .......... .......... .......... .......... 44%  116M 3s\n",
      " 35300K .......... .......... .......... .......... .......... 44%  118M 3s\n",
      " 35350K .......... .......... .......... .......... .......... 44%  113M 3s\n",
      " 35400K .......... .......... .......... .......... .......... 44% 84,3M 3s\n",
      " 35450K .......... .......... .......... .......... .......... 44% 90,8M 3s\n",
      " 35500K .......... .......... .......... .......... .......... 44%  128M 3s\n",
      " 35550K .......... .......... .......... .......... .......... 44%  117M 3s\n",
      " 35600K .......... .......... .......... .......... .......... 44% 91,4M 3s\n",
      " 35650K .......... .......... .......... .......... .......... 44%  117M 3s\n",
      " 35700K .......... .......... .......... .......... .......... 44% 70,2M 3s\n",
      " 35750K .......... .......... .......... .......... .......... 44%  117M 3s\n",
      " 35800K .......... .......... .......... .......... .......... 44% 35,3M 3s\n",
      " 35850K .......... .......... .......... .......... .......... 44%  118M 3s\n",
      " 35900K .......... .......... .......... .......... .......... 45%  117M 3s\n",
      " 35950K .......... .......... .......... .......... .......... 45%  117M 3s\n",
      " 36000K .......... .......... .......... .......... .......... 45% 89,5M 3s\n",
      " 36050K .......... .......... .......... .......... .......... 45%  119M 3s\n",
      " 36100K .......... .......... .......... .......... .......... 45%  116M 3s\n",
      " 36150K .......... .......... .......... .......... .......... 45%  117M 3s\n",
      " 36200K .......... .......... .......... .......... .......... 45%  111M 3s\n",
      " 36250K .......... .......... .......... .......... .......... 45% 90,0M 3s\n",
      " 36300K .......... .......... .......... .......... .......... 45%  119M 3s\n",
      " 36350K .......... .......... .......... .......... .......... 45%  114M 3s\n",
      " 36400K .......... .......... .......... .......... .......... 45% 91,7M 3s\n",
      " 36450K .......... .......... .......... .......... .......... 45%  117M 3s\n",
      " 36500K .......... .......... .......... .......... .......... 45% 5,53M 3s\n",
      " 36550K .......... .......... .......... .......... .......... 45% 35,4M 3s\n",
      " 36600K .......... .......... .......... .......... .......... 45%  585K 3s\n",
      " 36650K .......... .......... .......... .......... .......... 45% 24,0M 3s\n",
      " 36700K .......... .......... .......... .......... .......... 46%  117M 3s\n",
      " 36750K .......... .......... .......... .......... .......... 46% 25,5M 3s\n",
      " 36800K .......... .......... .......... .......... .......... 46% 27,4M 3s\n",
      " 36850K .......... .......... .......... .......... .......... 46%  139M 3s\n",
      " 36900K .......... .......... .......... .......... .......... 46%  131M 3s\n",
      " 36950K .......... .......... .......... .......... .......... 46%  133M 3s\n",
      " 37000K .......... .......... .......... .......... .......... 46%  100M 3s\n",
      " 37050K .......... .......... .......... .......... .......... 46%  120M 3s\n",
      " 37100K .......... .......... .......... .......... .......... 46%  119M 3s\n",
      " 37150K .......... .......... .......... .......... .......... 46%  114M 3s\n",
      " 37200K .......... .......... .......... .......... .......... 46% 93,1M 3s\n",
      " 37250K .......... .......... .......... .......... .......... 46% 74,6M 3s\n",
      " 37300K .......... .......... .......... .......... .......... 46%  136M 3s\n",
      " 37350K .......... .......... .......... .......... .......... 46% 47,7M 3s\n",
      " 37400K .......... .......... .......... .......... .......... 46% 83,3M 3s\n",
      " 37450K .......... .......... .......... .......... .......... 46% 68,6M 3s\n",
      " 37500K .......... .......... .......... .......... .......... 47%  130M 3s\n",
      " 37550K .......... .......... .......... .......... .......... 47%  127M 3s\n",
      " 37600K .......... .......... .......... .......... .......... 47%  126M 3s\n",
      " 37650K .......... .......... .......... .......... .......... 47% 92,7M 3s\n",
      " 37700K .......... .......... .......... .......... .......... 47%  156M 3s\n",
      " 37750K .......... .......... .......... .......... .......... 47%  104M 3s\n",
      " 37800K .......... .......... .......... .......... .......... 47% 96,5M 3s\n",
      " 37850K .......... .......... .......... .......... .......... 47%  159M 3s\n",
      " 37900K .......... .......... .......... .......... .......... 47%  100M 3s\n",
      " 37950K .......... .......... .......... .......... .......... 47%  106M 3s\n",
      " 38000K .......... .......... .......... .......... .......... 47%  105M 3s\n",
      " 38050K .......... .......... .......... .......... .......... 47%  129M 3s\n",
      " 38100K .......... .......... .......... .......... .......... 47%  120M 3s\n",
      " 38150K .......... .......... .......... .......... .......... 47%  119M 3s\n",
      " 38200K .......... .......... .......... .......... .......... 47%  106M 3s\n",
      " 38250K .......... .......... .......... .......... .......... 47%  115M 3s\n",
      " 38300K .......... .......... .......... .......... .......... 48%  117M 3s\n",
      " 38350K .......... .......... .......... .......... .......... 48%  116M 3s\n",
      " 38400K .......... .......... .......... .......... .......... 48% 90,5M 3s\n",
      " 38450K .......... .......... .......... .......... .......... 48%  122M 3s\n",
      " 38500K .......... .......... .......... .......... .......... 48%  113M 3s\n",
      " 38550K .......... .......... .......... .......... .......... 48%  121M 3s\n",
      " 38600K .......... .......... .......... .......... .......... 48%  106M 3s\n",
      " 38650K .......... .......... .......... .......... .......... 48%  115M 3s\n",
      " 38700K .......... .......... .......... .......... .......... 48%  113M 3s\n",
      " 38750K .......... .......... .......... .......... .......... 48%  122M 3s\n",
      " 38800K .......... .......... .......... .......... .......... 48% 60,9M 3s\n",
      " 38850K .......... .......... .......... .......... .......... 48% 49,2M 3s\n",
      " 38900K .......... .......... .......... .......... .......... 48%  124M 3s\n",
      " 38950K .......... .......... .......... .......... .......... 48%  119M 3s\n",
      " 39000K .......... .......... .......... .......... .......... 48%  107M 3s\n",
      " 39050K .......... .......... .......... .......... .......... 48%  117M 3s\n",
      " 39100K .......... .......... .......... .......... .......... 49%  135M 3s\n",
      " 39150K .......... .......... .......... .......... .......... 49%  127M 3s\n",
      " 39200K .......... .......... .......... .......... .......... 49%  106M 3s\n",
      " 39250K .......... .......... .......... .......... .......... 49%  118M 3s\n",
      " 39300K .......... .......... .......... .......... .......... 49%  124M 3s\n",
      " 39350K .......... .......... .......... .......... .......... 49%  127M 2s\n",
      " 39400K .......... .......... .......... .......... .......... 49%  114M 2s\n",
      " 39450K .......... .......... .......... .......... .......... 49%  129M 2s\n",
      " 39500K .......... .......... .......... .......... .......... 49% 41,3M 2s\n",
      " 39550K .......... .......... .......... .......... .......... 49% 6,39M 2s\n",
      " 39600K .......... .......... .......... .......... .......... 49% 42,4M 2s\n",
      " 39650K .......... .......... .......... .......... .......... 49%  583K 3s\n",
      " 39700K .......... .......... .......... .......... .......... 49% 35,4M 3s\n",
      " 39750K .......... .......... .......... .......... .......... 49% 52,0M 3s\n",
      " 39800K .......... .......... .......... .......... .......... 49% 24,2M 3s\n",
      " 39850K .......... .......... .......... .......... .......... 49% 25,1M 3s\n",
      " 39900K .......... .......... .......... .......... .......... 50% 61,6M 3s\n",
      " 39950K .......... .......... .......... .......... .......... 50% 66,7M 3s\n",
      " 40000K .......... .......... .......... .......... .......... 50% 74,9M 3s\n",
      " 40050K .......... .......... .......... .......... .......... 50%  100M 3s\n",
      " 40100K .......... .......... .......... .......... .......... 50%  155M 3s\n",
      " 40150K .......... .......... .......... .......... .......... 50%  194M 3s\n",
      " 40200K .......... .......... .......... .......... .......... 50%  112M 2s\n",
      " 40250K .......... .......... .......... .......... .......... 50%  118M 2s\n",
      " 40300K .......... .......... .......... .......... .......... 50%  121M 2s\n",
      " 40350K .......... .......... .......... .......... .......... 50%  114M 2s\n",
      " 40400K .......... .......... .......... .......... .......... 50% 91,2M 2s\n",
      " 40450K .......... .......... .......... .......... .......... 50%  107M 2s\n",
      " 40500K .......... .......... .......... .......... .......... 50% 39,2M 2s\n",
      " 40550K .......... .......... .......... .......... .......... 50%  124M 2s\n",
      " 40600K .......... .......... .......... .......... .......... 50%  114M 2s\n",
      " 40650K .......... .......... .......... .......... .......... 50%  117M 2s\n",
      " 40700K .......... .......... .......... .......... .......... 51%  118M 2s\n",
      " 40750K .......... .......... .......... .......... .......... 51%  113M 2s\n",
      " 40800K .......... .......... .......... .......... .......... 51% 91,1M 2s\n",
      " 40850K .......... .......... .......... .......... .......... 51%  117M 2s\n",
      " 40900K .......... .......... .......... .......... .......... 51%  113M 2s\n",
      " 40950K .......... .......... .......... .......... .......... 51%  111M 2s\n",
      " 41000K .......... .......... .......... .......... .......... 51% 50,7M 2s\n",
      " 41050K .......... .......... .......... .......... .......... 51% 90,2M 2s\n",
      " 41100K .......... .......... .......... .......... .......... 51%  102M 2s\n",
      " 41150K .......... .......... .......... .......... .......... 51%  121M 2s\n",
      " 41200K .......... .......... .......... .......... .......... 51% 94,0M 2s\n",
      " 41250K .......... .......... .......... .......... .......... 51%  160M 2s\n",
      " 41300K .......... .......... .......... .......... .......... 51%  161M 2s\n",
      " 41350K .......... .......... .......... .......... .......... 51%  158M 2s\n",
      " 41400K .......... .......... .......... .......... .......... 51%  140M 2s\n",
      " 41450K .......... .......... .......... .......... .......... 51%  161M 2s\n",
      " 41500K .......... .......... .......... .......... .......... 52%  169M 2s\n",
      " 41550K .......... .......... .......... .......... .......... 52%  118M 2s\n",
      " 41600K .......... .......... .......... .......... .......... 52% 92,6M 2s\n",
      " 41650K .......... .......... .......... .......... .......... 52%  117M 2s\n",
      " 41700K .......... .......... .......... .......... .......... 52%  114M 2s\n",
      " 41750K .......... .......... .......... .......... .......... 52%  118M 2s\n",
      " 41800K .......... .......... .......... .......... .......... 52%  108M 2s\n",
      " 41850K .......... .......... .......... .......... .......... 52%  133M 2s\n",
      " 41900K .......... .......... .......... .......... .......... 52%  117M 2s\n",
      " 41950K .......... .......... .......... .......... .......... 52%  115M 2s\n",
      " 42000K .......... .......... .......... .......... .......... 52% 89,1M 2s\n",
      " 42050K .......... .......... .......... .......... .......... 52%  118M 2s\n",
      " 42100K .......... .......... .......... .......... .......... 52% 73,0M 2s\n",
      " 42150K .......... .......... .......... .......... .......... 52%  116M 2s\n",
      " 42200K .......... .......... .......... .......... .......... 52%  115M 2s\n",
      " 42250K .......... .......... .......... .......... .......... 52%  118M 2s\n",
      " 42300K .......... .......... .......... .......... .......... 53%  118M 2s\n",
      " 42350K .......... .......... .......... .......... .......... 53%  117M 2s\n",
      " 42400K .......... .......... .......... .......... .......... 53% 87,9M 2s\n",
      " 42450K .......... .......... .......... .......... .......... 53%  117M 2s\n",
      " 42500K .......... .......... .......... .......... .......... 53%  117M 2s\n",
      " 42550K .......... .......... .......... .......... .......... 53% 6,43M 2s\n",
      " 42600K .......... .......... .......... .......... .......... 53% 26,5M 2s\n",
      " 42650K .......... .......... .......... .......... .......... 53% 1,17M 2s\n",
      " 42700K .......... .......... .......... .......... .......... 53% 1,12M 2s\n",
      " 42750K .......... .......... .......... .......... .......... 53% 28,1M 2s\n",
      " 42800K .......... .......... .......... .......... .......... 53% 24,4M 2s\n",
      " 42850K .......... .......... .......... .......... .......... 53% 19,9M 2s\n",
      " 42900K .......... .......... .......... .......... .......... 53% 43,3M 2s\n",
      " 42950K .......... .......... .......... .......... .......... 53%  122M 2s\n",
      " 43000K .......... .......... .......... .......... .......... 53% 46,6M 2s\n",
      " 43050K .......... .......... .......... .......... .......... 53%  152M 2s\n",
      " 43100K .......... .......... .......... .......... .......... 54% 81,0M 2s\n",
      " 43150K .......... .......... .......... .......... .......... 54%  117M 2s\n",
      " 43200K .......... .......... .......... .......... .......... 54% 83,4M 2s\n",
      " 43250K .......... .......... .......... .......... .......... 54%  115M 2s\n",
      " 43300K .......... .......... .......... .......... .......... 54%  115M 2s\n",
      " 43350K .......... .......... .......... .......... .......... 54% 54,0M 2s\n",
      " 43400K .......... .......... .......... .......... .......... 54% 76,9M 2s\n",
      " 43450K .......... .......... .......... .......... .......... 54% 73,8M 2s\n",
      " 43500K .......... .......... .......... .......... .......... 54%  107M 2s\n",
      " 43550K .......... .......... .......... .......... .......... 54%  144M 2s\n",
      " 43600K .......... .......... .......... .......... .......... 54%  117M 2s\n",
      " 43650K .......... .......... .......... .......... .......... 54%  158M 2s\n",
      " 43700K .......... .......... .......... .......... .......... 54%  167M 2s\n",
      " 43750K .......... .......... .......... .......... .......... 54%  150M 2s\n",
      " 43800K .......... .......... .......... .......... .......... 54%  148M 2s\n",
      " 43850K .......... .......... .......... .......... .......... 54%  163M 2s\n",
      " 43900K .......... .......... .......... .......... .......... 55%  149M 2s\n",
      " 43950K .......... .......... .......... .......... .......... 55%  164M 2s\n",
      " 44000K .......... .......... .......... .......... .......... 55% 92,4M 2s\n",
      " 44050K .......... .......... .......... .......... .......... 55%  116M 2s\n",
      " 44100K .......... .......... .......... .......... .......... 55% 38,1M 2s\n",
      " 44150K .......... .......... .......... .......... .......... 55%  123M 2s\n",
      " 44200K .......... .......... .......... .......... .......... 55% 55,9M 2s\n",
      " 44250K .......... .......... .......... .......... .......... 55%  116M 2s\n",
      " 44300K .......... .......... .......... .......... .......... 55%  119M 2s\n",
      " 44350K .......... .......... .......... .......... .......... 55%  119M 2s\n",
      " 44400K .......... .......... .......... .......... .......... 55% 89,9M 2s\n",
      " 44450K .......... .......... .......... .......... .......... 55%  117M 2s\n",
      " 44500K .......... .......... .......... .......... .......... 55%  118M 2s\n",
      " 44550K .......... .......... .......... .......... .......... 55%  117M 2s\n",
      " 44600K .......... .......... .......... .......... .......... 55%  109M 2s\n",
      " 44650K .......... .......... .......... .......... .......... 55%  117M 2s\n",
      " 44700K .......... .......... .......... .......... .......... 56%  118M 2s\n",
      " 44750K .......... .......... .......... .......... .......... 56%  120M 2s\n",
      " 44800K .......... .......... .......... .......... .......... 56% 87,9M 2s\n",
      " 44850K .......... .......... .......... .......... .......... 56%  120M 2s\n",
      " 44900K .......... .......... .......... .......... .......... 56% 97,8M 2s\n",
      " 44950K .......... .......... .......... .......... .......... 56%  136M 2s\n",
      " 45000K .......... .......... .......... .......... .......... 56%  111M 2s\n",
      " 45050K .......... .......... .......... .......... .......... 56%  111M 2s\n",
      " 45100K .......... .......... .......... .......... .......... 56%  125M 2s\n",
      " 45150K .......... .......... .......... .......... .......... 56% 97,6M 2s\n",
      " 45200K .......... .......... .......... .......... .......... 56%  106M 2s\n",
      " 45250K .......... .......... .......... .......... .......... 56% 57,1M 2s\n",
      " 45300K .......... .......... .......... .......... .......... 56% 85,5M 2s\n",
      " 45350K .......... .......... .......... .......... .......... 56%  101M 2s\n",
      " 45400K .......... .......... .......... .......... .......... 56% 71,5M 2s\n",
      " 45450K .......... .......... .......... .......... .......... 56%  114M 2s\n",
      " 45500K .......... .......... .......... .......... .......... 57%  124M 2s\n",
      " 45550K .......... .......... .......... .......... .......... 57%  174M 2s\n",
      " 45600K .......... .......... .......... .......... .......... 57% 6,84M 2s\n",
      " 45650K .......... .......... .......... .......... .......... 57% 38,5M 2s\n",
      " 45700K .......... .......... .......... .......... .......... 57% 1,19M 2s\n",
      " 45750K .......... .......... .......... .......... .......... 57% 1,08M 2s\n",
      " 45800K .......... .......... .......... .......... .......... 57%  106M 2s\n",
      " 45850K .......... .......... .......... .......... .......... 57% 29,7M 2s\n",
      " 45900K .......... .......... .......... .......... .......... 57% 20,3M 2s\n",
      " 45950K .......... .......... .......... .......... .......... 57% 32,0M 2s\n",
      " 46000K .......... .......... .......... .......... .......... 57% 43,1M 2s\n",
      " 46050K .......... .......... .......... .......... .......... 57%  119M 2s\n",
      " 46100K .......... .......... .......... .......... .......... 57% 64,2M 2s\n",
      " 46150K .......... .......... .......... .......... .......... 57% 53,1M 2s\n",
      " 46200K .......... .......... .......... .......... .......... 57% 61,1M 2s\n",
      " 46250K .......... .......... .......... .......... .......... 57%  116M 2s\n",
      " 46300K .......... .......... .......... .......... .......... 58%  118M 2s\n",
      " 46350K .......... .......... .......... .......... .......... 58%  117M 2s\n",
      " 46400K .......... .......... .......... .......... .......... 58% 93,2M 2s\n",
      " 46450K .......... .......... .......... .......... .......... 58%  116M 2s\n",
      " 46500K .......... .......... .......... .......... .......... 58%  116M 2s\n",
      " 46550K .......... .......... .......... .......... .......... 58%  117M 2s\n",
      " 46600K .......... .......... .......... .......... .......... 58%  109M 2s\n",
      " 46650K .......... .......... .......... .......... .......... 58%  119M 2s\n",
      " 46700K .......... .......... .......... .......... .......... 58%  101M 2s\n",
      " 46750K .......... .......... .......... .......... .......... 58%  131M 2s\n",
      " 46800K .......... .......... .......... .......... .......... 58% 92,2M 2s\n",
      " 46850K .......... .......... .......... .......... .......... 58%  116M 2s\n",
      " 46900K .......... .......... .......... .......... .......... 58%  119M 2s\n",
      " 46950K .......... .......... .......... .......... .......... 58% 62,4M 2s\n",
      " 47000K .......... .......... .......... .......... .......... 58%  145M 2s\n",
      " 47050K .......... .......... .......... .......... .......... 58%  153M 2s\n",
      " 47100K .......... .......... .......... .......... .......... 59%  181M 2s\n",
      " 47150K .......... .......... .......... .......... .......... 59% 92,3M 2s\n",
      " 47200K .......... .......... .......... .......... .......... 59% 78,0M 2s\n",
      " 47250K .......... .......... .......... .......... .......... 59%  114M 2s\n",
      " 47300K .......... .......... .......... .......... .......... 59% 68,5M 2s\n",
      " 47350K .......... .......... .......... .......... .......... 59% 65,4M 2s\n",
      " 47400K .......... .......... .......... .......... .......... 59%  108M 2s\n",
      " 47450K .......... .......... .......... .......... .......... 59%  123M 2s\n",
      " 47500K .......... .......... .......... .......... .......... 59%  113M 2s\n",
      " 47550K .......... .......... .......... .......... .......... 59%  119M 2s\n",
      " 47600K .......... .......... .......... .......... .......... 59% 94,2M 2s\n",
      " 47650K .......... .......... .......... .......... .......... 59%  114M 2s\n",
      " 47700K .......... .......... .......... .......... .......... 59%  117M 2s\n",
      " 47750K .......... .......... .......... .......... .......... 59%  118M 2s\n",
      " 47800K .......... .......... .......... .......... .......... 59%  111M 2s\n",
      " 47850K .......... .......... .......... .......... .......... 59%  121M 2s\n",
      " 47900K .......... .......... .......... .......... .......... 60%  117M 2s\n",
      " 47950K .......... .......... .......... .......... .......... 60%  115M 2s\n",
      " 48000K .......... .......... .......... .......... .......... 60% 91,0M 2s\n",
      " 48050K .......... .......... .......... .......... .......... 60%  117M 2s\n",
      " 48100K .......... .......... .......... .......... .......... 60%  115M 2s\n",
      " 48150K .......... .......... .......... .......... .......... 60%  116M 2s\n",
      " 48200K .......... .......... .......... .......... .......... 60%  115M 2s\n",
      " 48250K .......... .......... .......... .......... .......... 60%  117M 2s\n",
      " 48300K .......... .......... .......... .......... .......... 60%  117M 2s\n",
      " 48350K .......... .......... .......... .......... .......... 60%  114M 2s\n",
      " 48400K .......... .......... .......... .......... .......... 60% 90,9M 2s\n",
      " 48450K .......... .......... .......... .......... .......... 60%  117M 2s\n",
      " 48500K .......... .......... .......... .......... .......... 60% 74,3M 2s\n",
      " 48550K .......... .......... .......... .......... .......... 60% 8,82M 2s\n",
      " 48600K .......... .......... .......... .......... .......... 60% 51,5M 2s\n",
      " 48650K .......... .......... .......... .......... .......... 60%  118M 2s\n",
      " 48700K .......... .......... .......... .......... .......... 61% 17,1M 2s\n",
      " 48750K .......... .......... .......... .......... .......... 61%  602K 2s\n",
      " 48800K .......... .......... .......... .......... .......... 61% 28,3M 2s\n",
      " 48850K .......... .......... .......... .......... .......... 61% 22,2M 2s\n",
      " 48900K .......... .......... .......... .......... .......... 61% 29,0M 2s\n",
      " 48950K .......... .......... .......... .......... .......... 61%  123M 2s\n",
      " 49000K .......... .......... .......... .......... .......... 61% 28,9M 2s\n",
      " 49050K .......... .......... .......... .......... .......... 61% 21,9M 2s\n",
      " 49100K .......... .......... .......... .......... .......... 61% 53,8M 2s\n",
      " 49150K .......... .......... .......... .......... .......... 61%  131M 2s\n",
      " 49200K .......... .......... .......... .......... .......... 61% 55,2M 2s\n",
      " 49250K .......... .......... .......... .......... .......... 61% 74,8M 2s\n",
      " 49300K .......... .......... .......... .......... .......... 61% 91,0M 2s\n",
      " 49350K .......... .......... .......... .......... .......... 61%  130M 2s\n",
      " 49400K .......... .......... .......... .......... .......... 61%  101M 2s\n",
      " 49450K .......... .......... .......... .......... .......... 61% 99,0M 2s\n",
      " 49500K .......... .......... .......... .......... .......... 62% 73,0M 2s\n",
      " 49550K .......... .......... .......... .......... .......... 62% 84,0M 2s\n",
      " 49600K .......... .......... .......... .......... .......... 62% 71,6M 2s\n",
      " 49650K .......... .......... .......... .......... .......... 62% 85,4M 2s\n",
      " 49700K .......... .......... .......... .......... .......... 62% 82,7M 2s\n",
      " 49750K .......... .......... .......... .......... .......... 62%  148M 2s\n",
      " 49800K .......... .......... .......... .......... .......... 62%  148M 2s\n",
      " 49850K .......... .......... .......... .......... .......... 62%  160M 2s\n",
      " 49900K .......... .......... .......... .......... .......... 62%  142M 2s\n",
      " 49950K .......... .......... .......... .......... .......... 62%  153M 2s\n",
      " 50000K .......... .......... .......... .......... .......... 62%  126M 2s\n",
      " 50050K .......... .......... .......... .......... .......... 62%  168M 2s\n",
      " 50100K .......... .......... .......... .......... .......... 62%  166M 2s\n",
      " 50150K .......... .......... .......... .......... .......... 62%  216M 2s\n",
      " 50200K .......... .......... .......... .......... .......... 62%  115M 2s\n",
      " 50250K .......... .......... .......... .......... .......... 62%  117M 2s\n",
      " 50300K .......... .......... .......... .......... .......... 63%  116M 2s\n",
      " 50350K .......... .......... .......... .......... .......... 63%  119M 2s\n",
      " 50400K .......... .......... .......... .......... .......... 63% 55,9M 2s\n",
      " 50450K .......... .......... .......... .......... .......... 63% 66,8M 2s\n",
      " 50500K .......... .......... .......... .......... .......... 63% 70,2M 2s\n",
      " 50550K .......... .......... .......... .......... .......... 63%  114M 2s\n",
      " 50600K .......... .......... .......... .......... .......... 63%  113M 2s\n",
      " 50650K .......... .......... .......... .......... .......... 63%  117M 2s\n",
      " 50700K .......... .......... .......... .......... .......... 63%  118M 2s\n",
      " 50750K .......... .......... .......... .......... .......... 63%  118M 2s\n",
      " 50800K .......... .......... .......... .......... .......... 63% 88,9M 2s\n",
      " 50850K .......... .......... .......... .......... .......... 63%  117M 2s\n",
      " 50900K .......... .......... .......... .......... .......... 63%  118M 2s\n",
      " 50950K .......... .......... .......... .......... .......... 63%  118M 2s\n",
      " 51000K .......... .......... .......... .......... .......... 63%  110M 2s\n",
      " 51050K .......... .......... .......... .......... .......... 63%  109M 2s\n",
      " 51100K .......... .......... .......... .......... .......... 64%  127M 2s\n",
      " 51150K .......... .......... .......... .......... .......... 64%  117M 2s\n",
      " 51200K .......... .......... .......... .......... .......... 64% 89,0M 2s\n",
      " 51250K .......... .......... .......... .......... .......... 64%  117M 2s\n",
      " 51300K .......... .......... .......... .......... .......... 64%  117M 2s\n",
      " 51350K .......... .......... .......... .......... .......... 64%  117M 2s\n",
      " 51400K .......... .......... .......... .......... .......... 64%  111M 2s\n",
      " 51450K .......... .......... .......... .......... .......... 64%  112M 2s\n",
      " 51500K .......... .......... .......... .......... .......... 64%  102M 2s\n",
      " 51550K .......... .......... .......... .......... .......... 64%  133M 2s\n",
      " 51600K .......... .......... .......... .......... .......... 64% 12,3M 2s\n",
      " 51650K .......... .......... .......... .......... .......... 64% 18,8M 2s\n",
      " 51700K .......... .......... .......... .......... .......... 64% 44,4M 2s\n",
      " 51750K .......... .......... .......... .......... .......... 64% 29,8M 2s\n",
      " 51800K .......... .......... .......... .......... .......... 64% 4,29M 2s\n",
      " 51850K .......... .......... .......... .......... .......... 64%  678K 2s\n",
      " 51900K .......... .......... .......... .......... .......... 65% 31,5M 2s\n",
      " 51950K .......... .......... .......... .......... .......... 65% 17,6M 2s\n",
      " 52000K .......... .......... .......... .......... .......... 65% 41,7M 2s\n",
      " 52050K .......... .......... .......... .......... .......... 65%  123M 2s\n",
      " 52100K .......... .......... .......... .......... .......... 65% 16,2M 2s\n",
      " 52150K .......... .......... .......... .......... .......... 65% 74,8M 2s\n",
      " 52200K .......... .......... .......... .......... .......... 65% 62,4M 2s\n",
      " 52250K .......... .......... .......... .......... .......... 65%  105M 2s\n",
      " 52300K .......... .......... .......... .......... .......... 65%  115M 2s\n",
      " 52350K .......... .......... .......... .......... .......... 65%  152M 2s\n",
      " 52400K .......... .......... .......... .......... .......... 65% 62,2M 2s\n",
      " 52450K .......... .......... .......... .......... .......... 65%  121M 2s\n",
      " 52500K .......... .......... .......... .......... .......... 65%  117M 2s\n",
      " 52550K .......... .......... .......... .......... .......... 65%  114M 2s\n",
      " 52600K .......... .......... .......... .......... .......... 65%  112M 2s\n",
      " 52650K .......... .......... .......... .......... .......... 65%  113M 2s\n",
      " 52700K .......... .......... .......... .......... .......... 66% 64,5M 2s\n",
      " 52750K .......... .......... .......... .......... .......... 66%  117M 2s\n",
      " 52800K .......... .......... .......... .......... .......... 66% 77,7M 2s\n",
      " 52850K .......... .......... .......... .......... .......... 66%  155M 2s\n",
      " 52900K .......... .......... .......... .......... .......... 66%  103M 2s\n",
      " 52950K .......... .......... .......... .......... .......... 66%  116M 2s\n",
      " 53000K .......... .......... .......... .......... .......... 66%  107M 2s\n",
      " 53050K .......... .......... .......... .......... .......... 66%  116M 2s\n",
      " 53100K .......... .......... .......... .......... .......... 66% 68,0M 2s\n",
      " 53150K .......... .......... .......... .......... .......... 66%  153M 2s\n",
      " 53200K .......... .......... .......... .......... .......... 66%  127M 2s\n",
      " 53250K .......... .......... .......... .......... .......... 66% 97,5M 2s\n",
      " 53300K .......... .......... .......... .......... .......... 66%  140M 2s\n",
      " 53350K .......... .......... .......... .......... .......... 66%  141M 2s\n",
      " 53400K .......... .......... .......... .......... .......... 66%  108M 2s\n",
      " 53450K .......... .......... .......... .......... .......... 66%  117M 2s\n",
      " 53500K .......... .......... .......... .......... .......... 67%  113M 2s\n",
      " 53550K .......... .......... .......... .......... .......... 67%  120M 1s\n",
      " 53600K .......... .......... .......... .......... .......... 67% 92,4M 1s\n",
      " 53650K .......... .......... .......... .......... .......... 67%  120M 1s\n",
      " 53700K .......... .......... .......... .......... .......... 67% 98,2M 1s\n",
      " 53750K .......... .......... .......... .......... .......... 67%  112M 1s\n",
      " 53800K .......... .......... .......... .......... .......... 67%  108M 1s\n",
      " 53850K .......... .......... .......... .......... .......... 67%  117M 1s\n",
      " 53900K .......... .......... .......... .......... .......... 67%  115M 1s\n",
      " 53950K .......... .......... .......... .......... .......... 67%  130M 1s\n",
      " 54000K .......... .......... .......... .......... .......... 67% 92,1M 1s\n",
      " 54050K .......... .......... .......... .......... .......... 67%  114M 1s\n",
      " 54100K .......... .......... .......... .......... .......... 67%  117M 1s\n",
      " 54150K .......... .......... .......... .......... .......... 67%  114M 1s\n",
      " 54200K .......... .......... .......... .......... .......... 67%  118M 1s\n",
      " 54250K .......... .......... .......... .......... .......... 67%  117M 1s\n",
      " 54300K .......... .......... .......... .......... .......... 68%  114M 1s\n",
      " 54350K .......... .......... .......... .......... .......... 68%  118M 1s\n",
      " 54400K .......... .......... .......... .......... .......... 68% 90,7M 1s\n",
      " 54450K .......... .......... .......... .......... .......... 68%  115M 1s\n",
      " 54500K .......... .......... .......... .......... .......... 68%  117M 1s\n",
      " 54550K .......... .......... .......... .......... .......... 68%  119M 1s\n",
      " 54600K .......... .......... .......... .......... .......... 68% 24,3M 1s\n",
      " 54650K .......... .......... .......... .......... .......... 68% 30,0M 1s\n",
      " 54700K .......... .......... .......... .......... .......... 68% 17,6M 1s\n",
      " 54750K .......... .......... .......... .......... .......... 68% 36,6M 1s\n",
      " 54800K .......... .......... .......... .......... .......... 68% 26,2M 1s\n",
      " 54850K .......... .......... .......... .......... .......... 68%  606K 1s\n",
      " 54900K .......... .......... .......... .......... .......... 68% 10,8M 1s\n",
      " 54950K .......... .......... .......... .......... .......... 68% 83,0M 1s\n",
      " 55000K .......... .......... .......... .......... .......... 68% 19,3M 1s\n",
      " 55050K .......... .......... .......... .......... .......... 68% 42,3M 1s\n",
      " 55100K .......... .......... .......... .......... .......... 69% 20,9M 1s\n",
      " 55150K .......... .......... .......... .......... .......... 69% 89,9M 1s\n",
      " 55200K .......... .......... .......... .......... .......... 69% 40,6M 1s\n",
      " 55250K .......... .......... .......... .......... .......... 69% 51,9M 1s\n",
      " 55300K .......... .......... .......... .......... .......... 69%  126M 1s\n",
      " 55350K .......... .......... .......... .......... .......... 69%  156M 1s\n",
      " 55400K .......... .......... .......... .......... .......... 69%  129M 1s\n",
      " 55450K .......... .......... .......... .......... .......... 69% 49,0M 1s\n",
      " 55500K .......... .......... .......... .......... .......... 69%  184M 1s\n",
      " 55550K .......... .......... .......... .......... .......... 69%  110M 1s\n",
      " 55600K .......... .......... .......... .......... .......... 69%  121M 1s\n",
      " 55650K .......... .......... .......... .......... .......... 69% 99,3M 1s\n",
      " 55700K .......... .......... .......... .......... .......... 69%  119M 1s\n",
      " 55750K .......... .......... .......... .......... .......... 69%  130M 1s\n",
      " 55800K .......... .......... .......... .......... .......... 69%  109M 1s\n",
      " 55850K .......... .......... .......... .......... .......... 69%  115M 1s\n",
      " 55900K .......... .......... .......... .......... .......... 70%  117M 1s\n",
      " 55950K .......... .......... .......... .......... .......... 70% 52,6M 1s\n",
      " 56000K .......... .......... .......... .......... .......... 70% 94,2M 1s\n",
      " 56050K .......... .......... .......... .......... .......... 70%  118M 1s\n",
      " 56100K .......... .......... .......... .......... .......... 70%  114M 1s\n",
      " 56150K .......... .......... .......... .......... .......... 70%  117M 1s\n",
      " 56200K .......... .......... .......... .......... .......... 70%  112M 1s\n",
      " 56250K .......... .......... .......... .......... .......... 70%  117M 1s\n",
      " 56300K .......... .......... .......... .......... .......... 70%  119M 1s\n",
      " 56350K .......... .......... .......... .......... .......... 70%  114M 1s\n",
      " 56400K .......... .......... .......... .......... .......... 70% 92,7M 1s\n",
      " 56450K .......... .......... .......... .......... .......... 70%  117M 1s\n",
      " 56500K .......... .......... .......... .......... .......... 70%  113M 1s\n",
      " 56550K .......... .......... .......... .......... .......... 70%  120M 1s\n",
      " 56600K .......... .......... .......... .......... .......... 70%  112M 1s\n",
      " 56650K .......... .......... .......... .......... .......... 70%  117M 1s\n",
      " 56700K .......... .......... .......... .......... .......... 71%  118M 1s\n",
      " 56750K .......... .......... .......... .......... .......... 71%  115M 1s\n",
      " 56800K .......... .......... .......... .......... .......... 71% 88,2M 1s\n",
      " 56850K .......... .......... .......... .......... .......... 71%  122M 1s\n",
      " 56900K .......... .......... .......... .......... .......... 71%  113M 1s\n",
      " 56950K .......... .......... .......... .......... .......... 71%  119M 1s\n",
      " 57000K .......... .......... .......... .......... .......... 71%  114M 1s\n",
      " 57050K .......... .......... .......... .......... .......... 71%  118M 1s\n",
      " 57100K .......... .......... .......... .......... .......... 71%  118M 1s\n",
      " 57150K .......... .......... .......... .......... .......... 71%  117M 1s\n",
      " 57200K .......... .......... .......... .......... .......... 71% 88,5M 1s\n",
      " 57250K .......... .......... .......... .......... .......... 71%  117M 1s\n",
      " 57300K .......... .......... .......... .......... .......... 71%  117M 1s\n",
      " 57350K .......... .......... .......... .......... .......... 71%  114M 1s\n",
      " 57400K .......... .......... .......... .......... .......... 71%  115M 1s\n",
      " 57450K .......... .......... .......... .......... .......... 71%  117M 1s\n",
      " 57500K .......... .......... .......... .......... .......... 72%  118M 1s\n",
      " 57550K .......... .......... .......... .......... .......... 72%  118M 1s\n",
      " 57600K .......... .......... .......... .......... .......... 72% 88,0M 1s\n",
      " 57650K .......... .......... .......... .......... .......... 72% 20,8M 1s\n",
      " 57700K .......... .......... .......... .......... .......... 72% 15,1M 1s\n",
      " 57750K .......... .......... .......... .......... .......... 72% 30,0M 1s\n",
      " 57800K .......... .......... .......... .......... .......... 72%  122M 1s\n",
      " 57850K .......... .......... .......... .......... .......... 72% 34,4M 1s\n",
      " 57900K .......... .......... .......... .......... .......... 72%  612K 1s\n",
      " 57950K .......... .......... .......... .......... .......... 72% 13,2M 1s\n",
      " 58000K .......... .......... .......... .......... .......... 72% 13,0M 1s\n",
      " 58050K .......... .......... .......... .......... .......... 72%  118M 1s\n",
      " 58100K .......... .......... .......... .......... .......... 72% 40,3M 1s\n",
      " 58150K .......... .......... .......... .......... .......... 72% 25,9M 1s\n",
      " 58200K .......... .......... .......... .......... .......... 72% 19,6M 1s\n",
      " 58250K .......... .......... .......... .......... .......... 72%  118M 1s\n",
      " 58300K .......... .......... .......... .......... .......... 73% 53,8M 1s\n",
      " 58350K .......... .......... .......... .......... .......... 73% 60,4M 1s\n",
      " 58400K .......... .......... .......... .......... .......... 73% 88,0M 1s\n",
      " 58450K .......... .......... .......... .......... .......... 73%  119M 1s\n",
      " 58500K .......... .......... .......... .......... .......... 73%  116M 1s\n",
      " 58550K .......... .......... .......... .......... .......... 73%  115M 1s\n",
      " 58600K .......... .......... .......... .......... .......... 73%  113M 1s\n",
      " 58650K .......... .......... .......... .......... .......... 73%  118M 1s\n",
      " 58700K .......... .......... .......... .......... .......... 73%  118M 1s\n",
      " 58750K .......... .......... .......... .......... .......... 73%  118M 1s\n",
      " 58800K .......... .......... .......... .......... .......... 73% 88,0M 1s\n",
      " 58850K .......... .......... .......... .......... .......... 73%  116M 1s\n",
      " 58900K .......... .......... .......... .......... .......... 73%  118M 1s\n",
      " 58950K .......... .......... .......... .......... .......... 73%  118M 1s\n",
      " 59000K .......... .......... .......... .......... .......... 73%  111M 1s\n",
      " 59050K .......... .......... .......... .......... .......... 73%  117M 1s\n",
      " 59100K .......... .......... .......... .......... .......... 74%  114M 1s\n",
      " 59150K .......... .......... .......... .......... .......... 74%  122M 1s\n",
      " 59200K .......... .......... .......... .......... .......... 74% 87,8M 1s\n",
      " 59250K .......... .......... .......... .......... .......... 74%  116M 1s\n",
      " 59300K .......... .......... .......... .......... .......... 74%  118M 1s\n",
      " 59350K .......... .......... .......... .......... .......... 74%  118M 1s\n",
      " 59400K .......... .......... .......... .......... .......... 74%  106M 1s\n",
      " 59450K .......... .......... .......... .......... .......... 74%  119M 1s\n",
      " 59500K .......... .......... .......... .......... .......... 74%  119M 1s\n",
      " 59550K .......... .......... .......... .......... .......... 74%  113M 1s\n",
      " 59600K .......... .......... .......... .......... .......... 74% 91,5M 1s\n",
      " 59650K .......... .......... .......... .......... .......... 74%  118M 1s\n",
      " 59700K .......... .......... .......... .......... .......... 74%  112M 1s\n",
      " 59750K .......... .......... .......... .......... .......... 74%  118M 1s\n",
      " 59800K .......... .......... .......... .......... .......... 74%  112M 1s\n",
      " 59850K .......... .......... .......... .......... .......... 74%  121M 1s\n",
      " 59900K .......... .......... .......... .......... .......... 75%  118M 1s\n",
      " 59950K .......... .......... .......... .......... .......... 75%  118M 1s\n",
      " 60000K .......... .......... .......... .......... .......... 75% 87,5M 1s\n",
      " 60050K .......... .......... .......... .......... .......... 75%  117M 1s\n",
      " 60100K .......... .......... .......... .......... .......... 75%  117M 1s\n",
      " 60150K .......... .......... .......... .......... .......... 75%  114M 1s\n",
      " 60200K .......... .......... .......... .......... .......... 75%  115M 1s\n",
      " 60250K .......... .......... .......... .......... .......... 75%  114M 1s\n",
      " 60300K .......... .......... .......... .......... .......... 75%  117M 1s\n",
      " 60350K .......... .......... .......... .......... .......... 75%  117M 1s\n",
      " 60400K .......... .......... .......... .......... .......... 75% 87,2M 1s\n",
      " 60450K .......... .......... .......... .......... .......... 75%  117M 1s\n",
      " 60500K .......... .......... .......... .......... .......... 75%  118M 1s\n",
      " 60550K .......... .......... .......... .......... .......... 75%  117M 1s\n",
      " 60600K .......... .......... .......... .......... .......... 75%  112M 1s\n",
      " 60650K .......... .......... .......... .......... .......... 75%  113M 1s\n",
      " 60700K .......... .......... .......... .......... .......... 76%  116M 1s\n",
      " 60750K .......... .......... .......... .......... .......... 76% 12,5M 1s\n",
      " 60800K .......... .......... .......... .......... .......... 76% 16,7M 1s\n",
      " 60850K .......... .......... .......... .......... .......... 76% 32,3M 1s\n",
      " 60900K .......... .......... .......... .......... .......... 76% 35,3M 1s\n",
      " 60950K .......... .......... .......... .......... .......... 76%  627K 1s\n",
      " 61000K .......... .......... .......... .......... .......... 76% 13,6M 1s\n",
      " 61050K .......... .......... .......... .......... .......... 76% 8,98M 1s\n",
      " 61100K .......... .......... .......... .......... .......... 76% 75,0M 1s\n",
      " 61150K .......... .......... .......... .......... .......... 76%  116M 1s\n",
      " 61200K .......... .......... .......... .......... .......... 76% 19,7M 1s\n",
      " 61250K .......... .......... .......... .......... .......... 76% 22,8M 1s\n",
      " 61300K .......... .......... .......... .......... .......... 76% 46,9M 1s\n",
      " 61350K .......... .......... .......... .......... .......... 76% 63,3M 1s\n",
      " 61400K .......... .......... .......... .......... .......... 76% 82,6M 1s\n",
      " 61450K .......... .......... .......... .......... .......... 76% 89,7M 1s\n",
      " 61500K .......... .......... .......... .......... .......... 77% 72,7M 1s\n",
      " 61550K .......... .......... .......... .......... .......... 77%  121M 1s\n",
      " 61600K .......... .......... .......... .......... .......... 77%  107M 1s\n",
      " 61650K .......... .......... .......... .......... .......... 77%  132M 1s\n",
      " 61700K .......... .......... .......... .......... .......... 77%  128M 1s\n",
      " 61750K .......... .......... .......... .......... .......... 77%  129M 1s\n",
      " 61800K .......... .......... .......... .......... .......... 77%  117M 1s\n",
      " 61850K .......... .......... .......... .......... .......... 77%  128M 1s\n",
      " 61900K .......... .......... .......... .......... .......... 77%  132M 1s\n",
      " 61950K .......... .......... .......... .......... .......... 77%  132M 1s\n",
      " 62000K .......... .......... .......... .......... .......... 77%  109M 1s\n",
      " 62050K .......... .......... .......... .......... .......... 77%  126M 1s\n",
      " 62100K .......... .......... .......... .......... .......... 77%  126M 1s\n",
      " 62150K .......... .......... .......... .......... .......... 77%  117M 1s\n",
      " 62200K .......... .......... .......... .......... .......... 77%  106M 1s\n",
      " 62250K .......... .......... .......... .......... .......... 77%  119M 1s\n",
      " 62300K .......... .......... .......... .......... .......... 78%  114M 1s\n",
      " 62350K .......... .......... .......... .......... .......... 78%  116M 1s\n",
      " 62400K .......... .......... .......... .......... .......... 78% 95,7M 1s\n",
      " 62450K .......... .......... .......... .......... .......... 78%  118M 1s\n",
      " 62500K .......... .......... .......... .......... .......... 78%  119M 1s\n",
      " 62550K .......... .......... .......... .......... .......... 78%  117M 1s\n",
      " 62600K .......... .......... .......... .......... .......... 78%  110M 1s\n",
      " 62650K .......... .......... .......... .......... .......... 78%  118M 1s\n",
      " 62700K .......... .......... .......... .......... .......... 78%  114M 1s\n",
      " 62750K .......... .......... .......... .......... .......... 78%  119M 1s\n",
      " 62800K .......... .......... .......... .......... .......... 78% 91,8M 1s\n",
      " 62850K .......... .......... .......... .......... .......... 78%  115M 1s\n",
      " 62900K .......... .......... .......... .......... .......... 78%  117M 1s\n",
      " 62950K .......... .......... .......... .......... .......... 78%  103M 1s\n",
      " 63000K .......... .......... .......... .......... .......... 78%  116M 1s\n",
      " 63050K .......... .......... .......... .......... .......... 78%  131M 1s\n",
      " 63100K .......... .......... .......... .......... .......... 79%  110M 1s\n",
      " 63150K .......... .......... .......... .......... .......... 79%  119M 1s\n",
      " 63200K .......... .......... .......... .......... .......... 79% 92,8M 1s\n",
      " 63250K .......... .......... .......... .......... .......... 79%  117M 1s\n",
      " 63300K .......... .......... .......... .......... .......... 79%  115M 1s\n",
      " 63350K .......... .......... .......... .......... .......... 79%  117M 1s\n",
      " 63400K .......... .......... .......... .......... .......... 79% 99,1M 1s\n",
      " 63450K .......... .......... .......... .......... .......... 79%  127M 1s\n",
      " 63500K .......... .......... .......... .......... .......... 79%  126M 1s\n",
      " 63550K .......... .......... .......... .......... .......... 79%  114M 1s\n",
      " 63600K .......... .......... .......... .......... .......... 79% 91,2M 1s\n",
      " 63650K .......... .......... .......... .......... .......... 79%  117M 1s\n",
      " 63700K .......... .......... .......... .......... .......... 79%  115M 1s\n",
      " 63750K .......... .......... .......... .......... .......... 79% 16,6M 1s\n",
      " 63800K .......... .......... .......... .......... .......... 79%  108M 1s\n",
      " 63850K .......... .......... .......... .......... .......... 79% 12,2M 1s\n",
      " 63900K .......... .......... .......... .......... .......... 80% 35,2M 1s\n",
      " 63950K .......... .......... .......... .......... .......... 80% 90,0M 1s\n",
      " 64000K .......... .......... .......... .......... .......... 80% 1,16M 1s\n",
      " 64050K .......... .......... .......... .......... .......... 80% 1,10M 1s\n",
      " 64100K .......... .......... .......... .......... .......... 80% 17,5M 1s\n",
      " 64150K .......... .......... .......... .......... .......... 80% 45,9M 1s\n",
      " 64200K .......... .......... .......... .......... .......... 80% 44,6M 1s\n",
      " 64250K .......... .......... .......... .......... .......... 80% 64,7M 1s\n",
      " 64300K .......... .......... .......... .......... .......... 80% 12,4M 1s\n",
      " 64350K .......... .......... .......... .......... .......... 80% 84,3M 1s\n",
      " 64400K .......... .......... .......... .......... .......... 80% 77,3M 1s\n",
      " 64450K .......... .......... .......... .......... .......... 80% 88,7M 1s\n",
      " 64500K .......... .......... .......... .......... .......... 80% 90,5M 1s\n",
      " 64550K .......... .......... .......... .......... .......... 80% 90,9M 1s\n",
      " 64600K .......... .......... .......... .......... .......... 80% 83,1M 1s\n",
      " 64650K .......... .......... .......... .......... .......... 81% 91,6M 1s\n",
      " 64700K .......... .......... .......... .......... .......... 81% 73,7M 1s\n",
      " 64750K .......... .......... .......... .......... .......... 81%  113M 1s\n",
      " 64800K .......... .......... .......... .......... .......... 81% 99,8M 1s\n",
      " 64850K .......... .......... .......... .......... .......... 81%  122M 1s\n",
      " 64900K .......... .......... .......... .......... .......... 81%  122M 1s\n",
      " 64950K .......... .......... .......... .......... .......... 81%  151M 1s\n",
      " 65000K .......... .......... .......... .......... .......... 81% 83,9M 1s\n",
      " 65050K .......... .......... .......... .......... .......... 81%  178M 1s\n",
      " 65100K .......... .......... .......... .......... .......... 81%  179M 1s\n",
      " 65150K .......... .......... .......... .......... .......... 81%  182M 1s\n",
      " 65200K .......... .......... .......... .......... .......... 81%  156M 1s\n",
      " 65250K .......... .......... .......... .......... .......... 81%  239M 1s\n",
      " 65300K .......... .......... .......... .......... .......... 81%  122M 1s\n",
      " 65350K .......... .......... .......... .......... .......... 81%  128M 1s\n",
      " 65400K .......... .......... .......... .......... .......... 81% 87,0M 1s\n",
      " 65450K .......... .......... .......... .......... .......... 82%  138M 1s\n",
      " 65500K .......... .......... .......... .......... .......... 82%  122M 1s\n",
      " 65550K .......... .......... .......... .......... .......... 82% 85,5M 1s\n",
      " 65600K .......... .......... .......... .......... .......... 82% 93,5M 1s\n",
      " 65650K .......... .......... .......... .......... .......... 82%  107M 1s\n",
      " 65700K .......... .......... .......... .......... .......... 82%  136M 1s\n",
      " 65750K .......... .......... .......... .......... .......... 82%  111M 1s\n",
      " 65800K .......... .......... .......... .......... .......... 82%  116M 1s\n",
      " 65850K .......... .......... .......... .......... .......... 82%  117M 1s\n",
      " 65900K .......... .......... .......... .......... .......... 82%  117M 1s\n",
      " 65950K .......... .......... .......... .......... .......... 82%  119M 1s\n",
      " 66000K .......... .......... .......... .......... .......... 82% 88,9M 1s\n",
      " 66050K .......... .......... .......... .......... .......... 82%  117M 1s\n",
      " 66100K .......... .......... .......... .......... .......... 82%  117M 1s\n",
      " 66150K .......... .......... .......... .......... .......... 82%  113M 1s\n",
      " 66200K .......... .......... .......... .......... .......... 82%  116M 1s\n",
      " 66250K .......... .......... .......... .......... .......... 83%  117M 1s\n",
      " 66300K .......... .......... .......... .......... .......... 83%  118M 1s\n",
      " 66350K .......... .......... .......... .......... .......... 83%  117M 1s\n",
      " 66400K .......... .......... .......... .......... .......... 83% 88,1M 1s\n",
      " 66450K .......... .......... .......... .......... .......... 83%  117M 1s\n",
      " 66500K .......... .......... .......... .......... .......... 83%  118M 1s\n",
      " 66550K .......... .......... .......... .......... .......... 83%  117M 1s\n",
      " 66600K .......... .......... .......... .......... .......... 83%  112M 1s\n",
      " 66650K .......... .......... .......... .......... .......... 83%  117M 1s\n",
      " 66700K .......... .......... .......... .......... .......... 83%  117M 1s\n",
      " 66750K .......... .......... .......... .......... .......... 83%  116M 1s\n",
      " 66800K .......... .......... .......... .......... .......... 83% 24,9M 1s\n",
      " 66850K .......... .......... .......... .......... .......... 83% 11,9M 1s\n",
      " 66900K .......... .......... .......... .......... .......... 83%  117M 1s\n",
      " 66950K .......... .......... .......... .......... .......... 83% 36,0M 1s\n",
      " 67000K .......... .......... .......... .......... .......... 83% 32,7M 1s\n",
      " 67050K .......... .......... .......... .......... .......... 84% 1,19M 1s\n",
      " 67100K .......... .......... .......... .......... .......... 84% 1,05M 1s\n",
      " 67150K .......... .......... .......... .......... .......... 84% 56,1M 1s\n",
      " 67200K .......... .......... .......... .......... .......... 84% 23,1M 1s\n",
      " 67250K .......... .......... .......... .......... .......... 84% 48,6M 1s\n",
      " 67300K .......... .......... .......... .......... .......... 84% 15,3M 1s\n",
      " 67350K .......... .......... .......... .......... .......... 84% 21,1M 1s\n",
      " 67400K .......... .......... .......... .......... .......... 84%  110M 1s\n",
      " 67450K .......... .......... .......... .......... .......... 84%  115M 1s\n",
      " 67500K .......... .......... .......... .......... .......... 84%  112M 1s\n",
      " 67550K .......... .......... .......... .......... .......... 84%  120M 1s\n",
      " 67600K .......... .......... .......... .......... .......... 84% 95,6M 1s\n",
      " 67650K .......... .......... .......... .......... .......... 84%  108M 1s\n",
      " 67700K .......... .......... .......... .......... .......... 84%  102M 1s\n",
      " 67750K .......... .......... .......... .......... .......... 84%  115M 1s\n",
      " 67800K .......... .......... .......... .......... .......... 84%  102M 1s\n",
      " 67850K .......... .......... .......... .......... .......... 85%  107M 1s\n",
      " 67900K .......... .......... .......... .......... .......... 85%  112M 1s\n",
      " 67950K .......... .......... .......... .......... .......... 85% 86,1M 1s\n",
      " 68000K .......... .......... .......... .......... .......... 85% 89,8M 1s\n",
      " 68050K .......... .......... .......... .......... .......... 85%  107M 1s\n",
      " 68100K .......... .......... .......... .......... .......... 85%  122M 1s\n",
      " 68150K .......... .......... .......... .......... .......... 85%  117M 1s\n",
      " 68200K .......... .......... .......... .......... .......... 85%  110M 1s\n",
      " 68250K .......... .......... .......... .......... .......... 85%  118M 1s\n",
      " 68300K .......... .......... .......... .......... .......... 85%  128M 1s\n",
      " 68350K .......... .......... .......... .......... .......... 85%  107M 1s\n",
      " 68400K .......... .......... .......... .......... .......... 85% 89,4M 1s\n",
      " 68450K .......... .......... .......... .......... .......... 85%  109M 1s\n",
      " 68500K .......... .......... .......... .......... .......... 85%  110M 1s\n",
      " 68550K .......... .......... .......... .......... .......... 85%  115M 1s\n",
      " 68600K .......... .......... .......... .......... .......... 85%  104M 1s\n",
      " 68650K .......... .......... .......... .......... .......... 86%  111M 1s\n",
      " 68700K .......... .......... .......... .......... .......... 86%  129M 1s\n",
      " 68750K .......... .......... .......... .......... .......... 86%  137M 1s\n",
      " 68800K .......... .......... .......... .......... .......... 86%  113M 1s\n",
      " 68850K .......... .......... .......... .......... .......... 86%  137M 1s\n",
      " 68900K .......... .......... .......... .......... .......... 86%  113M 1s\n",
      " 68950K .......... .......... .......... .......... .......... 86%  119M 1s\n",
      " 69000K .......... .......... .......... .......... .......... 86%  119M 1s\n",
      " 69050K .......... .......... .......... .......... .......... 86%  126M 1s\n",
      " 69100K .......... .......... .......... .......... .......... 86% 64,7M 1s\n",
      " 69150K .......... .......... .......... .......... .......... 86%  120M 1s\n",
      " 69200K .......... .......... .......... .......... .......... 86% 93,0M 1s\n",
      " 69250K .......... .......... .......... .......... .......... 86%  123M 1s\n",
      " 69300K .......... .......... .......... .......... .......... 86%  150M 1s\n",
      " 69350K .......... .......... .......... .......... .......... 86%  119M 1s\n",
      " 69400K .......... .......... .......... .......... .......... 86%  111M 1s\n",
      " 69450K .......... .......... .......... .......... .......... 87%  120M 1s\n",
      " 69500K .......... .......... .......... .......... .......... 87%  124M 1s\n",
      " 69550K .......... .......... .......... .......... .......... 87%  107M 1s\n",
      " 69600K .......... .......... .......... .......... .......... 87%  105M 1s\n",
      " 69650K .......... .......... .......... .......... .......... 87%  143M 1s\n",
      " 69700K .......... .......... .......... .......... .......... 87%  142M 1s\n",
      " 69750K .......... .......... .......... .......... .......... 87%  144M 1s\n",
      " 69800K .......... .......... .......... .......... .......... 87%  113M 1s\n",
      " 69850K .......... .......... .......... .......... .......... 87% 16,8M 1s\n",
      " 69900K .......... .......... .......... .......... .......... 87% 19,8M 1s\n",
      " 69950K .......... .......... .......... .......... .......... 87% 27,4M 1s\n",
      " 70000K .......... .......... .......... .......... .......... 87% 53,0M 1s\n",
      " 70050K .......... .......... .......... .......... .......... 87%  127M 1s\n",
      " 70100K .......... .......... .......... .......... .......... 87%  631K 1s\n",
      " 70150K .......... .......... .......... .......... .......... 87% 7,06M 1s\n",
      " 70200K .......... .......... .......... .......... .......... 87% 20,4M 1s\n",
      " 70250K .......... .......... .......... .......... .......... 88% 29,3M 1s\n",
      " 70300K .......... .......... .......... .......... .......... 88% 39,0M 1s\n",
      " 70350K .......... .......... .......... .......... .......... 88% 27,5M 1s\n",
      " 70400K .......... .......... .......... .......... .......... 88% 15,9M 1s\n",
      " 70450K .......... .......... .......... .......... .......... 88% 63,8M 1s\n",
      " 70500K .......... .......... .......... .......... .......... 88%  118M 1s\n",
      " 70550K .......... .......... .......... .......... .......... 88% 58,7M 0s\n",
      " 70600K .......... .......... .......... .......... .......... 88%  114M 0s\n",
      " 70650K .......... .......... .......... .......... .......... 88%  118M 0s\n",
      " 70700K .......... .......... .......... .......... .......... 88%  114M 0s\n",
      " 70750K .......... .......... .......... .......... .......... 88%  117M 0s\n",
      " 70800K .......... .......... .......... .......... .......... 88% 91,2M 0s\n",
      " 70850K .......... .......... .......... .......... .......... 88%  114M 0s\n",
      " 70900K .......... .......... .......... .......... .......... 88%  116M 0s\n",
      " 70950K .......... .......... .......... .......... .......... 88%  118M 0s\n",
      " 71000K .......... .......... .......... .......... .......... 88%  111M 0s\n",
      " 71050K .......... .......... .......... .......... .......... 89%  109M 0s\n",
      " 71100K .......... .......... .......... .......... .......... 89%  133M 0s\n",
      " 71150K .......... .......... .......... .......... .......... 89%  111M 0s\n",
      " 71200K .......... .......... .......... .......... .......... 89% 91,9M 0s\n",
      " 71250K .......... .......... .......... .......... .......... 89%  117M 0s\n",
      " 71300K .......... .......... .......... .......... .......... 89%  114M 0s\n",
      " 71350K .......... .......... .......... .......... .......... 89%  117M 0s\n",
      " 71400K .......... .......... .......... .......... .......... 89%  115M 0s\n",
      " 71450K .......... .......... .......... .......... .......... 89%  119M 0s\n",
      " 71500K .......... .......... .......... .......... .......... 89%  117M 0s\n",
      " 71550K .......... .......... .......... .......... .......... 89%  114M 0s\n",
      " 71600K .......... .......... .......... .......... .......... 89% 89,7M 0s\n",
      " 71650K .......... .......... .......... .......... .......... 89%  117M 0s\n",
      " 71700K .......... .......... .......... .......... .......... 89%  114M 0s\n",
      " 71750K .......... .......... .......... .......... .......... 89%  117M 0s\n",
      " 71800K .......... .......... .......... .......... .......... 89%  116M 0s\n",
      " 71850K .......... .......... .......... .......... .......... 90%  117M 0s\n",
      " 71900K .......... .......... .......... .......... .......... 90%  117M 0s\n",
      " 71950K .......... .......... .......... .......... .......... 90%  114M 0s\n",
      " 72000K .......... .......... .......... .......... .......... 90% 89,8M 0s\n",
      " 72050K .......... .......... .......... .......... .......... 90%  118M 0s\n",
      " 72100K .......... .......... .......... .......... .......... 90%  114M 0s\n",
      " 72150K .......... .......... .......... .......... .......... 90%  117M 0s\n",
      " 72200K .......... .......... .......... .......... .......... 90%  116M 0s\n",
      " 72250K .......... .......... .......... .......... .......... 90%  117M 0s\n",
      " 72300K .......... .......... .......... .......... .......... 90%  118M 0s\n",
      " 72350K .......... .......... .......... .......... .......... 90%  114M 0s\n",
      " 72400K .......... .......... .......... .......... .......... 90% 89,5M 0s\n",
      " 72450K .......... .......... .......... .......... .......... 90%  117M 0s\n",
      " 72500K .......... .......... .......... .......... .......... 90%  117M 0s\n",
      " 72550K .......... .......... .......... .......... .......... 90%  114M 0s\n",
      " 72600K .......... .......... .......... .......... .......... 90%  116M 0s\n",
      " 72650K .......... .......... .......... .......... .......... 91%  117M 0s\n",
      " 72700K .......... .......... .......... .......... .......... 91%  117M 0s\n",
      " 72750K .......... .......... .......... .......... .......... 91%  108M 0s\n",
      " 72800K .......... .......... .......... .......... .......... 91% 92,8M 0s\n",
      " 72850K .......... .......... .......... .......... .......... 91% 23,6M 0s\n",
      " 72900K .......... .......... .......... .......... .......... 91%  118M 0s\n",
      " 72950K .......... .......... .......... .......... .......... 91% 16,7M 0s\n",
      " 73000K .......... .......... .......... .......... .......... 91% 28,7M 0s\n",
      " 73050K .......... .......... .......... .......... .......... 91% 1,20M 0s\n",
      " 73100K .......... .......... .......... .......... .......... 91% 1,25M 0s\n",
      " 73150K .......... .......... .......... .......... .......... 91% 97,6M 0s\n",
      " 73200K .......... .......... .......... .......... .......... 91% 8,16M 0s\n",
      " 73250K .......... .......... .......... .......... .......... 91% 13,4M 0s\n",
      " 73300K .......... .......... .......... .......... .......... 91% 30,0M 0s\n",
      " 73350K .......... .......... .......... .......... .......... 91% 23,2M 0s\n",
      " 73400K .......... .......... .......... .......... .......... 91%  110M 0s\n",
      " 73450K .......... .......... .......... .......... .......... 92% 10,9M 0s\n",
      " 73500K .......... .......... .......... .......... .......... 92%  108M 0s\n",
      " 73550K .......... .......... .......... .......... .......... 92% 85,4M 0s\n",
      " 73600K .......... .......... .......... .......... .......... 92% 90,6M 0s\n",
      " 73650K .......... .......... .......... .......... .......... 92%  124M 0s\n",
      " 73700K .......... .......... .......... .......... .......... 92% 99,3M 0s\n",
      " 73750K .......... .......... .......... .......... .......... 92%  141M 0s\n",
      " 73800K .......... .......... .......... .......... .......... 92% 44,2M 0s\n",
      " 73850K .......... .......... .......... .......... .......... 92%  140M 0s\n",
      " 73900K .......... .......... .......... .......... .......... 92%  158M 0s\n",
      " 73950K .......... .......... .......... .......... .......... 92%  144M 0s\n",
      " 74000K .......... .......... .......... .......... .......... 92%  125M 0s\n",
      " 74050K .......... .......... .......... .......... .......... 92%  167M 0s\n",
      " 74100K .......... .......... .......... .......... .......... 92%  164M 0s\n",
      " 74150K .......... .......... .......... .......... .......... 92%  120M 0s\n",
      " 74200K .......... .......... .......... .......... .......... 92%  106M 0s\n",
      " 74250K .......... .......... .......... .......... .......... 93%  122M 0s\n",
      " 74300K .......... .......... .......... .......... .......... 93%  114M 0s\n",
      " 74350K .......... .......... .......... .......... .......... 93%  119M 0s\n",
      " 74400K .......... .......... .......... .......... .......... 93% 91,2M 0s\n",
      " 74450K .......... .......... .......... .......... .......... 93%  116M 0s\n",
      " 74500K .......... .......... .......... .......... .......... 93%  119M 0s\n",
      " 74550K .......... .......... .......... .......... .......... 93%  115M 0s\n",
      " 74600K .......... .......... .......... .......... .......... 93%  109M 0s\n",
      " 74650K .......... .......... .......... .......... .......... 93%  121M 0s\n",
      " 74700K .......... .......... .......... .......... .......... 93%  111M 0s\n",
      " 74750K .......... .......... .......... .......... .......... 93%  127M 0s\n",
      " 74800K .......... .......... .......... .......... .......... 93% 90,0M 0s\n",
      " 74850K .......... .......... .......... .......... .......... 93%  120M 0s\n",
      " 74900K .......... .......... .......... .......... .......... 93%  116M 0s\n",
      " 74950K .......... .......... .......... .......... .......... 93%  117M 0s\n",
      " 75000K .......... .......... .......... .......... .......... 93%  109M 0s\n",
      " 75050K .......... .......... .......... .......... .......... 94%  112M 0s\n",
      " 75100K .......... .......... .......... .......... .......... 94%  123M 0s\n",
      " 75150K .......... .......... .......... .......... .......... 94% 99,1M 0s\n",
      " 75200K .......... .......... .......... .......... .......... 94% 43,5M 0s\n",
      " 75250K .......... .......... .......... .......... .......... 94% 51,9M 0s\n",
      " 75300K .......... .......... .......... .......... .......... 94%  131M 0s\n",
      " 75350K .......... .......... .......... .......... .......... 94%  128M 0s\n",
      " 75400K .......... .......... .......... .......... .......... 94%  118M 0s\n",
      " 75450K .......... .......... .......... .......... .......... 94%  130M 0s\n",
      " 75500K .......... .......... .......... .......... .......... 94%  119M 0s\n",
      " 75550K .......... .......... .......... .......... .......... 94%  187M 0s\n",
      " 75600K .......... .......... .......... .......... .......... 94%  192M 0s\n",
      " 75650K .......... .......... .......... .......... .......... 94%  236M 0s\n",
      " 75700K .......... .......... .......... .......... .......... 94%  227M 0s\n",
      " 75750K .......... .......... .......... .......... .......... 94%  196M 0s\n",
      " 75800K .......... .......... .......... .......... .......... 94%  111M 0s\n",
      " 75850K .......... .......... .......... .......... .......... 95%  119M 0s\n",
      " 75900K .......... .......... .......... .......... .......... 95%  119M 0s\n",
      " 75950K .......... .......... .......... .......... .......... 95% 15,0M 0s\n",
      " 76000K .......... .......... .......... .......... .......... 95% 22,5M 0s\n",
      " 76050K .......... .......... .......... .......... .......... 95% 42,6M 0s\n",
      " 76100K .......... .......... .......... .......... .......... 95% 1,18M 0s\n",
      " 76150K .......... .......... .......... .......... .......... 95% 1,29M 0s\n",
      " 76200K .......... .......... .......... .......... .......... 95% 8,19M 0s\n",
      " 76250K .......... .......... .......... .......... .......... 95%  141M 0s\n",
      " 76300K .......... .......... .......... .......... .......... 95% 9,20M 0s\n",
      " 76350K .......... .......... .......... .......... .......... 95% 37,5M 0s\n",
      " 76400K .......... .......... .......... .......... .......... 95% 37,7M 0s\n",
      " 76450K .......... .......... .......... .......... .......... 95% 15,9M 0s\n",
      " 76500K .......... .......... .......... .......... .......... 95% 72,8M 0s\n",
      " 76550K .......... .......... .......... .......... .......... 95% 23,8M 0s\n",
      " 76600K .......... .......... .......... .......... .......... 95% 58,8M 0s\n",
      " 76650K .......... .......... .......... .......... .......... 96% 89,1M 0s\n",
      " 76700K .......... .......... .......... .......... .......... 96%  117M 0s\n",
      " 76750K .......... .......... .......... .......... .......... 96%  119M 0s\n",
      " 76800K .......... .......... .......... .......... .......... 96% 88,0M 0s\n",
      " 76850K .......... .......... .......... .......... .......... 96%  117M 0s\n",
      " 76900K .......... .......... .......... .......... .......... 96%  118M 0s\n",
      " 76950K .......... .......... .......... .......... .......... 96%  117M 0s\n",
      " 77000K .......... .......... .......... .......... .......... 96%  112M 0s\n",
      " 77050K .......... .......... .......... .......... .......... 96%  117M 0s\n",
      " 77100K .......... .......... .......... .......... .......... 96% 81,9M 0s\n",
      " 77150K .......... .......... .......... .......... .......... 96%  119M 0s\n",
      " 77200K .......... .......... .......... .......... .......... 96% 91,0M 0s\n",
      " 77250K .......... .......... .......... .......... .......... 96%  112M 0s\n",
      " 77300K .......... .......... .......... .......... .......... 96%  119M 0s\n",
      " 77350K .......... .......... .......... .......... .......... 96%  116M 0s\n",
      " 77400K .......... .......... .......... .......... .......... 96%  116M 0s\n",
      " 77450K .......... .......... .......... .......... .......... 97%  118M 0s\n",
      " 77500K .......... .......... .......... .......... .......... 97%  113M 0s\n",
      " 77550K .......... .......... .......... .......... .......... 97%  119M 0s\n",
      " 77600K .......... .......... .......... .......... .......... 97% 90,0M 0s\n",
      " 77650K .......... .......... .......... .......... .......... 97%  115M 0s\n",
      " 77700K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 77750K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 77800K .......... .......... .......... .......... .......... 97%  116M 0s\n",
      " 77850K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 77900K .......... .......... .......... .......... .......... 97%  113M 0s\n",
      " 77950K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 78000K .......... .......... .......... .......... .......... 97% 89,8M 0s\n",
      " 78050K .......... .......... .......... .......... .......... 97%  114M 0s\n",
      " 78100K .......... .......... .......... .......... .......... 97%  118M 0s\n",
      " 78150K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 78200K .......... .......... .......... .......... .......... 97%  115M 0s\n",
      " 78250K .......... .......... .......... .......... .......... 98%  118M 0s\n",
      " 78300K .......... .......... .......... .......... .......... 98%  114M 0s\n",
      " 78350K .......... .......... .......... .......... .......... 98% 33,4M 0s\n",
      " 78400K .......... .......... .......... .......... .......... 98% 91,9M 0s\n",
      " 78450K .......... .......... .......... .......... .......... 98%  113M 0s\n",
      " 78500K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 78550K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 78600K .......... .......... .......... .......... .......... 98%  115M 0s\n",
      " 78650K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 78700K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 78750K .......... .......... .......... .......... .......... 98%  114M 0s\n",
      " 78800K .......... .......... .......... .......... .......... 98% 90,0M 0s\n",
      " 78850K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 78900K .......... .......... .......... .......... .......... 98%  114M 0s\n",
      " 78950K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 79000K .......... .......... .......... .......... .......... 98% 75,8M 0s\n",
      " 79050K .......... .......... .......... .......... .......... 99% 15,3M 0s\n",
      " 79100K .......... .......... .......... .......... .......... 99%  124M 0s\n",
      " 79150K .......... .......... .......... .......... .......... 99% 1,21M 0s\n",
      " 79200K .......... .......... .......... .......... .......... 99% 1,24M 0s\n",
      " 79250K .......... .......... .......... .......... .......... 99% 9,25M 0s\n",
      " 79300K .......... .......... .......... .......... .......... 99% 11,7M 0s\n",
      " 79350K .......... .......... .......... .......... .......... 99% 20,8M 0s\n",
      " 79400K .......... .......... .......... .......... .......... 99% 24,1M 0s\n",
      " 79450K .......... .......... .......... .......... .......... 99%  115M 0s\n",
      " 79500K .......... .......... .......... .......... .......... 99% 22,0M 0s\n",
      " 79550K .......... .......... .......... .......... .......... 99% 19,5M 0s\n",
      " 79600K .......... .......... .......... .......... .......... 99% 30,9M 0s\n",
      " 79650K .......... .......... .......... .......... .......... 99%  119M 0s\n",
      " 79700K .......... .......... .......... .......... .......... 99% 55,8M 0s\n",
      " 79750K .......... .......... .......... .......... .......... 99%  118M 0s\n",
      " 79800K .......... .......... .......... .......... .......... 99%  107M 0s\n",
      " 79850K .......... .......... .....                           100%  143M=4,2s\n",
      "\n",
      "2023-08-17 10:54:38 (18,6 MB/s) - “data/ClinVar/clinvar.vcf.gz” guardado [81792225/81792225]\n",
      "\n",
      "--2023-08-17 10:54:38--  https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/clinvar.vcf.gz.tbi\n",
      "Resolviendo ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.10, 130.14.250.12, 2607:f220:41e:250::10, ...\n",
      "Conectando con ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)[130.14.250.10]:443... conectado.\n",
      "Petición HTTP enviada, esperando respuesta... 200 OK\n",
      "Longitud: 506114 (494K) [application/x-gzip]\n",
      "Grabando a: “data/ClinVar/clinvar.vcf.gz.tbi”\n",
      "\n",
      "     0K .......... .......... .......... .......... .......... 10%  198K 2s\n",
      "    50K .......... .......... .......... .......... .......... 20%  398K 1s\n",
      "   100K .......... .......... .......... .......... .......... 30% 25,4M 1s\n",
      "   150K .......... .......... .......... .......... .......... 40%  402K 1s\n",
      "   200K .......... .......... .......... .......... .......... 50% 65,5M 0s\n",
      "   250K .......... .......... .......... .......... .......... 60% 77,4M 0s\n",
      "   300K .......... .......... .......... .......... .......... 70% 70,7M 0s\n",
      "   350K .......... .......... .......... .......... .......... 80%  403K 0s\n",
      "   400K .......... .......... .......... .......... .......... 91% 52,7M 0s\n",
      "   450K .......... .......... .......... .......... ....      100%  154M=0,6s\n",
      "\n",
      "2023-08-17 10:54:39 (782 KB/s) - “data/ClinVar/clinvar.vcf.gz.tbi” guardado [506114/506114]\n",
      "\n",
      "[Thu Aug 17 10:55:15 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T105432.103400.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:55:17 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753368_chr3.vcf\n",
      "    output: results/variants/vep/ERR753368_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753368_chr3.txt\n",
      "    wildcards: sample=ERR753368_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 10:55:52 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T105515.436344.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:55:54 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753369_chr3.vcf\n",
      "    output: results/variants/vep/ERR753369_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753369_chr3.txt\n",
      "    wildcards: sample=ERR753369_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 10:56:28 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T105552.861356.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:56:29 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753370_chr3.vcf\n",
      "    output: results/variants/vep/ERR753370_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753370_chr3.txt\n",
      "    wildcards: sample=ERR753370_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 10:57:07 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T105628.548386.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:57:09 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753371_chr3.vcf\n",
      "    output: results/variants/vep/ERR753371_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753371_chr3.txt\n",
      "    wildcards: sample=ERR753371_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 10:57:34 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T105708.131103.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:57:36 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753372_chr3.vcf\n",
      "    output: results/variants/vep/ERR753372_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753372_chr3.txt\n",
      "    wildcards: sample=ERR753372_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 10:58:09 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T105735.005590.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:58:11 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753373_chr3.vcf\n",
      "    output: results/variants/vep/ERR753373_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753373_chr3.txt\n",
      "    wildcards: sample=ERR753373_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 10:58:47 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T105810.143102.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:58:49 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753374_chr3.vcf\n",
      "    output: results/variants/vep/ERR753374_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753374_chr3.txt\n",
      "    wildcards: sample=ERR753374_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 10:59:16 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T105848.084624.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:59:17 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753375_chr3.vcf\n",
      "    output: results/variants/vep/ERR753375_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753375_chr3.txt\n",
      "    wildcards: sample=ERR753375_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 10:59:44 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T105916.364703.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 10:59:45 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753376_chr3.vcf\n",
      "    output: results/variants/vep/ERR753376_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753376_chr3.txt\n",
      "    wildcards: sample=ERR753376_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 11:00:13 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T105944.362135.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:00:14 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753377_chr3.vcf\n",
      "    output: results/variants/vep/ERR753377_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753377_chr3.txt\n",
      "    wildcards: sample=ERR753377_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 11:00:44 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T110013.661791.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:00:45 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753378_chr3.vcf\n",
      "    output: results/variants/vep/ERR753378_chr3.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753378_chr3.txt\n",
      "    wildcards: sample=ERR753378_chr3\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 11:01:19 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T110044.388394.snakemake.log\n"
     ]
    },
    {
     "ename": "TypeError",
     "evalue": "bufsize must be an integer",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[4], line 29\u001b[0m\n\u001b[1;32m     26\u001b[0m \u001b[39mfor\u001b[39;00m sample \u001b[39min\u001b[39;00m sample_list:\n\u001b[1;32m     27\u001b[0m     subprocess\u001b[39m.\u001b[39mrun([\u001b[39m\"\u001b[39m\u001b[39msnakemake\u001b[39m\u001b[39m\"\u001b[39m, \u001b[39m\"\u001b[39m\u001b[39m--cores\u001b[39m\u001b[39m\"\u001b[39m, \u001b[39m\"\u001b[39m\u001b[39m1\u001b[39m\u001b[39m\"\u001b[39m, \u001b[39m\"\u001b[39m\u001b[39m--use-conda\u001b[39m\u001b[39m\"\u001b[39m, \u001b[39m\"\u001b[39m\u001b[39mresults/variants/vep/\u001b[39m\u001b[39m\"\u001b[39m \u001b[39m+\u001b[39m sample \u001b[39m+\u001b[39m \u001b[39m\"\u001b[39m\u001b[39m_chr3.txt\u001b[39m\u001b[39m\"\u001b[39m])\n\u001b[0;32m---> 29\u001b[0m subprocess\u001b[39m.\u001b[39;49mrun(\u001b[39m\"\u001b[39;49m\u001b[39msnakemake\u001b[39;49m\u001b[39m\"\u001b[39;49m, \u001b[39m\"\u001b[39;49m\u001b[39m--cores\u001b[39;49m\u001b[39m\"\u001b[39;49m, \u001b[39m\"\u001b[39;49m\u001b[39m1\u001b[39;49m\u001b[39m\"\u001b[39;49m, \u001b[39m\"\u001b[39;49m\u001b[39mparsing_dataR\u001b[39;49m\u001b[39m\"\u001b[39;49m)\n",
      "File \u001b[0;32m~/miniconda3/envs/Greference_tools/lib/python3.11/subprocess.py:546\u001b[0m, in \u001b[0;36mrun\u001b[0;34m(input, capture_output, timeout, check, *popenargs, **kwargs)\u001b[0m\n\u001b[1;32m    543\u001b[0m     kwargs[\u001b[39m'\u001b[39m\u001b[39mstdout\u001b[39m\u001b[39m'\u001b[39m] \u001b[39m=\u001b[39m PIPE\n\u001b[1;32m    544\u001b[0m     kwargs[\u001b[39m'\u001b[39m\u001b[39mstderr\u001b[39m\u001b[39m'\u001b[39m] \u001b[39m=\u001b[39m PIPE\n\u001b[0;32m--> 546\u001b[0m \u001b[39mwith\u001b[39;00m Popen(\u001b[39m*\u001b[39;49mpopenargs, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mkwargs) \u001b[39mas\u001b[39;00m process:\n\u001b[1;32m    547\u001b[0m     \u001b[39mtry\u001b[39;00m:\n\u001b[1;32m    548\u001b[0m         stdout, stderr \u001b[39m=\u001b[39m process\u001b[39m.\u001b[39mcommunicate(\u001b[39minput\u001b[39m, timeout\u001b[39m=\u001b[39mtimeout)\n",
      "File \u001b[0;32m~/miniconda3/envs/Greference_tools/lib/python3.11/subprocess.py:833\u001b[0m, in \u001b[0;36mPopen.__init__\u001b[0;34m(self, args, bufsize, executable, stdin, stdout, stderr, preexec_fn, close_fds, shell, cwd, env, universal_newlines, startupinfo, creationflags, restore_signals, start_new_session, pass_fds, user, group, extra_groups, encoding, errors, text, umask, pipesize, process_group)\u001b[0m\n\u001b[1;32m    831\u001b[0m     bufsize \u001b[39m=\u001b[39m \u001b[39m-\u001b[39m\u001b[39m1\u001b[39m  \u001b[39m# Restore default\u001b[39;00m\n\u001b[1;32m    832\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39misinstance\u001b[39m(bufsize, \u001b[39mint\u001b[39m):\n\u001b[0;32m--> 833\u001b[0m     \u001b[39mraise\u001b[39;00m \u001b[39mTypeError\u001b[39;00m(\u001b[39m\"\u001b[39m\u001b[39mbufsize must be an integer\u001b[39m\u001b[39m\"\u001b[39m)\n\u001b[1;32m    835\u001b[0m \u001b[39mif\u001b[39;00m pipesize \u001b[39mis\u001b[39;00m \u001b[39mNone\u001b[39;00m:\n\u001b[1;32m    836\u001b[0m     pipesize \u001b[39m=\u001b[39m \u001b[39m-\u001b[39m\u001b[39m1\u001b[39m  \u001b[39m# Restore default\u001b[39;00m\n",
      "\u001b[0;31mTypeError\u001b[0m: bufsize must be an integer"
     ]
    }
   ],
   "source": [
    "## Running the workflow for the 3º chormosome\n",
    "\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"data/reference/genome.fa\"]) ## rule reference_genome \n",
    "\n",
    "for sample in sample_list:\n",
    "    ## Quality inspection\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/fastqc_result/\" + sample + \"_chr3_1_fastqc.html\",   \n",
    "                    \"data/processed/\" + sample + \"_chr3_1_fastp.fastq.gz\",   \n",
    "                    \"results/fastqc_result/trimmed/\" + sample + \"_chr3_1_fastp_fastqc.html\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    ## Mapping reads\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/mapped_reads/\" + sample + \"_chr3_sorted.sam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr3_sorted.bam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr3_dedup.bam\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    # Extracting variants  \n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/\" + sample + \"_chr3.vcf\"])   \n",
    "\n",
    "## Downloading VEP\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"vep_install_db\"])\n",
    "## VEP CLI\n",
    "for sample in sample_list:\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/vep/\" + sample + \"_chr3.txt\"])\n",
    "\n",
    "## Extracting the gene PIKCA3 from the VEP files\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"parsing_dataR\"])\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Changing the config file to do the chromosome 5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['sed', '-i', '50s/PIK3CA/APC/g', 'config.yaml'], returncode=0)"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subprocess.run([\"sed\", \"-i\", \"6,+12s/chr3/chr5/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\",\"24s/chr3/chr5/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"31s/chromosome.3.fa.gz/chromosome.5.fa.gz/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"50s/PIK3CA/APC/g\", \"config.yaml\"])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job                 count    min threads    max threads\n",
      "----------------  -------  -------------  -------------\n",
      "reference_genome        1              1              1\n",
      "total                   1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:19:12 2023]\n",
      "rule reference_genome:\n",
      "    output: data/reference/genome.fa\n",
      "    jobid: 0\n",
      "    reason: Forced execution\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "--2023-08-17 11:19:12--  https://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.5.fa.gz\n",
      "Resolviendo ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.169\n",
      "Conectando con ftp.ensembl.org (ftp.ensembl.org)[193.62.193.169]:443... conectado.\n",
      "Petición HTTP enviada, esperando respuesta... 200 OK\n",
      "Longitud: 54485176 (52M) [application/x-gzip]\n",
      "Guardando como: ‘data/reference/genome.fa.gz’\n",
      "\n",
      "     0K .......... .......... .......... .......... ..........  0%  407K 2m10s\n",
      "    50K .......... .......... .......... .......... ..........  0% 1,65M 81s\n",
      "   100K .......... .......... .......... .......... ..........  0% 1,48M 66s\n",
      "   150K .......... .......... .......... .......... ..........  0% 59,1M 49s\n",
      "   200K .......... .......... .......... .......... ..........  0%  834K 52s\n",
      "   250K .......... .......... .......... .......... ..........  0% 24,9M 44s\n",
      "   300K .......... .......... .......... .......... ..........  0% 48,3M 38s\n",
      "   350K .......... .......... .......... .......... ..........  0%  112M 33s\n",
      "   400K .......... .......... .......... .......... ..........  0% 1,86M 32s\n",
      "   450K .......... .......... .......... .......... ..........  0% 1,59M 32s\n",
      "   500K .......... .......... .......... .......... ..........  1% 40,0M 29s\n",
      "   550K .......... .......... .......... .......... ..........  1% 39,7M 27s\n",
      "   600K .......... .......... .......... .......... ..........  1%  102M 25s\n",
      "   650K .......... .......... .......... .......... ..........  1% 53,1M 23s\n",
      "   700K .......... .......... .......... .......... ..........  1% 57,5M 22s\n",
      "   750K .......... .......... .......... .......... ..........  1% 34,5M 20s\n",
      "   800K .......... .......... .......... .......... ..........  1%  114M 19s\n",
      "   850K .......... .......... .......... .......... ..........  1% 2,01M 20s\n",
      "   900K .......... .......... .......... .......... ..........  1% 5,32M 19s\n",
      "   950K .......... .......... .......... .......... ..........  1% 2,30M 19s\n",
      "  1000K .......... .......... .......... .......... ..........  1% 35,0M 18s\n",
      "  1050K .......... .......... .......... .......... ..........  2% 40,3M 18s\n",
      "  1100K .......... .......... .......... .......... ..........  2% 49,7M 17s\n",
      "  1150K .......... .......... .......... .......... ..........  2%  139M 16s\n",
      "  1200K .......... .......... .......... .......... ..........  2%  125M 15s\n",
      "  1250K .......... .......... .......... .......... ..........  2%  152M 15s\n",
      "  1300K .......... .......... .......... .......... ..........  2%  138M 14s\n",
      "  1350K .......... .......... .......... .......... ..........  2%  129M 14s\n",
      "  1400K .......... .......... .......... .......... ..........  2%  156M 13s\n",
      "  1450K .......... .......... .......... .......... ..........  2%  130M 13s\n",
      "  1500K .......... .......... .......... .......... ..........  2%  153M 12s\n",
      "  1550K .......... .......... .......... .......... ..........  3%  142M 12s\n",
      "  1600K .......... .......... .......... .......... ..........  3%  112M 12s\n",
      "  1650K .......... .......... .......... .......... ..........  3%  123M 11s\n",
      "  1700K .......... .......... .......... .......... ..........  3%  100M 11s\n",
      "  1750K .......... .......... .......... .......... ..........  3% 2,11M 11s\n",
      "  1800K .......... .......... .......... .......... ..........  3% 44,8M 11s\n",
      "  1850K .......... .......... .......... .......... ..........  3% 6,51M 11s\n",
      "  1900K .......... .......... .......... .......... ..........  3% 2,38M 11s\n",
      "  1950K .......... .......... .......... .......... ..........  3% 38,9M 11s\n",
      "  2000K .......... .......... .......... .......... ..........  3% 48,0M 11s\n",
      "  2050K .......... .......... .......... .......... ..........  3%  120M 10s\n",
      "  2100K .......... .......... .......... .......... ..........  4%  102M 10s\n",
      "  2150K .......... .......... .......... .......... ..........  4%  114M 10s\n",
      "  2200K .......... .......... .......... .......... ..........  4%  103M 10s\n",
      "  2250K .......... .......... .......... .......... ..........  4% 73,1M 10s\n",
      "  2300K .......... .......... .......... .......... ..........  4%  143M 9s\n",
      "  2350K .......... .......... .......... .......... ..........  4%  125M 9s\n",
      "  2400K .......... .......... .......... .......... ..........  4%  147M 9s\n",
      "  2450K .......... .......... .......... .......... ..........  4%  112M 9s\n",
      "  2500K .......... .......... .......... .......... ..........  4%  116M 9s\n",
      "  2550K .......... .......... .......... .......... ..........  4%  121M 8s\n",
      "  2600K .......... .......... .......... .......... ..........  4%  100M 8s\n",
      "  2650K .......... .......... .......... .......... ..........  5%  126M 8s\n",
      "  2700K .......... .......... .......... .......... ..........  5%  100M 8s\n",
      "  2750K .......... .......... .......... .......... ..........  5%  120M 8s\n",
      "  2800K .......... .......... .......... .......... ..........  5%  121M 8s\n",
      "  2850K .......... .......... .......... .......... ..........  5%  102M 8s\n",
      "  2900K .......... .......... .......... .......... ..........  5%  116M 7s\n",
      "  2950K .......... .......... .......... .......... ..........  5%  101M 7s\n",
      "  3000K .......... .......... .......... .......... ..........  5%  124M 7s\n",
      "  3050K .......... .......... .......... .......... ..........  5%  115M 7s\n",
      "  3100K .......... .......... .......... .......... ..........  5%  104M 7s\n",
      "  3150K .......... .......... .......... .......... ..........  6%  122M 7s\n",
      "  3200K .......... .......... .......... .......... ..........  6%  100M 7s\n",
      "  3250K .......... .......... .......... .......... ..........  6%  122M 7s\n",
      "  3300K .......... .......... .......... .......... ..........  6%  120M 7s\n",
      "  3350K .......... .......... .......... .......... ..........  6%  102M 6s\n",
      "  3400K .......... .......... .......... .......... ..........  6%  117M 6s\n",
      "  3450K .......... .......... .......... .......... ..........  6%  104M 6s\n",
      "  3500K .......... .......... .......... .......... ..........  6%  118M 6s\n",
      "  3550K .......... .......... .......... .......... ..........  6% 3,00M 6s\n",
      "  3600K .......... .......... .......... .......... ..........  6% 21,2M 6s\n",
      "  3650K .......... .......... .......... .......... ..........  6%  117M 6s\n",
      "  3700K .......... .......... .......... .......... ..........  7%  105M 6s\n",
      "  3750K .......... .......... .......... .......... ..........  7% 8,36M 6s\n",
      "  3800K .......... .......... .......... .......... ..........  7% 46,9M 6s\n",
      "  3850K .......... .......... .......... .......... ..........  7% 6,06M 6s\n",
      "  3900K .......... .......... .......... .......... ..........  7% 4,04M 6s\n",
      "  3950K .......... .......... .......... .......... ..........  7% 52,4M 6s\n",
      "  4000K .......... .......... .......... .......... ..........  7% 58,6M 6s\n",
      "  4050K .......... .......... .......... .......... ..........  7%  116M 6s\n",
      "  4100K .......... .......... .......... .......... ..........  7% 53,5M 6s\n",
      "  4150K .......... .......... .......... .......... ..........  7%  125M 6s\n",
      "  4200K .......... .......... .......... .......... ..........  7% 99,8M 6s\n",
      "  4250K .......... .......... .......... .......... ..........  8%  121M 6s\n",
      "  4300K .......... .......... .......... .......... ..........  8% 80,2M 6s\n",
      "  4350K .......... .......... .......... .......... ..........  8%  122M 5s\n",
      "  4400K .......... .......... .......... .......... ..........  8%  142M 5s\n",
      "  4450K .......... .......... .......... .......... ..........  8%  117M 5s\n",
      "  4500K .......... .......... .......... .......... ..........  8%  118M 5s\n",
      "  4550K .......... .......... .......... .......... ..........  8%  120M 5s\n",
      "  4600K .......... .......... .......... .......... ..........  8%  102M 5s\n",
      "  4650K .......... .......... .......... .......... ..........  8%  118M 5s\n",
      "  4700K .......... .......... .......... .......... ..........  8%  103M 5s\n",
      "  4750K .......... .......... .......... .......... ..........  9%  118M 5s\n",
      "  4800K .......... .......... .......... .......... ..........  9%  121M 5s\n",
      "  4850K .......... .......... .......... .......... ..........  9%  102M 5s\n",
      "  4900K .......... .......... .......... .......... ..........  9%  119M 5s\n",
      "  4950K .......... .......... .......... .......... ..........  9%  102M 5s\n",
      "  5000K .......... .......... .......... .......... ..........  9%  123M 5s\n",
      "  5050K .......... .......... .......... .......... ..........  9%  117M 5s\n",
      "  5100K .......... .......... .......... .......... ..........  9%  103M 5s\n",
      "  5150K .......... .......... .......... .......... ..........  9%  119M 5s\n",
      "  5200K .......... .......... .......... .......... ..........  9%  101M 5s\n",
      "  5250K .......... .......... .......... .......... ..........  9%  118M 5s\n",
      "  5300K .......... .......... .......... .......... .......... 10%  120M 5s\n",
      "  5350K .......... .......... .......... .......... .......... 10% 99,9M 4s\n",
      "  5400K .......... .......... .......... .......... .......... 10% 69,3M 4s\n",
      "  5450K .......... .......... .......... .......... .......... 10%  121M 4s\n",
      "  5500K .......... .......... .......... .......... .......... 10%  186M 4s\n",
      "  5550K .......... .......... .......... .......... .......... 10%  155M 4s\n",
      "  5600K .......... .......... .......... .......... .......... 10%  104M 4s\n",
      "  5650K .......... .......... .......... .......... .......... 10%  117M 4s\n",
      "  5700K .......... .......... .......... .......... .......... 10%  102M 4s\n",
      "  5750K .......... .......... .......... .......... .......... 10%  121M 4s\n",
      "  5800K .......... .......... .......... .......... .......... 10%  121M 4s\n",
      "  5850K .......... .......... .......... .......... .......... 11%  104M 4s\n",
      "  5900K .......... .......... .......... .......... .......... 11%  117M 4s\n",
      "  5950K .......... .......... .......... .......... .......... 11%  102M 4s\n",
      "  6000K .......... .......... .......... .......... .......... 11%  121M 4s\n",
      "  6050K .......... .......... .......... .......... .......... 11%  117M 4s\n",
      "  6100K .......... .......... .......... .......... .......... 11%  102M 4s\n",
      "  6150K .......... .......... .......... .......... .......... 11%  121M 4s\n",
      "  6200K .......... .......... .......... .......... .......... 11%  102M 4s\n",
      "  6250K .......... .......... .......... .......... .......... 11%  117M 4s\n",
      "  6300K .......... .......... .......... .......... .......... 11%  121M 4s\n",
      "  6350K .......... .......... .......... .......... .......... 12%  102M 4s\n",
      "  6400K .......... .......... .......... .......... .......... 12%  121M 4s\n",
      "  6450K .......... .......... .......... .......... .......... 12%  102M 4s\n",
      "  6500K .......... .......... .......... .......... .......... 12%  118M 4s\n",
      "  6550K .......... .......... .......... .......... .......... 12%  121M 4s\n",
      "  6600K .......... .......... .......... .......... .......... 12% 7,31M 4s\n",
      "  6650K .......... .......... .......... .......... .......... 12% 13,4M 4s\n",
      "  6700K .......... .......... .......... .......... .......... 12%  102M 4s\n",
      "  6750K .......... .......... .......... .......... .......... 12% 86,4M 4s\n",
      "  6800K .......... .......... .......... .......... .......... 12% 55,1M 4s\n",
      "  6850K .......... .......... .......... .......... .......... 12% 8,48M 4s\n",
      "  6900K .......... .......... .......... .......... .......... 13% 6,30M 4s\n",
      "  6950K .......... .......... .......... .......... .......... 13% 4,07M 4s\n",
      "  7000K .......... .......... .......... .......... .......... 13% 34,8M 4s\n",
      "  7050K .......... .......... .......... .......... .......... 13% 87,8M 4s\n",
      "  7100K .......... .......... .......... .......... .......... 13% 98,8M 4s\n",
      "  7150K .......... .......... .......... .......... .......... 13%  118M 4s\n",
      "  7200K .......... .......... .......... .......... .......... 13% 97,2M 4s\n",
      "  7250K .......... .......... .......... .......... .......... 13%  119M 3s\n",
      "  7300K .......... .......... .......... .......... .......... 13%  111M 3s\n",
      "  7350K .......... .......... .......... .......... .......... 13% 94,6M 3s\n",
      "  7400K .......... .......... .......... .......... .......... 14%  113M 3s\n",
      "  7450K .......... .......... .......... .......... .......... 14%  102M 3s\n",
      "  7500K .......... .......... .......... .......... .......... 14%  115M 3s\n",
      "  7550K .......... .......... .......... .......... .......... 14%  114M 3s\n",
      "  7600K .......... .......... .......... .......... .......... 14%  115M 3s\n",
      "  7650K .......... .......... .......... .......... .......... 14%  117M 3s\n",
      "  7700K .......... .......... .......... .......... .......... 14%  100M 3s\n",
      "  7750K .......... .......... .......... .......... .......... 14%  122M 3s\n",
      "  7800K .......... .......... .......... .......... .......... 14%  119M 3s\n",
      "  7850K .......... .......... .......... .......... .......... 14%  102M 3s\n",
      "  7900K .......... .......... .......... .......... .......... 14%  121M 3s\n",
      "  7950K .......... .......... .......... .......... .......... 15%  102M 3s\n",
      "  8000K .......... .......... .......... .......... .......... 15% 13,0M 3s\n",
      "  8050K .......... .......... .......... .......... .......... 15% 48,2M 3s\n",
      "  8100K .......... .......... .......... .......... .......... 15%  109K 6s\n",
      "  8150K .......... .......... .......... .......... .......... 15%  166M 6s\n",
      "  8200K .......... .......... .......... .......... .......... 15%  148M 6s\n",
      "  8250K .......... .......... .......... .......... .......... 15%  183M 6s\n",
      "  8300K .......... .......... .......... .......... .......... 15%  178M 6s\n",
      "  8350K .......... .......... .......... .......... .......... 15%  153M 5s\n",
      "  8400K .......... .......... .......... .......... .......... 15% 73,0M 5s\n",
      "  8450K .......... .......... .......... .......... .......... 15%  155M 5s\n",
      "  8500K .......... .......... .......... .......... .......... 16%  183M 5s\n",
      "  8550K .......... .......... .......... .......... .......... 16%  192M 5s\n",
      "  8600K .......... .......... .......... .......... .......... 16%  155M 5s\n",
      "  8650K .......... .......... .......... .......... .......... 16%  184M 5s\n",
      "  8700K .......... .......... .......... .......... .......... 16%  151M 5s\n",
      "  8750K .......... .......... .......... .......... .......... 16%  185M 5s\n",
      "  8800K .......... .......... .......... .......... .......... 16%  184M 5s\n",
      "  8850K .......... .......... .......... .......... .......... 16%  162M 5s\n",
      "  8900K .......... .......... .......... .......... .......... 16%  171M 5s\n",
      "  8950K .......... .......... .......... .......... .......... 16%  153M 5s\n",
      "  9000K .......... .......... .......... .......... .......... 17%  162M 5s\n",
      "  9050K .......... .......... .......... .......... .......... 17%  175M 5s\n",
      "  9100K .......... .......... .......... .......... .......... 17%  158M 5s\n",
      "  9150K .......... .......... .......... .......... .......... 17%  190M 5s\n",
      "  9200K .......... .......... .......... .......... .......... 17%  152M 5s\n",
      "  9250K .......... .......... .......... .......... .......... 17%  182M 5s\n",
      "  9300K .......... .......... .......... .......... .......... 17%  187M 5s\n",
      "  9350K .......... .......... .......... .......... .......... 17%  160M 5s\n",
      "  9400K .......... .......... .......... .......... .......... 17%  182M 5s\n",
      "  9450K .......... .......... .......... .......... .......... 17%  155M 5s\n",
      "  9500K .......... .......... .......... .......... .......... 17%  183M 5s\n",
      "  9550K .......... .......... .......... .......... .......... 18%  184M 5s\n",
      "  9600K .......... .......... .......... .......... .......... 18%  159M 5s\n",
      "  9650K .......... .......... .......... .......... .......... 18%  203M 5s\n",
      "  9700K .......... .......... .......... .......... .......... 18%  119M 5s\n",
      "  9750K .......... .......... .......... .......... .......... 18%  198M 5s\n",
      "  9800K .......... .......... .......... .......... .......... 18%  204M 5s\n",
      "  9850K .......... .......... .......... .......... .......... 18%  160M 5s\n",
      "  9900K .......... .......... .......... .......... .......... 18%  186M 5s\n",
      "  9950K .......... .......... .......... .......... .......... 18%  154M 4s\n",
      " 10000K .......... .......... .......... .......... .......... 18%  182M 4s\n",
      " 10050K .......... .......... .......... .......... .......... 18%  186M 4s\n",
      " 10100K .......... .......... .......... .......... .......... 19%  207M 4s\n",
      " 10150K .......... .......... .......... .......... .......... 19%  252M 4s\n",
      " 10200K .......... .......... .......... .......... .......... 19%  201M 4s\n",
      " 10250K .......... .......... .......... .......... .......... 19%  239M 4s\n",
      " 10300K .......... .......... .......... .......... .......... 19%  229M 4s\n",
      " 10350K .......... .......... .......... .......... .......... 19%  229M 4s\n",
      " 10400K .......... .......... .......... .......... .......... 19%  259M 4s\n",
      " 10450K .......... .......... .......... .......... .......... 19%  137M 4s\n",
      " 10500K .......... .......... .......... .......... .......... 19%  232M 4s\n",
      " 10550K .......... .......... .......... .......... .......... 19%  300M 4s\n",
      " 10600K .......... .......... .......... .......... .......... 20%  262M 4s\n",
      " 10650K .......... .......... .......... .......... .......... 20%  313M 4s\n",
      " 10700K .......... .......... .......... .......... .......... 20%  252M 4s\n",
      " 10750K .......... .......... .......... .......... .......... 20%  292M 4s\n",
      " 10800K .......... .......... .......... .......... .......... 20%  309M 4s\n",
      " 10850K .......... .......... .......... .......... .......... 20%  261M 4s\n",
      " 10900K .......... .......... .......... .......... .......... 20%  305M 4s\n",
      " 10950K .......... .......... .......... .......... .......... 20%  307M 4s\n",
      " 11000K .......... .......... .......... .......... .......... 20%  359M 4s\n",
      " 11050K .......... .......... .......... .......... .......... 20%  466K 4s\n",
      " 11100K .......... .......... .......... .......... .......... 20%  795K 5s\n",
      " 11150K .......... .......... .......... .......... .......... 21%  813K 5s\n",
      " 11200K .......... .......... .......... .......... .......... 21%  843K 5s\n",
      " 11250K .......... .......... .......... .......... .......... 21% 23,0M 5s\n",
      " 11300K .......... .......... .......... .......... .......... 21% 74,4M 5s\n",
      " 11350K .......... .......... .......... .......... .......... 21% 69,9M 5s\n",
      " 11400K .......... .......... .......... .......... .......... 21%  863K 5s\n",
      " 11450K .......... .......... .......... .......... .......... 21% 23,1M 5s\n",
      " 11500K .......... .......... .......... .......... .......... 21% 65,5M 5s\n",
      " 11550K .......... .......... .......... .......... .......... 21% 40,6M 5s\n",
      " 11600K .......... .......... .......... .......... .......... 21% 52,3M 5s\n",
      " 11650K .......... .......... .......... .......... .......... 21% 73,5M 5s\n",
      " 11700K .......... .......... .......... .......... .......... 22%  112M 5s\n",
      " 11750K .......... .......... .......... .......... .......... 22%  906K 5s\n",
      " 11800K .......... .......... .......... .......... .......... 22% 41,4M 5s\n",
      " 11850K .......... .......... .......... .......... .......... 22% 40,5M 5s\n",
      " 11900K .......... .......... .......... .......... .......... 22% 83,4M 5s\n",
      " 11950K .......... .......... .......... .......... .......... 22% 52,3M 5s\n",
      " 12000K .......... .......... .......... .......... .......... 22% 61,4M 5s\n",
      " 12050K .......... .......... .......... .......... .......... 22% 64,6M 5s\n",
      " 12100K .......... .......... .......... .......... .......... 22% 60,0M 5s\n",
      " 12150K .......... .......... .......... .......... .......... 22% 70,5M 5s\n",
      " 12200K .......... .......... .......... .......... .......... 23% 66,8M 5s\n",
      " 12250K .......... .......... .......... .......... .......... 23%  175M 5s\n",
      " 12300K .......... .......... .......... .......... .......... 23%  167M 5s\n",
      " 12350K .......... .......... .......... .......... .......... 23%  142M 5s\n",
      " 12400K .......... .......... .......... .......... .......... 23% 77,3M 5s\n",
      " 12450K .......... .......... .......... .......... .......... 23%  120M 5s\n",
      " 12500K .......... .......... .......... .......... .......... 23%  955K 5s\n",
      " 12550K .......... .......... .......... .......... .......... 23%  118M 5s\n",
      " 12600K .......... .......... .......... .......... .......... 23% 34,1M 5s\n",
      " 12650K .......... .......... .......... .......... .......... 23% 67,5M 5s\n",
      " 12700K .......... .......... .......... .......... .......... 23% 64,0M 5s\n",
      " 12750K .......... .......... .......... .......... .......... 24%  107M 5s\n",
      " 12800K .......... .......... .......... .......... .......... 24%  105M 5s\n",
      " 12850K .......... .......... .......... .......... .......... 24% 93,1M 5s\n",
      " 12900K .......... .......... .......... .......... .......... 24%  108M 5s\n",
      " 12950K .......... .......... .......... .......... .......... 24% 96,6M 5s\n",
      " 13000K .......... .......... .......... .......... .......... 24%  105M 5s\n",
      " 13050K .......... .......... .......... .......... .......... 24% 99,0M 5s\n",
      " 13100K .......... .......... .......... .......... .......... 24% 90,8M 5s\n",
      " 13150K .......... .......... .......... .......... .......... 24%  108M 5s\n",
      " 13200K .......... .......... .......... .......... .......... 24% 89,6M 5s\n",
      " 13250K .......... .......... .......... .......... .......... 24%  102M 5s\n",
      " 13300K .......... .......... .......... .......... .......... 25%  116M 5s\n",
      " 13350K .......... .......... .......... .......... .......... 25% 93,0M 5s\n",
      " 13400K .......... .......... .......... .......... .......... 25%  111M 5s\n",
      " 13450K .......... .......... .......... .......... .......... 25% 93,6M 4s\n",
      " 13500K .......... .......... .......... .......... .......... 25%  105M 4s\n",
      " 13550K .......... .......... .......... .......... .......... 25%  105M 4s\n",
      " 13600K .......... .......... .......... .......... .......... 25% 99,0M 4s\n",
      " 13650K .......... .......... .......... .......... .......... 25%  122M 4s\n",
      " 13700K .......... .......... .......... .......... .......... 25%  150M 4s\n",
      " 13750K .......... .......... .......... .......... .......... 25%  192M 4s\n",
      " 13800K .......... .......... .......... .......... .......... 26%  200M 4s\n",
      " 13850K .......... .......... .......... .......... .......... 26%  154M 4s\n",
      " 13900K .......... .......... .......... .......... .......... 26% 1,05M 4s\n",
      " 13950K .......... .......... .......... .......... .......... 26% 32,0M 4s\n",
      " 14000K .......... .......... .......... .......... .......... 26% 98,1M 4s\n",
      " 14050K .......... .......... .......... .......... .......... 26% 39,3M 4s\n",
      " 14100K .......... .......... .......... .......... .......... 26% 57,3M 4s\n",
      " 14150K .......... .......... .......... .......... .......... 26% 51,8M 4s\n",
      " 14200K .......... .......... .......... .......... .......... 26% 38,3M 4s\n",
      " 14250K .......... .......... .......... .......... .......... 26% 71,7M 4s\n",
      " 14300K .......... .......... .......... .......... .......... 26%  115M 4s\n",
      " 14350K .......... .......... .......... .......... .......... 27% 96,8M 4s\n",
      " 14400K .......... .......... .......... .......... .......... 27%  116M 4s\n",
      " 14450K .......... .......... .......... .......... .......... 27%  103M 4s\n",
      " 14500K .......... .......... .......... .......... .......... 27%  115M 4s\n",
      " 14550K .......... .......... .......... .......... .......... 27% 65,1M 4s\n",
      " 14600K .......... .......... .......... .......... .......... 27%  105M 4s\n",
      " 14650K .......... .......... .......... .......... .......... 27%  123M 4s\n",
      " 14700K .......... .......... .......... .......... .......... 27%  120M 4s\n",
      " 14750K .......... .......... .......... .......... .......... 27%  143M 4s\n",
      " 14800K .......... .......... .......... .......... .......... 27%  149M 4s\n",
      " 14850K .......... .......... .......... .......... .......... 28%  125M 4s\n",
      " 14900K .......... .......... .......... .......... .......... 28%  156M 4s\n",
      " 14950K .......... .......... .......... .......... .......... 28%  143M 4s\n",
      " 15000K .......... .......... .......... .......... .......... 28%  183M 4s\n",
      " 15050K .......... .......... .......... .......... .......... 28%  183M 4s\n",
      " 15100K .......... .......... .......... .......... .......... 28%  153M 4s\n",
      " 15150K .......... .......... .......... .......... .......... 28%  199M 4s\n",
      " 15200K .......... .......... .......... .......... .......... 28%  182M 4s\n",
      " 15250K .......... .......... .......... .......... .......... 28%  205M 4s\n",
      " 15300K .......... .......... .......... .......... .......... 28%  217M 4s\n",
      " 15350K .......... .......... .......... .......... .......... 28%  174M 4s\n",
      " 15400K .......... .......... .......... .......... .......... 29%  220M 4s\n",
      " 15450K .......... .......... .......... .......... .......... 29%  171M 4s\n",
      " 15500K .......... .......... .......... .......... .......... 29%  169M 4s\n",
      " 15550K .......... .......... .......... .......... .......... 29%  165M 4s\n",
      " 15600K .......... .......... .......... .......... .......... 29%  113M 4s\n",
      " 15650K .......... .......... .......... .......... .......... 29%  123M 4s\n",
      " 15700K .......... .......... .......... .......... .......... 29%  102M 4s\n",
      " 15750K .......... .......... .......... .......... .......... 29%  118M 4s\n",
      " 15800K .......... .......... .......... .......... .......... 29%  122M 4s\n",
      " 15850K .......... .......... .......... .......... .......... 29%  102M 4s\n",
      " 15900K .......... .......... .......... .......... .......... 29%  118M 4s\n",
      " 15950K .......... .......... .......... .......... .......... 30%  102M 4s\n",
      " 16000K .......... .......... .......... .......... .......... 30%  121M 4s\n",
      " 16050K .......... .......... .......... .......... .......... 30%  121M 4s\n",
      " 16100K .......... .......... .......... .......... .......... 30%  102M 4s\n",
      " 16150K .......... .......... .......... .......... .......... 30%  117M 4s\n",
      " 16200K .......... .......... .......... .......... .......... 30%  103M 4s\n",
      " 16250K .......... .......... .......... .......... .......... 30%  121M 4s\n",
      " 16300K .......... .......... .......... .......... .......... 30%  117M 4s\n",
      " 16350K .......... .......... .......... .......... .......... 30%  102M 4s\n",
      " 16400K .......... .......... .......... .......... .......... 30%  121M 4s\n",
      " 16450K .......... .......... .......... .......... .......... 31%  102M 4s\n",
      " 16500K .......... .......... .......... .......... .......... 31%  118M 4s\n",
      " 16550K .......... .......... .......... .......... .......... 31%  121M 4s\n",
      " 16600K .......... .......... .......... .......... .......... 31%  102M 4s\n",
      " 16650K .......... .......... .......... .......... .......... 31%  121M 4s\n",
      " 16700K .......... .......... .......... .......... .......... 31%  102M 3s\n",
      " 16750K .......... .......... .......... .......... .......... 31% 2,12M 4s\n",
      " 16800K .......... .......... .......... .......... .......... 31% 3,82M 4s\n",
      " 16850K .......... .......... .......... .......... .......... 31% 41,6M 4s\n",
      " 16900K .......... .......... .......... .......... .......... 31%  113M 4s\n",
      " 16950K .......... .......... .......... .......... .......... 31% 66,5M 3s\n",
      " 17000K .......... .......... .......... .......... .......... 32% 74,1M 3s\n",
      " 17050K .......... .......... .......... .......... .......... 32%  114M 3s\n",
      " 17100K .......... .......... .......... .......... .......... 32% 95,5M 3s\n",
      " 17150K .......... .......... .......... .......... .......... 32%  114M 3s\n",
      " 17200K .......... .......... .......... .......... .......... 32% 83,6M 3s\n",
      " 17250K .......... .......... .......... .......... .......... 32%  114M 3s\n",
      " 17300K .......... .......... .......... .......... .......... 32%  114M 3s\n",
      " 17350K .......... .......... .......... .......... .......... 32%  101M 3s\n",
      " 17400K .......... .......... .......... .......... .......... 32% 59,2M 3s\n",
      " 17450K .......... .......... .......... .......... .......... 32% 56,0M 3s\n",
      " 17500K .......... .......... .......... .......... .......... 32%  113M 3s\n",
      " 17550K .......... .......... .......... .......... .......... 33%  114M 3s\n",
      " 17600K .......... .......... .......... .......... .......... 33%  111M 3s\n",
      " 17650K .......... .......... .......... .......... .......... 33%  130M 3s\n",
      " 17700K .......... .......... .......... .......... .......... 33% 79,9M 3s\n",
      " 17750K .......... .......... .......... .......... .......... 33%  116M 3s\n",
      " 17800K .......... .......... .......... .......... .......... 33%  120M 3s\n",
      " 17850K .......... .......... .......... .......... .......... 33%  102M 3s\n",
      " 17900K .......... .......... .......... .......... .......... 33%  151M 3s\n",
      " 17950K .......... .......... .......... .......... .......... 33%  114M 3s\n",
      " 18000K .......... .......... .......... .......... .......... 33%  120M 3s\n",
      " 18050K .......... .......... .......... .......... .......... 34%  119M 3s\n",
      " 18100K .......... .......... .......... .......... .......... 34%  103M 3s\n",
      " 18150K .......... .......... .......... .......... .......... 34%  119M 3s\n",
      " 18200K .......... .......... .......... .......... .......... 34%  105M 3s\n",
      " 18250K .......... .......... .......... .......... .......... 34%  118M 3s\n",
      " 18300K .......... .......... .......... .......... .......... 34%  118M 3s\n",
      " 18350K .......... .......... .......... .......... .......... 34%  103M 3s\n",
      " 18400K .......... .......... .......... .......... .......... 34%  122M 3s\n",
      " 18450K .......... .......... .......... .......... .......... 34%  102M 3s\n",
      " 18500K .......... .......... .......... .......... .......... 34%  117M 3s\n",
      " 18550K .......... .......... .......... .......... .......... 34%  121M 3s\n",
      " 18600K .......... .......... .......... .......... .......... 35% 87,8M 3s\n",
      " 18650K .......... .......... .......... .......... .......... 35%  144M 3s\n",
      " 18700K .......... .......... .......... .......... .......... 35%  104M 3s\n",
      " 18750K .......... .......... .......... .......... .......... 35%  118M 3s\n",
      " 18800K .......... .......... .......... .......... .......... 35%  123M 3s\n",
      " 18850K .......... .......... .......... .......... .......... 35%  103M 3s\n",
      " 18900K .......... .......... .......... .......... .......... 35%  116M 3s\n",
      " 18950K .......... .......... .......... .......... .......... 35%  104M 3s\n",
      " 19000K .......... .......... .......... .......... .......... 35%  120M 3s\n",
      " 19050K .......... .......... .......... .......... .......... 35%  118M 3s\n",
      " 19100K .......... .......... .......... .......... .......... 35%  103M 3s\n",
      " 19150K .......... .......... .......... .......... .......... 36%  121M 3s\n",
      " 19200K .......... .......... .......... .......... .......... 36%  103M 3s\n",
      " 19250K .......... .......... .......... .......... .......... 36%  120M 3s\n",
      " 19300K .......... .......... .......... .......... .......... 36%  117M 3s\n",
      " 19350K .......... .......... .......... .......... .......... 36%  102M 3s\n",
      " 19400K .......... .......... .......... .......... .......... 36%  120M 3s\n",
      " 19450K .......... .......... .......... .......... .......... 36%  102M 3s\n",
      " 19500K .......... .......... .......... .......... .......... 36%  117M 3s\n",
      " 19550K .......... .......... .......... .......... .......... 36%  121M 3s\n",
      " 19600K .......... .......... .......... .......... .......... 36%  102M 3s\n",
      " 19650K .......... .......... .......... .......... .......... 37%  117M 3s\n",
      " 19700K .......... .......... .......... .......... .......... 37%  102M 3s\n",
      " 19750K .......... .......... .......... .......... .......... 37%  121M 3s\n",
      " 19800K .......... .......... .......... .......... .......... 37% 2,38M 3s\n",
      " 19850K .......... .......... .......... .......... .......... 37% 3,42M 3s\n",
      " 19900K .......... .......... .......... .......... .......... 37% 80,7M 3s\n",
      " 19950K .......... .......... .......... .......... .......... 37% 73,4M 3s\n",
      " 20000K .......... .......... .......... .......... .......... 37% 58,2M 3s\n",
      " 20050K .......... .......... .......... .......... .......... 37%  135M 3s\n",
      " 20100K .......... .......... .......... .......... .......... 37%  121M 3s\n",
      " 20150K .......... .......... .......... .......... .......... 37%  148M 3s\n",
      " 20200K .......... .......... .......... .......... .......... 38%  109M 3s\n",
      " 20250K .......... .......... .......... .......... .......... 38%  115M 3s\n",
      " 20300K .......... .......... .......... .......... .......... 38%  127M 3s\n",
      " 20350K .......... .......... .......... .......... .......... 38%  103M 3s\n",
      " 20400K .......... .......... .......... .......... .......... 38%  118M 3s\n",
      " 20450K .......... .......... .......... .......... .......... 38%  104M 3s\n",
      " 20500K .......... .......... .......... .......... .......... 38%  117M 3s\n",
      " 20550K .......... .......... .......... .......... .......... 38% 97,5M 3s\n",
      " 20600K .......... .......... .......... .......... .......... 38% 30,9M 3s\n",
      " 20650K .......... .......... .......... .......... .......... 38% 79,8M 3s\n",
      " 20700K .......... .......... .......... .......... .......... 38%  141M 3s\n",
      " 20750K .......... .......... .......... .......... .......... 39%  149M 3s\n",
      " 20800K .......... .......... .......... .......... .......... 39%  120M 3s\n",
      " 20850K .......... .......... .......... .......... .......... 39%  101M 3s\n",
      " 20900K .......... .......... .......... .......... .......... 39%  117M 3s\n",
      " 20950K .......... .......... .......... .......... .......... 39%  102M 3s\n",
      " 21000K .......... .......... .......... .......... .......... 39%  122M 3s\n",
      " 21050K .......... .......... .......... .......... .......... 39%  120M 3s\n",
      " 21100K .......... .......... .......... .......... .......... 39%  101M 3s\n",
      " 21150K .......... .......... .......... .......... .......... 39%  121M 3s\n",
      " 21200K .......... .......... .......... .......... .......... 39% 99,4M 3s\n",
      " 21250K .......... .......... .......... .......... .......... 40%  124M 3s\n",
      " 21300K .......... .......... .......... .......... .......... 40%  119M 3s\n",
      " 21350K .......... .......... .......... .......... .......... 40%  103M 3s\n",
      " 21400K .......... .......... .......... .......... .......... 40%  122M 3s\n",
      " 21450K .......... .......... .......... .......... .......... 40%  102M 3s\n",
      " 21500K .......... .......... .......... .......... .......... 40%  117M 3s\n",
      " 21550K .......... .......... .......... .......... .......... 40%  121M 3s\n",
      " 21600K .......... .......... .......... .......... .......... 40%  103M 3s\n",
      " 21650K .......... .......... .......... .......... .......... 40%  117M 2s\n",
      " 21700K .......... .......... .......... .......... .......... 40%  102M 2s\n",
      " 21750K .......... .......... .......... .......... .......... 40%  121M 2s\n",
      " 21800K .......... .......... .......... .......... .......... 41%  117M 2s\n",
      " 21850K .......... .......... .......... .......... .......... 41%  102M 2s\n",
      " 21900K .......... .......... .......... .......... .......... 41%  122M 2s\n",
      " 21950K .......... .......... .......... .......... .......... 41%  102M 2s\n",
      " 22000K .......... .......... .......... .......... .......... 41%  121M 2s\n",
      " 22050K .......... .......... .......... .......... .......... 41%  118M 2s\n",
      " 22100K .......... .......... .......... .......... .......... 41%  102M 2s\n",
      " 22150K .......... .......... .......... .......... .......... 41%  122M 2s\n",
      " 22200K .......... .......... .......... .......... .......... 41%  102M 2s\n",
      " 22250K .......... .......... .......... .......... .......... 41%  118M 2s\n",
      " 22300K .......... .......... .......... .......... .......... 42%  122M 2s\n",
      " 22350K .......... .......... .......... .......... .......... 42%  102M 2s\n",
      " 22400K .......... .......... .......... .......... .......... 42%  117M 2s\n",
      " 22450K .......... .......... .......... .......... .......... 42%  105M 2s\n",
      " 22500K .......... .......... .......... .......... .......... 42%  117M 2s\n",
      " 22550K .......... .......... .......... .......... .......... 42%  122M 2s\n",
      " 22600K .......... .......... .......... .......... .......... 42%  102M 2s\n",
      " 22650K .......... .......... .......... .......... .......... 42%  117M 2s\n",
      " 22700K .......... .......... .......... .......... .......... 42% 99,0M 2s\n",
      " 22750K .......... .......... .......... .......... .......... 42%  126M 2s\n",
      " 22800K .......... .......... .......... .......... .......... 42%  117M 2s\n",
      " 22850K .......... .......... .......... .......... .......... 43% 2,18M 2s\n",
      " 22900K .......... .......... .......... .......... .......... 43% 4,44M 2s\n",
      " 22950K .......... .......... .......... .......... .......... 43% 29,4M 2s\n",
      " 23000K .......... .......... .......... .......... .......... 43% 37,6M 2s\n",
      " 23050K .......... .......... .......... .......... .......... 43%  117M 2s\n",
      " 23100K .......... .......... .......... .......... .......... 43% 98,6M 2s\n",
      " 23150K .......... .......... .......... .......... .......... 43%  114M 2s\n",
      " 23200K .......... .......... .......... .......... .......... 43% 92,7M 2s\n",
      " 23250K .......... .......... .......... .......... .......... 43%  111M 2s\n",
      " 23300K .......... .......... .......... .......... .......... 43%  114M 2s\n",
      " 23350K .......... .......... .......... .......... .......... 43% 98,0M 2s\n",
      " 23400K .......... .......... .......... .......... .......... 44%  113M 2s\n",
      " 23450K .......... .......... .......... .......... .......... 44% 97,7M 2s\n",
      " 23500K .......... .......... .......... .......... .......... 44%  113M 2s\n",
      " 23550K .......... .......... .......... .......... .......... 44%  115M 2s\n",
      " 23600K .......... .......... .......... .......... .......... 44% 96,7M 2s\n",
      " 23650K .......... .......... .......... .......... .......... 44%  115M 2s\n",
      " 23700K .......... .......... .......... .......... .......... 44%  102M 2s\n",
      " 23750K .......... .......... .......... .......... .......... 44% 32,5M 2s\n",
      " 23800K .......... .......... .......... .......... .......... 44%  113M 2s\n",
      " 23850K .......... .......... .......... .......... .......... 44% 98,9M 2s\n",
      " 23900K .......... .......... .......... .......... .......... 45%  113M 2s\n",
      " 23950K .......... .......... .......... .......... .......... 45% 96,2M 2s\n",
      " 24000K .......... .......... .......... .......... .......... 45%  108M 2s\n",
      " 24050K .......... .......... .......... .......... .......... 45%  111M 2s\n",
      " 24100K .......... .......... .......... .......... .......... 45% 92,0M 2s\n",
      " 24150K .......... .......... .......... .......... .......... 45%  138M 2s\n",
      " 24200K .......... .......... .......... .......... .......... 45%  112M 2s\n",
      " 24250K .......... .......... .......... .......... .......... 45%  131M 2s\n",
      " 24300K .......... .......... .......... .......... .......... 45%  135M 2s\n",
      " 24350K .......... .......... .......... .......... .......... 45%  112M 2s\n",
      " 24400K .......... .......... .......... .......... .......... 45%  116M 2s\n",
      " 24450K .......... .......... .......... .......... .......... 46%  102M 2s\n",
      " 24500K .......... .......... .......... .......... .......... 46%  121M 2s\n",
      " 24550K .......... .......... .......... .......... .......... 46%  120M 2s\n",
      " 24600K .......... .......... .......... .......... .......... 46%  104M 2s\n",
      " 24650K .......... .......... .......... .......... .......... 46%  118M 2s\n",
      " 24700K .......... .......... .......... .......... .......... 46%  102M 2s\n",
      " 24750K .......... .......... .......... .......... .......... 46% 94,4M 2s\n",
      " 24800K .......... .......... .......... .......... .......... 46%  161M 2s\n",
      " 24850K .......... .......... .......... .......... .......... 46%  102M 2s\n",
      " 24900K .......... .......... .......... .......... .......... 46%  120M 2s\n",
      " 24950K .......... .......... .......... .......... .......... 46%  103M 2s\n",
      " 25000K .......... .......... .......... .......... .......... 47%  120M 2s\n",
      " 25050K .......... .......... .......... .......... .......... 47%  120M 2s\n",
      " 25100K .......... .......... .......... .......... .......... 47%  103M 2s\n",
      " 25150K .......... .......... .......... .......... .......... 47%  120M 2s\n",
      " 25200K .......... .......... .......... .......... .......... 47%  103M 2s\n",
      " 25250K .......... .......... .......... .......... .......... 47%  117M 2s\n",
      " 25300K .......... .......... .......... .......... .......... 47%  120M 2s\n",
      " 25350K .......... .......... .......... .......... .......... 47%  104M 2s\n",
      " 25400K .......... .......... .......... .......... .......... 47%  117M 2s\n",
      " 25450K .......... .......... .......... .......... .......... 47%  102M 2s\n",
      " 25500K .......... .......... .......... .......... .......... 48%  121M 2s\n",
      " 25550K .......... .......... .......... .......... .......... 48%  117M 2s\n",
      " 25600K .......... .......... .......... .......... .......... 48%  102M 2s\n",
      " 25650K .......... .......... .......... .......... .......... 48%  121M 2s\n",
      " 25700K .......... .......... .......... .......... .......... 48%  102M 2s\n",
      " 25750K .......... .......... .......... .......... .......... 48%  121M 2s\n",
      " 25800K .......... .......... .......... .......... .......... 48%  118M 2s\n",
      " 25850K .......... .......... .......... .......... .......... 48%  102M 2s\n",
      " 25900K .......... .......... .......... .......... .......... 48% 2,37M 2s\n",
      " 25950K .......... .......... .......... .......... .......... 48% 3,82M 2s\n",
      " 26000K .......... .......... .......... .......... .......... 48% 56,4M 2s\n",
      " 26050K .......... .......... .......... .......... .......... 49% 35,6M 2s\n",
      " 26100K .......... .......... .......... .......... .......... 49% 47,2M 2s\n",
      " 26150K .......... .......... .......... .......... .......... 49% 84,4M 2s\n",
      " 26200K .......... .......... .......... .......... .......... 49% 56,2M 2s\n",
      " 26250K .......... .......... .......... .......... .......... 49%  129M 2s\n",
      " 26300K .......... .......... .......... .......... .......... 49%  145M 2s\n",
      " 26350K .......... .......... .......... .......... .......... 49%  118M 2s\n",
      " 26400K .......... .......... .......... .......... .......... 49%  143M 2s\n",
      " 26450K .......... .......... .......... .......... .......... 49%  117M 2s\n",
      " 26500K .......... .......... .......... .......... .......... 49%  142M 2s\n",
      " 26550K .......... .......... .......... .......... .......... 49%  144M 2s\n",
      " 26600K .......... .......... .......... .......... .......... 50%  101M 2s\n",
      " 26650K .......... .......... .......... .......... .......... 50%  122M 2s\n",
      " 26700K .......... .......... .......... .......... .......... 50%  102M 2s\n",
      " 26750K .......... .......... .......... .......... .......... 50%  122M 2s\n",
      " 26800K .......... .......... .......... .......... .......... 50% 57,5M 2s\n",
      " 26850K .......... .......... .......... .......... .......... 50%  158M 2s\n",
      " 26900K .......... .......... .......... .......... .......... 50% 84,2M 2s\n",
      " 26950K .......... .......... .......... .......... .......... 50% 42,0M 2s\n",
      " 27000K .......... .......... .......... .......... .......... 50%  118M 2s\n",
      " 27050K .......... .......... .......... .......... .......... 50%  120M 2s\n",
      " 27100K .......... .......... .......... .......... .......... 51%  103M 2s\n",
      " 27150K .......... .......... .......... .......... .......... 51%  116M 2s\n",
      " 27200K .......... .......... .......... .......... .......... 51%  101M 2s\n",
      " 27250K .......... .......... .......... .......... .......... 51%  120M 2s\n",
      " 27300K .......... .......... .......... .......... .......... 51% 95,0M 2s\n",
      " 27350K .......... .......... .......... .......... .......... 51%  135M 2s\n",
      " 27400K .......... .......... .......... .......... .......... 51%  104M 2s\n",
      " 27450K .......... .......... .......... .......... .......... 51%  112M 2s\n",
      " 27500K .......... .......... .......... .......... .......... 51%  125M 2s\n",
      " 27550K .......... .......... .......... .......... .......... 51%  118M 2s\n",
      " 27600K .......... .......... .......... .......... .......... 51%  102M 2s\n",
      " 27650K .......... .......... .......... .......... .......... 52%  120M 2s\n",
      " 27700K .......... .......... .......... .......... .......... 52%  102M 2s\n",
      " 27750K .......... .......... .......... .......... .......... 52%  118M 2s\n",
      " 27800K .......... .......... .......... .......... .......... 52%  122M 2s\n",
      " 27850K .......... .......... .......... .......... .......... 52%  103M 2s\n",
      " 27900K .......... .......... .......... .......... .......... 52%  121M 2s\n",
      " 27950K .......... .......... .......... .......... .......... 52%  102M 2s\n",
      " 28000K .......... .......... .......... .......... .......... 52%  118M 2s\n",
      " 28050K .......... .......... .......... .......... .......... 52%  121M 2s\n",
      " 28100K .......... .......... .......... .......... .......... 52%  103M 2s\n",
      " 28150K .......... .......... .......... .......... .......... 52%  118M 2s\n",
      " 28200K .......... .......... .......... .......... .......... 53%  102M 2s\n",
      " 28250K .......... .......... .......... .......... .......... 53%  117M 2s\n",
      " 28300K .......... .......... .......... .......... .......... 53%  127M 2s\n",
      " 28350K .......... .......... .......... .......... .......... 53%  102M 2s\n",
      " 28400K .......... .......... .......... .......... .......... 53%  117M 2s\n",
      " 28450K .......... .......... .......... .......... .......... 53%  103M 2s\n",
      " 28500K .......... .......... .......... .......... .......... 53%  121M 2s\n",
      " 28550K .......... .......... .......... .......... .......... 53%  117M 2s\n",
      " 28600K .......... .......... .......... .......... .......... 53%  101M 2s\n",
      " 28650K .......... .......... .......... .......... .......... 53%  123M 2s\n",
      " 28700K .......... .......... .......... .......... .......... 54%  102M 2s\n",
      " 28750K .......... .......... .......... .......... .......... 54%  117M 2s\n",
      " 28800K .......... .......... .......... .......... .......... 54%  121M 2s\n",
      " 28850K .......... .......... .......... .......... .......... 54% 89,7M 2s\n",
      " 28900K .......... .......... .......... .......... .......... 54%  145M 2s\n",
      " 28950K .......... .......... .......... .......... .......... 54% 2,93M 2s\n",
      " 29000K .......... .......... .......... .......... .......... 54% 3,32M 2s\n",
      " 29050K .......... .......... .......... .......... .......... 54% 18,3M 2s\n",
      " 29100K .......... .......... .......... .......... .......... 54% 33,0M 2s\n",
      " 29150K .......... .......... .......... .......... .......... 54% 40,3M 2s\n",
      " 29200K .......... .......... .......... .......... .......... 54% 61,6M 2s\n",
      " 29250K .......... .......... .......... .......... .......... 55% 69,6M 2s\n",
      " 29300K .......... .......... .......... .......... .......... 55%  116M 2s\n",
      " 29350K .......... .......... .......... .......... .......... 55%  102M 2s\n",
      " 29400K .......... .......... .......... .......... .......... 55%  122M 2s\n",
      " 29450K .......... .......... .......... .......... .......... 55%  101M 2s\n",
      " 29500K .......... .......... .......... .......... .......... 55%  124M 2s\n",
      " 29550K .......... .......... .......... .......... .......... 55%  116M 2s\n",
      " 29600K .......... .......... .......... .......... .......... 55%  101M 2s\n",
      " 29650K .......... .......... .......... .......... .......... 55%  123M 2s\n",
      " 29700K .......... .......... .......... .......... .......... 55%  103M 1s\n",
      " 29750K .......... .......... .......... .......... .......... 56%  117M 1s\n",
      " 29800K .......... .......... .......... .......... .......... 56%  121M 1s\n",
      " 29850K .......... .......... .......... .......... .......... 56%  102M 1s\n",
      " 29900K .......... .......... .......... .......... .......... 56%  115M 1s\n",
      " 29950K .......... .......... .......... .......... .......... 56% 87,2M 1s\n",
      " 30000K .......... .......... .......... .......... .......... 56% 62,9M 1s\n",
      " 30050K .......... .......... .......... .......... .......... 56%  120M 1s\n",
      " 30100K .......... .......... .......... .......... .......... 56% 51,5M 1s\n",
      " 30150K .......... .......... .......... .......... .......... 56%  119M 1s\n",
      " 30200K .......... .......... .......... .......... .......... 56%  102M 1s\n",
      " 30250K .......... .......... .......... .......... .......... 56%  121M 1s\n",
      " 30300K .......... .......... .......... .......... .......... 57%  117M 1s\n",
      " 30350K .......... .......... .......... .......... .......... 57%  103M 1s\n",
      " 30400K .......... .......... .......... .......... .......... 57%  121M 1s\n",
      " 30450K .......... .......... .......... .......... .......... 57%  103M 1s\n",
      " 30500K .......... .......... .......... .......... .......... 57%  122M 1s\n",
      " 30550K .......... .......... .......... .......... .......... 57%  117M 1s\n",
      " 30600K .......... .......... .......... .......... .......... 57%  101M 1s\n",
      " 30650K .......... .......... .......... .......... .......... 57%  122M 1s\n",
      " 30700K .......... .......... .......... .......... .......... 57%  104M 1s\n",
      " 30750K .......... .......... .......... .......... .......... 57%  118M 1s\n",
      " 30800K .......... .......... .......... .......... .......... 57%  121M 1s\n",
      " 30850K .......... .......... .......... .......... .......... 58%  102M 1s\n",
      " 30900K .......... .......... .......... .......... .......... 58% 88,8M 1s\n",
      " 30950K .......... .......... .......... .......... .......... 58%  144M 1s\n",
      " 31000K .......... .......... .......... .......... .......... 58%  121M 1s\n",
      " 31050K .......... .......... .......... .......... .......... 58%  122M 1s\n",
      " 31100K .......... .......... .......... .......... .......... 58%  101M 1s\n",
      " 31150K .......... .......... .......... .......... .......... 58%  119M 1s\n",
      " 31200K .......... .......... .......... .......... .......... 58%  102M 1s\n",
      " 31250K .......... .......... .......... .......... .......... 58%  121M 1s\n",
      " 31300K .......... .......... .......... .......... .......... 58%  117M 1s\n",
      " 31350K .......... .......... .......... .......... .......... 59%  102M 1s\n",
      " 31400K .......... .......... .......... .......... .......... 59%  121M 1s\n",
      " 31450K .......... .......... .......... .......... .......... 59%  102M 1s\n",
      " 31500K .......... .......... .......... .......... .......... 59%  118M 1s\n",
      " 31550K .......... .......... .......... .......... .......... 59%  120M 1s\n",
      " 31600K .......... .......... .......... .......... .......... 59%  103M 1s\n",
      " 31650K .......... .......... .......... .......... .......... 59%  121M 1s\n",
      " 31700K .......... .......... .......... .......... .......... 59%  102M 1s\n",
      " 31750K .......... .......... .......... .......... .......... 59%  117M 1s\n",
      " 31800K .......... .......... .......... .......... .......... 59%  121M 1s\n",
      " 31850K .......... .......... .......... .......... .......... 59%  102M 1s\n",
      " 31900K .......... .......... .......... .......... .......... 60%  118M 1s\n",
      " 31950K .......... .......... .......... .......... .......... 60%  102M 1s\n",
      " 32000K .......... .......... .......... .......... .......... 60%  121M 1s\n",
      " 32050K .......... .......... .......... .......... .......... 60% 1,64M 1s\n",
      " 32100K .......... .......... .......... .......... .......... 60% 21,6M 1s\n",
      " 32150K .......... .......... .......... .......... .......... 60% 11,8M 1s\n",
      " 32200K .......... .......... .......... .......... .......... 60% 37,7M 1s\n",
      " 32250K .......... .......... .......... .......... .......... 60% 39,6M 1s\n",
      " 32300K .......... .......... .......... .......... .......... 60% 47,3M 1s\n",
      " 32350K .......... .......... .......... .......... .......... 60% 45,8M 1s\n",
      " 32400K .......... .......... .......... .......... .......... 60% 63,5M 1s\n",
      " 32450K .......... .......... .......... .......... .......... 61% 62,4M 1s\n",
      " 32500K .......... .......... .......... .......... .......... 61%  187M 1s\n",
      " 32550K .......... .......... .......... .......... .......... 61%  195M 1s\n",
      " 32600K .......... .......... .......... .......... .......... 61%  166M 1s\n",
      " 32650K .......... .......... .......... .......... .......... 61%  200M 1s\n",
      " 32700K .......... .......... .......... .......... .......... 61%  177M 1s\n",
      " 32750K .......... .......... .......... .......... .......... 61%  214M 1s\n",
      " 32800K .......... .......... .......... .......... .......... 61%  213M 1s\n",
      " 32850K .......... .......... .......... .......... .......... 61%  181M 1s\n",
      " 32900K .......... .......... .......... .......... .......... 61%  210M 1s\n",
      " 32950K .......... .......... .......... .......... .......... 62%  124M 1s\n",
      " 33000K .......... .......... .......... .......... .......... 62%  283M 1s\n",
      " 33050K .......... .......... .......... .......... .......... 62%  292M 1s\n",
      " 33100K .......... .......... .......... .......... .......... 62%  293M 1s\n",
      " 33150K .......... .......... .......... .......... .......... 62%  338M 1s\n",
      " 33200K .......... .......... .......... .......... .......... 62%  230M 1s\n",
      " 33250K .......... .......... .......... .......... .......... 62% 64,7M 1s\n",
      " 33300K .......... .......... .......... .......... .......... 62%  117M 1s\n",
      " 33350K .......... .......... .......... .......... .......... 62%  103M 1s\n",
      " 33400K .......... .......... .......... .......... .......... 62%  121M 1s\n",
      " 33450K .......... .......... .......... .......... .......... 62%  102M 1s\n",
      " 33500K .......... .......... .......... .......... .......... 63%  117M 1s\n",
      " 33550K .......... .......... .......... .......... .......... 63%  121M 1s\n",
      " 33600K .......... .......... .......... .......... .......... 63%  102M 1s\n",
      " 33650K .......... .......... .......... .......... .......... 63%  117M 1s\n",
      " 33700K .......... .......... .......... .......... .......... 63%  106M 1s\n",
      " 33750K .......... .......... .......... .......... .......... 63%  118M 1s\n",
      " 33800K .......... .......... .......... .......... .......... 63%  120M 1s\n",
      " 33850K .......... .......... .......... .......... .......... 63%  102M 1s\n",
      " 33900K .......... .......... .......... .......... .......... 63%  118M 1s\n",
      " 33950K .......... .......... .......... .......... .......... 63%  102M 1s\n",
      " 34000K .......... .......... .......... .......... .......... 63%  121M 1s\n",
      " 34050K .......... .......... .......... .......... .......... 64%  118M 1s\n",
      " 34100K .......... .......... .......... .......... .......... 64%  103M 1s\n",
      " 34150K .......... .......... .......... .......... .......... 64%  121M 1s\n",
      " 34200K .......... .......... .......... .......... .......... 64%  102M 1s\n",
      " 34250K .......... .......... .......... .......... .......... 64%  121M 1s\n",
      " 34300K .......... .......... .......... .......... .......... 64%  117M 1s\n",
      " 34350K .......... .......... .......... .......... .......... 64%  102M 1s\n",
      " 34400K .......... .......... .......... .......... .......... 64%  121M 1s\n",
      " 34450K .......... .......... .......... .......... .......... 64%  102M 1s\n",
      " 34500K .......... .......... .......... .......... .......... 64%  117M 1s\n",
      " 34550K .......... .......... .......... .......... .......... 65%  121M 1s\n",
      " 34600K .......... .......... .......... .......... .......... 65%  102M 1s\n",
      " 34650K .......... .......... .......... .......... .......... 65%  117M 1s\n",
      " 34700K .......... .......... .......... .......... .......... 65%  102M 1s\n",
      " 34750K .......... .......... .......... .......... .......... 65%  121M 1s\n",
      " 34800K .......... .......... .......... .......... .......... 65%  121M 1s\n",
      " 34850K .......... .......... .......... .......... .......... 65%  102M 1s\n",
      " 34900K .......... .......... .......... .......... .......... 65%  117M 1s\n",
      " 34950K .......... .......... .......... .......... .......... 65%  102M 1s\n",
      " 35000K .......... .......... .......... .......... .......... 65%  121M 1s\n",
      " 35050K .......... .......... .......... .......... .......... 65%  117M 1s\n",
      " 35100K .......... .......... .......... .......... .......... 66% 1,70M 1s\n",
      " 35150K .......... .......... .......... .......... .......... 66% 21,0M 1s\n",
      " 35200K .......... .......... .......... .......... .......... 66% 13,8M 1s\n",
      " 35250K .......... .......... .......... .......... .......... 66% 22,6M 1s\n",
      " 35300K .......... .......... .......... .......... .......... 66% 30,8M 1s\n",
      " 35350K .......... .......... .......... .......... .......... 66% 48,7M 1s\n",
      " 35400K .......... .......... .......... .......... .......... 66% 68,2M 1s\n",
      " 35450K .......... .......... .......... .......... .......... 66%  201M 1s\n",
      " 35500K .......... .......... .......... .......... .......... 66%  220M 1s\n",
      " 35550K .......... .......... .......... .......... .......... 66%  109M 1s\n",
      " 35600K .......... .......... .......... .......... .......... 67%  101M 1s\n",
      " 35650K .......... .......... .......... .......... .......... 67%  118M 1s\n",
      " 35700K .......... .......... .......... .......... .......... 67% 62,1M 1s\n",
      " 35750K .......... .......... .......... .......... .......... 67% 56,7M 1s\n",
      " 35800K .......... .......... .......... .......... .......... 67%  121M 1s\n",
      " 35850K .......... .......... .......... .......... .......... 67%  103M 1s\n",
      " 35900K .......... .......... .......... .......... .......... 67%  116M 1s\n",
      " 35950K .......... .......... .......... .......... .......... 67%  102M 1s\n",
      " 36000K .......... .......... .......... .......... .......... 67%  122M 1s\n",
      " 36050K .......... .......... .......... .......... .......... 67%  117M 1s\n",
      " 36100K .......... .......... .......... .......... .......... 67%  103M 1s\n",
      " 36150K .......... .......... .......... .......... .......... 68%  121M 1s\n",
      " 36200K .......... .......... .......... .......... .......... 68%  102M 1s\n",
      " 36250K .......... .......... .......... .......... .......... 68%  117M 1s\n",
      " 36300K .......... .......... .......... .......... .......... 68%  121M 1s\n",
      " 36350K .......... .......... .......... .......... .......... 68%  103M 1s\n",
      " 36400K .......... .......... .......... .......... .......... 68%  120M 1s\n",
      " 36450K .......... .......... .......... .......... .......... 68%  102M 1s\n",
      " 36500K .......... .......... .......... .......... .......... 68%  118M 1s\n",
      " 36550K .......... .......... .......... .......... .......... 68%  122M 1s\n",
      " 36600K .......... .......... .......... .......... .......... 68%  102M 1s\n",
      " 36650K .......... .......... .......... .......... .......... 68%  118M 1s\n",
      " 36700K .......... .......... .......... .......... .......... 69%  102M 1s\n",
      " 36750K .......... .......... .......... .......... .......... 69%  121M 1s\n",
      " 36800K .......... .......... .......... .......... .......... 69%  117M 1s\n",
      " 36850K .......... .......... .......... .......... .......... 69%  102M 1s\n",
      " 36900K .......... .......... .......... .......... .......... 69%  121M 1s\n",
      " 36950K .......... .......... .......... .......... .......... 69%  102M 1s\n",
      " 37000K .......... .......... .......... .......... .......... 69%  121M 1s\n",
      " 37050K .......... .......... .......... .......... .......... 69%  117M 1s\n",
      " 37100K .......... .......... .......... .......... .......... 69%  102M 1s\n",
      " 37150K .......... .......... .......... .......... .......... 69%  120M 1s\n",
      " 37200K .......... .......... .......... .......... .......... 70%  102M 1s\n",
      " 37250K .......... .......... .......... .......... .......... 70%  118M 1s\n",
      " 37300K .......... .......... .......... .......... .......... 70%  121M 1s\n",
      " 37350K .......... .......... .......... .......... .......... 70%  102M 1s\n",
      " 37400K .......... .......... .......... .......... .......... 70%  118M 1s\n",
      " 37450K .......... .......... .......... .......... .......... 70%  105M 1s\n",
      " 37500K .......... .......... .......... .......... .......... 70%  118M 1s\n",
      " 37550K .......... .......... .......... .......... .......... 70%  120M 1s\n",
      " 37600K .......... .......... .......... .......... .......... 70%  102M 1s\n",
      " 37650K .......... .......... .......... .......... .......... 70%  117M 1s\n",
      " 37700K .......... .......... .......... .......... .......... 70%  102M 1s\n",
      " 37750K .......... .......... .......... .......... .......... 71%  121M 1s\n",
      " 37800K .......... .......... .......... .......... .......... 71%  118M 1s\n",
      " 37850K .......... .......... .......... .......... .......... 71%  102M 1s\n",
      " 37900K .......... .......... .......... .......... .......... 71%  121M 1s\n",
      " 37950K .......... .......... .......... .......... .......... 71%  102M 1s\n",
      " 38000K .......... .......... .......... .......... .......... 71%  121M 1s\n",
      " 38050K .......... .......... .......... .......... .......... 71%  117M 1s\n",
      " 38100K .......... .......... .......... .......... .......... 71%  102M 1s\n",
      " 38150K .......... .......... .......... .......... .......... 71% 2,15M 1s\n",
      " 38200K .......... .......... .......... .......... .......... 71% 6,68M 1s\n",
      " 38250K .......... .......... .......... .......... .......... 71%  111M 1s\n",
      " 38300K .......... .......... .......... .......... .......... 72% 13,2M 1s\n",
      " 38350K .......... .......... .......... .......... .......... 72% 20,5M 1s\n",
      " 38400K .......... .......... .......... .......... .......... 72% 92,0M 1s\n",
      " 38450K .......... .......... .......... .......... .......... 72% 75,9M 1s\n",
      " 38500K .......... .......... .......... .......... .......... 72% 97,4M 1s\n",
      " 38550K .......... .......... .......... .......... .......... 72% 90,5M 1s\n",
      " 38600K .......... .......... .......... .......... .......... 72% 61,5M 1s\n",
      " 38650K .......... .......... .......... .......... .......... 72% 79,6M 1s\n",
      " 38700K .......... .......... .......... .......... .......... 72% 89,2M 1s\n",
      " 38750K .......... .......... .......... .......... .......... 72%  143M 1s\n",
      " 38800K .......... .......... .......... .......... .......... 73%  146M 1s\n",
      " 38850K .......... .......... .......... .......... .......... 73% 78,8M 1s\n",
      " 38900K .......... .......... .......... .......... .......... 73% 66,7M 1s\n",
      " 38950K .......... .......... .......... .......... .......... 73%  101M 1s\n",
      " 39000K .......... .......... .......... .......... .......... 73%  119M 1s\n",
      " 39050K .......... .......... .......... .......... .......... 73%  121M 1s\n",
      " 39100K .......... .......... .......... .......... .......... 73% 72,4M 1s\n",
      " 39150K .......... .......... .......... .......... .......... 73%  143M 1s\n",
      " 39200K .......... .......... .......... .......... .......... 73%  123M 1s\n",
      " 39250K .......... .......... .......... .......... .......... 73%  127M 1s\n",
      " 39300K .......... .......... .......... .......... .......... 73%  124M 1s\n",
      " 39350K .......... .......... .......... .......... .......... 74%  102M 1s\n",
      " 39400K .......... .......... .......... .......... .......... 74% 97,4M 1s\n",
      " 39450K .......... .......... .......... .......... .......... 74% 88,7M 1s\n",
      " 39500K .......... .......... .......... .......... .......... 74% 93,3M 1s\n",
      " 39550K .......... .......... .......... .......... .......... 74% 95,0M 1s\n",
      " 39600K .......... .......... .......... .......... .......... 74%  103M 1s\n",
      " 39650K .......... .......... .......... .......... .......... 74%  139M 1s\n",
      " 39700K .......... .......... .......... .......... .......... 74%  121M 1s\n",
      " 39750K .......... .......... .......... .......... .......... 74%  142M 1s\n",
      " 39800K .......... .......... .......... .......... .......... 74%  148M 1s\n",
      " 39850K .......... .......... .......... .......... .......... 74%  103M 1s\n",
      " 39900K .......... .......... .......... .......... .......... 75%  145M 1s\n",
      " 39950K .......... .......... .......... .......... .......... 75%  105M 1s\n",
      " 40000K .......... .......... .......... .......... .......... 75%  122M 1s\n",
      " 40050K .......... .......... .......... .......... .......... 75% 95,7M 1s\n",
      " 40100K .......... .......... .......... .......... .......... 75%  121M 1s\n",
      " 40150K .......... .......... .......... .......... .......... 75%  134M 1s\n",
      " 40200K .......... .......... .......... .......... .......... 75%  104M 1s\n",
      " 40250K .......... .......... .......... .......... .......... 75%  106M 1s\n",
      " 40300K .......... .......... .......... .......... .......... 75%  130M 1s\n",
      " 40350K .......... .......... .......... .......... .......... 75%  101M 1s\n",
      " 40400K .......... .......... .......... .......... .......... 76%  119M 1s\n",
      " 40450K .......... .......... .......... .......... .......... 76%  106M 1s\n",
      " 40500K .......... .......... .......... .......... .......... 76%  122M 1s\n",
      " 40550K .......... .......... .......... .......... .......... 76%  115M 1s\n",
      " 40600K .......... .......... .......... .......... .......... 76% 85,5M 1s\n",
      " 40650K .......... .......... .......... .......... .......... 76%  120M 1s\n",
      " 40700K .......... .......... .......... .......... .......... 76%  101M 1s\n",
      " 40750K .......... .......... .......... .......... .......... 76%  113M 1s\n",
      " 40800K .......... .......... .......... .......... .......... 76%  124M 1s\n",
      " 40850K .......... .......... .......... .......... .......... 76% 83,8M 1s\n",
      " 40900K .......... .......... .......... .......... .......... 76%  101M 1s\n",
      " 40950K .......... .......... .......... .......... .......... 77% 87,4M 1s\n",
      " 41000K .......... .......... .......... .......... .......... 77% 90,9M 1s\n",
      " 41050K .......... .......... .......... .......... .......... 77% 90,3M 1s\n",
      " 41100K .......... .......... .......... .......... .......... 77% 84,9M 1s\n",
      " 41150K .......... .......... .......... .......... .......... 77% 54,7M 1s\n",
      " 41200K .......... .......... .......... .......... .......... 77% 5,07M 1s\n",
      " 41250K .......... .......... .......... .......... .......... 77% 2,74M 1s\n",
      " 41300K .......... .......... .......... .......... .......... 77% 21,7M 1s\n",
      " 41350K .......... .......... .......... .......... .......... 77% 13,0M 1s\n",
      " 41400K .......... .......... .......... .......... .......... 77% 97,8M 1s\n",
      " 41450K .......... .......... .......... .......... .......... 77% 43,3M 1s\n",
      " 41500K .......... .......... .......... .......... .......... 78% 38,8M 1s\n",
      " 41550K .......... .......... .......... .......... .......... 78%  121M 1s\n",
      " 41600K .......... .......... .......... .......... .......... 78%  102M 1s\n",
      " 41650K .......... .......... .......... .......... .......... 78% 83,6M 1s\n",
      " 41700K .......... .......... .......... .......... .......... 78%  102M 1s\n",
      " 41750K .......... .......... .......... .......... .......... 78%  124M 1s\n",
      " 41800K .......... .......... .......... .......... .......... 78%  117M 1s\n",
      " 41850K .......... .......... .......... .......... .......... 78%  101M 1s\n",
      " 41900K .......... .......... .......... .......... .......... 78%  124M 1s\n",
      " 41950K .......... .......... .......... .......... .......... 78% 59,2M 1s\n",
      " 42000K .......... .......... .......... .......... .......... 79% 58,3M 1s\n",
      " 42050K .......... .......... .......... .......... .......... 79% 60,6M 1s\n",
      " 42100K .......... .......... .......... .......... .......... 79%  102M 1s\n",
      " 42150K .......... .......... .......... .......... .......... 79%  117M 1s\n",
      " 42200K .......... .......... .......... .......... .......... 79%  102M 1s\n",
      " 42250K .......... .......... .......... .......... .......... 79%  121M 1s\n",
      " 42300K .......... .......... .......... .......... .......... 79%  122M 1s\n",
      " 42350K .......... .......... .......... .......... .......... 79%  102M 1s\n",
      " 42400K .......... .......... .......... .......... .......... 79%  117M 1s\n",
      " 42450K .......... .......... .......... .......... .......... 79%  102M 1s\n",
      " 42500K .......... .......... .......... .......... .......... 79%  121M 1s\n",
      " 42550K .......... .......... .......... .......... .......... 80%  117M 1s\n",
      " 42600K .......... .......... .......... .......... .......... 80%  102M 1s\n",
      " 42650K .......... .......... .......... .......... .......... 80%  121M 1s\n",
      " 42700K .......... .......... .......... .......... .......... 80%  102M 1s\n",
      " 42750K .......... .......... .......... .......... .......... 80%  118M 1s\n",
      " 42800K .......... .......... .......... .......... .......... 80%  121M 1s\n",
      " 42850K .......... .......... .......... .......... .......... 80%  102M 1s\n",
      " 42900K .......... .......... .......... .......... .......... 80%  121M 1s\n",
      " 42950K .......... .......... .......... .......... .......... 80%  102M 1s\n",
      " 43000K .......... .......... .......... .......... .......... 80%  117M 1s\n",
      " 43050K .......... .......... .......... .......... .......... 81%  121M 1s\n",
      " 43100K .......... .......... .......... .......... .......... 81%  102M 1s\n",
      " 43150K .......... .......... .......... .......... .......... 81%  117M 1s\n",
      " 43200K .......... .......... .......... .......... .......... 81%  102M 0s\n",
      " 43250K .......... .......... .......... .......... .......... 81%  121M 0s\n",
      " 43300K .......... .......... .......... .......... .......... 81%  121M 0s\n",
      " 43350K .......... .......... .......... .......... .......... 81%  102M 0s\n",
      " 43400K .......... .......... .......... .......... .......... 81%  117M 0s\n",
      " 43450K .......... .......... .......... .......... .......... 81%  102M 0s\n",
      " 43500K .......... .......... .......... .......... .......... 81%  121M 0s\n",
      " 43550K .......... .......... .......... .......... .......... 81%  117M 0s\n",
      " 43600K .......... .......... .......... .......... .......... 82%  102M 0s\n",
      " 43650K .......... .......... .......... .......... .......... 82%  121M 0s\n",
      " 43700K .......... .......... .......... .......... .......... 82%  102M 0s\n",
      " 43750K .......... .......... .......... .......... .......... 82%  117M 0s\n",
      " 43800K .......... .......... .......... .......... .......... 82%  121M 0s\n",
      " 43850K .......... .......... .......... .......... .......... 82%  102M 0s\n",
      " 43900K .......... .......... .......... .......... .......... 82%  121M 0s\n",
      " 43950K .......... .......... .......... .......... .......... 82%  102M 0s\n",
      " 44000K .......... .......... .......... .......... .......... 82%  117M 0s\n",
      " 44050K .......... .......... .......... .......... .......... 82%  121M 0s\n",
      " 44100K .......... .......... .......... .......... .......... 82%  102M 0s\n",
      " 44150K .......... .......... .......... .......... .......... 83%  117M 0s\n",
      " 44200K .......... .......... .......... .......... .......... 83% 5,11M 0s\n",
      " 44250K .......... .......... .......... .......... .......... 83%  121M 0s\n",
      " 44300K .......... .......... .......... .......... .......... 83% 2,77M 0s\n",
      " 44350K .......... .......... .......... .......... .......... 83% 15,9M 0s\n",
      " 44400K .......... .......... .......... .......... .......... 83% 19,5M 0s\n",
      " 44450K .......... .......... .......... .......... .......... 83% 15,1M 0s\n",
      " 44500K .......... .......... .......... .......... .......... 83% 71,0M 0s\n",
      " 44550K .......... .......... .......... .......... .......... 83% 76,9M 0s\n",
      " 44600K .......... .......... .......... .......... .......... 83% 56,1M 0s\n",
      " 44650K .......... .......... .......... .......... .......... 84% 70,6M 0s\n",
      " 44700K .......... .......... .......... .......... .......... 84% 77,2M 0s\n",
      " 44750K .......... .......... .......... .......... .......... 84%  115M 0s\n",
      " 44800K .......... .......... .......... .......... .......... 84%  123M 0s\n",
      " 44850K .......... .......... .......... .......... .......... 84%  114M 0s\n",
      " 44900K .......... .......... .......... .......... .......... 84%  126M 0s\n",
      " 44950K .......... .......... .......... .......... .......... 84%  103M 0s\n",
      " 45000K .......... .......... .......... .......... .......... 84%  113M 0s\n",
      " 45050K .......... .......... .......... .......... .......... 84%  153M 0s\n",
      " 45100K .......... .......... .......... .......... .......... 84%  114M 0s\n",
      " 45150K .......... .......... .......... .......... .......... 84% 64,1M 0s\n",
      " 45200K .......... .......... .......... .......... .......... 85%  105M 0s\n",
      " 45250K .......... .......... .......... .......... .......... 85% 80,2M 0s\n",
      " 45300K .......... .......... .......... .......... .......... 85%  190M 0s\n",
      " 45350K .......... .......... .......... .......... .......... 85%  113M 0s\n",
      " 45400K .......... .......... .......... .......... .......... 85%  120M 0s\n",
      " 45450K .......... .......... .......... .......... .......... 85%  102M 0s\n",
      " 45500K .......... .......... .......... .......... .......... 85%  122M 0s\n",
      " 45550K .......... .......... .......... .......... .......... 85%  115M 0s\n",
      " 45600K .......... .......... .......... .......... .......... 85%  104M 0s\n",
      " 45650K .......... .......... .......... .......... .......... 85%  120M 0s\n",
      " 45700K .......... .......... .......... .......... .......... 85%  102M 0s\n",
      " 45750K .......... .......... .......... .......... .......... 86%  108M 0s\n",
      " 45800K .......... .......... .......... .......... .......... 86%  123M 0s\n",
      " 45850K .......... .......... .......... .......... .......... 86%  102M 0s\n",
      " 45900K .......... .......... .......... .......... .......... 86%  117M 0s\n",
      " 45950K .......... .......... .......... .......... .......... 86%  103M 0s\n",
      " 46000K .......... .......... .......... .......... .......... 86%  120M 0s\n",
      " 46050K .......... .......... .......... .......... .......... 86%  122M 0s\n",
      " 46100K .......... .......... .......... .......... .......... 86%  102M 0s\n",
      " 46150K .......... .......... .......... .......... .......... 86%  117M 0s\n",
      " 46200K .......... .......... .......... .......... .......... 86%  102M 0s\n",
      " 46250K .......... .......... .......... .......... .......... 87%  121M 0s\n",
      " 46300K .......... .......... .......... .......... .......... 87%  117M 0s\n",
      " 46350K .......... .......... .......... .......... .......... 87%  102M 0s\n",
      " 46400K .......... .......... .......... .......... .......... 87%  120M 0s\n",
      " 46450K .......... .......... .......... .......... .......... 87%  103M 0s\n",
      " 46500K .......... .......... .......... .......... .......... 87%  116M 0s\n",
      " 46550K .......... .......... .......... .......... .......... 87%  102M 0s\n",
      " 46600K .......... .......... .......... .......... .......... 87%  119M 0s\n",
      " 46650K .......... .......... .......... .......... .......... 87% 24,4M 0s\n",
      " 46700K .......... .......... .......... .......... .......... 87%  109M 0s\n",
      " 46750K .......... .......... .......... .......... .......... 87%  153M 0s\n",
      " 46800K .......... .......... .......... .......... .......... 88%  189M 0s\n",
      " 46850K .......... .......... .......... .......... .......... 88%  150M 0s\n",
      " 46900K .......... .......... .......... .......... .......... 88%  164M 0s\n",
      " 46950K .......... .......... .......... .......... .......... 88%  147M 0s\n",
      " 47000K .......... .......... .......... .......... .......... 88%  172M 0s\n",
      " 47050K .......... .......... .......... .......... .......... 88%  160M 0s\n",
      " 47100K .......... .......... .......... .......... .......... 88%  144M 0s\n",
      " 47150K .......... .......... .......... .......... .......... 88%  176M 0s\n",
      " 47200K .......... .......... .......... .......... .......... 88%  156M 0s\n",
      " 47250K .......... .......... .......... .......... .......... 88% 5,28M 0s\n",
      " 47300K .......... .......... .......... .......... .......... 88% 30,0M 0s\n",
      " 47350K .......... .......... .......... .......... .......... 89% 3,16M 0s\n",
      " 47400K .......... .......... .......... .......... .......... 89% 14,2M 0s\n",
      " 47450K .......... .......... .......... .......... .......... 89% 10,8M 0s\n",
      " 47500K .......... .......... .......... .......... .......... 89% 43,8M 0s\n",
      " 47550K .......... .......... .......... .......... .......... 89% 51,5M 0s\n",
      " 47600K .......... .......... .......... .......... .......... 89% 25,2M 0s\n",
      " 47650K .......... .......... .......... .......... .......... 89%  122M 0s\n",
      " 47700K .......... .......... .......... .......... .......... 89% 72,2M 0s\n",
      " 47750K .......... .......... .......... .......... .......... 89% 67,9M 0s\n",
      " 47800K .......... .......... .......... .......... .......... 89%  120M 0s\n",
      " 47850K .......... .......... .......... .......... .......... 90%  102M 0s\n",
      " 47900K .......... .......... .......... .......... .......... 90%  119M 0s\n",
      " 47950K .......... .......... .......... .......... .......... 90%  102M 0s\n",
      " 48000K .......... .......... .......... .......... .......... 90%  122M 0s\n",
      " 48050K .......... .......... .......... .......... .......... 90%  117M 0s\n",
      " 48100K .......... .......... .......... .......... .......... 90%  103M 0s\n",
      " 48150K .......... .......... .......... .......... .......... 90%  120M 0s\n",
      " 48200K .......... .......... .......... .......... .......... 90%  103M 0s\n",
      " 48250K .......... .......... .......... .......... .......... 90%  120M 0s\n",
      " 48300K .......... .......... .......... .......... .......... 90% 68,6M 0s\n",
      " 48350K .......... .......... .......... .......... .......... 90%  103M 0s\n",
      " 48400K .......... .......... .......... .......... .......... 91%  121M 0s\n",
      " 48450K .......... .......... .......... .......... .......... 91%  102M 0s\n",
      " 48500K .......... .......... .......... .......... .......... 91%  118M 0s\n",
      " 48550K .......... .......... .......... .......... .......... 91%  121M 0s\n",
      " 48600K .......... .......... .......... .......... .......... 91%  102M 0s\n",
      " 48650K .......... .......... .......... .......... .......... 91%  118M 0s\n",
      " 48700K .......... .......... .......... .......... .......... 91%  105M 0s\n",
      " 48750K .......... .......... .......... .......... .......... 91%  117M 0s\n",
      " 48800K .......... .......... .......... .......... .......... 91%  120M 0s\n",
      " 48850K .......... .......... .......... .......... .......... 91%  103M 0s\n",
      " 48900K .......... .......... .......... .......... .......... 91%  115M 0s\n",
      " 48950K .......... .......... .......... .......... .......... 92%  104M 0s\n",
      " 49000K .......... .......... .......... .......... .......... 92%  122M 0s\n",
      " 49050K .......... .......... .......... .......... .......... 92%  117M 0s\n",
      " 49100K .......... .......... .......... .......... .......... 92%  102M 0s\n",
      " 49150K .......... .......... .......... .......... .......... 92%  122M 0s\n",
      " 49200K .......... .......... .......... .......... .......... 92%  102M 0s\n",
      " 49250K .......... .......... .......... .......... .......... 92%  121M 0s\n",
      " 49300K .......... .......... .......... .......... .......... 92%  118M 0s\n",
      " 49350K .......... .......... .......... .......... .......... 92%  102M 0s\n",
      " 49400K .......... .......... .......... .......... .......... 92%  121M 0s\n",
      " 49450K .......... .......... .......... .......... .......... 93%  102M 0s\n",
      " 49500K .......... .......... .......... .......... .......... 93%  117M 0s\n",
      " 49550K .......... .......... .......... .......... .......... 93%  121M 0s\n",
      " 49600K .......... .......... .......... .......... .......... 93%  102M 0s\n",
      " 49650K .......... .......... .......... .......... .......... 93%  118M 0s\n",
      " 49700K .......... .......... .......... .......... .......... 93% 67,5M 0s\n",
      " 49750K .......... .......... .......... .......... .......... 93% 33,1M 0s\n",
      " 49800K .......... .......... .......... .......... .......... 93% 48,9M 0s\n",
      " 49850K .......... .......... .......... .......... .......... 93%  103M 0s\n",
      " 49900K .......... .......... .......... .......... .......... 93%  118M 0s\n",
      " 49950K .......... .......... .......... .......... .......... 93%  103M 0s\n",
      " 50000K .......... .......... .......... .......... .......... 94%  120M 0s\n",
      " 50050K .......... .......... .......... .......... .......... 94%  117M 0s\n",
      " 50100K .......... .......... .......... .......... .......... 94%  102M 0s\n",
      " 50150K .......... .......... .......... .......... .......... 94%  121M 0s\n",
      " 50200K .......... .......... .......... .......... .......... 94% 7,33M 0s\n",
      " 50250K .......... .......... .......... .......... .......... 94% 49,8M 0s\n",
      " 50300K .......... .......... .......... .......... .......... 94%  125M 0s\n",
      " 50350K .......... .......... .......... .......... .......... 94% 5,16M 0s\n",
      " 50400K .......... .......... .......... .......... .......... 94% 8,75M 0s\n",
      " 50450K .......... .......... .......... .......... .......... 94% 7,32M 0s\n",
      " 50500K .......... .......... .......... .......... .......... 95% 59,3M 0s\n",
      " 50550K .......... .......... .......... .......... .......... 95% 15,0M 0s\n",
      " 50600K .......... .......... .......... .......... .......... 95% 38,0M 0s\n",
      " 50650K .......... .......... .......... .......... .......... 95% 27,0M 0s\n",
      " 50700K .......... .......... .......... .......... .......... 95% 32,1M 0s\n",
      " 50750K .......... .......... .......... .......... .......... 95%  122M 0s\n",
      " 50800K .......... .......... .......... .......... .......... 95%  122M 0s\n",
      " 50850K .......... .......... .......... .......... .......... 95%  103M 0s\n",
      " 50900K .......... .......... .......... .......... .......... 95%  118M 0s\n",
      " 50950K .......... .......... .......... .......... .......... 95%  101M 0s\n",
      " 51000K .......... .......... .......... .......... .......... 95%  121M 0s\n",
      " 51050K .......... .......... .......... .......... .......... 96% 95,7M 0s\n",
      " 51100K .......... .......... .......... .......... .......... 96% 97,3M 0s\n",
      " 51150K .......... .......... .......... .......... .......... 96% 57,7M 0s\n",
      " 51200K .......... .......... .......... .......... .......... 96%  109M 0s\n",
      " 51250K .......... .......... .......... .......... .......... 96%  125M 0s\n",
      " 51300K .......... .......... .......... .......... .......... 96%  106M 0s\n",
      " 51350K .......... .......... .......... .......... .......... 96% 89,2M 0s\n",
      " 51400K .......... .......... .......... .......... .......... 96% 51,3M 0s\n",
      " 51450K .......... .......... .......... .......... .......... 96% 96,7M 0s\n",
      " 51500K .......... .......... .......... .......... .......... 96%  109M 0s\n",
      " 51550K .......... .......... .......... .......... .......... 96%  118M 0s\n",
      " 51600K .......... .......... .......... .......... .......... 97% 97,3M 0s\n",
      " 51650K .......... .......... .......... .......... .......... 97%  126M 0s\n",
      " 51700K .......... .......... .......... .......... .......... 97%  118M 0s\n",
      " 51750K .......... .......... .......... .......... .......... 97%  140M 0s\n",
      " 51800K .......... .......... .......... .......... .......... 97%  141M 0s\n",
      " 51850K .......... .......... .......... .......... .......... 97%  109M 0s\n",
      " 51900K .......... .......... .......... .......... .......... 97%  137M 0s\n",
      " 51950K .......... .......... .......... .......... .......... 97%  122M 0s\n",
      " 52000K .......... .......... .......... .......... .......... 97%  210M 0s\n",
      " 52050K .......... .......... .......... .......... .......... 97% 75,3M 0s\n",
      " 52100K .......... .......... .......... .......... .......... 98%  105M 0s\n",
      " 52150K .......... .......... .......... .......... .......... 98%  148M 0s\n",
      " 52200K .......... .......... .......... .......... .......... 98%  123M 0s\n",
      " 52250K .......... .......... .......... .......... .......... 98%  141M 0s\n",
      " 52300K .......... .......... .......... .......... .......... 98%  123M 0s\n",
      " 52350K .......... .......... .......... .......... .......... 98%  103M 0s\n",
      " 52400K .......... .......... .......... .......... .......... 98%  115M 0s\n",
      " 52450K .......... .......... .......... .......... .......... 98%  110M 0s\n",
      " 52500K .......... .......... .......... .......... .......... 98%  118M 0s\n",
      " 52550K .......... .......... .......... .......... .......... 98%  120M 0s\n",
      " 52600K .......... .......... .......... .......... .......... 98%  103M 0s\n",
      " 52650K .......... .......... .......... .......... .......... 99%  117M 0s\n",
      " 52700K .......... .......... .......... .......... .......... 99%  103M 0s\n",
      " 52750K .......... .......... .......... .......... .......... 99%  121M 0s\n",
      " 52800K .......... .......... .......... .......... .......... 99%  118M 0s\n",
      " 52850K .......... .......... .......... .......... .......... 99% 25,7M 0s\n",
      " 52900K .......... .......... .......... .......... .......... 99% 54,8M 0s\n",
      " 52950K .......... .......... .......... .......... .......... 99% 91,6M 0s\n",
      " 53000K .......... .......... .......... .......... .......... 99%  121M 0s\n",
      " 53050K .......... .......... .......... .......... .......... 99%  117M 0s\n",
      " 53100K .......... .......... .......... .......... .......... 99%  103M 0s\n",
      " 53150K .......... .......... .......... .......... .......... 99%  120M 0s\n",
      " 53200K ........                                              100%  181M=2,4s\n",
      "\n",
      "2023-08-17 11:19:15 (22,0 MB/s) - ‘data/reference/genome.fa.gz’ guardado [54485176/54485176]\n",
      "\n",
      "[Thu Aug 17 11:19:16 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T111912.607547.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:19:18 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR696683_chr5_1.fastq.gz, data/raw/ERR696683_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR696683_chr5_1_fastp.fastq.gz, data/processed/ERR696683_chr5_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR696683_chr5_1_fastp.fastq.gz, data/processed/ERR696683_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1745204\n",
      "total bases: 131934728\n",
      "Q20 bases: 131000292(99.2917%)\n",
      "Q30 bases: 127389751(96.5551%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1745204\n",
      "total bases: 131603364\n",
      "Q20 bases: 130011063(98.7901%)\n",
      "Q30 bases: 124792206(94.8245%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1090289\n",
      "total bases: 82806787\n",
      "Q20 bases: 82482544(99.6084%)\n",
      "Q30 bases: 81050211(97.8787%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1090289\n",
      "total bases: 82775596\n",
      "Q20 bases: 82215241(99.323%)\n",
      "Q30 bases: 80124032(96.7967%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2180578\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1309830\n",
      "reads with adapter trimmed: 1530\n",
      "bases trimmed due to adapters: 39157\n",
      "\n",
      "Duplication rate: 5.88378%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR696683_chr5_1.fastq.gz -I data/raw/ERR696683_chr5_2.fastq.gz -o data/processed/ERR696683_chr5_1_fastp.fastq.gz -O data/processed/ERR696683_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 7 seconds\n",
      "[Thu Aug 17 11:19:26 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:19:26 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR696683_chr5_1.fastq.gz, data/raw/ERR696683_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR696683_chr5_1_fastqc.html, results/fastqc_result/ERR696683_chr5_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR696683_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 11:19:41 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:19:41 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR696683_chr5_1_fastp.fastq.gz, data/processed/ERR696683_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR696683_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR696683_chr5_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR696683_chr5_1_fastp.fastq.gz, data/processed/ERR696683_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 11:19:49 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T111916.990603.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:19:51 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753368_chr5_1.fastq.gz, data/raw/ERR753368_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753368_chr5_1_fastp.fastq.gz, data/processed/ERR753368_chr5_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753368_chr5_2_fastp.fastq.gz, data/processed/ERR753368_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1754317\n",
      "total bases: 133207348\n",
      "Q20 bases: 132685869(99.6085%)\n",
      "Q30 bases: 128783661(96.6791%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1754317\n",
      "total bases: 133082490\n",
      "Q20 bases: 131930412(99.1343%)\n",
      "Q30 bases: 125959500(94.6477%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1515191\n",
      "total bases: 115148663\n",
      "Q20 bases: 114951118(99.8284%)\n",
      "Q30 bases: 113024140(98.155%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1515191\n",
      "total bases: 115139136\n",
      "Q20 bases: 114585523(99.5192%)\n",
      "Q30 bases: 111209688(96.5872%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3030382\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 478252\n",
      "reads with adapter trimmed: 2004\n",
      "bases trimmed due to adapters: 46944\n",
      "\n",
      "Duplication rate: 2.06194%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753368_chr5_1.fastq.gz -I data/raw/ERR753368_chr5_2.fastq.gz -o data/processed/ERR753368_chr5_1_fastp.fastq.gz -O data/processed/ERR753368_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Thu Aug 17 11:19:58 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:19:58 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753368_chr5_1.fastq.gz, data/raw/ERR753368_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753368_chr5_1_fastqc.html, results/fastqc_result/ERR753368_chr5_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753368_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753368_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 11:20:12 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:20:12 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753368_chr5_1_fastp.fastq.gz, data/processed/ERR753368_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753368_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753368_chr5_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753368_chr5_2_fastp.fastq.gz, data/processed/ERR753368_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 11:20:25 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T111950.121866.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:20:26 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753369_chr5_1.fastq.gz, data/raw/ERR753369_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753369_chr5_1_fastp.fastq.gz, data/processed/ERR753369_chr5_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753369_chr5_1_fastp.fastq.gz, data/processed/ERR753369_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1874038\n",
      "total bases: 142293627\n",
      "Q20 bases: 141712295(99.5915%)\n",
      "Q30 bases: 137456286(96.6005%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1874038\n",
      "total bases: 142166679\n",
      "Q20 bases: 140932525(99.1319%)\n",
      "Q30 bases: 134533978(94.6312%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1618376\n",
      "total bases: 122990005\n",
      "Q20 bases: 122765144(99.8172%)\n",
      "Q30 bases: 120638796(98.0883%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1618376\n",
      "total bases: 122980612\n",
      "Q20 bases: 122387247(99.5175%)\n",
      "Q30 bases: 118766072(96.573%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3236752\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 511324\n",
      "reads with adapter trimmed: 2190\n",
      "bases trimmed due to adapters: 51994\n",
      "\n",
      "Duplication rate: 2.41004%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753369_chr5_1.fastq.gz -I data/raw/ERR753369_chr5_2.fastq.gz -o data/processed/ERR753369_chr5_1_fastp.fastq.gz -O data/processed/ERR753369_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Thu Aug 17 11:20:33 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:20:33 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753369_chr5_1.fastq.gz, data/raw/ERR753369_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753369_chr5_1_fastqc.html, results/fastqc_result/ERR753369_chr5_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753369_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753369_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 11:20:47 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:20:47 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753369_chr5_1_fastp.fastq.gz, data/processed/ERR753369_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753369_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753369_chr5_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753369_chr5_1_fastp.fastq.gz, data/processed/ERR753369_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 11:21:00 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T112025.237800.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:21:02 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753370_chr5_1.fastq.gz, data/raw/ERR753370_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753370_chr5_1_fastqc.html, results/fastqc_result/ERR753370_chr5_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753370_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753370_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 11:21:27 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:21:27 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753370_chr5_1.fastq.gz, data/raw/ERR753370_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753370_chr5_1_fastp.fastq.gz, data/processed/ERR753370_chr5_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753370_chr5_2_fastp.fastq.gz, data/processed/ERR753370_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2518071\n",
      "total bases: 253490833\n",
      "Q20 bases: 251767367(99.3201%)\n",
      "Q30 bases: 244487028(96.4481%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2518071\n",
      "total bases: 253433925\n",
      "Q20 bases: 251051683(99.06%)\n",
      "Q30 bases: 241620433(95.3386%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 2517536\n",
      "total bases: 252452388\n",
      "Q20 bases: 250828184(99.3566%)\n",
      "Q30 bases: 243890545(96.6085%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 2517536\n",
      "total bases: 251656989\n",
      "Q20 bases: 249490937(99.1393%)\n",
      "Q30 bases: 240647742(95.6253%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 5035072\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1070\n",
      "reads with adapter trimmed: 14130\n",
      "bases trimmed due to adapters: 108801\n",
      "\n",
      "Duplication rate: 4.44886%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 159\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753370_chr5_1.fastq.gz -I data/raw/ERR753370_chr5_2.fastq.gz -o data/processed/ERR753370_chr5_1_fastp.fastq.gz -O data/processed/ERR753370_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 11 seconds\n",
      "[Thu Aug 17 11:21:38 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:21:38 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753370_chr5_1_fastp.fastq.gz, data/processed/ERR753370_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753370_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753370_chr5_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753370_chr5_2_fastp.fastq.gz, data/processed/ERR753370_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 11:22:01 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T112101.003322.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:22:03 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753371_chr5_1.fastq.gz, data/raw/ERR753371_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753371_chr5_1_fastp.fastq.gz, data/processed/ERR753371_chr5_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753371_chr5_2_fastp.fastq.gz, data/processed/ERR753371_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1647835\n",
      "total bases: 124169361\n",
      "Q20 bases: 123052427(99.1005%)\n",
      "Q30 bases: 118563904(95.4856%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1647835\n",
      "total bases: 123864691\n",
      "Q20 bases: 122402253(98.8193%)\n",
      "Q30 bases: 118123304(95.3648%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 856072\n",
      "total bases: 64982245\n",
      "Q20 bases: 64667623(99.5158%)\n",
      "Q30 bases: 63059172(97.0406%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 856072\n",
      "total bases: 64947026\n",
      "Q20 bases: 64500274(99.3121%)\n",
      "Q30 bases: 63094570(97.1477%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1712144\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1583526\n",
      "reads with adapter trimmed: 1440\n",
      "bases trimmed due to adapters: 32385\n",
      "\n",
      "Duplication rate: 22.2935%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 117\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753371_chr5_1.fastq.gz -I data/raw/ERR753371_chr5_2.fastq.gz -o data/processed/ERR753371_chr5_1_fastp.fastq.gz -O data/processed/ERR753371_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Thu Aug 17 11:22:09 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:22:09 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753371_chr5_1.fastq.gz, data/raw/ERR753371_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753371_chr5_1_fastqc.html, results/fastqc_result/ERR753371_chr5_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753371_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753371_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 11:22:23 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:22:23 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753371_chr5_1_fastp.fastq.gz, data/processed/ERR753371_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753371_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753371_chr5_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753371_chr5_2_fastp.fastq.gz, data/processed/ERR753371_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 11:22:32 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T112202.028060.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:22:33 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753372_chr5_1.fastq.gz, data/raw/ERR753372_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753372_chr5_1_fastqc.html, results/fastqc_result/ERR753372_chr5_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753372_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753372_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 11:22:49 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:22:49 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753372_chr5_1.fastq.gz, data/raw/ERR753372_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753372_chr5_1_fastp.fastq.gz, data/processed/ERR753372_chr5_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753372_chr5_1_fastp.fastq.gz, data/processed/ERR753372_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1849920\n",
      "total bases: 140463336\n",
      "Q20 bases: 139906933(99.6039%)\n",
      "Q30 bases: 135706572(96.6135%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1849920\n",
      "total bases: 140332630\n",
      "Q20 bases: 139116664(99.1335%)\n",
      "Q30 bases: 132851025(94.6687%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1592983\n",
      "total bases: 121060371\n",
      "Q20 bases: 120850328(99.8265%)\n",
      "Q30 bases: 118781603(98.1177%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1592983\n",
      "total bases: 121050636\n",
      "Q20 bases: 120473426(99.5232%)\n",
      "Q30 bases: 116971001(96.6298%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3185966\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 513874\n",
      "reads with adapter trimmed: 2008\n",
      "bases trimmed due to adapters: 48830\n",
      "\n",
      "Duplication rate: 2.0229%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753372_chr5_1.fastq.gz -I data/raw/ERR753372_chr5_2.fastq.gz -o data/processed/ERR753372_chr5_1_fastp.fastq.gz -O data/processed/ERR753372_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Thu Aug 17 11:22:56 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:22:56 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753372_chr5_1_fastp.fastq.gz, data/processed/ERR753372_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753372_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753372_chr5_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753372_chr5_1_fastp.fastq.gz, data/processed/ERR753372_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 11:23:09 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T112232.217987.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:23:10 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753373_chr5_1.fastq.gz, data/raw/ERR753373_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753373_chr5_1_fastqc.html, results/fastqc_result/ERR753373_chr5_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753373_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753373_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 11:23:27 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:23:27 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753373_chr5_1.fastq.gz, data/raw/ERR753373_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753373_chr5_1_fastp.fastq.gz, data/processed/ERR753373_chr5_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753373_chr5_2_fastp.fastq.gz, data/processed/ERR753373_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1769351\n",
      "total bases: 169422654\n",
      "Q20 bases: 168087237(99.2118%)\n",
      "Q30 bases: 156744268(92.5167%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1769351\n",
      "total bases: 165883501\n",
      "Q20 bases: 164108844(98.9302%)\n",
      "Q30 bases: 153648766(92.6245%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1727865\n",
      "total bases: 162896728\n",
      "Q20 bases: 161958252(99.4239%)\n",
      "Q30 bases: 152772191(93.7847%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1727865\n",
      "total bases: 157462847\n",
      "Q20 bases: 156729547(99.5343%)\n",
      "Q30 bases: 148765966(94.4769%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3455730\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 82972\n",
      "reads with adapter trimmed: 1628\n",
      "bases trimmed due to adapters: 56407\n",
      "\n",
      "Duplication rate: 1.9251%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 152\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753373_chr5_1.fastq.gz -I data/raw/ERR753373_chr5_2.fastq.gz -o data/processed/ERR753373_chr5_1_fastp.fastq.gz -O data/processed/ERR753373_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 8 seconds\n",
      "[Thu Aug 17 11:23:35 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:23:35 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753373_chr5_1_fastp.fastq.gz, data/processed/ERR753373_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753373_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753373_chr5_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753373_chr5_2_fastp.fastq.gz, data/processed/ERR753373_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 11:23:50 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T112309.196504.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:23:51 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753374_chr5_1.fastq.gz, data/raw/ERR753374_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753374_chr5_1_fastp.fastq.gz, data/processed/ERR753374_chr5_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753374_chr5_1_fastp.fastq.gz, data/processed/ERR753374_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1042969\n",
      "total bases: 99775475\n",
      "Q20 bases: 98851672(99.0741%)\n",
      "Q30 bases: 90441720(90.6452%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1042969\n",
      "total bases: 97644504\n",
      "Q20 bases: 96637952(98.9692%)\n",
      "Q30 bases: 89006849(91.154%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1011156\n",
      "total bases: 94571637\n",
      "Q20 bases: 93909953(99.3003%)\n",
      "Q30 bases: 87206490(92.2121%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1011156\n",
      "total bases: 92115109\n",
      "Q20 bases: 91596849(99.4374%)\n",
      "Q30 bases: 85744137(93.0837%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2022312\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 63626\n",
      "reads with adapter trimmed: 920\n",
      "bases trimmed due to adapters: 32756\n",
      "\n",
      "Duplication rate: 5.80338%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 148\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753374_chr5_1.fastq.gz -I data/raw/ERR753374_chr5_2.fastq.gz -o data/processed/ERR753374_chr5_1_fastp.fastq.gz -O data/processed/ERR753374_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Thu Aug 17 11:23:56 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:23:56 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753374_chr5_1_fastp.fastq.gz, data/processed/ERR753374_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753374_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753374_chr5_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753374_chr5_1_fastp.fastq.gz, data/processed/ERR753374_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 11:24:07 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:24:07 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753374_chr5_1.fastq.gz, data/raw/ERR753374_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753374_chr5_1_fastqc.html, results/fastqc_result/ERR753374_chr5_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753374_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753374_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 11:24:18 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T112350.464906.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:24:20 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753375_chr5_1.fastq.gz, data/raw/ERR753375_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753375_chr5_1_fastqc.html, results/fastqc_result/ERR753375_chr5_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753375_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753375_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 11:24:31 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:24:31 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753375_chr5_1.fastq.gz, data/raw/ERR753375_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753375_chr5_1_fastp.fastq.gz, data/processed/ERR753375_chr5_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753375_chr5_1_fastp.fastq.gz, data/processed/ERR753375_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1084744\n",
      "total bases: 81943993\n",
      "Q20 bases: 81438535(99.3832%)\n",
      "Q30 bases: 79595776(97.1344%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1084744\n",
      "total bases: 81928209\n",
      "Q20 bases: 80877145(98.7171%)\n",
      "Q30 bases: 78915671(96.323%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 711112\n",
      "total bases: 53994712\n",
      "Q20 bases: 53820596(99.6775%)\n",
      "Q30 bases: 53016677(98.1886%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 711112\n",
      "total bases: 54003767\n",
      "Q20 bases: 53488355(99.0456%)\n",
      "Q30 bases: 52659532(97.5108%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1422224\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 747264\n",
      "reads with adapter trimmed: 856\n",
      "bases trimmed due to adapters: 20807\n",
      "\n",
      "Duplication rate: 1.69512%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753375_chr5_1.fastq.gz -I data/raw/ERR753375_chr5_2.fastq.gz -o data/processed/ERR753375_chr5_1_fastp.fastq.gz -O data/processed/ERR753375_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 3 seconds\n",
      "[Thu Aug 17 11:24:35 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:24:35 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753375_chr5_1_fastp.fastq.gz, data/processed/ERR753375_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753375_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753375_chr5_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753375_chr5_1_fastp.fastq.gz, data/processed/ERR753375_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 11:24:43 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T112418.742868.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:24:44 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753376_chr5_1.fastq.gz, data/raw/ERR753376_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753376_chr5_1_fastqc.html, results/fastqc_result/ERR753376_chr5_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753376_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753376_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 11:24:57 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:24:57 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753376_chr5_1.fastq.gz, data/raw/ERR753376_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753376_chr5_1_fastp.fastq.gz, data/processed/ERR753376_chr5_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753376_chr5_1_fastp.fastq.gz, data/processed/ERR753376_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1402798\n",
      "total bases: 106050477\n",
      "Q20 bases: 105320649(99.3118%)\n",
      "Q30 bases: 102421962(96.5785%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1402798\n",
      "total bases: 105773011\n",
      "Q20 bases: 104493043(98.7899%)\n",
      "Q30 bases: 100281618(94.8083%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 875003\n",
      "total bases: 66456395\n",
      "Q20 bases: 66207960(99.6262%)\n",
      "Q30 bases: 65072453(97.9175%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 875003\n",
      "total bases: 66430859\n",
      "Q20 bases: 65983780(99.327%)\n",
      "Q30 bases: 64300470(96.7931%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1750006\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1055590\n",
      "reads with adapter trimmed: 1316\n",
      "bases trimmed due to adapters: 32140\n",
      "\n",
      "Duplication rate: 9.15894%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753376_chr5_1.fastq.gz -I data/raw/ERR753376_chr5_2.fastq.gz -o data/processed/ERR753376_chr5_1_fastp.fastq.gz -O data/processed/ERR753376_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Thu Aug 17 11:25:03 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:25:03 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753376_chr5_1_fastp.fastq.gz, data/processed/ERR753376_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753376_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753376_chr5_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753376_chr5_1_fastp.fastq.gz, data/processed/ERR753376_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "Approx 100% complete for ERR753376_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr5_2_fastp.fastq.gz\n",
      "Approx 100% complete for ERR753376_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 11:25:12 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T112443.599179.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:25:13 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753377_chr5_1.fastq.gz, data/raw/ERR753377_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753377_chr5_1_fastp.fastq.gz, data/processed/ERR753377_chr5_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753377_chr5_1_fastp.fastq.gz, data/processed/ERR753377_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1456925\n",
      "total bases: 110159308\n",
      "Q20 bases: 109389376(99.3011%)\n",
      "Q30 bases: 106378169(96.5676%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1456925\n",
      "total bases: 109900000\n",
      "Q20 bases: 108632355(98.8465%)\n",
      "Q30 bases: 104449264(95.0403%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 929607\n",
      "total bases: 70603984\n",
      "Q20 bases: 70333119(99.6164%)\n",
      "Q30 bases: 69114056(97.8897%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 929607\n",
      "total bases: 70578254\n",
      "Q20 bases: 70125662(99.3587%)\n",
      "Q30 bases: 68418786(96.9403%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1859214\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1054636\n",
      "reads with adapter trimmed: 1262\n",
      "bases trimmed due to adapters: 30893\n",
      "\n",
      "Duplication rate: 9.20978%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753377_chr5_1.fastq.gz -I data/raw/ERR753377_chr5_2.fastq.gz -o data/processed/ERR753377_chr5_1_fastp.fastq.gz -O data/processed/ERR753377_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Thu Aug 17 11:25:19 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:25:19 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753377_chr5_1_fastp.fastq.gz, data/processed/ERR753377_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753377_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753377_chr5_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753377_chr5_1_fastp.fastq.gz, data/processed/ERR753377_chr5_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 11:25:27 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:25:27 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753377_chr5_1.fastq.gz, data/raw/ERR753377_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753377_chr5_1_fastqc.html, results/fastqc_result/ERR753377_chr5_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753377_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753377_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 11:25:40 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T112512.596078.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:25:42 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753378_chr5_1.fastq.gz, data/raw/ERR753378_chr5_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753378_chr5_1_fastqc.html, results/fastqc_result/ERR753378_chr5_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753378_chr5_1_fastqc.html\n",
      "    wildcards: sample=ERR753378_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr5_1.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr5_1.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr5_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr5_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr5_2.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr5_2.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr5_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 11:25:55 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:25:55 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753378_chr5_1.fastq.gz, data/raw/ERR753378_chr5_2.fastq.gz\n",
      "    output: data/processed/ERR753378_chr5_1_fastp.fastq.gz, data/processed/ERR753378_chr5_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753378_chr5_2_fastp.fastq.gz, data/processed/ERR753378_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1582656\n",
      "total bases: 120172394\n",
      "Q20 bases: 119702143(99.6087%)\n",
      "Q30 bases: 116149817(96.6527%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1582656\n",
      "total bases: 120057743\n",
      "Q20 bases: 119018888(99.1347%)\n",
      "Q30 bases: 113653667(94.6658%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1368395\n",
      "total bases: 103992525\n",
      "Q20 bases: 103812087(99.8265%)\n",
      "Q30 bases: 102035807(98.1184%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1368395\n",
      "total bases: 103984175\n",
      "Q20 bases: 103483359(99.5184%)\n",
      "Q30 bases: 100427300(96.5794%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2736790\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 428522\n",
      "reads with adapter trimmed: 1908\n",
      "bases trimmed due to adapters: 45702\n",
      "\n",
      "Duplication rate: 2.03355%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753378_chr5_1.fastq.gz -I data/raw/ERR753378_chr5_2.fastq.gz -o data/processed/ERR753378_chr5_1_fastp.fastq.gz -O data/processed/ERR753378_chr5_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Thu Aug 17 11:26:01 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:26:01 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753378_chr5_1_fastp.fastq.gz, data/processed/ERR753378_chr5_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753378_chr5_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753378_chr5_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753378_chr5_2_fastp.fastq.gz, data/processed/ERR753378_chr5_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr5_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr5_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr5_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr5_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 11:26:12 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T112541.172271.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:26:13 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR696683_chr5_1_fastp.fastq.gz, data/processed/ERR696683_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR696683_chr5.sam, results/mapped_reads/ERR696683_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR696683_chr5_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR696683_chr5_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.71 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 70.81 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.55 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.38 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 40.43 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 118.117 sec; CPU: 112.887 sec\n",
      "[Thu Aug 17 11:29:49 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:29:49 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR696683_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr5.bam, results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR696683_chr5.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR696683_chr5_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 11:30:31 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:30:31 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR696683_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "11:30:32.969 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 11:30:32 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR696683_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR696683_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 11:30:32 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 11:30:32\tMarkDuplicates\tStart of doWork freeMemory: 528826664; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 11:30:32\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 11:30:32\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam\n",
      "INFO\t2023-08-17 11:30:34\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:81,252,689\n",
      "INFO\t2023-08-17 11:30:34\tMarkDuplicates\tTracking 736 as yet unmatched pairs. 736 records in RAM.\n",
      "INFO\t2023-08-17 11:30:36\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 5:168,199,830\n",
      "INFO\t2023-08-17 11:30:36\tMarkDuplicates\tTracking 288 as yet unmatched pairs. 288 records in RAM.\n",
      "INFO\t2023-08-17 11:30:36\tMarkDuplicates\tRead 2201603 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 11:30:36\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 552367712; totalMemory: 859832320; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 11:30:36\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 11:30:36\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 11:30:37\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 11:30:37\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 11:30:37\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 904912304; totalMemory: 1451229184; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 11:30:37\tMarkDuplicates\tMarking 201830 records as duplicates.\n",
      "INFO\t2023-08-17 11:30:37\tMarkDuplicates\tFound 99 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr5_sorted.bam\n",
      "INFO\t2023-08-17 11:30:37\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 11:30:51\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 11:30:51\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 11:30:51\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 11:30:51\tMarkDuplicates\tBefore output close freeMemory: 528124544; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 11:30:51\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 11:30:51\tMarkDuplicates\tAfter output close freeMemory: 528124544; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 11:30:51 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.30 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 11:30:52 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T112612.189018.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:30:54 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753368_chr5_1_fastp.fastq.gz, data/processed/ERR753368_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753368_chr5.sam, results/mapped_reads/ERR753368_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753368_chr5_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753368_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.61 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 71.34 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.54 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.37 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 40.44 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 119.202 sec; CPU: 113.310 sec\n",
      "[Thu Aug 17 11:35:10 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:35:10 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753368_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr5.bam, results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753368_chr5.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753368_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 11:35:59 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:35:59 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753368_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753368_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "11:36:00.338 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 11:36:00 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753368_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753368_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 11:36:00 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 11:36:00\tMarkDuplicates\tStart of doWork freeMemory: 528788528; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 11:36:00\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 11:36:00\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam\n",
      "INFO\t2023-08-17 11:36:01\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:72,252,503\n",
      "INFO\t2023-08-17 11:36:01\tMarkDuplicates\tTracking 479 as yet unmatched pairs. 479 records in RAM.\n",
      "INFO\t2023-08-17 11:36:03\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 5:138,184,159\n",
      "INFO\t2023-08-17 11:36:03\tMarkDuplicates\tTracking 440 as yet unmatched pairs. 440 records in RAM.\n",
      "INFO\t2023-08-17 11:36:04\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 5:179,891,658\n",
      "INFO\t2023-08-17 11:36:04\tMarkDuplicates\tTracking 24 as yet unmatched pairs. 24 records in RAM.\n",
      "INFO\t2023-08-17 11:36:04\tMarkDuplicates\tRead 3048194 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 11:36:05\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 625015560; totalMemory: 1024458752; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 11:36:05\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 11:36:05\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 11:36:05\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 11:36:05\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 11:36:05\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1014975656; totalMemory: 1561329664; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 11:36:05\tMarkDuplicates\tMarking 74584 records as duplicates.\n",
      "INFO\t2023-08-17 11:36:05\tMarkDuplicates\tFound 483 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr5_sorted.bam\n",
      "INFO\t2023-08-17 11:36:05\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 11:36:24\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 11:36:24\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 11:36:24\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 11:36:24\tMarkDuplicates\tBefore output close freeMemory: 528087248; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 11:36:24\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 11:36:24\tMarkDuplicates\tAfter output close freeMemory: 528087248; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 11:36:24 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.41 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 11:36:27 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T113052.782324.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:36:29 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753369_chr5_1_fastp.fastq.gz, data/processed/ERR753369_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753369_chr5.sam, results/mapped_reads/ERR753369_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753369_chr5_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753369_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.64 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 70.78 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.57 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.39 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 39.81 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 118.396 sec; CPU: 112.193 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 11:41:01 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:41:01 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753369_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr5.bam, results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753369_chr5.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753369_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 11:42:00 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:42:00 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753369_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753369_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "11:42:01.753 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 11:42:01 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753369_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753369_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 11:42:01 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 11:42:01\tMarkDuplicates\tStart of doWork freeMemory: 528806000; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 11:42:01\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 11:42:01\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam\n",
      "INFO\t2023-08-17 11:42:03\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:69,509,474\n",
      "INFO\t2023-08-17 11:42:03\tMarkDuplicates\tTracking 483 as yet unmatched pairs. 483 records in RAM.\n",
      "INFO\t2023-08-17 11:42:04\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 5:132,812,908\n",
      "INFO\t2023-08-17 11:42:04\tMarkDuplicates\tTracking 397 as yet unmatched pairs. 397 records in RAM.\n",
      "INFO\t2023-08-17 11:42:06\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 5:173,999,730\n",
      "INFO\t2023-08-17 11:42:06\tMarkDuplicates\tTracking 121 as yet unmatched pairs. 121 records in RAM.\n",
      "INFO\t2023-08-17 11:42:06\tMarkDuplicates\tRead 3250648 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 11:42:06\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 626866072; totalMemory: 1048576000; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 11:42:06\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 11:42:06\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 11:42:07\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 11:42:07\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 11:42:07\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1039110376; totalMemory: 1585446912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 11:42:07\tMarkDuplicates\tMarking 92908 records as duplicates.\n",
      "INFO\t2023-08-17 11:42:07\tMarkDuplicates\tFound 490 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr5_sorted.bam\n",
      "INFO\t2023-08-17 11:42:07\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 11:42:27\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 11:42:27\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 11:42:27\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 11:42:27\tMarkDuplicates\tBefore output close freeMemory: 528104880; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 11:42:27\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 11:42:27\tMarkDuplicates\tAfter output close freeMemory: 528104880; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 11:42:27 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.42 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 11:42:29 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T113627.781923.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:42:30 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753370_chr5_1_fastp.fastq.gz, data/processed/ERR753370_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753370_chr5.sam, results/mapped_reads/ERR753370_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753370_chr5_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753370_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.62 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 69.63 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.58 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.40 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 39.90 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 116.954 sec; CPU: 111.129 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 11:49:16 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:49:16 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753370_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr5.bam, results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753370_chr5.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753370_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 11:51:17 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:51:17 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753370_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753370_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "11:51:18.537 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 11:51:18 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753370_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753370_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 11:51:18 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 11:51:18\tMarkDuplicates\tStart of doWork freeMemory: 528788320; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 11:51:18\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 11:51:18\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam\n",
      "INFO\t2023-08-17 11:51:20\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:42,800,643\n",
      "INFO\t2023-08-17 11:51:20\tMarkDuplicates\tTracking 1041 as yet unmatched pairs. 1041 records in RAM.\n",
      "INFO\t2023-08-17 11:51:21\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 5:75,358,769\n",
      "INFO\t2023-08-17 11:51:21\tMarkDuplicates\tTracking 1655 as yet unmatched pairs. 1655 records in RAM.\n",
      "INFO\t2023-08-17 11:51:23\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 5:115,580,663\n",
      "INFO\t2023-08-17 11:51:23\tMarkDuplicates\tTracking 1486 as yet unmatched pairs. 1486 records in RAM.\n",
      "INFO\t2023-08-17 11:51:25\tMarkDuplicates\tRead     4,000,000 records.  Elapsed time: 00:00:06s.  Time for last 1,000,000:    1s.  Last read position: 5:143,041,833\n",
      "INFO\t2023-08-17 11:51:25\tMarkDuplicates\tTracking 958 as yet unmatched pairs. 958 records in RAM.\n",
      "INFO\t2023-08-17 11:51:26\tMarkDuplicates\tRead     5,000,000 records.  Elapsed time: 00:00:08s.  Time for last 1,000,000:    1s.  Last read position: 5:179,585,760\n",
      "INFO\t2023-08-17 11:51:26\tMarkDuplicates\tTracking 247 as yet unmatched pairs. 247 records in RAM.\n",
      "INFO\t2023-08-17 11:51:26\tMarkDuplicates\tRead 5095159 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 11:51:26\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 431103504; totalMemory: 1047527424; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 11:51:26\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 11:51:27\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 11:51:27\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 11:51:28\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 11:51:28\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1038042768; totalMemory: 1584398336; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 11:51:28\tMarkDuplicates\tMarking 357823 records as duplicates.\n",
      "INFO\t2023-08-17 11:51:28\tMarkDuplicates\tFound 194 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr5_sorted.bam\n",
      "INFO\t2023-08-17 11:51:28\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 11:52:06\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 11:52:06\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 11:52:06\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 11:52:06\tMarkDuplicates\tBefore output close freeMemory: 528085944; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 11:52:06\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 11:52:06\tMarkDuplicates\tAfter output close freeMemory: 528085944; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 11:52:06 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.80 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 11:52:10 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T114229.643446.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:52:16 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753371_chr5_1_fastp.fastq.gz, data/processed/ERR753371_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753371_chr5.sam, results/mapped_reads/ERR753371_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753371_chr5_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753371_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.64 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 72.66 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.56 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.38 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 40.90 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 121.568 sec; CPU: 115.145 sec\n",
      "[Thu Aug 17 11:55:34 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:55:34 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753371_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr5.bam, results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753371_chr5.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753371_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 11:56:06 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:56:06 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753371_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753371_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "11:56:07.025 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 11:56:07 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753371_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753371_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 11:56:07 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 11:56:07\tMarkDuplicates\tStart of doWork freeMemory: 528780696; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 11:56:07\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 11:56:07\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam\n",
      "INFO\t2023-08-17 11:56:08\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:109,819,561\n",
      "INFO\t2023-08-17 11:56:08\tMarkDuplicates\tTracking 139 as yet unmatched pairs. 139 records in RAM.\n",
      "INFO\t2023-08-17 11:56:09\tMarkDuplicates\tRead 1717822 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 11:56:09\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 280193104; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 11:56:09\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 11:56:09\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 11:56:10\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 11:56:10\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 11:56:10\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 790569448; totalMemory: 1336934400; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 11:56:10\tMarkDuplicates\tMarking 690666 records as duplicates.\n",
      "INFO\t2023-08-17 11:56:10\tMarkDuplicates\tFound 61 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr5_sorted.bam\n",
      "INFO\t2023-08-17 11:56:10\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 11:56:21\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 11:56:21\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 11:56:21\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 11:56:21\tMarkDuplicates\tBefore output close freeMemory: 528076304; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 11:56:21\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 11:56:21\tMarkDuplicates\tAfter output close freeMemory: 528076304; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 11:56:21 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.24 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 11:56:22 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T115211.289639.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 11:56:24 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753372_chr5_1_fastp.fastq.gz, data/processed/ERR753372_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753372_chr5.sam, results/mapped_reads/ERR753372_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753372_chr5_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753372_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.62 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 72.44 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.56 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.38 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 40.58 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 120.804 sec; CPU: 114.580 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 12:00:57 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:00:57 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753372_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr5.bam, results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753372_chr5.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753372_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 12:01:55 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:01:55 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753372_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753372_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "12:01:55.982 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 12:01:55 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753372_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753372_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 12:01:55 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 12:01:56\tMarkDuplicates\tStart of doWork freeMemory: 528785080; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:01:56\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 12:01:56\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam\n",
      "INFO\t2023-08-17 12:01:57\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:70,942,200\n",
      "INFO\t2023-08-17 12:01:57\tMarkDuplicates\tTracking 501 as yet unmatched pairs. 501 records in RAM.\n",
      "INFO\t2023-08-17 12:01:59\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 5:134,741,308\n",
      "INFO\t2023-08-17 12:01:59\tMarkDuplicates\tTracking 569 as yet unmatched pairs. 569 records in RAM.\n",
      "INFO\t2023-08-17 12:02:00\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 5:176,864,711\n",
      "INFO\t2023-08-17 12:02:00\tMarkDuplicates\tTracking 133 as yet unmatched pairs. 133 records in RAM.\n",
      "INFO\t2023-08-17 12:02:00\tMarkDuplicates\tRead 3202425 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 12:02:00\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 417453296; totalMemory: 833617920; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:02:00\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 12:02:00\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 12:02:01\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 12:02:01\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 12:02:01\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1088371552; totalMemory: 1634729984; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:02:01\tMarkDuplicates\tMarking 77152 records as duplicates.\n",
      "INFO\t2023-08-17 12:02:01\tMarkDuplicates\tFound 518 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr5_sorted.bam\n",
      "INFO\t2023-08-17 12:02:01\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 12:02:21\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 12:02:21\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 12:02:21\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 12:02:21\tMarkDuplicates\tBefore output close freeMemory: 528082832; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:02:21\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 12:02:21\tMarkDuplicates\tAfter output close freeMemory: 528082832; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 12:02:21 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.43 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 12:02:23 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T115622.932583.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:02:25 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753373_chr5_1_fastp.fastq.gz, data/processed/ERR753373_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753373_chr5.sam, results/mapped_reads/ERR753373_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753373_chr5_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753373_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.62 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 71.98 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.56 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.39 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 40.58 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 120.256 sec; CPU: 114.133 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 12:08:27 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:08:27 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753373_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr5.bam, results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753373_chr5.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753373_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 12:10:16 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:10:16 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753373_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753373_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "12:10:17.478 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 12:10:17 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753373_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753373_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 12:10:17 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 12:10:17\tMarkDuplicates\tStart of doWork freeMemory: 528776600; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:10:17\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 12:10:17\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam\n",
      "INFO\t2023-08-17 12:10:19\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:56,494,252\n",
      "INFO\t2023-08-17 12:10:19\tMarkDuplicates\tTracking 268 as yet unmatched pairs. 268 records in RAM.\n",
      "INFO\t2023-08-17 12:10:20\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 5:102,420,580\n",
      "INFO\t2023-08-17 12:10:20\tMarkDuplicates\tTracking 274 as yet unmatched pairs. 274 records in RAM.\n",
      "INFO\t2023-08-17 12:10:22\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 5:152,602,724\n",
      "INFO\t2023-08-17 12:10:22\tMarkDuplicates\tTracking 149 as yet unmatched pairs. 149 records in RAM.\n",
      "INFO\t2023-08-17 12:10:23\tMarkDuplicates\tRead 3481677 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 12:10:23\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 543313752; totalMemory: 988807168; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:10:23\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 12:10:23\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 12:10:23\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 12:10:24\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 12:10:24\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 979311512; totalMemory: 1525678080; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:10:24\tMarkDuplicates\tMarking 92710 records as duplicates.\n",
      "INFO\t2023-08-17 12:10:24\tMarkDuplicates\tFound 30640 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr5_sorted.bam\n",
      "INFO\t2023-08-17 12:10:24\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 12:10:55\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 12:10:55\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 12:10:55\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 12:10:55\tMarkDuplicates\tBefore output close freeMemory: 528074672; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:10:55\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 12:10:55\tMarkDuplicates\tAfter output close freeMemory: 528074672; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 12:10:55 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.63 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 12:10:58 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T120224.032492.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:10:59 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753374_chr5_1_fastp.fastq.gz, data/processed/ERR753374_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753374_chr5.sam, results/mapped_reads/ERR753374_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753374_chr5_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753374_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.60 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 69.75 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.56 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.40 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 40.19 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 117.278 sec; CPU: 111.497 sec\n",
      "[Thu Aug 17 12:15:01 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:15:01 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753374_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr5.bam, results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753374_chr5.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753374_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 12:15:56 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:15:56 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753374_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753374_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "12:15:57.365 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 12:15:57 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753374_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753374_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 12:15:57 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 12:15:57\tMarkDuplicates\tStart of doWork freeMemory: 528819592; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:15:57\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 12:15:57\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam\n",
      "INFO\t2023-08-17 12:15:59\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:88,348,059\n",
      "INFO\t2023-08-17 12:15:59\tMarkDuplicates\tTracking 230 as yet unmatched pairs. 230 records in RAM.\n",
      "INFO\t2023-08-17 12:16:00\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 5:177,739,566\n",
      "INFO\t2023-08-17 12:16:00\tMarkDuplicates\tTracking 65 as yet unmatched pairs. 65 records in RAM.\n",
      "INFO\t2023-08-17 12:16:00\tMarkDuplicates\tRead 2038926 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 12:16:00\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 270820624; totalMemory: 560988160; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:16:00\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 12:16:00\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 12:16:01\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 12:16:01\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 12:16:01\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 874495784; totalMemory: 1420820480; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:16:01\tMarkDuplicates\tMarking 219065 records as duplicates.\n",
      "INFO\t2023-08-17 12:16:01\tMarkDuplicates\tFound 73632 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr5_sorted.bam\n",
      "INFO\t2023-08-17 12:16:01\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 12:16:18\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 12:16:18\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 12:16:18\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 12:16:18\tMarkDuplicates\tBefore output close freeMemory: 528116784; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:16:18\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 12:16:18\tMarkDuplicates\tAfter output close freeMemory: 528116784; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 12:16:18 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.36 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 12:16:20 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T121058.620135.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:16:21 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753375_chr5_1_fastp.fastq.gz, data/processed/ERR753375_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753375_chr5.sam, results/mapped_reads/ERR753375_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753375_chr5_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753375_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.62 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 69.41 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.53 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.38 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 40.00 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 116.891 sec; CPU: 110.940 sec\n",
      "[Thu Aug 17 12:19:25 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:19:25 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753375_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr5.bam, results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753375_chr5.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753375_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 12:19:52 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:19:52 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753375_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753375_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "12:19:53.538 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 12:19:53 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753375_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753375_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 12:19:53 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 12:19:53\tMarkDuplicates\tStart of doWork freeMemory: 528785872; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:19:53\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 12:19:53\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam\n",
      "INFO\t2023-08-17 12:19:55\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:136,231,424\n",
      "INFO\t2023-08-17 12:19:55\tMarkDuplicates\tTracking 189 as yet unmatched pairs. 189 records in RAM.\n",
      "INFO\t2023-08-17 12:19:55\tMarkDuplicates\tRead 1429511 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 12:19:55\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 311484504; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:19:55\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 12:19:55\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 12:19:56\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 12:19:56\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 12:19:56\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 761216464; totalMemory: 1307574272; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:19:56\tMarkDuplicates\tMarking 67769 records as duplicates.\n",
      "INFO\t2023-08-17 12:19:56\tMarkDuplicates\tFound 153 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr5_sorted.bam\n",
      "INFO\t2023-08-17 12:19:56\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 12:20:05\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 12:20:05\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 12:20:05\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 12:20:05\tMarkDuplicates\tBefore output close freeMemory: 528083768; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:20:05\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 12:20:05\tMarkDuplicates\tAfter output close freeMemory: 528083768; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 12:20:05 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.20 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 12:20:06 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T121620.604769.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:20:08 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753376_chr5_1_fastp.fastq.gz, data/processed/ERR753376_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753376_chr5.sam, results/mapped_reads/ERR753376_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753376_chr5_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR753376_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.60 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 73.52 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.56 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.39 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 40.52 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 121.906 sec; CPU: 115.597 sec\n",
      "[Thu Aug 17 12:23:27 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:23:27 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753376_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr5.bam, results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753376_chr5.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753376_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 12:24:01 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:24:01 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753376_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753376_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "12:24:02.228 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 12:24:02 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753376_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753376_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 12:24:02 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 12:24:02\tMarkDuplicates\tStart of doWork freeMemory: 528826208; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:24:02\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 12:24:02\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam\n",
      "INFO\t2023-08-17 12:24:03\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:109,378,590\n",
      "INFO\t2023-08-17 12:24:03\tMarkDuplicates\tTracking 265 as yet unmatched pairs. 265 records in RAM.\n",
      "INFO\t2023-08-17 12:24:04\tMarkDuplicates\tRead 1758630 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 12:24:04\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 276166144; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:24:04\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 12:24:04\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 12:24:05\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 12:24:05\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 12:24:05\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 797956736; totalMemory: 1344274432; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:24:05\tMarkDuplicates\tMarking 249637 records as duplicates.\n",
      "INFO\t2023-08-17 12:24:05\tMarkDuplicates\tFound 80 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr5_sorted.bam\n",
      "INFO\t2023-08-17 12:24:05\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 12:24:16\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 12:24:16\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 12:24:16\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 12:24:16\tMarkDuplicates\tBefore output close freeMemory: 528123712; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:24:16\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 12:24:16\tMarkDuplicates\tAfter output close freeMemory: 528123712; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 12:24:16 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.24 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 12:24:17 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T122006.938912.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:24:19 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753377_chr5_1_fastp.fastq.gz, data/processed/ERR753377_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753377_chr5.sam, results/mapped_reads/ERR753377_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753377_chr5_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR753377_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.61 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 68.30 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.55 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.38 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 40.34 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 116.689 sec; CPU: 110.192 sec\n",
      "[Thu Aug 17 12:27:36 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:27:36 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753377_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr5.bam, results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753377_chr5.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753377_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 12:28:12 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:28:12 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753377_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753377_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "12:28:13.214 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 12:28:13 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753377_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753377_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 12:28:13 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 12:28:13\tMarkDuplicates\tStart of doWork freeMemory: 528790152; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:28:13\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 12:28:13\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam\n",
      "INFO\t2023-08-17 12:28:14\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:97,730,572\n",
      "INFO\t2023-08-17 12:28:14\tMarkDuplicates\tTracking 399 as yet unmatched pairs. 399 records in RAM.\n",
      "INFO\t2023-08-17 12:28:15\tMarkDuplicates\tRead 1869830 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 12:28:16\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 264108392; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:28:16\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 12:28:16\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 12:28:16\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 12:28:16\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 12:28:16\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 841961112; totalMemory: 1388314624; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:28:16\tMarkDuplicates\tMarking 263298 records as duplicates.\n",
      "INFO\t2023-08-17 12:28:16\tMarkDuplicates\tFound 72 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr5_sorted.bam\n",
      "INFO\t2023-08-17 12:28:16\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 12:28:28\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 12:28:28\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 12:28:28\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 12:28:28\tMarkDuplicates\tBefore output close freeMemory: 528087688; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:28:28\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 12:28:28\tMarkDuplicates\tAfter output close freeMemory: 528087688; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 12:28:28 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.25 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 12:28:29 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T122418.033648.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:28:31 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753378_chr5_1_fastp.fastq.gz, data/processed/ERR753378_chr5_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753378_chr5.sam, results/mapped_reads/ERR753378_chr5_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753378_chr5_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR753378_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.61 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=363076518, availableWord=37547260\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 61936310 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114423334 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161069910 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202525462 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239367270 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272108326 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301204614 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327061462 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350039062 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 70.92 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.56 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.38 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 40.21 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 118.775 sec; CPU: 112.679 sec\n",
      "[Thu Aug 17 12:32:32 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:32:32 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753378_chr5_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr5.bam, results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753378_chr5.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753378_chr5_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 12:33:14 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:33:14 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr5_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753378_chr5_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam\n",
      "    wildcards: sample=ERR753378_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "12:33:15.450 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 12:33:15 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753378_chr5_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753378_chr5_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 12:33:15 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 12:33:15\tMarkDuplicates\tStart of doWork freeMemory: 528819120; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:33:15\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 12:33:15\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam\n",
      "INFO\t2023-08-17 12:33:17\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 5:77,325,561\n",
      "INFO\t2023-08-17 12:33:17\tMarkDuplicates\tTracking 551 as yet unmatched pairs. 551 records in RAM.\n",
      "INFO\t2023-08-17 12:33:18\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 5:141,529,640\n",
      "INFO\t2023-08-17 12:33:18\tMarkDuplicates\tTracking 415 as yet unmatched pairs. 415 records in RAM.\n",
      "INFO\t2023-08-17 12:33:19\tMarkDuplicates\tRead 2755612 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 12:33:19\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 803580352; totalMemory: 1171259392; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:33:19\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 12:33:19\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 12:33:20\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 12:33:20\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 12:33:20\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 670024088; totalMemory: 1216348160; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:33:20\tMarkDuplicates\tMarking 66430 records as duplicates.\n",
      "INFO\t2023-08-17 12:33:20\tMarkDuplicates\tFound 462 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr5_sorted.bam\n",
      "INFO\t2023-08-17 12:33:20\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 12:33:36\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 12:33:36\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 12:33:36\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 12:33:36\tMarkDuplicates\tBefore output close freeMemory: 528117704; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 12:33:36\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 12:33:36\tMarkDuplicates\tAfter output close freeMemory: 528117704; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 12:33:36 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.35 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 12:33:38 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T122829.857580.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:33:40 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR696683_chr5_dedup.bam\n",
      "    output: results/variants/ERR696683_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR696683_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR696683_chr5.vcf\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "index file data/reference/genome.fa.fai not found, generating...\n",
      "[Thu Aug 17 12:34:29 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T123338.768892.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:34:31 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753368_chr5_dedup.bam\n",
      "    output: results/variants/ERR753368_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753368_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753368_chr5.vcf\n",
      "    wildcards: sample=ERR753368_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 12:35:36 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T123429.850422.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:35:38 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753369_chr5_dedup.bam\n",
      "    output: results/variants/ERR753369_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753369_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753369_chr5.vcf\n",
      "    wildcards: sample=ERR753369_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 12:36:45 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T123537.023729.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:36:46 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753370_chr5_dedup.bam\n",
      "    output: results/variants/ERR753370_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753370_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753370_chr5.vcf\n",
      "    wildcards: sample=ERR753370_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 12:38:33 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T123645.473928.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:38:34 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753371_chr5_dedup.bam\n",
      "    output: results/variants/ERR753371_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753371_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753371_chr5.vcf\n",
      "    wildcards: sample=ERR753371_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 12:39:09 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T123833.520385.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:39:11 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753372_chr5_dedup.bam\n",
      "    output: results/variants/ERR753372_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753372_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753372_chr5.vcf\n",
      "    wildcards: sample=ERR753372_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 12:40:18 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T123910.087707.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:40:19 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753373_chr5_dedup.bam\n",
      "    output: results/variants/ERR753373_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753373_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753373_chr5.vcf\n",
      "    wildcards: sample=ERR753373_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 12:41:47 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T124018.643621.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:41:48 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753374_chr5_dedup.bam\n",
      "    output: results/variants/ERR753374_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753374_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753374_chr5.vcf\n",
      "    wildcards: sample=ERR753374_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 12:42:46 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T124147.633815.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:42:47 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753375_chr5_dedup.bam\n",
      "    output: results/variants/ERR753375_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753375_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753375_chr5.vcf\n",
      "    wildcards: sample=ERR753375_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 12:43:27 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T124246.537412.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:43:29 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753376_chr5_dedup.bam\n",
      "    output: results/variants/ERR753376_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753376_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753376_chr5.vcf\n",
      "    wildcards: sample=ERR753376_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 12:44:10 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T124328.041437.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:44:11 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753377_chr5_dedup.bam\n",
      "    output: results/variants/ERR753377_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753377_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753377_chr5.vcf\n",
      "    wildcards: sample=ERR753377_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 12:44:53 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T124410.483998.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:44:55 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753378_chr5_dedup.bam\n",
      "    output: results/variants/ERR753378_chr5.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753378_chr5.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753378_chr5.vcf\n",
      "    wildcards: sample=ERR753378_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 12:45:55 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T124454.174729.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:45:57 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR696683_chr5.vcf\n",
      "    output: results/variants/vep/ERR696683_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR696683_chr5.txt\n",
      "    wildcards: sample=ERR696683_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 12:46:21 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T124555.941361.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:46:23 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753368_chr5.vcf\n",
      "    output: results/variants/vep/ERR753368_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753368_chr5.txt\n",
      "    wildcards: sample=ERR753368_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 12:46:50 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T124621.769431.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:46:51 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753369_chr5.vcf\n",
      "    output: results/variants/vep/ERR753369_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753369_chr5.txt\n",
      "    wildcards: sample=ERR753369_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 12:47:17 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T124650.670672.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:47:18 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753370_chr5.vcf\n",
      "    output: results/variants/vep/ERR753370_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753370_chr5.txt\n",
      "    wildcards: sample=ERR753370_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 12:47:46 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T124717.740815.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:47:48 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753371_chr5.vcf\n",
      "    output: results/variants/vep/ERR753371_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753371_chr5.txt\n",
      "    wildcards: sample=ERR753371_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 12:48:08 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T124746.943294.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:48:09 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753372_chr5.vcf\n",
      "    output: results/variants/vep/ERR753372_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753372_chr5.txt\n",
      "    wildcards: sample=ERR753372_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 12:48:35 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T124808.230333.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:48:37 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753373_chr5.vcf\n",
      "    output: results/variants/vep/ERR753373_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753373_chr5.txt\n",
      "    wildcards: sample=ERR753373_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 12:49:04 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T124836.058392.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:49:06 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753374_chr5.vcf\n",
      "    output: results/variants/vep/ERR753374_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753374_chr5.txt\n",
      "    wildcards: sample=ERR753374_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 12:49:27 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T124905.022072.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:49:29 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753375_chr5.vcf\n",
      "    output: results/variants/vep/ERR753375_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753375_chr5.txt\n",
      "    wildcards: sample=ERR753375_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 12:49:49 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T124927.782203.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:49:51 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753376_chr5.vcf\n",
      "    output: results/variants/vep/ERR753376_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753376_chr5.txt\n",
      "    wildcards: sample=ERR753376_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 12:50:13 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T124949.812364.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:50:14 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753377_chr5.vcf\n",
      "    output: results/variants/vep/ERR753377_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753377_chr5.txt\n",
      "    wildcards: sample=ERR753377_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 12:50:37 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T125013.232026.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:50:38 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753378_chr5.vcf\n",
      "    output: results/variants/vep/ERR753378_chr5.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753378_chr5.txt\n",
      "    wildcards: sample=ERR753378_chr5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 12:51:04 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T125037.233547.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job              count    min threads    max threads\n",
      "-------------  -------  -------------  -------------\n",
      "parsing_dataR        1              1              1\n",
      "total                1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:51:04 2023]\n",
      "rule parsing_dataR:\n",
      "    input: code/05parsing_vep_data.R\n",
      "    output: tasks/13parsing_dataR.done\n",
      "    jobid: 0\n",
      "    reason: Params have changed since last execution\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Warning message:\n",
      "package ‘glue’ was built under R version 4.2.3 \n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">>> This is your sample list: ERR696683_chr5\n",
      ">>> This is your sample list: ERR753368_chr5\n",
      ">>> This is your sample list: ERR753369_chr5\n",
      ">>> This is your sample list: ERR753370_chr5\n",
      ">>> This is your sample list: ERR753371_chr5\n",
      ">>> This is your sample list: ERR753372_chr5\n",
      ">>> This is your sample list: ERR753373_chr5\n",
      ">>> This is your sample list: ERR753374_chr5\n",
      ">>> This is your sample list: ERR753375_chr5\n",
      ">>> This is your sample list: ERR753376_chr5\n",
      ">>> This is your sample list: ERR753377_chr5\n",
      ">>> This is your sample list: ERR753378_chr5\n",
      ">>> This is your gene: APC\n",
      "# A tibble: 0 × 44\n",
      "# ℹ 44 variables: #uploaded_variation <chr>, location <chr>, allele <chr>,\n",
      "#   gene <chr>, feature <chr>, feature_type <chr>, consequence <chr>,\n",
      "#   cdna_position <chr>, cds_position <chr>, protein_position <chr>,\n",
      "#   amino_acids <chr>, codons <chr>, existing_variation <chr>, impact <chr>,\n",
      "#   distance <chr>, strand <chr>, flags <chr>, variant_class <chr>,\n",
      "#   symbol <chr>, symbol_source <chr>, hgnc_id <chr>, biotype <chr>,\n",
      "#   mane_select <chr>, mane_plus_clinical <chr>, tsl <chr>, appris <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 71 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112707850_C/T       5:1127078… T      ENSG… ENST00… Transcript   missense_v…\n",
      "2 5_112707850_C/T       5:1127078… T      ENSG… ENST00… Transcript   missense_v…\n",
      "3 5_112707850_C/T       5:1127078… T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 65 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 69 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112724597_G/C       5:1127245… C      ENSG… ENST00… Transcript   intron_var…\n",
      "2 5_112724597_G/C       5:1127245… C      ENSG… ENST00… Transcript   intron_var…\n",
      "3 5_112724597_G/C       5:1127245… C      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 63 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 86 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "2 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "3 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112778038_C/T       5:1127780… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 80 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 0 × 44\n",
      "# ℹ 44 variables: #uploaded_variation <chr>, location <chr>, allele <chr>,\n",
      "#   gene <chr>, feature <chr>, feature_type <chr>, consequence <chr>,\n",
      "#   cdna_position <chr>, cds_position <chr>, protein_position <chr>,\n",
      "#   amino_acids <chr>, codons <chr>, existing_variation <chr>, impact <chr>,\n",
      "#   distance <chr>, strand <chr>, flags <chr>, variant_class <chr>,\n",
      "#   symbol <chr>, symbol_source <chr>, hgnc_id <chr>, biotype <chr>,\n",
      "#   mane_select <chr>, mane_plus_clinical <chr>, tsl <chr>, appris <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 64 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112707850_C/T       5:1127078… T      ENSG… ENST00… Transcript   missense_v…\n",
      "2 5_112707850_C/T       5:1127078… T      ENSG… ENST00… Transcript   missense_v…\n",
      "3 5_112707850_C/T       5:1127078… T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 58 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 78 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112775373_G/A       5:1127753… A      ENSG… ENST00… Transcript   intron_var…\n",
      "2 5_112775373_G/A       5:1127753… A      ENSG… ENST00… Transcript   intron_var…\n",
      "3 5_112775373_G/A       5:1127753… A      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112775373_G/A       5:1127753… A      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112775373_G/A       5:1127753… A      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 72 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 59 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112729129_G/C       5:1127291… C      ENSG… ENST00… Transcript   intron_var…\n",
      "2 5_112729129_G/C       5:1127291… C      ENSG… ENST00… Transcript   intron_var…\n",
      "3 5_112729129_G/C       5:1127291… C      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112775373_G/A       5:1127753… A      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112775373_G/A       5:1127753… A      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112775373_G/A       5:1127753… A      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 53 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 61 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "2 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "3 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112778038_C/T       5:1127780… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 55 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 71 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "2 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "3 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112778038_C/T       5:1127780… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 65 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 78 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "2 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "3 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112778038_C/T       5:1127780… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 72 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 66 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "2 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "3 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 5_112777830_A/T       5:1127778… T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 5_112778038_C/T       5:1127780… T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 60 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "[1] \"##==========================##\"\n",
      "[1] \"##===> WORK FINISHED!!! <===##\"\n",
      "[1] \"##==========================##\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Touching output file tasks/13parsing_dataR.done.\n",
      "[Thu Aug 17 12:51:08 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T125104.219586.snakemake.log\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['snakemake', '--cores', '1', 'parsing_dataR'], returncode=0)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Running the workflow for the 5º chormosome\n",
    "\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--force\", \"data/reference/genome.fa\"]) ## rule reference_genome \n",
    "\n",
    "for sample in sample_list:\n",
    "    ## Quality inspection\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/fastqc_result/\" + sample + \"_chr5_1_fastqc.html\",   \n",
    "                    \"data/processed/\" + sample + \"_chr5_1_fastp.fastq.gz\",   \n",
    "                    \"results/fastqc_result/trimmed/\" + sample + \"_chr5_1_fastp_fastqc.html\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    ## Mapping reads\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/mapped_reads/\" + sample + \"_chr5_sorted.sam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr5_sorted.bam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr5_dedup.bam\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    # Extracting variants  \n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/\" + sample + \"_chr5.vcf\"])   \n",
    "\n",
    "## VEP CLI\n",
    "for sample in sample_list:\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/vep/\" + sample + \"_chr5.txt\"])\n",
    "\n",
    "## Extracting the gene PIKCA3 from the VEP files\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"parsing_dataR\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Changing the config file to do the chromosome 7 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['sed', '-i', '50s/APC/BRAF/g', 'config.yaml'], returncode=0)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subprocess.run([\"sed\", \"-i\", \"6,+12s/chr5/chr7/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\",\"24s/chr5/chr7/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"31s/chromosome.5.fa.gz/chromosome.7.fa.gz/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"50s/APC/BRAF/g\", \"config.yaml\"])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job                 count    min threads    max threads\n",
      "----------------  -------  -------------  -------------\n",
      "reference_genome        1              1              1\n",
      "total                   1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:55:48 2023]\n",
      "rule reference_genome:\n",
      "    output: data/reference/genome.fa\n",
      "    jobid: 0\n",
      "    reason: Forced execution\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "--2023-08-17 12:55:48--  https://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.7.fa.gz\n",
      "Resolviendo ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.169\n",
      "Conectando con ftp.ensembl.org (ftp.ensembl.org)[193.62.193.169]:443... conectado.\n",
      "Petición HTTP enviada, esperando respuesta... 200 OK\n",
      "Longitud: 47416846 (45M) [application/x-gzip]\n",
      "Guardando como: ‘data/reference/genome.fa.gz’\n",
      "\n",
      "     0K .......... .......... .......... .......... ..........  0%  428K 1m48s\n",
      "    50K .......... .......... .......... .......... ..........  0% 15,3M 55s\n",
      "   100K .......... .......... .......... .......... ..........  0%  865K 55s\n",
      "   150K .......... .......... .......... .......... ..........  0% 86,0M 41s\n",
      "   200K .......... .......... .......... .......... ..........  0%  900K 43s\n",
      "   250K .......... .......... .......... .......... ..........  0% 44,9M 36s\n",
      "   300K .......... .......... .......... .......... ..........  0% 28,5M 31s\n",
      "   350K .......... .......... .......... .......... ..........  0% 26,7M 27s\n",
      "   400K .......... .......... .......... .......... ..........  0%  126M 24s\n",
      "   450K .......... .......... .......... .......... ..........  1%  919K 27s\n",
      "   500K .......... .......... .......... .......... ..........  1% 98,4M 24s\n",
      "   550K .......... .......... .......... .......... ..........  1%  107M 22s\n",
      "   600K .......... .......... .......... .......... ..........  1% 92,6M 21s\n",
      "   650K .......... .......... .......... .......... ..........  1%  134M 19s\n",
      "   700K .......... .......... .......... .......... ..........  1% 21,2M 18s\n",
      "   750K .......... .......... .......... .......... ..........  1%  102M 17s\n",
      "   800K .......... .......... .......... .......... ..........  1%  105M 16s\n",
      "   850K .......... .......... .......... .......... ..........  1% 92,4M 15s\n",
      "   900K .......... .......... .......... .......... ..........  2% 1,31M 16s\n",
      "   950K .......... .......... .......... .......... ..........  2% 3,29M 16s\n",
      "  1000K .......... .......... .......... .......... ..........  2% 21,0M 15s\n",
      "  1050K .......... .......... .......... .......... ..........  2% 26,3M 15s\n",
      "  1100K .......... .......... .......... .......... ..........  2%  117M 14s\n",
      "  1150K .......... .......... .......... .......... ..........  2%  160M 13s\n",
      "  1200K .......... .......... .......... .......... ..........  2%  142M 13s\n",
      "  1250K .......... .......... .......... .......... ..........  2%  161M 12s\n",
      "  1300K .......... .......... .......... .......... ..........  2%  165M 12s\n",
      "  1350K .......... .......... .......... .......... ..........  3%  140M 11s\n",
      "  1400K .......... .......... .......... .......... ..........  3%  168M 11s\n",
      "  1450K .......... .......... .......... .......... ..........  3%  150M 11s\n",
      "  1500K .......... .......... .......... .......... ..........  3%  171M 10s\n",
      "  1550K .......... .......... .......... .......... ..........  3%  166M 10s\n",
      "  1600K .......... .......... .......... .......... ..........  3%  134M 10s\n",
      "  1650K .......... .......... .......... .......... ..........  3%  165M 9s\n",
      "  1700K .......... .......... .......... .......... ..........  3%  135M 9s\n",
      "  1750K .......... .......... .......... .......... ..........  3%  161M 9s\n",
      "  1800K .......... .......... .......... .......... ..........  3%  161M 9s\n",
      "  1850K .......... .......... .......... .......... ..........  4%  143M 8s\n",
      "  1900K .......... .......... .......... .......... ..........  4% 1,01M 9s\n",
      "  1950K .......... .......... .......... .......... ..........  4%  105M 9s\n",
      "  2000K .......... .......... .......... .......... ..........  4% 39,2M 9s\n",
      "  2050K .......... .......... .......... .......... ..........  4% 19,7M 9s\n",
      "  2100K .......... .......... .......... .......... ..........  4% 39,6M 8s\n",
      "  2150K .......... .......... .......... .......... ..........  4% 40,5M 8s\n",
      "  2200K .......... .......... .......... .......... ..........  4% 87,9M 8s\n",
      "  2250K .......... .......... .......... .......... ..........  4%  106M 8s\n",
      "  2300K .......... .......... .......... .......... ..........  5%  107M 8s\n",
      "  2350K .......... .......... .......... .......... ..........  5% 90,5M 8s\n",
      "  2400K .......... .......... .......... .......... ..........  5%  107M 7s\n",
      "  2450K .......... .......... .......... .......... ..........  5% 91,9M 7s\n",
      "  2500K .......... .......... .......... .......... ..........  5%  107M 7s\n",
      "  2550K .......... .......... .......... .......... ..........  5%  106M 7s\n",
      "  2600K .......... .......... .......... .......... ..........  5%  100M 7s\n",
      "  2650K .......... .......... .......... .......... ..........  5%  120M 7s\n",
      "  2700K .......... .......... .......... .......... ..........  5%  100M 7s\n",
      "  2750K .......... .......... .......... .......... ..........  6%  147M 7s\n",
      "  2800K .......... .......... .......... .......... ..........  6%  150M 6s\n",
      "  2850K .......... .......... .......... .......... ..........  6%  114M 6s\n",
      "  2900K .......... .......... .......... .......... ..........  6%  135M 6s\n",
      "  2950K .......... .......... .......... .......... ..........  6%  117M 6s\n",
      "  3000K .......... .......... .......... .......... ..........  6%  103M 6s\n",
      "  3050K .......... .......... .......... .......... ..........  6%  105M 6s\n",
      "  3100K .......... .......... .......... .......... ..........  6% 89,2M 6s\n",
      "  3150K .......... .......... .......... .......... ..........  6%  101M 6s\n",
      "  3200K .......... .......... .......... .......... ..........  7% 90,1M 6s\n",
      "  3250K .......... .......... .......... .......... ..........  7%  108M 6s\n",
      "  3300K .......... .......... .......... .......... ..........  7%  173M 5s\n",
      "  3350K .......... .......... .......... .......... ..........  7%  163M 5s\n",
      "  3400K .......... .......... .......... .......... ..........  7%  190M 5s\n",
      "  3450K .......... .......... .......... .......... ..........  7%  157M 5s\n",
      "  3500K .......... .......... .......... .......... ..........  7%  177M 5s\n",
      "  3550K .......... .......... .......... .......... ..........  7%  186M 5s\n",
      "  3600K .......... .......... .......... .......... ..........  7%  170M 5s\n",
      "  3650K .......... .......... .......... .......... ..........  7%  312M 5s\n",
      "  3700K .......... .......... .......... .......... ..........  8%  307M 5s\n",
      "  3750K .......... .......... .......... .......... ..........  8%  419M 5s\n",
      "  3800K .......... .......... .......... .......... ..........  8% 1,95M 5s\n",
      "  3850K .......... .......... .......... .......... ..........  8% 3,30M 5s\n",
      "  3900K .......... .......... .......... .......... ..........  8%  125M 5s\n",
      "  3950K .......... .......... .......... .......... ..........  8%  120M 5s\n",
      "  4000K .......... .......... .......... .......... ..........  8% 51,8M 5s\n",
      "  4050K .......... .......... .......... .......... ..........  8% 53,0M 5s\n",
      "  4100K .......... .......... .......... .......... ..........  8% 52,4M 5s\n",
      "  4150K .......... .......... .......... .......... ..........  9% 73,3M 5s\n",
      "  4200K .......... .......... .......... .......... ..........  9% 60,0M 5s\n",
      "  4250K .......... .......... .......... .......... ..........  9%  111M 5s\n",
      "  4300K .......... .......... .......... .......... ..........  9%  121M 5s\n",
      "  4350K .......... .......... .......... .......... ..........  9%  106M 5s\n",
      "  4400K .......... .......... .......... .......... ..........  9%  101M 4s\n",
      "  4450K .......... .......... .......... .......... ..........  9%  131M 4s\n",
      "  4500K .......... .......... .......... .......... ..........  9%  173M 4s\n",
      "  4550K .......... .......... .......... .......... ..........  9%  158M 4s\n",
      "  4600K .......... .......... .......... .......... .......... 10%  102M 4s\n",
      "  4650K .......... .......... .......... .......... .......... 10%  123M 4s\n",
      "  4700K .......... .......... .......... .......... .......... 10% 86,3M 4s\n",
      "  4750K .......... .......... .......... .......... .......... 10%  110M 4s\n",
      "  4800K .......... .......... .......... .......... .......... 10%  112M 4s\n",
      "  4850K .......... .......... .......... .......... .......... 10% 93,2M 4s\n",
      "  4900K .......... .......... .......... .......... .......... 10%  105M 4s\n",
      "  4950K .......... .......... .......... .......... .......... 10% 94,1M 4s\n",
      "  5000K .......... .......... .......... .......... .......... 10%  111M 4s\n",
      "  5050K .......... .......... .......... .......... .......... 11%  109M 4s\n",
      "  5100K .......... .......... .......... .......... .......... 11%  120M 4s\n",
      "  5150K .......... .......... .......... .......... .......... 11%  178M 4s\n",
      "  5200K .......... .......... .......... .......... .......... 11%  145M 4s\n",
      "  5250K .......... .......... .......... .......... .......... 11%  174M 4s\n",
      "  5300K .......... .......... .......... .......... .......... 11%  135M 4s\n",
      "  5350K .......... .......... .......... .......... .......... 11%  103M 4s\n",
      "  5400K .......... .......... .......... .......... .......... 11%  121M 4s\n",
      "  5450K .......... .......... .......... .......... .......... 11%  105M 4s\n",
      "  5500K .......... .......... .......... .......... .......... 11%  121M 4s\n",
      "  5550K .......... .......... .......... .......... .......... 12%  117M 4s\n",
      "  5600K .......... .......... .......... .......... .......... 12% 99,8M 3s\n",
      "  5650K .......... .......... .......... .......... .......... 12%  122M 3s\n",
      "  5700K .......... .......... .......... .......... .......... 12%  104M 3s\n",
      "  5750K .......... .......... .......... .......... .......... 12%  115M 3s\n",
      "  5800K .......... .......... .......... .......... .......... 12%  124M 3s\n",
      "  5850K .......... .......... .......... .......... .......... 12%  102M 3s\n",
      "  5900K .......... .......... .......... .......... .......... 12%  117M 3s\n",
      "  5950K .......... .......... .......... .......... .......... 12%  101M 3s\n",
      "  6000K .......... .......... .......... .......... .......... 13%  105M 3s\n",
      "  6050K .......... .......... .......... .......... .......... 13%  144M 3s\n",
      "  6100K .......... .......... .......... .......... .......... 13%  102M 3s\n",
      "  6150K .......... .......... .......... .......... .......... 13%  117M 3s\n",
      "  6200K .......... .......... .......... .......... .......... 13%  102M 3s\n",
      "  6250K .......... .......... .......... .......... .......... 13%  121M 3s\n",
      "  6300K .......... .......... .......... .......... .......... 13%  117M 3s\n",
      "  6350K .......... .......... .......... .......... .......... 13%  103M 3s\n",
      "  6400K .......... .......... .......... .......... .......... 13%  121M 3s\n",
      "  6450K .......... .......... .......... .......... .......... 14%  103M 3s\n",
      "  6500K .......... .......... .......... .......... .......... 14%  117M 3s\n",
      "  6550K .......... .......... .......... .......... .......... 14%  121M 3s\n",
      "  6600K .......... .......... .......... .......... .......... 14%  102M 3s\n",
      "  6650K .......... .......... .......... .......... .......... 14%  122M 3s\n",
      "  6700K .......... .......... .......... .......... .......... 14% 2,81M 3s\n",
      "  6750K .......... .......... .......... .......... .......... 14% 40,5M 3s\n",
      "  6800K .......... .......... .......... .......... .......... 14% 4,08M 3s\n",
      "  6850K .......... .......... .......... .......... .......... 14% 31,7M 3s\n",
      "  6900K .......... .......... .......... .......... .......... 15% 36,6M 3s\n",
      "  6950K .......... .......... .......... .......... .......... 15% 97,1M 3s\n",
      "  7000K .......... .......... .......... .......... .......... 15%  112M 3s\n",
      "  7050K .......... .......... .......... .......... .......... 15%  115M 3s\n",
      "  7100K .......... .......... .......... .......... .......... 15% 95,3M 3s\n",
      "  7150K .......... .......... .......... .......... .......... 15% 45,4M 3s\n",
      "  7200K .......... .......... .......... .......... .......... 15% 37,3M 3s\n",
      "  7250K .......... .......... .......... .......... .......... 15% 73,4M 3s\n",
      "  7300K .......... .......... .......... .......... .......... 15%  116M 3s\n",
      "  7350K .......... .......... .......... .......... .......... 15% 70,8M 3s\n",
      "  7400K .......... .......... .......... .......... .......... 16%  112M 3s\n",
      "  7450K .......... .......... .......... .......... .......... 16% 99,0M 3s\n",
      "  7500K .......... .......... .......... .......... .......... 16%  114M 3s\n",
      "  7550K .......... .......... .......... .......... .......... 16%  110M 3s\n",
      "  7600K .......... .......... .......... .......... .......... 16% 95,2M 3s\n",
      "  7650K .......... .......... .......... .......... .......... 16%  112M 3s\n",
      "  7700K .......... .......... .......... .......... .......... 16% 99,0M 3s\n",
      "  7750K .......... .......... .......... .......... .......... 16%  114M 3s\n",
      "  7800K .......... .......... .......... .......... .......... 16%  115M 3s\n",
      "  7850K .......... .......... .......... .......... .......... 17% 97,9M 3s\n",
      "  7900K .......... .......... .......... .......... .......... 17%  115M 3s\n",
      "  7950K .......... .......... .......... .......... .......... 17% 99,8M 3s\n",
      "  8000K .......... .......... .......... .......... .......... 17%  111M 3s\n",
      "  8050K .......... .......... .......... .......... .......... 17% 61,2M 3s\n",
      "  8100K .......... .......... .......... .......... .......... 17% 90,7M 3s\n",
      "  8150K .......... .......... .......... .......... .......... 17%  118M 3s\n",
      "  8200K .......... .......... .......... .......... .......... 17% 96,9M 2s\n",
      "  8250K .......... .......... .......... .......... .......... 17%  109M 2s\n",
      "  8300K .......... .......... .......... .......... .......... 18%  111M 2s\n",
      "  8350K .......... .......... .......... .......... .......... 18% 87,3M 2s\n",
      "  8400K .......... .......... .......... .......... .......... 18% 98,3M 2s\n",
      "  8450K .......... .......... .......... .......... .......... 18% 93,1M 2s\n",
      "  8500K .......... .......... .......... .......... .......... 18%  106M 2s\n",
      "  8550K .......... .......... .......... .......... .......... 18%  107M 2s\n",
      "  8600K .......... .......... .......... .......... .......... 18% 91,9M 2s\n",
      "  8650K .......... .......... .......... .......... .......... 18%  111M 2s\n",
      "  8700K .......... .......... .......... .......... .......... 18% 89,2M 2s\n",
      "  8750K .......... .......... .......... .......... .......... 19%  107M 2s\n",
      "  8800K .......... .......... .......... .......... .......... 19%  100M 2s\n",
      "  8850K .......... .......... .......... .......... .......... 19%  172M 2s\n",
      "  8900K .......... .......... .......... .......... .......... 19%  200M 2s\n",
      "  8950K .......... .......... .......... .......... .......... 19%  177M 2s\n",
      "  9000K .......... .......... .......... .......... .......... 19%  199M 2s\n",
      "  9050K .......... .......... .......... .......... .......... 19%  205M 2s\n",
      "  9100K .......... .......... .......... .......... .......... 19%  165M 2s\n",
      "  9150K .......... .......... .......... .......... .......... 19%  212M 2s\n",
      "  9200K .......... .......... .......... .......... .......... 19%  176M 2s\n",
      "  9250K .......... .......... .......... .......... .......... 20%  131M 2s\n",
      "  9300K .......... .......... .......... .......... .......... 20%  117M 2s\n",
      "  9350K .......... .......... .......... .......... .......... 20%  102M 2s\n",
      "  9400K .......... .......... .......... .......... .......... 20%  122M 2s\n",
      "  9450K .......... .......... .......... .......... .......... 20%  102M 2s\n",
      "  9500K .......... .......... .......... .......... .......... 20%  117M 2s\n",
      "  9550K .......... .......... .......... .......... .......... 20%  122M 2s\n",
      "  9600K .......... .......... .......... .......... .......... 20%  103M 2s\n",
      "  9650K .......... .......... .......... .......... .......... 20%  117M 2s\n",
      "  9700K .......... .......... .......... .......... .......... 21%  102M 2s\n",
      "  9750K .......... .......... .......... .......... .......... 21%  121M 2s\n",
      "  9800K .......... .......... .......... .......... .......... 21% 2,94M 2s\n",
      "  9850K .......... .......... .......... .......... .......... 21% 21,1M 2s\n",
      "  9900K .......... .......... .......... .......... .......... 21% 3,87M 2s\n",
      "  9950K .......... .......... .......... .......... .......... 21% 59,9M 2s\n",
      " 10000K .......... .......... .......... .......... .......... 21% 66,7M 2s\n",
      " 10050K .......... .......... .......... .......... .......... 21% 33,8M 2s\n",
      " 10100K .......... .......... .......... .......... .......... 21% 32,6M 2s\n",
      " 10150K .......... .......... .......... .......... .......... 22%  109M 2s\n",
      " 10200K .......... .......... .......... .......... .......... 22% 51,7M 2s\n",
      " 10250K .......... .......... .......... .......... .......... 22%  124M 2s\n",
      " 10300K .......... .......... .......... .......... .......... 22% 76,2M 2s\n",
      " 10350K .......... .......... .......... .......... .......... 22% 40,4M 2s\n",
      " 10400K .......... .......... .......... .......... .......... 22% 50,1M 2s\n",
      " 10450K .......... .......... .......... .......... .......... 22% 80,2M 2s\n",
      " 10500K .......... .......... .......... .......... .......... 22%  230M 2s\n",
      " 10550K .......... .......... .......... .......... .......... 22%  219M 2s\n",
      " 10600K .......... .......... .......... .......... .......... 22%  119M 2s\n",
      " 10650K .......... .......... .......... .......... .......... 23%  195M 2s\n",
      " 10700K .......... .......... .......... .......... .......... 23%  247M 2s\n",
      " 10750K .......... .......... .......... .......... .......... 23%  294M 2s\n",
      " 10800K .......... .......... .......... .......... .......... 23%  297M 2s\n",
      " 10850K .......... .......... .......... .......... .......... 23%  240M 2s\n",
      " 10900K .......... .......... .......... .......... .......... 23%  279M 2s\n",
      " 10950K .......... .......... .......... .......... .......... 23%  169M 2s\n",
      " 11000K .......... .......... .......... .......... .......... 23%  125M 2s\n",
      " 11050K .......... .......... .......... .......... .......... 23%  116M 2s\n",
      " 11100K .......... .......... .......... .......... .......... 24%  102M 2s\n",
      " 11150K .......... .......... .......... .......... .......... 24%  122M 2s\n",
      " 11200K .......... .......... .......... .......... .......... 24% 98,9M 2s\n",
      " 11250K .......... .......... .......... .......... .......... 24%  123M 2s\n",
      " 11300K .......... .......... .......... .......... .......... 24%  121M 2s\n",
      " 11350K .......... .......... .......... .......... .......... 24%  102M 2s\n",
      " 11400K .......... .......... .......... .......... .......... 24%  122M 2s\n",
      " 11450K .......... .......... .......... .......... .......... 24%  102M 2s\n",
      " 11500K .......... .......... .......... .......... .......... 24%  119M 2s\n",
      " 11550K .......... .......... .......... .......... .......... 25%  121M 2s\n",
      " 11600K .......... .......... .......... .......... .......... 25%  102M 2s\n",
      " 11650K .......... .......... .......... .......... .......... 25%  117M 2s\n",
      " 11700K .......... .......... .......... .......... .......... 25%  102M 2s\n",
      " 11750K .......... .......... .......... .......... .......... 25%  121M 2s\n",
      " 11800K .......... .......... .......... .......... .......... 25%  118M 2s\n",
      " 11850K .......... .......... .......... .......... .......... 25%  103M 2s\n",
      " 11900K .......... .......... .......... .......... .......... 25%  121M 2s\n",
      " 11950K .......... .......... .......... .......... .......... 25%  102M 2s\n",
      " 12000K .......... .......... .......... .......... .......... 26%  121M 2s\n",
      " 12050K .......... .......... .......... .......... .......... 26%  117M 2s\n",
      " 12100K .......... .......... .......... .......... .......... 26% 98,4M 2s\n",
      " 12150K .......... .......... .......... .......... .......... 26%  127M 2s\n",
      " 12200K .......... .......... .......... .......... .......... 26%  102M 2s\n",
      " 12250K .......... .......... .......... .......... .......... 26%  117M 2s\n",
      " 12300K .......... .......... .......... .......... .......... 26%  121M 2s\n",
      " 12350K .......... .......... .......... .......... .......... 26%  102M 2s\n",
      " 12400K .......... .......... .......... .......... .......... 26%  117M 2s\n",
      " 12450K .......... .......... .......... .......... .......... 26%  102M 2s\n",
      " 12500K .......... .......... .......... .......... .......... 27%  121M 2s\n",
      " 12550K .......... .......... .......... .......... .......... 27%  121M 2s\n",
      " 12600K .......... .......... .......... .......... .......... 27%  102M 2s\n",
      " 12650K .......... .......... .......... .......... .......... 27%  117M 2s\n",
      " 12700K .......... .......... .......... .......... .......... 27%  102M 2s\n",
      " 12750K .......... .......... .......... .......... .......... 27%  121M 2s\n",
      " 12800K .......... .......... .......... .......... .......... 27% 3,40M 2s\n",
      " 12850K .......... .......... .......... .......... .......... 27% 20,1M 2s\n",
      " 12900K .......... .......... .......... .......... .......... 27% 4,05M 2s\n",
      " 12950K .......... .......... .......... .......... .......... 28% 27,2M 2s\n",
      " 13000K .......... .......... .......... .......... .......... 28% 46,9M 2s\n",
      " 13050K .......... .......... .......... .......... .......... 28% 29,4M 2s\n",
      " 13100K .......... .......... .......... .......... .......... 28%  109M 2s\n",
      " 13150K .......... .......... .......... .......... .......... 28% 51,9M 2s\n",
      " 13200K .......... .......... .......... .......... .......... 28% 48,7M 2s\n",
      " 13250K .......... .......... .......... .......... .......... 28% 61,1M 2s\n",
      " 13300K .......... .......... .......... .......... .......... 28%  121M 2s\n",
      " 13350K .......... .......... .......... .......... .......... 28% 47,5M 2s\n",
      " 13400K .......... .......... .......... .......... .......... 29% 54,7M 2s\n",
      " 13450K .......... .......... .......... .......... .......... 29% 33,4M 2s\n",
      " 13500K .......... .......... .......... .......... .......... 29%  120M 2s\n",
      " 13550K .......... .......... .......... .......... .......... 29%  122M 2s\n",
      " 13600K .......... .......... .......... .......... .......... 29%  102M 2s\n",
      " 13650K .......... .......... .......... .......... .......... 29%  117M 2s\n",
      " 13700K .......... .......... .......... .......... .......... 29%  102M 2s\n",
      " 13750K .......... .......... .......... .......... .......... 29%  122M 2s\n",
      " 13800K .......... .......... .......... .......... .......... 29%  115M 2s\n",
      " 13850K .......... .......... .......... .......... .......... 30%  104M 2s\n",
      " 13900K .......... .......... .......... .......... .......... 30%  121M 2s\n",
      " 13950K .......... .......... .......... .......... .......... 30%  102M 2s\n",
      " 14000K .......... .......... .......... .......... .......... 30%  116M 2s\n",
      " 14050K .......... .......... .......... .......... .......... 30%  123M 2s\n",
      " 14100K .......... .......... .......... .......... .......... 30%  102M 1s\n",
      " 14150K .......... .......... .......... .......... .......... 30%  121M 1s\n",
      " 14200K .......... .......... .......... .......... .......... 30%  102M 1s\n",
      " 14250K .......... .......... .......... .......... .......... 30%  115M 1s\n",
      " 14300K .......... .......... .......... .......... .......... 30%  122M 1s\n",
      " 14350K .......... .......... .......... .......... .......... 31%  101M 1s\n",
      " 14400K .......... .......... .......... .......... .......... 31%  119M 1s\n",
      " 14450K .......... .......... .......... .......... .......... 31%  103M 1s\n",
      " 14500K .......... .......... .......... .......... .......... 31%  121M 1s\n",
      " 14550K .......... .......... .......... .......... .......... 31%  118M 1s\n",
      " 14600K .......... .......... .......... .......... .......... 31%  102M 1s\n",
      " 14650K .......... .......... .......... .......... .......... 31%  121M 1s\n",
      " 14700K .......... .......... .......... .......... .......... 31%  102M 1s\n",
      " 14750K .......... .......... .......... .......... .......... 31%  121M 1s\n",
      " 14800K .......... .......... .......... .......... .......... 32%  117M 1s\n",
      " 14850K .......... .......... .......... .......... .......... 32%  102M 1s\n",
      " 14900K .......... .......... .......... .......... .......... 32%  121M 1s\n",
      " 14950K .......... .......... .......... .......... .......... 32%  102M 1s\n",
      " 15000K .......... .......... .......... .......... .......... 32%  117M 1s\n",
      " 15050K .......... .......... .......... .......... .......... 32%  121M 1s\n",
      " 15100K .......... .......... .......... .......... .......... 32%  102M 1s\n",
      " 15150K .......... .......... .......... .......... .......... 32%  121M 1s\n",
      " 15200K .......... .......... .......... .......... .......... 32%  102M 1s\n",
      " 15250K .......... .......... .......... .......... .......... 33%  117M 1s\n",
      " 15300K .......... .......... .......... .......... .......... 33%  121M 1s\n",
      " 15350K .......... .......... .......... .......... .......... 33%  102M 1s\n",
      " 15400K .......... .......... .......... .......... .......... 33%  117M 1s\n",
      " 15450K .......... .......... .......... .......... .......... 33%  102M 1s\n",
      " 15500K .......... .......... .......... .......... .......... 33%  121M 1s\n",
      " 15550K .......... .......... .......... .......... .......... 33%  117M 1s\n",
      " 15600K .......... .......... .......... .......... .......... 33%  102M 1s\n",
      " 15650K .......... .......... .......... .......... .......... 33%  121M 1s\n",
      " 15700K .......... .......... .......... .......... .......... 34%  102M 1s\n",
      " 15750K .......... .......... .......... .......... .......... 34%  121M 1s\n",
      " 15800K .......... .......... .......... .......... .......... 34%  117M 1s\n",
      " 15850K .......... .......... .......... .......... .......... 34%  102M 1s\n",
      " 15900K .......... .......... .......... .......... .......... 34% 3,46M 1s\n",
      " 15950K .......... .......... .......... .......... .......... 34% 3,58M 1s\n",
      " 16000K .......... .......... .......... .......... .......... 34%  106M 1s\n",
      " 16050K .......... .......... .......... .......... .......... 34% 59,6M 1s\n",
      " 16100K .......... .......... .......... .......... .......... 34% 39,2M 1s\n",
      " 16150K .......... .......... .......... .......... .......... 34% 34,1M 1s\n",
      " 16200K .......... .......... .......... .......... .......... 35% 48,9M 1s\n",
      " 16250K .......... .......... .......... .......... .......... 35% 82,8M 1s\n",
      " 16300K .......... .......... .......... .......... .......... 35% 66,4M 1s\n",
      " 16350K .......... .......... .......... .......... .......... 35% 57,8M 1s\n",
      " 16400K .......... .......... .......... .......... .......... 35% 56,6M 1s\n",
      " 16450K .......... .......... .......... .......... .......... 35% 52,9M 1s\n",
      " 16500K .......... .......... .......... .......... .......... 35%  122M 1s\n",
      " 16550K .......... .......... .......... .......... .......... 35% 51,3M 1s\n",
      " 16600K .......... .......... .......... .......... .......... 35% 62,9M 1s\n",
      " 16650K .......... .......... .......... .......... .......... 36% 92,4M 1s\n",
      " 16700K .......... .......... .......... .......... .......... 36% 84,2M 1s\n",
      " 16750K .......... .......... .......... .......... .......... 36%  117M 1s\n",
      " 16800K .......... .......... .......... .......... .......... 36%  163M 1s\n",
      " 16850K .......... .......... .......... .......... .......... 36%  102M 1s\n",
      " 16900K .......... .......... .......... .......... .......... 36%  120M 1s\n",
      " 16950K .......... .......... .......... .......... .......... 36%  102M 1s\n",
      " 17000K .......... .......... .......... .......... .......... 36%  118M 1s\n",
      " 17050K .......... .......... .......... .......... .......... 36%  119M 1s\n",
      " 17100K .......... .......... .......... .......... .......... 37%  106M 1s\n",
      " 17150K .......... .......... .......... .......... .......... 37%  117M 1s\n",
      " 17200K .......... .......... .......... .......... .......... 37%  102M 1s\n",
      " 17250K .......... .......... .......... .......... .......... 37%  121M 1s\n",
      " 17300K .......... .......... .......... .......... .......... 37%  121M 1s\n",
      " 17350K .......... .......... .......... .......... .......... 37%  103M 1s\n",
      " 17400K .......... .......... .......... .......... .......... 37%  117M 1s\n",
      " 17450K .......... .......... .......... .......... .......... 37% 99,0M 1s\n",
      " 17500K .......... .......... .......... .......... .......... 37%  127M 1s\n",
      " 17550K .......... .......... .......... .......... .......... 38%  115M 1s\n",
      " 17600K .......... .......... .......... .......... .......... 38%  104M 1s\n",
      " 17650K .......... .......... .......... .......... .......... 38%  120M 1s\n",
      " 17700K .......... .......... .......... .......... .......... 38%  102M 1s\n",
      " 17750K .......... .......... .......... .......... .......... 38%  116M 1s\n",
      " 17800K .......... .......... .......... .......... .......... 38%  121M 1s\n",
      " 17850K .......... .......... .......... .......... .......... 38%  104M 1s\n",
      " 17900K .......... .......... .......... .......... .......... 38%  121M 1s\n",
      " 17950K .......... .......... .......... .......... .......... 38%  102M 1s\n",
      " 18000K .......... .......... .......... .......... .......... 38%  117M 1s\n",
      " 18050K .......... .......... .......... .......... .......... 39%  121M 1s\n",
      " 18100K .......... .......... .......... .......... .......... 39%  103M 1s\n",
      " 18150K .......... .......... .......... .......... .......... 39%  115M 1s\n",
      " 18200K .......... .......... .......... .......... .......... 39%  104M 1s\n",
      " 18250K .......... .......... .......... .......... .......... 39%  121M 1s\n",
      " 18300K .......... .......... .......... .......... .......... 39%  118M 1s\n",
      " 18350K .......... .......... .......... .......... .......... 39%  101M 1s\n",
      " 18400K .......... .......... .......... .......... .......... 39%  121M 1s\n",
      " 18450K .......... .......... .......... .......... .......... 39%  102M 1s\n",
      " 18500K .......... .......... .......... .......... .......... 40%  121M 1s\n",
      " 18550K .......... .......... .......... .......... .......... 40%  117M 1s\n",
      " 18600K .......... .......... .......... .......... .......... 40%  102M 1s\n",
      " 18650K .......... .......... .......... .......... .......... 40%  123M 1s\n",
      " 18700K .......... .......... .......... .......... .......... 40%  102M 1s\n",
      " 18750K .......... .......... .......... .......... .......... 40%  118M 1s\n",
      " 18800K .......... .......... .......... .......... .......... 40%  121M 1s\n",
      " 18850K .......... .......... .......... .......... .......... 40%  102M 1s\n",
      " 18900K .......... .......... .......... .......... .......... 40%  121M 1s\n",
      " 18950K .......... .......... .......... .......... .......... 41% 3,41M 1s\n",
      " 19000K .......... .......... .......... .......... .......... 41% 3,80M 1s\n",
      " 19050K .......... .......... .......... .......... .......... 41% 25,0M 1s\n",
      " 19100K .......... .......... .......... .......... .......... 41% 90,5M 1s\n",
      " 19150K .......... .......... .......... .......... .......... 41% 51,5M 1s\n",
      " 19200K .......... .......... .......... .......... .......... 41% 29,8M 1s\n",
      " 19250K .......... .......... .......... .......... .......... 41% 32,4M 1s\n",
      " 19300K .......... .......... .......... .......... .......... 41% 98,8M 1s\n",
      " 19350K .......... .......... .......... .......... .......... 41%  113M 1s\n",
      " 19400K .......... .......... .......... .......... .......... 42% 63,7M 1s\n",
      " 19450K .......... .......... .......... .......... .......... 42% 50,4M 1s\n",
      " 19500K .......... .......... .......... .......... .......... 42%  104M 1s\n",
      " 19550K .......... .......... .......... .......... .......... 42%  158M 1s\n",
      " 19600K .......... .......... .......... .......... .......... 42%  128M 1s\n",
      " 19650K .......... .......... .......... .......... .......... 42% 60,5M 1s\n",
      " 19700K .......... .......... .......... .......... .......... 42% 51,3M 1s\n",
      " 19750K .......... .......... .......... .......... .......... 42% 66,1M 1s\n",
      " 19800K .......... .......... .......... .......... .......... 42%  120M 1s\n",
      " 19850K .......... .......... .......... .......... .......... 42%  102M 1s\n",
      " 19900K .......... .......... .......... .......... .......... 43%  117M 1s\n",
      " 19950K .......... .......... .......... .......... .......... 43%  103M 1s\n",
      " 20000K .......... .......... .......... .......... .......... 43%  121M 1s\n",
      " 20050K .......... .......... .......... .......... .......... 43%  122M 1s\n",
      " 20100K .......... .......... .......... .......... .......... 43%  102M 1s\n",
      " 20150K .......... .......... .......... .......... .......... 43%  117M 1s\n",
      " 20200K .......... .......... .......... .......... .......... 43%  102M 1s\n",
      " 20250K .......... .......... .......... .......... .......... 43%  122M 1s\n",
      " 20300K .......... .......... .......... .......... .......... 43%  116M 1s\n",
      " 20350K .......... .......... .......... .......... .......... 44%  102M 1s\n",
      " 20400K .......... .......... .......... .......... .......... 44%  122M 1s\n",
      " 20450K .......... .......... .......... .......... .......... 44%  102M 1s\n",
      " 20500K .......... .......... .......... .......... .......... 44%  120M 1s\n",
      " 20550K .......... .......... .......... .......... .......... 44%  118M 1s\n",
      " 20600K .......... .......... .......... .......... .......... 44%  102M 1s\n",
      " 20650K .......... .......... .......... .......... .......... 44% 73,8M 1s\n",
      " 20700K .......... .......... .......... .......... .......... 44%  222M 1s\n",
      " 20750K .......... .......... .......... .......... .......... 44%  117M 1s\n",
      " 20800K .......... .......... .......... .......... .......... 45%  121M 1s\n",
      " 20850K .......... .......... .......... .......... .......... 45%  102M 1s\n",
      " 20900K .......... .......... .......... .......... .......... 45%  118M 1s\n",
      " 20950K .......... .......... .......... .......... .......... 45%  102M 1s\n",
      " 21000K .......... .......... .......... .......... .......... 45%  120M 1s\n",
      " 21050K .......... .......... .......... .......... .......... 45%  121M 1s\n",
      " 21100K .......... .......... .......... .......... .......... 45%  102M 1s\n",
      " 21150K .......... .......... .......... .......... .......... 45%  117M 1s\n",
      " 21200K .......... .......... .......... .......... .......... 45%  102M 1s\n",
      " 21250K .......... .......... .......... .......... .......... 45%  121M 1s\n",
      " 21300K .......... .......... .......... .......... .......... 46%  118M 1s\n",
      " 21350K .......... .......... .......... .......... .......... 46%  102M 1s\n",
      " 21400K .......... .......... .......... .......... .......... 46%  121M 1s\n",
      " 21450K .......... .......... .......... .......... .......... 46%  102M 1s\n",
      " 21500K .......... .......... .......... .......... .......... 46%  117M 1s\n",
      " 21550K .......... .......... .......... .......... .......... 46%  121M 1s\n",
      " 21600K .......... .......... .......... .......... .......... 46%  102M 1s\n",
      " 21650K .......... .......... .......... .......... .......... 46%  121M 1s\n",
      " 21700K .......... .......... .......... .......... .......... 46%  102M 1s\n",
      " 21750K .......... .......... .......... .......... .......... 47%  118M 1s\n",
      " 21800K .......... .......... .......... .......... .......... 47%  121M 1s\n",
      " 21850K .......... .......... .......... .......... .......... 47%  102M 1s\n",
      " 21900K .......... .......... .......... .......... .......... 47%  117M 1s\n",
      " 21950K .......... .......... .......... .......... .......... 47%  102M 1s\n",
      " 22000K .......... .......... .......... .......... .......... 47% 3,89M 1s\n",
      " 22050K .......... .......... .......... .......... .......... 47% 3,56M 1s\n",
      " 22100K .......... .......... .......... .......... .......... 47% 36,1M 1s\n",
      " 22150K .......... .......... .......... .......... .......... 47% 24,7M 1s\n",
      " 22200K .......... .......... .......... .......... .......... 48% 44,5M 1s\n",
      " 22250K .......... .......... .......... .......... .......... 48% 70,5M 1s\n",
      " 22300K .......... .......... .......... .......... .......... 48% 40,9M 1s\n",
      " 22350K .......... .......... .......... .......... .......... 48% 51,6M 1s\n",
      " 22400K .......... .......... .......... .......... .......... 48% 57,1M 1s\n",
      " 22450K .......... .......... .......... .......... .......... 48% 52,8M 1s\n",
      " 22500K .......... .......... .......... .......... .......... 48%  121M 1s\n",
      " 22550K .......... .......... .......... .......... .......... 48% 56,9M 1s\n",
      " 22600K .......... .......... .......... .......... .......... 48% 86,8M 1s\n",
      " 22650K .......... .......... .......... .......... .......... 49%  121M 1s\n",
      " 22700K .......... .......... .......... .......... .......... 49%  104M 1s\n",
      " 22750K .......... .......... .......... .......... .......... 49%  115M 1s\n",
      " 22800K .......... .......... .......... .......... .......... 49%  122M 1s\n",
      " 22850K .......... .......... .......... .......... .......... 49% 95,3M 1s\n",
      " 22900K .......... .......... .......... .......... .......... 49% 29,9M 1s\n",
      " 22950K .......... .......... .......... .......... .......... 49%  103M 1s\n",
      " 23000K .......... .......... .......... .......... .......... 49%  115M 1s\n",
      " 23050K .......... .......... .......... .......... .......... 49%  114M 1s\n",
      " 23100K .......... .......... .......... .......... .......... 49%  110M 1s\n",
      " 23150K .......... .......... .......... .......... .......... 50%  121M 1s\n",
      " 23200K .......... .......... .......... .......... .......... 50%  102M 1s\n",
      " 23250K .......... .......... .......... .......... .......... 50%  122M 1s\n",
      " 23300K .......... .......... .......... .......... .......... 50%  113M 1s\n",
      " 23350K .......... .......... .......... .......... .......... 50%  100M 1s\n",
      " 23400K .......... .......... .......... .......... .......... 50%  124M 1s\n",
      " 23450K .......... .......... .......... .......... .......... 50%  106M 1s\n",
      " 23500K .......... .......... .......... .......... .......... 50%  117M 1s\n",
      " 23550K .......... .......... .......... .......... .......... 50%  121M 1s\n",
      " 23600K .......... .......... .......... .......... .......... 51%  103M 1s\n",
      " 23650K .......... .......... .......... .......... .......... 51%  118M 1s\n",
      " 23700K .......... .......... .......... .......... .......... 51%  102M 1s\n",
      " 23750K .......... .......... .......... .......... .......... 51% 90,3M 1s\n",
      " 23800K .......... .......... .......... .......... .......... 51%  184M 1s\n",
      " 23850K .......... .......... .......... .......... .......... 51%  102M 1s\n",
      " 23900K .......... .......... .......... .......... .......... 51%  118M 1s\n",
      " 23950K .......... .......... .......... .......... .......... 51%  103M 1s\n",
      " 24000K .......... .......... .......... .......... .......... 51%  120M 1s\n",
      " 24050K .......... .......... .......... .......... .......... 52%  118M 1s\n",
      " 24100K .......... .......... .......... .......... .......... 52%  102M 1s\n",
      " 24150K .......... .......... .......... .......... .......... 52%  120M 1s\n",
      " 24200K .......... .......... .......... .......... .......... 52%  103M 1s\n",
      " 24250K .......... .......... .......... .......... .......... 52%  121M 1s\n",
      " 24300K .......... .......... .......... .......... .......... 52%  117M 1s\n",
      " 24350K .......... .......... .......... .......... .......... 52%  103M 1s\n",
      " 24400K .......... .......... .......... .......... .......... 52%  121M 1s\n",
      " 24450K .......... .......... .......... .......... .......... 52%  102M 1s\n",
      " 24500K .......... .......... .......... .......... .......... 53%  117M 1s\n",
      " 24550K .......... .......... .......... .......... .......... 53%  121M 1s\n",
      " 24600K .......... .......... .......... .......... .......... 53%  102M 1s\n",
      " 24650K .......... .......... .......... .......... .......... 53%  117M 1s\n",
      " 24700K .......... .......... .......... .......... .......... 53%  102M 1s\n",
      " 24750K .......... .......... .......... .......... .......... 53%  121M 1s\n",
      " 24800K .......... .......... .......... .......... .......... 53%  121M 1s\n",
      " 24850K .......... .......... .......... .......... .......... 53%  102M 1s\n",
      " 24900K .......... .......... .......... .......... .......... 53%  117M 1s\n",
      " 24950K .......... .......... .......... .......... .......... 53%  102M 1s\n",
      " 25000K .......... .......... .......... .......... .......... 54%  121M 1s\n",
      " 25050K .......... .......... .......... .......... .......... 54% 4,45M 1s\n",
      " 25100K .......... .......... .......... .......... .......... 54% 3,35M 1s\n",
      " 25150K .......... .......... .......... .......... .......... 54% 47,3M 1s\n",
      " 25200K .......... .......... .......... .......... .......... 54% 25,4M 1s\n",
      " 25250K .......... .......... .......... .......... .......... 54% 35,0M 1s\n",
      " 25300K .......... .......... .......... .......... .......... 54% 27,7M 1s\n",
      " 25350K .......... .......... .......... .......... .......... 54% 32,5M 1s\n",
      " 25400K .......... .......... .......... .......... .......... 54% 80,1M 1s\n",
      " 25450K .......... .......... .......... .......... .......... 55% 53,6M 1s\n",
      " 25500K .......... .......... .......... .......... .......... 55% 58,9M 1s\n",
      " 25550K .......... .......... .......... .......... .......... 55% 72,5M 1s\n",
      " 25600K .......... .......... .......... .......... .......... 55%  104M 1s\n",
      " 25650K .......... .......... .......... .......... .......... 55% 59,7M 1s\n",
      " 25700K .......... .......... .......... .......... .......... 55% 99,2M 1s\n",
      " 25750K .......... .......... .......... .......... .......... 55%  122M 1s\n",
      " 25800K .......... .......... .......... .......... .......... 55%  118M 1s\n",
      " 25850K .......... .......... .......... .......... .......... 55%  103M 1s\n",
      " 25900K .......... .......... .......... .......... .......... 56%  118M 1s\n",
      " 25950K .......... .......... .......... .......... .......... 56%  103M 1s\n",
      " 26000K .......... .......... .......... .......... .......... 56%  122M 1s\n",
      " 26050K .......... .......... .......... .......... .......... 56% 68,2M 1s\n",
      " 26100K .......... .......... .......... .......... .......... 56%  103M 1s\n",
      " 26150K .......... .......... .......... .......... .......... 56% 44,9M 1s\n",
      " 26200K .......... .......... .......... .......... .......... 56% 68,0M 1s\n",
      " 26250K .......... .......... .......... .......... .......... 56%  118M 1s\n",
      " 26300K .......... .......... .......... .......... .......... 56%  121M 1s\n",
      " 26350K .......... .......... .......... .......... .......... 57%  103M 1s\n",
      " 26400K .......... .......... .......... .......... .......... 57%  120M 1s\n",
      " 26450K .......... .......... .......... .......... .......... 57%  103M 1s\n",
      " 26500K .......... .......... .......... .......... .......... 57%  117M 1s\n",
      " 26550K .......... .......... .......... .......... .......... 57%  121M 1s\n",
      " 26600K .......... .......... .......... .......... .......... 57%  103M 1s\n",
      " 26650K .......... .......... .......... .......... .......... 57%  117M 1s\n",
      " 26700K .......... .......... .......... .......... .......... 57%  102M 1s\n",
      " 26750K .......... .......... .......... .......... .......... 57%  123M 1s\n",
      " 26800K .......... .......... .......... .......... .......... 57%  117M 1s\n",
      " 26850K .......... .......... .......... .......... .......... 58%  102M 1s\n",
      " 26900K .......... .......... .......... .......... .......... 58%  120M 1s\n",
      " 26950K .......... .......... .......... .......... .......... 58%  103M 1s\n",
      " 27000K .......... .......... .......... .......... .......... 58%  121M 1s\n",
      " 27050K .......... .......... .......... .......... .......... 58%  118M 1s\n",
      " 27100K .......... .......... .......... .......... .......... 58%  102M 1s\n",
      " 27150K .......... .......... .......... .......... .......... 58%  120M 1s\n",
      " 27200K .......... .......... .......... .......... .......... 58%  103M 1s\n",
      " 27250K .......... .......... .......... .......... .......... 58%  117M 1s\n",
      " 27300K .......... .......... .......... .......... .......... 59%  121M 1s\n",
      " 27350K .......... .......... .......... .......... .......... 59%  103M 1s\n",
      " 27400K .......... .......... .......... .......... .......... 59%  117M 1s\n",
      " 27450K .......... .......... .......... .......... .......... 59%  103M 1s\n",
      " 27500K .......... .......... .......... .......... .......... 59%  120M 1s\n",
      " 27550K .......... .......... .......... .......... .......... 59%  121M 1s\n",
      " 27600K .......... .......... .......... .......... .......... 59%  102M 1s\n",
      " 27650K .......... .......... .......... .......... .......... 59%  118M 1s\n",
      " 27700K .......... .......... .......... .......... .......... 59%  103M 1s\n",
      " 27750K .......... .......... .......... .......... .......... 60%  120M 1s\n",
      " 27800K .......... .......... .......... .......... .......... 60%  117M 1s\n",
      " 27850K .......... .......... .......... .......... .......... 60%  102M 1s\n",
      " 27900K .......... .......... .......... .......... .......... 60%  121M 1s\n",
      " 27950K .......... .......... .......... .......... .......... 60%  102M 1s\n",
      " 28000K .......... .......... .......... .......... .......... 60%  121M 1s\n",
      " 28050K .......... .......... .......... .......... .......... 60%  117M 1s\n",
      " 28100K .......... .......... .......... .......... .......... 60% 6,07M 1s\n",
      " 28150K .......... .......... .......... .......... .......... 60% 3,06M 1s\n",
      " 28200K .......... .......... .......... .......... .......... 61% 37,3M 1s\n",
      " 28250K .......... .......... .......... .......... .......... 61% 23,7M 1s\n",
      " 28300K .......... .......... .......... .......... .......... 61% 44,1M 1s\n",
      " 28350K .......... .......... .......... .......... .......... 61% 27,3M 1s\n",
      " 28400K .......... .......... .......... .......... .......... 61% 15,5M 1s\n",
      " 28450K .......... .......... .......... .......... .......... 61% 60,5M 1s\n",
      " 28500K .......... .......... .......... .......... .......... 61% 87,8M 1s\n",
      " 28550K .......... .......... .......... .......... .......... 61% 60,3M 1s\n",
      " 28600K .......... .......... .......... .......... .......... 61%  154M 1s\n",
      " 28650K .......... .......... .......... .......... .......... 61%  140M 1s\n",
      " 28700K .......... .......... .......... .......... .......... 62% 62,1M 1s\n",
      " 28750K .......... .......... .......... .......... .......... 62%  120M 1s\n",
      " 28800K .......... .......... .......... .......... .......... 62%  118M 1s\n",
      " 28850K .......... .......... .......... .......... .......... 62%  104M 1s\n",
      " 28900K .......... .......... .......... .......... .......... 62%  118M 1s\n",
      " 28950K .......... .......... .......... .......... .......... 62%  104M 1s\n",
      " 29000K .......... .......... .......... .......... .......... 62%  114M 1s\n",
      " 29050K .......... .......... .......... .......... .......... 62%  126M 1s\n",
      " 29100K .......... .......... .......... .......... .......... 62%  101M 1s\n",
      " 29150K .......... .......... .......... .......... .......... 63% 82,0M 1s\n",
      " 29200K .......... .......... .......... .......... .......... 63% 29,9M 1s\n",
      " 29250K .......... .......... .......... .......... .......... 63%  109M 1s\n",
      " 29300K .......... .......... .......... .......... .......... 63%  128M 1s\n",
      " 29350K .......... .......... .......... .......... .......... 63%  103M 1s\n",
      " 29400K .......... .......... .......... .......... .......... 63% 67,1M 1s\n",
      " 29450K .......... .......... .......... .......... .......... 63%  100M 1s\n",
      " 29500K .......... .......... .......... .......... .......... 63%  122M 1s\n",
      " 29550K .......... .......... .......... .......... .......... 63%  118M 1s\n",
      " 29600K .......... .......... .......... .......... .......... 64%  100M 1s\n",
      " 29650K .......... .......... .......... .......... .......... 64%  122M 1s\n",
      " 29700K .......... .......... .......... .......... .......... 64% 85,3M 1s\n",
      " 29750K .......... .......... .......... .......... .......... 64%  161M 1s\n",
      " 29800K .......... .......... .......... .......... .......... 64%  121M 1s\n",
      " 29850K .......... .......... .......... .......... .......... 64%  103M 1s\n",
      " 29900K .......... .......... .......... .......... .......... 64%  117M 1s\n",
      " 29950K .......... .......... .......... .......... .......... 64%  106M 1s\n",
      " 30000K .......... .......... .......... .......... .......... 64%  117M 1s\n",
      " 30050K .......... .......... .......... .......... .......... 65%  101M 1s\n",
      " 30100K .......... .......... .......... .......... .......... 65% 98,4M 1s\n",
      " 30150K .......... .......... .......... .......... .......... 65%  163M 1s\n",
      " 30200K .......... .......... .......... .......... .......... 65%  102M 1s\n",
      " 30250K .......... .......... .......... .......... .......... 65%  117M 1s\n",
      " 30300K .......... .......... .......... .......... .......... 65%  118M 1s\n",
      " 30350K .......... .......... .......... .......... .......... 65%  105M 1s\n",
      " 30400K .......... .......... .......... .......... .......... 65%  117M 1s\n",
      " 30450K .......... .......... .......... .......... .......... 65%  103M 1s\n",
      " 30500K .......... .......... .......... .......... .......... 65%  120M 0s\n",
      " 30550K .......... .......... .......... .......... .......... 66%  117M 0s\n",
      " 30600K .......... .......... .......... .......... .......... 66%  103M 0s\n",
      " 30650K .......... .......... .......... .......... .......... 66%  121M 0s\n",
      " 30700K .......... .......... .......... .......... .......... 66%  102M 0s\n",
      " 30750K .......... .......... .......... .......... .......... 66%  122M 0s\n",
      " 30800K .......... .......... .......... .......... .......... 66%  117M 0s\n",
      " 30850K .......... .......... .......... .......... .......... 66%  102M 0s\n",
      " 30900K .......... .......... .......... .......... .......... 66%  121M 0s\n",
      " 30950K .......... .......... .......... .......... .......... 66%  102M 0s\n",
      " 31000K .......... .......... .......... .......... .......... 67%  117M 0s\n",
      " 31050K .......... .......... .......... .......... .......... 67%  119M 0s\n",
      " 31100K .......... .......... .......... .......... .......... 67% 7,96M 0s\n",
      " 31150K .......... .......... .......... .......... .......... 67%  123M 0s\n",
      " 31200K .......... .......... .......... .......... .......... 67% 3,18M 0s\n",
      " 31250K .......... .......... .......... .......... .......... 67% 17,0M 0s\n",
      " 31300K .......... .......... .......... .......... .......... 67% 18,7M 0s\n",
      " 31350K .......... .......... .......... .......... .......... 67% 24,1M 0s\n",
      " 31400K .......... .......... .......... .......... .......... 67%  112M 0s\n",
      " 31450K .......... .......... .......... .......... .......... 68% 14,3M 0s\n",
      " 31500K .......... .......... .......... .......... .......... 68% 40,7M 0s\n",
      " 31550K .......... .......... .......... .......... .......... 68% 61,2M 0s\n",
      " 31600K .......... .......... .......... .......... .......... 68%  104M 0s\n",
      " 31650K .......... .......... .......... .......... .......... 68%  122M 0s\n",
      " 31700K .......... .......... .......... .......... .......... 68%  102M 0s\n",
      " 31750K .......... .......... .......... .......... .......... 68%  122M 0s\n",
      " 31800K .......... .......... .......... .......... .......... 68% 65,5M 0s\n",
      " 31850K .......... .......... .......... .......... .......... 68%  112M 0s\n",
      " 31900K .......... .......... .......... .......... .......... 68%  104M 0s\n",
      " 31950K .......... .......... .......... .......... .......... 69% 97,8M 0s\n",
      " 32000K .......... .......... .......... .......... .......... 69%  109M 0s\n",
      " 32050K .......... .......... .......... .......... .......... 69%  105M 0s\n",
      " 32100K .......... .......... .......... .......... .......... 69% 92,4M 0s\n",
      " 32150K .......... .......... .......... .......... .......... 69%  116M 0s\n",
      " 32200K .......... .......... .......... .......... .......... 69% 97,2M 0s\n",
      " 32250K .......... .......... .......... .......... .......... 69%  126M 0s\n",
      " 32300K .......... .......... .......... .......... .......... 69% 39,4M 0s\n",
      " 32350K .......... .......... .......... .......... .......... 69% 90,5M 0s\n",
      " 32400K .......... .......... .......... .......... .......... 70%  131M 0s\n",
      " 32450K .......... .......... .......... .......... .......... 70%  148M 0s\n",
      " 32500K .......... .......... .......... .......... .......... 70%  120M 0s\n",
      " 32550K .......... .......... .......... .......... .......... 70%  118M 0s\n",
      " 32600K .......... .......... .......... .......... .......... 70% 67,0M 0s\n",
      " 32650K .......... .......... .......... .......... .......... 70%  145M 0s\n",
      " 32700K .......... .......... .......... .......... .......... 70%  162M 0s\n",
      " 32750K .......... .......... .......... .......... .......... 70%  120M 0s\n",
      " 32800K .......... .......... .......... .......... .......... 70%  121M 0s\n",
      " 32850K .......... .......... .......... .......... .......... 71%  102M 0s\n",
      " 32900K .......... .......... .......... .......... .......... 71% 91,2M 0s\n",
      " 32950K .......... .......... .......... .......... .......... 71%  102M 0s\n",
      " 33000K .......... .......... .......... .......... .......... 71%  119M 0s\n",
      " 33050K .......... .......... .......... .......... .......... 71%  120M 0s\n",
      " 33100K .......... .......... .......... .......... .......... 71% 89,0M 0s\n",
      " 33150K .......... .......... .......... .......... .......... 71%  110M 0s\n",
      " 33200K .......... .......... .......... .......... .......... 71%  131M 0s\n",
      " 33250K .......... .......... .......... .......... .......... 71% 95,1M 0s\n",
      " 33300K .......... .......... .......... .......... .......... 72%  159M 0s\n",
      " 33350K .......... .......... .......... .......... .......... 72% 98,8M 0s\n",
      " 33400K .......... .......... .......... .......... .......... 72%  112M 0s\n",
      " 33450K .......... .......... .......... .......... .......... 72%  101M 0s\n",
      " 33500K .......... .......... .......... .......... .......... 72%  115M 0s\n",
      " 33550K .......... .......... .......... .......... .......... 72%  138M 0s\n",
      " 33600K .......... .......... .......... .......... .......... 72%  105M 0s\n",
      " 33650K .......... .......... .......... .......... .......... 72%  121M 0s\n",
      " 33700K .......... .......... .......... .......... .......... 72%  102M 0s\n",
      " 33750K .......... .......... .......... .......... .......... 72%  117M 0s\n",
      " 33800K .......... .......... .......... .......... .......... 73%  121M 0s\n",
      " 33850K .......... .......... .......... .......... .......... 73%  104M 0s\n",
      " 33900K .......... .......... .......... .......... .......... 73%  121M 0s\n",
      " 33950K .......... .......... .......... .......... .......... 73%  102M 0s\n",
      " 34000K .......... .......... .......... .......... .......... 73%  117M 0s\n",
      " 34050K .......... .......... .......... .......... .......... 73%  121M 0s\n",
      " 34100K .......... .......... .......... .......... .......... 73%  102M 0s\n",
      " 34150K .......... .......... .......... .......... .......... 73% 8,65M 0s\n",
      " 34200K .......... .......... .......... .......... .......... 73% 12,8M 0s\n",
      " 34250K .......... .......... .......... .......... .......... 74% 3,75M 0s\n",
      " 34300K .......... .......... .......... .......... .......... 74% 19,2M 0s\n",
      " 34350K .......... .......... .......... .......... .......... 74% 26,3M 0s\n",
      " 34400K .......... .......... .......... .......... .......... 74% 16,2M 0s\n",
      " 34450K .......... .......... .......... .......... .......... 74% 49,8M 0s\n",
      " 34500K .......... .......... .......... .......... .......... 74%  105M 0s\n",
      " 34550K .......... .......... .......... .......... .......... 74% 13,3M 0s\n",
      " 34600K .......... .......... .......... .......... .......... 74% 43,1M 0s\n",
      " 34650K .......... .......... .......... .......... .......... 74% 74,0M 0s\n",
      " 34700K .......... .......... .......... .......... .......... 75% 63,4M 0s\n",
      " 34750K .......... .......... .......... .......... .......... 75%  144M 0s\n",
      " 34800K .......... .......... .......... .......... .......... 75%  165M 0s\n",
      " 34850K .......... .......... .......... .......... .......... 75%  152M 0s\n",
      " 34900K .......... .......... .......... .......... .......... 75%  169M 0s\n",
      " 34950K .......... .......... .......... .......... .......... 75%  176M 0s\n",
      " 35000K .......... .......... .......... .......... .......... 75%  121M 0s\n",
      " 35050K .......... .......... .......... .......... .......... 75% 95,7M 0s\n",
      " 35100K .......... .......... .......... .......... .......... 75%  103M 0s\n",
      " 35150K .......... .......... .......... .......... .......... 76%  115M 0s\n",
      " 35200K .......... .......... .......... .......... .......... 76%  104M 0s\n",
      " 35250K .......... .......... .......... .......... .......... 76%  117M 0s\n",
      " 35300K .......... .......... .......... .......... .......... 76%  120M 0s\n",
      " 35350K .......... .......... .......... .......... .......... 76%  101M 0s\n",
      " 35400K .......... .......... .......... .......... .......... 76%  121M 0s\n",
      " 35450K .......... .......... .......... .......... .......... 76% 63,9M 0s\n",
      " 35500K .......... .......... .......... .......... .......... 76%  125M 0s\n",
      " 35550K .......... .......... .......... .......... .......... 76% 55,7M 0s\n",
      " 35600K .......... .......... .......... .......... .......... 76% 59,3M 0s\n",
      " 35650K .......... .......... .......... .......... .......... 77%  121M 0s\n",
      " 35700K .......... .......... .......... .......... .......... 77%  102M 0s\n",
      " 35750K .......... .......... .......... .......... .......... 77%  118M 0s\n",
      " 35800K .......... .......... .......... .......... .......... 77%  120M 0s\n",
      " 35850K .......... .......... .......... .......... .......... 77%  102M 0s\n",
      " 35900K .......... .......... .......... .......... .......... 77%  117M 0s\n",
      " 35950K .......... .......... .......... .......... .......... 77%  102M 0s\n",
      " 36000K .......... .......... .......... .......... .......... 77%  121M 0s\n",
      " 36050K .......... .......... .......... .......... .......... 77%  121M 0s\n",
      " 36100K .......... .......... .......... .......... .......... 78%  102M 0s\n",
      " 36150K .......... .......... .......... .......... .......... 78%  118M 0s\n",
      " 36200K .......... .......... .......... .......... .......... 78%  102M 0s\n",
      " 36250K .......... .......... .......... .......... .......... 78%  121M 0s\n",
      " 36300K .......... .......... .......... .......... .......... 78%  118M 0s\n",
      " 36350K .......... .......... .......... .......... .......... 78%  102M 0s\n",
      " 36400K .......... .......... .......... .......... .......... 78%  121M 0s\n",
      " 36450K .......... .......... .......... .......... .......... 78%  102M 0s\n",
      " 36500K .......... .......... .......... .......... .......... 78%  117M 0s\n",
      " 36550K .......... .......... .......... .......... .......... 79%  121M 0s\n",
      " 36600K .......... .......... .......... .......... .......... 79%  102M 0s\n",
      " 36650K .......... .......... .......... .......... .......... 79%  121M 0s\n",
      " 36700K .......... .......... .......... .......... .......... 79% 97,2M 0s\n",
      " 36750K .......... .......... .......... .......... .......... 79%  125M 0s\n",
      " 36800K .......... .......... .......... .......... .......... 79%  121M 0s\n",
      " 36850K .......... .......... .......... .......... .......... 79%  102M 0s\n",
      " 36900K .......... .......... .......... .......... .......... 79%  117M 0s\n",
      " 36950K .......... .......... .......... .......... .......... 79%  102M 0s\n",
      " 37000K .......... .......... .......... .......... .......... 80%  120M 0s\n",
      " 37050K .......... .......... .......... .......... .......... 80%  118M 0s\n",
      " 37100K .......... .......... .......... .......... .......... 80%  102M 0s\n",
      " 37150K .......... .......... .......... .......... .......... 80%  121M 0s\n",
      " 37200K .......... .......... .......... .......... .......... 80% 8,02M 0s\n",
      " 37250K .......... .......... .......... .......... .......... 80% 38,4M 0s\n",
      " 37300K .......... .......... .......... .......... .......... 80% 4,31M 0s\n",
      " 37350K .......... .......... .......... .......... .......... 80% 8,01M 0s\n",
      " 37400K .......... .......... .......... .......... .......... 80%  121M 0s\n",
      " 37450K .......... .......... .......... .......... .......... 80% 13,7M 0s\n",
      " 37500K .......... .......... .......... .......... .......... 81% 20,8M 0s\n",
      " 37550K .......... .......... .......... .......... .......... 81% 28,3M 0s\n",
      " 37600K .......... .......... .......... .......... .......... 81% 34,1M 0s\n",
      " 37650K .......... .......... .......... .......... .......... 81% 27,1M 0s\n",
      " 37700K .......... .......... .......... .......... .......... 81% 47,3M 0s\n",
      " 37750K .......... .......... .......... .......... .......... 81%  117M 0s\n",
      " 37800K .......... .......... .......... .......... .......... 81%  119M 0s\n",
      " 37850K .......... .......... .......... .......... .......... 81%  102M 0s\n",
      " 37900K .......... .......... .......... .......... .......... 81%  115M 0s\n",
      " 37950K .......... .......... .......... .......... .......... 82%  105M 0s\n",
      " 38000K .......... .......... .......... .......... .......... 82%  103M 0s\n",
      " 38050K .......... .......... .......... .......... .......... 82%  141M 0s\n",
      " 38100K .......... .......... .......... .......... .......... 82% 85,5M 0s\n",
      " 38150K .......... .......... .......... .......... .......... 82%  114M 0s\n",
      " 38200K .......... .......... .......... .......... .......... 82%  137M 0s\n",
      " 38250K .......... .......... .......... .......... .......... 82%  121M 0s\n",
      " 38300K .......... .......... .......... .......... .......... 82%  117M 0s\n",
      " 38350K .......... .......... .......... .......... .......... 82%  102M 0s\n",
      " 38400K .......... .......... .......... .......... .......... 83%  121M 0s\n",
      " 38450K .......... .......... .......... .......... .......... 83%  102M 0s\n",
      " 38500K .......... .......... .......... .......... .......... 83%  118M 0s\n",
      " 38550K .......... .......... .......... .......... .......... 83%  119M 0s\n",
      " 38600K .......... .......... .......... .......... .......... 83%  102M 0s\n",
      " 38650K .......... .......... .......... .......... .......... 83%  118M 0s\n",
      " 38700K .......... .......... .......... .......... .......... 83%  102M 0s\n",
      " 38750K .......... .......... .......... .......... .......... 83%  111M 0s\n",
      " 38800K .......... .......... .......... .......... .......... 83%  133M 0s\n",
      " 38850K .......... .......... .......... .......... .......... 84%  102M 0s\n",
      " 38900K .......... .......... .......... .......... .......... 84%  117M 0s\n",
      " 38950K .......... .......... .......... .......... .......... 84%  102M 0s\n",
      " 39000K .......... .......... .......... .......... .......... 84%  121M 0s\n",
      " 39050K .......... .......... .......... .......... .......... 84%  117M 0s\n",
      " 39100K .......... .......... .......... .......... .......... 84%  102M 0s\n",
      " 39150K .......... .......... .......... .......... .......... 84%  120M 0s\n",
      " 39200K .......... .......... .......... .......... .......... 84%  102M 0s\n",
      " 39250K .......... .......... .......... .......... .......... 84%  121M 0s\n",
      " 39300K .......... .......... .......... .......... .......... 84%  117M 0s\n",
      " 39350K .......... .......... .......... .......... .......... 85%  102M 0s\n",
      " 39400K .......... .......... .......... .......... .......... 85%  121M 0s\n",
      " 39450K .......... .......... .......... .......... .......... 85%  102M 0s\n",
      " 39500K .......... .......... .......... .......... .......... 85%  117M 0s\n",
      " 39550K .......... .......... .......... .......... .......... 85%  122M 0s\n",
      " 39600K .......... .......... .......... .......... .......... 85%  102M 0s\n",
      " 39650K .......... .......... .......... .......... .......... 85%  118M 0s\n",
      " 39700K .......... .......... .......... .......... .......... 85%  103M 0s\n",
      " 39750K .......... .......... .......... .......... .......... 85%  120M 0s\n",
      " 39800K .......... .......... .......... .......... .......... 86%  121M 0s\n",
      " 39850K .......... .......... .......... .......... .......... 86%  102M 0s\n",
      " 39900K .......... .......... .......... .......... .......... 86%  117M 0s\n",
      " 39950K .......... .......... .......... .......... .......... 86%  103M 0s\n",
      " 40000K .......... .......... .......... .......... .......... 86%  120M 0s\n",
      " 40050K .......... .......... .......... .......... .......... 86%  117M 0s\n",
      " 40100K .......... .......... .......... .......... .......... 86%  103M 0s\n",
      " 40150K .......... .......... .......... .......... .......... 86%  120M 0s\n",
      " 40200K .......... .......... .......... .......... .......... 86%  102M 0s\n",
      " 40250K .......... .......... .......... .......... .......... 87% 9,43M 0s\n",
      " 40300K .......... .......... .......... .......... .......... 87% 17,8M 0s\n",
      " 40350K .......... .......... .......... .......... .......... 87% 4,64M 0s\n",
      " 40400K .......... .......... .......... .......... .......... 87% 9,77M 0s\n",
      " 40450K .......... .......... .......... .......... .......... 87% 23,9M 0s\n",
      " 40500K .......... .......... .......... .......... .......... 87% 12,4M 0s\n",
      " 40550K .......... .......... .......... .......... .......... 87% 39,6M 0s\n",
      " 40600K .......... .......... .......... .......... .......... 87% 26,2M 0s\n",
      " 40650K .......... .......... .......... .......... .......... 87% 24,0M 0s\n",
      " 40700K .......... .......... .......... .......... .......... 88% 18,6M 0s\n",
      " 40750K .......... .......... .......... .......... .......... 88% 56,1M 0s\n",
      " 40800K .......... .......... .......... .......... .......... 88% 67,7M 0s\n",
      " 40850K .......... .......... .......... .......... .......... 88%  166M 0s\n",
      " 40900K .......... .......... .......... .......... .......... 88%  209M 0s\n",
      " 40950K .......... .......... .......... .......... .......... 88%  102M 0s\n",
      " 41000K .......... .......... .......... .......... .......... 88%  120M 0s\n",
      " 41050K .......... .......... .......... .......... .......... 88%  118M 0s\n",
      " 41100K .......... .......... .......... .......... .......... 88%  102M 0s\n",
      " 41150K .......... .......... .......... .......... .......... 88%  121M 0s\n",
      " 41200K .......... .......... .......... .......... .......... 89%  103M 0s\n",
      " 41250K .......... .......... .......... .......... .......... 89%  117M 0s\n",
      " 41300K .......... .......... .......... .......... .......... 89%  121M 0s\n",
      " 41350K .......... .......... .......... .......... .......... 89%  102M 0s\n",
      " 41400K .......... .......... .......... .......... .......... 89%  122M 0s\n",
      " 41450K .......... .......... .......... .......... .......... 89%  102M 0s\n",
      " 41500K .......... .......... .......... .......... .......... 89%  118M 0s\n",
      " 41550K .......... .......... .......... .......... .......... 89%  121M 0s\n",
      " 41600K .......... .......... .......... .......... .......... 89%  103M 0s\n",
      " 41650K .......... .......... .......... .......... .......... 90%  117M 0s\n",
      " 41700K .......... .......... .......... .......... .......... 90% 93,3M 0s\n",
      " 41750K .......... .......... .......... .......... .......... 90%  123M 0s\n",
      " 41800K .......... .......... .......... .......... .......... 90%  128M 0s\n",
      " 41850K .......... .......... .......... .......... .......... 90%  102M 0s\n",
      " 41900K .......... .......... .......... .......... .......... 90%  121M 0s\n",
      " 41950K .......... .......... .......... .......... .......... 90%  102M 0s\n",
      " 42000K .......... .......... .......... .......... .......... 90%  121M 0s\n",
      " 42050K .......... .......... .......... .......... .......... 90%  117M 0s\n",
      " 42100K .......... .......... .......... .......... .......... 91%  102M 0s\n",
      " 42150K .......... .......... .......... .......... .......... 91%  121M 0s\n",
      " 42200K .......... .......... .......... .......... .......... 91% 99,9M 0s\n",
      " 42250K .......... .......... .......... .......... .......... 91%  122M 0s\n",
      " 42300K .......... .......... .......... .......... .......... 91%  121M 0s\n",
      " 42350K .......... .......... .......... .......... .......... 91%  102M 0s\n",
      " 42400K .......... .......... .......... .......... .......... 91%  117M 0s\n",
      " 42450K .......... .......... .......... .......... .......... 91%  102M 0s\n",
      " 42500K .......... .......... .......... .......... .......... 91%  121M 0s\n",
      " 42550K .......... .......... .......... .......... .......... 91%  120M 0s\n",
      " 42600K .......... .......... .......... .......... .......... 92%  102M 0s\n",
      " 42650K .......... .......... .......... .......... .......... 92%  117M 0s\n",
      " 42700K .......... .......... .......... .......... .......... 92%  102M 0s\n",
      " 42750K .......... .......... .......... .......... .......... 92%  121M 0s\n",
      " 42800K .......... .......... .......... .......... .......... 92%  117M 0s\n",
      " 42850K .......... .......... .......... .......... .......... 92%  101M 0s\n",
      " 42900K .......... .......... .......... .......... .......... 92%  122M 0s\n",
      " 42950K .......... .......... .......... .......... .......... 92%  102M 0s\n",
      " 43000K .......... .......... .......... .......... .......... 92%  121M 0s\n",
      " 43050K .......... .......... .......... .......... .......... 93%  117M 0s\n",
      " 43100K .......... .......... .......... .......... .......... 93%  102M 0s\n",
      " 43150K .......... .......... .......... .......... .......... 93%  120M 0s\n",
      " 43200K .......... .......... .......... .......... .......... 93%  101M 0s\n",
      " 43250K .......... .......... .......... .......... .......... 93%  104M 0s\n",
      " 43300K .......... .......... .......... .......... .......... 93% 24,0M 0s\n",
      " 43350K .......... .......... .......... .......... .......... 93% 8,52M 0s\n",
      " 43400K .......... .......... .......... .......... .......... 93% 5,34M 0s\n",
      " 43450K .......... .......... .......... .......... .......... 93% 11,9M 0s\n",
      " 43500K .......... .......... .......... .......... .......... 94% 17,1M 0s\n",
      " 43550K .......... .......... .......... .......... .......... 94% 10,1M 0s\n",
      " 43600K .......... .......... .......... .......... .......... 94% 48,5M 0s\n",
      " 43650K .......... .......... .......... .......... .......... 94% 33,4M 0s\n",
      " 43700K .......... .......... .......... .......... .......... 94% 32,0M 0s\n",
      " 43750K .......... .......... .......... .......... .......... 94% 18,8M 0s\n",
      " 43800K .......... .......... .......... .......... .......... 94% 19,7M 0s\n",
      " 43850K .......... .......... .......... .......... .......... 94%  102M 0s\n",
      " 43900K .......... .......... .......... .......... .......... 94%  121M 0s\n",
      " 43950K .......... .......... .......... .......... .......... 95%  103M 0s\n",
      " 44000K .......... .......... .......... .......... .......... 95%  109M 0s\n",
      " 44050K .......... .......... .......... .......... .......... 95%  120M 0s\n",
      " 44100K .......... .......... .......... .......... .......... 95%  105M 0s\n",
      " 44150K .......... .......... .......... .......... .......... 95% 45,4M 0s\n",
      " 44200K .......... .......... .......... .......... .......... 95%  118M 0s\n",
      " 44250K .......... .......... .......... .......... .......... 95%  187M 0s\n",
      " 44300K .......... .......... .......... .......... .......... 95%  192M 0s\n",
      " 44350K .......... .......... .......... .......... .......... 95%  145M 0s\n",
      " 44400K .......... .......... .......... .......... .......... 95%  209M 0s\n",
      " 44450K .......... .......... .......... .......... .......... 96%  103M 0s\n",
      " 44500K .......... .......... .......... .......... .......... 96%  122M 0s\n",
      " 44550K .......... .......... .......... .......... .......... 96%  117M 0s\n",
      " 44600K .......... .......... .......... .......... .......... 96%  102M 0s\n",
      " 44650K .......... .......... .......... .......... .......... 96%  122M 0s\n",
      " 44700K .......... .......... .......... .......... .......... 96%  102M 0s\n",
      " 44750K .......... .......... .......... .......... .......... 96%  121M 0s\n",
      " 44800K .......... .......... .......... .......... .......... 96%  118M 0s\n",
      " 44850K .......... .......... .......... .......... .......... 96%  102M 0s\n",
      " 44900K .......... .......... .......... .......... .......... 97%  121M 0s\n",
      " 44950K .......... .......... .......... .......... .......... 97%  102M 0s\n",
      " 45000K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 45050K .......... .......... .......... .......... .......... 97%  120M 0s\n",
      " 45100K .......... .......... .......... .......... .......... 97%  103M 0s\n",
      " 45150K .......... .......... .......... .......... .......... 97%  121M 0s\n",
      " 45200K .......... .......... .......... .......... .......... 97%  103M 0s\n",
      " 45250K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 45300K .......... .......... .......... .......... .......... 97%  119M 0s\n",
      " 45350K .......... .......... .......... .......... .......... 98%  104M 0s\n",
      " 45400K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 45450K .......... .......... .......... .......... .......... 98%  103M 0s\n",
      " 45500K .......... .......... .......... .......... .......... 98%  121M 0s\n",
      " 45550K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 45600K .......... .......... .......... .......... .......... 98%  102M 0s\n",
      " 45650K .......... .......... .......... .......... .......... 98%  121M 0s\n",
      " 45700K .......... .......... .......... .......... .......... 98%  102M 0s\n",
      " 45750K .......... .......... .......... .......... .......... 98%  121M 0s\n",
      " 45800K .......... .......... .......... .......... .......... 99%  118M 0s\n",
      " 45850K .......... .......... .......... .......... .......... 99%  102M 0s\n",
      " 45900K .......... .......... .......... .......... .......... 99%  121M 0s\n",
      " 45950K .......... .......... .......... .......... .......... 99%  102M 0s\n",
      " 46000K .......... .......... .......... .......... .......... 99%  117M 0s\n",
      " 46050K .......... .......... .......... .......... .......... 99%  121M 0s\n",
      " 46100K .......... .......... .......... .......... .......... 99%  102M 0s\n",
      " 46150K .......... .......... .......... .......... .......... 99%  117M 0s\n",
      " 46200K .......... .......... .......... .......... .......... 99%  100M 0s\n",
      " 46250K .......... .......... .......... .......... .......... 99%  124M 0s\n",
      " 46300K .....                                                 100%  473M=1,3s\n",
      "\n",
      "2023-08-17 12:55:50 (35,7 MB/s) - ‘data/reference/genome.fa.gz’ guardado [47416846/47416846]\n",
      "\n",
      "[Thu Aug 17 12:55:51 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T125548.494598.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:55:52 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR696683_chr7_1.fastq.gz, data/raw/ERR696683_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR696683_chr7_1_fastqc.html, results/fastqc_result/ERR696683_chr7_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR696683_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 12:56:10 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:56:10 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR696683_chr7_1.fastq.gz, data/raw/ERR696683_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR696683_chr7_1_fastp.fastq.gz, data/processed/ERR696683_chr7_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR696683_chr7_1_fastp.fastq.gz, data/processed/ERR696683_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1935275\n",
      "total bases: 146263550\n",
      "Q20 bases: 145211647(99.2808%)\n",
      "Q30 bases: 141120433(96.4837%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1935275\n",
      "total bases: 145878500\n",
      "Q20 bases: 144064467(98.7565%)\n",
      "Q30 bases: 138098537(94.6668%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1177764\n",
      "total bases: 89449589\n",
      "Q20 bases: 89095661(99.6043%)\n",
      "Q30 bases: 87513697(97.8358%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1177764\n",
      "total bases: 89414052\n",
      "Q20 bases: 88793329(99.3058%)\n",
      "Q30 bases: 86459019(96.6951%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2355528\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1515022\n",
      "reads with adapter trimmed: 8864\n",
      "bases trimmed due to adapters: 230085\n",
      "\n",
      "Duplication rate: 5.65604%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR696683_chr7_1.fastq.gz -I data/raw/ERR696683_chr7_2.fastq.gz -o data/processed/ERR696683_chr7_1_fastp.fastq.gz -O data/processed/ERR696683_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Thu Aug 17 12:56:16 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:56:16 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR696683_chr7_1_fastp.fastq.gz, data/processed/ERR696683_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR696683_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR696683_chr7_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR696683_chr7_1_fastp.fastq.gz, data/processed/ERR696683_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 12:56:27 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T125551.481299.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:56:29 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753368_chr7_1.fastq.gz, data/raw/ERR753368_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753368_chr7_1_fastqc.html, results/fastqc_result/ERR753368_chr7_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753368_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753368_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 12:56:45 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:56:45 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753368_chr7_1.fastq.gz, data/raw/ERR753368_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753368_chr7_1_fastp.fastq.gz, data/processed/ERR753368_chr7_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753368_chr7_2_fastp.fastq.gz, data/processed/ERR753368_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1806088\n",
      "total bases: 137130775\n",
      "Q20 bases: 136588468(99.6045%)\n",
      "Q30 bases: 132513779(96.6331%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1806088\n",
      "total bases: 136987438\n",
      "Q20 bases: 135772877(99.1134%)\n",
      "Q30 bases: 129457863(94.5035%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1548330\n",
      "total bases: 117666684\n",
      "Q20 bases: 117461408(99.8255%)\n",
      "Q30 bases: 115455876(98.1211%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1548330\n",
      "total bases: 117656644\n",
      "Q20 bases: 117075719(99.5063%)\n",
      "Q30 bases: 113529570(96.4923%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3096660\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 515516\n",
      "reads with adapter trimmed: 5202\n",
      "bases trimmed due to adapters: 140710\n",
      "\n",
      "Duplication rate: 2.03607%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753368_chr7_1.fastq.gz -I data/raw/ERR753368_chr7_2.fastq.gz -o data/processed/ERR753368_chr7_1_fastp.fastq.gz -O data/processed/ERR753368_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Thu Aug 17 12:56:51 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:56:51 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753368_chr7_1_fastp.fastq.gz, data/processed/ERR753368_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753368_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753368_chr7_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753368_chr7_2_fastp.fastq.gz, data/processed/ERR753368_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 12:57:04 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T125627.786372.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:57:05 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753369_chr7_1.fastq.gz, data/raw/ERR753369_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753369_chr7_1_fastqc.html, results/fastqc_result/ERR753369_chr7_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753369_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753369_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 12:57:23 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:57:23 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753369_chr7_1.fastq.gz, data/raw/ERR753369_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753369_chr7_1_fastp.fastq.gz, data/processed/ERR753369_chr7_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753369_chr7_1_fastp.fastq.gz, data/processed/ERR753369_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1929269\n",
      "total bases: 146477595\n",
      "Q20 bases: 145872914(99.5872%)\n",
      "Q30 bases: 141418223(96.546%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1929269\n",
      "total bases: 146329545\n",
      "Q20 bases: 145027500(99.1102%)\n",
      "Q30 bases: 138248767(94.4777%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1651945\n",
      "total bases: 125540731\n",
      "Q20 bases: 125307776(99.8144%)\n",
      "Q30 bases: 123093775(98.0509%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1651945\n",
      "total bases: 125530164\n",
      "Q20 bases: 124909386(99.5055%)\n",
      "Q30 bases: 121110875(96.4795%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3303890\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 554648\n",
      "reads with adapter trimmed: 5782\n",
      "bases trimmed due to adapters: 156225\n",
      "\n",
      "Duplication rate: 2.35813%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753369_chr7_1.fastq.gz -I data/raw/ERR753369_chr7_2.fastq.gz -o data/processed/ERR753369_chr7_1_fastp.fastq.gz -O data/processed/ERR753369_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Thu Aug 17 12:57:30 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:57:30 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753369_chr7_1_fastp.fastq.gz, data/processed/ERR753369_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753369_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753369_chr7_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753369_chr7_1_fastp.fastq.gz, data/processed/ERR753369_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 12:57:43 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T125704.630827.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:57:45 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753370_chr7_1.fastq.gz, data/raw/ERR753370_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753370_chr7_1_fastqc.html, results/fastqc_result/ERR753370_chr7_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753370_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753370_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 12:58:12 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:58:12 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753370_chr7_1.fastq.gz, data/raw/ERR753370_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753370_chr7_1_fastp.fastq.gz, data/processed/ERR753370_chr7_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753370_chr7_2_fastp.fastq.gz, data/processed/ERR753370_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2673460\n",
      "total bases: 269090503\n",
      "Q20 bases: 267232327(99.3095%)\n",
      "Q30 bases: 259317581(96.3682%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2673460\n",
      "total bases: 269019918\n",
      "Q20 bases: 266435611(99.0394%)\n",
      "Q30 bases: 256135213(95.2105%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 2669965\n",
      "total bases: 267638922\n",
      "Q20 bases: 265893698(99.3479%)\n",
      "Q30 bases: 258369403(96.5366%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 2669965\n",
      "total bases: 266755891\n",
      "Q20 bases: 264414765(99.1224%)\n",
      "Q30 bases: 254782415(95.5114%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 5339930\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 6990\n",
      "reads with adapter trimmed: 22426\n",
      "bases trimmed due to adapters: 449745\n",
      "\n",
      "Duplication rate: 4.34628%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 159\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753370_chr7_1.fastq.gz -I data/raw/ERR753370_chr7_2.fastq.gz -o data/processed/ERR753370_chr7_1_fastp.fastq.gz -O data/processed/ERR753370_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 12 seconds\n",
      "[Thu Aug 17 12:58:24 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:58:24 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753370_chr7_1_fastp.fastq.gz, data/processed/ERR753370_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753370_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753370_chr7_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753370_chr7_2_fastp.fastq.gz, data/processed/ERR753370_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 12:58:48 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T125743.909092.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:58:50 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753371_chr7_1.fastq.gz, data/raw/ERR753371_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753371_chr7_1_fastqc.html, results/fastqc_result/ERR753371_chr7_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753371_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753371_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 12:59:05 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:59:05 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753371_chr7_1.fastq.gz, data/raw/ERR753371_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753371_chr7_1_fastp.fastq.gz, data/processed/ERR753371_chr7_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753371_chr7_2_fastp.fastq.gz, data/processed/ERR753371_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1777787\n",
      "total bases: 133923240\n",
      "Q20 bases: 132711766(99.0954%)\n",
      "Q30 bases: 127818746(95.4418%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1777787\n",
      "total bases: 133584939\n",
      "Q20 bases: 131975601(98.7953%)\n",
      "Q30 bases: 127258356(95.264%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 901910\n",
      "total bases: 68460468\n",
      "Q20 bases: 68125599(99.5109%)\n",
      "Q30 bases: 66410192(97.0052%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 901910\n",
      "total bases: 68422604\n",
      "Q20 bases: 67941529(99.2969%)\n",
      "Q30 bases: 66426314(97.0824%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1803820\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1751754\n",
      "reads with adapter trimmed: 7008\n",
      "bases trimmed due to adapters: 174651\n",
      "\n",
      "Duplication rate: 21.6777%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 118\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753371_chr7_1.fastq.gz -I data/raw/ERR753371_chr7_2.fastq.gz -o data/processed/ERR753371_chr7_1_fastp.fastq.gz -O data/processed/ERR753371_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Thu Aug 17 12:59:10 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:59:10 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753371_chr7_1_fastp.fastq.gz, data/processed/ERR753371_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753371_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753371_chr7_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753371_chr7_2_fastp.fastq.gz, data/processed/ERR753371_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 12:59:19 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T125848.774103.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:59:21 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753372_chr7_1.fastq.gz, data/raw/ERR753372_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753372_chr7_1_fastqc.html, results/fastqc_result/ERR753372_chr7_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753372_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753372_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 12:59:36 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:59:36 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753372_chr7_1.fastq.gz, data/raw/ERR753372_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753372_chr7_1_fastp.fastq.gz, data/processed/ERR753372_chr7_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753372_chr7_2_fastp.fastq.gz, data/processed/ERR753372_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1928165\n",
      "total bases: 146395509\n",
      "Q20 bases: 145810094(99.6001%)\n",
      "Q30 bases: 141369327(96.5667%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1928165\n",
      "total bases: 146239318\n",
      "Q20 bases: 144940401(99.1118%)\n",
      "Q30 bases: 138222059(94.5177%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1646996\n",
      "total bases: 125164638\n",
      "Q20 bases: 124943449(99.8233%)\n",
      "Q30 bases: 122763086(98.0813%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1646996\n",
      "total bases: 125153924\n",
      "Q20 bases: 124539813(99.5093%)\n",
      "Q30 bases: 120808917(96.5283%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3293992\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 562338\n",
      "reads with adapter trimmed: 6016\n",
      "bases trimmed due to adapters: 162856\n",
      "\n",
      "Duplication rate: 1.97989%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753372_chr7_1.fastq.gz -I data/raw/ERR753372_chr7_2.fastq.gz -o data/processed/ERR753372_chr7_1_fastp.fastq.gz -O data/processed/ERR753372_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Thu Aug 17 12:59:43 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:59:43 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753372_chr7_1_fastp.fastq.gz, data/processed/ERR753372_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753372_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753372_chr7_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753372_chr7_2_fastp.fastq.gz, data/processed/ERR753372_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 12:59:56 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T125919.760899.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 12:59:57 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753373_chr7_1.fastq.gz, data/raw/ERR753373_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753373_chr7_1_fastqc.html, results/fastqc_result/ERR753373_chr7_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753373_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753373_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 13:00:13 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:00:13 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753373_chr7_1.fastq.gz, data/raw/ERR753373_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753373_chr7_1_fastp.fastq.gz, data/processed/ERR753373_chr7_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753373_chr7_2_fastp.fastq.gz, data/processed/ERR753373_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1590509\n",
      "total bases: 152276057\n",
      "Q20 bases: 151053516(99.1972%)\n",
      "Q30 bases: 140667684(92.3768%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1590509\n",
      "total bases: 149042940\n",
      "Q20 bases: 147417333(98.9093%)\n",
      "Q30 bases: 137755782(92.4269%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1550999\n",
      "total bases: 146136096\n",
      "Q20 bases: 145283119(99.4163%)\n",
      "Q30 bases: 136885543(93.6699%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1550999\n",
      "total bases: 141207201\n",
      "Q20 bases: 140529494(99.5201%)\n",
      "Q30 bases: 133173977(94.311%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3101998\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 79020\n",
      "reads with adapter trimmed: 4140\n",
      "bases trimmed due to adapters: 154596\n",
      "\n",
      "Duplication rate: 1.84762%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 153\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753373_chr7_1.fastq.gz -I data/raw/ERR753373_chr7_2.fastq.gz -o data/processed/ERR753373_chr7_1_fastp.fastq.gz -O data/processed/ERR753373_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Thu Aug 17 13:00:20 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:00:20 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753373_chr7_1_fastp.fastq.gz, data/processed/ERR753373_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753373_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753373_chr7_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753373_chr7_2_fastp.fastq.gz, data/processed/ERR753373_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 13:00:35 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T125956.349681.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:00:37 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753374_chr7_1.fastq.gz, data/raw/ERR753374_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753374_chr7_1_fastqc.html, results/fastqc_result/ERR753374_chr7_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753374_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753374_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 13:00:49 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:00:49 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753374_chr7_1.fastq.gz, data/raw/ERR753374_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753374_chr7_1_fastp.fastq.gz, data/processed/ERR753374_chr7_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753374_chr7_1_fastp.fastq.gz, data/processed/ERR753374_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 937433\n",
      "total bases: 89668520\n",
      "Q20 bases: 88820397(99.0542%)\n",
      "Q30 bases: 81109687(90.455%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 937433\n",
      "total bases: 87729358\n",
      "Q20 bases: 86808200(98.95%)\n",
      "Q30 bases: 79810509(90.9735%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 907041\n",
      "total bases: 84770678\n",
      "Q20 bases: 84167468(99.2884%)\n",
      "Q30 bases: 78047285(92.0687%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 907041\n",
      "total bases: 82557850\n",
      "Q20 bases: 82082554(99.4243%)\n",
      "Q30 bases: 76724913(92.9347%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1814082\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 60784\n",
      "reads with adapter trimmed: 2678\n",
      "bases trimmed due to adapters: 102178\n",
      "\n",
      "Duplication rate: 5.64229%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 153\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753374_chr7_1.fastq.gz -I data/raw/ERR753374_chr7_2.fastq.gz -o data/processed/ERR753374_chr7_1_fastp.fastq.gz -O data/processed/ERR753374_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Thu Aug 17 13:00:54 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:00:54 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753374_chr7_1_fastp.fastq.gz, data/processed/ERR753374_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753374_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753374_chr7_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753374_chr7_1_fastp.fastq.gz, data/processed/ERR753374_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 13:01:03 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T130035.947073.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:01:04 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753375_chr7_1.fastq.gz, data/raw/ERR753375_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753375_chr7_1_fastqc.html, results/fastqc_result/ERR753375_chr7_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753375_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753375_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 13:01:15 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:01:15 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753375_chr7_1.fastq.gz, data/raw/ERR753375_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753375_chr7_1_fastp.fastq.gz, data/processed/ERR753375_chr7_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753375_chr7_2_fastp.fastq.gz, data/processed/ERR753375_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1203954\n",
      "total bases: 90919106\n",
      "Q20 bases: 90345397(99.369%)\n",
      "Q30 bases: 88249315(97.0636%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1203954\n",
      "total bases: 90896831\n",
      "Q20 bases: 89692945(98.6755%)\n",
      "Q30 bases: 87415245(96.1697%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 768315\n",
      "total bases: 58337245\n",
      "Q20 bases: 58144802(99.6701%)\n",
      "Q30 bases: 57253055(98.1415%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 768315\n",
      "total bases: 58346791\n",
      "Q20 bases: 57776292(99.0222%)\n",
      "Q30 bases: 56839488(97.4166%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1536630\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 871278\n",
      "reads with adapter trimmed: 4636\n",
      "bases trimmed due to adapters: 122070\n",
      "\n",
      "Duplication rate: 1.68137%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 119\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753375_chr7_1.fastq.gz -I data/raw/ERR753375_chr7_2.fastq.gz -o data/processed/ERR753375_chr7_1_fastp.fastq.gz -O data/processed/ERR753375_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 3 seconds\n",
      "[Thu Aug 17 13:01:19 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:01:19 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753375_chr7_1_fastp.fastq.gz, data/processed/ERR753375_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753375_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753375_chr7_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753375_chr7_2_fastp.fastq.gz, data/processed/ERR753375_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 13:01:28 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T130103.295013.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:01:30 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753376_chr7_1.fastq.gz, data/raw/ERR753376_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753376_chr7_1_fastqc.html, results/fastqc_result/ERR753376_chr7_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753376_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753376_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 13:01:43 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:01:43 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753376_chr7_1.fastq.gz, data/raw/ERR753376_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753376_chr7_1_fastp.fastq.gz, data/processed/ERR753376_chr7_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753376_chr7_2_fastp.fastq.gz, data/processed/ERR753376_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1556102\n",
      "total bases: 117603181\n",
      "Q20 bases: 116778781(99.299%)\n",
      "Q30 bases: 113491260(96.5036%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1556102\n",
      "total bases: 117282101\n",
      "Q20 bases: 115811340(98.746%)\n",
      "Q30 bases: 110980302(94.6268%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 941456\n",
      "total bases: 71502248\n",
      "Q20 bases: 71230059(99.6193%)\n",
      "Q30 bases: 69975404(97.8646%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 941456\n",
      "total bases: 71475352\n",
      "Q20 bases: 70978000(99.3042%)\n",
      "Q30 bases: 69107114(96.6866%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1882912\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1229292\n",
      "reads with adapter trimmed: 6714\n",
      "bases trimmed due to adapters: 177235\n",
      "\n",
      "Duplication rate: 8.84347%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 117\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753376_chr7_1.fastq.gz -I data/raw/ERR753376_chr7_2.fastq.gz -o data/processed/ERR753376_chr7_1_fastp.fastq.gz -O data/processed/ERR753376_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Thu Aug 17 13:01:48 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:01:48 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753376_chr7_1_fastp.fastq.gz, data/processed/ERR753376_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753376_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753376_chr7_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753376_chr7_2_fastp.fastq.gz, data/processed/ERR753376_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 13:01:58 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T130129.011712.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:01:59 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753377_chr7_1.fastq.gz, data/raw/ERR753377_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753377_chr7_1_fastqc.html, results/fastqc_result/ERR753377_chr7_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753377_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753377_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 13:02:12 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:02:12 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753377_chr7_1.fastq.gz, data/raw/ERR753377_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753377_chr7_1_fastp.fastq.gz, data/processed/ERR753377_chr7_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753377_chr7_2_fastp.fastq.gz, data/processed/ERR753377_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1631934\n",
      "total bases: 123352115\n",
      "Q20 bases: 122472837(99.2872%)\n",
      "Q30 bases: 119023024(96.4905%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1631934\n",
      "total bases: 123051439\n",
      "Q20 bases: 121582205(98.806%)\n",
      "Q30 bases: 116732556(94.8648%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1009848\n",
      "total bases: 76696745\n",
      "Q20 bases: 76397840(99.6103%)\n",
      "Q30 bases: 75043318(97.8442%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1009848\n",
      "total bases: 76668037\n",
      "Q20 bases: 76158958(99.336%)\n",
      "Q30 bases: 74237699(96.8301%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2019696\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1244172\n",
      "reads with adapter trimmed: 6972\n",
      "bases trimmed due to adapters: 178774\n",
      "\n",
      "Duplication rate: 8.81487%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753377_chr7_1.fastq.gz -I data/raw/ERR753377_chr7_2.fastq.gz -o data/processed/ERR753377_chr7_1_fastp.fastq.gz -O data/processed/ERR753377_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Thu Aug 17 13:02:18 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:02:18 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753377_chr7_1_fastp.fastq.gz, data/processed/ERR753377_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753377_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753377_chr7_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753377_chr7_2_fastp.fastq.gz, data/processed/ERR753377_chr7_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 13:02:27 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T130158.412735.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:02:28 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753378_chr7_1.fastq.gz, data/raw/ERR753378_chr7_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753378_chr7_1_fastqc.html, results/fastqc_result/ERR753378_chr7_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753378_chr7_1_fastqc.html\n",
      "    wildcards: sample=ERR753378_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr7_1.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr7_1.fastq.gz\n",
      "Approx 100% complete for ERR753378_chr7_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr7_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr7_2.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr7_2.fastq.gz\n",
      "Approx 100% complete for ERR753378_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr7_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 13:02:41 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:02:41 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753378_chr7_1.fastq.gz, data/raw/ERR753378_chr7_2.fastq.gz\n",
      "    output: data/processed/ERR753378_chr7_1_fastp.fastq.gz, data/processed/ERR753378_chr7_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753378_chr7_1_fastp.fastq.gz, data/processed/ERR753378_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1662012\n",
      "total bases: 126189423\n",
      "Q20 bases: 125691373(99.6053%)\n",
      "Q30 bases: 121897371(96.5987%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1662012\n",
      "total bases: 126054782\n",
      "Q20 bases: 124937025(99.1133%)\n",
      "Q30 bases: 119131691(94.5079%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1424617\n",
      "total bases: 108264659\n",
      "Q20 bases: 108073674(99.8236%)\n",
      "Q30 bases: 106186267(98.0803%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1424617\n",
      "total bases: 108255476\n",
      "Q20 bases: 107719628(99.505%)\n",
      "Q30 bases: 104436218(96.472%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2849234\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 474790\n",
      "reads with adapter trimmed: 5220\n",
      "bases trimmed due to adapters: 141481\n",
      "\n",
      "Duplication rate: 1.98966%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753378_chr7_1.fastq.gz -I data/raw/ERR753378_chr7_2.fastq.gz -o data/processed/ERR753378_chr7_1_fastp.fastq.gz -O data/processed/ERR753378_chr7_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Thu Aug 17 13:02:47 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:02:47 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753378_chr7_1_fastp.fastq.gz, data/processed/ERR753378_chr7_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753378_chr7_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753378_chr7_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753378_chr7_1_fastp.fastq.gz, data/processed/ERR753378_chr7_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr7_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr7_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr7_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr7_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 13:02:58 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T130227.322173.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:02:59 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR696683_chr7_1_fastp.fastq.gz, data/processed/ERR696683_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR696683_chr7.sam, results/mapped_reads/ERR696683_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR696683_chr7_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR696683_chr7_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.65 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 60.43 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.50 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.35 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 34.88 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 101.616 sec; CPU: 96.823 sec\n",
      "[Thu Aug 17 13:06:38 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:06:38 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR696683_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr7.bam, results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR696683_chr7.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR696683_chr7_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 13:07:23 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:07:23 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR696683_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "13:07:24.487 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 13:07:24 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR696683_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR696683_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 13:07:24 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 13:07:24\tMarkDuplicates\tStart of doWork freeMemory: 528788216; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:07:24\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 13:07:24\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam\n",
      "INFO\t2023-08-17 13:07:26\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:75,878,958\n",
      "INFO\t2023-08-17 13:07:26\tMarkDuplicates\tTracking 731 as yet unmatched pairs. 731 records in RAM.\n",
      "INFO\t2023-08-17 13:07:27\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 7:138,577,465\n",
      "INFO\t2023-08-17 13:07:27\tMarkDuplicates\tTracking 375 as yet unmatched pairs. 375 records in RAM.\n",
      "INFO\t2023-08-17 13:07:28\tMarkDuplicates\tRead 2393966 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 13:07:28\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 521988568; totalMemory: 848297984; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:07:28\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 13:07:28\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 13:07:28\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 13:07:29\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 13:07:29\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 936331088; totalMemory: 1482686464; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:07:29\tMarkDuplicates\tMarking 206367 records as duplicates.\n",
      "INFO\t2023-08-17 13:07:29\tMarkDuplicates\tFound 76 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr7_sorted.bam\n",
      "INFO\t2023-08-17 13:07:29\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 13:07:43\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 13:07:43\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 13:07:43\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 13:07:43\tMarkDuplicates\tBefore output close freeMemory: 528085984; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:07:43\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 13:07:43\tMarkDuplicates\tAfter output close freeMemory: 528085984; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 13:07:43 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.32 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 13:07:45 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T130258.732363.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:07:46 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753368_chr7_1_fastp.fastq.gz, data/processed/ERR753368_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753368_chr7.sam, results/mapped_reads/ERR753368_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753368_chr7_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753368_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.55 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 57.86 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.48 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.34 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 33.55 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 97.874 sec; CPU: 92.783 sec\n",
      "[Thu Aug 17 13:11:58 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:11:58 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753368_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr7.bam, results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753368_chr7.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753368_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 13:12:48 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:12:48 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753368_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753368_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "13:12:49.558 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 13:12:49 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753368_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753368_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 13:12:49 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 13:12:49\tMarkDuplicates\tStart of doWork freeMemory: 528786104; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:12:49\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 13:12:49\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam\n",
      "INFO\t2023-08-17 13:12:51\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:56,377,876\n",
      "INFO\t2023-08-17 13:12:51\tMarkDuplicates\tTracking 419 as yet unmatched pairs. 419 records in RAM.\n",
      "INFO\t2023-08-17 13:12:52\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 7:105,465,321\n",
      "INFO\t2023-08-17 13:12:52\tMarkDuplicates\tTracking 474 as yet unmatched pairs. 474 records in RAM.\n",
      "INFO\t2023-08-17 13:12:54\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 7:152,248,153\n",
      "INFO\t2023-08-17 13:12:54\tMarkDuplicates\tTracking 277 as yet unmatched pairs. 277 records in RAM.\n",
      "INFO\t2023-08-17 13:12:54\tMarkDuplicates\tRead 3118785 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 13:12:54\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 453234120; totalMemory: 859832320; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:12:54\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 13:12:54\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 13:12:54\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 13:12:55\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 13:12:55\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1051673144; totalMemory: 1598029824; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:12:55\tMarkDuplicates\tMarking 74930 records as duplicates.\n",
      "INFO\t2023-08-17 13:12:55\tMarkDuplicates\tFound 515 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr7_sorted.bam\n",
      "INFO\t2023-08-17 13:12:55\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 13:13:14\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 13:13:14\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 13:13:14\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 13:13:14\tMarkDuplicates\tBefore output close freeMemory: 528084400; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:13:14\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 13:13:14\tMarkDuplicates\tAfter output close freeMemory: 528084400; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 13:13:14 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.41 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 13:13:17 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T130745.637057.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:13:20 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753369_chr7_1_fastp.fastq.gz, data/processed/ERR753369_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753369_chr7.sam, results/mapped_reads/ERR753369_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753369_chr7_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR753369_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.55 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 59.56 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.49 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.35 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 34.06 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 100.105 sec; CPU: 94.997 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 13:17:43 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:17:43 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753369_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr7.bam, results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753369_chr7.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753369_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 13:18:43 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:18:43 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753369_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753369_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "13:18:44.696 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 13:18:44 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753369_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753369_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 13:18:44 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 13:18:44\tMarkDuplicates\tStart of doWork freeMemory: 528792256; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:18:44\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 13:18:44\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam\n",
      "INFO\t2023-08-17 13:18:46\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:53,274,020\n",
      "INFO\t2023-08-17 13:18:46\tMarkDuplicates\tTracking 382 as yet unmatched pairs. 382 records in RAM.\n",
      "INFO\t2023-08-17 13:18:47\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 7:101,057,919\n",
      "INFO\t2023-08-17 13:18:47\tMarkDuplicates\tTracking 524 as yet unmatched pairs. 524 records in RAM.\n",
      "INFO\t2023-08-17 13:18:49\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 7:144,095,190\n",
      "INFO\t2023-08-17 13:18:49\tMarkDuplicates\tTracking 239 as yet unmatched pairs. 239 records in RAM.\n",
      "INFO\t2023-08-17 13:18:49\tMarkDuplicates\tRead 3329144 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 13:18:49\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 393092896; totalMemory: 822083584; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:18:49\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 13:18:49\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 13:18:50\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 13:18:50\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 13:18:50\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1112495728; totalMemory: 1658847232; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:18:50\tMarkDuplicates\tMarking 92624 records as duplicates.\n",
      "INFO\t2023-08-17 13:18:50\tMarkDuplicates\tFound 529 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr7_sorted.bam\n",
      "INFO\t2023-08-17 13:18:50\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 13:19:09\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 13:19:09\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 13:19:09\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 13:19:09\tMarkDuplicates\tBefore output close freeMemory: 528090248; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:19:09\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 13:19:09\tMarkDuplicates\tAfter output close freeMemory: 528090248; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 13:19:09 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.42 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 13:19:12 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T131318.053038.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:19:13 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753370_chr7_1_fastp.fastq.gz, data/processed/ERR753370_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753370_chr7.sam, results/mapped_reads/ERR753370_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753370_chr7_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753370_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.56 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 59.58 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.50 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.35 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 34.13 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 100.731 sec; CPU: 95.107 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 13:26:39 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:26:39 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753370_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr7.bam, results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753370_chr7.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753370_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 13:28:41 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:28:41 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753370_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753370_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "13:28:42.147 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 13:28:42 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753370_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753370_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 13:28:42 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 13:28:42\tMarkDuplicates\tStart of doWork freeMemory: 528785928; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:28:42\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 13:28:42\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam\n",
      "INFO\t2023-08-17 13:28:43\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:33,152,779\n",
      "INFO\t2023-08-17 13:28:43\tMarkDuplicates\tTracking 1012 as yet unmatched pairs. 1012 records in RAM.\n",
      "INFO\t2023-08-17 13:28:45\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 7:72,939,724\n",
      "INFO\t2023-08-17 13:28:45\tMarkDuplicates\tTracking 1944 as yet unmatched pairs. 1944 records in RAM.\n",
      "INFO\t2023-08-17 13:28:47\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 7:95,299,055\n",
      "INFO\t2023-08-17 13:28:47\tMarkDuplicates\tTracking 1672 as yet unmatched pairs. 1672 records in RAM.\n",
      "INFO\t2023-08-17 13:28:48\tMarkDuplicates\tRead     4,000,000 records.  Elapsed time: 00:00:06s.  Time for last 1,000,000:    1s.  Last read position: 7:117,115,568\n",
      "INFO\t2023-08-17 13:28:48\tMarkDuplicates\tTracking 1114 as yet unmatched pairs. 1114 records in RAM.\n",
      "INFO\t2023-08-17 13:28:50\tMarkDuplicates\tRead     5,000,000 records.  Elapsed time: 00:00:07s.  Time for last 1,000,000:    1s.  Last read position: 7:144,268,946\n",
      "INFO\t2023-08-17 13:28:50\tMarkDuplicates\tTracking 730 as yet unmatched pairs. 730 records in RAM.\n",
      "INFO\t2023-08-17 13:28:50\tMarkDuplicates\tRead 5445839 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 13:28:50\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 446546784; totalMemory: 1095761920; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:28:50\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 13:28:51\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 13:28:51\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 13:28:52\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 13:28:52\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1086275208; totalMemory: 1632632832; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:28:52\tMarkDuplicates\tMarking 368354 records as duplicates.\n",
      "INFO\t2023-08-17 13:28:52\tMarkDuplicates\tFound 219 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr7_sorted.bam\n",
      "INFO\t2023-08-17 13:28:52\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 13:29:31\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 13:29:31\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 13:29:31\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 13:29:31\tMarkDuplicates\tBefore output close freeMemory: 528084232; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:29:31\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 13:29:31\tMarkDuplicates\tAfter output close freeMemory: 528084232; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 13:29:31 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.83 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 13:29:35 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T131912.497746.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:29:37 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753371_chr7_1_fastp.fastq.gz, data/processed/ERR753371_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753371_chr7.sam, results/mapped_reads/ERR753371_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753371_chr7_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753371_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.53 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 60.80 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.48 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.33 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 34.32 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 101.955 sec; CPU: 96.467 sec\n",
      "[Thu Aug 17 13:32:46 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:32:46 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753371_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr7.bam, results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753371_chr7.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753371_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 13:33:19 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:33:19 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753371_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753371_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "13:33:20.905 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 13:33:20 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753371_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753371_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 13:33:20 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 13:33:20\tMarkDuplicates\tStart of doWork freeMemory: 528785672; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:33:20\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 13:33:20\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam\n",
      "INFO\t2023-08-17 13:33:22\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:94,599,614\n",
      "INFO\t2023-08-17 13:33:22\tMarkDuplicates\tTracking 173 as yet unmatched pairs. 173 records in RAM.\n",
      "INFO\t2023-08-17 13:33:23\tMarkDuplicates\tRead 1812344 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 13:33:23\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 335230368; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:33:23\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 13:33:23\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 13:33:24\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 13:33:24\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 13:33:24\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 813643120; totalMemory: 1360003072; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:33:24\tMarkDuplicates\tMarking 702765 records as duplicates.\n",
      "INFO\t2023-08-17 13:33:24\tMarkDuplicates\tFound 57 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr7_sorted.bam\n",
      "INFO\t2023-08-17 13:33:24\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 13:33:35\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 13:33:35\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 13:33:35\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 13:33:35\tMarkDuplicates\tBefore output close freeMemory: 528081736; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:33:35\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 13:33:35\tMarkDuplicates\tAfter output close freeMemory: 528081736; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 13:33:35 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.24 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 13:33:36 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T132936.124420.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:33:38 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753372_chr7_1_fastp.fastq.gz, data/processed/ERR753372_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753372_chr7.sam, results/mapped_reads/ERR753372_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753372_chr7_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR753372_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.56 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 57.56 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.49 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.33 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 34.41 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 98.700 sec; CPU: 93.354 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 13:38:06 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:38:06 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753372_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr7.bam, results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753372_chr7.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753372_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 13:39:06 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:39:06 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753372_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753372_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "13:39:07.668 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 13:39:07 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753372_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753372_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 13:39:07 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 13:39:07\tMarkDuplicates\tStart of doWork freeMemory: 528788632; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:39:07\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 13:39:07\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam\n",
      "INFO\t2023-08-17 13:39:09\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:51,799,456\n",
      "INFO\t2023-08-17 13:39:09\tMarkDuplicates\tTracking 415 as yet unmatched pairs. 415 records in RAM.\n",
      "INFO\t2023-08-17 13:39:10\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 7:101,087,627\n",
      "INFO\t2023-08-17 13:39:10\tMarkDuplicates\tTracking 487 as yet unmatched pairs. 487 records in RAM.\n",
      "INFO\t2023-08-17 13:39:12\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 7:144,371,098\n",
      "INFO\t2023-08-17 13:39:12\tMarkDuplicates\tTracking 225 as yet unmatched pairs. 225 records in RAM.\n",
      "INFO\t2023-08-17 13:39:12\tMarkDuplicates\tRead 3322476 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 13:39:12\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 409081328; totalMemory: 836763648; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:39:12\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 13:39:12\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 13:39:13\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 13:39:13\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 13:39:13\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 827279584; totalMemory: 1373634560; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:39:13\tMarkDuplicates\tMarking 78247 records as duplicates.\n",
      "INFO\t2023-08-17 13:39:13\tMarkDuplicates\tFound 501 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr7_sorted.bam\n",
      "INFO\t2023-08-17 13:39:13\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 13:39:33\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 13:39:33\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 13:39:33\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 13:39:33\tMarkDuplicates\tBefore output close freeMemory: 528087032; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:39:33\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 13:39:33\tMarkDuplicates\tAfter output close freeMemory: 528087032; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 13:39:33 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.43 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 13:39:35 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T133336.934908.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:39:37 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753373_chr7_1_fastp.fastq.gz, data/processed/ERR753373_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753373_chr7.sam, results/mapped_reads/ERR753373_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753373_chr7_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753373_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.55 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 59.57 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.51 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.34 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 34.02 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 100.560 sec; CPU: 94.983 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 13:45:49 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:45:49 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753373_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr7.bam, results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753373_chr7.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753373_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 13:47:23 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:47:23 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753373_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753373_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "13:47:24.365 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 13:47:24 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753373_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753373_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 13:47:24 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 13:47:24\tMarkDuplicates\tStart of doWork freeMemory: 528832904; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:47:24\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 13:47:24\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam\n",
      "INFO\t2023-08-17 13:47:26\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:54,553,807\n",
      "INFO\t2023-08-17 13:47:26\tMarkDuplicates\tTracking 256 as yet unmatched pairs. 256 records in RAM.\n",
      "INFO\t2023-08-17 13:47:27\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 7:102,664,978\n",
      "INFO\t2023-08-17 13:47:27\tMarkDuplicates\tTracking 231 as yet unmatched pairs. 231 records in RAM.\n",
      "INFO\t2023-08-17 13:47:29\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 7:150,822,446\n",
      "INFO\t2023-08-17 13:47:29\tMarkDuplicates\tTracking 56 as yet unmatched pairs. 56 records in RAM.\n",
      "INFO\t2023-08-17 13:47:29\tMarkDuplicates\tRead 3137876 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 13:47:29\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 483145624; totalMemory: 890241024; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:47:29\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 13:47:29\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 13:47:29\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 13:47:30\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 13:47:30\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 880802328; totalMemory: 1427111936; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:47:30\tMarkDuplicates\tMarking 79801 records as duplicates.\n",
      "INFO\t2023-08-17 13:47:30\tMarkDuplicates\tFound 26039 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr7_sorted.bam\n",
      "INFO\t2023-08-17 13:47:30\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 13:47:56\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 13:47:56\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 13:47:56\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 13:47:56\tMarkDuplicates\tBefore output close freeMemory: 528132024; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:47:56\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 13:47:56\tMarkDuplicates\tAfter output close freeMemory: 528132024; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 13:47:56 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.54 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 13:47:59 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T133935.987878.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:48:00 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753374_chr7_1_fastp.fastq.gz, data/processed/ERR753374_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753374_chr7.sam, results/mapped_reads/ERR753374_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753374_chr7_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753374_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.54 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 55.41 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.48 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.33 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 32.52 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 94.531 sec; CPU: 89.285 sec\n",
      "[Thu Aug 17 13:52:18 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:52:18 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753374_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr7.bam, results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753374_chr7.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753374_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 13:53:05 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:53:05 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753374_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753374_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "13:53:06.854 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 13:53:06 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753374_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753374_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 13:53:06 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 13:53:06\tMarkDuplicates\tStart of doWork freeMemory: 528788768; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:53:06\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 13:53:06\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam\n",
      "INFO\t2023-08-17 13:53:08\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:91,121,525\n",
      "INFO\t2023-08-17 13:53:08\tMarkDuplicates\tTracking 227 as yet unmatched pairs. 227 records in RAM.\n",
      "INFO\t2023-08-17 13:53:09\tMarkDuplicates\tRead 1838287 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 13:53:09\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 269049136; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:53:09\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 13:53:10\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 13:53:10\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 13:53:10\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 13:53:10\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 810502120; totalMemory: 1356857344; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:53:10\tMarkDuplicates\tMarking 189110 records as duplicates.\n",
      "INFO\t2023-08-17 13:53:10\tMarkDuplicates\tFound 63001 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr7_sorted.bam\n",
      "INFO\t2023-08-17 13:53:10\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 13:53:25\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 13:53:25\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 13:53:25\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 13:53:25\tMarkDuplicates\tBefore output close freeMemory: 528086088; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:53:25\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 13:53:25\tMarkDuplicates\tAfter output close freeMemory: 528086088; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 13:53:25 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.31 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 13:53:27 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T134759.623817.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:53:28 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753375_chr7_1_fastp.fastq.gz, data/processed/ERR753375_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753375_chr7.sam, results/mapped_reads/ERR753375_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753375_chr7_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753375_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.55 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 55.51 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.48 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.33 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 32.97 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 95.115 sec; CPU: 89.834 sec\n",
      "[Thu Aug 17 13:56:17 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:56:17 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753375_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr7.bam, results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753375_chr7.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753375_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 13:56:44 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:56:44 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753375_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753375_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "13:56:45.696 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 13:56:45 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753375_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753375_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 13:56:45 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 13:56:45\tMarkDuplicates\tStart of doWork freeMemory: 528775048; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:56:45\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 13:56:45\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam\n",
      "INFO\t2023-08-17 13:56:47\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:102,659,630\n",
      "INFO\t2023-08-17 13:56:47\tMarkDuplicates\tTracking 230 as yet unmatched pairs. 230 records in RAM.\n",
      "INFO\t2023-08-17 13:56:48\tMarkDuplicates\tRead 1549276 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 13:56:48\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 299152720; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:56:48\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 13:56:48\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 13:56:48\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 13:56:48\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 13:56:48\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 761205568; totalMemory: 1307574272; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:56:48\tMarkDuplicates\tMarking 70163 records as duplicates.\n",
      "INFO\t2023-08-17 13:56:48\tMarkDuplicates\tFound 171 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr7_sorted.bam\n",
      "INFO\t2023-08-17 13:56:48\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 13:56:58\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 13:56:58\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 13:56:58\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 13:56:58\tMarkDuplicates\tBefore output close freeMemory: 528072984; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 13:56:58\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 13:56:58\tMarkDuplicates\tAfter output close freeMemory: 528072984; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 13:56:58 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.22 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 13:56:59 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T135327.364425.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 13:57:01 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753376_chr7_1_fastp.fastq.gz, data/processed/ERR753376_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753376_chr7.sam, results/mapped_reads/ERR753376_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753376_chr7_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR753376_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.56 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 60.27 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.50 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.34 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 34.19 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 101.001 sec; CPU: 95.868 sec\n",
      "[Thu Aug 17 14:00:13 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:00:13 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753376_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr7.bam, results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753376_chr7.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753376_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 14:00:50 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:00:50 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753376_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753376_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "14:00:51.463 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 14:00:51 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753376_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753376_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 14:00:51 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 14:00:51\tMarkDuplicates\tStart of doWork freeMemory: 528775088; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:00:51\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 14:00:51\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam\n",
      "INFO\t2023-08-17 14:00:53\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:93,135,587\n",
      "INFO\t2023-08-17 14:00:53\tMarkDuplicates\tTracking 288 as yet unmatched pairs. 288 records in RAM.\n",
      "INFO\t2023-08-17 14:00:54\tMarkDuplicates\tRead 1892327 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 14:00:54\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 264683152; totalMemory: 540016640; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:00:54\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 14:00:54\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 14:00:54\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 14:00:55\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 14:00:55\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 848237216; totalMemory: 1394606080; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:00:55\tMarkDuplicates\tMarking 255018 records as duplicates.\n",
      "INFO\t2023-08-17 14:00:55\tMarkDuplicates\tFound 82 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr7_sorted.bam\n",
      "INFO\t2023-08-17 14:00:55\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 14:01:07\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 14:01:07\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 14:01:07\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 14:01:07\tMarkDuplicates\tBefore output close freeMemory: 528073072; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:01:07\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 14:01:07\tMarkDuplicates\tAfter output close freeMemory: 528073072; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 14:01:07 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.26 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 14:01:08 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T135659.983494.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:01:09 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753377_chr7_1_fastp.fastq.gz, data/processed/ERR753377_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753377_chr7.sam, results/mapped_reads/ERR753377_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753377_chr7_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR753377_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.57 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 59.49 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.50 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.35 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 32.81 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 98.875 sec; CPU: 93.719 sec\n",
      "[Thu Aug 17 14:04:21 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:04:21 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753377_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr7.bam, results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753377_chr7.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753377_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 14:04:58 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:04:58 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753377_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753377_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "14:04:59.599 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 14:04:59 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753377_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753377_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 14:04:59 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 14:04:59\tMarkDuplicates\tStart of doWork freeMemory: 528784840; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:04:59\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 14:04:59\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam\n",
      "INFO\t2023-08-17 14:05:01\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:88,812,800\n",
      "INFO\t2023-08-17 14:05:01\tMarkDuplicates\tTracking 416 as yet unmatched pairs. 416 records in RAM.\n",
      "INFO\t2023-08-17 14:05:02\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 7:154,976,188\n",
      "INFO\t2023-08-17 14:05:02\tMarkDuplicates\tTracking 144 as yet unmatched pairs. 144 records in RAM.\n",
      "INFO\t2023-08-17 14:05:02\tMarkDuplicates\tRead 2043633 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 14:05:02\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 311898448; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:05:02\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 14:05:02\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 14:05:03\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 14:05:03\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 14:05:03\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 848247592; totalMemory: 1394606080; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:05:03\tMarkDuplicates\tMarking 269869 records as duplicates.\n",
      "INFO\t2023-08-17 14:05:03\tMarkDuplicates\tFound 71 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr7_sorted.bam\n",
      "INFO\t2023-08-17 14:05:03\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 14:05:15\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 14:05:15\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 14:05:15\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 14:05:15\tMarkDuplicates\tBefore output close freeMemory: 528083528; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:05:15\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 14:05:15\tMarkDuplicates\tAfter output close freeMemory: 528083528; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 14:05:15 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.27 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 14:05:16 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T140108.721224.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:05:18 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753378_chr7_1_fastp.fastq.gz, data/processed/ERR753378_chr7_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753378_chr7.sam, results/mapped_reads/ERR753378_chr7_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753378_chr7_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR753378_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.54 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
      "[bwa_index] 55.50 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.48 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.33 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 32.59 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 94.615 sec; CPU: 89.432 sec\n",
      "[Thu Aug 17 14:09:01 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:09:01 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753378_chr7_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr7.bam, results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753378_chr7.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753378_chr7_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 14:09:44 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:09:44 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr7_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753378_chr7_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam\n",
      "    wildcards: sample=ERR753378_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "14:09:45.671 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 14:09:45 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753378_chr7_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753378_chr7_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 14:09:45 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 14:09:45\tMarkDuplicates\tStart of doWork freeMemory: 528786880; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:09:45\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 14:09:45\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam\n",
      "INFO\t2023-08-17 14:09:47\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 7:65,356,727\n",
      "INFO\t2023-08-17 14:09:47\tMarkDuplicates\tTracking 549 as yet unmatched pairs. 549 records in RAM.\n",
      "INFO\t2023-08-17 14:09:48\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 7:116,559,170\n",
      "INFO\t2023-08-17 14:09:48\tMarkDuplicates\tTracking 424 as yet unmatched pairs. 424 records in RAM.\n",
      "INFO\t2023-08-17 14:09:49\tMarkDuplicates\tRead 2871707 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 14:09:49\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 843926016; totalMemory: 1223688192; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:09:49\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 14:09:50\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 14:09:50\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 14:09:50\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 14:09:50\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1214203120; totalMemory: 1760559104; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:09:50\tMarkDuplicates\tMarking 67008 records as duplicates.\n",
      "INFO\t2023-08-17 14:09:50\tMarkDuplicates\tFound 448 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr7_sorted.bam\n",
      "INFO\t2023-08-17 14:09:50\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 14:10:06\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 14:10:06\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 14:10:06\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 14:10:06\tMarkDuplicates\tBefore output close freeMemory: 528085864; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:10:06\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 14:10:06\tMarkDuplicates\tAfter output close freeMemory: 528085864; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 14:10:06 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.36 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 14:10:08 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T140517.086226.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:10:10 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR696683_chr7_dedup.bam\n",
      "    output: results/variants/ERR696683_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR696683_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR696683_chr7.vcf\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "index file data/reference/genome.fa.fai not found, generating...\n",
      "[Thu Aug 17 14:10:57 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T141009.095388.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:10:58 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753368_chr7_dedup.bam\n",
      "    output: results/variants/ERR753368_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753368_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753368_chr7.vcf\n",
      "    wildcards: sample=ERR753368_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 14:12:01 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T141057.233680.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:12:02 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753369_chr7_dedup.bam\n",
      "    output: results/variants/ERR753369_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753369_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753369_chr7.vcf\n",
      "    wildcards: sample=ERR753369_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 14:13:07 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T141201.406630.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:13:08 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753370_chr7_dedup.bam\n",
      "    output: results/variants/ERR753370_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753370_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753370_chr7.vcf\n",
      "    wildcards: sample=ERR753370_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 14:14:55 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T141307.461615.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:14:56 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753371_chr7_dedup.bam\n",
      "    output: results/variants/ERR753371_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753371_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753371_chr7.vcf\n",
      "    wildcards: sample=ERR753371_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 14:15:30 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T141455.435987.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:15:31 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753372_chr7_dedup.bam\n",
      "    output: results/variants/ERR753372_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753372_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753372_chr7.vcf\n",
      "    wildcards: sample=ERR753372_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 14:16:39 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T141530.598035.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:16:41 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753373_chr7_dedup.bam\n",
      "    output: results/variants/ERR753373_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753373_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753373_chr7.vcf\n",
      "    wildcards: sample=ERR753373_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 14:18:00 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T141639.870926.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:18:01 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753374_chr7_dedup.bam\n",
      "    output: results/variants/ERR753374_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753374_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753374_chr7.vcf\n",
      "    wildcards: sample=ERR753374_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 14:18:54 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T141800.736114.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:18:55 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753375_chr7_dedup.bam\n",
      "    output: results/variants/ERR753375_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753375_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753375_chr7.vcf\n",
      "    wildcards: sample=ERR753375_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 14:19:34 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T141854.262451.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:19:36 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753376_chr7_dedup.bam\n",
      "    output: results/variants/ERR753376_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753376_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753376_chr7.vcf\n",
      "    wildcards: sample=ERR753376_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 14:20:17 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T141935.178714.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:20:18 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753377_chr7_dedup.bam\n",
      "    output: results/variants/ERR753377_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753377_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753377_chr7.vcf\n",
      "    wildcards: sample=ERR753377_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 14:21:00 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T142017.675098.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:21:02 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753378_chr7_dedup.bam\n",
      "    output: results/variants/ERR753378_chr7.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753378_chr7.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753378_chr7.vcf\n",
      "    wildcards: sample=ERR753378_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 14:22:02 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T142100.906387.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:22:03 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR696683_chr7.vcf\n",
      "    output: results/variants/vep/ERR696683_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR696683_chr7.txt\n",
      "    wildcards: sample=ERR696683_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 14:22:30 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T142202.435294.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:22:32 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753368_chr7.vcf\n",
      "    output: results/variants/vep/ERR753368_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753368_chr7.txt\n",
      "    wildcards: sample=ERR753368_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 14:23:03 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T142230.392844.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:23:05 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753369_chr7.vcf\n",
      "    output: results/variants/vep/ERR753369_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753369_chr7.txt\n",
      "    wildcards: sample=ERR753369_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 14:23:35 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T142303.767784.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:23:36 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753370_chr7.vcf\n",
      "    output: results/variants/vep/ERR753370_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753370_chr7.txt\n",
      "    wildcards: sample=ERR753370_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 14:24:11 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T142335.729820.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:24:12 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753371_chr7.vcf\n",
      "    output: results/variants/vep/ERR753371_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753371_chr7.txt\n",
      "    wildcards: sample=ERR753371_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 14:24:34 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T142411.736668.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:24:36 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753372_chr7.vcf\n",
      "    output: results/variants/vep/ERR753372_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753372_chr7.txt\n",
      "    wildcards: sample=ERR753372_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 14:25:05 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T142434.782608.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:25:07 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753373_chr7.vcf\n",
      "    output: results/variants/vep/ERR753373_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753373_chr7.txt\n",
      "    wildcards: sample=ERR753373_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 14:25:39 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T142506.146099.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:25:40 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753374_chr7.vcf\n",
      "    output: results/variants/vep/ERR753374_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753374_chr7.txt\n",
      "    wildcards: sample=ERR753374_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 14:26:03 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T142539.397123.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:26:05 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753375_chr7.vcf\n",
      "    output: results/variants/vep/ERR753375_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753375_chr7.txt\n",
      "    wildcards: sample=ERR753375_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 14:26:28 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T142603.985218.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:26:29 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753376_chr7.vcf\n",
      "    output: results/variants/vep/ERR753376_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753376_chr7.txt\n",
      "    wildcards: sample=ERR753376_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 14:26:53 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T142628.438708.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:26:54 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753377_chr7.vcf\n",
      "    output: results/variants/vep/ERR753377_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753377_chr7.txt\n",
      "    wildcards: sample=ERR753377_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 14:27:17 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T142653.204848.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:27:19 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753378_chr7.vcf\n",
      "    output: results/variants/vep/ERR753378_chr7.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753378_chr7.txt\n",
      "    wildcards: sample=ERR753378_chr7\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 14:27:47 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T142718.162459.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job              count    min threads    max threads\n",
      "-------------  -------  -------------  -------------\n",
      "parsing_dataR        1              1              1\n",
      "total                1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:27:48 2023]\n",
      "rule parsing_dataR:\n",
      "    input: code/05parsing_vep_data.R\n",
      "    output: tasks/13parsing_dataR.done\n",
      "    jobid: 0\n",
      "    reason: Params have changed since last execution\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Warning message:\n",
      "package ‘glue’ was built under R version 4.2.3 \n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">>> This is your sample list: ERR696683_chr7\n",
      ">>> This is your sample list: ERR753368_chr7\n",
      ">>> This is your sample list: ERR753369_chr7\n",
      ">>> This is your sample list: ERR753370_chr7\n",
      ">>> This is your sample list: ERR753371_chr7\n",
      ">>> This is your sample list: ERR753372_chr7\n",
      ">>> This is your sample list: ERR753373_chr7\n",
      ">>> This is your sample list: ERR753374_chr7\n",
      ">>> This is your sample list: ERR753375_chr7\n",
      ">>> This is your sample list: ERR753376_chr7\n",
      ">>> This is your sample list: ERR753377_chr7\n",
      ">>> This is your sample list: ERR753378_chr7\n",
      ">>> This is your gene: BRAF\n",
      "# A tibble: 44 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "3 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "4 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   downstream…\n",
      "6 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 38 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 52 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "2 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "3 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "4 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "5 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "6 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 46 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 5 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "3 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "4 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   downstream…\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>,\n",
      "#   freqs <chr>, clin_sig <chr>, somatic <chr>, pheno <chr>, pubmed <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 5 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "3 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "4 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   downstream…\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>,\n",
      "#   freqs <chr>, clin_sig <chr>, somatic <chr>, pheno <chr>, pubmed <chr>, …\n",
      "# A tibble: 31 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "3 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "4 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   downstream…\n",
      "6 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 25 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 31 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "3 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "4 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   downstream…\n",
      "6 7_140749350_T/C       7:1407493… C      ENSG… ENST00… Transcript   synonymous…\n",
      "# ℹ 25 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 52 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "2 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "3 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "4 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "5 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "6 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 46 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 39 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "2 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "3 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "4 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "5 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "6 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 33 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 44 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "3 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "4 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   downstream…\n",
      "6 7_140749271_C/G       7:1407492… G      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 38 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 5 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "3 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   non_coding…\n",
      "4 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 7_140726457_A/G       7:1407264… G      ENSG… ENST00… Transcript   downstream…\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>,\n",
      "#   freqs <chr>, clin_sig <chr>, somatic <chr>, pheno <chr>, pubmed <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 0 × 44\n",
      "# ℹ 44 variables: #uploaded_variation <chr>, location <chr>, allele <chr>,\n",
      "#   gene <chr>, feature <chr>, feature_type <chr>, consequence <chr>,\n",
      "#   cdna_position <chr>, cds_position <chr>, protein_position <chr>,\n",
      "#   amino_acids <chr>, codons <chr>, existing_variation <chr>, impact <chr>,\n",
      "#   distance <chr>, strand <chr>, flags <chr>, variant_class <chr>,\n",
      "#   symbol <chr>, symbol_source <chr>, hgnc_id <chr>, biotype <chr>,\n",
      "#   mane_select <chr>, mane_plus_clinical <chr>, tsl <chr>, appris <chr>, …\n",
      "# A tibble: 26 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 7_140749350_T/C       7:1407493… C      ENSG… ENST00… Transcript   synonymous…\n",
      "2 7_140749350_T/C       7:1407493… C      ENSG… ENST00… Transcript   synonymous…\n",
      "3 7_140749350_T/C       7:1407493… C      ENSG… ENST00… Transcript   synonymous…\n",
      "4 7_140749350_T/C       7:1407493… C      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 7_140749350_T/C       7:1407493… C      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 7_140749350_T/C       7:1407493… C      ENSG… ENST00… Transcript   non_coding…\n",
      "# ℹ 20 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "[1] \"##==========================##\"\n",
      "[1] \"##===> WORK FINISHED!!! <===##\"\n",
      "[1] \"##==========================##\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "Touching output file tasks/13parsing_dataR.done.\n",
      "[Thu Aug 17 14:27:52 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T142748.118768.snakemake.log\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['snakemake', '--cores', '1', 'parsing_dataR'], returncode=0)"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Running the workflow for the 7º chormosome\n",
    "\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--force\", \"data/reference/genome.fa\"]) ## rule reference_genome \n",
    "\n",
    "for sample in sample_list:\n",
    "    ## Quality inspection\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/fastqc_result/\" + sample + \"_chr7_1_fastqc.html\",   \n",
    "                    \"data/processed/\" + sample + \"_chr7_1_fastp.fastq.gz\",   \n",
    "                    \"results/fastqc_result/trimmed/\" + sample + \"_chr7_1_fastp_fastqc.html\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    ## Mapping reads\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/mapped_reads/\" + sample + \"_chr7_sorted.sam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr7_sorted.bam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr7_dedup.bam\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    # Extracting variants  \n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/\" + sample + \"_chr7.vcf\"])   \n",
    "\n",
    "## VEP CLI\n",
    "for sample in sample_list:\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/vep/\" + sample + \"_chr7.txt\"])\n",
    "\n",
    "## Extracting the gene PIKCA3 from the VEP files\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"parsing_dataR\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Changing the config file to do the chromosome 12 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['sed', '-i', '50s/BRAF/KRAS/g', 'config.yaml'], returncode=0)"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subprocess.run([\"sed\", \"-i\", \"6,+12s/chr7/chr12/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\",\"24s/chr7/chr12/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"31s/chromosome.7.fa.gz/chromosome.12.fa.gz/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"50s/BRAF/KRAS/g\", \"config.yaml\"])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job                 count    min threads    max threads\n",
      "----------------  -------  -------------  -------------\n",
      "reference_genome        1              1              1\n",
      "total                   1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:29:34 2023]\n",
      "rule reference_genome:\n",
      "    output: data/reference/genome.fa\n",
      "    jobid: 0\n",
      "    reason: Forced execution\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "--2023-08-17 14:29:34--  https://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.12.fa.gz\n",
      "Resolviendo ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.169\n",
      "Conectando con ftp.ensembl.org (ftp.ensembl.org)[193.62.193.169]:443... conectado.\n",
      "Petición HTTP enviada, esperando respuesta... 200 OK\n",
      "Longitud: 39795542 (38M) [application/x-gzip]\n",
      "Guardando como: ‘data/reference/genome.fa.gz’\n",
      "\n",
      "     0K .......... .......... .......... .......... ..........  0%  444K 87s\n",
      "    50K .......... .......... .......... .......... ..........  0% 4,40M 48s\n",
      "   100K .......... .......... .......... .......... ..........  0% 1,04M 44s\n",
      "   150K .......... .......... .......... .......... ..........  0% 5,44M 35s\n",
      "   200K .......... .......... .......... .......... ..........  0% 1,05M 35s\n",
      "   250K .......... .......... .......... .......... ..........  0% 40,3M 29s\n",
      "   300K .......... .......... .......... .......... ..........  0%  118M 25s\n",
      "   350K .......... .......... .......... .......... ..........  1%  102M 22s\n",
      "   400K .......... .......... .......... .......... ..........  1% 5,53M 20s\n",
      "   450K .......... .......... .......... .......... ..........  1% 1,07M 22s\n",
      "   500K .......... .......... .......... .......... ..........  1% 49,8M 20s\n",
      "   550K .......... .......... .......... .......... ..........  1% 29,8M 18s\n",
      "   600K .......... .......... .......... .......... ..........  1%  120M 17s\n",
      "   650K .......... .......... .......... .......... ..........  1%  128M 16s\n",
      "   700K .......... .......... .......... .......... ..........  1%  104M 15s\n",
      "   750K .......... .......... .......... .......... ..........  2%  125M 14s\n",
      "   800K .......... .......... .......... .......... ..........  2%  106M 13s\n",
      "   850K .......... .......... .......... .......... ..........  2% 6,70M 12s\n",
      "   900K .......... .......... .......... .......... ..........  2%  134M 12s\n",
      "   950K .......... .......... .......... .......... ..........  2% 1,10M 13s\n",
      "  1000K .......... .......... .......... .......... ..........  2% 96,8M 12s\n",
      "  1050K .......... .......... .......... .......... ..........  2%  123M 12s\n",
      "  1100K .......... .......... .......... .......... ..........  2% 66,3M 11s\n",
      "  1150K .......... .......... .......... .......... ..........  3% 39,9M 11s\n",
      "  1200K .......... .......... .......... .......... ..........  3% 42,6M 10s\n",
      "  1250K .......... .......... .......... .......... ..........  3% 58,5M 10s\n",
      "  1300K .......... .......... .......... .......... ..........  3% 24,6M 10s\n",
      "  1350K .......... .......... .......... .......... ..........  3% 96,1M 9s\n",
      "  1400K .......... .......... .......... .......... ..........  3%  208M 9s\n",
      "  1450K .......... .......... .......... .......... ..........  3%  200M 9s\n",
      "  1500K .......... .......... .......... .......... ..........  3%  200M 8s\n",
      "  1550K .......... .......... .......... .......... ..........  4%  180M 8s\n",
      "  1600K .......... .......... .......... .......... ..........  4%  158M 8s\n",
      "  1650K .......... .......... .......... .......... ..........  4%  186M 8s\n",
      "  1700K .......... .......... .......... .......... ..........  4% 36,5M 7s\n",
      "  1750K .......... .......... .......... .......... ..........  4% 43,6M 7s\n",
      "  1800K .......... .......... .......... .......... ..........  4% 55,3M 7s\n",
      "  1850K .......... .......... .......... .......... ..........  4%  102M 7s\n",
      "  1900K .......... .......... .......... .......... ..........  5% 8,13M 7s\n",
      "  1950K .......... .......... .......... .......... ..........  5% 1,28M 7s\n",
      "  2000K .......... .......... .......... .......... ..........  5%  121M 7s\n",
      "  2050K .......... .......... .......... .......... ..........  5%  127M 7s\n",
      "  2100K .......... .......... .......... .......... ..........  5% 99,9M 7s\n",
      "  2150K .......... .......... .......... .......... ..........  5% 45,5M 7s\n",
      "  2200K .......... .......... .......... .......... ..........  5%  129M 7s\n",
      "  2250K .......... .......... .......... .......... ..........  5%  163M 6s\n",
      "  2300K .......... .......... .......... .......... ..........  6%  170M 6s\n",
      "  2350K .......... .......... .......... .......... ..........  6%  144M 6s\n",
      "  2400K .......... .......... .......... .......... ..........  6%  165M 6s\n",
      "  2450K .......... .......... .......... .......... ..........  6%  122M 6s\n",
      "  2500K .......... .......... .......... .......... ..........  6%  122M 6s\n",
      "  2550K .......... .......... .......... .......... ..........  6%  118M 6s\n",
      "  2600K .......... .......... .......... .......... ..........  6%  104M 6s\n",
      "  2650K .......... .......... .......... .......... ..........  6%  120M 5s\n",
      "  2700K .......... .......... .......... .......... ..........  7%  103M 5s\n",
      "  2750K .......... .......... .......... .......... ..........  7%  118M 5s\n",
      "  2800K .......... .......... .......... .......... ..........  7%  121M 5s\n",
      "  2850K .......... .......... .......... .......... ..........  7%  104M 5s\n",
      "  2900K .......... .......... .......... .......... ..........  7%  113M 5s\n",
      "  2950K .......... .......... .......... .......... ..........  7%  102M 5s\n",
      "  3000K .......... .......... .......... .......... ..........  7%  122M 5s\n",
      "  3050K .......... .......... .......... .......... ..........  7%  115M 5s\n",
      "  3100K .......... .......... .......... .......... ..........  8%  107M 5s\n",
      "  3150K .......... .......... .......... .......... ..........  8%  120M 5s\n",
      "  3200K .......... .......... .......... .......... ..........  8%  102M 4s\n",
      "  3250K .......... .......... .......... .......... ..........  8%  122M 4s\n",
      "  3300K .......... .......... .......... .......... ..........  8%  114M 4s\n",
      "  3350K .......... .......... .......... .......... ..........  8% 99,0M 4s\n",
      "  3400K .......... .......... .......... .......... ..........  8%  126M 4s\n",
      "  3450K .......... .......... .......... .......... ..........  9%  101M 4s\n",
      "  3500K .......... .......... .......... .......... ..........  9%  117M 4s\n",
      "  3550K .......... .......... .......... .......... ..........  9%  127M 4s\n",
      "  3600K .......... .......... .......... .......... ..........  9%  103M 4s\n",
      "  3650K .......... .......... .......... .......... ..........  9%  117M 4s\n",
      "  3700K .......... .......... .......... .......... ..........  9%  105M 4s\n",
      "  3750K .......... .......... .......... .......... ..........  9% 51,6M 4s\n",
      "  3800K .......... .......... .......... .......... ..........  9%  119M 4s\n",
      "  3850K .......... .......... .......... .......... .......... 10% 24,2M 4s\n",
      "  3900K .......... .......... .......... .......... .......... 10% 1,31M 4s\n",
      "  3950K .......... .......... .......... .......... .......... 10% 31,7M 4s\n",
      "  4000K .......... .......... .......... .......... .......... 10% 56,7M 4s\n",
      "  4050K .......... .......... .......... .......... .......... 10%  111M 4s\n",
      "  4100K .......... .......... .......... .......... .......... 10% 94,9M 4s\n",
      "  4150K .......... .......... .......... .......... .......... 10%  118M 4s\n",
      "  4200K .......... .......... .......... .......... .......... 10% 56,2M 4s\n",
      "  4250K .......... .......... .......... .......... .......... 11%  128M 4s\n",
      "  4300K .......... .......... .......... .......... .......... 11%  129M 4s\n",
      "  4350K .......... .......... .......... .......... .......... 11%  115M 4s\n",
      "  4400K .......... .......... .......... .......... .......... 11%  136M 4s\n",
      "  4450K .......... .......... .......... .......... .......... 11%  115M 4s\n",
      "  4500K .......... .......... .......... .......... .......... 11%  137M 3s\n",
      "  4550K .......... .......... .......... .......... .......... 11%  148M 3s\n",
      "  4600K .......... .......... .......... .......... .......... 11%  125M 3s\n",
      "  4650K .......... .......... .......... .......... .......... 12%  140M 3s\n",
      "  4700K .......... .......... .......... .......... .......... 12%  100M 3s\n",
      "  4750K .......... .......... .......... .......... .......... 12%  124M 3s\n",
      "  4800K .......... .......... .......... .......... .......... 12%  120M 3s\n",
      "  4850K .......... .......... .......... .......... .......... 12%  103M 3s\n",
      "  4900K .......... .......... .......... .......... .......... 12%  118M 3s\n",
      "  4950K .......... .......... .......... .......... .......... 12%  102M 3s\n",
      "  5000K .......... .......... .......... .......... .......... 12%  120M 3s\n",
      "  5050K .......... .......... .......... .......... .......... 13%  118M 3s\n",
      "  5100K .......... .......... .......... .......... .......... 13%  102M 3s\n",
      "  5150K .......... .......... .......... .......... .......... 13%  121M 3s\n",
      "  5200K .......... .......... .......... .......... .......... 13% 79,9M 3s\n",
      "  5250K .......... .......... .......... .......... .......... 13%  159M 3s\n",
      "  5300K .......... .......... .......... .......... .......... 13% 86,7M 3s\n",
      "  5350K .......... .......... .......... .......... .......... 13%  113M 3s\n",
      "  5400K .......... .......... .......... .......... .......... 14%  143M 3s\n",
      "  5450K .......... .......... .......... .......... .......... 14%  112M 3s\n",
      "  5500K .......... .......... .......... .......... .......... 14%  129M 3s\n",
      "  5550K .......... .......... .......... .......... .......... 14%  114M 3s\n",
      "  5600K .......... .......... .......... .......... .......... 14%  105M 3s\n",
      "  5650K .......... .......... .......... .......... .......... 14%  120M 3s\n",
      "  5700K .......... .......... .......... .......... .......... 14%  100M 3s\n",
      "  5750K .......... .......... .......... .......... .......... 14%  121M 3s\n",
      "  5800K .......... .......... .......... .......... .......... 15%  122M 3s\n",
      "  5850K .......... .......... .......... .......... .......... 15%  102M 3s\n",
      "  5900K .......... .......... .......... .......... .......... 15%  120M 3s\n",
      "  5950K .......... .......... .......... .......... .......... 15%  102M 3s\n",
      "  6000K .......... .......... .......... .......... .......... 15%  121M 3s\n",
      "  6050K .......... .......... .......... .......... .......... 15%  118M 3s\n",
      "  6100K .......... .......... .......... .......... .......... 15%  103M 3s\n",
      "  6150K .......... .......... .......... .......... .......... 15%  119M 2s\n",
      "  6200K .......... .......... .......... .......... .......... 16%  104M 2s\n",
      "  6250K .......... .......... .......... .......... .......... 16% 97,6M 2s\n",
      "  6300K .......... .......... .......... .......... .......... 16%  152M 2s\n",
      "  6350K .......... .......... .......... .......... .......... 16%  102M 2s\n",
      "  6400K .......... .......... .......... .......... .......... 16%  117M 2s\n",
      "  6450K .......... .......... .......... .......... .......... 16%  103M 2s\n",
      "  6500K .......... .......... .......... .......... .......... 16%  114M 2s\n",
      "  6550K .......... .......... .......... .......... .......... 16%  124M 2s\n",
      "  6600K .......... .......... .......... .......... .......... 17%  106M 2s\n",
      "  6650K .......... .......... .......... .......... .......... 17%  117M 2s\n",
      "  6700K .......... .......... .......... .......... .......... 17%  103M 2s\n",
      "  6750K .......... .......... .......... .......... .......... 17%  120M 2s\n",
      "  6800K .......... .......... .......... .......... .......... 17%  115M 2s\n",
      "  6850K .......... .......... .......... .......... .......... 17%  102M 2s\n",
      "  6900K .......... .......... .......... .......... .......... 17%  114M 2s\n",
      "  6950K .......... .......... .......... .......... .......... 18% 1,62M 2s\n",
      "  7000K .......... .......... .......... .......... .......... 18% 35,4M 2s\n",
      "  7050K .......... .......... .......... .......... .......... 18% 57,5M 2s\n",
      "  7100K .......... .......... .......... .......... .......... 18% 31,2M 2s\n",
      "  7150K .......... .......... .......... .......... .......... 18% 48,0M 2s\n",
      "  7200K .......... .......... .......... .......... .......... 18% 62,2M 2s\n",
      "  7250K .......... .......... .......... .......... .......... 18%  148M 2s\n",
      "  7300K .......... .......... .......... .......... .......... 18%  157M 2s\n",
      "  7350K .......... .......... .......... .......... .......... 19%  124M 2s\n",
      "  7400K .......... .......... .......... .......... .......... 19%  157M 2s\n",
      "  7450K .......... .......... .......... .......... .......... 19%  140M 2s\n",
      "  7500K .......... .......... .......... .......... .......... 19%  148M 2s\n",
      "  7550K .......... .......... .......... .......... .......... 19%  159M 2s\n",
      "  7600K .......... .......... .......... .......... .......... 19%  133M 2s\n",
      "  7650K .......... .......... .......... .......... .......... 19%  143M 2s\n",
      "  7700K .......... .......... .......... .......... .......... 19%  102M 2s\n",
      "  7750K .......... .......... .......... .......... .......... 20%  111M 2s\n",
      "  7800K .......... .......... .......... .......... .......... 20%  114M 2s\n",
      "  7850K .......... .......... .......... .......... .......... 20% 93,2M 2s\n",
      "  7900K .......... .......... .......... .......... .......... 20%  116M 2s\n",
      "  7950K .......... .......... .......... .......... .......... 20%  101M 2s\n",
      "  8000K .......... .......... .......... .......... .......... 20%  104M 2s\n",
      "  8050K .......... .......... .......... .......... .......... 20%  104M 2s\n",
      "  8100K .......... .......... .......... .......... .......... 20%  111M 2s\n",
      "  8150K .......... .......... .......... .......... .......... 21%  171M 2s\n",
      "  8200K .......... .......... .......... .......... .......... 21%  146M 2s\n",
      "  8250K .......... .......... .......... .......... .......... 21%  166M 2s\n",
      "  8300K .......... .......... .......... .......... .......... 21% 58,5M 2s\n",
      "  8350K .......... .......... .......... .......... .......... 21%  143M 2s\n",
      "  8400K .......... .......... .......... .......... .......... 21%  248M 2s\n",
      "  8450K .......... .......... .......... .......... .......... 21%  220M 2s\n",
      "  8500K .......... .......... .......... .......... .......... 22%  235M 2s\n",
      "  8550K .......... .......... .......... .......... .......... 22%  262M 2s\n",
      "  8600K .......... .......... .......... .......... .......... 22%  212M 2s\n",
      "  8650K .......... .......... .......... .......... .......... 22%  255M 2s\n",
      "  8700K .......... .......... .......... .......... .......... 22%  200M 2s\n",
      "  8750K .......... .......... .......... .......... .......... 22%  124M 2s\n",
      "  8800K .......... .......... .......... .......... .......... 22%  117M 2s\n",
      "  8850K .......... .......... .......... .......... .......... 22%  103M 2s\n",
      "  8900K .......... .......... .......... .......... .......... 23%  121M 2s\n",
      "  8950K .......... .......... .......... .......... .......... 23%  103M 2s\n",
      "  9000K .......... .......... .......... .......... .......... 23%  117M 2s\n",
      "  9050K .......... .......... .......... .......... .......... 23%  120M 2s\n",
      "  9100K .......... .......... .......... .......... .......... 23%  103M 2s\n",
      "  9150K .......... .......... .......... .......... .......... 23%  120M 2s\n",
      "  9200K .......... .......... .......... .......... .......... 23%  103M 2s\n",
      "  9250K .......... .......... .......... .......... .......... 23%  117M 2s\n",
      "  9300K .......... .......... .......... .......... .......... 24%  121M 2s\n",
      "  9350K .......... .......... .......... .......... .......... 24% 69,8M 2s\n",
      "  9400K .......... .......... .......... .......... .......... 24%  234M 2s\n",
      "  9450K .......... .......... .......... .......... .......... 24%  105M 2s\n",
      "  9500K .......... .......... .......... .......... .......... 24%  121M 2s\n",
      "  9550K .......... .......... .......... .......... .......... 24%  121M 2s\n",
      "  9600K .......... .......... .......... .......... .......... 24%  102M 2s\n",
      "  9650K .......... .......... .......... .......... .......... 24%  117M 2s\n",
      "  9700K .......... .......... .......... .......... .......... 25%  102M 2s\n",
      "  9750K .......... .......... .......... .......... .......... 25%  121M 2s\n",
      "  9800K .......... .......... .......... .......... .......... 25%  117M 2s\n",
      "  9850K .......... .......... .......... .......... .......... 25%  102M 2s\n",
      "  9900K .......... .......... .......... .......... .......... 25%  121M 2s\n",
      "  9950K .......... .......... .......... .......... .......... 25%  102M 2s\n",
      " 10000K .......... .......... .......... .......... .......... 25% 1,74M 2s\n",
      " 10050K .......... .......... .......... .......... .......... 25% 21,4M 2s\n",
      " 10100K .......... .......... .......... .......... .......... 26% 31,3M 2s\n",
      " 10150K .......... .......... .......... .......... .......... 26% 32,4M 2s\n",
      " 10200K .......... .......... .......... .......... .......... 26% 41,5M 2s\n",
      " 10250K .......... .......... .......... .......... .......... 26%  105M 2s\n",
      " 10300K .......... .......... .......... .......... .......... 26% 79,5M 2s\n",
      " 10350K .......... .......... .......... .......... .......... 26% 32,5M 2s\n",
      " 10400K .......... .......... .......... .......... .......... 26% 68,6M 2s\n",
      " 10450K .......... .......... .......... .......... .......... 27% 59,6M 2s\n",
      " 10500K .......... .......... .......... .......... .......... 27% 78,1M 2s\n",
      " 10550K .......... .......... .......... .......... .......... 27% 86,5M 2s\n",
      " 10600K .......... .......... .......... .......... .......... 27% 72,5M 2s\n",
      " 10650K .......... .......... .......... .......... .......... 27%  141M 2s\n",
      " 10700K .......... .......... .......... .......... .......... 27%  121M 2s\n",
      " 10750K .......... .......... .......... .......... .......... 27%  150M 2s\n",
      " 10800K .......... .......... .......... .......... .......... 27%  146M 1s\n",
      " 10850K .......... .......... .......... .......... .......... 28%  124M 1s\n",
      " 10900K .......... .......... .......... .......... .......... 28%  146M 1s\n",
      " 10950K .......... .......... .......... .......... .......... 28%  124M 1s\n",
      " 11000K .......... .......... .......... .......... .......... 28%  179M 1s\n",
      " 11050K .......... .......... .......... .......... .......... 28%  201M 1s\n",
      " 11100K .......... .......... .......... .......... .......... 28%  165M 1s\n",
      " 11150K .......... .......... .......... .......... .......... 28%  199M 1s\n",
      " 11200K .......... .......... .......... .......... .......... 28%  158M 1s\n",
      " 11250K .......... .......... .......... .......... .......... 29%  191M 1s\n",
      " 11300K .......... .......... .......... .......... .......... 29%  181M 1s\n",
      " 11350K .......... .......... .......... .......... .......... 29%  154M 1s\n",
      " 11400K .......... .......... .......... .......... .......... 29%  178M 1s\n",
      " 11450K .......... .......... .......... .......... .......... 29%  118M 1s\n",
      " 11500K .......... .......... .......... .......... .......... 29%  123M 1s\n",
      " 11550K .......... .......... .......... .......... .......... 29%  116M 1s\n",
      " 11600K .......... .......... .......... .......... .......... 29%  103M 1s\n",
      " 11650K .......... .......... .......... .......... .......... 30%  121M 1s\n",
      " 11700K .......... .......... .......... .......... .......... 30%  102M 1s\n",
      " 11750K .......... .......... .......... .......... .......... 30%  123M 1s\n",
      " 11800K .......... .......... .......... .......... .......... 30%  117M 1s\n",
      " 11850K .......... .......... .......... .......... .......... 30%  102M 1s\n",
      " 11900K .......... .......... .......... .......... .......... 30%  121M 1s\n",
      " 11950K .......... .......... .......... .......... .......... 30%  102M 1s\n",
      " 12000K .......... .......... .......... .......... .......... 31%  118M 1s\n",
      " 12050K .......... .......... .......... .......... .......... 31%  120M 1s\n",
      " 12100K .......... .......... .......... .......... .......... 31%  103M 1s\n",
      " 12150K .......... .......... .......... .......... .......... 31%  117M 1s\n",
      " 12200K .......... .......... .......... .......... .......... 31%  102M 1s\n",
      " 12250K .......... .......... .......... .......... .......... 31%  121M 1s\n",
      " 12300K .......... .......... .......... .......... .......... 31%  121M 1s\n",
      " 12350K .......... .......... .......... .......... .......... 31%  102M 1s\n",
      " 12400K .......... .......... .......... .......... .......... 32%  117M 1s\n",
      " 12450K .......... .......... .......... .......... .......... 32%  103M 1s\n",
      " 12500K .......... .......... .......... .......... .......... 32%  121M 1s\n",
      " 12550K .......... .......... .......... .......... .......... 32%  118M 1s\n",
      " 12600K .......... .......... .......... .......... .......... 32%  102M 1s\n",
      " 12650K .......... .......... .......... .......... .......... 32%  120M 1s\n",
      " 12700K .......... .......... .......... .......... .......... 32%  103M 1s\n",
      " 12750K .......... .......... .......... .......... .......... 32%  116M 1s\n",
      " 12800K .......... .......... .......... .......... .......... 33%  121M 1s\n",
      " 12850K .......... .......... .......... .......... .......... 33% 98,2M 1s\n",
      " 12900K .......... .......... .......... .......... .......... 33%  127M 1s\n",
      " 12950K .......... .......... .......... .......... .......... 33%  102M 1s\n",
      " 13000K .......... .......... .......... .......... .......... 33%  117M 1s\n",
      " 13050K .......... .......... .......... .......... .......... 33%  121M 1s\n",
      " 13100K .......... .......... .......... .......... .......... 33% 1,70M 1s\n",
      " 13150K .......... .......... .......... .......... .......... 33% 47,8M 1s\n",
      " 13200K .......... .......... .......... .......... .......... 34% 78,2M 1s\n",
      " 13250K .......... .......... .......... .......... .......... 34% 36,5M 1s\n",
      " 13300K .......... .......... .......... .......... .......... 34% 60,2M 1s\n",
      " 13350K .......... .......... .......... .......... .......... 34%  101M 1s\n",
      " 13400K .......... .......... .......... .......... .......... 34%  120M 1s\n",
      " 13450K .......... .......... .......... .......... .......... 34% 43,9M 1s\n",
      " 13500K .......... .......... .......... .......... .......... 34% 42,5M 1s\n",
      " 13550K .......... .......... .......... .......... .......... 34% 69,8M 1s\n",
      " 13600K .......... .......... .......... .......... .......... 35%  102M 1s\n",
      " 13650K .......... .......... .......... .......... .......... 35%  120M 1s\n",
      " 13700K .......... .......... .......... .......... .......... 35% 94,2M 1s\n",
      " 13750K .......... .......... .......... .......... .......... 35%  119M 1s\n",
      " 13800K .......... .......... .......... .......... .......... 35%  118M 1s\n",
      " 13850K .......... .......... .......... .......... .......... 35%  103M 1s\n",
      " 13900K .......... .......... .......... .......... .......... 35%  122M 1s\n",
      " 13950K .......... .......... .......... .......... .......... 36%  102M 1s\n",
      " 14000K .......... .......... .......... .......... .......... 36%  118M 1s\n",
      " 14050K .......... .......... .......... .......... .......... 36%  123M 1s\n",
      " 14100K .......... .......... .......... .......... .......... 36%  102M 1s\n",
      " 14150K .......... .......... .......... .......... .......... 36% 92,4M 1s\n",
      " 14200K .......... .......... .......... .......... .......... 36%  134M 1s\n",
      " 14250K .......... .......... .......... .......... .......... 36%  120M 1s\n",
      " 14300K .......... .......... .......... .......... .......... 36%  115M 1s\n",
      " 14350K .......... .......... .......... .......... .......... 37%  104M 1s\n",
      " 14400K .......... .......... .......... .......... .......... 37%  119M 1s\n",
      " 14450K .......... .......... .......... .......... .......... 37% 71,2M 1s\n",
      " 14500K .......... .......... .......... .......... .......... 37%  182M 1s\n",
      " 14550K .......... .......... .......... .......... .......... 37%  145M 1s\n",
      " 14600K .......... .......... .......... .......... .......... 37%  102M 1s\n",
      " 14650K .......... .......... .......... .......... .......... 37%  120M 1s\n",
      " 14700K .......... .......... .......... .......... .......... 37%  102M 1s\n",
      " 14750K .......... .......... .......... .......... .......... 38%  117M 1s\n",
      " 14800K .......... .......... .......... .......... .......... 38%  125M 1s\n",
      " 14850K .......... .......... .......... .......... .......... 38%  102M 1s\n",
      " 14900K .......... .......... .......... .......... .......... 38%  118M 1s\n",
      " 14950K .......... .......... .......... .......... .......... 38%  104M 1s\n",
      " 15000K .......... .......... .......... .......... .......... 38%  117M 1s\n",
      " 15050K .......... .......... .......... .......... .......... 38%  121M 1s\n",
      " 15100K .......... .......... .......... .......... .......... 38%  103M 1s\n",
      " 15150K .......... .......... .......... .......... .......... 39%  117M 1s\n",
      " 15200K .......... .......... .......... .......... .......... 39%  102M 1s\n",
      " 15250K .......... .......... .......... .......... .......... 39%  121M 1s\n",
      " 15300K .......... .......... .......... .......... .......... 39%  117M 1s\n",
      " 15350K .......... .......... .......... .......... .......... 39%  102M 1s\n",
      " 15400K .......... .......... .......... .......... .......... 39%  121M 1s\n",
      " 15450K .......... .......... .......... .......... .......... 39%  103M 1s\n",
      " 15500K .......... .......... .......... .......... .......... 40%  121M 1s\n",
      " 15550K .......... .......... .......... .......... .......... 40%  118M 1s\n",
      " 15600K .......... .......... .......... .......... .......... 40%  102M 1s\n",
      " 15650K .......... .......... .......... .......... .......... 40%  118M 1s\n",
      " 15700K .......... .......... .......... .......... .......... 40% 95,3M 1s\n",
      " 15750K .......... .......... .......... .......... .......... 40%  131M 1s\n",
      " 15800K .......... .......... .......... .......... .......... 40%  121M 1s\n",
      " 15850K .......... .......... .......... .......... .......... 40%  103M 1s\n",
      " 15900K .......... .......... .......... .......... .......... 41%  117M 1s\n",
      " 15950K .......... .......... .......... .......... .......... 41%  102M 1s\n",
      " 16000K .......... .......... .......... .......... .......... 41%  121M 1s\n",
      " 16050K .......... .......... .......... .......... .......... 41%  121M 1s\n",
      " 16100K .......... .......... .......... .......... .......... 41%  102M 1s\n",
      " 16150K .......... .......... .......... .......... .......... 41% 1,86M 1s\n",
      " 16200K .......... .......... .......... .......... .......... 41% 19,7M 1s\n",
      " 16250K .......... .......... .......... .......... .......... 41% 68,3M 1s\n",
      " 16300K .......... .......... .......... .......... .......... 42% 70,9M 1s\n",
      " 16350K .......... .......... .......... .......... .......... 42% 78,5M 1s\n",
      " 16400K .......... .......... .......... .......... .......... 42% 61,6M 1s\n",
      " 16450K .......... .......... .......... .......... .......... 42% 57,5M 1s\n",
      " 16500K .......... .......... .......... .......... .......... 42% 59,3M 1s\n",
      " 16550K .......... .......... .......... .......... .......... 42%  120M 1s\n",
      " 16600K .......... .......... .......... .......... .......... 42%  103M 1s\n",
      " 16650K .......... .......... .......... .......... .......... 42% 72,4M 1s\n",
      " 16700K .......... .......... .......... .......... .......... 43% 47,7M 1s\n",
      " 16750K .......... .......... .......... .......... .......... 43% 72,8M 1s\n",
      " 16800K .......... .......... .......... .......... .......... 43% 98,5M 1s\n",
      " 16850K .......... .......... .......... .......... .......... 43% 90,3M 1s\n",
      " 16900K .......... .......... .......... .......... .......... 43%  123M 1s\n",
      " 16950K .......... .......... .......... .......... .......... 43%  172M 1s\n",
      " 17000K .......... .......... .......... .......... .......... 43%  137M 1s\n",
      " 17050K .......... .......... .......... .......... .......... 44%  121M 1s\n",
      " 17100K .......... .......... .......... .......... .......... 44%  103M 1s\n",
      " 17150K .......... .......... .......... .......... .......... 44%  117M 1s\n",
      " 17200K .......... .......... .......... .......... .......... 44%  103M 1s\n",
      " 17250K .......... .......... .......... .......... .......... 44%  121M 1s\n",
      " 17300K .......... .......... .......... .......... .......... 44%  117M 1s\n",
      " 17350K .......... .......... .......... .......... .......... 44%  102M 1s\n",
      " 17400K .......... .......... .......... .......... .......... 44%  121M 1s\n",
      " 17450K .......... .......... .......... .......... .......... 45%  103M 1s\n",
      " 17500K .......... .......... .......... .......... .......... 45%  118M 1s\n",
      " 17550K .......... .......... .......... .......... .......... 45%  120M 1s\n",
      " 17600K .......... .......... .......... .......... .......... 45%  103M 1s\n",
      " 17650K .......... .......... .......... .......... .......... 45%  121M 1s\n",
      " 17700K .......... .......... .......... .......... .......... 45%  102M 1s\n",
      " 17750K .......... .......... .......... .......... .......... 45%  117M 1s\n",
      " 17800K .......... .......... .......... .......... .......... 45%  121M 1s\n",
      " 17850K .......... .......... .......... .......... .......... 46%  103M 1s\n",
      " 17900K .......... .......... .......... .......... .......... 46%  117M 1s\n",
      " 17950K .......... .......... .......... .......... .......... 46%  103M 1s\n",
      " 18000K .......... .......... .......... .......... .......... 46%  121M 1s\n",
      " 18050K .......... .......... .......... .......... .......... 46%  117M 1s\n",
      " 18100K .......... .......... .......... .......... .......... 46%  102M 1s\n",
      " 18150K .......... .......... .......... .......... .......... 46%  121M 1s\n",
      " 18200K .......... .......... .......... .......... .......... 46%  102M 1s\n",
      " 18250K .......... .......... .......... .......... .......... 47%  117M 1s\n",
      " 18300K .......... .......... .......... .......... .......... 47%  121M 1s\n",
      " 18350K .......... .......... .......... .......... .......... 47%  103M 1s\n",
      " 18400K .......... .......... .......... .......... .......... 47%  121M 1s\n",
      " 18450K .......... .......... .......... .......... .......... 47% 94,7M 1s\n",
      " 18500K .......... .......... .......... .......... .......... 47%  129M 1s\n",
      " 18550K .......... .......... .......... .......... .......... 47%  121M 1s\n",
      " 18600K .......... .......... .......... .......... .......... 47%  102M 1s\n",
      " 18650K .......... .......... .......... .......... .......... 48%  117M 1s\n",
      " 18700K .......... .......... .......... .......... .......... 48%  104M 1s\n",
      " 18750K .......... .......... .......... .......... .......... 48%  119M 1s\n",
      " 18800K .......... .......... .......... .......... .......... 48%  122M 1s\n",
      " 18850K .......... .......... .......... .......... .......... 48%  102M 1s\n",
      " 18900K .......... .......... .......... .......... .......... 48%  117M 1s\n",
      " 18950K .......... .......... .......... .......... .......... 48%  103M 1s\n",
      " 19000K .......... .......... .......... .......... .......... 49%  120M 1s\n",
      " 19050K .......... .......... .......... .......... .......... 49%  117M 1s\n",
      " 19100K .......... .......... .......... .......... .......... 49%  103M 1s\n",
      " 19150K .......... .......... .......... .......... .......... 49%  120M 1s\n",
      " 19200K .......... .......... .......... .......... .......... 49% 1,99M 1s\n",
      " 19250K .......... .......... .......... .......... .......... 49% 15,8M 1s\n",
      " 19300K .......... .......... .......... .......... .......... 49% 24,3M 1s\n",
      " 19350K .......... .......... .......... .......... .......... 49% 53,1M 1s\n",
      " 19400K .......... .......... .......... .......... .......... 50% 55,6M 1s\n",
      " 19450K .......... .......... .......... .......... .......... 50%  105M 1s\n",
      " 19500K .......... .......... .......... .......... .......... 50% 37,8M 1s\n",
      " 19550K .......... .......... .......... .......... .......... 50%  119M 1s\n",
      " 19600K .......... .......... .......... .......... .......... 50%  186M 1s\n",
      " 19650K .......... .......... .......... .......... .......... 50% 84,2M 1s\n",
      " 19700K .......... .......... .......... .......... .......... 50%  106M 1s\n",
      " 19750K .......... .......... .......... .......... .......... 50%  121M 1s\n",
      " 19800K .......... .......... .......... .......... .......... 51%  122M 1s\n",
      " 19850K .......... .......... .......... .......... .......... 51%  102M 1s\n",
      " 19900K .......... .......... .......... .......... .......... 51%  118M 1s\n",
      " 19950K .......... .......... .......... .......... .......... 51%  102M 1s\n",
      " 20000K .......... .......... .......... .......... .......... 51%  121M 1s\n",
      " 20050K .......... .......... .......... .......... .......... 51%  117M 1s\n",
      " 20100K .......... .......... .......... .......... .......... 51%  102M 1s\n",
      " 20150K .......... .......... .......... .......... .......... 51%  110M 1s\n",
      " 20200K .......... .......... .......... .......... .......... 52%  112M 1s\n",
      " 20250K .......... .......... .......... .......... .......... 52%  117M 1s\n",
      " 20300K .......... .......... .......... .......... .......... 52%  121M 1s\n",
      " 20350K .......... .......... .......... .......... .......... 52% 96,5M 1s\n",
      " 20400K .......... .......... .......... .......... .......... 52%  100M 1s\n",
      " 20450K .......... .......... .......... .......... .......... 52%  133M 1s\n",
      " 20500K .......... .......... .......... .......... .......... 52%  118M 1s\n",
      " 20550K .......... .......... .......... .......... .......... 53%  122M 1s\n",
      " 20600K .......... .......... .......... .......... .......... 53%  102M 1s\n",
      " 20650K .......... .......... .......... .......... .......... 53%  117M 1s\n",
      " 20700K .......... .......... .......... .......... .......... 53%  102M 1s\n",
      " 20750K .......... .......... .......... .......... .......... 53%  120M 1s\n",
      " 20800K .......... .......... .......... .......... .......... 53%  122M 1s\n",
      " 20850K .......... .......... .......... .......... .......... 53%  102M 1s\n",
      " 20900K .......... .......... .......... .......... .......... 53%  117M 1s\n",
      " 20950K .......... .......... .......... .......... .......... 54%  103M 1s\n",
      " 21000K .......... .......... .......... .......... .......... 54%  121M 1s\n",
      " 21050K .......... .......... .......... .......... .......... 54%  117M 1s\n",
      " 21100K .......... .......... .......... .......... .......... 54%  102M 1s\n",
      " 21150K .......... .......... .......... .......... .......... 54%  121M 1s\n",
      " 21200K .......... .......... .......... .......... .......... 54%  102M 1s\n",
      " 21250K .......... .......... .......... .......... .......... 54%  117M 1s\n",
      " 21300K .......... .......... .......... .......... .......... 54%  121M 1s\n",
      " 21350K .......... .......... .......... .......... .......... 55%  103M 1s\n",
      " 21400K .......... .......... .......... .......... .......... 55%  120M 1s\n",
      " 21450K .......... .......... .......... .......... .......... 55%  102M 1s\n",
      " 21500K .......... .......... .......... .......... .......... 55%  118M 1s\n",
      " 21550K .......... .......... .......... .......... .......... 55%  120M 1s\n",
      " 21600K .......... .......... .......... .......... .......... 55%  102M 1s\n",
      " 21650K .......... .......... .......... .......... .......... 55%  118M 1s\n",
      " 21700K .......... .......... .......... .......... .......... 55%  102M 1s\n",
      " 21750K .......... .......... .......... .......... .......... 56%  121M 1s\n",
      " 21800K .......... .......... .......... .......... .......... 56%  117M 1s\n",
      " 21850K .......... .......... .......... .......... .......... 56%  102M 1s\n",
      " 21900K .......... .......... .......... .......... .......... 56%  121M 1s\n",
      " 21950K .......... .......... .......... .......... .......... 56%  102M 1s\n",
      " 22000K .......... .......... .......... .......... .......... 56%  121M 1s\n",
      " 22050K .......... .......... .......... .......... .......... 56%  118M 1s\n",
      " 22100K .......... .......... .......... .......... .......... 56%  102M 1s\n",
      " 22150K .......... .......... .......... .......... .......... 57%  121M 1s\n",
      " 22200K .......... .......... .......... .......... .......... 57%  102M 1s\n",
      " 22250K .......... .......... .......... .......... .......... 57% 2,07M 1s\n",
      " 22300K .......... .......... .......... .......... .......... 57% 17,7M 1s\n",
      " 22350K .......... .......... .......... .......... .......... 57% 13,4M 1s\n",
      " 22400K .......... .......... .......... .......... .......... 57% 34,1M 1s\n",
      " 22450K .......... .......... .......... .......... .......... 57% 53,7M 1s\n",
      " 22500K .......... .......... .......... .......... .......... 58% 78,2M 1s\n",
      " 22550K .......... .......... .......... .......... .......... 58% 80,7M 1s\n",
      " 22600K .......... .......... .......... .......... .......... 58% 77,3M 1s\n",
      " 22650K .......... .......... .......... .......... .......... 58% 53,1M 1s\n",
      " 22700K .......... .......... .......... .......... .......... 58% 65,5M 1s\n",
      " 22750K .......... .......... .......... .......... .......... 58% 80,6M 1s\n",
      " 22800K .......... .......... .......... .......... .......... 58% 82,8M 1s\n",
      " 22850K .......... .......... .......... .......... .......... 58% 67,8M 1s\n",
      " 22900K .......... .......... .......... .......... .......... 59% 79,3M 1s\n",
      " 22950K .......... .......... .......... .......... .......... 59% 68,5M 1s\n",
      " 23000K .......... .......... .......... .......... .......... 59%  160M 1s\n",
      " 23050K .......... .......... .......... .......... .......... 59%  206M 1s\n",
      " 23100K .......... .......... .......... .......... .......... 59%  173M 1s\n",
      " 23150K .......... .......... .......... .......... .......... 59%  208M 1s\n",
      " 23200K .......... .......... .......... .......... .......... 59%  187M 1s\n",
      " 23250K .......... .......... .......... .......... .......... 59%  207M 1s\n",
      " 23300K .......... .......... .......... .......... .......... 60%  211M 1s\n",
      " 23350K .......... .......... .......... .......... .......... 60%  180M 1s\n",
      " 23400K .......... .......... .......... .......... .......... 60%  182M 1s\n",
      " 23450K .......... .......... .......... .......... .......... 60% 59,2M 1s\n",
      " 23500K .......... .......... .......... .......... .......... 60%  109M 1s\n",
      " 23550K .......... .......... .......... .......... .......... 60% 96,6M 1s\n",
      " 23600K .......... .......... .......... .......... .......... 60%  119M 1s\n",
      " 23650K .......... .......... .......... .......... .......... 60%  129M 1s\n",
      " 23700K .......... .......... .......... .......... .......... 61%  103M 1s\n",
      " 23750K .......... .......... .......... .......... .......... 61%  107M 1s\n",
      " 23800K .......... .......... .......... .......... .......... 61%  120M 1s\n",
      " 23850K .......... .......... .......... .......... .......... 61%  114M 1s\n",
      " 23900K .......... .......... .......... .......... .......... 61%  135M 1s\n",
      " 23950K .......... .......... .......... .......... .......... 61%  119M 1s\n",
      " 24000K .......... .......... .......... .......... .......... 61%  125M 1s\n",
      " 24050K .......... .......... .......... .......... .......... 62%  118M 1s\n",
      " 24100K .......... .......... .......... .......... .......... 62%  103M 0s\n",
      " 24150K .......... .......... .......... .......... .......... 62%  124M 0s\n",
      " 24200K .......... .......... .......... .......... .......... 62%  102M 0s\n",
      " 24250K .......... .......... .......... .......... .......... 62%  116M 0s\n",
      " 24300K .......... .......... .......... .......... .......... 62%  128M 0s\n",
      " 24350K .......... .......... .......... .......... .......... 62%  103M 0s\n",
      " 24400K .......... .......... .......... .......... .......... 62%  117M 0s\n",
      " 24450K .......... .......... .......... .......... .......... 63%  102M 0s\n",
      " 24500K .......... .......... .......... .......... .......... 63%  120M 0s\n",
      " 24550K .......... .......... .......... .......... .......... 63%  121M 0s\n",
      " 24600K .......... .......... .......... .......... .......... 63%  103M 0s\n",
      " 24650K .......... .......... .......... .......... .......... 63%  117M 0s\n",
      " 24700K .......... .......... .......... .......... .......... 63%  102M 0s\n",
      " 24750K .......... .......... .......... .......... .......... 63%  120M 0s\n",
      " 24800K .......... .......... .......... .......... .......... 63%  117M 0s\n",
      " 24850K .......... .......... .......... .......... .......... 64%  103M 0s\n",
      " 24900K .......... .......... .......... .......... .......... 64%  120M 0s\n",
      " 24950K .......... .......... .......... .......... .......... 64%  103M 0s\n",
      " 25000K .......... .......... .......... .......... .......... 64%  118M 0s\n",
      " 25050K .......... .......... .......... .......... .......... 64%  121M 0s\n",
      " 25100K .......... .......... .......... .......... .......... 64%  102M 0s\n",
      " 25150K .......... .......... .......... .......... .......... 64%  121M 0s\n",
      " 25200K .......... .......... .......... .......... .......... 64%  103M 0s\n",
      " 25250K .......... .......... .......... .......... .......... 65%  117M 0s\n",
      " 25300K .......... .......... .......... .......... .......... 65%  121M 0s\n",
      " 25350K .......... .......... .......... .......... .......... 65% 1,91M 0s\n",
      " 25400K .......... .......... .......... .......... .......... 65% 17,2M 0s\n",
      " 25450K .......... .......... .......... .......... .......... 65% 22,5M 0s\n",
      " 25500K .......... .......... .......... .......... .......... 65% 81,8M 0s\n",
      " 25550K .......... .......... .......... .......... .......... 65% 67,4M 0s\n",
      " 25600K .......... .......... .......... .......... .......... 66% 54,0M 0s\n",
      " 25650K .......... .......... .......... .......... .......... 66% 62,2M 0s\n",
      " 25700K .......... .......... .......... .......... .......... 66% 97,7M 0s\n",
      " 25750K .......... .......... .......... .......... .......... 66%  205M 0s\n",
      " 25800K .......... .......... .......... .......... .......... 66%  188M 0s\n",
      " 25850K .......... .......... .......... .......... .......... 66% 52,8M 0s\n",
      " 25900K .......... .......... .......... .......... .......... 66% 57,1M 0s\n",
      " 25950K .......... .......... .......... .......... .......... 66%  104M 0s\n",
      " 26000K .......... .......... .......... .......... .......... 67% 56,3M 0s\n",
      " 26050K .......... .......... .......... .......... .......... 67%  117M 0s\n",
      " 26100K .......... .......... .......... .......... .......... 67%  103M 0s\n",
      " 26150K .......... .......... .......... .......... .......... 67%  115M 0s\n",
      " 26200K .......... .......... .......... .......... .......... 67%  104M 0s\n",
      " 26250K .......... .......... .......... .......... .......... 67%  118M 0s\n",
      " 26300K .......... .......... .......... .......... .......... 67% 50,5M 0s\n",
      " 26350K .......... .......... .......... .......... .......... 67% 64,8M 0s\n",
      " 26400K .......... .......... .......... .......... .......... 68%  116M 0s\n",
      " 26450K .......... .......... .......... .......... .......... 68%  120M 0s\n",
      " 26500K .......... .......... .......... .......... .......... 68%  143M 0s\n",
      " 26550K .......... .......... .......... .......... .......... 68%  143M 0s\n",
      " 26600K .......... .......... .......... .......... .......... 68%  124M 0s\n",
      " 26650K .......... .......... .......... .......... .......... 68%  145M 0s\n",
      " 26700K .......... .......... .......... .......... .......... 68% 75,4M 0s\n",
      " 26750K .......... .......... .......... .......... .......... 68%  209M 0s\n",
      " 26800K .......... .......... .......... .......... .......... 69%  211M 0s\n",
      " 26850K .......... .......... .......... .......... .......... 69%  180M 0s\n",
      " 26900K .......... .......... .......... .......... .......... 69%  170M 0s\n",
      " 26950K .......... .......... .......... .......... .......... 69%  102M 0s\n",
      " 27000K .......... .......... .......... .......... .......... 69%  117M 0s\n",
      " 27050K .......... .......... .......... .......... .......... 69%  121M 0s\n",
      " 27100K .......... .......... .......... .......... .......... 69%  102M 0s\n",
      " 27150K .......... .......... .......... .......... .......... 69%  118M 0s\n",
      " 27200K .......... .......... .......... .......... .......... 70%  102M 0s\n",
      " 27250K .......... .......... .......... .......... .......... 70%  121M 0s\n",
      " 27300K .......... .......... .......... .......... .......... 70%  122M 0s\n",
      " 27350K .......... .......... .......... .......... .......... 70%  103M 0s\n",
      " 27400K .......... .......... .......... .......... .......... 70%  117M 0s\n",
      " 27450K .......... .......... .......... .......... .......... 70%  102M 0s\n",
      " 27500K .......... .......... .......... .......... .......... 70%  121M 0s\n",
      " 27550K .......... .......... .......... .......... .......... 71%  117M 0s\n",
      " 27600K .......... .......... .......... .......... .......... 71%  103M 0s\n",
      " 27650K .......... .......... .......... .......... .......... 71%  120M 0s\n",
      " 27700K .......... .......... .......... .......... .......... 71%  103M 0s\n",
      " 27750K .......... .......... .......... .......... .......... 71%  117M 0s\n",
      " 27800K .......... .......... .......... .......... .......... 71% 48,8M 0s\n",
      " 27850K .......... .......... .......... .......... .......... 71%  391M 0s\n",
      " 27900K .......... .......... .......... .......... .......... 71%  275M 0s\n",
      " 27950K .......... .......... .......... .......... .......... 72%  106M 0s\n",
      " 28000K .......... .......... .......... .......... .......... 72% 46,4M 0s\n",
      " 28050K .......... .......... .......... .......... .......... 72%  221M 0s\n",
      " 28100K .......... .......... .......... .......... .......... 72%  198M 0s\n",
      " 28150K .......... .......... .......... .......... .......... 72%  228M 0s\n",
      " 28200K .......... .......... .......... .......... .......... 72%  107M 0s\n",
      " 28250K .......... .......... .......... .......... .......... 72%  123M 0s\n",
      " 28300K .......... .......... .......... .......... .......... 72%  120M 0s\n",
      " 28350K .......... .......... .......... .......... .......... 73%  103M 0s\n",
      " 28400K .......... .......... .......... .......... .......... 73% 2,06M 0s\n",
      " 28450K .......... .......... .......... .......... .......... 73% 18,0M 0s\n",
      " 28500K .......... .......... .......... .......... .......... 73% 14,1M 0s\n",
      " 28550K .......... .......... .......... .......... .......... 73% 56,1M 0s\n",
      " 28600K .......... .......... .......... .......... .......... 73% 72,7M 0s\n",
      " 28650K .......... .......... .......... .......... .......... 73% 84,9M 0s\n",
      " 28700K .......... .......... .......... .......... .......... 73% 27,0M 0s\n",
      " 28750K .......... .......... .......... .......... .......... 74% 35,0M 0s\n",
      " 28800K .......... .......... .......... .......... .......... 74% 81,5M 0s\n",
      " 28850K .......... .......... .......... .......... .......... 74% 55,3M 0s\n",
      " 28900K .......... .......... .......... .......... .......... 74% 71,6M 0s\n",
      " 28950K .......... .......... .......... .......... .......... 74%  175M 0s\n",
      " 29000K .......... .......... .......... .......... .......... 74%  210M 0s\n",
      " 29050K .......... .......... .......... .......... .......... 74%  216M 0s\n",
      " 29100K .......... .......... .......... .......... .......... 75%  186M 0s\n",
      " 29150K .......... .......... .......... .......... .......... 75%  231M 0s\n",
      " 29200K .......... .......... .......... .......... .......... 75%  190M 0s\n",
      " 29250K .......... .......... .......... .......... .......... 75%  224M 0s\n",
      " 29300K .......... .......... .......... .......... .......... 75%  238M 0s\n",
      " 29350K .......... .......... .......... .......... .......... 75%  188M 0s\n",
      " 29400K .......... .......... .......... .......... .......... 75%  224M 0s\n",
      " 29450K .......... .......... .......... .......... .......... 75%  191M 0s\n",
      " 29500K .......... .......... .......... .......... .......... 76% 57,2M 0s\n",
      " 29550K .......... .......... .......... .......... .......... 76%  120M 0s\n",
      " 29600K .......... .......... .......... .......... .......... 76% 85,3M 0s\n",
      " 29650K .......... .......... .......... .......... .......... 76%  121M 0s\n",
      " 29700K .......... .......... .......... .......... .......... 76%  101M 0s\n",
      " 29750K .......... .......... .......... .......... .......... 76%  117M 0s\n",
      " 29800K .......... .......... .......... .......... .......... 76%  121M 0s\n",
      " 29850K .......... .......... .......... .......... .......... 76%  103M 0s\n",
      " 29900K .......... .......... .......... .......... .......... 77%  117M 0s\n",
      " 29950K .......... .......... .......... .......... .......... 77%  105M 0s\n",
      " 30000K .......... .......... .......... .......... .......... 77%  118M 0s\n",
      " 30050K .......... .......... .......... .......... .......... 77% 80,6M 0s\n",
      " 30100K .......... .......... .......... .......... .......... 77% 92,9M 0s\n",
      " 30150K .......... .......... .......... .......... .......... 77%  152M 0s\n",
      " 30200K .......... .......... .......... .......... .......... 77%  121M 0s\n",
      " 30250K .......... .......... .......... .......... .......... 77% 67,5M 0s\n",
      " 30300K .......... .......... .......... .......... .......... 78%  111M 0s\n",
      " 30350K .......... .......... .......... .......... .......... 78% 93,2M 0s\n",
      " 30400K .......... .......... .......... .......... .......... 78%  113M 0s\n",
      " 30450K .......... .......... .......... .......... .......... 78%  101M 0s\n",
      " 30500K .......... .......... .......... .......... .......... 78%  213M 0s\n",
      " 30550K .......... .......... .......... .......... .......... 78%  209M 0s\n",
      " 30600K .......... .......... .......... .......... .......... 78%  148M 0s\n",
      " 30650K .......... .......... .......... .......... .......... 78%  121M 0s\n",
      " 30700K .......... .......... .......... .......... .......... 79% 97,5M 0s\n",
      " 30750K .......... .......... .......... .......... .......... 79%  127M 0s\n",
      " 30800K .......... .......... .......... .......... .......... 79%  122M 0s\n",
      " 30850K .......... .......... .......... .......... .......... 79%  102M 0s\n",
      " 30900K .......... .......... .......... .......... .......... 79%  117M 0s\n",
      " 30950K .......... .......... .......... .......... .......... 79%  103M 0s\n",
      " 31000K .......... .......... .......... .......... .......... 79%  120M 0s\n",
      " 31050K .......... .......... .......... .......... .......... 80%  122M 0s\n",
      " 31100K .......... .......... .......... .......... .......... 80%  102M 0s\n",
      " 31150K .......... .......... .......... .......... .......... 80%  118M 0s\n",
      " 31200K .......... .......... .......... .......... .......... 80%  102M 0s\n",
      " 31250K .......... .......... .......... .......... .......... 80%  121M 0s\n",
      " 31300K .......... .......... .......... .......... .......... 80%  117M 0s\n",
      " 31350K .......... .......... .......... .......... .......... 80%  102M 0s\n",
      " 31400K .......... .......... .......... .......... .......... 80% 3,39M 0s\n",
      " 31450K .......... .......... .......... .......... .......... 81% 5,49M 0s\n",
      " 31500K .......... .......... .......... .......... .......... 81% 24,9M 0s\n",
      " 31550K .......... .......... .......... .......... .......... 81% 9,94M 0s\n",
      " 31600K .......... .......... .......... .......... .......... 81% 41,8M 0s\n",
      " 31650K .......... .......... .......... .......... .......... 81% 68,0M 0s\n",
      " 31700K .......... .......... .......... .......... .......... 81% 65,3M 0s\n",
      " 31750K .......... .......... .......... .......... .......... 81% 73,9M 0s\n",
      " 31800K .......... .......... .......... .......... .......... 81% 97,3M 0s\n",
      " 31850K .......... .......... .......... .......... .......... 82% 85,5M 0s\n",
      " 31900K .......... .......... .......... .......... .......... 82% 33,3M 0s\n",
      " 31950K .......... .......... .......... .......... .......... 82%  100M 0s\n",
      " 32000K .......... .......... .......... .......... .......... 82% 71,0M 0s\n",
      " 32050K .......... .......... .......... .......... .......... 82% 55,1M 0s\n",
      " 32100K .......... .......... .......... .......... .......... 82%  100M 0s\n",
      " 32150K .......... .......... .......... .......... .......... 82%  123M 0s\n",
      " 32200K .......... .......... .......... .......... .......... 82% 99,8M 0s\n",
      " 32250K .......... .......... .......... .......... .......... 83%  124M 0s\n",
      " 32300K .......... .......... .......... .......... .......... 83%  114M 0s\n",
      " 32350K .......... .......... .......... .......... .......... 83%  104M 0s\n",
      " 32400K .......... .......... .......... .......... .......... 83%  122M 0s\n",
      " 32450K .......... .......... .......... .......... .......... 83%  102M 0s\n",
      " 32500K .......... .......... .......... .......... .......... 83%  119M 0s\n",
      " 32550K .......... .......... .......... .......... .......... 83%  120M 0s\n",
      " 32600K .......... .......... .......... .......... .......... 84%  103M 0s\n",
      " 32650K .......... .......... .......... .......... .......... 84%  119M 0s\n",
      " 32700K .......... .......... .......... .......... .......... 84%  104M 0s\n",
      " 32750K .......... .......... .......... .......... .......... 84%  107M 0s\n",
      " 32800K .......... .......... .......... .......... .......... 84%  119M 0s\n",
      " 32850K .......... .......... .......... .......... .......... 84% 80,6M 0s\n",
      " 32900K .......... .......... .......... .......... .......... 84%  197M 0s\n",
      " 32950K .......... .......... .......... .......... .......... 84%  103M 0s\n",
      " 33000K .......... .......... .......... .......... .......... 85%  122M 0s\n",
      " 33050K .......... .......... .......... .......... .......... 85%  118M 0s\n",
      " 33100K .......... .......... .......... .......... .......... 85%  102M 0s\n",
      " 33150K .......... .......... .......... .......... .......... 85%  122M 0s\n",
      " 33200K .......... .......... .......... .......... .......... 85%  102M 0s\n",
      " 33250K .......... .......... .......... .......... .......... 85%  121M 0s\n",
      " 33300K .......... .......... .......... .......... .......... 85%  117M 0s\n",
      " 33350K .......... .......... .......... .......... .......... 85%  103M 0s\n",
      " 33400K .......... .......... .......... .......... .......... 86%  121M 0s\n",
      " 33450K .......... .......... .......... .......... .......... 86%  102M 0s\n",
      " 33500K .......... .......... .......... .......... .......... 86%  117M 0s\n",
      " 33550K .......... .......... .......... .......... .......... 86%  121M 0s\n",
      " 33600K .......... .......... .......... .......... .......... 86%  102M 0s\n",
      " 33650K .......... .......... .......... .......... .......... 86%  117M 0s\n",
      " 33700K .......... .......... .......... .......... .......... 86%  105M 0s\n",
      " 33750K .......... .......... .......... .......... .......... 86%  118M 0s\n",
      " 33800K .......... .......... .......... .......... .......... 87%  121M 0s\n",
      " 33850K .......... .......... .......... .......... .......... 87%  102M 0s\n",
      " 33900K .......... .......... .......... .......... .......... 87%  117M 0s\n",
      " 33950K .......... .......... .......... .......... .......... 87%  103M 0s\n",
      " 34000K .......... .......... .......... .......... .......... 87%  120M 0s\n",
      " 34050K .......... .......... .......... .......... .......... 87%  115M 0s\n",
      " 34100K .......... .......... .......... .......... .......... 87% 89,6M 0s\n",
      " 34150K .......... .......... .......... .......... .......... 88%  147M 0s\n",
      " 34200K .......... .......... .......... .......... .......... 88%  102M 0s\n",
      " 34250K .......... .......... .......... .......... .......... 88%  121M 0s\n",
      " 34300K .......... .......... .......... .......... .......... 88%  117M 0s\n",
      " 34350K .......... .......... .......... .......... .......... 88%  102M 0s\n",
      " 34400K .......... .......... .......... .......... .......... 88%  121M 0s\n",
      " 34450K .......... .......... .......... .......... .......... 88% 3,94M 0s\n",
      " 34500K .......... .......... .......... .......... .......... 88% 6,94M 0s\n",
      " 34550K .......... .......... .......... .......... .......... 89% 11,2M 0s\n",
      " 34600K .......... .......... .......... .......... .......... 89% 11,5M 0s\n",
      " 34650K .......... .......... .......... .......... .......... 89% 26,9M 0s\n",
      " 34700K .......... .......... .......... .......... .......... 89% 40,7M 0s\n",
      " 34750K .......... .......... .......... .......... .......... 89% 55,8M 0s\n",
      " 34800K .......... .......... .......... .......... .......... 89% 57,0M 0s\n",
      " 34850K .......... .......... .......... .......... .......... 89% 97,8M 0s\n",
      " 34900K .......... .......... .......... .......... .......... 89% 64,3M 0s\n",
      " 34950K .......... .......... .......... .......... .......... 90%  109M 0s\n",
      " 35000K .......... .......... .......... .......... .......... 90%  102M 0s\n",
      " 35050K .......... .......... .......... .......... .......... 90% 65,0M 0s\n",
      " 35100K .......... .......... .......... .......... .......... 90% 94,8M 0s\n",
      " 35150K .......... .......... .......... .......... .......... 90% 57,7M 0s\n",
      " 35200K .......... .......... .......... .......... .......... 90% 96,6M 0s\n",
      " 35250K .......... .......... .......... .......... .......... 90%  113M 0s\n",
      " 35300K .......... .......... .......... .......... .......... 90%  121M 0s\n",
      " 35350K .......... .......... .......... .......... .......... 91% 73,6M 0s\n",
      " 35400K .......... .......... .......... .......... .......... 91%  112M 0s\n",
      " 35450K .......... .......... .......... .......... .......... 91% 94,7M 0s\n",
      " 35500K .......... .......... .......... .......... .......... 91%  123M 0s\n",
      " 35550K .......... .......... .......... .......... .......... 91%  121M 0s\n",
      " 35600K .......... .......... .......... .......... .......... 91%  108M 0s\n",
      " 35650K .......... .......... .......... .......... .......... 91%  126M 0s\n",
      " 35700K .......... .......... .......... .......... .......... 91%  110M 0s\n",
      " 35750K .......... .......... .......... .......... .......... 92%  119M 0s\n",
      " 35800K .......... .......... .......... .......... .......... 92%  123M 0s\n",
      " 35850K .......... .......... .......... .......... .......... 92%  103M 0s\n",
      " 35900K .......... .......... .......... .......... .......... 92%  118M 0s\n",
      " 35950K .......... .......... .......... .......... .......... 92%  101M 0s\n",
      " 36000K .......... .......... .......... .......... .......... 92%  124M 0s\n",
      " 36050K .......... .......... .......... .......... .......... 92%  117M 0s\n",
      " 36100K .......... .......... .......... .......... .......... 93%  102M 0s\n",
      " 36150K .......... .......... .......... .......... .......... 93%  121M 0s\n",
      " 36200K .......... .......... .......... .......... .......... 93%  103M 0s\n",
      " 36250K .......... .......... .......... .......... .......... 93%  117M 0s\n",
      " 36300K .......... .......... .......... .......... .......... 93%  121M 0s\n",
      " 36350K .......... .......... .......... .......... .......... 93%  103M 0s\n",
      " 36400K .......... .......... .......... .......... .......... 93%  121M 0s\n",
      " 36450K .......... .......... .......... .......... .......... 93%  103M 0s\n",
      " 36500K .......... .......... .......... .......... .......... 94%  117M 0s\n",
      " 36550K .......... .......... .......... .......... .......... 94%  121M 0s\n",
      " 36600K .......... .......... .......... .......... .......... 94%  102M 0s\n",
      " 36650K .......... .......... .......... .......... .......... 94%  118M 0s\n",
      " 36700K .......... .......... .......... .......... .......... 94%  102M 0s\n",
      " 36750K .......... .......... .......... .......... .......... 94%  121M 0s\n",
      " 36800K .......... .......... .......... .......... .......... 94%  117M 0s\n",
      " 36850K .......... .......... .......... .......... .......... 94%  102M 0s\n",
      " 36900K .......... .......... .......... .......... .......... 95%  121M 0s\n",
      " 36950K .......... .......... .......... .......... .......... 95%  102M 0s\n",
      " 37000K .......... .......... .......... .......... .......... 95%  121M 0s\n",
      " 37050K .......... .......... .......... .......... .......... 95%  117M 0s\n",
      " 37100K .......... .......... .......... .......... .......... 95%  102M 0s\n",
      " 37150K .......... .......... .......... .......... .......... 95%  121M 0s\n",
      " 37200K .......... .......... .......... .......... .......... 95%  102M 0s\n",
      " 37250K .......... .......... .......... .......... .......... 95%  117M 0s\n",
      " 37300K .......... .......... .......... .......... .......... 96%  120M 0s\n",
      " 37350K .......... .......... .......... .......... .......... 96%  103M 0s\n",
      " 37400K .......... .......... .......... .......... .......... 96%  117M 0s\n",
      " 37450K .......... .......... .......... .......... .......... 96% 4,26M 0s\n",
      " 37500K .......... .......... .......... .......... .......... 96% 28,6M 0s\n",
      " 37550K .......... .......... .......... .......... .......... 96% 14,2M 0s\n",
      " 37600K .......... .......... .......... .......... .......... 96% 6,26M 0s\n",
      " 37650K .......... .......... .......... .......... .......... 97% 23,0M 0s\n",
      " 37700K .......... .......... .......... .......... .......... 97% 12,3M 0s\n",
      " 37750K .......... .......... .......... .......... .......... 97% 55,7M 0s\n",
      " 37800K .......... .......... .......... .......... .......... 97%  117M 0s\n",
      " 37850K .......... .......... .......... .......... .......... 97% 57,6M 0s\n",
      " 37900K .......... .......... .......... .......... .......... 97% 57,3M 0s\n",
      " 37950K .......... .......... .......... .......... .......... 97% 59,9M 0s\n",
      " 38000K .......... .......... .......... .......... .......... 97%  126M 0s\n",
      " 38050K .......... .......... .......... .......... .......... 98%  116M 0s\n",
      " 38100K .......... .......... .......... .......... .......... 98%  103M 0s\n",
      " 38150K .......... .......... .......... .......... .......... 98% 59,4M 0s\n",
      " 38200K .......... .......... .......... .......... .......... 98% 78,0M 0s\n",
      " 38250K .......... .......... .......... .......... .......... 98%  118M 0s\n",
      " 38300K .......... .......... .......... .......... .......... 98%  122M 0s\n",
      " 38350K .......... .......... .......... .......... .......... 98%  102M 0s\n",
      " 38400K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 38450K .......... .......... .......... .......... .......... 99%  102M 0s\n",
      " 38500K .......... .......... .......... .......... .......... 99%  106M 0s\n",
      " 38550K .......... .......... .......... .......... .......... 99% 68,2M 0s\n",
      " 38600K .......... .......... .......... .......... .......... 99% 57,4M 0s\n",
      " 38650K .......... .......... .......... .......... .......... 99%  117M 0s\n",
      " 38700K .......... .......... .......... .......... .......... 99%  101M 0s\n",
      " 38750K .......... .......... .......... .......... .......... 99%  122M 0s\n",
      " 38800K .......... .......... .......... .......... .......... 99%  118M 0s\n",
      " 38850K .......... ..                                         100%  123M=1,1s\n",
      "\n",
      "2023-08-17 14:29:36 (34,4 MB/s) - ‘data/reference/genome.fa.gz’ guardado [39795542/39795542]\n",
      "\n",
      "[Thu Aug 17 14:29:37 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T142934.232128.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:29:38 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR696683_chr12_1.fastq.gz, data/raw/ERR696683_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR696683_chr12_1_fastqc.html, results/fastqc_result/ERR696683_chr12_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR696683_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 14:29:54 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:29:54 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR696683_chr12_1.fastq.gz, data/raw/ERR696683_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR696683_chr12_1_fastp.fastq.gz, data/processed/ERR696683_chr12_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR696683_chr12_1_fastp.fastq.gz, data/processed/ERR696683_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1814551\n",
      "total bases: 137166702\n",
      "Q20 bases: 136196362(99.2926%)\n",
      "Q30 bases: 132450329(96.5616%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1814551\n",
      "total bases: 136812703\n",
      "Q20 bases: 135133064(98.7723%)\n",
      "Q30 bases: 129660948(94.7726%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1124306\n",
      "total bases: 85390403\n",
      "Q20 bases: 85056739(99.6092%)\n",
      "Q30 bases: 83576597(97.8759%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1124306\n",
      "total bases: 85356933\n",
      "Q20 bases: 84768167(99.3102%)\n",
      "Q30 bases: 82580401(96.7472%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2248612\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1380490\n",
      "reads with adapter trimmed: 812\n",
      "bases trimmed due to adapters: 18871\n",
      "\n",
      "Duplication rate: 5.76741%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR696683_chr12_1.fastq.gz -I data/raw/ERR696683_chr12_2.fastq.gz -o data/processed/ERR696683_chr12_1_fastp.fastq.gz -O data/processed/ERR696683_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Thu Aug 17 14:30:00 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:30:00 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR696683_chr12_1_fastp.fastq.gz, data/processed/ERR696683_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR696683_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR696683_chr12_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR696683_chr12_1_fastp.fastq.gz, data/processed/ERR696683_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 14:30:11 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T142937.258497.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:30:13 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753368_chr12_1.fastq.gz, data/raw/ERR753368_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753368_chr12_1_fastp.fastq.gz, data/processed/ERR753368_chr12_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753368_chr12_2_fastp.fastq.gz, data/processed/ERR753368_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1897940\n",
      "total bases: 144109638\n",
      "Q20 bases: 143542594(99.6065%)\n",
      "Q30 bases: 139301752(96.6637%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1897940\n",
      "total bases: 143961401\n",
      "Q20 bases: 142692339(99.1185%)\n",
      "Q30 bases: 136129314(94.5596%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1632596\n",
      "total bases: 124070769\n",
      "Q20 bases: 123855197(99.8263%)\n",
      "Q30 bases: 121763665(98.1405%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1632596\n",
      "total bases: 124060324\n",
      "Q20 bases: 123450972(99.5088%)\n",
      "Q30 bases: 119750755(96.5262%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3265192\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 530688\n",
      "reads with adapter trimmed: 1148\n",
      "bases trimmed due to adapters: 26680\n",
      "\n",
      "Duplication rate: 2.03273%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753368_chr12_1.fastq.gz -I data/raw/ERR753368_chr12_2.fastq.gz -o data/processed/ERR753368_chr12_1_fastp.fastq.gz -O data/processed/ERR753368_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Thu Aug 17 14:30:20 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:30:20 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753368_chr12_1.fastq.gz, data/raw/ERR753368_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753368_chr12_1_fastqc.html, results/fastqc_result/ERR753368_chr12_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753368_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753368_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 14:30:34 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:30:34 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753368_chr12_1_fastp.fastq.gz, data/processed/ERR753368_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753368_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753368_chr12_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753368_chr12_2_fastp.fastq.gz, data/processed/ERR753368_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 14:30:48 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T143011.980153.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:30:49 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753369_chr12_1.fastq.gz, data/raw/ERR753369_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753369_chr12_1_fastqc.html, results/fastqc_result/ERR753369_chr12_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753369_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753369_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 14:31:05 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:31:05 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753369_chr12_1.fastq.gz, data/raw/ERR753369_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753369_chr12_1_fastp.fastq.gz, data/processed/ERR753369_chr12_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753369_chr12_2_fastp.fastq.gz, data/processed/ERR753369_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2053990\n",
      "total bases: 155951416\n",
      "Q20 bases: 155309500(99.5884%)\n",
      "Q30 bases: 150604540(96.5714%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2053990\n",
      "total bases: 155798651\n",
      "Q20 bases: 154421725(99.1162%)\n",
      "Q30 bases: 147287629(94.5372%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1766738\n",
      "total bases: 134264590\n",
      "Q20 bases: 134014650(99.8138%)\n",
      "Q30 bases: 131659977(98.0601%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1766738\n",
      "total bases: 134253748\n",
      "Q20 bases: 133591459(99.5067%)\n",
      "Q30 bases: 129560868(96.5045%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3533476\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 574504\n",
      "reads with adapter trimmed: 1156\n",
      "bases trimmed due to adapters: 27060\n",
      "\n",
      "Duplication rate: 2.36787%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753369_chr12_1.fastq.gz -I data/raw/ERR753369_chr12_2.fastq.gz -o data/processed/ERR753369_chr12_1_fastp.fastq.gz -O data/processed/ERR753369_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Thu Aug 17 14:31:12 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:31:12 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753369_chr12_1_fastp.fastq.gz, data/processed/ERR753369_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753369_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753369_chr12_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753369_chr12_2_fastp.fastq.gz, data/processed/ERR753369_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 14:31:25 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T143048.420336.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:31:27 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753370_chr12_1.fastq.gz, data/raw/ERR753370_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753370_chr12_1_fastp.fastq.gz, data/processed/ERR753370_chr12_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753370_chr12_1_fastp.fastq.gz, data/processed/ERR753370_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2549417\n",
      "total bases: 256637892\n",
      "Q20 bases: 254886215(99.3175%)\n",
      "Q30 bases: 247502750(96.4405%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2549417\n",
      "total bases: 256574888\n",
      "Q20 bases: 254137882(99.0502%)\n",
      "Q30 bases: 244519685(95.3015%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 2548864\n",
      "total bases: 255580768\n",
      "Q20 bases: 253929976(99.3541%)\n",
      "Q30 bases: 246895663(96.6018%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 2548864\n",
      "total bases: 254756297\n",
      "Q20 bases: 252541907(99.1308%)\n",
      "Q30 bases: 243527730(95.5924%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 5097728\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1106\n",
      "reads with adapter trimmed: 13432\n",
      "bases trimmed due to adapters: 104382\n",
      "\n",
      "Duplication rate: 4.35053%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 160\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753370_chr12_1.fastq.gz -I data/raw/ERR753370_chr12_2.fastq.gz -o data/processed/ERR753370_chr12_1_fastp.fastq.gz -O data/processed/ERR753370_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 12 seconds\n",
      "[Thu Aug 17 14:31:39 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:31:39 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753370_chr12_1.fastq.gz, data/raw/ERR753370_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753370_chr12_1_fastqc.html, results/fastqc_result/ERR753370_chr12_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753370_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753370_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 14:32:01 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:32:01 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753370_chr12_1_fastp.fastq.gz, data/processed/ERR753370_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753370_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753370_chr12_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753370_chr12_1_fastp.fastq.gz, data/processed/ERR753370_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 14:32:24 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T143125.854648.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:32:26 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753371_chr12_1.fastq.gz, data/raw/ERR753371_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753371_chr12_1_fastqc.html, results/fastqc_result/ERR753371_chr12_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753371_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753371_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 14:32:39 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:32:39 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753371_chr12_1.fastq.gz, data/raw/ERR753371_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753371_chr12_1_fastp.fastq.gz, data/processed/ERR753371_chr12_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753371_chr12_1_fastp.fastq.gz, data/processed/ERR753371_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1690709\n",
      "total bases: 127390591\n",
      "Q20 bases: 126242409(99.0987%)\n",
      "Q30 bases: 121643046(95.4882%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1690709\n",
      "total bases: 127070345\n",
      "Q20 bases: 125554387(98.807%)\n",
      "Q30 bases: 121137819(95.3313%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 872488\n",
      "total bases: 66228440\n",
      "Q20 bases: 65905625(99.5126%)\n",
      "Q30 bases: 64262329(97.0313%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 872488\n",
      "total bases: 66192098\n",
      "Q20 bases: 65729465(99.3011%)\n",
      "Q30 bases: 64277244(97.1071%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1744976\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1636442\n",
      "reads with adapter trimmed: 710\n",
      "bases trimmed due to adapters: 15017\n",
      "\n",
      "Duplication rate: 22.0286%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 116\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753371_chr12_1.fastq.gz -I data/raw/ERR753371_chr12_2.fastq.gz -o data/processed/ERR753371_chr12_1_fastp.fastq.gz -O data/processed/ERR753371_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Thu Aug 17 14:32:44 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:32:44 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753371_chr12_1_fastp.fastq.gz, data/processed/ERR753371_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753371_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753371_chr12_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753371_chr12_1_fastp.fastq.gz, data/processed/ERR753371_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 14:32:53 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T143224.973240.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:32:54 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753372_chr12_1.fastq.gz, data/raw/ERR753372_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753372_chr12_1_fastqc.html, results/fastqc_result/ERR753372_chr12_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753372_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753372_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 14:33:12 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:33:12 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753372_chr12_1.fastq.gz, data/raw/ERR753372_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753372_chr12_1_fastp.fastq.gz, data/processed/ERR753372_chr12_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753372_chr12_1_fastp.fastq.gz, data/processed/ERR753372_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2028383\n",
      "total bases: 154006268\n",
      "Q20 bases: 153393425(99.6021%)\n",
      "Q30 bases: 148774276(96.6027%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2028383\n",
      "total bases: 153852859\n",
      "Q20 bases: 152494427(99.1171%)\n",
      "Q30 bases: 145498335(94.5698%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1738874\n",
      "total bases: 132147137\n",
      "Q20 bases: 131915460(99.8247%)\n",
      "Q30 bases: 129641227(98.1037%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1738874\n",
      "total bases: 132136009\n",
      "Q20 bases: 131490557(99.5115%)\n",
      "Q30 bases: 127584864(96.5557%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3477748\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 579018\n",
      "reads with adapter trimmed: 1250\n",
      "bases trimmed due to adapters: 26990\n",
      "\n",
      "Duplication rate: 2.003%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753372_chr12_1.fastq.gz -I data/raw/ERR753372_chr12_2.fastq.gz -o data/processed/ERR753372_chr12_1_fastp.fastq.gz -O data/processed/ERR753372_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Thu Aug 17 14:33:19 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:33:19 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753372_chr12_1_fastp.fastq.gz, data/processed/ERR753372_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753372_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753372_chr12_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753372_chr12_1_fastp.fastq.gz, data/processed/ERR753372_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 14:33:32 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T143253.431753.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:33:33 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753373_chr12_1.fastq.gz, data/raw/ERR753373_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753373_chr12_1_fastqc.html, results/fastqc_result/ERR753373_chr12_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753373_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753373_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 14:33:46 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:33:46 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753373_chr12_1.fastq.gz, data/raw/ERR753373_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753373_chr12_1_fastp.fastq.gz, data/processed/ERR753373_chr12_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753373_chr12_1_fastp.fastq.gz, data/processed/ERR753373_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1413533\n",
      "total bases: 135329456\n",
      "Q20 bases: 134237138(99.1928%)\n",
      "Q30 bases: 125040132(92.3968%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1413533\n",
      "total bases: 132444072\n",
      "Q20 bases: 130991936(98.9036%)\n",
      "Q30 bases: 122419166(92.4308%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1378941\n",
      "total bases: 129926291\n",
      "Q20 bases: 129164081(99.4134%)\n",
      "Q30 bases: 121734716(93.6952%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1378941\n",
      "total bases: 125531307\n",
      "Q20 bases: 124923753(99.516%)\n",
      "Q30 bases: 118402277(94.3209%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2757882\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 69184\n",
      "reads with adapter trimmed: 1192\n",
      "bases trimmed due to adapters: 41078\n",
      "\n",
      "Duplication rate: 1.84895%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 153\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753373_chr12_1.fastq.gz -I data/raw/ERR753373_chr12_2.fastq.gz -o data/processed/ERR753373_chr12_1_fastp.fastq.gz -O data/processed/ERR753373_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Thu Aug 17 14:33:53 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:33:53 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753373_chr12_1_fastp.fastq.gz, data/processed/ERR753373_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753373_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753373_chr12_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753373_chr12_1_fastp.fastq.gz, data/processed/ERR753373_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 14:34:07 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T143332.715087.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:34:08 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753374_chr12_1.fastq.gz, data/raw/ERR753374_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753374_chr12_1_fastqc.html, results/fastqc_result/ERR753374_chr12_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753374_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753374_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 14:34:18 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:34:18 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753374_chr12_1.fastq.gz, data/raw/ERR753374_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753374_chr12_1_fastp.fastq.gz, data/processed/ERR753374_chr12_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753374_chr12_1_fastp.fastq.gz, data/processed/ERR753374_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 849844\n",
      "total bases: 81288111\n",
      "Q20 bases: 80518260(99.0529%)\n",
      "Q30 bases: 73547935(90.4781%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 849844\n",
      "total bases: 79522059\n",
      "Q20 bases: 78682606(98.9444%)\n",
      "Q30 bases: 72336674(90.9643%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 822905\n",
      "total bases: 76905295\n",
      "Q20 bases: 76358183(99.2886%)\n",
      "Q30 bases: 70820041(92.0873%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 822905\n",
      "total bases: 74885957\n",
      "Q20 bases: 74450348(99.4183%)\n",
      "Q30 bases: 69591241(92.9296%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1645810\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 53878\n",
      "reads with adapter trimmed: 802\n",
      "bases trimmed due to adapters: 29662\n",
      "\n",
      "Duplication rate: 5.61036%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 154\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753374_chr12_1.fastq.gz -I data/raw/ERR753374_chr12_2.fastq.gz -o data/processed/ERR753374_chr12_1_fastp.fastq.gz -O data/processed/ERR753374_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 3 seconds\n",
      "[Thu Aug 17 14:34:22 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:34:22 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753374_chr12_1_fastp.fastq.gz, data/processed/ERR753374_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753374_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753374_chr12_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753374_chr12_1_fastp.fastq.gz, data/processed/ERR753374_chr12_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 14:34:31 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T143407.513340.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:34:33 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753375_chr12_1.fastq.gz, data/raw/ERR753375_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753375_chr12_1_fastqc.html, results/fastqc_result/ERR753375_chr12_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753375_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753375_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 14:34:44 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:34:44 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753375_chr12_1.fastq.gz, data/raw/ERR753375_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753375_chr12_1_fastp.fastq.gz, data/processed/ERR753375_chr12_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753375_chr12_2_fastp.fastq.gz, data/processed/ERR753375_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1133521\n",
      "total bases: 85620774\n",
      "Q20 bases: 85092338(99.3828%)\n",
      "Q30 bases: 83171029(97.1388%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1133521\n",
      "total bases: 85602587\n",
      "Q20 bases: 84492851(98.7036%)\n",
      "Q30 bases: 82429218(96.2929%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 738967\n",
      "total bases: 56110081\n",
      "Q20 bases: 55927294(99.6742%)\n",
      "Q30 bases: 55087312(98.1772%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 738967\n",
      "total bases: 56118788\n",
      "Q20 bases: 55577475(99.0354%)\n",
      "Q30 bases: 54704557(97.4799%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1477934\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 789108\n",
      "reads with adapter trimmed: 472\n",
      "bases trimmed due to adapters: 10223\n",
      "\n",
      "Duplication rate: 1.69099%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753375_chr12_1.fastq.gz -I data/raw/ERR753375_chr12_2.fastq.gz -o data/processed/ERR753375_chr12_1_fastp.fastq.gz -O data/processed/ERR753375_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Thu Aug 17 14:34:48 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:34:48 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753375_chr12_1_fastp.fastq.gz, data/processed/ERR753375_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753375_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753375_chr12_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753375_chr12_2_fastp.fastq.gz, data/processed/ERR753375_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 14:34:55 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T143432.044414.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:34:56 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753376_chr12_1.fastq.gz, data/raw/ERR753376_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753376_chr12_1_fastqc.html, results/fastqc_result/ERR753376_chr12_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753376_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753376_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 14:35:09 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:35:09 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753376_chr12_1.fastq.gz, data/raw/ERR753376_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753376_chr12_1_fastp.fastq.gz, data/processed/ERR753376_chr12_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753376_chr12_2_fastp.fastq.gz, data/processed/ERR753376_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1465726\n",
      "total bases: 110800464\n",
      "Q20 bases: 110033992(99.3082%)\n",
      "Q30 bases: 107011804(96.5806%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1465726\n",
      "total bases: 110496460\n",
      "Q20 bases: 109134987(98.7679%)\n",
      "Q30 bases: 104686265(94.7417%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 904519\n",
      "total bases: 68697720\n",
      "Q20 bases: 68438927(99.6233%)\n",
      "Q30 bases: 67261210(97.9089%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 904519\n",
      "total bases: 68671754\n",
      "Q20 bases: 68200419(99.3136%)\n",
      "Q30 bases: 66433000(96.7399%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1809038\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1122414\n",
      "reads with adapter trimmed: 700\n",
      "bases trimmed due to adapters: 15961\n",
      "\n",
      "Duplication rate: 9.00958%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753376_chr12_1.fastq.gz -I data/raw/ERR753376_chr12_2.fastq.gz -o data/processed/ERR753376_chr12_1_fastp.fastq.gz -O data/processed/ERR753376_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Thu Aug 17 14:35:14 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:35:14 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753376_chr12_1_fastp.fastq.gz, data/processed/ERR753376_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753376_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753376_chr12_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753376_chr12_2_fastp.fastq.gz, data/processed/ERR753376_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 14:35:23 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T143455.612305.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:35:25 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753377_chr12_1.fastq.gz, data/raw/ERR753377_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753377_chr12_1_fastqc.html, results/fastqc_result/ERR753377_chr12_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753377_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753377_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 14:35:38 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:35:38 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753377_chr12_1.fastq.gz, data/raw/ERR753377_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753377_chr12_1_fastp.fastq.gz, data/processed/ERR753377_chr12_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753377_chr12_2_fastp.fastq.gz, data/processed/ERR753377_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1534776\n",
      "total bases: 116035701\n",
      "Q20 bases: 115218791(99.296%)\n",
      "Q30 bases: 112042570(96.5587%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1534776\n",
      "total bases: 115756078\n",
      "Q20 bases: 114393805(98.8232%)\n",
      "Q30 bases: 109938521(94.9743%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 970154\n",
      "total bases: 73683410\n",
      "Q20 bases: 73398394(99.6132%)\n",
      "Q30 bases: 72118967(97.8768%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 970154\n",
      "total bases: 73656183\n",
      "Q20 bases: 73172261(99.343%)\n",
      "Q30 bases: 71360589(96.8834%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1940308\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1129244\n",
      "reads with adapter trimmed: 748\n",
      "bases trimmed due to adapters: 16700\n",
      "\n",
      "Duplication rate: 9.03239%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 119\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753377_chr12_1.fastq.gz -I data/raw/ERR753377_chr12_2.fastq.gz -o data/processed/ERR753377_chr12_1_fastp.fastq.gz -O data/processed/ERR753377_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Thu Aug 17 14:35:43 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:35:43 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753377_chr12_1_fastp.fastq.gz, data/processed/ERR753377_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753377_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753377_chr12_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753377_chr12_2_fastp.fastq.gz, data/processed/ERR753377_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 14:35:52 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T143523.951716.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:35:53 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753378_chr12_1.fastq.gz, data/raw/ERR753378_chr12_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753378_chr12_1_fastqc.html, results/fastqc_result/ERR753378_chr12_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753378_chr12_1_fastqc.html\n",
      "    wildcards: sample=ERR753378_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr12_1.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr12_1.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr12_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr12_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr12_2.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr12_2.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr12_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 14:36:08 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:36:08 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753378_chr12_1.fastq.gz, data/raw/ERR753378_chr12_2.fastq.gz\n",
      "    output: data/processed/ERR753378_chr12_1_fastp.fastq.gz, data/processed/ERR753378_chr12_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753378_chr12_2_fastp.fastq.gz, data/processed/ERR753378_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1753717\n",
      "total bases: 133156565\n",
      "Q20 bases: 132634253(99.6077%)\n",
      "Q30 bases: 128678299(96.6368%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1753717\n",
      "total bases: 133019021\n",
      "Q20 bases: 131845170(99.1175%)\n",
      "Q30 bases: 125793123(94.5678%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1509829\n",
      "total bases: 114740679\n",
      "Q20 bases: 114539895(99.825%)\n",
      "Q30 bases: 112565716(98.1045%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1509829\n",
      "total bases: 114731150\n",
      "Q20 bases: 114164648(99.5062%)\n",
      "Q30 bases: 110726794(96.5098%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3019658\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 487776\n",
      "reads with adapter trimmed: 1118\n",
      "bases trimmed due to adapters: 25907\n",
      "\n",
      "Duplication rate: 1.96604%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753378_chr12_1.fastq.gz -I data/raw/ERR753378_chr12_2.fastq.gz -o data/processed/ERR753378_chr12_1_fastp.fastq.gz -O data/processed/ERR753378_chr12_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Thu Aug 17 14:36:14 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:36:14 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753378_chr12_1_fastp.fastq.gz, data/processed/ERR753378_chr12_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753378_chr12_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753378_chr12_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753378_chr12_2_fastp.fastq.gz, data/processed/ERR753378_chr12_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr12_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr12_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr12_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr12_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 14:36:28 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T143552.400748.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:36:29 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR696683_chr12_1_fastp.fastq.gz, data/processed/ERR696683_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR696683_chr12.sam, results/mapped_reads/ERR696683_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR696683_chr12_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR696683_chr12_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.57 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 47.73 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.40 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.27 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 28.93 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 81.723 sec; CPU: 77.903 sec\n",
      "[Thu Aug 17 14:39:19 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:39:19 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR696683_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr12.bam, results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR696683_chr12.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR696683_chr12_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 14:40:03 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:40:03 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR696683_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "14:40:04.323 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 14:40:04 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR696683_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR696683_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 14:40:04 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 14:40:04\tMarkDuplicates\tStart of doWork freeMemory: 528831416; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:40:04\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 14:40:04\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam\n",
      "INFO\t2023-08-17 14:40:05\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:53,179,808\n",
      "INFO\t2023-08-17 14:40:05\tMarkDuplicates\tTracking 639 as yet unmatched pairs. 639 records in RAM.\n",
      "INFO\t2023-08-17 14:40:07\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 12:113,957,746\n",
      "INFO\t2023-08-17 14:40:07\tMarkDuplicates\tTracking 278 as yet unmatched pairs. 278 records in RAM.\n",
      "INFO\t2023-08-17 14:40:07\tMarkDuplicates\tRead 2264185 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 14:40:07\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 287345520; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:40:07\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 14:40:07\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 14:40:08\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 14:40:08\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 14:40:08\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 890237512; totalMemory: 1436549120; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:40:08\tMarkDuplicates\tMarking 210588 records as duplicates.\n",
      "INFO\t2023-08-17 14:40:08\tMarkDuplicates\tFound 91 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr12_sorted.bam\n",
      "INFO\t2023-08-17 14:40:08\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 14:40:23\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 14:40:23\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 14:40:23\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 14:40:23\tMarkDuplicates\tBefore output close freeMemory: 528130008; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:40:23\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 14:40:23\tMarkDuplicates\tAfter output close freeMemory: 528130008; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 14:40:23 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.31 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 14:40:24 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T143628.230905.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:40:26 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753368_chr12_1_fastp.fastq.gz, data/processed/ERR753368_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753368_chr12.sam, results/mapped_reads/ERR753368_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753368_chr12_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753368_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.45 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 46.56 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.43 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.30 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 29.34 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 81.646 sec; CPU: 77.079 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 14:44:18 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:44:18 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753368_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr12.bam, results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753368_chr12.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753368_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 14:45:19 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:45:19 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753368_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753368_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "14:45:20.734 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 14:45:20 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753368_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753368_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 14:45:20 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 14:45:20\tMarkDuplicates\tStart of doWork freeMemory: 528793384; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:45:20\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 14:45:20\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam\n",
      "INFO\t2023-08-17 14:45:22\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:43,551,130\n",
      "INFO\t2023-08-17 14:45:22\tMarkDuplicates\tTracking 481 as yet unmatched pairs. 481 records in RAM.\n",
      "INFO\t2023-08-17 14:45:23\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 12:76,860,355\n",
      "INFO\t2023-08-17 14:45:23\tMarkDuplicates\tTracking 510 as yet unmatched pairs. 510 records in RAM.\n",
      "INFO\t2023-08-17 14:45:25\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 12:122,377,489\n",
      "INFO\t2023-08-17 14:45:25\tMarkDuplicates\tTracking 227 as yet unmatched pairs. 227 records in RAM.\n",
      "INFO\t2023-08-17 14:45:25\tMarkDuplicates\tRead 3277664 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 14:45:25\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 435178536; totalMemory: 859832320; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:45:25\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 14:45:25\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 14:45:26\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 14:45:26\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 14:45:26\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1075797504; totalMemory: 1622147072; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:45:26\tMarkDuplicates\tMarking 79700 records as duplicates.\n",
      "INFO\t2023-08-17 14:45:26\tMarkDuplicates\tFound 488 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr12_sorted.bam\n",
      "INFO\t2023-08-17 14:45:26\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 14:45:46\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 14:45:46\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 14:45:46\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 14:45:46\tMarkDuplicates\tBefore output close freeMemory: 528091936; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:45:46\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 14:45:46\tMarkDuplicates\tAfter output close freeMemory: 528091936; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 14:45:46 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.43 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 14:45:49 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T144024.801761.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:45:50 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753369_chr12_1_fastp.fastq.gz, data/processed/ERR753369_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753369_chr12.sam, results/mapped_reads/ERR753369_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753369_chr12_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753369_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.46 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 50.77 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.39 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.27 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 29.53 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 85.739 sec; CPU: 81.431 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 14:49:44 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:49:44 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753369_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr12.bam, results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753369_chr12.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753369_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 14:50:45 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:50:45 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753369_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753369_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "14:50:46.315 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 14:50:46 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753369_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753369_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 14:50:46 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 14:50:46\tMarkDuplicates\tStart of doWork freeMemory: 528826688; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:50:46\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 14:50:46\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam\n",
      "INFO\t2023-08-17 14:50:47\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:40,104,739\n",
      "INFO\t2023-08-17 14:50:47\tMarkDuplicates\tTracking 392 as yet unmatched pairs. 392 records in RAM.\n",
      "INFO\t2023-08-17 14:50:49\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 12:67,306,225\n",
      "INFO\t2023-08-17 14:50:49\tMarkDuplicates\tTracking 515 as yet unmatched pairs. 515 records in RAM.\n",
      "INFO\t2023-08-17 14:50:50\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 12:112,948,761\n",
      "INFO\t2023-08-17 14:50:50\tMarkDuplicates\tTracking 273 as yet unmatched pairs. 273 records in RAM.\n",
      "INFO\t2023-08-17 14:50:51\tMarkDuplicates\tRead 3547704 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 14:50:51\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 773250040; totalMemory: 1227882496; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:50:51\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 14:50:51\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 14:50:52\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 14:50:52\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 14:50:52\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1218437384; totalMemory: 1764753408; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:50:52\tMarkDuplicates\tMarking 100939 records as duplicates.\n",
      "INFO\t2023-08-17 14:50:52\tMarkDuplicates\tFound 534 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr12_sorted.bam\n",
      "INFO\t2023-08-17 14:50:52\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 14:51:12\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 14:51:12\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 14:51:12\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 14:51:12\tMarkDuplicates\tBefore output close freeMemory: 528125648; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:51:12\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 14:51:12\tMarkDuplicates\tAfter output close freeMemory: 528125648; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 14:51:12 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.43 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 14:51:14 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T144549.221048.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:51:15 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753370_chr12_1_fastp.fastq.gz, data/processed/ERR753370_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753370_chr12.sam, results/mapped_reads/ERR753370_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753370_chr12_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753370_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.45 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 43.95 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.41 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.27 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 26.72 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 76.275 sec; CPU: 71.802 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 14:56:42 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:56:42 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753370_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr12.bam, results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753370_chr12.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753370_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 14:58:38 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:58:38 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753370_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753370_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "14:58:39.850 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 14:58:39 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753370_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753370_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 14:58:39 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 14:58:39\tMarkDuplicates\tStart of doWork freeMemory: 528827552; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:58:39\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 14:58:39\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam\n",
      "INFO\t2023-08-17 14:58:41\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:21,504,461\n",
      "INFO\t2023-08-17 14:58:41\tMarkDuplicates\tTracking 798 as yet unmatched pairs. 798 records in RAM.\n",
      "INFO\t2023-08-17 14:58:43\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 12:49,752,445\n",
      "INFO\t2023-08-17 14:58:43\tMarkDuplicates\tTracking 1422 as yet unmatched pairs. 1422 records in RAM.\n",
      "INFO\t2023-08-17 14:58:44\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 12:69,697,586\n",
      "INFO\t2023-08-17 14:58:44\tMarkDuplicates\tTracking 1413 as yet unmatched pairs. 1413 records in RAM.\n",
      "INFO\t2023-08-17 14:58:46\tMarkDuplicates\tRead     4,000,000 records.  Elapsed time: 00:00:06s.  Time for last 1,000,000:    1s.  Last read position: 12:103,793,216\n",
      "INFO\t2023-08-17 14:58:46\tMarkDuplicates\tTracking 1253 as yet unmatched pairs. 1253 records in RAM.\n",
      "INFO\t2023-08-17 14:58:48\tMarkDuplicates\tRead     5,000,000 records.  Elapsed time: 00:00:08s.  Time for last 1,000,000:    1s.  Last read position: 12:128,502,441\n",
      "INFO\t2023-08-17 14:58:48\tMarkDuplicates\tTracking 275 as yet unmatched pairs. 275 records in RAM.\n",
      "INFO\t2023-08-17 14:58:48\tMarkDuplicates\tRead 5146509 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 14:58:48\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 430419360; totalMemory: 1053818880; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:58:48\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 14:58:48\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 14:58:49\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 14:58:49\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 14:58:49\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1044374456; totalMemory: 1590689792; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:58:49\tMarkDuplicates\tMarking 368832 records as duplicates.\n",
      "INFO\t2023-08-17 14:58:49\tMarkDuplicates\tFound 219 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr12_sorted.bam\n",
      "INFO\t2023-08-17 14:58:49\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 14:59:28\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 14:59:28\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 14:59:28\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 14:59:28\tMarkDuplicates\tBefore output close freeMemory: 528125744; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 14:59:28\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 14:59:28\tMarkDuplicates\tAfter output close freeMemory: 528125744; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 14:59:28 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.81 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 14:59:32 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T145114.722180.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 14:59:33 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753371_chr12_1_fastp.fastq.gz, data/processed/ERR753371_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753371_chr12.sam, results/mapped_reads/ERR753371_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753371_chr12_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753371_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.44 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 50.77 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.39 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.27 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 29.49 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 85.682 sec; CPU: 81.356 sec\n",
      "[Thu Aug 17 15:02:10 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:02:10 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753371_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr12.bam, results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753371_chr12.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753371_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 15:02:43 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:02:43 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753371_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753371_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "15:02:44.098 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 15:02:44 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753371_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753371_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 15:02:44 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 15:02:44\tMarkDuplicates\tStart of doWork freeMemory: 528776536; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:02:44\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 15:02:44\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam\n",
      "INFO\t2023-08-17 15:02:45\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:68,764,488\n",
      "INFO\t2023-08-17 15:02:45\tMarkDuplicates\tTracking 110 as yet unmatched pairs. 110 records in RAM.\n",
      "INFO\t2023-08-17 15:02:46\tMarkDuplicates\tRead 1748836 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 15:02:46\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 276480456; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:02:46\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 15:02:46\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 15:02:47\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 15:02:47\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 15:02:47\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 818879920; totalMemory: 1365245952; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:02:47\tMarkDuplicates\tMarking 714304 records as duplicates.\n",
      "INFO\t2023-08-17 15:02:47\tMarkDuplicates\tFound 65 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr12_sorted.bam\n",
      "INFO\t2023-08-17 15:02:47\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 15:02:58\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 15:02:58\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 15:02:58\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 15:02:58\tMarkDuplicates\tBefore output close freeMemory: 528073088; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:02:58\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 15:02:58\tMarkDuplicates\tAfter output close freeMemory: 528073088; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 15:02:58 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.24 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 15:02:59 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T145932.679371.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:03:01 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753372_chr12_1_fastp.fastq.gz, data/processed/ERR753372_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753372_chr12.sam, results/mapped_reads/ERR753372_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753372_chr12_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR753372_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.45 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 44.61 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.40 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.27 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 28.07 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 78.502 sec; CPU: 73.816 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 15:06:59 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:06:59 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753372_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr12.bam, results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753372_chr12.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753372_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 15:07:59 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:07:59 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753372_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753372_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "15:08:00.966 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 15:08:00 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753372_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753372_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 15:08:00 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 15:08:00\tMarkDuplicates\tStart of doWork freeMemory: 528822160; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:08:00\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 15:08:00\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam\n",
      "INFO\t2023-08-17 15:08:02\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:40,527,729\n",
      "INFO\t2023-08-17 15:08:02\tMarkDuplicates\tTracking 548 as yet unmatched pairs. 548 records in RAM.\n",
      "INFO\t2023-08-17 15:08:03\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 12:69,257,720\n",
      "INFO\t2023-08-17 15:08:03\tMarkDuplicates\tTracking 530 as yet unmatched pairs. 530 records in RAM.\n",
      "INFO\t2023-08-17 15:08:05\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 12:116,750,201\n",
      "INFO\t2023-08-17 15:08:05\tMarkDuplicates\tTracking 302 as yet unmatched pairs. 302 records in RAM.\n",
      "INFO\t2023-08-17 15:08:05\tMarkDuplicates\tRead 3494913 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 15:08:06\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 744473992; totalMemory: 1192230912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:08:06\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 15:08:06\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 15:08:06\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 15:08:06\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 15:08:06\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1182781792; totalMemory: 1729101824; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:08:06\tMarkDuplicates\tMarking 84319 records as duplicates.\n",
      "INFO\t2023-08-17 15:08:06\tMarkDuplicates\tFound 567 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr12_sorted.bam\n",
      "INFO\t2023-08-17 15:08:06\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 15:08:26\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 15:08:26\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 15:08:26\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 15:08:26\tMarkDuplicates\tBefore output close freeMemory: 528122048; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:08:26\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 15:08:26\tMarkDuplicates\tAfter output close freeMemory: 528122048; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 15:08:26 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.42 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 15:08:28 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T150300.003803.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:08:30 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753373_chr12_1_fastp.fastq.gz, data/processed/ERR753373_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753373_chr12.sam, results/mapped_reads/ERR753373_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753373_chr12_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753373_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.42 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 42.81 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.40 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.27 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 26.15 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 74.392 sec; CPU: 70.057 sec\n",
      "[Thu Aug 17 15:12:13 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:12:13 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753373_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr12.bam, results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753373_chr12.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753373_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 15:13:25 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:13:25 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753373_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753373_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "15:13:26.861 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 15:13:26 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753373_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753373_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 15:13:26 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 15:13:26\tMarkDuplicates\tStart of doWork freeMemory: 528786632; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:13:26\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 15:13:26\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam\n",
      "INFO\t2023-08-17 15:13:28\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:46,204,254\n",
      "INFO\t2023-08-17 15:13:28\tMarkDuplicates\tTracking 309 as yet unmatched pairs. 309 records in RAM.\n",
      "INFO\t2023-08-17 15:13:30\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 12:92,184,591\n",
      "INFO\t2023-08-17 15:13:30\tMarkDuplicates\tTracking 219 as yet unmatched pairs. 219 records in RAM.\n",
      "INFO\t2023-08-17 15:13:31\tMarkDuplicates\tRead 2777001 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 15:13:31\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 638903728; totalMemory: 1008730112; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:13:31\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 15:13:31\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 15:13:31\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 15:13:31\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 15:13:31\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 999244992; totalMemory: 1545601024; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:13:31\tMarkDuplicates\tMarking 73157 records as duplicates.\n",
      "INFO\t2023-08-17 15:13:31\tMarkDuplicates\tFound 23782 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr12_sorted.bam\n",
      "INFO\t2023-08-17 15:13:31\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 15:13:55\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 15:13:55\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 15:13:55\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 15:13:55\tMarkDuplicates\tBefore output close freeMemory: 528085408; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:13:55\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 15:13:55\tMarkDuplicates\tAfter output close freeMemory: 528085408; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 15:13:55 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.47 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 15:13:57 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T150828.832207.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:13:58 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753374_chr12_1_fastp.fastq.gz, data/processed/ERR753374_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753374_chr12.sam, results/mapped_reads/ERR753374_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753374_chr12_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR753374_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.44 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 42.83 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.39 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.27 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 26.07 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 74.427 sec; CPU: 70.003 sec\n",
      "[Thu Aug 17 15:16:33 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:16:33 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753374_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr12.bam, results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753374_chr12.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753374_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 15:17:13 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:17:13 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753374_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753374_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "15:17:14.859 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 15:17:14 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753374_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753374_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 15:17:14 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 15:17:14\tMarkDuplicates\tStart of doWork freeMemory: 528832680; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:17:14\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 15:17:14\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam\n",
      "INFO\t2023-08-17 15:17:16\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:75,838,860\n",
      "INFO\t2023-08-17 15:17:16\tMarkDuplicates\tTracking 162 as yet unmatched pairs. 162 records in RAM.\n",
      "INFO\t2023-08-17 15:17:17\tMarkDuplicates\tRead 1659483 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 15:17:17\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 287418256; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:17:17\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 15:17:17\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 15:17:17\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 15:17:18\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 15:17:18\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 784332344; totalMemory: 1330642944; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:17:18\tMarkDuplicates\tMarking 175808 records as duplicates.\n",
      "INFO\t2023-08-17 15:17:18\tMarkDuplicates\tFound 58563 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr12_sorted.bam\n",
      "INFO\t2023-08-17 15:17:18\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 15:17:31\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 15:17:31\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 15:17:31\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 15:17:31\tMarkDuplicates\tBefore output close freeMemory: 528130680; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:17:31\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 15:17:31\tMarkDuplicates\tAfter output close freeMemory: 528130680; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 15:17:31 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.28 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 15:17:33 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T151357.447374.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:17:34 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753375_chr12_1_fastp.fastq.gz, data/processed/ERR753375_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753375_chr12.sam, results/mapped_reads/ERR753375_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753375_chr12_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR753375_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.44 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 42.92 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.39 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.26 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 26.08 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 74.665 sec; CPU: 70.090 sec\n",
      "[Thu Aug 17 15:19:40 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:19:40 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753375_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr12.bam, results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753375_chr12.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753375_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 15:20:05 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:20:05 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753375_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753375_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "15:20:06.577 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 15:20:06 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753375_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753375_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 15:20:06 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 15:20:06\tMarkDuplicates\tStart of doWork freeMemory: 528826944; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:20:06\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 15:20:06\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam\n",
      "INFO\t2023-08-17 15:20:08\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:87,996,510\n",
      "INFO\t2023-08-17 15:20:08\tMarkDuplicates\tTracking 188 as yet unmatched pairs. 188 records in RAM.\n",
      "INFO\t2023-08-17 15:20:08\tMarkDuplicates\tRead 1483975 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 15:20:08\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 370635256; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:20:08\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 15:20:08\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 15:20:09\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 15:20:09\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 15:20:09\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 769648112; totalMemory: 1315962880; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:20:09\tMarkDuplicates\tMarking 72150 records as duplicates.\n",
      "INFO\t2023-08-17 15:20:09\tMarkDuplicates\tFound 188 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr12_sorted.bam\n",
      "INFO\t2023-08-17 15:20:09\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 15:20:18\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 15:20:18\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 15:20:18\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 15:20:18\tMarkDuplicates\tBefore output close freeMemory: 528126080; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:20:18\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 15:20:18\tMarkDuplicates\tAfter output close freeMemory: 528126080; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 15:20:18 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.20 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 15:20:19 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T151733.229028.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:20:20 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753376_chr12_1_fastp.fastq.gz, data/processed/ERR753376_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753376_chr12.sam, results/mapped_reads/ERR753376_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753376_chr12_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753376_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.45 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 42.81 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.40 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.27 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 26.20 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 73.363 sec; CPU: 70.142 sec\n",
      "[Thu Aug 17 15:22:43 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:22:43 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753376_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr12.bam, results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753376_chr12.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753376_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 15:23:17 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:23:17 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753376_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753376_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "15:23:18.312 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 15:23:18 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753376_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753376_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 15:23:18 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 15:23:18\tMarkDuplicates\tStart of doWork freeMemory: 528788408; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:23:18\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 15:23:18\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam\n",
      "INFO\t2023-08-17 15:23:19\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:64,464,346\n",
      "INFO\t2023-08-17 15:23:19\tMarkDuplicates\tTracking 275 as yet unmatched pairs. 275 records in RAM.\n",
      "INFO\t2023-08-17 15:23:20\tMarkDuplicates\tRead 1814904 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 15:23:20\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 334843144; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:23:20\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 15:23:21\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 15:23:21\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 15:23:21\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 15:23:21\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 818890656; totalMemory: 1365245952; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:23:21\tMarkDuplicates\tMarking 260471 records as duplicates.\n",
      "INFO\t2023-08-17 15:23:21\tMarkDuplicates\tFound 83 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr12_sorted.bam\n",
      "INFO\t2023-08-17 15:23:21\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 15:23:32\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 15:23:32\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 15:23:32\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 15:23:32\tMarkDuplicates\tBefore output close freeMemory: 528085944; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:23:32\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 15:23:32\tMarkDuplicates\tAfter output close freeMemory: 528085944; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 15:23:32 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.24 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 15:23:34 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T152019.579754.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:23:35 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753377_chr12_1_fastp.fastq.gz, data/processed/ERR753377_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753377_chr12.sam, results/mapped_reads/ERR753377_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753377_chr12_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753377_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.44 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 42.93 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.39 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.27 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 26.05 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 74.328 sec; CPU: 70.083 sec\n",
      "[Thu Aug 17 15:25:55 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:25:55 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753377_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr12.bam, results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753377_chr12.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753377_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 15:26:27 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:26:27 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753377_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753377_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "15:26:28.565 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 15:26:28 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753377_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753377_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 15:26:28 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 15:26:28\tMarkDuplicates\tStart of doWork freeMemory: 528762280; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:26:28\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 15:26:28\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam\n",
      "INFO\t2023-08-17 15:26:30\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:57,029,291\n",
      "INFO\t2023-08-17 15:26:30\tMarkDuplicates\tTracking 396 as yet unmatched pairs. 396 records in RAM.\n",
      "INFO\t2023-08-17 15:26:31\tMarkDuplicates\tRead 1953622 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 15:26:31\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 320405152; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:26:31\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 15:26:31\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 15:26:32\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 15:26:32\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 15:26:32\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 832495488; totalMemory: 1378877440; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:26:32\tMarkDuplicates\tMarking 278982 records as duplicates.\n",
      "INFO\t2023-08-17 15:26:32\tMarkDuplicates\tFound 103 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr12_sorted.bam\n",
      "INFO\t2023-08-17 15:26:32\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 15:26:43\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 15:26:43\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 15:26:43\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 15:26:43\tMarkDuplicates\tBefore output close freeMemory: 528059568; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:26:43\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 15:26:43\tMarkDuplicates\tAfter output close freeMemory: 528059568; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 15:26:43 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.25 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 15:26:45 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T152334.195019.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:26:46 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753378_chr12_1_fastp.fastq.gz, data/processed/ERR753378_chr12_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753378_chr12.sam, results/mapped_reads/ERR753378_chr12_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753378_chr12_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR753378_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.44 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 42.84 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.40 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.26 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 26.14 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 70.889 sec; CPU: 70.090 sec\n",
      "[Thu Aug 17 15:29:56 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:29:56 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753378_chr12_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr12.bam, results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753378_chr12.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753378_chr12_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 15:30:41 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:30:41 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr12_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753378_chr12_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam\n",
      "    wildcards: sample=ERR753378_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "15:30:41.999 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 15:30:42 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753378_chr12_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753378_chr12_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 15:30:42 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 15:30:42\tMarkDuplicates\tStart of doWork freeMemory: 528781600; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:30:42\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 15:30:42\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam\n",
      "INFO\t2023-08-17 15:30:43\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 12:47,983,102\n",
      "INFO\t2023-08-17 15:30:43\tMarkDuplicates\tTracking 460 as yet unmatched pairs. 460 records in RAM.\n",
      "INFO\t2023-08-17 15:30:44\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 12:90,841,669\n",
      "INFO\t2023-08-17 15:30:44\tMarkDuplicates\tTracking 410 as yet unmatched pairs. 410 records in RAM.\n",
      "INFO\t2023-08-17 15:30:46\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 12:132,671,826\n",
      "INFO\t2023-08-17 15:30:46\tMarkDuplicates\tTracking 25 as yet unmatched pairs. 25 records in RAM.\n",
      "INFO\t2023-08-17 15:30:46\tMarkDuplicates\tRead 3037191 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 15:30:46\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 575868872; totalMemory: 974127104; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:30:46\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 15:30:46\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 15:30:47\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 15:30:47\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 15:30:47\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 964637144; totalMemory: 1510998016; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:30:47\tMarkDuplicates\tMarking 71538 records as duplicates.\n",
      "INFO\t2023-08-17 15:30:47\tMarkDuplicates\tFound 500 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr12_sorted.bam\n",
      "INFO\t2023-08-17 15:30:47\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 15:31:04\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 15:31:04\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 15:31:04\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 15:31:04\tMarkDuplicates\tBefore output close freeMemory: 528080360; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 15:31:04\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 15:31:04\tMarkDuplicates\tAfter output close freeMemory: 528080360; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 15:31:04 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.37 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 15:31:06 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T152645.168110.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:31:07 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR696683_chr12_dedup.bam\n",
      "    output: results/variants/ERR696683_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR696683_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR696683_chr12.vcf\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "index file data/reference/genome.fa.fai not found, generating...\n",
      "[Thu Aug 17 15:31:52 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T153106.358469.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:31:53 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753368_chr12_dedup.bam\n",
      "    output: results/variants/ERR753368_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753368_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753368_chr12.vcf\n",
      "    wildcards: sample=ERR753368_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 15:32:55 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T153152.307800.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:32:57 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753369_chr12_dedup.bam\n",
      "    output: results/variants/ERR753369_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753369_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753369_chr12.vcf\n",
      "    wildcards: sample=ERR753369_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 15:34:02 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T153255.815763.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:34:03 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753370_chr12_dedup.bam\n",
      "    output: results/variants/ERR753370_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753370_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753370_chr12.vcf\n",
      "    wildcards: sample=ERR753370_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 15:35:46 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T153402.441633.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:35:47 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753371_chr12_dedup.bam\n",
      "    output: results/variants/ERR753371_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753371_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753371_chr12.vcf\n",
      "    wildcards: sample=ERR753371_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 15:36:18 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T153546.601097.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:36:19 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753372_chr12_dedup.bam\n",
      "    output: results/variants/ERR753372_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753372_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753372_chr12.vcf\n",
      "    wildcards: sample=ERR753372_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 15:37:24 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T153618.513757.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:37:26 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753373_chr12_dedup.bam\n",
      "    output: results/variants/ERR753373_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753373_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753373_chr12.vcf\n",
      "    wildcards: sample=ERR753373_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 15:38:33 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T153724.894205.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:38:34 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753374_chr12_dedup.bam\n",
      "    output: results/variants/ERR753374_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753374_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753374_chr12.vcf\n",
      "    wildcards: sample=ERR753374_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 15:39:19 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T153833.694770.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:39:20 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753375_chr12_dedup.bam\n",
      "    output: results/variants/ERR753375_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753375_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753375_chr12.vcf\n",
      "    wildcards: sample=ERR753375_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 15:39:56 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T153919.429573.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:39:57 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753376_chr12_dedup.bam\n",
      "    output: results/variants/ERR753376_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753376_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753376_chr12.vcf\n",
      "    wildcards: sample=ERR753376_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 15:40:35 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T153956.611526.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:40:36 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753377_chr12_dedup.bam\n",
      "    output: results/variants/ERR753377_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753377_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753377_chr12.vcf\n",
      "    wildcards: sample=ERR753377_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 15:41:15 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T154035.308920.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:41:16 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753378_chr12_dedup.bam\n",
      "    output: results/variants/ERR753378_chr12.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753378_chr12.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753378_chr12.vcf\n",
      "    wildcards: sample=ERR753378_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 15:42:15 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T154115.581399.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:42:16 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR696683_chr12.vcf\n",
      "    output: results/variants/vep/ERR696683_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR696683_chr12.txt\n",
      "    wildcards: sample=ERR696683_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 15:42:40 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T154215.299348.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:42:42 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753368_chr12.vcf\n",
      "    output: results/variants/vep/ERR753368_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753368_chr12.txt\n",
      "    wildcards: sample=ERR753368_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 15:43:10 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T154240.900317.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:43:11 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753369_chr12.vcf\n",
      "    output: results/variants/vep/ERR753369_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753369_chr12.txt\n",
      "    wildcards: sample=ERR753369_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 15:43:39 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T154310.479311.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:43:40 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753370_chr12.vcf\n",
      "    output: results/variants/vep/ERR753370_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753370_chr12.txt\n",
      "    wildcards: sample=ERR753370_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 15:44:10 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T154339.337805.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:44:12 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753371_chr12.vcf\n",
      "    output: results/variants/vep/ERR753371_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753371_chr12.txt\n",
      "    wildcards: sample=ERR753371_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 15:44:31 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T154410.855348.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:44:33 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753372_chr12.vcf\n",
      "    output: results/variants/vep/ERR753372_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753372_chr12.txt\n",
      "    wildcards: sample=ERR753372_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 15:45:00 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T154432.028529.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:45:01 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753373_chr12.vcf\n",
      "    output: results/variants/vep/ERR753373_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753373_chr12.txt\n",
      "    wildcards: sample=ERR753373_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 15:45:27 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T154500.653388.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:45:28 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753374_chr12.vcf\n",
      "    output: results/variants/vep/ERR753374_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753374_chr12.txt\n",
      "    wildcards: sample=ERR753374_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 15:45:48 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T154527.763572.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:45:50 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753375_chr12.vcf\n",
      "    output: results/variants/vep/ERR753375_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753375_chr12.txt\n",
      "    wildcards: sample=ERR753375_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 15:46:11 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T154549.157699.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:46:12 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753376_chr12.vcf\n",
      "    output: results/variants/vep/ERR753376_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753376_chr12.txt\n",
      "    wildcards: sample=ERR753376_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 15:46:34 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T154611.737058.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:46:35 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753377_chr12.vcf\n",
      "    output: results/variants/vep/ERR753377_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753377_chr12.txt\n",
      "    wildcards: sample=ERR753377_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 15:46:57 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T154634.599831.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:46:58 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753378_chr12.vcf\n",
      "    output: results/variants/vep/ERR753378_chr12.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753378_chr12.txt\n",
      "    wildcards: sample=ERR753378_chr12\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 15:47:24 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T154657.404009.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job              count    min threads    max threads\n",
      "-------------  -------  -------------  -------------\n",
      "parsing_dataR        1              1              1\n",
      "total                1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 15:47:25 2023]\n",
      "rule parsing_dataR:\n",
      "    input: code/05parsing_vep_data.R\n",
      "    output: tasks/13parsing_dataR.done\n",
      "    jobid: 0\n",
      "    reason: Params have changed since last execution\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Warning message:\n",
      "package ‘glue’ was built under R version 4.2.3 \n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">>> This is your sample list: ERR696683_chr12\n",
      ">>> This is your sample list: ERR753368_chr12\n",
      ">>> This is your sample list: ERR753369_chr12\n",
      ">>> This is your sample list: ERR753370_chr12\n",
      ">>> This is your sample list: ERR753371_chr12\n",
      ">>> This is your sample list: ERR753372_chr12\n",
      ">>> This is your sample list: ERR753373_chr12\n",
      ">>> This is your sample list: ERR753374_chr12\n",
      ">>> This is your sample list: ERR753375_chr12\n",
      ">>> This is your sample list: ERR753376_chr12\n",
      ">>> This is your sample list: ERR753377_chr12\n",
      ">>> This is your sample list: ERR753378_chr12\n",
      ">>> This is your gene: KRAS\n",
      "# A tibble: 0 × 44\n",
      "# ℹ 44 variables: #uploaded_variation <chr>, location <chr>, allele <chr>,\n",
      "#   gene <chr>, feature <chr>, feature_type <chr>, consequence <chr>,\n",
      "#   cdna_position <chr>, cds_position <chr>, protein_position <chr>,\n",
      "#   amino_acids <chr>, codons <chr>, existing_variation <chr>, impact <chr>,\n",
      "#   distance <chr>, strand <chr>, flags <chr>, variant_class <chr>,\n",
      "#   symbol <chr>, symbol_source <chr>, hgnc_id <chr>, biotype <chr>,\n",
      "#   mane_select <chr>, mane_plus_clinical <chr>, tsl <chr>, appris <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 24 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "3 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "4 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "# ℹ 18 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 36 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "3 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "4 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "5 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "# ℹ 30 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 36 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25209531_G/A       12:252095… A      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 12_25209531_G/A       12:252095… A      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "3 12_25209531_G/A       12:252095… A      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "4 12_25209531_G/A       12:252095… A      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 12_25209531_G/A       12:252095… A      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 12_25209531_G/A       12:252095… A      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "# ℹ 30 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 0 × 44\n",
      "# ℹ 44 variables: #uploaded_variation <chr>, location <chr>, allele <chr>,\n",
      "#   gene <chr>, feature <chr>, feature_type <chr>, consequence <chr>,\n",
      "#   cdna_position <chr>, cds_position <chr>, protein_position <chr>,\n",
      "#   amino_acids <chr>, codons <chr>, existing_variation <chr>, impact <chr>,\n",
      "#   distance <chr>, strand <chr>, flags <chr>, variant_class <chr>,\n",
      "#   symbol <chr>, symbol_source <chr>, hgnc_id <chr>, biotype <chr>,\n",
      "#   mane_select <chr>, mane_plus_clinical <chr>, tsl <chr>, appris <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 36 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "3 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "4 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 12_25209283_A/G       12:252092… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "# ℹ 30 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 25 × 44\n",
      "  `#uploaded_variation`   location allele gene  feature feature_type consequence\n",
      "  <chr>                   <chr>    <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25230796_AATTAATTA/… 12:2523… AATTA  ENSG… ENST00… Transcript   intron_var…\n",
      "2 12_25230796_AATTAATTA/… 12:2523… AATTA  ENSG… ENST00… Transcript   intron_var…\n",
      "3 12_25230796_AATTAATTA/… 12:2523… AATTA  ENSG… ENST00… Transcript   downstream…\n",
      "4 12_25230796_AATTAATTA/… 12:2523… AATTA  ENSG… ENST00… Transcript   intron_var…\n",
      "5 12_25230796_AATTAATTA/… 12:2523… AATTA  ENSG… ENST00… Transcript   intron_var…\n",
      "6 12_25230796_AATTAATTA/… 12:2523… AATTA  ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 19 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 0 × 44\n",
      "# ℹ 44 variables: #uploaded_variation <chr>, location <chr>, allele <chr>,\n",
      "#   gene <chr>, feature <chr>, feature_type <chr>, consequence <chr>,\n",
      "#   cdna_position <chr>, cds_position <chr>, protein_position <chr>,\n",
      "#   amino_acids <chr>, codons <chr>, existing_variation <chr>, impact <chr>,\n",
      "#   distance <chr>, strand <chr>, flags <chr>, variant_class <chr>,\n",
      "#   symbol <chr>, symbol_source <chr>, hgnc_id <chr>, biotype <chr>,\n",
      "#   mane_select <chr>, mane_plus_clinical <chr>, tsl <chr>, appris <chr>, …\n",
      "# A tibble: 24 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "3 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "4 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "5 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "# ℹ 18 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 12 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25209618_A/C       12:252096… C      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 12_25209618_A/C       12:252096… C      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "3 12_25209618_A/C       12:252096… C      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "4 12_25209618_A/C       12:252096… C      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "5 12_25209618_A/C       12:252096… C      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 12_25209618_A/C       12:252096… C      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "# ℹ 6 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 12 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "3 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "4 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "5 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "# ℹ 6 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 36 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "2 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "3 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "4 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   synonymous…\n",
      "5 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "6 12_25209843_A/G       12:252098… G      ENSG… ENST00… Transcript   3_prime_UT…\n",
      "# ℹ 30 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "[1] \"##==========================##\"\n",
      "[1] \"##===> WORK FINISHED!!! <===##\"\n",
      "[1] \"##==========================##\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Touching output file tasks/13parsing_dataR.done.\n",
      "[Thu Aug 17 15:47:29 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T154725.135820.snakemake.log\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['snakemake', '--cores', '1', 'parsing_dataR'], returncode=0)"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Running the workflow for the 12º chormosome\n",
    "\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--force\", \"data/reference/genome.fa\"]) ## rule reference_genome \n",
    "\n",
    "for sample in sample_list:\n",
    "    ## Quality inspection\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/fastqc_result/\" + sample + \"_chr12_1_fastqc.html\",   \n",
    "                    \"data/processed/\" + sample + \"_chr12_1_fastp.fastq.gz\",   \n",
    "                    \"results/fastqc_result/trimmed/\" + sample + \"_chr12_1_fastp_fastqc.html\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    ## Mapping reads\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/mapped_reads/\" + sample + \"_chr12_sorted.sam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr12_sorted.bam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr12_dedup.bam\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    # Extracting variants  \n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/\" + sample + \"_chr12.vcf\"])   \n",
    "\n",
    "## VEP CLI\n",
    "for sample in sample_list:\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/vep/\" + sample + \"_chr12.txt\"])\n",
    "\n",
    "## Extracting the gene PIKCA3 from the VEP files\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"parsing_dataR\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Changing the config file to do the chromosome 17 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['sed', '-i', '50s/KRAS/TP53/g', 'config.yaml'], returncode=0)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subprocess.run([\"sed\", \"-i\", \"6,+12s/chr12/chr17/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\",\"24s/chr12/chr17/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"31s/chromosome.12.fa.gz/chromosome.17.fa.gz/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"50s/KRAS/TP53/g\", \"config.yaml\"])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job                 count    min threads    max threads\n",
      "----------------  -------  -------------  -------------\n",
      "reference_genome        1              1              1\n",
      "total                   1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:44:22 2023]\n",
      "rule reference_genome:\n",
      "    output: data/reference/genome.fa\n",
      "    jobid: 0\n",
      "    reason: Forced execution\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "--2023-08-17 17:44:22--  https://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.17.fa.gz\n",
      "Resolviendo ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.169\n",
      "Conectando con ftp.ensembl.org (ftp.ensembl.org)[193.62.193.169]:443... conectado.\n",
      "Petición HTTP enviada, esperando respuesta... 200 OK\n",
      "Longitud: 24104573 (23M) [application/x-gzip]\n",
      "Guardando como: ‘data/reference/genome.fa.gz’\n",
      "\n",
      "     0K .......... .......... .......... .......... ..........  0%  446K 53s\n",
      "    50K .......... .......... .......... .......... ..........  0% 4,62M 29s\n",
      "   100K .......... .......... .......... .......... ..........  0% 1,02M 27s\n",
      "   150K .......... .......... .......... .......... ..........  0% 6,68M 21s\n",
      "   200K .......... .......... .......... .......... ..........  1% 1,06M 21s\n",
      "   250K .......... .......... .......... .......... ..........  1% 24,4M 18s\n",
      "   300K .......... .......... .......... .......... ..........  1% 50,0M 15s\n",
      "   350K .......... .......... .......... .......... ..........  1% 40,9M 13s\n",
      "   400K .......... .......... .......... .......... ..........  1% 6,68M 12s\n",
      "   450K .......... .......... .......... .......... ..........  2% 1,09M 13s\n",
      "   500K .......... .......... .......... .......... ..........  2% 36,1M 12s\n",
      "   550K .......... .......... .......... .......... ..........  2% 57,4M 11s\n",
      "   600K .......... .......... .......... .......... ..........  2% 57,0M 10s\n",
      "   650K .......... .......... .......... .......... ..........  2%  107M 9s\n",
      "   700K .......... .......... .......... .......... ..........  3%  105M 9s\n",
      "   750K .......... .......... .......... .......... ..........  3%  115M 8s\n",
      "   800K .......... .......... .......... .......... ..........  3% 35,4M 8s\n",
      "   850K .......... .......... .......... .......... ..........  3% 8,18M 7s\n",
      "   900K .......... .......... .......... .......... ..........  4% 1,17M 8s\n",
      "   950K .......... .......... .......... .......... ..........  4% 16,6M 8s\n",
      "  1000K .......... .......... .......... .......... ..........  4% 43,1M 7s\n",
      "  1050K .......... .......... .......... .......... ..........  4% 34,5M 7s\n",
      "  1100K .......... .......... .......... .......... ..........  4% 94,3M 7s\n",
      "  1150K .......... .......... .......... .......... ..........  5%  114M 6s\n",
      "  1200K .......... .......... .......... .......... ..........  5% 99,9M 6s\n",
      "  1250K .......... .......... .......... .......... ..........  5%  118M 6s\n",
      "  1300K .......... .......... .......... .......... ..........  5%  119M 6s\n",
      "  1350K .......... .......... .......... .......... ..........  5% 93,4M 5s\n",
      "  1400K .......... .......... .......... .......... ..........  6%  112M 5s\n",
      "  1450K .......... .......... .......... .......... ..........  6%  101M 5s\n",
      "  1500K .......... .......... .......... .......... ..........  6%  119M 5s\n",
      "  1550K .......... .......... .......... .......... ..........  6%  118M 5s\n",
      "  1600K .......... .......... .......... .......... ..........  7% 99,6M 5s\n",
      "  1650K .......... .......... .......... .......... ..........  7%  131M 4s\n",
      "  1700K .......... .......... .......... .......... ..........  7% 11,7M 4s\n",
      "  1750K .......... .......... .......... .......... ..........  7%  113M 4s\n",
      "  1800K .......... .......... .......... .......... ..........  7% 98,5M 4s\n",
      "  1850K .......... .......... .......... .......... ..........  8% 12,6M 4s\n",
      "  1900K .......... .......... .......... .......... ..........  8% 1,31M 4s\n",
      "  1950K .......... .......... .......... .......... ..........  8% 39,3M 4s\n",
      "  2000K .......... .......... .......... .......... ..........  8% 52,9M 4s\n",
      "  2050K .......... .......... .......... .......... ..........  8% 64,6M 4s\n",
      "  2100K .......... .......... .......... .......... ..........  9% 44,3M 4s\n",
      "  2150K .......... .......... .......... .......... ..........  9% 92,2M 4s\n",
      "  2200K .......... .......... .......... .......... ..........  9% 55,2M 4s\n",
      "  2250K .......... .......... .......... .......... ..........  9%  114M 4s\n",
      "  2300K .......... .......... .......... .......... ..........  9%  117M 4s\n",
      "  2350K .......... .......... .......... .......... .......... 10% 96,1M 4s\n",
      "  2400K .......... .......... .......... .......... .......... 10%  110M 3s\n",
      "  2450K .......... .......... .......... .......... .......... 10%  102M 3s\n",
      "  2500K .......... .......... .......... .......... .......... 10%  115M 3s\n",
      "  2550K .......... .......... .......... .......... .......... 11%  116M 3s\n",
      "  2600K .......... .......... .......... .......... .......... 11% 99,5M 3s\n",
      "  2650K .......... .......... .......... .......... .......... 11%  109M 3s\n",
      "  2700K .......... .......... .......... .......... .......... 11% 96,1M 3s\n",
      "  2750K .......... .......... .......... .......... .......... 11%  110M 3s\n",
      "  2800K .......... .......... .......... .......... .......... 12%  107M 3s\n",
      "  2850K .......... .......... .......... .......... .......... 12% 91,3M 3s\n",
      "  2900K .......... .......... .......... .......... .......... 12%  105M 3s\n",
      "  2950K .......... .......... .......... .......... .......... 12% 91,7M 3s\n",
      "  3000K .......... .......... .......... .......... .......... 12%  107M 3s\n",
      "  3050K .......... .......... .......... .......... .......... 13%  110M 3s\n",
      "  3100K .......... .......... .......... .......... .......... 13% 94,3M 3s\n",
      "  3150K .......... .......... .......... .......... .......... 13%  107M 3s\n",
      "  3200K .......... .......... .......... .......... .......... 13% 90,8M 3s\n",
      "  3250K .......... .......... .......... .......... .......... 14%  108M 2s\n",
      "  3300K .......... .......... .......... .......... .......... 14%  109M 2s\n",
      "  3350K .......... .......... .......... .......... .......... 14% 91,7M 2s\n",
      "  3400K .......... .......... .......... .......... .......... 14%  100M 2s\n",
      "  3450K .......... .......... .......... .......... .......... 14% 94,1M 2s\n",
      "  3500K .......... .......... .......... .......... .......... 15%  107M 2s\n",
      "  3550K .......... .......... .......... .......... .......... 15%  111M 2s\n",
      "  3600K .......... .......... .......... .......... .......... 15% 90,4M 2s\n",
      "  3650K .......... .......... .......... .......... .......... 15%  127M 2s\n",
      "  3700K .......... .......... .......... .......... .......... 15%  122M 2s\n",
      "  3750K .......... .......... .......... .......... .......... 16%  183M 2s\n",
      "  3800K .......... .......... .......... .......... .......... 16% 1,36M 2s\n",
      "  3850K .......... .......... .......... .......... .......... 16% 42,3M 2s\n",
      "  3900K .......... .......... .......... .......... .......... 16% 74,5M 2s\n",
      "  3950K .......... .......... .......... .......... .......... 16%  102M 2s\n",
      "  4000K .......... .......... .......... .......... .......... 17%  121M 2s\n",
      "  4050K .......... .......... .......... .......... .......... 17%  117M 2s\n",
      "  4100K .......... .......... .......... .......... .......... 17%  102M 2s\n",
      "  4150K .......... .......... .......... .......... .......... 17%  121M 2s\n",
      "  4200K .......... .......... .......... .......... .......... 18% 91,6M 2s\n",
      "  4250K .......... .......... .......... .......... .......... 18%  139M 2s\n",
      "  4300K .......... .......... .......... .......... .......... 18%  118M 2s\n",
      "  4350K .......... .......... .......... .......... .......... 18%  102M 2s\n",
      "  4400K .......... .......... .......... .......... .......... 18%  121M 2s\n",
      "  4450K .......... .......... .......... .......... .......... 19%  102M 2s\n",
      "  4500K .......... .......... .......... .......... .......... 19%  116M 2s\n",
      "  4550K .......... .......... .......... .......... .......... 19%  120M 2s\n",
      "  4600K .......... .......... .......... .......... .......... 19%  103M 2s\n",
      "  4650K .......... .......... .......... .......... .......... 19%  118M 2s\n",
      "  4700K .......... .......... .......... .......... .......... 20%  102M 2s\n",
      "  4750K .......... .......... .......... .......... .......... 20%  121M 2s\n",
      "  4800K .......... .......... .......... .......... .......... 20%  119M 2s\n",
      "  4850K .......... .......... .......... .......... .......... 20%  104M 2s\n",
      "  4900K .......... .......... .......... .......... .......... 21%  117M 2s\n",
      "  4950K .......... .......... .......... .......... .......... 21%  102M 2s\n",
      "  5000K .......... .......... .......... .......... .......... 21%  120M 2s\n",
      "  5050K .......... .......... .......... .......... .......... 21%  119M 2s\n",
      "  5100K .......... .......... .......... .......... .......... 21%  103M 2s\n",
      "  5150K .......... .......... .......... .......... .......... 22%  121M 2s\n",
      "  5200K .......... .......... .......... .......... .......... 22%  102M 2s\n",
      "  5250K .......... .......... .......... .......... .......... 22%  117M 2s\n",
      "  5300K .......... .......... .......... .......... .......... 22%  122M 2s\n",
      "  5350K .......... .......... .......... .......... .......... 22% 99,0M 2s\n",
      "  5400K .......... .......... .......... .......... .......... 23%  120M 2s\n",
      "  5450K .......... .......... .......... .......... .......... 23%  103M 2s\n",
      "  5500K .......... .......... .......... .......... .......... 23%  118M 2s\n",
      "  5550K .......... .......... .......... .......... .......... 23%  126M 1s\n",
      "  5600K .......... .......... .......... .......... .......... 24%  102M 1s\n",
      "  5650K .......... .......... .......... .......... .......... 24%  117M 1s\n",
      "  5700K .......... .......... .......... .......... .......... 24%  103M 1s\n",
      "  5750K .......... .......... .......... .......... .......... 24%  121M 1s\n",
      "  5800K .......... .......... .......... .......... .......... 24%  121M 1s\n",
      "  5850K .......... .......... .......... .......... .......... 25%  102M 1s\n",
      "  5900K .......... .......... .......... .......... .......... 25%  117M 1s\n",
      "  5950K .......... .......... .......... .......... .......... 25%  102M 1s\n",
      "  6000K .......... .......... .......... .......... .......... 25%  120M 1s\n",
      "  6050K .......... .......... .......... .......... .......... 25%  118M 1s\n",
      "  6100K .......... .......... .......... .......... .......... 26%  103M 1s\n",
      "  6150K .......... .......... .......... .......... .......... 26%  120M 1s\n",
      "  6200K .......... .......... .......... .......... .......... 26%  102M 1s\n",
      "  6250K .......... .......... .......... .......... .......... 26%  118M 1s\n",
      "  6300K .......... .......... .......... .......... .......... 26%  120M 1s\n",
      "  6350K .......... .......... .......... .......... .......... 27%  103M 1s\n",
      "  6400K .......... .......... .......... .......... .......... 27%  121M 1s\n",
      "  6450K .......... .......... .......... .......... .......... 27%  103M 1s\n",
      "  6500K .......... .......... .......... .......... .......... 27%  118M 1s\n",
      "  6550K .......... .......... .......... .......... .......... 28%  120M 1s\n",
      "  6600K .......... .......... .......... .......... .......... 28%  102M 1s\n",
      "  6650K .......... .......... .......... .......... .......... 28%  118M 1s\n",
      "  6700K .......... .......... .......... .......... .......... 28%  102M 1s\n",
      "  6750K .......... .......... .......... .......... .......... 28% 35,5M 1s\n",
      "  6800K .......... .......... .......... .......... .......... 29% 1,73M 1s\n",
      "  6850K .......... .......... .......... .......... .......... 29% 38,2M 1s\n",
      "  6900K .......... .......... .......... .......... .......... 29% 29,7M 1s\n",
      "  6950K .......... .......... .......... .......... .......... 29%  117M 1s\n",
      "  7000K .......... .......... .......... .......... .......... 29%  237M 1s\n",
      "  7050K .......... .......... .......... .......... .......... 30%  155M 1s\n",
      "  7100K .......... .......... .......... .......... .......... 30%  100M 1s\n",
      "  7150K .......... .......... .......... .......... .......... 30%  122M 1s\n",
      "  7200K .......... .......... .......... .......... .......... 30%  102M 1s\n",
      "  7250K .......... .......... .......... .......... .......... 31%  118M 1s\n",
      "  7300K .......... .......... .......... .......... .......... 31%  121M 1s\n",
      "  7350K .......... .......... .......... .......... .......... 31%  102M 1s\n",
      "  7400K .......... .......... .......... .......... .......... 31%  117M 1s\n",
      "  7450K .......... .......... .......... .......... .......... 31%  105M 1s\n",
      "  7500K .......... .......... .......... .......... .......... 32%  117M 1s\n",
      "  7550K .......... .......... .......... .......... .......... 32%  123M 1s\n",
      "  7600K .......... .......... .......... .......... .......... 32%  102M 1s\n",
      "  7650K .......... .......... .......... .......... .......... 32%  117M 1s\n",
      "  7700K .......... .......... .......... .......... .......... 32%  102M 1s\n",
      "  7750K .......... .......... .......... .......... .......... 33%  121M 1s\n",
      "  7800K .......... .......... .......... .......... .......... 33%  117M 1s\n",
      "  7850K .......... .......... .......... .......... .......... 33%  102M 1s\n",
      "  7900K .......... .......... .......... .......... .......... 33%  122M 1s\n",
      "  7950K .......... .......... .......... .......... .......... 33%  102M 1s\n",
      "  8000K .......... .......... .......... .......... .......... 34%  121M 1s\n",
      "  8050K .......... .......... .......... .......... .......... 34%  117M 1s\n",
      "  8100K .......... .......... .......... .......... .......... 34%  103M 1s\n",
      "  8150K .......... .......... .......... .......... .......... 34%  120M 1s\n",
      "  8200K .......... .......... .......... .......... .......... 35%  104M 1s\n",
      "  8250K .......... .......... .......... .......... .......... 35%  117M 1s\n",
      "  8300K .......... .......... .......... .......... .......... 35%  120M 1s\n",
      "  8350K .......... .......... .......... .......... .......... 35%  103M 1s\n",
      "  8400K .......... .......... .......... .......... .......... 35%  117M 1s\n",
      "  8450K .......... .......... .......... .......... .......... 36%  102M 1s\n",
      "  8500K .......... .......... .......... .......... .......... 36%  121M 1s\n",
      "  8550K .......... .......... .......... .......... .......... 36%  121M 1s\n",
      "  8600K .......... .......... .......... .......... .......... 36%  102M 1s\n",
      "  8650K .......... .......... .......... .......... .......... 36%  117M 1s\n",
      "  8700K .......... .......... .......... .......... .......... 37%  102M 1s\n",
      "  8750K .......... .......... .......... .......... .......... 37%  121M 1s\n",
      "  8800K .......... .......... .......... .......... .......... 37%  117M 1s\n",
      "  8850K .......... .......... .......... .......... .......... 37%  102M 1s\n",
      "  8900K .......... .......... .......... .......... .......... 38%  121M 1s\n",
      "  8950K .......... .......... .......... .......... .......... 38%  102M 1s\n",
      "  9000K .......... .......... .......... .......... .......... 38%  117M 1s\n",
      "  9050K .......... .......... .......... .......... .......... 38%  120M 1s\n",
      "  9100K .......... .......... .......... .......... .......... 38%  103M 1s\n",
      "  9150K .......... .......... .......... .......... .......... 39%  120M 1s\n",
      "  9200K .......... .......... .......... .......... .......... 39%  102M 1s\n",
      "  9250K .......... .......... .......... .......... .......... 39%  118M 1s\n",
      "  9300K .......... .......... .......... .......... .......... 39%  121M 1s\n",
      "  9350K .......... .......... .......... .......... .......... 39%  102M 1s\n",
      "  9400K .......... .......... .......... .......... .......... 40%  117M 1s\n",
      "  9450K .......... .......... .......... .......... .......... 40%  102M 1s\n",
      "  9500K .......... .......... .......... .......... .......... 40%  121M 1s\n",
      "  9550K .......... .......... .......... .......... .......... 40%  121M 1s\n",
      "  9600K .......... .......... .......... .......... .......... 40%  102M 1s\n",
      "  9650K .......... .......... .......... .......... .......... 41%  117M 1s\n",
      "  9700K .......... .......... .......... .......... .......... 41%  102M 1s\n",
      "  9750K .......... .......... .......... .......... .......... 41%  121M 1s\n",
      "  9800K .......... .......... .......... .......... .......... 41%  119M 1s\n",
      "  9850K .......... .......... .......... .......... .......... 42% 1,74M 1s\n",
      "  9900K .......... .......... .......... .......... .......... 42% 21,9M 1s\n",
      "  9950K .......... .......... .......... .......... .......... 42% 29,7M 1s\n",
      " 10000K .......... .......... .......... .......... .......... 42%  122M 1s\n",
      " 10050K .......... .......... .......... .......... .......... 42%  120M 1s\n",
      " 10100K .......... .......... .......... .......... .......... 43%  103M 1s\n",
      " 10150K .......... .......... .......... .......... .......... 43%  117M 1s\n",
      " 10200K .......... .......... .......... .......... .......... 43%  104M 1s\n",
      " 10250K .......... .......... .......... .......... .......... 43%  117M 1s\n",
      " 10300K .......... .......... .......... .......... .......... 43%  119M 1s\n",
      " 10350K .......... .......... .......... .......... .......... 44% 77,0M 1s\n",
      " 10400K .......... .......... .......... .......... .......... 44%  144M 1s\n",
      " 10450K .......... .......... .......... .......... .......... 44%  122M 1s\n",
      " 10500K .......... .......... .......... .......... .......... 44%  124M 1s\n",
      " 10550K .......... .......... .......... .......... .......... 45%  114M 1s\n",
      " 10600K .......... .......... .......... .......... .......... 45%  105M 1s\n",
      " 10650K .......... .......... .......... .......... .......... 45%  122M 1s\n",
      " 10700K .......... .......... .......... .......... .......... 45%  103M 1s\n",
      " 10750K .......... .......... .......... .......... .......... 45%  118M 1s\n",
      " 10800K .......... .......... .......... .......... .......... 46%  116M 1s\n",
      " 10850K .......... .......... .......... .......... .......... 46%  104M 1s\n",
      " 10900K .......... .......... .......... .......... .......... 46%  121M 1s\n",
      " 10950K .......... .......... .......... .......... .......... 46%  102M 1s\n",
      " 11000K .......... .......... .......... .......... .......... 46%  119M 1s\n",
      " 11050K .......... .......... .......... .......... .......... 47%  124M 1s\n",
      " 11100K .......... .......... .......... .......... .......... 47%  101M 1s\n",
      " 11150K .......... .......... .......... .......... .......... 47%  116M 1s\n",
      " 11200K .......... .......... .......... .......... .......... 47%  105M 1s\n",
      " 11250K .......... .......... .......... .......... .......... 48%  117M 1s\n",
      " 11300K .......... .......... .......... .......... .......... 48%  121M 1s\n",
      " 11350K .......... .......... .......... .......... .......... 48%  103M 1s\n",
      " 11400K .......... .......... .......... .......... .......... 48%  117M 1s\n",
      " 11450K .......... .......... .......... .......... .......... 48%  100M 1s\n",
      " 11500K .......... .......... .......... .......... .......... 49%  124M 1s\n",
      " 11550K .......... .......... .......... .......... .......... 49%  123M 1s\n",
      " 11600K .......... .......... .......... .......... .......... 49%  102M 1s\n",
      " 11650K .......... .......... .......... .......... .......... 49%  122M 1s\n",
      " 11700K .......... .......... .......... .......... .......... 49%  102M 1s\n",
      " 11750K .......... .......... .......... .......... .......... 50%  122M 1s\n",
      " 11800K .......... .......... .......... .......... .......... 50%  116M 1s\n",
      " 11850K .......... .......... .......... .......... .......... 50%  102M 1s\n",
      " 11900K .......... .......... .......... .......... .......... 50%  121M 1s\n",
      " 11950K .......... .......... .......... .......... .......... 50%  102M 1s\n",
      " 12000K .......... .......... .......... .......... .......... 51%  118M 1s\n",
      " 12050K .......... .......... .......... .......... .......... 51%  120M 1s\n",
      " 12100K .......... .......... .......... .......... .......... 51%  103M 1s\n",
      " 12150K .......... .......... .......... .......... .......... 51%  117M 1s\n",
      " 12200K .......... .......... .......... .......... .......... 52%  102M 1s\n",
      " 12250K .......... .......... .......... .......... .......... 52%  121M 1s\n",
      " 12300K .......... .......... .......... .......... .......... 52%  121M 1s\n",
      " 12350K .......... .......... .......... .......... .......... 52%  102M 1s\n",
      " 12400K .......... .......... .......... .......... .......... 52%  117M 1s\n",
      " 12450K .......... .......... .......... .......... .......... 53%  103M 1s\n",
      " 12500K .......... .......... .......... .......... .......... 53%  121M 1s\n",
      " 12550K .......... .......... .......... .......... .......... 53%  117M 1s\n",
      " 12600K .......... .......... .......... .......... .......... 53%  102M 1s\n",
      " 12650K .......... .......... .......... .......... .......... 53%  121M 1s\n",
      " 12700K .......... .......... .......... .......... .......... 54%  103M 0s\n",
      " 12750K .......... .......... .......... .......... .......... 54%  117M 0s\n",
      " 12800K .......... .......... .......... .......... .......... 54%  121M 0s\n",
      " 12850K .......... .......... .......... .......... .......... 54%  102M 0s\n",
      " 12900K .......... .......... .......... .......... .......... 55% 1,90M 1s\n",
      " 12950K .......... .......... .......... .......... .......... 55% 13,8M 1s\n",
      " 13000K .......... .......... .......... .......... .......... 55% 39,1M 0s\n",
      " 13050K .......... .......... .......... .......... .......... 55% 28,9M 0s\n",
      " 13100K .......... .......... .......... .......... .......... 55% 88,7M 0s\n",
      " 13150K .......... .......... .......... .......... .......... 56%  147M 0s\n",
      " 13200K .......... .......... .......... .......... .......... 56% 99,3M 0s\n",
      " 13250K .......... .......... .......... .......... .......... 56%  119M 0s\n",
      " 13300K .......... .......... .......... .......... .......... 56%  128M 0s\n",
      " 13350K .......... .......... .......... .......... .......... 56% 98,4M 0s\n",
      " 13400K .......... .......... .......... .......... .......... 57%  122M 0s\n",
      " 13450K .......... .......... .......... .......... .......... 57%  102M 0s\n",
      " 13500K .......... .......... .......... .......... .......... 57%  120M 0s\n",
      " 13550K .......... .......... .......... .......... .......... 57%  118M 0s\n",
      " 13600K .......... .......... .......... .......... .......... 57%  103M 0s\n",
      " 13650K .......... .......... .......... .......... .......... 58%  124M 0s\n",
      " 13700K .......... .......... .......... .......... .......... 58%  101M 0s\n",
      " 13750K .......... .......... .......... .......... .......... 58%  114M 0s\n",
      " 13800K .......... .......... .......... .......... .......... 58%  124M 0s\n",
      " 13850K .......... .......... .......... .......... .......... 59%  103M 0s\n",
      " 13900K .......... .......... .......... .......... .......... 59%  121M 0s\n",
      " 13950K .......... .......... .......... .......... .......... 59%  102M 0s\n",
      " 14000K .......... .......... .......... .......... .......... 59%  117M 0s\n",
      " 14050K .......... .......... .......... .......... .......... 59%  121M 0s\n",
      " 14100K .......... .......... .......... .......... .......... 60%  102M 0s\n",
      " 14150K .......... .......... .......... .......... .......... 60%  119M 0s\n",
      " 14200K .......... .......... .......... .......... .......... 60%  103M 0s\n",
      " 14250K .......... .......... .......... .......... .......... 60%  120M 0s\n",
      " 14300K .......... .......... .......... .......... .......... 60%  117M 0s\n",
      " 14350K .......... .......... .......... .......... .......... 61%  103M 0s\n",
      " 14400K .......... .......... .......... .......... .......... 61%  115M 0s\n",
      " 14450K .......... .......... .......... .......... .......... 61%  106M 0s\n",
      " 14500K .......... .......... .......... .......... .......... 61%  122M 0s\n",
      " 14550K .......... .......... .......... .......... .......... 62%  117M 0s\n",
      " 14600K .......... .......... .......... .......... .......... 62%  102M 0s\n",
      " 14650K .......... .......... .......... .......... .......... 62%  122M 0s\n",
      " 14700K .......... .......... .......... .......... .......... 62%  102M 0s\n",
      " 14750K .......... .......... .......... .......... .......... 62%  117M 0s\n",
      " 14800K .......... .......... .......... .......... .......... 63%  122M 0s\n",
      " 14850K .......... .......... .......... .......... .......... 63%  102M 0s\n",
      " 14900K .......... .......... .......... .......... .......... 63%  117M 0s\n",
      " 14950K .......... .......... .......... .......... .......... 63%  105M 0s\n",
      " 15000K .......... .......... .......... .......... .......... 63%  117M 0s\n",
      " 15050K .......... .......... .......... .......... .......... 64%  122M 0s\n",
      " 15100K .......... .......... .......... .......... .......... 64%  102M 0s\n",
      " 15150K .......... .......... .......... .......... .......... 64%  118M 0s\n",
      " 15200K .......... .......... .......... .......... .......... 64%  102M 0s\n",
      " 15250K .......... .......... .......... .......... .......... 64%  122M 0s\n",
      " 15300K .......... .......... .......... .......... .......... 65%  117M 0s\n",
      " 15350K .......... .......... .......... .......... .......... 65%  102M 0s\n",
      " 15400K .......... .......... .......... .......... .......... 65%  122M 0s\n",
      " 15450K .......... .......... .......... .......... .......... 65%  102M 0s\n",
      " 15500K .......... .......... .......... .......... .......... 66%  121M 0s\n",
      " 15550K .......... .......... .......... .......... .......... 66%  117M 0s\n",
      " 15600K .......... .......... .......... .......... .......... 66%  102M 0s\n",
      " 15650K .......... .......... .......... .......... .......... 66%  121M 0s\n",
      " 15700K .......... .......... .......... .......... .......... 66%  102M 0s\n",
      " 15750K .......... .......... .......... .......... .......... 67%  117M 0s\n",
      " 15800K .......... .......... .......... .......... .......... 67%  122M 0s\n",
      " 15850K .......... .......... .......... .......... .......... 67%  102M 0s\n",
      " 15900K .......... .......... .......... .......... .......... 67%  117M 0s\n",
      " 15950K .......... .......... .......... .......... .......... 67% 4,15M 0s\n",
      " 16000K .......... .......... .......... .......... .......... 68% 2,91M 0s\n",
      " 16050K .......... .......... .......... .......... .......... 68% 36,7M 0s\n",
      " 16100K .......... .......... .......... .......... .......... 68% 27,5M 0s\n",
      " 16150K .......... .......... .......... .......... .......... 68% 44,1M 0s\n",
      " 16200K .......... .......... .......... .......... .......... 69%  102M 0s\n",
      " 16250K .......... .......... .......... .......... .......... 69%  121M 0s\n",
      " 16300K .......... .......... .......... .......... .......... 69%  117M 0s\n",
      " 16350K .......... .......... .......... .......... .......... 69%  101M 0s\n",
      " 16400K .......... .......... .......... .......... .......... 69%  121M 0s\n",
      " 16450K .......... .......... .......... .......... .......... 70% 88,7M 0s\n",
      " 16500K .......... .......... .......... .......... .......... 70%  142M 0s\n",
      " 16550K .......... .......... .......... .......... .......... 70%  121M 0s\n",
      " 16600K .......... .......... .......... .......... .......... 70%  103M 0s\n",
      " 16650K .......... .......... .......... .......... .......... 70%  121M 0s\n",
      " 16700K .......... .......... .......... .......... .......... 71%  103M 0s\n",
      " 16750K .......... .......... .......... .......... .......... 71%  117M 0s\n",
      " 16800K .......... .......... .......... .......... .......... 71%  121M 0s\n",
      " 16850K .......... .......... .......... .......... .......... 71%  102M 0s\n",
      " 16900K .......... .......... .......... .......... .......... 72%  117M 0s\n",
      " 16950K .......... .......... .......... .......... .......... 72%  102M 0s\n",
      " 17000K .......... .......... .......... .......... .......... 72%  120M 0s\n",
      " 17050K .......... .......... .......... .......... .......... 72%  124M 0s\n",
      " 17100K .......... .......... .......... .......... .......... 72%  103M 0s\n",
      " 17150K .......... .......... .......... .......... .......... 73%  117M 0s\n",
      " 17200K .......... .......... .......... .......... .......... 73%  103M 0s\n",
      " 17250K .......... .......... .......... .......... .......... 73%  121M 0s\n",
      " 17300K .......... .......... .......... .......... .......... 73%  115M 0s\n",
      " 17350K .......... .......... .......... .......... .......... 73%  103M 0s\n",
      " 17400K .......... .......... .......... .......... .......... 74%  121M 0s\n",
      " 17450K .......... .......... .......... .......... .......... 74%  103M 0s\n",
      " 17500K .......... .......... .......... .......... .......... 74%  117M 0s\n",
      " 17550K .......... .......... .......... .......... .......... 74% 86,2M 0s\n",
      " 17600K .......... .......... .......... .......... .......... 74%  155M 0s\n",
      " 17650K .......... .......... .......... .......... .......... 75%  122M 0s\n",
      " 17700K .......... .......... .......... .......... .......... 75%  102M 0s\n",
      " 17750K .......... .......... .......... .......... .......... 75%  117M 0s\n",
      " 17800K .......... .......... .......... .......... .......... 75%  121M 0s\n",
      " 17850K .......... .......... .......... .......... .......... 76%  102M 0s\n",
      " 17900K .......... .......... .......... .......... .......... 76%  118M 0s\n",
      " 17950K .......... .......... .......... .......... .......... 76%  103M 0s\n",
      " 18000K .......... .......... .......... .......... .......... 76%  120M 0s\n",
      " 18050K .......... .......... .......... .......... .......... 76%  117M 0s\n",
      " 18100K .......... .......... .......... .......... .......... 77%  103M 0s\n",
      " 18150K .......... .......... .......... .......... .......... 77%  120M 0s\n",
      " 18200K .......... .......... .......... .......... .......... 77%  103M 0s\n",
      " 18250K .......... .......... .......... .......... .......... 77%  120M 0s\n",
      " 18300K .......... .......... .......... .......... .......... 77%  117M 0s\n",
      " 18350K .......... .......... .......... .......... .......... 78%  102M 0s\n",
      " 18400K .......... .......... .......... .......... .......... 78%  121M 0s\n",
      " 18450K .......... .......... .......... .......... .......... 78%  101M 0s\n",
      " 18500K .......... .......... .......... .......... .......... 78%  118M 0s\n",
      " 18550K .......... .......... .......... .......... .......... 79%  121M 0s\n",
      " 18600K .......... .......... .......... .......... .......... 79%  102M 0s\n",
      " 18650K .......... .......... .......... .......... .......... 79%  117M 0s\n",
      " 18700K .......... .......... .......... .......... .......... 79%  105M 0s\n",
      " 18750K .......... .......... .......... .......... .......... 79%  117M 0s\n",
      " 18800K .......... .......... .......... .......... .......... 80%  121M 0s\n",
      " 18850K .......... .......... .......... .......... .......... 80%  102M 0s\n",
      " 18900K .......... .......... .......... .......... .......... 80%  117M 0s\n",
      " 18950K .......... .......... .......... .......... .......... 80%  102M 0s\n",
      " 19000K .......... .......... .......... .......... .......... 80% 4,60M 0s\n",
      " 19050K .......... .......... .......... .......... .......... 81% 2,71M 0s\n",
      " 19100K .......... .......... .......... .......... .......... 81% 49,0M 0s\n",
      " 19150K .......... .......... .......... .......... .......... 81%  118M 0s\n",
      " 19200K .......... .......... .......... .......... .......... 81% 18,2M 0s\n",
      " 19250K .......... .......... .......... .......... .......... 81%  121M 0s\n",
      " 19300K .......... .......... .......... .......... .......... 82%  116M 0s\n",
      " 19350K .......... .......... .......... .......... .......... 82%  103M 0s\n",
      " 19400K .......... .......... .......... .......... .......... 82%  122M 0s\n",
      " 19450K .......... .......... .......... .......... .......... 82%  102M 0s\n",
      " 19500K .......... .......... .......... .......... .......... 83%  117M 0s\n",
      " 19550K .......... .......... .......... .......... .......... 83%  121M 0s\n",
      " 19600K .......... .......... .......... .......... .......... 83% 60,4M 0s\n",
      " 19650K .......... .......... .......... .......... .......... 83%  118M 0s\n",
      " 19700K .......... .......... .......... .......... .......... 83%  102M 0s\n",
      " 19750K .......... .......... .......... .......... .......... 84%  123M 0s\n",
      " 19800K .......... .......... .......... .......... .......... 84%  122M 0s\n",
      " 19850K .......... .......... .......... .......... .......... 84%  102M 0s\n",
      " 19900K .......... .......... .......... .......... .......... 84%  117M 0s\n",
      " 19950K .......... .......... .......... .......... .......... 84%  103M 0s\n",
      " 20000K .......... .......... .......... .......... .......... 85%  122M 0s\n",
      " 20050K .......... .......... .......... .......... .......... 85%  118M 0s\n",
      " 20100K .......... .......... .......... .......... .......... 85% 97,5M 0s\n",
      " 20150K .......... .......... .......... .......... .......... 85%  126M 0s\n",
      " 20200K .......... .......... .......... .......... .......... 86%  102M 0s\n",
      " 20250K .......... .......... .......... .......... .......... 86%  116M 0s\n",
      " 20300K .......... .......... .......... .......... .......... 86%  120M 0s\n",
      " 20350K .......... .......... .......... .......... .......... 86%  100M 0s\n",
      " 20400K .......... .......... .......... .......... .......... 86%  125M 0s\n",
      " 20450K .......... .......... .......... .......... .......... 87%  104M 0s\n",
      " 20500K .......... .......... .......... .......... .......... 87%  118M 0s\n",
      " 20550K .......... .......... .......... .......... .......... 87%  101M 0s\n",
      " 20600K .......... .......... .......... .......... .......... 87%  123M 0s\n",
      " 20650K .......... .......... .......... .......... .......... 87%  118M 0s\n",
      " 20700K .......... .......... .......... .......... .......... 88%  102M 0s\n",
      " 20750K .......... .......... .......... .......... .......... 88%  121M 0s\n",
      " 20800K .......... .......... .......... .......... .......... 88%  118M 0s\n",
      " 20850K .......... .......... .......... .......... .......... 88%  104M 0s\n",
      " 20900K .......... .......... .......... .......... .......... 88%  117M 0s\n",
      " 20950K .......... .......... .......... .......... .......... 89%  103M 0s\n",
      " 21000K .......... .......... .......... .......... .......... 89%  122M 0s\n",
      " 21050K .......... .......... .......... .......... .......... 89%  107M 0s\n",
      " 21100K .......... .......... .......... .......... .......... 89%  110M 0s\n",
      " 21150K .......... .......... .......... .......... .......... 90% 92,2M 0s\n",
      " 21200K .......... .......... .......... .......... .......... 90%  142M 0s\n",
      " 21250K .......... .......... .......... .......... .......... 90%  117M 0s\n",
      " 21300K .......... .......... .......... .......... .......... 90%  122M 0s\n",
      " 21350K .......... .......... .......... .......... .......... 90%  102M 0s\n",
      " 21400K .......... .......... .......... .......... .......... 91%  122M 0s\n",
      " 21450K .......... .......... .......... .......... .......... 91%  103M 0s\n",
      " 21500K .......... .......... .......... .......... .......... 91%  117M 0s\n",
      " 21550K .......... .......... .......... .......... .......... 91%  120M 0s\n",
      " 21600K .......... .......... .......... .......... .......... 91%  103M 0s\n",
      " 21650K .......... .......... .......... .......... .......... 92%  116M 0s\n",
      " 21700K .......... .......... .......... .......... .......... 92%  102M 0s\n",
      " 21750K .......... .......... .......... .......... .......... 92%  122M 0s\n",
      " 21800K .......... .......... .......... .......... .......... 92%  117M 0s\n",
      " 21850K .......... .......... .......... .......... .......... 93%  104M 0s\n",
      " 21900K .......... .......... .......... .......... .......... 93%  120M 0s\n",
      " 21950K .......... .......... .......... .......... .......... 93%  103M 0s\n",
      " 22000K .......... .......... .......... .......... .......... 93% 5,23M 0s\n",
      " 22050K .......... .......... .......... .......... .......... 93% 23,8M 0s\n",
      " 22100K .......... .......... .......... .......... .......... 94% 3,19M 0s\n",
      " 22150K .......... .......... .......... .......... .......... 94% 18,5M 0s\n",
      " 22200K .......... .......... .......... .......... .......... 94% 23,1M 0s\n",
      " 22250K .......... .......... .......... .......... .......... 94% 29,5M 0s\n",
      " 22300K .......... .......... .......... .......... .......... 94% 77,7M 0s\n",
      " 22350K .......... .......... .......... .......... .......... 95%  138M 0s\n",
      " 22400K .......... .......... .......... .......... .......... 95%  154M 0s\n",
      " 22450K .......... .......... .......... .......... .......... 95%  108M 0s\n",
      " 22500K .......... .......... .......... .......... .......... 95%  115M 0s\n",
      " 22550K .......... .......... .......... .......... .......... 96%  124M 0s\n",
      " 22600K .......... .......... .......... .......... .......... 96% 78,4M 0s\n",
      " 22650K .......... .......... .......... .......... .......... 96%  177M 0s\n",
      " 22700K .......... .......... .......... .......... .......... 96% 96,9M 0s\n",
      " 22750K .......... .......... .......... .......... .......... 96%  131M 0s\n",
      " 22800K .......... .......... .......... .......... .......... 97%  118M 0s\n",
      " 22850K .......... .......... .......... .......... .......... 97%  101M 0s\n",
      " 22900K .......... .......... .......... .......... .......... 97% 57,3M 0s\n",
      " 22950K .......... .......... .......... .......... .......... 97%  102M 0s\n",
      " 23000K .......... .......... .......... .......... .......... 97%  122M 0s\n",
      " 23050K .......... .......... .......... .......... .......... 98%  116M 0s\n",
      " 23100K .......... .......... .......... .......... .......... 98%  102M 0s\n",
      " 23150K .......... .......... .......... .......... .......... 98%  120M 0s\n",
      " 23200K .......... .......... .......... .......... .......... 98%  103M 0s\n",
      " 23250K .......... .......... .......... .......... .......... 98%  117M 0s\n",
      " 23300K .......... .......... .......... .......... .......... 99%  123M 0s\n",
      " 23350K .......... .......... .......... .......... .......... 99%  101M 0s\n",
      " 23400K .......... .......... .......... .......... .......... 99%  117M 0s\n",
      " 23450K .......... .......... .......... .......... .......... 99%  102M 0s\n",
      " 23500K .......... .......... .......... .........            100%  131M=0,8s\n",
      "\n",
      "2023-08-17 17:44:24 (28,6 MB/s) - ‘data/reference/genome.fa.gz’ guardado [24104573/24104573]\n",
      "\n",
      "[Thu Aug 17 17:44:25 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T174422.156293.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:44:26 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR696683_chr17_1.fastq.gz, data/raw/ERR696683_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR696683_chr17_1_fastqc.html, results/fastqc_result/ERR696683_chr17_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR696683_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 17:44:39 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:44:39 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR696683_chr17_1.fastq.gz, data/raw/ERR696683_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR696683_chr17_1_fastp.fastq.gz, data/processed/ERR696683_chr17_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR696683_chr17_1_fastp.fastq.gz, data/processed/ERR696683_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1662053\n",
      "total bases: 125542423\n",
      "Q20 bases: 124610599(99.2578%)\n",
      "Q30 bases: 120902230(96.3039%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1662053\n",
      "total bases: 125166559\n",
      "Q20 bases: 123458540(98.6354%)\n",
      "Q30 bases: 117801905(94.1161%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 941109\n",
      "total bases: 71474317\n",
      "Q20 bases: 71185568(99.596%)\n",
      "Q30 bases: 69857451(97.7378%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 941109\n",
      "total bases: 71445379\n",
      "Q20 bases: 70903174(99.2411%)\n",
      "Q30 bases: 68848719(96.3655%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1882218\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1441888\n",
      "reads with adapter trimmed: 1178\n",
      "bases trimmed due to adapters: 33118\n",
      "\n",
      "Duplication rate: 5.2002%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 119\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR696683_chr17_1.fastq.gz -I data/raw/ERR696683_chr17_2.fastq.gz -o data/processed/ERR696683_chr17_1_fastp.fastq.gz -O data/processed/ERR696683_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Thu Aug 17 17:44:44 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:44:44 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR696683_chr17_1_fastp.fastq.gz, data/processed/ERR696683_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR696683_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR696683_chr17_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR696683_chr17_1_fastp.fastq.gz, data/processed/ERR696683_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR696683_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR696683_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR696683_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 17:44:53 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T174425.215139.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:44:55 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753368_chr17_1.fastq.gz, data/raw/ERR753368_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753368_chr17_1_fastqc.html, results/fastqc_result/ERR753368_chr17_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753368_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753368_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 17:45:07 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:45:07 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753368_chr17_1.fastq.gz, data/raw/ERR753368_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753368_chr17_1_fastp.fastq.gz, data/processed/ERR753368_chr17_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753368_chr17_2_fastp.fastq.gz, data/processed/ERR753368_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1720078\n",
      "total bases: 130586939\n",
      "Q20 bases: 130062674(99.5985%)\n",
      "Q30 bases: 126025291(96.5068%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1720078\n",
      "total bases: 130408498\n",
      "Q20 bases: 129167916(99.0487%)\n",
      "Q30 bases: 122638409(94.0417%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1449801\n",
      "total bases: 110178424\n",
      "Q20 bases: 109981943(99.8217%)\n",
      "Q30 bases: 108017025(98.0383%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1449801\n",
      "total bases: 110167371\n",
      "Q20 bases: 109584045(99.4705%)\n",
      "Q30 bases: 105965050(96.1855%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2899602\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 540554\n",
      "reads with adapter trimmed: 1280\n",
      "bases trimmed due to adapters: 33381\n",
      "\n",
      "Duplication rate: 1.94673%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753368_chr17_1.fastq.gz -I data/raw/ERR753368_chr17_2.fastq.gz -o data/processed/ERR753368_chr17_1_fastp.fastq.gz -O data/processed/ERR753368_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Thu Aug 17 17:45:13 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:45:13 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753368_chr17_1_fastp.fastq.gz, data/processed/ERR753368_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753368_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753368_chr17_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753368_chr17_2_fastp.fastq.gz, data/processed/ERR753368_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753368_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753368_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753368_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753368_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 17:45:24 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T174453.887853.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:45:25 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753369_chr17_1.fastq.gz, data/raw/ERR753369_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753369_chr17_1_fastqc.html, results/fastqc_result/ERR753369_chr17_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753369_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753369_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 17:45:40 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:45:40 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753369_chr17_1.fastq.gz, data/raw/ERR753369_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753369_chr17_1_fastp.fastq.gz, data/processed/ERR753369_chr17_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753369_chr17_1_fastp.fastq.gz, data/processed/ERR753369_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1834306\n",
      "total bases: 139253654\n",
      "Q20 bases: 138670028(99.5809%)\n",
      "Q30 bases: 134266277(96.4185%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1834306\n",
      "total bases: 139072650\n",
      "Q20 bases: 137744780(99.0452%)\n",
      "Q30 bases: 130771387(94.031%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1545614\n",
      "total bases: 117459470\n",
      "Q20 bases: 117236110(99.8098%)\n",
      "Q30 bases: 115069005(97.9649%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1545614\n",
      "total bases: 117448151\n",
      "Q20 bases: 116824626(99.4691%)\n",
      "Q30 bases: 112958776(96.1776%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3091228\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 577384\n",
      "reads with adapter trimmed: 1210\n",
      "bases trimmed due to adapters: 30992\n",
      "\n",
      "Duplication rate: 2.23205%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753369_chr17_1.fastq.gz -I data/raw/ERR753369_chr17_2.fastq.gz -o data/processed/ERR753369_chr17_1_fastp.fastq.gz -O data/processed/ERR753369_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Thu Aug 17 17:45:47 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:45:47 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753369_chr17_1_fastp.fastq.gz, data/processed/ERR753369_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753369_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753369_chr17_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753369_chr17_1_fastp.fastq.gz, data/processed/ERR753369_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753369_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753369_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753369_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753369_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 17:45:59 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T174524.781817.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:46:01 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753370_chr17_1.fastq.gz, data/raw/ERR753370_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753370_chr17_1_fastp.fastq.gz, data/processed/ERR753370_chr17_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753370_chr17_1_fastp.fastq.gz, data/processed/ERR753370_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 2195518\n",
      "total bases: 220903960\n",
      "Q20 bases: 219316637(99.2814%)\n",
      "Q30 bases: 212394069(96.1477%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2195518\n",
      "total bases: 220842678\n",
      "Q20 bases: 218611707(98.9898%)\n",
      "Q30 bases: 209497343(94.8627%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 2194882\n",
      "total bases: 219845045\n",
      "Q20 bases: 218360012(99.3245%)\n",
      "Q30 bases: 211791798(96.3369%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 2194882\n",
      "total bases: 219058701\n",
      "Q20 bases: 217044793(99.0807%)\n",
      "Q30 bases: 208534267(95.1956%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 4389764\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1272\n",
      "reads with adapter trimmed: 11652\n",
      "bases trimmed due to adapters: 109912\n",
      "\n",
      "Duplication rate: 4.00859%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 157\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753370_chr17_1.fastq.gz -I data/raw/ERR753370_chr17_2.fastq.gz -o data/processed/ERR753370_chr17_1_fastp.fastq.gz -O data/processed/ERR753370_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 10 seconds\n",
      "[Thu Aug 17 17:46:11 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:46:11 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753370_chr17_1_fastp.fastq.gz, data/processed/ERR753370_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753370_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753370_chr17_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753370_chr17_1_fastp.fastq.gz, data/processed/ERR753370_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753370_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 17:46:31 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:46:31 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753370_chr17_1.fastq.gz, data/raw/ERR753370_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753370_chr17_1_fastqc.html, results/fastqc_result/ERR753370_chr17_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753370_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753370_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753370_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753370_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753370_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753370_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 17:46:52 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T174600.039157.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:46:53 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753371_chr17_1.fastq.gz, data/raw/ERR753371_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753371_chr17_1_fastqc.html, results/fastqc_result/ERR753371_chr17_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753371_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753371_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 17:47:06 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:47:06 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753371_chr17_1.fastq.gz, data/raw/ERR753371_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753371_chr17_1_fastp.fastq.gz, data/processed/ERR753371_chr17_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753371_chr17_2_fastp.fastq.gz, data/processed/ERR753371_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1479982\n",
      "total bases: 111397900\n",
      "Q20 bases: 110370854(99.078%)\n",
      "Q30 bases: 106159867(95.2979%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1479982\n",
      "total bases: 111084814\n",
      "Q20 bases: 109633816(98.6938%)\n",
      "Q30 bases: 105347657(94.8353%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 690305\n",
      "total bases: 52398376\n",
      "Q20 bases: 52131601(99.4909%)\n",
      "Q30 bases: 50765095(96.883%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 690305\n",
      "total bases: 52368274\n",
      "Q20 bases: 51965373(99.2306%)\n",
      "Q30 bases: 50683984(96.7838%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1380610\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1579354\n",
      "reads with adapter trimmed: 980\n",
      "bases trimmed due to adapters: 24008\n",
      "\n",
      "Duplication rate: 19.606%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 117\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753371_chr17_1.fastq.gz -I data/raw/ERR753371_chr17_2.fastq.gz -o data/processed/ERR753371_chr17_1_fastp.fastq.gz -O data/processed/ERR753371_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Thu Aug 17 17:47:11 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:47:11 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753371_chr17_1_fastp.fastq.gz, data/processed/ERR753371_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753371_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753371_chr17_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753371_chr17_2_fastp.fastq.gz, data/processed/ERR753371_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753371_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753371_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753371_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753371_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 17:47:18 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T174652.587077.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:47:19 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753372_chr17_1.fastq.gz, data/raw/ERR753372_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753372_chr17_1_fastp.fastq.gz, data/processed/ERR753372_chr17_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753372_chr17_2_fastp.fastq.gz, data/processed/ERR753372_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1891163\n",
      "total bases: 143570978\n",
      "Q20 bases: 142991472(99.5964%)\n",
      "Q30 bases: 138476280(96.4514%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1891163\n",
      "total bases: 143376057\n",
      "Q20 bases: 142013761(99.0498%)\n",
      "Q30 bases: 134881566(94.0754%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1589698\n",
      "total bases: 120809730\n",
      "Q20 bases: 120591835(99.8196%)\n",
      "Q30 bases: 118389976(97.9971%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1589698\n",
      "total bases: 120797532\n",
      "Q20 bases: 120159672(99.472%)\n",
      "Q30 bases: 116232633(96.221%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3179396\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 602930\n",
      "reads with adapter trimmed: 1446\n",
      "bases trimmed due to adapters: 35677\n",
      "\n",
      "Duplication rate: 1.89044%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753372_chr17_1.fastq.gz -I data/raw/ERR753372_chr17_2.fastq.gz -o data/processed/ERR753372_chr17_1_fastp.fastq.gz -O data/processed/ERR753372_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 6 seconds\n",
      "[Thu Aug 17 17:47:26 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:47:26 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753372_chr17_1_fastp.fastq.gz, data/processed/ERR753372_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753372_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753372_chr17_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753372_chr17_2_fastp.fastq.gz, data/processed/ERR753372_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753372_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 17:47:39 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:47:39 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753372_chr17_1.fastq.gz, data/raw/ERR753372_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753372_chr17_1_fastqc.html, results/fastqc_result/ERR753372_chr17_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753372_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753372_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753372_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753372_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753372_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753372_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 17:47:54 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T174718.575096.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:47:55 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753373_chr17_1.fastq.gz, data/raw/ERR753373_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753373_chr17_1_fastp.fastq.gz, data/processed/ERR753373_chr17_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753373_chr17_2_fastp.fastq.gz, data/processed/ERR753373_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 843274\n",
      "total bases: 80686867\n",
      "Q20 bases: 79980120(99.1241%)\n",
      "Q30 bases: 74138420(91.8841%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 843274\n",
      "total bases: 78848808\n",
      "Q20 bases: 77912736(98.8128%)\n",
      "Q30 bases: 72303629(91.6991%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 819531\n",
      "total bases: 77028198\n",
      "Q20 bases: 76544780(99.3724%)\n",
      "Q30 bases: 71847320(93.2741%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 819531\n",
      "total bases: 74314773\n",
      "Q20 bases: 73905877(99.4498%)\n",
      "Q30 bases: 69630249(93.6964%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1639062\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 47486\n",
      "reads with adapter trimmed: 1874\n",
      "bases trimmed due to adapters: 72905\n",
      "\n",
      "Duplication rate: 1.52937%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 154\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753373_chr17_1.fastq.gz -I data/raw/ERR753373_chr17_2.fastq.gz -o data/processed/ERR753373_chr17_1_fastp.fastq.gz -O data/processed/ERR753373_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 4 seconds\n",
      "[Thu Aug 17 17:48:00 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:48:00 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753373_chr17_1.fastq.gz, data/raw/ERR753373_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753373_chr17_1_fastqc.html, results/fastqc_result/ERR753373_chr17_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753373_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753373_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 17:48:09 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:48:09 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753373_chr17_1_fastp.fastq.gz, data/processed/ERR753373_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753373_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753373_chr17_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753373_chr17_2_fastp.fastq.gz, data/processed/ERR753373_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753373_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753373_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753373_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753373_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 17:48:18 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T174754.646434.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:48:19 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753374_chr17_1.fastq.gz, data/raw/ERR753374_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753374_chr17_1_fastp.fastq.gz, data/processed/ERR753374_chr17_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753374_chr17_2_fastp.fastq.gz, data/processed/ERR753374_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 492332\n",
      "total bases: 47062824\n",
      "Q20 bases: 46581392(98.977%)\n",
      "Q30 bases: 42272899(89.8223%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 492332\n",
      "total bases: 45990975\n",
      "Q20 bases: 45452935(98.8301%)\n",
      "Q30 bases: 41465767(90.1607%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 473782\n",
      "total bases: 44142721\n",
      "Q20 bases: 43810033(99.2463%)\n",
      "Q30 bases: 40441158(91.6146%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 473782\n",
      "total bases: 42957145\n",
      "Q20 bases: 42673614(99.34%)\n",
      "Q30 bases: 39641425(92.2813%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 947564\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 37100\n",
      "reads with adapter trimmed: 1076\n",
      "bases trimmed due to adapters: 40498\n",
      "\n",
      "Duplication rate: 4.87372%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 148\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753374_chr17_1.fastq.gz -I data/raw/ERR753374_chr17_2.fastq.gz -o data/processed/ERR753374_chr17_1_fastp.fastq.gz -O data/processed/ERR753374_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 2 seconds\n",
      "[Thu Aug 17 17:48:22 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:48:22 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753374_chr17_1.fastq.gz, data/raw/ERR753374_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753374_chr17_1_fastqc.html, results/fastqc_result/ERR753374_chr17_2_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/fastqc_result/ERR753374_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753374_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 17:48:29 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:48:29 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753374_chr17_1_fastp.fastq.gz, data/processed/ERR753374_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753374_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753374_chr17_2_fastp_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Input files updated by another job: data/processed/ERR753374_chr17_2_fastp.fastq.gz, data/processed/ERR753374_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753374_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753374_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753374_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753374_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 17:48:36 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T174818.255216.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:48:37 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753375_chr17_1.fastq.gz, data/raw/ERR753375_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753375_chr17_1_fastp.fastq.gz, data/processed/ERR753375_chr17_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753375_chr17_1_fastp.fastq.gz, data/processed/ERR753375_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1026269\n",
      "total bases: 77458309\n",
      "Q20 bases: 76947062(99.34%)\n",
      "Q30 bases: 75053112(96.8948%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1026269\n",
      "total bases: 77416125\n",
      "Q20 bases: 76301088(98.5597%)\n",
      "Q30 bases: 74090313(95.704%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 615090\n",
      "total bases: 46703360\n",
      "Q20 bases: 46543091(99.6568%)\n",
      "Q30 bases: 45786092(98.036%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 615090\n",
      "total bases: 46709566\n",
      "Q20 bases: 46221428(98.955%)\n",
      "Q30 bases: 45368202(97.1283%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1230180\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 822358\n",
      "reads with adapter trimmed: 702\n",
      "bases trimmed due to adapters: 19004\n",
      "\n",
      "Duplication rate: 1.55687%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753375_chr17_1.fastq.gz -I data/raw/ERR753375_chr17_2.fastq.gz -o data/processed/ERR753375_chr17_1_fastp.fastq.gz -O data/processed/ERR753375_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 3 seconds\n",
      "[Thu Aug 17 17:48:41 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:48:41 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753375_chr17_1_fastp.fastq.gz, data/processed/ERR753375_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753375_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753375_chr17_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753375_chr17_1_fastp.fastq.gz, data/processed/ERR753375_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753375_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 17:48:48 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:48:48 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753375_chr17_1.fastq.gz, data/raw/ERR753375_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753375_chr17_1_fastqc.html, results/fastqc_result/ERR753375_chr17_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753375_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753375_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753375_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753375_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753375_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753375_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 17:48:57 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T174836.365645.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:48:58 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753376_chr17_1.fastq.gz, data/raw/ERR753376_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753376_chr17_1_fastqc.html, results/fastqc_result/ERR753376_chr17_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753376_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753376_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 17:49:09 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:49:09 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753376_chr17_1.fastq.gz, data/raw/ERR753376_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753376_chr17_1_fastp.fastq.gz, data/processed/ERR753376_chr17_2_fastp.fastq.gz\n",
      "    jobid: 1\n",
      "    reason: Missing output files: data/processed/ERR753376_chr17_2_fastp.fastq.gz, data/processed/ERR753376_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1319555\n",
      "total bases: 99673497\n",
      "Q20 bases: 98950424(99.2746%)\n",
      "Q30 bases: 96007070(96.3216%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1319555\n",
      "total bases: 99357655\n",
      "Q20 bases: 97988993(98.6225%)\n",
      "Q30 bases: 93463715(94.068%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 743097\n",
      "total bases: 56436043\n",
      "Q20 bases: 56214517(99.6075%)\n",
      "Q30 bases: 55168884(97.7547%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 743097\n",
      "total bases: 56414017\n",
      "Q20 bases: 55982344(99.2348%)\n",
      "Q30 bases: 54346819(96.3357%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1486194\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1152916\n",
      "reads with adapter trimmed: 998\n",
      "bases trimmed due to adapters: 26275\n",
      "\n",
      "Duplication rate: 8.00792%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 119\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753376_chr17_1.fastq.gz -I data/raw/ERR753376_chr17_2.fastq.gz -o data/processed/ERR753376_chr17_1_fastp.fastq.gz -O data/processed/ERR753376_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 3 seconds\n",
      "[Thu Aug 17 17:49:13 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:49:13 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753376_chr17_1_fastp.fastq.gz, data/processed/ERR753376_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753376_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753376_chr17_2_fastp_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: data/processed/ERR753376_chr17_2_fastp.fastq.gz, data/processed/ERR753376_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753376_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753376_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753376_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753376_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 17:49:21 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T174857.351677.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:49:23 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753377_chr17_1.fastq.gz, data/raw/ERR753377_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753377_chr17_1_fastp.fastq.gz, data/processed/ERR753377_chr17_2_fastp.fastq.gz\n",
      "    jobid: 0\n",
      "    reason: Missing output files: data/processed/ERR753377_chr17_1_fastp.fastq.gz, data/processed/ERR753377_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1385666\n",
      "total bases: 104679071\n",
      "Q20 bases: 103910027(99.2653%)\n",
      "Q30 bases: 100812888(96.3066%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1385666\n",
      "total bases: 104385271\n",
      "Q20 bases: 103014264(98.6866%)\n",
      "Q30 bases: 98468698(94.332%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 800065\n",
      "total bases: 60762969\n",
      "Q20 bases: 60520165(99.6004%)\n",
      "Q30 bases: 59388451(97.7379%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 800065\n",
      "total bases: 60740129\n",
      "Q20 bases: 60296945(99.2704%)\n",
      "Q30 bases: 58618077(96.5063%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1600130\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1171202\n",
      "reads with adapter trimmed: 1020\n",
      "bases trimmed due to adapters: 26943\n",
      "\n",
      "Duplication rate: 8.08439%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 120\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753377_chr17_1.fastq.gz -I data/raw/ERR753377_chr17_2.fastq.gz -o data/processed/ERR753377_chr17_1_fastp.fastq.gz -O data/processed/ERR753377_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Thu Aug 17 17:49:28 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:49:28 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753377_chr17_1_fastp.fastq.gz, data/processed/ERR753377_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753377_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753377_chr17_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753377_chr17_1_fastp.fastq.gz, data/processed/ERR753377_chr17_2_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753377_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 17:49:36 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:49:36 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753377_chr17_1.fastq.gz, data/raw/ERR753377_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753377_chr17_1_fastqc.html, results/fastqc_result/ERR753377_chr17_2_fastqc.html\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/fastqc_result/ERR753377_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753377_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753377_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753377_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753377_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753377_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 17:49:49 2023]\n",
      "Finished job 2.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T174921.955273.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job               count    min threads    max threads\n",
      "--------------  -------  -------------  -------------\n",
      "fastp                 1              1              1\n",
      "fastqc                1              1              1\n",
      "fastqc_trimmed        1              1              1\n",
      "total                 3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:49:51 2023]\n",
      "rule fastqc:\n",
      "    input: data/raw/ERR753378_chr17_1.fastq.gz, data/raw/ERR753378_chr17_2.fastq.gz\n",
      "    output: results/fastqc_result/ERR753378_chr17_1_fastqc.html, results/fastqc_result/ERR753378_chr17_2_fastqc.html\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/fastqc_result/ERR753378_chr17_1_fastqc.html\n",
      "    wildcards: sample=ERR753378_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr17_1.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr17_1.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr17_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr17_1.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr17_2.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr17_2.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr17_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 17:50:04 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:50:04 2023]\n",
      "rule fastp:\n",
      "    input: data/raw/ERR753378_chr17_1.fastq.gz, data/raw/ERR753378_chr17_2.fastq.gz\n",
      "    output: data/processed/ERR753378_chr17_1_fastp.fastq.gz, data/processed/ERR753378_chr17_2_fastp.fastq.gz\n",
      "    jobid: 2\n",
      "    reason: Missing output files: data/processed/ERR753378_chr17_2_fastp.fastq.gz, data/processed/ERR753378_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "Read1 before filtering:\n",
      "total reads: 1654676\n",
      "total bases: 125620372\n",
      "Q20 bases: 125114317(99.5972%)\n",
      "Q30 bases: 121181168(96.4662%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1654676\n",
      "total bases: 125447036\n",
      "Q20 bases: 124251291(99.0468%)\n",
      "Q30 bases: 118001271(94.0646%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1397009\n",
      "total bases: 106166313\n",
      "Q20 bases: 105973483(99.8184%)\n",
      "Q30 bases: 104029622(97.9874%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 1397009\n",
      "total bases: 106155410\n",
      "Q20 bases: 105589103(99.4665%)\n",
      "Q30 bases: 102096673(96.1766%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2794018\n",
      "reads failed due to low quality: 0\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 515334\n",
      "reads with adapter trimmed: 1108\n",
      "bases trimmed due to adapters: 27200\n",
      "\n",
      "Duplication rate: 1.89109%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 121\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i data/raw/ERR753378_chr17_1.fastq.gz -I data/raw/ERR753378_chr17_2.fastq.gz -o data/processed/ERR753378_chr17_1_fastp.fastq.gz -O data/processed/ERR753378_chr17_2_fastp.fastq.gz --cut_tail 30 --cut_front 30 --cut_mean_quality 30 -l 75 \n",
      "fastp v0.22.0, time used: 5 seconds\n",
      "[Thu Aug 17 17:50:09 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:50:09 2023]\n",
      "rule fastqc_trimmed:\n",
      "    input: data/processed/ERR753378_chr17_1_fastp.fastq.gz, data/processed/ERR753378_chr17_2_fastp.fastq.gz\n",
      "    output: results/fastqc_result/trimmed/ERR753378_chr17_1_fastp_fastqc.html, results/fastqc_result/trimmed/ERR753378_chr17_2_fastp_fastqc.html\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: data/processed/ERR753378_chr17_2_fastp.fastq.gz, data/processed/ERR753378_chr17_1_fastp.fastq.gz\n",
      "    wildcards: sample=ERR753378_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "Approx 100% complete for ERR753378_chr17_1_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr17_1_fastp.fastq.gz\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 5% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 10% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 15% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 20% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 25% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 30% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 35% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 40% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 45% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 50% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 55% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 60% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 65% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 70% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 75% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 80% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 85% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 90% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 95% complete for ERR753378_chr17_2_fastp.fastq.gz\n",
      "Approx 100% complete for ERR753378_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR753378_chr17_2_fastp.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 17:50:20 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T174949.904081.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:50:22 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR696683_chr17_1_fastp.fastq.gz, data/processed/ERR696683_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR696683_chr17.sam, results/mapped_reads/ERR696683_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR696683_chr17_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR696683_chr17_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.29 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 27.02 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.26 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.18 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 15.77 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 43.863 sec; CPU: 43.515 sec\n",
      "[Thu Aug 17 17:52:21 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:52:21 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR696683_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr17.bam, results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR696683_chr17.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR696683_chr17_sorted.sam\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 17:52:54 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:52:54 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR696683_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR696683_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "17:52:55.737 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 17:52:55 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR696683_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR696683_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 17:52:55 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 17:52:55\tMarkDuplicates\tStart of doWork freeMemory: 528790720; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 17:52:55\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 17:52:55\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam\n",
      "INFO\t2023-08-17 17:52:57\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:40,133,231\n",
      "INFO\t2023-08-17 17:52:57\tMarkDuplicates\tTracking 320 as yet unmatched pairs. 320 records in RAM.\n",
      "INFO\t2023-08-17 17:52:58\tMarkDuplicates\tRead 1926106 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 17:52:58\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 264951528; totalMemory: 540016640; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 17:52:58\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 17:52:58\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 17:52:59\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 17:52:59\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 17:52:59\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 822038832; totalMemory: 1368391680; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 17:52:59\tMarkDuplicates\tMarking 153600 records as duplicates.\n",
      "INFO\t2023-08-17 17:52:59\tMarkDuplicates\tFound 63 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR696683_chr17_sorted.bam\n",
      "INFO\t2023-08-17 17:52:59\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 17:53:11\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 17:53:11\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 17:53:11\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 17:53:11\tMarkDuplicates\tBefore output close freeMemory: 528088248; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 17:53:11\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 17:53:11\tMarkDuplicates\tAfter output close freeMemory: 528088248; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 17:53:11 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.27 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 17:53:12 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T175020.888932.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:53:14 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753368_chr17_1_fastp.fastq.gz, data/processed/ERR753368_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753368_chr17.sam, results/mapped_reads/ERR753368_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753368_chr17_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR753368_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.28 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 27.30 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.26 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.19 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 15.83 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 44.288 sec; CPU: 43.859 sec\n",
      "[Thu Aug 17 17:55:48 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:55:48 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753368_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr17.bam, results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753368_chr17.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753368_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753368_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 17:56:32 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:56:32 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753368_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753368_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753368_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "17:56:33.789 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 17:56:33 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753368_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753368_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 17:56:33 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 17:56:33\tMarkDuplicates\tStart of doWork freeMemory: 528789328; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 17:56:33\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 17:56:33\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam\n",
      "INFO\t2023-08-17 17:56:35\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:28,617,150\n",
      "INFO\t2023-08-17 17:56:35\tMarkDuplicates\tTracking 428 as yet unmatched pairs. 428 records in RAM.\n",
      "INFO\t2023-08-17 17:56:36\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 17:50,643,460\n",
      "INFO\t2023-08-17 17:56:36\tMarkDuplicates\tTracking 337 as yet unmatched pairs. 337 records in RAM.\n",
      "INFO\t2023-08-17 17:56:38\tMarkDuplicates\tRead 2989104 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 17:56:38\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 464332216; totalMemory: 849346560; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 17:56:38\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 17:56:38\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 17:56:38\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 17:56:38\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 17:56:38\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1010781160; totalMemory: 1557135360; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 17:56:38\tMarkDuplicates\tMarking 67248 records as duplicates.\n",
      "INFO\t2023-08-17 17:56:38\tMarkDuplicates\tFound 468 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753368_chr17_sorted.bam\n",
      "INFO\t2023-08-17 17:56:38\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 17:56:55\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 17:56:55\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 17:56:55\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 17:56:55\tMarkDuplicates\tBefore output close freeMemory: 528087792; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 17:56:55\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 17:56:55\tMarkDuplicates\tAfter output close freeMemory: 528087792; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 17:56:55 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.36 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 17:56:57 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T175313.113242.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:56:58 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753369_chr17_1_fastp.fastq.gz, data/processed/ERR753369_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753369_chr17.sam, results/mapped_reads/ERR753369_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753369_chr17_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753369_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.28 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 24.91 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.26 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.18 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 14.74 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 45.182 sec; CPU: 40.362 sec\n",
      "[Thu Aug 17 17:59:44 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 17:59:44 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753369_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr17.bam, results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753369_chr17.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753369_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753369_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 18:00:30 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:00:30 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753369_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753369_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753369_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:00:31.136 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 18:00:31 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753369_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753369_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 18:00:31 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 18:00:31\tMarkDuplicates\tStart of doWork freeMemory: 528766072; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:00:31\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 18:00:31\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam\n",
      "INFO\t2023-08-17 18:00:32\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:28,638,433\n",
      "INFO\t2023-08-17 18:00:32\tMarkDuplicates\tTracking 412 as yet unmatched pairs. 412 records in RAM.\n",
      "INFO\t2023-08-17 18:00:34\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 17:49,206,538\n",
      "INFO\t2023-08-17 18:00:34\tMarkDuplicates\tTracking 294 as yet unmatched pairs. 294 records in RAM.\n",
      "INFO\t2023-08-17 18:00:35\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 17:80,386,226\n",
      "INFO\t2023-08-17 18:00:35\tMarkDuplicates\tTracking 114 as yet unmatched pairs. 114 records in RAM.\n",
      "INFO\t2023-08-17 18:00:35\tMarkDuplicates\tRead 3123173 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 18:00:35\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 453932176; totalMemory: 859832320; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:00:35\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 18:00:35\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 18:00:36\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 18:00:36\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 18:00:36\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1044312856; totalMemory: 1590689792; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:00:36\tMarkDuplicates\tMarking 82969 records as duplicates.\n",
      "INFO\t2023-08-17 18:00:36\tMarkDuplicates\tFound 436 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753369_chr17_sorted.bam\n",
      "INFO\t2023-08-17 18:00:36\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 18:00:53\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 18:00:53\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 18:00:53\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 18:00:53\tMarkDuplicates\tBefore output close freeMemory: 528064320; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:00:53\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 18:00:53\tMarkDuplicates\tAfter output close freeMemory: 528064320; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 18:00:53 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.38 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 18:00:55 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T175657.726772.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:00:57 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753370_chr17_1_fastp.fastq.gz, data/processed/ERR753370_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753370_chr17.sam, results/mapped_reads/ERR753370_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753370_chr17_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753370_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.27 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 24.72 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.25 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.18 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 14.83 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 42.787 sec; CPU: 40.255 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 18:05:34 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:05:34 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753370_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr17.bam, results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753370_chr17.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753370_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753370_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 18:07:16 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:07:16 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753370_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753370_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753370_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:07:17.864 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 18:07:17 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753370_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753370_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 18:07:17 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 18:07:17\tMarkDuplicates\tStart of doWork freeMemory: 528782312; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:07:17\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 18:07:17\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam\n",
      "INFO\t2023-08-17 18:07:19\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:17,848,783\n",
      "INFO\t2023-08-17 18:07:19\tMarkDuplicates\tTracking 639 as yet unmatched pairs. 639 records in RAM.\n",
      "INFO\t2023-08-17 18:07:21\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 17:36,310,502\n",
      "INFO\t2023-08-17 18:07:21\tMarkDuplicates\tTracking 1013 as yet unmatched pairs. 1013 records in RAM.\n",
      "INFO\t2023-08-17 18:07:22\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 17:47,368,559\n",
      "INFO\t2023-08-17 18:07:22\tMarkDuplicates\tTracking 955 as yet unmatched pairs. 955 records in RAM.\n",
      "INFO\t2023-08-17 18:07:24\tMarkDuplicates\tRead     4,000,000 records.  Elapsed time: 00:00:06s.  Time for last 1,000,000:    1s.  Last read position: 17:68,941,907\n",
      "INFO\t2023-08-17 18:07:24\tMarkDuplicates\tTracking 487 as yet unmatched pairs. 487 records in RAM.\n",
      "INFO\t2023-08-17 18:07:25\tMarkDuplicates\tRead 4523067 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 18:07:25\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 453976976; totalMemory: 1000341504; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:07:25\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 18:07:25\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 18:07:25\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 18:07:26\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 18:07:26\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 990851152; totalMemory: 1537212416; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:07:26\tMarkDuplicates\tMarking 290934 records as duplicates.\n",
      "INFO\t2023-08-17 18:07:26\tMarkDuplicates\tFound 164 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753370_chr17_sorted.bam\n",
      "INFO\t2023-08-17 18:07:26\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 18:07:59\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 18:07:59\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 18:07:59\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 18:07:59\tMarkDuplicates\tBefore output close freeMemory: 528080152; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:07:59\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 18:07:59\tMarkDuplicates\tAfter output close freeMemory: 528080152; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 18:07:59 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.69 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 18:08:02 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T180055.923586.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:08:04 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753371_chr17_1_fastp.fastq.gz, data/processed/ERR753371_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753371_chr17.sam, results/mapped_reads/ERR753371_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753371_chr17_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753371_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.27 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 28.29 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.28 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.19 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 16.87 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 48.468 sec; CPU: 45.906 sec\n",
      "[Thu Aug 17 18:09:57 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:09:57 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753371_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr17.bam, results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753371_chr17.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753371_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753371_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 18:10:22 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:10:22 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753371_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753371_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753371_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:10:23.643 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 18:10:23 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753371_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753371_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 18:10:23 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 18:10:23\tMarkDuplicates\tStart of doWork freeMemory: 528787280; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:10:23\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 18:10:23\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam\n",
      "INFO\t2023-08-17 18:10:25\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:51,502,813\n",
      "INFO\t2023-08-17 18:10:25\tMarkDuplicates\tTracking 116 as yet unmatched pairs. 116 records in RAM.\n",
      "INFO\t2023-08-17 18:10:25\tMarkDuplicates\tRead 1400902 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 18:10:25\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 316078208; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:10:25\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 18:10:25\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 18:10:26\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 18:10:26\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 18:10:26\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 755973712; totalMemory: 1302331392; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:10:26\tMarkDuplicates\tMarking 496406 records as duplicates.\n",
      "INFO\t2023-08-17 18:10:26\tMarkDuplicates\tFound 34 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753371_chr17_sorted.bam\n",
      "INFO\t2023-08-17 18:10:26\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 18:10:35\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 18:10:35\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 18:10:35\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 18:10:35\tMarkDuplicates\tBefore output close freeMemory: 528083880; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:10:35\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 18:10:35\tMarkDuplicates\tAfter output close freeMemory: 528083880; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 18:10:35 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.19 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 18:10:36 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T180803.121144.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:10:37 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753372_chr17_1_fastp.fastq.gz, data/processed/ERR753372_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753372_chr17.sam, results/mapped_reads/ERR753372_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753372_chr17_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR753372_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.29 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 28.36 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.28 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.19 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 16.54 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 48.497 sec; CPU: 45.659 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 18:13:52 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:13:52 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753372_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr17.bam, results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753372_chr17.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753372_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753372_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[Thu Aug 17 18:14:50 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:14:50 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753372_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753372_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753372_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:14:51.133 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 18:14:51 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753372_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753372_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 18:14:51 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 18:14:51\tMarkDuplicates\tStart of doWork freeMemory: 528787112; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:14:51\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 18:14:51\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam\n",
      "INFO\t2023-08-17 18:14:52\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:26,199,666\n",
      "INFO\t2023-08-17 18:14:52\tMarkDuplicates\tTracking 488 as yet unmatched pairs. 488 records in RAM.\n",
      "INFO\t2023-08-17 18:14:54\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    1s.  Last read position: 17:47,247,754\n",
      "INFO\t2023-08-17 18:14:54\tMarkDuplicates\tTracking 325 as yet unmatched pairs. 325 records in RAM.\n",
      "INFO\t2023-08-17 18:14:55\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: 17:78,126,195\n",
      "INFO\t2023-08-17 18:14:55\tMarkDuplicates\tTracking 96 as yet unmatched pairs. 96 records in RAM.\n",
      "INFO\t2023-08-17 18:14:55\tMarkDuplicates\tRead 3244765 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 18:14:55\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 411954960; totalMemory: 827326464; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:14:55\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 18:14:56\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 18:14:56\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 18:14:56\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 18:14:56\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 817841416; totalMemory: 1364197376; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:14:56\tMarkDuplicates\tMarking 72265 records as duplicates.\n",
      "INFO\t2023-08-17 18:14:56\tMarkDuplicates\tFound 452 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753372_chr17_sorted.bam\n",
      "INFO\t2023-08-17 18:14:56\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 18:15:15\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 18:15:15\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 18:15:15\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 18:15:15\tMarkDuplicates\tBefore output close freeMemory: 528085384; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:15:15\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 18:15:15\tMarkDuplicates\tAfter output close freeMemory: 528085384; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 18:15:15 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.40 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 18:15:17 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T181036.259604.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:15:19 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753373_chr17_1_fastp.fastq.gz, data/processed/ERR753373_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753373_chr17.sam, results/mapped_reads/ERR753373_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753373_chr17_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR753373_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.28 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 25.70 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.26 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.17 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 15.25 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 44.525 sec; CPU: 41.664 sec\n",
      "[Thu Aug 17 18:18:12 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:18:12 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753373_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr17.bam, results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753373_chr17.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753373_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753373_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 18:19:01 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:19:01 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753373_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753373_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753373_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:19:02.691 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 18:19:02 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753373_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753373_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 18:19:02 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 18:19:02\tMarkDuplicates\tStart of doWork freeMemory: 528787704; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:19:02\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 18:19:02\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam\n",
      "INFO\t2023-08-17 18:19:04\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:34,038,083\n",
      "INFO\t2023-08-17 18:19:04\tMarkDuplicates\tTracking 175 as yet unmatched pairs. 175 records in RAM.\n",
      "INFO\t2023-08-17 18:19:05\tMarkDuplicates\tRead 1994716 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 18:19:05\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 288043744; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:19:05\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 18:19:05\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 18:19:06\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 18:19:06\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 18:19:06\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 780093112; totalMemory: 1326448640; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:19:06\tMarkDuplicates\tMarking 36871 records as duplicates.\n",
      "INFO\t2023-08-17 18:19:06\tMarkDuplicates\tFound 11383 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753373_chr17_sorted.bam\n",
      "INFO\t2023-08-17 18:19:06\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 18:19:21\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 18:19:21\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 18:19:21\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 18:19:21\tMarkDuplicates\tBefore output close freeMemory: 528086184; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:19:21\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 18:19:21\tMarkDuplicates\tAfter output close freeMemory: 528086184; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 18:19:21 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.31 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 18:19:22 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T181517.816179.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:19:24 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753374_chr17_1_fastp.fastq.gz, data/processed/ERR753374_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753374_chr17.sam, results/mapped_reads/ERR753374_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753374_chr17_infosam.out\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/ERR753374_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.30 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 29.02 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.27 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.18 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 16.39 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 49.116 sec; CPU: 46.163 sec\n",
      "[Thu Aug 17 18:21:27 2023]\n",
      "Finished job 3.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:21:27 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753374_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr17.bam, results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753374_chr17.flagstats\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753374_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753374_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 18:21:54 2023]\n",
      "Finished job 2.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:21:54 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753374_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753374_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 1\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753374_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:21:55.769 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 18:21:55 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753374_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753374_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 18:21:55 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 18:21:55\tMarkDuplicates\tStart of doWork freeMemory: 528787864; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:21:55\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 18:21:55\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam\n",
      "INFO\t2023-08-17 18:21:57\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:62,737,004\n",
      "INFO\t2023-08-17 18:21:57\tMarkDuplicates\tTracking 91 as yet unmatched pairs. 91 records in RAM.\n",
      "INFO\t2023-08-17 18:21:57\tMarkDuplicates\tRead 1192512 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 18:21:57\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 362816272; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:21:57\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 18:21:57\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 18:21:57\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 18:21:58\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 18:21:58\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 669991616; totalMemory: 1216348160; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:21:58\tMarkDuplicates\tMarking 88619 records as duplicates.\n",
      "INFO\t2023-08-17 18:21:58\tMarkDuplicates\tFound 28784 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753374_chr17_sorted.bam\n",
      "INFO\t2023-08-17 18:21:58\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 18:22:07\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 18:22:07\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 18:22:07\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 18:22:07\tMarkDuplicates\tBefore output close freeMemory: 528085096; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:22:07\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 18:22:07\tMarkDuplicates\tAfter output close freeMemory: 528085096; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 18:22:07 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.19 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 18:22:07 2023]\n",
      "Finished job 1.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T181923.136475.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:22:09 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753375_chr17_1_fastp.fastq.gz, data/processed/ERR753375_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753375_chr17.sam, results/mapped_reads/ERR753375_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753375_chr17_infosam.out\n",
      "    jobid: 2\n",
      "    reason: Missing output files: results/mapped_reads/ERR753375_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.29 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 28.36 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.28 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.19 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 16.51 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 48.360 sec; CPU: 45.629 sec\n",
      "[Thu Aug 17 18:23:56 2023]\n",
      "Finished job 2.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:23:56 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753375_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr17.bam, results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753375_chr17.flagstats\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753375_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753375_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 18:24:19 2023]\n",
      "Finished job 1.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:24:19 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753375_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753375_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753375_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:24:20.571 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 18:24:20 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753375_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753375_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 18:24:20 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 18:24:20\tMarkDuplicates\tStart of doWork freeMemory: 528785432; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:24:20\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 18:24:20\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam\n",
      "INFO\t2023-08-17 18:24:22\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:61,924,970\n",
      "INFO\t2023-08-17 18:24:22\tMarkDuplicates\tTracking 107 as yet unmatched pairs. 107 records in RAM.\n",
      "INFO\t2023-08-17 18:24:22\tMarkDuplicates\tRead 1249072 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 18:24:22\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 332371408; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:24:22\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 18:24:22\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 18:24:22\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 18:24:23\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 18:24:23\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 727661848; totalMemory: 1274019840; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:24:23\tMarkDuplicates\tMarking 53205 records as duplicates.\n",
      "INFO\t2023-08-17 18:24:23\tMarkDuplicates\tFound 132 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753375_chr17_sorted.bam\n",
      "INFO\t2023-08-17 18:24:23\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 18:24:31\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 18:24:31\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 18:24:31\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 18:24:31\tMarkDuplicates\tBefore output close freeMemory: 528083888; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:24:31\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 18:24:31\tMarkDuplicates\tAfter output close freeMemory: 528083888; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 18:24:31 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.18 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 18:24:32 2023]\n",
      "Finished job 0.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T182208.117578.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:24:33 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753376_chr17_1_fastp.fastq.gz, data/processed/ERR753376_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753376_chr17.sam, results/mapped_reads/ERR753376_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753376_chr17_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753376_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.29 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 28.02 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.26 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.19 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 16.37 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 47.986 sec; CPU: 45.122 sec\n",
      "[Thu Aug 17 18:26:28 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:26:28 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753376_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr17.bam, results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753376_chr17.flagstats\n",
      "    jobid: 3\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753376_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753376_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 18:26:57 2023]\n",
      "Finished job 3.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:26:57 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753376_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753376_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753376_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:26:58.687 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 18:26:58 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753376_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753376_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 18:26:58 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 18:26:58\tMarkDuplicates\tStart of doWork freeMemory: 528834280; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:26:58\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 18:26:58\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam\n",
      "INFO\t2023-08-17 18:27:00\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:46,640,085\n",
      "INFO\t2023-08-17 18:27:00\tMarkDuplicates\tTracking 176 as yet unmatched pairs. 176 records in RAM.\n",
      "INFO\t2023-08-17 18:27:01\tMarkDuplicates\tRead 1528048 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 18:27:01\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 369766192; totalMemory: 601882624; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:27:01\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 18:27:01\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 18:27:01\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 18:27:01\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 18:27:01\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 773848248; totalMemory: 1320157184; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:27:01\tMarkDuplicates\tMarking 187048 records as duplicates.\n",
      "INFO\t2023-08-17 18:27:01\tMarkDuplicates\tFound 58 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753376_chr17_sorted.bam\n",
      "INFO\t2023-08-17 18:27:01\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 18:27:11\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 18:27:11\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 18:27:11\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 18:27:11\tMarkDuplicates\tBefore output close freeMemory: 528132728; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:27:11\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 18:27:11\tMarkDuplicates\tAfter output close freeMemory: 528132728; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 18:27:11 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.21 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 18:27:12 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T182432.232858.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:27:13 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753377_chr17_1_fastp.fastq.gz, data/processed/ERR753377_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753377_chr17.sam, results/mapped_reads/ERR753377_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753377_chr17_infosam.out\n",
      "    jobid: 1\n",
      "    reason: Missing output files: results/mapped_reads/ERR753377_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.29 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 28.99 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.27 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.19 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 16.51 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 49.146 sec; CPU: 46.250 sec\n",
      "[Thu Aug 17 18:29:14 2023]\n",
      "Finished job 1.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:29:14 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753377_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr17.bam, results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753377_chr17.flagstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753377_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753377_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 18:29:45 2023]\n",
      "Finished job 0.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:29:45 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753377_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753377_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 4\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753377_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:29:46.244 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 18:29:46 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753377_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753377_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 18:29:46 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 18:29:46\tMarkDuplicates\tStart of doWork freeMemory: 528791832; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:29:46\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 18:29:46\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam\n",
      "INFO\t2023-08-17 18:29:47\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:45,177,396\n",
      "INFO\t2023-08-17 18:29:47\tMarkDuplicates\tTracking 190 as yet unmatched pairs. 190 records in RAM.\n",
      "INFO\t2023-08-17 18:29:48\tMarkDuplicates\tRead 1629046 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 18:29:48\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 292321984; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:29:48\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 18:29:49\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 18:29:49\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 18:29:49\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 18:29:49\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 795825360; totalMemory: 1342177280; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:29:49\tMarkDuplicates\tMarking 200618 records as duplicates.\n",
      "INFO\t2023-08-17 18:29:49\tMarkDuplicates\tFound 53 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753377_chr17_sorted.bam\n",
      "INFO\t2023-08-17 18:29:49\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 18:30:00\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 18:30:00\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 18:30:00\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 18:30:00\tMarkDuplicates\tBefore output close freeMemory: 528089400; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:30:00\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 18:30:00\tMarkDuplicates\tAfter output close freeMemory: 528089400; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 18:30:00 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.24 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 18:30:01 2023]\n",
      "Finished job 4.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T182712.678067.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                  count    min threads    max threads\n",
      "-----------------  -------  -------------  -------------\n",
      "bwa_mapping              1              1              1\n",
      "delete_duplicates        1              1              1\n",
      "sam_to_bam               1              1              1\n",
      "total                    3              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:30:03 2023]\n",
      "rule bwa_mapping:\n",
      "    input: data/reference/genome.fa, data/processed/ERR753378_chr17_1_fastp.fastq.gz, data/processed/ERR753378_chr17_2_fastp.fastq.gz\n",
      "    output: results/mapped_reads/ERR753378_chr17.sam, results/mapped_reads/ERR753378_chr17_sorted.sam\n",
      "    log: metadata/logs/sam/ERR753378_chr17_infosam.out\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/mapped_reads/ERR753378_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[bwa_index] Pack FASTA... 0.28 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=166514882, availableWord=23716276\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 39120754 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 72271554 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 101732034 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 127912546 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 151177842 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 58 iterations.\n",
      "[bwa_index] 30.68 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.26 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.18 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 17.67 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index data/reference/genome.fa\n",
      "[main] Real time: 51.907 sec; CPU: 49.061 sec\n",
      "[Thu Aug 17 18:32:56 2023]\n",
      "Finished job 0.\n",
      "1 of 3 steps (33%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:32:56 2023]\n",
      "rule sam_to_bam:\n",
      "    input: results/mapped_reads/ERR753378_chr17_sorted.sam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr17.bam, results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam\n",
      "    log: metadata/logs/flagstats/ERR753378_chr17.flagstats\n",
      "    jobid: 4\n",
      "    reason: Missing output files: results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam; Input files updated by another job: results/mapped_reads/ERR753378_chr17_sorted.sam\n",
      "    wildcards: sample=ERR753378_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 18:33:40 2023]\n",
      "Finished job 4.\n",
      "2 of 3 steps (67%) done\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:33:40 2023]\n",
      "rule delete_duplicates:\n",
      "    input: results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam\n",
      "    output: results/mapped_reads/bam_files/ERR753378_chr17_dedup.bam\n",
      "    log: metadata/logs/markduplicates/ERR753378_chr17_markDuplicatesMetrics.txt\n",
      "    jobid: 3\n",
      "    reason: Input files updated by another job: results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam\n",
      "    wildcards: sample=ERR753378_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "18:33:41.825 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/juancarlos/Escritorio/TFM/.snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_/share/picard-3.0.0-1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Aug 17 18:33:41 WEST 2023] MarkDuplicates --INPUT results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam --OUTPUT results/mapped_reads/bam_files/ERR753378_chr17_dedup.bam --METRICS_FILE metadata/logs/markduplicates/ERR753378_chr17_markDuplicatesMetrics.txt --ASSUME_SORTED true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Aug 17 18:33:41 WEST 2023] Executing as juancarlos@juancarlos-System-Product-Name on Linux 5.19.0-46-generic amd64; OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:3.0.0\n",
      "INFO\t2023-08-17 18:33:41\tMarkDuplicates\tStart of doWork freeMemory: 528828856; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:33:41\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2023-08-17 18:33:41\tMarkDuplicates\tWill retain up to 7780737 data points before spilling to disk.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam\n",
      "INFO\t2023-08-17 18:33:43\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:01s.  Time for last 1,000,000:    1s.  Last read position: 17:30,209,106\n",
      "INFO\t2023-08-17 18:33:43\tMarkDuplicates\tTracking 410 as yet unmatched pairs. 410 records in RAM.\n",
      "INFO\t2023-08-17 18:33:44\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:03s.  Time for last 1,000,000:    1s.  Last read position: 17:58,462,920\n",
      "INFO\t2023-08-17 18:33:44\tMarkDuplicates\tTracking 397 as yet unmatched pairs. 397 records in RAM.\n",
      "INFO\t2023-08-17 18:33:46\tMarkDuplicates\tRead 2826651 records. 0 pairs never matched.\n",
      "INFO\t2023-08-17 18:33:46\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 290151952; totalMemory: 663748608; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:33:46\tMarkDuplicates\tWill retain up to 67108864 duplicate indices before spilling to disk.\n",
      "INFO\t2023-08-17 18:33:46\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2023-08-17 18:33:46\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2023-08-17 18:33:46\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2023-08-17 18:33:46\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 955247360; totalMemory: 1501560832; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:33:46\tMarkDuplicates\tMarking 63262 records as duplicates.\n",
      "INFO\t2023-08-17 18:33:46\tMarkDuplicates\tFound 446 optical duplicate clusters.\n",
      "WARNING: BAM index file /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam.bai is older than BAM /home/juancarlos/Escritorio/TFM/results/mapped_reads/bam_files/ERR753378_chr17_sorted.bam\n",
      "INFO\t2023-08-17 18:33:46\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2023-08-17 18:34:03\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2023-08-17 18:34:03\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2023-08-17 18:34:03\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2023-08-17 18:34:03\tMarkDuplicates\tBefore output close freeMemory: 528128104; totalMemory: 536870912; maxMemory: 2147483648\n",
      "INFO\t2023-08-17 18:34:03\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2023-08-17 18:34:03\tMarkDuplicates\tAfter output close freeMemory: 528128104; totalMemory: 536870912; maxMemory: 2147483648\n",
      "[Thu Aug 17 18:34:03 WEST 2023] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.36 minutes.\n",
      "Runtime.totalMemory()=536870912\n",
      "[Thu Aug 17 18:34:05 2023]\n",
      "Finished job 3.\n",
      "3 of 3 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T183001.833234.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:34:07 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR696683_chr17_dedup.bam\n",
      "    output: results/variants/ERR696683_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR696683_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR696683_chr17.vcf\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "index file data/reference/genome.fa.fai not found, generating...\n",
      "[Thu Aug 17 18:34:42 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T183405.823475.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:34:43 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753368_chr17_dedup.bam\n",
      "    output: results/variants/ERR753368_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753368_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753368_chr17.vcf\n",
      "    wildcards: sample=ERR753368_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 18:35:37 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T183442.253524.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:35:39 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753369_chr17_dedup.bam\n",
      "    output: results/variants/ERR753369_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753369_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753369_chr17.vcf\n",
      "    wildcards: sample=ERR753369_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 18:36:36 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T183538.074992.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:36:37 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753370_chr17_dedup.bam\n",
      "    output: results/variants/ERR753370_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753370_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753370_chr17.vcf\n",
      "    wildcards: sample=ERR753370_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 18:38:04 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T183636.380516.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:38:05 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753371_chr17_dedup.bam\n",
      "    output: results/variants/ERR753371_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753371_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753371_chr17.vcf\n",
      "    wildcards: sample=ERR753371_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 18:38:29 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T183804.301257.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:38:30 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753372_chr17_dedup.bam\n",
      "    output: results/variants/ERR753372_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753372_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753372_chr17.vcf\n",
      "    wildcards: sample=ERR753372_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 18:39:30 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T183829.285727.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:39:32 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753373_chr17_dedup.bam\n",
      "    output: results/variants/ERR753373_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753373_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753373_chr17.vcf\n",
      "    wildcards: sample=ERR753373_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 18:40:14 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T183930.789713.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:40:16 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753374_chr17_dedup.bam\n",
      "    output: results/variants/ERR753374_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753374_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753374_chr17.vcf\n",
      "    wildcards: sample=ERR753374_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 18:40:43 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T184015.053826.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:40:45 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753375_chr17_dedup.bam\n",
      "    output: results/variants/ERR753375_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753375_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753375_chr17.vcf\n",
      "    wildcards: sample=ERR753375_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 18:41:13 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T184044.109690.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:41:14 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753376_chr17_dedup.bam\n",
      "    output: results/variants/ERR753376_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753376_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753376_chr17.vcf\n",
      "    wildcards: sample=ERR753376_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 18:41:43 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T184113.474029.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:41:45 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753377_chr17_dedup.bam\n",
      "    output: results/variants/ERR753377_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753377_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753377_chr17.vcf\n",
      "    wildcards: sample=ERR753377_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 18:42:15 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T184143.805062.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                    count    min threads    max threads\n",
      "-------------------  -------  -------------  -------------\n",
      "extracting_variants        1              1              1\n",
      "total                      1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:42:16 2023]\n",
      "rule extracting_variants:\n",
      "    input: data/reference/genome.fa, results/mapped_reads/bam_files/ERR753378_chr17_dedup.bam\n",
      "    output: results/variants/ERR753378_chr17.vcf\n",
      "    log: metadata/logs/vcfstats/ERR753378_chr17.vcfstats\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/ERR753378_chr17.vcf\n",
      "    wildcards: sample=ERR753378_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/048a3a8f16f93fa02ed16849ed907ced_\n",
      "[Thu Aug 17 18:43:09 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T184215.592670.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:43:11 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR696683_chr17.vcf\n",
      "    output: results/variants/vep/ERR696683_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR696683_chr17.txt\n",
      "    wildcards: sample=ERR696683_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 18:43:42 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T184309.769065.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:43:44 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753368_chr17.vcf\n",
      "    output: results/variants/vep/ERR753368_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753368_chr17.txt\n",
      "    wildcards: sample=ERR753368_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 18:44:19 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T184342.460104.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:44:20 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753369_chr17.vcf\n",
      "    output: results/variants/vep/ERR753369_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753369_chr17.txt\n",
      "    wildcards: sample=ERR753369_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 18:44:54 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T184419.651367.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:44:56 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753370_chr17.vcf\n",
      "    output: results/variants/vep/ERR753370_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753370_chr17.txt\n",
      "    wildcards: sample=ERR753370_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 18:45:32 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T184455.017405.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:45:34 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753371_chr17.vcf\n",
      "    output: results/variants/vep/ERR753371_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753371_chr17.txt\n",
      "    wildcards: sample=ERR753371_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 18:45:55 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T184532.795362.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:45:56 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753372_chr17.vcf\n",
      "    output: results/variants/vep/ERR753372_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753372_chr17.txt\n",
      "    wildcards: sample=ERR753372_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 18:46:32 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T184555.269817.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:46:33 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753373_chr17.vcf\n",
      "    output: results/variants/vep/ERR753373_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753373_chr17.txt\n",
      "    wildcards: sample=ERR753373_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 18:47:00 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T184632.335464.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:47:02 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753374_chr17.vcf\n",
      "    output: results/variants/vep/ERR753374_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753374_chr17.txt\n",
      "    wildcards: sample=ERR753374_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 18:47:23 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T184700.772265.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:47:25 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753375_chr17.vcf\n",
      "    output: results/variants/vep/ERR753375_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753375_chr17.txt\n",
      "    wildcards: sample=ERR753375_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 18:47:48 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T184723.685499.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:47:50 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753376_chr17.vcf\n",
      "    output: results/variants/vep/ERR753376_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753376_chr17.txt\n",
      "    wildcards: sample=ERR753376_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 18:48:14 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T184748.972205.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:48:16 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753377_chr17.vcf\n",
      "    output: results/variants/vep/ERR753377_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753377_chr17.txt\n",
      "    wildcards: sample=ERR753377_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 18:48:42 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T184815.104781.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job        count    min threads    max threads\n",
      "-------  -------  -------------  -------------\n",
      "vep_cli        1              1              1\n",
      "total          1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:48:43 2023]\n",
      "rule vep_cli:\n",
      "    input: code/04vep.sh, results/variants/ERR753378_chr17.vcf\n",
      "    output: results/variants/vep/ERR753378_chr17.txt\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/variants/vep/ERR753378_chr17.txt\n",
      "    wildcards: sample=ERR753378_chr17\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Activating conda environment: .snakemake/conda/0f112e57cb9da2b58652c2b24195a7e6_\n",
      "[Thu Aug 17 18:49:18 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T184842.260549.snakemake.log\n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job              count    min threads    max threads\n",
      "-------------  -------  -------------  -------------\n",
      "parsing_dataR        1              1              1\n",
      "total                1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:49:19 2023]\n",
      "rule parsing_dataR:\n",
      "    input: code/05parsing_vep_data.R\n",
      "    output: tasks/13parsing_dataR.done\n",
      "    jobid: 0\n",
      "    reason: Params have changed since last execution\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "Warning message:\n",
      "package ‘glue’ was built under R version 4.2.3 \n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">>> This is your sample list: ERR696683_chr17\n",
      ">>> This is your sample list: ERR753368_chr17\n",
      ">>> This is your sample list: ERR753369_chr17\n",
      ">>> This is your sample list: ERR753370_chr17\n",
      ">>> This is your sample list: ERR753371_chr17\n",
      ">>> This is your sample list: ERR753372_chr17\n",
      ">>> This is your sample list: ERR753373_chr17\n",
      ">>> This is your sample list: ERR753374_chr17\n",
      ">>> This is your sample list: ERR753375_chr17\n",
      ">>> This is your sample list: ERR753376_chr17\n",
      ">>> This is your sample list: ERR753377_chr17\n",
      ">>> This is your sample list: ERR753378_chr17\n",
      ">>> This is your gene: TP53\n",
      "# A tibble: 26 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "2 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "3 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "4 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "5 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "6 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "# ℹ 20 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 152 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7674109_G/A        17:7674109 A      ENSG… ENST00… Transcript   intron_var…\n",
      "2 17_7674109_G/A        17:7674109 A      ENSG… ENST00… Transcript   intron_var…\n",
      "3 17_7674109_G/A        17:7674109 A      ENSG… ENST00… Transcript   intron_var…\n",
      "4 17_7674109_G/A        17:7674109 A      ENSG… ENST00… Transcript   intron_var…\n",
      "5 17_7674109_G/A        17:7674109 A      ENSG… ENST00… Transcript   intron_var…\n",
      "6 17_7674109_G/A        17:7674109 A      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 146 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 249 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   downstream…\n",
      "2 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   intron_var…\n",
      "3 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   intron_var…\n",
      "4 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   downstream…\n",
      "5 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   downstream…\n",
      "6 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   downstream…\n",
      "# ℹ 243 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 130 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "2 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "3 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "4 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "5 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "6 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 124 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 58 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "2 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "3 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "4 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "5 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "6 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "# ℹ 52 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 126 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "2 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "3 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "4 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "5 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "6 17_7674797_T/C        17:7674797 C      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 120 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 68 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "2 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "3 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 62 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 0 × 44\n",
      "# ℹ 44 variables: #uploaded_variation <chr>, location <chr>, allele <chr>,\n",
      "#   gene <chr>, feature <chr>, feature_type <chr>, consequence <chr>,\n",
      "#   cdna_position <chr>, cds_position <chr>, protein_position <chr>,\n",
      "#   amino_acids <chr>, codons <chr>, existing_variation <chr>, impact <chr>,\n",
      "#   distance <chr>, strand <chr>, flags <chr>, variant_class <chr>,\n",
      "#   symbol <chr>, symbol_source <chr>, hgnc_id <chr>, biotype <chr>,\n",
      "#   mane_select <chr>, mane_plus_clinical <chr>, tsl <chr>, appris <chr>, …\n",
      "# A tibble: 26 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "2 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "3 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "4 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "5 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "6 17_7676301_G/T        17:7676301 T      ENSG… ENST00… Transcript   intron_var…\n",
      "# ℹ 20 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 52 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "2 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "3 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "4 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "5 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "6 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "# ℹ 46 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "# A tibble: 52 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "2 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "3 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "4 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "5 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "6 17_7676154_G/C        17:7676154 C      ENSG… ENST00… Transcript   missense_v…\n",
      "# ℹ 46 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n",
      "\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "cols(\n",
      "  .default = col_character()\n",
      ")\n",
      "ℹ Use `spec()` for the full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 149 × 44\n",
      "  `#uploaded_variation` location   allele gene  feature feature_type consequence\n",
      "  <chr>                 <chr>      <chr>  <chr> <chr>   <chr>        <chr>      \n",
      "1 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   downstream…\n",
      "2 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   intron_var…\n",
      "3 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   intron_var…\n",
      "4 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   downstream…\n",
      "5 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   downstream…\n",
      "6 17_7666380_A/C        17:7666380 C      ENSG… ENST00… Transcript   downstream…\n",
      "# ℹ 143 more rows\n",
      "# ℹ 37 more variables: cdna_position <chr>, cds_position <chr>,\n",
      "#   protein_position <chr>, amino_acids <chr>, codons <chr>,\n",
      "#   existing_variation <chr>, impact <chr>, distance <chr>, strand <chr>,\n",
      "#   flags <chr>, variant_class <chr>, symbol <chr>, symbol_source <chr>,\n",
      "#   hgnc_id <chr>, biotype <chr>, mane_select <chr>, mane_plus_clinical <chr>,\n",
      "#   tsl <chr>, appris <chr>, source <chr>, sift <chr>, polyphen <chr>, …\n",
      "[1] \"##==========================##\"\n",
      "[1] \"##===> WORK FINISHED!!! <===##\"\n",
      "[1] \"##==========================##\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Touching output file tasks/13parsing_dataR.done.\n",
      "[Thu Aug 17 18:49:23 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T184918.808115.snakemake.log\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['snakemake', '--cores', '1', 'parsing_dataR'], returncode=0)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Running the workflow for the 12º chormosome\n",
    "\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--force\", \"data/reference/genome.fa\"]) ## rule reference_genome \n",
    "\n",
    "for sample in sample_list:\n",
    "    ## Quality inspection\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/fastqc_result/\" + sample + \"_chr17_1_fastqc.html\",   \n",
    "                    \"data/processed/\" + sample + \"_chr17_1_fastp.fastq.gz\",   \n",
    "                    \"results/fastqc_result/trimmed/\" + sample + \"_chr17_1_fastp_fastqc.html\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    ## Mapping reads\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\",\n",
    "                    \"results/mapped_reads/\" + sample + \"_chr17_sorted.sam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr17_sorted.bam\",   \n",
    "                    \"results/mapped_reads/bam_files/\" + sample + \"_chr17_dedup.bam\"])\n",
    "    \n",
    "for sample in sample_list:\n",
    "    # Extracting variants  \n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/\" + sample + \"_chr17.vcf\"])   \n",
    "\n",
    "## VEP CLI\n",
    "for sample in sample_list:\n",
    "    subprocess.run([\"snakemake\", \"--cores\", \"1\", \"--use-conda\", \"results/variants/vep/\" + sample + \"_chr17.txt\"])\n",
    "\n",
    "## Extracting the gene PIKCA3 from the VEP files\n",
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"parsing_dataR\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['sed', '-i', '50s/TP53/PIK3CA/g', 'config.yaml'], returncode=0)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## reset config \n",
    "subprocess.run([\"sed\", \"-i\", \"6,+12s/chr17/chr3/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"24s/chr17/chr3/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"31s/chromosome.17.fa.gz/chromosome.3.fa.gz/g\", \"config.yaml\"])\n",
    "subprocess.run([\"sed\", \"-i\", \"50s/TP53/PIK3CA/g\", \"config.yaml\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## **Last rule**\n",
    "\n",
    "### **rule R_plotting**\n",
    "\n",
    "Plotting the data of the 5 gene tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 1 (use --cores to define parallelism)\n",
      "Rules claiming more threads will be scaled down.\n",
      "Conda environments: ignored\n",
      "Job stats:\n",
      "job           count    min threads    max threads\n",
      "----------  -------  -------------  -------------\n",
      "R_plotting        1              1              1\n",
      "total             1              1              1\n",
      "\n",
      "Select jobs to execute...\n",
      "\n",
      "[Thu Aug 17 18:53:21 2023]\n",
      "rule R_plotting:\n",
      "    input: code/06final_plot.R\n",
      "    output: results/biostatistics/plots/final_plot.png, results/biostatistics/plots/other_plot.png\n",
      "    jobid: 0\n",
      "    reason: Missing output files: results/biostatistics/plots/other_plot.png, results/biostatistics/plots/final_plot.png\n",
      "    resources: tmpdir=/tmp\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"##==========================##\"\n",
      "[1] \"##===> WORK FINISHED!!! <===##\"\n",
      "[1] \"##==========================##\"\n",
      "THE SCRIPT RAN WELL congrats :)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Thu Aug 17 18:53:23 2023]\n",
      "Finished job 0.\n",
      "1 of 1 steps (100%) done\n",
      "Complete log: .snakemake/log/2023-08-17T185320.887286.snakemake.log\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['snakemake', '--cores', '1', 'R_plotting'], returncode=0)"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subprocess.run([\"snakemake\", \"--cores\", \"1\", \"R_plotting\"])"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Greference_tools",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
