{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Import libreries\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Downloading the table 3, for supporting information found in the article.\n",
    "\n",
    "After that we will use the function in pandas ```pd.read_table()``` to download the data.\n",
    "\n",
    "https://onlinelibrary.wiley.com/action/downloadSupplement?doi=10.1002%2Fgcc.22314&file=gcc22314-sup-0005-supptable5.docx"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Raw Reads</th>\n",
       "      <th>Sequencing depth (X)</th>\n",
       "      <th>Reads after QC</th>\n",
       "      <th>Reads after QC# %</th>\n",
       "      <th>Mapped reads</th>\n",
       "      <th>Reads after QC %</th>\n",
       "      <th>% On-target (30X)</th>\n",
       "      <th>Mean coverage (X)</th>\n",
       "      <th>Genes covered (%)</th>\n",
       "      <th>SNPs</th>\n",
       "      <th>Indels</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IV-5^</td>\n",
       "      <td>29 191 671</td>\n",
       "      <td>84.95</td>\n",
       "      <td>22 978 888</td>\n",
       "      <td>79,00 %</td>\n",
       "      <td>45 770 776</td>\n",
       "      <td>99.59%</td>\n",
       "      <td>49,00 %</td>\n",
       "      <td>17.79x(1x-856x)</td>\n",
       "      <td>98.48%</td>\n",
       "      <td>114 651</td>\n",
       "      <td>6 935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IV-2^</td>\n",
       "      <td>40 672 795</td>\n",
       "      <td>122.02</td>\n",
       "      <td>39 920 701</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>71 507 872</td>\n",
       "      <td>99.54%</td>\n",
       "      <td>79,00 %</td>\n",
       "      <td>26.76x(1x-860x)</td>\n",
       "      <td>99.70%</td>\n",
       "      <td>226 812</td>\n",
       "      <td>10 226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IV-6^</td>\n",
       "      <td>43 486 476</td>\n",
       "      <td>130.46</td>\n",
       "      <td>38 353 481</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>76 404 013</td>\n",
       "      <td>99.61%</td>\n",
       "      <td>81,00 %</td>\n",
       "      <td>29.66x(1x-817x)</td>\n",
       "      <td>99.71%</td>\n",
       "      <td>207 627</td>\n",
       "      <td>10 263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IV-4^</td>\n",
       "      <td>42 838 012</td>\n",
       "      <td>128.51</td>\n",
       "      <td>37 729 408</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>75 125 893</td>\n",
       "      <td>99.56%</td>\n",
       "      <td>81,00 %</td>\n",
       "      <td>28.76x(1x-877x)</td>\n",
       "      <td>99.91%</td>\n",
       "      <td>229 099</td>\n",
       "      <td>10 325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IV-7^</td>\n",
       "      <td>377 188 93</td>\n",
       "      <td>113.16</td>\n",
       "      <td>33 250 705</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>66 224 206</td>\n",
       "      <td>99.58%</td>\n",
       "      <td>77,00 %</td>\n",
       "      <td>25.82x(1x-847x)</td>\n",
       "      <td>99.70%</td>\n",
       "      <td>203 338</td>\n",
       "      <td>9 474</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Sample   Raw Reads Sequencing depth (X) Reads after QC Reads after QC# %  \\\n",
       "0  IV-5^  29 191 671                84.95     22 978 888           79,00 %   \n",
       "1  IV-2^  40 672 795               122.02     39 920 701           88,00 %   \n",
       "2  IV-6^  43 486 476               130.46     38 353 481           88,00 %   \n",
       "3  IV-4^  42 838 012               128.51     37 729 408           88,00 %   \n",
       "4  IV-7^  377 188 93               113.16     33 250 705           88,00 %   \n",
       "\n",
       "  Mapped reads Reads after QC % % On-target (30X) Mean coverage (X)  \\\n",
       "0   45 770 776           99.59%           49,00 %   17.79x(1x-856x)   \n",
       "1   71 507 872           99.54%           79,00 %   26.76x(1x-860x)   \n",
       "2   76 404 013           99.61%           81,00 %   29.66x(1x-817x)   \n",
       "3   75 125 893           99.56%           81,00 %   28.76x(1x-877x)   \n",
       "4   66 224 206           99.58%           77,00 %   25.82x(1x-847x)   \n",
       "\n",
       "  Genes covered (%)     SNPs  Indels  \n",
       "0            98.48%  114 651   6 935  \n",
       "1            99.70%  226 812  10 226  \n",
       "2            99.71%  207 627  10 263  \n",
       "3            99.91%  229 099  10 325  \n",
       "4            99.70%  203 338   9 474  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table3=pd.read_clipboard()\n",
    "table3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_34176/3769299083.py:4: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.\n",
      "  ).str.replace(\n",
      "/tmp/ipykernel_34176/3769299083.py:6: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.\n",
      "  ).str.replace(\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Raw Reads</th>\n",
       "      <th>Sequencing depth (X)</th>\n",
       "      <th>Reads after QC</th>\n",
       "      <th>Reads after QC# %</th>\n",
       "      <th>Mapped reads</th>\n",
       "      <th>Reads after QC %</th>\n",
       "      <th>% On-target (30X)</th>\n",
       "      <th>Mean coverage (X)</th>\n",
       "      <th>Genes covered (%)</th>\n",
       "      <th>SNPs</th>\n",
       "      <th>Indels</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IV_5_method1</td>\n",
       "      <td>29 191 671</td>\n",
       "      <td>84.95</td>\n",
       "      <td>22 978 888</td>\n",
       "      <td>79,00 %</td>\n",
       "      <td>45 770 776</td>\n",
       "      <td>99.59%</td>\n",
       "      <td>49,00 %</td>\n",
       "      <td>17.79x(1x-856x)</td>\n",
       "      <td>98.48%</td>\n",
       "      <td>114 651</td>\n",
       "      <td>6 935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IV_2_method1</td>\n",
       "      <td>40 672 795</td>\n",
       "      <td>122.02</td>\n",
       "      <td>39 920 701</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>71 507 872</td>\n",
       "      <td>99.54%</td>\n",
       "      <td>79,00 %</td>\n",
       "      <td>26.76x(1x-860x)</td>\n",
       "      <td>99.70%</td>\n",
       "      <td>226 812</td>\n",
       "      <td>10 226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IV_6_method1</td>\n",
       "      <td>43 486 476</td>\n",
       "      <td>130.46</td>\n",
       "      <td>38 353 481</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>76 404 013</td>\n",
       "      <td>99.61%</td>\n",
       "      <td>81,00 %</td>\n",
       "      <td>29.66x(1x-817x)</td>\n",
       "      <td>99.71%</td>\n",
       "      <td>207 627</td>\n",
       "      <td>10 263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IV_4_method1</td>\n",
       "      <td>42 838 012</td>\n",
       "      <td>128.51</td>\n",
       "      <td>37 729 408</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>75 125 893</td>\n",
       "      <td>99.56%</td>\n",
       "      <td>81,00 %</td>\n",
       "      <td>28.76x(1x-877x)</td>\n",
       "      <td>99.91%</td>\n",
       "      <td>229 099</td>\n",
       "      <td>10 325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IV_7_method1</td>\n",
       "      <td>377 188 93</td>\n",
       "      <td>113.16</td>\n",
       "      <td>33 250 705</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>66 224 206</td>\n",
       "      <td>99.58%</td>\n",
       "      <td>77,00 %</td>\n",
       "      <td>25.82x(1x-847x)</td>\n",
       "      <td>99.70%</td>\n",
       "      <td>203 338</td>\n",
       "      <td>9 474</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Sample   Raw Reads Sequencing depth (X) Reads after QC  \\\n",
       "0  IV_5_method1  29 191 671                84.95     22 978 888   \n",
       "1  IV_2_method1  40 672 795               122.02     39 920 701   \n",
       "2  IV_6_method1  43 486 476               130.46     38 353 481   \n",
       "3  IV_4_method1  42 838 012               128.51     37 729 408   \n",
       "4  IV_7_method1  377 188 93               113.16     33 250 705   \n",
       "\n",
       "  Reads after QC# % Mapped reads Reads after QC % % On-target (30X)  \\\n",
       "0           79,00 %   45 770 776           99.59%           49,00 %   \n",
       "1           88,00 %   71 507 872           99.54%           79,00 %   \n",
       "2           88,00 %   76 404 013           99.61%           81,00 %   \n",
       "3           88,00 %   75 125 893           99.56%           81,00 %   \n",
       "4           88,00 %   66 224 206           99.58%           77,00 %   \n",
       "\n",
       "  Mean coverage (X) Genes covered (%)     SNPs  Indels  \n",
       "0   17.79x(1x-856x)            98.48%  114 651   6 935  \n",
       "1   26.76x(1x-860x)            99.70%  226 812  10 226  \n",
       "2   29.66x(1x-817x)            99.71%  207 627  10 263  \n",
       "3   28.76x(1x-877x)            99.91%  229 099  10 325  \n",
       "4   25.82x(1x-847x)            99.70%  203 338   9 474  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## We wolud do some tranformations to the first field (\"Sample\")\n",
    "table3[\"Sample\"] = table3[\"Sample\"].str.replace(\n",
    "    \"-\", \"_\"\n",
    "    ).str.replace(\n",
    "        \"^\", \"_method1\"\n",
    "        ).str.replace(\n",
    "            \"*\", \"_method2\"\n",
    "            )\n",
    "table3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "table3.to_csv(\"../metadata/table3.csv\", index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "envpython",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
