{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Import libreries\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Downloading the table 3, for supporting information found in the article.\n",
    "\n",
    "After that we will use the function in pandas ```pd.read_clipboard()``` to download the data.\n",
    "\n",
    "https://onlinelibrary.wiley.com/action/downloadSupplement?doi=10.1002%2Fgcc.22314&file=gcc22314-sup-0005-supptable5.docx"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Raw Reads</th>\n",
       "      <th>Sequencing depth (X)</th>\n",
       "      <th>Reads after QC</th>\n",
       "      <th>Reads after QC# %</th>\n",
       "      <th>Mapped reads</th>\n",
       "      <th>Reads after QC %</th>\n",
       "      <th>% On-target (30X)</th>\n",
       "      <th>Mean coverage (X)</th>\n",
       "      <th>Genes covered (%)</th>\n",
       "      <th>SNPs</th>\n",
       "      <th>Indels</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IV-5^</td>\n",
       "      <td>29 191 671</td>\n",
       "      <td>84.95</td>\n",
       "      <td>22 978 888</td>\n",
       "      <td>79,00 %</td>\n",
       "      <td>45 770 776</td>\n",
       "      <td>99.59%</td>\n",
       "      <td>49,00 %</td>\n",
       "      <td>17.79x(1x-856x)</td>\n",
       "      <td>98.48%</td>\n",
       "      <td>114 651</td>\n",
       "      <td>6 935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IV-2^</td>\n",
       "      <td>40 672 795</td>\n",
       "      <td>122.02</td>\n",
       "      <td>39 920 701</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>71 507 872</td>\n",
       "      <td>99.54%</td>\n",
       "      <td>79,00 %</td>\n",
       "      <td>26.76x(1x-860x)</td>\n",
       "      <td>99.70%</td>\n",
       "      <td>226 812</td>\n",
       "      <td>10 226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IV-6^</td>\n",
       "      <td>43 486 476</td>\n",
       "      <td>130.46</td>\n",
       "      <td>38 353 481</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>76 404 013</td>\n",
       "      <td>99.61%</td>\n",
       "      <td>81,00 %</td>\n",
       "      <td>29.66x(1x-817x)</td>\n",
       "      <td>99.71%</td>\n",
       "      <td>207 627</td>\n",
       "      <td>10 263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IV-4^</td>\n",
       "      <td>42 838 012</td>\n",
       "      <td>128.51</td>\n",
       "      <td>37 729 408</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>75 125 893</td>\n",
       "      <td>99.56%</td>\n",
       "      <td>81,00 %</td>\n",
       "      <td>28.76x(1x-877x)</td>\n",
       "      <td>99.91%</td>\n",
       "      <td>229 099</td>\n",
       "      <td>10 325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IV-7^</td>\n",
       "      <td>377 188 93</td>\n",
       "      <td>113.16</td>\n",
       "      <td>33 250 705</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>66 224 206</td>\n",
       "      <td>99.58%</td>\n",
       "      <td>77,00 %</td>\n",
       "      <td>25.82x(1x-847x)</td>\n",
       "      <td>99.70%</td>\n",
       "      <td>203 338</td>\n",
       "      <td>9 474</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Sample   Raw Reads Sequencing depth (X) Reads after QC Reads after QC# %  \\\n",
       "0  IV-5^  29 191 671                84.95     22 978 888           79,00 %   \n",
       "1  IV-2^  40 672 795               122.02     39 920 701           88,00 %   \n",
       "2  IV-6^  43 486 476               130.46     38 353 481           88,00 %   \n",
       "3  IV-4^  42 838 012               128.51     37 729 408           88,00 %   \n",
       "4  IV-7^  377 188 93               113.16     33 250 705           88,00 %   \n",
       "\n",
       "  Mapped reads Reads after QC % % On-target (30X) Mean coverage (X)  \\\n",
       "0   45 770 776           99.59%           49,00 %   17.79x(1x-856x)   \n",
       "1   71 507 872           99.54%           79,00 %   26.76x(1x-860x)   \n",
       "2   76 404 013           99.61%           81,00 %   29.66x(1x-817x)   \n",
       "3   75 125 893           99.56%           81,00 %   28.76x(1x-877x)   \n",
       "4   66 224 206           99.58%           77,00 %   25.82x(1x-847x)   \n",
       "\n",
       "  Genes covered (%)     SNPs  Indels  \n",
       "0            98.48%  114 651   6 935  \n",
       "1            99.70%  226 812  10 226  \n",
       "2            99.71%  207 627  10 263  \n",
       "3            99.91%  229 099  10 325  \n",
       "4            99.70%  203 338   9 474  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table3=pd.read_clipboard()\n",
    "table3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_34176/3769299083.py:4: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.\n",
      "  ).str.replace(\n",
      "/tmp/ipykernel_34176/3769299083.py:6: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.\n",
      "  ).str.replace(\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Raw Reads</th>\n",
       "      <th>Sequencing depth (X)</th>\n",
       "      <th>Reads after QC</th>\n",
       "      <th>Reads after QC# %</th>\n",
       "      <th>Mapped reads</th>\n",
       "      <th>Reads after QC %</th>\n",
       "      <th>% On-target (30X)</th>\n",
       "      <th>Mean coverage (X)</th>\n",
       "      <th>Genes covered (%)</th>\n",
       "      <th>SNPs</th>\n",
       "      <th>Indels</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IV_5_method1</td>\n",
       "      <td>29 191 671</td>\n",
       "      <td>84.95</td>\n",
       "      <td>22 978 888</td>\n",
       "      <td>79,00 %</td>\n",
       "      <td>45 770 776</td>\n",
       "      <td>99.59%</td>\n",
       "      <td>49,00 %</td>\n",
       "      <td>17.79x(1x-856x)</td>\n",
       "      <td>98.48%</td>\n",
       "      <td>114 651</td>\n",
       "      <td>6 935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IV_2_method1</td>\n",
       "      <td>40 672 795</td>\n",
       "      <td>122.02</td>\n",
       "      <td>39 920 701</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>71 507 872</td>\n",
       "      <td>99.54%</td>\n",
       "      <td>79,00 %</td>\n",
       "      <td>26.76x(1x-860x)</td>\n",
       "      <td>99.70%</td>\n",
       "      <td>226 812</td>\n",
       "      <td>10 226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IV_6_method1</td>\n",
       "      <td>43 486 476</td>\n",
       "      <td>130.46</td>\n",
       "      <td>38 353 481</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>76 404 013</td>\n",
       "      <td>99.61%</td>\n",
       "      <td>81,00 %</td>\n",
       "      <td>29.66x(1x-817x)</td>\n",
       "      <td>99.71%</td>\n",
       "      <td>207 627</td>\n",
       "      <td>10 263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IV_4_method1</td>\n",
       "      <td>42 838 012</td>\n",
       "      <td>128.51</td>\n",
       "      <td>37 729 408</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>75 125 893</td>\n",
       "      <td>99.56%</td>\n",
       "      <td>81,00 %</td>\n",
       "      <td>28.76x(1x-877x)</td>\n",
       "      <td>99.91%</td>\n",
       "      <td>229 099</td>\n",
       "      <td>10 325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IV_7_method1</td>\n",
       "      <td>377 188 93</td>\n",
       "      <td>113.16</td>\n",
       "      <td>33 250 705</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>66 224 206</td>\n",
       "      <td>99.58%</td>\n",
       "      <td>77,00 %</td>\n",
       "      <td>25.82x(1x-847x)</td>\n",
       "      <td>99.70%</td>\n",
       "      <td>203 338</td>\n",
       "      <td>9 474</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Sample   Raw Reads Sequencing depth (X) Reads after QC  \\\n",
       "0  IV_5_method1  29 191 671                84.95     22 978 888   \n",
       "1  IV_2_method1  40 672 795               122.02     39 920 701   \n",
       "2  IV_6_method1  43 486 476               130.46     38 353 481   \n",
       "3  IV_4_method1  42 838 012               128.51     37 729 408   \n",
       "4  IV_7_method1  377 188 93               113.16     33 250 705   \n",
       "\n",
       "  Reads after QC# % Mapped reads Reads after QC % % On-target (30X)  \\\n",
       "0           79,00 %   45 770 776           99.59%           49,00 %   \n",
       "1           88,00 %   71 507 872           99.54%           79,00 %   \n",
       "2           88,00 %   76 404 013           99.61%           81,00 %   \n",
       "3           88,00 %   75 125 893           99.56%           81,00 %   \n",
       "4           88,00 %   66 224 206           99.58%           77,00 %   \n",
       "\n",
       "  Mean coverage (X) Genes covered (%)     SNPs  Indels  \n",
       "0   17.79x(1x-856x)            98.48%  114 651   6 935  \n",
       "1   26.76x(1x-860x)            99.70%  226 812  10 226  \n",
       "2   29.66x(1x-817x)            99.71%  207 627  10 263  \n",
       "3   28.76x(1x-877x)            99.91%  229 099  10 325  \n",
       "4   25.82x(1x-847x)            99.70%  203 338   9 474  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## We wolud do some tranformations to the first field (\"Sample\")\n",
    "table3[\"Sample\"] = table3[\"Sample\"].str.replace(\n",
    "    \"-\", \"_\"\n",
    "    ).str.replace(\n",
    "        \"^\", \"_method1\"\n",
    "        ).str.replace(\n",
    "            \"*\", \"_method2\"\n",
    "            )\n",
    "table3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "table3.to_csv(\"../metadata/table3.csv\", index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Raw Reads</th>\n",
       "      <th>Sequencing depth (X)</th>\n",
       "      <th>Reads after QC</th>\n",
       "      <th>Reads after QC# %</th>\n",
       "      <th>Mapped reads</th>\n",
       "      <th>Reads after QC %</th>\n",
       "      <th>% On-target (30X)</th>\n",
       "      <th>Mean coverage (X)</th>\n",
       "      <th>Genes covered (%)</th>\n",
       "      <th>SNPs</th>\n",
       "      <th>Indels</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IV_5_method1</td>\n",
       "      <td>29 191 671</td>\n",
       "      <td>84.95</td>\n",
       "      <td>22 978 888</td>\n",
       "      <td>79,00 %</td>\n",
       "      <td>45 770 776</td>\n",
       "      <td>99.59%</td>\n",
       "      <td>49,00 %</td>\n",
       "      <td>17.79x(1x-856x)</td>\n",
       "      <td>98.48%</td>\n",
       "      <td>114 651</td>\n",
       "      <td>6 935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IV_2_method1</td>\n",
       "      <td>40 672 795</td>\n",
       "      <td>122.02</td>\n",
       "      <td>39 920 701</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>71 507 872</td>\n",
       "      <td>99.54%</td>\n",
       "      <td>79,00 %</td>\n",
       "      <td>26.76x(1x-860x)</td>\n",
       "      <td>99.70%</td>\n",
       "      <td>226 812</td>\n",
       "      <td>10 226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IV_6_method1</td>\n",
       "      <td>43 486 476</td>\n",
       "      <td>130.46</td>\n",
       "      <td>38 353 481</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>76 404 013</td>\n",
       "      <td>99.61%</td>\n",
       "      <td>81,00 %</td>\n",
       "      <td>29.66x(1x-817x)</td>\n",
       "      <td>99.71%</td>\n",
       "      <td>207 627</td>\n",
       "      <td>10 263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IV_4_method1</td>\n",
       "      <td>42 838 012</td>\n",
       "      <td>128.51</td>\n",
       "      <td>37 729 408</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>75 125 893</td>\n",
       "      <td>99.56%</td>\n",
       "      <td>81,00 %</td>\n",
       "      <td>28.76x(1x-877x)</td>\n",
       "      <td>99.91%</td>\n",
       "      <td>229 099</td>\n",
       "      <td>10 325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IV_7_method1</td>\n",
       "      <td>377 188 93</td>\n",
       "      <td>113.16</td>\n",
       "      <td>33 250 705</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>66 224 206</td>\n",
       "      <td>99.58%</td>\n",
       "      <td>77,00 %</td>\n",
       "      <td>25.82x(1x-847x)</td>\n",
       "      <td>99.70%</td>\n",
       "      <td>203 338</td>\n",
       "      <td>9 474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>V_8_method1</td>\n",
       "      <td>55 720 364</td>\n",
       "      <td>222.88</td>\n",
       "      <td>51 583 390</td>\n",
       "      <td>93,00 %</td>\n",
       "      <td>102 574 419</td>\n",
       "      <td>99.43%</td>\n",
       "      <td>82,00 %</td>\n",
       "      <td>49.28x(1x-997x)</td>\n",
       "      <td>99.57%</td>\n",
       "      <td>195 153</td>\n",
       "      <td>12 768</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>V_2_method1</td>\n",
       "      <td>55 054 777</td>\n",
       "      <td>160.22</td>\n",
       "      <td>37 529 435</td>\n",
       "      <td>68,00 %</td>\n",
       "      <td>74 739 463</td>\n",
       "      <td>99.57%</td>\n",
       "      <td>64,00 %</td>\n",
       "      <td>27.24x(1x-998x)</td>\n",
       "      <td>98.62%</td>\n",
       "      <td>137 021</td>\n",
       "      <td>8 655</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>V_1_method1</td>\n",
       "      <td>57 215 132</td>\n",
       "      <td>166.51</td>\n",
       "      <td>34 333 842</td>\n",
       "      <td>60,00 %</td>\n",
       "      <td>68 423 638</td>\n",
       "      <td>99.64%</td>\n",
       "      <td>38,00 %</td>\n",
       "      <td>14.35x(1x-885x)</td>\n",
       "      <td>98.21%</td>\n",
       "      <td>99 034</td>\n",
       "      <td>5 679</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>IV_6_method2</td>\n",
       "      <td>43486476</td>\n",
       "      <td>130,46</td>\n",
       "      <td>38 353 481</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>76 404 013</td>\n",
       "      <td>99,61 %</td>\n",
       "      <td>81,00 %</td>\n",
       "      <td>29.66x(1x-817x)</td>\n",
       "      <td>99,71 %</td>\n",
       "      <td>207 626</td>\n",
       "      <td>10 263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>IV_2_method2</td>\n",
       "      <td>40672795</td>\n",
       "      <td>122,2</td>\n",
       "      <td>35920701</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>71 507 872</td>\n",
       "      <td>99,54 %</td>\n",
       "      <td>79,00 %</td>\n",
       "      <td>26.76x(1x-860x)</td>\n",
       "      <td>99,70 %</td>\n",
       "      <td>226 812</td>\n",
       "      <td>10 226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>IV_4_method2</td>\n",
       "      <td>42838012</td>\n",
       "      <td>125,51</td>\n",
       "      <td>37729408</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>75 125 893</td>\n",
       "      <td>99,56 %</td>\n",
       "      <td>81,00 %</td>\n",
       "      <td>28.76x(1x-877x)</td>\n",
       "      <td>99,91 %</td>\n",
       "      <td>229 099</td>\n",
       "      <td>10 325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>IV_7_method2</td>\n",
       "      <td>37718893</td>\n",
       "      <td>113,16</td>\n",
       "      <td>33250705</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>66 224 205</td>\n",
       "      <td>99,58 %</td>\n",
       "      <td>77,00 %</td>\n",
       "      <td>25.82x(1x-847x)</td>\n",
       "      <td>99,70 %</td>\n",
       "      <td>203 338</td>\n",
       "      <td>9 474</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Sample   Raw Reads Sequencing depth (X) Reads after QC  \\\n",
       "0   IV_5_method1  29 191 671                84.95     22 978 888   \n",
       "1   IV_2_method1  40 672 795               122.02     39 920 701   \n",
       "2   IV_6_method1  43 486 476               130.46     38 353 481   \n",
       "3   IV_4_method1  42 838 012               128.51     37 729 408   \n",
       "4   IV_7_method1  377 188 93               113.16     33 250 705   \n",
       "5    V_8_method1  55 720 364               222.88     51 583 390   \n",
       "6    V_2_method1  55 054 777               160.22     37 529 435   \n",
       "7    V_1_method1  57 215 132               166.51     34 333 842   \n",
       "8   IV_6_method2    43486476               130,46     38 353 481   \n",
       "9   IV_2_method2    40672795                122,2       35920701   \n",
       "10  IV_4_method2    42838012               125,51       37729408   \n",
       "11  IV_7_method2    37718893               113,16       33250705   \n",
       "\n",
       "   Reads after QC# % Mapped reads Reads after QC % % On-target (30X)  \\\n",
       "0            79,00 %   45 770 776           99.59%           49,00 %   \n",
       "1            88,00 %   71 507 872           99.54%           79,00 %   \n",
       "2            88,00 %   76 404 013           99.61%           81,00 %   \n",
       "3            88,00 %   75 125 893           99.56%           81,00 %   \n",
       "4            88,00 %   66 224 206           99.58%           77,00 %   \n",
       "5            93,00 %  102 574 419           99.43%           82,00 %   \n",
       "6            68,00 %   74 739 463           99.57%           64,00 %   \n",
       "7            60,00 %   68 423 638           99.64%           38,00 %   \n",
       "8            88,00 %   76 404 013          99,61 %           81,00 %   \n",
       "9            88,00 %   71 507 872          99,54 %           79,00 %   \n",
       "10           88,00 %   75 125 893          99,56 %           81,00 %   \n",
       "11           88,00 %   66 224 205          99,58 %           77,00 %   \n",
       "\n",
       "   Mean coverage (X) Genes covered (%)     SNPs  Indels  \n",
       "0    17.79x(1x-856x)            98.48%  114 651   6 935  \n",
       "1    26.76x(1x-860x)            99.70%  226 812  10 226  \n",
       "2    29.66x(1x-817x)            99.71%  207 627  10 263  \n",
       "3    28.76x(1x-877x)            99.91%  229 099  10 325  \n",
       "4    25.82x(1x-847x)            99.70%  203 338   9 474  \n",
       "5    49.28x(1x-997x)            99.57%  195 153  12 768  \n",
       "6    27.24x(1x-998x)            98.62%  137 021   8 655  \n",
       "7    14.35x(1x-885x)            98.21%   99 034   5 679  \n",
       "8    29.66x(1x-817x)           99,71 %  207 626  10 263  \n",
       "9    26.76x(1x-860x)           99,70 %  226 812  10 226  \n",
       "10   28.76x(1x-877x)           99,91 %  229 099  10 325  \n",
       "11   25.82x(1x-847x)           99,70 %  203 338   9 474  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table3_csv = pd.read_csv(\"../metadata/table3.csv\")\n",
    "table3_csv"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Amount of reads after removing low quality sequences. \n",
    "### Low quality sequences are considered when\n",
    "\n",
    "1. bases with Q20 account for >85% of the total sequence length \n",
    "2. reads with N-rate >10%\n",
    "\n",
    "### Methods for seguencing\n",
    "\n",
    "* ***method1*** run with Agilent Sureselect XT Human All Exon 50Mb\n",
    "* ***method2*** run with Agilent SureSelect_XT_Human_All_Exon_V5"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ensembl Variation -Calculated variant consequences\n",
    "\n",
    "from this table wi will extract the most severes consequences of Variations\n",
    "\n",
    "https://www.ensembl.org/info/genome/variation/prediction/predicted_data.html#consequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>*</th>\n",
       "      <th>SO term</th>\n",
       "      <th>SO description</th>\n",
       "      <th>SO accession</th>\n",
       "      <th>Display term</th>\n",
       "      <th>IMPACT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>transcript_ablation</td>\n",
       "      <td>A feature ablation whereby the deleted region ...</td>\n",
       "      <td>SO:0001893</td>\n",
       "      <td>Transcript ablation</td>\n",
       "      <td>HIGH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>splice_acceptor_variant</td>\n",
       "      <td>A splice variant that changes the 2 base regio...</td>\n",
       "      <td>SO:0001574</td>\n",
       "      <td>Splice acceptor variant</td>\n",
       "      <td>HIGH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>splice_donor_variant</td>\n",
       "      <td>A splice variant that changes the 2 base regio...</td>\n",
       "      <td>SO:0001575</td>\n",
       "      <td>Splice donor variant</td>\n",
       "      <td>HIGH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>A sequence variant whereby at least one base o...</td>\n",
       "      <td>SO:0001587</td>\n",
       "      <td>Stop gained</td>\n",
       "      <td>HIGH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>A sequence variant which causes a disruption o...</td>\n",
       "      <td>SO:0001589</td>\n",
       "      <td>Frameshift variant</td>\n",
       "      <td>HIGH</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    *                  SO term  \\\n",
       "0 NaN      transcript_ablation   \n",
       "1 NaN  splice_acceptor_variant   \n",
       "2 NaN     splice_donor_variant   \n",
       "3 NaN              stop_gained   \n",
       "4 NaN       frameshift_variant   \n",
       "\n",
       "                                      SO description SO accession  \\\n",
       "0  A feature ablation whereby the deleted region ...   SO:0001893   \n",
       "1  A splice variant that changes the 2 base regio...   SO:0001574   \n",
       "2  A splice variant that changes the 2 base regio...   SO:0001575   \n",
       "3  A sequence variant whereby at least one base o...   SO:0001587   \n",
       "4  A sequence variant which causes a disruption o...   SO:0001589   \n",
       "\n",
       "              Display term IMPACT  \n",
       "0      Transcript ablation   HIGH  \n",
       "1  Splice acceptor variant   HIGH  \n",
       "2     Splice donor variant   HIGH  \n",
       "3              Stop gained   HIGH  \n",
       "4       Frameshift variant   HIGH  "
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "severe_consequences=pd.read_clipboard()\n",
    "severe_consequences.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>so_term</th>\n",
       "      <th>so_description</th>\n",
       "      <th>so_accession</th>\n",
       "      <th>display_term</th>\n",
       "      <th>impact</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>transcript_ablation</td>\n",
       "      <td>A feature ablation whereby the deleted region ...</td>\n",
       "      <td>SO:0001893</td>\n",
       "      <td>Transcript ablation</td>\n",
       "      <td>HIGH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>splice_acceptor_variant</td>\n",
       "      <td>A splice variant that changes the 2 base regio...</td>\n",
       "      <td>SO:0001574</td>\n",
       "      <td>Splice acceptor variant</td>\n",
       "      <td>HIGH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>splice_donor_variant</td>\n",
       "      <td>A splice variant that changes the 2 base regio...</td>\n",
       "      <td>SO:0001575</td>\n",
       "      <td>Splice donor variant</td>\n",
       "      <td>HIGH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>stop_gained</td>\n",
       "      <td>A sequence variant whereby at least one base o...</td>\n",
       "      <td>SO:0001587</td>\n",
       "      <td>Stop gained</td>\n",
       "      <td>HIGH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>A sequence variant which causes a disruption o...</td>\n",
       "      <td>SO:0001589</td>\n",
       "      <td>Frameshift variant</td>\n",
       "      <td>HIGH</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   so_term                                     so_description  \\\n",
       "0      transcript_ablation  A feature ablation whereby the deleted region ...   \n",
       "1  splice_acceptor_variant  A splice variant that changes the 2 base regio...   \n",
       "2     splice_donor_variant  A splice variant that changes the 2 base regio...   \n",
       "3              stop_gained  A sequence variant whereby at least one base o...   \n",
       "4       frameshift_variant  A sequence variant which causes a disruption o...   \n",
       "\n",
       "  so_accession             display_term impact  \n",
       "0   SO:0001893      Transcript ablation   HIGH  \n",
       "1   SO:0001574  Splice acceptor variant   HIGH  \n",
       "2   SO:0001575     Splice donor variant   HIGH  \n",
       "3   SO:0001587              Stop gained   HIGH  \n",
       "4   SO:0001589       Frameshift variant   HIGH  "
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "severe_consequences=severe_consequences.drop(\"*\", axis=1)\n",
    "\n",
    "severe_consequences.columns=severe_consequences.columns.str.replace(\n",
    "    \" \", \"_\"\n",
    "    ).str.lower()\n",
    "\n",
    "severe_consequences.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "severe_consequences.to_csv(\"../metadata/severe_consequences.csv\",\n",
    "                           index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "envpython",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
