{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "7e490edb",
   "metadata": {},
   "source": [
    "# Final anlalysis\n",
    "\n",
    "Now that we have some of the sample we are going to do some biostatistical analysis."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0f514192",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“package ‘ggplot2’ was built under R version 4.2.3”\n",
      "Warning message:\n",
      "“package ‘tibble’ was built under R version 4.2.3”\n",
      "Warning message:\n",
      "“package ‘purrr’ was built under R version 4.2.3”\n",
      "Warning message:\n",
      "“package ‘dplyr’ was built under R version 4.2.3”\n",
      "Warning message:\n",
      "“package ‘stringr’ was built under R version 4.2.3”\n",
      "Warning message:\n",
      "“package ‘forcats’ was built under R version 4.2.3”\n",
      "Warning message:\n",
      "“package ‘lubridate’ was built under R version 4.2.3”\n",
      "── \u001b[1mAttaching core tidyverse packages\u001b[22m ──────────────────────────────────────────────────────────── tidyverse 2.0.0 ──\n",
      "\u001b[32m✔\u001b[39m \u001b[34mdplyr    \u001b[39m 1.1.2     \u001b[32m✔\u001b[39m \u001b[34mreadr    \u001b[39m 2.1.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mforcats  \u001b[39m 1.0.0     \u001b[32m✔\u001b[39m \u001b[34mstringr  \u001b[39m 1.5.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2  \u001b[39m 3.4.2     \u001b[32m✔\u001b[39m \u001b[34mtibble   \u001b[39m 3.2.1\n",
      "\u001b[32m✔\u001b[39m \u001b[34mlubridate\u001b[39m 1.9.2     \u001b[32m✔\u001b[39m \u001b[34mtidyr    \u001b[39m 1.3.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mpurrr    \u001b[39m 1.0.1     \n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[36mℹ\u001b[39m Use the conflicted package (\u001b[3m\u001b[34m<http://conflicted.r-lib.org/>\u001b[39m\u001b[23m) to force all conflicts to become errors\n"
     ]
    }
   ],
   "source": [
    "## Libreries\n",
    "library(tidyverse)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "5e9a22d6",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1mRows: \u001b[22m\u001b[34m30\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m3\u001b[39m\n",
      "\u001b[36m──\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m────────────────────────────────────────────────────────────────────────────────────────────\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \"\\t\"\n",
      "\u001b[31mchr\u001b[39m (2): biotype, sample\n",
      "\u001b[32mdbl\u001b[39m (1): n\n",
      "\n",
      "\u001b[36mℹ\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36mℹ\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n",
      "\u001b[1mRows: \u001b[22m\u001b[34m41\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m3\u001b[39m\n",
      "\u001b[36m──\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m────────────────────────────────────────────────────────────────────────────────────────────\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \"\\t\"\n",
      "\u001b[31mchr\u001b[39m (2): biotype, sample\n",
      "\u001b[32mdbl\u001b[39m (1): n\n",
      "\n",
      "\u001b[36mℹ\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36mℹ\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n",
      "\u001b[1mRows: \u001b[22m\u001b[34m33\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m3\u001b[39m\n",
      "\u001b[36m──\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m────────────────────────────────────────────────────────────────────────────────────────────\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \"\\t\"\n",
      "\u001b[31mchr\u001b[39m (2): biotype, sample\n",
      "\u001b[32mdbl\u001b[39m (1): n\n",
      "\n",
      "\u001b[36mℹ\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36mℹ\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n",
      "\u001b[1mRows: \u001b[22m\u001b[34m44\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m3\u001b[39m\n",
      "\u001b[36m──\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m────────────────────────────────────────────────────────────────────────────────────────────\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \"\\t\"\n",
      "\u001b[31mchr\u001b[39m (2): biotype, sample\n",
      "\u001b[32mdbl\u001b[39m (1): n\n",
      "\n",
      "\u001b[36mℹ\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36mℹ\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n"
     ]
    }
   ],
   "source": [
    "## reading the data ##\n",
    "\n",
    "# Biotype\n",
    "apc_biotype <- read_tsv(\"../results/results_workflow/chromosome5/biotype.tsv\") %>%\n",
    "    mutate(gene=\"APC\") \n",
    "braf_biotype <- read_tsv(\"../results/results_workflow/chromosome7/biotype.tsv\") %>%\n",
    "    mutate(gene=\"BRAF\")\n",
    "kras_biotype <- read_tsv(\"../results/results_workflow/chromosome12/biotype.tsv\") %>%\n",
    "    mutate(gene=\"KRAS\")\n",
    "tp53_biotype <- read_tsv(\"../results/results_workflow/chromosome17/biotype.tsv\") %>%\n",
    "    mutate(gene=\"TP53\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "2b90ece6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[90m# A tibble: 148 × 4\u001b[39m\n",
      "    biotype                            n sample          gene \n",
      "    \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                          \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m           \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m\n",
      "\u001b[90m  1\u001b[39m protein_coding                    45 ERR753368_chr5  APC  \n",
      "\u001b[90m  2\u001b[39m nonsense_mediated_decay           17 ERR753368_chr5  APC  \n",
      "\u001b[90m  3\u001b[39m retained_intron                    2 ERR753368_chr5  APC  \n",
      "\u001b[90m  4\u001b[39m protein_coding                    49 ERR753369_chr5  APC  \n",
      "\u001b[90m  5\u001b[39m nonsense_mediated_decay           18 ERR753369_chr5  APC  \n",
      "\u001b[90m  6\u001b[39m retained_intron                    2 ERR753369_chr5  APC  \n",
      "\u001b[90m  7\u001b[39m protein_coding                    61 ERR753370_chr5  APC  \n",
      "\u001b[90m  8\u001b[39m nonsense_mediated_decay           22 ERR753370_chr5  APC  \n",
      "\u001b[90m  9\u001b[39m retained_intron                    3 ERR753370_chr5  APC  \n",
      "\u001b[90m 10\u001b[39m protein_coding                    45 ERR753372_chr5  APC  \n",
      "\u001b[90m 11\u001b[39m nonsense_mediated_decay           17 ERR753372_chr5  APC  \n",
      "\u001b[90m 12\u001b[39m retained_intron                    2 ERR753372_chr5  APC  \n",
      "\u001b[90m 13\u001b[39m protein_coding                    69 ERR753373_chr5  APC  \n",
      "\u001b[90m 14\u001b[39m nonsense_mediated_decay           24 ERR753373_chr5  APC  \n",
      "\u001b[90m 15\u001b[39m retained_intron                    3 ERR753373_chr5  APC  \n",
      "\u001b[90m 16\u001b[39m protein_coding                    43 ERR753374_chr5  APC  \n",
      "\u001b[90m 17\u001b[39m nonsense_mediated_decay           16 ERR753374_chr5  APC  \n",
      "\u001b[90m 18\u001b[39m retained_intron                    2 ERR753374_chr5  APC  \n",
      "\u001b[90m 19\u001b[39m protein_coding                    48 ERR753375_chr5  APC  \n",
      "\u001b[90m 20\u001b[39m nonsense_mediated_decay           18 ERR753375_chr5  APC  \n",
      "\u001b[90m 21\u001b[39m retained_intron                    3 ERR753375_chr5  APC  \n",
      "\u001b[90m 22\u001b[39m protein_coding                    61 ERR753376_chr5  APC  \n",
      "\u001b[90m 23\u001b[39m nonsense_mediated_decay           22 ERR753376_chr5  APC  \n",
      "\u001b[90m 24\u001b[39m retained_intron                    2 ERR753376_chr5  APC  \n",
      "\u001b[90m 25\u001b[39m protein_coding                    61 ERR753377_chr5  APC  \n",
      "\u001b[90m 26\u001b[39m nonsense_mediated_decay           22 ERR753377_chr5  APC  \n",
      "\u001b[90m 27\u001b[39m retained_intron                    2 ERR753377_chr5  APC  \n",
      "\u001b[90m 28\u001b[39m protein_coding                    52 ERR753378_chr5  APC  \n",
      "\u001b[90m 29\u001b[39m nonsense_mediated_decay           19 ERR753378_chr5  APC  \n",
      "\u001b[90m 30\u001b[39m retained_intron                    2 ERR753378_chr5  APC  \n",
      "\u001b[90m 31\u001b[39m nonsense_mediated_decay           20 ERR696683_chr7  BRAF \n",
      "\u001b[90m 32\u001b[39m protein_coding                    20 ERR696683_chr7  BRAF \n",
      "\u001b[90m 33\u001b[39m retained_intron                   11 ERR696683_chr7  BRAF \n",
      "\u001b[90m 34\u001b[39m protein_coding_CDS_not_defined     6 ERR696683_chr7  BRAF \n",
      "\u001b[90m 35\u001b[39m nonsense_mediated_decay           20 ERR753368_chr7  BRAF \n",
      "\u001b[90m 36\u001b[39m protein_coding                    17 ERR753368_chr7  BRAF \n",
      "\u001b[90m 37\u001b[39m retained_intron                   10 ERR753368_chr7  BRAF \n",
      "\u001b[90m 38\u001b[39m protein_coding_CDS_not_defined     5 ERR753368_chr7  BRAF \n",
      "\u001b[90m 39\u001b[39m protein_coding                     3 ERR753369_chr7  BRAF \n",
      "\u001b[90m 40\u001b[39m protein_coding_CDS_not_defined     1 ERR753369_chr7  BRAF \n",
      "\u001b[90m 41\u001b[39m retained_intron                    1 ERR753369_chr7  BRAF \n",
      "\u001b[90m 42\u001b[39m protein_coding                     3 ERR753370_chr7  BRAF \n",
      "\u001b[90m 43\u001b[39m protein_coding_CDS_not_defined     1 ERR753370_chr7  BRAF \n",
      "\u001b[90m 44\u001b[39m retained_intron                    1 ERR753370_chr7  BRAF \n",
      "\u001b[90m 45\u001b[39m nonsense_mediated_decay           12 ERR753371_chr7  BRAF \n",
      "\u001b[90m 46\u001b[39m protein_coding                    11 ERR753371_chr7  BRAF \n",
      "\u001b[90m 47\u001b[39m retained_intron                    5 ERR753371_chr7  BRAF \n",
      "\u001b[90m 48\u001b[39m protein_coding_CDS_not_defined     3 ERR753371_chr7  BRAF \n",
      "\u001b[90m 49\u001b[39m nonsense_mediated_decay           11 ERR753372_chr7  BRAF \n",
      "\u001b[90m 50\u001b[39m protein_coding                    11 ERR753372_chr7  BRAF \n",
      "\u001b[90m 51\u001b[39m retained_intron                    6 ERR753372_chr7  BRAF \n",
      "\u001b[90m 52\u001b[39m protein_coding_CDS_not_defined     3 ERR753372_chr7  BRAF \n",
      "\u001b[90m 53\u001b[39m nonsense_mediated_decay           20 ERR753373_chr7  BRAF \n",
      "\u001b[90m 54\u001b[39m protein_coding                    17 ERR753373_chr7  BRAF \n",
      "\u001b[90m 55\u001b[39m retained_intron                   10 ERR753373_chr7  BRAF \n",
      "\u001b[90m 56\u001b[39m protein_coding_CDS_not_defined     5 ERR753373_chr7  BRAF \n",
      "\u001b[90m 57\u001b[39m nonsense_mediated_decay           17 ERR753374_chr7  BRAF \n",
      "\u001b[90m 58\u001b[39m protein_coding                    12 ERR753374_chr7  BRAF \n",
      "\u001b[90m 59\u001b[39m retained_intron                    7 ERR753374_chr7  BRAF \n",
      "\u001b[90m 60\u001b[39m protein_coding_CDS_not_defined     3 ERR753374_chr7  BRAF \n",
      "\u001b[90m 61\u001b[39m nonsense_mediated_decay           17 ERR753375_chr7  BRAF \n",
      "\u001b[90m 62\u001b[39m protein_coding                    15 ERR753375_chr7  BRAF \n",
      "\u001b[90m 63\u001b[39m retained_intron                    8 ERR753375_chr7  BRAF \n",
      "\u001b[90m 64\u001b[39m protein_coding_CDS_not_defined     4 ERR753375_chr7  BRAF \n",
      "\u001b[90m 65\u001b[39m protein_coding                     3 ERR753376_chr7  BRAF \n",
      "\u001b[90m 66\u001b[39m protein_coding_CDS_not_defined     1 ERR753376_chr7  BRAF \n",
      "\u001b[90m 67\u001b[39m retained_intron                    1 ERR753376_chr7  BRAF \n",
      "\u001b[90m 68\u001b[39m nonsense_mediated_decay           11 ERR753378_chr7  BRAF \n",
      "\u001b[90m 69\u001b[39m protein_coding                     8 ERR753378_chr7  BRAF \n",
      "\u001b[90m 70\u001b[39m retained_intron                    5 ERR753378_chr7  BRAF \n",
      "\u001b[90m 71\u001b[39m protein_coding_CDS_not_defined     2 ERR753378_chr7  BRAF \n",
      "\u001b[90m 72\u001b[39m protein_coding                    69 ERR696683_chr17 TP53 \n",
      "\u001b[90m 73\u001b[39m nonsense_mediated_decay            3 ERR696683_chr17 TP53 \n",
      "\u001b[90m 74\u001b[39m protein_coding_CDS_not_defined     3 ERR696683_chr17 TP53 \n",
      "\u001b[90m 75\u001b[39m retained_intron                    3 ERR696683_chr17 TP53 \n",
      "\u001b[90m 76\u001b[39m protein_coding                   131 ERR753368_chr17 TP53 \n",
      "\u001b[90m 77\u001b[39m retained_intron                   10 ERR753368_chr17 TP53 \n",
      "\u001b[90m 78\u001b[39m nonsense_mediated_decay            7 ERR753368_chr17 TP53 \n",
      "\u001b[90m 79\u001b[39m protein_coding_CDS_not_defined     4 ERR753368_chr17 TP53 \n",
      "\u001b[90m 80\u001b[39m protein_coding                   218 ERR753369_chr17 TP53 \n",
      "\u001b[90m 81\u001b[39m retained_intron                   13 ERR753369_chr17 TP53 \n",
      "\u001b[90m 82\u001b[39m nonsense_mediated_decay           11 ERR753369_chr17 TP53 \n",
      "\u001b[90m 83\u001b[39m protein_coding_CDS_not_defined     7 ERR753369_chr17 TP53 \n",
      "\u001b[90m 84\u001b[39m protein_coding                   115 ERR753370_chr17 TP53 \n",
      "\u001b[90m 85\u001b[39m nonsense_mediated_decay            5 ERR753370_chr17 TP53 \n",
      "\u001b[90m 86\u001b[39m protein_coding_CDS_not_defined     5 ERR753370_chr17 TP53 \n",
      "\u001b[90m 87\u001b[39m retained_intron                    5 ERR753370_chr17 TP53 \n",
      "\u001b[90m 88\u001b[39m protein_coding                    72 ERR753371_chr17 TP53 \n",
      "\u001b[90m 89\u001b[39m retained_intron                    6 ERR753371_chr17 TP53 \n",
      "\u001b[90m 90\u001b[39m nonsense_mediated_decay            4 ERR753371_chr17 TP53 \n",
      "\u001b[90m 91\u001b[39m protein_coding_CDS_not_defined     2 ERR753371_chr17 TP53 \n",
      "\u001b[90m 92\u001b[39m protein_coding                   131 ERR753372_chr17 TP53 \n",
      "\u001b[90m 93\u001b[39m retained_intron                   10 ERR753372_chr17 TP53 \n",
      "\u001b[90m 94\u001b[39m nonsense_mediated_decay            7 ERR753372_chr17 TP53 \n",
      "\u001b[90m 95\u001b[39m protein_coding_CDS_not_defined     4 ERR753372_chr17 TP53 \n",
      "\u001b[90m 96\u001b[39m protein_coding                    59 ERR753373_chr17 TP53 \n",
      "\u001b[90m 97\u001b[39m retained_intron                    4 ERR753373_chr17 TP53 \n",
      "\u001b[90m 98\u001b[39m nonsense_mediated_decay            3 ERR753373_chr17 TP53 \n",
      "\u001b[90m 99\u001b[39m protein_coding_CDS_not_defined     2 ERR753373_chr17 TP53 \n",
      "\u001b[90m100\u001b[39m protein_coding                   105 ERR753375_chr17 TP53 \n",
      "\u001b[90m101\u001b[39m retained_intron                    6 ERR753375_chr17 TP53 \n",
      "\u001b[90m102\u001b[39m nonsense_mediated_decay            5 ERR753375_chr17 TP53 \n",
      "\u001b[90m103\u001b[39m protein_coding_CDS_not_defined     4 ERR753375_chr17 TP53 \n",
      "\u001b[90m104\u001b[39m protein_coding                    92 ERR753376_chr17 TP53 \n",
      "\u001b[90m105\u001b[39m nonsense_mediated_decay            4 ERR753376_chr17 TP53 \n",
      "\u001b[90m106\u001b[39m protein_coding_CDS_not_defined     4 ERR753376_chr17 TP53 \n",
      "\u001b[90m107\u001b[39m retained_intron                    4 ERR753376_chr17 TP53 \n",
      "\u001b[90m108\u001b[39m protein_coding                    69 ERR753377_chr17 TP53 \n",
      "\u001b[90m109\u001b[39m nonsense_mediated_decay            3 ERR753377_chr17 TP53 \n",
      "\u001b[90m110\u001b[39m protein_coding_CDS_not_defined     3 ERR753377_chr17 TP53 \n",
      "\u001b[90m111\u001b[39m retained_intron                    3 ERR753377_chr17 TP53 \n",
      "\u001b[90m112\u001b[39m protein_coding                   133 ERR753378_chr17 TP53 \n",
      "\u001b[90m113\u001b[39m nonsense_mediated_decay            6 ERR753378_chr17 TP53 \n",
      "\u001b[90m114\u001b[39m protein_coding_CDS_not_defined     5 ERR753378_chr17 TP53 \n",
      "\u001b[90m115\u001b[39m retained_intron                    5 ERR753378_chr17 TP53 \n",
      "\u001b[90m116\u001b[39m protein_coding                     7 ERR696683_chr12 KRAS \n",
      "\u001b[90m117\u001b[39m nonsense_mediated_decay            4 ERR696683_chr12 KRAS \n",
      "\u001b[90m118\u001b[39m retained_intron                    1 ERR696683_chr12 KRAS \n",
      "\u001b[90m119\u001b[39m protein_coding                    16 ERR753368_chr12 KRAS \n",
      "\u001b[90m120\u001b[39m nonsense_mediated_decay            7 ERR753368_chr12 KRAS \n",
      "\u001b[90m121\u001b[39m retained_intron                    1 ERR753368_chr12 KRAS \n",
      "\u001b[90m122\u001b[39m protein_coding                    25 ERR753369_chr12 KRAS \n",
      "\u001b[90m123\u001b[39m nonsense_mediated_decay           10 ERR753369_chr12 KRAS \n",
      "\u001b[90m124\u001b[39m retained_intron                    1 ERR753369_chr12 KRAS \n",
      "\u001b[90m125\u001b[39m protein_coding                    21 ERR753370_chr12 KRAS \n",
      "\u001b[90m126\u001b[39m nonsense_mediated_decay           12 ERR753370_chr12 KRAS \n",
      "\u001b[90m127\u001b[39m retained_intron                    3 ERR753370_chr12 KRAS \n",
      "\u001b[90m128\u001b[39m protein_coding                     7 ERR753371_chr12 KRAS \n",
      "\u001b[90m129\u001b[39m nonsense_mediated_decay            4 ERR753371_chr12 KRAS \n",
      "\u001b[90m130\u001b[39m retained_intron                    1 ERR753371_chr12 KRAS \n",
      "\u001b[90m131\u001b[39m protein_coding                    23 ERR753372_chr12 KRAS \n",
      "\u001b[90m132\u001b[39m nonsense_mediated_decay           11 ERR753372_chr12 KRAS \n",
      "\u001b[90m133\u001b[39m retained_intron                    2 ERR753372_chr12 KRAS \n",
      "\u001b[90m134\u001b[39m protein_coding                    18 ERR753373_chr12 KRAS \n",
      "\u001b[90m135\u001b[39m nonsense_mediated_decay            6 ERR753373_chr12 KRAS \n",
      "\u001b[90m136\u001b[39m retained_intron                    1 ERR753373_chr12 KRAS \n",
      "\u001b[90m137\u001b[39m protein_coding                    14 ERR753375_chr12 KRAS \n",
      "\u001b[90m138\u001b[39m nonsense_mediated_decay            8 ERR753375_chr12 KRAS \n",
      "\u001b[90m139\u001b[39m retained_intron                    2 ERR753375_chr12 KRAS \n",
      "\u001b[90m140\u001b[39m protein_coding                     7 ERR753376_chr12 KRAS \n",
      "\u001b[90m141\u001b[39m nonsense_mediated_decay            4 ERR753376_chr12 KRAS \n",
      "\u001b[90m142\u001b[39m retained_intron                    1 ERR753376_chr12 KRAS \n",
      "\u001b[90m143\u001b[39m protein_coding                     7 ERR753377_chr12 KRAS \n",
      "\u001b[90m144\u001b[39m nonsense_mediated_decay            4 ERR753377_chr12 KRAS \n",
      "\u001b[90m145\u001b[39m retained_intron                    1 ERR753377_chr12 KRAS \n",
      "\u001b[90m146\u001b[39m protein_coding                    25 ERR753378_chr12 KRAS \n",
      "\u001b[90m147\u001b[39m nonsense_mediated_decay           10 ERR753378_chr12 KRAS \n",
      "\u001b[90m148\u001b[39m retained_intron                    1 ERR753378_chr12 KRAS \n"
     ]
    }
   ],
   "source": [
    "## Joining the data\n",
    "biotype <- rbind(apc_biotype,braf_biotype,tp53_biotype,kras_biotype)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "d3030daa",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Processing the data\n",
    "df_biotype <- expand.grid(biotype=as.character(unique(as.character(biotype$biotype))),\n",
    "                          sample=as.character(unique(as.character(biotype$sample))),\n",
    "                          gene=as.character(unique(as.character(biotype$gene)))) %>% \n",
    "    left_join(\n",
    "        biotype, by=c(\"biotype\",\"sample\",\"gene\")\n",
    "        ) %>% \n",
    "    mutate(n=ifelse(is.na(n),0,n)) %>%\n",
    "    separate_wider_delim(sample,delim = \"_\", names = c(\"sample\", \"chr\")) %>%\n",
    "    select(-\"chr\") \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "a8f6a131",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'gene'. You can override using the `.groups` argument.\n",
      "Warning message:\n",
      "“\u001b[1m\u001b[22mRemoved 2 rows containing missing values (`geom_bar()`).”\n"
     ]
    }
   ],
   "source": [
    "## Ploting ggplot2 biotype-genes\n",
    "biotype_plot <- df_biotype %>%\n",
    "    group_by(gene,biotype) %>%\n",
    "    summarise(n=sum(n)) %>%\n",
    "    mutate(biotype=factor(biotype,\n",
    "                          levels = c(\"protein_coding\", \"protein_coding_CDS_not_defined\", \n",
    "                                     \"nonsense_mediated_decay\", \"retained_intron\"),\n",
    "                          labels = c(\"Portein coding\", \"Protein coding (not define)\",\n",
    "                                     \"Nonsense mediated decay\", \"Retained intron\"))) %>%\n",
    "    ggplot(aes(reorder(gene,n),n, fill=biotype)) +\n",
    "    geom_bar(stat=\"identity\",position=position_dodge(.65), width=.5,\n",
    "             color=\"black\") +\n",
    "    scale_y_continuous(expand=expansion(0),\n",
    "                       limits=c(0,270),\n",
    "                       breaks=seq(0,250,25)) +\n",
    "    scale_fill_manual(values=c(\"darkgray\",\"red\", \"blue\", \"forestgreen\")) +\n",
    "    labs(\n",
    "        title=\"Variant biotypes per genes\",\n",
    "        x=\"Samples\",\n",
    "        y=\"Number of variants\",\n",
    "        fill=\"Biotype\"\n",
    "    ) +\n",
    "    theme_classic() +\n",
    "    theme(\n",
    "        plot.title=element_text(hjust=.5, face=\"bold\", size=14),\n",
    "        axis.title=element_text(face=\"bold\", size=12),\n",
    "        axis.text=element_text(color=\"black\", size=10),\n",
    "        axis.ticks=element_blank(),\n",
    "        legend.position=c(.3,.8),\n",
    "        legend.background=element_rect(color=\"black\")\n",
    "    )\n",
    "\n",
    "\n",
    "ggsave(file = \"../results_workflow/plots_workflow/biotype.png\",\n",
    "       plot = biotype_plot,\n",
    "       width = 7,\n",
    "       heigh = 5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "9cfc68e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "?pivot_wider()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.2.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
