{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Annotations of the workflow\n",
    "\n",
    "The idea is to get more content about the project/workflow.\n",
    "\n",
    "* Conda environment for this Notebook: [Greference_tools](../environments/Greference_tools.yml)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Import libreries\n",
    "\n",
    "# Base packages\n",
    "import subprocess\n",
    "import os\n",
    "import re\n",
    "# Data manipulation\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Metadata/report.csv\n",
    "\n",
    "Lets see our Project and all the fields.\n",
    "\n",
    "For this project we will be interested in download the Bam files in the ***8 field***: **submitted_ftp**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_accession</th>\n",
       "      <th>sample_accession</th>\n",
       "      <th>experiment_accession</th>\n",
       "      <th>run_accession</th>\n",
       "      <th>tax_id</th>\n",
       "      <th>scientific_name</th>\n",
       "      <th>fastq_ftp</th>\n",
       "      <th>submitted_ftp</th>\n",
       "      <th>sra_ftp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PRJEB7926</td>\n",
       "      <td>SAMEA3168442</td>\n",
       "      <td>ERX641122</td>\n",
       "      <td>ERR696683</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/fastq/ERR696/ERR696683/...</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR696/ERR696683/C2...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PRJEB7926</td>\n",
       "      <td>SAMEA3168443</td>\n",
       "      <td>ERX697048</td>\n",
       "      <td>ERR753368</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/fastq/ERR753/ERR753368/...</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR753/ERR753368/C2...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PRJEB7926</td>\n",
       "      <td>SAMEA3168444</td>\n",
       "      <td>ERX697049</td>\n",
       "      <td>ERR753369</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/fastq/ERR753/ERR753369/...</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR753/ERR753369/C2...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PRJEB7926</td>\n",
       "      <td>SAMEA3168445</td>\n",
       "      <td>ERX697050</td>\n",
       "      <td>ERR753370</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/fastq/ERR753/ERR753370/...</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR753/ERR753370/C3...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PRJEB7926</td>\n",
       "      <td>SAMEA3168446</td>\n",
       "      <td>ERX697051</td>\n",
       "      <td>ERR753371</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/fastq/ERR753/ERR753371/...</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR753/ERR753371/C3...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PRJEB7926</td>\n",
       "      <td>SAMEA3168447</td>\n",
       "      <td>ERX697052</td>\n",
       "      <td>ERR753372</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/fastq/ERR753/ERR753372/...</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR753/ERR753372/C3...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PRJEB7926</td>\n",
       "      <td>SAMEA3168448</td>\n",
       "      <td>ERX697053</td>\n",
       "      <td>ERR753373</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/fastq/ERR753/ERR753373/...</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR753/ERR753373/NG...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PRJEB7926</td>\n",
       "      <td>SAMEA3168449</td>\n",
       "      <td>ERX697054</td>\n",
       "      <td>ERR753374</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/fastq/ERR753/ERR753374/...</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR753/ERR753374/NG...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PRJEB7926</td>\n",
       "      <td>SAMEA3168450</td>\n",
       "      <td>ERX697055</td>\n",
       "      <td>ERR753375</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/fastq/ERR753/ERR753375/...</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR753/ERR753375/C-...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PRJEB7926</td>\n",
       "      <td>SAMEA3168451</td>\n",
       "      <td>ERX697056</td>\n",
       "      <td>ERR753376</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/fastq/ERR753/ERR753376/...</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR753/ERR753376/C2...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PRJEB7926</td>\n",
       "      <td>SAMEA3168452</td>\n",
       "      <td>ERX697057</td>\n",
       "      <td>ERR753377</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/fastq/ERR753/ERR753377/...</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR753/ERR753377/C8...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PRJEB7926</td>\n",
       "      <td>SAMEA3168453</td>\n",
       "      <td>ERX697058</td>\n",
       "      <td>ERR753378</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/fastq/ERR753/ERR753378/...</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR753/ERR753378/C3...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   study_accession sample_accession experiment_accession run_accession  \\\n",
       "0        PRJEB7926     SAMEA3168442            ERX641122     ERR696683   \n",
       "1        PRJEB7926     SAMEA3168443            ERX697048     ERR753368   \n",
       "2        PRJEB7926     SAMEA3168444            ERX697049     ERR753369   \n",
       "3        PRJEB7926     SAMEA3168445            ERX697050     ERR753370   \n",
       "4        PRJEB7926     SAMEA3168446            ERX697051     ERR753371   \n",
       "5        PRJEB7926     SAMEA3168447            ERX697052     ERR753372   \n",
       "6        PRJEB7926     SAMEA3168448            ERX697053     ERR753373   \n",
       "7        PRJEB7926     SAMEA3168449            ERX697054     ERR753374   \n",
       "8        PRJEB7926     SAMEA3168450            ERX697055     ERR753375   \n",
       "9        PRJEB7926     SAMEA3168451            ERX697056     ERR753376   \n",
       "10       PRJEB7926     SAMEA3168452            ERX697057     ERR753377   \n",
       "11       PRJEB7926     SAMEA3168453            ERX697058     ERR753378   \n",
       "\n",
       "    tax_id scientific_name                                          fastq_ftp  \\\n",
       "0     9606    Homo sapiens  ftp.sra.ebi.ac.uk/vol1/fastq/ERR696/ERR696683/...   \n",
       "1     9606    Homo sapiens  ftp.sra.ebi.ac.uk/vol1/fastq/ERR753/ERR753368/...   \n",
       "2     9606    Homo sapiens  ftp.sra.ebi.ac.uk/vol1/fastq/ERR753/ERR753369/...   \n",
       "3     9606    Homo sapiens  ftp.sra.ebi.ac.uk/vol1/fastq/ERR753/ERR753370/...   \n",
       "4     9606    Homo sapiens  ftp.sra.ebi.ac.uk/vol1/fastq/ERR753/ERR753371/...   \n",
       "5     9606    Homo sapiens  ftp.sra.ebi.ac.uk/vol1/fastq/ERR753/ERR753372/...   \n",
       "6     9606    Homo sapiens  ftp.sra.ebi.ac.uk/vol1/fastq/ERR753/ERR753373/...   \n",
       "7     9606    Homo sapiens  ftp.sra.ebi.ac.uk/vol1/fastq/ERR753/ERR753374/...   \n",
       "8     9606    Homo sapiens  ftp.sra.ebi.ac.uk/vol1/fastq/ERR753/ERR753375/...   \n",
       "9     9606    Homo sapiens  ftp.sra.ebi.ac.uk/vol1/fastq/ERR753/ERR753376/...   \n",
       "10    9606    Homo sapiens  ftp.sra.ebi.ac.uk/vol1/fastq/ERR753/ERR753377/...   \n",
       "11    9606    Homo sapiens  ftp.sra.ebi.ac.uk/vol1/fastq/ERR753/ERR753378/...   \n",
       "\n",
       "                                        submitted_ftp  sra_ftp  \n",
       "0   ftp.sra.ebi.ac.uk/vol1/run/ERR696/ERR696683/C2...      NaN  \n",
       "1   ftp.sra.ebi.ac.uk/vol1/run/ERR753/ERR753368/C2...      NaN  \n",
       "2   ftp.sra.ebi.ac.uk/vol1/run/ERR753/ERR753369/C2...      NaN  \n",
       "3   ftp.sra.ebi.ac.uk/vol1/run/ERR753/ERR753370/C3...      NaN  \n",
       "4   ftp.sra.ebi.ac.uk/vol1/run/ERR753/ERR753371/C3...      NaN  \n",
       "5   ftp.sra.ebi.ac.uk/vol1/run/ERR753/ERR753372/C3...      NaN  \n",
       "6   ftp.sra.ebi.ac.uk/vol1/run/ERR753/ERR753373/NG...      NaN  \n",
       "7   ftp.sra.ebi.ac.uk/vol1/run/ERR753/ERR753374/NG...      NaN  \n",
       "8   ftp.sra.ebi.ac.uk/vol1/run/ERR753/ERR753375/C-...      NaN  \n",
       "9   ftp.sra.ebi.ac.uk/vol1/run/ERR753/ERR753376/C2...      NaN  \n",
       "10  ftp.sra.ebi.ac.uk/vol1/run/ERR753/ERR753377/C8...      NaN  \n",
       "11  ftp.sra.ebi.ac.uk/vol1/run/ERR753/ERR753378/C3...      NaN  "
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "report = pd.read_csv(\"../../metadata/report.tsv\", sep=\"\\t\")\n",
    "report"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Downloading the table 3, for supporting information found in the article.\n",
    "\n",
    "After that we will use the function in pandas ```pd.read_clipboard()``` to download the data.\n",
    "\n",
    "https://onlinelibrary.wiley.com/action/downloadSupplement?doi=10.1002%2Fgcc.22314&file=gcc22314-sup-0005-supptable5.docx"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Raw Reads</th>\n",
       "      <th>Sequencing depth (X)</th>\n",
       "      <th>Reads after QC</th>\n",
       "      <th>Reads after QC# %</th>\n",
       "      <th>Mapped reads</th>\n",
       "      <th>Reads after QC %</th>\n",
       "      <th>% On-target (30X)</th>\n",
       "      <th>Mean coverage (X)</th>\n",
       "      <th>Genes covered (%)</th>\n",
       "      <th>SNPs</th>\n",
       "      <th>Indels</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IV-5^</td>\n",
       "      <td>29 191 671</td>\n",
       "      <td>84.95</td>\n",
       "      <td>22 978 888</td>\n",
       "      <td>79,00 %</td>\n",
       "      <td>45 770 776</td>\n",
       "      <td>99.59%</td>\n",
       "      <td>49,00 %</td>\n",
       "      <td>17.79x(1x-856x)</td>\n",
       "      <td>98.48%</td>\n",
       "      <td>114 651</td>\n",
       "      <td>6 935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IV-2^</td>\n",
       "      <td>40 672 795</td>\n",
       "      <td>122.02</td>\n",
       "      <td>39 920 701</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>71 507 872</td>\n",
       "      <td>99.54%</td>\n",
       "      <td>79,00 %</td>\n",
       "      <td>26.76x(1x-860x)</td>\n",
       "      <td>99.70%</td>\n",
       "      <td>226 812</td>\n",
       "      <td>10 226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IV-6^</td>\n",
       "      <td>43 486 476</td>\n",
       "      <td>130.46</td>\n",
       "      <td>38 353 481</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>76 404 013</td>\n",
       "      <td>99.61%</td>\n",
       "      <td>81,00 %</td>\n",
       "      <td>29.66x(1x-817x)</td>\n",
       "      <td>99.71%</td>\n",
       "      <td>207 627</td>\n",
       "      <td>10 263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IV-4^</td>\n",
       "      <td>42 838 012</td>\n",
       "      <td>128.51</td>\n",
       "      <td>37 729 408</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>75 125 893</td>\n",
       "      <td>99.56%</td>\n",
       "      <td>81,00 %</td>\n",
       "      <td>28.76x(1x-877x)</td>\n",
       "      <td>99.91%</td>\n",
       "      <td>229 099</td>\n",
       "      <td>10 325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IV-7^</td>\n",
       "      <td>377 188 93</td>\n",
       "      <td>113.16</td>\n",
       "      <td>33 250 705</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>66 224 206</td>\n",
       "      <td>99.58%</td>\n",
       "      <td>77,00 %</td>\n",
       "      <td>25.82x(1x-847x)</td>\n",
       "      <td>99.70%</td>\n",
       "      <td>203 338</td>\n",
       "      <td>9 474</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Sample   Raw Reads Sequencing depth (X) Reads after QC Reads after QC# %  \\\n",
       "0  IV-5^  29 191 671                84.95     22 978 888           79,00 %   \n",
       "1  IV-2^  40 672 795               122.02     39 920 701           88,00 %   \n",
       "2  IV-6^  43 486 476               130.46     38 353 481           88,00 %   \n",
       "3  IV-4^  42 838 012               128.51     37 729 408           88,00 %   \n",
       "4  IV-7^  377 188 93               113.16     33 250 705           88,00 %   \n",
       "\n",
       "  Mapped reads Reads after QC % % On-target (30X) Mean coverage (X)  \\\n",
       "0   45 770 776           99.59%           49,00 %   17.79x(1x-856x)   \n",
       "1   71 507 872           99.54%           79,00 %   26.76x(1x-860x)   \n",
       "2   76 404 013           99.61%           81,00 %   29.66x(1x-817x)   \n",
       "3   75 125 893           99.56%           81,00 %   28.76x(1x-877x)   \n",
       "4   66 224 206           99.58%           77,00 %   25.82x(1x-847x)   \n",
       "\n",
       "  Genes covered (%)     SNPs  Indels  \n",
       "0            98.48%  114 651   6 935  \n",
       "1            99.70%  226 812  10 226  \n",
       "2            99.71%  207 627  10 263  \n",
       "3            99.91%  229 099  10 325  \n",
       "4            99.70%  203 338   9 474  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table3=pd.read_clipboard()\n",
    "table3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_34176/3769299083.py:4: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.\n",
      "  ).str.replace(\n",
      "/tmp/ipykernel_34176/3769299083.py:6: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.\n",
      "  ).str.replace(\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Raw Reads</th>\n",
       "      <th>Sequencing depth (X)</th>\n",
       "      <th>Reads after QC</th>\n",
       "      <th>Reads after QC# %</th>\n",
       "      <th>Mapped reads</th>\n",
       "      <th>Reads after QC %</th>\n",
       "      <th>% On-target (30X)</th>\n",
       "      <th>Mean coverage (X)</th>\n",
       "      <th>Genes covered (%)</th>\n",
       "      <th>SNPs</th>\n",
       "      <th>Indels</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IV_5_method1</td>\n",
       "      <td>29 191 671</td>\n",
       "      <td>84.95</td>\n",
       "      <td>22 978 888</td>\n",
       "      <td>79,00 %</td>\n",
       "      <td>45 770 776</td>\n",
       "      <td>99.59%</td>\n",
       "      <td>49,00 %</td>\n",
       "      <td>17.79x(1x-856x)</td>\n",
       "      <td>98.48%</td>\n",
       "      <td>114 651</td>\n",
       "      <td>6 935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IV_2_method1</td>\n",
       "      <td>40 672 795</td>\n",
       "      <td>122.02</td>\n",
       "      <td>39 920 701</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>71 507 872</td>\n",
       "      <td>99.54%</td>\n",
       "      <td>79,00 %</td>\n",
       "      <td>26.76x(1x-860x)</td>\n",
       "      <td>99.70%</td>\n",
       "      <td>226 812</td>\n",
       "      <td>10 226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IV_6_method1</td>\n",
       "      <td>43 486 476</td>\n",
       "      <td>130.46</td>\n",
       "      <td>38 353 481</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>76 404 013</td>\n",
       "      <td>99.61%</td>\n",
       "      <td>81,00 %</td>\n",
       "      <td>29.66x(1x-817x)</td>\n",
       "      <td>99.71%</td>\n",
       "      <td>207 627</td>\n",
       "      <td>10 263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IV_4_method1</td>\n",
       "      <td>42 838 012</td>\n",
       "      <td>128.51</td>\n",
       "      <td>37 729 408</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>75 125 893</td>\n",
       "      <td>99.56%</td>\n",
       "      <td>81,00 %</td>\n",
       "      <td>28.76x(1x-877x)</td>\n",
       "      <td>99.91%</td>\n",
       "      <td>229 099</td>\n",
       "      <td>10 325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IV_7_method1</td>\n",
       "      <td>377 188 93</td>\n",
       "      <td>113.16</td>\n",
       "      <td>33 250 705</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>66 224 206</td>\n",
       "      <td>99.58%</td>\n",
       "      <td>77,00 %</td>\n",
       "      <td>25.82x(1x-847x)</td>\n",
       "      <td>99.70%</td>\n",
       "      <td>203 338</td>\n",
       "      <td>9 474</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Sample   Raw Reads Sequencing depth (X) Reads after QC  \\\n",
       "0  IV_5_method1  29 191 671                84.95     22 978 888   \n",
       "1  IV_2_method1  40 672 795               122.02     39 920 701   \n",
       "2  IV_6_method1  43 486 476               130.46     38 353 481   \n",
       "3  IV_4_method1  42 838 012               128.51     37 729 408   \n",
       "4  IV_7_method1  377 188 93               113.16     33 250 705   \n",
       "\n",
       "  Reads after QC# % Mapped reads Reads after QC % % On-target (30X)  \\\n",
       "0           79,00 %   45 770 776           99.59%           49,00 %   \n",
       "1           88,00 %   71 507 872           99.54%           79,00 %   \n",
       "2           88,00 %   76 404 013           99.61%           81,00 %   \n",
       "3           88,00 %   75 125 893           99.56%           81,00 %   \n",
       "4           88,00 %   66 224 206           99.58%           77,00 %   \n",
       "\n",
       "  Mean coverage (X) Genes covered (%)     SNPs  Indels  \n",
       "0   17.79x(1x-856x)            98.48%  114 651   6 935  \n",
       "1   26.76x(1x-860x)            99.70%  226 812  10 226  \n",
       "2   29.66x(1x-817x)            99.71%  207 627  10 263  \n",
       "3   28.76x(1x-877x)            99.91%  229 099  10 325  \n",
       "4   25.82x(1x-847x)            99.70%  203 338   9 474  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## We wolud do some tranformations to the first field (\"Sample\")\n",
    "table3[\"Sample\"] = table3[\"Sample\"].str.replace(\n",
    "    \"-\", \"_\"\n",
    "    ).str.replace(\n",
    "        \"^\", \"_method1\"\n",
    "        ).str.replace(\n",
    "            \"*\", \"_method2\"\n",
    "            )\n",
    "table3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "table3.to_csv(\"../metadata/table3.csv\", index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Raw Reads</th>\n",
       "      <th>Sequencing depth (X)</th>\n",
       "      <th>Reads after QC</th>\n",
       "      <th>Reads after QC# %</th>\n",
       "      <th>Mapped reads</th>\n",
       "      <th>Reads after QC %</th>\n",
       "      <th>% On-target (30X)</th>\n",
       "      <th>Mean coverage (X)</th>\n",
       "      <th>Genes covered (%)</th>\n",
       "      <th>SNPs</th>\n",
       "      <th>Indels</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IV_5_method1</td>\n",
       "      <td>29 191 671</td>\n",
       "      <td>84.95</td>\n",
       "      <td>22 978 888</td>\n",
       "      <td>79,00 %</td>\n",
       "      <td>45 770 776</td>\n",
       "      <td>99.59%</td>\n",
       "      <td>49,00 %</td>\n",
       "      <td>17.79x(1x-856x)</td>\n",
       "      <td>98.48%</td>\n",
       "      <td>114 651</td>\n",
       "      <td>6 935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IV_2_method1</td>\n",
       "      <td>40 672 795</td>\n",
       "      <td>122.02</td>\n",
       "      <td>39 920 701</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>71 507 872</td>\n",
       "      <td>99.54%</td>\n",
       "      <td>79,00 %</td>\n",
       "      <td>26.76x(1x-860x)</td>\n",
       "      <td>99.70%</td>\n",
       "      <td>226 812</td>\n",
       "      <td>10 226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IV_6_method1</td>\n",
       "      <td>43 486 476</td>\n",
       "      <td>130.46</td>\n",
       "      <td>38 353 481</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>76 404 013</td>\n",
       "      <td>99.61%</td>\n",
       "      <td>81,00 %</td>\n",
       "      <td>29.66x(1x-817x)</td>\n",
       "      <td>99.71%</td>\n",
       "      <td>207 627</td>\n",
       "      <td>10 263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IV_4_method1</td>\n",
       "      <td>42 838 012</td>\n",
       "      <td>128.51</td>\n",
       "      <td>37 729 408</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>75 125 893</td>\n",
       "      <td>99.56%</td>\n",
       "      <td>81,00 %</td>\n",
       "      <td>28.76x(1x-877x)</td>\n",
       "      <td>99.91%</td>\n",
       "      <td>229 099</td>\n",
       "      <td>10 325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IV_7_method1</td>\n",
       "      <td>377 188 93</td>\n",
       "      <td>113.16</td>\n",
       "      <td>33 250 705</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>66 224 206</td>\n",
       "      <td>99.58%</td>\n",
       "      <td>77,00 %</td>\n",
       "      <td>25.82x(1x-847x)</td>\n",
       "      <td>99.70%</td>\n",
       "      <td>203 338</td>\n",
       "      <td>9 474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>V_8_method1</td>\n",
       "      <td>55 720 364</td>\n",
       "      <td>222.88</td>\n",
       "      <td>51 583 390</td>\n",
       "      <td>93,00 %</td>\n",
       "      <td>102 574 419</td>\n",
       "      <td>99.43%</td>\n",
       "      <td>82,00 %</td>\n",
       "      <td>49.28x(1x-997x)</td>\n",
       "      <td>99.57%</td>\n",
       "      <td>195 153</td>\n",
       "      <td>12 768</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>V_2_method1</td>\n",
       "      <td>55 054 777</td>\n",
       "      <td>160.22</td>\n",
       "      <td>37 529 435</td>\n",
       "      <td>68,00 %</td>\n",
       "      <td>74 739 463</td>\n",
       "      <td>99.57%</td>\n",
       "      <td>64,00 %</td>\n",
       "      <td>27.24x(1x-998x)</td>\n",
       "      <td>98.62%</td>\n",
       "      <td>137 021</td>\n",
       "      <td>8 655</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>V_1_method1</td>\n",
       "      <td>57 215 132</td>\n",
       "      <td>166.51</td>\n",
       "      <td>34 333 842</td>\n",
       "      <td>60,00 %</td>\n",
       "      <td>68 423 638</td>\n",
       "      <td>99.64%</td>\n",
       "      <td>38,00 %</td>\n",
       "      <td>14.35x(1x-885x)</td>\n",
       "      <td>98.21%</td>\n",
       "      <td>99 034</td>\n",
       "      <td>5 679</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>IV_6_method2</td>\n",
       "      <td>43486476</td>\n",
       "      <td>130,46</td>\n",
       "      <td>38 353 481</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>76 404 013</td>\n",
       "      <td>99,61 %</td>\n",
       "      <td>81,00 %</td>\n",
       "      <td>29.66x(1x-817x)</td>\n",
       "      <td>99,71 %</td>\n",
       "      <td>207 626</td>\n",
       "      <td>10 263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>IV_2_method2</td>\n",
       "      <td>40672795</td>\n",
       "      <td>122,2</td>\n",
       "      <td>35920701</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>71 507 872</td>\n",
       "      <td>99,54 %</td>\n",
       "      <td>79,00 %</td>\n",
       "      <td>26.76x(1x-860x)</td>\n",
       "      <td>99,70 %</td>\n",
       "      <td>226 812</td>\n",
       "      <td>10 226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>IV_4_method2</td>\n",
       "      <td>42838012</td>\n",
       "      <td>125,51</td>\n",
       "      <td>37729408</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>75 125 893</td>\n",
       "      <td>99,56 %</td>\n",
       "      <td>81,00 %</td>\n",
       "      <td>28.76x(1x-877x)</td>\n",
       "      <td>99,91 %</td>\n",
       "      <td>229 099</td>\n",
       "      <td>10 325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>IV_7_method2</td>\n",
       "      <td>37718893</td>\n",
       "      <td>113,16</td>\n",
       "      <td>33250705</td>\n",
       "      <td>88,00 %</td>\n",
       "      <td>66 224 205</td>\n",
       "      <td>99,58 %</td>\n",
       "      <td>77,00 %</td>\n",
       "      <td>25.82x(1x-847x)</td>\n",
       "      <td>99,70 %</td>\n",
       "      <td>203 338</td>\n",
       "      <td>9 474</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Sample   Raw Reads Sequencing depth (X) Reads after QC  \\\n",
       "0   IV_5_method1  29 191 671                84.95     22 978 888   \n",
       "1   IV_2_method1  40 672 795               122.02     39 920 701   \n",
       "2   IV_6_method1  43 486 476               130.46     38 353 481   \n",
       "3   IV_4_method1  42 838 012               128.51     37 729 408   \n",
       "4   IV_7_method1  377 188 93               113.16     33 250 705   \n",
       "5    V_8_method1  55 720 364               222.88     51 583 390   \n",
       "6    V_2_method1  55 054 777               160.22     37 529 435   \n",
       "7    V_1_method1  57 215 132               166.51     34 333 842   \n",
       "8   IV_6_method2    43486476               130,46     38 353 481   \n",
       "9   IV_2_method2    40672795                122,2       35920701   \n",
       "10  IV_4_method2    42838012               125,51       37729408   \n",
       "11  IV_7_method2    37718893               113,16       33250705   \n",
       "\n",
       "   Reads after QC# % Mapped reads Reads after QC % % On-target (30X)  \\\n",
       "0            79,00 %   45 770 776           99.59%           49,00 %   \n",
       "1            88,00 %   71 507 872           99.54%           79,00 %   \n",
       "2            88,00 %   76 404 013           99.61%           81,00 %   \n",
       "3            88,00 %   75 125 893           99.56%           81,00 %   \n",
       "4            88,00 %   66 224 206           99.58%           77,00 %   \n",
       "5            93,00 %  102 574 419           99.43%           82,00 %   \n",
       "6            68,00 %   74 739 463           99.57%           64,00 %   \n",
       "7            60,00 %   68 423 638           99.64%           38,00 %   \n",
       "8            88,00 %   76 404 013          99,61 %           81,00 %   \n",
       "9            88,00 %   71 507 872          99,54 %           79,00 %   \n",
       "10           88,00 %   75 125 893          99,56 %           81,00 %   \n",
       "11           88,00 %   66 224 205          99,58 %           77,00 %   \n",
       "\n",
       "   Mean coverage (X) Genes covered (%)     SNPs  Indels  \n",
       "0    17.79x(1x-856x)            98.48%  114 651   6 935  \n",
       "1    26.76x(1x-860x)            99.70%  226 812  10 226  \n",
       "2    29.66x(1x-817x)            99.71%  207 627  10 263  \n",
       "3    28.76x(1x-877x)            99.91%  229 099  10 325  \n",
       "4    25.82x(1x-847x)            99.70%  203 338   9 474  \n",
       "5    49.28x(1x-997x)            99.57%  195 153  12 768  \n",
       "6    27.24x(1x-998x)            98.62%  137 021   8 655  \n",
       "7    14.35x(1x-885x)            98.21%   99 034   5 679  \n",
       "8    29.66x(1x-817x)           99,71 %  207 626  10 263  \n",
       "9    26.76x(1x-860x)           99,70 %  226 812  10 226  \n",
       "10   28.76x(1x-877x)           99,91 %  229 099  10 325  \n",
       "11   25.82x(1x-847x)           99,70 %  203 338   9 474  "
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table3_csv = pd.read_csv(\"../../metadata/table3.csv\")\n",
    "table3_csv"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Amount of reads after removing low quality sequences. \n",
    "### Low quality sequences are considered when\n",
    "\n",
    "1. bases with Q20 account for >85% of the total sequence length \n",
    "2. reads with N-rate >10%\n",
    "\n",
    "### Methods for seguencing\n",
    "\n",
    "* **method1** run with Agilent Sureselect XT Human All Exon 50Mb\n",
    "* **method2** run with Agilent SureSelect_XT_Human_All_Exon_V5"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Mean depth of the project"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "135.00333333333336"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table3_csv[\"Sequencing depth (X)\"].str.replace(\",\",\".\").astype(float).mean()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## **Calculating the AVERAGE sequence length for each sample**  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['bash', 'joining_files.sh'], returncode=0)"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## heavy computing +25 min -> you need all fastq.gz files in data/raw/\n",
    "\n",
    "for chr in [\"chr3\", \"chr5\", \"chr7\", \"chr12\", \"chr17\"]:\n",
    "    subprocess.run([\"bash\", \"calculating_length.sh\", chr])\n",
    "\n",
    "subprocess.run([\"bash\", \"joining_files.sh\"])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "list_files = os.listdir(\"../../results/annotations/\")\n",
    "\n",
    "list_files_path = []\n",
    "\n",
    "for file in range(len(list_files)):\n",
    "    path = \"../../results/annotations/\" + list_files[file]\n",
    "    list_files_path.append(path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mean value of  ERR753374  ->  94.61 +/- 2.29\n",
      "mean value of  ERR753373  ->  94.72 +/- 2.18\n",
      "mean value of  ERR753375  ->  75.51 +/- 1.26\n",
      "mean value of  ERR753372  ->  75.89 +/- 0.73\n",
      "mean value of  ERR753370  ->  100.64 +/- 1.27\n",
      "mean value of  ERR696683  ->  75.48 +/- 1.31\n",
      "mean value of  ERR753376  ->  75.48 +/- 1.31\n",
      "mean value of  ERR753378  ->  75.89 +/- 0.73\n",
      "mean value of  ERR753371  ->  75.24 +/- 1.53\n",
      "mean value of  ERR753369  ->  75.89 +/- 0.73\n",
      "mean value of  ERR753377  ->  75.5 +/- 1.29\n",
      "mean value of  ERR753368  ->  75.89 +/- 0.73\n"
     ]
    }
   ],
   "source": [
    "list_mean_len = []\n",
    "list_sd_len = []\n",
    "\n",
    "for file_path in list_files_path:\n",
    "\n",
    "    with open(file_path) as file:\n",
    "        len_list = file.readlines()\n",
    "    \n",
    "    mean_seq_len = np.array([s.strip() for s in len_list]).astype(int).mean().round(2)\n",
    "    sd_seq_len = np.array([s.strip() for s in len_list]).astype(int).std().round(2)\n",
    "    \n",
    "    print(\"mean value of \", file_path.replace(\n",
    "        \"../../results/annotations/\", \"\"\n",
    "        ).replace(\n",
    "            \"_len.txt\", \"\"\n",
    "            ), \" -> \", mean_seq_len, \"+/-\", sd_seq_len)\n",
    "\n",
    "    list_mean_len.append(mean_seq_len)\n",
    "    list_sd_len.append(sd_seq_len)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_4425/3359814889.py:7: FutureWarning: The default value of regex will change from True to False in a future version.\n",
      "  df_len_seq[\"sample\"] = df_len_seq[\"sample\"].str.replace(\"../../results/annotations/\", \"\")\n",
      "/tmp/ipykernel_4425/3359814889.py:8: FutureWarning: The default value of regex will change from True to False in a future version.\n",
      "  df_len_seq[\"sample\"] = df_len_seq[\"sample\"].str.replace(\"_len.txt\", \"\")\n"
     ]
    }
   ],
   "source": [
    "df_len_seq = pd.DataFrame({\n",
    "    \"sample\": pd.Series(list_files_path),\n",
    "    \"mean_len\": pd.Series(list_mean_len),\n",
    "    \"sd_len\": pd.Series(list_sd_len)\n",
    "})\n",
    "\n",
    "df_len_seq[\"sample\"] = df_len_seq[\"sample\"].str.replace(\"../../results/annotations/\", \"\")\n",
    "df_len_seq[\"sample\"] = df_len_seq[\"sample\"].str.replace(\"_len.txt\", \"\")\n",
    "\n",
    "df_len_seq.to_csv(\"len_mean.csv\", index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Rows: 12 Columns: 3\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "Delimiter: \",\"\n",
      "chr (1): sample\n",
      "dbl (2): mean_len, sd_len\n",
      "\n",
      "ℹ Use `spec()` to retrieve the full column specification for this data.\n",
      "ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['Rscript', 'plot_mean_len.R'], returncode=0)"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subprocess.run([\"Rscript\", \"plot_mean_len.R\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<p align=\"center\">\n",
    " \n",
    "  <img src=\"../../results/objective/mean_len.png\" alt=\"800\" width=\"900\" />\n",
    "\n",
    "</p>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## **Number of Seqs before and after fastp**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "subprocess.run([\"bash\", \"fastp_analysis.sh\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.read_csv(\"fastp_analysis.tsv\", sep=\"\\t\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "envpython",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
